// @ths/design/es/components/progress-base/src/index.vue2.js
import { defineComponent as tt, computed as Je, onMounted as at, watch as rt } from "vue";
import { getDomRandomClass as lt, setStyle as it, getCssPropertyString as a } from "../../../utils/index.js";
const ut = tt({
  name: "TProgressBase",
  props: {
    // 数据 支持传入一个对象，或一个数组
    data: {
      type: Object,
      default: () => ({})
    },
    // 显示顶栏
    showHeader: {
      type: Boolean,
      default: !1
    },
    // 别名
    alias: {
      type: Object,
      default: () => ({
        // 状态字段
        stateText: "stateText",
        stateText2: "stateText2",
        stateImg: "stateImg",
        titleText: "titleText",
        titleText2: "titleText2",
        titleImg: "titleImg",
        iconText: "iconText",
        iconText2: "iconText2",
        imgSrc: "imgSrc"
      })
    },
    // 状态样式配置选项
    stateOptions: {
      type: Object,
      default: () => ({
        // 文本
        text: {},
        // 背景
        background: {}
      })
    },
    // 比较值1
    comparableValueField1: {
      type: String,
      default: "barValue"
    },
    // 比较值2
    comparableValueField2: {
      type: String,
      default: "modifyValue1"
    },
    // 进度条背景色
    barBackground: {
      type: Object,
      default: () => ({
        // 比较值1等于比较值2
        equal: "var(--t-progress-base-inner-bg)",
        // 等于
        // 比较值1大于比较值2
        greater: "var(--t-progress-base-inner-bg)",
        // 大于
        // 比较值1小于比较值2
        less: "var(--t-progress-base-inner-bg)"
        // 小于
      })
    },
    // 进度条容器背景色
    barWrapperBackground: {
      type: Object,
      default: () => ({
        // 比较值1等于比较值2
        equal: "var(--t-progress-base-outer-bg)",
        // 等于
        // 比较值1大于比较值2
        greater: "var(--t-progress-base-outer-bg)",
        // 大于
        // 比较值1小于比较值2
        less: "var(--t-progress-base-outer-bg)"
        // 小于
      })
    },
    // 目标值文字颜色
    currentValueColor: {
      type: Object,
      default: () => ({
        // 比较值1等于比较值2
        equal: "var(--t-progress-base-num-current-color)",
        // 等于
        // 比较值1大于比较值2
        greater: "var(--t-progress-base-num-current-color)",
        // 大于
        // 比较值1小于比较值2
        less: "var(--t-progress-base-num-current-color)"
        // 小于
      })
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 列表项
        item: {
          default: {},
          hover: {}
        },
        // 列表项的顶栏头部
        itemHeader: {
          default: {},
          hover: {}
        },
        // 列表项的顶栏图标容器
        itemHeaderIcon: {
          default: {},
          hover: {}
        },
        // 列表项的顶栏图标容器中的文本
        itemHeaderIconText: {
          default: {},
          hover: {}
        },
        // 列表项的顶栏图标容器中的背景
        itemHeaderIconBg: {
          default: {},
          hover: {}
        },
        // 列表项的顶栏标题容器
        itemHeaderTitle: {
          default: {},
          hover: {}
        },
        // 列表项的顶栏标题容器中的文本
        itemHeaderTitleText: {
          default: {},
          hover: {}
        },
        // 列表项的顶栏标题容器中的背景
        itemHeaderTitleBg: {
          default: {},
          hover: {}
        },
        // 列表项的顶栏状态容器
        itemHeaderState: {
          default: {},
          hover: {}
        },
        // 列表项的顶栏状态容器中的文本
        itemHeaderStateText: {
          default: {},
          hover: {}
        },
        // 列表项的顶栏状态容器中的背景
        itemHeaderStateBg: {
          default: {},
          hover: {}
        },
        // 名称
        label: {
          default: {}
        },
        // 当前值文本
        currentText: {
          default: {}
        },
        // 当前值
        currentValue: {
          default: {}
        },
        // 目标值文本
        targetText: {
          default: {}
        },
        // 目标值
        targetValue: {
          default: {}
        },
        // 进度条容器
        barOuter: {
          default: {}
        },
        // 进度条填充
        barInner: {
          default: {}
        },
        // 修饰器1
        modify1: {
          default: {}
        },
        // 修饰器2
        modify2: {
          default: {}
        },
        // 下方描述
        desc: {
          default: {}
        }
      })
    }
  },
  emits: ["item-click"],
  setup(e, { emit: Ke }) {
    const m = Je(() => e.cStyle);
    function y(t, i) {
      var d, o, n;
      let l = "";
      if (e.comparableValueField1 in i && e.comparableValueField2 in i) {
        let c = i[e.comparableValueField1], u = i[e.comparableValueField2];
        const f = typeof i[e.comparableValueField1], s = typeof i[e.comparableValueField2];
        f === "string" && (c = parseFloat(c)), s === "string" && (u = parseFloat(u)), c > u ? l = (d = e[t]) == null ? void 0 : d.greater : c === u ? l = (o = e[t]) == null ? void 0 : o.equal : l = (n = e[t]) == null ? void 0 : n.less;
      }
      return l;
    }
    function Qe(t) {
      var l;
      const i = y("barBackground", t);
      return { width: `${t.barValue || 0}%`, background: i, ...(l = m.value.barInner) == null ? void 0 : l.default };
    }
    function Ue(t) {
      var l, d;
      return { background: y("barWrapperBackground", t), ...(d = (l = m.value) == null ? void 0 : l.barOuter) == null ? void 0 : d.default };
    }
    function Xe(t) {
      var l, d;
      return { color: y("currentValueColor", t), ...(d = (l = m.value) == null ? void 0 : l.currentValue) == null ? void 0 : d.default };
    }
    const Ye = Je(() => Array.isArray(e.data) ? e.data : Object.prototype.toString.call(e.data) === "[object Object]" ? [e.data] : []), r = lt("progress-base"), Ze = () => {
      let t = "";
      return m.value.wrapper && (t += it(m.value.wrapper, `.${r}`)), t;
    }, pe = () => {
      var i, l, d, o, n, c, u, f, s, S, b, h, H, v, _, $, T, x, B, I, V, k, F, O, j, C, w, q, P, A, L, M, W, D, N, R, z, E, G, J, K, Q, U, X, Y, Z, p, ee, te, ae, re, le, ie, de, ce, ue, me, oe, ne, fe, se, ye, ge, Se, be, he, He, ve, _e, $e, Te, xe, Be, Ie, Ve, ke, Fe, Oe, je, Ce, we, qe, Pe, Ae, Le, Me, We, De, Ne, Re, ze, Ee, Ge;
      let t = "";
      return (l = (i = e.cStyle) == null ? void 0 : i.item) != null && l.default && (t += `.${r}.t-progress-base .item {
          ${a((o = (d = e.cStyle) == null ? void 0 : d.item) == null ? void 0 : o.default)}
        }`), (c = (n = e.cStyle) == null ? void 0 : n.item) != null && c.hover && (t += `.${r}.t-progress-base .item:hover {
          ${a((f = (u = e.cStyle) == null ? void 0 : u.item) == null ? void 0 : f.hover)}
        }`), (S = (s = e.cStyle) == null ? void 0 : s.itemHeader) != null && S.default && (t += `.${r}.t-progress-base .item__header {
          ${a((h = (b = e.cStyle) == null ? void 0 : b.itemHeader) == null ? void 0 : h.default)}
        }`), (v = (H = e.cStyle) == null ? void 0 : H.itemHeader) != null && v.hover && (t += `.${r}.t-progress-base .item__header:hover {
          ${a(($ = (_ = e.cStyle) == null ? void 0 : _.itemHeader) == null ? void 0 : $.hover)}

          ${a((x = (T = e.cStyle) == null ? void 0 : T.itemHeaderIcon) == null ? void 0 : x.hover)}
          ${a((I = (B = e.cStyle) == null ? void 0 : B.itemHeaderIconText) == null ? void 0 : I.hover)}
          ${a((k = (V = e.cStyle) == null ? void 0 : V.itemHeaderIconBg) == null ? void 0 : k.hover)}

          ${a((O = (F = e.cStyle) == null ? void 0 : F.itemHeaderTitle) == null ? void 0 : O.hover)}
          ${a((C = (j = e.cStyle) == null ? void 0 : j.itemHeaderTitleText) == null ? void 0 : C.hover)}
          ${a((q = (w = e.cStyle) == null ? void 0 : w.itemHeaderTitleBg) == null ? void 0 : q.hover)}

          ${a((A = (P = e.cStyle) == null ? void 0 : P.itemHeaderState) == null ? void 0 : A.hover)}
          ${a((M = (L = e.cStyle) == null ? void 0 : L.itemHeaderStateText) == null ? void 0 : M.hover)}
          ${a((D = (W = e.cStyle) == null ? void 0 : W.itemHeaderStateBg) == null ? void 0 : D.hover)}
        }`), (R = (N = e.cStyle) == null ? void 0 : N.itemHeaderIcon) != null && R.default && (t += `.${r}.t-progress-base .item__header .item__header-icon {
          ${a((E = (z = e.cStyle) == null ? void 0 : z.itemHeaderIcon) == null ? void 0 : E.default)}
        }`), (J = (G = e.cStyle) == null ? void 0 : G.itemHeaderIcon) != null && J.hover && (t += `.${r}.t-progress-base .item__header .item__header-icon:hover {
          ${a((Q = (K = e.cStyle) == null ? void 0 : K.itemHeaderIcon) == null ? void 0 : Q.hover)}
          ${a((X = (U = e.cStyle) == null ? void 0 : U.itemHeaderIconText) == null ? void 0 : X.hover)}
          ${a((Z = (Y = e.cStyle) == null ? void 0 : Y.itemHeaderIconBg) == null ? void 0 : Z.hover)}
        }`), (ee = (p = e.cStyle) == null ? void 0 : p.itemHeaderIconText) != null && ee.default && (t += `.${r}.t-progress-base .item__header .item__header-icon--text {
          ${a((ae = (te = e.cStyle) == null ? void 0 : te.itemHeaderIconText) == null ? void 0 : ae.default)}
        }`), (le = (re = e.cStyle) == null ? void 0 : re.itemHeaderIconBg) != null && le.default && (t += `.${r}.t-progress-base .item__header .item__header-icon--background {
          ${a((de = (ie = e.cStyle) == null ? void 0 : ie.itemHeaderIconBg) == null ? void 0 : de.default)}
        }`), (ue = (ce = e.cStyle) == null ? void 0 : ce.itemHeaderTitle) != null && ue.default && (t += `.${r}.t-progress-base .item__header .item__header-title {
          ${a((oe = (me = e.cStyle) == null ? void 0 : me.itemHeaderTitle) == null ? void 0 : oe.default)}
        }`), (fe = (ne = e.cStyle) == null ? void 0 : ne.itemHeaderTitle) != null && fe.hover && (t += `.${r}.t-progress-base .item__header .item__header-title:hover {
          ${a((ye = (se = e.cStyle) == null ? void 0 : se.itemHeaderTitle) == null ? void 0 : ye.hover)}
          ${a((Se = (ge = e.cStyle) == null ? void 0 : ge.itemHeaderTitleText) == null ? void 0 : Se.hover)}
          ${a((he = (be = e.cStyle) == null ? void 0 : be.itemHeaderTitleBg) == null ? void 0 : he.hover)}
        }`), (ve = (He = e.cStyle) == null ? void 0 : He.itemHeaderTitleText) != null && ve.default && (t += `.${r}.t-progress-base .item__header .item__header-title--text {
          ${a(($e = (_e = e.cStyle) == null ? void 0 : _e.itemHeaderTitleText) == null ? void 0 : $e.default)}
        }`), (xe = (Te = e.cStyle) == null ? void 0 : Te.itemHeaderTitleBg) != null && xe.default && (t += `.${r}.t-progress-base .item__header .item__header-title--background {
          ${a((Ie = (Be = e.cStyle) == null ? void 0 : Be.itemHeaderTitleBg) == null ? void 0 : Ie.default)}
        }`), (ke = (Ve = e.cStyle) == null ? void 0 : Ve.itemHeaderState) != null && ke.default && (t += `.${r}.t-progress-base .item__header .item__header-state {
          ${a((Oe = (Fe = e.cStyle) == null ? void 0 : Fe.itemHeaderState) == null ? void 0 : Oe.default)}
        }`), (Ce = (je = e.cStyle) == null ? void 0 : je.itemHeaderState) != null && Ce.hover && (t += `.${r}.t-progress-base .item__header .item__header-state:hover {
          ${a((qe = (we = e.cStyle) == null ? void 0 : we.itemHeaderState) == null ? void 0 : qe.hover)}
          ${a((Ae = (Pe = e.cStyle) == null ? void 0 : Pe.itemHeaderStateText) == null ? void 0 : Ae.hover)}
          ${a((Me = (Le = e.cStyle) == null ? void 0 : Le.itemHeaderStateBg) == null ? void 0 : Me.hover)}
        }`), (De = (We = e.cStyle) == null ? void 0 : We.itemHeaderStateText) != null && De.default && (t += `.${r}.t-progress-base .item__header .item__header-state--text {
          ${a((Re = (Ne = e.cStyle) == null ? void 0 : Ne.itemHeaderStateText) == null ? void 0 : Re.default)}
        }`), (Ee = (ze = e.cStyle) == null ? void 0 : ze.itemHeaderStateBg) != null && Ee.default && (t += `.${r}.t-progress-base .item__header .item__header-state--background {
          ${a((Ge = e.cStyle) == null ? void 0 : Ge.itemHeaderStateBg.default)}
        }`), t;
    }, g = () => {
      const t = document.querySelector(`.${r} .style-container`);
      t && (t.innerHTML = `<style>${Ze()}${pe()}</style>`);
    }, et = (t, i) => {
      Ke("item-click", t, i);
    };
    return at(() => {
      g();
    }), rt(
      () => e.cStyle,
      () => {
        g();
      },
      { deep: !0 }
    ), {
      curStyle: m,
      dataList: Ye,
      domOnlyClass: r,
      getProgressBarInnerStyle: Qe,
      getProgressBarOuterStyle: Ue,
      getCurrentValueStyle: Xe,
      clickProgress: et
    };
  }
});
export {
  ut as default
};
