// @ths/design/es/components/progress-base/src/index.vue.js
import ue from "./index.vue2.js";
import { openBlock as r, createElementBlock as o, normalizeClass as ye, normalizeStyle as l, Fragment as ce, renderList as ge, createElementVNode as a, toDisplayString as t, createCommentVNode as d, renderSlot as fe } from "vue";
import "./index.vue3.js";
import pe from "../../../_virtual/_plugin-vue_export-helper.js";
const me = ["onClick"], _e = {
  key: 0,
  class: "item__header"
}, ke = { class: "item__header-icon" }, be = { class: "item__header-title" }, ve = { class: "item__header-state" }, he = { class: "item__top" }, Se = { class: "item__top__left" }, Te = { class: "item__top__span" }, Ie = { class: "item__center progress-bar" }, Ve = ["innerHTML"];
function $e(e, i, Oe, Ce, Be, Le) {
  var n, u;
  return r(), o("div", {
    class: ye(["t-progress-base wrapper", e.domOnlyClass]),
    style: l({ ...(u = (n = e.curStyle) == null ? void 0 : n.wrapper) == null ? void 0 : u.default })
  }, [
    (r(!0), o(ce, null, ge(e.dataList, (s, y) => {
      var c, g, f, p, m, _, k, b, v, h, S, T, I, V, $, O, C, B, L, P, w, H, z, E, M, N, D, F, j, q, A, G, J, K, Q, R, U, W, X, Y, Z, x, ee, se, ae, le, te, re, oe, de, ie, ne;
      return r(), o("div", {
        key: y + s.label,
        class: "item",
        onClick: (Pe) => e.clickProgress(s, y)
      }, [
        e.showHeader ? (r(), o("div", _e, [
          a("div", ke, [
            a("span", {
              class: "item__header-icon--text",
              style: l((f = (c = e.stateOptions) == null ? void 0 : c.text) == null ? void 0 : f[s[(g = e.alias) == null ? void 0 : g.iconText]])
            }, t(s[(p = e.alias) == null ? void 0 : p.iconText]), 5),
            a("span", {
              class: "item__header-icon--background",
              style: l({ backgroundImage: `url(${s[e.alias.imgSrc]})`, ...(k = (m = e.stateOptions) == null ? void 0 : m.background) == null ? void 0 : k[s[(_ = e.alias) == null ? void 0 : _.iconText2]] })
            }, t(s[(b = e.alias) == null ? void 0 : b.iconText2]), 5)
          ]),
          a("div", be, [
            a("span", {
              class: "item__header-title--text",
              style: l((S = (v = e.stateOptions) == null ? void 0 : v.text) == null ? void 0 : S[s[(h = e.alias) == null ? void 0 : h.titleText]])
            }, t(s[(T = e.alias) == null ? void 0 : T.titleText]), 5),
            a("span", {
              class: "item__header-title--background",
              style: l({ backgroundImage: `url(${s[e.alias.titleImg]})`, ...($ = (I = e.stateOptions) == null ? void 0 : I.background) == null ? void 0 : $[s[(V = e.alias) == null ? void 0 : V.titleText2]] })
            }, t(s[(O = e.alias) == null ? void 0 : O.titleText2]), 5)
          ]),
          a("div", ve, [
            a("span", {
              class: "item__header-state--text",
              style: l((L = (C = e.stateOptions) == null ? void 0 : C.text) == null ? void 0 : L[s[(B = e.alias) == null ? void 0 : B.stateText]])
            }, t(s[(P = e.alias) == null ? void 0 : P.stateText]), 5),
            a("span", {
              class: "item__header-state--background",
              style: l({ backgroundImage: `url(${s[e.alias.stateImg]})`, ...(z = (w = e.stateOptions) == null ? void 0 : w.background) == null ? void 0 : z[s[(H = e.alias) == null ? void 0 : H.stateText2]] })
            }, t(s[(E = e.alias) == null ? void 0 : E.stateText2]), 5)
          ])
        ])) : d("", !0),
        a("div", he, [
          a("div", Se, [
            a("div", {
              class: "name",
              style: l({ ...(N = (M = e.curStyle) == null ? void 0 : M.label) == null ? void 0 : N.default, ...(F = (D = e.curStyle) == null ? void 0 : D.currentText) == null ? void 0 : F.default })
            }, t(s.label || ""), 5),
            a("div", {
              class: "num",
              style: l(e.getCurrentValueStyle(s))
            }, t(s.value || ""), 5)
          ]),
          a("div", Te, [
            a("div", {
              class: "name",
              style: l({ ...(q = (j = e.curStyle) == null ? void 0 : j.label) == null ? void 0 : q.default, ...(G = (A = e.curStyle) == null ? void 0 : A.targetText) == null ? void 0 : G.default })
            }, t(s.targetLabel || ""), 5),
            a("div", {
              class: "num",
              style: l({ ...(K = (J = e.curStyle) == null ? void 0 : J.targetValue) == null ? void 0 : K.default })
            }, t(s.targetValue || ""), 5)
          ])
        ]),
        a("div", Ie, [
          a("div", {
            class: "progress-bar__outer",
            style: l(e.getProgressBarOuterStyle(s))
          }, [
            a("div", {
              class: "progress-bar__inner",
              style: l(e.getProgressBarInnerStyle(s))
            }, null, 4),
            s.modifyValue1 || s.modifyValue1 === 0 ? (r(), o("div", {
              key: 0,
              class: "progress-bar__modify progress-bar__modify1",
              style: l({
                left: (s.modifyValue1 || 0) + "%",
                ...(R = (Q = e.curStyle) == null ? void 0 : Q.modify1) == null ? void 0 : R.default,
                maskImage: (X = (W = (U = e.curStyle) == null ? void 0 : U.modify1) == null ? void 0 : W.default) != null && X.maskImage ? `url(${(x = (Z = (Y = e.curStyle) == null ? void 0 : Y.modify1) == null ? void 0 : Z.default) == null ? void 0 : x.maskImage})` : ""
              })
            }, null, 4)) : d("", !0),
            s.modifyValue2 || s.modifyValue2 === 0 ? (r(), o("div", {
              key: 1,
              class: "progress-bar__modify progress-bar__modify2",
              style: l({
                left: (s.modifyValue2 || 0) + "%",
                ...(se = (ee = e.curStyle) == null ? void 0 : ee.modify2) == null ? void 0 : se.default,
                maskImage: (te = (le = (ae = e.curStyle) == null ? void 0 : ae.modify2) == null ? void 0 : le.default) != null && te.maskImage ? `url(${(de = (oe = (re = e.curStyle) == null ? void 0 : re.modify2) == null ? void 0 : oe.default) == null ? void 0 : de.maskImage})` : ""
              })
            }, null, 4)) : d("", !0)
          ], 4)
        ]),
        s.desc ? (r(), o("div", {
          key: 1,
          class: "item__bottom",
          style: l({ ...(ne = (ie = e.curStyle) == null ? void 0 : ie.desc) == null ? void 0 : ne.default }),
          innerHTML: s.desc
        }, null, 12, Ve)) : d("", !0),
        s.desc ? d("", !0) : fe(e.$slots, "default", { key: 2 })
      ], 8, me);
    }), 128)),
    i[0] || (i[0] = a("div", { class: "style-container" }, null, -1))
  ], 6);
}
const Me = /* @__PURE__ */ pe(ue, [["render", $e]]);
export {
  Me as default
};
