// @ths/design/es/components/two-dimension-pie/src/index.vue2.js
import { defineComponent as T, ref as x, computed as O, reactive as R, onMounted as V, onUnmounted as W, watch as y, nextTick as w, toRefs as z } from "vue";
import * as j from "echarts";
import h from "lodash/merge";
import C from "lodash/cloneDeep";
import { getDomRandomClass as $, setStyle as E } from "../../../utils/index.js";
import "../../two-dimension-container/index.js";
import F from "../../two-dimension-container/src/index.vue.js";
const p = T({
  name: "TTwoDimensionPie",
  components: {
    TTwoDimensionContainer: F
  },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外容器
        wrapper: {
          default: {}
        },
        // 中心内容
        innerWrapper: {
          default: {}
        },
        // 左容器
        sideLeftWrapper: {
          default: {}
        },
        // 右容器
        sideRightWrapper: {
          default: {}
        },
        // 中间容器
        centerWrapper: {
          default: {}
        },
        // 标题
        title: {
          default: {}
        },
        // 标识线
        line: {
          default: {}
        },
        // 内标识线
        lineInner: {
          default: {}
        },
        // 左值
        leftNum: {
          default: {}
        },
        // 右值
        rightNum: {
          default: {}
        },
        // 单位
        unit: {
          default: {}
        },
        // 百分比
        percent: {
          default: {}
        }
      })
    },
    // 左侧数据
    leftData: {
      type: Object,
      default: () => ({})
    },
    // 右侧数据
    rightData: {
      type: Object,
      default: () => ({})
    },
    // 中心背景
    bgUrl: {
      type: String,
      default: ""
    },
    // 中心图标
    innerUrl: {
      type: String,
      default: ""
    },
    // 饼图series样式设置
    pieStyle: {
      type: Object,
      default: () => ({})
    },
    // 百分比保留小数位数
    fixNumber: {
      type: Number,
      default: () => 2
    },
    // 间隔值
    intervalValue: {
      type: Number,
      default: () => 1
    }
  },
  setup(e) {
    const f = x(null);
    let l = null;
    const d = O(() => e.cStyle), s = $("two-dimension-pie"), a = R({
      chartData: [],
      // 图表数据
      curLeftData: {},
      // 左侧数据
      curRightData: {}
      // 右侧数据
    }), g = (t, r) => {
      const n = (t / r * 100).toFixed(e.fixNumber <= 0 ? 0 : e.fixNumber);
      return n && n !== "NaN" ? `${n}%` : "-";
    }, b = () => {
      var t, r, n, i, u, c, o, m;
      if (a.chartData = [], a.curRightData = {}, a.curLeftData = {}, e.leftData && e.rightData) {
        a.chartData = [
          {
            value: (t = e.leftData) == null ? void 0 : t.num,
            itemStyle: {
              color: ((r = e.leftData) == null ? void 0 : r.lineColor) || "#FFD543"
            }
          },
          {
            value: (n = e.rightData) == null ? void 0 : n.num,
            itemStyle: {
              color: ((i = e.rightData) == null ? void 0 : i.lineColor) || "#13EB9C"
            }
          }
        ];
        const S = Number(((u = e.leftData) == null ? void 0 : u.num) || "") + Number(((c = e.rightData) == null ? void 0 : c.num) || "");
        a.curRightData = h(C(e.rightData), { percent: g(Number((o = e.rightData) == null ? void 0 : o.num), S) }), a.curLeftData = h(C(e.leftData), { percent: g(Number((m = e.leftData) == null ? void 0 : m.num), S) });
      }
    }, D = () => {
      var i, u, c, o;
      let t = [];
      const r = {
        type: "pie",
        radius: ["78%", "92%"],
        avoidLabelOverlap: !0,
        clockwise: !0,
        label: {
          show: !1
        },
        labelLine: {
          show: !1
        },
        silent: !0,
        data: [
          {
            ...a.chartData[1]
          },
          {
            // 有一个值为0时，无间隔
            value: Number(((i = e.leftData) == null ? void 0 : i.num) || "") === 0 || Number(((u = e.rightData) == null ? void 0 : u.num) || "") === 0 ? 0 : e.intervalValue,
            itemStyle: {
              color: "transparent"
            }
          },
          {
            ...a.chartData[0]
          },
          {
            // 有一个值为0时，无间隔
            value: Number(((c = e.leftData) == null ? void 0 : c.num) || "") === 0 || Number(((o = e.rightData) == null ? void 0 : o.num) || "") === 0 ? 0 : e.intervalValue,
            itemStyle: {
              color: "transparent"
            }
          }
        ]
      };
      t = h(r, e.pieStyle);
      const n = {
        series: t
      };
      l.clear();
      try {
        l.setOption(n);
      } catch (m) {
        console.log(m);
      }
    }, v = () => {
      w(() => {
        l.resize();
      });
    }, L = () => {
      let t = "";
      return d.value.innerWrapper && (t += E(d.value.innerWrapper, `.${s} .center-content`)), t;
    }, N = () => {
      const t = document.querySelector(`.${s} > .style-container`);
      t && (t.innerHTML = `<style>${L()}</style>`);
    };
    return V(() => {
      N(), f.value && (l = j.init(f.value), b(), D()), window.addEventListener("resize", v);
    }), W(() => {
      window.removeEventListener("resize", v);
    }), y(() => [e.rightData, e.leftData, e.fixNumber, e.pieStyle], () => {
      b(), D();
    }, {
      deep: !0
    }), y(() => [e.cStyle], () => {
      N(), w(() => {
        l.resize();
      });
    }, {
      deep: !0
    }), y(() => [e.intervalValue], () => {
      D();
    }, {
      deep: !0
    }), {
      ...z(a),
      chart: f,
      curStyle: d,
      domOnlyClass: s
    };
  }
});
export {
  p as default
};
