// @ths/design/es/components/steps-base/src/index.vue.js
import f from "./index.vue2.js";
import { resolveComponent as a, openBlock as t, createElementBlock as o, normalizeClass as l, createBlock as c, mergeProps as g, withCtx as i, Fragment as m, renderList as k, createSlots as v, createCommentVNode as u, createElementVNode as C } from "vue";
import "./index.vue3.js";
import h from "../../../_virtual/_plugin-vue_export-helper.js";
const y = ["src"], D = ["src"];
function b(e, n, I, S, $, B) {
  const p = a("el-step"), d = a("el-steps");
  return t(), o("div", {
    class: l(["t-steps-base", e.domOnlyClass])
  }, [
    (t(), c(d, g({
      key: e.count,
      "align-center": e.curDirection === "horizontal",
      "finish-status": "finish",
      "process-status": e.curStatus ? "error" : "process",
      direction: e.curDirection
    }, e.filterAttrs(e.$attrs)), {
      default: i(() => [
        (t(!0), o(m, null, k(e.listData, (s, r) => (t(), c(p, {
          key: r,
          title: s.title,
          description: s.desc,
          status: s.status,
          class: l([{
            "no-serial": !e.curHasSerial,
            "no-border": e.curHasSerial && e.curDesign,
            "no-bg": e.curDesign && (s.activeIcon !== "" || s.defaultIcon !== ""),
            "t-steps-base-item-checked": e.stateCollection[r]
          }]),
          onClick: (A) => e.handleClick(r)
        }, v({ _: 2 }, [
          e.curHasSerial ? {
            name: "icon",
            fn: i(() => [
              s.activeIcon !== "" && r <= e.curActive && e.curDesign ? (t(), o("img", {
                key: 0,
                class: "icon-active",
                src: s.activeIcon,
                draggable: "false"
              }, null, 8, y)) : u("", !0),
              s.defaultIcon !== "" && r > e.curActive && e.curDesign ? (t(), o("img", {
                key: 1,
                class: "icon-default",
                src: s.defaultIcon,
                draggable: "false"
              }, null, 8, D)) : u("", !0)
            ]),
            key: "0"
          } : void 0
        ]), 1032, ["title", "description", "status", "class", "onClick"]))), 128))
      ]),
      _: 1
    }, 16, ["align-center", "process-status", "direction"])),
    n[0] || (n[0] = C("div", { class: "style-container" }, null, -1))
  ], 2);
}
const V = /* @__PURE__ */ h(f, [["render", b]]);
export {
  V as default
};
