// @ths/design/es/components/steps-base/src/index.vue2.js
import { defineComponent as L, computed as f, ref as $, reactive as x, watch as _, nextTick as E, onMounted as F, toRefs as G } from "vue";
import K from "lodash/cloneDeep";
import { filterAttrs as Q, getDomRandomClass as U, setStyle as N, getCssPropertyString as i } from "../../../utils/index.js";
const Z = L({
  name: "TStepsBase",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        },
        itemWrapper: {
          default: {},
          hover: {},
          active: {}
        },
        // 步骤标题
        title: {
          default: {},
          active: {},
          finish: {},
          error: {}
        },
        // 步骤连接线
        line: {
          default: {},
          active: {},
          finish: {},
          error: {}
        },
        // 步骤描述
        desc: {
          default: {},
          active: {},
          finish: {},
          error: {}
        },
        // 步骤标记
        icon: {
          default: {},
          active: {},
          finish: {},
          error: {}
        },
        // 自定义图标
        designIcon: {
          default: {},
          active: {}
        }
      })
    },
    // 关键字，用于设置默认选中
    keyWord: {
      type: String,
      default: ""
    },
    /**
         * 设置默认选中字段，当通过关键字，以及defaultChecked设置默认选中
         */
    defaultChecked: {
      type: String,
      default: ""
    },
    // 是否开启默认选中
    tab: {
      type: Boolean,
      default: !1
    },
    // 当前激活项
    active: {
      type: Number,
      default: () => 1
    },
    // 是否有序号
    hasSerial: {
      type: Boolean,
      default: () => !0
    },
    // 是否开启自定义图标
    onDesign: {
      type: Boolean,
      default: () => !1
    },
    // 是否为错误项
    isError: {
      type: Boolean,
      default: () => !1
    },
    // 步骤条方向 水平或垂直
    direction: {
      type: String,
      default: () => "horizontal"
    },
    // 数据项
    data: {
      type: Array,
      default: () => []
    }
  },
  emits: ["step-click"],
  setup(a, { emit: T }) {
    const e = f(() => a.cStyle), J = f(() => a.hasSerial), M = f(() => a.onDesign), u = $(0), P = f(() => a.direction), p = f(() => a.isError), l = U("steps-base"), r = x({
      listData: [],
      // 步骤条数据
      count: 1
      // 计数器-确保DOM重新渲染
    }), c = $([]), o = $(-1), R = () => {
      var n, s, v, d;
      let t = "";
      return e.value.wrapper && (t += N(e.value.wrapper, `.${l}`)), e.value.itemWrapper && (t += N(e.value.itemWrapper, `.${l} .el-step`)), (n = e.value.itemWrapper) != null && n.hover && (t += `.${l} .el-step:hover {
          ${i((s = e.value.itemWrapper) == null ? void 0 : s.hover)}
        }`), (v = e.value.itemWrapper) != null && v.active && (t += `.${l} .t-steps-base-item-checked {
          ${i((d = e.value.itemWrapper) == null ? void 0 : d.active)}
        }`), t;
    }, j = () => {
      var n, s, v, d, y, g, S, k, C, I, W, D, b, w, B, A, H, O;
      let t = "";
      return e.value.title && ((n = e.value.title) != null && n.default) && (t += `.${l} .el-step__main .el-step__title {${i(e.value.title.default)}}`), e.value.title && ((s = e.value.title) != null && s.active) && (t += `.${l} .el-step__main .el-step__title.is-process {${i(e.value.title.active)}}`), e.value.title && ((v = e.value.title) != null && v.finish) && (t += `.${l} .el-step__main .el-step__title.is-finish {${i(e.value.title.finish)}}`), e.value.title && ((d = e.value.title) != null && d.error) && (t += `.${l} .el-step__main .el-step__title.is-error {${i(e.value.title.error)}}`), e.value.line && ((y = e.value.line) != null && y.default) && (t += `.${l} .el-step__head .el-step__line {${i(e.value.line.default)}}`), e.value.line && ((g = e.value.line) != null && g.active) && (t += `.${l} .el-step__head.is-process .el-step__line {${i(e.value.line.active)}}`), e.value.line && ((S = e.value.line) != null && S.finish) && (t += `.${l} .el-step__head.is-finish .el-step__line {${i(e.value.line.finish)}}`), e.value.line && ((k = e.value.line) != null && k.error) && (t += `.${l} .el-step__head.is-error .el-step__line {${i(e.value.line.error)}}`), e.value.desc && ((C = e.value.desc) != null && C.default) && (t += `.${l} .el-step__main .el-step__description {${i(e.value.desc.default)}}`), e.value.desc && ((I = e.value.desc) != null && I.active) && (t += `.${l} .el-step__main .el-step__description.is-process {${i(e.value.desc.active)}}`), e.value.desc && ((W = e.value.desc) != null && W.finish) && (t += `.${l} .el-step__main .el-step__description.is-finish {${i(e.value.desc.finish)}}`), e.value.desc && ((D = e.value.desc) != null && D.error) && (t += `.${l} .el-step__main .el-step__description.is-error {${i(e.value.desc.error)}}`), e.value.icon && ((b = e.value.icon) != null && b.default) && (t += `.${l} .el-step__head .el-step__icon {${i(e.value.icon.default)}}`), e.value.icon && ((w = e.value.icon) != null && w.active) && (t += `.${l} .el-step__head.is-process .el-step__icon {${i(e.value.icon.active)}}`), e.value.icon && ((B = e.value.icon) != null && B.finish) && (t += `.${l} .el-step__head.is-finish .el-step__icon {${i(e.value.icon.finish)}}`), e.value.icon && ((A = e.value.icon) != null && A.error) && (t += `.${l} .el-step__head.is-error .el-step__icon {${i(e.value.icon.error)}}`), e.value.designIcon && ((H = e.value.designIcon) != null && H.default) && (t += `
        .${l} .el-step__icon img.icon-default {${i(e.value.designIcon.default)}}`), e.value.designIcon && ((O = e.value.designIcon) != null && O.active) && (t += `
        .${l} .el-step__icon img.icon-active {${i(e.value.designIcon.active)}}`), t;
    }, m = () => {
      const t = document.querySelector(`.${l} .style-container`);
      t && (t.innerHTML = `<style>${R()}${j()}</style>`);
    }, q = (t) => {
      t !== o.value && (c.value[o.value] = !1, c.value[t] = !0, o.value = t), h(a.data), u.value = t, T("step-click", JSON.parse(JSON.stringify(r.listData[t])));
    }, z = (t) => {
      let n = "";
      return t < u.value ? n = "finish" : t === u.value ? n = p.value ? "error" : "process" : n = "wait", n;
    }, h = (t) => {
      r.listData = [], t.length && (r.count += 1, r.listData = t.map((n, s) => ({
        title: n.title || "-",
        desc: n.desc || "-",
        defaultIcon: n.normalSrc || "",
        activeIcon: n.activeSrc || "",
        status: z(s)
      })));
    };
    return _(() => a.cStyle, () => {
      m();
    }, {
      deep: !0
    }), _(() => [a.data, a.active, a.onDesign, a.isError], () => {
      h(a.data);
    }, {
      deep: !0
    }), _(() => [a.data, a.keyWord, a.defaultChecked, a.tab], () => {
      if (a.data.length && (c.value = new Array(a.data.length).fill(!1)), a.data.length > 0 && a.keyWord !== "" && a.defaultChecked !== "") {
        const t = a.data.findIndex((n) => String(n[a.keyWord]) === String(a.defaultChecked));
        t >= 0 && E(() => {
          c.value[t] = !0, o.value = t;
        });
      } else
        a.tab && E(() => {
          c.value[0] = !0, o.value = 0;
        });
    }, { deep: !0, immediate: !0 }), _(() => a.active, () => {
      u.value = a.active - 1;
    }, { immediate: !0 }), F(() => {
      m();
      const t = K(a.data);
      h(t);
    }), {
      ...G(r),
      curStyle: e,
      curHasSerial: J,
      curDesign: M,
      curActive: u,
      curStatus: p,
      curDirection: P,
      filterAttrs: Q,
      domOnlyClass: l,
      handleClick: q,
      stateCollection: c
    };
  }
});
export {
  Z as default
};
