// @ths/c-common-base-button/lib/index.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('https://unpkg.com/vue@next'), require('element-plus')) :
  typeof define === 'function' && define.amd ? define(['exports', 'https://unpkg.com/vue@next', 'element-plus'], factory) :
  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global["C-common-base-button"] = {}, global.Vue, global.elementPlus));
})(this, (function (exports, vue, elementPlus) { 'use strict';

  var script$3 = vue.defineComponent({
      name: 'TButton',
      props: {
          // 按钮文字
          text: {
              type: String,
              required: true,
              default: '按钮',
          },
          type: {
              type: String,
              default: '',
          },
          // 按钮尺寸
          size: {
              type: String,
              default: 'mini',
          },
          plain: {
              type: Boolean,
              default: false,
          },
          // 按钮图标名称
          icon: {
              type: String,
              default: '',
          },
          // 按钮图标尺寸
          iconSize: {
              type: Number,
              default: 13,
          },
      },
      emits: ['click'],
      setup(props, { emit }) {
          // 响应式数据
          const state = vue.reactive({});
          /**
           * 处理按钮点击事件
           */
          const handleButtonClick = () => {
              emit('click');
          };
          return {
              ...vue.toRefs(state),
              handleButtonClick,
          };
      },
  });

  const _hoisted_1$3 = { key: 1 };

  function render$3(_ctx, _cache, $props, $setup, $data, $options) {
    const _component_el_icon = vue.resolveComponent("el-icon");
    const _component_el_button = vue.resolveComponent("el-button");

    return (vue.openBlock(), vue.createBlock(_component_el_button, {
      type: _ctx.type,
      plain: _ctx.plain,
      size: _ctx.size,
      onClick: _ctx.handleButtonClick
    }, {
      default: vue.withCtx(() => [
        vue.createCommentVNode(" 按钮图标 "),
        (_ctx.icon)
          ? (vue.openBlock(), vue.createBlock(_component_el_icon, {
              key: 0,
              size: _ctx.iconSize
            }, {
              default: vue.withCtx(() => [
                (vue.openBlock(), vue.createBlock(vue.resolveDynamicComponent(_ctx.icon)))
              ]),
              _: 1 /* STABLE */
            }, 8 /* PROPS */, ["size"]))
          : vue.createCommentVNode("v-if", true),
        vue.createCommentVNode(" 自定义内容 "),
        vue.renderSlot(_ctx.$slots, "default"),
        vue.createCommentVNode(" 按钮文字 "),
        (_ctx.text)
          ? (vue.openBlock(), vue.createElementBlock("span", _hoisted_1$3, vue.toDisplayString(_ctx.text), 1 /* TEXT */))
          : vue.createCommentVNode("v-if", true)
      ]),
      _: 3 /* FORWARDED */
    }, 8 /* PROPS */, ["type", "plain", "size", "onClick"]))
  }

  script$3.render = render$3;
  script$3.__file = "package/ths/components/c-common-base-button/src/view/t-button.vue";

  var script$2 = vue.defineComponent({
      name: 'TExcelButton',
      props: {
          // 导出Excel对应的url,get请求，参数拼到url后
          exportExcelUrl: {
              type: String,
              default: '',
          },
          // 通过模板导出Excel对应的url，get请求，参数拼到url后
          exportExcelByTemplateUrl: {
              type: String,
              default: '',
          },
          // excel模板下载url
          templateDownloadUrl: {
              type: String,
              default: '',
          },
          // Excel导入url，post请求，参数通过uploadRequestParam传递
          uploadExcelUrl: {
              type: String,
              default: '',
          },
          // 导入携带的额外参数
          uploadRequestParam: {
              type: Object,
              default: () => ({}),
          },
          // 文件上传携带请求头
          uploadHeaders: {
              type: Object,
              default: () => ({}),
          },
          // 错误文件下载Url
          errorFileDownloadUrl: {
              type: String,
              default: '',
          },
          // 导出excel按钮文字&下拉框内容
          excelTextArr: {
              type: Array,
              default: () => ([
                  { text: 'Excel导入/导出示例' },
                  { text: '导入Excel' },
                  { text: '导出Excel' },
                  { text: '导出Excel(by模板)' },
              ])
          }
      },
      emits: ['on-upload-success'],
      setup(props, { emit }) {
          const uploadRef = vue.ref();
          const state = vue.reactive({
              showDialog: false,
              accept: '.xls, .xlsx',
              fileList: [],
              fileName: '',
              errorFileName: '',
              loading: false, // 导入loading
          });
          /**
           * Excel操作列表触发事件
           */
          function handleCommandExcel(command) {
              if (command === 'export') {
                  window.location.href = `${props.exportExcelUrl}`;
              }
              else if (command === 'exportByTemplate') {
                  window.location.href = `${props.exportExcelByTemplateUrl}`;
              }
              else {
                  state.showDialog = true;
              }
          }
          /**
           * 关闭导入excel弹框
           */
          function onClose() {
              state.showDialog = false;
          }
          /**
           * input框点击事件
           */
          function focusInput(event) {
              document.getElementById('singleSelectButton')?.click();
              event.target.blur();
          }
          /**
           * 下载模板
           */
          function handleDownloadModel() {
              if (props.templateDownloadUrl?.length) {
                  window.location.href = props.templateDownloadUrl;
              }
              else {
                  elementPlus.ElMessage.warning('请设置【templateDownloadUrl】');
              }
          }
          /**
           * 导入
           */
          function handleImport() {
              if (state.fileList.length && state.fileList[0].status === 'ready') {
                  if (props.uploadExcelUrl.length === 0) {
                      elementPlus.ElMessage.warning('请设置【uploadExcelUrl】对应值');
                      return;
                  }
                  state.loading = true;
                  uploadRef.value.submit();
              }
              else {
                  elementPlus.ElMessage.warning('请先选择文件！！！');
              }
          }
          /**
           * 上传文件前调用，如果返回false则停止上传
           */
          function beforeUpload(file) {
              const fileType = file.name.substring(file.name.lastIndexOf('.') + 1);
              if (!state.accept.includes(fileType)) {
                  elementPlus.ElMessage.warning(`只允许上传${state.accept}类型文件！！！`);
                  return false;
              }
              return true;
          }
          /**
           * 文件上传成功时调用
           */
          function onSuccess(response, file) {
              state.loading = false;
              if (response.code === 500) {
                  elementPlus.ElMessage.error(`导入失败，${response.message}`);
                  const tempFile = file;
                  tempFile.status = 'ready';
                  return;
              }
              if (response.code === 200 && response.data.checked) {
                  elementPlus.ElMessage.success('导入成功！！！');
                  state.showDialog = false;
                  emit('on-upload-success');
              }
              else {
                  const tempFile = file;
                  tempFile.status = 'ready';
                  state.errorFileName = response.data.errorTempFileName;
                  elementPlus.ElMessage.error(`导入失败，${response.data.errorMessage}`);
              }
          }
          /**
           * 文件上传失败时触发事件
           */
          function onError(file) {
              state.loading = false;
              const tempFile = file;
              tempFile.status = 'ready';
              state.fileList.push(tempFile);
              elementPlus.ElMessage.error('导入失败！！！');
          }
          /**
           * 添加文件、上传成功和上传失败时都会被调用
           * 此处主要处理文件名展示问题
           */
          function onChange(file) {
              if (file.status === 'ready') {
                  state.fileList = [];
                  state.fileName = file.name;
                  state.fileList.push(file);
              }
          }
          /**
           * 打开弹窗是触发
           */
          function onOpen() {
              state.fileList = [];
              state.fileName = '';
              state.errorFileName = '';
          }
          /**
           * 下载错误文件
           */
          function downLoadErrorFile() {
              if (!props.errorFileDownloadUrl || !props.errorFileDownloadUrl.length) {
                  elementPlus.ElMessage.warning('请设置错误文件下载URL');
                  return;
              }
              let errorFileUrl = '';
              if (props.errorFileDownloadUrl.indexOf('?') !== -1) {
                  errorFileUrl = `${props.errorFileDownloadUrl}&fileName=${state.errorFileName}`;
              }
              else {
                  errorFileUrl = `${props.errorFileDownloadUrl}?fileName=${state.errorFileName}`;
              }
              window.location.href = errorFileUrl;
          }
          return {
              ...vue.toRefs(state),
              handleCommandExcel,
              onClose,
              onOpen,
              focusInput,
              handleDownloadModel,
              handleImport,
              beforeUpload,
              onSuccess,
              onError,
              onChange,
              uploadRef,
              downLoadErrorFile,
          };
      },
  });

  const _hoisted_1$2 = { class: "common-base-excel" };
  const _hoisted_2 = /*#__PURE__*/vue.createTextVNode(" 导入Excel数据");
  const _hoisted_3 = { class: "m-t-sm m-b-sm flex flex-end" };
  const _hoisted_4 = /*#__PURE__*/vue.createTextVNode(" 下载模板 ");
  const _hoisted_5 = /*#__PURE__*/vue.createTextVNode(" 导入 ");
  const _hoisted_6 = /*#__PURE__*/vue.createTextVNode(" 取消 ");
  const _hoisted_7 = /*#__PURE__*/vue.createTextVNode(" 选择 ");
  const _hoisted_8 = { key: 0 };
  const _hoisted_9 = /*#__PURE__*/vue.createElementVNode("span", null, "错误文件：", -1 /* HOISTED */);

  function render$2(_ctx, _cache, $props, $setup, $data, $options) {
    const _component_document = vue.resolveComponent("document");
    const _component_el_icon = vue.resolveComponent("el-icon");
    const _component_arrow_down = vue.resolveComponent("arrow-down");
    const _component_el_button = vue.resolveComponent("el-button");
    const _component_upload = vue.resolveComponent("upload");
    const _component_el_dropdown_item = vue.resolveComponent("el-dropdown-item");
    const _component_download = vue.resolveComponent("download");
    const _component_tickets = vue.resolveComponent("tickets");
    const _component_el_dropdown_menu = vue.resolveComponent("el-dropdown-menu");
    const _component_el_dropdown = vue.resolveComponent("el-dropdown");
    const _component_info_filled = vue.resolveComponent("info-filled");
    const _component_upload_filled = vue.resolveComponent("upload-filled");
    const _component_el_input = vue.resolveComponent("el-input");
    const _component_el_upload = vue.resolveComponent("el-upload");
    const _component_el_dialog = vue.resolveComponent("el-dialog");

    return (vue.openBlock(), vue.createElementBlock("div", _hoisted_1$2, [
      vue.createVNode(_component_el_dropdown, {
        trigger: "click",
        onCommand: _ctx.handleCommandExcel
      }, {
        dropdown: vue.withCtx(() => [
          vue.createVNode(_component_el_dropdown_menu, { class: "dropdown-menu" }, {
            default: vue.withCtx(() => [
              (_ctx.uploadExcelUrl.length)
                ? (vue.openBlock(), vue.createBlock(_component_el_dropdown_item, {
                    key: 0,
                    command: "import"
                  }, {
                    default: vue.withCtx(() => [
                      vue.createVNode(_component_el_icon, { size: 13 }, {
                        default: vue.withCtx(() => [
                          vue.createVNode(_component_upload)
                        ]),
                        _: 1 /* STABLE */
                      }),
                      vue.createTextVNode(" " + vue.toDisplayString(_ctx.excelTextArr[1].text), 1 /* TEXT */)
                    ]),
                    _: 1 /* STABLE */
                  }))
                : vue.createCommentVNode("v-if", true),
              (_ctx.exportExcelUrl.length)
                ? (vue.openBlock(), vue.createBlock(_component_el_dropdown_item, {
                    key: 1,
                    command: "export"
                  }, {
                    default: vue.withCtx(() => [
                      vue.createVNode(_component_el_icon, { size: 13 }, {
                        default: vue.withCtx(() => [
                          vue.createVNode(_component_download)
                        ]),
                        _: 1 /* STABLE */
                      }),
                      vue.createTextVNode(" " + vue.toDisplayString(_ctx.excelTextArr[2].text), 1 /* TEXT */)
                    ]),
                    _: 1 /* STABLE */
                  }))
                : vue.createCommentVNode("v-if", true),
              (_ctx.exportExcelByTemplateUrl.length)
                ? (vue.openBlock(), vue.createBlock(_component_el_dropdown_item, {
                    key: 2,
                    command: "exportByTemplate"
                  }, {
                    default: vue.withCtx(() => [
                      vue.createVNode(_component_el_icon, { size: 13 }, {
                        default: vue.withCtx(() => [
                          vue.createVNode(_component_tickets)
                        ]),
                        _: 1 /* STABLE */
                      }),
                      vue.createTextVNode(" " + vue.toDisplayString(_ctx.excelTextArr[3].text), 1 /* TEXT */)
                    ]),
                    _: 1 /* STABLE */
                  }))
                : vue.createCommentVNode("v-if", true)
            ]),
            _: 1 /* STABLE */
          })
        ]),
        default: vue.withCtx(() => [
          vue.createVNode(_component_el_button, {
            class: "dropdown-button",
            size: "mini",
            plain: ""
          }, {
            default: vue.withCtx(() => [
              vue.createVNode(_component_el_icon, { size: 13 }, {
                default: vue.withCtx(() => [
                  vue.createVNode(_component_document)
                ]),
                _: 1 /* STABLE */
              }),
              vue.createElementVNode("span", null, vue.toDisplayString(_ctx.excelTextArr[0].text), 1 /* TEXT */),
              vue.createVNode(_component_el_icon, {
                size: 13,
                class: "el-icon--right"
              }, {
                default: vue.withCtx(() => [
                  vue.createVNode(_component_arrow_down)
                ]),
                _: 1 /* STABLE */
              })
            ]),
            _: 1 /* STABLE */
          })
        ]),
        _: 1 /* STABLE */
      }, 8 /* PROPS */, ["onCommand"]),
      vue.createCommentVNode(" 导入Excel弹框 "),
      vue.createVNode(_component_el_dialog, {
        "custom-class": "excel-import",
        "model-value": _ctx.showDialog,
        "close-on-click-modal": false,
        "close-on-press-escape": false,
        onClose: _ctx.onClose,
        onOpen: _ctx.onOpen
      }, {
        title: vue.withCtx(() => [
          vue.createElementVNode("span", null, [
            vue.createVNode(_component_el_icon, {
              size: 16,
              style: {"vertical-align":"-2px"}
            }, {
              default: vue.withCtx(() => [
                vue.createVNode(_component_info_filled)
              ]),
              _: 1 /* STABLE */
            }),
            _hoisted_2
          ])
        ]),
        default: vue.withCtx(() => [
          vue.createElementVNode("div", _hoisted_3, [
            vue.createVNode(_component_el_button, {
              size: "small",
              type: "primary",
              plain: "",
              onClick: _ctx.handleDownloadModel
            }, {
              default: vue.withCtx(() => [
                _hoisted_4
              ]),
              _: 1 /* STABLE */
            }, 8 /* PROPS */, ["onClick"]),
            vue.createVNode(_component_el_button, {
              size: "small",
              type: "primary",
              plain: "",
              loading: _ctx.loading,
              onClick: _ctx.handleImport
            }, {
              default: vue.withCtx(() => [
                _hoisted_5
              ]),
              _: 1 /* STABLE */
            }, 8 /* PROPS */, ["loading", "onClick"]),
            vue.createVNode(_component_el_button, {
              size: "small",
              type: "primary",
              plain: "",
              onClick: _ctx.onClose
            }, {
              default: vue.withCtx(() => [
                _hoisted_6
              ]),
              _: 1 /* STABLE */
            }, 8 /* PROPS */, ["onClick"])
          ]),
          vue.createVNode(_component_el_input, {
            modelValue: _ctx.fileName,
            "onUpdate:modelValue": _cache[0] || (_cache[0] = $event => ((_ctx.fileName) = $event)),
            placeholder: "请选择excel文件...",
            readonly: "",
            onFocus: _ctx.focusInput
          }, {
            prefix: vue.withCtx(() => [
              vue.createVNode(_component_el_icon, { class: "el-input__icon" }, {
                default: vue.withCtx(() => [
                  vue.createVNode(_component_upload_filled)
                ]),
                _: 1 /* STABLE */
              })
            ]),
            append: vue.withCtx(() => [
              vue.createVNode(_component_el_button, {
                size: "small",
                onClick: _ctx.focusInput
              }, {
                default: vue.withCtx(() => [
                  _hoisted_7
                ]),
                _: 1 /* STABLE */
              }, 8 /* PROPS */, ["onClick"])
            ]),
            _: 1 /* STABLE */
          }, 8 /* PROPS */, ["modelValue", "onFocus"]),
          vue.createVNode(_component_el_upload, {
            ref: "uploadRef",
            accept: _ctx.accept,
            "show-file-list": false,
            multiple: false,
            action: _ctx.uploadExcelUrl,
            data: _ctx.uploadRequestParam,
            "auto-upload": false,
            "before-upload": _ctx.beforeUpload,
            "on-error": _ctx.onError,
            "on-success": _ctx.onSuccess,
            "on-change": _ctx.onChange,
            "file-list": _ctx.fileList
          }, {
            default: vue.withCtx(() => [
              vue.createVNode(_component_el_button, {
                id: "singleSelectButton",
                style: {"display":"none"}
              })
            ]),
            _: 1 /* STABLE */
          }, 8 /* PROPS */, ["accept", "action", "data", "before-upload", "on-error", "on-success", "on-change", "file-list"]),
          (_ctx.errorFileName.length)
            ? (vue.openBlock(), vue.createElementBlock("div", _hoisted_8, [
                _hoisted_9,
                vue.createElementVNode("a", {
                  href: "javascript:void(0)",
                  onClick: _cache[1] || (_cache[1] = (...args) => (_ctx.downLoadErrorFile && _ctx.downLoadErrorFile(...args)))
                }, vue.toDisplayString(_ctx.errorFileName), 1 /* TEXT */)
              ]))
            : vue.createCommentVNode("v-if", true)
        ]),
        _: 1 /* STABLE */
      }, 8 /* PROPS */, ["model-value", "onClose", "onOpen"])
    ]))
  }

  script$2.render = render$2;
  script$2.__file = "package/ths/components/c-common-base-button/src/view/t-excel-button.vue";

  var script$1 = vue.defineComponent({
      name: 'TSearchButton',
      props: {
          // 搜索条件是否展开
          searchShowAll: {
              type: Boolean,
              required: true,
              default: false,
          },
          // 是否隐藏展开按钮
          isShowFoldBtn: {
              type: Boolean,
              default: true,
          },
          // 查询按钮名称
          searchBtnName: {
              type: String,
              default: '查询',
          },
      },
      emits: ['search', 'reset', 'toggle-fold-state'],
      setup(props, { emit }) {
          // 响应式数据
          const state = vue.reactive({
              // 展开/收起按钮名称
              buttonName: '展开',
              // 展开/收起按钮icon标签
              buttonIcon: 'arrow-down',
          });
          /**
           * 查询
           */
          const handleSearch = () => {
              emit('search');
          };
          /**
           * 重置
           */
          const handleReset = () => {
              emit('reset');
          };
          /**
           * 切换 查询条件的 展开与折叠状态
           */
          const toggleFoldState = () => {
              if (!props.searchShowAll) {
                  state.buttonName = '收起';
                  state.buttonIcon = 'arrow-up';
              }
              else {
                  state.buttonName = '展开';
                  state.buttonIcon = 'arrow-down';
              }
              emit('toggle-fold-state', !props.searchShowAll);
          };
          return {
              ...vue.toRefs(state),
              handleSearch,
              handleReset,
              toggleFoldState,
          };
      },
  });

  const _hoisted_1$1 = /*#__PURE__*/vue.createTextVNode(" 重置 ");

  function render$1(_ctx, _cache, $props, $setup, $data, $options) {
    const _component_el_button = vue.resolveComponent("el-button");
    const _component_el_icon = vue.resolveComponent("el-icon");

    return (vue.openBlock(), vue.createElementBlock(vue.Fragment, null, [
      vue.createVNode(_component_el_button, {
        type: "primary",
        size: "small",
        onClick: _ctx.handleSearch
      }, {
        default: vue.withCtx(() => [
          vue.createTextVNode(vue.toDisplayString(_ctx.searchBtnName), 1 /* TEXT */)
        ]),
        _: 1 /* STABLE */
      }, 8 /* PROPS */, ["onClick"]),
      vue.createVNode(_component_el_button, {
        size: "small",
        plain: "",
        onClick: _ctx.handleReset
      }, {
        default: vue.withCtx(() => [
          _hoisted_1$1
        ]),
        _: 1 /* STABLE */
      }, 8 /* PROPS */, ["onClick"]),
      (_ctx.isShowFoldBtn)
        ? (vue.openBlock(), vue.createBlock(_component_el_button, {
            key: 0,
            type: "text",
            size: "small",
            onClick: _ctx.toggleFoldState
          }, {
            default: vue.withCtx(() => [
              vue.createVNode(_component_el_icon, { size: 14 }, {
                default: vue.withCtx(() => [
                  (vue.openBlock(), vue.createBlock(vue.resolveDynamicComponent(_ctx.buttonIcon)))
                ]),
                _: 1 /* STABLE */
              }),
              vue.createTextVNode(" " + vue.toDisplayString(_ctx.buttonName), 1 /* TEXT */)
            ]),
            _: 1 /* STABLE */
          }, 8 /* PROPS */, ["onClick"]))
        : vue.createCommentVNode("v-if", true)
    ], 64 /* STABLE_FRAGMENT */))
  }

  script$1.render = render$1;
  script$1.__file = "package/ths/components/c-common-base-button/src/view/t-search-button.vue";

  var script = vue.defineComponent({
      name: 'TTextButton',
      props: {
          // 按钮文字
          text: {
              type: String,
              required: true,
              default: '按钮',
          },
          // 是否显示右侧竖线（最后一个按钮不需要）
          showSplit: {
              type: Boolean,
              required: false,
              default: true,
          },
          // 是否禁用
          isDisabled: {
              type: Boolean,
              default: false,
          },
          // 删除按钮
          isDelete: {
              type: Boolean,
              default: false,
          },
      },
      emits: ['click'],
      setup(props, { emit }) {
          // 响应式数据
          const state = vue.reactive({});
          /**
           * 处理按钮点击事件
           */
          const handleButtonClick = () => {
              emit('click');
          };
          return {
              ...vue.toRefs(state),
              handleButtonClick,
          };
      },
  });

  const _hoisted_1 = {
    key: 0,
    class: "text-group-line"
  };

  function render(_ctx, _cache, $props, $setup, $data, $options) {
    const _component_el_button = vue.resolveComponent("el-button");

    return (vue.openBlock(), vue.createElementBlock(vue.Fragment, null, [
      vue.createCommentVNode(" 文字按钮 "),
      vue.createVNode(_component_el_button, {
        class: vue.normalizeClass(["text-group-button", _ctx.isDelete ? 'delete-text' : '']),
        size: "mini",
        type: "text",
        disabled: _ctx.isDisabled,
        onClick: _cache[0] || (_cache[0] = $event => (_ctx.handleButtonClick()))
      }, {
        default: vue.withCtx(() => [
          vue.createElementVNode("span", null, vue.toDisplayString(_ctx.text), 1 /* TEXT */)
        ]),
        _: 1 /* STABLE */
      }, 8 /* PROPS */, ["disabled", "class"]),
      vue.createCommentVNode(" 右侧竖线 "),
      (_ctx.showSplit)
        ? (vue.openBlock(), vue.createElementBlock("span", _hoisted_1))
        : vue.createCommentVNode("v-if", true)
    ], 64 /* STABLE_FRAGMENT */))
  }

  script.render = render;
  script.__file = "package/ths/components/c-common-base-button/src/view/t-text-button.vue";

  // 存储组件列表
  const components = [
      script$3,
      script$2,
      script$1,
      script,
  ];
  const install = (app) => {
      components.forEach((component) => {
          app.component(component.name, component);
      });
  };
  var index = {
      install,
  };

  exports.TButton = script$3;
  exports.TExcelButton = script$2;
  exports.TSearchButton = script$1;
  exports.TTextButton = script;
  exports["default"] = index;
  exports.install = install;

  Object.defineProperty(exports, '__esModule', { value: true });

}));
