// @ths/design/es/components/area-selector-data/src/index.vue2.js
import { defineComponent as ka, computed as et, reactive as at, onMounted as tt, watch as za, toRefs as ot, nextTick as rt } from "vue";
import lt from "lodash/cloneDeep";
import it from "lodash/merge";
import { pinyin as nt } from "pinyin-pro";
import { getDomRandomClass as ct, setStyle as ut, getCssPropertyString as r } from "../../../utils/index.js";
const Ua = {
  REGION_NAME: "REGION_NAME",
  REGION_CODE: "REGION_CODE",
  REGION_PY: "REGION_PY",
  PARENT_REGION_CODE: "PARENT_REGION_CODE"
}, bt = ka({
  name: "TAreaSelectorData",
  props: {
    // 面包屑箭头样子
    separatorType: {
      type: String,
      default: ""
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 标题
        areaSelector: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域选择按钮
        selector: {
          default: {},
          hover: {},
          focus: {},
          active: {}
        },
        // 最后区域高亮
        lastRegion: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询容器
        areaBox: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询标题
        areaTitle: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询标题图标
        titleIcon: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询标题关闭按钮
        titleClose: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域容器
        areaContainer: {
          default: {},
          hover: {},
          active: {}
        },
        // 按钮容器
        areaButton: {
          default: {},
          hover: {},
          active: {}
        },
        // radio按钮
        areaSwitch: {
          default: {},
          hover: {},
          active: {}
        },
        // 操作按钮
        handleButton: {
          default: {},
          hover: {},
          focus: {},
          active: {}
        },
        // 选中高亮
        Active: {
          default: {},
          hover: {}
        }
      })
    },
    selected: {
      type: String,
      default: ""
    },
    // 默认省份区域
    thsDefaultRegionCode: {
      type: String,
      default: ""
    },
    // 默认省份区域名称
    regionCode: {
      type: String,
      default: ""
    },
    // 行政区划数据别名
    alias: {
      type: Object,
      default: () => Ua
    },
    // 行政区划数据
    divisionData: {
      type: Array,
      default: () => []
    },
    // 重点流域数据
    basinData: {
      type: Array,
      default: () => []
    },
    // 是否展示首字母排序
    showInitialSort: {
      type: Boolean,
      default: !0
    }
  },
  emits: ["search", "resetRegion", "select"],
  setup(c, { emit: R }) {
    const a = et(() => c.cStyle), o = ct("area-selector"), e = at({
      // 区域选择的显隐
      isShowBox: !1,
      // 选中区域
      radio1: "xz",
      // 区域列表dom
      list: null,
      // 默认选中类型
      selected: c.selected,
      // 默认渲染顶级区域编码
      thsDefaultRegionCode: c.thsDefaultRegionCode,
      // 当前选中区域编码
      REGION_CODE: c.regionCode,
      // 当前数据
      regionTool: {
        thsDefaultLevel: "",
        // 默认首字母编码
        currentProvinceName: "",
        // 当前渲染省名称
        currentProvinceCode: "",
        // 当前渲染省编码
        currentCityName: "",
        // 当前渲染市名称
        currentCityCode: "",
        // 当前渲染市编码
        currentCountyName: "",
        // 当前渲染县名称
        currentCountyCode: "",
        // 当前渲染县编码
        currentTownCode: "",
        // 当前乡镇编码
        currentTownName: "",
        // 当前乡镇名称
        currentRegionCode: "",
        // 当前渲染行政区编码
        currentRegionName: "",
        // 当前渲染行政区名称
        currentRegionType: "",
        // 当前渲染行政区类型
        tempRegionCode: "",
        // 临时行政区编码，选中未执行查询
        allRegionMap: [],
        // 所有行政区map
        regionLevelShowList: [],
        // 行政区的树结构
        regionType: ""
      },
      // 面包屑数据
      breadcrumbObj: {
        path: [{ name: "", code: c.thsDefaultRegionCode }],
        code: ""
      }
    }), Fa = () => {
      let t = "";
      return a.value.wrapper && (t += ut(a.value.wrapper, `.${o}`)), t;
    }, Za = () => {
      var i, l, u, v, n, d, s, h, E, $, O, N, m, y, _, I, p, x, D, G, S, M, w, A, B, P, L, j, Y, H, z, U, F, Z, q, J, K, Q, V, W, X, k, ee, ae, te, oe, re, le, ie, ne, ce, ue, se, de, ve, ge, fe, be, Re, Te, Ce, he, Ee, $e, Oe, Ne, me, ye, _e, Ie, pe, xe, De, Ge, Se, Me, we, Ae, Be, Pe, Le, je, Ye, He, ze, Ue, Fe, Ze, qe, Je, Ke, Qe, Ve, We, Xe, ke, ea, aa, ta, oa, ra, la, ia, na, ca, ua, sa, da, va, ga, fa, ba, Ra, Ta, Ca, ha, Ea, $a, Oa, Na, ma, ya, _a, Ia, pa, xa, Da, Ga, Sa, Ma, wa, Aa, Ba, Pa, La, ja, Ya, Ha;
      let t = "";
      return (l = (i = a.value) == null ? void 0 : i.areaSelector) != null && l.default && (t += `.${o}.t-area-selector-data .area-selector {
          ${r((u = a.value) == null ? void 0 : u.areaSelector.default)}
        }`), (n = (v = a.value) == null ? void 0 : v.areaSelector) != null && n.hover && (t += `.${o}.t-area-selector-data .area-selector:hover {
          ${r((d = a.value) == null ? void 0 : d.areaSelector.hover)}
        }`), (h = (s = a.value) == null ? void 0 : s.areaSelector) != null && h.active && (t += `.${o}.t-area-selector-data .area-selector.active {
          ${r((E = a.value) == null ? void 0 : E.areaSelector.active)}
        }`), (O = ($ = a.value) == null ? void 0 : $.selector) != null && O.default && (t += `.${o}.t-area-selector-data .area-selector .el-button{
          ${r((N = a.value) == null ? void 0 : N.selector.default)}
        }`), (y = (m = a.value) == null ? void 0 : m.selector) != null && y.hover && (t += `.${o}.t-area-selector-data .area-selector .el-button:hover {
          ${r((_ = a.value) == null ? void 0 : _.selector.hover)}
        }`), (p = (I = a.value) == null ? void 0 : I.selector) != null && p.focus && (t += `.${o}.t-area-selector-data .area-selector .el-button:focus {
          ${r((x = a.value) == null ? void 0 : x.selector.focus)}
        }`), (G = (D = a.value) == null ? void 0 : D.selector) != null && G.active && (t += `.${o}.t-area-selector-data .area-selector .el-button:active {
          ${r((S = a.value) == null ? void 0 : S.selector.active)}
        }`), (w = (M = a.value) == null ? void 0 : M.lastRegion) != null && w.default && (t += `.${o}.t-area-selector-data .area-selector .area-breadcrumb .el-breadcrumb__item.last-region .el-breadcrumb__inner{
          ${r((A = a.value) == null ? void 0 : A.lastRegion.default)}
        }`), (P = (B = a.value) == null ? void 0 : B.lastRegion) != null && P.hover && (t += `.${o}.t-area-selector-data .area-selector .area-breadcrumb .el-breadcrumb__item.last-region .el-breadcrumb__inner:hover {
          ${r((L = a.value) == null ? void 0 : L.lastRegion.hover)}
        }`), (Y = (j = a.value) == null ? void 0 : j.lastRegion) != null && Y.active && (t += `.${o}.t-area-selector-data .area-selector .area-breadcrumb .el-breadcrumb__item.last-region .el-breadcrumb__inner:active {
          ${r((H = a.value) == null ? void 0 : H.lastRegion.active)}
        }`), (U = (z = a.value) == null ? void 0 : z.areaBox) != null && U.default && (t += `.${o}.t-area-selector-data .area-box {
          ${r((F = a.value) == null ? void 0 : F.areaBox.default)}
        }`), (q = (Z = a.value) == null ? void 0 : Z.areaBox) != null && q.hover && (t += `.${o}.t-area-selector-data .area-box:hover {
          ${r((J = a.value) == null ? void 0 : J.areaBox.hover)}
        }`), (Q = (K = a.value) == null ? void 0 : K.areaBox) != null && Q.active && (t += `.${o}.t-area-selector-data .area-box.active {
          ${r((V = a.value) == null ? void 0 : V.areaBox.active)}
        }`), (X = (W = a.value) == null ? void 0 : W.areaTitle) != null && X.default && (t += `.${o}.t-area-selector-data .area-box .area-title{
          ${r((k = a.value) == null ? void 0 : k.areaTitle.default)}
        }`), (ae = (ee = a.value) == null ? void 0 : ee.areaTitle) != null && ae.hover && (t += `.${o}.t-area-selector-data .area-box .area-title:hover {
          ${r((te = a.value) == null ? void 0 : te.areaTitle.hover)}
        }`), (re = (oe = a.value) == null ? void 0 : oe.areaTitle) != null && re.active && (t += `.${o}.t-area-selector-data .area-box .area-title.active {
          ${r((le = a.value) == null ? void 0 : le.areaTitle.active)}
        }`), (ne = (ie = a.value) == null ? void 0 : ie.titleIcon) != null && ne.default && (t += `.${o}.t-area-selector-data .area-box .area-title .title-icon{
          ${r((ce = a.value) == null ? void 0 : ce.titleIcon.default)}
        }`), (se = (ue = a.value) == null ? void 0 : ue.titleIcon) != null && se.hover && (t += `.${o}.t-area-selector-data .area-box .area-title .title-icon:hover {
          ${r((de = a.value) == null ? void 0 : de.titleIcon.hover)}
        }`), (ge = (ve = a.value) == null ? void 0 : ve.titleIcon) != null && ge.active && (t += `.${o}.t-area-selector-data .area-box .area-title .title-icon.active {
          ${r((fe = a.value) == null ? void 0 : fe.titleIcon.active)}
        }`), (Re = (be = a.value) == null ? void 0 : be.titleClose) != null && Re.default && (t += `.${o}.t-area-selector-data .area-box .area-title .title-close{
          ${r((Te = a.value) == null ? void 0 : Te.titleClose.default)}
        }`), (he = (Ce = a.value) == null ? void 0 : Ce.titleClose) != null && he.hover && (t += `.${o}.t-area-selector-data .area-box .area-title .title-close:hover {
          ${r((Ee = a.value) == null ? void 0 : Ee.titleClose.hover)}
        }`), (Oe = ($e = a.value) == null ? void 0 : $e.titleClose) != null && Oe.active && (t += `.${o}.t-area-selector-data .area-box .area-title .title-close.active {
          ${r((Ne = a.value) == null ? void 0 : Ne.titleClose.active)}
        }`), (ye = (me = a.value) == null ? void 0 : me.areaContainer) != null && ye.default && (t += `.${o}.t-area-selector-data .area-box .area-container{
          ${r((_e = a.value) == null ? void 0 : _e.areaContainer.default)}
        }`), (pe = (Ie = a.value) == null ? void 0 : Ie.areaContainer) != null && pe.hover && (t += `.${o}.t-area-selector-data .area-box .area-container:hover {
          ${r((xe = a.value) == null ? void 0 : xe.areaContainer.hover)}
        }`), (Ge = (De = a.value) == null ? void 0 : De.areaContainer) != null && Ge.active && (t += `.${o}.t-area-selector-data .area-box .area-container.active {
          ${r((Se = a.value) == null ? void 0 : Se.areaContainer.active)}
        }`), (we = (Me = a.value) == null ? void 0 : Me.areaButton) != null && we.default && (t += `.${o}.t-area-selector-data .area-box .area-container .area-button{
          ${r((Ae = a.value) == null ? void 0 : Ae.areaButton.default)}
        }`), (Pe = (Be = a.value) == null ? void 0 : Be.areaButton) != null && Pe.hover && (t += `.${o}.t-area-selector-data .area-box .area-container .area-button:hover {
          ${r((Le = a.value) == null ? void 0 : Le.areaButton.hover)}
        }`), (Ye = (je = a.value) == null ? void 0 : je.areaButton) != null && Ye.active && (t += `.${o}.t-area-selector-data .area-box .area-container .area-button.active {
          ${r((He = a.value) == null ? void 0 : He.areaButton.active)}
        }`), (Ue = (ze = a.value) == null ? void 0 : ze.areaSwitch) != null && Ue.default && (t += `.${o}.t-area-selector-data .area-box .area-container .area-button .el-radio-button__inner{
          ${r((Fe = a.value) == null ? void 0 : Fe.areaSwitch.default)}
        }`), (qe = (Ze = a.value) == null ? void 0 : Ze.areaSwitch) != null && qe.hover && (t += `.${o}.t-area-selector-data .area-box .area-container .area-button .el-radio-button__inner:hover {
          ${r((Je = a.value) == null ? void 0 : Je.areaSwitch.hover)}
        }`), (Qe = (Ke = a.value) == null ? void 0 : Ke.areaSwitch) != null && Qe.active && (t += `.${o}  .area-box .area-container .area-button .el-radio-button__original-radio:checked + .el-radio-button__inner {
          ${r((Ve = a.value) == null ? void 0 : Ve.areaSwitch.active)}
        }`), (Xe = (We = a.value) == null ? void 0 : We.handleButton) != null && Xe.default && (t += `.${o}.t-area-selector-data .area-box .area-container .area-button .handle-button .el-button{
          ${r((ke = a.value) == null ? void 0 : ke.handleButton.default)}
        }`), (aa = (ea = a.value) == null ? void 0 : ea.handleButton) != null && aa.hover && (t += `.${o}.t-area-selector-data .area-box .area-container .area-button .handle-button .el-button:hover {
          ${r((ta = a.value) == null ? void 0 : ta.handleButton.hover)}
        }`), (ra = (oa = a.value) == null ? void 0 : oa.handleButton) != null && ra.focus && (t += `.${o}.t-area-selector-data .area-box .area-container .area-button .handle-button .el-button:focus {
          ${r((la = a.value) == null ? void 0 : la.handleButton.focus)}
        }`), (na = (ia = a.value) == null ? void 0 : ia.handleButton) != null && na.active && (t += `.${o}.t-area-selector-data .area-box .area-container .area-button .handle-button .el-button:active{
          ${r((ca = a.value) == null ? void 0 : ca.handleButton.active)}
        }`), (sa = (ua = a.value) == null ? void 0 : ua.cityContainer) != null && sa.default && (t += `.${o}.t-area-selector-data .area-box .area-container .area-city-list{
          ${r((da = a.value) == null ? void 0 : da.cityContainer.default)}
        }`), (ga = (va = a.value) == null ? void 0 : va.breadcrumbItem) != null && ga.default && (t += `.${o}.t-area-selector-data .area-selector .area-breadcrumb .el-breadcrumb__item .el-breadcrumb__inner{
          ${r((fa = a.value) == null ? void 0 : fa.breadcrumbItem.default)}
        }`), (Ra = (ba = a.value) == null ? void 0 : ba.litters) != null && Ra.default && (t += `.${o}.t-area-selector-data .area-box .area-container .area-litter{
          ${r((Ta = a.value) == null ? void 0 : Ta.litters.default)}
        }`), (ha = (Ca = a.value) == null ? void 0 : Ca.litterItem) != null && ha.default && (t += `.${o}.t-area-selector-data .area-box .area-container .area-litter .litter{
          ${r((Ea = a.value) == null ? void 0 : Ea.litterItem.default)}
        }`), (Oa = ($a = a.value) == null ? void 0 : $a.litterItemTag) != null && Oa.default && (t += `.${o}.t-area-selector-data .area-box .area-container .area-city-list .row .tag{
          ${r((Na = a.value) == null ? void 0 : Na.litterItemTag.default)}
        }`), (ya = (ma = a.value) == null ? void 0 : ma.countyTop) != null && ya.default && (t += `.${o}.t-area-selector-data .area-box .area-container .area-city-list .row .citys .city-container .city {
          ${r((_a = a.value) == null ? void 0 : _a.countyTop.default)}
        }`), (pa = (Ia = a.value) == null ? void 0 : Ia.countyItem) != null && pa.default && (t += `.${o}.t-area-selector-data .area-box .area-container .area-city-list .row .citys .city-container .countys .county {
          ${r((xa = a.value) == null ? void 0 : xa.countyItem.default)}
        }`), (Ga = (Da = a.value) == null ? void 0 : Da.Active) != null && Ga.default && (t += `.${o}.t-area-selector-data .area-box .area-container .area-litter .litter-active{
          ${r((Sa = a.value) == null ? void 0 : Sa.Active.default)}
        }`, t += `.${o}.t-area-selector-data .area-box .area-container .area-city-list .row .tag-active{
          ${r((Ma = a.value) == null ? void 0 : Ma.Active.default)}
        }`, t += `.${o}.t-area-selector-data .area-box .area-container .area-city-list .row .citys .city-container .city-active {
          ${r((wa = a.value) == null ? void 0 : wa.Active.default)}
        }`, t += `.${o}.t-area-selector-data .area-box .area-container .area-city-list .row .citys .city-container .countys .county-active {
          ${r((Aa = a.value) == null ? void 0 : Aa.Active.default)}
        }`), (Pa = (Ba = a.value) == null ? void 0 : Ba.Active) != null && Pa.hover && (t += `.${o}.t-area-selector-data .area-box .area-container .area-litter .litter-active:hover{
          ${r((La = a.value) == null ? void 0 : La.Active.default)}
        }`, t += `.${o}.t-area-selector-data .area-box .area-container .area-city-list .row .tag-active:hover{
          ${r((ja = a.value) == null ? void 0 : ja.Active.default)}
        }`, t += `.${o}.t-area-selector-data .area-box .area-container .area-city-list .row .citys .city-container .city-active:hover {
          ${r((Ya = a.value) == null ? void 0 : Ya.Active.default)}
        }`, t += `.${o}.t-area-selector-data .area-box .area-container .area-city-list .row .citys .city-container .countys .county-active:hover {
          ${r((Ha = a.value) == null ? void 0 : Ha.Active.default)}
        }`), t;
    }, T = () => {
      const t = document.querySelector(`.${o} .style-container`);
      t && (t.innerHTML = `<style>${Fa()}${Za()}</style>`);
    }, qa = () => {
      e.isShowBox = !e.isShowBox, rt(() => {
        var i;
        const t = document.getElementById(`${e.breadcrumbObj.code}`);
        t && ((i = e.list) == null || i.scroll(0, (t == null ? void 0 : t.offsetTop) - e.list.offsetTop));
      });
    }, Ja = (t) => {
      var l;
      e.breadcrumbObj.code = t;
      const i = document.getElementById(`${t}`);
      i && ((l = e.list) == null || l.scroll(0, (i == null ? void 0 : i.offsetTop) - e.list.offsetTop));
    }, g = (t) => {
      let i;
      return t.substr(4, 2) !== "00" ? i = "country" : t.substr(2, 2) !== "00" ? i = "city" : i = "province", i;
    }, f = (t) => {
      var l, u;
      if (e.breadcrumbObj.path = [], e.regionTool.currentRegionCode = t, e.regionTool.currentRegionName = ((u = (l = e.regionTool) == null ? void 0 : l.allRegionMap[t]) == null ? void 0 : u.REGION_NAME) || "", e.regionTool.currentRegionType = g(t), !e.regionTool.currentRegionName)
        return;
      e.regionTool.currentProvinceCode = `${e.regionTool.currentRegionCode.substr(0, 2)}0000000000`, t.substr(2, 2) !== "00" && e.regionTool.allRegionMap[e.regionTool.currentProvinceCode] && (e.regionTool.currentProvinceName = e.regionTool.allRegionMap[e.regionTool.currentProvinceCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentProvinceCode, name: e.regionTool.currentProvinceName })), e.regionTool.currentCityCode = `${e.regionTool.currentRegionCode.substr(0, 4)}00000000`, t.substr(4, 2) !== "00" && e.regionTool.allRegionMap[e.regionTool.currentCityCode] && (e.regionTool.currentCityName = e.regionTool.allRegionMap[e.regionTool.currentCityCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentCityCode, name: e.regionTool.currentCityName })), e.regionTool.currentCountyCode = `${e.regionTool.currentRegionCode.substr(0, 6)}000000`, t.substr(6, 3) !== "000" && e.regionTool.allRegionMap[e.regionTool.currentCountyCode] && (e.regionTool.currentCountyName = e.regionTool.allRegionMap[e.regionTool.currentCountyCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentCountyCode, name: e.regionTool.currentCountyName })), e.regionTool.currentTownCode = `${e.regionTool.currentRegionCode.substr(0, 9)}000`, t.substr(9, 3) !== "000" && e.regionTool.allRegionMap[e.regionTool.currentTownCode] && (e.regionTool.currentTownName = e.regionTool.allRegionMap[e.regionTool.currentTownCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentTownCode, name: e.regionTool.currentTownName }));
      const i = e.regionTool.allRegionMap[e.regionTool.currentRegionCode].REGION_NAME;
      i.length > 8 && `${i.substr(0, 8)}`, e.breadcrumbObj.path.push({ code: e.regionTool.currentRegionCode, name: e.regionTool.currentRegionName });
    }, C = (t) => {
      if (e.regionTool.tempRegionCode = t, !e.regionTool.allRegionMap[t] || t === e.thsDefaultRegionCode)
        return;
      const i = e.regionTool.allRegionMap[t].PARENT_REGION_CODE;
      e.breadcrumbObj.code = i === e.thsDefaultRegionCode ? e.regionTool.allRegionMap[t].REGION_PY.substring(0, 1) : e.regionTool.allRegionMap[i].REGION_PY.substring(0, 1);
    }, b = () => {
      var v;
      if (!c.divisionData.length)
        return;
      e.regionTool.regionLevelShowList = [], e.regionTool.allRegionMap = [];
      const t = [
        "A",
        "B",
        "C",
        "D",
        "E",
        "F",
        "G",
        "H",
        "I",
        "J",
        "K",
        "L",
        "M",
        "N",
        "O",
        "P",
        "Q",
        "R",
        "S",
        "T",
        "U",
        "V",
        "W",
        "X",
        "Y",
        "Z",
        "港澳"
      ], i = it(lt(Ua), c.alias), l = c.divisionData.map((n) => ({
        ...n,
        REGION_NAME: n[i.REGION_NAME] || "",
        REGION_CODE: n[i.REGION_CODE] || "",
        PARENT_REGION_CODE: n[i.PARENT_REGION_CODE] ?? "",
        REGION_PY: n[i.REGION_PY] || nt(n[i.REGION_NAME], { pattern: "initial" })[0].toUpperCase()
      }));
      e.regionTool.currentRegionCode = l[0].REGION_CODE, e.regionTool.currentRegionName = l[0].REGION_NAME;
      for (let n = 0; n < l.length; n++)
        l[n] && l[n].REGION_PY && (l[n].CHILD = []), e.regionTool.allRegionMap[l[n].REGION_CODE] = l[n];
      for (let n = 0; n < l.length; n++) {
        const d = l[n].PARENT_REGION_CODE, s = l[n].REGION_PY.substring(0, 1);
        l[n] && d && (d === e.thsDefaultRegionCode ? (e.regionTool.allRegionMap[s] || (e.regionTool.allRegionMap[s] = {}, e.regionTool.allRegionMap[s].CHILD = [], e.regionTool.allRegionMap[s].REGION_CODE = s, e.regionTool.allRegionMap[s].REGION_NAME = s), e.regionTool.allRegionMap[s].CHILD.push(l[n])) : e.regionTool.allRegionMap[d] && e.regionTool.allRegionMap[d].CHILD.push(l[n]));
      }
      for (let n = 0; n < t.length; n++) {
        const d = t[n];
        if (e.regionTool.allRegionMap[d] && e.regionTool.allRegionMap[d].CHILD != null && e.regionTool.allRegionMap[d].CHILD.length > 0) {
          const s = e.regionTool.allRegionMap[d];
          e.regionTool.regionLevelShowList.push(s);
        }
      }
      e.REGION_CODE && C(e.REGION_CODE), f(e.REGION_CODE ? e.REGION_CODE : e.thsDefaultRegionCode);
      const u = document.getElementById(`${e.breadcrumbObj.code}`);
      u && ((v = e.list) == null || v.scroll(0, (u == null ? void 0 : u.offsetTop) - e.list.offsetTop));
    }, Ka = (t) => {
      t !== "xz" ? e.regionTool.regionLevelShowList = [] : b();
    }, Qa = () => {
      e.isShowBox = !e.isShowBox;
    }, Va = () => {
      if (e.selected === "REGION_CODE") {
        f(e.regionTool.tempRegionCode);
        const t = {
          selected: "REGION_CODE",
          REGION_CODE: e.regionTool.currentRegionCode,
          REGION_NAME: e.regionTool.currentRegionName,
          REGION_TYPE: g(e.regionTool.currentRegionCode)
        };
        R("search", t);
      }
      e.isShowBox = !e.isShowBox;
    }, Wa = (t, i) => {
      var u;
      if (e.regionTool.tempRegionCode = t.code, e.selected === "REGION_CODE") {
        f(e.regionTool.tempRegionCode);
        const v = {
          selected: "REGION_CODE",
          REGION_CODE: e.regionTool.currentRegionCode,
          REGION_NAME: e.regionTool.currentRegionName,
          REGION_TYPE: g(e.regionTool.currentRegionCode)
        };
        R("select", v);
      }
      const l = document.getElementById(`${e.breadcrumbObj.code}`);
      l && ((u = e.list) == null || u.scroll(0, (l == null ? void 0 : l.offsetTop) - e.list.offsetTop));
    }, Xa = () => {
      var t;
      if (e.selected === "REGION_CODE") {
        f(e.REGION_CODE);
        const i = {
          selected: "REGION_CODE",
          REGION_CODE: e.regionTool.currentRegionCode,
          REGION_NAME: e.regionTool.currentRegionName,
          REGION_TYPE: g(e.regionTool.currentRegionCode)
        };
        R("resetRegion", i);
        const l = document.getElementById(`${e.breadcrumbObj.code}`);
        l && ((t = e.list) == null || t.scroll(0, (l == null ? void 0 : l.offsetTop) - e.list.offsetTop));
      }
      e.isShowBox = !e.isShowBox, b();
    };
    return tt(() => {
      T(), b();
    }), za(
      () => c.cStyle,
      () => {
        T();
      },
      { deep: !0 }
    ), za(() => [c.thsDefaultRegionCode, c.regionCode, c.selected, c.divisionData], () => {
      e.thsDefaultRegionCode = c.thsDefaultRegionCode, e.REGION_CODE = c.regionCode, e.selected = c.selected, b();
    }, { deep: !0 }), {
      curStyle: a,
      domOnlyClass: o,
      checkRegion: C,
      select: Wa,
      search: Va,
      resetRegion: Xa,
      switchSelector: qa,
      close: Qa,
      selectLitter: Ja,
      changeSearch: Ka,
      ...ot(e)
    };
  }
});
export {
  bt as default
};
