// @ths/design/es/components/area-selector-data/src/index.vue.js
import R from "./index.vue2.js";
import { resolveComponent as u, openBlock as s, createElementBlock as n, normalizeClass as c, createElementVNode as l, createVNode as a, withCtx as r, createTextVNode as _, Fragment as C, renderList as p, createBlock as N, toDisplayString as v, withDirectives as O, vShow as g, createCommentVNode as h } from "vue";
import "./index.vue3.js";
import I from "../../../_virtual/_plugin-vue_export-helper.js";
const G = { class: "area-selector" }, y = { class: "area-breadcrumb" }, w = {
  key: 0,
  class: "area-box"
}, S = { class: "area-title" }, A = { class: "area-container" }, L = { class: "area-button" }, T = { class: "handle-button" }, $ = { class: "area-litter" }, B = ["onClick"], V = {
  ref: "list",
  class: "area-city-list"
}, F = ["id"], j = { class: "citys" }, M = ["onClick"], z = { class: "city_text" }, H = { class: "countys" }, U = ["onClick"], q = { class: "county_text" };
function J(e, o, K, P, Q, W) {
  const m = u("el-button"), k = u("el-breadcrumb-item"), f = u("el-breadcrumb"), E = u("el-radio-button"), D = u("el-radio-group"), b = u("el-tooltip");
  return s(), n("div", {
    class: c(["t-area-selector-data", e.domOnlyClass])
  }, [
    l("div", G, [
      a(m, {
        class: "selector",
        onClick: e.switchSelector
      }, {
        default: r(() => o[2] || (o[2] = [
          _(" 区域选择 ")
        ])),
        _: 1,
        __: [2]
      }, 8, ["onClick"]),
      l("div", y, [
        a(f, { separator: e.separatorType }, {
          default: r(() => [
            (s(!0), n(C, null, p(e.breadcrumbObj.path, (t, i) => (s(), N(k, {
              key: t,
              class: c(i + 1 === e.breadcrumbObj.path.length ? "last-region" : ""),
              onClick: (d) => e.select(t, i)
            }, {
              default: r(() => [
                _(v(t.name), 1)
              ]),
              _: 2
            }, 1032, ["class", "onClick"]))), 128))
          ]),
          _: 1
        }, 8, ["separator"])
      ])
    ]),
    e.isShowBox ? (s(), n("div", w, [
      l("div", S, [
        o[3] || (o[3] = l("div", { class: "title-icon" }, null, -1)),
        o[4] || (o[4] = l("div", { class: "title-text" }, " 区域查询 ", -1)),
        l("div", {
          class: "title-close",
          onClick: o[0] || (o[0] = (...t) => e.close && e.close(...t))
        }, " x ")
      ]),
      l("div", A, [
        l("div", L, [
          a(D, {
            modelValue: e.radio1,
            "onUpdate:modelValue": o[1] || (o[1] = (t) => e.radio1 = t),
            onChange: e.changeSearch
          }, {
            default: r(() => [
              a(E, { label: "xz" }, {
                default: r(() => o[5] || (o[5] = [
                  _(" 行政区划 ")
                ])),
                _: 1,
                __: [5]
              }),
              a(E, { label: "ly" }, {
                default: r(() => o[6] || (o[6] = [
                  _(" 重点流域 ")
                ])),
                _: 1,
                __: [6]
              })
            ]),
            _: 1
          }, 8, ["modelValue", "onChange"]),
          l("div", T, [
            a(m, {
              label: "cx",
              onClick: e.search
            }, {
              default: r(() => o[7] || (o[7] = [
                _(" 查询 ")
              ])),
              _: 1,
              __: [7]
            }, 8, ["onClick"]),
            a(m, {
              label: "cz",
              onClick: e.resetRegion
            }, {
              default: r(() => o[8] || (o[8] = [
                _(" 重置 ")
              ])),
              _: 1,
              __: [8]
            }, 8, ["onClick"])
          ])
        ]),
        O(l("div", $, [
          (s(!0), n(C, null, p(e.regionTool.regionLevelShowList, (t) => (s(), n("a", {
            key: t,
            href: "javascript:void(0)",
            class: c(["litter", e.breadcrumbObj.code === t.REGION_CODE && e.regionTool.tempRegionCode != e.thsDefaultRegionCode ? "litter-active" : ""]),
            onClick: (i) => e.selectLitter(t.REGION_CODE)
          }, v(t.REGION_CODE), 11, B))), 128))
        ], 512), [
          [g, e.showInitialSort]
        ]),
        l("div", V, [
          (s(!0), n(C, null, p(e.regionTool.regionLevelShowList, (t) => (s(), n("div", {
            id: t.REGION_CODE,
            key: t,
            class: "row"
          }, [
            O(l("div", {
              class: c(["tag", e.breadcrumbObj.code === t.REGION_CODE && e.regionTool.tempRegionCode != e.thsDefaultRegionCode ? "tag-active" : ""])
            }, v(t.REGION_CODE), 3), [
              [g, e.showInitialSort]
            ]),
            l("div", j, [
              (s(!0), n(C, null, p(t.CHILD, (i) => (s(), n("div", {
                key: i,
                class: "city-container"
              }, [
                l("div", {
                  class: c(["city", e.regionTool.tempRegionCode === i.REGION_CODE ? "city-active" : ""]),
                  onClick: (d) => e.checkRegion(i.REGION_CODE)
                }, [
                  a(b, {
                    class: "box-item",
                    effect: "dark",
                    content: i.REGION_NAME,
                    placement: "top"
                  }, {
                    default: r(() => [
                      l("div", z, v(i.REGION_NAME), 1)
                    ]),
                    _: 2
                  }, 1032, ["content"])
                ], 10, M),
                l("div", H, [
                  (s(!0), n(C, null, p(i.CHILD, (d) => (s(), n("div", {
                    key: d,
                    class: c(["county", e.regionTool.tempRegionCode === d.REGION_CODE ? "county-active" : ""]),
                    onClick: (X) => e.checkRegion(d.REGION_CODE)
                  }, [
                    a(b, {
                      class: "box-item",
                      effect: "dark",
                      content: d.REGION_NAME,
                      placement: "top"
                    }, {
                      default: r(() => [
                        l("div", q, v(d.REGION_NAME), 1)
                      ]),
                      _: 2
                    }, 1032, ["content"])
                  ], 10, U))), 128))
                ])
              ]))), 128))
            ])
          ], 8, F))), 128))
        ], 512)
      ])
    ])) : h("", !0),
    o[9] || (o[9] = l("div", { class: "style-container" }, null, -1))
  ], 2);
}
const oe = /* @__PURE__ */ I(R, [["render", J]]);
export {
  oe as default
};
