// @ths/design/es/components/header-menu/src/index.vue.js
import m from "./index.vue2.js";
import { resolveComponent as l, openBlock as o, createBlock as t, normalizeClass as h, withCtx as a, createElementBlock as r, Fragment as d, renderList as p, createTextVNode as f, toDisplayString as y, createCommentVNode as u } from "vue";
import "./index.vue3.js";
import b from "../../../_virtual/_plugin-vue_export-helper.js";
function v(e, S, w, k, g, C) {
  const c = l("t-menu-item"), s = l("submenu"), i = l("t-menu");
  return o(), t(i, {
    class: h(["t-header-menu", e.themeStyle]),
    "theme-style": e.themeStyle,
    "menu-style": e.menuStyle,
    mode: e.mode,
    "menu-layout": e.menuLayout,
    "default-active": e.defaultActive,
    "active-menus": e.selfActiveMenus,
    width: e.width,
    "border-line": e.borderLine,
    collapsible: e.collapsible,
    folded: e.folded,
    "show-icon": e.showIcon,
    "c-style": e.cStyle,
    onSelect: e.handleSelect
  }, {
    default: a(() => [
      (o(!0), r(d, null, p(e.data, (n) => (o(), r(d, {
        key: n.label
      }, [
        !n.children || n.children && !n.children.length ? (o(), t(c, {
          key: 0,
          index: n.label,
          icon: n.icon,
          route: n
        }, {
          default: a(() => [
            f(y(n.label), 1)
          ]),
          _: 2
        }, 1032, ["index", "icon", "route"])) : u("", !0),
        n.children && n.children.length ? (o(), t(s, {
          key: 1,
          data: n,
          icon: n.icon,
          active: e.defaultActive
        }, null, 8, ["data", "icon", "active"])) : u("", !0)
      ], 64))), 128))
    ]),
    _: 1
  }, 8, ["theme-style", "class", "menu-style", "mode", "menu-layout", "default-active", "active-menus", "width", "border-line", "collapsible", "folded", "show-icon", "c-style", "onSelect"]);
}
const _ = /* @__PURE__ */ b(m, [["render", v]]);
export {
  _ as default
};
