// @ths/design/es/components/header-menu/src/index.vue2.js
import { defineComponent as r, ref as i, reactive as f, watch as u, toRefs as c } from "vue";
import d from "./sub-menu.vue.js";
import "../../menu-item/index.js";
import "../../menu/index.js";
import m from "../../menu/src/menu-item.vue.js";
import s from "../../menu/src/menu.vue.js";
const A = r({
  name: "THeaderMenu",
  components: {
    Submenu: d,
    TMenuItem: m,
    TMenu: s
  },
  props: {
    // // 主题风格
    themeStyle: {
      type: String,
      default: "light"
    },
    // 自定义主题色
    themeColor: {
      type: String,
      default: "var(--theme-color)"
    },
    // 模式 horizontal-水平 vertical-垂直
    mode: {
      type: String,
      default: "horizontal"
    },
    // 是否显示图标
    showIcon: {
      type: Boolean,
      default: !0
    },
    // 图标文字排列方向 horizontal-水平 vertical-垂直
    menuLayout: {
      type: String,
      default: "horizontal"
    },
    // 风格样式  下划线 line | 背景色 surface
    menuStyle: {
      type: String,
      default: "surface"
    },
    // 纵向菜单 菜单宽度
    width: {
      type: Number,
      default: 200
    },
    // 纵向菜单 是否显示下划线
    borderLine: {
      type: Boolean,
      default: !0
    },
    // 纵向菜单 是否可以折叠
    collapsible: {
      type: Boolean,
      default: !0
    },
    // 纵向菜单 折叠状态
    folded: {
      type: Boolean,
      default: !1
    },
    activeMenu: {
      type: String,
      default: ""
    },
    activeMenus: {
      type: Array,
      default: () => []
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        // 菜单
        wrapper: {
          default: {}
        }
      })
    },
    data: Array
  },
  emits: ["menu-select"],
  setup(n, { emit: a }) {
    const l = i("#333"), t = f({
      // 当前选中项
      defaultActive: n.activeMenu,
      // 选中项数组
      selfActiveMenus: n.activeMenus
    });
    function o(e) {
      t.defaultActive = e.index, t.selfActiveMenus = e.route.menus, a("menu-select", { activeMenus: t.selfActiveMenus, activeItem: e });
    }
    return u(() => n.activeMenu, (e) => {
      t.defaultActive = e;
    }), u(() => n.activeMenus, (e) => {
      t.selfActiveMenus = e;
    }), {
      ...c(t),
      color: l,
      handleSelect: o
    };
  }
});
export {
  A as default
};
