// @ths/design/es/components/header-menu/src/sub-menu.vue.js
import b from "./sub-menu.vue2.js";
import { resolveComponent as t, openBlock as o, createBlock as a, withCtx as r, createTextVNode as c, toDisplayString as l, createElementBlock as u, Fragment as i, renderList as f, createCommentVNode as d } from "vue";
import h from "../../../_virtual/_plugin-vue_export-helper.js";
function _(n, v, k, g, y, $) {
  const s = t("submenu"), m = t("t-submenu-item"), p = t("t-submenu");
  return o(), a(p, {
    icon: n.data.icon || "fa-home",
    index: n.data.label,
    route: n.data,
    active: n.active
  }, {
    title: r(() => [
      c(l(n.data.label), 1)
    ]),
    default: r(() => [
      (o(!0), u(i, null, f(n.data.children, (e) => (o(), u(i, {
        key: e.label
      }, [
        e.children && e.children.length ? (o(), a(s, {
          key: 0,
          data: e,
          icon: e.icon,
          active: n.active
        }, null, 8, ["data", "icon", "active"])) : d("", !0),
        !e.children || e.children && !e.children.length ? (o(), a(m, {
          key: 1,
          icon: e.icon || "fa-home",
          route: e,
          index: e.label
        }, {
          default: r(() => [
            c(l(e.label), 1)
          ]),
          _: 2
        }, 1032, ["icon", "route", "index"])) : d("", !0)
      ], 64))), 128))
    ]),
    _: 1
  }, 8, ["icon", "index", "route", "active"]);
}
const N = /* @__PURE__ */ h(b, [["render", _]]);
export {
  N as default
};
