// @ths/design/es/components/switch/src/index.vue2.js
import { defineComponent as _, computed as m, ref as d, watch as h, onMounted as y } from "vue";
import { filterAttrs as $, getDomRandomClass as p, getCssPropertyString as a, setStyle as c } from "../../../utils/index.js";
const x = _({
  name: "TSwitch",
  props: {
    value: {
      type: [String, Number, Boolean],
      default: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 开关背景
        core: {
          default: {},
          active: {}
        },
        // 开关点
        action: {
          default: {},
          active: {}
        },
        leftText: {
          default: {},
          active: {}
        },
        rightText: {
          default: {},
          active: {}
        }
      })
    }
  },
  emits: ["title-click"],
  setup(i, { emit: S }) {
    const e = m(() => i.cStyle), l = p("switch"), s = d(i.value);
    h(() => i.value, (t) => {
      s.value = t;
    });
    const f = () => {
      var o, n, u, v;
      let t = "";
      return (o = e.value.core) != null && o.active && (t += `
          .${l} .el-switch.is-checked .el-switch__core {
            ${a(e.value.core.active)}
          }
        `), (n = e.value.action) != null && n.active && (t += `
          .${l} .el-switch.is-checked .el-switch__core .el-switch__action {
            ${a(e.value.action.active)}
          }
        `), (u = e.value.leftText) != null && u.active && (t += `
          .${l} .el-switch.is-checked .el-switch__label--left {
            ${a(e.value.leftText.active)}
          }
        `), (v = e.value.rightText) != null && v.active && (t += `
          .${l} .el-switch.is-checked .el-switch__label--right {
            ${a(e.value.rightText.active)}
          }
        `), t;
    }, w = () => {
      let t = "";
      return e.value.wrapper && (t += c(e.value.wrapper, `.${l} .t-switch`)), e.value.core && (t += c(e.value.core, `.${l} .el-switch .el-switch__core`)), e.value.action && (t += c(e.value.action, `.${l} .el-switch .el-switch__core .el-switch__action`)), e.value.leftText && (t += c(e.value.leftText, `.${l} .el-switch .el-switch__label--left`)), e.value.rightText && (t += c(e.value.rightText, `.${l} .el-switch .el-switch__label--right`)), t;
    }, r = () => {
      const t = document.querySelector(`.${l} .style-container`);
      t && (t.innerHTML = `<style>${w()}${f()}</style>`);
    };
    return y(() => {
      r();
    }), h(() => i.cStyle, () => {
      r();
    }), {
      curValue: s,
      curStyle: e,
      domOnlyClass: l,
      filterAttrs: $
    };
  }
});
export {
  x as default
};
