// @ths/design/es/components/dialog-single-tree/src/index.vue.js
import u from "./index.vue2.js";
import { resolveComponent as o, openBlock as d, createBlock as r, withCtx as l, createElementVNode as n, createVNode as i, createTextVNode as s, toDisplayString as f, mergeProps as g, createCommentVNode as C } from "vue";
import "./index.vue3.js";
import b from "../../../_virtual/_plugin-vue_export-helper.js";
const k = { class: "dialog-body" }, h = { class: "dialog-main" }, T = { class: "dialog-button-box" };
function v(e, t, y, V, N, $) {
  const c = o("info-filled"), p = o("el-icon"), m = o("t-tree-single"), a = o("el-button"), _ = o("el-dialog");
  return d(), r(_, {
    "model-value": e.dialogVisible,
    "close-on-press-escape": !1,
    "before-close": e.handleCancel,
    "append-to-body": "",
    width: e.dialogWidth,
    "custom-class": "business-tree-dialog"
  }, {
    title: l(() => [
      n("span", null, [
        i(p, { size: 16 }, {
          default: l(() => [
            i(c)
          ]),
          _: 1
        }),
        s(" " + f(e.dialogTitle), 1)
      ])
    ]),
    default: l(() => [
      n("div", k, [
        n("div", h, [
          e.dialogVisible ? (d(), r(m, g({
            key: 0,
            ref: "singleTreeRef",
            "root-id": "root",
            "result-data-path": "mapList"
          }, e.$attrs, { onOnClickTree: e.handleTreeNodeClick }), null, 16, ["onOnClickTree"])) : C("", !0)
        ]),
        n("div", T, [
          i(a, {
            size: "small",
            onClick: e.handleCancel
          }, {
            default: l(() => t[0] || (t[0] = [
              s(" 取消 ")
            ])),
            _: 1,
            __: [0]
          }, 8, ["onClick"]),
          i(a, {
            size: "small",
            type: "primary",
            onClick: e.handleSure
          }, {
            default: l(() => t[1] || (t[1] = [
              s(" 确定 ")
            ])),
            _: 1,
            __: [1]
          }, 8, ["onClick"])
        ])
      ])
    ]),
    _: 1
  }, 8, ["model-value", "before-close", "width"]);
}
const S = /* @__PURE__ */ b(u, [["render", v]]);
export {
  S as default
};
