// @ths/design/es/components/dialog-single-tree/src/index.vue2.js
import { defineComponent as n, reactive as r, ref as c, watch as g, toRefs as p } from "vue";
import f from "./t-tree-single.vue.js";
const T = n({
  name: "TDialogSingleTree",
  components: {
    TTreeSingle: f
  },
  props: {
    // 弹窗是否显示
    dialogVisible: {
      type: Boolean,
      default: !1
    },
    // 弹窗标题
    dialogTitle: {
      type: String,
      default: "选择"
    },
    // 弹窗宽度
    dialogWidth: {
      type: String,
      default: "500px"
    }
  },
  emits: ["update:dialogVisible", "dialogClose", "dialogSure"],
  setup(u, { emit: a }) {
    const t = r({
      selectedData: {}
    }), l = c();
    g(() => {
      var e, o;
      return (o = (e = l.value) == null ? void 0 : e.tTreeSingleRef) == null ? void 0 : o.getCurrentNode();
    }, (e) => {
      !Object.keys(t.selectedData).length && e && (t.selectedData = e);
    }, {
      immediate: !0,
      deep: !0
    });
    const i = () => {
      a("update:dialogVisible", !1), a("dialogClose");
    }, d = (e) => {
      t.selectedData = e;
    }, s = () => {
      a("dialogSure", t.selectedData);
    };
    return {
      singleTreeRef: l,
      ...p(t),
      handleSure: s,
      handleTreeNodeClick: d,
      handleCancel: i
    };
  }
});
export {
  T as default
};
