// @ths/design/es/components/dialog-single-tree/src/t-tree-single.vue.js
import b from "./t-tree-single.vue2.js";
import { resolveComponent as t, resolveDirective as z, openBlock as s, createElementBlock as f, normalizeClass as u, createVNode as n, withCtx as i, createTextVNode as S, withDirectives as c, createBlock as m, createElementVNode as a, createCommentVNode as h, resolveDynamicComponent as w, toDisplayString as y, withModifiers as k, vShow as C } from "vue";
import B from "../../../_virtual/_plugin-vue_export-helper.js";
const I = ["title", "onMouseover", "onMouseleave"], K = { class: "ths-tree__node-append-text" };
function M(e, d, V, _, F, L) {
  const v = t("search"), r = t("el-icon"), D = t("el-button"), N = t("el-input"), g = t("Edit"), T = t("Delete"), E = t("el-tree"), $ = z("loading");
  return s(), f("div", {
    class: u(["height-full", { "ths-structure-tree": e.openStructureLine }])
  }, [
    n(N, {
      modelValue: e.filterText,
      "onUpdate:modelValue": d[0] || (d[0] = (l) => e.filterText = l),
      class: "m-t-sm m-b-xs",
      placeholder: "请输入关键字搜索",
      clearable: "",
      onClear: e.filterNodeByText
    }, {
      append: i(() => [
        n(D, {
          size: "small",
          onClick: e.filterNodeByText
        }, {
          default: i(() => [
            n(r, { size: 13 }, {
              default: i(() => [
                n(v)
              ]),
              _: 1
            }),
            d[1] || (d[1] = S(" 搜索 "))
          ]),
          _: 1,
          __: [1]
        }, 8, ["onClick"])
      ]),
      _: 1
    }, 8, ["modelValue", "onClear"]),
    c((s(), m(E, {
      ref: "tTreeSingleRef",
      "check-on-click-node": !0,
      "node-key": e.simpleData.idKey,
      data: e.treeNodes,
      props: e.treeProps,
      "default-expanded-keys": e.defaultExpandKeys,
      "default-expand-all": e.isDefaultExpandAll,
      "highlight-current": !0,
      "current-node-key": e.hiddenId,
      lazy: e.lazy,
      indent: e.openStructureLine ? 0 : 20,
      load: e.loadLazyTree,
      "empty-text": e.emptyText,
      "expand-on-click-node": e.expandOnClickNode,
      "filter-node-method": e.filterNode,
      onNodeClick: e.handleNodeClick
    }, {
      default: i(({ node: l, data: o }) => [
        a("span", {
          title: l.label,
          class: u(e.getTreeNodeClass(l, o)),
          onMouseover: (p) => e.handleMouseover(o),
          onMouseleave: (p) => e.handleMouseLeave(o)
        }, [
          a("span", null, [
            e.isCustomIcon ? (s(), f("i", {
              key: 0,
              class: u(`${o[e.simpleData.iconKey]} ths-tree__node-icon`)
            }, null, 2)) : h("", !0),
            e.isCustomIcon ? h("", !0) : (s(), m(r, {
              key: 1,
              size: 14,
              class: "ths-tree__node-icon"
            }, {
              default: i(() => [
                (s(), m(w(
                  o[e.simpleData.iconKey] ? o[e.simpleData.iconKey] : l.expanded ? e.defaultIcon.expandIcon : e.defaultIcon.collapseIcon
                )))
              ]),
              _: 2
            }, 1024)),
            a("span", null, y(e.textOverflowHidden && l.label.length > 10 ? l.label.slice(0, 8) + "…" : l.label), 1),
            a("span", K, y(o[e.simpleData.spanKey] ? o[e.simpleData.spanKey] : ""), 1)
          ]),
          a("div", null, [
            c(n(r, {
              onClick: k((p) => e.handleNodeEdit(o), ["stop"])
            }, {
              default: i(() => [
                n(g)
              ]),
              _: 2
            }, 1032, ["onClick"]), [
              [C, o[e.isShowEditField]]
            ]),
            c(n(r, {
              style: { "margin-left": "8px" },
              color: "#f56c6c",
              onClick: k((p) => e.handleNodeDel(o), ["stop"])
            }, {
              default: i(() => [
                n(T)
              ]),
              _: 2
            }, 1032, ["onClick"]), [
              [C, o[e.isShowDeleteField]]
            ])
          ])
        ], 42, I)
      ]),
      _: 1
    }, 8, ["node-key", "data", "props", "default-expanded-keys", "default-expand-all", "current-node-key", "lazy", "indent", "load", "empty-text", "expand-on-click-node", "filter-node-method", "onNodeClick"])), [
      [$, e.loading]
    ])
  ], 2);
}
const P = /* @__PURE__ */ B(b, [["render", M]]);
export {
  P as default
};
