// @ths/design/es/components/dialog-single-tree/src/t-tree-single.vue2.js
import { defineComponent as j, ref as F, reactive as A, computed as _, watch as D, toRefs as p, nextTick as q } from "vue";
import V from "qs";
import { Delete as v, Edit as G } from "@element-plus/icons";
import L from "./utils/http.js";
const Q = j({
  name: "TTreeSingle",
  components: {
    Delete: v,
    Edit: G
  },
  props: {
    // 请求 url 地址
    url: {
      type: String,
      required: !0
    },
    // 请求方式（GET、POST），目前只支持这两种方式
    type: {
      type: String,
      default: "GET"
    },
    // 请求参数
    param: {
      type: Object,
      default: () => ({})
    },
    // 是否使用 requestBody 接收参数（post 请求时使用），如果传 false，则会使用 qs.stringify() 对参数进行序列化处理
    useRequestBody: {
      type: Boolean,
      default: !0
    },
    // 返回数据在 data 中的位置，如：mapList
    resultDataPath: {
      type: String,
      default: "mapList"
    },
    // 主要业务字段
    simpleData: {
      type: Object,
      default: () => ({
        idKey: "treeId",
        // 数据中 id 对应的 key
        nameKey: "treeName",
        // 要展示的名称（节点文本）
        pIdKey: "treePId",
        // 数据中父节点标识 key
        iconKey: "icon",
        // 树节点图标 key
        spanKey: "appendInfo",
        // 在节点后拼接内容 key
        leafKey: "isLeaf",
        // 是否叶子节点的标识（只有异步树生效）
        leafValue: "true"
        // 是否叶子节点的值（只有异步树生效）
      })
    },
    // 根节点id，默认 null
    rootId: {
      type: String,
      required: !0
    },
    // 树节点前的图标
    defaultIcon: {
      type: Object,
      default: () => ({
        expandIcon: "folder-opened",
        // 展开图标
        collapseIcon: "folder"
        // 关闭时图标，叶子节点统一使用关闭时图标
      })
    },
    // 是否默认展开所有节点，默认 false
    defaultExpandAll: {
      type: Boolean,
      default: !1
    },
    // 默认展开的层数，默认值 0
    defaultExpandLayer: {
      type: Number,
      default: 0
    },
    // 是否只允许选中叶子节点，默认 false
    onlyCheckLeaf: {
      type: Boolean,
      default: !1
    },
    // 是否懒加载
    lazy: {
      type: Boolean,
      default: !1
    },
    // 文字溢出隐藏
    textOverflowHidden: {
      type: Boolean,
      default: !1
    },
    // 逗号分隔的默认选中的数据id
    hiddenId: {
      type: String,
      default: ""
    },
    // 是否展示删除按钮
    isShowDeleteButton: {
      type: Boolean,
      default: !1
    },
    // 配置删除的字段, isShowDeleteButton为true时必填
    isShowDeleteField: {
      type: String,
      default: "_delFlag"
    },
    // 是否暂时编辑按钮
    isShowEditButton: {
      type: Boolean,
      default: !1
    },
    // 配置删除的字段, isShowEditButton为true时必填
    isShowEditField: {
      type: String,
      default: "_editFlag"
    },
    // 是否是树形结构
    isTreeStructure: {
      type: Boolean,
      default: !1
    },
    // 是否开启结构线
    openStructureLine: {
      type: Boolean,
      default: !0
    },
    // 是否使用 elementplus 之外的自定义图标
    isCustomIcon: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["pre-load-tree", "handle-node-del", "tree-load-finished", "handle-node-edit", "on-click-tree"],
  setup(t, { emit: s, attrs: c }) {
    const u = F(), a = A({
      // 默认展开的节点 key，根据展开的层数计算得来
      defaultExpandKeys: [],
      // 点击节点的时候是否展开，在只允许勾选子节点时候展开
      expandOnClickNode: !1,
      // 树对应的 prop 属性，正常加载和懒加载不同，在初始化页面时初始化该值
      treeProps: {},
      // 平行结构数据
      listNodes: [],
      // 树型数据，根据 listNodes 转换来的数据
      treeNodes: [],
      // 是否展示加载动画
      loading: !1,
      // 异步树第一个节点内容
      lazyNodes: [],
      // 内容为空的时候展示的文本
      emptyText: "暂无数据",
      // 是否默认展开全部节点
      isDefaultExpandAll: _(() => t.hiddenId.length > 0 && !t.lazy ? !0 : t.defaultExpandAll),
      filterText: ""
    }), N = (e) => {
      let l = !1;
      e.children ? t.simpleData.leafKey && e[t.simpleData.leafKey] && e[t.simpleData.leafKey] === t.simpleData.leafValue && (l = !0) : l = !0, !(t.onlyCheckLeaf && !l) && s("on-click-tree", e);
    }, y = (e) => new Promise((l, n) => {
      try {
        l(L.get(t.url, { params: e }));
      } catch (i) {
        n(i);
      }
    }), h = (e) => {
      let l = e;
      return t.useRequestBody || (l = V.stringify(e)), new Promise((n, i) => {
        try {
          n(L.post(t.url, l));
        } catch (d) {
          i(d);
        }
      });
    }, E = () => {
      let e = [];
      if (a.treeNodes.length > 0 && e.push(a.treeNodes[0]), e.length > 0)
        for (let l = 1; l <= t.defaultExpandLayer; l += 1) {
          const n = [];
          e.forEach((i) => {
            i.children && (a.defaultExpandKeys.push(i[t.simpleData.idKey]), i[t.simpleData.iconKey] || (i[t.simpleData.iconKey] = t.defaultIcon.expandIcon), i.children.forEach((d) => {
              n.push(d);
            }));
          }), e = n;
        }
    }, w = (e) => {
      const l = [], { idKey: n, pIdKey: i } = t.simpleData;
      if (!Array.isArray(e))
        return l;
      const d = {};
      return e.forEach((f) => {
        f[n] && (d[f[n]] = f);
      }), e.forEach((f) => {
        const o = d[f[i]];
        o ? (o.children || (o.children = [])).push(f) : l.push(f);
      }), l;
    };
    function m(e, l) {
      let n = e, i = l;
      i = i.replace(/\[(\w+)\]/g, ".$1"), i = i.replace(/^\./, "");
      const d = i.split(".");
      for (let f = 0; f < d.length && n; f++) {
        const o = d[f];
        if (o in n)
          n = n[o];
        else
          break;
      }
      return n;
    }
    const K = async () => {
      let e = {};
      t.type === "GET" ? e = await y(t.param) : t.type === "POST" && (e = await h(t.param)), e != null && e.data && (e == null ? void 0 : e.status) === 200 ? a.listNodes = m(e.data, t.resultDataPath) || [] : a.listNodes = [];
    }, g = async () => {
      a.treeNodes = await w(a.listNodes), E();
    }, T = async () => {
      if (a.loading = !0, a.treeProps = {
        label: t.simpleData.nameKey,
        children: "children",
        disabled: t.simpleData.disabled,
        isLeaf: t.simpleData.isLeaf
      }, t.onlyCheckLeaf && (a.expandOnClickNode = !0), await K(), a.listNodes.length === 0) {
        a.treeNodes = [], a.defaultExpandKeys = [], a.loading = !1;
        return;
      }
      c.onPreLoadTree ? s("pre-load-tree", a.listNodes, (e) => {
        e.length > 0 && (a.listNodes = e), g();
      }) : g(), a.loading = !1;
    }, x = async () => {
      a.treeProps = {
        label: t.simpleData.nameKey,
        isLeaf: "isLeaf"
      }, a.emptyText = "", t.onlyCheckLeaf && (a.expandOnClickNode = !0);
    }, S = async (e) => {
      const l = { ...t.param };
      e.level === 0 ? l[t.simpleData.pIdKey] = t.rootId : Object.keys(t.param).forEach((i) => {
        l[i] = e.data[t.param[i]] ? e.data[t.param[i]] : l[i];
      });
      let n;
      if (t.type === "GET" && (n = await y(l)), t.type === "POST" && (n = await h(l)), n != null && n.data && (n == null ? void 0 : n.status) === 200) {
        const i = m(n.data, t.resultDataPath) || [];
        i.length > 0 ? c.onPreLoadTree ? s("pre-load-tree", i, (d) => {
          d.length > 0 && (d.forEach((f) => {
            f[t.simpleData.leafKey] === t.simpleData.leafValue ? f.isLeaf = !0 : (f.isLeaf = !1, f.children || (f.children = []));
          }), a.lazyNodes = d);
        }) : (i.forEach((d) => {
          d[t.simpleData.leafKey] === t.simpleData.leafValue ? d.isLeaf = !0 : (d.isLeaf = !1, d.children || (d.children = []));
        }), a.lazyNodes = i) : a.lazyNodes = [];
      } else
        a.lazyNodes = [];
    }, B = (e) => {
      t.isShowDeleteButton && (e[t.isShowDeleteField] = !0), t.isShowEditButton && (e[t.isShowEditField] = !0);
    }, P = (e) => {
      e[t.isShowDeleteField] = !1, e[t.isShowEditField] = !1;
    }, k = (e) => {
      s("handle-node-del", e);
    }, I = (e) => {
      s("handle-node-edit", e);
    }, b = async (e, l) => {
      a.loading = !0, await S(e), l(a.lazyNodes), a.loading = !1;
    }, r = async () => {
      t.lazy ? await x() : await T(), q(() => {
        s("tree-load-finished");
      });
    };
    r(), D(
      () => t.url,
      () => {
        r();
      }
    ), D(
      () => t.param,
      () => {
        r();
      }
    );
    const O = () => {
      u.value.filter(a.filterText);
    }, z = (e, l) => e ? l[t.simpleData.nameKey].includes(e) : !0, C = (e, l) => ({
      "ths-tree__node": !t.onlyCheckLeaf || !l.children
    });
    return {
      ...p(a),
      handleNodeClick: N,
      loadLazyTree: b,
      handleMouseover: B,
      handleMouseLeave: P,
      handleNodeDel: k,
      handleNodeEdit: I,
      tTreeSingleRef: u,
      filterNode: z,
      filterNodeByText: O,
      getTreeNodeClass: C
    };
  }
});
export {
  Q as default
};
