// @ths/design/es/components/pagination-base/src/index.vue.js
import u from "./index.vue2.js";
import { resolveComponent as d, openBlock as f, createElementBlock as S, normalizeClass as y, normalizeStyle as P, createVNode as c, mergeProps as z, createElementVNode as B } from "vue";
import "./index.vue3.js";
import $ from "../../../_virtual/_plugin-vue_export-helper.js";
function w(e, r, C, v, b, k) {
  var n, t, o, p, i, l, s, g;
  const m = d("el-pagination");
  return f(), S("div", {
    class: y(["t-pagination-base", e.domOnlyClass]),
    style: P({
      marginTop: (t = (n = e.curStyle.wrapper) == null ? void 0 : n.default) == null ? void 0 : t.marginTop,
      marginLeft: (p = (o = e.curStyle.wrapper) == null ? void 0 : o.default) == null ? void 0 : p.marginLeft,
      marginRight: (l = (i = e.curStyle.wrapper) == null ? void 0 : i.default) == null ? void 0 : l.marginRight,
      marginBottom: (g = (s = e.curStyle.wrapper) == null ? void 0 : s.default) == null ? void 0 : g.marginBottom
    })
  }, [
    c(m, z({
      currentPage: e.curPage,
      "onUpdate:currentPage": r[0] || (r[0] = (a) => e.curPage = a),
      pageSize: e.curPageSize,
      "onUpdate:pageSize": r[1] || (r[1] = (a) => e.curPageSize = a)
    }, e.filterAttrs(e.$attrs), {
      "popper-class": e.popperOnlyClass + " t-pagination-base"
    }), null, 16, ["currentPage", "pageSize", "popper-class"]),
    r[2] || (r[2] = B("div", { class: "style-container" }, null, -1))
  ], 6);
}
const R = /* @__PURE__ */ $(u, [["render", w]]);
export {
  R as default
};
