// @ths/design/es/components/pagination-base/src/index.vue2.js
import { defineComponent as f, computed as o, onMounted as g, watch as d } from "vue";
import { filterAttrs as _, getDomRandomClass as u, getCssPropertyString as l, setStyle as p } from "../../../utils/index.js";
const y = f({
  name: "TPaginationBase",
  props: {
    currentPage: {
      type: Number,
      default: 1
    },
    pageSize: {
      type: Number,
      default: 10
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        pagerItem: {
          default: {},
          hover: {},
          active: {}
        },
        pager: {
          default: {},
          hover: {}
        },
        more: {
          default: {},
          hover: {}
        },
        total: {
          default: {}
        },
        sizesWrapper: {
          default: {}
        },
        sizesInput: {
          default: {},
          focus: {},
          hover: {}
        },
        sizesDropdownWrapper: {
          default: {}
        },
        sizesDropdownItem: {
          default: {},
          active: {},
          hover: {}
        },
        jumpWrapper: {
          default: {}
        },
        jumpInput: {
          default: {},
          focus: {},
          hover: {}
        }
      })
    }
  },
  emits: ["title-click"],
  setup(n, { emit: z }) {
    const e = o(() => n.cStyle), s = o(() => n.currentPage), c = o(() => n.pageSize), a = u("pagination-base"), i = u("pagination-base-popper"), m = () => {
      let t = "";
      return e.value.pagerItem && e.value.pagerItem.active && (t += `
          .${a} .el-pagination .el-pager .number.active {
            ${l(e.value.pagerItem.active)}
          }
        `), e.value.jumpInput && e.value.jumpInput.focus && (t += `
          .${a} .el-pagination .el-pagination__jump .el-input.is-focus .el-input__inner {
            ${l(e.value.sizesInput.focus)}
          }
        `), e.value.sizesInput && e.value.sizesInput.focus && (t += `
          .${a} .el-pagination .el-pagination__sizes .el-input.is-focus .el-input__inner {
            ${l(e.value.sizesInput.focus)}
          }
        `), e.value.sizesDropdownItem && e.value.sizesDropdownItem.active && (t += `
          .${i}.el-select__popper.el-popper[role=tooltip] .el-select-dropdown__item.selected {
            ${l(e.value.sizesDropdownItem.active)}
          }
        `), t;
    }, v = () => {
      let t = "";
      return e.value.wrapper && (t += p(e.value.wrapper, `.${a} .el-pagination`)), e.value.pagerItem && (t += p(e.value.pagerItem, `.${a} .el-pagination .el-pager .number:not(.active)`)), e.value.pager && (t += p(
        e.value.pager,
        `.${a} .el-pagination .btn-prev:not(.active), .${a} .el-pagination .btn-next:not(.active)`
      )), e.value.more && (t += p(e.value.more, `.${a} .el-pagination .el-pager .more`)), e.value.total && (t += p(e.value.total, `.${a} .el-pagination .el-pagination__total`)), e.value.jumpWrapper && (t += p(e.value.jumpWrapper, `.${a} .el-pagination .el-pagination__jump`)), e.value.jumpInput && (t += p(e.value.jumpInput, `.${a} .el-pagination .el-pagination__jump .el-input__inner`)), e.value.sizesWrapper && (t += p(e.value.sizesWrapper, `.${a} .el-pagination .el-pagination__sizes`)), e.value.sizesInput && (t += p(e.value.sizesInput, `.${a} .el-pagination .el-pagination__sizes .el-input__inner`)), e.value.sizesDropdownWrapper && (t += p(e.value.sizesDropdownWrapper, `.${i}.el-select__popper.el-popper[role=tooltip]`)), e.value.sizesDropdownItem && (t += p(e.value.sizesDropdownItem, `.${i}.el-select__popper.el-popper[role=tooltip] .el-select-dropdown__item`)), t;
    }, r = () => {
      const t = document.querySelector(`.${a} .style-container`);
      t && (t.innerHTML = `<style>${v()}${m()}</style>`);
    };
    return g(() => {
      r();
    }), d(() => n.cStyle, () => {
      r();
    }), {
      curStyle: e,
      domOnlyClass: a,
      popperOnlyClass: i,
      curPageSize: c,
      curPage: s,
      filterAttrs: _
    };
  }
});
export {
  y as default
};
