// @ths/design/es/components/input/src/index.vue2.js
import { defineComponent as c, computed as f, ref as y, watch as d, onMounted as h } from "vue";
import { filterAttrs as m, getDomRandomClass as _, setStyle as a } from "../../../utils/index.js";
const b = c({
  name: "TInput",
  props: {
    // 输入值
    inputValue: {
      type: String,
      default: ""
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {},
          focus: {},
          hover: {}
        },
        placeholder: {
          default: {}
        },
        disabledWrapper: {
          default: {}
        },
        disabledPlaceholder: {
          default: {}
        }
      })
    },
    // 类型
    type: {
      type: String,
      default: "text"
    }
  },
  emits: ["input", "change"],
  setup(n, { emit: u }) {
    const t = f(() => n.cStyle), l = _("input"), r = y(n.inputValue);
    d(() => n.inputValue, (e) => {
      r.value = e;
    });
    const p = () => {
      let e = "";
      return t.value.wrapper && (e += a(t.value.wrapper, `.${l} .el-input__inner`), e += a(t.value.wrapper, `.${l} .el-textarea .el-textarea__inner`)), t.value.placeholder && (e += a(t.value.placeholder, `.${l} .el-input__inner::placeholder`)), t.value.disabledWrapper && (e += a(t.value.disabledWrapper, `.${l} .is-disabled .el-input__inner`)), t.value.disabledPlaceholder && (e += a(t.value.disabledPlaceholder, `.${l} .is-disabled .el-input__inner::placeholder`)), e;
    }, i = () => {
      const e = document.querySelector(`.${l} .style-container`);
      e && (e.innerHTML = `<style>${p()}</style>`);
    }, o = (e) => {
      u("change", e);
    }, s = (e) => {
      r.value = e, u("input", e);
    };
    return d(() => n.cStyle, () => {
      i();
    }), h(() => {
      i();
    }), {
      curInputVal: r,
      curStyle: t,
      domOnlyClass: l,
      filterAttrs: m,
      handleChange: o,
      handleInputChange: s
    };
  }
});
export {
  b as default
};
