// @ths/design/es/components/input/src/index.vue.js
import d from "./index.vue2.js";
import { resolveComponent as f, openBlock as g, createElementBlock as y, normalizeClass as V, normalizeStyle as C, createVNode as I, mergeProps as h, withModifiers as w, createElementVNode as S } from "vue";
import "./index.vue3.js";
import $ from "../../../_virtual/_plugin-vue_export-helper.js";
function c(e, t, B, v, k, z) {
  var n, r, l, o, a, p, i, u;
  const m = f("el-input");
  return g(), y("div", {
    class: V(["t-input", e.domOnlyClass]),
    style: C({
      marginTop: (r = (n = e.curStyle.wrapper) == null ? void 0 : n.default) == null ? void 0 : r.marginTop,
      marginLeft: (o = (l = e.curStyle.wrapper) == null ? void 0 : l.default) == null ? void 0 : o.marginLeft,
      marginRight: (p = (a = e.curStyle.wrapper) == null ? void 0 : a.default) == null ? void 0 : p.marginRight,
      marginBottom: (u = (i = e.curStyle.wrapper) == null ? void 0 : i.default) == null ? void 0 : u.marginBottom
    })
  }, [
    I(m, h({
      modelValue: e.curInputVal,
      "onUpdate:modelValue": t[0] || (t[0] = (s) => e.curInputVal = s)
    }, e.filterAttrs(e.$attrs), {
      type: e.type,
      title: e.curInputVal,
      onInput: e.handleInputChange,
      onChange: w(e.handleChange, ["self"])
    }), null, 16, ["modelValue", "type", "title", "onInput", "onChange"]),
    t[1] || (t[1] = S("div", { class: "style-container" }, null, -1))
  ], 6);
}
const T = /* @__PURE__ */ $(d, [["render", c]]);
export {
  T as default
};
