// @ths/design/es/components/chart-stacked-column/src/index.vue.js
import a from "./index.vue2.js";
import { openBlock as n, createElementBlock as c, normalizeClass as i, normalizeStyle as o, withDirectives as l, createElementVNode as e, vShow as r } from "vue";
import "./index.vue3.js";
import m from "../../../_virtual/_plugin-vue_export-helper.js";
const p = {
  ref: "chartId",
  class: "chart"
}, d = { class: "empty" }, u = ["src"];
function y(t, s, f, h, k, v) {
  return n(), c("div", {
    class: i(["t-chart-stacked-column", t.domOnlyClass]),
    style: o({ ...t.cStyle.wrapper.default })
  }, [
    l(e("div", p, null, 512), [
      [r, t.checkEmpty]
    ]),
    l(e("div", d, [
      e("img", {
        class: "empty-img",
        style: o({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, u),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [r, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const $ = /* @__PURE__ */ m(a, [["render", y]]);
export {
  $ as default
};
