// @ths/design/es/components/chart-stacked-column/src/index.vue2.js
import { defineComponent as N, computed as E, ref as W, reactive as _, onMounted as R, onUnmounted as q, watch as L, nextTick as j, toRefs as P } from "vue";
import * as B from "echarts";
import c from "lodash/merge";
import H from "lodash/invertBy";
import p from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as U, DEFAULT_OPTIONS_AXIS as G } from "../../../utils/constant.js";
import { getDomRandomClass as K, setStyle as A } from "../../../utils/index.js";
const V = c(p(U), p(G)), m = c(
  p(V),
  {
    // backgroundColor: 'rgb(19, 38, 85)',
    tooltip: {
      axisPointer: {
        type: "shadow",
        color: "#FFFfff80"
      },
      padding: [0, 10],
      formatter: (e) => {
        let h = `<span>${e[0].axisValueLabel}</span>`;
        const u = e;
        return u.splice(e.length - 1, 1), u.forEach((o) => {
          h += `<p><span>${o.seriesName}:</span>  <span style="font-weight: 600;">${o.data}</span></p>`;
        }), h;
      }
    },
    legend: {
      left: "center",
      x: "33%",
      itemWidth: 15,
      itemGap: 60
    },
    calculable: !0,
    grid: {
      borderWidth: 0,
      top: "13%",
      bottom: "10%",
      left: "6%",
      right: "2%"
    },
    xAxis: {
      boundaryGap: !0,
      splitArea: {
        show: !1
      }
    },
    yAxis: {
      nameTextStyle: {
        align: "right",
        padding: [0, 15, 10, 0],
        fontSize: 16
      },
      splitNumber: 3,
      splitLine: {
        show: !0,
        lineStyle: {}
      },
      axisLabel: {
        margin: 8
      }
    }
  }
), te = N({
  name: "TChartStackedColumn",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 设置主轴
    mainAxis: {
      type: String,
      default: "x"
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 计数文本
    countLabel: {
      type: Object,
      default: () => ({})
    },
    bar: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: h }) {
    const u = K("chart-stacked-column"), o = E(() => e.cStyle), D = W(null);
    let f = null;
    const n = _({
      option: {},
      // 图表数据
      chartData: [],
      selectedLegendData: []
    });
    function w() {
      var t, i;
      (i = (t = e.seriesSetting) == null ? void 0 : t.data) != null && i.length && e.data && (n.chartData = e.seriesSetting.data.map((r) => {
        const { name: l, dataKey: y } = r;
        return {
          name: l || "--",
          data: e.data.map((s) => s[y])
        };
      }), n.selectedLegendData = n.chartData), n.selectedLegendData = e.options.series || n.chartData;
    }
    const z = () => {
      let t = "";
      return o.value.wrapper && (t += A(o.value.wrapper, `.${u}`)), o.value.emptyContainer && (t += A(o.value.emptyContainer, `.${u} .empty`)), o.value.emptyImg && (t += A(o.value.emptyImg, `.${u} .empty .empty-img`)), o.value.emptyText && (t += A(o.value.emptyText, `.${u} .empty .empty-text`)), t;
    }, v = () => {
      const t = document.querySelector(`.${u} .style-container`);
      t && (t.innerHTML = `<style>${z()}</style>`);
    }, M = E(() => {
      var t, i, r, l;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((s) => {
            var b;
            return (s == null ? void 0 : s.data) && ((b = s == null ? void 0 : s.data) == null ? void 0 : b.length) !== 0;
          });
        if (typeof ((i = e.options) == null ? void 0 : i.series) == "object")
          return ((l = (r = e.options.series) == null ? void 0 : r.data) == null ? void 0 : l.length) === 0;
      }
      return !!e.data.length;
    });
    function C() {
      var y, s, b, k, T;
      f.clear();
      const { color: t, series: i } = e.options;
      let r = [], l = {};
      i ? r = i.map((d, a) => {
        var g, S, x;
        const F = c(
          c(
            {
              type: "bar",
              stack: "总量",
              barMaxWidth: 40,
              emphasis: {
                focus: "none"
              },
              itemStyle: {
                color: {
                  type: "linear",
                  x: 1,
                  y: 1,
                  r: 1,
                  colorStops: [
                    {
                      offset: 0,
                      color: "rgba(255,255,255,0)"
                      // 0% 处的颜色
                    },
                    {
                      offset: 1,
                      color: t && t[a] || m.color[a] || "#FFFFFF00"
                      // 100% 处的颜色
                    }
                  ]
                },
                opacity: 0.7,
                borderColor: t && t[a] || m.color[a] || "#FFF",
                // 边框颜色
                borderWidth: 2,
                borderRadius: 2,
                shadowColor: t && t[a] || m.color[a] || "#FFF",
                // 图形阴影颜色
                shadowBlur: 10
                // 图形阴影
              }
            },
            e == null ? void 0 : e.bar
          ),
          (x = (S = (g = e.seriesSetting) == null ? void 0 : g.data) == null ? void 0 : S[a]) == null ? void 0 : x.style
        );
        return c({}, F, d);
      }) : e.data && ((s = (y = e.seriesSetting) == null ? void 0 : y.data) != null && s.length) && (r = (b = e.seriesSetting) == null ? void 0 : b.data.map((d, a) => {
        var g, S, x, I, $;
        const F = c(
          c(
            {
              name: (g = n.chartData[a]) == null ? void 0 : g.name,
              type: "bar",
              stack: "总量",
              barMaxWidth: 40,
              emphasis: {
                focus: "none"
              },
              itemStyle: {
                color: {
                  type: "linear",
                  x: 1,
                  y: 1,
                  r: 1,
                  colorStops: [
                    {
                      offset: 0,
                      color: "rgba(255,255,255,0)"
                      // 0% 处的颜色
                    },
                    {
                      offset: 1,
                      color: t && t[a] || m.color[a] || "#FFFFFF00"
                      // 100% 处的颜色
                    }
                  ]
                },
                opacity: 0.7,
                borderColor: t && t[a] || m.color[a] || "#FFF",
                // 边框颜色
                borderWidth: 2,
                borderRadius: 2,
                shadowColor: t && t[a] || m.color[a] || "#FFF",
                // 图形阴影颜色
                shadowBlur: 10
                // 图形阴影
              },
              data: e.data.length ? (S = n.chartData[a]) == null ? void 0 : S.data : []
            },
            e == null ? void 0 : e.bar
          ),
          ($ = (I = (x = e.seriesSetting) == null ? void 0 : x.data) == null ? void 0 : I[a]) == null ? void 0 : $.style
        );
        return c({}, F, d);
      }), (k = e.seriesSetting) != null && k.xAxisLabel ? l = (T = e.data) != null && T.length ? e.data.map((d) => d[e.seriesSetting.xAxisLabel]) : [] : l = e.data.map((d, a) => a)), n.selectedLegendData.length && r.push({
        name: "",
        type: "bar",
        stack: "总量",
        label: {
          show: !0,
          color: "#FFF",
          lineHeight: 16,
          padding: [5, 12],
          // 设置文字位置
          fontSize: 16,
          offset: [0, 0],
          // 是否对文字进行偏移
          backgroundColor: "#05AFC3 ",
          borderRadius: 5,
          position: "top",
          formatter(d) {
            let a = n.selectedLegendData.reduce((F, g) => F + g.data[d.dataIndex], 0);
            return a = Number.isInteger(a) ? a : a.toFixed(2), a;
          },
          ...e.countLabel
        },
        data: e.data.length ? Array(e.data.length).fill(0) : []
      }), e.mainAxis === "x" ? n.option = c(
        {
          series: r,
          xAxis: {
            data: l
          }
        },
        m
      ) : e.mainAxis === "y" && (n.option = c(
        {
          series: r,
          yAxis: {
            data: l
          }
        },
        m
      )), n.option = c(n.option, e.options), e.mainAxis === "y" && (n.option.xAxis.type = "value", n.option.yAxis.type = "category");
      try {
        f.setOption(n.option);
      } catch (d) {
        console.log(d);
      }
    }
    function O() {
      j(() => {
        f.resize();
      });
    }
    return R(() => {
      v(), D.value && (f = B.init(D.value), w(), C(), f.on("click", (t) => {
        h("com-click", t);
      }), f.on("legendselectchanged", (t) => {
        var l;
        const i = H(t.selected).true, r = ((l = e.options) == null ? void 0 : l.series) || n.chartData;
        n.selectedLegendData = r.filter((y) => i == null ? void 0 : i.includes(y.name)), f.setOption(n.option), h("legend-select", t);
      })), window.addEventListener("resize", O);
    }), q(() => {
      window.removeEventListener("resize", O);
    }), L(
      [
        () => e.data,
        () => e.seriesSetting,
        () => e.countLabel,
        () => e.options,
        () => e.mainAxis
      ],
      () => {
        w(), C();
      },
      {
        deep: !0
      }
    ), L(
      () => e.bar,
      () => {
        C();
      },
      {
        deep: !0
      }
    ), L(
      () => e.cStyle,
      () => {
        v(), j(() => {
          f.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      ...P(n),
      chartId: D,
      curStyle: o,
      checkEmpty: M,
      domOnlyClass: u
    };
  }
});
export {
  te as default
};
