// @ths/design/es/components/title/src/index.vue.js
import a from "./index.vue2.js";
import { resolveComponent as c, openBlock as t, createElementBlock as l, normalizeClass as r, createElementVNode as i, createCommentVNode as o, createVNode as d, createBlock as f } from "vue";
import "./index.vue3.js";
import u from "../../../_virtual/_plugin-vue_export-helper.js";
const m = { class: "t-title-lf" }, p = ["src"], _ = { class: "t-title-rt" }, y = ["src"];
function k(e, s, C, h, T, g) {
  const n = c("t-text");
  return t(), l("header", {
    class: r(["t-title", e.domOnlyClass])
  }, [
    i("div", m, [
      e.curPrefixIcon ? (t(), l("img", {
        key: 0,
        class: "t-title-lf__icon",
        src: e.curPrefixIcon,
        alt: "图标",
        draggable: "false"
      }, null, 8, p)) : o("", !0),
      e.curPrefixIconType ? (t(), l("div", {
        key: 1,
        class: r(`t-title__shape t-title__${e.curPrefixIconType}`)
      }, null, 2)) : o("", !0),
      d(n, {
        tip: !1,
        "edit-key": "title",
        label: e.curTitle,
        class: "t-title-lf__text",
        onClick: e.handleTitleClick,
        onModifyText: e.handleModifyTextChange
      }, null, 8, ["label", "onClick", "onModifyText"])
    ]),
    i("div", _, [
      e.suffixIcon ? (t(), l("img", {
        key: 0,
        class: "t-title__icon t-title-rt__icon",
        src: e.suffixIcon,
        alt: "图标",
        draggable: "false"
      }, null, 8, y)) : o("", !0),
      e.curDesc ? (t(), f(n, {
        key: 1,
        tip: !1,
        "edit-key": "desc",
        label: e.curDesc,
        class: "t-title-rt__text",
        onModifyText: e.handleModifyTextChange,
        onClick: e.handleDescClick
      }, null, 8, ["label", "onModifyText", "onClick"])) : o("", !0)
    ]),
    s[0] || (s[0] = i("div", { class: "style-container" }, null, -1))
  ], 2);
}
const x = /* @__PURE__ */ u(a, [["render", k]]);
export {
  x as default
};
