// @ths/design/es/components/title/src/index.vue2.js
import { defineComponent as I, computed as n, watch as v, onMounted as S } from "vue";
import { getDomRandomClass as T, setStyle as i } from "../../../utils/index.js";
import "../../text/index.js";
import _ from "../../text/src/index.vue.js";
const k = I({
  name: "TTitle",
  components: {
    TText: _
  },
  props: {
    // 左侧图标(图片地址)
    prefixIcon: {
      type: String,
      default: ""
    },
    // 左侧图标-形状
    prefixIconType: {
      type: String,
      default: ""
    },
    // 右侧图标(图片地址)
    suffixIcon: {
      type: String,
      default: ""
    },
    // 标题
    title: {
      type: String,
      default: "标题"
    },
    // 右侧文字-描述
    desc: {
      type: String,
      default: ""
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        prefixIcon: {
          default: {}
        },
        customIcon: {
          default: {}
        },
        suffixIcon: {
          default: {}
        },
        title: {
          default: {},
          hover: {}
        },
        desc: {
          default: {},
          hover: {}
        }
      })
    }
  },
  emits: ["modify-text", "title-click", "desc-click"],
  setup(l, { emit: o }) {
    const t = n(() => l.cStyle), u = n(() => l.prefixIcon), r = n(() => l.prefixIconType), s = n(() => l.suffixIcon), a = n(() => l.title), d = n(() => l.desc), c = T("title"), y = (e) => {
      o("modify-text", e);
    }, m = () => {
      o("title-click");
    }, p = () => {
      o("desc-click");
    }, x = () => {
      let e = "";
      return t.value.wrapper && (e += i(t.value.wrapper, `.${c}`)), t.value.customIcon && (e += i(t.value.customIcon, `.${c} .t-title__shape`)), t.value.prefixIcon && (e += i(t.value.prefixIcon, `.${c} .t-title-lf__icon`)), t.value.suffixIcon && (e += i(t.value.suffixIcon, `.${c} .t-title-rt__icon`)), t.value.title && (e += i(t.value.title, `.${c} .t-title-lf__text`)), t.value.desc && (e += i(t.value.desc, `.${c} .t-title-rt__text`)), e;
    }, f = () => {
      const e = document.querySelector(`.${c} .style-container`);
      e && (e.innerHTML = `<style>${x()}</style>`);
    };
    return v(() => l.cStyle, () => {
      f();
    }), S(() => {
      f();
    }), {
      handleModifyTextChange: y,
      handleTitleClick: m,
      handleDescClick: p,
      curStyle: t,
      curPrefixIcon: u,
      curSuffixIcon: s,
      curTitle: a,
      curDesc: d,
      domOnlyClass: c,
      curPrefixIconType: r
    };
  }
});
export {
  k as default
};
