// @ths/design/es/components/chart-tower-diagram/src/index.vue2.js
import { defineComponent as $, computed as G, ref as H, reactive as N, onMounted as P, onUnmounted as q, watch as v, nextTick as I, toRefs as F } from "vue";
import * as h from "echarts";
import f from "lodash/merge";
import p from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as U, DEFAULT_OPTIONS_AXIS as k } from "../../../utils/constant.js";
import { getDomRandomClass as R, setStyle as S } from "../../../utils/index.js";
const K = f(p(U), p(k)), b = f(
  p(K),
  {
    tooltip: {
      axisPointer: {
        type: "shadow"
      }
    },
    legend: {
      show: !1,
      left: "center",
      x: "33%",
      itemWidth: 15,
      itemGap: 60
    },
    grid: {
      left: "1%",
      right: "2%",
      bottom: "2%",
      top: "13%",
      containLabel: !0
    },
    xAxis: {
      triggerEvent: !0,
      boundaryGap: !0
    },
    yAxis: {
      name: "(吨)",
      splitNumber: 3,
      triggerEvent: !0
    }
  }
), Y = $({
  name: "TChartTowerDiagram",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => []
    },
    // 包含 系列样式 、别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // x 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    pictorialBar: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["legend-select"],
  setup(e, { emit: j }) {
    const y = R("chart-tower-diagram"), o = G(() => e.cStyle), w = H(null);
    let d = null;
    const u = N({
      // 定义图表数据
      chartData: []
    }), x = () => {
      var t, l, n;
      (l = (t = e.seriesSetting) == null ? void 0 : t.data) != null && l.length && ((n = e.data) != null && n.length) ? u.chartData = e.seriesSetting.data.map((s) => {
        const { name: m, dataKey: a } = s;
        return {
          name: m || "--",
          data: e.data.map((c) => c[a])
        };
      }) : u.chartData = [];
    }, B = () => {
      let t = "";
      return o.value.wrapper && (t += S(o.value.wrapper, `.${y}`)), o.value.emptyContainer && (t += S(o.value.emptyContainer, `.${y} .empty`)), o.value.emptyImg && (t += S(o.value.emptyImg, `.${y} .empty .empty-img`)), o.value.emptyText && (t += S(o.value.emptyText, `.${y} .empty .empty-text`)), t;
    }, A = () => {
      const t = document.querySelector(`.${y} .style-container`);
      t && (t.innerHTML = `<style>${B()}</style>`);
    }, _ = G(() => {
      var t, l, n, s;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((a) => {
            var c;
            return (a == null ? void 0 : a.data) && ((c = a == null ? void 0 : a.data) == null ? void 0 : c.length) !== 0;
          });
        if (typeof ((l = e.options) == null ? void 0 : l.series) == "object")
          return ((s = (n = e.options.series) == null ? void 0 : n.data) == null ? void 0 : s.length) === 0;
      }
      return !!e.data.length;
    }), C = () => {
      var a, c, L, O, z, T;
      d.clear();
      const { color: t, series: l } = e.options;
      let n = [], s = {};
      l ? n = l.map((r, i) => {
        var g;
        return f(
          {
            name: "数量",
            type: "pictorialBar",
            barMinHeight: 10,
            // 柱条最小高度，可用于防止某数据项的值过小而影响交互。
            barCategoryGap: "60%",
            // 同一系列的柱间距离，默认为类目间距的20%，可设固定值
            symbolSize: ["280%", "100%"],
            // 图形的大小
            symbol: "path://M0,10 L10,10 C5.5,10 5.5,5 5,0 C4.5,5 4.5,10 0,10 z",
            itemStyle: {
              color: new h.graphic.LinearGradient(0, 0, 0, 1, [
                {
                  offset: 1,
                  color: ((g = r == null ? void 0 : r.itemStyle) == null ? void 0 : g.color) || t && t[i] || b.color[i] || "#fff"
                },
                {
                  offset: 0,
                  color: "rgba(255,255,255,0)"
                }
              ]),
              opacity: 0.9
            },
            z: 10,
            ...e == null ? void 0 : e.pictorialBar
          },
          r
        );
      }) : (a = e.data) != null && a.length && ((L = (c = e.seriesSetting) == null ? void 0 : c.data) != null && L.length) && (n = (O = e.seriesSetting) == null ? void 0 : O.data.map((r, i) => {
        var E, M;
        const g = f(
          {
            name: (E = u.chartData[i]) == null ? void 0 : E.name,
            type: "pictorialBar",
            barMinHeight: 10,
            // 柱条最小高度，可用于防止某数据项的值过小而影响交互。
            barCategoryGap: "60%",
            // 同一系列的柱间距离，默认为类目间距的20%，可设固定值
            symbolSize: [r.width || "280%", "100%"],
            // 图形的大小
            symbol: "path://M0,10 L10,10 C5.5,10 5.5,5 5,0 C4.5,5 4.5,10 0,10 z",
            itemStyle: {
              color: new h.graphic.LinearGradient(0, 0, 0, 1, [
                {
                  offset: 1,
                  color: t && t[i] || b.color[i] || "#fff"
                },
                {
                  offset: 0,
                  color: "rgba(255,255,255,0)"
                }
              ]),
              opacity: 0.7
            },
            z: 10,
            data: ((M = u.chartData[i]) == null ? void 0 : M.data) || []
          },
          p(e == null ? void 0 : e.pictorialBar)
        );
        return f({}, g, {
          symbolSize: [r.width, "100%"],
          itemStyle: {
            color: new h.graphic.LinearGradient(0, 0, 0, 1, [
              {
                offset: 1,
                color: r.color || t && t[i] || b.color[i] || "#fff"
              },
              {
                offset: 0,
                color: "rgba(255,255,255,0)"
              }
            ]),
            opacity: 0.7
          }
        });
      }), (z = e.seriesSetting) != null && z.xAxisLabel ? s = (T = e.data) != null && T.length ? e.data.map((r) => r[e.seriesSetting.xAxisLabel]) : [] : s = e.data.map((r, i) => i));
      let m = f(
        {
          series: n,
          xAxis: {
            data: s
          }
        },
        b
      );
      m = f(m, e.options), d.setOption(m);
    }, D = () => {
      I(() => {
        d.resize();
      });
    };
    return P(() => {
      A(), w.value && (d = h.init(w.value), x(), C(), d.on("legendselectchanged", (t) => {
        j("legend-select", t);
      }), window.addEventListener("resize", D));
    }), q(() => {
      window.removeEventListener("resize", D);
    }), v(
      () => e.pictorialBar,
      (t) => {
        C();
      },
      {
        deep: !0
      }
    ), v(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting
      ],
      () => {
        x(), C();
      },
      {
        deep: !0
      }
    ), v(
      () => e.cStyle,
      () => {
        A(), I(() => {
          d.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      ...F(u),
      chartId: w,
      checkEmpty: _,
      domOnlyClass: y
    };
  }
});
export {
  Y as default
};
