// @ths/design/es/components/list-card-preview/src/index.vue.js
import M from "./index.vue2.js";
import { resolveComponent as V, openBlock as o, createElementBlock as r, normalizeClass as b, createElementVNode as s, Fragment as z, renderList as B, toDisplayString as n, withDirectives as E, normalizeStyle as S, vShow as N, withModifiers as U, createBlock as D, createCommentVNode as T } from "vue";
import "./index.vue3.js";
import O from "../../../_virtual/_plugin-vue_export-helper.js";
const P = { class: "wrapper" }, j = ["onClick", "onMouseenter"], q = { class: "list-hover-wrapper" }, A = { class: "list-item__title" }, G = { class: "title" }, H = { class: "list-item__content" }, J = ["src", "onClick"], Q = ["title"];
function R(e, t, W, X, Y, Z) {
  const L = V("el-image-viewer");
  return o(), r("div", {
    class: b(["t-list-card-preview", e.domOnlyClass])
  }, [
    s("div", P, [
      (o(!0), r(z, null, B(e.data, (i, d) => {
        var v, u, p, y, h, C, f, m, F, g, K, k, w, I, c, $;
        return o(), r("div", {
          key: i[(v = e.dataKeyField) == null ? void 0 : v.id],
          class: "list-item",
          onClick: (l) => e.handleCardClick(i, d, e.data),
          onMouseenter: (l) => {
            var a;
            return e.hoverContainerId = i[(a = e.dataKeyField) == null ? void 0 : a.id];
          },
          onMouseleave: t[0] || (t[0] = (l) => e.hoverContainerId = "")
        }, [
          s("div", q, [
            s("div", A, [
              s("div", G, n(i[(u = e.dataKeyField) == null ? void 0 : u.title]), 1),
              E(s("div", {
                class: "status",
                style: S({
                  ...(y = e.cStyle.status) == null ? void 0 : y[e.hoverContainerId === i[(p = e.dataKeyField) == null ? void 0 : p.id] ? "hover" : "default"],
                  ...(f = (h = e.statusList.find((l) => {
                    var a;
                    return l.status === i[(a = e.dataKeyField) == null ? void 0 : a.type];
                  })) == null ? void 0 : h.containerStyle) == null ? void 0 : f[e.hoverContainerId === i[(C = e.dataKeyField) == null ? void 0 : C.id] ? "hover" : "default"]
                })
              }, [
                s("span", {
                  style: S({
                    ...(F = e.cStyle.status) == null ? void 0 : F[e.hoverContainerId === i[(m = e.dataKeyField) == null ? void 0 : m.id] ? "hover" : "default"],
                    ...(k = (g = e.statusList.find((l) => {
                      var a;
                      return l.status === i[(a = e.dataKeyField) == null ? void 0 : a.type];
                    })) == null ? void 0 : g.fontStyle) == null ? void 0 : k[e.hoverContainerId === i[(K = e.dataKeyField) == null ? void 0 : K.id] ? "hover" : "default"]
                  })
                }, n(i[(w = e.dataKeyField) == null ? void 0 : w.typeName]), 5)
              ], 4), [
                [N, e.showType]
              ])
            ]),
            s("div", H, [
              s("img", {
                src: i[(I = e.dataKeyField) == null ? void 0 : I.imageUrl],
                alt: "图片",
                draggable: "false",
                onClick: U((l) => {
                  var a;
                  e.handleImageClick(i[(a = e.dataKeyField) == null ? void 0 : a.imageUrl], d), e.handleCardClick(i, d, e.data);
                }, ["stop"])
              }, null, 8, J)
            ]),
            s("div", {
              title: i[(c = e.dataKeyField) == null ? void 0 : c.desc],
              class: "list-item__footer"
            }, n(i[($ = e.dataKeyField) == null ? void 0 : $.desc]), 9, Q)
          ])
        ], 40, j);
      }), 128))
    ]),
    t[2] || (t[2] = s("div", { class: "style-container" }, null, -1)),
    e.previewImageVisible ? (o(), D(L, {
      key: 0,
      "hide-on-click-modal": !0,
      "close-on-press-escape": !0,
      "url-list": e.imageUrls,
      "z-index": "99",
      teleported: !0,
      "initial-index": e.previewImageIndex,
      fit: "cover",
      onClose: t[1] || (t[1] = (i) => e.previewImageVisible = !1)
    }, null, 8, ["url-list", "initial-index"])) : T("", !0)
  ], 2);
}
const ae = /* @__PURE__ */ O(M, [["render", R]]);
export {
  ae as default
};
