// @ths/design/es/components/list-card-preview/src/index.vue2.js
import { defineComponent as V, computed as j, reactive as D, onMounted as z, watch as A, toRefs as E } from "vue";
import { getDomRandomClass as G, setStyle as o, getCssPropertyString as i } from "../../../utils/index.js";
const X = V({
  name: "TListCardPreview",
  props: {
    // 是否显示卡片类型
    showType: {
      default: !0,
      type: Boolean
    },
    // 是否预览图片
    showPreview: {
      default: !0,
      type: Boolean
    },
    // 数据
    data: {
      type: Array,
      required: !0,
      default: () => []
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({})
    },
    // 数据字段别名
    dataKeyField: {
      type: Object,
      default: () => ({
        id: "id",
        title: "title",
        desc: "desc",
        type: "type",
        typeName: "typeName",
        imageUrl: "imageUrl"
      }),
      required: !0
    },
    // 右上角状态配置
    statusList: {
      type: Array,
      default: () => []
    }
  },
  emits: ["card-click"],
  setup(r, { emit: B }) {
    const e = j(() => r.cStyle), a = G("t-list-card-preview"), s = D({
      // 需要预览的图片数组
      imageUrls: r.showPreview ? j(() => r.data.map((t) => {
        var l;
        return t[((l = r.dataKeyField) == null ? void 0 : l.imageUrl) || "imageUrl"];
      })) : [],
      hoverContainerId: "",
      // 记录右上角状态容器的主键id
      previewImageIndex: 0,
      // 当前预览的图片下标
      previewImageVisible: !1
      // 是否预览图片
    }), M = () => {
      var l, u, n, m, p, d, f, c, w, y, h, $, _, g, I, C, S, U, P, H, W, b, k, q, x, F, K, L, O, T;
      let t = "";
      return (l = e.value.wrapper) != null && l.default && (t += o((u = e.value) == null ? void 0 : u.wrapper, `.${a} .wrapper`)), (n = e.value.listItem) != null && n.default && (t += `.${a} .wrapper .list-item {
          ${i((m = e.value.listItem) == null ? void 0 : m.default)}
        }`), (p = e.value.listItem) != null && p.default && (t += `.${a} .wrapper .list-item {
          ${i((d = e.value.listItem) == null ? void 0 : d.default)}
        }`), ((f = e.value.listItem) != null && f.hover || (c = e.value.listHoverWrapper) != null && c.hover) && (t += `.${a} .wrapper .list-item:hover .list-hover-wrapper {
          ${i((w = e.value.listHoverWrapper) == null ? void 0 : w.hover)}
        }`, t += `.${a} .wrapper .list-item:hover {
          ${i((y = e.value.listItem) == null ? void 0 : y.hover)}
        }`), (h = e.value.listHoverWrapper) != null && h.default && (t += `.${a} .wrapper .list-hover-wrapper {
          ${i(($ = e.value.listHoverWrapper) == null ? void 0 : $.default)}
        }`), (_ = e.value.titleWrapper) != null && _.default && (t += o((g = e.value) == null ? void 0 : g.titleWrapper, `.${a} .wrapper .list-item__title`)), (I = e.value.title) != null && I.default && (t += o((C = e.value) == null ? void 0 : C.title, `.${a} .wrapper .list-item__title .title`)), (S = e.value.status) != null && S.default && (t += o((U = e.value) == null ? void 0 : U.status, `.${a} .wrapper .list-item__title .status`)), (P = e.value.content) != null && P.default && (t += o((H = e.value) == null ? void 0 : H.content, `.${a} .wrapper .list-item__content`)), (W = e.value.content) != null && W.hover && (t += `.${a} .wrapper .list-item:hover .list-item__content {
          ${i((b = e.value.content) == null ? void 0 : b.hover)}
        }`), (k = e.value.content) != null && k.hover && (t += `.${a} .wrapper .list-item:hover .list-item__content {
          ${i((q = e.value.content) == null ? void 0 : q.hover)}
        }`), (x = e.value.image) != null && x.default && (t += o((F = e.value) == null ? void 0 : F.image, `.${a} .list-item__content img`)), (K = e.value.footer) != null && K.default && (t += o((L = e.value) == null ? void 0 : L.footer, `.${a} .list-item__footer`)), (O = e.value.footer) != null && O.hover && (t += `.${a} .wrapper .list-item:hover .list-item__footer {
          ${i((T = e.value.footer) == null ? void 0 : T.hover)}
        }`), t;
    }, N = (t, l) => {
      r.showPreview && (s.previewImageIndex = l, s.previewImageVisible = !0);
    }, v = () => {
      const t = document.querySelector(`.${a} .style-container`);
      t && (t.innerHTML = `<style>${M()}</style>`);
    };
    z(() => {
      v();
    }), A(() => r.cStyle, () => {
      v();
    }, { deep: !0 }), A(() => r.showPreview, () => {
      s.imageUrls = r.showPreview ? r.data.map((t) => {
        var l;
        return t[((l = r.dataKeyField) == null ? void 0 : l.imageUrl) || "imageUrl"];
      }) : [];
    });
    const R = (t, l, u) => {
      B("card-click", t, l, u);
    };
    return {
      ...E(s),
      curStyle: e,
      domOnlyClass: a,
      handleCardClick: R,
      handleImageClick: N
    };
  }
});
export {
  X as default
};
