// @ths/design/es/components/area-selector/src/index.vue2.js
import { defineComponent as Mo, computed as mo, reactive as Go, onMounted as wo, watch as No, toRefs as Bo, nextTick as Ao } from "vue";
import Lo from "axios";
import Po from "qs";
import { getDomRandomClass as jo, setStyle as Ho, getCssPropertyString as r } from "../../../utils/index.js";
const Fo = Mo({
  name: "TAreaSelector",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 标题
        areaSelector: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域选择按钮
        selector: {
          default: {},
          hover: {},
          focus: {},
          active: {}
        },
        // 最后区域高亮
        lastRegion: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询容器
        areaBox: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询标题
        areaTitle: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询标题图标
        titleIcon: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询标题关闭按钮
        titleClose: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域容器
        areaContainer: {
          default: {},
          hover: {},
          active: {}
        },
        // 按钮容器
        areaButton: {
          default: {},
          hover: {},
          active: {}
        },
        // radio按钮
        areaSwitch: {
          default: {},
          hover: {},
          active: {}
        },
        // 操作按钮
        handleButton: {
          default: {},
          hover: {},
          focus: {},
          active: {}
        },
        // 选中高亮
        Active: {
          default: {},
          hover: {}
        }
      })
    },
    // 默认省份区域
    url: {
      type: String,
      default: ""
    },
    selected: {
      type: String,
      default: ""
    },
    // 默认省份区域
    thsDefaultRegionCode: {
      type: String,
      default: ""
    },
    // 默认省份区域名称
    regionCode: {
      type: String,
      default: ""
    },
    // 数据请求方式
    autoRequest: {
      type: Boolean,
      default: !1
    },
    // 是否显示首字母排序，默认 true
    showInitialSort: {
      type: Boolean,
      default: !0
    }
  },
  emits: ["search", "resetRegion", "select"],
  setup(c, { emit: h }) {
    const a = mo(() => c.cStyle), t = jo("area-selector"), e = Go({
      // 区域选择的显隐
      isShowBox: !1,
      // 选中区域
      radio1: "xz",
      // 区域列表dom
      list: null,
      // 默认选中类型
      selected: c.selected,
      // 默认渲染顶级区域编码
      thsDefaultRegionCode: c.thsDefaultRegionCode,
      // 当前选中区域编码
      REGION_CODE: c.regionCode,
      // 当前数据
      regionTool: {
        thsDefaultLevel: "",
        // 默认首字母编码
        currentProvinceName: "",
        // 当前渲染省名称
        currentProvinceCode: "",
        // 当前渲染省编码
        currentCityName: "",
        // 当前渲染市名称
        currentCityCode: "",
        // 当前渲染市编码
        currentCountyName: "",
        // 当前渲染县名称
        currentCountyCode: "",
        // 当前渲染县编码
        currentTownCode: "",
        // 当前乡镇编码
        currentTownName: "",
        // 当前乡镇名称
        currentRegionCode: "",
        // 当前渲染行政区编码
        currentRegionName: "",
        // 当前渲染行政区名称
        currentRegionType: "",
        // 当前渲染行政区类型
        tempRegionCode: "",
        // 临时行政区编码，选中未执行查询
        allRegionMap: {},
        // 所有行政区map
        regionLevelShowList: [],
        // 行政区的树结构
        regionType: ""
      },
      // 面包屑数据
      breadcrumbObj: {
        path: [{ name: "", code: c.thsDefaultRegionCode }],
        code: ""
      },
      url: c.url
    }), yo = () => {
      let o = "";
      return a.value.wrapper && (o += Ho(a.value.wrapper, `.${t}`)), o;
    }, Io = () => {
      var l, i, v, n, u, s, f, $, E, O, m, N, y, I, p, _, x, S, D, M, G, w, B, A, L, P, j, H, Y, q, z, Z, F, J, K, Q, U, V, W, X, k, ee, oe, ae, te, re, le, ne, ie, ce, ue, se, ve, ge, de, fe, he, Re, be, Te, Ce, $e, Ee, Oe, me, Ne, ye, Ie, pe, _e, xe, Se, De, Me, Ge, we, Be, Ae, Le, Pe, je, He, Ye, qe, ze, Ze, Fe, Je, Ke, Qe, Ue, Ve, We, Xe, ke, eo, oo, ao, to, ro, lo, no, io, co, uo, so, vo, go, fo, ho, Ro, bo, To, Co, $o, Eo, Oo;
      let o = "";
      return (i = (l = a.value) == null ? void 0 : l.areaSelector) != null && i.default && (o += `.${t}.t-area-selector .area-selector {
          ${r((v = a.value) == null ? void 0 : v.areaSelector.default)}
        }`), (u = (n = a.value) == null ? void 0 : n.areaSelector) != null && u.hover && (o += `.${t}.t-area-selector .area-selector:hover {
          ${r((s = a.value) == null ? void 0 : s.areaSelector.hover)}
        }`), ($ = (f = a.value) == null ? void 0 : f.areaSelector) != null && $.active && (o += `.${t}.t-area-selector .area-selector.active {
          ${r((E = a.value) == null ? void 0 : E.areaSelector.active)}
        }`), (m = (O = a.value) == null ? void 0 : O.selector) != null && m.default && (o += `.${t}.t-area-selector .area-selector .el-button{
          ${r((N = a.value) == null ? void 0 : N.selector.default)}
        }`), (I = (y = a.value) == null ? void 0 : y.selector) != null && I.hover && (o += `.${t}.t-area-selector .area-selector .el-button:hover {
          ${r((p = a.value) == null ? void 0 : p.selector.hover)}
        }`), (x = (_ = a.value) == null ? void 0 : _.selector) != null && x.focus && (o += `.${t}.t-area-selector .area-selector .el-button:focus {
          ${r((S = a.value) == null ? void 0 : S.selector.focus)}
        }`), (M = (D = a.value) == null ? void 0 : D.selector) != null && M.active && (o += `.${t}.t-area-selector .area-selector .el-button:active {
          ${r((G = a.value) == null ? void 0 : G.selector.active)}
        }`), (B = (w = a.value) == null ? void 0 : w.lastRegion) != null && B.default && (o += `.${t}.t-area-selector .area-selector .area-breadcrumb .last-region .el-breadcrumb__inner{
          ${r((A = a.value) == null ? void 0 : A.lastRegion.default)}
        }`), (P = (L = a.value) == null ? void 0 : L.lastRegion) != null && P.hover && (o += `.${t}.t-area-selector .area-selector .area-breadcrumb .last-region .el-breadcrumb__inner:hover {
          ${r((j = a.value) == null ? void 0 : j.lastRegion.hover)}
        }`), (Y = (H = a.value) == null ? void 0 : H.lastRegion) != null && Y.active && (o += `.${t}.t-area-selector .area-selector .area-breadcrumb .last-region .el-breadcrumb__inner:active {
          ${r((q = a.value) == null ? void 0 : q.lastRegion.active)}
        }`), (Z = (z = a.value) == null ? void 0 : z.areaBox) != null && Z.default && (o += `.${t}.t-area-selector .area-box {
          ${r((F = a.value) == null ? void 0 : F.areaBox.default)}
        }`), (K = (J = a.value) == null ? void 0 : J.areaBox) != null && K.hover && (o += `.${t}.t-area-selector .area-box:hover {
          ${r((Q = a.value) == null ? void 0 : Q.areaBox.hover)}
        }`), (V = (U = a.value) == null ? void 0 : U.areaBox) != null && V.active && (o += `.${t}.t-area-selector .area-box.active {
          ${r((W = a.value) == null ? void 0 : W.areaBox.active)}
        }`), (k = (X = a.value) == null ? void 0 : X.areaTitle) != null && k.default && (o += `.${t}.t-area-selector .area-box .area-title{
          ${r((ee = a.value) == null ? void 0 : ee.areaTitle.default)}
        }`), (ae = (oe = a.value) == null ? void 0 : oe.areaTitle) != null && ae.hover && (o += `.${t}.t-area-selector .area-box .area-title:hover {
          ${r((te = a.value) == null ? void 0 : te.areaTitle.hover)}
        }`), (le = (re = a.value) == null ? void 0 : re.areaTitle) != null && le.active && (o += `.${t}.t-area-selector .area-box .area-title.active {
          ${r((ne = a.value) == null ? void 0 : ne.areaTitle.active)}
        }`), (ce = (ie = a.value) == null ? void 0 : ie.titleIcon) != null && ce.default && (o += `.${t}.t-area-selector .area-box .area-title .title-icon{
          ${r((ue = a.value) == null ? void 0 : ue.titleIcon.default)}
        }`), (ve = (se = a.value) == null ? void 0 : se.titleIcon) != null && ve.hover && (o += `.${t}.t-area-selector .area-box .area-title .title-icon:hover {
          ${r((ge = a.value) == null ? void 0 : ge.titleIcon.hover)}
        }`), (fe = (de = a.value) == null ? void 0 : de.titleIcon) != null && fe.active && (o += `.${t}.t-area-selector .area-box .area-title .title-icon.active {
          ${r((he = a.value) == null ? void 0 : he.titleIcon.active)}
        }`), (be = (Re = a.value) == null ? void 0 : Re.titleClose) != null && be.default && (o += `.${t}.t-area-selector .area-box .area-title .title-close{
          ${r((Te = a.value) == null ? void 0 : Te.titleClose.default)}
        }`), ($e = (Ce = a.value) == null ? void 0 : Ce.titleClose) != null && $e.hover && (o += `.${t}.t-area-selector .area-box .area-title .title-close:hover {
          ${r((Ee = a.value) == null ? void 0 : Ee.titleClose.hover)}
        }`), (me = (Oe = a.value) == null ? void 0 : Oe.titleClose) != null && me.active && (o += `.${t}.t-area-selector .area-box .area-title .title-close.active {
          ${r((Ne = a.value) == null ? void 0 : Ne.titleClose.active)}
        }`), (Ie = (ye = a.value) == null ? void 0 : ye.areaContainer) != null && Ie.default && (o += `.${t}.t-area-selector .area-box .area-container{
          ${r((pe = a.value) == null ? void 0 : pe.areaContainer.default)}
        }`), (xe = (_e = a.value) == null ? void 0 : _e.areaContainer) != null && xe.hover && (o += `.${t}.t-area-selector .area-box .area-container:hover {
          ${r((Se = a.value) == null ? void 0 : Se.areaContainer.hover)}
        }`), (Me = (De = a.value) == null ? void 0 : De.areaContainer) != null && Me.active && (o += `.${t}.t-area-selector .area-box .area-container.active {
          ${r((Ge = a.value) == null ? void 0 : Ge.areaContainer.active)}
        }`), (Be = (we = a.value) == null ? void 0 : we.areaButton) != null && Be.default && (o += `.${t}.t-area-selector .area-box .area-container .area-button{
          ${r((Ae = a.value) == null ? void 0 : Ae.areaButton.default)}
        }`), (Pe = (Le = a.value) == null ? void 0 : Le.areaButton) != null && Pe.hover && (o += `.${t}.t-area-selector .area-box .area-container .area-button:hover {
          ${r((je = a.value) == null ? void 0 : je.areaButton.hover)}
        }`), (Ye = (He = a.value) == null ? void 0 : He.areaButton) != null && Ye.active && (o += `.${t}.t-area-selector .area-box .area-container .area-button.active {
          ${r((qe = a.value) == null ? void 0 : qe.areaButton.active)}
        }`), (Ze = (ze = a.value) == null ? void 0 : ze.areaSwitch) != null && Ze.default && (o += `.${t}.t-area-selector .area-box .area-container .area-button .el-radio-button__inner{
          ${r((Fe = a.value) == null ? void 0 : Fe.areaSwitch.default)}
        }`), (Ke = (Je = a.value) == null ? void 0 : Je.areaSwitch) != null && Ke.hover && (o += `.${t}.t-area-selector .area-box .area-container .area-button .el-radio-button__inner:hover {
          ${r((Qe = a.value) == null ? void 0 : Qe.areaSwitch.hover)}
        }`), (Ve = (Ue = a.value) == null ? void 0 : Ue.areaSwitch) != null && Ve.active && (o += `.${t}  .area-box .area-container .area-button .el-radio-button__original-radio:checked + .el-radio-button__inner {
          ${r((We = a.value) == null ? void 0 : We.areaSwitch.active)}
        }`), (ke = (Xe = a.value) == null ? void 0 : Xe.handleButton) != null && ke.default && (o += `.${t}.t-area-selector .area-box .area-container .area-button .handle-button .el-button{
          ${r((eo = a.value) == null ? void 0 : eo.handleButton.default)}
        }`), (ao = (oo = a.value) == null ? void 0 : oo.handleButton) != null && ao.hover && (o += `.${t}.t-area-selector .area-box .area-container .area-button .handle-button .el-button:hover {
          ${r((to = a.value) == null ? void 0 : to.handleButton.hover)}
        }`), (lo = (ro = a.value) == null ? void 0 : ro.handleButton) != null && lo.focus && (o += `.${t}.t-area-selector .area-box .area-container .area-button .handle-button .el-button:focus {
          ${r((no = a.value) == null ? void 0 : no.handleButton.focus)}
        }`), (co = (io = a.value) == null ? void 0 : io.handleButton) != null && co.active && (o += `.${t}.t-area-selector .area-box .area-container .area-button .handle-button .el-button:active{
          ${r((uo = a.value) == null ? void 0 : uo.handleButton.active)}
        }`), (vo = (so = a.value) == null ? void 0 : so.Active) != null && vo.default && (o += `.${t}.t-area-selector .area-box .area-container .area-litter .litter-active{
          ${r((go = a.value) == null ? void 0 : go.Active.default)}
        }`, o += `.${t}.t-area-selector .area-box .area-container .area-city-list .row .tag-active{
          ${r((fo = a.value) == null ? void 0 : fo.Active.default)}
        }`, o += `.${t}.t-area-selector .area-box .area-container .area-city-list .row .citys .city-container .city-active {
          ${r((ho = a.value) == null ? void 0 : ho.Active.default)}
        }`, o += `.${t}.t-area-selector .area-box .area-container .area-city-list .row .citys .city-container .countys .county-active {
          ${r((Ro = a.value) == null ? void 0 : Ro.Active.default)}
        }`), (To = (bo = a.value) == null ? void 0 : bo.Active) != null && To.hover && (o += `.${t}.t-area-selector .area-box .area-container .area-litter .litter-active:hover{
          ${r((Co = a.value) == null ? void 0 : Co.Active.default)}
        }`, o += `.${t}.t-area-selector .area-box .area-container .area-city-list .row .tag-active:hover{
          ${r(($o = a.value) == null ? void 0 : $o.Active.default)}
        }`, o += `.${t}.t-area-selector .area-box .area-container .area-city-list .row .citys .city-container .city-active:hover {
          ${r((Eo = a.value) == null ? void 0 : Eo.Active.default)}
        }`, o += `.${t}.t-area-selector .area-box .area-container .area-city-list .row .citys .city-container .countys .county-active:hover {
          ${r((Oo = a.value) == null ? void 0 : Oo.Active.default)}
        }`), o;
    }, b = () => {
      const o = document.querySelector(`.${t} .style-container`);
      o && (o.innerHTML = `<style>${yo()}${Io()}</style>`);
    }, po = () => {
      e.isShowBox = !e.isShowBox, Ao(() => {
        var l;
        const o = document.getElementById(`${e.breadcrumbObj.code}`);
        o && ((l = e.list) == null || l.scroll(0, (o == null ? void 0 : o.offsetTop) - e.list.offsetTop));
      });
    }, _o = (o) => {
      var i;
      e.breadcrumbObj.code = o;
      const l = document.getElementById(`${o}`);
      l && ((i = e.list) == null || i.scroll(0, (l == null ? void 0 : l.offsetTop) - e.list.offsetTop));
    }, g = (o) => {
      let l;
      return o.substr(4, 2) !== "00" ? l = "country" : o.substr(2, 2) !== "00" ? l = "city" : l = "province", l;
    }, d = (o) => {
      if (e.breadcrumbObj.path = [], e.regionTool.currentRegionCode = o, e.regionTool.currentRegionName = e.regionTool.allRegionMap[o].REGION_NAME, e.regionTool.currentRegionType = g(o), !e.regionTool.currentRegionName)
        return;
      e.regionTool.currentProvinceCode = `${e.regionTool.currentRegionCode.substr(0, 2)}0000000000`, o.substr(2, 2) !== "00" && e.regionTool.allRegionMap[e.regionTool.currentProvinceCode] && (e.regionTool.currentProvinceName = e.regionTool.allRegionMap[e.regionTool.currentProvinceCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentProvinceCode, name: e.regionTool.currentProvinceName })), e.regionTool.currentCityCode = `${e.regionTool.currentRegionCode.substr(0, 4)}00000000`, o.substr(4, 2) !== "00" && e.regionTool.allRegionMap[e.regionTool.currentCityCode] && (e.regionTool.currentCityName = e.regionTool.allRegionMap[e.regionTool.currentCityCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentCityCode, name: e.regionTool.currentCityName })), e.regionTool.currentCountyCode = `${e.regionTool.currentRegionCode.substr(0, 6)}000000`, o.substr(6, 3) !== "000" && e.regionTool.allRegionMap[e.regionTool.currentCountyCode] && (e.regionTool.currentCountyName = e.regionTool.allRegionMap[e.regionTool.currentCountyCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentCountyCode, name: e.regionTool.currentCountyName })), e.regionTool.currentTownCode = `${e.regionTool.currentRegionCode.substr(0, 9)}000`, o.substr(9, 3) !== "000" && e.regionTool.allRegionMap[e.regionTool.currentTownCode] && (e.regionTool.currentTownName = e.regionTool.allRegionMap[e.regionTool.currentTownCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentTownCode, name: e.regionTool.currentTownName }));
      const l = e.regionTool.allRegionMap[e.regionTool.currentRegionCode].REGION_NAME;
      l.length > 8 && `${l.substr(0, 8)}`, e.breadcrumbObj.path.push({ code: e.regionTool.currentRegionCode, name: e.regionTool.currentRegionName });
    }, T = () => {
      if (e.selected === "REGION_CODE") {
        d(e.regionTool.tempRegionCode);
        const o = {
          selected: "REGION_CODE",
          REGION_CODE: e.regionTool.currentRegionCode,
          REGION_NAME: e.regionTool.currentRegionName,
          REGION_TYPE: g(e.regionTool.currentRegionCode)
        };
        h("search", o);
      }
      e.isShowBox = !e.isShowBox;
    }, C = (o) => {
      if (e.regionTool.tempRegionCode = o, !e.regionTool.allRegionMap[o] || o === e.thsDefaultRegionCode)
        return;
      const l = e.regionTool.allRegionMap[o].PARENT_REGION_CODE;
      e.breadcrumbObj.code = l === e.thsDefaultRegionCode ? e.regionTool.allRegionMap[o].REGION_PY.substring(0, 1) : e.regionTool.allRegionMap[l].REGION_PY.substring(0, 1), c.autoRequest && T();
    }, R = () => {
      Lo({
        url: e.url,
        method: "post",
        data: Po.stringify({ REGION_PARENT_CODE: c.thsDefaultRegionCode })
      }).then((o) => {
        var v;
        e.regionTool.currentRegionCode = o.data.data[0].REGION_CODE, e.regionTool.currentRegionName = o.data.data[0].REGION_NAME, e.regionTool.regionLevelShowList = [], e.regionTool.allRegionMap = [];
        const l = o.data.data;
        for (let n = 0; n < l.length; n++)
          l[n] && l[n].REGION_PY && (l[n].CHILD = []), e.regionTool.allRegionMap[l[n].REGION_CODE] = l[n];
        for (let n = 0; n < l.length; n++) {
          const u = l[n].PARENT_REGION_CODE, s = l[n].REGION_PY.substring(0, 1);
          l[n] && u && (u === e.thsDefaultRegionCode ? (e.regionTool.allRegionMap[s] || (e.regionTool.allRegionMap[s] = {}, e.regionTool.allRegionMap[s].CHILD = [], e.regionTool.allRegionMap[s].REGION_CODE = s, e.regionTool.allRegionMap[s].REGION_NAME = s), e.regionTool.allRegionMap[s].CHILD.push(l[n])) : e.regionTool.allRegionMap[u] && e.regionTool.allRegionMap[u].CHILD.push(l[n]));
        }
        if (c.showInitialSort) {
          const n = [
            "A",
            "B",
            "C",
            "D",
            "E",
            "F",
            "G",
            "H",
            "I",
            "J",
            "K",
            "L",
            "M",
            "N",
            "O",
            "P",
            "Q",
            "R",
            "S",
            "T",
            "U",
            "V",
            "W",
            "X",
            "Y",
            "Z",
            "港澳"
          ];
          for (let u = 0; u < n.length; u++) {
            const s = n[u];
            if (e.regionTool.allRegionMap[s] && e.regionTool.allRegionMap[s].CHILD != null && e.regionTool.allRegionMap[s].CHILD.length > 0) {
              const f = e.regionTool.allRegionMap[s];
              e.regionTool.regionLevelShowList.push(f);
            }
          }
        } else {
          const n = [];
          for (let u = 0; u < l.length; u++)
            l[u].PARENT_REGION_CODE === e.thsDefaultRegionCode && n.push(l[u]);
          e.regionTool.regionLevelShowList.push({
            REGION_CODE: "",
            REGION_NAME: "",
            CHILD: n
          });
        }
        e.REGION_CODE && C(e.REGION_CODE), d(e.REGION_CODE ? e.REGION_CODE : e.thsDefaultRegionCode);
        const i = document.getElementById(`${e.breadcrumbObj.code}`);
        i && ((v = e.list) == null || v.scroll(0, (i == null ? void 0 : i.offsetTop) - e.list.offsetTop));
      });
    }, xo = () => {
      e.isShowBox = !e.isShowBox;
    }, So = (o, l) => {
      var v;
      if (e.regionTool.tempRegionCode = o.code, e.selected === "REGION_CODE") {
        d(e.regionTool.tempRegionCode);
        const n = {
          selected: "REGION_CODE",
          REGION_CODE: e.regionTool.currentRegionCode,
          REGION_NAME: e.regionTool.currentRegionName,
          REGION_TYPE: g(e.regionTool.currentRegionCode)
        };
        h("select", n);
      }
      const i = document.getElementById(`${e.breadcrumbObj.code}`);
      i && ((v = e.list) == null || v.scroll(0, (i == null ? void 0 : i.offsetTop) - e.list.offsetTop));
    }, Do = () => {
      var o;
      if (e.selected === "REGION_CODE") {
        d(e.REGION_CODE);
        const l = {
          selected: "REGION_CODE",
          REGION_CODE: e.regionTool.currentRegionCode,
          REGION_NAME: e.regionTool.currentRegionName,
          REGION_TYPE: g(e.regionTool.currentRegionCode)
        };
        h("resetRegion", l);
        const i = document.getElementById(`${e.breadcrumbObj.code}`);
        i && ((o = e.list) == null || o.scroll(0, (i == null ? void 0 : i.offsetTop) - e.list.offsetTop));
      }
      e.isShowBox = !e.isShowBox, R();
    };
    return wo(() => {
      b(), R();
    }), No(
      () => c.cStyle,
      () => {
        b();
      },
      { deep: !0 }
    ), No(() => [c.url, c.thsDefaultRegionCode, c.regionCode, c.selected], () => {
      e.thsDefaultRegionCode = c.thsDefaultRegionCode, e.REGION_CODE = c.regionCode, e.selected = c.selected, e.url = c.url, R();
    }, { deep: !0 }), {
      curStyle: a,
      domOnlyClass: t,
      checkRegion: C,
      select: So,
      search: T,
      resetRegion: Do,
      switchSelector: po,
      close: xo,
      selectLitter: _o,
      ...Bo(e),
      showInitialSort: mo(() => c.showInitialSort)
    };
  }
});
export {
  Fo as default
};
