// @ths/design/es/components/area-selector/src/index.vue.js
import R from "./index.vue2.js";
import { resolveComponent as c, openBlock as s, createElementBlock as n, normalizeClass as u, createElementVNode as l, createVNode as a, withCtx as r, createTextVNode as _, Fragment as C, renderList as m, createBlock as N, toDisplayString as p, createCommentVNode as v } from "vue";
import "./index.vue3.js";
import h from "../../../_virtual/_plugin-vue_export-helper.js";
const D = { class: "area-selector" }, I = { class: "area-breadcrumb" }, G = {
  key: 0,
  class: "area-box"
}, y = { class: "area-title" }, w = { class: "area-container" }, A = { class: "area-button" }, L = {
  key: 0,
  class: "handle-button"
}, S = {
  key: 0,
  class: "area-litter"
}, $ = ["onClick"], T = {
  ref: "list",
  class: "area-city-list"
}, V = ["id"], B = { class: "citys" }, j = ["onClick"], F = { class: "city_text" }, M = { class: "countys" }, z = ["onClick"], H = { class: "county_text" };
function q(e, o, U, J, K, P) {
  const E = c("el-button"), k = c("el-breadcrumb-item"), O = c("el-breadcrumb"), f = c("el-radio-button"), g = c("el-radio-group"), b = c("el-tooltip");
  return s(), n("div", {
    class: u(["t-area-selector", e.domOnlyClass])
  }, [
    l("div", D, [
      a(E, {
        class: "selector",
        onClick: e.switchSelector
      }, {
        default: r(() => o[2] || (o[2] = [
          _(" 区域选择 ")
        ])),
        _: 1,
        __: [2]
      }, 8, ["onClick"]),
      l("div", I, [
        a(O, { separator: ">" }, {
          default: r(() => [
            (s(!0), n(C, null, m(e.breadcrumbObj.path, (t, i) => (s(), N(k, {
              key: t,
              class: u(i + 1 === e.breadcrumbObj.path.length ? "last-region" : ""),
              onClick: (d) => e.select(t, i)
            }, {
              default: r(() => [
                _(p(t.name), 1)
              ]),
              _: 2
            }, 1032, ["class", "onClick"]))), 128))
          ]),
          _: 1
        })
      ])
    ]),
    e.isShowBox ? (s(), n("div", G, [
      l("div", y, [
        o[3] || (o[3] = l("div", { class: "title-icon" }, null, -1)),
        o[4] || (o[4] = l("div", { class: "title-text" }, " 区域查询 ", -1)),
        l("div", {
          class: "title-close",
          onClick: o[0] || (o[0] = (...t) => e.close && e.close(...t))
        }, " x ")
      ]),
      l("div", w, [
        l("div", A, [
          a(g, {
            modelValue: e.radio1,
            "onUpdate:modelValue": o[1] || (o[1] = (t) => e.radio1 = t)
          }, {
            default: r(() => [
              a(f, { label: "xz" }, {
                default: r(() => o[5] || (o[5] = [
                  _(" 行政区划 ")
                ])),
                _: 1,
                __: [5]
              })
            ]),
            _: 1
          }, 8, ["modelValue"]),
          e.autoRequest ? v("", !0) : (s(), n("div", L, [
            a(E, {
              label: "cx",
              class: "search",
              onClick: e.search
            }, {
              default: r(() => o[6] || (o[6] = [
                _(" 查询 ")
              ])),
              _: 1,
              __: [6]
            }, 8, ["onClick"]),
            a(E, {
              label: "cz",
              class: "reset",
              onClick: e.resetRegion
            }, {
              default: r(() => o[7] || (o[7] = [
                _(" 重置 ")
              ])),
              _: 1,
              __: [7]
            }, 8, ["onClick"])
          ]))
        ]),
        e.showInitialSort ? (s(), n("div", S, [
          (s(!0), n(C, null, m(e.regionTool.regionLevelShowList, (t) => (s(), n("a", {
            key: t,
            href: "javascript:void(0)",
            class: u(["litter", e.breadcrumbObj.code === t.REGION_CODE && e.regionTool.tempRegionCode != e.thsDefaultRegionCode ? "litter-active" : ""]),
            onClick: (i) => e.selectLitter(t.REGION_CODE)
          }, p(t.REGION_CODE), 11, $))), 128))
        ])) : v("", !0),
        l("div", T, [
          (s(!0), n(C, null, m(e.regionTool.regionLevelShowList, (t) => (s(), n("div", {
            id: t.REGION_CODE,
            key: t,
            class: "row"
          }, [
            e.showInitialSort ? (s(), n("div", {
              key: 0,
              class: u(["tag", e.breadcrumbObj.code === t.REGION_CODE && e.regionTool.tempRegionCode != e.thsDefaultRegionCode ? "tag-active" : ""])
            }, p(t.REGION_CODE), 3)) : v("", !0),
            l("div", B, [
              (s(!0), n(C, null, m(t.CHILD, (i) => (s(), n("div", {
                key: i,
                class: "city-container"
              }, [
                l("div", {
                  class: u(["city", e.regionTool.tempRegionCode === i.REGION_CODE ? "city-active" : ""]),
                  onClick: (d) => e.checkRegion(i.REGION_CODE)
                }, [
                  a(b, {
                    class: "box-item",
                    effect: "dark",
                    content: i.REGION_NAME,
                    placement: "top"
                  }, {
                    default: r(() => [
                      l("div", F, p(i.REGION_NAME), 1)
                    ]),
                    _: 2
                  }, 1032, ["content"])
                ], 10, j),
                l("div", M, [
                  (s(!0), n(C, null, m(i.CHILD, (d) => (s(), n("div", {
                    key: d,
                    class: u(["county", e.regionTool.tempRegionCode === d.REGION_CODE ? "county-active" : ""]),
                    onClick: (Q) => e.checkRegion(d.REGION_CODE)
                  }, [
                    a(b, {
                      class: "box-item",
                      effect: "dark",
                      content: d.REGION_NAME,
                      placement: "top"
                    }, {
                      default: r(() => [
                        l("div", H, p(d.REGION_NAME), 1)
                      ]),
                      _: 2
                    }, 1032, ["content"])
                  ], 10, z))), 128))
                ])
              ]))), 128))
            ])
          ], 8, V))), 128))
        ], 512)
      ])
    ])) : v("", !0),
    o[8] || (o[8] = l("div", { class: "style-container" }, null, -1))
  ], 2);
}
const x = /* @__PURE__ */ h(R, [["render", q]]);
export {
  x as default
};
