// @ths/design/es/components/regulatory-target-selector/src/point-select/index.vue2.js
import { defineComponent as a, ref as c, reactive as l, onMounted as m, onActivated as d, toRefs as g } from "vue";
import { ElMessage as p } from "element-plus";
import f from "element-plus/lib/locale/lang/zh-cn";
import "@ths/c-common-base-button/es/index.css";
import "@ths/c-common-base-paging/es/index.css";
import h from "./point-select.vue.js";
const R = a({
  name: "TRegulatoryTargetSelector",
  components: {
    PointSelect: h
  },
  emits: ["on-close", "on-sure"],
  setup(u, t) {
    const n = c(), o = l({
      // 控制弹框显示
      dialogVisible: !0,
      // 弹框高度
      dialogContentHeight: 300,
      dataIndex: 0
    }), s = () => {
      t.emit("on-close");
    }, r = () => {
      const e = n.value.getSelectDataList();
      e.length > 0 ? (t.emit("on-close"), t.emit("on-sure", e)) : p.error("请选择一条数据");
    }, i = () => {
      const e = window.innerHeight;
      o.dialogContentHeight = e * 0.8 - 120;
    };
    return m(async () => {
      i();
    }), d(() => {
      o.dataIndex++, i();
    }), {
      ...g(o),
      locale: {
        ...f
      },
      closePop: s,
      chooseEvt: r,
      pointSelectRef: n
    };
  }
});
export {
  R as default
};
