// @ths/design/es/components/regulatory-target-selector/src/point-select/index.vue.js
import m from "./index.vue2.js";
import { resolveComponent as l, openBlock as s, createBlock as a, withCtx as t, createElementVNode as n, createVNode as i, createTextVNode as p, normalizeStyle as f, mergeProps as _ } from "vue";
import "./index.vue3.js";
import g from "../../../../_virtual/_plugin-vue_export-helper.js";
const C = { class: "btn" };
function v(e, o, k, y, b, B) {
  const d = l("point-select"), c = l("el-config-provider"), r = l("el-button"), u = l("el-dialog");
  return s(), a(u, {
    "model-value": e.dialogVisible,
    width: "85%",
    top: "0",
    "append-to-body": !0,
    "custom-class": "point-select-dialog",
    onClose: e.closePop
  }, {
    title: t(() => o[0] || (o[0] = [
      n("header", null, [
        n("div", { class: "title" }, " 选择监管对象 ")
      ], -1)
    ])),
    footer: t(() => [
      n("div", C, [
        i(r, { onClick: e.closePop }, {
          default: t(() => o[1] || (o[1] = [
            p(" 取消 ")
          ])),
          _: 1,
          __: [1]
        }, 8, ["onClick"]),
        i(r, {
          type: "primary",
          onClick: e.chooseEvt
        }, {
          default: t(() => o[2] || (o[2] = [
            p(" 确认 ")
          ])),
          _: 1,
          __: [2]
        }, 8, ["onClick"])
      ])
    ]),
    default: t(() => [
      n("div", {
        class: "point-select-dialog-content",
        style: f({ height: e.dialogContentHeight + "px" })
      }, [
        i(c, { locale: e.locale }, {
          default: t(() => [
            (s(), a(d, _(e.$attrs, {
              ref: "pointSelectRef",
              key: e.dataIndex
            }), null, 16))
          ]),
          _: 1
        }, 8, ["locale"])
      ], 4)
    ]),
    _: 1
  }, 8, ["model-value", "onClose"]);
}
const D = /* @__PURE__ */ g(m, [["render", v]]);
export {
  D as default
};
