// @ths/design/es/components/regulatory-target-selector/src/point-select/point-table.vue.js
import p from "./point-table.vue2.js";
import { resolveComponent as d, openBlock as a, createBlock as o, mergeProps as h, withCtx as i, createCommentVNode as t, createElementVNode as g, toDisplayString as b, createElementBlock as w, Fragment as S, renderList as y, createSlots as c, renderSlot as u, createTextVNode as C } from "vue";
import "./point-table.vue3.js";
import k from "../../../../_virtual/_plugin-vue_export-helper.js";
function z(e, s, A, $, N, V) {
  const m = d("el-radio"), r = d("el-table-column"), f = d("el-table");
  return a(), o(f, h({
    ref: "tableRef",
    data: e.tableData,
    size: e.tableSize,
    "tooltip-effect": "dark",
    sortable: "",
    border: "",
    stripe: e.openStripe,
    class: "table-component"
  }, e.$attrs, {
    onSelectAll: e.handleSelectionAllChange,
    onSelect: e.handleSelectionChange
  }), {
    default: i(() => [
      e.openSingleSelectColumn && !e.openMultipleSelectColumn ? (a(), o(r, {
        key: 0,
        label: "",
        width: "50",
        align: "center",
        "class-name": "radio-select-column"
      }, {
        default: i((l) => [
          (a(), o(m, {
            key: l.row[e.tableProps.idKey || e.defaultTableProps.idKey],
            modelValue: e.tableRadioValue,
            "onUpdate:modelValue": s[0] || (s[0] = (n) => e.tableRadioValue = n),
            label: l.row[e.tableProps.idKey || e.defaultTableProps.idKey],
            value: l.row[e.tableProps.idKey || e.defaultTableProps.idKey],
            onChange: (n) => e.handleRadioChange(l)
          }, null, 8, ["modelValue", "label", "value", "onChange"]))
        ]),
        _: 1
      })) : t("", !0),
      e.openMultipleSelectColumn && !e.openSingleSelectColumn ? (a(), o(r, {
        key: 1,
        type: "selection",
        "header-align": "center",
        align: "center",
        fixed: "left",
        selectable: e.selectable,
        resizable: !1
      }, null, 8, ["selectable"])) : t("", !0),
      e.openIndexColumn && e.pageSize && e.pageNum ? (a(), o(r, {
        key: 2,
        type: "index",
        align: "center",
        fixed: "left",
        label: "序号",
        width: e.orderNoWidth,
        "min-width": e.orderNoWidth,
        resizable: !1
      }, {
        default: i((l) => [
          g("span", null, b((e.pageNum - 1) * e.pageSize + l.$index + 1), 1)
        ]),
        _: 1
      }, 8, ["width", "min-width"])) : t("", !0),
      (a(!0), w(S, null, y(e.tableNormalOptions, (l) => (a(), o(r, {
        key: l.prop,
        prop: l.prop,
        label: l.label,
        width: l.width ? l.width : "",
        "min-width": l.minWidth ? l.minWidth : "",
        align: l.align ? l.align : "center",
        "header-align": l.headerAlign ? l.headerAlign : "center",
        fixed: l.fixed ? l.fixed : !1,
        sortable: l.sortable ? l.sortable : !1,
        "show-overflow-tooltip": !l.slot,
        resizable: l.resizable !== null ? l.resizable : !0
      }, c({ _: 2 }, [
        l.slot ? {
          name: "default",
          fn: i((n) => [
            u(e.$slots, l.slot, { scope: n })
          ]),
          key: "0"
        } : {
          name: "default",
          fn: i((n) => [
            C(b(`${e.isNoVal(n.row[l.prop])}`), 1)
          ]),
          key: "1"
        }
      ]), 1032, ["prop", "label", "width", "min-width", "align", "header-align", "fixed", "sortable", "show-overflow-tooltip", "resizable"]))), 128)),
      e.tableActionOptions ? (a(), o(r, {
        key: 3,
        width: e.tableActionOptions.width ? e.tableActionOptions.width : "",
        "min-width": e.tableActionOptions.minWidth ? e.tableActionOptions.minWidth : "",
        "header-align": "center",
        align: "center",
        fixed: e.tableActionOptions.fixed ? e.tableActionOptions.fixed : "right",
        label: "操作",
        resizable: !1
      }, {
        default: i((l) => [
          u(e.$slots, "actionSolt", { scope: l })
        ]),
        _: 3
      }, 8, ["width", "min-width", "fixed"])) : t("", !0)
    ]),
    _: 3
  }, 16, ["data", "size", "stripe", "onSelectAll", "onSelect"]);
}
const T = /* @__PURE__ */ k(p, [["render", z]]);
export {
  T as default
};
