// @ths/design/es/components/regulatory-target-selector/src/point-select/point-table.vue2.js
import { defineComponent as h, ref as g, computed as u, reactive as K, watch as s, toRefs as x, nextTick as R } from "vue";
const i = {
  idKey: "id"
}, N = h({
  name: "TRegulatoryTargetTable",
  components: {},
  props: {
    // 表格数据
    tableData: {
      type: Array,
      required: !0,
      default: () => []
    },
    // 表头配置项
    tableHeaderOptions: {
      type: Array,
      required: !0,
      default: () => []
    },
    // 判断是否禁用多选方法（返回true不禁用）
    selectable: {
      type: Function,
      default: (e, d) => !0
    },
    // 页码
    pageNum: {
      type: Number,
      default: 1
    },
    // 每页条数
    pageSize: {
      type: Number,
      default: 10
    },
    // 是否开启序号列(开启的话,必须传入每页条数/页码)
    openIndexColumn: {
      type: Boolean,
      default: !1
    },
    // 是否开启多选选择列
    openMultipleSelectColumn: {
      type: Boolean,
      default: !1
    },
    // 是否开启单选选择列
    openSingleSelectColumn: {
      type: Boolean,
      default: !1
    },
    // 操作列是否显示
    showOperationColumn: {
      type: Boolean,
      default: !0
    },
    // 是否开启斑马纹
    openStripe: {
      type: Boolean,
      default: !0
    },
    // 序号列宽度
    orderNoWidth: {
      type: Number,
      default: 55
    },
    // 选中的tab项
    activeTabKey: {
      type: String,
      default: ""
    },
    // 选中的数据
    selectedData: {
      type: Array,
      default: () => []
    },
    // 人员列表数据-数据别名
    tableProps: {
      type: Object,
      default: () => i
    }
  },
  emits: ["table-radio-change", "table-multiple-change"],
  setup(e, { emit: d }) {
    const c = g(), r = u(() => e.tableHeaderOptions.filter((t) => !t.action)), f = u(() => e.tableHeaderOptions.find((t) => t.action)), l = K({
      // 表格尺寸
      tableSize: "mini",
      // 单选
      tableRadioValue: "",
      // 单选选中数据
      singleSelectData: void 0,
      // 多选选择的列表项数组
      multipleSelection: e.selectedData
    }), y = (t) => {
      let a = t;
      return !t && t !== 0 && (a = "-"), a;
    }, b = (t) => {
      l.singleSelectData = JSON.parse(JSON.stringify(t.row)), d("table-radio-change", [t.row]);
    }, m = () => {
      c.value.clearSelection(), l.multipleSelection.forEach((t, a) => {
        const n = e.tableData.find(
          (o) => o[e.tableProps.idKey || i.idKey] === t[e.tableProps.idKey || i.idKey]
        );
        n && (l.multipleSelection[a] = JSON.parse(JSON.stringify(n)), R(() => c.value.toggleRowSelection(n, !0)));
      });
    }, p = (t) => {
      t.length ? t.forEach((a) => {
        l.multipleSelection.findIndex(
          (o) => o[e.tableProps.idKey || i.idKey] === a[e.tableProps.idKey || i.idKey]
        ) === -1 && l.multipleSelection.push(a);
      }) : (e.tableData || []).forEach((a) => {
        const n = l.multipleSelection.findIndex(
          (o) => o[e.tableProps.idKey || i.idKey] === a[e.tableProps.idKey || i.idKey]
        );
        l.multipleSelection.splice(n, 1);
      }), d("table-multiple-change", l.multipleSelection);
    }, S = (t, a) => {
      const n = l.multipleSelection.findIndex(
        (o) => o[e.tableProps.idKey || i.idKey] === a[e.tableProps.idKey || i.idKey]
      );
      n !== -1 ? l.multipleSelection.splice(n, 1) : l.multipleSelection.push(a), d("table-multiple-change", l.multipleSelection);
    };
    return s(() => e.activeTabKey, (t, a) => {
      t !== a && (l.tableRadioValue = e.selectedData.length && e.selectedData[0][e.tableProps.idKey || i.idKey] || "");
    }), s([
      () => e.tableData,
      () => e.openMultipleSelectColumn
    ], () => {
      e.openMultipleSelectColumn && m();
    }), {
      ...x(l),
      tableNormalOptions: r,
      tableActionOptions: f,
      isNoVal: y,
      handleRadioChange: b,
      tableRef: c,
      handleSelectionAllChange: p,
      handleSelectionChange: S,
      defaultTableProps: i
    };
  }
});
export {
  N as default
};
