// @ths/design/es/components/regulatory-target-selector/src/point-select/point-select.vue.js
import m from "./point-select.js";
import { resolveComponent as o, openBlock as l, createElementBlock as r, createVNode as t, withCtx as s, Fragment as h, renderList as C, createBlock as i, createCommentVNode as f } from "vue";
import "./point-select.vue2.js";
import S from "../../../../_virtual/_plugin-vue_export-helper.js";
const T = { class: "point-select-container" };
function y(e, p, v, k, _, N) {
  const c = o("t-search"), d = o("point-table"), g = o("t-paging"), u = o("el-tab-pane"), b = o("el-tabs");
  return l(), r("div", T, [
    t(b, {
      modelValue: e.activeTabKey,
      "onUpdate:modelValue": p[0] || (p[0] = (a) => e.activeTabKey = a),
      class: "demo-tabs",
      onTabClick: e.handleTabChange
    }, {
      default: s(() => [
        (l(!0), r(h, null, C(e.pointSelectOptions, (a) => (l(), i(u, {
          key: a.entityId,
          label: a.entityName,
          name: a.entityId
        }, {
          default: s(() => [
            e.showSearch ? (l(), i(c, {
              key: 0,
              option: a,
              "is-ignore-user-region": !0,
              "region-tree-url": e.regionTreeUrl,
              "user-region-url": e.userRegionUrl,
              "top-region-code": e.userRegionCode,
              onHandleSearch: (n) => e.searchPage(a)
            }, null, 8, ["option", "region-tree-url", "user-region-url", "top-region-code", "onHandleSearch"])) : f("", !0),
            t(d, {
              ref_for: !0,
              ref: (n) => e.setChooseRef(n, a.entityId),
              "table-data": a.page.list,
              "table-header-options": a.tableHeaderOptions,
              "open-index-column": !1,
              "open-select-column": !0,
              "page-size": a.page.pageSize,
              "page-num": a.page.pageNum,
              "is-disable-selection-all": !1,
              "open-single-select-column": e.openSingleSelectColumn,
              "open-multiple-select-column": e.openMultipleSelectColumn,
              "active-tab-key": e.activeTabKey,
              "selected-data": e.selectedData,
              "table-props": e.tableProps,
              onTableRadioChange: e.onSelectChange,
              onTableMultipleChange: e.onSelectChange
            }, null, 8, ["table-data", "table-header-options", "page-size", "page-num", "open-single-select-column", "open-multiple-select-column", "active-tab-key", "selected-data", "table-props", "onTableRadioChange", "onTableMultipleChange"]),
            t(g, {
              page: a.page,
              onHandleSearch: (n) => e.searchPage(a),
              "onUpdate:pageNum": (n) => a.page.pageNum = n,
              "onUpdate:pageSize": (n) => a.page.pageSize = n
            }, null, 8, ["page", "onHandleSearch", "onUpdate:pageNum", "onUpdate:pageSize"])
          ]),
          _: 2
        }, 1032, ["label", "name"]))), 128))
      ]),
      _: 1
    }, 8, ["modelValue", "onTabClick"])
  ]);
}
const R = /* @__PURE__ */ S(m, [["render", y], ["__scopeId", "data-v-78cadc1b"]]);
export {
  R as default
};
