// @ths/design/es/components/regulatory-target-selector/src/point-select/point-select.js
import { defineComponent as O, ref as f, reactive as R, onMounted as D, onActivated as P, toRefs as w } from "vue";
import U from "@ths/c-common-base-table";
import v from "@ths/c-common-base-paging";
import { ElTabs as I, ElTabPane as K } from "element-plus";
import { getDataListPage as B, initPointSelectOptions as E, getUserRegion as N, getTagSearchs as q } from "../services/point-select-api.js";
import J from "./search.vue.js";
import k from "./point-table.vue.js";
const _ = O({
  name: "PointSelect",
  components: {
    TTable: U,
    // 普通表格组件
    TPaging: v,
    // 分页组件
    TSearch: J,
    // 搜索组件
    ElTabs: I,
    // 搜索组件
    ElTabPane: K,
    // 搜索组件
    PointTable: k
  },
  props: {
    // 是否单选
    isSelectOne: {
      type: Boolean,
      default: !0
    },
    // 是否忽略用户对应区域
    isIgnoreUserRegion: {
      type: Boolean,
      default: !1
    },
    // 顶级行政区域
    topRegionCode: {
      type: String,
      default: ""
    },
    // tab切换数据及表格配置数据url
    tabTableConfigUrl: {
      type: String,
      default: "",
      required: !0
    },
    // 搜索条件配置url
    searchConfigUrl: {
      type: String,
      default: "",
      required: !0
    },
    // 表格数据url
    tableDataUrl: {
      type: String,
      default: "",
      required: !0
    },
    // 行政区树-数据url
    regionTreeUrl: {
      type: String,
      default: "",
      required: !0
    },
    // 获取当前用户行政区信息（通过请求头的userToken)isIgnoreUserRegion为false时必传
    userRegionUrl: {
      type: String,
      default: ""
    },
    // 是否开启多选选择列
    openMultipleSelectColumn: {
      type: Boolean,
      default: !1
    },
    // 是否开启单选选择列
    openSingleSelectColumn: {
      type: Boolean,
      default: !1
    },
    // 选中的数据
    selectedData: {
      type: Array,
      default: () => []
    },
    // 人员列表数据-数据别名
    tableProps: {
      type: Object,
      default: () => ({
        idKey: "id"
      })
    }
  },
  emits: ["on-close", "on-sure"],
  setup(n) {
    const d = f(), t = R({
      // 监管对象类型清单
      pointSelectOptions: [],
      // 监管对象类型清单缓存
      pointSelectOptionCache: {},
      // 已选择的监管对象缓存(一级key对应点位类型，二级key对应点位编码)
      selectPointCache: {},
      // 外界已选择的监管对象缓存(一级key对应点位类型，二级key对应点位编码)
      oldSelectPointCache: {},
      // 当前展示的监管对象类型
      activeTabKey: "",
      // 用户默认行政区
      userRegionCode: "",
      // 是否展示搜索栏
      showSearch: !1,
      // 表格 顶部左侧列 按钮的显隐控制
      tableLeftBtnSetting: {
        densityBtn: !1,
        // 表格间距设置（默认为 true）
        fullScreenBtn: !1
        // 表格全屏（默认为 true）
      },
      // 表格选中数据
      selectedData: [],
      // 选中点位数据（暴露出组件的数据）
      selectedPointData: []
    }), p = (e) => {
    }, u = (e, a) => {
    }, r = f(null), y = async (e) => {
      if (e.tagSearchs.length > 0)
        return;
      const { topicCode: a } = e.config.tagSearch;
      if (a) {
        const o = await q(n.searchConfigUrl, a);
        e.tagSearchs = o;
      }
    }, c = async (e) => {
      r.value = null;
      const a = e.regionCode || t.userRegionCode, o = await B(n.tableDataUrl, {
        ENTITY_CODE: e.config.tagSearch.entityCode,
        sourcefields: e.config.tagSearch.sourcefields,
        queryField: e.config.tagSearch.queryField,
        pageNum: e.page.pageNum,
        pageSize: e.page.pageSize,
        tagJson: JSON.stringify(e.tagJson),
        keyword: e.keyWord,
        REGION_CODE: a,
        orderBy: e.config.tagSearch.orderBy
      });
      e.page = o.page, e.page.list = e.page.list.map((i) => ({
        ...i,
        activeTabKey: t.activeTabKey
      }));
    }, s = async (e) => {
      t.activeTabKey = e.props.name;
      const a = t.pointSelectOptionCache[t.activeTabKey];
      await y(a), await c(a);
    }, S = async () => {
      t.pointSelectOptions = await E(n.tabTableConfigUrl);
      for (const e of t.pointSelectOptions)
        t.pointSelectOptionCache[e.entityId] = e;
    }, h = (e) => {
      const a = [];
      return (e || []).forEach((o, i) => {
        const l = t.pointSelectOptionCache[o.activeTabKey].config.tagSearchKeyMapping, m = t.pointSelectOptionCache[o.activeTabKey].config.defaultValue;
        a.push({ ...m });
        for (const g of Object.keys(l))
          a.forEach(() => {
            a[i][g] = e[i][l[g]];
          });
      }), a;
    }, C = (e) => {
      t.selectedData = e, r.value = e, e && e.length && (t.selectedPointData = h(JSON.parse(JSON.stringify(e))));
    }, T = async () => {
      if (n.topRegionCode)
        t.userRegionCode = n.topRegionCode;
      else if (!n.isIgnoreUserRegion) {
        const e = await N(n.userRegionUrl);
        e && e.regionCode && (t.userRegionCode = e.regionCode);
      }
    }, b = () => t.selectedPointData.length ? t.selectedPointData : [];
    return D(async () => {
      await T(), t.showSearch = !0, await S(), await s({ props: { name: t.pointSelectOptions[0].entityId } });
    }), P(() => {
    }), {
      ...w(t),
      searchComponentRef: d,
      setChooseRef: u,
      searchPage: c,
      onSelectChange: C,
      handleTabChange: s,
      getSelectDataList: b,
      handleRadioChange: p
    };
  }
});
export {
  _ as default
};
