// @ths/design/es/components/regulatory-target-selector/src/point-select/search.vue2.js
import { defineComponent as l, ref as p, reactive as d, onMounted as m, toRefs as c } from "vue";
import { TSearchButton as f } from "@ths/c-common-base-button";
import { getUserRegion as u } from "../services/point-select-api.js";
import R from "../../../dialog-single-tree-input/src/index.vue.js";
const T = l({
  name: "Search",
  components: {
    TSearchButton: f,
    // 搜索条件的按钮
    TDialogSingleTreeInput: R
  },
  props: {
    // 下拉列表(如果搜索条件中有下拉列表, 则必传)
    option: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 是否忽略用户对应区域
    isIgnoreUserRegion: {
      type: Boolean,
      default: !1
    },
    // 顶级行政区域
    topRegionCode: {
      type: String,
      default: ""
    },
    // 行政区树url
    regionTreeUrl: {
      type: String,
      default: ""
    },
    // 获取当前用户行政区信息（通过请求头的userToken) isIgnoreUserRegion为false时必传
    userRegionUrl: {
      type: String,
      default: ""
    }
  },
  emits: ["handle-search"],
  setup(n, { emit: r }) {
    const a = p(), e = d({
      tempOption: n.option,
      // 更多查询条件是否可见（搜索条件是否展开）
      searchShowAll: !1,
      // 是否展示行政区选择条件
      showRegionSelect: !1,
      // 行政区单选树
      regionSingleTreeParams: {
        defaultExpandLayer: 1,
        url: n.regionTreeUrl,
        rootId: "510000000000",
        simpleData: {
          idKey: "regionCode",
          // 数据中id对应的key
          nameKey: "regionName",
          // 要展示的名称
          pIdKey: "parentRegionCode",
          // 数据中父节点标识key，懒加载时候使用 PARENT
          spanKey: ""
          // 在节点后拼接内容 key
        },
        param: { parentRegionCode: "510000000000", isAll: "1" },
        placeholder: "请选择行政区"
      }
    }), t = () => {
      r("handle-search");
    }, i = async () => {
      e.tempOption.page.pageNum = 1, e.tempOption.page.pageSize = 10, e.tempOption.keyWord = "", e.tempOption.regionCode = "", e.tempOption.regionName = "";
      const o = e.tempOption.tagJson;
      for (const s of Object.keys(o))
        o[s] = "";
      t();
    }, g = (o) => {
      e.searchShowAll = o;
    };
    return m(async () => {
      if (n.topRegionCode)
        e.regionSingleTreeParams.rootId = n.topRegionCode, e.regionSingleTreeParams.param.parentRegionCode = n.topRegionCode;
      else if (!n.isIgnoreUserRegion) {
        const o = await u(n.userRegionUrl);
        o && o.regionCode && (e.regionSingleTreeParams.rootId = o.regionCode, e.regionSingleTreeParams.param.parentRegionCode = o.regionCode);
      }
      e.showRegionSelect = !0;
    }), {
      ...c(e),
      searchFormRef: a,
      handleSearch: t,
      handleReset: i,
      toggleFoldState: g
    };
  }
});
export {
  T as default
};
