// @ths/design/es/components/list-status/src/index.vue.js
import _ from "./index.vue2.js";
import { openBlock as l, createElementBlock as o, normalizeClass as c, Fragment as d, renderList as p, normalizeStyle as n, createElementVNode as s, createTextVNode as v, toDisplayString as a, createCommentVNode as m } from "vue";
import "./index.vue3.js";
import g from "../../../_virtual/_plugin-vue_export-helper.js";
const h = ["title"], f = ["onClick"], k = { class: "t-list-status__item-point-name" }, y = { class: "t-list-status__item-buttons" }, V = ["onClick"], C = ["onClick"], b = { class: "t-list-status__item-bottom" }, N = { class: "t-list-status__item-bottom--left" }, $ = { class: "t-list-status__item-pollutant-desc" }, A = { class: "t-list-status__item-pollutant" }, S = { class: "t-list-status__item-bottom--right" };
function B(t, u, D, O, w, E) {
  return l(), o("div", {
    class: c(["t-list-status", t.domOnlyClass])
  }, [
    (l(!0), o(d, null, p(t.data, (e, i) => (l(), o("div", {
      key: i,
      class: "t-list-status__item",
      style: n({ marginBottom: t.getRowGap(i) })
    }, [
      s("div", {
        class: "t-list-status__item-top",
        title: e[t.dataAlias.pointName]
      }, [
        t.locationImageUrl ? (l(), o("div", {
          key: 0,
          style: n({ backgroundImage: `url(${t.locationImageUrl})` }),
          class: "t-list-status__item-location-image",
          alt: "定位图标",
          onClick: (r) => t.handleLocation(e, i)
        }, [
          t.showOrderNumber ? (l(), o(d, { key: 0 }, [
            v(a(i + 1), 1)
          ], 64)) : m("", !0)
        ], 12, f)) : m("", !0),
        s("div", k, a(t.getValueOrDefaultValue(e[t.dataAlias.pointName], "-")), 1),
        s("div", y, [
          s("div", {
            class: "t-list-status__item-buttons-push cursor",
            onClick: (r) => t.itemPush(e)
          }, " 推送 ", 8, V),
          s("div", {
            class: "t-list-status__item-buttons-view cursor",
            onClick: (r) => t.itemView(e)
          }, " 查看 ", 8, C)
        ])
      ], 8, h),
      s("div", b, [
        s("div", N, [
          s("span", {
            class: "t-list-status__item-status",
            style: n(t.getStatusStyle(e))
          }, a(t.getStatusName(e)), 5),
          s("span", $, a(t.pollutantName ?? ""), 1),
          s("span", A, a(t.getValueOrDefaultValue(e[t.dataAlias.pollutant], "-")), 1)
        ]),
        s("div", S, a(t.getValueOrDefaultValue(e[t.dataAlias.time], "-")), 1)
      ])
    ], 4))), 128)),
    u[0] || (u[0] = s("div", { class: "style-container" }, null, -1))
  ], 2);
}
const U = /* @__PURE__ */ g(_, [["render", B]]);
export {
  U as default
};
