// @ths/design/es/components/list-status/src/index.vue2.js
import { defineComponent as A, computed as B, watch as W, onMounted as D, onBeforeUnmount as P, useCssVars as M } from "vue";
import { merge as V, cloneDeep as x } from "lodash";
import E from "./assets/images/location.png.js";
import { getValueOrDefaultValue as U, getDomRandomClass as F, setStyle as s } from "../../../utils/index.js";
const f = {
  /* 容器 */
  wrapper: {
    default: {
      width: "432px"
    }
  },
  /* 单项容器 */
  itemWrapper: {
    default: {
      width: "400px"
    },
    hover: {
      width: "432px"
    }
  },
  /* 上部分的容器 */
  topWrapper: {
    default: {}
  },
  /* 上部分的图片 */
  topImage: {
    default: {}
  },
  /* 上部分的文本 */
  topText: {
    default: {}
  },
  /* 上部分的按钮组 */
  topButtons: {
    default: {}
  },
  /* 上部分的按钮-推送 */
  topButtonsPush: {
    default: {}
  },
  /* 上部分的按钮-查看 */
  topButtonsView: {
    default: {}
  },
  /* 下部分的容器 */
  bottomWrapper: {
    default: {}
  },
  /* 下部分左边容器（包裹着状态和污染物文本的容器） */
  bottomLeftWrapper: {
    default: {}
  },
  /* 下部分的状态 */
  bottomStatus: {
    default: {}
  },
  /* 下部分的污染物 */
  bottomPollutant: {
    default: {}
  },
  /* 下部分的时间 */
  bottomTime: {
    default: {}
  },
  /* 污染物名称 */
  pollutantName: {
    default: ""
  }
}, b = {
  pointName: "pointName",
  // 点位名称
  statusType: "statusType",
  // 状态类型
  pollutant: "pollutant",
  // 污染物
  time: "time"
  // 时间
}, r = {
  data: () => [],
  // 列表数据
  dataAlias: b,
  // 数据别名
  // 状态类型
  statusType: [
    {
      color: "#fff",
      bgColor: "rgba(119, 119, 119, 0.3)",
      name: "-",
      status: "0"
    },
    {
      color: "#FFDD00",
      bgColor: "rgba(255, 221, 0, 0.3)",
      name: "数据突变",
      status: "1"
    },
    {
      color: "#FF2626",
      bgColor: "rgba(255, 38, 38, 0.3)",
      name: "告警",
      status: "2"
    }
  ],
  // 定位图标
  locationImageUrl: E,
  // 列表项之间的间距
  rowGap: "12px"
}, v = A({
  name: "TListStatus",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => f
    },
    // 列表数据
    data: {
      type: Array,
      default: r.data
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: r.dataAlias
    },
    // 状态类型
    statusType: {
      type: Array,
      default: r.statusType
    },
    // 定位图标url
    locationImageUrl: {
      type: String,
      default: r.locationImageUrl
    },
    // 列表项之间的间距
    rowGap: {
      type: String,
      default: r.rowGap
    },
    // 污染物名称
    pollutantName: {
      type: String,
      default: ""
    },
    // 列表是否展示序号
    showOrderNumber: {
      type: Boolean,
      default: !1
    }
  },
  emits: [
    "point-location",
    // 点位定位
    "load-more",
    // 加载更多
    "item-view",
    // 查看
    "item-push"
    // 推送
  ],
  setup(l, { emit: i }) {
    const e = B(() => V(
      x(f),
      l.cStyle
    )), a = F("list-status");
    let u = null;
    const y = (t) => t === l.data.length - 1 ? 0 : l.rowGap, m = () => {
      if (u) {
        const { scrollTop: t, clientHeight: o, scrollHeight: n } = u;
        t && Math.round(t) + Math.round(o) === Math.round(n) && i("load-more");
      }
    }, g = () => {
      u && u.addEventListener("scroll", m);
    }, S = () => {
      let t = "";
      return e.value.wrapper && (t += s(e.value.wrapper, `.${a}`)), e.value.itemWrapper && (t += s(e.value.itemWrapper, `.${a} .t-list-status__item`)), e.value.topWrapper && (t += s(e.value.topWrapper, `.${a} .t-list-status__item .t-list-status__item-top`)), e.value.topImage && (t += s(e.value.topImage, `.${a} .t-list-status__item .t-list-status__item-top .t-list-status__item-location-image`)), e.value.topText && (t += s(e.value.topText, `.${a} .t-list-status__item .t-list-status__item-top .t-list-status__item-point-name`)), e.value.topButtons && (t += s(e.value.topButtons, `.${a} .t-list-status__item .t-list-status__item-top .t-list-status__item-buttons`)), e.value.topButtonsPush && (t += s(
        e.value.topButtonsPush,
        `.${a} .t-list-status__item .t-list-status__item-top .t-list-status__item-buttons-push`
      )), e.value.topButtonsView && (t += s(
        e.value.topButtonsView,
        `.${a} .t-list-status__item .t-list-status__item-top .t-list-status__item-buttons-view`
      )), e.value.bottomWrapper && (t += s(e.value.bottomWrapper, `.${a} .t-list-status__item .t-list-status__item-bottom`)), e.value.bottomLeftWrapper && (t += s(e.value.bottomLeftWrapper, `.${a} .t-list-status__item .t-list-status__item-bottom--left`)), e.value.bottomStatus && (t += s(e.value.bottomStatus, `.${a} .t-list-status__item .t-list-status__item-status`)), e.value.bottomPollutant && (t += s(
        e.value.bottomPollutant,
        `.${a} .t-list-status__item .t-list-status__item-bottom .t-list-status__item-pollutant`
      )), e.value.bottomTime && (t += s(e.value.bottomTime, `.${a} .t-list-status__item .t-list-status__item-bottom--right`)), e.value.pollutantDesc && (t += s(e.value.pollutantDesc, `.${a} .t-list-status__item .t-list-status__item-bottom .t-list-status__item-pollutant-desc`)), t;
    }, p = (t) => {
      const o = t[l.dataAlias.statusType];
      return l.statusType.find((n) => n.status === o);
    }, T = (t) => {
      const o = p(t);
      if (o) {
        const {
          bgColor: n,
          color: L
        } = o;
        return {
          background: n,
          color: L
        };
      }
      return {
        display: "none"
      };
    }, h = (t) => {
      const o = p(t);
      return (o == null ? void 0 : o.name) || "-";
    }, _ = () => {
      const t = document.querySelector(`.${a} .style-container`);
      t && (t.innerHTML = `<style>${S()}</style>`);
    }, w = (t) => {
      i("item-push", t);
    }, $ = (t) => {
      i("item-view", t);
    }, C = (t, o) => {
      i("point-location", {
        item: t,
        index: o
      });
    };
    return W(
      () => [l.cStyle],
      () => {
        _();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), D(() => {
      u = document.getElementsByClassName(a)[0] ?? null, g(), _();
    }), P(() => {
      u && u.removeEventListener("scroll", m);
    }), {
      curStyle: e,
      domOnlyClass: a,
      /* 配置 */
      DATA_FIELD_MAP: b,
      /* 方法 */
      getStatusStyle: T,
      getStatusName: h,
      handleLocation: C,
      getRowGap: y,
      getValueOrDefaultValue: U,
      itemView: $,
      itemPush: w
    };
  }
}), c = () => {
  M((l) => ({
    "2ab8275e": l.rowgap
  }));
}, d = v.setup;
v.setup = d ? (l, i) => (c(), d(l, i)) : c;
export {
  v as default
};
