// @ths/design/es/components/quality-statistics/src/index.vue2.js
import { defineComponent as w, computed as c, reactive as v, watch as d, onMounted as S, toRefs as U } from "vue";
import b from "lodash/merge";
import f from "lodash/cloneDeep";
import { getDomRandomClass as C, setStyle as m } from "../../../utils/index.js";
const s = {
  name: "name",
  value: "value",
  valueUnit: "valueUnit",
  duration: "duration",
  durationUnit: "durationUnit",
  color: "color",
  tooltip: "tooltip"
}, D = w({
  name: "TQualityStatistics",
  props: {
    // 设置提示框主题
    effect: {
      type: String,
      default: () => "dark"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        wrapperTop: {
          default: {}
        },
        wrapperBottom: {
          default: {}
        },
        // 单项容器
        itemWrapper: {
          default: {}
        },
        icon: {
          default: {}
        },
        name: {
          default: {}
        },
        value: {
          default: {}
        },
        valueUnit: {
          default: {}
        },
        duration: {
          default: {}
        },
        durationUnit: {
          default: {}
        }
      })
    },
    // 栅格间隔
    spacingX: {
      default: 10,
      type: Number
    },
    // 栅格列数
    col: {
      default: 5,
      type: Number
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => f(s),
      require: !1
    },
    // tooltip位置
    tooltipPlacement: {
      type: String,
      default: () => "top"
    }
  },
  setup(e, { emit: $ }) {
    const r = c(() => e.cStyle), y = c(() => e.effect), n = C("quality-statistics"), o = {
      wrapper: ".t-quality-statistics",
      wrapperTop: ".wrapper__info",
      icon: ".wrapper__info--name .icon",
      name: ".wrapper__info--name .name",
      value: ".wrapper__info--num .num",
      valueUnit: ".wrapper__info--num .unit",
      wrapperBottom: ".wrapper__duration",
      itemWrapper: ".wrapper__col",
      duration: ".wrapper__duration .num",
      durationUnit: ".wrapper__duration .unit"
    }, i = v({
      curData: e.data
    }), _ = () => {
      let t = "";
      return Object.keys(o).forEach((a) => {
        r.value[a] && (a === "wrapper" ? t += m(r.value[a], `.${n}${o[a]}`) : t += m(r.value[a], `.${n} ${o[a]}`));
      }), t;
    }, u = () => {
      const t = document.querySelector(`.${n} .style-container`);
      t && (t.innerHTML = `<style>${_()}</style>`);
    };
    return d(() => e.cStyle, () => {
      u();
    }), d(() => [e.data, e.alias], () => {
      const t = b(f(s), e.alias);
      i.curData = e.data.map((a) => {
        const l = {};
        return Object.keys(t).forEach((p) => {
          l[p] = a[t[p]];
        }), l;
      });
    }, {
      deep: !0,
      immediate: !0
    }), S(() => {
      u();
    }), {
      curEffect: y,
      ...U(i),
      curStyle: r,
      domOnlyClass: n,
      curStyleMap: o
    };
  }
});
export {
  D as default
};
