// @ths/design/es/components/quality-statistics/src/index.vue.js
import r from "./index.vue2.js";
import { resolveComponent as p, openBlock as a, createElementBlock as e, normalizeClass as d, createElementVNode as s, Fragment as u, renderList as h, normalizeStyle as c, createBlock as f, withCtx as m, toDisplayString as t } from "vue";
import "./index.vue3.js";
import v from "../../../_virtual/_plugin-vue_export-helper.js";
const w = { class: "wrapper" }, y = { class: "wrapper__content" }, k = { class: "wrapper__info" }, g = { class: "wrapper__info--name" }, C = { class: "name" }, U = { class: "wrapper__info--num" }, $ = { class: "num" }, B = { class: "unit" }, E = { class: "wrapper__duration" }, S = { class: "num" }, b = { class: "unit" }, z = {
  key: 1,
  class: "wrapper__content"
}, D = { class: "wrapper__info" }, q = { class: "wrapper__info--name" }, F = { class: "name" }, L = { class: "wrapper__info--num" }, N = { class: "num" }, O = { class: "unit" }, P = { class: "wrapper__duration" }, Q = { class: "num" }, V = { class: "unit" };
function X(o, l, j, A, G, H) {
  const i = p("el-tooltip");
  return a(), e("div", {
    class: d(["t-quality-statistics", o.domOnlyClass])
  }, [
    s("div", w, [
      (a(!0), e(u, null, h(o.curData, (n, _) => (a(), e("div", {
        key: _,
        class: "wrapper__col",
        style: c({ width: 100 / o.col + "%", padding: "0 " + o.spacingX / 2 + "px" })
      }, [
        n.tooltip ? (a(), f(i, {
          key: 0,
          class: "item",
          effect: o.curEffect,
          content: n.tooltip || "",
          placement: o.tooltipPlacement
        }, {
          default: m(() => [
            s("div", y, [
              s("div", k, [
                s("div", g, [
                  s("span", {
                    class: "icon",
                    style: c({ background: n.color })
                  }, null, 4),
                  s("span", C, t(n.name), 1)
                ]),
                s("div", U, [
                  s("span", $, t(n.value), 1),
                  s("span", B, t(n.valueUnit), 1)
                ])
              ]),
              s("div", E, [
                s("span", S, t(n.duration), 1),
                s("span", b, t(n.durationUnit), 1)
              ])
            ])
          ]),
          _: 2
        }, 1032, ["effect", "content", "placement"])) : (a(), e("div", z, [
          s("div", D, [
            s("div", q, [
              s("span", {
                class: "icon",
                style: c({ background: n.color })
              }, null, 4),
              s("span", F, t(n.name), 1)
            ]),
            s("div", L, [
              s("span", N, t(n.value), 1),
              s("span", O, t(n.valueUnit), 1)
            ])
          ]),
          s("div", P, [
            s("span", Q, t(n.duration), 1),
            s("span", V, t(n.durationUnit), 1)
          ])
        ]))
      ], 4))), 128))
    ]),
    l[0] || (l[0] = s("div", { class: "style-container" }, null, -1))
  ], 2);
}
const R = /* @__PURE__ */ v(r, [["render", X]]);
export {
  R as default
};
