// @ths/design/es/components/chart-border-rose/src/index.vue.js
import N from "./index.vue2.js";
import { openBlock as s, createElementBlock as i, normalizeClass as L, normalizeStyle as l, withDirectives as D, createElementVNode as t, toDisplayString as S, createCommentVNode as m, renderSlot as T, Fragment as u, renderList as p, vShow as E } from "vue";
import "./index.vue3.js";
import j from "../../../_virtual/_plugin-vue_export-helper.js";
const F = {
  ref: "borderRose",
  class: "chart"
}, M = { class: "slot" }, V = ["onClick"], O = { class: "list-item__wrap" }, q = ["title"], A = ["onClick"], G = ["title"], H = { class: "empty" }, J = ["src"];
function K(e, d, P, Q, U, W) {
  var c, f, h, g, v, k, $, b, w, C, z, B;
  return s(), i("div", {
    class: L(["t-chart-border-rose", e.domOnlyClass]),
    style: l({
      ...(c = e.listDataStyle) == null ? void 0 : c.style,
      ...e.cStyle,
      ...(f = e.cStyles) == null ? void 0 : f.root,
      ...(g = (h = e.cStyles) == null ? void 0 : h.wrapper) == null ? void 0 : g.default
    })
  }, [
    D(t("main", {
      style: l({
        ...(v = e.listDataStyle) == null ? void 0 : v.style
      })
    }, [
      t("div", {
        style: l({ ...e.style, ...(k = e.cStyles) == null ? void 0 : k.chart, ...(b = ($ = e.cStyles) == null ? void 0 : $.chart) == null ? void 0 : b.default }),
        class: "chart-box"
      }, [
        t("div", F, null, 512),
        t("div", M, [
          e.curInnerText ? (s(), i("div", {
            key: 0,
            class: "inner-text",
            style: l({ ...(C = (w = e.cStyles) == null ? void 0 : w.innerText) == null ? void 0 : C.default })
          }, S(e.curInnerText), 5)) : m("", !0),
          T(e.$slots, "default")
        ])
      ], 4),
      e.listDataStyle.show === "row" && e.showList ? (s(), i("div", {
        key: 0,
        style: l({
          ...e.lStyle,
          ...e.cStyles.legendRoot,
          ...(z = e.cStyles.legendRoot) == null ? void 0 : z.default
        }),
        class: "row-list"
      }, [
        (s(!0), i(u, null, p(e.chartDataRow, (n, o) => {
          var a;
          return s(), i("div", {
            key: o,
            style: l({ ...e.cStyles.legendRow, ...(a = e.cStyles.legendRow) == null ? void 0 : a.default }),
            class: "list-wrap"
          }, [
            (s(!0), i(u, null, p(n, (r, y) => {
              var I, R;
              return s(), i("div", {
                key: r.name,
                class: "list-item",
                style: l({
                  ...e.listStyle.cStyle,
                  ...e.cStyles.legendItem,
                  ...(I = e.cStyles.legendItem) == null ? void 0 : I.default
                }),
                onClick: (X) => e.handleLegendItemClick(
                  r,
                  o * Math.ceil(e.data.length / e.listStyle.col) + y
                )
              }, [
                t("div", O, [
                  t("div", {
                    class: "list-item__box",
                    style: l({
                      background: e.borderColors[o * Math.ceil(e.data.length / e.listStyle.col) + y],
                      "flex-basis": `${e.listStyle.iconSize}px`,
                      "border-radius": e.listStyle.iconBorder === "string" ? `${e.listStyle.iconBorder}` : `${e.listStyle.iconBorder}px`,
                      height: `${e.listStyle.iconSize}px`
                    })
                  }, null, 4),
                  t("div", {
                    style: l({
                      fontSize: e.listStyle.fontSize ? e.listStyle.fontSize + "px" : "14px",
                      ...e.cStyles.legendName,
                      ...(R = e.cStyles.legendName) == null ? void 0 : R.default
                    }),
                    class: "list-item__address",
                    title: r.name
                  }, S(r.name), 13, q)
                ])
              ], 12, V);
            }), 128))
          ], 4);
        }), 128))
      ], 4)) : m("", !0),
      e.listDataStyle.show === "col" && e.showList ? (s(), i("div", {
        key: 1,
        style: l({
          ...e.lStyle,
          ...e.cStyles.legendRoot,
          ...(B = e.cStyles.legendRoot) == null ? void 0 : B.default,
          gridTemplateColumns: `repeat(${e.col}, minmax(120px, 1fr))`
        }),
        class: "col-list"
      }, [
        (s(!0), i(u, null, p(e.curData, (n, o) => {
          var a, r;
          return s(), i("div", {
            key: o,
            style: l({
              ...e.listStyle.cStyle,
              ...e.cStyles.legendItem,
              ...(a = e.cStyles.legendItem) == null ? void 0 : a.default
            }),
            class: "list-item",
            onClick: (y) => e.handleLegendItemClick(n, o)
          }, [
            t("div", {
              class: "list-item__box",
              style: l({
                background: e.borderColors[o],
                "flex-basis": `${e.listStyle.iconSize}px`,
                "border-radius": typeof e.listStyle.iconBorder == "string" ? `${e.listStyle.iconBorder}` : `${e.listStyle.iconBorder}px`,
                height: `${e.listStyle.iconSize}px`
              })
            }, null, 4),
            t("div", {
              style: l({
                fontSize: e.listStyle.fontSize ? e.listStyle.fontSize + "px" : "14px",
                ...e.cStyles.legendName,
                ...(r = e.cStyles.legendName) == null ? void 0 : r.default
              }),
              class: "list-item__address",
              title: n.name
            }, S(n.name), 13, G)
          ], 12, A);
        }), 128))
      ], 4)) : m("", !0)
    ], 4), [
      [E, e.checkEmpty]
    ]),
    D(t("div", H, [
      t("img", {
        class: "empty-img",
        style: l({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, J),
      d[0] || (d[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [E, !e.checkEmpty]
    ]),
    d[1] || (d[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const ee = /* @__PURE__ */ j(N, [["render", K]]);
export {
  ee as default
};
