// @ths/design/es/components/chart-border-rose/src/index.vue2.js
import { defineComponent as W, ref as m, computed as s, onMounted as V, onUnmounted as G, watch as J, nextTick as L, useCssVars as K } from "vue";
import O from "lodash/merge";
import Q from "lodash/cloneDeep";
import * as X from "echarts";
import { splitArray as Y, getDomRandomClass as Z, rgb2rgba as ee, replacePollutionHtml as te, setStyle as f } from "../../../utils/index.js";
import { POLLUTION_HTML as le } from "../../../utils/constant.js";
const q = { name: "name", value: "value" }, T = [
  "rgba(79, 158, 254, 1)",
  "rgba(10, 204, 204, 1)",
  "rgba(19, 235, 156, 1)",
  "rgba(255, 213, 67, 1)",
  "rgba(252, 90, 90, 1)",
  "rgba(233, 67, 255, 1)",
  "rgba(255, 166, 79, 1)"
], re = [
  ["rgba(255,255,255,0)", "rgba(79, 158, 254, 0.8)"],
  ["rgba(10, 204, 204, 0)", "rgba(10, 204, 204, 0.8)"],
  ["rgba(19, 235, 156, 0)", "rgba(19, 235, 156, 0.8)"],
  ["rgba(255, 213, 67, 0)", "rgba(255, 213, 67, 0.8)"],
  ["rgba(252, 90, 90, 0)", "rgba(252, 90, 90, 0.8)"],
  ["rgba(233, 67, 255, 0)", "rgba(233, 67, 255, 0.8)"],
  ["rgba(255, 166, 79, 0)", "rgba(255, 166, 79, 0.8)"]
], A = {
  wrapper: {
    default: {}
  },
  chart: {
    default: {}
  },
  legendName: {
    default: {}
  }
}, h = {
  small: [0.8, 2.5],
  middle: [0.7, 2.9],
  big: [0.6, 3.4]
}, k = W({
  name: "TChartBorderRose",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 饼图数据
    data: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    alias: {
      type: Object,
      default: () => q,
      required: !1
    },
    splitType: {
      type: String,
      default: "small",
      required: !1
    },
    // 中心文字
    innerText: {
      type: String,
      default: "",
      required: !1
    },
    // 显示数据列表
    showList: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 分割线颜色
    splitLineColor: {
      type: String,
      default: "rgba(255,255,255,0.1)",
      required: !1
    },
    // 数据列表排列方式
    listStyle: {
      type: Object,
      default: () => ({
        align: "right",
        fontSize: 14,
        iconSize: 14,
        iconBorder: 0,
        col: 2,
        cStyle: {}
      }),
      required: !1,
      validator: (e) => !!e
    },
    // 列表容器自定义样式
    lStyle: {
      type: Object,
      default: () => ({}),
      required: !1,
      validator: (e) => !!e
    },
    // 整个图形宽度
    width: {
      type: String,
      default: "100%",
      required: !1
    },
    // 整个图形高度
    height: {
      type: String,
      default: "300px",
      required: !1
    },
    // 整个图形命名
    title: {
      type: String,
      default: "",
      required: !1
    },
    // 图表扇形颜色
    fillColors: {
      type: Array,
      default: () => re,
      required: !1
    },
    // 图表边框颜色
    borderColors: {
      type: Array,
      default: () => T,
      required: !1
    },
    // 饼图半径
    radius: {
      type: Array,
      default: () => ["30%", "80%"],
      required: !1,
      validator: (e) => e.length === 2
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({ ...A }),
      required: !1,
      validator: (e) => !!e
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    }
  },
  emits: ["com-click", "legend-click"],
  setup(e, { emit: y }) {
    const g = m(null), c = m(e.data), z = s(() => e.innerText), b = {
      dataColor: "#666666",
      fontColor: "#666666",
      separationColor: "#EEE",
      separationColorOpacity: 1,
      maskColor: "#FFF",
      maskColorOpacity: 0.8
    }, n = s(() => e.cStyle), u = Z("chart-border-rose"), D = s(() => ({
      ...A,
      ...e.cStyle
    })), E = () => e.showList ? ["left", "right"].includes(e.listStyle.align) ? {
      show: "row",
      style: {
        flexDirection: e.listStyle.align === "right" ? "row" : "row-reverse",
        alignItems: "center"
      },
      cStyle: e.listStyle.cStyle
    } : {
      show: "col",
      style: {
        flexDirection: e.listStyle.align === "bottom" ? "column" : "column-reverse",
        alignItems: "center"
      },
      cStyle: e.listStyle.cStyle
    } : {
      show: "",
      value: !1,
      style: {},
      cStyle: {}
    }, j = (t = "") => {
      const l = t.toUpperCase().replace(/[-.+_]/g, ""), r = le.filter((a) => a.name === l);
      return r.length > 0 ? r[0].html : t;
    }, R = () => {
      let t = "";
      return n.value.wrapper && (t += f(n.value.wrapper, `.${u}`)), n.value.emptyContainer && (t += f(n.value.emptyContainer, `.${u} .empty`)), n.value.emptyImg && (t += f(n.value.emptyImg, `.${u} .empty .empty-img`)), n.value.emptyText && (t += f(n.value.emptyText, `.${u} .empty .empty-text`)), t;
    }, S = () => {
      const t = document.querySelector(`.${u} .style-container`);
      t && (t.innerHTML = `<style>${R()}</style>`);
    }, $ = s(() => {
      var t, l, r, a;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((d) => {
            var x;
            return (d == null ? void 0 : d.data) && ((x = d == null ? void 0 : d.data) == null ? void 0 : x.length) !== 0;
          });
        if (typeof ((l = e.options) == null ? void 0 : l.series) == "object")
          return ((a = (r = e.options.series) == null ? void 0 : r.data) == null ? void 0 : a.length) === 0;
      }
      return !!e.data.length;
    }), F = (t, l = T) => [...t].map((a, o) => ({
      ...a,
      name: j(a.name),
      style: {
        color: l[o % l.length],
        legend: {
          nameColor: b.fontColor,
          valueColor: b.dataColor,
          backgroundColor: l[o % l.length],
          borderColor: ee(l[o % l.length], 0.5)
        }
      }
    })), H = (t, l) => {
      const r = [];
      return t.forEach((a, o) => {
        r.push({
          type: "radial",
          x: l.getWidth() / 2,
          y: l.getHeight() / 2,
          r: 200,
          colorStops: [
            {
              offset: 0,
              color: e.fillColors[o][0]
              // 0% 处的颜色
            },
            {
              offset: 1,
              color: e.fillColors[o][1]
              // 100% 处的颜色
            }
          ],
          global: !0
          // 缺省为 false
        });
      }), r;
    };
    function P(t, l) {
      const r = [l.value(0), l.value(1)], a = l.coord(r), o = l.size([1, 1], r);
      return {
        type: "sector",
        // 装饰
        shape: {
          cx: t.coordSys.cx,
          cy: t.coordSys.cy,
          r0: a[2] - o[0] / 1.5,
          r: a[2] + o[0] / 2,
          startAngle: -(a[3] + o[1] / h[e.splitType][1]),
          endAngle: -(a[3] - o[1] / h[e.splitType][1])
        },
        style: l.style()
      };
    }
    function C() {
      const t = O(Q(q), e.alias);
      c.value = e.data.map((l) => {
        const r = {};
        return Object.keys(t).forEach((a) => {
          a === "name" ? r[a] = te(l[t[a]]) : r[a] = l[t[a]];
        }), { ...l, ...r };
      });
    }
    const B = s(() => {
      var t;
      return {
        width: e.showList ? e.height : e.width,
        height: e.height,
        background: (t = e.options) == null ? void 0 : t.backgroundColor
      };
    });
    let i = null;
    const v = () => {
      L(() => {
        i.resize();
      });
    }, M = s(() => Y(F(c.value, e.borderColors), e.listStyle.col || 2)), w = () => {
      if (g.value) {
        i || (i = X.init(g.value));
        const t = h[e.splitType][0], l = O({
          title: {
            text: e.title,
            left: "center",
            padding: [0, 0, 40, 0]
          },
          xAxis: {
            show: !1
          },
          yAxis: {
            show: !1
          },
          polar: {
            radius: e.radius
          },
          radiusAxis: {
            axisLine: {
              show: !1
            },
            axisTick: {
              show: !1
            },
            axisLabel: {
              show: !1
            },
            splitLine: {
              lineStyle: {
                color: e.splitLineColor
              }
            }
          },
          angleAxis: {
            show: !1,
            type: "category",
            startAngle: 75
          },
          tooltip: {
            show: !0,
            backgroundColor: "rgba(7, 26, 55, 0.8)",
            borderColor: "rgba(255, 255, 255, 0.2)",
            textStyle: {
              fontSize: 14,
              color: "rgba(201, 232, 255, 1)"
            },
            axisPointer: {
              lineStyle: {
                color: "rgba(255,255,255,0.2)",
                width: 1
              },
              crossStyle: {
                color: "red",
                width: 1
              }
            }
          },
          series: [
            {
              type: "bar",
              coordinateSystem: "polar",
              barWidth: `${t * 100}%`,
              data: c.value,
              zlevel: 3,
              itemStyle: {
                color: (r) => H(e.data, i)[r.dataIndex]
              },
              label: {}
            },
            {
              type: "custom",
              coordinateSystem: "polar",
              renderItem: P,
              data: c.value,
              zlevel: 2,
              tooltip: {
                show: !1
              },
              itemStyle: {
                color(r) {
                  return e.borderColors[r.dataIndex];
                },
                opacity: 1
              }
            }
          ]
        }, e.options);
        setTimeout(() => {
          i.setOption(l);
        }, 300);
      }
    }, p = m(e.listStyle.col);
    V(() => {
      S(), C(), w(), window.addEventListener("resize", v), i.on("click", (t) => {
        y("com-click", t);
      });
    }), G(() => {
      window.removeEventListener("resize", v);
    });
    function N(t, l) {
      y("legend-click", {
        current: t,
        currentIndex: l,
        all: c.value
      });
    }
    const U = s(() => E());
    return J(() => e, () => {
      try {
        p.value = e.listStyle.col, S(), C(), w(), L(() => {
          i.resize();
        });
      } catch (t) {
        console.log(t);
      }
    }, { deep: !0 }), {
      borderRose: g,
      curStyle: n,
      domOnlyClass: u,
      listDataStyle: U,
      cStyles: D,
      curData: c,
      style: B,
      curInnerText: z,
      chartDataRow: M,
      col: p,
      handleLegendItemClick: N,
      checkEmpty: $
    };
  }
}), _ = () => {
  K((e) => ({
    "4f8f6a92": e.col
  }));
}, I = k.setup;
k.setup = I ? (e, y) => (_(), I(e, y)) : _;
export {
  k as default
};
