// @ths/design/es/components/compare/src/index.vue2.js
import { defineComponent as p, computed as f, reactive as y, watch as u, onMounted as b, toRefs as C } from "vue";
import S from "lodash/merge";
import i from "lodash/cloneDeep";
import { getDomRandomClass as _, setStyle as m } from "../../../utils/index.js";
const d = {
  label: "label",
  unit: "unit",
  value: "value",
  compareLabel: "compareLabel",
  compareUnit: "compareUnit",
  compareValue: "compareValue",
  compareResult: "compareResult",
  upIconColor: "upIconColor",
  downIconColor: "downIconColor",
  equalIconColor: "equalIconColor"
}, O = p({
  name: "TCompare",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        label: {
          default: {}
        },
        unit: {
          default: {}
        },
        value: {
          default: {}
        },
        compareValue: {
          default: {}
        },
        line: {
          default: {}
        },
        lineDecorate: {
          default: {}
        }
      })
    },
    // 单位是否换行
    unitBreak: {
      default: !0,
      type: Boolean
    },
    // 排列方向 horizontal  vertical
    direction: {
      default: "vertical",
      type: String
    },
    // 数据
    data: {
      type: Object,
      default: () => ({})
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => i(d),
      require: !1
    }
  },
  setup(a) {
    const l = f(() => a.cStyle), o = _("t-compare"), n = {
      wrapper: ".t-compare",
      label: ".card-content .label",
      unit: ".card-content .unit",
      value: ".card-content__left .num",
      compareValue: ".card-content__right .num",
      line: ".card-content__border .line",
      lineDecorate: ".card-content__border span"
    }, r = y({
      curData: a.data
    }), s = () => {
      let e = "";
      return Object.keys(n).forEach((t) => {
        l.value[t] && (t === "wrapper" ? e += m(l.value[t], `.${o}${n[t]}`) : e += m(l.value[t], `.${o} ${n[t]}`));
      }), e;
    }, c = () => {
      const e = document.querySelector(`.${o} .style-container`);
      e && (e.innerHTML = `<style>${s()}</style>`);
    };
    return u(() => a.cStyle, () => {
      c();
    }), u(() => [a.data, a.alias], () => {
      const e = S(i(d), a.alias);
      Object.keys(e).forEach((t) => {
        r.curData[t] = a.data[e[t]];
      });
    }, {
      deep: !0,
      immediate: !0
    }), b(() => {
      c();
    }), {
      ...C(r),
      curStyle: l,
      domOnlyClass: o,
      curStyleMap: n
    };
  }
});
export {
  O as default
};
