// @ths/design/es/components/tree-panel/src/index.vue2.js
import { defineComponent as H, computed as N, ref as M, reactive as P, onMounted as R, watch as B, toRefs as q } from "vue";
import { getDomRandomClass as A, setStyle as D, getCssPropertyString as c } from "../../../utils/index.js";
const F = H({
  name: "TTreePanel",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 箭头容器
        arrow: {
          default: {}
        },
        // 非叶子节点箭头图标
        arrowNonLeaf: {
          default: {}
        },
        // 叶子节点箭头图标
        arrowLeaf: {
          default: {}
        },
        // 复选框容器
        checkbox: {
          default: {},
          hover: {},
          active: {}
        },
        // 节点文字容器
        nodeTextContainer: {
          default: {}
        },
        // 文本
        text: {
          default: {},
          active: {}
        }
      })
    },
    data: {
      type: Array,
      default: () => []
    },
    defaultProps: {
      type: Object,
      default: () => ({
        children: "children",
        label: "label"
      })
    },
    nodeKey: {
      type: String,
      default: ""
    },
    moreBtn: {
      type: Boolean,
      default: !1
    },
    // 类型（单选、多选）
    type: {
      type: String,
      default: "single"
    }
  },
  emits: ["node-click"],
  setup(d, { emit: L }) {
    const t = N(() => d.cStyle), l = A("tree-panel"), T = M(), i = P({
      activeNodeKey: "",
      // 当前激活的节点key
      isCheckbox: N(() => d.type === "multiple")
      // 是否多选
    }), K = () => {
      let e = "";
      const n = `.${l}.t-tree-panel`, o = {
        wrapper: `${n} .el-tree`,
        arrow: `${n} .el-tree .el-tree-node__expand-icon::before`,
        arrowNonLeaf: `${n} .el-tree .el-tree-node__expand-icon:not(.is-leaf)::before`,
        arrowLeaf: `${n} .el-tree .el-tree-node__expand-icon.is-leaf::before`,
        nodeTextContainer: `${n} .el-tree .custom-tree-node`
      };
      return Object.entries(t.value).forEach(([r, a]) => {
        o[r] && a && (e += D(a, o[r]));
      }), e;
    }, O = () => {
      var n, o, r, a, u, s, p, _, y, m, v, b, x, h, $, k, S, C, w, g;
      let e = "";
      return (o = (n = t.value) == null ? void 0 : n.checkbox) != null && o.default && (e += `.${l}.t-tree-panel .el-tree .el-tree-node__content .el-checkbox__inner{
          ${c((a = (r = t.value) == null ? void 0 : r.checkbox) == null ? void 0 : a.default)}
        }`), (s = (u = t.value) == null ? void 0 : u.checkbox) != null && s.hover && (e += `.${l}.t-tree-panel .el-tree .el-tree-node__content .el-checkbox__inner:hover{
          ${c((_ = (p = t.value) == null ? void 0 : p.checkbox) == null ? void 0 : _.hover)}
        }`), (m = (y = t.value) == null ? void 0 : y.checkbox) != null && m.active && (e += `.${l}.t-tree-panel .el-tree .el-tree-node__content .el-checkbox__input.is-checked .el-checkbox__inner{
          ${c((b = (v = t.value) == null ? void 0 : v.checkbox) == null ? void 0 : b.active)}
        }`), (h = (x = t.value) == null ? void 0 : x.text) != null && h.default && (e += `.${l}.t-tree-panel .el-tree .el-tree-node__content .custom-tree-node__label{
          ${c((k = ($ = t.value) == null ? void 0 : $.text) == null ? void 0 : k.default)}
        }`), (C = (S = t.value) == null ? void 0 : S.text) != null && C.active && (e += `.${l}.t-tree-panel .el-tree .el-tree-node__content .custom-tree-node__label.active{
          ${c((g = (w = t.value) == null ? void 0 : w.text) == null ? void 0 : g.active)}
        }`), e;
    }, f = () => {
      const e = document.querySelector(`.${l} .style-container`);
      e && (e.innerHTML = `<style>${K()}${O()}</style>`);
    };
    R(() => {
      f();
    }), B(() => d.cStyle, () => {
      f();
    }, { deep: !0 });
    const j = (e) => {
      i.activeNodeKey = e[d.nodeKey], L("node-click", e);
    };
    return {
      ...q(i),
      curStyle: t,
      domOnlyClass: l,
      treeRef: T,
      handleNodeClick: j
    };
  }
});
export {
  F as default
};
