// @ths/design/es/components/tree-panel/src/index.vue.js
import u from "./index.vue2.js";
import { resolveComponent as i, openBlock as d, createElementBlock as p, normalizeClass as m, createVNode as r, mergeProps as C, withCtx as l, createElementVNode as s, toDisplayString as y, createTextVNode as c, createCommentVNode as v } from "vue";
import "./index.vue3.js";
import b from "../../../_virtual/_plugin-vue_export-helper.js";
const N = { class: "custom-tree-node" }, _ = { key: 0 };
function $(e, o, h, g, V, w) {
  const a = i("el-button"), k = i("el-tree");
  return d(), p("div", {
    class: m(["t-tree-panel", e.domOnlyClass])
  }, [
    r(k, C(e.$attrs, {
      ref: "treeRef",
      data: e.data,
      props: e.defaultProps,
      "node-key": e.nodeKey,
      "show-checkbox": e.isCheckbox,
      onNodeClick: e.handleNodeClick
    }), {
      default: l(({ node: t, data: n }) => [
        s("div", N, [
          s("span", {
            class: m(["custom-tree-node__label", { active: n[e.nodeKey] === e.activeNodeKey && t.isLeaf && !e.isCheckbox }])
          }, y(t.label), 3),
          e.moreBtn ? (d(), p("div", _, [
            r(a, {
              type: "primary",
              link: "",
              onClick: (f) => e.append(n)
            }, {
              default: l(() => o[0] || (o[0] = [
                c(" Append ")
              ])),
              _: 2,
              __: [0]
            }, 1032, ["onClick"]),
            r(a, {
              style: { "margin-left": "4px" },
              type: "danger",
              link: "",
              onClick: (f) => e.remove(t, n)
            }, {
              default: l(() => o[1] || (o[1] = [
                c(" Delete ")
              ])),
              _: 2,
              __: [1]
            }, 1032, ["onClick"])
          ])) : v("", !0)
        ])
      ]),
      _: 1
    }, 16, ["data", "props", "node-key", "show-checkbox", "onNodeClick"]),
    o[2] || (o[2] = s("div", { class: "style-container" }, null, -1))
  ], 2);
}
const E = /* @__PURE__ */ b(u, [["render", $]]);
export {
  E as default
};
