// @ths/design/es/components/timer/src/index.vue2.js
import { defineComponent as $, computed as H, reactive as D, onBeforeUnmount as p, watch as b, onMounted as R, toRefs as C } from "vue";
import { getDomRandomClass as E } from "../../../utils/index.js";
const O = $({
  name: "TTimer",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 时间图标
        icon: {
          default: {}
        },
        // 年月日时间文本容器
        yearMdWrapper: {
          default: {}
        },
        // 星期几时间文本容器
        weekDayWrapper: {
          default: {}
        },
        // 时分秒时间文本容器
        HourMSWrapper: {
          default: {}
        }
      })
    },
    // 时间格式
    timeFormat: {
      type: String,
      default: "YYYY-MM-dd hh:mm:ss"
    },
    // 是否显示星期几
    showDay: {
      type: Boolean,
      default: !1
    },
    // 文字阴影样式
    textShadowStyle: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e, { emit: T }) {
    const M = H(() => e.cStyle), l = E("timer");
    let m = null;
    const o = D({
      yearMd: "",
      // 年月日
      HourMS: "",
      // 时分秒
      weekDay: ""
      // 星期几
    }), s = ["星期日", "星期一", "星期二", "星期三", "星期四", "星期五", "星期六"], i = (t) => {
      const a = /* @__PURE__ */ new Date(), y = a.getDay();
      let r = t, S = "", c = "";
      const n = {
        "[Mm]+": a.getMonth() + 1,
        // 月份
        "[Dd]+": a.getDate(),
        // 日
        "[Hh]+": a.getHours(),
        // 小时
        "[m]+": a.getMinutes(),
        // 分
        "[Ss]+": a.getSeconds(),
        // 秒
        "[Qq]+": Math.floor((a.getMonth() + 3) / 3),
        // 季度
        S: a.getMilliseconds()
        // 毫秒
      };
      return /([Yy]+)/.test(r) && (r = r.replace(RegExp.$1, `${a.getFullYear()}`.substr(4 - RegExp.$1.length))), Object.keys(n).forEach((d) => {
        new RegExp(`(${d})`).test(r) && (r = r.replace(RegExp.$1, RegExp.$1.length === 1 ? n[d] : `00${n[d]}`.substr(`${n[d]}`.length)), [S, c] = r.split(" "));
      }), {
        yearMd: S,
        HourMS: c,
        day: y
      };
    }, f = () => {
      m = setInterval(() => {
        const { yearMd: t, HourMS: a, day: y } = i(e.timeFormat);
        o.yearMd = t, o.HourMS = a, o.weekDay = s[y];
      }, 1e3);
    }, { yearMd: h, HourMS: w, day: g } = i(e.timeFormat);
    o.yearMd = h, o.HourMS = w, o.weekDay = s[g], f(), p(() => {
      clearInterval(m);
    });
    const x = () => {
      let t = "";
      return e.textShadowStyle.x && (t += `.${l}.t-timer .time {
          --t-timer-shadow-x: ${e.textShadowStyle.x || 0};
        }`), e.textShadowStyle.y && (t += `.${l}.t-timer .time {
          --t-timer-shadow-y: ${e.textShadowStyle.y || 0};
        }`), e.textShadowStyle.blur && (t += `.${l}.t-timer .time {
          --t-timer-shadow-blur: ${e.textShadowStyle.blur || 0};
        }`), e.textShadowStyle.color && (t += `.${l}.t-timer .time {
          --t-timer-shadow-color: ${e.textShadowStyle.color || "transparent"};
        }`), t;
    }, u = () => {
      const t = document.querySelector(`.${l} > .style-container`);
      t && (t.innerHTML = `<style>${x()}}</style>`);
    };
    return b(() => e.textShadowStyle, () => {
      u();
    }, {
      deep: !0
    }), R(() => {
      u();
    }), {
      ...C(o),
      curStyle: M,
      domOnlyClass: l
    };
  }
});
export {
  O as default
};
