// @ths/design/es/components/iframe/src/index.vue2.js
import { defineComponent as s, computed as c, reactive as i, watch as m, toRefs as y } from "vue";
import g from "lodash/merge";
const $ = s({
  name: "TIframe",
  props: {
    // 加载中的状态
    loading: {
      type: Boolean,
      default: !1
    },
    width: {
      type: String,
      default: "100%"
    },
    height: {
      type: String,
      default: "100%"
    },
    scrolling: {
      type: String,
      default: "no"
    },
    src: {
      type: String,
      default: ""
    },
    srcdoc: {
      type: String,
      default: ""
    },
    // socket链接
    socketUrl: {
      type: String,
      default: ""
    },
    // 房间号
    roomId: {
      type: String,
      default: ""
    },
    // 拼接的参数
    params: {
      type: Array,
      default: () => []
    },
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    }
  },
  setup(e) {
    const l = c(() => g(e.cStyle.wrapper.default, {})), o = c(() => e.scrolling), r = i({
      contentKey: "",
      contentValue: ""
    }), f = ["socketUrl", "roomId"];
    return m([() => e.src, () => e.srcdoc, () => e.params, () => e.socketUrl, () => e.roomId], () => {
      const n = e.params.map((a) => a);
      if (e.srcdoc)
        r.contentKey = "srcdoc", r.contentValue = e.srcdoc;
      else {
        r.contentKey = "src";
        let { src: a } = e;
        if (!a)
          return;
        const u = a.split("?")[1] ?? "";
        f.forEach((t) => {
          e[t] && n.unshift({
            name: t,
            value: e[t]
          });
        }), u ? n.forEach((t) => {
          a += `&${t.name}=${t.value}`;
        }) : n.forEach((t, d) => {
          a += d === 0 ? `?${t.name}=${t.value}` : `&${t.name}=${t.value}`;
        }), r.contentValue = a;
      }
    }, {
      deep: !0,
      immediate: !0
    }), {
      ...y(r),
      currentStyle: l,
      currentScrolling: o
    };
  }
});
export {
  $ as default
};
