// @ths/design/es/components/iframe/src/index.vue.js
import i from "./index.vue2.js";
import { openBlock as n, createElementBlock as s, normalizeClass as a, normalizeStyle as c, createElementVNode as r, createCommentVNode as m, normalizeProps as d } from "vue";
import "./index.vue3.js";
import p from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  key: 0,
  class: "el-loading-spinner"
}, f = ["width", "height", "scrolling"];
function h(e, o, g, y, S, v) {
  var t, l;
  return n(), s("div", {
    class: a(["t-iframe", { empty: !e.src && !e.srcdoc }]),
    style: c({ ...e.currentStyle })
  }, [
    e.loading && !(e.src || e.srcdoc) ? (n(), s("div", u, o[0] || (o[0] = [
      r("svg", {
        class: "circular",
        viewBox: "25 25 50 50"
      }, [
        r("circle", {
          class: "path",
          cx: "50",
          cy: "50",
          r: "20",
          fill: "none"
        })
      ], -1),
      r("p", { class: "el-loading-text" }, " 加载中... ", -1)
    ]))) : m("", !0),
    r("iframe", d({
      width: ((t = e.currentStyle) == null ? void 0 : t.width) || "100%",
      height: ((l = e.currentStyle) == null ? void 0 : l.height) || "100%",
      [e.contentKey || ""]: e.contentValue,
      scrolling: e.currentScrolling,
      frameborder: "0"
    }), null, 16, f)
  ], 6);
}
const B = /* @__PURE__ */ p(i, [["render", h]]);
export {
  B as default
};
