// @ths/design/es/components/image-base/src/index.vue2.js
import { defineComponent as f, computed as n, watch as g, onMounted as S } from "vue";
import { getDomRandomClass as v, getCssPropertyString as b, setStyle as k } from "../../../utils/index.js";
const $ = f({
  name: "TImageBase",
  // components: { PictureFilled },
  props: {
    src: {
      type: String,
      default: ""
    },
    // 确定图片如何适应容器框
    fit: {
      type: String,
      default: "fill"
    },
    // 选中状态
    active: {
      type: Boolean,
      default: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {},
          active: {},
          hover: {}
        }
      })
    }
  },
  setup(r) {
    const i = n(() => r.active), t = n(() => r.cStyle), a = v("image-base"), u = ["background", "backgroundImage", "backgroundColor", "background-image", "background-color"], p = n(() => u.reduce((e, m) => {
      var c, l, s;
      return ((s = Object.keys((l = (c = r.cStyle) == null ? void 0 : c.wrapper) == null ? void 0 : l.default)) == null ? void 0 : s.includes(m)) || e;
    }, !1)), d = () => {
      let e = "";
      return t.value.wrapper && t.value.wrapper.active && (e += `
          .${a}.active {
            ${b(t.value.wrapper.active)}
          }
        `), e;
    }, y = () => {
      let e = "";
      return t.value.wrapper && (e += k(t.value.wrapper, `.${a}`)), e;
    }, o = () => {
      const e = document.querySelector(`.${a} .style-container`);
      e && (e.innerHTML = `<style>${y()}${d()}</style>`);
    };
    return g(() => r.cStyle, () => {
      o();
    }), S(() => {
      o();
    }), {
      isActive: i,
      domOnlyClass: a,
      includeBackgroundProperty: p
    };
  }
});
export {
  $ as default
};
