// @ths/design/es/components/chart-bar-simple/src/index.vue.js
import m from "./index.vue2.js";
import { openBlock as n, createElementBlock as p, normalizeClass as c, normalizeStyle as o, withDirectives as i, createElementVNode as e, vShow as a } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  ref: "chartId",
  class: "chart"
}, y = { class: "empty" }, f = ["src"];
function h(t, s, v, E, k, S) {
  var r, l;
  return n(), p("div", {
    class: c(["t-chart-bar-simple", t.domOnlyClass]),
    style: o({ ...(l = (r = t.cStyle) == null ? void 0 : r.wrapper) == null ? void 0 : l.default })
  }, [
    i(e("div", u, null, 512), [
      [a, t.checkEmpty]
    ]),
    i(e("div", y, [
      e("img", {
        class: "empty-img",
        style: o({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [a, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const w = /* @__PURE__ */ d(m, [["render", h]]);
export {
  w as default
};
