// @ths/design/es/components/chart-bar-simple/src/index.vue2.js
import { defineComponent as N, computed as q, ref as P, reactive as U, nextTick as L, onMounted as W, onUnmounted as H, watch as p, toRefs as K } from "vue";
import * as X from "echarts";
import y from "lodash/merge";
import A from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as J, DEFAULT_OPTIONS_AXIS as Q } from "../../../utils/constant.js";
import { getDomRandomClass as V, setStyle as w } from "../../../utils/index.js";
const Y = y(A(J), A(Q)), g = y(
  A(Y),
  {
    // x轴
    xAxis: {
      boundaryGap: !0
    },
    yAxis: {
      axisLabel: {
        formatter: (e) => String(e).indexOf(".") === -1 ? e : ""
      }
    },
    // 图例
    legend: {
      show: !0
    },
    // 网格线
    grid: {
      top: "12%",
      bottom: "2%",
      left: "1%",
      right: "2%",
      containLabel: !0
    }
  }
), ne = N({
  name: "TChartBarSimple",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外容器
        wrapper: {
          default: {}
        }
      })
    },
    // 设置主轴
    mainAxis: {
      type: String,
      default: "x"
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => g,
      required: !1
    },
    // 传入数组数据 如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 柱状图主题样式，用于平台里设置
    bar: {
      type: Object,
      default: () => ({})
    }
  },
  // 事件名定义
  emits: ["com-click", "legend-select", "resize"],
  setup(e, { emit: T }) {
    const b = V("chart-bar-compare"), d = q(() => e.cStyle), s = P(null);
    let h = null, f = null;
    const i = U({
      // 图表数据
      chartData: [],
      // 柱状图配置
      barOptions: A(e.bar)
    }), C = () => {
      var t, o;
      (o = (t = e.seriesSetting) == null ? void 0 : t.data) != null && o.length && e.data && (i.chartData = e.seriesSetting.data.map((l) => {
        const { name: c, dataKey: n } = l;
        return {
          name: c || "--",
          data: e.data.map((r) => r[n])
        };
      }));
    }, M = () => {
      let t = "";
      return d.value.wrapper && (t += w(d.value.wrapper, `.${b}`)), d.value.emptyContainer && (t += w(d.value.emptyContainer, `.${b} .empty`)), d.value.emptyImg && (t += w(d.value.emptyImg, `.${b} .empty .empty-img`)), d.value.emptyText && (t += w(d.value.emptyText, `.${b} .empty .empty-text`)), t;
    }, z = () => {
      const t = document.querySelector(`.${b} .style-container`);
      t && (t.innerHTML = `<style>${M()}</style>`);
    }, m = () => {
      if (s.value) {
        const t = s.value.getBoundingClientRect().width ?? 0;
        console.log("柱图=================>", s.value.getBoundingClientRect().width), t && L(() => {
          f.resize();
        });
      }
    };
    function B() {
      s.value && (h = new ResizeObserver((t) => {
        for (const o of t)
          o.target === s.value && m();
      }), h.observe(s.value));
    }
    const G = q(() => {
      var t, o, l, c;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((r) => {
            var S;
            return (r == null ? void 0 : r.data) && ((S = r == null ? void 0 : r.data) == null ? void 0 : S.length) !== 0;
          });
        if (typeof ((o = e.options) == null ? void 0 : o.series) == "object")
          return ((c = (l = e.options.series) == null ? void 0 : l.data) == null ? void 0 : c.length) === 0;
      }
      return e.data.length ? (L(() => {
        m();
      }), !0) : !1;
    }), O = () => {
      var r, S, E, I, j, R, _;
      const t = (r = e.options) == null ? void 0 : r.color, o = (S = e.options) == null ? void 0 : S.series;
      let l = [], c = [];
      o ? l = o.map((u, a) => {
        var F, x, v;
        const D = y(
          {
            type: "bar",
            barWidth: 14,
            barGap: "80%",
            itemStyle: {
              color: {
                type: "linear",
                x: 0,
                y: 1,
                x2: 0,
                y2: 0,
                colorStops: [
                  {
                    offset: 1,
                    color: t && t[a] || g.color[a] || "#FFFFFF00"
                    // 100% 处的颜色
                  },
                  {
                    offset: 0,
                    color: t && t[a] || g.color[a] || "#FFFFFF00"
                    // 0% 处的颜色
                  }
                ]
              }
            }
          },
          i == null ? void 0 : i.barOptions,
          (v = (x = (F = e.seriesSetting) == null ? void 0 : F.data) == null ? void 0 : x[a]) == null ? void 0 : v.style
        );
        return y({}, D, u);
      }) : e.data && ((I = (E = e.seriesSetting) == null ? void 0 : E.data) != null && I.length) && (l = (j = e.seriesSetting) == null ? void 0 : j.data.map((u, a) => {
        var F, x, v, $, k;
        const D = y(
          {
            name: (F = i.chartData[a]) == null ? void 0 : F.name,
            type: "bar",
            barWidth: 14,
            barGap: "80%",
            label: {
              show: !1,
              position: "top"
            },
            itemStyle: {
              color: {
                type: "linear",
                x: 0,
                y: 1,
                x2: 0,
                y2: 0,
                colorStops: [
                  {
                    offset: 1,
                    color: t && t[a] || g.color[a] || "#FFFFFF00"
                    // 100% 处的颜色
                  },
                  {
                    offset: 0,
                    color: t && t[a] || g.color[a] || "#FFFFFF00"
                    // 0% 处的颜色
                  }
                ]
              }
            },
            data: e.data.length ? (x = i.chartData[a]) == null ? void 0 : x.data : []
          },
          i == null ? void 0 : i.barOptions,
          (k = ($ = (v = e.seriesSetting) == null ? void 0 : v.data) == null ? void 0 : $[a]) == null ? void 0 : k.style
        );
        return y({}, D, u);
      }), (R = e.seriesSetting) != null && R.xAxisLabel ? c = (_ = e.data) != null && _.length ? e.data.map((u) => u[e.seriesSetting.xAxisLabel]) : [] : c = e.data.map((u, a) => a));
      let n;
      e.mainAxis === "x" ? n = y(
        {
          series: l,
          xAxis: {
            data: c
          }
        },
        g
      ) : e.mainAxis === "y" && (n = y(
        {
          series: l,
          yAxis: {
            data: c
          }
        },
        g
      )), n = y(n, e.options), e.mainAxis === "y" && (n.xAxis.type = "value", n.yAxis.type = "category"), f.clear();
      try {
        console.log(n, "基础柱状图"), f.setOption(n);
      } catch (u) {
        console.log(u);
      }
    };
    return W(() => {
      z(), s.value && (f = X.init(s.value), C(), O(), f.on("click", (t) => {
        T("com-click", t);
      }), f.on("legendselectchanged", (t) => {
        T("legend-select", t);
      })), setTimeout(() => {
        m();
      }, 500), window.addEventListener("resize", m), B();
    }), H(() => {
      window.removeEventListener("resize", m), h && (h.disconnect(), h = null);
    }), p(
      [
        () => e.bar,
        () => e.mainAxis
      ],
      () => {
        i.barOptions = A(e.bar), O(), m();
      },
      {
        deep: !0
      }
    ), p(() => e.options, () => {
      C(), O();
    }, { deep: !0 }), p(
      [
        () => e.data,
        () => e.seriesSetting
      ],
      () => {
        C(), O(), m();
      },
      {
        deep: !0
      }
    ), p(
      () => e.cStyle,
      () => {
        z(), L(() => {
          f.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      ...K(i),
      chartId: s,
      checkEmpty: G,
      domOnlyClass: b
    };
  }
});
export {
  ne as default
};
