// @ths/design/es/components/layer-toolbox/src/index.vue.js
import f from "./index.vue2.js";
import { resolveComponent as r, openBlock as s, createElementBlock as d, normalizeClass as t, createElementVNode as a, withDirectives as k, createVNode as u, mergeProps as V, withCtx as i, Fragment as p, renderList as m, createBlock as v, toDisplayString as c, createTextVNode as L, vShow as y } from "vue";
import "./index.vue3.js";
import A from "../../../_virtual/_plugin-vue_export-helper.js";
const $ = { class: "title-text" };
function w(e, l, N, B, T, D) {
  const g = r("el-checkbox"), C = r("el-checkbox-group"), h = r("el-collapse-item"), b = r("el-collapse");
  return s(), d("div", {
    class: t(["t-layer-toolbox", e.domOnlyClass])
  }, [
    a("div", {
      class: t(e.openControl ? "active-button" : "control-button"),
      onClick: l[0] || (l[0] = (...o) => e.changeControl && e.changeControl(...o))
    }, l[3] || (l[3] = [
      a("div", { class: "img-control" }, null, -1)
    ]), 2),
    k(u(b, V({
      modelValue: e.titleList,
      "onUpdate:modelValue": l[2] || (l[2] = (o) => e.titleList = o)
    }, e.filterAttrs(e.$attrs), {
      class: "collapse-wrapper",
      onChange: e.changeTitle
    }), {
      default: i(() => [
        (s(!0), d(p, null, m(e.data, (o) => (s(), v(h, {
          key: o[e.alias.value],
          name: o[e.alias.value],
          class: t(["collapse-title", e.handleActive(o[e.alias.children]) ? "isActive" : ""])
        }, {
          title: i(() => [
            l[4] || (l[4] = a("div", { class: "img-title" }, null, -1)),
            a("div", {
              class: t(["title-line", e.handleActive(o[e.alias.children]) ? "active" : ""])
            }, null, 2),
            a("div", $, c(o[e.alias.label]), 1),
            a("div", {
              class: t(["collapse-icon", e.handleCollapseActive(o[e.alias.value]) ? "animation" : ""])
            }, null, 2)
          ]),
          default: i(() => [
            a("div", null, [
              u(C, {
                modelValue: e.checkedList,
                "onUpdate:modelValue": l[1] || (l[1] = (n) => e.checkedList = n),
                onChange: e.changeCheck
              }, {
                default: i(() => [
                  (s(!0), d(p, null, m(o.children, (n) => (s(), v(g, {
                    key: n[e.alias.value],
                    label: n[e.alias.value],
                    disabled: e.checkedList.length === e.maxCheckedNum && !e.checkedList.includes(n[e.alias.value])
                  }, {
                    default: i(() => [
                      L(c(n[e.alias.label]), 1)
                    ]),
                    _: 2
                  }, 1032, ["label", "disabled"]))), 128))
                ]),
                _: 2
              }, 1032, ["modelValue", "onChange"])
            ])
          ]),
          _: 2
        }, 1032, ["name", "class"]))), 128))
      ]),
      _: 1
    }, 16, ["modelValue", "onChange"]), [
      [y, e.openControl]
    ]),
    l[5] || (l[5] = a("div", { class: "style-container" }, null, -1))
  ], 2);
}
const F = /* @__PURE__ */ A(f, [["render", w]]);
export {
  F as default
};
