// @ths/design/es/components/layer-toolbox/src/index.vue2.js
import { defineComponent as it, computed as rt, reactive as vt, onMounted as ut, watch as u, toRefs as nt } from "vue";
import { filterAttrs as ht, getDomRandomClass as ft, setStyle as st, getCssPropertyString as o } from "../../../utils/index.js";
const yt = it({
  name: "TLayerToolbox",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 控制按钮默认样式
        controlButton: {
          default: {},
          hover: {},
          active: {}
        },
        // 控制按钮图标
        imgControl: {
          default: {},
          hover: {},
          active: {}
        },
        // 图层工具容器
        collapseWrapper: {
          default: {},
          hover: {},
          active: {}
        },
        // 图层分类标题
        collapseTitle: {
          default: {},
          hover: {},
          active: {}
        },
        // 标题背景
        imgTitle: {
          default: {},
          hover: {},
          active: {}
        },
        // 折叠icon
        collapseIcon: {
          default: {},
          hover: {},
          active: {}
        },
        // 分类标题页签装饰
        titleLine: {
          default: {},
          hover: {},
          active: {}
        },
        // 分类标题文字
        titleText: {
          default: {},
          hover: {},
          active: {}
        },
        // 单个容器
        itemWrapper: {
          default: {},
          hover: {},
          active: {}
        },
        // 选项文本
        label: {
          default: {},
          hover: {},
          active: {}
        },
        // 矩形
        rectangle: {
          default: {},
          hover: {},
          active: {}
        }
      })
    },
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => ({}),
      require: !1
    },
    checked: {
      type: Array,
      default: () => []
    },
    // 限制最大选中数量
    maxCheckedNum: {
      type: Number,
      default: 5
    }
  },
  emits: ["change-title", "change-control", "change-check"],
  setup(a, { emit: v }) {
    var f;
    const e = rt(() => a.cStyle), l = ft("layer-toolbox"), i = vt({
      titleList: [],
      checkedList: (f = a.checked) != null && f.length ? a.checked : [],
      openControl: !1
    }), lt = (t) => {
      try {
        return !!t.find((c) => i.checkedList.includes(c[a.alias.value]));
      } catch {
        return console.log("数据格式有误"), !1;
      }
    }, ot = (t) => {
      try {
        return !!(i.titleList === t || i.titleList.includes(t));
      } catch {
        return console.log("数据格式有误"), !1;
      }
    }, n = () => {
      i.titleList = [], i.checkedList = a.checked ? a.checked : [];
    }, at = () => {
      var c, r;
      let t = "";
      return (c = e.value) != null && c.wrapper && (t += st((r = e.value) == null ? void 0 : r.wrapper, `.${l}`)), t;
    }, ct = () => {
      var c, r, s, d, p, $, b, x, y, m, g, k, T, C, _, L, W, S, w, B, I, N, A, O, H, j, q, D, J, M, R, P, z, E, F, G, K, Q, U, V, X, Y, Z, ee, te, le, oe, ae, ce, ie, re, ve, ue, ne, he, fe, se, de, pe, $e, be, xe, ye, me, ge, ke, Te, Ce, _e, Le, We, Se, we, Be, Ie, Ne, Ae, Oe, He, je, qe, De, Je, Me, Re, Pe, ze, Ee, Fe, Ge, Ke, Qe, Ue, Ve, Xe, Ye, Ze, et, tt;
      let t = "";
      return (r = (c = e.value) == null ? void 0 : c.controlButton) != null && r.default && (t += `.${l}.t-layer-toolbox .control-button {
          ${o((s = e.value) == null ? void 0 : s.controlButton.default)}
        }`), (p = (d = e.value) == null ? void 0 : d.controlButton) != null && p.hover && (t += `.${l}.t-layer-toolbox .control-button:hover {
          ${o(($ = e.value) == null ? void 0 : $.controlButton.hover)}
        }`), (x = (b = e.value) == null ? void 0 : b.controlButton) != null && x.active && (t += `.${l}.t-layer-toolbox .active-button {
          ${o((y = e.value) == null ? void 0 : y.controlButton.active)}
        }`), (g = (m = e.value) == null ? void 0 : m.imgControl) != null && g.default && (t += `.${l}.t-layer-toolbox .img-control {
          ${o((k = e.value) == null ? void 0 : k.imgControl.default)}
        }`), (C = (T = e.value) == null ? void 0 : T.imgControl) != null && C.hover && (t += `.${l}.t-layer-toolbox .img-control:hover {
          ${o((_ = e.value) == null ? void 0 : _.imgControl.hover)}
        }`), (W = (L = e.value) == null ? void 0 : L.imgControl) != null && W.active && (t += `.${l}.t-layer-toolbox .img-control.active {
          ${o((S = e.value) == null ? void 0 : S.imgControl.active)}
        }`), (B = (w = e.value) == null ? void 0 : w.collapseWrapper) != null && B.default && (t += `.${l}.t-layer-toolbox .collapse-wrapper {
          ${o((I = e.value) == null ? void 0 : I.collapseWrapper.default)}
        }`), (A = (N = e.value) == null ? void 0 : N.collapseWrapper) != null && A.hover && (t += `.${l}.t-layer-toolbox .collapse-wrapper:hover {
          ${o((O = e.value) == null ? void 0 : O.collapseWrapper.hover)}
        }`), (j = (H = e.value) == null ? void 0 : H.collapseWrapper) != null && j.active && (t += `.${l}.t-layer-toolbox .collapse-wrapper.active {
          ${o((q = e.value) == null ? void 0 : q.collapseWrapper.active)}
        }`), (J = (D = e.value) == null ? void 0 : D.collapseTitle) != null && J.default && (t += `.${l}.t-layer-toolbox .collapse-title {
          ${o((M = e.value) == null ? void 0 : M.collapseTitle.default)}
        }`), (P = (R = e.value) == null ? void 0 : R.collapseTitle) != null && P.hover && (t += `.${l}.t-layer-toolbox .collapse-title:hover {
          ${o((z = e.value) == null ? void 0 : z.collapseTitle.hover)}
        }`), (F = (E = e.value) == null ? void 0 : E.collapseTitle) != null && F.active && (t += `.${l}.t-layer-toolbox .collapse-title.active {
          ${o((G = e.value) == null ? void 0 : G.collapseTitle.active)}
        }`), (Q = (K = e.value) == null ? void 0 : K.imgTitle) != null && Q.default && (t += `.${l}.t-layer-toolbox .img-title {
          ${o((U = e.value) == null ? void 0 : U.imgTitle.default)}
        }`), (X = (V = e.value) == null ? void 0 : V.imgTitle) != null && X.hover && (t += `.${l}.t-layer-toolbox .img-title:hover {
          ${o((Y = e.value) == null ? void 0 : Y.imgTitle.hover)}
        }`), (ee = (Z = e.value) == null ? void 0 : Z.imgTitle) != null && ee.active && (t += `.${l}.t-layer-toolbox .img-title.active {
          ${o((te = e.value) == null ? void 0 : te.imgTitle.active)}
        }`), (oe = (le = e.value) == null ? void 0 : le.collapseIcon) != null && oe.default && (t += `.${l}.t-layer-toolbox .collapse-icon {
          ${o((ae = e.value) == null ? void 0 : ae.collapseIcon.default)}
        }`), (ie = (ce = e.value) == null ? void 0 : ce.collapseIcon) != null && ie.hover && (t += `.${l}.t-layer-toolbox .collapse-icon:hover {
          ${o((re = e.value) == null ? void 0 : re.collapseIcon.hover)}
        }`), (ue = (ve = e.value) == null ? void 0 : ve.collapseIcon) != null && ue.active && (t += `.${l}.t-layer-toolbox .collapse-icon.active {
          ${o((ne = e.value) == null ? void 0 : ne.collapseIcon.active)}
        }`), (fe = (he = e.value) == null ? void 0 : he.titleLine) != null && fe.default && (t += `.${l}.t-layer-toolbox .title-line {
          ${o((se = e.value) == null ? void 0 : se.titleLine.default)}
        }`), (pe = (de = e.value) == null ? void 0 : de.titleLine) != null && pe.hover && (t += `.${l}.t-layer-toolbox .title-line:hover {
          ${o(($e = e.value) == null ? void 0 : $e.titleLine.hover)}
        }`), (xe = (be = e.value) == null ? void 0 : be.titleLine) != null && xe.active && (t += `.${l}.t-layer-toolbox .title-line.active {
          ${o((ye = e.value) == null ? void 0 : ye.titleLine.active)}
        }`), (ge = (me = e.value) == null ? void 0 : me.titleText) != null && ge.default && (t += `.${l}.t-layer-toolbox .title-text {
          ${o((ke = e.value) == null ? void 0 : ke.titleText.default)}
        }`), (Ce = (Te = e.value) == null ? void 0 : Te.titleText) != null && Ce.hover && (t += `.${l}.t-layer-toolbox .title-text:hover {
          ${o((_e = e.value) == null ? void 0 : _e.titleText.hover)}
        }`), (We = (Le = e.value) == null ? void 0 : Le.titleText) != null && We.active && (t += `.${l}.t-layer-toolbox .title-text.active {
          ${o((Se = e.value) == null ? void 0 : Se.titleText.active)}
        }`), (Be = (we = e.value) == null ? void 0 : we.label) != null && Be.default && (t += `.${l}.t-layer-toolbox .el-checkbox .el-checkbox__label {
          ${o((Ie = e.value) == null ? void 0 : Ie.label.default)}
        }`), (Ae = (Ne = e.value) == null ? void 0 : Ne.label) != null && Ae.hover && (t += `.${l}.t-layer-toolbox .el-checkbox .el-checkbox__label:hover {
          ${o((Oe = e.value) == null ? void 0 : Oe.label.hover)}
        }`), (je = (He = e.value) == null ? void 0 : He.label) != null && je.active && (t += `.${l}.t-layer-toolbox .el-checkbox__input.is-checked + .el-checkbox__label {
          ${o((qe = e.value) == null ? void 0 : qe.label.active)}
        }`), (Je = (De = e.value) == null ? void 0 : De.itemWrapper) != null && Je.default && (t += `.${l}.t-layer-toolbox .el-checkbox {
          ${o((Me = e.value) == null ? void 0 : Me.itemWrapper.default)}
        }`), (Pe = (Re = e.value) == null ? void 0 : Re.itemWrapper) != null && Pe.hover && (t += `.${l}.t-layer-toolbox .el-checkbox:hover {
          ${o((ze = e.value) == null ? void 0 : ze.itemWrapper.hover)}
        }`), (Fe = (Ee = e.value) == null ? void 0 : Ee.itemWrapper) != null && Fe.active && (t += `.${l}.t-layer-toolbox .el-checkbox.is-checked {
          ${o((Ge = e.value) == null ? void 0 : Ge.itemWrapper.active)}
        }`), (Qe = (Ke = e.value) == null ? void 0 : Ke.rectangle) != null && Qe.default && (t += `.${l}.t-layer-toolbox .el-checkbox__inner {
          ${o((Ue = e.value) == null ? void 0 : Ue.rectangle.default)}
        }`), (Xe = (Ve = e.value) == null ? void 0 : Ve.rectangle) != null && Xe.hover && (t += `.${l}.t-layer-toolbox .el-checkbox:not(.is-checked):hover .el-checkbox__inner {
          ${o((Ye = e.value) == null ? void 0 : Ye.rectangle.hover)}
        }`), (et = (Ze = e.value) == null ? void 0 : Ze.rectangle) != null && et.active && (t += `.${l}.t-layer-toolbox .el-checkbox__input.is-checked .el-checkbox__inner {
          ${o((tt = e.value) == null ? void 0 : tt.rectangle.active)}
        }`), t;
    }, h = () => {
      const t = document.querySelector(`.${l} .style-container`);
      t && (t.innerHTML = `<style>${at()}${ct()}</style>`);
    };
    return ut(() => {
      h(), n();
    }), u(
      () => a.cStyle,
      () => {
        h();
      },
      { deep: !0 }
    ), u(
      () => a.data,
      (t, c) => {
        JSON.stringify(t) !== JSON.stringify(c) && n();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), u(
      () => a.checked,
      () => {
        var t;
        i.checkedList = (t = a.checked) != null && t.length ? a.checked : [];
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      curStyle: e,
      domOnlyClass: l,
      filterAttrs: ht,
      changeTitle: (t) => {
        v("change-title", t);
      },
      changeControl: (t) => {
        i.openControl = !i.openControl, v("change-control", t);
      },
      changeCheck: (t) => {
        t.length > a.maxCheckedNum || v("change-check", t);
      },
      handleActive: lt,
      handleCollapseActive: ot,
      ...nt(i)
    };
  }
});
export {
  yt as default
};
