// @ths/design/es/components/text-common/src/index.vue.js
import l from "./index.vue2.js";
import { openBlock as d, createElementBlock as t, normalizeClass as s, createTextVNode as r, toDisplayString as i, createElementVNode as a } from "vue";
import "./index.vue3.js";
import p from "../../../_virtual/_plugin-vue_export-helper.js";
const u = ["contenteditable", "title"];
function m(e, n, y, b, f, v) {
  return d(), t("div", {
    ref: "textRef",
    class: s(["t-text-common", e.domOnlyClass]),
    contenteditable: e.editableElements.label.editable,
    title: e.showTitle ? e.label : "",
    onDblclick: n[0] || (n[0] = (...o) => e.handleDbclick && e.handleDbclick(...o)),
    onInput: n[1] || (n[1] = (...o) => e.handleInput && e.handleInput(...o)),
    onBlur: n[2] || (n[2] = (...o) => e.handleBlur && e.handleBlur(...o)),
    onMousedown: n[3] || (n[3] = (...o) => e.handleMouseDown && e.handleMouseDown(...o)),
    onMouseenter: n[4] || (n[4] = (o) => e.idHover = !0),
    onMouseleave: n[5] || (n[5] = (o) => e.idHover = !1),
    onKeydown: n[6] || (n[6] = (...o) => e.handleKeydown && e.handleKeydown(...o)),
    onKeyup: n[7] || (n[7] = (...o) => e.handleKeyup && e.handleKeyup(...o)),
    onCopy: n[8] || (n[8] = (...o) => e.handleCopy && e.handleCopy(...o)),
    onPaste: n[9] || (n[9] = (...o) => e.handlePaste && e.handlePaste(...o))
  }, [
    r(i(e.editableElements.label.value) + " ", 1),
    n[10] || (n[10] = a("div", { class: "style-container" }, null, -1))
  ], 42, u);
}
const $ = /* @__PURE__ */ p(l, [["render", m]]);
export {
  $ as default
};
