// @ths/design/es/components/text-common/src/index.vue2.js
import { defineComponent as K, ref as R, reactive as q, computed as d, watch as v, nextTick as s, toRefs as M } from "vue";
import { getDomRandomClass as _, setStyle as A, getCssPropertyString as g } from "../../../utils/index.js";
const L = K({
  name: "TTextCommon",
  props: {
    // 文本内容
    label: {
      type: [String, Number, Boolean],
      default: ""
    },
    // 是否显示标题
    showTitle: {
      type: Boolean,
      default: !1
    },
    // 要修改的字段名
    editKey: {
      type: String,
      default: "label"
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {},
          focus: {
            cursor: "text"
          }
        }
      })
    },
    // 是否显示双击编辑提示 默认显示 例如矩形组件双击编辑文本 失去焦点时不需要显示默认文本
    tip: {
      type: Boolean,
      default: !0
    },
    // 在 body 上设置为不能编辑的情况下 文本是否可以编辑
    editable: {
      type: Boolean,
      default: !1
    },
    // 渐变色
    gradientColor: {
      type: String,
      default: ""
    },
    // 文字阴影样式数据
    shadowStyle: {
      type: Object,
      default: () => ({})
    },
    // 描边
    outline: {
      type: Object,
      default: () => ({
        width: 0,
        color: "transparent"
      })
    }
  },
  // 修改文本事件
  emits: ["modify-text", "blur-text"],
  setup(t, { emit: f }) {
    const b = R(null), l = q({
      // 组件被鼠标滑过状态 用于修改组件滑入滑出样式
      idHover: !1,
      // 可以编辑的元素列表
      editableElements: {
        // 可编辑的变量在props中定义的名称
        label: {
          // 可编辑状态
          editable: !1,
          // 变量值
          value: t.label,
          // 输入框变更时临时存储数据的变量
          template: t.label
        }
      }
    }), a = _("text"), i = d(() => t.shadowStyle), u = d(() => t.gradientColor), o = d(() => t.cStyle);
    v(() => t.label, () => {
      l.editableElements.label.value = t.label, l.editableElements.label.template = t.label;
    });
    const w = d(() => {
      var y, m, h;
      const { hover: e, focus: n } = (y = t.cStyle) == null ? void 0 : y.wrapper, r = { ...(h = (m = t.cStyle) == null ? void 0 : m.wrapper) == null ? void 0 : h.default }, B = { ...r, ...n }, H = { ...r, ...e };
      let c = r;
      return l.idHover && e && Object.keys(e).length && (c = H), l.editableElements.label.editable && n && Object.keys(n).length && (c = B), c;
    });
    function p() {
      const e = {
        [t.editKey]: l.editableElements.label
      };
      f("modify-text", e);
    }
    function S(e) {
      var n;
      l.editableElements.label.template = (n = e == null ? void 0 : e.target) == null ? void 0 : n.innerText;
    }
    function $() {
      f("blur-text"), l.editableElements.label.editable && (l.editableElements.label.editable = !1, l.editableElements.label.value = "", s(() => {
        l.editableElements.label.value = l.editableElements.label.template || t.tip && "双击编辑文本" || "", p();
      }));
    }
    function x(e) {
      l.editableElements.label.editable && e.stopPropagation();
    }
    function C(e) {
      e.stopPropagation();
    }
    function E(e) {
      l.editableElements.label.editable && e.stopPropagation();
    }
    function k(e) {
      e.stopPropagation();
    }
    function T(e) {
      e.preventDefault();
      const n = e.clipboardData, r = (n == null ? void 0 : n.getData("text/plain")) || "";
      r !== "" && document.execCommand("insertText", !1, r);
    }
    function D() {
      const e = document.querySelector("body");
      !(e != null && e.getAttribute("editable")) && !t.editable || (l.editableElements.label.editable = !0, s(() => {
        var n;
        (n = b.value) == null || n.focus();
      }));
    }
    const O = () => {
      let e = "";
      return o.value.wrapper && (e += A(o.value.wrapper, `.${a}`)), i.value.x && (e += `.${a}{
          --t-gradient-shadow-x: ${i.value.x || 0};
        }`), i.value.y && (e += `.${a}{
          --t-gradient-shadow-y: ${i.value.y || 0};
        }`), i.value.blur && (e += `.${a}{
          --t-gradient-shadow-blur: ${i.value.blur || 0};
        }`), i.value.color && (e += `.${a}{
          --t-gradient-shadow-color: ${i.value.color || "transparent"};
        }`), u.value && (u.value.includes("linear-gradient") || u.value.includes("radial-gradient") ? e += `.${a}{
          --t-gradient-text-color: ${u.value};
           background-clip: text;
           color: transparent;
        }` : e += `.${a}{
          color: ${u.value};
        }`), t.outline.width && (e += `.${a}{
          --t-text-stroke-width: ${t.outline.width || 0};
        }`), t.outline.color && (e += `.${a}{
          --t-text-stroke-color: ${t.outline.color};
        }`), e;
    }, P = () => {
      let e = "";
      return o.value.wrapper && o.value.wrapper.hover && (e += `
          .${a}:hover {
            ${g(o.value.wrapper.hover)}
          }
        `), o.value.wrapper && o.value.wrapper.focus && (e += `
          .${a}:focus {
            ${g(o.value.wrapper.focus)}
          }
        `), e;
    }, j = () => {
      const e = document.querySelector(`.${a} .style-container`);
      e && (e.innerHTML = `<style>${O()}${P()}}</style>`);
    };
    return v(() => [t.cStyle, t.shadowStyle, t.gradientColor, t.outline], () => {
      s(() => {
        j();
      });
    }, {
      deep: !0,
      immediate: !0
    }), {
      textRef: b,
      customStyle: w,
      domOnlyClass: a,
      curStyle: o,
      ...M(l),
      handleInput: S,
      handleBlur: $,
      handleMouseDown: x,
      handleKeydown: C,
      handleKeyup: E,
      handleCopy: k,
      handlePaste: T,
      handleDbclick: D
    };
  }
});
export {
  L as default
};
