// @ths/design/es/components/table/src/index.vue2.js
import { defineComponent as Z, computed as x, ref as C, h as ee, watch as h, nextTick as B, onMounted as te, onUnmounted as le } from "vue";
import ae from "lodash/merge";
import oe from "lodash/cloneDeep";
import { filterAttrs as re, getDomRandomClass as ne, getCssPropertyString as g, setStyle as u } from "../../../utils/index.js";
const I = 10, me = Z({
  name: "TTable",
  props: {
    // 表格数据
    data: {
      type: Array,
      default: () => []
    },
    // 高度
    height: {
      type: [Number, String],
      default: () => null
    },
    // 列属性
    columns: {
      type: Array,
      default: () => []
    },
    // 开启tooltip 之后，tooltip 的属性设置 表格级别优先级小于列级别
    tooltipProps: {
      type: Object,
      default: () => {
      }
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        headerRowWrapper: {
          default: {}
        },
        headerCell: {
          default: {}
        },
        headerSortContainer: {
          default: {}
        },
        headerSortTop: {
          default: {
            borderBottomColor: "#E0EAFF"
          },
          active: {
            borderBottomColor: "#00FFF7"
          }
        },
        headerSortBottom: {
          default: {
            borderTopColor: "#E0EAFF"
          },
          active: {
            borderTopColor: "#00FFF7"
          }
        },
        rowWrapper: {
          default: {},
          hover: {}
        },
        bodyCell: {
          default: {},
          hover: {}
        },
        bodyCellInner: {
          default: {}
        },
        stripeBodyCell: {
          default: {},
          hover: {}
        },
        stripeBodyRow: {
          default: {},
          hover: {}
        },
        // 高亮行
        highlightRow: {
          default: {}
        },
        // 描述icon icon 容器
        infoCellIcon: {
          default: {},
          hover: {}
        },
        // desc 文本容器
        tooltipInfo: {
          default: {}
        },
        // desc 文本容器子元素
        tooltipInfoInner: {
          default: {}
        },
        // 表格tooltip 样式
        tableTooltip: {
          default: {}
        },
        // 表格tooltip 箭头
        tableTooltipArrow: {
          default: {}
        },
        // 表尾行
        footerRowWrapper: {
          default: {}
        },
        // 表尾单元格
        footerCell: {
          default: {}
        },
        // 合计容器
        totalWrapper: {
          default: {}
        },
        // 暂无数据容器
        emptyWrapper: {
          default: {}
        },
        // 暂无数据文本
        emptyText: {
          default: {}
        }
      })
    },
    // 开启自动滚动
    autoScroll: {
      type: String,
      default: ""
    },
    // 滚动速度 数字越大越快
    speed: {
      type: Number,
      default: () => 3
    },
    showAnimation: {
      type: Boolean,
      default: !1
    },
    animationDelay: {
      type: Number,
      default: 0.1
    },
    animationType: {
      type: String,
      default: ""
    },
    maxHeight: {
      type: Number,
      default: null
    },
    // 是否支持树形数据
    showTreeData: {
      type: Object,
      default: () => ({
        open: !1,
        idKey: "",
        parentIdKey: ""
      })
    },
    // 合计数据
    total: {
      type: Array,
      default: () => []
    },
    // 合计文字
    sumText: {
      type: String,
      default: "合计"
    },
    // 合计行合并列数
    sumMergeCol: {
      type: Number,
      default: 1
    }
  },
  emits: ["operation-click"],
  setup(a, { emit: k, attrs: ie }) {
    const t = x(() => a.cStyle), i = ne("table"), R = x(() => a.columns), v = C(), E = C(!1), _ = C(null);
    let w = null;
    const S = C(""), d = C(null), H = (e, l) => k("operation-click", e, l), F = (e, l) => {
      let o;
      return S.value === l ? o = e.hoverBackgroundImage || e["background-image"] || e.backgroundImage || "" : o = e["background-image"] || e.backgroundImage || "", o ? {
        ...e,
        "background-image": o.indexOf("url") === -1 ? `url(${o})` : o
      } : e;
    }, K = (e, l, o) => {
      let r = o;
      return (e === "circle" && o.includes("gradient") || o === "") && (l === "backgroundColor" ? r = "rgb(235, 238, 245)" : l === "color" && (r = "rgb(46,124,246)")), r;
    }, L = ({ rowIndex: e }) => {
      let l = `rowIndex${e}`;
      return a.showAnimation && (l += ` ${a.animationType} animate__animated`), l;
    }, M = ({ rowIndex: e }) => ({
      "animation-delay": `${e * a.animationDelay}s`
    }), O = ({ row: e, columnIndex: l }) => {
      var r, c, n;
      const o = R.value[l];
      if (o.type === "state" && o.stateList) {
        const s = o.stateKey ? e[o.stateKey] : "", f = o.stateList.find((m) => m.name === s);
        return g(f == null ? void 0 : f.cellStyle);
      }
      return `${g(((r = o.style) == null ? void 0 : r.cellStyle) || {})}; color: ${(n = (c = o.style) == null ? void 0 : c.fontStyle) == null ? void 0 : n.color}`;
    }, P = () => {
      var l, o, r, c, n, s;
      let e = "";
      return t.value.bodyCell && t.value.bodyCell.hover && (e += `
          .${i} .el-table .el-table__body tr:hover>td,
          .${i} .el-table .el-table__body tr:hover>td.el-table__cell {
            ${g(t.value.bodyCell.hover)}
            color: ${(l = t.value.bodyCell.hover) == null ? void 0 : l.color} !important;
            font-size: ${(o = t.value.bodyCell.hover) == null ? void 0 : o.fontSize} !important;
          }
        `), (r = t.value.headerSortTop) != null && r.active && (e += `.${i} .el-table .el-table__header .ascending .cell .caret-wrapper .sort-caret.ascending {
          ${g((c = t.value.headerSortTop) == null ? void 0 : c.active)}
        }`), (n = t.value.headerSortBottom) != null && n.active && (e += `.${i} .el-table .el-table__header .descending .cell .caret-wrapper .sort-caret.descending {
          ${g((s = t.value.headerSortBottom) == null ? void 0 : s.active)}
        }`), e;
    }, z = () => {
      var l, o;
      let e = "";
      return t.value.wrapper && (e += u(t.value.wrapper, `.${i}.t-table`)), t.value.headerRowWrapper && (e += u(t.value.headerRowWrapper, `.${i}.t-table .el-table__header tr`)), t.value.headerCell && (e += u(t.value.headerCell, `.${i} .el-table .el-table__header th`)), t.value.headerSortContainer && (e += u(t.value.headerSortContainer, `.${i} .el-table .el-table__header th .cell .caret-wrapper`)), t.value.headerSortTop && (e += u(t.value.headerSortTop, `.${i} .el-table .el-table__header th .cell .caret-wrapper .sort-caret.ascending`)), t.value.headerSortBottom && (e += u(t.value.headerSortBottom, `.${i} .el-table .el-table__header th .cell .caret-wrapper .sort-caret.descending`)), t.value.rowWrapper && (e += u(t.value.rowWrapper, `.${i} .el-table .el-table__body tr`)), t.value.bodyCell && (e += u({ default: t.value.bodyCell.default }, `.${i} .el-table .el-table__body tr>td`)), t.value.bodyCellInner && (e += u({ default: t.value.bodyCellInner.default }, `.${i} .el-table .el-table__body tr>td .cell`)), t.value.stripeBodyCell && (e += u(t.value.stripeBodyCell, `.${i} .el-table--striped .el-table__body tr.el-table__row--striped td`)), t.value.stripeBodyRow && (e += u(t.value.stripeBodyRow, `.${i} .el-table--striped .el-table__body tr.el-table__row--striped`)), t.value.footerRowWrapper && (e += u(t.value.footerRowWrapper, `.${i}.t-table .el-table__footer-wrapper`)), t.value.footerCell && (e += u(t.value.footerCell, `.${i}.t-table .el-table__footer td`)), t.value.totalWrapper && (e += u(t.value.totalWrapper, `.${i}.t-table .el-table__footer td:first-child`)), t.value.emptyWrapper && (e += u(t.value.emptyWrapper, `.${i}.t-table .el-table__empty-block`)), t.value.emptyText && (e += u(t.value.emptyText, `.${i}.t-table .el-table__empty-text`)), t.value.highlightRow && (e += u(t.value.highlightRow, `.${i} .el-table .el-table__body tr.current-row > td`), e += u(t.value.highlightRow, `.${i} .el-table .el-table__body tr.current-row > td.el-table__cell`)), t.value.infoCellIcon && (e += u({ default: t.value.infoCellIcon.default }, `.${i} .info-cell-icon`)), (l = t.value.infoCellIcon) != null && l.hover && (e += `.${i} .info-cell-icon:hover {
          ${g((o = t.value.infoCellIcon) == null ? void 0 : o.hover)}
        }`), t.value.tooltipInfo && (e += u({ default: t.value.tooltipInfo.default }, ".t-tooltip-text")), t.value.tooltipInfoInner && (e += u({ default: t.value.tooltipInfoInner.default }, ".t-tooltip-text-inner")), t.value.tableTooltip && (e += u({ default: t.value.tableTooltip.default }, ".t-table-tooltip.el-popper")), t.value.tableTooltipArrow && (e += u({ default: t.value.tableTooltipArrow.default }, ".t-table-tooltip.el-popper>.el-popper__arrow::before")), e;
    }, D = () => {
      const e = document.querySelector(`.${i} .style-container`);
      e && (e.innerHTML = `<style>${z()}${P()}</style>`);
    }, A = (e) => {
      const l = e.$refs.bodyWrapper;
      w = setInterval(() => {
        l.scrollTop += 1.5, l.clientHeight + l.scrollTop === l.scrollHeight && (l.scrollTop = 0);
      }, I * (I / (a.speed || 3)));
    }, N = () => {
      const e = _.value, l = e.$refs.bodyWrapper.getElementsByClassName("el-table__body")[0];
      A(e), l.onmouseenter = () => {
        clearInterval(w);
      }, l.onmouseleave = () => {
        clearInterval(w), A(e);
      };
    };
    let y, p;
    const W = (e) => {
      var c, n;
      const o = _.value.$refs.bodyWrapper, r = Array.from(((n = (c = e == null ? void 0 : e.children) == null ? void 0 : c[1]) == null ? void 0 : n.children) ?? []) ?? [];
      if (r.length) {
        const s = (r == null ? void 0 : r[0]) ?? null, f = (s == null ? void 0 : s.getBoundingClientRect().height) ?? 0;
        y = setInterval(() => j(o, f), I * (I / (a.speed || 3)));
      }
    }, T = () => {
      const l = _.value.$refs.bodyWrapper.getElementsByClassName("el-table__body")[0];
      l.onmouseenter = () => {
        clearInterval(y), clearTimeout(p);
      }, l.onmouseleave = () => {
        clearInterval(y), clearTimeout(p), p = setTimeout(() => {
          W(l);
        }, 1e3);
      }, p = setTimeout(() => {
        W(l);
      }, 1e3);
    };
    let b = 0;
    const j = (e, l) => {
      e.scrollTop += 1.3, e.scrollTop / (b + 1) >= l ? (b++, clearInterval(y), v.value.push(v.value.slice(b - 1, b)[0]), b > 1 && (v.value.shift(), v.value.shift(), e.scrollTop = 0, b = 0), T()) : e.scrollTop + e.clientHeight === e.scrollHeight && (clearInterval(y), e.scrollTop = 0, b = 0, T());
    }, q = (e, l, o) => {
      var n;
      const c = ((n = o.parentElement) == null ? void 0 : n.classList.value).match(/rowIndex(\d*)/)[1];
      S.value = Number(c);
    }, G = () => {
      S.value = "";
    }, $ = () => window.pageYOffset || document.documentElement.scrollTop || document.body.scrollTop || 0, Y = (e, l, o) => {
      d.value.firstChild.innerHTML = e || "", d.value.style.display = "block";
      const r = o.srcElement.getBoundingClientRect().x, c = o.srcElement.getBoundingClientRect().y, n = d.value.getBoundingClientRect().width, s = d.value.getBoundingClientRect().height, f = o.srcElement.clientWidth, m = o.srcElement.clientHeight;
      switch (l || "top") {
        case "top":
          d.value.style.left = `${r + f / 2 - n / 2}px`, d.value.style.top = `${c - s - 4 + $()}px`;
          break;
        case "bottom":
          d.value.style.left = `${r + f / 2 - n / 2}px`, d.value.style.top = `${c + m + 4 + $()}px`;
          break;
        case "left":
          d.value.style.left = `${r - 4 - n}px`, d.value.style.top = `${c + m / 2 - s / 2 + $()}px`;
          break;
        case "right":
          d.value.style.left = `${r + 4 + f}px`, d.value.style.top = `${c + m / 2 - s / 2 + $()}px`;
          break;
      }
    }, U = () => {
      d.value.firstChild.innerHTML = "", d.value.style.display = "none";
    }, X = (e, l, o) => {
      const r = [];
      if (!Array.isArray(e))
        return r;
      const c = {};
      return e.forEach((n) => {
        n[l] && (c[n[l]] = n);
      }), e.forEach((n) => {
        const s = c[n[o]];
        s ? (s.children || (s.children = [])).push(n) : r.push(n);
      }), r;
    }, J = x(() => (e) => {
      const { columns: l, data: o } = e;
      let r = [];
      return l.forEach((c, n) => {
        if (n === 0) {
          r[n] = ee("div", {}, [
            a.sumText
          ]);
          return;
        }
        const s = o.map((f) => Number(f[c.property]));
        s.every((f) => Number.isNaN(f)) ? r[n] = "" : r[n] = `${s.reduce((f, m) => {
          const V = Number(m);
          return Number.isNaN(V) ? f : f + m;
        }, 0)}`;
      }), a.total.length && (r = [...a.total]), r;
    }), Q = () => {
      const e = document.querySelectorAll(`.${i} .el-table .el-table__footer td`);
      if (e.length) {
        e[0].colSpan = a.sumMergeCol;
        for (let l = 1; l < a.sumMergeCol; l++)
          e[l].style.display = "none";
      }
    };
    return h([() => a.total, () => a.sumMergeCol], () => {
      B(() => {
        Q();
      });
    }, { immediate: !0 }), h(() => a.data, () => {
      v.value = a.data;
    }, { immediate: !0, deep: !0 }), h(() => a.showTreeData, () => {
      a.showTreeData.open && a.showTreeData.idKey && a.showTreeData.parentIdKey && (v.value = X(a.data, a.showTreeData.idKey, a.showTreeData.parentIdKey));
    }, { immediate: !0, deep: !0 }), h(
      () => a.columns,
      () => {
        let e = !1;
        if (a.columns.length)
          for (let l = 0; l < a.columns.length; l++)
            a.columns[l].tooltipKey && (e = !0);
        if (e && !document.querySelector(".tooltip")) {
          d.value = document.createElement("div"), d.value.className = "t-tooltip-text", d.value.style.display = "none";
          const o = document.createElement("div");
          o.className = "t-tooltip-text-inner", d.value.appendChild(o), document.body.appendChild(d.value);
        }
      },
      { immediate: !0, deep: !0 }
    ), h(
      () => a == null ? void 0 : a.defaultSort,
      () => {
        E.value = !E.value;
      },
      { deep: !0 }
    ), h([() => a.cStyle, a.animationType, a.animationDelay, a.showAnimation], () => {
      D();
    }), h(
      () => [a.autoScroll, a.data, a.speed, a.animationType, a.animationDelay, a.showAnimation],
      () => {
        var e;
        clearInterval(w), clearTimeout(p), clearInterval(y), a.autoScroll === "bodyScroll" && ((e = a.data) != null && e.length) && N(), a.autoScroll === "rowScroll" && B(() => {
          T();
        });
      },
      {
        deep: !0
      }
    ), te(() => {
      var e;
      D(), a.autoScroll === "bodyScroll" && ((e = a.data) != null && e.length) && N(), a.autoScroll === "rowScroll" && B(() => {
        T();
      });
    }), le(() => {
      clearInterval(w), clearTimeout(p), clearInterval(y);
    }), {
      tableScroll: _,
      refreshTable: E,
      curStyle: t,
      domOnlyClass: i,
      curColumns: R,
      curData: v,
      handleOperationClick: H,
      handleCellStyle: O,
      filterAttrs: re,
      formatImageStyle: F,
      merge: ae,
      cloneDeep: oe,
      rowClass: L,
      rowStyle: M,
      hoverData: q,
      hoverLeaveData: G,
      hoverRowIndex: S,
      mouseEnterInfoCell: Y,
      mouseLeaveInfoCell: U,
      handleProgressStyle: K,
      summaryData: J,
      normalizeImages: (e) => {
        if (!e)
          return [];
        if (Array.isArray(e))
          return e.filter(Boolean);
        const l = String(e).trim();
        return l ? l.split(",").map((o) => o.trim()).filter(Boolean) : [];
      }
    };
  }
});
export {
  me as default
};
