// @ths/design/es/components/table/src/index.vue.js
import Ke from "./index.vue2.js";
import { resolveComponent as K, openBlock as s, createElementBlock as t, normalizeClass as c, createBlock as D, mergeProps as L, withCtx as h, Fragment as g, renderList as M, createElementVNode as i, normalizeStyle as n, createCommentVNode as y, toDisplayString as p, withModifiers as he, createVNode as De } from "vue";
import "./index.vue3.js";
import Te from "../../../_virtual/_plugin-vue_export-helper.js";
const Me = ["innerHTML"], Le = ["onMouseenter"], $e = ["title"], Oe = ["onClick"], me = ["onClick"], Be = ["title"], Ee = ["title"], ze = ["title"], Ve = ["onMouseenter"], He = ["title"];
function Ne(l, w, Ae, We, Fe, je) {
  const $ = K("el-tooltip"), be = K("el-image"), ve = K("el-progress"), Ce = K("el-table-column"), Pe = K("el-table");
  return s(), t("div", {
    class: c(["t-table", l.domOnlyClass])
  }, [
    (s(), D(Pe, L({ ref: "tableScroll" }, l.filterAttrs(l.$attrs), {
      key: l.refreshTable,
      style: { width: "100%" },
      border: !1,
      data: l.curData,
      height: l.height,
      "max-height": l.maxHeight,
      "cell-style": l.handleCellStyle,
      class: { "hidden-scroll-y": l.autoScroll !== "" },
      "row-class-name": l.rowClass,
      "row-style": l.rowStyle,
      "summary-method": l.summaryData,
      onCellMouseEnter: l.hoverData,
      onCellMouseLeave: l.hoverLeaveData
    }), {
      default: h(() => [
        (s(!0), t(g, null, M(l.curColumns, (e, Ie) => (s(), D(Ce, {
          key: Ie,
          label: e.label,
          prop: e.key,
          width: e.width,
          "min-width": e.minWidth,
          "header-align": e.headerAlign || "center",
          sortable: e.sortable || !1,
          fixed: e.fixed || !1
        }, {
          header: h(() => [
            i("span", {
              innerHTML: e.label
            }, null, 8, Me)
          ]),
          default: h((r) => {
            var O, m, B, E, z, V, H, N, A, W, F, j, q, G, J, Q, R, U, X, Y, Z, _, x, ee, le, oe, re, se, te, ne, ae, ye, ge, ie, pe, fe;
            return [
              i("div", {
                style: n({ "text-align": e.align || "center" })
              }, [
                (e.infoKey || e.tooltipKey) && (!e.infoPosition || e.infoPosition === "left") ? (s(), t(g, { key: 0 }, [
                  (m = (O = e.infoStyle) == null ? void 0 : O.imageStyle) != null && m["background-image"] || (E = (B = e.infoStyle) == null ? void 0 : B.imageStyle) != null && E.backgroundImage ? (s(), t("span", {
                    key: 0,
                    class: "cell-icon",
                    style: n(l.formatImageStyle((z = e.infoStyle) == null ? void 0 : z.imageStyle))
                  }, null, 4)) : y("", !0),
                  e.tooltipKey ? (s(), t("span", {
                    key: 1,
                    class: "info-cell-icon",
                    onMouseenter: (o) => {
                      var a;
                      return l.mouseEnterInfoCell((a = r.row[e.tooltipKey]) == null ? void 0 : a.toString(), e.tooltipPosition, o);
                    },
                    onMouseleave: w[0] || (w[0] = (o) => l.mouseLeaveInfoCell(o))
                  }, null, 40, Le)) : y("", !0),
                  r.row[e.infoKey] ? (s(), t("span", {
                    key: 2,
                    class: c(["cell-text", { "overflow-ellipsis": e.showOverflowTooltip || !1 }]),
                    title: (V = r.row[e.infoKey]) == null ? void 0 : V.toString(),
                    style: n((H = e.infoStyle) == null ? void 0 : H.fontStyle)
                  }, p((N = r.row[e.infoKey]) == null ? void 0 : N.toString()), 15, $e)) : y("", !0)
                ], 64)) : y("", !0),
                e.type === "operation" && e.operationList ? (s(!0), t(g, { key: 1 }, M(e.operationList, (o, a) => {
                  var f, u, d, S, k, b, v, C, P, I;
                  return s(), t(g, { key: a }, [
                    (u = (f = r.row[e.key]) == null ? void 0 : f.toString().split(",")) != null && u.includes(o.id) && ((o == null ? void 0 : o.visible) ?? !0) ? (s(), t(g, { key: 0 }, [
                      (d = o.imageStyle) != null && d["background-image"] || (S = o.imageStyle) != null && S.backgroundImage || (b = (k = e.style) == null ? void 0 : k.imageStyle) != null && b["background-image"] || (C = (v = e.style) == null ? void 0 : v.imageStyle) != null && C.backgroundImage ? (s(), t("span", {
                        key: 0,
                        class: "cell-icon",
                        style: n(l.formatImageStyle(l.merge(l.cloneDeep(((P = e.style) == null ? void 0 : P.imageStyle) || {}), l.cloneDeep(o == null ? void 0 : o.imageStyle)), r.$index)),
                        onClick: he((T) => l.handleOperationClick(r.row, o.id), ["prevent"])
                      }, null, 12, Oe)) : y("", !0),
                      o.name ? (s(), t("span", {
                        key: 1,
                        class: c(["cell-text cell-btn", { "overflow-ellipsis": e.showOverflowTooltip || !1 }]),
                        style: n(l.merge(l.cloneDeep(((I = e.style) == null ? void 0 : I.fontStyle) || {}), o.fontStyle)),
                        onClick: he((T) => l.handleOperationClick(r.row, o.id), ["prevent"])
                      }, p(o.name), 15, me)) : y("", !0)
                    ], 64)) : y("", !0)
                  ], 64);
                }), 128)) : e.type === "state" && e.stateList ? (s(!0), t(g, { key: 2 }, M(e.stateList, (o, a) => {
                  var f, u, d, S, k, b, v, C, P, I, T, ue, de, Se, ke, ce;
                  return s(), t(g, { key: a }, [
                    e.stateKey && r.row[e.stateKey] === o.name && !e.showStateTooltip ? (s(), t(g, { key: 0 }, [
                      (f = o.imageStyle) != null && f["background-image"] || (u = o.imageStyle) != null && u.backgroundImage || (S = (d = e.style) == null ? void 0 : d.imageStyle) != null && S["background-image"] || (b = (k = e.style) == null ? void 0 : k.imageStyle) != null && b.backgroundImage ? (s(), t("span", {
                        key: 0,
                        class: "cell-icon",
                        style: n(l.formatImageStyle(l.merge(l.cloneDeep(((v = e.style) == null ? void 0 : v.imageStyle) || {}), l.cloneDeep(o.imageStyle)), r.$index))
                      }, null, 4)) : y("", !0),
                      i("span", {
                        class: c(["cell-text cell-btn state-cell", { "overflow-ellipsis": e.showOverflowTooltip || !1 }]),
                        title: (C = r.row[e.key]) == null ? void 0 : C.toString(),
                        style: n(l.formatImageStyle(l.merge(l.cloneDeep(((P = e.style) == null ? void 0 : P.fontStyle) || {}), l.cloneDeep(o.fontStyle))))
                      }, p(r.row[e.key]), 15, Be)
                    ], 64)) : y("", !0),
                    e.stateKey && r.row[e.stateKey] === o.name && e.showStateTooltip ? (s(), t(g, { key: 1 }, [
                      (I = o.imageStyle) != null && I["background-image"] || (T = o.imageStyle) != null && T.backgroundImage || (de = (ue = e.style) == null ? void 0 : ue.imageStyle) != null && de["background-image"] || (ke = (Se = e.style) == null ? void 0 : Se.imageStyle) != null && ke.backgroundImage ? (s(), t("span", {
                        key: 0,
                        class: "cell-icon",
                        style: n(l.formatImageStyle(l.merge(l.cloneDeep(((ce = e.style) == null ? void 0 : ce.imageStyle) || {}), l.cloneDeep(o.imageStyle)), r.$index))
                      }, null, 4)) : y("", !0),
                      De($, L({
                        effect: "dark",
                        "popper-class": "t-table-tooltip",
                        content: r.row[e.stateTooltipKey] || o.name,
                        placement: "bottom-start",
                        ref_for: !0
                      }, { ...l.tooltipProps, ...e.tooltipProps }), {
                        default: h(() => {
                          var we;
                          return [
                            i("span", {
                              class: c(["cell-text cell-btn state-cell", { "overflow-ellipsis": e.showOverflowTooltip || !1 }]),
                              style: n(l.formatImageStyle(l.merge(l.cloneDeep(((we = e.style) == null ? void 0 : we.fontStyle) || {}), l.cloneDeep(o.fontStyle))))
                            }, p(r.row[e.key]), 7)
                          ];
                        }),
                        _: 2
                      }, 1040, ["content"])
                    ], 64)) : y("", !0)
                  ], 64);
                }), 128)) : e.type === "image" ? (s(!0), t(g, { key: 3 }, M(l.normalizeImages(r.row[e.key]), (o, a) => (s(), D(be, {
                  key: a,
                  class: "cell-image",
                  src: o,
                  style: n({ ...e == null ? void 0 : e.imageStyle }),
                  "preview-src-list": l.normalizeImages(r.row[e.key]),
                  "initial-index": a,
                  "preview-teleported": !0,
                  fit: "cover"
                }, null, 8, ["src", "style", "preview-src-list", "initial-index"]))), 128)) : e.type === "color" ? (s(), t("div", {
                  key: 4,
                  class: "cell-color",
                  title: (A = r.row[e.key]) == null ? void 0 : A.toString(),
                  style: n({
                    ...e == null ? void 0 : e.colorStyle,
                    borderColor: (W = r.row[e.key]) == null ? void 0 : W.toString(),
                    backgroundColor: `${e.fillColorBlock ? (F = r.row[e.key]) == null ? void 0 : F.toString() : ""}`
                  })
                }, null, 12, Ee)) : e.type === "progress" ? (s(), t(g, { key: 5 }, [
                  e.type === "progress" ? (s(), D(ve, {
                    key: 0,
                    class: "table-progress",
                    type: (j = e.progressProps) == null ? void 0 : j.type,
                    percentage: (q = r.row[e.key]) == null ? void 0 : q.toString(),
                    color: (G = e.progressProps) == null ? void 0 : G.color,
                    "stroke-width": (J = e.progressProps) == null ? void 0 : J.strokeWidth,
                    style: n([
                      { "--t-table-progress-outer-bg": l.handleProgressStyle((Q = e.progressProps) == null ? void 0 : Q.type, "backgroundColor", ((R = e.progressProps) == null ? void 0 : R.backgroundColor) || "") },
                      { "--t-table-progress-inner-bg": l.handleProgressStyle((U = e.progressProps) == null ? void 0 : U.type, "color", ((X = e.progressProps) == null ? void 0 : X.color) || "") },
                      (Y = e.progressProps) == null ? void 0 : Y.progressBar
                    ])
                  }, {
                    default: h(() => {
                      var o, a, f, u, d, S, k;
                      return [
                        i("div", {
                          class: "progress_font",
                          style: n((o = e.progressProps) == null ? void 0 : o.progressBarText)
                        }, [
                          i("span", {
                            class: "text",
                            style: n(l.merge(l.cloneDeep(((a = e.style) == null ? void 0 : a.fontStyle) || {}), (f = e.progressProps) == null ? void 0 : f.textStyle))
                          }, p((u = r.row[e.key]) == null ? void 0 : u.toString()), 5),
                          i("span", {
                            class: "unit",
                            style: n(l.merge(l.cloneDeep(((d = e.style) == null ? void 0 : d.fontStyle) || {}), (S = e.progressProps) == null ? void 0 : S.unitStyle))
                          }, p(((k = e.progressProps) == null ? void 0 : k.unit) || "%"), 5)
                        ], 4)
                      ];
                    }),
                    _: 2
                  }, 1032, ["type", "percentage", "color", "stroke-width", "style"])) : y("", !0)
                ], 64)) : (s(), t(g, { key: 6 }, [
                  (_ = (Z = e.style) == null ? void 0 : Z.imageStyle) != null && _["background-image"] || (ee = (x = e.style) == null ? void 0 : x.imageStyle) != null && ee.backgroundImage ? (s(), t("span", {
                    key: 0,
                    class: "cell-icon",
                    style: n(l.formatImageStyle((le = e.style) == null ? void 0 : le.imageStyle, r.$index))
                  }, null, 4)) : y("", !0),
                  e.showTooltip ? (s(), D($, L({
                    key: 1,
                    effect: "dark",
                    "popper-class": "t-table-tooltip",
                    content: r.row[e.tooltipContentKey],
                    placement: "bottom-start",
                    ref_for: !0
                  }, { ...l.tooltipProps, ...e.tooltipProps }), {
                    default: h(() => {
                      var o, a;
                      return [
                        i("span", {
                          class: c(["cell-text", { "overflow-ellipsis": e.showOverflowTooltip || !1 }]),
                          style: n((o = e.style) == null ? void 0 : o.fontStyle)
                        }, p((a = r.row[e.key]) == null ? void 0 : a.toString()), 7)
                      ];
                    }),
                    _: 2
                  }, 1040, ["content"])) : (s(), t("span", {
                    key: 2,
                    class: c(["cell-text", { "overflow-ellipsis": e.showOverflowTooltip || !1 }]),
                    title: (oe = r.row[e.key]) == null ? void 0 : oe.toString(),
                    style: n((re = e.style) == null ? void 0 : re.fontStyle)
                  }, p((se = r.row[e.key]) == null ? void 0 : se.toString()), 15, ze))
                ], 64)),
                (e.infoKey || e.tooltipKey) && e.infoPosition === "right" ? (s(), t(g, { key: 7 }, [
                  ((ne = (te = e.infoStyle) == null ? void 0 : te.imageStyle) != null && ne["background-image"] || (ye = (ae = e.infoStyle) == null ? void 0 : ae.imageStyle) != null && ye.backgroundImage) && r.row[e == null ? void 0 : e.infoIconVisibleKey] !== !1 ? (s(), t("span", {
                    key: 0,
                    class: "cell-icon",
                    style: n(l.formatImageStyle((ge = e.infoStyle) == null ? void 0 : ge.imageStyle))
                  }, null, 4)) : y("", !0),
                  e.tooltipKey ? (s(), t("span", {
                    key: 1,
                    class: "info-cell-icon",
                    onMouseenter: (o) => {
                      var a;
                      return l.mouseEnterInfoCell((a = r.row[e.tooltipKey]) == null ? void 0 : a.toString(), e.tooltipPosition, o);
                    },
                    onMouseleave: w[1] || (w[1] = (o) => l.mouseLeaveInfoCell(o))
                  }, null, 40, Ve)) : y("", !0),
                  r.row[e.infoKey] ? (s(), t("span", {
                    key: 2,
                    class: c(["cell-text", { "overflow-ellipsis": e.showOverflowTooltip || !1 }]),
                    title: (ie = r.row[e.infoKey]) == null ? void 0 : ie.toString(),
                    style: n((pe = e.infoStyle) == null ? void 0 : pe.fontStyle)
                  }, p((fe = r.row[e.infoKey]) == null ? void 0 : fe.toString()), 15, He)) : y("", !0)
                ], 64)) : y("", !0)
              ], 4)
            ];
          }),
          _: 2
        }, 1032, ["label", "prop", "width", "min-width", "header-align", "sortable", "fixed"]))), 128))
      ]),
      _: 1
    }, 16, ["data", "height", "max-height", "cell-style", "class", "row-class-name", "row-style", "summary-method", "onCellMouseEnter", "onCellMouseLeave"])),
    w[2] || (w[2] = i("div", { class: "style-container" }, null, -1))
  ], 2);
}
const Re = /* @__PURE__ */ Te(Ke, [["render", Ne]]);
export {
  Re as default
};
