// @ths/design/es/components/list-card-weather-forecast/src/index.vue.js
import ue from "./index.vue2.js";
import { openBlock as t, createElementBlock as u, normalizeClass as le, normalizeStyle as r, Fragment as s, renderList as te, createElementVNode as l, toDisplayString as n, createTextVNode as p, createCommentVNode as d } from "vue";
import "./index.vue3.js";
import ne from "../../../_virtual/_plugin-vue_export-helper.js";
const se = ["onMouseenter"], oe = ["src"], pe = { class: "temperature__max temperature__label" }, de = { class: "temperature__min temperature__label" };
function ge(e, g, ye, ie, fe, he) {
  var y, i;
  return t(), u("ul", {
    class: le(["t-list-card-weather-forecast", e.domOnlyClass]),
    style: r({ ...(i = (y = e.curStyle) == null ? void 0 : y.wrapper) == null ? void 0 : i.default, columnGap: `${e.gap}px` })
  }, [
    (t(!0), u(s, null, te(e.list, (a, f) => {
      var h, V, m, S, v, I, L, c, A, C, O, b, q, B, D, R, $, k, M, U, E, P, T, Q, w, F, H, W, N, z, G, j, J, K, X, Y, Z, _, x;
      return t(), u("li", {
        key: a.date,
        class: "card",
        style: r(e.getCardStyle(f)),
        onMouseenter: (o) => e.handleChangeHoveringCardIndex(f),
        onMouseleave: g[0] || (g[0] = (o) => e.handleChangeHoveringCardIndex(void 0))
      }, [
        l("span", {
          class: "day",
          style: r((V = (h = e.curStyle) == null ? void 0 : h.day) == null ? void 0 : V.default)
        }, n(e.getValueOrDefaultValue(a.day)), 5),
        l("span", {
          class: "date",
          style: r((S = (m = e.curStyle) == null ? void 0 : m.date) == null ? void 0 : S.default)
        }, n(e.getValueOrDefaultValue(a.date)), 5),
        l("img", {
          class: le(["image", { "image--none": !a.weatherIconUrl }]),
          style: r((I = (v = e.curStyle) == null ? void 0 : v.image) == null ? void 0 : I.default),
          src: a.weatherIconUrl
        }, null, 14, oe),
        l("span", {
          class: "desc",
          style: r((c = (L = e.curStyle) == null ? void 0 : L.desc) == null ? void 0 : c.default)
        }, n(e.getValueOrDefaultValue(a.weatherName)), 5),
        l("div", pe, [
          typeof a.temperatureMax == "number" ? (t(), u(s, { key: 0 }, [
            l("span", {
              style: r((C = (A = e.curStyle) == null ? void 0 : A.temperatureRangeMaxLabel) == null ? void 0 : C.default)
            }, n(e.getValueOrDefaultValue(a.temperatureMax)), 5),
            l("span", {
              class: "temperature__label--unit",
              style: r((b = (O = e.curStyle) == null ? void 0 : O.temperatureUnit) == null ? void 0 : b.default)
            }, n(e.temperatureUnit), 5)
          ], 64)) : (t(), u(s, { key: 1 }, [
            p(" - ")
          ], 64))
        ]),
        l("div", {
          class: "temperature__range",
          style: r(
            {
              ...(B = (q = e.curStyle) == null ? void 0 : q.temperatureRangeWrapper) == null ? void 0 : B.default,
              height: `${e.getValueOrDefaultValue(e.temperatureBarHeight, e.DEFAULT_PROPS_VALUE.temperatureBarHeight)}px`
            }
          )
        }, [
          l("div", {
            class: "temperature__range--current",
            style: r({
              width: `${e.getValueOrDefaultValue(($ = (R = (D = e.curStyle) == null ? void 0 : D.temperatureRangeInset) == null ? void 0 : R.default) == null ? void 0 : $.width, e.DEFAULT_PROPS_VALUE.temperatureBarInsetWidth)}`,
              height: a.height,
              top: `${a.top}px`
            })
          }, [
            l("div", {
              class: "temperature__range--current-inset",
              style: r({
                ...(M = (k = e.curStyle) == null ? void 0 : k.temperatureRangeInset) == null ? void 0 : M.default,
                top: `${-a.top}px`,
                height: `${e.getValueOrDefaultValue(e.temperatureBarHeight, e.DEFAULT_PROPS_VALUE.temperatureBarHeight)}px`,
                background: `linear-gradient(180deg,
                ${e.getValueOrDefaultValue(e.temperatureBarInsetBgColorTop, e.DEFAULT_PROPS_VALUE.temperatureBarInsetBgColorTop)} 0%,
                ${e.getValueOrDefaultValue(e.temperatureBarInsetBgColorBottom, e.DEFAULT_PROPS_VALUE.temperatureBarInsetBgColorBottom)} 100%)`
              })
            }, null, 4)
          ], 4)
        ], 4),
        l("div", de, [
          typeof a.temperatureMin == "number" ? (t(), u(s, { key: 0 }, [
            l("span", {
              style: r((E = (U = e.curStyle) == null ? void 0 : U.temperatureRangeMinLabel) == null ? void 0 : E.default)
            }, n(e.getValueOrDefaultValue(a.temperatureMin)), 5),
            l("span", {
              class: "temperature__label--unit",
              style: r((T = (P = e.curStyle) == null ? void 0 : P.temperatureUnit) == null ? void 0 : T.default)
            }, n(e.temperatureUnit), 5)
          ], 64)) : (t(), u(s, { key: 1 }, [
            p(" - ")
          ], 64))
        ]),
        Array.isArray(a.otherIndicator) ? (t(!0), u(s, { key: 0 }, te(a.otherIndicator, (o, ee) => {
          var ae, re;
          return t(), u("div", {
            key: o + ee,
            class: "other-indicator",
            style: r({ ...(re = (ae = e.curStyle) == null ? void 0 : ae.otherIndicator) == null ? void 0 : re.default, ...e.getOtherIndicatorStyle(ee) })
          }, n(o || "-"), 5);
        }), 128)) : d("", !0),
        l("div", {
          class: "aqi-number",
          style: r({ ...(w = (Q = e.curStyle) == null ? void 0 : Q.AQIValueRangeWrapper) == null ? void 0 : w.default })
        }, [
          typeof a.aqiMax != "number" && typeof a.aqiMin != "number" ? (t(), u(s, { key: 0 }, [
            p(" - ")
          ], 64)) : (t(), u(s, { key: 1 }, [
            l("span", {
              style: r({ ...(H = (F = e.curStyle) == null ? void 0 : F.AQIValueRangeMin) == null ? void 0 : H.default })
            }, n(e.getValueOrDefaultValue(a.aqiMin, "-")), 5),
            l("span", {
              style: r({ ...(N = (W = e.curStyle) == null ? void 0 : W.AQIValueRangeTilde) == null ? void 0 : N.default })
            }, "~", 4),
            l("span", {
              style: r({ ...(G = (z = e.curStyle) == null ? void 0 : z.AQIValueRangeMax) == null ? void 0 : G.default })
            }, n(e.getValueOrDefaultValue(a.aqiMax, "-")), 5)
          ], 64))
        ], 4),
        a.aqiLevel.length && a.aqiColor.length ? (t(), u("div", {
          key: 1,
          class: "aqi-level",
          style: r({ ...(J = (j = e.curStyle) == null ? void 0 : j.AQIRangeLevelWrapper) == null ? void 0 : J.default, background: e.getAQILevelBackgroundColor(a.aqiColor) })
        }, [
          l("span", {
            style: r({ ...(X = (K = e.curStyle) == null ? void 0 : K.AQIRangeLevelMin) == null ? void 0 : X.default, color: `rgb(${a.aqiColor[0]})` })
          }, n(a.aqiLevel[0]), 5),
          a.aqiLevel.length > 1 && a.aqiColor.length > 1 ? (t(), u(s, { key: 0 }, [
            l("span", {
              class: "aqi-level__tilde",
              style: r({
                ...(Z = (Y = e.curStyle) == null ? void 0 : Y.AQIRangeLevelTilde) == null ? void 0 : Z.default,
                backgroundImage: `linear-gradient(90deg, rgb(${a.aqiColor[0]}) 0%, rgb(${a.aqiColor[1]}) 100%)`
              })
            }, "~", 4),
            l("span", {
              style: r({ ...(x = (_ = e.curStyle) == null ? void 0 : _.AQIRangeLevelMax) == null ? void 0 : x.default, color: `rgb(${a.aqiColor[1]})` })
            }, n(a.aqiLevel[1]), 5)
          ], 64)) : d("", !0)
        ], 4)) : d("", !0)
      ], 44, se);
    }), 128))
  ], 6);
}
const Ie = /* @__PURE__ */ ne(ue, [["render", ge]]);
export {
  Ie as default
};
