// @ths/design/es/components/list-card-weather-forecast/src/index.vue2.js
import { defineComponent as W, computed as I, reactive as j, watch as _, onMounted as z, toRefs as P, useCssVars as k } from "vue";
import { AQI_LEVEL_NAME as x, AQI_LEVEL_COLOR as L } from "../../../utils/constant.js";
import { deepCloneFun as A, deepMergeFun as Y, getValueOrDefaultValue as G, getDomRandomClass as J, setStyle as K, colorRgb as b, getAQILevel as R } from "../../../utils/index.js";
const S = {
  // 容器
  wrapper: {
    default: {
      width: "400px"
    }
  },
  // 单项卡片
  card: {
    default: {}
  },
  // 图片
  image: {
    default: {}
  },
  // 星期
  day: {
    default: {}
  },
  // 日期
  date: {
    default: {}
  },
  // 天气描述
  desc: {
    default: {}
  },
  // 气温范围-最高气温文本
  temperatureRangeMaxLabel: {
    default: {}
  },
  // 气温范围-最低气温文本
  temperatureRangeMinLabel: {
    default: {}
  },
  // 气温单位
  temperatureUnit: {
    default: {}
  },
  // 气温范围容器
  temperatureRangeWrapper: {
    default: {
      height: "48px"
    }
  },
  // 气温范围容器内部柱条
  temperatureRangeInset: {
    default: {
      width: "5px"
    }
  },
  // 其他天气指标
  otherIndicator: {
    default: {}
  },
  // AQI取值范围容器
  AQIValueRangeWrapper: {
    default: {}
  },
  // AQI取值范围-最大值文本
  AQIValueRangeMax: {
    default: {}
  },
  // AQI取值范围-最小值文本
  AQIValueRangeMin: {
    default: {}
  },
  // AQI取值范围-波浪线文本
  AQIValueRangeTilde: {
    default: {}
  },
  // AQI等级范围容器
  AQIRangeLevelWrapper: {
    default: {}
  },
  // AQI等级范围-最大值文本
  AQIRangeLevelMax: {
    default: {}
  },
  // AQI等级范围-最小值文本
  AQIRangeLevelMin: {
    default: {}
  },
  // AQI等级范围-波浪线文本
  AQIRangeLevelTilde: {
    default: {}
  }
}, c = {
  temperatureUnit: "°",
  gap: 5,
  otherIndicatorStyle: [
    {
      fontSize: "12px",
      lineHeight: "20px"
    }
  ],
  temperatureBarHeight: 48,
  temperatureBarInsetWidth: "5px",
  temperatureBarInsetBgColorTop: "#D93D47",
  temperatureBarInsetBgColorBottom: "#2B5CFF"
}, X = {
  weatherName: "weatherName",
  // 天气名称
  weatherIconUrl: "weatherIconUrl",
  // 天气图标url
  temperatureMax: "temperatureMax",
  // 最高气温
  temperatureMin: "temperatureMin",
  // 最低气温
  aqiMax: "aqiMax",
  // AQI最大值
  aqiMin: "aqiMin",
  // AQI最小值
  day: "day",
  // 星期
  date: "date",
  // 日期
  otherIndicator: "otherIndicator"
  // 其他天气指标（如风力、湿度等）
}, q = W({
  name: "TListCardWeatherForecast",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        ...A(S)
      })
    },
    // 卡片组数据
    data: {
      type: Array,
      default: () => []
    },
    // 温度单位
    temperatureUnit: {
      type: String,
      default: c.temperatureUnit
    },
    // 卡片和卡片之间的间距
    gap: {
      type: Number,
      default: c.gap
    },
    // 其他天气指标（如风力、湿度等）样式
    otherIndicatorStyle: {
      type: Array,
      default: () => c.otherIndicatorStyle
    },
    // 温度柱条的高度
    temperatureBarHeight: {
      type: Number,
      default: c.temperatureBarHeight
    },
    // 温度柱条的背景色-上
    temperatureBarInsetBgColorTop: {
      type: String,
      default: c.temperatureBarInsetBgColorTop
    },
    // 温度柱条的背景色-下
    temperatureBarInsetBgColorBottom: {
      type: String,
      default: c.temperatureBarInsetBgColorBottom
    },
    // 数据字段别名
    dataAlias: {
      type: Object,
      default: X
    }
  },
  emits: [],
  setup(a, { emit: B }) {
    const m = I(() => {
      const e = A(S);
      return a != null && a.cStyle ? Y(e, a == null ? void 0 : a.cStyle) : e;
    }), y = J("list-card-weather-forecast"), o = j({
      list: [],
      // 当前光标悬停所在卡片的索引值
      hoveringCardIndex: void 0
    }), D = I(() => a.data.length), E = I(() => {
      var e, r, n, u, i, l, t, g, s, f, M, h;
      return (n = (r = (e = m.value) == null ? void 0 : e.image) == null ? void 0 : r.default) != null && n.width || (l = (i = (u = m.value) == null ? void 0 : u.image) == null ? void 0 : i.default) != null && l.height ? ((s = (g = (t = m.value) == null ? void 0 : t.image) == null ? void 0 : g.default) == null ? void 0 : s.width) || ((h = (M = (f = m.value) == null ? void 0 : f.image) == null ? void 0 : M.default) == null ? void 0 : h.height) : "32px";
    }), Q = (e) => {
      o.hoveringCardIndex = e;
    }, U = (e) => {
      var n, u, i, l;
      const r = {};
      return e === o.list.length - 1 && (r.marginRight = "0"), o.hoveringCardIndex === e ? { ...(u = (n = m.value) == null ? void 0 : n.card) == null ? void 0 : u.hover, ...r } : { ...(l = (i = m.value) == null ? void 0 : i.card) == null ? void 0 : l.default, ...r };
    }, H = (e) => e > a.otherIndicatorStyle.length - 1 ? {} : a.otherIndicatorStyle[e], p = (e, r = !0) => {
      if (Array.isArray(e)) {
        let n = [];
        return r ? (n = e.filter((u) => typeof u.temperatureMin == "number").map((u) => u.temperatureMin), Math.min(...n)) : (n = e.filter((u) => typeof u.temperatureMax == "number").map((u) => u.temperatureMax), Math.max(...n));
      }
      return 0;
    }, N = (e) => e.length > 1 ? `linear-gradient(90deg, rgba(${e[0]}, 0.2) 0%, rgba(${e[1]}, 0.2) 100%)` : e.length === 1 ? `rgba(${e[0]}, 0.2)` : "unset", C = (e) => {
      const r = "temperatureBarHeight" in a ? a == null ? void 0 : a.temperatureBarHeight : c.temperatureBarHeight;
      if (Number.isNaN(r))
        return 0;
      const n = p(e, !1) - p(e);
      return r ? n / r : 1;
    }, V = (e) => {
      const r = [], n = [], u = typeof e.aqiMax == "number", i = typeof e.aqiMin == "number";
      if (!u && !i)
        r.push("-"), n.push(b(L.level7));
      else {
        const l = R(e.aqiMax), t = R(e.aqiMin);
        r.push(x[t]), n.push(b(L[t])), l !== t && (r.push(x[l]), n.push(b(L[l])));
      }
      return {
        aqiLevel: r,
        aqiColor: n
      };
    }, d = (e) => a.dataAlias[e] ? a.dataAlias[e] : e, T = (e) => {
      const r = e.map((t) => ({
        weatherName: t[d("weatherName")],
        // 天气名称
        weatherIconUrl: t[d("weatherIconUrl")],
        // 天气图标url
        temperatureMax: t[d("temperatureMax")],
        // 最高气温
        temperatureMin: t[d("temperatureMin")],
        // 最低气温
        aqiMax: t[d("aqiMax")],
        // AQI最大值
        aqiMin: t[d("aqiMin")],
        // AQI最小值
        day: t[d("day")],
        // 星期
        date: t[d("date")],
        // 日期
        otherIndicator: t[d("otherIndicator")]
        // 其他天气指标的样式
      })), n = C(r), u = p(r, !1), i = p(r), l = [];
      r.forEach((t) => {
        const { aqiLevel: g, aqiColor: s } = V(t), f = typeof t.temperatureMax == "number" ? t.temperatureMax : u, M = typeof t.temperatureMin == "number" ? t.temperatureMin : i;
        let h = `${(f - M) / n}px`;
        typeof t.temperatureMax != "number" && typeof t.temperatureMin != "number" && (h = "0");
        const $ = {
          ...t,
          height: h,
          top: (u - f) / n,
          aqiLevel: g,
          aqiColor: s
        };
        l.push($);
      }), o.list = l;
    }, F = () => {
      const e = C(o.list), r = p(o.list, !1), n = p(o.list);
      A(o.list).forEach((i, l) => {
        const t = typeof i.temperatureMax == "number" ? i.temperatureMax : r, g = typeof i.temperatureMin == "number" ? i.temperatureMin : n;
        let s = `${(t - g) / e}px`;
        typeof i.temperatureMax != "number" && typeof i.temperatureMin != "number" && (s = "0"), o.list[l].height = s;
      });
    }, O = () => {
      let e = "";
      return m.value.wrapper && (e += K(m.value.wrapper, `.${y}`)), e;
    };
    return T(a.data), _(() => a.data, (e) => {
      T(e);
    }, { deep: !0 }), _(() => a == null ? void 0 : a.temperatureBarHeight, () => {
      F();
    }, { deep: !0 }), z(() => {
      const e = document.querySelector(`.${y} .style-container`);
      e && (e.innerHTML = `<style>${O()}</style>`);
    }), {
      /* 样式 */
      curStyle: m,
      domOnlyClass: y,
      /* 配置 */
      AQI_LEVEL_NAME: x,
      DEFAULT_MODULE_STYLE: S,
      DEFAULT_PROPS_VALUE: c,
      /* 变量 */
      ...P(o),
      imageSize: E,
      dataLength: D,
      /* 方法 */
      getAQILevelBackgroundColor: N,
      getValueOrDefaultValue: G,
      getOtherIndicatorStyle: H,
      getCardStyle: U,
      handleChangeHoveringCardIndex: Q
    };
  }
}), v = () => {
  k((a) => ({
    "4d45cab1": a.dataLength,
    "7986485b": a.imageSize
  }));
}, w = q.setup;
q.setup = w ? (a, B) => (v(), w(a, B)) : v;
export {
  q as default
};
