// @ths/design/es/components/chart-bar-3d-cylindrical/src/index.vue2.js
import { defineComponent as q, computed as j, ref as P, reactive as R, onMounted as F, onUnmounted as U, watch as z, toRefs as H, nextTick as V } from "vue";
import * as G from "echarts";
import O from "lodash/merge";
import T from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as K, DEFAULT_OPTIONS_AXIS as X } from "../../../utils/constant.js";
import { getDomRandomClass as J, setStyle as C } from "../../../utils/index.js";
const Q = {
  wrapper: {
    default: {}
  },
  // 暂无数据容器
  emptyContainer: {
    default: {}
  },
  // 暂无数据图片
  emptyImg: {
    default: {}
  },
  // 暂无数据文字
  emptyText: {
    default: {}
  }
}, Y = O(T(K), T(X)), x = O(T(Y), {
  xAxis: { boundaryGap: !0 },
  legend: {
    selectedMode: !1
    // 图例是否允许点击
  },
  tooltip: {
    formatter: (e) => {
      const p = e.map((c) => c.seriesType !== "pictorialBar" ? `<div>${c.seriesName}: ${c.value}</div>` : "").join("");
      return `
            <div>${e[0].axisValue}</div>
            ${p}
        `;
    }
  },
  grid: {
    top: "13%",
    bottom: "10%",
    left: "6%",
    right: "3%"
  }
}), ne = q({
  name: "TChartBar3dCylindrical",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => {
      },
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => Q
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 柱宽
        barWidth: {
          type: Number,
          default: 20
        },
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: p }) {
    const c = J("chart-bar-3d-cylindrical"), d = j(() => e.cStyle), f = P(null);
    let g = null, h = null;
    const a = R({
      // 图表数据
      chartData: [],
      maxValue: 0,
      // 柱子宽度
      barWidth: e.seriesSetting.barWidth || 20
    }), b = (t, i = "#43e97b") => {
      var r, m;
      return ((r = (e.options || x).color) == null ? void 0 : r[t]) || ((m = x.color) == null ? void 0 : m[t]) || i;
    }, D = () => {
      var t, i, r;
      (t = e.data) != null && t.length && ((r = (i = e.seriesSetting) == null ? void 0 : i.data) != null && r.length) && (a.chartData = e.seriesSetting.data.map((m) => {
        const { name: v, dataKey: l } = m;
        return {
          name: v || "--",
          value: e.data.map((y) => y[l])
        };
      }));
    }, _ = () => {
      let t = "";
      return d.value.wrapper && (t += C(d.value.wrapper, `.${c}`)), d.value.emptyContainer && (t += C(d.value.emptyContainer, `.${c} .empty`)), d.value.emptyImg && (t += C(d.value.emptyImg, `.${c} .empty .empty-img`)), d.value.emptyText && (t += C(d.value.emptyText, `.${c} .empty .empty-text`)), t;
    }, w = () => {
      const t = document.querySelector(`.${c} .style-container`);
      t && (t.innerHTML = `<style>${_()}</style>`);
    }, k = j(() => {
      var t, i, r, m;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((l) => {
            var y;
            return (l == null ? void 0 : l.data) && ((y = l == null ? void 0 : l.data) == null ? void 0 : y.length) !== 0;
          });
        if (typeof ((i = e.options) == null ? void 0 : i.series) == "object")
          return ((m = (r = e.options.series) == null ? void 0 : r.data) == null ? void 0 : m.length) === 0;
      }
      return !!e.data.length;
    }), A = () => {
      var v, l, y, $, E, I, M, N;
      h.clear();
      const { series: t } = e.options;
      let i = [], r = [];
      t ? i = t.map((s, n) => {
        var o;
        return s.type === "pictorialBar" ? O({
          label: {
            normal: {
              color: b(n)
            }
          },
          color: b(n),
          data: (o = a.chartData[n]) == null ? void 0 : o.value
        }, s) : s;
      }) : (v = e.data) != null && v.length && ((y = (l = e.seriesSetting) == null ? void 0 : l.data) != null && y.length) && (i = ($ = e.seriesSetting) == null ? void 0 : $.data.map((s, n) => {
        var o, u;
        return {
          name: (o = a.chartData[n]) == null ? void 0 : o.name,
          type: "bar",
          itemStyle: {
            opacity: 0.85,
            color: b(n)
          },
          z: 11,
          barWidth: a.barWidth,
          data: ((u = a.chartData[n]) == null ? void 0 : u.value) || []
        };
      }), i = i.concat((E = e.seriesSetting) == null ? void 0 : E.data.map((s, n) => {
        var S;
        let o = 0;
        const u = e.seriesSetting.data.length;
        return o = ((u % 2 ? 0 : 0.5) + Math.ceil(u / 2) - n - 1) * (a.barWidth + a.barWidth / 20 * 4), {
          name: "上椭圆",
          type: "pictorialBar",
          symbolSize: [a.barWidth, a.barWidth * 0.5],
          symbolOffset: [-o, "-50%"],
          symbolPosition: "end",
          z: 12,
          color: b(n),
          data: ((S = a.chartData[n]) == null ? void 0 : S.value) || []
        };
      })), i = i.concat((I = e.seriesSetting) == null ? void 0 : I.data.map((s, n) => {
        var S;
        let o = 0;
        const u = e.seriesSetting.data.length;
        return o = ((u % 2 ? 0 : 0.5) + Math.ceil(u / 2) - n - 1) * (a.barWidth + a.barWidth / 20 * 4), {
          name: "下椭圆",
          type: "pictorialBar",
          symbolSize: [a.barWidth, a.barWidth * 0.5],
          symbolOffset: [-o, "50%"],
          z: 12,
          color: b(n),
          data: ((S = a.chartData[n]) == null ? void 0 : S.value) || []
        };
      })), (M = e.seriesSetting) != null && M.xAxisLabel ? r = (N = e.data) != null && N.length ? e.data.map((s) => s[e.seriesSetting.xAxisLabel]) : [] : r = e.data.map((s, n) => n));
      const m = O(
        {
          series: i,
          xAxis: {
            data: r
          },
          legend: {
            data: a.chartData.map((s) => s.name)
          }
        },
        x,
        e.options
      );
      h.setOption(m);
    }, W = () => {
      V(() => {
        h.resize();
      });
    };
    function L() {
      f.value && (g = new ResizeObserver((t) => {
        t.forEach((i) => {
          i.target === f.value && W();
        });
      }), g.observe(f.value));
    }
    return F(() => {
      w(), f.value && (h = G.init(f.value), D(), A(), h.on("click", (t) => {
        p("com-click", t);
      }), h.on("legendselectchanged", (t) => {
        p("legend-select", t);
      })), window.addEventListener("resize", W), L();
    }), U(() => {
      window.removeEventListener("resize", W), g && (g.disconnect(), g = null);
    }), z(
      () => e.options,
      (t) => {
        D(), a.barWidth = e.seriesSetting.barWidth, A();
      },
      {
        deep: !0
      }
    ), z(
      [
        () => e.data,
        () => e.seriesSetting
      ],
      () => {
        D(), a.barWidth = e.seriesSetting.barWidth, A();
      },
      {
        deep: !0
      }
    ), z(
      () => e.cStyle,
      () => {
        w(), W();
      },
      {
        deep: !0
      }
    ), {
      ...H(a),
      curStyle: d,
      chartId: f,
      checkEmpty: k,
      domOnlyClass: c,
      initResizeObserver: L
    };
  }
});
export {
  ne as default
};
