// @ths/design/es/components/chart-bar-horizontal-stacked/src/index.vue2.js
import { defineComponent as M, computed as E, ref as N, reactive as P, onMounted as q, onUnmounted as F, watch as A, nextTick as j } from "vue";
import * as G from "echarts";
import y from "lodash/merge";
import O from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as H, DEFAULT_OPTIONS_AXIS as U } from "../../../utils/constant.js";
import { getDomRandomClass as B, setStyle as h } from "../../../utils/index.js";
const K = y(O(H), O(U)), g = y(
  O(K),
  {
    // 气泡框
    tooltip: {
      axisPointer: {
        type: "line"
      }
    },
    // 网格配置
    grid: {
      top: "13%",
      left: "2%",
      right: "7%",
      bottom: "2%",
      containLabel: !0
    },
    // y轴
    yAxis: {
      name: "",
      type: "category",
      nameGap: 5,
      axisLabel: {
        margin: 10,
        interval: 0
      },
      boundaryGap: !0,
      splitLine: {
        show: !1
      }
    },
    // x轴
    xAxis: {
      name: "单位",
      nameLocation: "end",
      type: "value",
      nameGap: 10,
      inverse: !1,
      splitNumber: 5,
      splitLine: {
        show: !0
      }
    }
  }
), Y = M({
  name: "TChartBarHorizontalStacked",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 条形图数据；如果Options设置了series属性，则会忽略data的数据
    data: {
      type: Array,
      default: () => []
    },
    // 包含 系列样式 、别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // y 轴类目取resData数据中的别名设置
        yAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => g,
      required: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外容器
        wrapper: {
          default: {}
        }
      })
    },
    // 柱状图主题样式，用于平台里设置
    bar: {
      type: Object,
      default: () => ({})
    }
  },
  // 事件名定义
  emits: ["col-click"],
  setup(t, { emit: I }) {
    const d = B("chart-bar-horizontal-stacked"), i = E(() => t.cStyle), b = N(null);
    let m = null;
    const S = P({
      // 图表的数据
      chartData: []
    }), L = () => {
      var e, o, r;
      (o = (e = t.seriesSetting) == null ? void 0 : e.data) != null && o.length && t.data && (S.chartData = (r = t.seriesSetting) == null ? void 0 : r.data.map((s) => {
        const { name: u, dataKey: n } = s;
        return {
          name: u || "--",
          data: t.data.map((c) => c[n])
        };
      }));
    }, W = () => {
      let e = "";
      return i.value.wrapper && (e += h(i.value.wrapper, `.${d}`)), i.value.emptyContainer && (e += h(i.value.emptyContainer, `.${d} .empty`)), i.value.emptyImg && (e += h(i.value.emptyImg, `.${d} .empty .empty-img`)), i.value.emptyText && (e += h(i.value.emptyText, `.${d} .empty .empty-text`)), e;
    }, C = () => {
      const e = document.querySelector(`.${d} .style-container`);
      e && (e.innerHTML = `<style>${W()}</style>`);
    }, _ = E(() => {
      var e, o, r, s;
      if (t.options) {
        if (Array.isArray((e = t.options) == null ? void 0 : e.series))
          return t.options.series.some((n) => {
            var c;
            return (n == null ? void 0 : n.data) && ((c = n == null ? void 0 : n.data) == null ? void 0 : c.length) !== 0;
          });
        if (typeof ((o = t.options) == null ? void 0 : o.series) == "object")
          return ((s = (r = t.options.series) == null ? void 0 : r.data) == null ? void 0 : s.length) === 0;
      }
      return !!t.data.length;
    }), v = () => {
      var n, c, D, w, x, k, T;
      m.clear();
      const e = (n = t.options) == null ? void 0 : n.color, o = (c = t.options) == null ? void 0 : c.series;
      let r = [], s = [];
      o ? r = o.map((a, l) => y(
        {
          type: "bar",
          stack: "total",
          barWidth: (a == null ? void 0 : a.barWidth) || 12,
          itemStyle: {
            color: e && e[l] || g.color[l] || "#fff"
          },
          ...t == null ? void 0 : t.bar
        },
        a
      )) : t.data && ((w = (D = t.seriesSetting) == null ? void 0 : D.data) != null && w.length) && (r = (x = t.seriesSetting) == null ? void 0 : x.data.map((a, l) => {
        var p, z;
        const $ = y({
          name: (p = S.chartData[l]) == null ? void 0 : p.name,
          stack: "total",
          type: "bar",
          barWidth: a.barWidth || 12,
          itemStyle: {
            color: e && e[l] || g.color[l] || "#fff"
          },
          data: ((z = S.chartData[l]) == null ? void 0 : z.data) || []
        }, t == null ? void 0 : t.bar);
        return y({}, $, { barWidth: a.barWidth, itemStyle: a.itemStyle });
      }), (k = t.seriesSetting) != null && k.yAxisLabel ? s = (T = t.data) != null && T.length ? t.data.map((a) => a[t.seriesSetting.yAxisLabel]) : [] : s = t.data.map((a, l) => l));
      let u = y(
        {
          series: r,
          yAxis: {
            data: s
          }
        },
        g
      );
      u = y(u, t.options);
      try {
        m.setOption(u);
      } catch (a) {
        console.log(a);
      }
    }, f = () => {
      j(() => {
        m.resize();
      });
    };
    return q(() => {
      C(), b.value && (m = G.init(b.value), L(), v(), m.on("click", (e) => {
        I("col-click", e);
      })), f(), window.addEventListener("resize", f);
    }), F(() => {
      window.removeEventListener("resize", f);
    }), A(
      () => [
        t.options,
        t.bar
      ],
      () => {
        v(), f();
      },
      {
        deep: !0
      }
    ), A(
      [
        () => t.data,
        () => t.seriesSetting
      ],
      () => {
        L(), v(), f();
      },
      {
        deep: !0
      }
    ), A(
      () => [t.cStyle],
      () => {
        C(), j(() => {
          m.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      chart: b,
      curStyle: i,
      checkEmpty: _,
      domOnlyClass: d
    };
  }
});
export {
  Y as default
};
