// @ths/design/es/components/chart-yaxis-inverse/src/index.vue2.js
import { defineComponent as U, computed as M, ref as G, reactive as H, nextTick as W, onMounted as K, onUnmounted as X, watch as T, toRefs as Y } from "vue";
import * as j from "echarts";
import S from "lodash/merge";
import m from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as Q, DEFAULT_OPTIONS_AXIS as V } from "../../../utils/constant.js";
import { getDomRandomClass as Z, setStyle as C } from "../../../utils/index.js";
const x = S(m(Q), m(V), {
  yAxis: {
    axisLine: {
      show: !1
    },
    inverse: !0
  },
  grid: {
    containLabel: !0,
    top: "20%",
    bottom: "3%",
    left: "1%",
    right: "2%"
  }
}), le = U({
  name: "TChartYaxisInverse",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // line主题配置
    line: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // bar主题配置
    bar: {
      type: Object,
      default: () => ({}),
      required: !1
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: z }) {
    const b = Z("chart-yaxis-inverse"), f = M(() => e.cStyle), p = G(null);
    let h = null;
    const r = H({
      // 图表数据
      chartData: [],
      lineOptions: {},
      barOptions: {}
    }), E = () => {
      var t, i;
      (i = (t = e.seriesSetting) == null ? void 0 : t.data) != null && i.length && e.data && (r.chartData = e.seriesSetting.data.map((s) => {
        const { name: c, dataKey: w } = s;
        return {
          name: c || "--",
          data: e.data.map((y) => y[w])
        };
      }));
    }, J = () => {
      let t = "";
      return f.value.wrapper && (t += C(f.value.wrapper, `.${b}`)), f.value.emptyContainer && (t += C(f.value.emptyContainer, `.${b} .empty`)), f.value.emptyImg && (t += C(f.value.emptyImg, `.${b} .empty .empty-img`)), f.value.emptyText && (t += C(f.value.emptyText, `.${b} .empty .empty-text`)), t;
    }, I = () => {
      const t = document.querySelector(`.${b} .style-container`);
      t && (t.innerHTML = `<style>${J()}</style>`);
    }, g = () => {
      p.value && (p.value.getBoundingClientRect().width ?? 0) && W(() => {
        h.resize();
      });
    }, B = M(() => {
      var t, i, s, c;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((y) => {
            var v;
            return (y == null ? void 0 : y.data) && ((v = y == null ? void 0 : y.data) == null ? void 0 : v.length) !== 0;
          });
        if (typeof ((i = e.options) == null ? void 0 : i.series) == "object")
          return ((c = (s = e.options.series) == null ? void 0 : s.data) == null ? void 0 : c.length) === 0;
      }
      return e.data.length ? (W(() => {
        g();
      }), !0) : !1;
    }), l = (t, i = "#FFF") => {
      var s, c;
      return ((s = e.options.color) == null ? void 0 : s[t]) || ((c = x.color) == null ? void 0 : c[t]) || i;
    }, L = () => {
      var y, v, N, $;
      if (!h)
        return;
      const { series: t } = e.options;
      let i = [], s = [], c = [];
      t ? i = t.map((o, a) => {
        var n, F, A;
        return S(
          {
            name: "",
            type: "line",
            smooth: !0,
            symbol: "circle",
            symbolSize: 8,
            areaStyle: {
              color: new j.graphic.LinearGradient(0, 0, 0, 1, [
                {
                  offset: 0,
                  color: l(a)
                  // 0% 处的颜色
                },
                {
                  offset: 1,
                  color: "transparent"
                  // 100% 处的颜色
                }
              ])
            },
            itemStyle: {
              color: {
                type: "linear",
                x: 1,
                y: 1,
                r: 1,
                colorStops: [
                  {
                    offset: 0,
                    color: l(a, "#FFFFFF00")
                    // 0% 处的颜色
                  },
                  {
                    offset: 1,
                    color: l(a, "#FFFFFF00")
                    // 100% 处的颜色
                  }
                ]
              },
              borderColor: l(a),
              // 边框颜色
              borderWidth: 2,
              borderRadius: 2,
              shadowColor: l(a),
              // 图形阴影颜色
              shadowBlur: 10
              // 图形阴影
            },
            lineStyle: {
              color: l(a)
            }
          },
          // 根据type合并主题传入的配置
          o.type === "bar" ? r == null ? void 0 : r.barOptions : r == null ? void 0 : r.lineOptions,
          // 合并seriesSetting中设置的style
          (A = (F = (n = e.seriesSetting) == null ? void 0 : n.data) == null ? void 0 : F[a]) == null ? void 0 : A.style,
          m(o)
        );
      }) : (v = (y = e.seriesSetting) == null ? void 0 : y.data) != null && v.length && e.data && (e.seriesSetting.data.forEach((o, a) => {
        var F, A, q, R, _, k;
        const n = S(
          {
            name: o.name,
            type: "line",
            smooth: !0,
            symbol: "circle",
            symbolSize: 8,
            areaStyle: {
              color: new j.graphic.LinearGradient(0, 0, 0, 1, [
                {
                  offset: 0,
                  color: l(a)
                },
                {
                  offset: 1,
                  color: "transparent"
                }
              ])
            },
            itemStyle: {
              color: {
                type: "linear",
                x: 1,
                y: 1,
                r: 0,
                colorStops: [
                  {
                    offset: 0,
                    color: l(a, "#FFFFFF00")
                    // 0% 处的颜色
                  },
                  {
                    offset: 1,
                    color: "rgba(255, 255, 255, 0)"
                    // 100% 处的颜色
                  }
                ]
              },
              borderColor: l(a),
              // 边框颜色
              borderWidth: 2,
              borderRadius: 2,
              shadowColor: l(a),
              // 图形阴影颜色
              shadowBlur: 10
              // 图形阴影
            },
            lineStyle: {
              color: l(a)
            },
            barWidth: 20,
            data: e.data.length ? (F = r.chartData[a]) == null ? void 0 : F.data : []
          },
          // 根据type合并主题传入的配置
          ((R = (q = (A = e.seriesSetting.data) == null ? void 0 : A[a]) == null ? void 0 : q.style) == null ? void 0 : R.type) === "bar" ? r.barOptions : r.lineOptions,
          // 合并seriesSetting中设置的style
          (k = (_ = e.seriesSetting.data) == null ? void 0 : _[a]) == null ? void 0 : k.style,
          m(o)
        );
        if (n.type === "bar") {
          let d = [];
          for (let u = 0; u < n.data.length; u++)
            if (n.data[u] < 0) {
              if (d.length === 0) {
                d = JSON.parse(JSON.stringify(n.itemStyle.color.colorStops));
                const D = d.length;
                for (let O = 0; O < Math.floor(D / 2); O++) {
                  const P = d[O].color;
                  d[O].color = d[D - O - 1].color, d[D - O - 1].color = P;
                }
              }
              n.data[u] = {
                value: n.data[u],
                itemStyle: JSON.parse(JSON.stringify(n.itemStyle))
              }, n.data[u].itemStyle.color.colorStops = d;
            }
        }
        if (i.push(n), e.data.length > 1) {
          const d = S({
            type: "lines",
            coordinateSystem: "cartesian2d",
            zlevel: 1,
            polyline: !0,
            smooth: !0,
            symbol: "circle",
            data: [
              {
                coords: e.data.map((u) => [u[e.seriesSetting.xAxisLabel], u[o.dataKey]])
              }
            ],
            effect: {
              show: !1,
              // 拖尾长度0-1
              trailLength: 0.4,
              symbol: "circle",
              // 配置特效图形的移动动画是否是固定速度，
              // 单位像素/秒，设置为大于 0 的值后会忽略 period 配置项。
              // constantSpeed: 280,
              period: 8,
              symbolSize: 4
            },
            lineStyle: {
              color: l(a),
              width: 0,
              cap: "round",
              opacity: 0,
              curveness: 0
            }
          }, o == null ? void 0 : o.linesStyle);
          i.push(d);
        }
      }), (N = e.seriesSetting) != null && N.xAxisLabel ? s = ($ = e.data) != null && $.length ? e.data.map((o) => o[e.seriesSetting.xAxisLabel]) : [] : s = e.data.map((o, a) => `${a}`)), e.options.yAxis && Array.isArray(e.options.yAxis) ? c = e.options.yAxis.map((o, a) => S(m(x.yAxis), o, {
        // 控制第二根轴以上axisLabel左对齐
        axisLabel: {
          align: a ? "left" : "right"
        }
      })) : c = S(m(x.yAxis), e.options.yAxis || {});
      const w = S(
        {
          series: i,
          xAxis: {
            data: s
          },
          yAxis: c
        },
        m(x),
        e.options
      );
      h.clear();
      try {
        h.setOption(w);
      } catch (o) {
        console.log(o);
      }
    };
    return K(() => {
      I(), p.value && (h = j.init(p.value), E(), L(), h.on("click", (t) => {
        z("com-click", t);
      }), h.on("legendselectchanged", (t) => {
        z("legend-select", t);
      })), setTimeout(() => {
        g();
      }, 500), window.addEventListener("resize", g);
    }), X(() => {
      window.removeEventListener("resize", g);
    }), T(
      () => [e.line, e.bar],
      (t) => {
        r.lineOptions = m(e.line), r.barOptions = m(e.bar), L(), g();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), T(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting
      ],
      () => {
        E(), L(), g();
      },
      {
        deep: !0
      }
    ), T(
      () => e.cStyle,
      () => {
        I(), g();
      },
      {
        deep: !0
      }
    ), {
      ...Y(r),
      chartId: p,
      curStyle: f,
      checkEmpty: B,
      domOnlyClass: b
    };
  }
});
export {
  le as default
};
