// @ths/design/es/components/header-base/src/index.vue2.js
import { defineComponent as g, computed as i, reactive as y, ref as S, watch as l, toRefs as x } from "vue";
import "../../text/index.js";
import p from "../../text/src/index.vue.js";
const h = g({
  name: "THeaderBase",
  components: {
    TText: p
  },
  props: {
    // 图标(图片地址)
    icon: {
      type: String,
      default: ""
    },
    // 标题
    title: {
      type: String,
      default: "标题"
    },
    // 导航背景图
    backgroundImage: {
      type: String,
      default: ""
    },
    // 标题是否显示手型（为标题添加点击事件的时候提供鼠标显示手型样式）
    pointer: {
      type: Boolean,
      default: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // logo 图标
        icon: {
          default: {}
        },
        // 标题文字
        title: {
          default: {}
        }
      })
    }
  },
  emits: ["modify-text", "title-click"],
  setup(t, { emit: o }) {
    const r = i(() => t.cStyle), u = i(() => t.title), n = y({
      customStyle: t.cStyle
      // 组件样式
    }), c = S(!0), a = () => {
      if (t.icon) {
        const e = t.icon.lastIndexOf("."), m = t.icon.substr(e + 1);
        c.value = m === "svg";
      }
    }, d = (e) => {
      n.customStyle.wrapper.default["background-image"] = `url(${e})`;
    };
    a(), l(
      () => t.icon,
      () => {
        a();
      }
    ), l(
      () => t.backgroundImage,
      (e) => {
        d(e);
      }
    );
    const f = (e) => {
      o("modify-text", e);
    }, s = () => {
      o("title-click");
    };
    return {
      curTitle: u,
      ...x(n),
      handleModifyTextChange: f,
      handClick: s,
      isSvgOfLogo: c,
      currentStyle: r
    };
  }
});
export {
  h as default
};
