// @ths/design/es/components/header-base/src/index.vue.js
import g from "./index.vue2.js";
import { resolveComponent as b, openBlock as l, createElementBlock as a, normalizeStyle as t, createElementVNode as h, createCommentVNode as c, createVNode as k, normalizeClass as S, renderSlot as C } from "vue";
import "./index.vue3.js";
import $ from "../../../_virtual/_plugin-vue_export-helper.js";
const v = ["src"];
function T(e, o, B, E, M, N) {
  var r, s, n, i, d, f, u, m;
  const p = b("t-text");
  return l(), a("header", {
    class: "t-header-base",
    style: t({
      "background-image": `url(${e.backgroundImage})`,
      ...(s = (r = e.currentStyle) == null ? void 0 : r.wrapper) == null ? void 0 : s.default
    })
  }, [
    h("div", {
      class: "t-header-base-left",
      onClick: o[0] || (o[0] = (...y) => e.handClick && e.handClick(...y))
    }, [
      e.icon && !e.isSvgOfLogo ? (l(), a("img", {
        key: 0,
        class: "t-header-base-left__img",
        src: e.icon,
        style: t({
          ...(i = (n = e.currentStyle) == null ? void 0 : n.icon) == null ? void 0 : i.default
        }),
        alt: "图标",
        draggable: "false"
      }, null, 12, v)) : c("", !0),
      e.icon && e.isSvgOfLogo ? (l(), a("div", {
        key: 1,
        class: "t-header-base-left__logo",
        style: t({
          maskImage: `url(${e.icon})`,
          ...(f = (d = e.currentStyle) == null ? void 0 : d.icon) == null ? void 0 : f.default
        })
      }, null, 4)) : c("", !0),
      k(p, {
        tip: !1,
        label: e.curTitle,
        "edit-key": "title",
        class: S(["t-header-base-left__text", { "t-header-base-left__text--pointer": e.pointer }]),
        style: t({ ...(m = (u = e.currentStyle) == null ? void 0 : u.title) == null ? void 0 : m.default }),
        onModifyText: e.handleModifyTextChange
      }, null, 8, ["label", "class", "style", "onModifyText"])
    ]),
    C(e.$slots, "default")
  ], 4);
}
const O = /* @__PURE__ */ $(g, [["render", T]]);
export {
  O as default
};
