// @ths/design/es/components/select/src/index.vue2.js
import { defineComponent as O, computed as V, reactive as k, ref as P, watch as v, onMounted as j, toRefs as H } from "vue";
import M from "lodash/merge";
import D from "lodash/cloneDeep";
import { filterAttrs as N, getDomRandomClass as T, getCssPropertyString as i, setStyle as d } from "../../../utils/index.js";
const C = {
  label: "label",
  value: "value",
  disabled: "disabled"
}, z = O({
  name: "TSelect",
  props: {
    // 高亮导航项
    activeValue: {
      type: [String, Number, Boolean, Object, Array],
      default: ""
    },
    // 数据
    options: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => D(C),
      require: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {},
          focus: {},
          hover: {}
        },
        arrow: {
          default: {}
        },
        placeholder: {
          default: {}
        },
        tag: {
          default: {}
        },
        tagText: {
          default: {}
        },
        tagClose: {
          default: {}
        },
        disabledWrapper: {
          default: {}
        },
        disabledArrow: {
          default: {}
        },
        disabledPlaceholder: {
          default: {}
        },
        dropdownWrapper: {
          default: {}
        },
        dropdownItem: {
          default: {},
          active: {},
          hover: {}
        },
        dropdownItemText: {
          default: {},
          active: {},
          hover: {}
        },
        dropdownArrow: {
          default: {}
        }
      })
    },
    // 滚动条轨道颜色
    scrollbarTrackColor: {
      type: String,
      default: ""
    },
    // 滚动条滑块颜色
    scrollbarThumbColor: {
      type: String,
      default: ""
    },
    // 滚动条宽度
    scrollbarWidth: {
      type: Number,
      default: 4
    }
  },
  emits: ["change", "search"],
  setup(o, { emit: f, attrs: I }) {
    const l = V(() => o.cStyle), s = k({
      curOptions: o.options
    }), u = P(o.activeValue), t = T("select"), r = T("select-popper");
    v(() => o.activeValue, (e) => {
      u.value = e;
    }), v(() => [o.options, o.alias], () => {
      const e = M(C, o.alias);
      try {
        s.curOptions = o.options.map((a) => ({
          ...a,
          disabled: a[e.disabled],
          label: a[e.label],
          value: a[e.value]
        }));
      } catch (a) {
        console.error(a);
      }
    }, {
      deep: !0,
      immediate: !0
    });
    const S = (e) => {
      let a = "", p = [], c = [];
      u.value = e, I.multiple ? (a = e.toString(), p = e, c = s.curOptions.filter((n) => e.includes(n.value))) : (a = e, p = [e], c = s.curOptions.find((n) => n.value === e)), f("change", a, p, c);
    }, A = (e) => {
      console.log(e), f("search", e);
    }, x = () => {
      var a, p, c, n, $, w, b, _, h, g, y;
      let e = "";
      return (a = l.value.wrapper) != null && a.focus && (e += `
          .${t} .el-select .el-input.is-focus .el-input__inner {
            ${i(l.value.wrapper.focus)}
          }
        `), (p = l.value.dropdownArrow) != null && p.default && (e += `
          .${t} .${r}.el-popper[role=tooltip]>.el-popper__arrow::before {
            ${i((c = l.value.dropdownArrow) == null ? void 0 : c.default)}
          }
        `), (n = l.value.dropdownItem) != null && n.active && (e += `
          .${t} .${r}.el-popper[role=tooltip] .el-select-dropdown__item.selected {
            ${i(l.value.dropdownItem.active)}
          }
        `), ($ = l.value.dropdownItem) != null && $.hover ? e += `
         .${t} .${r}.el-popper[role=tooltip] .el-select-dropdown__item:hover {
            ${i(l.value.dropdownItem.hover)}
          }
        ` : (w = l.value.dropdownItem) != null && w.active && (e += `
         .${t} .${r}.el-popper[role=tooltip] .el-select-dropdown__item:hover {
            ${i((b = l.value.dropdownItem) == null ? void 0 : b.active)}
          }
        `), (_ = l.value.dropdownItemText) != null && _.active && (e += `
          .${t} .${r}.el-popper[role=tooltip] .el-select-dropdown__item.selected span{
            ${i(l.value.dropdownItemText.active)}
          }
        `), (h = l.value.dropdownItemText) != null && h.hover ? e += `
         .${t} .${r}.el-popper[role=tooltip] .el-select-dropdown__item:hover span{
            ${i(l.value.dropdownItemText.hover)}
          }
        ` : (g = l.value.dropdownItemText) != null && g.active && (e += `
         .${t} .${r}.el-popper[role=tooltip] .el-select-dropdown__item:hover span{
            ${i((y = l.value.dropdownItemText) == null ? void 0 : y.active)}
          }
        `), o.scrollbarTrackColor && (e += `
        .${t} .${r}.el-popper[role=tooltip]{
          --scrollbar-track-color: ${o.scrollbarTrackColor};
        }
        `), o.scrollbarThumbColor && (e += `
        .${t} .${r}.el-popper[role=tooltip]{
          --scrollbar-thumb-color: ${o.scrollbarThumbColor};
        }
        `), o.scrollbarWidth && (e += `
       .${t} .${r}.el-popper[role=tooltip]{
          --scrollbar-width: ${o.scrollbarWidth};
        }
        `), e;
    }, W = () => {
      let e = "";
      return l.value.wrapper && (e += d(l.value.wrapper, `.${t} .el-input__inner`)), l.value.arrow && (e += d(l.value.arrow, `.${t} .el-input .el-select__caret`)), l.value.tag && (e += d(l.value.tag, `.${t} .el-select__tags .el-tag`)), l.value.tagText && (e += d(l.value.tagText, `.${t} .el-select__tags .el-select__tags-text,
         .${t} .el-select__tags .el-tag .el-icon-close`)), l.value.tagClose && (e += d(l.value.tagClose, `.${t} .el-select__tags .el-tag .el-icon-close`)), l.value.dropdownWrapper && (e += d(l.value.dropdownWrapper, `.${t} .${r}.el-popper[role=tooltip]`)), l.value.dropdownItem && (e += d(l.value.dropdownItem, `.${t} .${r}.el-popper[role=tooltip] .el-select-dropdown__item`)), l.value.dropdownItemText && (e += d(l.value.dropdownItemText, `.${t} .${r}.el-popper[role=tooltip] .el-select-dropdown__item span`)), l.value.placeholder && (e += d(l.value.placeholder, `.${t} .el-input__inner::placeholder`)), l.value.disabledWrapper && (e += d(l.value.disabledWrapper, `.${t} .is-disabled .el-input__inner`)), l.value.disabledPlaceholder && (e += d(l.value.disabledPlaceholder, `.${t} .is-disabled .el-input__inner::placeholder`)), l.value.disabledArrow && (e += d(l.value.disabledArrow, `.${t} .el-input.is-disabled .el-select__caret`)), e;
    }, m = () => {
      const e = document.querySelector(`.${t} .style-container`);
      e && (e.innerHTML = `<style>${W()}${x()}</style>`);
    };
    return v([() => o.cStyle, o.scrollbarTrackColor, o.scrollbarThumbColor, o.scrollbarWidth], () => {
      m();
    }), j(() => {
      m();
    }), {
      ...H(s),
      curActiveValue: u,
      curStyle: l,
      domOnlyClass: t,
      popperOnlyClass: r,
      filterAttrs: N,
      handleSelectChange: S,
      remoteMethod: A
    };
  }
});
export {
  z as default
};
