// @ths/design/es/components/select/src/index.vue.js
import y from "./index.vue2.js";
import { resolveComponent as u, openBlock as t, createElementBlock as f, normalizeClass as v, normalizeStyle as b, createVNode as C, mergeProps as h, withCtx as S, Fragment as V, renderList as w, createBlock as B, createElementVNode as $ } from "vue";
import "./index.vue3.js";
import k from "../../../_virtual/_plugin-vue_export-helper.js";
function A(e, r, L, O, z, E) {
  var o, a, n, p, s, d, m, i;
  const c = u("el-option"), g = u("el-select");
  return t(), f("div", {
    class: v(["t-select", e.domOnlyClass]),
    style: b({
      marginTop: (a = (o = e.curStyle.wrapper) == null ? void 0 : o.default) == null ? void 0 : a.marginTop,
      marginLeft: (p = (n = e.curStyle.wrapper) == null ? void 0 : n.default) == null ? void 0 : p.marginLeft,
      marginRight: (d = (s = e.curStyle.wrapper) == null ? void 0 : s.default) == null ? void 0 : d.marginRight,
      marginBottom: (i = (m = e.curStyle.wrapper) == null ? void 0 : m.default) == null ? void 0 : i.marginBottom
    })
  }, [
    C(g, h({
      modelValue: e.curActiveValue,
      "onUpdate:modelValue": r[0] || (r[0] = (l) => e.curActiveValue = l),
      "popper-append-to-body": !1
    }, e.filterAttrs(e.$attrs), {
      "popper-class": e.popperOnlyClass,
      teleported: !1,
      "remote-method": e.remoteMethod,
      onChange: e.handleSelectChange
    }), {
      default: S(() => [
        (t(!0), f(V, null, w(e.curOptions, (l) => (t(), B(c, {
          key: l.value,
          label: l.label,
          title: l.label,
          value: l.value,
          disabled: l.disabled || !1
        }, null, 8, ["label", "title", "value", "disabled"]))), 128))
      ]),
      _: 1
    }, 16, ["modelValue", "popper-class", "remote-method", "onChange"]),
    r[1] || (r[1] = $("div", { class: "style-container" }, null, -1))
  ], 6);
}
const M = /* @__PURE__ */ k(y, [["render", A]]);
export {
  M as default
};
