// @ths/design/es/components/dialog-single-tree-input/src/index.vue.js
import u from "./index.vue2.js";
import { resolveComponent as l, openBlock as m, createElementBlock as c, createVNode as n, withCtx as d, createTextVNode as g, mergeProps as b } from "vue";
import _ from "../../../_virtual/_plugin-vue_export-helper.js";
const f = { class: "input-wrapper" };
function V(e, o, S, C, D, h) {
  const a = l("search"), r = l("el-icon"), t = l("el-button"), s = l("el-input"), p = l("t-dialog-single-tree");
  return m(), c("div", f, [
    n(s, {
      modelValue: e.curSelectedName,
      "onUpdate:modelValue": o[0] || (o[0] = (i) => e.curSelectedName = i),
      placeholder: e.placeholder,
      clearable: "",
      disabled: e.disabled,
      onFocus: e.openDialog,
      onClear: e.clearInput
    }, {
      append: d(() => [
        n(t, {
          size: "small",
          disabled: e.disabled,
          onClick: e.openDialog
        }, {
          default: d(() => [
            n(r, { size: 13 }, {
              default: d(() => [
                n(a)
              ]),
              _: 1
            }),
            o[2] || (o[2] = g(" 选择 "))
          ]),
          _: 1,
          __: [2]
        }, 8, ["disabled", "onClick"])
      ]),
      _: 1
    }, 8, ["modelValue", "placeholder", "disabled", "onFocus", "onClear"]),
    n(p, b({
      dialogVisible: e.visible,
      "onUpdate:dialogVisible": o[1] || (o[1] = (i) => e.visible = i),
      "hidden-id": e.curSelectedId
    }, e.$attrs, { onDialogSure: e.handleDialogSure }), null, 16, ["dialogVisible", "hidden-id", "onDialogSure"])
  ]);
}
const N = /* @__PURE__ */ _(u, [["render", V]]);
export {
  N as default
};
