// @ths/design/es/components/dialog-single-tree-input/src/index.vue2.js
import { defineComponent as p, reactive as m, watch as r, toRefs as f } from "vue";
import g from "../../dialog-single-tree/src/index.vue.js";
const N = p({
  name: "TDialogSingleTreeInput",
  components: {
    TDialogSingleTree: g
  },
  props: {
    // 选中的节点id
    selectedId: {
      type: String,
      default: ""
    },
    // 选中的节点名称
    selectedName: {
      type: String,
      default: ""
    },
    // 是否不可编辑
    disabled: {
      type: Boolean,
      default: !1
    },
    // 输入框提示文字
    placeholder: {
      type: String,
      default: "请选择"
    }
  },
  emits: ["update:selectedId", "update:selectedName", "selected", "clear"],
  setup(a, { emit: t, attrs: d }) {
    const l = m({
      // 单选角色弹框是否可见
      visible: !1,
      // 选中的节点名称
      curSelectedName: "",
      // 选中的节点id
      curSelectedId: ""
    });
    r(() => a.selectedName, (e) => {
      l.curSelectedName = e;
    }, {
      deep: !0,
      immediate: !0
    }), r(() => a.selectedId, (e) => {
      l.curSelectedId = e;
    }, {
      deep: !0,
      immediate: !0
    });
    const i = (e) => {
      l.visible = !0, e.target.blur();
    }, o = () => {
      l.visible = !1;
    }, n = () => {
      t("update:selectedId", ""), t("update:selectedName", ""), t("clear");
    }, u = (e) => {
      var s, c;
      t("update:selectedId", e[((s = d["simple-data"]) == null ? void 0 : s.idKey) || "treeId"]), t("update:selectedName", e[((c = d["simple-data"]) == null ? void 0 : c.nameKey) || "treeName"]), t("selected", e), l.visible = !1;
    };
    return {
      ...f(l),
      clearInput: n,
      handleDialogSure: u,
      openDialog: i,
      closeDialog: o
    };
  }
});
export {
  N as default
};
