// @ths/design/es/components/trigger-menu/src/index.vue.js
import M from "./index.vue2.js";
import { openBlock as o, createElementBlock as a, normalizeClass as s, createElementVNode as n, normalizeStyle as d, toDisplayString as r, withDirectives as $, Fragment as w, renderList as I, createCommentVNode as k, vShow as A } from "vue";
import "./index.vue3.js";
import C from "../../../_virtual/_plugin-vue_export-helper.js";
const L = { class: "relative" }, S = ["src"], T = ["src"], B = ["src"], D = ["onClick"], F = { class: "menu-item-text clip" }, O = ["onClick"], R = { class: "clip" };
function z(e, t, E, N, V, W) {
  var u, g, m, v, p, f;
  return o(), a("div", {
    class: s(["t-trigger-menu", e.domOnlyClass])
  }, [
    n("div", L, [
      n("div", {
        class: s(["trigger-title menu-title", { "title-active": e.showAnimation, vertical: e.triggerTextWriteMode === "vertical", "title-transition": e.titleTransition }]),
        style: d({
          "--menu-width": e.menuWidth,
          "--menu-height": e.menuHeight,
          opacity: e.imgOnloaded ? 1 : 0
        }),
        onClick: t[3] || (t[3] = (...l) => e.changeShowMenu && e.changeShowMenu(...l))
      }, [
        n("img", {
          ref: "topImgRef",
          class: "title-front",
          draggable: "false",
          src: e.showAnimation ? (u = e.titleFrontImg) == null ? void 0 : u.active : (g = e.titleFrontImg) == null ? void 0 : g.default,
          alt: "",
          onLoad: t[0] || (t[0] = (l) => e.onLoad())
        }, null, 40, S),
        t[4] || (t[4] = n("div", { class: "title-fill" }, null, -1)),
        n("img", {
          class: "title-mid",
          src: e.showAnimation ? (m = e.titleMidImg) == null ? void 0 : m.active : (v = e.titleMidImg) == null ? void 0 : v.default,
          alt: "",
          draggable: "false",
          onLoad: t[1] || (t[1] = (l) => e.onLoad())
        }, null, 40, T),
        t[5] || (t[5] = n("div", { class: "title-fill" }, null, -1)),
        n("img", {
          ref: "bottomImgRef",
          class: "title-below",
          src: e.showAnimation ? (p = e.titleBelowImg) == null ? void 0 : p.active : (f = e.titleBelowImg) == null ? void 0 : f.default,
          alt: "",
          draggable: "false",
          onLoad: t[2] || (t[2] = (l) => e.onLoad())
        }, null, 40, B),
        n("span", null, r(e.currentItem), 1),
        n("div", {
          class: s(["title-arrow", { "arrow-active": e.showAnimation }])
        }, null, 2)
      ], 6),
      n("span", {
        ref: "autoTextRef",
        class: "text"
      }, r(e.currentItem), 513),
      t[7] || (t[7] = n("div", { class: "style-container" }, null, -1)),
      $(n("div", {
        class: "trigger-menu translate",
        style: d({
          "--translate-x": e.showMenuAnimation ? `${e.translateX}px` : "0",
          "--opacity": e.showMenuAnimation ? "1" : "0"
        })
      }, [
        (o(!0), a(w, null, I(e.curData, (l, b) => {
          var h, c;
          return o(), a("div", {
            key: b,
            class: "menu-collapse"
          }, [
            n("div", {
              class: s(["menu-item", {
                active: ((h = l == null ? void 0 : l.children) == null ? void 0 : h.length) > 0 ? l.children.findIndex((i) => (i == null ? void 0 : i.label) === e.currentItem) > -1 : e.currentItem === l.label
              }]),
              onClick: (i) => e.toggleItemOverflow(l)
            }, [
              n("span", F, r(l.label), 1),
              k("", !0)
            ], 10, D),
            n("div", {
              class: s(["menu-content", { hidden: !e.hiddenMenuItems.includes(l.label) }]),
              style: d({
                paddingTop: (c = l == null ? void 0 : l.children) != null && c.length ? "4px" : 0
              })
            }, [
              (o(!0), a(w, null, I(l.children, (i, y) => (o(), a("div", {
                key: y,
                class: s(["menu-children", { active: e.currentItem === i.label }]),
                onClick: (H) => e.changeActive(i)
              }, [
                n("span", R, r(i.label), 1)
              ], 10, O))), 128))
            ], 6)
          ]);
        }), 128))
      ], 4), [
        [A, e.showMenu]
      ])
    ])
  ], 2);
}
const J = /* @__PURE__ */ C(M, [["render", z]]);
export {
  J as default
};
