// @ths/design/es/components/trigger-menu/src/index.vue2.js
import { defineComponent as oe, computed as U, ref as h, reactive as ce, watch as g, onMounted as se, toRefs as ge, nextTick as ve } from "vue";
import de from "lodash/merge";
import Y from "lodash/cloneDeep";
import { getDomRandomClass as fe, useDebounce as he, getCssPropertyString as u, setStyle as r } from "../../../utils/index.js";
const Z = {
  label: "label",
  value: "value",
  children: "children"
}, Me = oe({
  name: "TTriggerMenu",
  props: {
    // 触发按钮文字排位方式
    triggerTextWriteMode: {
      type: String,
      default: "vertical"
    },
    // 标题按钮前部分图片
    titleFrontImg: {
      type: Object,
      default: () => ({})
    },
    // 标题按钮后部分图片
    titleBelowImg: {
      type: Object,
      default: () => ({})
    },
    // 标题按钮中部分图片
    titleMidImg: {
      type: Object,
      default: () => ({})
    },
    // 手风琴模式
    accordion: {
      type: Boolean,
      default: !1
    },
    // 已选中的菜单项
    activeMenuItem: {
      type: String,
      default: ""
    },
    // 位移距离
    translateX: {
      type: Number,
      default: 0
    },
    // 展开的菜单
    showedMenuItem: {
      type: String,
      default: ""
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => Y(Z),
      require: !1
    },
    // 默认展开菜单
    showMenuItem: {
      type: Boolean,
      default: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        title: {
          default: {},
          hover: {},
          active: {}
        },
        menu: {
          default: {}
        },
        menuCollapse: {
          default: {}
        },
        menuItem: {
          default: {},
          hover: {},
          active: {}
        },
        menuItemText: {
          default: {},
          hover: {},
          active: {}
        },
        menuItemArrow: {
          default: {},
          active: {}
        },
        menuContent: {
          default: {}
        },
        subMenuItem: {
          default: {},
          hover: {},
          active: {}
        },
        titleFrontImg: {
          default: {},
          active: {}
        },
        titleBelowImg: {
          default: {},
          active: {}
        },
        titleMidImg: {
          default: {},
          active: {}
        },
        titleFillImg: {
          default: {},
          active: {}
        },
        titleArrow: {
          default: {},
          active: {}
        }
      })
    }
  },
  emits: ["item-click"],
  setup(m, { emit: v }) {
    const t = U(() => m.cStyle), i = fe("trigger-menu"), I = h(), $ = h(), w = h(), l = ce({
      hiddenMenuItems: [],
      // 收起的菜单
      showMenuAnimation: !1,
      // 控制菜单的展开收起动画
      showMenu: !1,
      // 控制菜单的展开收起
      curData: [],
      // 菜单列表
      currentItem: "",
      // 选中的菜单项
      menuWidth: "",
      // 菜单项动态宽度
      menuHeight: "",
      // 菜单项动态高度
      imgOnloaded: !1,
      // 菜单按钮图片加载完毕
      titleTransition: !1,
      showAnimation: !1
    }), ee = he(() => {
      l.showMenu ? (l.showAnimation = !1, l.showMenuAnimation = !1, setTimeout(() => {
        l.showMenu = !1;
      }, 1e3)) : (l.showAnimation = !0, l.showMenu = !0, setTimeout(() => {
        l.showMenuAnimation = !0;
      }, 100));
    }, 200);
    g(() => m.showMenuItem, (e) => {
      e ? (l.showAnimation = !0, l.showMenu = !0, setTimeout(() => {
        l.showMenuAnimation = !0;
      }, 100)) : (l.showAnimation = !1, l.showMenuAnimation = !1, setTimeout(() => {
        l.showMenu = !1;
      }, 1e3));
    }, { immediate: !0 });
    const p = (e) => {
      try {
        const a = de(Z, m.alias);
        if (!Array.isArray(e))
          return;
        e.forEach((n, o) => {
          n.label = n[a.label], n.value = n[a.value], n.children = n[a.children], n != null && n.children && n.children.length && p(n.children);
        });
      } catch (a) {
        console.warn(a);
      }
    }, s = () => {
      ve(() => {
        const e = I.value;
        if (e) {
          l.menuWidth = `${e.clientWidth}px`;
          const a = $.value.clientHeight, n = w.value.clientHeight;
          e.style.paddingTop = `${a}px`, e.style.paddingBottom = `${n}px`, l.menuHeight = `${e.clientHeight}px`;
        }
      });
    }, te = (e) => {
      l.currentItem !== e.label && (l.currentItem = e.label, v("item-click", e)), s();
    }, le = U(() => (e) => {
    });
    function ie(e, a) {
      var c;
      const n = JSON.stringify(e), o = new RegExp(`label":"([^"]+)"[^\\}\\]]+label":"${a}`);
      return o.test(n) ? (c = n.match(o)) == null ? void 0 : c[1] : void 0;
    }
    const ne = (e) => {
      l.currentItem = e;
      const a = ie(l.curData, e);
      a && (l.hiddenMenuItems.push(a), l.hiddenMenuItems = [...new Set(l.hiddenMenuItems)]);
    }, M = (e) => {
      l.hiddenMenuItems[0] = e;
    }, ae = (e) => {
      e.children && e.children.length ? l.hiddenMenuItems.includes(e.label) ? l.hiddenMenuItems.splice(l.hiddenMenuItems.indexOf(e.label), 1) : m.accordion ? M(e.label) : l.hiddenMenuItems.push(e.label) : m.accordion ? (M(e.label), l.currentItem = e.label, s(), v("item-click", e)) : (l.hiddenMenuItems.push(e.label), l.currentItem = e.label, s(), v("item-click", e)), l.hiddenMenuItems = [...new Set(l.hiddenMenuItems)];
    };
    let d = 0;
    const ue = () => {
      l.imgOnloaded = !1, d++, d === 1 && (l.titleTransition = !0, d = 0, l.imgOnloaded = !0, s());
    }, re = () => {
      var a, n, o, c, y, A, S, T, C, O, F, k, x, B, D, H, R, j, W, N, q, E, L, P, _, z, J, V, X, G, K, Q;
      let e = "";
      if ((a = t.value.title) != null && a.hover && (e += `
          .${i} .trigger-title:hover {
            ${u((n = t.value.title) == null ? void 0 : n.hover)}
          }
        `), (o = t.value.title) != null && o.active && (e += `
          .${i} .trigger-title.title-active {
            ${u((c = t.value.title) == null ? void 0 : c.active)}
          }
        `), (y = t.value.menuItem) != null && y.hover && (e += `
          .${i} .trigger-menu .menu-collapse .menu-item:hover {
            ${u((A = t.value.menuItem) == null ? void 0 : A.hover)}
          }
        `), (S = t.value.menuItem) != null && S.active && (e += `
          .${i} .trigger-menu .menu-collapse .menu-item.active {
            ${u((T = t.value.menuItem) == null ? void 0 : T.active)}
          }
        `), (C = t.value.menuItemText) != null && C.hover && (e += `
          .${i} .trigger-menu .menu-collapse .menu-item:hover .menu-item-text {
            ${u((O = t.value.menuItemText) == null ? void 0 : O.hover)}
          }
        `), (F = t.value.menuItemText) != null && F.active && (e += `
          .${i} .trigger-menu .menu-collapse .menu-item.active .menu-item-text {
            ${u((k = t.value.menuItemText) == null ? void 0 : k.active)}
          }
        `), (x = t.value.menuItemArrow) != null && x.hover && (e += `
          .${i} .trigger-menu .menu-collapse .menu-item:hover .menu-item-arrow {
            ${u((B = t.value.menuItemArrow) == null ? void 0 : B.hover)}
          }
        `), (D = t.value.menuItemArrow) != null && D.active && (e += `
          .${i} .trigger-menu .menu-collapse .menu-item.active .menu-item-arrow {
            ${u((H = t.value.menuItemArrow) == null ? void 0 : H.active)}
          }
        `), (R = t.value.subMenuItem) != null && R.hover && (e += `
          .${i} .trigger-menu .menu-collapse .menu-content .menu-children:hover {
            ${u((j = t.value.subMenuItem) == null ? void 0 : j.hover)}
          }
        `), (W = t.value.subMenuItem) != null && W.active && (e += `
          .${i} .trigger-menu .menu-collapse .menu-content .menu-children.active {
            ${u((N = t.value.subMenuItem) == null ? void 0 : N.active)}
          }
        `), (q = t.value.titleFrontImg) != null && q.active && (e += `
          .${i} .trigger-title.title-active .title-front {
            ${u((E = t.value.titleFrontImg) == null ? void 0 : E.active)}
          }
        `), (L = t.value.titleBelowImg) != null && L.active && (e += `
          .${i} .trigger-title.title-active .title-below {
            ${u((P = t.value.titleBelowImg) == null ? void 0 : P.active)}
          }
        `), (_ = t.value.titleMidImg) != null && _.active && (e += `
          .${i} .trigger-title.title-active .title-mid {
            ${u((z = t.value.titleMidImg) == null ? void 0 : z.active)}
          }
        `), (J = t.value.titleFillImg) != null && J.active) {
        const f = ((X = (V = t.value.titleFillImg) == null ? void 0 : V.active) == null ? void 0 : X.backgroundImage) || "";
        e += `
          .${i} .trigger-title.title-active .title-fill {
            ${u((G = t.value.titleFillImg) == null ? void 0 : G.active)}
            ${f.indexOf("url(") === -1 ? `background-image: url(${f})` : `background-image: ${f}`}
          }
        `;
      }
      return (K = t.value.titleArrow) != null && K.active && (e += `
          .${i} .trigger-title .title-arrow.arrow-active {
            ${u((Q = t.value.titleArrow) == null ? void 0 : Q.active)}
          }
        `), e;
    }, me = () => {
      var a, n, o;
      let e = "";
      if (t.value.wrapper && (e += r(t.value.wrapper, `.${i}`)), t.value.title && (e += r(t.value.title, `.${i} .trigger-title`), e += r(t.value.title, `.${i} .relative .text`)), t.value.titleFrontImg && (e += r(t.value.titleFrontImg, `.${i} .trigger-title .title-front`)), t.value.titleBelowImg && (e += r(t.value.titleBelowImg, `.${i} .trigger-title .title-below`)), t.value.titleMidImg && (e += r(t.value.titleMidImg, `.${i} .trigger-title .title-mid`)), t.value.titleFillImg) {
        const c = ((n = (a = t.value.titleFillImg) == null ? void 0 : a.default) == null ? void 0 : n.backgroundImage) || "";
        e += `
          .${i} .trigger-title .title-fill {
            ${u((o = t.value.titleFillImg) == null ? void 0 : o.default)}
            ${c.indexOf("url(") === -1 ? `background-image: url(${c})` : `background-image: ${c}`}
          }
        `;
      }
      return t.value.titleArrow && (e += r(t.value.titleArrow, `.${i} .trigger-title .title-arrow`)), t.value.menu && (e += r(t.value.menu, `.${i} .trigger-menu`)), t.value.menuCollapse && (e += r(t.value.menuCollapse, `.${i} .trigger-menu .menu-collapse`)), t.value.menuItem && (e += r(t.value.menuItem, `.${i} .trigger-menu .menu-collapse .menu-item`)), t.value.menuItemText && (e += r(t.value.menuItemText, `.${i} .trigger-menu .menu-collapse .menu-item .menu-item-text`)), t.value.menuContent && (e += r(t.value.menuContent, `.${i} .trigger-menu .menu-collapse .menu-content`)), t.value.subMenuItem && (e += r(t.value.subMenuItem, `.${i} .trigger-menu .menu-collapse .menu-content .menu-children`)), e;
    }, b = () => {
      const e = document.querySelector(`.${i} .style-container`);
      e && (e.innerHTML = `<style>${me()}${re()}</style>`);
    };
    return se(() => {
      b();
    }), g([() => m.data, () => m.alias], () => {
      l.hiddenMenuItems = [], l.curData = Y(m.data), p(l.curData);
    }, {
      deep: !0,
      immediate: !0
    }), g(() => m.activeMenuItem, (e) => {
      ne(e), s();
    }, {
      immediate: !0
    }), g(() => m.cStyle, () => {
      b();
    }, { deep: !0 }), {
      ...ge(l),
      curStyle: t,
      domOnlyClass: i,
      changeShowMenu: ee,
      changeActive: te,
      toggleItemOverflow: ae,
      autoTextRef: I,
      bottomImgRef: w,
      topImgRef: $,
      onLoad: ue,
      toggleClick: le
    };
  }
});
export {
  Me as default
};
