// @ths/design/es/components/chart-excellent-rate/src/index.vue2.js
import { defineComponent as H, ref as Z, reactive as J, onMounted as K, onUnmounted as Q, watch as C, nextTick as W, toRefs as V } from "vue";
import * as X from "echarts";
import P from "lodash/merge";
const o = {
  // 颜色为默认颜色
  color: ["#25B200", "#0066FF", "#E0E0E0"],
  // tooltip提示气泡框
  tooltip: {
    show: !0,
    trigger: "item"
  },
  // 图表标题
  title: {
    show: !0,
    text: "",
    left: "center",
    top: "top",
    padding: [0, 0, 40, 0],
    textStyle: {
      color: "#000",
      fontSize: 12,
      fontWeight: "normal"
    }
  },
  // 图例
  legend: {
    show: !0,
    orient: "horizontal",
    bottom: 0,
    itemGap: 15,
    icon: "circle",
    itemWidth: 10,
    itemHeight: 10,
    textStyle: {
      color: "#333",
      fontSize: 12,
      fontWeight: "bold"
    },
    data: ["2020年同期", "2021年"]
  },
  dataZoom: [
    {
      // 禁用缩放
      disabled: !0,
      type: "inside",
      start: 0,
      end: 100,
      preventDefaultMouseMove: !0
    },
    {
      // 隐藏滑块
      show: !1,
      type: "slider",
      start: 0,
      end: 100,
      height: 12,
      borderColor: "transparent",
      backgroundColor: "#f8f8f8",
      dataBackground: {
        lineStyle: {
          width: 0
        },
        areaStyle: {
          color: "transparent"
        }
      },
      fillerColor: "#e1e1e1",
      handleIcon: "path://path://M100, 100m -75, 0a75,75 0 1,0 150,0a75,75 0 1,0 -150,0",
      handleStyle: {
        color: "#fff",
        shadowColor: "rgba(0, 0, 0, 0.2)",
        shadowBlur: 4
      },
      handleSize: "100%",
      textStyle: {
        color: "#888"
      },
      bottom: "10px"
    }
  ]
}, F = {
  type: "pie",
  selectedMode: "single",
  clockwise: !0,
  center: ["50%", "40%"],
  label: {
    show: !1
  },
  labelLine: {
    show: !1
  }
}, v = [
  {
    radius: ["0%", "50%"],
    ...F
  },
  {
    radius: ["60%", "70%"],
    ...F
  }
], p = H({
  name: "TChartExcellentRate",
  props: {
    // 传入图表数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      }),
      required: !1
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => o,
      required: !1
    },
    // 别名 及 样式设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 圆半径
        radius: {
          type: Array,
          default: () => [],
          required: !1
        },
        // 饼图颜色
        color: {
          type: Array,
          default: () => [],
          required: !1
        },
        // 饼图阴影颜色
        shadowColor: {
          type: Array,
          default: () => [],
          required: !1
        },
        // 饼图的中心（圆心）坐标
        center: {
          type: Array,
          default: () => [],
          required: !1
        },
        // 取data数据中的别名设置
        data: {
          lastExcellentNum: 0,
          thisExcellentNum: 0,
          lastPollutionNum: 0,
          thisPollutionNum: 0
        }
      }),
      required: !1
    },
    // 页面展示的数据
    // 标题
    title: {
      type: String,
      default: "鄂尔多斯空气优良率同比分析",
      required: !1
    },
    // 截止日期
    deadline: {
      type: String,
      default: "2021年04月19日",
      required: !1
    },
    // 详情链接
    detailUrl: {
      type: String,
      default: "http://www.baidu.com",
      required: !1
    },
    // 标题图片
    titleIcon: {
      type: String,
      default: "",
      required: !1
    },
    excellentIcon: {
      type: String,
      default: "",
      required: !1
    },
    // 主题风格
    themeStyle: {
      type: String,
      default: "light",
      required: !1
    }
  },
  emits: ["legend-select"],
  setup(e, { emit: U }) {
    function _(t) {
      window.open(t);
    }
    const b = Z(null);
    let d = null;
    const i = J({
      // 页面展示数据
      pageData: {
        excellentRate: "--",
        lastExcellentDays: "--",
        thisExcellentDays: "--",
        thisPollutionDays: "--",
        compareDay: "--"
      }
    });
    function E() {
      var h, g, y, S;
      const { series: t } = e.options;
      if (t)
        i.pageData = {
          lastExcellentDays: ((h = t[1]) == null ? void 0 : h.value[1]) >= 0 ? t[1].value[1] : 0,
          thisExcellentDays: ((g = t[0]) == null ? void 0 : g.value[1]) >= 0 ? t[0].value[1] : 0,
          lastPollutionDays: ((y = t[1]) == null ? void 0 : y.value[0]) >= 0 ? t[1].value[0] : 0,
          thisPollutionDays: ((S = t[0]) == null ? void 0 : S.value[0]) >= 0 ? t[0].value[0] : 0
        };
      else if (e.data) {
        const f = e.data[e.seriesSetting.data.lastExcellentNum], m = e.data[e.seriesSetting.data.thisExcellentNum], w = e.data[e.seriesSetting.data.thisPollutionNum], D = e.data[e.seriesSetting.data.lastPollutionNum];
        i.pageData = {
          lastExcellentDays: f >= 0 ? f : 0,
          thisExcellentDays: m >= 0 ? m : 0,
          thisPollutionDays: w >= 0 ? w : 0,
          lastPollutionDays: D >= 0 ? D : 0
        };
      }
      const { lastExcellentDays: l, thisExcellentDays: a, thisPollutionDays: u } = i.pageData;
      let r = Math.round(a / (a + u) * 100);
      Number.isNaN(r) && (r = 0);
      let s = l - a, n = "";
      s > 0 ? n = "减少" : (n = "增加", s = Math.abs(s));
      const x = {
        excellentRate: r,
        days: s,
        compareDay: n
      };
      i.pageData = P(
        i.pageData,
        x
      );
    }
    function G() {
      e.themeStyle === "dark" ? o.legend.textStyle.color = "#fff" : e.themeStyle === "light" && (o.legend.textStyle.color = "#333");
    }
    function c() {
      var r, s, n, x, h, g, y, S, f, m, w, D, q, B, z, M, O, k, j, A, I, R, L, T;
      let t = [];
      const { series: l, legend: a } = e.options;
      l ? t = [
        {
          ...v[0],
          radius: l[0].radius || ["0%", "50%"],
          center: l[0].center || ["50%", "45%"],
          data: [
            {
              value: i.pageData.thisPollutionDays,
              itemStyle: {
                color: ((r = l[0].itemStyle) == null ? void 0 : r.color[1]) || o.color[2],
                shadowBlur: 12,
                shadowColor: (s = l[0].itemStyle) != null && s.shadowColor ? l[0].itemStyle.shadowColor[1] : "rgba(224, 224, 224, 0.5)"
              },
              emphasis: {
                scale: !1,
                itemStyle: {
                  color: "inherit"
                }
              }
            },
            {
              name: ((n = a == null ? void 0 : a.data) == null ? void 0 : n[1]) || o.legend.data[1],
              value: i.pageData.thisExcellentDays,
              itemStyle: {
                color: ((x = l[0].itemStyle) == null ? void 0 : x.color[0]) || o.color[0],
                shadowBlur: 12,
                shadowColor: (h = l[0].itemStyle) != null && h.shadowColor ? l[0].itemStyle.shadowColor[0] : "rgba(37, 178, 0, 0.7)"
              }
            }
          ]
        },
        {
          ...v[1],
          radius: l[1].radius || ["60%", "70%"],
          center: l[1].center || ["50%", "45%"],
          data: [
            {
              value: i.pageData.lastPollutionDays,
              itemStyle: {
                color: ((g = l[1].itemStyle) == null ? void 0 : g.color[1]) || o.color[2],
                shadowBlur: 12,
                shadowColor: (y = l[1].itemStyle) != null && y.shadowColor ? l[1].itemStyle.shadowColor[1] : "rgba(224, 224, 224, 0.5)"
              },
              emphasis: {
                scale: !1,
                itemStyle: {
                  color: "inherit"
                }
              }
            },
            {
              name: ((S = a == null ? void 0 : a.data) == null ? void 0 : S[0]) || o.legend.data[0],
              value: i.pageData.lastExcellentDays,
              itemStyle: {
                color: ((f = l[1].itemStyle) == null ? void 0 : f.color[0]) || o.color[1],
                shadowBlur: 12,
                shadowColor: (m = l[1].itemStyle) != null && m.shadowColor ? l[1].itemStyle.shadowColor[0] : "rgba(0, 102, 255, 0.7)"
              }
            }
          ]
        }
      ] : e.data && (t = [
        {
          ...v[0],
          radius: (w = e.seriesSetting.radius) != null && w.length ? e.seriesSetting.radius[0] : ["0%", "50%"],
          center: (D = e.seriesSetting.center) != null && D.length ? e.seriesSetting.center : ["50%", "45%"],
          data: [
            {
              value: i.pageData.thisPollutionDays,
              itemStyle: {
                color: (q = e.seriesSetting.color) != null && q.length ? e.seriesSetting.color[2] : o.color[2],
                shadowBlur: 12,
                shadowColor: (B = e.seriesSetting.shadowColor) != null && B.length ? e.seriesSetting.shadowColor[2] : "rgba(224, 224, 224, 0.5)"
              },
              emphasis: {
                scale: !1,
                itemStyle: {
                  color: "inherit"
                }
              }
            },
            {
              name: ((z = a == null ? void 0 : a.data) == null ? void 0 : z[1]) || o.legend.data[1],
              value: i.pageData.thisExcellentDays,
              itemStyle: {
                color: (M = e.seriesSetting.color) != null && M.length ? e.seriesSetting.color[0] : o.color[0],
                shadowBlur: 12,
                shadowColor: (O = e.seriesSetting.shadowColor) != null && O.length ? e.seriesSetting.shadowColor[0] : " rgba(37, 178, 0, 0.7)"
              }
            }
          ]
        },
        {
          ...v[1],
          radius: (k = e.seriesSetting.radius) != null && k.length ? e.seriesSetting.radius[1] : ["60%", "70%"],
          center: (j = e.seriesSetting.center) != null && j.length ? e.seriesSetting.center : ["50%", "45%"],
          data: [
            {
              value: i.pageData.lastPollutionDays,
              itemStyle: {
                color: (A = e.seriesSetting.color) != null && A.length ? e.seriesSetting.color[2] : o.color[2],
                shadowBlur: 12,
                shadowColor: (I = e.seriesSetting.shadowColor) != null && I.length ? e.seriesSetting.shadowColor[2] : "rgba(224, 224, 224, 0.5)"
              },
              emphasis: {
                scale: !1,
                itemStyle: {
                  color: "inherit"
                }
              }
            },
            {
              name: ((R = a == null ? void 0 : a.data) == null ? void 0 : R[0]) || o.legend.data[0],
              value: i.pageData.lastExcellentDays,
              itemStyle: {
                color: (L = e.seriesSetting.color) != null && L.length ? e.seriesSetting.color[1] : o.color[1],
                shadowBlur: 12,
                shadowColor: (T = e.seriesSetting.shadowColor) != null && T.length ? e.seriesSetting.shadowColor[1] : "rgba(0, 102, 255, 0.7)"
              }
            }
          ]
        }
      ]), G();
      let u = P(
        {
          series: t
        },
        o
      );
      u = P(u, e.options), d.setOption(u);
    }
    function N() {
      W(() => {
        d.resize();
      });
    }
    return K(() => {
      b.value && (d = X.init(b.value), E(), c(), d.on("legendselectchanged", (t) => {
        U("legend-select", t);
      })), window.addEventListener("resize", N);
    }), Q(() => {
      window.removeEventListener("resize", N);
    }), C(
      () => e.options,
      (t) => {
        E(), c();
      },
      {
        deep: !0
      }
    ), C(
      () => e.data,
      (t) => {
        E(), c();
      },
      {
        deep: !0
      }
    ), C(
      () => e.themeStyle,
      () => {
        c();
      },
      {
        deep: !0
      }
    ), C(
      () => e.cStyle,
      () => {
        W(() => {
          d.resize();
        });
      },
      {
        deep: !0
      }
    ), C(
      () => e.seriesSetting,
      () => {
        c();
      },
      {
        deep: !0
      }
    ), {
      ...V(i),
      chartId: b,
      goToPage: _
    };
  }
});
export {
  p as default
};
