// @ths/design/es/components/chart-bar-across/src/index.vue2.js
import { defineComponent as ne, computed as V, ref as re, reactive as oe, onMounted as se, onUnmounted as ye, watch as I, nextTick as Z, toRefs as ce } from "vue";
import * as fe from "echarts";
import c from "lodash/merge";
import z from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as me, DEFAULT_OPTIONS_AXIS as ue } from "../../../utils/constant.js";
import { getDomRandomClass as de, setStyle as L } from "../../../utils/index.js";
const Se = c(z(me), z(ue.xAxis)), y = {
  markData: 60,
  type: "dashed",
  color: "rgba(255, 166, 79, 1)",
  fontOpacity: 1,
  fontSize: 14,
  lineOpacity: 1,
  lineWidth: 1
}, A = c(
  z(Se),
  {
    legend: {
      show: !1
    },
    tooltip: {
      trigger: "axis"
    },
    xAxis: {
      type: "value",
      splitLine: {
        lineStyle: {
          type: "dashed",
          color: "rgba(216, 216, 216, 0.5)"
        }
      },
      axisLabel: {
        fontSize: 12,
        color: "rgba(255, 255, 255, 0.8)"
      }
    },
    yAxis: c({}, {
      type: "category",
      inverse: !0,
      axisTick: {
        show: !1
      },
      axisLabel: {
        fontSize: 12,
        lineHeight: 15,
        overflow: "break",
        color: "#666",
        opacity: 0.6
      }
    })
  }
), ge = {
  name: "条",
  type: "bar",
  barWidth: 6,
  borderRadius: [0, 0, 20, 0],
  itemStyle: {
    color: {
      type: "linear",
      x: 0,
      y: 0,
      x2: 1,
      y2: 0,
      colorStops: [
        {
          offset: 0,
          color: "rgb(252, 237, 37,0)"
        },
        {
          offset: 0.3,
          color: "rgb(252, 237, 37,0.8)"
        },
        {
          offset: 1,
          color: "rgb(252, 237, 37,1)"
        }
      ],
      global: !1
      // 缺省为 false
    }
  },
  emphasis: {
    label: {
      show: !0,
      width: 10,
      height: 10,
      backgroundColor: "#538FFF",
      position: "right",
      distance: -10,
      fontSize: 16,
      borderRadius: 6,
      shadowColor: "#538FFF",
      shadowBlur: 8,
      formatter(e) {
        return "";
      }
    }
  }
}, we = ne({
  name: "TChartBarAcross",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => {
      },
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 开启或关闭标记线
    isMark: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // Y 轴类目取resData数据中的别名设置
        yAxisLeftLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    markStyle: {
      type: Object,
      default: () => y,
      required: !1
    },
    // 单位
    unit: {
      type: String,
      default: ""
    },
    bar: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["legend-select"],
  setup(e, { emit: ee }) {
    var W, B, N, U, H, P, p, K, X;
    const f = V(() => e.cStyle), u = de("chart-bar-across"), x = re(null);
    let d = null;
    const S = oe({
      // 图表数据
      chartData: []
    }), $ = {
      show: !1,
      type: "line",
      symbol: "none",
      inverse: !0,
      silent: !0,
      data: [((W = e.markStyle) == null ? void 0 : W.markData) || y.markData],
      tooltip: {
        show: !1
      },
      markLine: {
        data: [{ type: "average" }],
        symbol: ["none", "none"],
        // 去掉箭头
        itemStyle: {
          color: ((B = e.markStyle) == null ? void 0 : B.color) || y.color
        },
        lineStyle: {
          type: ((N = e.markStyle) == null ? void 0 : N.type) || y.type,
          opacity: ((U = e.markStyle) == null ? void 0 : U.lineOpacity) || y.lineOpacity,
          width: ((H = e.markStyle) == null ? void 0 : H.lineWidth) || y.lineWidth
        },
        label: {
          show: !0,
          position: "start",
          formatter: `${((P = e.markStyle) == null ? void 0 : P.markData) || y.markData}${e.unit}`,
          fontSize: ((p = e.markStyle) == null ? void 0 : p.fontSize) || y.fontSize,
          opacity: ((K = e.markStyle) == null ? void 0 : K.fontOpacity) || y.fontOpacity,
          color: ((X = e.markStyle) == null ? void 0 : X.color) || y.color
        }
      }
    };
    function T() {
      var t, i;
      (i = (t = e.seriesSetting) == null ? void 0 : t.data) != null && i.length && e.data && (S.chartData = e.seriesSetting.data.map((a) => {
        const { name: r, dataKey: n } = a;
        return {
          name: r || "--",
          data: e.data.map((l) => l[n])
        };
      }));
    }
    const te = () => {
      let t = "";
      return f.value.wrapper && (t += L(f.value.wrapper, `.${u}`)), f.value.emptyContainer && (t += L(f.value.emptyContainer, `.${u} .empty`)), f.value.emptyImg && (t += L(f.value.emptyImg, `.${u} .empty .empty-img`)), f.value.emptyText && (t += L(f.value.emptyText, `.${u} .empty .empty-text`)), t;
    }, R = () => {
      const t = document.querySelector(`.${u} .style-container`);
      t && (t.innerHTML = `<style>${te()}</style>`);
    }, ae = V(() => {
      var t, i, a, r;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((l) => {
            var m;
            return (l == null ? void 0 : l.data) && ((m = l == null ? void 0 : l.data) == null ? void 0 : m.length) !== 0;
          });
        if (typeof ((i = e.options) == null ? void 0 : i.series) == "object")
          return ((r = (a = e.options.series) == null ? void 0 : a.data) == null ? void 0 : r.length) === 0;
      }
      return !!e.data.length;
    });
    function g(t, i) {
      let a = t == null ? void 0 : t.toLowerCase();
      if (a && a.slice(0, 1) === "#") {
        if (a.length === 4) {
          let n = "#";
          for (let l = 1; l < 4; l += 1)
            n += a.slice(l, l + 1).concat(a.slice(l, l + 1));
          a = n;
        }
        const r = [];
        for (let n = 1; n < 7; n += 2)
          r.push(parseInt(`0x${a.slice(n, n + 2)}`, 16));
        return `rgba(${r.join(",")},${i})`;
      }
      return a;
    }
    function M(t, i) {
      const a = z(ge);
      return c(
        {
          ...a,
          itemStyle: {
            color: {
              type: "linear",
              x: 0,
              y: 0,
              x2: 1,
              y2: 0,
              colorStops: [
                {
                  offset: 0,
                  color: g(t == null ? void 0 : t[i], 0) || g(A.color[i], 0) || "#FFF"
                },
                {
                  offset: 0.5,
                  color: g(t == null ? void 0 : t[i], 0.8) || g(A.color[i], 0.8) || "#FFF"
                },
                {
                  offset: 1,
                  color: g(t == null ? void 0 : t[i], 1) || g(A.color[i], 1) || "#FFF"
                }
              ],
              global: !1
              // 缺省为 false
            },
            borderRadius: [0, 0, 20, 0]
          }
        },
        e == null ? void 0 : e.bar
      );
    }
    function ie(t) {
      var a;
      let i = [];
      return t ? i = (a = e.data) != null && a.length ? e.data.map((r) => r[t]) : [] : i = e.data.map((r, n) => n), i;
    }
    function _(t) {
      var i, a, r, n, l, m, h, k, b, s, o, E, O, D, w, v, C, F;
      return c($, {
        show: !1,
        type: "line",
        symbol: "none",
        inverse: !0,
        silent: !0,
        data: [((i = t.markStyle) == null ? void 0 : i.markData) || ((a = e.markStyle) == null ? void 0 : a.markData)],
        tooltip: {
          show: !1
        },
        markLine: {
          data: [{ type: "average" }],
          symbol: ["none", "none"],
          // 去掉箭头
          itemStyle: {
            color: ((r = t.markStyle) == null ? void 0 : r.color) || ((n = e.markStyle) == null ? void 0 : n.color)
          },
          lineStyle: {
            type: ((l = t.markStyle) == null ? void 0 : l.type) || ((m = e.markStyle) == null ? void 0 : m.type),
            opacity: ((h = t.markStyle) == null ? void 0 : h.lineOpacity) || ((k = e.markStyle) == null ? void 0 : k.lineOpacity),
            width: ((b = t.markStyle) == null ? void 0 : b.lineWidth) || ((s = e.markStyle) == null ? void 0 : s.lineWidth)
          },
          label: {
            show: !0,
            position: "start",
            formatter: `${((o = t.markStyle) == null ? void 0 : o.markData) || ((E = e.markStyle) == null ? void 0 : E.markData)}${e.unit}`,
            fontSize: ((O = t.markStyle) == null ? void 0 : O.fontSize) || ((D = e.markStyle) == null ? void 0 : D.fontSize),
            opacity: ((w = t.markStyle) == null ? void 0 : w.fontOpacity) || ((v = e.markStyle) == null ? void 0 : v.fontOpacity),
            color: ((C = t.markStyle) == null ? void 0 : C.color) || ((F = e.markStyle) == null ? void 0 : F.color)
          }
        }
      });
    }
    function j() {
      var l, m, h, k, b;
      const { color: t, series: i } = e.options;
      let a = [], r = [];
      i ? (a = i.map((s, o) => c(M(t, o), s)), e.isMark && a.push(_($))) : (l = e.data) != null && l.length && ((h = (m = e.seriesSetting) == null ? void 0 : m.data) != null && h.length) && (a = (k = e.seriesSetting) == null ? void 0 : k.data.map((s, o) => {
        var O, D, w, v, C, F, Y, G, J, Q;
        if ((s == null ? void 0 : s.highLight) === "img") {
          const le = {
            emphasis: {
              label: {
                backgroundColor: {
                  image: (v = (w = (D = (O = e.seriesSetting) == null ? void 0 : O.data) == null ? void 0 : D[o]) == null ? void 0 : w.img) == null ? void 0 : v.backgroundColor
                }
              }
            }
          };
          return c({
            ...M(t, o),
            name: ((C = S.chartData[o]) == null ? void 0 : C.name) || "",
            data: S.chartData[o].data || []
          }, (G = (Y = (F = e.seriesSetting) == null ? void 0 : F.data) == null ? void 0 : Y[o]) == null ? void 0 : G.style, le);
        }
        return c({
          ...M(t, o),
          name: ((J = S.chartData[o]) == null ? void 0 : J.name) || "",
          data: ((Q = S.chartData[o]) == null ? void 0 : Q.data) || []
        }, s == null ? void 0 : s.style);
      }), r = ie((b = e.seriesSetting) == null ? void 0 : b.yAxisLeftLabel), e.isMark && a.push(_($)));
      let n = c(
        {
          series: a,
          yAxis: {
            data: r
          }
        },
        A
      );
      n = c(n, e.options), d.clear(), d.setOption(n);
    }
    function q() {
      Z(() => {
        d.resize();
      });
    }
    return se(() => {
      R(), x.value && (d = fe.init(x.value), T(), j(), d.on("legendselectchanged", (t) => {
        ee("legend-select", t);
      })), window.addEventListener("resize", q);
    }), ye(() => {
      window.removeEventListener("resize", q);
    }), I(
      [() => e.options, () => e.data, () => e.seriesSetting, () => e.markStyle],
      () => {
        T(), j();
      },
      {
        deep: !0
      }
    ), I(
      [() => e.unit, () => e.bar, () => e.isMark],
      () => {
        j();
      }
    ), I(
      () => e.cStyle,
      () => {
        R(), Z(() => {
          d.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      ...ce(S),
      chartId: x,
      dataProcessing: T,
      checkEmpty: ae,
      domOnlyClass: u
    };
  }
});
export {
  we as default
};
