// @ths/design/es/components/text/src/index.vue.js
import o from "./index.vue2.js";
import { openBlock as d, createElementBlock as t, normalizeStyle as r, toDisplayString as s } from "vue";
import "./index.vue3.js";
import i from "../../../_virtual/_plugin-vue_export-helper.js";
const p = ["contenteditable", "title"];
function u(e, n, a, m, y, b) {
  return d(), t("div", {
    ref: "textRef",
    class: "t-text",
    style: r(e.customStyle),
    contenteditable: e.editableElements.label.editable,
    title: e.showTitle ? e.label : "",
    onDblclick: n[0] || (n[0] = (...l) => e.handleDbclick && e.handleDbclick(...l)),
    onInput: n[1] || (n[1] = (...l) => e.handleInput && e.handleInput(...l)),
    onBlur: n[2] || (n[2] = (...l) => e.handleBlur && e.handleBlur(...l)),
    onMousedown: n[3] || (n[3] = (...l) => e.handleMouseDown && e.handleMouseDown(...l)),
    onMouseenter: n[4] || (n[4] = (l) => e.idHover = !0),
    onMouseleave: n[5] || (n[5] = (l) => e.idHover = !1),
    onKeydown: n[6] || (n[6] = (...l) => e.handleKeydown && e.handleKeydown(...l)),
    onKeyup: n[7] || (n[7] = (...l) => e.handleKeyup && e.handleKeyup(...l)),
    onCopy: n[8] || (n[8] = (...l) => e.handleCopy && e.handleCopy(...l)),
    onPaste: n[9] || (n[9] = (...l) => e.handlePaste && e.handlePaste(...l))
  }, s(e.editableElements.label.value), 45, p);
}
const K = /* @__PURE__ */ i(o, [["render", u]]);
export {
  K as default
};
