// @ths/design/es/components/text/src/index.vue2.js
import { defineComponent as T, ref as B, reactive as D, watch as P, computed as k, toRefs as C, nextTick as c } from "vue";
const j = T({
  name: "TText",
  props: {
    // 文本内容
    label: {
      type: [String, Number, Boolean],
      default: ""
    },
    // 是否显示标题
    showTitle: {
      type: Boolean,
      default: !1
    },
    // 要修改的字段名
    editKey: {
      type: String,
      default: "label"
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {},
          focus: {
            cursor: "text"
          }
        }
      })
    },
    // 是否显示双击编辑提示 默认显示 例如矩形组件双击编辑文本 失去焦点时不需要显示默认文本
    tip: {
      type: Boolean,
      default: !0
    },
    // 在 body 上设置为不能编辑的情况下 文本是否可以编辑
    editable: {
      type: Boolean,
      default: !1
    }
  },
  // 修改文本事件
  emits: ["modify-text", "blur-text"],
  setup(a, { emit: o }) {
    const d = B(null), t = D({
      // 组件被鼠标滑过状态 用于修改组件滑入滑出样式
      idHover: !1,
      // 可以编辑的元素列表
      editableElements: {
        // 可编辑的变量在props中定义的名称
        label: {
          // 可编辑状态
          editable: !1,
          // 变量值
          value: a.label,
          // 输入框变更时临时存储数据的变量
          template: a.label
        }
      }
    });
    P(() => a.label, () => {
      t.editableElements.label.value = a.label, t.editableElements.label.template = a.label;
    });
    const f = k(() => {
      var b, u, s;
      const { hover: e, focus: l } = (b = a.cStyle) == null ? void 0 : b.wrapper, n = { ...(s = (u = a.cStyle) == null ? void 0 : u.wrapper) == null ? void 0 : s.default }, w = { ...n, ...l }, v = { ...n, ...e };
      let i = n;
      return t.idHover && e && Object.keys(e).length && (i = v), t.editableElements.label.editable && l && Object.keys(l).length && (i = w), i;
    });
    function r() {
      const e = {
        [a.editKey]: t.editableElements.label
      };
      o("modify-text", e);
    }
    function m(e) {
      var l;
      t.editableElements.label.template = (l = e == null ? void 0 : e.target) == null ? void 0 : l.innerText;
    }
    function y() {
      o("blur-text"), t.editableElements.label.editable && (t.editableElements.label.editable = !1, t.editableElements.label.value = "", c(() => {
        t.editableElements.label.value = t.editableElements.label.template || a.tip && "双击编辑文本" || "", r();
      }));
    }
    function p(e) {
      t.editableElements.label.editable && e.stopPropagation();
    }
    function h(e) {
      e.stopPropagation();
    }
    function x(e) {
      t.editableElements.label.editable && e.stopPropagation();
    }
    function E(e) {
      e.stopPropagation();
    }
    function g(e) {
      e.preventDefault();
      const l = e.clipboardData, n = (l == null ? void 0 : l.getData("text/plain")) || "";
      n !== "" && document.execCommand("insertText", !1, n);
    }
    function S() {
      const e = document.querySelector("body");
      !(e != null && e.getAttribute("editable")) && !a.editable || (t.editableElements.label.editable = !0, c(() => {
        var l;
        (l = d.value) == null || l.focus();
      }));
    }
    return {
      textRef: d,
      customStyle: f,
      ...C(t),
      handleInput: m,
      handleBlur: y,
      handleMouseDown: p,
      handleKeydown: h,
      handleKeyup: x,
      handleCopy: E,
      handlePaste: g,
      handleDbclick: S
    };
  }
});
export {
  j as default
};
