// @ths/design/es/components/chart-bar-base/src/index.vue2.js
import { defineComponent as T, ref as h, computed as g, watch as y, nextTick as S, onMounted as O } from "vue";
import * as B from "echarts";
import { splitArray as W, getStyle as m, rgb2rgba as x } from "../../../utils/index.js";
import { POLLUTION_HTML as V } from "../../../utils/constant.js";
var w = /* @__PURE__ */ ((e) => (e.LIGHT = "light", e.DARK = "dark", e))(w || {});
const C = [
  "#43d401",
  "#ffea00",
  "#00f6ff",
  "#be4cff",
  "#0167ff",
  "#fd7800",
  "#5c25d2",
  "#ffb540"
], _ = {
  light: {
    axisLine: "rgba(202, 202, 202, .8)",
    lineColor: "rgba(202, 202, 202, .3)",
    fontColor: "#666",
    tooltipBackgroundColor: "rgba(255, 255, 255, 1)",
    tooltipFontColor: "#666"
  },
  dark: {
    axisLine: "rgba(202, 202, 202, .3)",
    lineColor: "rgba(202, 202, 202, .1)",
    fontColor: "#fff",
    tooltipBackgroundColor: "rgba(34, 68, 99, 1)",
    tooltipFontColor: "#fff"
  }
}, b = {
  root: {},
  wrapper: {
    default: {}
  },
  chart: {
    default: {}
  },
  legendRoot: {
    default: {}
  },
  legendItem: {
    default: {}
  },
  legendIcon: {
    default: {}
  },
  legendName: {
    default: {}
  }
}, N = T({
  name: "TChartBarBase",
  props: {
    // 数据
    data: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    // 横坐标数据
    dataX: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    // 每个系列命名
    names: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    // 整个图形宽度
    width: {
      type: String,
      default: "100%",
      required: !1
    },
    // 整个图形高度
    height: {
      type: String,
      default: "300px",
      required: !1
    },
    // 整个图形命名
    title: {
      type: String,
      default: "",
      required: !1
    },
    // 图像颜色
    colors: {
      type: Array,
      default: () => C,
      required: !1,
      validator: (e) => !e.some((d) => !/#[0-9a-fA-F]{6}/.test(d.toString()))
    },
    // 柱形宽度
    barWidth: {
      type: String,
      default: void 0,
      required: !1
    },
    // 不同系列间距
    barGap: {
      type: String,
      default: "30%",
      required: !1
    },
    // 数据单位
    unit: {
      type: String,
      default: "",
      required: !1
    },
    // 标线
    markLine: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 显示默认图例
    showLegend: {
      type: Boolean,
      default: !0,
      required: !1
    },
    legendIcon: {
      type: String,
      default: "default"
    },
    // 默认图例排列方式
    legendStyle: {
      type: Object,
      default: () => ({
        position: "top",
        align: "center",
        fontSize: 12,
        iconSize: 12,
        col: 1
      }),
      required: !1,
      validator: (e) => !!e
    },
    // 显示默认tooltip
    showTooltip: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 柱形上是否显示数据
    showValue: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 区域锁定
    zoomLock: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 横向显示区域
    scope: {
      type: Array,
      default: () => ["0%", "100%"],
      required: !1,
      validator: (e) => e.length === 2 ? /%/.test(e[0].toString()) ? /%/.test(e[1].toString()) : !/%/.test(e[1].toString()) : !1
    },
    // 数据极值区间
    range: {
      type: Array,
      default: () => [0, void 0],
      required: !1,
      validator: (e) => !(e.length > 2)
    },
    yIndexs: {
      type: Array,
      default: () => [],
      required: !1
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({
        title: {},
        grid: {},
        xAxis: {},
        yAxis: {},
        tooltip: {},
        color: null,
        dataZoom: void 0
      }),
      required: !1
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => b,
      required: !1,
      validator: (e) => !!e
    },
    // 主题风格深浅
    themeStyle: {
      type: String,
      default: "light",
      validator: (e) => e.toUpperCase() in w
    }
  },
  emits: ["col-click"],
  setup(e, d) {
    const f = h(null), n = h({
      width: e.width,
      height: e.height
    }), L = g(() => ({
      ...b,
      ...e.cStyle
    })), u = (a) => a ? String(a).split(" ").map((o) => `${o}px`).join(" ") : "", A = g(() => ({
      position: e.legendStyle.position || "top",
      align: e.legendStyle.align || "center",
      margin: u(e.legendStyle.margin),
      fontSize: e.legendStyle.fontSize ? u(e.legendStyle.fontSize) : "14px",
      iconSize: e.legendStyle.iconSize ? u(e.legendStyle.iconSize) : "14px",
      cStyle: e.legendStyle.cStyle
    })), v = () => /%/.test(e.scope[0].toString()) ? {
      start: Number(e.scope[0].toString().replace(/%/, "")),
      end: Number(e.scope[1].toString().replace(/%/, ""))
    } : {
      startValue: e.scope[0],
      endValue: e.scope[1]
    }, l = () => _[e.themeStyle], k = (a) => {
      const t = a.toUpperCase().replace(/[-.+_]/g, ""), o = V.filter((i) => i.name === t);
      return o.length > 0 ? o[0].html : a;
    }, z = (a, t = C) => [...a].map((i, c) => ({
      ...i,
      name: k(i),
      style: {
        color: t[c % t.length],
        legend: {
          nameColor: l().fontColor,
          backgroundColor: t[c % t.length],
          borderColor: x(t[c % t.length], 0.5)
        }
      }
    })), q = (a) => {
      const t = {
        dataX: a.name,
        dataIndex: a.dataIndex,
        seriesIndex: a.seriesIndex,
        data: e.data.map((o, i) => ({
          name: e.names[i],
          value: o[a.dataIndex]
        }))
      };
      d.emit("col-click", t);
    };
    let r = null;
    const I = g(() => W(z(e.names, e.colors), e.legendStyle.col || 1));
    function s() {
      if (f.value) {
        r || (r = B.init(f.value));
        const a = {
          type: "inside",
          zoomLock: e.zoomLock,
          // 锁定区域：只能平移，不能放大缩小
          ...v()
        }, t = {
          ...e.options,
          dataZoom: e.options.dataZoom || [a],
          grid: {
            top: "16%",
            left: 10,
            right: 10,
            bottom: "15%",
            containLabel: !0,
            ...e.options.grid
          },
          tooltip: {
            show: e.showTooltip,
            trigger: "axis",
            confine: !0,
            backgroundColor: l().tooltipBackgroundColor,
            borderColor: l().tooltipBackgroundColor,
            textStyle: {
              color: l().fontColor
            },
            axisPointer: {
              type: "line",
              lineStyle: {
                color: x(l().lineColor, 0.4)
              }
            },
            ...e.options.tooltip
          },
          xAxis: {
            type: "category",
            axisLabel: {
              fontSize: 12,
              color: l().fontColor,
              align: "center"
            },
            axisLine: {
              show: !0,
              lineStyle: {
                color: l().axisLine
              }
            },
            axisTick: { show: !1 },
            data: e.dataX,
            ...e.options.xAxis
          },
          yAxis: Array.isArray(e.options.yAxis) ? e.options.yAxis : [{
            type: "value",
            min: e.range[0],
            max: e.range[1],
            name: e.unit || "",
            nameTextStyle: {
              align: "center",
              fontSize: 12,
              color: l().fontColor,
              fontWeight: "normal",
              padding: [0, 0, 0, 10]
            },
            axisLabel: {
              show: !0,
              fontSize: 12,
              color: l().fontColor
            },
            axisLine: {
              show: !0,
              lineStyle: {
                color: l().axisLine
              }
            },
            splitLine: {
              show: !1,
              lineStyle: {
                color: [l().lineColor]
              }
            },
            ...e.options.yAxis
          }],
          color: e.options.color || e.colors,
          series: e.data.map((o, i) => ({
            name: e.names[i],
            yAxisIndex: e.yIndexs.length > 0 ? e.yIndexs[i] : 0,
            type: "bar",
            data: o,
            label: {
              show: !!e.showValue[i],
              color: l().fontColor,
              position: "top"
            },
            barWidth: e.barWidth,
            barGap: e.barGap,
            markLine: e.markLine[i] ? {
              silent: !0,
              symbol: ["none"],
              label: { show: !1 },
              lineStyle: e.markLine[i].lineStyle,
              data: [{
                name: "",
                // TODO: 标注线标题
                yAxis: e.markLine[i].data
              }]
            } : void 0,
            ...e.options.barSeries
          }))
        };
        e.title && (t.title = {
          text: e.title,
          left: "50%",
          padding: [0, 5, 0, 5],
          textAlign: "center",
          textStyle: {
            color: l().fontColor,
            fontSize: 16,
            fontWeight: "bold"
          },
          ...e.options.title
        }), e.legendIcon !== "default" && (t.legend = {
          show: e.showLegend,
          icon: e.legendIcon,
          textStyle: {
            color: l().fontColor,
            fontSize: e.legendStyle.fontSize
          },
          itemWidth: e.legendStyle.iconSize,
          itemHeight: e.legendStyle.iconSize,
          [e.legendStyle.position]: "6%",
          left: e.legendStyle.align
        }), r.setOption(t), r.on("click", (o) => {
          q(o);
        }), window.onresize = () => {
          r.resize();
        };
      }
    }
    return y(() => [e.width, e.height], () => {
      n.value = { ...n.value, ...m({ width: e.width }) }, n.value = { ...n.value, ...m({ width: e.height }) }, S(() => {
        r.resize();
      });
    }), y(() => [e.cStyle], () => {
      e.cStyle.wrapper.default.width && (n.value.width = e.cStyle.wrapper.default.width), e.cStyle.wrapper.default.height && (n.value.height = e.cStyle.wrapper.default.height), S(() => {
        r.resize();
      });
    }), y(() => [
      e.data,
      e.dataX,
      e.names,
      e.title,
      e.options,
      e.colors,
      e.markLine,
      e.showTooltip,
      e.showValue,
      e.zoomLock,
      e.barGap,
      e.barWidth,
      e.yIndexs,
      e.range
    ], () => {
      s();
    }), O(() => {
      s();
    }), {
      chart: f,
      chartDataRow: I,
      style: n,
      legendStyles: A,
      cStyles: L
    };
  }
});
export {
  N as default
};
