// @ths/design/es/components/chart-bar-base/src/index.vue.js
import j from "./index.vue2.js";
import { openBlock as n, createElementBlock as t, normalizeStyle as l, createElementVNode as o, Fragment as H, renderList as M, createCommentVNode as F } from "vue";
import "./index.vue3.js";
import q from "../../../_virtual/_plugin-vue_export-helper.js";
const A = {
  key: 0,
  class: "legend"
}, G = ["innerHTML"];
function J(e, K, O, P, Q, U) {
  var r, a, d, i, y, c;
  return n(), t("div", {
    style: l({
      flexDirection: e.legendStyles.position === "top" ? "column-reverse" : "column",
      ...(r = e.cStyles) == null ? void 0 : r.root,
      ...(d = (a = e.cStyles) == null ? void 0 : a.wrapper) == null ? void 0 : d.default
    }),
    class: "t-chart-bar-base"
  }, [
    o("div", {
      ref: "chart",
      class: "chart-box",
      style: l({
        ...e.style,
        ...(i = e.cStyles) == null ? void 0 : i.chart,
        ...(c = (y = e.cStyles) == null ? void 0 : y.chart) == null ? void 0 : c.default
      })
    }, null, 4),
    e.showLegend && e.legendIcon === "default" ? (n(), t("div", A, [
      (n(!0), t(H, null, M(e.chartDataRow, (T, V) => {
        var g, u, m;
        return n(), t("div", {
          key: V,
          style: l({
            justifyContent: e.legendStyles.align === "center" ? "center" : e.legendStyles.align === "left" ? "flex-start" : "flex-end",
            ...(g = e.cStyles) == null ? void 0 : g.legendRoot,
            ...(m = (u = e.cStyles) == null ? void 0 : u.legendRoot) == null ? void 0 : m.default
          }),
          class: "list-wrap"
        }, [
          (n(!0), t(H, null, M(T, (s) => {
            var S, f, p, h, v, k, C, b, w, z, I, B, L, N, $, R, D, E;
            return n(), t("div", {
              key: s,
              style: l({
                margin: (S = e.legendStyles) == null ? void 0 : S.margin,
                ...(f = e.cStyles) == null ? void 0 : f.legendItem,
                ...(h = (p = e.cStyles) == null ? void 0 : p.legendItem) == null ? void 0 : h.default
              }),
              class: "list-item"
            }, [
              o("div", {
                style: l({
                  backgroundColor: (k = (v = s.style) == null ? void 0 : v.legend) == null ? void 0 : k.borderColor,
                  width: (C = e.legendStyles) == null ? void 0 : C.iconSize,
                  height: (b = e.legendStyles) == null ? void 0 : b.iconSize,
                  ...(w = e.cStyles) == null ? void 0 : w.legendIcon,
                  ...(I = (z = e.cStyles) == null ? void 0 : z.legendIcon) == null ? void 0 : I.default
                }),
                class: "list-item__icon"
              }, [
                o("div", {
                  class: "icon-inner",
                  style: l({
                    backgroundColor: (L = (B = s.style) == null ? void 0 : B.legend) == null ? void 0 : L.backgroundColor
                  })
                }, null, 4)
              ], 4),
              o("div", {
                style: l({
                  color: ($ = (N = s.style) == null ? void 0 : N.legend) == null ? void 0 : $.nameColor,
                  fontSize: (R = e.legendStyles) == null ? void 0 : R.fontSize,
                  ...(D = e.cStyles) == null ? void 0 : D.legendName,
                  ...(E = e.cStyles) == null ? void 0 : E.legendName.default
                }),
                class: "list-item__name",
                innerHTML: s.name
              }, null, 12, G)
            ], 4);
          }), 128))
        ], 4);
      }), 128))
    ])) : F("", !0)
  ], 4);
}
const _ = /* @__PURE__ */ q(j, [["render", J]]);
export {
  _ as default
};
