// @ths/c-common-base-paging/lib/index.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory(require('https://unpkg.com/vue@next')) :
  typeof define === 'function' && define.amd ? define(['https://unpkg.com/vue@next'], factory) :
  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, global["C-common-base-paging"] = factory(global.Vue));
})(this, (function (vue) { 'use strict';

  var script = vue.defineComponent({
      name: 'TPaging',
      props: {
          page: {
              required: true,
              type: Object,
              default: () => ({
                  pageNum: 1,
                  pageSize: 10,
                  total: 0,
              }),
          },
          // 每页显示个数选择器的选项设置
          pageSizes: {
              required: false,
              type: Array,
              default: () => [5, 10, 20, 50, 100],
          },
          // 组件布局，子组件名用逗号分隔
          layout: {
              type: String,
              default: 'total, sizes, prev, pager, next, jumper',
          },
          // 总页数 total 和 page-count 设置任意一个就可以达到显示页码的功能；如果要支持 page-sizes 的更改，则需要使用 total 属性
          pagerCount: {
              required: false,
              type: Number,
              default: 7,
          },
          /*
           * autoScroll: {
           *   type: Boolean,
           *   default: true,
           * },
           */
      },
      emits: ['update:page-num', 'update:page-size', 'handle-search'],
      setup(props, { emit }) {
          console.log(props, 'pageProps');
          // 响应式数据
          const state = vue.reactive({
              // 当前页码
              currentPage: 1,
              // 页码
              pageNum: props.page.pageNum,
              // 每页数据条数
              pageSize: 10,
              // 是否显示加载动画
              loading: false,
          });
          /**
           * 更改每页数量
           * @param val 页码
           */
          const handleSizeChange = async (val) => {
              state.loading = true;
              // 更新每页数据条数及页码
              emit('update:page-size', val);
              emit('update:page-num', props.page.pageNum);
              // 告诉父组件，分页参数更新完成，可以执行搜索了
              await emit('handle-search');
              state.loading = false;
              // scrollTo(0, 800);
          };
          /**
           * 翻页
           * @param val 页码
           */
          const handleCurrentChange = async (val) => {
              state.loading = true;
              // 更新每页数据条数及页码
              emit('update:page-size', props.page.pageSize);
              emit('update:page-num', val);
              // 告诉父组件，分页参数更新完成，可以执行搜索了
              await emit('handle-search');
              state.loading = false;
              /*
               * if (props.autoScroll) {
               *    scrollTo(0, 800);
               * }
               */
          };
          // 监测外部页码变化
          vue.watch(() => props.page.pageNum, async (newValue) => {
              state.pageNum = newValue;
          });
          vue.watch(() => props.page.pageSize, async (newValue) => {
              state.pageSize = newValue;
          }, {
              immediate: true,
          });
          return {
              ...vue.toRefs(state),
              handleSizeChange,
              handleCurrentChange,
          };
      },
  });

  const _hoisted_1 = { class: "pagination-container" };

  function render(_ctx, _cache, $props, $setup, $data, $options) {
    const _component_el_pagination = vue.resolveComponent("el-pagination");
    const _directive_loading = vue.resolveDirective("loading");

    return (vue.openBlock(), vue.createElementBlock("div", _hoisted_1, [
      vue.createCommentVNode(" current-page: 当前页数，支持 v-model 双向绑定 "),
      vue.createCommentVNode(" page-size: 每页显示条目个数，支持 v-model 双向绑定 "),
      vue.createCommentVNode(" page-sizes: 每页显示个数选择器的选项设置 "),
      vue.createCommentVNode(" pager-count: 设置最大页码按钮数。 页码按钮的数量，当总页数超过该值时会折叠 "),
      vue.createCommentVNode(" background: 是否为分页按钮添加背景色 "),
      vue.createCommentVNode(" layout: 组件布局，子组件名用逗号分隔 "),
      vue.createCommentVNode(" total: 总条目数 "),
      vue.createCommentVNode(" @size-change: pageSize 改变时触发 "),
      vue.createCommentVNode(" @current-change: current-change 改变时触发 "),
      vue.withDirectives(vue.createVNode(_component_el_pagination, vue.mergeProps({
        "current-page": _ctx.pageNum,
        "onUpdate:current-page": _cache[0] || (_cache[0] = $event => ((_ctx.pageNum) = $event)),
        "page-size": _ctx.pageSize,
        "onUpdate:page-size": _cache[1] || (_cache[1] = $event => ((_ctx.pageSize) = $event)),
        "element-loading-spinner": "el-icon-loading",
        background: "",
        layout: _ctx.layout,
        "page-sizes": _ctx.pageSizes,
        total: _ctx.page.total || 0,
        "pager-count": _ctx.pagerCount
      }, _ctx.$attrs, {
        onSizeChange: _ctx.handleSizeChange,
        onCurrentChange: _ctx.handleCurrentChange
      }), null, 16 /* FULL_PROPS */, ["current-page", "page-size", "layout", "page-sizes", "total", "pager-count", "onSizeChange", "onCurrentChange"]), [
        [
          _directive_loading,
          _ctx.loading,
          void 0,
          {
            fullscreen: true,
            lock: true
          }
        ]
      ])
    ]))
  }

  script.render = render;
  script.__file = "package/ths/components/c-common-base-paging/src/view/paging.vue";

  script.install = (app) => {
      app.component(script.name, script);
  };

  return script;

}));
