// @ths/design/es/components/html/src/index.vue.js
import u from "./index.vue2.js";
import { openBlock as d, createElementBlock as f, normalizeClass as c, normalizeStyle as g, createElementVNode as s } from "vue";
import "./index.vue3.js";
import y from "../../../_virtual/_plugin-vue_export-helper.js";
const h = ["innerHTML"];
function S(r, e, w, B, L, T) {
  var t, a, l, n, o, m, i, p;
  return d(), f("div", {
    class: c(["t-html", { [r.domOnlyClass]: "true", empty: !r.htmlData }]),
    style: g({
      marginTop: (a = (t = r.curStyle.wrapper) == null ? void 0 : t.default) == null ? void 0 : a.marginTop,
      marginLeft: (n = (l = r.curStyle.wrapper) == null ? void 0 : l.default) == null ? void 0 : n.marginLeft,
      marginRight: (m = (o = r.curStyle.wrapper) == null ? void 0 : o.default) == null ? void 0 : m.marginRight,
      marginBottom: (p = (i = r.curStyle.wrapper) == null ? void 0 : i.default) == null ? void 0 : p.marginBottom
    })
  }, [
    s("div", {
      class: "html-container",
      innerHTML: r.htmlData
    }, null, 8, h),
    e[0] || (e[0] = s("div", { class: "style-container" }, null, -1))
  ], 6);
}
const z = /* @__PURE__ */ y(u, [["render", S]]);
export {
  z as default
};
