// @ths/design/es/components/html/src/index.vue2.js
import { defineComponent as a, computed as m, onMounted as s, watch as c } from "vue";
import { getDomRandomClass as p, setStyle as u } from "../../../utils/index.js";
const d = a({
  name: "THtml",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {},
          hover: {}
        }
      })
    },
    // 自定义html内容
    htmlData: {
      type: String,
      default: ""
    }
  },
  setup(o) {
    const e = m(() => o.cStyle), n = p("html"), r = () => {
      let t = "";
      return e.value.wrapper && (t += u(e.value.wrapper, `.${n} .html-container`)), t;
    }, l = () => {
      const t = document.querySelector(`.${n} .style-container`);
      t && (t.innerHTML = `<style>${r()}</style>`);
    };
    return s(() => {
      l();
    }), c(
      () => [o.cStyle],
      () => {
        l();
      },
      {
        deep: !0
      }
    ), {
      curStyle: e,
      domOnlyClass: n
    };
  }
});
export {
  d as default
};
