// @ths/design/es/components/chart-pie-base/src/index.vue.js
import n from "./index.vue2.js";
import { openBlock as m, createElementBlock as c, normalizeStyle as l, normalizeClass as p, withDirectives as i, createElementVNode as t, vShow as a } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const y = {
  ref: "chart",
  class: "chart"
}, u = { class: "empty" }, f = ["src"];
function h(e, s, v, E, k, S) {
  var o, r;
  return m(), c("div", {
    style: l({
      ...e.cStyle,
      ...(r = (o = e.cStyles) == null ? void 0 : o.wrapper) == null ? void 0 : r.default
    }),
    class: p([e.domOnlyClass, "t-chart-pie-base"])
  }, [
    i(t("div", y, null, 512), [
      [a, e.checkEmpty]
    ]),
    i(t("div", u, [
      t("img", {
        class: "empty-img",
        style: l({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [a, !e.checkEmpty]
    ]),
    s[1] || (s[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const B = /* @__PURE__ */ d(n, [["render", h]]);
export {
  B as default
};
