// @ths/design/es/components/chart-pie-base/src/index.vue2.js
import { defineComponent as P, computed as h, ref as T, reactive as A, nextTick as L, onMounted as E, onUnmounted as K, watch as $ } from "vue";
import * as k from "echarts";
import u from "lodash/merge";
import m from "lodash/cloneDeep";
import { DEFAULT_OPTIONS as z } from "../../../utils/constant.js";
import { getDomRandomClass as N, setStyle as y } from "../../../utils/index.js";
const x = {
  wrapper: {
    default: {}
  },
  chart: {
    default: {}
  }
}, g = u(
  m(z),
  {
    tooltip: {
      trigger: "item",
      formatter: (e) => {
        const { data: o, color: a } = e;
        return `<div style="display: flex;justify-content: center;align-items: center;">
        <span style="display: block;width: 8px;height: 8px;margin-right: 5px;border-radius: 50%;background: ${a};"></span>
          ${o.name} ${e.percent}%
        </div>`;
      }
    },
    legend: {
      padding: 0,
      orient: "vertical",
      left: "left",
      top: "top",
      icon: "circle"
    },
    grid: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    }
  }
), F = P({
  name: "TChartPieBase",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 系列样式设置
    seriesStyle: {
      type: Object,
      default: () => ({})
    },
    // 名称别名
    nameKey: {
      type: String,
      default: () => "name"
    },
    // 值别名
    dataKey: {
      type: String,
      default: () => "revenue"
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({ ...x }),
      required: !1
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => g,
      required: !1
    },
    // 饼图主题设置
    pie: {
      type: Object,
      default: () => ({})
    },
    // 是否展示数值
    isShowLabelNumber: {
      type: Boolean,
      default: !1
    },
    // 是否展示占比
    isShowLabelProportion: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const o = N("chart-pie-base"), a = h(() => e.cStyle), c = T(null), O = h(() => ({
      ...x,
      ...e.cStyle
    }));
    let d = null;
    const b = A({
      // 图表数据
      chartData: []
    }), S = () => {
      var t;
      if (e.data && e.data.length) {
        const { nameKey: r, dataKey: l } = e, n = (t = e.options) == null ? void 0 : t.color;
        b.chartData = e.data.map((s, i) => ({
          name: s[r] || "-",
          value: s[l] || 0,
          itemStyle: {
            color: n && n[i] || g.color[i] || "#fff"
          }
        }));
      }
    }, D = () => {
      let t = "";
      return a.value.wrapper && (t += y(a.value.wrapper, `.${o}`)), a.value.emptyContainer && (t += y(a.value.emptyContainer, `.${o} .empty`)), a.value.emptyImg && (t += y(a.value.emptyImg, `.${o} .empty .empty-img`)), a.value.emptyText && (t += y(a.value.emptyText, `.${o} .empty .empty-text`)), t;
    }, v = () => {
      const t = document.querySelector(`.${o} .style-container`);
      t && (t.innerHTML = `<style>${D()}</style>`);
    }, f = () => {
      c.value && (c.value.getBoundingClientRect().width ?? 0) && L(() => {
        d.resize();
      });
    }, j = h(() => {
      var t, r, l, n;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((i) => {
            var C;
            return (i == null ? void 0 : i.data) && ((C = i == null ? void 0 : i.data) == null ? void 0 : C.length) !== 0;
          });
        if (typeof ((r = e.options) == null ? void 0 : r.series) == "object")
          return ((n = (l = e.options.series) == null ? void 0 : l.data) == null ? void 0 : n.length) === 0;
      }
      return e.data.length ? (L(() => {
        f();
      }), !0) : !1;
    }), w = () => ({
      type: "pie",
      radius: [0, "50%"],
      avoidLabelOverlap: !0,
      clockwise: !0,
      label: {
        show: !0,
        position: "outside",
        color: "#000",
        fontSize: 12,
        fontWeight: "normal"
      },
      itemStyle: {
        borderColor: "#fff",
        borderWidth: 3
      },
      labelLine: {
        show: !0,
        length: 12,
        length2: 12
      }
    }), p = () => {
      var t;
      if (c.value) {
        d || (d = k.init(c.value));
        const r = (t = e.options) == null ? void 0 : t.series;
        let l = {};
        r ? l = r.map((s) => u(
          m(w()),
          e == null ? void 0 : e.pie,
          s,
          /* TODO：暂时添加上了数值，需要添加文本和数值的样式配置 */
          {
            label: {
              formatter: `{name|{b}}  {value|${e.isShowLabelNumber ? "{c}" : ""}} {proportion|${e.isShowLabelProportion ? "{d}%" : ""}}`,
              rich: {
                name: {},
                value: {},
                proportion: {}
              }
            }
          }
        )) : e.data && e.data.length && (l = u(
          m(w()),
          {
            data: b.chartData,
            label: {
              formatter: `{name|{b}}  {value|${e.isShowLabelNumber ? "{c}" : ""}} {proportion|${e.isShowLabelProportion ? "{d}%" : ""}}`,
              rich: {
                name: {},
                value: {},
                proportion: {}
              }
            }
          },
          m(e.pie),
          e == null ? void 0 : e.seriesStyle
        ));
        let n = u(
          {
            series: l
          },
          g
        );
        n = u(n, e.options), d.clear();
        try {
          d.setOption(n);
        } catch (s) {
          console.log(s);
        }
      }
    };
    return E(() => {
      v(), S(), p(), window.addEventListener("resize", f);
    }), K(() => {
      window.removeEventListener("resize", f);
    }), $(
      () => [
        e.pie,
        e.options,
        e.data,
        e.nameKey,
        e.dataKey,
        e.seriesStyle,
        e.isShowLabelNumber,
        e.isShowLabelProportion
      ],
      () => {
        S(), p();
      },
      {
        deep: !0
      }
    ), $(
      () => [e.cStyle],
      () => {
        v(), f();
      },
      {
        deep: !0
      }
    ), {
      chart: c,
      cStyles: O,
      curStyle: a,
      checkEmpty: j,
      domOnlyClass: o
    };
  }
});
export {
  F as default
};
