// @ths/design/es/components/grid/src/index.vue.js
import n from "./index.vue2.js";
import { resolveComponent as s, openBlock as t, createElementBlock as l, normalizeStyle as u, normalizeClass as i, Fragment as m, renderList as c, createBlock as p, mergeProps as f, createElementVNode as y } from "vue";
import "./index.vue3.js";
import g from "../../../_virtual/_plugin-vue_export-helper.js";
function S(e, r, $, b, k, B) {
  const a = s("t-grid-item");
  return t(), l("div", {
    style: u(e.rootStyle),
    class: i(e.classes)
  }, [
    (t(!0), l(m, null, c(e.curData, (d, o) => (t(), p(a, f({
      key: o,
      "item-data": d,
      "item-index": o,
      gutter: e.curGutter,
      border: e.hasBorder,
      ref_for: !0
    }, e.$attrs, {
      "c-style": {
        icon: e.curStyle.icon || { default: {} },
        text: e.curStyle.text || { default: {} },
        dot: e.curStyle.dot || { default: {} },
        badge: e.curStyle.badge || { default: {} },
        wrapper: e.curStyle.itemWrapper || { default: {} },
        desc: e.curStyle.desc || { default: {} }
      }
    }), null, 16, ["item-data", "item-index", "gutter", "border", "c-style"]))), 128)),
    r[0] || (r[0] = y("div", { class: "style-container" }, null, -1))
  ], 6);
}
const G = /* @__PURE__ */ g(n, [["render", S]]);
export {
  G as default
};
