// @ths/design/es/components/grid/src/index.vue2.js
import { defineComponent as p, computed as a, ref as b, watch as g } from "vue";
import l from "lodash/cloneDeep";
import y from "lodash/merge";
import O from "./item.vue.js";
import { addUnit as x } from "../../../utils/index.js";
import { createNamespace as S } from "../../../utils/create-class.js";
import { BORDER_TOP as B, BORDER_LEFT as D } from "../../../utils/constant.js";
const f = {
  text: "text",
  icon: "icon",
  dot: "dot",
  badge: "badge",
  desc: "desc"
}, A = p({
  name: "TGrid",
  components: { TGridItem: O },
  props: {
    // 宫格数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => l(f),
      require: !1
    },
    // 是否显示边框
    border: {
      type: Boolean,
      default: !1
    },
    // 格子间距
    gutter: {
      type: [String, Number],
      default: 0
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 单项容器
        itemWrapper: {
          default: {}
        },
        // 图标
        icon: {
          default: {}
        },
        // 宫格文字
        text: {
          default: {}
        },
        // 右上角小点
        dot: {
          default: {}
        },
        // 右上角徽标
        badge: {
          default: {}
        },
        // 宫格描述
        desc: {
          default: {}
        }
      })
    }
  },
  setup(t) {
    const o = a(() => t.cStyle), c = b(t.data), n = a(() => t.gutter), m = a(() => t.border), i = a(() => {
      var e;
      return {
        ...((e = o.value.wrapper) == null ? void 0 : e.default) || { default: {} },
        paddingLeft: x(t.gutter)
        // gutter利用paddingLeft实现
      };
    }), s = a(() => {
      const { border: e, gutter: r } = t, [d] = S("grid");
      return [d(""), { [B]: e && !r, [D]: e && !r }];
    });
    return g(() => [t.data, t.alias], () => {
      const e = y(l(f), t.alias);
      c.value = t.data.map((r) => {
        const d = {};
        return Object.keys(e).forEach((u) => {
          d[u] = r[e[u]];
        }), d;
      });
    }, {
      deep: !0,
      immediate: !0
    }), {
      curStyle: o,
      curData: c,
      curGutter: n,
      hasBorder: m,
      rootStyle: i,
      classes: s
    };
  }
});
export {
  A as default
};
