// @ths/design/es/components/grid/src/item.vue.js
import k from "./item.vue2.js";
import { openBlock as s, createElementBlock as r, normalizeStyle as t, createElementVNode as a, normalizeClass as v, createCommentVNode as d, toDisplayString as i } from "vue";
import "./item.vue3.js";
import C from "../../../_virtual/_plugin-vue_export-helper.js";
const $ = { class: "t-badge-wrapper" }, h = ["src"], w = ["title"], z = ["title"];
function B(e, l, E, N, V, G) {
  var o, n, u, m, c, y, p, f, g, I, S, b;
  return s(), r("div", {
    class: "t-grid-item",
    style: t(e.rootStyle),
    onClick: l[0] || (l[0] = (...D) => e.handleItemClick && e.handleItemClick(...D))
  }, [
    a("div", {
      class: v(e.classes),
      style: t({ ...e.contentStyle, ...(n = (o = e.curStyle) == null ? void 0 : o.wrapper) == null ? void 0 : n.default })
    }, [
      a("div", $, [
        a("img", {
          src: e.curItemData.icon,
          style: t({ ...e.iconStyle, ...(m = (u = e.curStyle) == null ? void 0 : u.icon) == null ? void 0 : m.default }),
          draggable: "false"
        }, null, 12, h),
        e.curItemData.dot ? (s(), r("div", {
          key: 0,
          class: "t-badge t-badge--dot t-badge--fixed",
          style: t({ ...(y = (c = e.curStyle) == null ? void 0 : c.dot) == null ? void 0 : y.default })
        }, null, 4)) : d("", !0),
        e.curItemData.badge ? (s(), r("div", {
          key: 1,
          style: t({ ...(f = (p = e.curStyle) == null ? void 0 : p.badge) == null ? void 0 : f.default }),
          class: "t-badge t-badge--fixed"
        }, i(e.curItemData.badge), 5)) : d("", !0)
      ]),
      a("span", {
        class: "t-grid-item__text",
        style: t({ ...(I = (g = e.curStyle) == null ? void 0 : g.text) == null ? void 0 : I.default }),
        title: e.curItemData.text
      }, i(e.curItemData.text), 13, w),
      e.curItemData.desc ? (s(), r("span", {
        key: 0,
        class: "t-grid-item__text",
        style: t({ ...(b = (S = e.curStyle) == null ? void 0 : S.desc) == null ? void 0 : b.default }),
        title: e.curItemData.desc
      }, i(e.curItemData.desc), 13, z)) : d("", !0)
    ], 6),
    l[1] || (l[1] = a("div", { class: "style-container" }, null, -1))
  ], 4);
}
const F = /* @__PURE__ */ C(k, [["render", B]]);
export {
  F as default
};
