// @ths/design/es/components/grid/src/item.vue2.js
import { defineComponent as S, computed as a } from "vue";
import { addUnit as l } from "../../../utils/index.js";
import { createNamespace as b } from "../../../utils/create-class.js";
import { BORDER as h } from "../../../utils/constant.js";
const T = S({
  name: "TGridItem",
  inheritAttrs: !1,
  props: {
    // 宫格单项数据
    itemData: {
      type: Object,
      required: !0
    },
    // 宫格序号
    itemIndex: {
      type: Number,
      default: 0
    },
    // 列数
    columnNum: {
      type: [String, Number],
      default: 4
    },
    // 图标大小，默认单位为 px
    iconSize: {
      type: [String, Number],
      default: "53px"
    },
    // 格子之间的间距，默认单位为 px
    gutter: {
      type: [String, Number],
      default: 0
    },
    // 是否显示边框
    border: {
      type: Boolean,
      default: !1
    },
    // 格子内容排列的方向，可选值为 horizontal、vertical
    direction: {
      type: String,
      default: "vertical"
    },
    // 是否调换图标和文本的位置
    reverse: {
      type: Boolean,
      default: !1
    },
    // 是否将格子固定为正方形
    square: {
      type: Boolean,
      default: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        icon: {
          default: {}
        },
        text: {
          default: {}
        },
        dot: {
          default: {}
        },
        badge: {
          default: {}
        },
        desc: {
          default: {}
        }
      })
    }
  },
  emits: ["grid-item-click"],
  setup(n, { emit: s }) {
    const m = a(() => n.cStyle), c = a(() => n.itemData), f = a(() => {
      const {
        square: e,
        gutter: t,
        columnNum: r,
        itemIndex: u
      } = n, o = `${100 / +r}%`, i = {
        flexBasis: o,
        paddingTop: "",
        paddingRight: "",
        marginTop: ""
      };
      if (e)
        i.paddingTop = o;
      else if (t) {
        const d = l(t);
        i.paddingRight = d, u >= r && (i.marginTop = d);
      }
      return i;
    }), g = a(() => {
      const { square: e, gutter: t } = n;
      if (e && t) {
        const r = l(t);
        return {
          right: r,
          bottom: r,
          height: "auto"
        };
      }
      return {};
    }), p = a(() => {
      const { iconSize: e } = n;
      return e ? {
        width: e,
        height: e
      } : {};
    }), y = a(() => {
      const {
        border: e,
        square: t,
        gutter: r,
        reverse: u,
        direction: o
      } = n, [i] = b("grid-item");
      return [
        i("content", [
          o,
          {
            square: t,
            reverse: u,
            surround: e && r
          }
        ]),
        { [h]: e }
      ];
    });
    return {
      curStyle: m,
      rootStyle: f,
      contentStyle: g,
      classes: y,
      iconStyle: p,
      curItemData: c,
      handleItemClick: () => {
        s("grid-item-click", c.value);
      }
    };
  }
});
export {
  T as default
};
