// @ths/design/es/components/pagination/src/components/pager.vue.js
import d from "./pager.vue2.js";
import { openBlock as n, createElementBlock as t, Fragment as m, renderList as v, normalizeClass as s, normalizeStyle as l, withModifiers as a, createElementVNode as r, toDisplayString as y, createCommentVNode as u } from "vue";
import "./pager.vue3.js";
import f from "../../../../_virtual/_plugin-vue_export-helper.js";
const k = { class: "page-link-box" }, S = ["onClick", "onMouseover", "onMouseleave", "onMousedown", "onMouseup"], g = { key: 0 };
function c(e, M, $, h, C, p) {
  return n(), t("div", k, [
    (n(!0), t(m, null, v(e.pageListValue, (o) => (n(), t("div", {
      key: o.val,
      class: s(["page-link", e.miniClass]),
      style: l([e.mini ? e.miniClass : ""]),
      onClick: (i) => o.val === "..." ? e.quickJump(o) : e.jumpTo(o),
      onMouseover: a((i) => o.val === "..." ? e.onOver(o) : "", ["stop"]),
      onMouseleave: a((i) => o.text === "》" || o.text === "《" ? e.onLeave(o) : "", ["stop"]),
      onMousedown: (i) => e.onFocus(o),
      onMouseup: (i) => e.onUp(o)
    }, [
      r("div", {
        style: l([
          o.text === "《" ? e.leftStyle : "",
          o.text === "》" ? e.rightStyle : "",
          o.val === "..." ? e.eliStyle : e.normalStyle,
          o.selected ? e.selectedStyle : "",
          e.mini ? e.miniClass : ""
        ]),
        class: s(["page-text", [e.pageClass]])
      }, [
        r("div", {
          class: s([o.val === "..." ? e.innerEliStyle : ""])
        }, [
          !e.isUpEllipsis || o.text !== "》" && o.text !== "《" ? (n(), t("div", g, y(o.text), 1)) : (n(), t("div", {
            key: 1,
            style: l([
              o.text === "《" ? e.leftStyle : "",
              o.text === "》" ? e.rightStyle : ""
            ]),
            class: s(e.mini ? e.miniIconStyle : e.iconStyle)
          }, [
            e.direction === "left" && o.text === "《" ? (n(), t("span", {
              key: 0,
              class: s(e.directionIcon)
            }, null, 2)) : u("", !0),
            e.direction === "right" && o.text === "》" ? (n(), t("span", {
              key: 1,
              class: s(e.directionIcon)
            }, null, 2)) : u("", !0)
          ], 6))
        ], 2)
      ], 6)
    ], 46, S))), 128))
  ]);
}
const w = /* @__PURE__ */ f(d, [["render", c], ["__scopeId", "data-v-aa5cfb63"]]);
export {
  w as default
};
