// @ths/design/es/components/pagination/src/components/pager.vue2.js
import { defineComponent as S, reactive as t, ref as n } from "vue";
const E = S({
  name: "Pager",
  props: {
    // 展示数组
    pageListValue: {
      type: Array,
      default: () => []
    },
    // 盒子样式
    pageClass: {
      type: String,
      default: ""
    },
    // 迷你盒子样式
    mini: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["jump-orinary", "jump-quick"],
  setup(d, s) {
    const p = t({
      width: "24px",
      height: "24px",
      fontSize: "12px",
      lineHeight: "24px"
    }), f = n("mini-icon"), g = n("icon"), m = n("innerEliStyle"), c = n(!1), r = n(""), i = n(""), x = t({
      border: "none",
      "background-color": "transparent"
    }), b = t({
      "background-color": "#fefefe"
    }), v = t({
      "background-color": "var(--theme-color)",
      color: "#fff",
      border: "1px solid var(--theme-color)",
      "border-radius": "3px"
    }), l = t({
      "background-color": "#eee",
      borderRadius: "3px"
    }), a = t({
      "background-color": "#eee",
      borderRadius: "3px"
    });
    return {
      leftStyle: a,
      rightStyle: l,
      eliStyle: x,
      normalStyle: b,
      selectedStyle: v,
      miniClass: p,
      innerEliStyle: m,
      directionIcon: r,
      direction: i,
      isUpEllipsis: c,
      miniIconStyle: f,
      iconStyle: g,
      onOver: (e) => {
        const o = e;
        o.text = "middle";
        let u = 0;
        const y = (k) => k.text === o.text;
        u = d.pageListValue.findIndex(y), u === 1 ? (o.text = "《", c.value = !0, r.value = "icon-pagination-double-left", i.value = "left") : (o.text = "》", c.value = !0, r.value = "icon-pagination-double-right", i.value = "right");
      },
      onLeave: (e) => {
        const o = e;
        c.value = !1, o.text = "...";
      },
      onFocus: (e) => {
        e.text === "》" ? l["background-color"] = "#dcdcdc" : a["background-color"] = "#dcdcdc";
      },
      onUp: (e) => {
        e.text === "》" ? l["background-color"] = "#eee" : a["background-color"] = "#eee";
      },
      jumpTo: (e) => {
        s.emit("jump-orinary", e.val);
      },
      quickJump: (e) => {
        s.emit("jump-quick", e);
      }
    };
  }
});
export {
  E as default
};
