// @ths/design/es/components/chart-pie-3d/src/index.vue.js
import a from "./index.vue2.js";
import { openBlock as m, createElementBlock as n, normalizeClass as p, normalizeStyle as o, withDirectives as l, createElementVNode as t, vShow as i } from "vue";
import "./index.vue3.js";
import c from "../../../_virtual/_plugin-vue_export-helper.js";
const d = ["id"], y = { class: "empty" }, f = ["src"];
function u(e, s, h, v, E, k) {
  var r;
  return m(), n("div", {
    ref: "myChartRef",
    class: p(["t-chart-pie-3d", e.domOnlyClass]),
    style: o({ ...(r = e.cStyle) == null ? void 0 : r.wrapper.default })
  }, [
    l(t("div", {
      id: e.echartsId,
      class: "chart"
    }, null, 8, d), [
      [i, e.checkEmpty]
    ]),
    l(t("div", y, [
      t("img", {
        class: "empty-img",
        style: o({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [i, !e.checkEmpty]
    ]),
    s[1] || (s[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const S = /* @__PURE__ */ c(a, [["render", u]]);
export {
  S as default
};
