// @ths/design/es/components/chart-pie-3d/src/index.vue2.js
import { defineComponent as Q, computed as B, reactive as Z, ref as ee, onMounted as te, watch as T, nextTick as V, toRefs as ae } from "vue";
import p from "highcharts";
import O from "lodash/merge";
import ne from "lodash/cloneDeep";
import { DEFAULT_OPTIONS as G } from "../../../utils/constant.js";
import { getDomRandomClass as se, setStyle as j } from "../../../utils/index.js";
const ie = {
  wrapper: {
    default: {}
  }
}, k = {
  // 图表系列默认颜色数组。当所有颜色都用完后，新的颜色会重新从头开始
  colors: G.color,
  // 图表常规选项
  chart: {
    type: "pie",
    animation: !1,
    backgroundColor: G.backgroundColor,
    events: {
      // 图表加载时触发的事件
      load() {
        const { each: e } = p, { points: D } = this.series[0];
        e(D, (l, d) => {
          l.graphic.attr({
            translateY: -l.shapeArgs.ran
          }), l.graphic.side1.attr({
            translateY: -l.shapeArgs.ran
          }), l.graphic.side2.attr({
            translateY: -l.shapeArgs.ran
          });
        });
      }
    },
    options3d: {
      enabled: !0,
      alpha: 78,
      beta: 0
    }
  },
  // 标题
  title: {
    text: ""
  },
  // 提示框
  tooltip: {
    // headerFormat: '',
    // pointFormat: '{point.name}: <b>{point.percentage:.1f}%</b>',
    backgroundColor: "#fff"
  },
  // 图例
  legend: {
    align: "center",
    symbolHeight: 16,
    symbolWidth: 16,
    symbolRadius: 0,
    itemStyle: {
      color: "rgba(237, 85, 85, 1)"
    },
    width: 0,
    padding: 0
  },
  plotOptions: {
    pie: {
      allowPointSelect: !1,
      cursor: "pointer",
      depth: 35,
      // 引线样式
      dataLabels: {
        enabled: !0,
        format: "{point.name}"
      },
      showInLegend: !0,
      // 数据图例
      innerSize: 0,
      point: {
        events: {}
      }
      // size: '65%',
      // x: 400,
    },
    series: {
      events: {
        // click: () => {},
        // legendItemClick: () => {},
      }
      // 显示系列时禁用动画
      // animation: false,
    }
  }
}, ye = Q({
  name: "TChartPie3d",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ie
    },
    // 矩形图例样式
    legendStyle: {
      type: Array,
      default: () => []
    },
    // highcharts 原生配置
    options: {
      type: Object,
      default: () => ({})
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 饼图的内径大小，可以是百分比或像素值；大于0是圆环图,0是实心饼图
    innerSize: {
      type: Number,
      default: () => 0
    },
    pie3d: {
      type: Object,
      default: () => ({})
    },
    // 是否展示数值
    isShowLabelNumber: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: D }) {
    const l = se("chart-pie-3d"), d = B(() => e.cStyle);
    let M = null;
    const b = Z({
      echartsId: "",
      chartData: []
    }), x = ee();
    function U() {
      const t = [], i = "0123456789";
      for (let r = 0; r < 16; r++)
        t[r] = i.substr(Math.floor(Math.random() * 16), 1);
      const n = t.join("");
      b.echartsId = n;
    }
    function N() {
      var t, i;
      (i = (t = e.seriesSetting) == null ? void 0 : t.data) != null && i.length && e.data && (b.chartData = e.seriesSetting.data.map((n) => {
        var u;
        const {
          nameKey: r,
          valueKey: g,
          depthKey: y
        } = n;
        return {
          name: ((u = e.seriesSetting) == null ? void 0 : u.name) || "--",
          data: e.data.map((S) => ({
            name: S[r],
            y: S[g],
            h: S[y]
          }))
        };
      }));
    }
    const W = () => {
      let t = "";
      return d.value.wrapper && (t += j(d.value.wrapper, `.${l}`)), d.value.emptyContainer && (t += j(d.value.emptyContainer, `.${l} .empty`)), d.value.emptyImg && (t += j(d.value.emptyImg, `.${l} .empty .empty-img`)), d.value.emptyText && (t += j(d.value.emptyText, `.${l} .empty .empty-text`)), t;
    }, Y = () => {
      const t = document.querySelector(`.${l} .style-container`);
      t && (t.innerHTML = `<style>${W()}</style>`);
    }, X = B(() => {
      var t, i, n, r;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((y) => {
            var u;
            return (y == null ? void 0 : y.data) && ((u = y == null ? void 0 : y.data) == null ? void 0 : u.length) !== 0;
          });
        if (typeof ((i = e.options) == null ? void 0 : i.series) == "object")
          return ((r = (n = e.options.series) == null ? void 0 : n.data) == null ? void 0 : r.length) === 0;
      }
      return !!e.data.length;
    }), $ = () => {
      let t = 0;
      const i = Array.from(x.value.getElementsByTagName("rect"));
      if (e.legendStyle.length) {
        for (let n = 0; n < i.length; n++)
          if (i[n].className.baseVal === "highcharts-point") {
            for (let r = 0; r < Object.keys(e.legendStyle[t]).length; r++) {
              const g = Object.keys(e.legendStyle[t])[r];
              i[n].style[g] = e.legendStyle[t][g];
            }
            t++;
          }
      }
    };
    function A() {
      var E, P, R, F, H, K, _;
      const { each: t } = p, { round: i } = Math, { cos: n } = Math, { sin: r } = Math, { deg2rad: g } = Math;
      p.wrap(p.seriesTypes.pie.prototype, "translate", function(o) {
        if (o.apply(this, [].slice.call(arguments, 1)), !this.chart.is3d())
          return;
        const s = this, { chart: a } = s, { options: m } = a, c = s.options, h = c.depth || 0, { options3d: w } = m.chart, { alpha: C } = w, { beta: J } = w;
        let z = c.stacking ? (c.stack || 0) * h : s._i * h;
        z += h / 2, c.grouping !== !1 && (z = 0), t(s.data, (I) => {
          const { shapeArgs: f } = I;
          let L;
          I.shapeType = "arc3d";
          const q = I.options.h;
          f.z = z, f.depth = h * 0.75 + q, f.alpha = C, f.beta = J, f.center = s.center, f.ran = q, L = (f.end + f.start) / 2, I.slicedTranslation = {
            translateX: i(n(L) * c.slicedOffset * n(C * g)),
            translateY: i(r(L) * c.slicedOffset * n(C * g))
          };
        });
      }), function(o) {
        o.wrap(p.SVGRenderer.prototype, "arc3dPath", function(s) {
          const a = s.apply(this, [].slice.call(arguments, 1));
          return a.zTop = (a.zOut + 0.5) / 100, a;
        });
      }(p);
      const { color: y, series: u } = e.options;
      let S = [];
      u ? S = u.map((o, s) => {
        var m, c, h;
        const a = O(
          {
            type: "pie",
            ...e == null ? void 0 : e.pie3d
          },
          (h = (c = (m = e.seriesSetting) == null ? void 0 : m.data) == null ? void 0 : c[s]) == null ? void 0 : h.style
        );
        return O({}, a, o);
      }) : e.data && ((P = (E = e.seriesSetting) == null ? void 0 : E.data) != null && P.length) && (S = (R = e.seriesSetting) == null ? void 0 : R.data.map((o, s) => {
        var m, c, h, w, C;
        const a = O(
          {
            type: "pie",
            name: (m = b.chartData[s]) == null ? void 0 : m.name,
            data: e.data.length ? (c = b.chartData[s]) == null ? void 0 : c.data : [],
            ...e == null ? void 0 : e.pie3d
          },
          (C = (w = (h = e.seriesSetting) == null ? void 0 : h.data) == null ? void 0 : w[s]) == null ? void 0 : C.style
        );
        return O({}, a, o);
      })), k.plotOptions.pie.innerSize = e.innerSize, k.plotOptions.series.events.click = (o) => {
        D("com-click", o);
      }, k.plotOptions.pie.point.events.legendItemClick = (o) => {
        V(() => {
          const { points: s } = o.target.series;
          t(s, (a, m) => {
            a.graphic.attr({
              translateY: -a.shapeArgs.ran
            }), a.graphic.side1.attr({
              translateY: -a.shapeArgs.ran
            }), a.graphic.side2.attr({
              translateY: -a.shapeArgs.ran
            });
          });
        }), D("legend-select", o);
      };
      let v = O(
        ne(k),
        {
          series: S,
          colors: ((F = e.options) == null ? void 0 : F.color) || k.colors,
          chart: {
            backgroundColor: ((H = e.options) == null ? void 0 : H.backgroundColor) || k.chart.backgroundColor
          }
        },
        {
          plotOptions: {
            pie: {
              // 引线样式
              dataLabels: {
                format: `{point.name} ${e.isShowLabelNumber ? "{y}" : ""}`,
                style: {
                  textOutline: "0"
                }
              }
            }
          }
        },
        (K = e.seriesSetting) == null ? void 0 : K.labelStyle
      );
      v = O(v, e.options), (_ = e.seriesSetting) != null && _.name || (v.tooltip.headerFormat = "", v.tooltip.pointFormat = "{point.name}: <b>{point.percentage:.1f}%</b>"), setTimeout(() => {
        M = p.chart(b.echartsId, v), M.reflow(), $();
      }, 50);
    }
    return te(() => {
      Y(), U(), N(), A();
    }), T(
      () => [
        e.options,
        e.pie3d
      ],
      () => {
        A();
      },
      {
        deep: !0
      }
    ), T(() => e.legendStyle, () => {
      e.legendStyle.length && V(() => {
        $();
      });
    }, { immediate: !0 }), T(
      [
        () => e.data,
        () => e.seriesSetting,
        e.isShowLabelNumber
      ],
      () => {
        N(), A();
      },
      {
        deep: !0
      }
    ), T(() => e.innerSize, () => {
      A();
    }), T(
      () => [e.cStyle],
      () => {
        Y(), A();
      },
      {
        deep: !0
      }
    ), {
      ...ae(b),
      curStyle: d,
      checkEmpty: X,
      domOnlyClass: l,
      myChartRef: x
    };
  }
});
export {
  ye as default
};
