// @ths/design/es/components/radio/src/index.vue.js
import w from "./index.vue2.js";
import { resolveComponent as t, openBlock as o, createElementBlock as d, normalizeClass as c, normalizeStyle as h, createBlock as a, withCtx as n, Fragment as C, renderList as V, mergeProps as k, createTextVNode as B, toDisplayString as S, createElementVNode as M } from "vue";
import "./index.vue3.js";
import L from "../../../_virtual/_plugin-vue_export-helper.js";
function O(e, l, R, T, U, z) {
  var i, s, p, m, f, g, b, y;
  const $ = t("el-radio"), u = t("el-radio-group"), v = t("el-radio-button");
  return o(), d("div", {
    class: c(["t-radio", e.domOnlyClass]),
    style: h({
      marginTop: (s = (i = e.curStyle.wrapper) == null ? void 0 : i.default) == null ? void 0 : s.marginTop,
      marginLeft: (m = (p = e.curStyle.wrapper) == null ? void 0 : p.default) == null ? void 0 : m.marginLeft,
      marginRight: (g = (f = e.curStyle.wrapper) == null ? void 0 : f.default) == null ? void 0 : g.marginRight,
      marginBottom: (y = (b = e.curStyle.wrapper) == null ? void 0 : b.default) == null ? void 0 : y.marginBottom
    })
  }, [
    e.isUseButton ? (o(), a(u, {
      key: 1,
      modelValue: e.curModel,
      "onUpdate:modelValue": l[1] || (l[1] = (r) => e.curModel = r),
      onChange: e.change
    }, {
      default: n(() => [
        (o(!0), d(C, null, V(e.curOptions, (r) => (o(), a(v, k({ ref_for: !0 }, e.filterAttrs(e.$attrs), {
          key: r,
          label: r.label,
          disabled: r.disabled,
          border: e.border
        }), {
          default: n(() => [
            B(S(r.value), 1)
          ]),
          _: 2
        }, 1040, ["label", "disabled", "border"]))), 128))
      ]),
      _: 1
    }, 8, ["modelValue", "onChange"])) : (o(), a(u, {
      key: 0,
      modelValue: e.curModel,
      "onUpdate:modelValue": l[0] || (l[0] = (r) => e.curModel = r),
      onChange: e.change
    }, {
      default: n(() => [
        (o(!0), d(C, null, V(e.curOptions, (r) => (o(), a($, k({ ref_for: !0 }, e.filterAttrs(e.$attrs), {
          key: r,
          label: r.label,
          disabled: r.disabled,
          border: e.border
        }), {
          default: n(() => [
            B(S(r.value), 1)
          ]),
          _: 2
        }, 1040, ["label", "disabled", "border"]))), 128))
      ]),
      _: 1
    }, 8, ["modelValue", "onChange"])),
    l[2] || (l[2] = M("div", { class: "style-container" }, null, -1))
  ], 6);
}
const F = /* @__PURE__ */ L(w, [["render", O]]);
export {
  F as default
};
