// @ths/design/es/components/radio/src/index.vue2.js
import { defineComponent as F, computed as R, reactive as j, watch as c, onMounted as A, toRefs as P } from "vue";
import { filterAttrs as q, getDomRandomClass as z, setStyle as v, getCssPropertyString as r } from "../../../utils/index.js";
const U = F({
  name: "TRadio",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        circle: () => ({
          default: {},
          hover: {},
          active: {}
        }),
        label: () => ({
          default: {},
          hover: {},
          active: {}
        }),
        radioBorder: () => ({
          default: {},
          hover: {},
          active: {}
        }),
        radioButton: () => ({
          default: {},
          hover: {},
          active: {}
        }),
        // 单选按钮文本
        radioButtonText: () => ({
          default: {},
          hover: {},
          active: {}
        }),
        // 第一个按钮元素
        itemFirstWrapper: {
          default: {}
        },
        // 最后一个按钮元素
        itemLastWrapper: {
          default: {}
        }
      })
    },
    active: {
      type: String,
      default: ""
    },
    options: {
      type: Array,
      default: () => []
    },
    alias: {
      type: Object,
      default: () => ({
        label: "",
        value: "",
        disabled: ""
      })
    },
    size: {
      type: String,
      default: "default"
    },
    border: {
      type: Boolean,
      default: !1
    },
    isUseButton: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["change"],
  setup(l, { emit: O }) {
    const e = R(() => l.cStyle), t = z("radio"), o = j({
      curModel: "image",
      curOptions: []
      // border: true,
    });
    function f() {
      o.curOptions = [], l.options && l.options.length && l.options.forEach((a, d) => {
        var i, u, n;
        o.curOptions.push({
          label: a[((i = l == null ? void 0 : l.alias) == null ? void 0 : i.label) || "label"] || "",
          value: a[((u = l == null ? void 0 : l.alias) == null ? void 0 : u.value) || "value"] || "",
          disabled: a[((n = l == null ? void 0 : l.alias) == null ? void 0 : n.disabled) || "disabled"] || !1
        });
      });
    }
    const W = () => {
      let a = "";
      return e.value.wrapper && (a += v(e.value.wrapper, `.${t} .el-radio-group`)), e.value.itemWrapper && (a += v(e.value.itemWrapper, `.${t} .el-radio`)), a;
    }, M = () => {
      let a = "";
      return e.value.itemFirstWrapper && (a += v(e.value.itemFirstWrapper, `.${t} .el-radio-button:first-child`)), e.value.itemLastWrapper && (a += v(e.value.itemLastWrapper, `.${t} .el-radio-button:last-child`)), a;
    }, w = () => {
      var d, i, u, n, $, B, h, b, m, _, y, S, g, x, T, k, C, p;
      let a = "";
      return e.value.radioBorder && ((d = e.value.radioBorder) != null && d.default) && (a += `
          .${t} .el-radio__inner {
            ${r(e.value.radioBorder.default)}
          }
        `), e.value.radioBorder && ((i = e.value.radioBorder) != null && i.hover) && (a += `
          .${t} .el-radio:not(.is-checked):hover .el-radio__inner {
            ${r(e.value.radioBorder.hover)}
          }
        `), e.value.radioBorder && ((u = e.value.radioBorder) != null && u.active) && (a += `
          .${t} .is-checked .el-radio__inner {
            ${r(e.value.radioBorder.active)}
          }
        `), e.value.circle && ((n = e.value.circle) != null && n.default) && (a += `
          .${t} .el-radio__inner:after {
            ${r(e.value.circle.default)}
          }
        `), e.value.circle && (($ = e.value.circle) != null && $.hover) && (a += `
          .${t} .el-radio:not(.is-checked):hover .el-radio__inner:after {
            ${r(e.value.circle.hover)}
          }
        `), e.value.circle && ((B = e.value.circle) != null && B.active) && (a += `
          .${t} .is-checked .el-radio__inner:after {
            ${r(e.value.circle.active)}
          }
        `), e.value.label && ((h = e.value.label) != null && h.default) && (a += `
          .${t} .el-radio__label {
            ${r(e.value.label.default)}
          }
        `), e.value.label && ((b = e.value.label) != null && b.hover) && (a += `
          .${t} .el-radio:not(.is-checked):hover .el-radio__label {
            ${r(e.value.label.hover)}
          }
        `), e.value.label && ((m = e.value.label) != null && m.active) && (a += `
          .${t} .is-checked .el-radio__label {
            ${r(e.value.label.active)}
          }
        `), e.value.radioButton && ((_ = e.value.radioButton) != null && _.default) && (a += `
          .${t} .el-radio-button {
            ${r(e.value.radioButton.default)}
          }
        `), e.value.radioButton && ((y = e.value.radioButton) != null && y.hover) && (a += `
          .${t} .el-radio-button:not(.is-active):hover {
            ${r(e.value.radioButton.hover)}
          }
        `), e.value.radioButton && ((S = e.value.radioButton) != null && S.active) && (a += `
          .${t} .el-radio-button.is-active {
            ${r(e.value.radioButton.active)}
          }
        `), e.value.radioButtonText && ((g = e.value.radioButtonText) != null && g.default) && (a += `
          .${t} .el-radio-button .el-radio-button__inner {
            ${r(e.value.radioButtonText.default)}
          }
        `), e.value.radioButtonText && ((x = e.value.radioButtonText) != null && x.hover) && (a += `
          .${t} .el-radio-button:not(.is-active):hover .el-radio-button__inner {
            ${r(e.value.radioButtonText.hover)}
          }
        `), e.value.radioButtonText && ((T = e.value.radioButtonText) != null && T.active) && (a += `
          .${t} .el-radio-button.is-active .el-radio-button__inner {
            ${r(e.value.radioButtonText.active)}
          }
        `), e.value.radioBorder && ((k = e.value.radioBorder) != null && k.default) && (a += `.${t}.t-radio .el-radio.is-bordered {
           ${r(e.value.radioBorder.default)}
        }`), e.value.radioBorder && ((C = e.value.radioBorder) != null && C.hover) && (a += `.${t}.t-radio .el-radio.is-bordered:not(.is-checked):hover {
           ${r(e.value.radioBorder.hover)}
        }`), e.value.radioBorder && ((p = e.value.radioBorder) != null && p.active) && (a += `.${t}.t-radio .el-radio.is-bordered.is-checked {
           ${r(e.value.radioBorder.active)}
        }`), a;
    }, s = () => {
      const a = document.querySelector(`.${t} .style-container`);
      a && (a.innerHTML = `<style>${W()}${w()}${M()}</style>`);
    }, H = () => {
      o.curModel = l.active;
    }, L = (a) => {
      const d = o.curOptions.find((i) => i.label === a);
      O("change", a, d);
    };
    return c(() => l.cStyle, () => {
      s();
    }, { deep: !0 }), c(() => l.active, (a) => {
      o.curModel = a;
    }), c(
      [
        () => l.options,
        () => l.alias
      ],
      () => {
        f();
      },
      {
        deep: !0
      }
    ), A(() => {
      f(), s(), H();
    }), {
      ...P(o),
      curStyle: e,
      domOnlyClass: t,
      filterAttrs: q,
      change: L
    };
  }
});
export {
  U as default
};
