// @ths/design/es/components/carousel-text/src/index.vue2.js
import { defineComponent as O, ref as W, reactive as N, computed as q, watch as T, onMounted as H, toRefs as R, nextTick as U } from "vue";
import { getDomRandomClass as P } from "../../../utils/index.js";
import "../../text/index.js";
import X from "../../text/src/index.vue.js";
const x = 1e3, B = O({
  name: "TCarouselText",
  components: {
    TText: X
  },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器样式
        wrapper: {
          default: {},
          focus: {},
          hover: {}
        },
        // 文本样式
        text: {
          default: {},
          focus: {},
          hover: {}
        }
      })
    },
    // 跑马灯展示文本
    data: {
      type: String,
      default: "",
      required: !0
    },
    // 是否滚动
    scroll: {
      type: Boolean,
      default: !0
    },
    // 滚动速度
    speed: {
      type: Number,
      default: 30
    },
    // 前期停留时间
    earlyStage: {
      type: Number,
      default: 2e3
    },
    // 后期停留时间
    laterStage: {
      type: Number,
      default: 2e3
    },
    // 超链接标题
    linkTitle: {
      type: String,
      default: ""
    },
    // 超链接地址
    linkUrl: {
      type: String,
      default: ""
    },
    // 超链接打开方式
    linkOpenMode: {
      type: String,
      default: "newWin"
    }
  },
  emits: ["modify-text"],
  setup(t, { emit: w }) {
    const F = W(null), i = P("carousel");
    let d, o, h;
    const n = N({
      isFocused: !1,
      // 是否是聚焦状态
      isExist: "",
      // 元素是否存在
      text: t.data,
      // 滚动文本
      keyframes: "",
      // 滚动动画
      isHover: !1,
      // 组件被鼠标滑过状态 用于修改组件滑入滑出样式
      editable: !1
      // 双击是否可编辑
    }), v = q(() => {
      var r, c, f, u, y, S;
      const e = { ...(c = (r = t.cStyle) == null ? void 0 : r.wrapper) == null ? void 0 : c.default }, a = { ...e, ...(u = (f = t.cStyle) == null ? void 0 : f.wrapper) == null ? void 0 : u.focus }, l = { ...e, ...(S = (y = t.cStyle) == null ? void 0 : y.wrapper) == null ? void 0 : S.hover };
      let s = e;
      return n.isHover && (s = l), n.isFocused && (s = a), s;
    }), k = (e, a, l, s, r) => {
      U(() => {
        const c = d == null ? void 0 : d.clientWidth, f = o[0].clientWidth, u = f / l + s / x + r / x;
        if (f <= c && o) {
          o.forEach((g) => {
            g.style["animation-name"] = "none";
          }), n.isFocused = !0, n.isExist = "first";
          return;
        }
        const y = s / x / u, S = r / x / u;
        n.keyframes = `@keyframes ${e} {
          0% {
            transform: translateX(0);
          }
          ${y * 100}% {
            transform: translateX(0);
          }
          ${(1 - S) * 100}% {
            transform: translateX(-100%);
          }
          100% {
            transform: translateX(-100%);
          }
        }`, o && o.forEach((g) => {
          g.style["animation-name"] = a ? e : "none", g.style["animation-duration"] = `${u}s`;
        });
      });
    }, C = (e) => {
      n.text = e.data.value, k(i, t.scroll, t.speed, t.earlyStage, t.laterStage), w("modify-text", e);
    }, m = (e, a, l) => {
      e.style["animation-name"] = a, e.style["animation-play-state"] = l;
    }, b = () => {
      n.isHover = !0;
      const e = t.scroll ? i : "none";
      n.isFocused || o.forEach((a) => {
        m(a, e, "paused");
      });
    }, A = () => {
      n.isHover = !1;
      const e = t.scroll ? i : "none";
      n.isFocused || o.forEach((a) => {
        m(a, e, "running");
      });
    }, E = (e) => {
      n.editable && (clearTimeout(h), n.isExist = e, n.isFocused = !0, n.isFocused && o && o.forEach((a) => {
        m(a, "none", "paused");
      }));
    }, $ = () => {
      n.isExist = "", n.isFocused = !1, n.isFocused || o.forEach((e) => {
        m(e, i, "running");
      });
    }, M = () => {
      clearTimeout(h), h = setTimeout(() => {
        n.isFocused || (t.linkUrl && t.linkOpenMode === "newWin" ? window.open(t.linkUrl) : t.linkUrl && t.linkOpenMode === "currWin" && window.open(t.linkUrl, "_self"));
      }, 300);
    };
    return T(() => t.data, (e) => {
      n.text = e;
    }), T(() => [n.text, t.scroll, t.speed, t.earlyStage, t.laterStage], (e, a) => {
      const [, l, s, r, c] = e;
      k(i, l, s, r, c);
    }), T(() => n.keyframes, (e, a) => {
      const l = document.querySelector(`.${i} .style-container`);
      l && (l.innerHTML = `<style>${e}</style>`);
    }), H(() => {
      const e = document.querySelector("body");
      e && (n.editable = (e == null ? void 0 : e.getAttribute("editable")) === "true"), d = document.querySelector(`.${i}`), o = document.querySelectorAll(`.${i} .content-text`), k(i, t.scroll, t.speed, t.earlyStage, t.laterStage);
    }), {
      ...R(n),
      customStyle: v,
      handleModifyTextChange: C,
      pauseAnimation: b,
      playAnimation: A,
      stopAnimation: E,
      startAnimation: $,
      jumpPage: M,
      carouselRef: F,
      domOnlyClass: i
    };
  }
});
export {
  B as default
};
