// @ths/design/es/components/carousel-text/src/index.vue.js
import m from "./index.vue2.js";
import { resolveComponent as p, openBlock as u, createElementBlock as y, normalizeClass as f, normalizeStyle as v, createElementVNode as n, withDirectives as r, createVNode as a, vShow as d } from "vue";
import "./index.vue3.js";
import T from "../../../_virtual/_plugin-vue_export-helper.js";
const k = ["title"], A = { class: "content" };
function M(e, t, C, b, B, E) {
  var l, s;
  const i = p("t-text");
  return u(), y("div", {
    class: f(["t-carousel-text", e.domOnlyClass]),
    style: v(e.customStyle),
    title: e.linkTitle,
    onClick: t[2] || (t[2] = (...o) => e.jumpPage && e.jumpPage(...o)),
    onMouseenter: t[3] || (t[3] = (...o) => e.pauseAnimation && e.pauseAnimation(...o)),
    onMouseleave: t[4] || (t[4] = (...o) => e.playAnimation && e.playAnimation(...o))
  }, [
    n("div", A, [
      r(n("div", {
        class: "content-text",
        onDblclick: t[0] || (t[0] = (o) => e.stopAnimation("first"))
      }, [
        a(i, {
          ref: "carouselRef",
          label: e.text,
          "edit-key": "data",
          "c-style": {
            wrapper: {
              ...(l = e.cStyle) == null ? void 0 : l.text,
              focus: {},
              hover: {}
            }
          },
          onModifyText: e.handleModifyTextChange,
          onBlurText: e.startAnimation
        }, null, 8, ["label", "c-style", "onModifyText", "onBlurText"])
      ], 544), [
        [d, e.isExist === "first" || e.isExist === ""]
      ]),
      r(n("div", {
        class: "content-text",
        onDblclick: t[1] || (t[1] = (o) => e.stopAnimation("second"))
      }, [
        a(i, {
          label: e.text,
          "edit-key": "data",
          "c-style": {
            wrapper: {
              ...(s = e.cStyle) == null ? void 0 : s.text,
              focus: {},
              hover: {}
            }
          },
          onModifyText: e.handleModifyTextChange,
          onBlurText: e.startAnimation
        }, null, 8, ["label", "c-style", "onModifyText", "onBlurText"])
      ], 544), [
        [d, e.isExist === "second" || e.isExist === ""]
      ])
    ]),
    t[5] || (t[5] = n("div", { class: "style-container" }, null, -1))
  ], 46, k);
}
const x = /* @__PURE__ */ T(m, [["render", M]]);
export {
  x as default
};
