// @ths/design/es/components/loop-video/src/index.vue2.js
import { defineComponent as v, reactive as S, computed as g, onBeforeUnmount as w, watch as a, onMounted as C, toRefs as x, nextTick as M } from "vue";
import c from "video.js";
import V from "video.js/dist/lang/zh-CN.json";
import "video.js/dist/video-js.css";
import { getDomRandomClass as J, setStyle as O } from "../../../utils/index.js";
c.addLanguage("zh-CN", V);
const U = v({
  name: "TLoopVideo",
  props: {
    // 视频路径
    videoSrc: {
      type: String,
      default: ""
    },
    // 混合模式
    mixBlendMode: {
      type: String,
      default: "normal"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    }
  },
  setup(n, { emit: b }) {
    const o = S({
      autoplayVideo: {},
      // 自动播放video元素
      player: null,
      // video.js video对象
      videoJs: {}
      // video.js参数设置
    }), m = window.videojs || c, s = g(() => n.cStyle), l = J("loop-video"), p = (e) => {
      let i = "";
      const f = e.lastIndexOf("."), t = e.substr(f);
      return t === ".MP4" || t === ".mp4" ? i = "video/mp4" : t === ".m3u8" || t === ".M3U8" ? i = "application/x-mpegURL" : t === ".ogg" || t === ".OGG" ? i = "video/ogg" : (t === ".webm" || t === ".WEBM") && (i = "video/webm"), i;
    }, u = () => {
      M(() => {
        o.player = m(o.autoplayVideo, o.videoJs, () => {
        });
      });
    }, d = () => {
      o.videoJs = {
        aspectRatio: "16:7",
        // 宽高比
        controls: !1,
        // 显示控制条
        sources: [
          {
            src: n.videoSrc,
            // 视频路径
            type: p(n.videoSrc)
            // 视频类型
          }
        ]
      }, u();
    }, y = () => {
      let e = "";
      return s.value.wrapper && (e += O(s.value.wrapper, `.${l} .video`)), e;
    }, r = () => {
      const e = document.querySelector(`.${l} .style-container`);
      e && (e.innerHTML = `<style>${y()}</style>`);
    };
    return w(() => {
      o.player && o.player.dispose();
    }), a(() => [n.cStyle, n.mixBlendMode], () => {
      r();
    }), a(() => n.videoSrc, () => {
      d();
    }), C(() => {
      r(), d();
    }), {
      ...x(o),
      curStyle: s,
      domOnlyClass: l
    };
  }
});
export {
  U as default
};
