// @ths/design/es/components/loop-video/src/index.vue.js
import c from "./index.vue2.js";
import { openBlock as y, createElementBlock as g, normalizeClass as v, normalizeStyle as f, createElementVNode as S } from "vue";
import "./index.vue3.js";
import h from "../../../_virtual/_plugin-vue_export-helper.js";
const w = ["src"];
function B(e, C, E, $, L, V) {
  var r, o, t, a, l, i, p, d, u, m, n, s;
  return y(), g("div", {
    class: v(["t-loop-video", e.domOnlyClass]),
    style: f({
      marginTop: (o = (r = e.curStyle.wrapper) == null ? void 0 : r.default) == null ? void 0 : o.marginTop,
      marginLeft: (a = (t = e.curStyle.wrapper) == null ? void 0 : t.default) == null ? void 0 : a.marginLeft,
      marginRight: (i = (l = e.curStyle.wrapper) == null ? void 0 : l.default) == null ? void 0 : i.marginRight,
      marginBottom: (d = (p = e.curStyle.wrapper) == null ? void 0 : p.default) == null ? void 0 : d.marginBottom,
      ...(u = e.curStyle.wrapper) == null ? void 0 : u.default
    })
  }, [
    S("video", {
      ref: "autoplayVideo",
      style: f({
        mixBlendMode: e.mixBlendMode,
        height: (n = (m = e.curStyle.wrapper) == null ? void 0 : m.default) == null ? void 0 : n.height,
        ...(s = e.curStyle.video) == null ? void 0 : s.default
      }),
      class: "video-js preview-video",
      src: e.videoSrc,
      autoplay: "",
      loop: "",
      muted: ""
    }, " 浏览器不支持视频标签 ", 12, w)
  ], 6);
}
const M = /* @__PURE__ */ h(c, [["render", B], ["__scopeId", "data-v-46447f0d"]]);
export {
  M as default
};
