// @ths/design/es/components/pollution-forecast/src/index.vue2.js
import { defineComponent as S, computed as g, reactive as h, onMounted as A, watch as s, toRefs as B } from "vue";
import D from "lodash/merge";
import p from "lodash/cloneDeep";
import { getDomRandomClass as W, setStyle as n } from "../../../utils/index.js";
import { POLLUTION_HTML as O } from "../../../utils/constant.js";
const f = {
  weekday: "weekday",
  time: "time",
  startValue: "startValue",
  endValue: "endValue",
  startType: "startType",
  endType: "endType",
  startColor: "startColor",
  endColor: "endColor",
  name: "name"
}, N = S({
  name: "TPollutionForecast",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        },
        // 单个容器
        itemWrapper: {
          default: {},
          hover: {}
        },
        // 工作日
        weekday: {
          default: {}
        },
        // 时间
        time: {
          default: {}
        },
        // 值
        value: {
          default: {}
        },
        // 值区间波浪线
        valueBreak: {
          default: {}
        },
        // 类型容器
        typeWrapper: {
          default: {}
        },
        // 类型值
        type: {
          default: {}
        },
        // 类型区间波浪线
        typeBreak: {
          default: {}
        },
        // 名称
        name: {
          default: {}
        },
        // 渐变色柱
        line: {
          default: {}
        }
      })
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => p(f)
    },
    // 展示方向，横纵向 vertical/horizontal
    direction: {
      type: String,
      default: "vertical"
    }
  },
  setup(o) {
    const t = g(() => o.cStyle), a = W("pollution-forecast"), c = h({
      listData: []
      // 预报数据
    }), r = (e) => {
      let u = e;
      return !e && e !== 0 && (u = "-"), u;
    }, y = (e = "") => {
      const u = e.toUpperCase().replace(/[-.+_]/g, ""), i = O.filter((_) => _.name === u);
      return i.length > 0 ? i[0].html : e;
    }, m = () => {
      const e = D(p(f), o.dataAlias), {
        weekday: u,
        time: i,
        name: _,
        startValue: k,
        endValue: C,
        startType: w,
        endType: $,
        startColor: T,
        endColor: V
      } = e;
      c.listData = [], o.data && o.data.length && (c.listData = o.data.map((l) => ({
        weekday: r(l[u]),
        time: r(l[i]),
        name: y(l[_]),
        startValue: r(l[k]),
        endValue: r(l[C]),
        startType: r(l[w]),
        endType: r(l[$]),
        startColor: l[T] || "#fff",
        endColor: l[V] || "#fff"
      })));
    }, v = () => {
      let e = "";
      return t.value.wrapper && (e += n(t.value.wrapper, `.${a}`)), t.value.itemWrapper && (e += n(t.value.itemWrapper, `.${a} .content .content__item`)), t.value.weekday && (e += n(t.value.weekday, `.${a} .content .content__item .content__item__weekday`)), t.value.time && (e += n(t.value.time, `.${a} .content .content__item .content__item__time`)), t.value.value && (e += n(t.value.value, `.${a} .content .content__item .content__item__value .value`)), t.value.valueBreak && (e += n(t.value.valueBreak, `.${a} .content .content__item .content__item__value .value--break`)), t.value.typeWrapper && (e += n(t.value.typeWrapper, `.${a} .content .content__item .content__item__value .type-container--horizontal`)), t.value.type && (e += n(t.value.type, `.${a} .content .content__item .content__item__value .type`)), t.value.typeBreak && (e += n(t.value.typeBreak, `.${a} .content .content__item .content__item__value .type--break`)), t.value.name && (e += n(t.value.name, `.${a} .content .content__item .content__item__name`)), t.value.line && (e += n(t.value.line, `.${a} .content .content__item .content__item__value .line`)), e;
    }, d = () => {
      const e = document.querySelector(`.${a} .style-container`);
      e && (e.innerHTML = `<style>${v()}</style>`);
    };
    return A(() => {
      d(), m();
    }), s(() => [o.cStyle], () => {
      d();
    }, {
      deep: !0
    }), s(() => [o.data, o.dataAlias], () => {
      m();
    }, {
      deep: !0
    }), {
      curStyle: t,
      domOnlyClass: a,
      ...B(c)
    };
  }
});
export {
  N as default
};
