// @ths/design/es/components/list-card-common/src/index.vue2.js
import { defineComponent as j, computed as L, ref as n, onMounted as J, watch as D, nextTick as h } from "vue";
import { getValueOrDefaultValue as M, getDomRandomClass as W, setStyle as l, getCssPropertyString as g } from "../../../utils/index.js";
const _ = {
  header: "header",
  // 卡片标题
  icon: "icon",
  // 是否存在卡片Icon
  state: "state",
  // 卡片状态文本
  stateColor: "stateColor",
  // 卡片状态颜色
  stateBgColor: "stateBgColor",
  // 卡片状态背景颜色
  list: "list",
  // 卡片数据
  type: "type",
  // list 信息项类型
  label: "label",
  // 卡片信息项的文本
  labelColor: "labelColor",
  // 卡片信息项lavel文本颜色
  labelBgColor: "labelBgColor",
  // 卡片信息项label的背景颜色
  value: "value",
  // 卡片信息项的数值
  src: "src"
  // 当list中的信息项为的类型不是text时的src
}, P = j({
  name: "TListCardCommon",
  props: {
    // key 指定关键字字段，用于设置默认选中项,当且仅当radio为true时生效
    keyWord: {
      type: String,
      default: ""
    },
    // 默认选中项,即默认选中的keyWord 对应的值，
    defaultChecked: {
      type: String,
      default: ""
    },
    // 选项卡状态
    tab: {
      type: Boolean,
      default: !1
    },
    // collapse 是否开启折叠
    collapse: {
      type: Boolean,
      default: !1
    },
    // 开启折叠后，是否默认折叠
    defaultCollapse: {
      type: Boolean,
      default: !1
    },
    // 折叠按钮文本
    collapseText: {
      type: String,
      default: "数据"
    },
    // 折叠时展示多少行信息,默认折叠时只展示卡片header
    displayCount: {
      type: [String, Number],
      default: "none"
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 暂无数据时的提示信息
    noDataInfo: {
      type: String,
      default: "暂无告警信息"
    },
    // 标签样式集合
    tagStyleList: {
      type: Array,
      default: () => []
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => {
      }
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 卡片容器的父级元素
        listBody: {
          default: {}
        },
        // 卡片容器
        card: {
          default: {},
          hover: {},
          active: {}
        },
        // 卡片容器数据集合的父元素
        cardList: {
          default: {}
        },
        // 选中卡片icon
        checkedCardIcon: {
          default: {}
        },
        // 非最后一个卡片样式
        notLastCard: {
          default: {}
        },
        // 卡片标题
        cardHeader: {
          default: {},
          hover: {}
        },
        // 卡片标题图标
        cardHeaderIcon: {
          default: {},
          hover: {}
        },
        // 卡片标题文本
        cardHeaderText: {
          default: {}
        },
        // 卡片图标
        cardIcon: {
          default: {},
          active: {},
          hover: {}
        },
        // 卡片状态容器
        cardStatus: {
          default: {}
        },
        // 卡片信息项集合容器
        cardListInfo: {
          default: {}
        },
        // 卡片单行信息容器
        cardItemInfo: {
          default: {}
        },
        // 卡片中单行信息label容器
        cardItemLabel: {
          default: {}
        },
        // 卡片中单行信息Value容器
        cardItemValue: {
          default: {}
        },
        // 卡片中单行信息类型为link 时的value容器的样式
        linkItemValue: {
          default: {}
        },
        // 折叠容器
        cardCollapse: {
          default: {}
        },
        // 折叠按钮文本
        collapseText: {
          default: {}
        },
        // 折叠容器图标
        collapseIcon: {
          default: {}
        },
        // 图片类型单行信息
        cardImageItem: {
          default: {}
        },
        // 图片类型标签值
        imageItemValue: {
          default: {}
        },
        // 暂无数据
        noData: {
          default: {}
        },
        // 暂无数据图标
        noDataIcon: {
          default: {}
        },
        // 暂无数据文本
        noDataText: {
          default: {}
        },
        // 单个标签容器
        tagItem: {
          default: {}
        }
      })
    }
  },
  emits: ["card-click", "icon-click"],
  setup(c, { emit: I }) {
    const a = L(() => c.cStyle), t = W("list-card-common"), i = n([]), s = n([]), r = n([]), C = n([]), m = n([]), u = n(-1), T = L(() => ({ ..._, ...c.dataAlias })), A = () => {
      var o, d, v, y, f, p, b, H;
      let e = "";
      return a.value.wrapper && (e += l(a.value.wrapper, `.${t} `)), a.value.listBody && (e += l(a.value.listBody, `.${t} .t-list-card-common-body`)), a.value.card && (e += l(a.value.card, `.${t} .t-list-card-common-card`)), (o = a.value.card) != null && o.hover && (e += `.${t} .t-list-card-common-card:hover {
          ${g((d = a.value.card) == null ? void 0 : d.hover)}
        }`), (v = a.value.card) != null && v.active && (e += `.${t} .t-list-card-common-card-checked {
          ${g((y = a.value.card) == null ? void 0 : y.active)}
        }`), a.value.checkedCardIcon && (e += l(a.value.checkedCardIcon, `.${t} .t-list-card-common-radio-card-icon`)), a.value.notLastCard && (e += l(a.value.notLastCard, `.${t} .t-list-card-common-card:not(:last-child)`)), a.value.cardHeader && (e += l(a.value.cardHeader, `.${t} .t-list-card-common-card-head`)), (f = a.value.cardHeader) != null && f.hover && (e += `.${t} .t-list-card-common-card-head:hover {
          ${g((p = a.value.card) == null ? void 0 : p.hover)}
        }`), a.value.cardHeaderIcon && (e += l(a.value.cardHeaderIcon, `.${t} .t-list-card-common-card-head-icon`)), (b = a.value.cardHeaderIcon) != null && b.hover && (e += `.${t} .t-list-card-common-card-head-icon:hover {
          ${g((H = a.value.card) == null ? void 0 : H.hover)}
        }`), a.value.cardHeaderText && (e += l(a.value.cardHeaderText, `.${t} .t-list-card-common-card-head-text`)), a.value.cardIcon && (e += l(a.value.cardIcon, `.${t} .t-list-card-common-card-icon`)), a.value.cardStatus && (e += l(a.value.cardStatus, `.${t} .t-list-card-common-state`)), a.value.cardListInfo && (e += l(a.value.cardListInfo, `.${t} .t-list-card-common-list`)), a.value.cardItemInfo && (e += l(a.value.cardItemInfo, `.${t} .t-list-card-common-card-item`)), a.value.cardImageItem && (e += l(a.value.cardImageItem, `.${t} .t-list-card-common-card-image-item`)), a.value.imageItemValue && (e += l(a.value.imageItemValue, `.${t} .t-list-card-common-card-item-image`)), a.value.cardItemLabel && (e += l(a.value.cardItemLabel, `.${t} .t-list-card-common-card-item-label`)), a.value.cardItemValue && (e += l(a.value.cardItemValue, `.${t} .t-list-card-common-card-item-value`)), a.value.linkItemValue && (e += l(a.value.linkItemValue, `.${t} .t-list-card-common-card-item-link`)), a.value.cardItemImage && (e += l(a.value.cardItemImage, `.${t} .t-list-card-common-card-item-image`)), a.value.cardCollapse && (e += l(a.value.cardCollapse, `.${t} .t-list-card-common-collapse`)), a.value.collapseText && (e += l(a.value.collapseText, `.${t} .t-list-card-common-collapse-text`)), a.value.collapseIcon && (e += l(a.value.collapseIcon, `.${t} .t-list-card-common-collapse-icon`)), a.value.noData && (e += l(a.value.noData, `.${t} .t-list-card-common-no-data`)), a.value.noDataIcon && (e += l(a.value.noDataIcon, `.${t} .t-list-card-common-no-data-icon`)), a.value.noDataText && (e += l(a.value.noDataText, `.${t} .t-list-card-common-no-data-text`)), a.value.tagItem && (e += l(a.value.tagItem, `.${t} .t-list-card-common-tag-item`)), e;
    }, $ = () => {
      const e = document.querySelector(`.${t} .style-container`);
      e && (e.innerHTML = `<style>${A()}</style>`);
    }, O = (e) => {
      var o;
      return (o = c.tagStyleList.find((d) => d.name === e)) == null ? void 0 : o.style;
    }, V = (e) => {
      I("icon-click", JSON.parse(JSON.stringify(e)));
    }, x = (e) => {
      u.value !== -1 && e !== u.value && (m.value[u.value] = !1), e !== u.value && (m.value[e] = !0, u.value = e), I("card-click", JSON.parse(JSON.stringify(i.value[e])));
    }, B = (e) => {
      window.open(e);
    }, S = (e, o) => {
      let d;
      if (Object.prototype.toString.call(r.value[o].value) === "[object Array]" ? d = r.value[o].value[0] : d = r.value[o].value, d.style.transition = "all .5s ease-in-out", e) {
        if (C.value[o] = d.style.height, d.style.height = "0px", c.displayCount === "none" || Number(c.displayCount) === 0)
          d.style.height = "0px";
        else if (Number(c.displayCount) > 0 && Number(c.displayCount) < c.data[o].list.length) {
          let v = 0, y = 0;
          for (let f = 0; f < c.displayCount; f++)
            y = d.children[f].offsetHeight, v += y;
          d.style.height = `${v}px`;
        }
      } else
        d.style.height = C.value[o];
    }, N = (e) => {
      Array.isArray(e) && h(() => {
        e.forEach((o, d) => {
          S(o, d);
        });
      });
    }, w = (e) => {
      s.value[e] = !s.value[e], S(s.value[e], e);
    }, k = () => {
      let e;
      for (let o = 0; o < r.value.length; o++)
        Object.prototype.toString.call(r.value[o].value) === "[object Array]" ? e = r.value[o].value[0] : e = r.value[o].value, e.style.height = `${e.clientHeight}px`;
    };
    return J(() => {
      $(), k();
    }), D(() => c.cStyle, () => {
      $();
    }, { deep: !0 }), D(() => [c.data, c.keyWord, c.defaultChecked, c.tab], () => {
      if (c.data.length) {
        i.value = JSON.parse(JSON.stringify(c.data)), s.value = new Array(i.value.length).fill(c.defaultCollapse), m.value = new Array(i.value.length).fill(!1);
        for (let e = 0; e < i.value.length; e++)
          r.value[e] = n(null);
        h(() => {
          k(), N(s.value);
        });
      } else
        i.value = [];
      if (c.data.length > 0 && c.keyWord !== "" && c.defaultChecked !== "") {
        const e = i.value.findIndex((o) => String(o[c.keyWord]) === String(c.defaultChecked));
        e >= 0 && h(() => {
          m.value[e] = !0, u.value = e;
        });
      } else
        c.tab && h(() => {
          m.value[0] = !0, u.value = 0;
        });
    }, { deep: !0, immediate: !0 }), {
      curStyle: a,
      domOnlyClass: t,
      dataList: i,
      getClickCardInfo: x,
      clickIconHandler: V,
      getValueOrDefaultValue: M,
      downloadFile: B,
      clickCollapseHandler: w,
      listRef: r,
      isCollapse: s,
      stateCollection: m,
      curAlias: T,
      getMyTagStyle: O
    };
  }
});
export {
  P as default
};
