// @ths/design/es/components/list-card-common/src/index.vue.js
import V from "./index.vue2.js";
import { openBlock as e, createElementBlock as s, normalizeClass as c, Fragment as d, renderList as m, createElementVNode as o, toDisplayString as r, createCommentVNode as u, normalizeStyle as g } from "vue";
import k from "./assets/images/no-data.png.js";
import "./index.vue3.js";
import p from "../../../_virtual/_plugin-vue_export-helper.js";
const y = {
  key: 0,
  class: "t-list-card-common-body"
}, v = ["onClick"], C = {
  key: 0,
  class: "t-list-card-common-card-head"
}, D = { class: "t-list-card-common-card-head-text" }, O = ["onClick"], A = {
  key: 2,
  class: "t-list-card-common-radio-card-icon"
}, b = ["onClick"], $ = { class: "t-list-card-common-collapse-text" }, h = ["src"], L = ["onClick"], B = {
  key: 2,
  class: "t-list-card-common-card-item-value"
}, N = {
  key: 1,
  class: "t-list-card-common-no-data"
}, S = { class: "t-list-card-common-no-data-text" };
function w(l, n, z, E, F, H) {
  return e(), s("div", {
    class: c(["t-list-card-common", l.domOnlyClass])
  }, [
    l.dataList && l.dataList.length > 0 ? (e(), s("div", y, [
      (e(!0), s(d, null, m(l.dataList, (t, i) => (e(), s("div", {
        key: i,
        class: c(["t-list-card-common-card", { "t-list-card-common-card-checked": l.stateCollection[i] }]),
        onClick: (a) => l.getClickCardInfo(i)
      }, [
        l.getValueOrDefaultValue(t[l.curAlias.header], !1) ? (e(), s("div", C, [
          n[0] || (n[0] = o("span", { class: "t-list-card-common-card-head-icon" }, null, -1)),
          o("span", D, r(l.getValueOrDefaultValue(t[l.curAlias.header], "-")), 1)
        ])) : u("", !0),
        l.getValueOrDefaultValue(t[l.curAlias.icon], !1) ? (e(), s("i", {
          key: 1,
          class: "t-list-card-common-card-icon",
          onClick: (a) => l.clickIconHandler(t)
        }, null, 8, O)) : u("", !0),
        l.stateCollection[i] ? (e(), s("i", A)) : u("", !0),
        o("div", {
          class: "t-list-card-common-state",
          style: g({ background: l.getValueOrDefaultValue(t[l.curAlias.stateBgColor], ""), color: l.getValueOrDefaultValue(t[l.curAlias.stateColor], "") })
        }, r(l.getValueOrDefaultValue(t[l.curAlias.state], "")), 5),
        l.collapse ? (e(), s("div", {
          key: 3,
          class: "t-list-card-common-collapse",
          onClick: (a) => l.clickCollapseHandler(i)
        }, [
          o("div", $, r(l.collapseText), 1),
          o("i", {
            class: c(["el-icon-arrow-up t-list-card-common-collapse-icon", l.isCollapse[i] ? "collapse-icon-rotate" : ""])
          }, null, 2)
        ], 8, b)) : u("", !0),
        (e(!0), s(d, null, m(t.tagList, (a) => (e(), s("div", {
          key: a.text,
          class: "t-list-card-common-tag-item",
          style: g(l.getMyTagStyle(a.styleName))
        }, r(a.text), 5))), 128)),
        o("div", {
          ref_for: !0,
          ref: l.listRef[i],
          class: "t-list-card-common-list"
        }, [
          (e(!0), s(d, null, m(l.getValueOrDefaultValue(t[l.curAlias.list]), (a, f) => (e(), s("div", {
            key: f,
            class: c(["t-list-card-common-card-item", a.type === "image" ? "t-list-card-common-card-image-item" : ""])
          }, [
            o("div", {
              class: "t-list-card-common-card-item-label",
              style: g({
                color: l.getValueOrDefaultValue(a[l.curAlias.labelColor], ""),
                background: l.getValueOrDefaultValue(a[l.curAlias.labelBgColor], "")
              })
            }, r(l.getValueOrDefaultValue(a[l.curAlias.label], "-")), 5),
            l.getValueOrDefaultValue(a[l.curAlias.type], "") === "image" ? (e(), s("img", {
              key: 0,
              class: "t-list-card-common-card-item-image",
              src: l.getValueOrDefaultValue(a[l.curAlias.src], ""),
              alt: "",
              draggable: "false"
            }, null, 8, h)) : l.getValueOrDefaultValue(a[l.curAlias.type], "") === "link" ? (e(), s("div", {
              key: 1,
              class: "t-list-card-common-card-item-link",
              onClick: (T) => l.downloadFile(l.getValueOrDefaultValue(a[l.curAlias.src], ""))
            }, r(l.getValueOrDefaultValue(a[l.curAlias.value], "-")), 9, L)) : (e(), s("div", B, r(l.getValueOrDefaultValue(a[l.curAlias.value], "-")), 1))
          ], 2))), 128))
        ], 512)
      ], 10, v))), 128))
    ])) : (e(), s("div", N, [
      n[1] || (n[1] = o("img", {
        class: "t-list-card-common-no-data-icon",
        src: k,
        draggable: "false"
      }, null, -1)),
      o("div", S, r(l.noDataInfo), 1)
    ])),
    n[2] || (n[2] = o("div", { class: "style-container" }, null, -1))
  ], 2);
}
const G = /* @__PURE__ */ p(V, [["render", w]]);
export {
  G as default
};
