// @ths/design/es/components/chart-double-xaxis/src/index.vue.js
import n from "./index.vue2.js";
import { openBlock as m, createElementBlock as c, normalizeClass as p, normalizeStyle as r, withDirectives as i, createElementVNode as e, vShow as a } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  ref: "chartId",
  class: "chart"
}, y = { class: "empty" }, f = ["src"];
function h(t, s, v, E, b, k) {
  var o, l;
  return m(), c("div", {
    class: p(["t-chart-double-xaxis", t.domOnlyClass]),
    style: r({ ...(l = (o = t.cStyle) == null ? void 0 : o.wrapper) == null ? void 0 : l.default })
  }, [
    i(e("div", u, null, 512), [
      [a, t.checkEmpty]
    ]),
    i(e("div", y, [
      e("img", {
        class: "empty-img",
        style: r({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [a, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const C = /* @__PURE__ */ d(n, [["render", h]]);
export {
  C as default
};
