// @ths/design/es/components/chart-double-xaxis/src/index.vue2.js
import { defineComponent as N, computed as q, ref as P, reactive as U, onMounted as B, onUnmounted as G, watch as w, toRefs as H, nextTick as W } from "vue";
import * as D from "echarts";
import u from "lodash/merge";
import d from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as X, DEFAULT_OPTIONS_AXIS as K } from "../../../utils/constant.js";
import { getDomRandomClass as J, setStyle as p } from "../../../utils/index.js";
const O = u(d(X), d(K), {
  xAxis: {
    axisLine: {
      show: !0
    },
    type: "value"
  },
  yAxis: {
    type: "category",
    inverse: !0
  },
  grid: {
    containLabel: !0,
    top: "8%",
    bottom: "8%",
    left: "2%",
    right: "8%"
  }
}), te = N({
  name: "TChartDoubleXaxis",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // Y 轴类目取resData数据中的别名设置
        yAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 折线图主题配置
    line: {
      type: Object,
      default: () => ({})
    },
    // 柱状图主题配置
    bar: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: L }) {
    const m = J("chart-double-xaxis"), y = q(() => e.cStyle), x = P(null);
    let f = null;
    const o = U({
      // 图表数据
      chartData: [],
      lineOptions: {},
      barOptions: {}
    }), T = () => {
      var t, n;
      (n = (t = e.seriesSetting) == null ? void 0 : t.data) != null && n.length && e.data && (o.chartData = e.seriesSetting.data.map((l) => {
        const { name: s, dataKey: A } = l;
        return {
          name: s || "--",
          data: e.data.map((c) => c[A])
        };
      }));
    }, M = () => {
      let t = "";
      return y.value.wrapper && (t += p(y.value.wrapper, `.${m}`)), y.value.emptyContainer && (t += p(y.value.emptyContainer, `.${m} .empty`)), y.value.emptyImg && (t += p(y.value.emptyImg, `.${m} .empty .empty-img`)), y.value.emptyText && (t += p(y.value.emptyText, `.${m} .empty .empty-text`)), t;
    }, j = () => {
      const t = document.querySelector(`.${m} .style-container`);
      t && (t.innerHTML = `<style>${M()}</style>`);
    }, R = q(() => {
      var t, n, l, s;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((c) => {
            var g;
            return (c == null ? void 0 : c.data) && ((g = c == null ? void 0 : c.data) == null ? void 0 : g.length) !== 0;
          });
        if (typeof ((n = e.options) == null ? void 0 : n.series) == "object")
          return ((s = (l = e.options.series) == null ? void 0 : l.data) == null ? void 0 : s.length) === 0;
      }
      return !!e.data.length;
    }), r = (t, n = "#FFF") => {
      var l, s;
      return ((l = e.options.color) == null ? void 0 : l[t]) || ((s = O.color) == null ? void 0 : s[t]) || n;
    }, v = () => {
      var c, g, E, I;
      const { series: t } = e.options;
      let n = [], l = [], s = [];
      t ? n = t.map((i, a) => {
        var S, b, h;
        return u(
          {
            name: "",
            type: "line",
            smooth: !0,
            symbol: "circle",
            symbolSize: 8,
            areaStyle: {
              color: new D.graphic.LinearGradient(0, 0, 0, 1, [
                {
                  offset: 0,
                  color: r(a)
                  // 100% 处的颜色
                },
                {
                  offset: 1,
                  color: "transparent"
                }
              ])
            },
            itemStyle: {
              color: {
                type: "linear",
                x: 0,
                y: 1,
                r: 1,
                colorStops: [
                  {
                    offset: 0,
                    color: r(a)
                    // 0% 处的颜色
                  },
                  {
                    offset: 1,
                    color: r(a)
                    // 100% 处的颜色
                  }
                ]
              },
              opacity: 0.7,
              borderColor: r(a),
              // 边框颜色
              borderWidth: 2,
              borderRadius: 2,
              shadowColor: r(a),
              // 图形阴影颜色
              shadowBlur: 10
              // 图形阴影
            },
            lineStyle: {
              color: r(a)
            }
          },
          i.type === "bar" ? o == null ? void 0 : o.barOptions : o == null ? void 0 : o.lineOptions,
          (h = (b = (S = e.seriesSetting) == null ? void 0 : S.data) == null ? void 0 : b[a]) == null ? void 0 : h.style,
          d(i)
        );
      }) : (g = (c = e.seriesSetting) == null ? void 0 : c.data) != null && g.length && e.data && (n = e.seriesSetting.data.map((i, a) => {
        var S, b, h, z, F, _, $, k;
        return u(
          {
            name: i.name,
            type: "line",
            smooth: !0,
            symbol: "circle",
            symbolSize: 8,
            areaStyle: {
              color: new D.graphic.LinearGradient(0, 0, 0, 1, [
                {
                  offset: 0,
                  color: r(a)
                },
                {
                  offset: 1,
                  color: "transparent"
                }
              ])
            },
            itemStyle: {
              color: {
                type: "linear",
                x: 0,
                y: 1,
                r: 1,
                colorStops: [
                  {
                    offset: 0,
                    color: r(a)
                    // 0% 处的颜色
                  },
                  {
                    offset: 1,
                    color: r(a)
                    // 100% 处的颜色
                  }
                ]
              },
              borderColor: r(a),
              // 边框颜色
              borderWidth: 2,
              borderRadius: 2,
              shadowColor: r(a),
              // 图形阴影颜色
              shadowBlur: 10
              // 图形阴影
            },
            lineStyle: {
              color: r(a)
            },
            data: e.data.length ? (S = o.chartData[a]) == null ? void 0 : S.data : []
          },
          ((F = (z = (h = (b = e.seriesSetting) == null ? void 0 : b.data) == null ? void 0 : h[a]) == null ? void 0 : z.style) == null ? void 0 : F.type) === "bar" ? o == null ? void 0 : o.barOptions : o == null ? void 0 : o.lineOptions,
          (k = ($ = (_ = e.seriesSetting) == null ? void 0 : _.data) == null ? void 0 : $[a]) == null ? void 0 : k.style,
          d(i)
        );
      }), (E = e.seriesSetting) != null && E.yAxisLabel ? l = (I = e.data) != null && I.length ? e.data.map((i) => i[e.seriesSetting.yAxisLabel]) : [] : l = e.data.map((i, a) => a)), e.options.xAxis && Array.isArray(e.options.xAxis) ? s = e.options.xAxis.map((i) => u(d(O.xAxis), i)) : s = u(d(O.xAxis), e.options.xAxis || {});
      const A = u(
        {
          series: n,
          yAxis: {
            data: l
          },
          xAxis: s
        },
        d(O),
        e.options
      );
      f.clear();
      try {
        f.setOption(A);
      } catch (i) {
        console.log(i);
      }
    }, C = () => {
      W(() => {
        f.resize();
      });
    };
    return B(() => {
      j(), x.value && (f = D.init(x.value), T(), v(), f.on("click", (t) => {
        L("com-click", t);
      }), f.on("legendselectchanged", (t) => {
        L("legend-select", t);
      })), window.addEventListener("resize", C);
    }), G(() => {
      window.removeEventListener("resize", C);
    }), w(
      () => [e.line, e.bar],
      (t) => {
        o.lineOptions = d(e.line), o.barOptions = d(e.bar), v();
      },
      {
        deep: !0
      }
    ), w(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting
      ],
      () => {
        T(), v();
      },
      {
        deep: !0
      }
    ), w(
      () => e.cStyle,
      () => {
        j(), C();
      },
      {
        deep: !0
      }
    ), {
      ...H(o),
      chartId: x,
      curStyle: y,
      checkEmpty: R,
      domOnlyClass: m
    };
  }
});
export {
  te as default
};
