// @ths/design/es/components/slider/src/index.vue.js
import i from "./index.vue2.js";
import { resolveComponent as u, openBlock as r, createElementBlock as o, normalizeClass as p, Fragment as c, renderList as f, createElementVNode as s, toDisplayString as m, createVNode as b, mergeProps as v } from "vue";
import "./index.vue3.js";
import V from "../../../_virtual/_plugin-vue_export-helper.js";
const g = { class: "t-slider-item-label" }, A = { class: "t-slider-item-box" };
function C(e, a, y, $, h, k) {
  const d = u("el-slider");
  return r(), o("div", {
    class: p(["t-slider", e.domOnlyClass])
  }, [
    (r(!0), o(c, null, f(e.dataArray, (l) => {
      var n;
      return r(), o("div", {
        key: l.label,
        class: "t-slider-item"
      }, [
        s("div", g, m(e.getValueOrDefaultValue(l[e.curAlias.label], "-")), 1),
        s("div", A, [
          b(d, v({
            modelValue: l[e.curAlias.value],
            "onUpdate:modelValue": (t) => l[e.curAlias.value] = t,
            ref_for: !0
          }, { ...e.filterAttrs(e.$attrs), ...l == null ? void 0 : l.option }, {
            disabled: ((n = l == null ? void 0 : l.option) == null ? void 0 : n.disable) || l[e.curAlias.disable],
            onChange: e.sliderHandlerChange,
            onInput: e.sliderHandlerInput
          }), null, 16, ["modelValue", "onUpdate:modelValue", "disabled", "onChange", "onInput"])
        ])
      ]);
    }), 128)),
    a[0] || (a[0] = s("div", { class: "style-container" }, null, -1))
  ], 2);
}
const E = /* @__PURE__ */ V(i, [["render", C]]);
export {
  E as default
};
