// @ths/design/es/components/slider/src/index.vue2.js
import { defineComponent as c, computed as d, ref as m, onMounted as y, watch as n } from "vue";
import _ from "lodash/debounce";
import { getValueOrDefaultValue as b, filterAttrs as S, getDomRandomClass as $, setStyle as a } from "../../../utils/index.js";
const I = {
  label: "label",
  // 滑块前面的文本
  value: "value",
  // 滑块绑定的数值
  disable: "disable"
  // 是否禁用
}, O = c({
  name: "TSlider",
  props: {
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => {
      }
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 单个滑块容器
        sliderItem: {
          default: {}
        },
        // 滑块文本容器
        sliderLabel: {
          default: {}
        },
        // 滑块盒子容器
        sliderBox: {
          default: {}
        },
        // 滑块输入框外层容器  当props.showInput为true时生效
        sliderInputWrapper: {
          default: {}
        },
        // 滑块输入框容器
        sliderInput: {
          default: {}
        },
        // 滑块进度条容器
        sliderProgress: {
          default: {}
        },
        // 滑块断点容器
        sliderStop: {
          default: {}
        },
        // 刻度字体容器
        stopText: {
          default: {}
        },
        // 滑过刻度容器
        sliderOver: {
          default: {}
        },
        // 滑块按钮容器
        sliderButton: {
          default: {}
        }
      })
    }
  },
  emits: ["change", "input"],
  setup(r, { emit: s }) {
    const e = d(() => r.cStyle), t = $("slider"), o = d(() => ({ ...I, ...r.dataAlias })), i = m([]), p = () => {
      let l = "";
      return e.value.wrapper && (l += a(e.value.wrapper, `.${t}`)), e.value.sliderItem && (l += a(e.value.sliderItem, `.${t} .t-slider-item`)), e.value.sliderLabel && (l += a(e.value.sliderLabel, `.${t} .t-slider-item-label`)), e.value.sliderBox && (l += a(e.value.sliderBox, `.${t} .t-slider-item-box`)), e.value.sliderInputWrapper && (l += a(e.value.sliderInputWrapper, `.${t} .el-slider__input`)), e.value.sliderInput && (l += a(e.value.sliderInput, `.${t} .el-input-number.is-without-controls .el-input__inner`)), e.value.sliderProgress && (l += a(e.value.sliderProgress, `.${t} .el-slider__runway.show-input`)), e.value.sliderStop && (l += a(e.value.sliderStop, `.${t} .el-slider__stop`)), e.value.stopText && (l += a(e.value.stopText, `.${t} .el-slider__marks-text`)), e.value.sliderOver && (l += a(e.value.sliderOver, `.${t} .el-slider__bar`)), e.value.sliderButton && (l += a(e.value.sliderButton, `.${t} .el-slider__button`)), l;
    }, u = () => {
      const l = document.querySelector(`.${t} .style-container`);
      l && (l.innerHTML = `<style>${p()}</style>`);
    }, f = (l) => {
      s("change", i.value);
    }, v = _((l) => {
      s("input", i.value);
    }, 500);
    return y(() => {
      u();
    }), n(() => r.cStyle, () => {
      u();
    }, { deep: !0 }), n(() => r.data, () => {
      r.data.length > 0 ? i.value = JSON.parse(JSON.stringify(r.data)) : i.value = [];
    }, { immediate: !0, deep: !0 }), {
      curStyle: e,
      domOnlyClass: t,
      sliderHandlerChange: f,
      sliderHandlerInput: v,
      curAlias: o,
      getValueOrDefaultValue: b,
      filterAttrs: S,
      dataArray: i
    };
  }
});
export {
  O as default
};
