// @ths/design/es/components/list-card/src/index.vue2.js
import { defineComponent as s, computed as m, ref as y, onMounted as v, watch as u } from "vue";
import { getDomRandomClass as S, setStyle as o } from "../../../utils/index.js";
import "../../list-card-item/index.js";
import C from "../../list-card-item/src/index.vue.js";
const b = s({
  name: "TListCard",
  components: {
    TListCardItem: C
  },
  props: {
    // 横向间隔
    spacingX: {
      default: 8,
      type: Number
    },
    // 列数
    col: {
      default: 4,
      type: Number
    },
    // 排列方向 horizontal  vertical
    direction: {
      default: "vertical",
      type: String
    },
    // 单位是否换行
    unitBreak: {
      default: !1,
      type: Boolean
    },
    // 交换标题和值的排列顺序
    exchangeOrder: {
      default: !1,
      type: Boolean
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 字段关键词
    keyField: {
      default: () => ({
        icon: "icon",
        title: "title",
        num: "num",
        unit: "unit",
        trend: "trend",
        color: "color",
        special: "special",
        selectIcon: "selectIcon",
        unitStyle: "unitStyle"
      }),
      type: Object
    },
    // 数据中唯一属性key
    activeKeyField: {
      type: String,
      default: "title"
    },
    // 选中项
    active: {
      type: String,
      default: ""
    },
    // // 自定义样式
    cStyle: {
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 内层容器
        innerWrapper: {
          default: {
            background: "red"
          }
        },
        // 单项容器最外层容器
        outerWrapper: {
          default: {}
        },
        // 单项容器
        itemWrapper: {
          default: {},
          active: {}
        },
        // 右侧文本区域容器
        rightContent: {
          default: {}
        },
        // 特殊状态下的单项容器
        specialItemWrapper: {
          default: {}
        },
        // 图标
        icon: {
          default: {},
          active: {}
        },
        // 卡片图标
        cardIcon: {
          default: {}
        },
        // 特殊状态下的卡片图标
        specialCardIcon: {
          default: {}
        },
        // 标题
        title: {
          default: {},
          active: {}
        },
        // 数值
        num: {
          default: {},
          active: {}
        },
        // 单位
        unit: {
          default: {},
          active: {}
        }
      }),
      type: Object
    }
  },
  // 事件定义
  emits: ["card-click"],
  setup(t, { emit: d }) {
    const a = m(() => t.cStyle), r = y(t.active), p = (e) => {
      r.value = e[t.activeKeyField], d("card-click", e);
    }, l = S("list-card"), f = () => {
      var c, i;
      let e = "";
      return (c = a.value) != null && c.wrapper && (e += o(a.value.wrapper, `.${l}`)), (i = a.value) != null && i.innerWrapper && (e += o(a.value.innerWrapper, `.${l} .wrapper`)), e;
    }, n = () => {
      const e = document.querySelector(`.${l} .style-container`);
      e && (e.innerHTML = `<style>${f()}</style>`);
    };
    return v(() => {
      n();
    }), u(() => t.cStyle, () => {
      n();
    }, { deep: !0 }), u(() => t.active, () => {
      r.value = t.active;
    }), {
      domOnlyClass: l,
      curStyle: a,
      curActive: r,
      clickListCard: p
    };
  }
});
export {
  b as default
};
