// @ths/design/es/components/list-card/src/index.vue.js
import B from "./index.vue2.js";
import { resolveComponent as I, openBlock as r, createElementBlock as t, normalizeClass as $, normalizeStyle as v, createElementVNode as g, Fragment as W, renderList as F, createVNode as L } from "vue";
import "./index.vue3.js";
import b from "../../../_virtual/_plugin-vue_export-helper.js";
const z = { class: "wrapper" };
function E(e, i, N, O, V, A) {
  var a, l;
  const h = I("t-list-card-item");
  return r(), t("div", {
    class: $(["t-list-card", e.domOnlyClass]),
    style: v({ ...(l = (a = e.curStyle) == null ? void 0 : a.wrapper) == null ? void 0 : l.default })
  }, [
    g("div", z, [
      (r(!0), t(W, null, F(e.data, (n, w) => {
        var o, d, c, s, p, u, y, m, f, k, C, S;
        return r(), t("div", {
          key: w,
          class: "wrapper__col",
          style: v({ width: 100 / e.col + "%", padding: "0 " + e.spacingX / 2 + "px", ...(d = (o = e.curStyle) == null ? void 0 : o.outerWrapper) == null ? void 0 : d.default })
        }, [
          L(h, {
            direction: e.direction,
            "unit-break": e.unitBreak,
            data: n,
            "key-field": e.keyField,
            "exchange-order": e.exchangeOrder,
            "c-style": {
              wrapper: (c = e.curStyle) == null ? void 0 : c.itemWrapper,
              specialWrapper: (s = e.curStyle) == null ? void 0 : s.specialItemWrapper,
              icon: (p = e.curStyle) == null ? void 0 : p.icon,
              title: (u = e.curStyle) == null ? void 0 : u.title,
              num: (y = e.curStyle) == null ? void 0 : y.num,
              unit: (m = e.curStyle) == null ? void 0 : m.unit,
              numBox: (f = e.curStyle) == null ? void 0 : f.numBox,
              cardIcon: (k = e.curStyle) == null ? void 0 : k.cardIcon,
              specialCardIcon: (C = e.curStyle) == null ? void 0 : C.specialCardIcon,
              rightContent: (S = e.curStyle) == null ? void 0 : S.rightContent
            },
            "active-key-field": e.activeKeyField,
            active: e.curActive,
            onCardClick: (K) => e.clickListCard(n)
          }, null, 8, ["direction", "unit-break", "data", "key-field", "exchange-order", "c-style", "active-key-field", "active", "onCardClick"])
        ], 4);
      }), 128))
    ]),
    i[0] || (i[0] = g("div", { class: "style-container" }, null, -1))
  ], 6);
}
const G = /* @__PURE__ */ b(B, [["render", E]]);
export {
  G as default
};
