// @ths/c-common-base-button/es/index.js
import { defineComponent, reactive, toRefs, resolveComponent, openBlock, createBlock, withCtx, createCommentVNode, resolveDynamicComponent, renderSlot, createElementBlock, toDisplayString, ref, createVNode, createTextVNode, createElementVNode, Fragment, normalizeClass } from 'vue';
import { ElMessage } from 'element-plus';

var script$3 = defineComponent({
    name: 'TButton',
    props: {
        // 按钮文字
        text: {
            type: String,
            required: true,
            default: '按钮',
        },
        type: {
            type: String,
            default: '',
        },
        // 按钮尺寸
        size: {
            type: String,
            default: 'mini',
        },
        plain: {
            type: Boolean,
            default: false,
        },
        // 按钮图标名称
        icon: {
            type: String,
            default: '',
        },
        // 按钮图标尺寸
        iconSize: {
            type: Number,
            default: 13,
        },
    },
    emits: ['click'],
    setup(props, { emit }) {
        // 响应式数据
        const state = reactive({});
        /**
         * 处理按钮点击事件
         */
        const handleButtonClick = () => {
            emit('click');
        };
        return {
            ...toRefs(state),
            handleButtonClick,
        };
    },
});

const _hoisted_1$3 = { key: 1 };

function render$3(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_el_icon = resolveComponent("el-icon");
  const _component_el_button = resolveComponent("el-button");

  return (openBlock(), createBlock(_component_el_button, {
    type: _ctx.type,
    plain: _ctx.plain,
    size: _ctx.size,
    onClick: _ctx.handleButtonClick
  }, {
    default: withCtx(() => [
      createCommentVNode(" 按钮图标 "),
      (_ctx.icon)
        ? (openBlock(), createBlock(_component_el_icon, {
            key: 0,
            size: _ctx.iconSize
          }, {
            default: withCtx(() => [
              (openBlock(), createBlock(resolveDynamicComponent(_ctx.icon)))
            ]),
            _: 1 /* STABLE */
          }, 8 /* PROPS */, ["size"]))
        : createCommentVNode("v-if", true),
      createCommentVNode(" 自定义内容 "),
      renderSlot(_ctx.$slots, "default"),
      createCommentVNode(" 按钮文字 "),
      (_ctx.text)
        ? (openBlock(), createElementBlock("span", _hoisted_1$3, toDisplayString(_ctx.text), 1 /* TEXT */))
        : createCommentVNode("v-if", true)
    ]),
    _: 3 /* FORWARDED */
  }, 8 /* PROPS */, ["type", "plain", "size", "onClick"]))
}

script$3.render = render$3;
script$3.__file = "package/ths/components/c-common-base-button/src/view/t-button.vue";

var script$2 = defineComponent({
    name: 'TExcelButton',
    props: {
        // 导出Excel对应的url,get请求，参数拼到url后
        exportExcelUrl: {
            type: String,
            default: '',
        },
        // 通过模板导出Excel对应的url，get请求，参数拼到url后
        exportExcelByTemplateUrl: {
            type: String,
            default: '',
        },
        // excel模板下载url
        templateDownloadUrl: {
            type: String,
            default: '',
        },
        // Excel导入url，post请求，参数通过uploadRequestParam传递
        uploadExcelUrl: {
            type: String,
            default: '',
        },
        // 导入携带的额外参数
        uploadRequestParam: {
            type: Object,
            default: () => ({}),
        },
        // 文件上传携带请求头
        uploadHeaders: {
            type: Object,
            default: () => ({}),
        },
        // 错误文件下载Url
        errorFileDownloadUrl: {
            type: String,
            default: '',
        },
        // 导出excel按钮文字&下拉框内容
        excelTextArr: {
            type: Array,
            default: () => ([
                { text: 'Excel导入/导出示例' },
                { text: '导入Excel' },
                { text: '导出Excel' },
                { text: '导出Excel(by模板)' },
            ])
        }
    },
    emits: ['on-upload-success'],
    setup(props, { emit }) {
        const uploadRef = ref();
        const state = reactive({
            showDialog: false,
            accept: '.xls, .xlsx',
            fileList: [],
            fileName: '',
            errorFileName: '',
            loading: false, // 导入loading
        });
        /**
         * Excel操作列表触发事件
         */
        function handleCommandExcel(command) {
            if (command === 'export') {
                window.location.href = `${props.exportExcelUrl}`;
            }
            else if (command === 'exportByTemplate') {
                window.location.href = `${props.exportExcelByTemplateUrl}`;
            }
            else {
                state.showDialog = true;
            }
        }
        /**
         * 关闭导入excel弹框
         */
        function onClose() {
            state.showDialog = false;
        }
        /**
         * input框点击事件
         */
        function focusInput(event) {
            document.getElementById('singleSelectButton')?.click();
            event.target.blur();
        }
        /**
         * 下载模板
         */
        function handleDownloadModel() {
            if (props.templateDownloadUrl?.length) {
                window.location.href = props.templateDownloadUrl;
            }
            else {
                ElMessage.warning('请设置【templateDownloadUrl】');
            }
        }
        /**
         * 导入
         */
        function handleImport() {
            if (state.fileList.length && state.fileList[0].status === 'ready') {
                if (props.uploadExcelUrl.length === 0) {
                    ElMessage.warning('请设置【uploadExcelUrl】对应值');
                    return;
                }
                state.loading = true;
                uploadRef.value.submit();
            }
            else {
                ElMessage.warning('请先选择文件！！！');
            }
        }
        /**
         * 上传文件前调用，如果返回false则停止上传
         */
        function beforeUpload(file) {
            const fileType = file.name.substring(file.name.lastIndexOf('.') + 1);
            if (!state.accept.includes(fileType)) {
                ElMessage.warning(`只允许上传${state.accept}类型文件！！！`);
                return false;
            }
            return true;
        }
        /**
         * 文件上传成功时调用
         */
        function onSuccess(response, file) {
            state.loading = false;
            if (response.code === 500) {
                ElMessage.error(`导入失败，${response.message}`);
                const tempFile = file;
                tempFile.status = 'ready';
                return;
            }
            if (response.code === 200 && response.data.checked) {
                ElMessage.success('导入成功！！！');
                state.showDialog = false;
                emit('on-upload-success');
            }
            else {
                const tempFile = file;
                tempFile.status = 'ready';
                state.errorFileName = response.data.errorTempFileName;
                ElMessage.error(`导入失败，${response.data.errorMessage}`);
            }
        }
        /**
         * 文件上传失败时触发事件
         */
        function onError(file) {
            state.loading = false;
            const tempFile = file;
            tempFile.status = 'ready';
            state.fileList.push(tempFile);
            ElMessage.error('导入失败！！！');
        }
        /**
         * 添加文件、上传成功和上传失败时都会被调用
         * 此处主要处理文件名展示问题
         */
        function onChange(file) {
            if (file.status === 'ready') {
                state.fileList = [];
                state.fileName = file.name;
                state.fileList.push(file);
            }
        }
        /**
         * 打开弹窗是触发
         */
        function onOpen() {
            state.fileList = [];
            state.fileName = '';
            state.errorFileName = '';
        }
        /**
         * 下载错误文件
         */
        function downLoadErrorFile() {
            if (!props.errorFileDownloadUrl || !props.errorFileDownloadUrl.length) {
                ElMessage.warning('请设置错误文件下载URL');
                return;
            }
            let errorFileUrl = '';
            if (props.errorFileDownloadUrl.indexOf('?') !== -1) {
                errorFileUrl = `${props.errorFileDownloadUrl}&fileName=${state.errorFileName}`;
            }
            else {
                errorFileUrl = `${props.errorFileDownloadUrl}?fileName=${state.errorFileName}`;
            }
            window.location.href = errorFileUrl;
        }
        return {
            ...toRefs(state),
            handleCommandExcel,
            onClose,
            onOpen,
            focusInput,
            handleDownloadModel,
            handleImport,
            beforeUpload,
            onSuccess,
            onError,
            onChange,
            uploadRef,
            downLoadErrorFile,
        };
    },
});

const _hoisted_1$2 = { class: "common-base-excel" };
const _hoisted_2 = /*#__PURE__*/createTextVNode(" 导入Excel数据");
const _hoisted_3 = { class: "m-t-sm m-b-sm flex flex-end" };
const _hoisted_4 = /*#__PURE__*/createTextVNode(" 下载模板 ");
const _hoisted_5 = /*#__PURE__*/createTextVNode(" 导入 ");
const _hoisted_6 = /*#__PURE__*/createTextVNode(" 取消 ");
const _hoisted_7 = /*#__PURE__*/createTextVNode(" 选择 ");
const _hoisted_8 = { key: 0 };
const _hoisted_9 = /*#__PURE__*/createElementVNode("span", null, "错误文件：", -1 /* HOISTED */);

function render$2(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_document = resolveComponent("document");
  const _component_el_icon = resolveComponent("el-icon");
  const _component_arrow_down = resolveComponent("arrow-down");
  const _component_el_button = resolveComponent("el-button");
  const _component_upload = resolveComponent("upload");
  const _component_el_dropdown_item = resolveComponent("el-dropdown-item");
  const _component_download = resolveComponent("download");
  const _component_tickets = resolveComponent("tickets");
  const _component_el_dropdown_menu = resolveComponent("el-dropdown-menu");
  const _component_el_dropdown = resolveComponent("el-dropdown");
  const _component_info_filled = resolveComponent("info-filled");
  const _component_upload_filled = resolveComponent("upload-filled");
  const _component_el_input = resolveComponent("el-input");
  const _component_el_upload = resolveComponent("el-upload");
  const _component_el_dialog = resolveComponent("el-dialog");

  return (openBlock(), createElementBlock("div", _hoisted_1$2, [
    createVNode(_component_el_dropdown, {
      trigger: "click",
      onCommand: _ctx.handleCommandExcel
    }, {
      dropdown: withCtx(() => [
        createVNode(_component_el_dropdown_menu, { class: "dropdown-menu" }, {
          default: withCtx(() => [
            (_ctx.uploadExcelUrl.length)
              ? (openBlock(), createBlock(_component_el_dropdown_item, {
                  key: 0,
                  command: "import"
                }, {
                  default: withCtx(() => [
                    createVNode(_component_el_icon, { size: 13 }, {
                      default: withCtx(() => [
                        createVNode(_component_upload)
                      ]),
                      _: 1 /* STABLE */
                    }),
                    createTextVNode(" " + toDisplayString(_ctx.excelTextArr[1].text), 1 /* TEXT */)
                  ]),
                  _: 1 /* STABLE */
                }))
              : createCommentVNode("v-if", true),
            (_ctx.exportExcelUrl.length)
              ? (openBlock(), createBlock(_component_el_dropdown_item, {
                  key: 1,
                  command: "export"
                }, {
                  default: withCtx(() => [
                    createVNode(_component_el_icon, { size: 13 }, {
                      default: withCtx(() => [
                        createVNode(_component_download)
                      ]),
                      _: 1 /* STABLE */
                    }),
                    createTextVNode(" " + toDisplayString(_ctx.excelTextArr[2].text), 1 /* TEXT */)
                  ]),
                  _: 1 /* STABLE */
                }))
              : createCommentVNode("v-if", true),
            (_ctx.exportExcelByTemplateUrl.length)
              ? (openBlock(), createBlock(_component_el_dropdown_item, {
                  key: 2,
                  command: "exportByTemplate"
                }, {
                  default: withCtx(() => [
                    createVNode(_component_el_icon, { size: 13 }, {
                      default: withCtx(() => [
                        createVNode(_component_tickets)
                      ]),
                      _: 1 /* STABLE */
                    }),
                    createTextVNode(" " + toDisplayString(_ctx.excelTextArr[3].text), 1 /* TEXT */)
                  ]),
                  _: 1 /* STABLE */
                }))
              : createCommentVNode("v-if", true)
          ]),
          _: 1 /* STABLE */
        })
      ]),
      default: withCtx(() => [
        createVNode(_component_el_button, {
          class: "dropdown-button",
          size: "mini",
          plain: ""
        }, {
          default: withCtx(() => [
            createVNode(_component_el_icon, { size: 13 }, {
              default: withCtx(() => [
                createVNode(_component_document)
              ]),
              _: 1 /* STABLE */
            }),
            createElementVNode("span", null, toDisplayString(_ctx.excelTextArr[0].text), 1 /* TEXT */),
            createVNode(_component_el_icon, {
              size: 13,
              class: "el-icon--right"
            }, {
              default: withCtx(() => [
                createVNode(_component_arrow_down)
              ]),
              _: 1 /* STABLE */
            })
          ]),
          _: 1 /* STABLE */
        })
      ]),
      _: 1 /* STABLE */
    }, 8 /* PROPS */, ["onCommand"]),
    createCommentVNode(" 导入Excel弹框 "),
    createVNode(_component_el_dialog, {
      "custom-class": "excel-import",
      "model-value": _ctx.showDialog,
      "close-on-click-modal": false,
      "close-on-press-escape": false,
      onClose: _ctx.onClose,
      onOpen: _ctx.onOpen
    }, {
      title: withCtx(() => [
        createElementVNode("span", null, [
          createVNode(_component_el_icon, {
            size: 16,
            style: {"vertical-align":"-2px"}
          }, {
            default: withCtx(() => [
              createVNode(_component_info_filled)
            ]),
            _: 1 /* STABLE */
          }),
          _hoisted_2
        ])
      ]),
      default: withCtx(() => [
        createElementVNode("div", _hoisted_3, [
          createVNode(_component_el_button, {
            size: "small",
            type: "primary",
            plain: "",
            onClick: _ctx.handleDownloadModel
          }, {
            default: withCtx(() => [
              _hoisted_4
            ]),
            _: 1 /* STABLE */
          }, 8 /* PROPS */, ["onClick"]),
          createVNode(_component_el_button, {
            size: "small",
            type: "primary",
            plain: "",
            loading: _ctx.loading,
            onClick: _ctx.handleImport
          }, {
            default: withCtx(() => [
              _hoisted_5
            ]),
            _: 1 /* STABLE */
          }, 8 /* PROPS */, ["loading", "onClick"]),
          createVNode(_component_el_button, {
            size: "small",
            type: "primary",
            plain: "",
            onClick: _ctx.onClose
          }, {
            default: withCtx(() => [
              _hoisted_6
            ]),
            _: 1 /* STABLE */
          }, 8 /* PROPS */, ["onClick"])
        ]),
        createVNode(_component_el_input, {
          modelValue: _ctx.fileName,
          "onUpdate:modelValue": _cache[0] || (_cache[0] = $event => ((_ctx.fileName) = $event)),
          placeholder: "请选择excel文件...",
          readonly: "",
          onFocus: _ctx.focusInput
        }, {
          prefix: withCtx(() => [
            createVNode(_component_el_icon, { class: "el-input__icon" }, {
              default: withCtx(() => [
                createVNode(_component_upload_filled)
              ]),
              _: 1 /* STABLE */
            })
          ]),
          append: withCtx(() => [
            createVNode(_component_el_button, {
              size: "small",
              onClick: _ctx.focusInput
            }, {
              default: withCtx(() => [
                _hoisted_7
              ]),
              _: 1 /* STABLE */
            }, 8 /* PROPS */, ["onClick"])
          ]),
          _: 1 /* STABLE */
        }, 8 /* PROPS */, ["modelValue", "onFocus"]),
        createVNode(_component_el_upload, {
          ref: "uploadRef",
          accept: _ctx.accept,
          "show-file-list": false,
          multiple: false,
          action: _ctx.uploadExcelUrl,
          data: _ctx.uploadRequestParam,
          "auto-upload": false,
          "before-upload": _ctx.beforeUpload,
          "on-error": _ctx.onError,
          "on-success": _ctx.onSuccess,
          "on-change": _ctx.onChange,
          "file-list": _ctx.fileList
        }, {
          default: withCtx(() => [
            createVNode(_component_el_button, {
              id: "singleSelectButton",
              style: {"display":"none"}
            })
          ]),
          _: 1 /* STABLE */
        }, 8 /* PROPS */, ["accept", "action", "data", "before-upload", "on-error", "on-success", "on-change", "file-list"]),
        (_ctx.errorFileName.length)
          ? (openBlock(), createElementBlock("div", _hoisted_8, [
              _hoisted_9,
              createElementVNode("a", {
                href: "javascript:void(0)",
                onClick: _cache[1] || (_cache[1] = (...args) => (_ctx.downLoadErrorFile && _ctx.downLoadErrorFile(...args)))
              }, toDisplayString(_ctx.errorFileName), 1 /* TEXT */)
            ]))
          : createCommentVNode("v-if", true)
      ]),
      _: 1 /* STABLE */
    }, 8 /* PROPS */, ["model-value", "onClose", "onOpen"])
  ]))
}

script$2.render = render$2;
script$2.__file = "package/ths/components/c-common-base-button/src/view/t-excel-button.vue";

var script$1 = defineComponent({
    name: 'TSearchButton',
    props: {
        // 搜索条件是否展开
        searchShowAll: {
            type: Boolean,
            required: true,
            default: false,
        },
        // 是否隐藏展开按钮
        isShowFoldBtn: {
            type: Boolean,
            default: true,
        },
        // 查询按钮名称
        searchBtnName: {
            type: String,
            default: '查询',
        },
    },
    emits: ['search', 'reset', 'toggle-fold-state'],
    setup(props, { emit }) {
        // 响应式数据
        const state = reactive({
            // 展开/收起按钮名称
            buttonName: '展开',
            // 展开/收起按钮icon标签
            buttonIcon: 'arrow-down',
        });
        /**
         * 查询
         */
        const handleSearch = () => {
            emit('search');
        };
        /**
         * 重置
         */
        const handleReset = () => {
            emit('reset');
        };
        /**
         * 切换 查询条件的 展开与折叠状态
         */
        const toggleFoldState = () => {
            if (!props.searchShowAll) {
                state.buttonName = '收起';
                state.buttonIcon = 'arrow-up';
            }
            else {
                state.buttonName = '展开';
                state.buttonIcon = 'arrow-down';
            }
            emit('toggle-fold-state', !props.searchShowAll);
        };
        return {
            ...toRefs(state),
            handleSearch,
            handleReset,
            toggleFoldState,
        };
    },
});

const _hoisted_1$1 = /*#__PURE__*/createTextVNode(" 重置 ");

function render$1(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_el_button = resolveComponent("el-button");
  const _component_el_icon = resolveComponent("el-icon");

  return (openBlock(), createElementBlock(Fragment, null, [
    createVNode(_component_el_button, {
      type: "primary",
      size: "small",
      onClick: _ctx.handleSearch
    }, {
      default: withCtx(() => [
        createTextVNode(toDisplayString(_ctx.searchBtnName), 1 /* TEXT */)
      ]),
      _: 1 /* STABLE */
    }, 8 /* PROPS */, ["onClick"]),
    createVNode(_component_el_button, {
      size: "small",
      plain: "",
      onClick: _ctx.handleReset
    }, {
      default: withCtx(() => [
        _hoisted_1$1
      ]),
      _: 1 /* STABLE */
    }, 8 /* PROPS */, ["onClick"]),
    (_ctx.isShowFoldBtn)
      ? (openBlock(), createBlock(_component_el_button, {
          key: 0,
          type: "text",
          size: "small",
          onClick: _ctx.toggleFoldState
        }, {
          default: withCtx(() => [
            createVNode(_component_el_icon, { size: 14 }, {
              default: withCtx(() => [
                (openBlock(), createBlock(resolveDynamicComponent(_ctx.buttonIcon)))
              ]),
              _: 1 /* STABLE */
            }),
            createTextVNode(" " + toDisplayString(_ctx.buttonName), 1 /* TEXT */)
          ]),
          _: 1 /* STABLE */
        }, 8 /* PROPS */, ["onClick"]))
      : createCommentVNode("v-if", true)
  ], 64 /* STABLE_FRAGMENT */))
}

script$1.render = render$1;
script$1.__file = "package/ths/components/c-common-base-button/src/view/t-search-button.vue";

var script = defineComponent({
    name: 'TTextButton',
    props: {
        // 按钮文字
        text: {
            type: String,
            required: true,
            default: '按钮',
        },
        // 是否显示右侧竖线（最后一个按钮不需要）
        showSplit: {
            type: Boolean,
            required: false,
            default: true,
        },
        // 是否禁用
        isDisabled: {
            type: Boolean,
            default: false,
        },
        // 删除按钮
        isDelete: {
            type: Boolean,
            default: false,
        },
    },
    emits: ['click'],
    setup(props, { emit }) {
        // 响应式数据
        const state = reactive({});
        /**
         * 处理按钮点击事件
         */
        const handleButtonClick = () => {
            emit('click');
        };
        return {
            ...toRefs(state),
            handleButtonClick,
        };
    },
});

const _hoisted_1 = {
  key: 0,
  class: "text-group-line"
};

function render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_el_button = resolveComponent("el-button");

  return (openBlock(), createElementBlock(Fragment, null, [
    createCommentVNode(" 文字按钮 "),
    createVNode(_component_el_button, {
      class: normalizeClass(["text-group-button", _ctx.isDelete ? 'delete-text' : '']),
      size: "mini",
      type: "text",
      disabled: _ctx.isDisabled,
      onClick: _cache[0] || (_cache[0] = $event => (_ctx.handleButtonClick()))
    }, {
      default: withCtx(() => [
        createElementVNode("span", null, toDisplayString(_ctx.text), 1 /* TEXT */)
      ]),
      _: 1 /* STABLE */
    }, 8 /* PROPS */, ["disabled", "class"]),
    createCommentVNode(" 右侧竖线 "),
    (_ctx.showSplit)
      ? (openBlock(), createElementBlock("span", _hoisted_1))
      : createCommentVNode("v-if", true)
  ], 64 /* STABLE_FRAGMENT */))
}

script.render = render;
script.__file = "package/ths/components/c-common-base-button/src/view/t-text-button.vue";

// 存储组件列表
const components = [
    script$3,
    script$2,
    script$1,
    script,
];
const install = (app) => {
    components.forEach((component) => {
        app.component(component.name, component);
    });
};
var index = {
    install,
};

export { script$3 as TButton, script$2 as TExcelButton, script$1 as TSearchButton, script as TTextButton, index as default, install };
