// @ths/design/es/components/chart-map-scatter/src/index.vue2.js
import { defineComponent as P, ref as E, computed as _, reactive as j, watch as x, nextTick as w, onMounted as Y, onUnmounted as X } from "vue";
import * as L from "echarts";
import y from "lodash/merge";
import G from "lodash/cloneDeep";
import J from "axios";
import { getDomRandomClass as Q, setStyle as c } from "../../../utils/index.js";
const V = {
  longitudeKey: "longitude",
  latitudeKey: "latitude",
  areaNameKey: "regionName",
  numKey: "num",
  unitKey: "unit"
}, re = P({
  name: "TChartMapScatter",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据； 如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => ({
        longitudeKey: "longitude",
        latitudeKey: "latitude",
        areaNameKey: "regionName",
        numKey: "num",
        unitKey: "unit"
      })
    },
    // 地图背景图
    mapBackgroundUrl: {
      type: String,
      default: ""
    },
    // 点位图标SVG路径
    pointIconSvgPath: {
      type: String,
      default: ""
    },
    // 点位颜色
    pointIconColor: {
      type: String,
      default: "#54FFFF"
    },
    // 点位高亮颜色
    pointIconColorHover: {
      type: String,
      default: "#FFE400"
    },
    // 点位图标宽度
    pointIconWidth: {
      type: Number,
      default: 13
    },
    // 点位图标高度
    pointIconHeight: {
      type: Number,
      default: 16
    },
    // 高亮点位图标宽度
    pointIconWidthHover: {
      type: Number,
      default: 20
    },
    // 高亮点位图标高度
    pointIconHeightHover: {
      type: Number,
      default: 26
    },
    // 开启自动滚动
    autoScroll: {
      type: Boolean,
      default: !0
    },
    // 滚动速度 数字越大越快
    speed: {
      type: Number,
      default: () => 3
    },
    // 左侧地图是否显示
    isLeftMapShow: {
      type: Boolean,
      default: !0
    },
    // 右侧卡片是否显示
    isRightCardShow: {
      type: Boolean,
      default: !0
    },
    // 地图选中区域
    currentRegionName: {
      type: String,
      default: ""
    },
    // x偏移量
    xOffset: {
      type: Number,
      default: 48
    },
    // y偏移量
    yOffset: {
      type: Number,
      default: 40
    }
  },
  emits: [
    "change"
    // 卡片滚动事件
  ],
  setup(t, { emit: O }) {
    const o = E(null), s = E(null);
    let g = null;
    const a = _(() => t.cStyle), n = Q("chart-map-scatter"), d = j({
      // 点位数据
      pointData: [],
      // 当前选中区域
      currentAreaName: "",
      // 地图背景图
      mapBackground: ""
    });
    let h = null;
    const U = () => {
      let e = "";
      return a.value.wrapper && (e += c(a.value.wrapper, `.${n}`)), a.value.leftMapContainer && (e += c(a.value.leftMapContainer, `.${n} .container .left-map-container`)), a.value.rightCardContainer && (e += c(a.value.rightCardContainer, `.${n} .container .right-card-container`)), a.value.itemWrapper && (e += c(a.value.itemWrapper, `.${n} .container .right-card-container .card-item`)), a.value.itemText && (e += c(a.value.itemText, `.${n} .container .right-card-container .card-item .card-item-left`)), a.value.itemNum && (e += c(a.value.itemNum, `.${n} .container .right-card-container .card-item .card-item-right .num`)), a.value.itemUnit && (e += c(a.value.itemUnit, `.${n} .container .right-card-container .card-item .card-item-right .unit`)), a.value.itemWrapperSecond && (e += c(a.value.itemWrapperSecond, `.${n} .container .right-card-container .card-item:nth-child(3),
        .${n} .container .right-card-container .card-item:nth-child(5)`)), a.value.itemTextSecond && (e += c(a.value.itemTextSecond, `.${n} .container .right-card-container .card-item:nth-child(3) .card-item-left.name,
        .${n} .container .right-card-container .card-item:nth-child(5) .card-item-left.name`)), a.value.itemNumSecond && (e += c(a.value.itemNumSecond, `.${n} .container .right-card-container .card-item:nth-child(3) .card-item-right .num,
        .${n} .container .right-card-container .card-item:nth-child(5) .card-item-right .num`)), a.value.itemUnitSecond && (e += c(a.value.itemUnitSecond, `.${n} .container .right-card-container .card-item:nth-child(3) .card-item-right .unit,
        .${n} .container .right-card-container .card-item:nth-child(5) .card-item-right .unit`)), a.value.itemWrapperOuter && (e += c(a.value.itemWrapperOuter, `.${n} .container .right-card-container .card-item:nth-child(2),
        .${n} .container .right-card-container .card-item:nth-child(6)`)), a.value.itemTextOuter && (e += c(a.value.itemTextOuter, `.${n} .container .right-card-container .card-item:nth-child(2) .card-item-left.name,
        .${n} .container .right-card-container .card-item:nth-child(6) .card-item-left.name`)), a.value.itemNumOuter && (e += c(a.value.itemNumOuter, `.${n} .container .right-card-container .card-item:nth-child(2) .card-item-right .num,
        .${n} .container .right-card-container .card-item:nth-child(6) .card-item-right .num`)), a.value.itemUnitOuter && (e += c(a.value.itemUnitOuter, `.${n} .container .right-card-container .card-item:nth-child(2) .card-item-right .unit,
        .${n} .container .right-card-container .card-item:nth-child(6) .card-item-right .unit`)), e;
    }, A = () => {
      const e = document.querySelector(`.${n} .style-container`);
      e && (e.innerHTML = `<style>${U()}</style>`);
    }, T = (e, r, i) => {
      const C = t.xOffset, k = 410, M = 39.44, K = t.yOffset, D = 260, B = (k - C) / (117.5 - 115.41), W = (D - K) / (M - 41.05), z = C - 115.41 * B, R = K - 41.05 * W, q = e * B + z, F = r * W + R;
      return [q, F, 10, i];
    }, p = () => {
      if (t.data.length && t.dataAlias) {
        const e = y(G(V), t.dataAlias);
        d.pointData = t.data.map((r) => {
          const { longitudeKey: i, latitudeKey: u, areaNameKey: l } = e;
          return d.currentAreaName = t.currentRegionName !== "" ? t.currentRegionName : t.data[3][l], T(
            r[i],
            r[u],
            r[l]
          );
        });
      }
    }, H = () => new Promise((e, r) => {
      J.get(t.mapBackgroundUrl).then((i) => {
        i.data && i.data.length ? e(i.data) : r(new Error("地图背景图获取失败"));
      });
    }), f = () => {
      var l;
      const e = (l = t.options) == null ? void 0 : l.series;
      let r = [];
      L.registerMap("iceland_svg", { png: d.mapBackground });
      const i = {
        graphic: {
          type: "image",
          left: "0",
          // 横向位置
          top: "0",
          // 纵向位置
          z: -10,
          // 层级，-10 表示在所有图形元素的底部
          bounding: "all",
          // 定义图形的包围盒，'all' 表示整个图表区域
          style: {
            image: t.mapBackgroundUrl,
            // 背景图片的 URL
            width: s.value.offsetWidth,
            // 图片宽度
            height: s.value.offsetHeight * 0.67,
            // 图片高度
            opacity: 1
            // 透明度
          }
        },
        geo: {
          tooltip: {
            show: !1,
            formatter(m) {
              return m.data[3];
            }
          },
          map: "iceland_svg"
        },
        series: {
          type: "scatter",
          coordinateSystem: "geo",
          geoIndex: 0,
          hoverAnimation: !1,
          symbol() {
            return `path://${t.pointIconSvgPath}`;
          },
          symbolSize(m) {
            return m[3] === d.currentAreaName ? [t.pointIconWidthHover, t.pointIconHeightHover] : [t.pointIconWidth, t.pointIconHeight];
          },
          data: d.pointData,
          itemStyle: {
            color(m) {
              return m.data[3] === d.currentAreaName ? `${t.pointIconColorHover}` : `${t.pointIconColor}`;
            }
          }
        }
      };
      e && (r = e.map((m, C) => y(
        {
          type: "scatter",
          coordinateSystem: "geo",
          geoIndex: 0
        },
        m
      )));
      let u = y(
        {
          series: r
        },
        i
      );
      u = y(i, t.options);
      try {
        g.setOption(u, !0);
      } catch (m) {
        console.error(m);
      }
    }, v = () => {
      w(() => {
        g.resize();
      });
    }, S = () => {
      var r, i, u, l;
      const e = ((r = o.value) == null ? void 0 : r.querySelectorAll(".card-item")) ?? [];
      d.currentAreaName = (u = (i = e[4]) == null ? void 0 : i.querySelector(".name")) == null ? void 0 : u.innerText, O("change", d.currentAreaName), e.length > 0 && ((l = o.value) == null || l.appendChild(e[0])), f();
    }, b = () => {
      var r, i, u, l;
      const e = ((r = o.value) == null ? void 0 : r.querySelectorAll(".card-item")) ?? [];
      d.currentAreaName = (u = (i = e[2]) == null ? void 0 : i.querySelector(".name")) == null ? void 0 : u.innerText, O("change", d.currentAreaName), e.length > 0 && ((l = o.value) == null || l.prepend(e[e.length - 1])), f();
    }, N = (e) => {
      e.deltaY > 0 ? S() : b();
    }, $ = () => {
      clearInterval(h), h = setInterval(() => {
        S();
      }, 1e3 * t.speed);
    }, I = () => {
      w(() => {
        const e = o.value;
        $(), e.addEventListener("mouseenter", () => {
          clearInterval(h);
        }), e.addEventListener("mouseleave", () => {
          clearInterval(h), $();
        }), e.addEventListener("wheel", N);
      });
    };
    return x(() => t.cStyle, () => {
      A(), w(() => {
        g.resize();
      });
    }, { deep: !0 }), Y(() => {
      A(), s.value && (g = L.init(s.value), p(), H().then((e) => {
        d.mapBackground = e, f();
      }).catch((e) => {
        console.error(e);
      })), v(), t.autoScroll && o.value && I(), window.addEventListener("resize", v);
    }), x(() => [t.autoScroll, t.data, t.speed], () => {
      var e;
      clearImmediate(h), t.autoScroll && ((e = t.data) != null && e.length) && o.value ? I() : clearInterval(h);
    }, { immediate: !0 }), x(() => [
      t.data,
      t.mapBackgroundUrl,
      t.pointIconSvgPath,
      t.pointIconColorHover,
      t.pointIconColor,
      t.currentRegionName,
      t.isLeftMapShow
    ], () => {
      p(), f();
    }), X(() => {
      window.removeEventListener("resize", v), o.value && (o.value.removeEventListener("wheel", N), o.value.removeEventListener("mouseenter", () => {
        clearInterval(h);
      }), o.value.removeEventListener("mouseleave", () => {
        clearInterval(h), $();
      }));
    }), {
      cardRef: o,
      mapBase: s,
      curStyle: a,
      domOnlyClass: n,
      drawChart: f,
      transformCoordinates: T,
      dataProcessing: p,
      resize: v,
      handleWheel: N,
      movePrev: b,
      moveNext: S,
      cardScroll: I,
      getMapBackground: H
    };
  }
});
export {
  re as default
};
