// @ths/design/es/components/chart-dashboard-aqi-base/src/index.vue2.js
import { defineComponent as x, computed as g, ref as N, onMounted as O, onUnmounted as L, watch as C, nextTick as w } from "vue";
import * as A from "echarts";
import n from "lodash/merge";
import r from "lodash/cloneDeep";
import { getAQILevel as t } from "../../../utils/index.js";
import { DEFAULT_OPTIONS_COMMON as z, AQI_COLOR as d } from "../../../utils/constant.js";
const b = n(
  r(z),
  {
    tooltip: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    }
  }
), S = "#6E7683", D = "level7", B = x({
  name: "TChartDashboardAqiBase",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 仪表盘数据
    data: {
      type: String,
      default: () => ""
    },
    // 仪表盘文字数据
    name: {
      type: String,
      default: () => ""
    },
    // 仪表盘配置
    seriesStyle: {
      type: Object,
      default: () => {
      }
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => b,
      required: !1
    },
    // 仪表盘主题设置
    gauge: {
      type: Object,
      default: () => ({})
    },
    // 等级颜色
    levelColors: {
      type: Object,
      default: () => r(d)
    },
    // 填充使用计算颜色
    lineColorComputed: {
      type: Boolean,
      default: () => !1
    },
    // 填充色
    lineColor: {
      type: String,
      default: () => S
    },
    // 刻度值是否需要保留小数
    isFixed: {
      type: Boolean,
      default: !1
    },
    // 刻度值保留小数位数
    fixedNumber: {
      type: Number,
      default: 0
    }
  },
  setup(e) {
    const v = g(() => e.cStyle), u = g(() => n(r(d), e.levelColors)), c = N(null);
    let a = null;
    const s = () => {
      var o, f, l;
      return {
        type: "gauge",
        radius: "100%",
        startAngle: 235,
        endAngle: -55,
        splitNumber: 10,
        min: 0,
        max: 300,
        progress: {
          show: !0,
          roundCap: !0,
          width: 25,
          itemStyle: {
            // 获取值颜色 如果传的等级颜色为空 则显示默认的值
            color: ((o = u.value) == null ? void 0 : o[t(e.data)]) || d[t(e.data)]
          }
        },
        axisLine: {
          roundCap: !0,
          lineStyle: {
            width: 25,
            color: [
              // 判断填充色使用等级颜色或自定义颜色 如果传的等级颜色为空 则显示默认的值
              [1, e.lineColorComputed ? ((f = u.value) == null ? void 0 : f[t(e.data)]) || d[t(e.data)] : e.lineColor || S]
            ]
          }
        },
        pointer: {
          show: !1,
          itemStyle: {
            color: "#005fff"
          },
          icon: "",
          offsetCenter: [0, 0],
          length: "80%",
          width: 8
        },
        axisTick: {
          show: !1,
          distance: -25,
          length: 8,
          lineStyle: {
            color: "#fff",
            width: 2
          }
        },
        splitLine: {
          show: !1,
          distance: -25,
          length: 25,
          lineStyle: {
            color: "#fff",
            width: 2
          }
        },
        axisLabel: {
          show: !1,
          color: "#000",
          distance: 40,
          fontSize: 20,
          fontWeight: "normal",
          // 刻度值保留小数位
          formatter: (i) => e.isFixed ? i.toFixed(Number(e.fixedNumber) <= 0 ? 0 : Number(e.fixedNumber)) : i
        },
        anchor: {
          show: !1,
          showAbove: !0,
          size: 6,
          icon: "circle",
          itemStyle: {
            color: "#fff"
          }
        },
        detail: {
          valueAnimation: !0,
          formatter: "{value}",
          // 数值颜色为等级色 如果传的等级颜色为空 则显示默认的值
          color: ((l = u.value) == null ? void 0 : l[t(e.data)]) || d[t(e.data)],
          fontSize: 40,
          fontWeight: "bold",
          offsetCenter: [0, -30]
        },
        title: {
          fontSize: 40,
          color: "#000",
          fontWeight: "normal",
          offsetCenter: [0, 47]
        }
      };
    }, h = () => {
      var o;
      if (c.value) {
        a || (a = A.init(c.value));
        const f = (o = e.options) == null ? void 0 : o.series;
        let l = {};
        f ? l = f.map((m) => n(
          r(s()),
          e == null ? void 0 : e.gauge,
          m
        )) : e.data !== "" && (l = n(
          r(s()),
          {
            data: [
              {
                name: e.name || "-",
                value: Number.isNaN(Number(e.data)) ? "0" : Number(e.data)
              }
            ]
          },
          e == null ? void 0 : e.gauge,
          e == null ? void 0 : e.seriesStyle
        ));
        let i = n(
          {
            series: l
          },
          b
        );
        i = n(i, e.options), a.clear();
        try {
          t(e.data) !== D && a.setOption(i);
        } catch (m) {
          console.log(m);
        }
      }
    }, y = () => {
      w(() => {
        a.resize();
      });
    };
    return O(() => {
      h(), window.addEventListener("resize", y);
    }), L(() => {
      window.removeEventListener("resize", y);
    }), C(
      () => [
        e.gauge,
        e.options,
        e.seriesStyle,
        e.data,
        e.name,
        e.lineColorComputed,
        e.lineColor,
        e.isFixed,
        e.fixedNumber
      ],
      () => {
        h();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), C(
      () => [e.cStyle],
      () => {
        w(() => {
          a.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      chart: c,
      curStyle: v
    };
  }
});
export {
  B as default
};
