// @ths/design/es/components/chart-bar-stacked/src/index.vue2.js
import { defineComponent as q, computed as M, ref as B, reactive as U, watch as x, nextTick as w, onMounted as H, onUnmounted as R } from "vue";
import * as G from "echarts";
import f from "lodash/merge";
import v from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as K, DEFAULT_OPTIONS_AXIS as X } from "../../../utils/constant.js";
import { getDomRandomClass as J, setStyle as F } from "../../../utils/index.js";
const Q = f(v(K), v(X)), l = f(
  v(Q),
  {
    // 气泡框
    tooltip: {
      axisPointer: {
        type: "line"
      }
    },
    // 网格线
    grid: {
      top: "13%",
      left: "1%",
      right: "2%",
      bottom: "2%",
      containLabel: !0
    },
    // x轴
    xAxis: {
      boundaryGap: !0
    },
    // y轴
    yAxis: {
      name: "(单位)",
      splitNumber: 3,
      splitLine: {
        show: !0
      }
    }
  }
), at = q({
  name: "TChartBarStacked",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 传入数组数据 如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => []
    },
    // 设置主轴
    mainAxis: {
      type: String,
      default: "x"
    },
    // 包含 系列样式 、别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // x 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => l,
      required: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        }
      })
    },
    // 柱图主题设置
    bar: {
      type: Object,
      default: () => ({})
    },
    // 折线主题配置项
    line: {
      type: Object,
      default: () => ({})
    }
  },
  // 方法名
  emits: ["com-click"],
  setup(t, { emit: N }) {
    const u = J("chart-bar-compare"), c = M(() => t.cStyle), A = B(null);
    let d = null;
    const p = U({
      // 图表数据
      chartData: []
    }), O = () => {
      var e, y, s, i;
      (y = (e = t.seriesSetting) == null ? void 0 : e.data) != null && y.length && ((s = t.data) != null && s.length) && (p.chartData = (i = t.seriesSetting) == null ? void 0 : i.data.map((n) => {
        const { name: r, dataKey: m } = n;
        return {
          name: r || "--",
          data: t.data.map((S) => S[m])
        };
      }));
    }, P = () => {
      let e = "";
      return c.value.wrapper && (e += F(c.value.wrapper, `.${u}`)), c.value.emptyContainer && (e += F(c.value.emptyContainer, `.${u} .empty`)), c.value.emptyImg && (e += F(c.value.emptyImg, `.${u} .empty .empty-img`)), c.value.emptyText && (e += F(c.value.emptyText, `.${u} .empty .empty-text`)), e;
    }, D = () => {
      const e = document.querySelector(`.${u} .style-container`);
      e && (e.innerHTML = `<style>${P()}</style>`);
    }, W = M(() => {
      var e, y, s, i;
      if (t.options) {
        if (Array.isArray((e = t.options) == null ? void 0 : e.series))
          return t.options.series.some((r) => {
            var m;
            return (r == null ? void 0 : r.data) && ((m = r == null ? void 0 : r.data) == null ? void 0 : m.length) !== 0;
          });
        if (typeof ((y = t.options) == null ? void 0 : y.series) == "object")
          return ((i = (s = t.options.series) == null ? void 0 : s.data) == null ? void 0 : i.length) === 0;
      }
      return !!t.data.length;
    }), C = () => {
      var r, m, S, L, T, k, j, E;
      d.clear();
      const e = (r = t.options) == null ? void 0 : r.color, y = (m = t.options) == null ? void 0 : m.series;
      let s = [], i = [];
      y ? s = y.map((o, a) => {
        var b, g;
        return f(
          {
            type: "bar",
            stack: "总量",
            barWidth: 12,
            itemStyle: {
              color: {
                type: "linear",
                x: 0,
                y: 1,
                x2: 0,
                y2: 0,
                colorStops: [
                  {
                    offset: 1,
                    color: e && e[a] || l.color[a] || "#FFFFFF00"
                    // 100% 处的颜色
                  },
                  {
                    offset: 0,
                    color: e && e[a] || l.color[a] || "#FFFFFF00"
                    // 0% 处的颜色
                  }
                ]
              },
              borderColor: ((b = o == null ? void 0 : o.itemStyle) == null ? void 0 : b.borderColor) || e && e[a] || l.color[a] || "#fff",
              // 边框颜色
              borderWidth: 2,
              shadowColor: ((g = o == null ? void 0 : o.itemStyle) == null ? void 0 : g.shadowColor) || e && e[a] || l.color[a] || "#fff",
              // 图形阴影颜色
              shadowBlur: 5
              // 图形阴影
            },
            ...t[o.type]
          },
          o
        );
      }) : (S = t.data) != null && S.length && ((T = (L = t.seriesSetting) == null ? void 0 : L.data) != null && T.length) && (s = (k = t.seriesSetting) == null ? void 0 : k.data.map((o, a) => {
        var g, I, z, _, $;
        const b = f({
          name: (g = p.chartData[a]) == null ? void 0 : g.name,
          type: o.type || "bar",
          stack: o.type === "line" ? "" : "总量",
          symbol: "circle",
          barWidth: 12,
          smooth: o.smooth && !0,
          showSymbol: !0,
          // 在 hover 时，显示 icon
          itemStyle: {
            color: {
              type: "linear",
              x: 0,
              y: 1,
              x2: 0,
              y2: 0,
              colorStops: [
                {
                  offset: 1,
                  color: e && e[a] || l.color[a] || "#FFFFFF00"
                  // 100% 处的颜色
                },
                {
                  offset: 0,
                  color: e && e[a] || l.color[a] || "#FFFFFF00"
                  // 0% 处的颜色
                }
              ]
            },
            borderColor: e && e[a] || l.color[a] || "#fff",
            // 边框颜色
            borderWidth: 2,
            borderRadius: 0,
            shadowColor: e && e[a] || l.color[a] || "#fff",
            // 图形阴影颜色
            shadowBlur: 5
            // 图形阴影
          },
          data: ((I = p.chartData[a]) == null ? void 0 : I.data) || []
        }, t[o.type]);
        return f({}, b, ($ = (_ = (z = t.seriesSetting) == null ? void 0 : z.data) == null ? void 0 : _[a]) == null ? void 0 : $.style);
      }), (j = t.seriesSetting) != null && j.xAxisLabel ? i = (E = t.data) != null && E.length ? t.data.map((o) => o[t.seriesSetting.xAxisLabel]) : [] : i = t.data.map((o, a) => a));
      let n;
      t.mainAxis === "x" ? n = f(
        {
          series: s,
          xAxis: {
            data: i
          }
        },
        l
      ) : t.mainAxis === "y" && (n = f(
        {
          series: s,
          yAxis: {
            data: i
          }
        },
        l
      )), n = f(n, t.options), t.mainAxis === "y" && (n.xAxis.type = "value", n.yAxis.type = "category");
      try {
        w(() => {
          d.setOption(n), d.resize();
        });
      } catch (o) {
        console.log(o);
      }
    }, h = () => {
      w(() => {
        d.resize();
      });
    };
    return x(
      () => [
        t.options,
        t.bar,
        t.line,
        t.mainAxis
      ],
      () => {
        C(), h();
      },
      {
        deep: !0
      }
    ), x(
      () => [
        t.data,
        t.seriesSetting
      ],
      () => {
        O(), C(), h();
      },
      {
        deep: !0
      }
    ), x(
      () => [t.cStyle],
      () => {
        D(), w(() => {
          d.resize();
        });
      },
      {
        deep: !0
      }
    ), H(() => {
      D(), A.value && (d = G.init(A.value), O(), C(), setTimeout(() => {
        h();
      }, 500), d.on("click", (e) => {
        N("com-click", e);
      })), window.addEventListener("resize", h);
    }), R(() => {
      window.removeEventListener("resize", h);
    }), {
      chartId: A,
      curStyle: c,
      checkEmpty: W,
      domOnlyClass: u
    };
  }
});
export {
  at as default
};
