// @ths/design/es/components/chart-radar/src/index.vue2.js
import { defineComponent as L, computed as b, ref as q, reactive as z, nextTick as T, onMounted as N, onUnmounted as R, watch as w, toRefs as V } from "vue";
import * as _ from "echarts";
import E from "lodash/cloneDeep";
import S from "lodash/merge";
import { DEFAULT_OPTIONS as F } from "../../../utils/constant.js";
import { getDomRandomClass as H, setStyle as v } from "../../../utils/index.js";
const M = {
  wrapper: {
    default: {}
  }
}, P = S(
  E(F),
  {
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    },
    tooltip: {
      trigger: "item"
    },
    // 雷达图配置
    radar: {
      center: ["50%", "55%"]
    }
  }
), A = {
  width: 1,
  opacity: 0.5
}, j = {
  opacity: 0.2
}, J = L({
  name: "TChartRadar",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 指示器名称取resData数据中的别名设置
        indicatorName: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => M
    },
    // 雷达图主题配置
    radar: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["com-click"],
  setup(e, { emit: $ }) {
    const d = H("chart-radar"), i = b(() => e.cStyle), h = q(null);
    let m = null;
    const u = z({
      // 图表数据
      chartData: [],
      maxValue: 0
    });
    function C() {
      var t, r, s;
      if ((t = e.data) != null && t.length && ((s = (r = e.seriesSetting) == null ? void 0 : r.data) != null && s.length)) {
        let n = [];
        u.chartData = e.seriesSetting.data.map((l) => {
          const { name: a, dataKey: o } = l;
          return n = n.concat(e.data.map((y) => y[o])), {
            name: a || "--",
            value: e.data.map((y) => y[o])
          };
        }), n.length && (u.maxValue = n.sort((l, a) => l - a)[n.length - 1]);
      }
    }
    const k = () => {
      let t = "";
      return i.value.wrapper && (t += v(i.value.wrapper, `.${d}`)), i.value.emptyContainer && (t += v(i.value.emptyContainer, `.${d} .empty`)), i.value.emptyImg && (t += v(i.value.emptyImg, `.${d} .empty .empty-img`)), i.value.emptyText && (t += v(i.value.emptyText, `.${d} .empty .empty-text`)), t;
    }, p = () => {
      const t = document.querySelector(`.${d} .style-container`);
      t && (t.innerHTML = `<style>${k()}</style>`);
    }, c = () => {
      h.value && (h.value.getBoundingClientRect().width ?? 0) && T(() => {
        m.resize();
      });
    }, I = b(() => {
      var t, r, s, n;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series)) {
          const l = e.options.series.some((a) => {
            var o;
            return (a == null ? void 0 : a.data) && ((o = a == null ? void 0 : a.data) == null ? void 0 : o.length) !== 0;
          });
          return console.log(l, "isEmptyData"), l;
        }
        if (typeof ((r = e.options) == null ? void 0 : r.series) == "object")
          return ((n = (s = e.options.series) == null ? void 0 : s.data) == null ? void 0 : n.length) === 0;
      }
      return e.data.length ? (T(() => {
        c();
      }), !0) : !1;
    });
    function x() {
      var l, a, o, y, D, O;
      if (!m)
        return;
      const { series: t } = e.options;
      let r = [], s = [];
      t ? r = t.map((f) => S(
        {
          type: "radar",
          areaStyle: j,
          symbol: "none",
          lineStyle: A,
          emphasis: {
            focus: "series"
          }
        },
        f
      )) : (l = e.data) != null && l.length && ((o = (a = e.seriesSetting) == null ? void 0 : a.data) != null && o.length) && (r = (y = e.seriesSetting) == null ? void 0 : y.data.map((f, g) => S({
        type: "radar",
        data: [u.chartData[g]],
        areaStyle: j,
        symbol: "none",
        lineStyle: A,
        emphasis: {
          focus: "series"
        }
      }, f.style)), (D = e.seriesSetting) != null && D.indicatorName ? s = (O = e.data) != null && O.length ? e.data.map((f) => {
        var g;
        return {
          name: f[(g = e.seriesSetting) == null ? void 0 : g.indicatorName],
          max: u.maxValue
        };
      }) : [] : s = e.data.map((f, g) => ({
        name: g + 1,
        max: u.maxValue
      })));
      const n = S(
        E(P),
        {
          series: r,
          radar: S({
            indicator: s
          }, e == null ? void 0 : e.radar)
        },
        e.options
      );
      m.clear(), m.setOption(n);
    }
    return N(() => {
      p(), h.value && (m = _.init(h.value), C(), x(), m.on("click", (t) => {
        $("com-click", t);
      })), setTimeout(() => {
        c();
      }, 1500), window.addEventListener("resize", c);
    }), R(() => {
      window.removeEventListener("resize", c);
    }), w(
      [
        () => e.options,
        () => e.radar
      ],
      (t) => {
        x(), c();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), w(
      [
        () => e.data,
        () => e.seriesSetting
      ],
      () => {
        C(), x(), c();
      },
      {
        deep: !0
      }
    ), w(
      () => e.cStyle,
      () => {
        p(), c();
      },
      {
        deep: !0
      }
    ), {
      ...V(u),
      curStyle: i,
      chartId: h,
      checkEmpty: I,
      domOnlyClass: d
    };
  }
});
export {
  J as default
};
