// @ths/design/es/components/chart-heatmap-base/src/index.vue.js
import l from "./index.vue2.js";
import { openBlock as i, createElementBlock as m, normalizeClass as n, normalizeStyle as r, withDirectives as o, createElementVNode as e, vShow as a } from "vue";
import "./index.vue3.js";
import p from "../../../_virtual/_plugin-vue_export-helper.js";
const c = {
  ref: "chartId",
  class: "chart"
}, d = { class: "empty" }, u = ["src"];
function y(t, s, f, h, v, E) {
  return i(), m("div", {
    class: n(["t-chart-heatmap-base", t.domOnlyClass]),
    style: r({ ...t.curStyle.wrapper.default })
  }, [
    o(e("div", c, null, 512), [
      [a, t.checkEmpty]
    ]),
    o(e("div", d, [
      e("img", {
        class: "empty-img",
        style: r({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, u),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [a, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const g = /* @__PURE__ */ p(l, [["render", y]]);
export {
  g as default
};
