// @ths/design/es/components/chart-heatmap-base/src/index.vue2.js
import { defineComponent as R, computed as I, ref as q, reactive as G, onMounted as B, onUnmounted as F, watch as x, nextTick as M, toRefs as H } from "vue";
import * as N from "echarts";
import S from "lodash/cloneDeep";
import s from "lodash/merge";
import { DEFAULT_OPTIONS_COMMON as z, DEFAULT_OPTIONS_AXIS as P } from "../../../utils/constant.js";
import { getDomRandomClass as U, setStyle as v } from "../../../utils/index.js";
const V = {
  wrapper: {
    default: {}
  }
}, X = s(S(z), S(P)), j = s(S(X), {
  tooltip: {
    position: "top",
    trigger: "item"
  },
  grid: {
    top: "5%",
    right: "2%",
    bottom: "15%",
    left: "11%"
  },
  legend: {
    show: !1
  },
  xAxis: {
    type: "category",
    boundaryGap: !0,
    // data: ['北京', '上海', '广东', '成都'],
    // 隐藏坐标线
    axisLine: {
      show: !1
    },
    // 隐藏刻度
    axisTick: {
      show: !1
    }
  },
  yAxis: {
    type: "category",
    boundaryGap: !0,
    // 隐藏坐标线
    axisLine: {
      show: !1
    },
    // 隐藏刻度
    axisTick: {
      show: !1
    }
  },
  visualMap: {
    type: "piecewise",
    orient: "horizontal",
    left: "center",
    top: "auto",
    itemGap: 5,
    textGap: 5,
    inRange: {
      color: z.color
    }
  }
}), Z = R({
  name: "TChartHeatmapBase",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => V
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    heatmap: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: D }) {
    const m = U("chart-heatmap-base"), n = I(() => e.cStyle), b = q(null);
    let r = null;
    const w = G({
      // 图表数据
      chartData: [],
      maxValue: 0
    });
    function T() {
      var t, i;
      (i = (t = e.seriesSetting) == null ? void 0 : t.data) != null && i.length && e.data && (w.chartData = e.seriesSetting.data.map((o) => {
        const {
          name: l,
          xLabel: d,
          yLabel: a,
          value: c
        } = o;
        return {
          name: l || "--",
          data: e.data.map((u) => [u[d], u[a], u[c]])
        };
      }));
    }
    const _ = () => {
      let t = "";
      return n.value.wrapper && (t += v(n.value.wrapper, `.${m}`)), n.value.emptyContainer && (t += v(n.value.emptyContainer, `.${m} .empty`)), n.value.emptyImg && (t += v(n.value.emptyImg, `.${m} .empty .empty-img`)), n.value.emptyText && (t += v(n.value.emptyText, `.${m} .empty .empty-text`)), t;
    }, A = () => {
      const t = document.querySelector(`.${m} .style-container`);
      t && (t.innerHTML = `<style>${_()}</style>`);
    }, $ = I(() => {
      var t, i, o, l;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((a) => {
            var c;
            return (a == null ? void 0 : a.data) && ((c = a == null ? void 0 : a.data) == null ? void 0 : c.length) !== 0;
          });
        if (typeof ((i = e.options) == null ? void 0 : i.series) == "object")
          return ((l = (o = e.options.series) == null ? void 0 : o.data) == null ? void 0 : l.length) === 0;
      }
      return !!e.data.length;
    });
    function C() {
      var d, a, c, u;
      const { color: t, series: i } = e.options;
      let o = [];
      i ? o = i.map((y, f) => {
        var h, g, p;
        const O = s(
          {
            type: "heatmap",
            label: {
              // 是否显示标签，也就是热力图方块上的文字
              show: !0
            },
            // 高亮状态的图形样式
            emphasis: {
              itemStyle: {
                shadowBlur: 10,
                shadowColor: "rgba(0, 0, 0, 0.5)"
              }
            },
            ...e == null ? void 0 : e.heatmap
          },
          (p = (g = (h = e.seriesSetting) == null ? void 0 : h.data) == null ? void 0 : g[f]) == null ? void 0 : p.style
        );
        return s({}, O, y);
      }) : e.data && ((a = (d = e.seriesSetting) == null ? void 0 : d.data) != null && a.length) && (o = (c = e.seriesSetting) == null ? void 0 : c.data.map((y, f) => {
        var h, g, p, k, E;
        const O = s(
          {
            name: (h = w.chartData[f]) == null ? void 0 : h.name,
            type: "heatmap",
            label: {
              show: !0
            },
            emphasis: {
              itemStyle: {
                shadowBlur: 10,
                shadowColor: "rgba(0, 0, 0, 0.5)"
              }
            },
            data: e.data.length ? (g = w.chartData[f]) == null ? void 0 : g.data : [],
            ...e == null ? void 0 : e.heatmap
          },
          (E = (k = (p = e.seriesSetting) == null ? void 0 : p.data) == null ? void 0 : k[f]) == null ? void 0 : E.style
        );
        return s({}, O, y);
      }));
      let l = s(
        S(j),
        {
          series: o,
          visualMap: {
            inRange: {
              color: ((u = e.options) == null ? void 0 : u.color) || j.visualMap.inRange.color
            }
          }
        }
      );
      l = s(S(l), e.options), r.clear();
      try {
        r.setOption(l);
      } catch (y) {
        console.log(y);
      }
    }
    function L() {
      M(() => {
        r.resize();
      });
    }
    return B(() => {
      A(), b.value && (r = N.init(b.value), T(), C(), r.on("click", (t) => {
        D("com-click", t);
      }), r.on("datarangeselected", (t) => {
        D("legend-select", t);
      })), window.addEventListener("resize", L);
    }), F(() => {
      window.removeEventListener("resize", L);
    }), x(
      [
        () => e.options,
        () => e.heatmap
      ],
      (t) => {
        C();
      },
      {
        deep: !0
      }
    ), x(
      [
        () => e.data,
        () => e.seriesSetting
      ],
      () => {
        T(), C();
      },
      {
        deep: !0
      }
    ), x(
      () => e.cStyle,
      () => {
        A(), M(() => {
          r.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      ...H(w),
      curStyle: n,
      chartId: b,
      checkEmpty: $,
      domOnlyClass: m
    };
  }
});
export {
  Z as default
};
