// @ths/design/es/components/chart-rose/src/index.vue2.js
import { defineComponent as I, computed as E, ref as q, reactive as z, onMounted as M, onUnmounted as _, watch as p, toRefs as R, nextTick as F } from "vue";
import * as H from "echarts";
import o from "lodash/merge";
import N from "lodash/cloneDeep";
import { getDomRandomClass as P, setStyle as w } from "../../../utils/index.js";
import { DEFAULT_OPTIONS_COMMON as U } from "../../../utils/constant.js";
const K = o(N(U), {
  tooltip: {
    trigger: "item"
  },
  xAxis: {
    show: !1
  },
  yAxis: {
    show: !1
  },
  // 极坐标系角度轴
  angleAxis: {
    type: "category",
    // data: ['内蒙古', '浙江', '辽宁', '吉林', '安徽', '福建'],
    axisTick: {
      show: !1
      // 是否显示坐标轴刻度
    },
    splitLine: {
      show: !0,
      lineStyle: {}
    },
    axisLabel: {
      show: !0
    },
    axisLine: {
      show: !0,
      lineStyle: {}
    }
  },
  // 极坐标系径向轴
  radiusAxis: {
    axisLabel: {
      show: !0
    },
    axisTick: {
      show: !1
      // 是否显示坐标轴刻度
    },
    axisLine: {
      show: !0
      // 是否显示坐标轴轴线
    },
    splitLine: {
      show: !0
    }
  },
  // 极坐标系
  polar: {
    center: ["50%", "56%"]
  }
}), X = I({
  name: "TChartRose",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 玫瑰图柱条宽度
    barWidth: {
      type: String,
      default: "28%"
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 角度轴类目取resData数据中的别名设置
        angleAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: L }) {
    const c = P("chart-pie-rose"), i = E(() => e.cStyle), x = q(null);
    let d = null;
    const b = z({
      chartData: []
    }), $ = () => {
      let t = "";
      return i.value.wrapper && (t += w(i.value.wrapper, `.${c}`)), i.value.emptyContainer && (t += w(i.value.emptyContainer, `.${c} .empty`)), i.value.emptyImg && (t += w(i.value.emptyImg, `.${c} .empty .empty-img`)), i.value.emptyText && (t += w(i.value.emptyText, `.${c} .empty .empty-text`)), t;
    }, C = () => {
      const t = document.querySelector(`.${c} .style-container`);
      t && (t.innerHTML = `<style>${$()}</style>`);
    }, j = E(() => {
      var t, n, s, r;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((a) => {
            var f;
            return (a == null ? void 0 : a.data) && ((f = a == null ? void 0 : a.data) == null ? void 0 : f.length) !== 0;
          });
        if (typeof ((n = e.options) == null ? void 0 : n.series) == "object")
          return ((r = (s = e.options.series) == null ? void 0 : s.data) == null ? void 0 : r.length) === 0;
      }
      return !!e.data.length;
    });
    function D() {
      var t, n;
      (n = (t = e.seriesSetting) == null ? void 0 : t.data) != null && n.length && e.data && (b.chartData = e.seriesSetting.data.map((s) => {
        const { name: r, dataKey: y } = s;
        return {
          name: r || "--",
          data: e.data.map((a) => a[y])
        };
      }));
    }
    function v() {
      var y, a, f, O, T;
      const { series: t } = e.options;
      let n = [], s = [];
      t ? n = t.map((m, l) => {
        var g, h, S;
        const A = o(
          {
            type: "bar",
            stack: "a",
            coordinateSystem: "polar",
            barWidth: e.barWidth
          },
          (S = (h = (g = e.seriesSetting) == null ? void 0 : g.data) == null ? void 0 : h[l]) == null ? void 0 : S.style
        );
        return o({}, A, m);
      }) : e.data && ((a = (y = e.seriesSetting) == null ? void 0 : y.data) != null && a.length) && (n = (f = e.seriesSetting) == null ? void 0 : f.data.map((m, l) => {
        var g, h, S, k, W;
        const A = o(
          {
            type: "bar",
            // 数据堆叠
            stack: "a",
            // 使用极坐标系 cartesian2d->二维直角坐标系
            coordinateSystem: "polar",
            // 柱条的宽度
            barWidth: e.barWidth,
            name: (g = b.chartData[l]) == null ? void 0 : g.name,
            data: e.data.length ? (h = b.chartData[l]) == null ? void 0 : h.data : []
          },
          (W = (k = (S = e.seriesSetting) == null ? void 0 : S.data) == null ? void 0 : k[l]) == null ? void 0 : W.style
        );
        return o({}, A, m);
      }), (O = e.seriesSetting) != null && O.angleAxisLabel ? s = (T = e.data) != null && T.length ? e.data.map((m) => m[e.seriesSetting.angleAxisLabel]) : [] : s = e.data.map((m, l) => l));
      let r = o(
        {
          series: n,
          angleAxis: {
            data: s
          }
        },
        K
      );
      r = o(r, e.options), d.clear(), d.setOption(r);
    }
    function u() {
      F(() => {
        d.resize();
      });
    }
    return M(() => {
      C(), x.value && (d = H.init(x.value), D(), v(), d.on("click", (t) => {
        L("com-click", t);
      }), d.on("legendselectchanged", (t) => {
        L("legend-select", t);
      })), setTimeout(() => {
        u();
      }, 500), window.addEventListener("resize", u);
    }), _(() => {
      window.removeEventListener("resize", u);
    }), p(
      [
        () => e.options,
        () => e.barWidth
      ],
      () => {
        v(), u();
      },
      {
        deep: !0
      }
    ), p(
      [
        () => e.data,
        () => e.seriesSetting,
        () => e.barWidth
      ],
      () => {
        D(), v(), u();
      },
      {
        deep: !0
      }
    ), p(() => e.cStyle, () => {
      C(), u();
    }, { deep: !0 }), {
      ...R(b),
      chartId: x,
      curStyle: i,
      checkEmpty: j,
      domOnlyClass: c
    };
  }
});
export {
  X as default
};
