// @ths/design/es/components/chart-rose/src/index.vue.js
import a from "./index.vue2.js";
import { openBlock as n, createElementBlock as m, normalizeClass as c, normalizeStyle as r, withDirectives as l, createElementVNode as e, vShow as i } from "vue";
import "./index.vue3.js";
import p from "../../../_virtual/_plugin-vue_export-helper.js";
const d = {
  ref: "chartId",
  class: "chart"
}, u = { class: "empty" }, y = ["src"];
function f(t, s, h, v, E, k) {
  var o;
  return n(), m("div", {
    class: c(["t-chart-rose", t.domOnlyClass]),
    style: r({ ...(o = t.cStyle) == null ? void 0 : o.wrapper.default })
  }, [
    l(e("div", d, null, 512), [
      [i, t.checkEmpty]
    ]),
    l(e("div", u, [
      e("img", {
        class: "empty-img",
        style: r({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, y),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [i, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const C = /* @__PURE__ */ p(a, [["render", f]]);
export {
  C as default
};
