// @ths/design/es/components/header-menu-scroll/src/index.vue.js
import k from "./index.vue2.js";
import { resolveComponent as g, openBlock as l, createElementBlock as t, normalizeClass as d, createElementVNode as s, createCommentVNode as a, Fragment as y, renderList as f, normalizeStyle as u, toDisplayString as I, createVNode as v, Transition as M, withCtx as p } from "vue";
import "./index.vue3.js";
import b from "../../../_virtual/_plugin-vue_export-helper.js";
const $ = {
  ref: "main",
  class: "main"
}, h = {
  ref: "menu",
  class: "main-menu"
}, C = ["onClick", "onMouseenter"], F = { class: "main-menu--item__text" };
function S(e, o, B, N, w, L) {
  const m = g("TNavigationItem");
  return l(), t("div", {
    class: d([e.domOnlyClass, "t-header-menu-scroll"])
  }, [
    s("div", $, [
      e.showScroll ? (l(), t("div", {
        key: 0,
        class: "main-btn--pre",
        onClick: o[0] || (o[0] = (...n) => e.moveLeft && e.moveLeft(...n)),
        onMouseenter: o[1] || (o[1] = (n) => e.hoverButtonIndex = 0),
        onMouseleave: o[2] || (o[2] = (n) => e.hoverButtonIndex = -1)
      }, null, 32)) : a("", !0),
      s("div", h, [
        (l(!0), t(y, null, f(e.menusData, (n, r) => (l(), t("div", {
          key: r,
          class: d([{ active: e.currentActiveIndex[0] === r }, "main-menu--item"]),
          style: u({
            "background-image": e.currentActiveIndex[0] === r ? `url(${n[e.keyField.backgroundLightImg]}` : `url(${n[e.keyField.backgroundDefaultImg]})`
          }),
          onClick: (i) => e.handleMenuClick(n, r),
          onMouseenter: (i) => e.handleMenuMouseenter(r),
          onMouseleave: o[3] || (o[3] = (...i) => e.handleMenuMouseleave && e.handleMenuMouseleave(...i))
        }, [
          e.showIcon && n[e.keyField.startIcon] ? (l(), t("div", {
            key: 0,
            style: u({
              "background-image": "url(" + n[e.keyField.startIcon] + ")"
            }),
            class: "main-menu--item__leftIcon",
            draggable: "false"
          }, null, 4)) : a("", !0),
          s("span", F, I(n[e.keyField.label]), 1),
          n[e.keyField.children] && n[e.keyField.endIcon] ? (l(), t("div", {
            key: 1,
            class: "main-menu--item__rightIcon",
            style: u({
              "background-image": "url(" + n[e.keyField.endIcon] + ")",
              transform: e.hoverItemIndex === r ? "rotate(180deg)" : "rotate(0deg)"
            }),
            draggable: "false"
          }, null, 4)) : a("", !0)
        ], 46, C))), 128))
      ], 512),
      e.showScroll ? (l(), t("div", {
        key: 1,
        class: "main-btn--next",
        onClick: o[4] || (o[4] = (...n) => e.moveRight && e.moveRight(...n)),
        onMouseenter: o[5] || (o[5] = (n) => e.hoverButtonIndex = 0),
        onMouseleave: o[6] || (o[6] = (n) => e.hoverButtonIndex = -1)
      }, null, 32)) : a("", !0)
    ], 512),
    v(M, { "name:fade": "" }, {
      default: p(() => {
        var n;
        return [
          v(m, {
            "current-active-index": e.currentActiveIndexTemp.slice(1),
            "current-hover-index": e.currentHoverIndex,
            data: ((n = e.currentSubData) == null ? void 0 : n.children) || [],
            "key-field": e.keyField,
            style: u({
              left: `${e.getNavigationItemLeft}px`,
              opacity: e.showSub || e.mouseoverSub ? 1 : 0
            }),
            class: "navigation",
            onMouseleave: e.handleSubMenuMouseleave,
            onMouseover: o[7] || (o[7] = (r) => e.mouseoverSub = !0)
          }, null, 8, ["current-active-index", "current-hover-index", "data", "key-field", "style", "onMouseleave"])
        ];
      }),
      _: 1
    }),
    o[8] || (o[8] = s("div", { class: "style-container" }, null, -1))
  ], 2);
}
const z = /* @__PURE__ */ b(k, [["render", S]]);
export {
  z as default
};
