// @ths/design/es/components/header-menu-scroll/src/index.vue2.js
import { defineComponent as xe, ref as m, computed as ue, reactive as ye, watch as w, provide as Te, onUpdated as $e, onMounted as Ce, toRefs as be } from "vue";
import Se from "mitt";
import pe from "./navigationItem.vue.js";
import { getDomRandomClass as _e, setStyle as o, getCssPropertyString as O } from "../../../utils/index.js";
const Me = {
  /* 容器 */
  wrapper: {
    default: {
      width: "1000px"
    }
  }
}, ke = xe({
  name: "THeaderMenuScroll",
  components: { TNavigationItem: pe },
  props: {
    // 数据
    data: {
      type: Object,
      default: () => ({})
    },
    // 当前选中项路径
    activePath: {
      type: Array,
      default: () => []
    },
    // 是否显示图标
    showIcon: {
      type: Boolean,
      default: !0
    },
    // 字段关键词
    keyField: {
      type: Object,
      default: () => ({
        label: "label",
        startIcon: "startIcon",
        endIcon: "endIcon",
        path: "path",
        backgroundDefaultImg: "backgroundDefaultImg",
        backgroundLightImg: "backgroundLightImg",
        children: "children"
      })
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => Me
    }
  },
  emits: ["menu-click"],
  setup(c, F) {
    const a = _e("header-menu-scroll"), v = m(null), A = m(null), E = m(!1), T = m(!1), y = m(!1), d = m(-1), ve = [], $ = Se(), B = m([]), C = m([]), t = ue(() => c.cStyle), W = m(), h = ye({
      menusData: ve,
      hoverButtonIndex: -1,
      // hover的索引
      currentLeftIndex: -1,
      // 当前最左元素索引
      currentRightIndex: -1
      // 当前最右元素索引
    });
    Array.isArray(c.data) ? h.menusData = [...c.data] : console.log("传入的数据不是数组");
    const s = m([]), re = (e) => {
      s.value = [...e.path], C.value = [...e.path], F.emit("menu-click", C.value, e.item);
    }, q = (e, n) => {
      let l = h.menusData;
      s.value = [];
      for (let r = 0; r < n; r++)
        l == null || l.some((g, f) => {
          if (g[c.keyField.label] === e[r])
            return s.value.push(f), !0;
        }), l && l[s.value[r]].children && (l = l[s.value[r]].children);
    };
    Array.isArray(c.activePath) ? q(c.activePath, c.activePath.length) : console.log("类型错误");
    const ce = ue(() => {
      var e, n, l;
      if (v.value && A.value)
        return ((n = (e = v.value.children[d.value]) == null ? void 0 : e.getBoundingClientRect()) == null ? void 0 : n.left) - ((l = A.value.getBoundingClientRect()) == null ? void 0 : l.left);
    }), D = (e) => {
      var n;
      e === -1 && !y.value ? T.value = !1 : (n = h.menusData[e]) != null && n.children || y.value ? T.value = !0 : T.value = !1;
    }, me = (e) => {
      if (s.value[0] === e ? C.value = s.value : C.value = [], B.value = [e], d.value = e, D(d.value), W.value = h.menusData[e], !T.value) {
        const n = document.querySelector(".navigation");
        n == null || n.classList.add("stop"), setTimeout(() => {
          n == null || n.classList.remove("stop");
        }, 1);
      }
    }, se = () => {
      y.value || setTimeout(() => {
        y.value || (d.value = -1), D(d.value);
      }, 1);
    }, fe = () => {
      y.value = !1, d.value = -1, B.value = [], D(d.value);
    }, _ = () => {
      const e = document.querySelector(".main");
      v.value && e && (E.value = v.value.scrollWidth > e.scrollWidth);
    }, k = () => {
      var e, n, l;
      if (v.value) {
        const r = v.value.children, { length: g } = r;
        let f = -1, I = -1, b = -1, S = -1;
        const u = (e = v.value) == null ? void 0 : e.getBoundingClientRect(), x = (i) => {
          var p;
          return i >= 0 && i < g ? (p = r[i]) == null ? void 0 : p.getBoundingClientRect() : { right: 0, left: 0 };
        }, M = (i) => x(i).left - (u == null ? void 0 : u.left), L = (i) => x(i).right - (u == null ? void 0 : u.right);
        for (let i = 0; i < g; i += 1)
          f === -1 && M(i) >= 0 && (f = i), I === -1 && L(i) > 1 && (I = i - 1);
        return b = ((n = x(f - 1)) == null ? void 0 : n.left) - (u == null ? void 0 : u.left), S = ((l = x(I + 1)) == null ? void 0 : l.right) - (u == null ? void 0 : u.right), h.currentLeftIndex = f, h.currentRightIndex = I, {
          scrollLastWidth: b,
          scrollNextWidth: S
        };
      }
    }, de = () => {
      if (v.value) {
        const { scrollLastWidth: e } = k();
        v.value.scrollBy({
          left: e,
          behavior: "smooth"
        });
      }
    }, he = () => {
      if (v.value) {
        const { scrollNextWidth: e } = k();
        v.value.scrollBy({
          left: e,
          behavior: "smooth"
        });
      }
    }, ge = (e, n) => {
      e.children || (s.value = [n], F.emit("menu-click", e, n));
    }, Ie = () => {
      var n, l, r, g, f, I, b, S, u, x, M, L, i, p, P, H, N, j, U, z, Y, G, J, K, Q, X, Z, V, ee, te, ne, ae, oe, ie, le;
      let e = "";
      return t.value.wrapper && (e += o(t.value.wrapper, `.${a}`)), t.value.main && (e += o(t.value.main, `.${a} .main`)), t.value.leftBtn && (e += o(t.value.leftBtn, `.${a} .main-btn--pre`)), (n = t.value) != null && n.rightBtn && (e += o(t.value.rightBtn, `.${a} .main-btn--next`)), t.value.menu && (e += o(t.value.menu, `.${a} .main-menu`)), (l = t.value) != null && l.menuItem && (e += o(t.value.menuItem, `.${a} .main-menu--item`)), (g = (r = t.value) == null ? void 0 : r.menuItem) != null && g.active && (e += `
          .${a} .main-menu--item.active {
             ${O((I = (f = t.value) == null ? void 0 : f.menuItem) == null ? void 0 : I.active)}
           }
        `), (S = (b = t.value) == null ? void 0 : b.menuItem) != null && S.hover && (e += o((x = (u = t.value) == null ? void 0 : u.menuItem) == null ? void 0 : x.hover, `.${a} .main-menu--item:hover`)), (M = t.value) != null && M.menuItemText && (e += o(t.value.menuItemText, `.${a} .main-menu--item__text`)), (i = (L = t.value) == null ? void 0 : L.menuItemText) != null && i.active && (e += `
          .${a} .main-menu--item__text.active {
             ${O((P = (p = t.value) == null ? void 0 : p.menuItemText) == null ? void 0 : P.active)}
           }
        `), (N = (H = t.value) == null ? void 0 : H.menuItemText) != null && N.hover && (e += o((U = (j = t.value) == null ? void 0 : j.menuItemText) == null ? void 0 : U.hover, `.${a} .main-menu--item__text:hover`)), t.value.leftIcon && (e += o(t.value.leftIcon, `.${a} .main-menu--item__leftIcon`)), t.value.rightIcon && (e += o(t.value.rightIcon, `.${a} .main-menu--item__rightIcon`)), t.value.navigation && (e += o(t.value.navigation, `.${a} .navigation`)), t.value.navigationContent && (e += o(t.value.navigationContent, `.${a} .navigation-content`)), (Y = (z = t.value) == null ? void 0 : z.navigationContent) != null && Y.active && (e += `
          .${a} .navigation-content.active {
             ${O((J = (G = t.value) == null ? void 0 : G.navigationContent) == null ? void 0 : J.active)}
           }
        `), t.value.navigationContentText && (e += o(t.value.navigationContentText, `.${a} .navigation-content--text`)), (Q = (K = t.value) == null ? void 0 : K.navigationContentText) != null && Q.hover && (e += o((Z = (X = t.value) == null ? void 0 : X.navigationContentText) == null ? void 0 : Z.hover, `.${a} .navigation-content--text:hover`)), t.value.navigationContentTextFont && (e += o(t.value.navigationContentTextFont, `.${a} .navigation-content--text__font`)), (ee = (V = t.value) == null ? void 0 : V.navigationContentTextFont) != null && ee.hover && (e += o((ne = (te = t.value) == null ? void 0 : te.navigationContentTextFont) == null ? void 0 : ne.hover, `.${a} .navigation-content--text__font:hover`)), t.value.navigationContentTextIcon && (e += o(t.value.navigationContentTextIcon, `.${a} .navigation-content--text__font`)), (oe = (ae = t.value) == null ? void 0 : ae.navigationContentTextIcon) != null && oe.hover && (e += o((le = (ie = t.value) == null ? void 0 : ie.navigationContentTextIcon) == null ? void 0 : le.hover, `.${a} .navigation-content--text__font:hover`)), e;
    }, R = () => {
      const e = document.querySelector(`.${a} .style-container`);
      e && (e.innerHTML = `<style>${Ie()}</style>`);
    };
    return w(
      () => c.data,
      (e) => {
        Array.isArray(e) ? h.menusData = [...e] : console.log("类型错误");
        const n = setTimeout(() => {
          _(), clearTimeout(n);
        }, 10);
      },
      {
        deep: !0
      }
    ), w(
      () => c.activePath,
      (e) => {
        Array.isArray(e) ? q(e, e.length) : console.log("类型错误");
      },
      {
        deep: !0
      }
    ), w(
      () => [c.cStyle],
      () => {
        R();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), Te("rootMenu", {
      rootMenuEmit: $.emit,
      rootMenuOn: $.on
    }), $e(() => {
      document.querySelector(".main-btn--pre"), document.querySelector(".main-btn--next"), R(), _();
    }), Ce(() => {
      R(), _(), window.onresize = () => {
        _();
      }, $.on("submenuItem:item-click", (e) => {
        re(e);
      });
    }), {
      ...be(h),
      showScroll: E,
      showSub: T,
      domOnlyClass: a,
      moveLeft: de,
      moveRight: he,
      handleMenuClick: ge,
      menu: v,
      currentActiveIndex: s,
      getFirstAndLastIndex: k,
      getNavigationItemLeft: ce,
      hoverItemIndex: d,
      handleMenuMouseenter: me,
      handleMenuMouseleave: se,
      mouseoverSub: y,
      handleSubMenuMouseleave: fe,
      rootMenuEmit: $.emit,
      rootMenuOn: $.on,
      currentHoverIndex: B,
      currentActiveIndexTemp: C,
      currentSubData: W,
      main: A
    };
  }
});
export {
  ke as default
};
