// @ths/design/es/components/header-menu-scroll/src/navigationItem.vue2.js
import { defineComponent as I, inject as g, ref as n, onUnmounted as b } from "vue";
const k = I({
  name: "TNavigationItem",
  props: {
    // 定义 props
    data: {
      type: Array,
      default: () => []
    },
    currentActiveIndex: {
      type: Array,
      default: () => []
    },
    currentHoverIndex: {
      type: Array,
      default: () => []
    },
    subMenuLeft: {
      type: String,
      default: "left : 100%"
    },
    keyField: {
      type: Object,
      default: () => ({
        label: "label",
        startIcon: "startIcon",
        endIcon: "endIcon",
        path: "path",
        backgroundDefaultImg: "backgroundDefaultImg",
        backgroundLightImg: "backgroundLightImg",
        children: "children"
      })
    }
  },
  setup(v, y) {
    const d = g("rootMenu"), t = n(!1), u = n(-1), a = n(!1), l = n([]), o = n(void 0), r = n("left : 100%"), c = (e) => {
      var s;
      e === -1 && !a.value ? t.value = !1 : (s = v.data[e]) != null && s.children || a.value ? t.value = !0 : t.value = !1;
    }, f = (e, s) => {
      if (!e.children) {
        const p = l.value;
        d == null || d.rootMenuEmit("submenuItem:item-click", {
          item: e,
          path: p
        });
      }
      s.stopPropagation();
    }, i = (e) => {
      l.value = v.currentHoverIndex.concat([e]), u.value = e, c(e), a.value = !0, t.value && o.value && (window.innerWidth - o.value.getBoundingClientRect().right < o.value.clientWidth ? r.value = "left: -101%" : r.value = "left : 100%");
    }, m = () => {
      l.value.pop(), a.value || (u.value = -1, c(u.value)), t.value = !1;
    }, h = () => {
      a.value = !1, c(u.value);
    };
    return b(() => {
      r.value = "left : 100%";
    }), {
      showSub: t,
      showSubByIndex: c,
      handleMenuMouseenter: i,
      handleMenuMouseleave: m,
      handleSubMenuMouseleave: h,
      mouseoverSub: a,
      handleSubMenuClick: f,
      currentHoverIndexTemp: l,
      hoverItemIndex: u,
      dom: o,
      subLeft: r
    };
  }
});
export {
  k as default
};
