// @ths/design/es/components/header-menu-scroll/src/navigationItem.vue.js
import d from "./navigationItem.vue2.js";
import { resolveComponent as v, openBlock as a, createElementBlock as s, normalizeStyle as l, Fragment as m, renderList as c, normalizeClass as p, createElementVNode as u, toDisplayString as M, createVNode as f } from "vue";
import "./navigationItem.vue3.js";
import g from "../../../_virtual/_plugin-vue_export-helper.js";
const h = ["onMouseenter", "onMouseover", "onClick"], b = { class: "navigation-content--text" }, I = { class: "navigation-content--text__font" };
function y(e, r, k, N, S, $) {
  const i = v("TNavigationItem");
  return a(), s("div", {
    ref: "dom",
    class: "navigation",
    style: l(e.subMenuLeft)
  }, [
    (a(!0), s(m, null, c(e.data, (n, o) => (a(), s("div", {
      key: o,
      class: p(["navigation-content", { active: e.currentActiveIndex[0] === Number(o) }]),
      onMouseenter: (t) => e.handleMenuMouseenter(Number(o)),
      onMouseleave: r[1] || (r[1] = (...t) => e.handleMenuMouseleave && e.handleMenuMouseleave(...t)),
      onMouseover: (t) => e.hoverItemIndex = Number(o),
      onClick: (t) => e.handleSubMenuClick(n, t)
    }, [
      u("div", b, [
        u("span", I, M(n[e.keyField.label]), 1),
        u("div", {
          class: "navigation-content--text__icon",
          style: l({
            "background-image": "url(" + n[e.keyField.endIcon] + ")",
            transform: e.showSub && e.hoverItemIndex === o ? "rotate(90deg)" : "rotate(270deg)"
          })
        }, null, 4)
      ]),
      f(i, {
        data: n[e.keyField.children] ? n[e.keyField.children] : [],
        "sub-menu-left": e.subLeft,
        "current-active-index": e.currentActiveIndex.slice(1),
        "current-hover-index": e.currentHoverIndexTemp,
        class: "navigation",
        style: l([{
          opacity: e.showSub && e.hoverItemIndex === o && n[e.keyField.children] ? 1 : 0
        }, { top: "0" }]),
        onMouseover: r[0] || (r[0] = (t) => e.mouseoverSub = !0),
        onMouseleave: e.handleSubMenuMouseleave
      }, null, 8, ["data", "sub-menu-left", "current-active-index", "current-hover-index", "style", "onMouseleave"])
    ], 42, h))), 128))
  ], 4);
}
const w = /* @__PURE__ */ g(d, [["render", y], ["__scopeId", "data-v-2993ce38"]]);
export {
  w as default
};
