// @ths/design/es/components/upload-button/src/index.vue2.js
import { defineComponent as D, computed as A, ref as F, watch as I, onMounted as P, nextTick as G } from "vue";
import { ElMessage as p } from "element-plus";
import { getDomRandomClass as J, setStyle as K, getCssPropertyString as u } from "../../../utils/index.js";
const oe = D({
  name: "TUploadButton",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        buttonWrapper: {
          default: {}
        },
        text: {
          default: {}
        }
      })
    },
    // 按钮文字
    label: {
      type: String,
      default: "上传"
    },
    // 上传地址
    uploadUrl: {
      type: String,
      default: ""
    },
    // 文件类型
    fileType: {
      type: String,
      default: "zip"
    },
    // 上传参数
    params: {
      type: Object,
      default: () => ({})
    },
    // 是否支持多选
    isMultiple: {
      type: Boolean,
      default: !1
    },
    // 是否开启文件数量限制
    isLimit: {
      type: Boolean,
      default: !1
    },
    // 上传文件数量限制
    fileLimit: {
      type: Number,
      default: 100
    }
  },
  emits: ["error", "success"],
  setup(a, { emit: f }) {
    const t = A(() => a.cStyle), n = J("upload-button"), N = A(() => a.label), y = F(), R = () => {
      let e = "";
      return t.value.wrapper && (e += K(t.value.wrapper, `.${n}.t-upload-button`)), e;
    }, _ = () => {
      var l, o, r, c, d, s, i, m, v, h, $, S, g, x, W, w, C, T, L, E, O, U, B, M, j, k, H, q;
      let e = "";
      return (o = (l = t.value) == null ? void 0 : l.buttonWrapper) != null && o.default && (e += `.${n}.t-upload-button .el-button{
          ${u((c = (r = t.value) == null ? void 0 : r.buttonWrapper) == null ? void 0 : c.default)}
          width: 100%;
          height: 100%;
        }`), (s = (d = t.value) == null ? void 0 : d.buttonWrapper) != null && s.hover && (e += `.${n}.t-upload-button .el-button:hover{
          ${u((m = (i = t.value) == null ? void 0 : i.buttonWrapper) == null ? void 0 : m.hover)}
        }`), (h = (v = t.value) == null ? void 0 : v.buttonWrapper) != null && h.focus && (e += `.${n}.t-upload-button .el-button:focus{
          ${u((S = ($ = t.value) == null ? void 0 : $.buttonWrapper) == null ? void 0 : S.focus)}
        }`), (x = (g = t.value) == null ? void 0 : g.buttonWrapper) != null && x.active && (e += `.${n}.t-upload-button .el-button:active{
          ${u((w = (W = t.value) == null ? void 0 : W.buttonWrapper) == null ? void 0 : w.active)}
        }`), (T = (C = t.value) == null ? void 0 : C.text) != null && T.default && (e += `.${n}.t-upload-button .el-button span{
          ${u((E = (L = t.value) == null ? void 0 : L.text) == null ? void 0 : E.default)}
        }`), (U = (O = t.value) == null ? void 0 : O.text) != null && U.hover && (e += `.${n}.t-upload-button.el-button:hover span{
          ${u((M = (B = t.value) == null ? void 0 : B.text) == null ? void 0 : M.hover)}
        }`), (k = (j = t.value) == null ? void 0 : j.text) != null && k.focus && (e += `.${n}.t-upload-button.el-button:focus span{
          ${u((q = (H = t.value) == null ? void 0 : H.text) == null ? void 0 : q.focus)}
        }`), e;
    }, b = () => {
      const e = document.querySelector(`.${n} .style-container`);
      e && (e.innerHTML = `<style>${R()}${_()}</style>`);
    };
    I(
      () => [a.cStyle],
      () => {
        b();
      },
      { deep: !0 }
    );
    const z = () => {
      const e = document.querySelector(`.${n}.t-upload-button`);
      if (e) {
        const l = e.firstChild;
        l && (l.style.width = "100%", l.style.height = "100%");
      }
    };
    return P(() => {
      b(), G(() => {
        z();
      });
    }), {
      curStyle: t,
      domOnlyClass: n,
      filterAttrs: (e, l = []) => {
        const o = {};
        return Object.keys(e).forEach((r) => {
          ["data-id", "cStyle", "class", "style", "onClick", "id"].concat(l).reduce((s, i) => s || r === i, !1) || (o[r] = e[r]);
        }), o;
      },
      uploadRef: y,
      curLabel: N,
      handleUpload: () => {
      },
      handleBeforeUpload: (e) => {
        const l = e.name.substring(e.name.lastIndexOf(".") + 1);
        return a.fileType && !a.fileType.includes(l) ? (p.warning(`只允许导入${a.fileType}类型文件！！！`), !1) : !0;
      },
      handleError: (e, l, o) => {
        p.error("上传失败"), console.error(e), f("error", e, l, o);
      },
      handleSuccess: (e, l, o) => {
        y.value.clearFiles(), f("success", e, l, o);
      },
      handleExceed: (e, l) => {
        a.isLimit && p.warning(`最多只能上传${a.fileLimit}个文件!`);
      }
    };
  }
});
export {
  oe as default
};
