// @ths/design/es/components/upload-button/src/index.vue.js
import s from "./index.vue2.js";
import { resolveComponent as l, openBlock as p, createElementBlock as i, normalizeClass as d, createVNode as t, mergeProps as u, withCtx as r, createTextVNode as m, toDisplayString as f, createElementVNode as c } from "vue";
import "./index.vue3.js";
import b from "../../../_virtual/_plugin-vue_export-helper.js";
function y(e, o, h, E, $, B) {
  const a = l("el-button"), n = l("el-upload");
  return p(), i("div", {
    class: d(["t-upload-button", e.domOnlyClass])
  }, [
    t(n, u(e.filterAttrs(e.$attrs), {
      ref: "uploadRef",
      multiple: e.isMultiple,
      "show-file-list": !1,
      data: e.params,
      action: e.uploadUrl,
      accept: e.fileType,
      limit: e.fileLimit,
      "before-upload": e.handleBeforeUpload,
      "on-error": e.handleError,
      "on-success": e.handleSuccess,
      "on-exceed": e.handleExceed
    }), {
      default: r(() => [
        t(a, { type: "primary" }, {
          default: r(() => [
            m(f(e.label || "上传"), 1)
          ]),
          _: 1
        })
      ]),
      _: 1
    }, 16, ["multiple", "data", "action", "accept", "limit", "before-upload", "on-error", "on-success", "on-exceed"]),
    o[0] || (o[0] = c("div", { class: "style-container" }, null, -1))
  ], 2);
}
const V = /* @__PURE__ */ b(s, [["render", y]]);
export {
  V as default
};
