// @ths/design/es/components/rect/src/index.vue2.js
import { defineComponent as p, computed as n, watch as s, onMounted as m } from "vue";
import { filterAttrs as f, getDomRandomClass as y, getCssPropertyString as l, setStyle as c } from "../../../utils/index.js";
import "../../text/index.js";
import x from "../../text/src/index.vue.js";
const h = p({
  name: "TRect",
  components: {
    TText: x
  },
  props: {
    // 文字
    label: {
      type: String,
      default: ""
    },
    // 选中状态
    active: {
      type: Boolean,
      default: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {},
          active: {},
          hover: {}
        },
        text: {
          default: {},
          hover: {},
          active: {}
        }
      })
    }
  },
  setup(r) {
    const i = n(() => r.active), e = n(() => r.cStyle), a = y("rect"), v = () => {
      let t = "";
      return e.value.wrapper && (e.value.wrapper.active && (t += `
            .${a}.active {
              ${l(e.value.wrapper.active)}
            }

            .${a}.active .t-text {
              ${e.value.text && e.value.text.active && l(e.value.text.active)}
            }
          `), e.value.wrapper.hover && (t += `
            .${a}:hover .t-text {
              ${e.value.text && e.value.text.hover && l(e.value.text.hover)}
            }
          `)), t;
    }, u = () => {
      let t = "";
      return e.value.wrapper && (t += c(e.value.wrapper, `.${a}`)), e.value.text && (t += c(e.value.text, `.${a} .t-text`)), t;
    }, o = () => {
      const t = document.querySelector(`.${a} .style-container`);
      t && (t.innerHTML = `<style>${u()}${v()}</style>`);
    };
    return s(() => r.cStyle, () => {
      o();
    }, {
      deep: !0
    }), m(() => {
      o();
    }), {
      isActive: i,
      domOnlyClass: a,
      filterAttrs: f,
      curStyle: e
    };
  }
});
export {
  h as default
};
