// @ths/design/es/components/weather-day/src/index.vue2.js
import { defineComponent as m, computed as c, onMounted as i, watch as u } from "vue";
import { getDomRandomClass as p, setStyle as s } from "../../../utils/index.js";
const h = m({
  name: "TWeatherDay",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 标题：今日天气
        title: {
          default: {}
        },
        // 天气状态：晴
        weather: {
          default: {}
        },
        // 图标
        imgUrl: {
          default: {}
        },
        // 气温25
        tmp: {
          default: {}
        },
        // 单位：℃
        unit: {
          default: {}
        }
      })
    },
    // 数据
    data: {
      type: Object,
      default: () => ({
        title: "",
        weather: "",
        tmp: "",
        imgUrl: "",
        unit: ""
      })
    }
  },
  setup(l, { emit: y }) {
    const a = c(() => l.cStyle), n = p("weather-day"), r = {
      wrapper: ".t-weather-day",
      title: ".weather-day-content__top-title",
      weather: ".weather-day-content__top-weather",
      imgUrl: ".weather-day-content__middle-imgUrl",
      tmp: ".weather-day-content__middle-tmp",
      unit: ".weather-day-content__footer-unit"
    }, d = () => {
      let t = "";
      return Object.keys(r).forEach((e) => {
        a.value[e] && (t += s(a.value[e], `.${n}${e === "wrapper" ? "" : " "}${r[e]}`));
      }), t;
    }, o = () => {
      const t = document.querySelector(`.${n} .style-container`);
      t && (t.innerHTML = `<style>${d()}</style>`);
    };
    return i(() => {
      o();
    }), u(
      () => l.cStyle,
      () => {
        o();
      },
      { deep: !0, immediate: !0 }
    ), {
      curStyle: a,
      domOnlyClass: n,
      curStyleMap: r
    };
  }
});
export {
  h as default
};
