// @ths/design/es/components/back-top/src/index.vue2.js
import { defineComponent as s, computed as a, watch as c, onMounted as p } from "vue";
import { filterAttrs as m, getDomRandomClass as u, setStyle as y } from "../../../utils/index.js";
const f = s({
  name: "TBackTop",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {},
          // 默认
          hover: {}
          // 滑过
        }
      })
    }
  },
  setup(n) {
    const t = a(() => n.cStyle), o = u("back-top"), l = () => {
      let e = "";
      return t.value.wrapper && (e += y(t.value.wrapper, `.${o} .el-backtop`)), e;
    }, r = () => {
      const e = document.querySelector(`.${o} .style-container`);
      e && (e.innerHTML = `<style>${l()}</style>`);
    };
    return c(
      () => [n.cStyle],
      () => {
        r();
      },
      {
        deep: !0
      }
    ), p(() => {
      r();
    }), {
      curStyle: t,
      domOnlyClass: o,
      filterAttrs: m
    };
  }
});
export {
  f as default
};
