// @ths/design/es/components/chart-bar-3d/src/index.vue2.js
import { defineComponent as K, computed as H, ref as X, reactive as J, onMounted as Q, onUnmounted as Y, watch as V, toRefs as Z, nextTick as ee } from "vue";
import * as d from "echarts";
import k from "lodash/merge";
import I from "lodash/cloneDeep";
import { getDomRandomClass as te, rgb2rgba as u, setStyle as L } from "../../../utils/index.js";
import { DEFAULT_OPTIONS_COMMON as oe, DEFAULT_OPTIONS_AXIS as ae } from "../../../utils/constant.js";
const ne = {
  wrapper: {
    default: {}
  },
  // 暂无数据容器
  emptyContainer: {
    default: {}
  },
  // 暂无数据图片
  emptyImg: {
    default: {}
  },
  // 暂无数据文字
  emptyText: {
    default: {}
  }
}, le = k(I(oe), I(ae)), T = k(
  I(le),
  {
    xAxis: { boundaryGap: !0 },
    tooltip: {
      formatter: (o) => {
        let p = "";
        return o.forEach((f) => {
          f.seriesType !== "custom" && (p += `${f.name}: ${f.value}`);
        }), p;
      }
    },
    legend: {
      show: !0,
      selectedMode: !1
      // 图例是否允许点击
    },
    grid: {
      top: "13%",
      bottom: "10%",
      left: "6%",
      right: "3%"
    }
  }
), ye = K({
  name: "TChartBar3d",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => T,
      required: !1
    },
    // 图表数据，传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ne
    }
  },
  emits: ["com-click", "legend-select"],
  setup(o, { emit: p }) {
    const f = te("chart-bar-3d"), m = H(() => o.cStyle), C = X(null);
    let A = null, b = null;
    const F = d.graphic.extendShape({
      shape: {
        x: 0,
        y: 0
      },
      buildPath(e, t) {
        const { xAxisPoint: a } = t, r = [t.x, t.y], c = [t.x - 13, t.y - 13], s = [a[0] - 13, a[1] - 13], y = [a[0], a[1]];
        e.moveTo(r[0], r[1]).lineTo(c[0], c[1]).lineTo(s[0], s[1]).lineTo(y[0], y[1]).closePath();
      }
    }), N = d.graphic.extendShape({
      shape: {
        x: 0,
        y: 0
      },
      buildPath(e, t) {
        const { xAxisPoint: a } = t, r = [t.x, t.y], c = [a[0], a[1]], s = [a[0] + 18, a[1] - 9], y = [t.x + 18, t.y - 9];
        e.moveTo(r[0], r[1]).lineTo(c[0], c[1]).lineTo(s[0], s[1]).lineTo(y[0], y[1]).closePath();
      }
    }), U = d.graphic.extendShape({
      shape: {
        x: 0,
        y: 0
      },
      buildPath(e, t) {
        const a = [t.x, t.y], r = [t.x + 18, t.y - 9], c = [t.x + 5, t.y - 22], s = [t.x - 13, t.y - 13];
        e.moveTo(a[0], a[1]).lineTo(r[0], r[1]).lineTo(c[0], c[1]).lineTo(s[0], s[1]).closePath();
      }
    });
    d.graphic.registerShape("CubeLeft", F), d.graphic.registerShape("CubeRight", N), d.graphic.registerShape("CubeTop", U);
    const w = J({
      // 图表数据
      chartData: []
    }), z = () => {
      var e, t;
      (t = (e = o.seriesSetting) == null ? void 0 : e.data) != null && t.length && o.data && (w.chartData = o.seriesSetting.data.map((a) => {
        const { name: r, dataKey: c } = a;
        return {
          name: r || "--",
          data: o.data.map((s) => s[c])
        };
      }));
    }, W = () => {
      let e = "";
      return m.value.wrapper && (e += L(m.value.wrapper, `.${f}`)), m.value.emptyContainer && (e += L(m.value.emptyContainer, `.${f} .empty`)), m.value.emptyImg && (e += L(m.value.emptyImg, `.${f} .empty .empty-img`)), m.value.emptyText && (e += L(m.value.emptyText, `.${f} .empty .empty-text`)), e;
    }, E = () => {
      const e = document.querySelector(`.${f} .style-container`);
      e && (e.innerHTML = `<style>${W()}</style>`);
    }, O = (e, t = "#43e97b") => {
      var a, r;
      return ((a = (o.options || T).color) == null ? void 0 : a[e]) || ((r = T.color) == null ? void 0 : r[e]) || t;
    }, D = () => {
      var s, y, R, _, j, G;
      b.clear();
      const { series: e, color: t } = o.options || T;
      let a = [], r = [];
      e ? (a = e.map((v, i) => k(
        {
          type: "custom",
          renderItem: (g, l) => {
            const S = l.coord([l.value(0), l.value(1)]), n = O(i), x = new d.graphic.LinearGradient(0, 0, 0, 1, [
              {
                offset: 0,
                color: u(n, 0)
              },
              {
                offset: 1,
                color: u(n, 1)
              }
            ]), h = new d.graphic.LinearGradient(0, 0, 0, 1, [
              {
                offset: 0,
                color: u(n, 1)
              },
              {
                offset: 1,
                color: u(n, 0)
              }
            ]);
            return {
              type: "group",
              children: [{
                type: "CubeLeft",
                shape: {
                  api: l,
                  xValue: l.value(0),
                  yValue: l.value(1),
                  x: S[0],
                  y: S[1],
                  xAxisPoint: l.coord([l.value(0), 0])
                },
                style: {
                  fill: x,
                  stroke: h
                }
              }, {
                type: "CubeRight",
                shape: {
                  api: l,
                  xValue: l.value(0),
                  yValue: l.value(1),
                  x: S[0],
                  y: S[1],
                  xAxisPoint: l.coord([l.value(0), 0])
                },
                style: {
                  fill: x,
                  stroke: h
                }
              }, {
                type: "CubeTop",
                shape: {
                  api: l,
                  xValue: l.value(0),
                  yValue: l.value(1),
                  x: S[0],
                  y: S[1],
                  xAxisPoint: l.coord([l.value(0), 0])
                },
                style: {
                  fill: u(n, 0),
                  stroke: u(n, 1)
                }
              }]
            };
          }
        },
        v
      )), a = a.concat(e.map((v, i) => {
        var g;
        return {
          type: "bar",
          label: {
            normal: {
              show: !0,
              position: "top",
              fontWeight: 400,
              fontSize: 14,
              color: O(i),
              offset: [2, -25]
            }
          },
          itemStyle: {
            color: "transparent"
          },
          data: (g = a[i]) == null ? void 0 : g.data
        };
      }))) : o.data && ((y = (s = o.seriesSetting) == null ? void 0 : s.data) != null && y.length) && (a = (R = o.seriesSetting) == null ? void 0 : R.data.map((v, i) => {
        var g, l;
        return {
          name: (g = w.chartData[i]) == null ? void 0 : g.name,
          type: "custom",
          renderItem: (S, n) => {
            const x = n.coord([n.value(0), n.value(1)]), h = O(i), M = new d.graphic.LinearGradient(0, 0, 0, 1, [
              {
                offset: 0,
                color: u(h, 0)
              },
              {
                offset: 1,
                color: u(h, 1)
              }
            ]), q = new d.graphic.LinearGradient(0, 0, 0, 1, [
              {
                offset: 0,
                color: u(h, 1)
              },
              {
                offset: 1,
                color: u(h, 0)
              }
            ]);
            return {
              type: "group",
              children: [{
                type: "CubeLeft",
                shape: {
                  api: n,
                  xValue: n.value(0),
                  yValue: n.value(1),
                  x: x[0],
                  y: x[1],
                  xAxisPoint: n.coord([n.value(0), 0])
                },
                style: {
                  fill: M,
                  stroke: q
                }
              }, {
                type: "CubeRight",
                shape: {
                  api: n,
                  xValue: n.value(0),
                  yValue: n.value(1),
                  x: x[0],
                  y: x[1],
                  xAxisPoint: n.coord([n.value(0), 0])
                },
                style: {
                  fill: M,
                  stroke: q
                }
              }, {
                type: "CubeTop",
                shape: {
                  api: n,
                  xValue: n.value(0),
                  yValue: n.value(1),
                  x: x[0],
                  y: x[1],
                  xAxisPoint: n.coord([n.value(0), 0])
                },
                style: {
                  fill: u(h, 0),
                  stroke: u(h, 1)
                }
              }]
            };
          },
          data: o.data.length ? (l = w.chartData[i]) == null ? void 0 : l.data : []
        };
      }), a = a.concat((_ = o.seriesSetting) == null ? void 0 : _.data.map((v, i) => {
        var g;
        return {
          type: "bar",
          label: {
            normal: {
              show: !0,
              position: "top",
              fontWeight: 400,
              fontSize: 14,
              color: O(i),
              offset: [2, -25]
            }
          },
          itemStyle: {
            color: "transparent"
          },
          data: (g = w.chartData[i]) == null ? void 0 : g.data
        };
      })), (j = o.seriesSetting) != null && j.xAxisLabel ? r = (G = o.data) != null && G.length ? o.data.map((v) => v[o.seriesSetting.xAxisLabel]) : [] : r = o.data.map((v, i) => `${i}`));
      const c = k(
        {
          series: a,
          xAxis: {
            data: r
          }
        },
        T,
        o.options
      );
      try {
        b.setOption(c);
      } catch {
        b.setOption(T);
      }
    }, B = H(() => {
      var e, t, a, r;
      if (o.options) {
        if (Array.isArray((e = o.options) == null ? void 0 : e.series))
          return o.options.series.some((s) => {
            var y;
            return (s == null ? void 0 : s.data) && ((y = s == null ? void 0 : s.data) == null ? void 0 : y.length) !== 0;
          });
        if (typeof ((t = o.options) == null ? void 0 : t.series) == "object")
          return ((r = (a = o.options.series) == null ? void 0 : a.data) == null ? void 0 : r.length) === 0;
      }
      return !!o.data.length;
    }), P = () => {
      ee(() => {
        b.resize();
      });
    };
    function $() {
      C.value && (A = new ResizeObserver((e) => {
        e.forEach((t) => {
          t.target === C.value && P();
        });
      }), A.observe(C.value));
    }
    return Q(() => {
      E(), C.value && (b = d.init(C.value), z(), D(), b.on("click", (e) => {
        p("com-click", e);
      }), b.on("legendselectchanged", (e) => {
        p("legend-select", e);
      })), window.addEventListener("resize", P), $();
    }), Y(() => {
      window.removeEventListener("resize", P), A && (A.disconnect(), A = null);
    }), V(
      () => o.options,
      (e) => {
        D();
      },
      {
        deep: !0
      }
    ), V(
      [
        () => o.data,
        () => o.seriesSetting
      ],
      () => {
        z(), D();
      },
      {
        deep: !0
      }
    ), V(
      () => o.cStyle,
      () => {
        E(), P();
      },
      {
        deep: !0
      }
    ), {
      ...Z(w),
      curStyle: m,
      chartId: C,
      checkEmpty: B,
      domOnlyClass: f,
      initResizeObserver: $
    };
  }
});
export {
  ye as default
};
