// @ths/design/es/components/chart-muti-ring-pie/src/index.vue2.js
import { defineComponent as le, computed as u, ref as D, reactive as ae, watch as $, onMounted as re, onUnmounted as ne, nextTick as q, useCssVars as ie } from "vue";
import * as oe from "echarts";
import x from "lodash/merge";
import E from "lodash/cloneDeep";
import { splitArray as ue, getDomRandomClass as se, rgb2rgba as ce, replacePollutionHtml as de, setStyle as S } from "../../../utils/index.js";
import { DEFAULT_OPTIONS as U, POLLUTION_HTML as fe } from "../../../utils/constant.js";
const T = { name: "name", value: "value" }, m = E(U.color), F = {
  root: {},
  wrapper: {
    default: {}
  },
  chart: {
    default: {}
  },
  legendRoot: {
    default: {}
  },
  legendRow: {
    default: {}
  },
  legendItem: {
    default: {}
  },
  legendIcon: {
    default: {}
  },
  legendName: {
    default: {}
  },
  legendValue: {
    default: {}
  }
}, ye = 1, A = {
  borderWidth: 0,
  borderRadius: 0
}, p = le({
  name: "TChartMutiRingPie",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 饼图数据
    data: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    alias: {
      type: Object,
      default: () => T,
      required: !1
    },
    // 图形填充透明度
    opacity: {
      type: Number,
      default: 1
    },
    // 图形边框宽度
    borderStyle: {
      type: Object,
      default: A
    },
    // 整个图形宽度
    width: {
      type: String,
      default: "100%",
      required: !1
    },
    // 整个图形高度
    height: {
      type: String,
      default: "100%",
      required: !1
    },
    // 整个图形命名
    title: {
      type: String,
      default: "",
      required: !1
    },
    // 饼图半径
    radius: {
      type: Array,
      default: () => ["72%", "75%"],
      required: !1,
      validator: (e) => e.length === 2
    },
    // 数据单位
    unit: {
      type: String,
      default: "",
      required: !1
    },
    // 总数单位
    allCountUnit: {
      type: String,
      default: "",
      required: !1
    },
    // 显示数据列表
    showList: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 数据列表排列方式
    listStyle: {
      type: Object,
      default: () => ({
        align: "right",
        value: !0,
        fontSize: 14,
        iconSize: 14,
        col: 2,
        cStyle: {}
      }),
      required: !1,
      validator: (e) => !!e
    },
    // 分隔线样式
    splitLineStyle: {
      type: Object,
      default: () => ({
        value: !1,
        cStyle: {}
      })
    },
    // 总数样式
    allCountStyle: {
      type: Object,
      default: () => ({
        value: !1,
        cStyle: {}
      })
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({ ...F }),
      required: !1,
      validator: (e) => !!e
    },
    // 列表容器自定义样式
    lStyle: {
      type: Object,
      default: () => ({}),
      required: !1,
      validator: (e) => !!e
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 环形间隔
    ringGap: {
      type: Number,
      default: 9
    },
    // 环形背景色
    ringBackground: {
      type: String,
      default: "#013561"
    },
    // 按百分比排
    noSum: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["com-click", "legend-click"],
  setup(e, { emit: g }) {
    const o = u(() => e.cStyle), d = se("chart-muti-ring-pie"), b = E(U.color), v = D(null), f = D([]), z = ae({
      label: "",
      value: ""
    }), L = {
      dataColor: "#666666",
      fontColor: "#666666",
      separationColor: "#EEE",
      separationColorOpacity: 1,
      maskColor: "#FFF",
      maskColorOpacity: 0.8
    }, B = u(() => ({
      ...F,
      ...e.cStyle
    })), G = u(() => (t) => {
      var a, l;
      return (a = t == null ? void 0 : t.default) != null && a.margin ? ((l = t == null ? void 0 : t.default) == null ? void 0 : l.margin).map((n) => `${n}px`).join(" ") : null;
    }), W = u(() => (t) => t != null && t.margin ? (t == null ? void 0 : t.margin).map((r) => `${r}px`).join(" ") : null), C = (t) => {
      if (!/^#([0-9A-F]{3}|[0-9A-F]{6})$/i.test(t))
        return !1;
      let l = t.substr(1);
      if (l.length === 3) {
        let k = "";
        for (let h = 0; h < 3; h++)
          k += l[h] + l[h];
        l = k;
      }
      l = `0x${l}`;
      const r = l >> 16, i = l >> 8 & 255, n = l & 255, c = `rgb(${r}, ${i}, ${n})`, te = `rgba(${r}, ${i}, ${n}, ${e.opacity || ye})`;
      return {
        rgb: c,
        rgba: te
      };
    }, H = () => e.showList ? ["left", "right"].includes(e.listStyle.align) ? {
      show: "row",
      value: !!e.listStyle.value,
      style: {
        flexDirection: e.listStyle.align === "right" ? "row" : "row-reverse",
        alignItems: "center"
      },
      cStyle: e.listStyle.cStyle
    } : {
      show: "col",
      value: !!e.listStyle.value,
      style: {
        flexDirection: e.listStyle.align === "bottom" ? "column" : "column-reverse",
        alignItems: "center"
      },
      cStyle: e.listStyle.cStyle
    } : {
      show: "",
      value: !1,
      style: {},
      cStyle: {}
    }, M = (t = "") => {
      const a = t.toUpperCase().replace(/[-.+_]/g, ""), l = fe.filter((r) => r.name === a);
      return l.length > 0 ? l[0].html : t;
    }, V = () => {
      let t = "";
      return o.value.wrapper && (t += S(o.value.wrapper, `.${d}`)), o.value.emptyContainer && (t += S(o.value.emptyContainer, `.${d} .empty`)), o.value.emptyImg && (t += S(o.value.emptyImg, `.${d} .empty .empty-img`)), o.value.emptyText && (t += S(o.value.emptyText, `.${d} .empty .empty-text`)), t;
    }, w = () => {
      const t = document.querySelector(`.${d} .style-container`);
      t && (t.innerHTML = `<style>${V()}</style>`);
    }, J = u(() => {
      var t, a, l, r;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((n) => {
            var c;
            return (n == null ? void 0 : n.data) && ((c = n == null ? void 0 : n.data) == null ? void 0 : c.length) !== 0;
          });
        if (typeof ((a = e.options) == null ? void 0 : a.series) == "object")
          return ((r = (l = e.options.series) == null ? void 0 : l.data) == null ? void 0 : r.length) === 0;
      }
      return !!e.data.length;
    }), _ = (t, a = m) => [...t].map((r, i) => ({
      ...r,
      index: i,
      name: M(r.name),
      style: {
        color: a[i % a.length],
        legend: {
          nameColor: L.fontColor,
          valueColor: L.dataColor,
          backgroundColor: a[i % a.length],
          borderColor: ce(a[i % a.length], 0.5)
        }
      }
    })), Y = (t) => typeof t == "string" && t.includes("[") && t.includes("]") ? JSON.parse(t) : t, K = () => {
      const t = [];
      return f.value.reduce((a, l) => typeof l == "object" ? a + Number(l.value || 0) : a + Number(l || 0), 0), f.value.forEach((a, l) => {
        C((e.options.color || m)[l]) && t.push({
          value: typeof a == "object" ? Number(a.value) || 0 : Number(a) || 0,
          name: typeof a == "object" && a.name || "",
          itemStyle: {
            color: C((e.options.color ?? m)[l]).rgba,
            borderColor: C((e.options.color ?? m)[l]).rgb,
            borderWidth: e.borderStyle.borderWidth ?? A.borderWidth,
            borderRadius: Y(e.borderStyle.borderRadius ?? A.borderRadius)
          }
        });
      }), t;
    };
    let s = null;
    const Q = u(() => ue(_(f.value, e.options.color || b), e.listStyle.col || 2)), X = u(() => _(f.value, e.options.color || b)), y = () => {
      q(() => {
        s.resize();
      });
    };
    function O() {
      if (v.value) {
        s || (s = oe.init(v.value)), s.clear();
        const t = x({
          color: e.options.color || m,
          title: {
            text: e.title,
            left: "center",
            padding: [0, 0, 40, 0]
          },
          legend: {
            show: !1
          },
          tooltip: {
            show: !0
          },
          xAxis: {
            show: !1
          },
          yAxis: {
            show: !1
          },
          series: []
        }, e.options, { animation: !0 }), a = K() || [], l = [];
        let r = 0;
        a.forEach((i, n) => {
          r += i.value;
        }), a.forEach((i, n) => {
          var c;
          l.push({
            type: "pie",
            clockWise: !0,
            hoverAnimation: !1,
            radius: [`${parseInt(e.radius[0], 10) - n * e.ringGap}%`, `${parseInt(e.radius[1], 10) - n * e.ringGap}%`],
            label: { show: !1 },
            data: [{
              value: i.value,
              name: i.name,
              tooltip: { show: (c = e.options.tooltip) == null ? void 0 : c.show },
              ...i
            }, {
              value: (e.noSum ? 100 : r) - i.value,
              tooltip: { show: !1 },
              name: "",
              itemStyle: { color: "rgba(0,0,0,0)" }
            }]
          }), l.push({
            type: "pie",
            silent: !0,
            z: 1,
            clockWise: !0,
            hoverAnimation: !1,
            radius: [`${parseInt(e.radius[0], 10) - n * e.ringGap}%`, `${parseInt(e.radius[1], 10) - n * e.ringGap}%`],
            label: { show: !1 },
            data: [{ value: e.noSum ? 100 : r, tooltip: { show: !1 }, itemStyle: { color: e.ringBackground } }]
          });
        }), t.series = l, s.setOption(t), y();
      }
    }
    function Z(t, a) {
      g("legend-click", {
        current: t,
        currentIndex: a,
        all: f.value
      });
    }
    function j() {
      const t = x(E(T), e.alias);
      f.value = e.data.map((a) => {
        const l = {};
        return Object.keys(t).forEach((r) => {
          r === "name" ? l[r] = de(a[t[r]]) : l[r] = a[t[r]];
        }), { ...a, ...l };
      });
    }
    $(() => [
      e.data,
      e.alias
    ], () => {
      try {
        j(), O(), y();
      } catch (t) {
        console.log(t);
      }
    }, { deep: !0 });
    const I = D(e.listStyle.col);
    $(() => [
      e.title,
      e.options,
      e.radius,
      e.borderStyle,
      e.opacity,
      e.cStyle,
      e.listStyle,
      e.allCountStyle,
      e.splitLineStyle,
      e.allCountUnit,
      e.ringBackground,
      e.ringGap
    ], () => {
      try {
        w(), I.value = e.listStyle.col, O();
      } catch (t) {
        console.log(t);
      }
    }, { deep: !0 }), re(() => {
      w(), j(), O(), y(), window.addEventListener("resize", y), s.on("click", (t) => {
        g("com-click", t);
      });
    }), ne(() => {
      window.removeEventListener("resize", y);
    });
    const R = u(() => H()), ee = u(() => {
      var t;
      return {
        width: e.showList ? e.height : e.width,
        height: e.height,
        background: (t = e.options) == null ? void 0 : t.backgroundColor
      };
    });
    return $(
      () => [e.width, e.height, e.cStyle],
      () => {
        w(), q(() => {
          s.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      currentContent: z,
      chart: v,
      chartDataRow: Q,
      chartData: X,
      listDataStyle: R,
      style: ee,
      cStyles: B,
      col: I,
      handleLegendItemClick: Z,
      marginAddPx: G,
      marginAddPxLine: W,
      defaultColors: b,
      checkEmpty: J,
      curStyle: o,
      domOnlyClass: d
    };
  }
}), N = () => {
  ie((e) => ({
    dada7422: e.col
  }));
}, P = p.setup;
p.setup = P ? (e, g) => (N(), P(e, g)) : N;
export {
  p as default
};
