// @ths/design/es/components/checkbox/src/index.vue2.js
import { defineComponent as F, computed as M, reactive as O, onMounted as j, watch as p, toRefs as N } from "vue";
import P from "lodash/merge";
import T from "lodash/cloneDeep";
import { filterAttrs as R, getDomRandomClass as q, setStyle as d, getCssPropertyString as l } from "../../../utils/index.js";
const S = {
  label: "label",
  value: "value",
  disabled: "disabled"
}, K = F({
  name: "TCheckbox",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        },
        // 单个容器
        itemWrapper: {
          default: {},
          hover: {},
          active: {},
          disabled: {},
          disabledActive: {}
        },
        // 文本
        label: {
          default: {},
          hover: {},
          active: {},
          disabled: {},
          disabledActive: {}
        },
        // 矩形
        rectangle: {
          default: {},
          hover: {},
          active: {},
          disabled: {},
          disabledActive: {}
        },
        // 图标
        icon: {
          default: {},
          disabledActive: {}
        },
        // 按钮多选容器
        buttonWrapper: {
          default: {}
        },
        // 按钮多选框单项容器
        itemButtonWrapper: {
          default: {},
          hover: {},
          active: {},
          disabled: {},
          disabledActive: {}
        },
        // 按钮多选框单项容器中的文本
        itemButtonText: {
          default: {},
          hover: {},
          active: {},
          disabled: {},
          disabledActive: {}
        },
        // 第一个按钮元素
        itemFirstWrapper: {
          default: {}
        },
        // 最后一个按钮元素
        itemLastWrapper: {
          default: {}
        }
      })
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => T(S)
    },
    // 当前选中
    active: {
      type: Array,
      default: () => []
    },
    // 是否采用button多选形式
    useButton: {
      type: Boolean,
      default: () => !1
    }
  },
  // 事件名定义
  emits: ["change"],
  setup(i, { emit: C }) {
    const e = M(() => i.cStyle), a = q("checkbox"), u = O({
      curCheckList: [],
      // 多选框值列表
      checkedList: i.active.length ? i.active : []
      // 已经选中的项
    }), m = () => {
      u.curCheckList = [];
      const t = P(T(S), i.dataAlias), { label: r, value: o, disabled: n } = t;
      i.data && i.data.length ? u.curCheckList = i.data.map((c) => ({
        label: c[o] || "-",
        value: c[r] || "-",
        disabled: c[n] || !1
      })) : (u.curCheckList = [], u.checkedList = []);
    }, L = () => {
      let t = "";
      return e.value.wrapper && (t += d(e.value.wrapper, `.${a} .el-checkbox-group`)), t;
    }, w = () => {
      var r, o, n, c, v, b, s, h, x, f, $, _, W, B, g, y, A;
      let t = "";
      return e.value.label && ((r = e.value.label) != null && r.default) && (t += `.${a}.t-checkbox .el-checkbox .el-checkbox__label {
          ${l(e.value.label.default)}
        }`), e.value.rectangle && ((o = e.value.rectangle) != null && o.default) && (t += `.${a}.t-checkbox .el-checkbox .el-checkbox__inner {
          ${l(e.value.rectangle.default)}
        }`), e.value.label && ((n = e.value.label) != null && n.hover) && (t += `.${a}.t-checkbox .el-checkbox:not(.is-checked):hover .el-checkbox__label {
          ${l(e.value.label.hover)}
        }`), e.value.rectangle && ((c = e.value.rectangle) != null && c.hover) && (t += `.${a}.t-checkbox .el-checkbox:not(.is-checked):hover .el-checkbox__inner {
          ${l(e.value.rectangle.hover)}
        }`), e.value.itemWrapper && ((v = e.value.itemWrapper) != null && v.default) && (t += `.${a}.t-checkbox .el-checkbox {
          ${l(e.value.itemWrapper.default)}
        }`), e.value.itemWrapper && ((b = e.value.itemWrapper) != null && b.hover) && (t += `.${a}.t-checkbox .el-checkbox:not(.is-checked):hover {
          ${l(e.value.itemWrapper.hover)}
        }`), e.value.itemWrapper && ((s = e.value.itemWrapper) != null && s.active) && (t += `.${a}.t-checkbox .el-checkbox.is-checked {
          ${l(e.value.itemWrapper.active)}
        }`), e.value.itemWrapper && ((h = e.value.itemWrapper) != null && h.disabled) && (t += `.${a}.t-checkbox .el-checkbox.is-disabled {
          ${l(e.value.itemWrapper.disabled)}
        }`), e.value.itemWrapper && ((x = e.value.itemWrapper) != null && x.disabledActive) && (t += `.${a}.t-checkbox .el-checkbox.is-disabled.is-checked {
          ${l(e.value.itemWrapper.disabledActive)}
        }`), e.value.label && ((f = e.value.label) != null && f.active) && (t += `.${a} .el-checkbox__input.is-checked + .el-checkbox__label {
          ${l(e.value.label.active)}
        }`), e.value.label && (($ = e.value.label) != null && $.disabled) && (t += `.${a} .el-checkbox__input.is-disabled + span.el-checkbox__label {
          ${l(e.value.label.disabled)}
        }`), e.value.label && ((_ = e.value.label) != null && _.disabledActive) && (t += `.${a} .el-checkbox__input.is-disabled.is-checked + span.el-checkbox__label {
          ${l(e.value.label.disabledActive)}
        }`), e.value.rectangle && ((W = e.value.rectangle) != null && W.active) && (t += `.${a} .el-checkbox__input.is-checked .el-checkbox__inner {
          ${l(e.value.rectangle.active)}
        }`), e.value.rectangle && ((B = e.value.rectangle) != null && B.disabled) && (t += `.${a} .el-checkbox__input.is-disabled .el-checkbox__inner {
          ${l(e.value.rectangle.disabled)}
        }`), e.value.rectangle && ((g = e.value.rectangle) != null && g.disabledActive) && (t += `.${a} .el-checkbox__input.is-disabled.is-checked .el-checkbox__inner {
          ${l(e.value.rectangle.disabledActive)}
        }`), e.value.icon && ((y = e.value.icon) != null && y.default) && (t += `.${a} .el-checkbox__inner::after {
          ${l(e.value.icon.default)}
        }`), e.value.icon && ((A = e.value.icon) != null && A.disabledActive) && (t += `.${a} .el-checkbox__input.is-disabled.is-checked .el-checkbox__inner::after {
          ${l(e.value.icon.disabledActive)}
        }`), t;
    }, H = () => {
      let t = "";
      return e.value.wrapper && (t += d(e.value.wrapper, `.${a}`)), e.value.buttonWrapper && (t += d(e.value.buttonWrapper, `.${a} .el-checkbox-group`)), e.value.itemButtonWrapper && (t += d(e.value.itemButtonWrapper, `.${a} .el-checkbox-button`)), e.value.itemButtonText && (t += d(e.value.itemButtonText, `.${a} .el-checkbox-button .el-checkbox-button__inner`)), e.value.itemFirstWrapper && (t += d(e.value.itemFirstWrapper, `.${a} .el-checkbox-button:first-child`)), e.value.itemLastWrapper && (t += d(e.value.itemLastWrapper, `.${a} .el-checkbox-button:last-child`)), t;
    }, D = () => {
      var r, o, n, c, v, b, s, h;
      let t = "";
      return e.value.itemButtonWrapper && ((r = e.value.itemButtonWrapper) != null && r.active) && (t += `.${a} .el-checkbox-button.is-checked {
          ${l(e.value.itemButtonWrapper.active)}
        }`), e.value.itemButtonWrapper && ((o = e.value.itemButtonWrapper) != null && o.disabled) && (t += `.${a} .el-checkbox-button.is-disabled {
          ${l(e.value.itemButtonWrapper.disabled)}
        }`), e.value.itemButtonWrapper && ((n = e.value.itemButtonWrapper) != null && n.hover) && (t += `.${a} .el-checkbox-button:hover {
          ${l(e.value.itemButtonWrapper.hover)}
        }`), e.value.itemButtonWrapper && ((c = e.value.itemButtonWrapper) != null && c.disabledActive) && (t += `.${a} .el-checkbox-button.is-disabled.is-checked {
          ${l(e.value.itemButtonWrapper.disabledActive)}
        }`), e.value.itemButtonText && ((v = e.value.itemButtonText) != null && v.active) && (t += `.${a} .el-checkbox-button.is-checked .el-checkbox-button__inner {
          ${l(e.value.itemButtonText.active)}
        }`), e.value.itemButtonText && ((b = e.value.itemButtonText) != null && b.hover) && (t += `.${a} .el-checkbox-button:hover .el-checkbox-button__inner {
          ${l(e.value.itemButtonText.hover)}
        }`), e.value.itemButtonText && ((s = e.value.itemButtonText) != null && s.disabled) && (t += `.${a} .el-checkbox-button.is-disabled .el-checkbox-button__inner {
          ${l(e.value.itemButtonText.disabled)}
        }`), e.value.itemButtonText && ((h = e.value.itemButtonText) != null && h.disabledActive) && (t += `.${a} .el-checkbox-button.is-disabled.is-checked .el-checkbox-button__inner {
          ${l(e.value.itemButtonText.disabledActive)}
        }`), t;
    }, k = () => {
      const t = document.querySelector(`.${a} > .style-container`);
      t && !i.useButton ? t.innerHTML = `<style>${L()}${w()}</style>` : t && i.useButton && (t.innerHTML = `<style>${H()}${D()}</style>`);
    };
    return j(() => {
      m(), k();
    }), p(
      () => [i.cStyle, i.useButton],
      () => {
        k();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), p(
      () => [i.data, i.dataAlias],
      () => {
        m(), u.checkedList = i.active.length ? i.active : [];
      },
      {
        deep: !0,
        immediate: !0
      }
    ), p(
      () => [i.active],
      () => {
        u.checkedList = i.active.length ? i.active : [];
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      filterAttrs: R,
      curStyle: e,
      domOnlyClass: a,
      change: (t) => {
        C("change", t);
      },
      ...N(u)
    };
  }
});
export {
  K as default
};
