// @ths/design/es/components/checkbox/src/index.vue.js
import S from "./index.vue2.js";
import { resolveComponent as n, openBlock as r, createElementBlock as d, normalizeClass as $, normalizeStyle as v, createBlock as t, withCtx as a, Fragment as k, renderList as h, mergeProps as y, createTextVNode as C, toDisplayString as L, createElementVNode as w } from "vue";
import "./index.vue3.js";
import T from "../../../_virtual/_plugin-vue_export-helper.js";
function z(e, o, A, E, N, R) {
  var u, i, p, m, c, f, b, g;
  const V = n("el-checkbox"), s = n("el-checkbox-group"), B = n("el-checkbox-button");
  return r(), d("div", {
    class: $(["t-checkbox", e.domOnlyClass]),
    style: v({
      marginTop: (i = (u = e.curStyle.wrapper) == null ? void 0 : u.default) == null ? void 0 : i.marginTop,
      marginLeft: (m = (p = e.curStyle.wrapper) == null ? void 0 : p.default) == null ? void 0 : m.marginLeft,
      marginRight: (f = (c = e.curStyle.wrapper) == null ? void 0 : c.default) == null ? void 0 : f.marginRight,
      marginBottom: (g = (b = e.curStyle.wrapper) == null ? void 0 : b.default) == null ? void 0 : g.marginBottom
    })
  }, [
    e.useButton ? (r(), t(s, {
      key: 1,
      modelValue: e.checkedList,
      "onUpdate:modelValue": o[1] || (o[1] = (l) => e.checkedList = l),
      onChange: e.change
    }, {
      default: a(() => [
        (r(!0), d(k, null, h(e.curCheckList, (l) => (r(), t(B, y({
          key: l,
          label: l.label,
          disabled: l.disabled,
          ref_for: !0
        }, e.filterAttrs(e.$attrs)), {
          default: a(() => [
            C(L(l.value), 1)
          ]),
          _: 2
        }, 1040, ["label", "disabled"]))), 128))
      ]),
      _: 1
    }, 8, ["modelValue", "onChange"])) : (r(), t(s, {
      key: 0,
      modelValue: e.checkedList,
      "onUpdate:modelValue": o[0] || (o[0] = (l) => e.checkedList = l),
      onChange: e.change
    }, {
      default: a(() => [
        (r(!0), d(k, null, h(e.curCheckList, (l) => (r(), t(V, y({
          key: l,
          label: l.label,
          disabled: l.disabled,
          ref_for: !0
        }, e.filterAttrs(e.$attrs)), {
          default: a(() => [
            C(L(l.value), 1)
          ]),
          _: 2
        }, 1040, ["label", "disabled"]))), 128))
      ]),
      _: 1
    }, 8, ["modelValue", "onChange"])),
    o[2] || (o[2] = w("div", { class: "style-container" }, null, -1))
  ], 6);
}
const P = /* @__PURE__ */ T(S, [["render", z]]);
export {
  P as default
};
