// @ths/design/es/components/chart-scatter-base/src/index.vue2.js
import { defineComponent as $, computed as L, ref as q, reactive as M, onMounted as p, onUnmounted as F, watch as C, nextTick as j, toRefs as N } from "vue";
import * as P from "echarts";
import b from "lodash/cloneDeep";
import o from "lodash/merge";
import { DEFAULT_OPTIONS_COMMON as U, DEFAULT_OPTIONS_AXIS as H } from "../../../utils/constant.js";
import { getDomRandomClass as R, setStyle as h } from "../../../utils/index.js";
const B = o(b(U), b(H), {
  grid: {
    top: "13%",
    bottom: "10%",
    left: "6%",
    right: "3%"
  },
  xAxis: {
    type: "value",
    splitLine: {
      show: !0
    }
  },
  yAxis: {
    axisTick: {
      show: !0
    },
    splitLine: {
      show: !0
    }
  }
}), Q = $({
  name: "TChartScatterBase",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    scatter: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: T }) {
    const d = R("chart-scatter-base"), n = L(() => e.cStyle), v = q(null);
    let c = null;
    const S = M({
      // 图表数据
      chartData: []
    });
    function A() {
      var t, i;
      (i = (t = e.seriesSetting) == null ? void 0 : t.data) != null && i.length && e.data && (S.chartData = e.seriesSetting.data.map((s) => {
        const { name: r, xData: m, yData: a } = s;
        return {
          name: r || "--",
          data: e.data.map((l) => [l[m], l[a]])
        };
      }));
    }
    const k = () => {
      console.log(n.value);
      let t = "";
      return n.value.wrapper && (t += h(n.value.wrapper, `.${d}`)), n.value.emptyContainer && (t += h(n.value.emptyContainer, `.${d} .empty`)), n.value.emptyImg && (t += h(n.value.emptyImg, `.${d} .empty .empty-img`)), n.value.emptyText && (t += h(n.value.emptyText, `.${d} .empty .empty-text`)), t;
    }, x = () => {
      const t = document.querySelector(`.${d} .style-container`);
      t && (t.innerHTML = `<style>${k()}</style>`);
    }, _ = L(() => {
      var t, i, s, r;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((a) => {
            var l;
            return (a == null ? void 0 : a.data) && ((l = a == null ? void 0 : a.data) == null ? void 0 : l.length) !== 0;
          });
        if (typeof ((i = e.options) == null ? void 0 : i.series) == "object")
          return ((r = (s = e.options.series) == null ? void 0 : s.data) == null ? void 0 : r.length) === 0;
      }
      return !!e.data.length;
    }), w = () => {
      var m, a, l;
      const { color: t, series: i } = e.options;
      let s = [];
      i ? s = i.map((D, u) => {
        var y, f, g;
        const O = o(
          {
            type: "scatter",
            symbolSize: 20,
            ...e == null ? void 0 : e.scatter
          },
          (g = (f = (y = e.seriesSetting) == null ? void 0 : y.data) == null ? void 0 : f[u]) == null ? void 0 : g.style
        );
        return o({}, O, D);
      }) : e.data && ((a = (m = e.seriesSetting) == null ? void 0 : m.data) != null && a.length) && (s = (l = e.seriesSetting) == null ? void 0 : l.data.map((D, u) => {
        var y, f, g, E, I;
        const O = o(
          {
            type: "scatter",
            name: (y = S.chartData[u]) == null ? void 0 : y.name,
            symbolSize: 20,
            data: e.data.length ? (f = S.chartData[u]) == null ? void 0 : f.data : [],
            ...b(e == null ? void 0 : e.scatter)
          },
          (I = (E = (g = e.seriesSetting) == null ? void 0 : g.data) == null ? void 0 : E[u]) == null ? void 0 : I.style
        );
        return o({}, O, D);
      }));
      let r = o(
        {
          series: s
        },
        B
      );
      r = o(r, e.options), c.clear(), c.setOption(r);
    }, z = () => {
      j(() => {
        c.resize();
      });
    };
    return p(() => {
      x(), v.value && (c = P.init(v.value), A(), w(), c.on("click", (t) => {
        T("com-click", t);
      }), c.on("legendselectchanged", (t) => {
        T("legend-select", t);
      })), window.addEventListener("resize", z);
    }), F(() => {
      window.removeEventListener("resize", z);
    }), C(
      [
        () => e.options,
        () => e.scatter
      ],
      (t) => {
        w();
      },
      {
        deep: !0
      }
    ), C(
      () => [
        e.data,
        e.seriesSetting
      ],
      (t) => {
        A(), w();
      },
      {
        deep: !0
      }
    ), C(
      () => [e.cStyle],
      () => {
        x(), j(() => {
          c.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      ...N(S),
      chartId: v,
      curStyle: n,
      checkEmpty: _,
      domOnlyClass: d
    };
  }
});
export {
  Q as default
};
