// @ths/design/es/components/chart-bar-3d-gradual/src/index.vue2.js
import { defineComponent as ve, ref as v, computed as m, onMounted as ze, watch as Ae, nextTick as we } from "vue";
import * as xe from "echarts";
import z from "lodash/merge";
import { getDomRandomClass as ke, rgb2rgba as Le, setStyle as h } from "../../../utils/index.js";
var ce = /* @__PURE__ */ ((e) => (e.LIGHT = "light", e.DARK = "dark", e))(ce || {});
const qe = [
  "#43d401",
  "#ffea00",
  "#00f6ff",
  "#be4cff",
  "#0167ff",
  "#fd7800",
  "#5c25d2",
  "#ffb540"
], Oe = {
  light: {
    axisLine: "rgba(202, 202, 202, .8)",
    lineColor: "rgba(202, 202, 202, .3)",
    fontColor: "#666",
    tooltipBackgroundColor: "rgba(255, 255, 255, 1)",
    tooltipFontColor: "#666"
  },
  dark: {
    axisLine: "rgba(202, 202, 202, .3)",
    lineColor: "rgba(202, 202, 202, .1)",
    fontColor: "#fff",
    tooltipBackgroundColor: "rgba(34, 68, 99, 1)",
    tooltipFontColor: "#fff"
  }
}, Ie = {
  root: {},
  wrapper: {
    default: {}
  },
  chart: {
    default: {}
  },
  legendRoot: {
    default: {}
  },
  legendItem: {
    default: {}
  },
  legendIcon: {
    default: {}
  },
  legendName: {
    default: {}
  }
}, De = ve({
  name: "TChartBar3dGradual",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 数据
    data: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    // 渐变填充颜色
    gradualColors: {
      type: Array,
      default: () => [],
      required: !0
    },
    // 柱图顶部颜色
    topColors: {
      type: Array,
      default: () => [],
      required: !0
    },
    // 背景色透明度
    shadowOpacity: {
      type: String,
      default: "0.2",
      required: !1
    },
    // 整个图形命名
    title: {
      type: String,
      default: "",
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 图像颜色
    colors: {
      type: Array,
      default: () => qe,
      required: !1,
      validator: (e) => !e.some((y) => !/#[0-9a-fA-F]{6}/.test(y.toString()))
    },
    // 区域锁定
    zoomLock: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 横向显示区域
    scope: {
      type: Array,
      default: () => ["0%", "100%"],
      required: !1,
      validator: (e) => e.length === 2 ? /%/.test(e[0].toString()) ? /%/.test(e[1].toString()) : !/%/.test(e[1].toString()) : !1
    },
    // 整个图形宽度
    width: {
      type: String,
      default: "100%",
      required: !1
    },
    // 整个图形高度
    height: {
      type: String,
      default: "300px",
      required: !1
    },
    // 默认图例排列方式
    legendStyle: {
      type: Object,
      default: () => ({
        position: "top",
        align: "center",
        fontSize: 12,
        iconSize: 12,
        col: 1
      }),
      required: !1,
      validator: (e) => !!e
    },
    // 柱形宽度
    barWidth: {
      type: Number,
      default: 20,
      required: !1
    },
    // 不同系列间距
    barGap: {
      type: String,
      default: "30%",
      required: !1
    },
    // 单位
    unit: {
      type: String,
      default: "",
      required: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 暂无数据容器
        emptyContainer: {
          default: {}
        },
        // 暂无数据图片
        emptyImg: {
          default: {}
        },
        // 暂无数据文字
        emptyText: {
          default: {}
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 主题风格深浅
    themeStyle: {
      type: String,
      default: "light",
      validator: (e) => e.toUpperCase() in ce
    }
  },
  emits: ["col-click", "legend-select"],
  setup(e, { emit: y }) {
    const f = ke("chart-bar-3d-gradual"), s = v([]);
    let S = {};
    const C = v(null), c = m(() => e.cStyle);
    let A = [];
    const ue = m(() => ({
      ...Ie,
      ...e.cStyle
    })), fe = v({
      width: e.width,
      height: e.height
    }), d = () => Oe[e.themeStyle], ge = (a) => {
      const r = {
        dataX: a.name,
        dataIndex: a.dataIndex,
        seriesIndex: a.seriesIndex,
        data: s.value.map((l, n) => ({
          name: s.value[n].name,
          value: s.value[n].data
        }))
      };
      y("col-click", r);
    }, b = (a) => a ? String(a).split(" ").map((l) => `${l}px`).join(" ") : "", ye = m(() => ({
      position: e.legendStyle.position || "top",
      align: e.legendStyle.align || "center",
      margin: b(e.legendStyle.margin),
      fontSize: e.legendStyle.fontSize ? b(e.legendStyle.fontSize) : "14px",
      iconSize: e.legendStyle.iconSize ? b(e.legendStyle.iconSize) : "14px",
      cStyle: e.legendStyle.cStyle
    })), me = () => {
      let a = "";
      return c.value.wrapper && (a += h(c.value.wrapper, `.${f}`)), c.value.emptyContainer && (a += h(c.value.emptyContainer, `.${f} .empty`)), c.value.emptyImg && (a += h(c.value.emptyImg, `.${f} .empty .empty-img`)), c.value.emptyText && (a += h(c.value.emptyText, `.${f} .empty .empty-text`)), a;
    }, w = () => {
      const a = document.querySelector(`.${f} .style-container`);
      a && (a.innerHTML = `<style>${me()}</style>`);
    }, he = m(() => {
      var a, r, l, n;
      if (e.options) {
        if (Array.isArray((a = e.options) == null ? void 0 : a.series))
          return e.options.series.some((i) => {
            var g;
            return (i == null ? void 0 : i.data) && ((g = i == null ? void 0 : i.data) == null ? void 0 : g.length) !== 0;
          });
        if (typeof ((r = e.options) == null ? void 0 : r.series) == "object")
          return ((n = (l = e.options.series) == null ? void 0 : l.data) == null ? void 0 : n.length) === 0;
      }
      return !!e.data.length;
    }), Se = () => /%/.test(e.scope[0].toString()) ? {
      start: Number(e.scope[0].toString().replace(/%/, "")),
      end: Number(e.scope[1].toString().replace(/%/, ""))
    } : {
      startValue: e.scope[0],
      endValue: e.scope[1]
    }, o = () => e.gradualColors.map((r) => {
      const l = [];
      return r.forEach((n) => {
        const t = n.split(",");
        t[t.length - 1] = `${e.shadowOpacity})`, l.push(t.join(","));
      }), l;
    });
    let x = [];
    const Ce = () => {
      const a = [];
      s.value.forEach((r, l) => {
        var n, t;
        (t = (n = s.value) == null ? void 0 : n[l]) != null && t.name && a.push(
          {
            name: s.value[l].name,
            itemStyle: {
              color: e.topColors[l] || e.topColors[e.topColors.length - 1]
            }
          }
        );
      }), x = a;
    }, be = () => {
      const a = [], r = [];
      s.value.forEach((n, t) => {
        var i, g, q, O, I, T, $, B, D, E, W, j, G, F, H, N, R, Z, _, p, M, P, V, K, U, X, J, Q, Y, ee, te, ae, le, oe, re, ne, ie, se, de;
        a.push(
          z({
            name: n.name,
            type: "bar",
            barWidth: ((O = (q = (g = (i = e.seriesSetting) == null ? void 0 : i.data) == null ? void 0 : g[t]) == null ? void 0 : q.style) == null ? void 0 : O.barWidth) || e.barWidth,
            barGap: `${e.barGap}%`,
            data: s.value[t].data,
            showBackground: !0,
            zlevel: -1,
            backgroundStyle: {
              color: {
                type: "linear",
                x: 0,
                x2: 1,
                y: 0,
                y2: 0,
                colorStops: [
                  {
                    offset: 0,
                    color: ((T = (I = o()) == null ? void 0 : I[t]) == null ? void 0 : T[0]) || o()[o().length - 1][0]
                  },
                  {
                    offset: 0.5,
                    color: ((B = ($ = o()) == null ? void 0 : $[t]) == null ? void 0 : B[0]) || o()[o().length - 1][0]
                  },
                  {
                    offset: 0.5,
                    color: ((E = (D = o()) == null ? void 0 : D[t]) == null ? void 0 : E[1]) || o()[o().length - 1][1]
                  },
                  {
                    offset: 1,
                    color: ((j = (W = o()) == null ? void 0 : W[t]) == null ? void 0 : j[0]) || o()[o().length - 1][0]
                  },
                  {
                    offset: 0,
                    color: ((F = (G = o()) == null ? void 0 : G[t]) == null ? void 0 : F[0]) || o()[o().length - 1][0]
                  },
                  {
                    offset: 0,
                    color: ((N = (H = o()) == null ? void 0 : H[t]) == null ? void 0 : N[1]) || o()[o().length - 1][1]
                  }
                ]
              }
            },
            itemStyle: {
              color: {
                type: "linear",
                x: 0,
                x2: 1,
                y: 0,
                y2: 0,
                colorStops: [
                  {
                    offset: 0,
                    color: ((Z = (R = e.gradualColors) == null ? void 0 : R[t]) == null ? void 0 : Z[0]) || e.gradualColors[e.gradualColors.length - 1][0]
                  },
                  {
                    offset: 0.5,
                    color: ((p = (_ = e.gradualColors) == null ? void 0 : _[t]) == null ? void 0 : p[0]) || e.gradualColors[e.gradualColors.length - 1][0]
                  },
                  {
                    offset: 0.5,
                    color: ((P = (M = e.gradualColors) == null ? void 0 : M[t]) == null ? void 0 : P[1]) || e.gradualColors[e.gradualColors.length - 1][1]
                  },
                  {
                    offset: 1,
                    color: ((K = (V = e.gradualColors) == null ? void 0 : V[t]) == null ? void 0 : K[0]) || e.gradualColors[e.gradualColors.length - 1][0]
                  },
                  {
                    offset: 0,
                    color: ((X = (U = e.gradualColors) == null ? void 0 : U[t]) == null ? void 0 : X[0]) || e.gradualColors[e.gradualColors.length - 1][0]
                  },
                  {
                    offset: 0,
                    color: ((Q = (J = e.gradualColors) == null ? void 0 : J[t]) == null ? void 0 : Q[1]) || e.gradualColors[e.gradualColors.length - 1][1]
                  }
                ]
              }
            }
          }, (te = (ee = (Y = e.seriesSetting) == null ? void 0 : Y.data) == null ? void 0 : ee[t]) == null ? void 0 : te.style)
        ), r.push(
          z({
            name: n.name,
            type: "pictorialBar",
            barWidth: ((re = (oe = (le = (ae = e.seriesSetting) == null ? void 0 : ae.data) == null ? void 0 : le[t]) == null ? void 0 : oe.style) == null ? void 0 : re.barWidth) || 20,
            symbol: "diamond",
            symbolOffset: [0, "-50%"],
            symbolPosition: "end",
            symbolSize: [((de = (se = (ie = (ne = e.seriesSetting) == null ? void 0 : ne.data) == null ? void 0 : ie[t]) == null ? void 0 : se.style) == null ? void 0 : de.barWidth) || 20, 10],
            zlevel: 0,
            data: s.value[t].data,
            barGap: `${e.barGap}%`,
            itemStyle: {
              color: e.topColors[t]
            },
            tooltip: {
              show: !1
            }
          })
        );
      });
      let l = [];
      l = [...a, ...r], A = l;
    };
    let u = null;
    function k() {
      var a, r;
      (r = (a = e.seriesSetting) == null ? void 0 : a.data) != null && r.length && e.data && (s.value = e.seriesSetting.data.map((l) => {
        const { name: n, dataKey: t } = l;
        return {
          name: n || "--",
          data: e.data.map((i) => i[t])
        };
      }));
    }
    const L = () => {
      var a, r;
      if (C.value) {
        Ce(), be(), u || (u = xe.init(C.value));
        const l = {
          type: "inside",
          zoomLock: e.zoomLock,
          // 锁定区域：只能平移，不能放大缩小
          ...Se()
        };
        (a = e.seriesSetting) != null && a.xAxisLabel ? S = (r = e.data) != null && r.length ? e.data.map((t) => t[e.seriesSetting.xAxisLabel]) : [] : S = e.data.map((t, i) => i);
        const n = z({
          title: {
            text: e.title,
            left: "50%",
            padding: [0, 5, 0, 5],
            textAlign: "center",
            textStyle: {
              color: d().fontColor,
              fontSize: 16,
              fontWeight: "bold"
            }
          },
          tooltip: {
            trigger: "axis",
            backgroundColor: d().tooltipBackgroundColor,
            borderColor: d().tooltipBackgroundColor,
            textStyle: {
              color: d().fontColor
            },
            axisPointer: {
              type: "line",
              label: {
                backgroundColor: "#6a7985"
              },
              lineStyle: {
                color: Le(d().lineColor, 0.4)
              }
            }
          },
          legend: {
            itemWidth: 15,
            itemHeight: 15,
            itemGap: 20,
            top: "top",
            left: "left",
            orient: "horizontal",
            textStyle: {
              color: d().fontColor,
              fontSize: 16,
              height: 14,
              verticalAlign: "top",
              rich: {
                a: {
                  verticalAlign: "top"
                }
              }
            },
            data: x
          },
          grid: {
            left: 0,
            right: "0%",
            bottom: "1%",
            containLabel: !0
          },
          xAxis: {
            type: "category",
            data: S,
            axisTick: {
              show: !1
            },
            axisLine: {
              show: !0,
              lineStyle: {
                color: d().axisLine
              }
            },
            axisLabel: {
              color: d().fontColor,
              align: "center",
              fontSize: 16
            }
          },
          yAxis: {
            type: "value",
            name: e.unit,
            nameTextStyle: {
              color: d().fontColor,
              align: "center",
              fontSize: 14
            },
            axisLabel: {
              color: d().fontColor,
              align: "center",
              fontSize: 16,
              margin: 20
            },
            splitLine: {
              show: !0,
              lineStyle: {
                color: [d().lineColor],
                type: "dashed"
              }
            }
          },
          dataZoom: e.options.dataZoom || [l],
          series: A
        }, e.options);
        u.clear(), u.setOption(n), u.on("click", (t) => {
          ge(t);
        }), u.on("legendselectchanged", (t) => {
          y("legend-select", t);
        }), window.onresize = () => {
          u.resize();
        };
      }
    };
    return ze(() => {
      w(), k(), L();
    }), Ae(() => e, () => {
      s.value = [], w(), k(), L(), we(() => {
        u.resize(), window.onresize = () => {
          u.resize();
        };
      });
    }, { deep: !0 }), {
      chart: C,
      style: fe,
      curStyle: c,
      legendStyles: ye,
      cStyles: ue,
      checkEmpty: he,
      domOnlyClass: f
    };
  }
});
export {
  De as default
};
