// @ths/design/es/components/chart-bar-3d-gradual/src/index.vue.js
import m from "./index.vue2.js";
import { openBlock as p, createElementBlock as c, normalizeStyle as l, normalizeClass as d, withDirectives as i, createElementVNode as t, vShow as n } from "vue";
import "./index.vue3.js";
import u from "../../../_virtual/_plugin-vue_export-helper.js";
const y = { class: "empty" }, f = ["src"];
function h(e, s, v, E, g, S) {
  var r, o, a;
  return p(), c("div", {
    style: l({
      flexDirection: e.legendStyles.position === "top" ? "column-reverse" : "column",
      ...(r = e.cStyles) == null ? void 0 : r.root,
      ...(a = (o = e.cStyles) == null ? void 0 : o.wrapper) == null ? void 0 : a.default
    }),
    class: d([e.domOnlyClass, "t-chart-bar-3d-gradual"])
  }, [
    i(t("div", {
      ref: "chart",
      style: l({ ...e.style }),
      class: "chart-box"
    }, null, 4), [
      [n, e.checkEmpty]
    ]),
    i(t("div", y, [
      t("img", {
        class: "empty-img",
        style: l({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [n, !e.checkEmpty]
    ]),
    s[1] || (s[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const B = /* @__PURE__ */ u(m, [["render", h]]);
export {
  B as default
};
