// @ths/design/es/components/checkbox-menu/src/index.vue2.js
import { defineComponent as j, computed as k, ref as d, onMounted as H, watch as b } from "vue";
import q from "lodash/merge";
import C from "lodash/cloneDeep";
import { getDomRandomClass as L, getCssPropertyString as f, setStyle as m } from "../../../utils/index.js";
const S = {
  date: "date",
  label: "label",
  children: "children"
}, z = j({
  name: "TRadioMenu",
  props: {
    // 标题
    title: {
      type: String,
      default: "年份预览"
    },
    // 选中的数据
    selectedData: {
      type: String,
      default: ""
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => C(S),
      require: !1
    },
    // 手风琴模式
    accordion: {
      type: Boolean,
      default: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        title: {
          default: {}
        },
        menu: {
          default: {}
        },
        menuCollapse: {
          default: {}
        },
        menuItem: {
          default: {}
        },
        menuItemText: {
          default: {}
        },
        menuItemArrow: {
          default: {}
        },
        menuContent: {
          default: {}
        },
        subMenuItem: {
          default: {},
          active: {}
        }
      })
    }
  },
  /**
   * item-click: 点击事件
   */
  emits: ["item-click", "cancel-one"],
  setup(u, { emit: v }) {
    const t = k(() => u.cStyle), a = L("year-contrast"), o = d([]), D = d([]), c = d({}), r = d([]), p = (e) => {
      try {
        const n = q(S, u.alias);
        if (!Array.isArray(e))
          return;
        e.forEach((l, i) => {
          typeof l != "object" ? e[i] = {
            date: l
          } : (l.date = l[n.date], l.label = l[n.label], l.children = l[n.children]), l != null && l.children && l.children.length && p(l.children);
        });
      } catch (n) {
        console.warn(n);
      }
    }, g = (e) => {
      r.value[0] = e;
    }, A = (e) => {
      e.children && e.children.length ? r.value.includes(e.label) ? r.value.splice(r.value.indexOf(e.label), 1) : u.accordion ? g(e.label) : r.value.push(e.label) : (c.value = e, v("item-click", c.value));
    }, M = (e) => {
      c.value = e, v("item-click", c.value);
    }, T = (e) => {
      try {
        o.value.forEach((n) => {
          var l;
          if (n.date === e)
            throw c.value = n, new Error("跳出循环");
          if (n.children) {
            const i = ((l = n == null ? void 0 : n.children) == null ? void 0 : l.find((s) => s.date === e.trim())) || "";
            if (i)
              throw c.value = i, new Error("跳出循环");
          }
        });
      } catch (n) {
        if (n.message !== "跳出循环")
          throw n;
      }
    }, E = () => {
      var n, l, i, s, h, I, $, w;
      let e = "";
      return (n = t.value.menuItem) != null && n.active && (e += `
          .${a} .menu .menu-collapse .menu-item.active {
            ${f((l = t.value.menuItem) == null ? void 0 : l.active)}
          }
        `), (i = t.value.menuItemText) != null && i.active && (e += `
          .${a} .menu .menu-collapse .menu-item.active .menu-item-text {
            ${f((s = t.value.menuItemText) == null ? void 0 : s.active)}
          }
        `), (h = t.value.subMenuItem) != null && h.active && (e += `
          .${a} .menu .menu-collapse .menu-content .menu-children.active {
            ${f((I = t.value.subMenuItem) == null ? void 0 : I.active)}
          }
        `), ($ = t.value.menuItemArrow) != null && $.active && (e += `
          .${a} .menu .menu-collapse .menu-item.active .menu-item-arrow.rotate {
            ${f((w = t.value.menuItemArrow) == null ? void 0 : w.active)}
          }
        `), e;
    }, O = () => {
      let e = "";
      return t.value.wrapper && (e += m(t.value.wrapper, `.${a}`)), t.value.title && (e += m(t.value.title, `.${a} .title`)), t.value.menu && (e += m(t.value.menu, `.${a} .menu`)), t.value.menuCollapse && (e += m(t.value.menuCollapse, `.${a} .menu .menu-collapse`)), t.value.menuItem && (e += m(t.value.menuItem, `.${a} .menu .menu-collapse .menu-item`)), t.value.menuItemText && (e += m(t.value.menuItemText, `.${a} .menu .menu-collapse .menu-item .menu-item-text`)), t.value.menuItemArrow && (e += m(t.value.menuItemArrow, `.${a} .menu .menu-collapse .menu-item .menu-item-arrow`)), t.value.menuContent && (e += m(t.value.menuContent, `.${a} .menu .menu-collapse .menu-content`)), t.value.subMenuItem && (e += m(t.value.subMenuItem, `.${a} .menu .menu-collapse .menu-content .menu-children`)), e;
    }, y = () => {
      const e = document.querySelector(`.${a} .style-container`);
      e && (e.innerHTML = `<style>${O()}${E()}</style>`);
    };
    return H(() => {
      y();
    }), b(() => u.cStyle, () => {
      y();
    }, { deep: !0 }), b([() => u.data, () => u.alias, () => u.selectedData], () => {
      c.value = {}, o.value = C(u.data), p(o.value), T(u.selectedData);
    }, {
      deep: !0,
      immediate: !0
    }), {
      curStyle: t,
      domOnlyClass: a,
      curData: o,
      showContent: A,
      addDate: M,
      selectedDates: D,
      selectedItem: c,
      hiddenMenuItems: r
    };
  }
});
export {
  z as default
};
