// @ths/design/es/components/chart-liquidfill-base/src/index.vue2.js
import { defineComponent as v, ref as O, computed as s, watch as c, nextTick as C, onMounted as S } from "vue";
import * as q from "echarts";
import "echarts-liquidfill";
import { rgb2rgba as d } from "../../../utils/index.js";
const f = {
  wrapper: {
    default: {}
  },
  chart: {
    default: {}
  }
}, g = "#00FF54", F = v({
  name: "TChartLiquidfillBase",
  props: {
    // 水波数据
    data: {
      type: Number,
      default: 0.4,
      required: !1,
      validator: (e) => e >= 0 && e <= 1
    },
    // 波浪数量
    wave: {
      type: Number,
      default: 2,
      required: !1
    },
    // 整个图形大小
    size: {
      type: Number,
      default: 200,
      required: !1
    },
    // 球半径
    radius: {
      type: String,
      default: "90%",
      required: !1,
      validator: (e) => /%$/.test(e)
    },
    // 波浪幅度
    amplitude: {
      type: String,
      default: "3%",
      required: !1,
      validator: (e) => /%$/.test(e)
    },
    // 图像颜色
    color: {
      type: String,
      default: g,
      required: !1,
      validator: (e) => /#[0-9a-fA-F]{6}/.test(e)
    },
    // 是否显示默认外阴影
    showOuter: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 外部阴影大小
    outerWidth: {
      type: Number,
      default: 20,
      required: !1
    },
    // 是否显示默认内阴影
    showInner: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 波浪透明度
    waveOpacity: {
      type: Number,
      default: 0.8,
      required: !1,
      validator: (e) => e >= 0 && e <= 1
    },
    // 外部阴影透明度
    outerOpacity: {
      type: Number,
      default: 0.2,
      required: !1,
      validator: (e) => e >= 0 && e <= 1
    },
    // 内部阴影透明度
    innerOpacity: {
      type: Number,
      default: 0.3,
      required: !1,
      validator: (e) => e >= 0 && e <= 1
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => f
    }
  },
  setup(e) {
    const o = O(null), y = s(() => ({
      width: `${Number(e.size)}px`,
      height: `${Number(e.size)}px`
    })), m = s(() => ({
      ...f,
      ...e.cStyle
    })), i = () => ({
      color: e.color,
      borderColor: e.color,
      waveColor: d(e.color, e.waveOpacity),
      outerColor: d(e.color, e.outerOpacity),
      innerColor: d(e.color, e.innerOpacity)
    }), h = () => {
      const t = Number(e.radius.split("%")[0]), r = Number(e.size) / 100, a = 5;
      return e.showOuter ? (t * r + e.outerWidth) / r - e.outerWidth / r * 2 - a : t - a;
    }, b = (t, r = 1) => {
      const a = [];
      for (let u = 0; u < r; u++)
        a.push(t);
      return a;
    };
    let l = null;
    function n() {
      if (o.value) {
        l || (l = q.init(o.value, void 0, { renderer: "svg" }));
        const t = {
          ...e.options,
          series: []
        }, r = {
          type: "liquidFill",
          data: b(e.data, e.wave),
          color: [i().waveColor],
          itemStyle: {
            // opacity: 0.6,
            shadowBlur: 0
            // 波浪上的阴影
            // shadowColor: 'rgba(0, 0, 0, 0.4)',
          },
          radius: e.radius,
          amplitude: e.amplitude,
          direction: "right",
          period: (u, w) => w % 2 === 0 ? 1200 : 1500,
          // 周期：交错效果占时通过不同周期实现
          // phase: Math.PI / 4, // 相位 FIXME: 相位暂时无法分别控制，所以出不了交错效果
          // waveAnimation: 1,
          // waveLength: '80%',
          animationDuration: 2e3,
          // 向上填充的时间
          animationDurationUpdate: 2e3,
          // animationEasingUpdate: 'cubicOut',
          backgroundStyle: {
            borderWidth: 1,
            borderColor: i().borderColor,
            color: i().outerColor
          },
          outline: {
            show: e.showOuter,
            borderDistance: 0,
            itemStyle: {
              borderWidth: e.outerWidth,
              borderColor: i().outerColor
            }
          },
          label: {
            show: !1
          },
          ...e.options.dataSeries
          // FIXME: 没办法，该插件没有定义类型，就算定义了echarts也不识别
        }, a = {
          z: 1,
          name: "内部阴影",
          type: "pie",
          radius: ["0%", `${h()}%`],
          avoidLabelOverlap: !0,
          silent: !0,
          clockwise: !0,
          label: { show: !1 },
          itemStyle: {
            color: i().innerColor
          },
          data: [
            { value: 100, name: "内部阴影" }
          ],
          ...e.options.innerSeries
        };
        t.series = [
          r
          // 水波series
        ], e.showInner && (t.series = [
          ...t.series,
          a
        ]), l.setOption(t), window.onresize = () => {
          l.resize();
        };
      }
    }
    return c(() => [
      e.data,
      e.color,
      e.options,
      e.wave,
      e.outerOpacity,
      e.innerOpacity,
      e.size,
      e.radius,
      e.amplitude,
      e.showOuter,
      e.outerWidth,
      e.showInner,
      e.waveOpacity
    ], () => {
      n();
    }), c(() => e.size, () => {
      C(() => {
        l.resize();
      });
    }), S(() => {
      n();
    }), {
      chart: o,
      style: y,
      cStyles: m
    };
  }
});
export {
  F as default
};
