// @ths/design/es/components/chart-pie-percent/src/index.vue2.js
import { defineComponent as _, computed as W, ref as q, nextTick as v, onMounted as H, onUnmounted as M, watch as b } from "vue";
import * as N from "echarts";
import u from "lodash/merge";
import R from "lodash/cloneDeep";
import { DEFAULT_OPTIONS as U } from "../../../utils/constant.js";
import { getDomRandomClass as B, setStyle as f } from "../../../utils/index.js";
const C = u(
  R(U),
  {
    grid: {
      show: !1
    },
    tooltip: {
      show: !0,
      trigger: "item"
    },
    title: {
      show: !1
    },
    legend: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    }
  }
), Y = _({
  name: "TChartPiePercent",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 饼图数据
    data: {
      type: Number,
      default: () => 0
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => C,
      required: !1
    },
    // series样式设置
    seriesStyle: {
      type: Object,
      default: () => ({})
    },
    // 饼图主题配置
    pie: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const a = B("chart-pie-percent"), i = W(() => e.cStyle), o = q(null), I = () => {
      let t = "";
      return i.value.wrapper && (t += f(i.value.wrapper, `.${a}`)), i.value.emptyContainer && (t += f(i.value.emptyContainer, `.${a} .empty`)), i.value.emptyImg && (t += f(i.value.emptyImg, `.${a} .empty .empty-img`)), i.value.emptyText && (t += f(i.value.emptyText, `.${a} .empty .empty-text`)), t;
    }, A = () => {
      const t = document.querySelector(`.${a} .style-container`);
      t && (t.innerHTML = `<style>${I()}</style>`);
    };
    let n = null;
    const m = () => {
      o.value && (o.value.getBoundingClientRect().width ?? 0) && v(() => {
        n.resize();
      });
    }, P = W(() => {
      var t, s, r, c;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((l) => {
            var y;
            return (l == null ? void 0 : l.data) && ((y = l == null ? void 0 : l.data) == null ? void 0 : y.length) !== 0;
          });
        if (typeof ((s = e.options) == null ? void 0 : s.series) == "object")
          return ((c = (r = e.options.series) == null ? void 0 : r.data) == null ? void 0 : c.length) === 0;
      }
      return typeof e.data != "number" ? !1 : (v(() => {
        m();
      }), !0);
    }), d = () => {
      var t, s, r, c, S, l, y, O, x, k, z, T;
      if (o.value) {
        n || (n = N.init(o.value));
        const j = (t = e.options) == null ? void 0 : t.series, D = (s = e.options) == null ? void 0 : s.color;
        let h = [];
        j ? h = j.map((w, G) => {
          var E, L, $, F;
          return u(
            {
              type: "pie",
              animation: !1,
              silent: !0,
              radius: ["50%", "65%"],
              avoidLabelOverlap: !0,
              clockwise: typeof ((E = e.seriesStyle) == null ? void 0 : E.clockwise) == "boolean" ? (L = e.seriesStyle) == null ? void 0 : L.clockwise : !0,
              // 饼图的扇区是否是顺时针排布
              startAngle: typeof (($ = e.seriesStyle) == null ? void 0 : $.startAngle) == "number" ? (F = e.seriesStyle) == null ? void 0 : F.startAngle : 90,
              // 起始角度，支持范围[0, 360]
              label: {
                show: !1,
                position: "center"
              },
              ...e == null ? void 0 : e.pie
            },
            w
          );
        }) : e.data.toString() && (h = [
          u(
            {
              type: "pie",
              animation: !1,
              silent: !0,
              radius: ((r = e.seriesStyle) == null ? void 0 : r.radius) || ["50%", "65%"],
              avoidLabelOverlap: !0,
              clockwise: typeof ((c = e.seriesStyle) == null ? void 0 : c.clockwise) == "boolean" ? (S = e.seriesStyle) == null ? void 0 : S.clockwise : !0,
              // 饼图的扇区是否是顺时针排布
              startAngle: typeof ((l = e.seriesStyle) == null ? void 0 : l.startAngle) == "number" ? (y = e.seriesStyle) == null ? void 0 : y.startAngle : 90,
              // 起始角度，支持范围[0, 360
              label: {
                show: !1,
                position: "center"
              },
              data: [
                {
                  value: 100 - e.data,
                  name: "背景",
                  label: {
                    show: !1
                  },
                  itemStyle: {
                    color: ((O = e.seriesStyle) == null ? void 0 : O.bgColor) || "#F1F1F1"
                  }
                },
                {
                  value: e.data,
                  name: "单值",
                  label: {
                    show: !0,
                    position: "center",
                    formatter: "{c}%",
                    fontSize: ((x = e.seriesStyle) == null ? void 0 : x.fontSize) || 12,
                    color: (k = e.seriesStyle) == null ? void 0 : k.fontColor,
                    fontWeight: ((z = e.seriesStyle) == null ? void 0 : z.fontWeight) || "normal"
                  },
                  itemStyle: {
                    color: ((T = e.seriesStyle) == null ? void 0 : T.valueColor) || D && D[0] || C.color[0]
                  }
                }
              ],
              ...e == null ? void 0 : e.pie
            },
            {
              itemStyle: {
                borderWidth: 0
              }
            }
          )
        ]);
        let g = u(
          {
            series: h
          },
          C
        );
        g = u(g, e.options), n.clear();
        try {
          n.setOption(g);
        } catch (w) {
          console.log(w);
        }
      }
    };
    return H(() => {
      A(), d(), window.addEventListener("resize", m);
    }), M(() => {
      window.removeEventListener("resize", m);
    }), b(
      () => [
        e.pie,
        e.options,
        e.seriesStyle
      ],
      () => {
        d();
      },
      {
        deep: !0
      }
    ), b(
      () => [
        e.data
      ],
      () => {
        d();
      },
      {
        deep: !0
      }
    ), b(
      () => [e.cStyle],
      () => {
        A(), v(() => {
          n.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      chart: o,
      curStyle: i,
      checkEmpty: P,
      domOnlyClass: a
    };
  }
});
export {
  Y as default
};
