// @ths/design/es/components/chart-pie-percent/src/index.vue.js
import a from "./index.vue2.js";
import { openBlock as c, createElementBlock as m, normalizeClass as p, normalizeStyle as l, withDirectives as i, createElementVNode as t, vShow as n } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  ref: "chart",
  class: "chart"
}, y = { class: "empty" }, f = ["src"];
function h(e, s, v, E, k, _) {
  var r, o;
  return c(), m("div", {
    class: p(["t-chart-pie-percent", e.domOnlyClass]),
    style: l({ ...(o = (r = e.curStyle) == null ? void 0 : r.wrapper) == null ? void 0 : o.default })
  }, [
    i(t("div", u, null, 512), [
      [n, e.checkEmpty]
    ]),
    i(t("div", y, [
      t("img", {
        class: "empty-img",
        style: l({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [n, !e.checkEmpty]
    ]),
    s[1] || (s[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const S = /* @__PURE__ */ d(a, [["render", h]]);
export {
  S as default
};
