// @ths/design/es/components/two-dimension-image/src/index.vue.js
import l from "./index.vue2.js";
import { resolveComponent as c, openBlock as d, createElementBlock as m, normalizeClass as p, normalizeStyle as u, createVNode as f, withCtx as g, createElementVNode as t, toDisplayString as h } from "vue";
import "./index.vue3.js";
import w from "../../../_virtual/_plugin-vue_export-helper.js";
const y = { class: "center-content" }, _ = ["src"], C = ["title"];
function D(e, r, S, b, v, $) {
  var n, a, o, s;
  const i = c("t-two-dimension-container");
  return d(), m("div", {
    class: p(["t-two-dimension-image", e.domOnlyClass]),
    style: u({ width: (a = (n = e.curStyle.wrapper) == null ? void 0 : n.default) == null ? void 0 : a.width, height: (s = (o = e.curStyle.wrapper) == null ? void 0 : o.default) == null ? void 0 : s.height })
  }, [
    f(i, {
      class: "two-dimension-container",
      "left-data": e.curLeftData,
      "right-data": e.curRightData,
      "bg-url": e.bgUrl,
      "c-style": e.curStyle,
      "is-show-percent": !1
    }, {
      default: g(() => [
        t("div", y, [
          t("img", {
            src: e.curCenterData.innerUrl,
            alt: "",
            draggable: "false"
          }, null, 8, _),
          t("span", {
            class: "desc",
            title: e.curCenterData.title
          }, h(e.curCenterData.title || "-"), 9, C)
        ])
      ]),
      _: 1
    }, 8, ["left-data", "right-data", "bg-url", "c-style"]),
    r[0] || (r[0] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const N = /* @__PURE__ */ w(l, [["render", D]]);
export {
  N as default
};
