// @ths/design/es/components/trace-path/src/index.vue2.js
import { defineComponent as T, reactive as R, ref as A, computed as p, onMounted as B, nextTick as w, onBeforeUnmount as W, watch as G } from "vue";
import { getDomRandomClass as I, setStyle as N } from "../../../utils/index.js";
import D from "./round.js";
import E from "./round-reverse.js";
const H = T({
  name: "TTracePath",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 路径
    path: {
      type: String,
      default: ""
    },
    // 线条长度
    lineLength: {
      type: Number,
      default: 0
    },
    // 线条宽度
    lineWidth: {
      type: Number,
      default: 1
    },
    // 速度
    basicSpeed: {
      type: Number,
      default: 0
    },
    // 加速度
    acceleratedSpeed: {
      type: Number,
      default: 0
    },
    // 运动轨迹类型
    motionTrailType: {
      type: String,
      default: "yunsu"
    },
    // 渐变色
    gradientList: {
      type: Array,
      default: () => []
    },
    // 是否展示背景路径
    showBackgroundPath: {
      type: Boolean,
      default: !1
    },
    // 背景路径线条宽度
    backgroundPathLineWidth: {
      type: Number,
      default: 0
    },
    // 背景路径渐变色
    backgroundPathGradientList: {
      type: Array,
      default: () => []
    },
    // 是否反向播放
    isReverse: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["title-click"],
  setup(n, { emit: F }) {
    const e = R({
      canvas: null,
      ctx: null,
      pathLength: 0,
      isChangeSpeed: !1
    }), v = 10, r = A(), g = p(() => n.cStyle), u = I("trace-path"), k = () => {
      let a = "";
      return g.value.wrapper && (a += N(g.value.wrapper, `.${u}`)), a;
    };
    function C() {
      r.value && (e.pathLength = r.value.getTotalLength());
    }
    function L() {
      e.canvas = document.getElementById(u), e.canvas && (e.ctx = e.canvas.getContext("2d"), C());
    }
    function f() {
      var i, b;
      const a = new Path2D(n.path), l = (i = r.value) == null ? void 0 : i.getPointAtLength(0), c = (b = r.value) == null ? void 0 : b.getPointAtLength(e.pathLength - 1), s = e.ctx.createLinearGradient(
        l.x,
        l.y,
        c.x,
        c.y
      );
      n.backgroundPathGradientList.forEach((d) => {
        s.addColorStop((d == null ? void 0 : d.point) || 0, (d == null ? void 0 : d.color) || "rgba(0, 0, 0, 0)");
      }), e.ctx.strokeStyle = s, e.ctx.lineWidth = n.backgroundPathLineWidth, e.ctx.stroke(a);
    }
    let o, t;
    function y() {
      var a;
      if (o = (a = r.value) == null ? void 0 : a.getPointAtLength(0), o) {
        let l = 0;
        n.motionTrailType !== "yunsu" && (l = n.acceleratedSpeed), t = new D(o.x, o.y, n.lineWidth, 0, v, n.lineLength, n.basicSpeed, l, r.value);
        for (let c = 0; c < n.lineLength; c++)
          t.updateInitParam(), t.initDraw();
      }
    }
    let h;
    function P() {
      function a() {
        var l;
        e.ctx.clearRect(0, 0, e.canvas.width, e.canvas.height), n.showBackgroundPath && f(), t.moveCirclesPosition(n.motionTrailType);
        for (const c in t.circles) {
          t.circles[c].position > e.pathLength && (t.circles[c].x += 1e7), e.ctx.beginPath(), e.ctx.arc(t.circles[c].x, t.circles[c].y, t.circles[c].r, 0, Math.PI * 2);
          const s = e.ctx.createLinearGradient(
            t.circles[0].x,
            t.circles[0].y,
            t.circles[t.circles.length - 1].x,
            t.circles[t.circles.length - 1].y
          );
          n.gradientList.forEach((i) => {
            s.addColorStop((i == null ? void 0 : i.point) || 0, (i == null ? void 0 : i.color) || "rgba(0, 0, 0, 0)");
          }), e.ctx.fillStyle = s, e.ctx.fill();
        }
        ((l = t.circles[0]) == null ? void 0 : l.position) > e.pathLength && (e.ctx.clearRect(0, 0, e.canvas.width, e.canvas.height), n.showBackgroundPath && f(), t.circles.length = 0, t.t = 0, y()), h = requestAnimationFrame(a);
      }
      a();
    }
    function x() {
      var a;
      if (o = (a = r.value) == null ? void 0 : a.getPointAtLength(e.pathLength), o) {
        let l = 0;
        n.motionTrailType !== "yunsu" && (l = n.acceleratedSpeed), t = new E(o.x, o.y, n.lineWidth, 0, v, e.canvas.width, n.basicSpeed, l, r.value);
        for (let c = 0; c < n.lineLength; c++)
          t.updateInitParam(), t.initDraw();
      }
    }
    function m() {
      function a() {
        var l;
        e.ctx.clearRect(0, 0, e.canvas.width, e.canvas.height), n.showBackgroundPath && f(), t.moveCirclesPositionReverse(n.motionTrailType);
        for (const c in t.circles) {
          t.circles[c].position < 0 && (t.circles[c].x -= 5), e.ctx.beginPath(), e.ctx.arc(t.circles[c].x, t.circles[c].y, t.circles[c].r, 0, Math.PI * 2);
          const s = e.ctx.createLinearGradient(
            t.circles[t.circles.length - 1].x,
            t.circles[t.circles.length - 1].y,
            t.circles[0].x,
            t.circles[0].y
          );
          n.gradientList.forEach((i) => {
            s.addColorStop((i == null ? void 0 : i.point) || 0, (i == null ? void 0 : i.color) || "rgba(0, 0, 0, 0)");
          }), e.ctx.fillStyle = s, e.ctx.fill();
        }
        ((l = t.circles[n.lineLength - 1]) == null ? void 0 : l.x) < r.value.getPointAtLength(0).x && (e.ctx.clearRect(0, 0, e.canvas.width, e.canvas.height), n.showBackgroundPath && f(), t.circles.length = 0, t.t = 0, x()), h = requestAnimationFrame(a);
      }
      a();
    }
    const S = () => {
      const a = document.querySelector(`.${u} .style-container`);
      a && (a.innerHTML = `<style>${k()}</style>`);
    };
    return B(() => {
      S(), w(() => {
        L(), n.lineLength && (n.isReverse ? (x(), e.ctx && m()) : (y(), e.ctx && P()));
      });
    }), W(() => {
      h && cancelAnimationFrame(h);
    }), G(() => n, () => {
      S(), h && cancelAnimationFrame(h), w(() => {
        L(), n.lineLength && (n.isReverse ? (x(), e.ctx && m()) : (y(), e.ctx && P()));
      });
    }, { deep: !0 }), {
      curStyle: g,
      domOnlyClass: u,
      pathRef: r
    };
  }
});
export {
  H as default
};
