// @ths/design/es/components/trace-path/src/round.js
var E = Object.defineProperty;
var l = (o, t, s) => t in o ? E(o, t, { enumerable: !0, configurable: !0, writable: !0, value: s }) : o[t] = s;
var i = (o, t, s) => (l(o, typeof t != "symbol" ? t + "" : t, s), s);
const e = class e {
  constructor(t, s, h, n, c, a, p, P, S) {
    i(this, "circles", []);
    i(this, "initX");
    i(this, "initY");
    i(this, "initPosition");
    i(this, "x");
    i(this, "y");
    i(this, "r");
    i(this, "circlePosition");
    i(this, "direction");
    i(this, "speed");
    i(this, "basicSpeed");
    i(this, "jiaSpeed");
    i(this, "path");
    i(this, "t", 0);
    this.initX = -(a + 1), this.initY = 0, this.initPosition = -(a + 1), this.x = t, this.y = s, this.r = h, this.circlePosition = n, this.direction = c, this.speed = this.direction * 0.07, this.basicSpeed = p, this.jiaSpeed = P, this.path = S;
  }
  // 填充小球
  initDraw() {
    this.circles.push({
      x: this.initX,
      y: this.initY,
      r: this.r,
      position: this.initPosition
    });
  }
  updateInitParam() {
    this.initX++, this.initPosition++;
  }
  /**
   * 当小球数组填充完毕之后，开始给每个小球设置加速度
   */
  moveCirclesPosition(t) {
    let s = this.speed + this.basicSpeed / e.CONSTANT_SPEED + (t === "jiansu" ? 0 - this.jiaSpeed : this.jiaSpeed) * this.t;
    s < e.MINIMUM_SPEED && (s = e.MINIMUM_SPEED), this.circles.forEach((h) => {
      if (this.basicSpeed > 0) {
        h.position += s;
        const [n, c] = this.realPosition(h.position);
        h.x = n, h.y = c;
      }
    }), this.t += 1e-3;
  }
  /**
   * 当小球数组填充完毕之后，开始给每个小球设置加速度（反向）
   */
  moveCirclesPositionReverse(t) {
    let s = this.speed + this.basicSpeed / e.CONSTANT_SPEED + (t === "jiansu" ? 0 - this.jiaSpeed : this.jiaSpeed) * this.t;
    s < e.MINIMUM_SPEED && (s = e.MINIMUM_SPEED), this.circles.forEach((h) => {
      if (this.basicSpeed > 0) {
        h.position -= s;
        const [n, c] = this.realPosition(h.position);
        h.x = n, h.y = c;
      }
    }), this.t += 1e-3;
  }
  /**
   * 设置小球点之间的间隔
   */
  setCirclePosition() {
    this.circlePosition += this.speed;
  }
  /**
   * 获取当前点的横纵坐标
   */
  realMove() {
    const t = this.path.getPointAtLength(this.circlePosition);
    this.x = t.x, this.y = t.y;
  }
  /**
   * 获取每个小球当前点的横纵坐标
   */
  realPosition(t) {
    const s = this.path.getPointAtLength(t);
    return [s.x, s.y];
  }
};
i(e, "MINIMUM_SPEED", 0.4), // 加速度系数
i(e, "ACCELERATED_SPEED", 1e3), // 调整经加速度变化的速度系数
i(e, "REAL_SPEED", 900), // 恒定速度系数
i(e, "CONSTANT_SPEED", 5);
let r = e;
export {
  r as default
};
