// @ths/design/es/components/list-card-item/src/index.vue2.js
import { defineComponent as d, computed as a, ref as u } from "vue";
const p = d({
  name: "TListCardItem",
  props: {
    // 排列方向 horizontal  vertical
    direction: {
      default: "vertical",
      type: String
    },
    // 交换标题和值的排列顺序
    exchangeOrder: {
      default: !1,
      type: Boolean
    },
    // 单位是否换行
    unitBreak: {
      default: !1,
      type: Boolean
    },
    // 数据
    data: {
      type: Object,
      default: () => ({})
    },
    // 字段关键词
    keyField: {
      default: () => ({
        icon: "icon",
        title: "title",
        num: "num",
        unit: "unit",
        trend: "trend",
        color: "color",
        special: "special",
        selectIcon: "selectIcon",
        unitStyle: "unitStyle"
      }),
      type: Object
    },
    // 数据中唯一属性key
    activeKeyField: {
      type: String,
      default: "title"
    },
    // 选中项
    active: {
      type: String,
      default: ""
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {},
          active: {}
        },
        // 右侧文本区域容器
        rightContent: {
          default: {}
        },
        // 特殊状态的容器
        specialWrapper: {
          default: {}
        },
        // 图标
        icon: {
          default: {},
          active: {}
        },
        // 卡片图标
        cardIcon: {
          default: {}
        },
        // 特殊状态卡片图标
        specialCardIcon: {
          default: {}
        },
        // 标题
        title: {
          default: {},
          active: {}
        },
        // 数值
        num: {
          default: {},
          active: {}
        },
        // 数值容器
        numBox: {
          default: {},
          active: {}
        },
        // 单位
        unit: {
          default: {},
          active: {}
        }
      })
    }
  },
  // 事件名定义
  emits: ["card-click"],
  setup(e, { emit: i }) {
    const l = a(() => e.active), c = u(""), n = a(() => e.cStyle), r = a(() => e.exchangeOrder);
    return {
      clickListCard: (t) => {
        i("card-click", t);
      },
      activeTitle: l,
      currentStyle: n,
      currentExchangeOrder: r,
      hoverTitle: c,
      hoverCard: (t) => {
        e.active && (c.value = t);
      }
    };
  }
});
export {
  p as default
};
