// @ths/design/es/components/list-card-air-quality-forecast/src/index.vue.js
import m from "./index.vue2.js";
import { openBlock as i, createElementBlock as s, normalizeClass as x, normalizeStyle as t, Fragment as _, renderList as ee, createElementVNode as l, toDisplayString as r } from "vue";
import "./index.vue3.js";
import ae from "../../../_virtual/_plugin-vue_export-helper.js";
const le = ["onClick"], te = { class: "left" }, re = ["title"];
function ie(e, u, se, ue, oe, de) {
  var o, d;
  return i(), s("div", {
    class: x(["t-list-card-air-quality-forecast", e.domOnlyClass]),
    style: t({ ...(d = (o = e.curStyle) == null ? void 0 : o.wrapper) == null ? void 0 : d.default })
  }, [
    (i(!0), s(_, null, ee(e.list, (a, n) => {
      var c, y, v, p, S, q, f, g, $, C, L, b, k, V, M, h, B, T, w, O, z, E, F, A, D, I, N, Q, j, G, H, J, K, P, R, U, W, X, Y, Z;
      return i(), s("div", {
        key: n,
        class: "container",
        style: t({ ...e.active === a.date ? (y = (c = e.curStyle) == null ? void 0 : c.card) == null ? void 0 : y.active : "" }),
        onClick: (ne) => e.clickItem(a, n)
      }, [
        l("div", te, [
          l("div", {
            class: "date",
            style: t({ ...(p = (v = e.curStyle) == null ? void 0 : v.date) == null ? void 0 : p.default, ...e.active === a.date ? (q = (S = e.curStyle) == null ? void 0 : S.date) == null ? void 0 : q.active : "" })
          }, r(a.date), 5),
          l("div", {
            class: "aqi-value",
            style: t({ ...(g = (f = e.curStyle) == null ? void 0 : f.aqiValue) == null ? void 0 : g.default, ...e.active === a.date ? (C = ($ = e.curStyle) == null ? void 0 : $.aqiValue) == null ? void 0 : C.active : "" })
          }, [
            l("span", null, r(a.aqiMin), 1),
            l("span", {
              style: t({ ...(b = (L = e.curStyle) == null ? void 0 : L.aqiValueTilde) == null ? void 0 : b.default, ...e.active === a.date ? (V = (k = e.curStyle) == null ? void 0 : k.aqiValueTilde) == null ? void 0 : V.active : "" })
            }, "~", 4),
            l("span", null, r(a.aqiMax), 1)
          ], 4)
        ]),
        l("div", {
          class: "mediate",
          style: t((h = (M = e.curStyle) == null ? void 0 : M.mediate) == null ? void 0 : h.default)
        }, [
          l("div", {
            class: "text-wrapper",
            style: t({
              color: `rgb(${a.aqiColor[0]})`,
              background: `rgba(${a.aqiColor[0]}, ${e.aqiLevelBgOpacity})`,
              ...(T = (B = e.curStyle) == null ? void 0 : B.aqiLevelMin) == null ? void 0 : T.default,
              ...e.active === a.date ? (O = (w = e.curStyle) == null ? void 0 : w.aqiLevelMin) == null ? void 0 : O.active : ""
            })
          }, r(a.aqiLevel[0]), 5),
          l("div", {
            class: "status-strip",
            style: t({ background: `linear-gradient(rgb(${a.aqiColor[0]}) 0%, rgb(${a.aqiColor[1]}) 100%)`, ...(E = (z = e.curStyle) == null ? void 0 : z.aqiStatusBar) == null ? void 0 : E.default })
          }, null, 4),
          l("div", {
            class: "text-wrapper text-wrapper-right",
            style: t({
              color: `rgb(${a.aqiColor[1]})`,
              background: `rgba(${a.aqiColor[1]}, ${e.aqiLevelBgOpacity})`,
              ...(A = (F = e.curStyle) == null ? void 0 : F.aqiLevelMax) == null ? void 0 : A.default,
              ...e.active === a.date ? (I = (D = e.curStyle) == null ? void 0 : D.aqiLevelMax) == null ? void 0 : I.active : ""
            })
          }, r(a.aqiLevel[1]), 5)
        ], 4),
        l("div", {
          class: "line",
          style: t({ ...(Q = (N = e.curStyle) == null ? void 0 : N.line) == null ? void 0 : Q.default, ...e.active === a.date ? (G = (j = e.curStyle) == null ? void 0 : j.line) == null ? void 0 : G.active : "" })
        }, null, 4),
        l("div", {
          class: "right",
          style: t((J = (H = e.curStyle) == null ? void 0 : H.right) == null ? void 0 : J.default)
        }, [
          l("div", {
            class: "text",
            style: t({ ...(P = (K = e.curStyle) == null ? void 0 : K.pollutantText) == null ? void 0 : P.default, ...e.active === a.date ? (U = (R = e.curStyle) == null ? void 0 : R.pollutantText) == null ? void 0 : U.active : "" })
          }, r(e.pollutantText), 5),
          l("div", {
            style: t({ ...(X = (W = e.curStyle) == null ? void 0 : W.pollutantValue) == null ? void 0 : X.default, ...e.active === a.date ? (Z = (Y = e.curStyle) == null ? void 0 : Y.pollutantValue) == null ? void 0 : Z.active : "" }),
            title: a.pollutantValue
          }, r(a.pollutantValue), 13, re)
        ], 4)
      ], 12, le);
    }), 128)),
    u[0] || (u[0] = l("div", { class: "style-container" }, null, -1))
  ], 6);
}
const Se = /* @__PURE__ */ ae(m, [["render", ie]]);
export {
  Se as default
};
