// @ths/design/es/components/list-card-air-quality-forecast/src/index.vue2.js
import { defineComponent as C, reactive as g, computed as b, watch as S, onMounted as x, toRefs as A } from "vue";
import { AQI_LEVEL_COLOR as u, AQI_LEVEL_NAME as m } from "../../../utils/constant.js";
import { getDomRandomClass as I, setStyle as v, colorRgb as s, getAQILevel as y } from "../../../utils/index.js";
const w = C({
  name: "TListCardAirQualityForecast",
  props: {
    // иҮӘе®ҡд№үз»„д»¶ж ·ејҸ
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        /* ж•°жҚ®йЎ№ */
        card: {
          default: {},
          active: {},
          hover: {}
        },
        /* ж—Ҙжңҹ */
        date: {
          default: {},
          active: {}
        },
        /* AQIеҖј */
        aqiValue: {
          default: {},
          active: {}
        },
        /* AQIеҖјжіўжөӘеҸ· */
        aqiValueTilde: {
          default: {},
          active: {}
        },
        /* дёӯй—ҙе®№еҷЁ */
        mediate: {
          default: {}
        },
        /* AQIзә§еҲ«жңҖе°ҸеҖј */
        aqiLevelMin: {
          default: {
            background: "transparent"
          },
          active: {}
        },
        /* AQIзҠ¶жҖҒж Ҹ */
        aqiStatusBar: {
          default: {}
        },
        /* AQIжңҖеӨ§еҖј */
        aqiLevelMax: {
          default: {
            background: "transparent"
          },
          active: {}
        },
        /* иҫ№зәҝ */
        line: {
          default: {},
          active: {}
        },
        /* еҸідҫ§е®№еҷЁ */
        right: {
          default: {}
        },
        /* жұЎжҹ“зү©ж–Үжң¬ */
        pollutantText: {
          default: {},
          active: {}
        },
        /* жұЎжҹ“зү©еҖј */
        pollutantValue: {
          default: {},
          active: {}
        }
      })
    },
    // еҚЎзүҮз»„ж•°жҚ®
    data: {
      type: Array,
      default: () => []
    },
    // йҖүдёӯйЎ№пјҲж—Ҙжңҹпјү
    active: {
      type: String,
      default: ""
    },
    // йҰ–иҰҒжұЎжҹ“зү©ж–Үжң¬
    pollutantText: {
      type: String,
      default: "йҰ–иҰҒжұЎжҹ“зү©"
    },
    // AQIзә§еҲ«ж–Үжң¬иғҢжҷҜиүІзҡ„йҖҸжҳҺеәҰ
    aqiLevelBgOpacity: {
      type: Number,
      default: 0.2
    }
  },
  emits: ["item-click"],
  setup(n, { emit: q }) {
    const r = g({
      list: []
    }), l = b(() => n.cStyle), i = I("list-card-air-quality-forecast"), L = (e, t) => {
      q("item-click", e, t);
    }, M = (e) => {
      const t = [], a = [], o = typeof e.aqiMax == "number", c = typeof e.aqiMin == "number";
      if (!o && !c)
        t.push("-"), a.push(s(u.level7));
      else {
        const f = y(e.aqiMax), p = y(e.aqiMin);
        t.push(m[p]), a.push(s(u[p])), t.push(m[f]), a.push(s(u[f]));
      }
      return {
        aqiLevel: t,
        aqiColor: a
      };
    }, d = (e) => {
      r.list = [], e.forEach((t) => {
        const { aqiLevel: a, aqiColor: o } = M(t), c = {
          ...t,
          aqiLevel: a,
          aqiColor: o
        };
        r.list.push(c);
      });
    }, h = () => {
      var t, a;
      let e = "";
      return (t = l.value) != null && t.wrapper && (e += v(l.value.wrapper, `.${i}`)), (a = l.value) != null && a.card && (e += v(l.value.card, `.${i} .container`)), e;
    };
    return d(n.data), S(() => n.data, (e) => {
      d(e);
    }, { deep: !0 }), x(() => {
      const e = document.querySelector(`.${i} .style-container`);
      e && (e.innerHTML = `<style>${h()}</style>`);
    }), {
      ...A(r),
      curStyle: l,
      domOnlyClass: i,
      clickItem: L
    };
  }
});
export {
  w as default
};
