// @ths/design/es/components/chart-dashboard-aqi/src/index.vue2.js
import { defineComponent as T, computed as D, ref as V, reactive as M, onMounted as U, watch as _, nextTick as q, onUnmounted as j, toRefs as I } from "vue";
import * as H from "echarts";
import n from "lodash/merge";
import f from "lodash/cloneDeep";
import { getPollutionHtml as P, getDomRandomClass as W, getAQILevel as $, setStyle as s } from "../../../utils/index.js";
import { DEFAULT_OPTIONS_COMMON as E, AQI_COLOR as C, POLLUTION_HTML as b } from "../../../utils/constant.js";
const x = n(
  f(E),
  {
    legend: {
      show: !1
    },
    tooltip: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    }
  }
), K = {
  aqiName: "AQI",
  // aqi名称
  aqiValue: void 0,
  // aqi值
  textMes: "首要污染物:",
  // 描述文字
  pollutantName: ""
  // 描述文字后面跟着的内容
}, J = T({
  name: "TChartDashboardAqi",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        },
        // 图标
        icon: {
          default: {}
        },
        // 名称
        itemName: {
          default: {}
        },
        // 值
        itemValue: {
          default: {}
        },
        // 下方描述
        primaryWrapper: {
          default: {}
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => x,
      required: !1
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名设置及样式设置
    seriesSetting: {
      type: Object,
      default: () => ({
        valueKey: "",
        // 污染物数值别名
        nameKey: "",
        // 污染物名称别名
        colorKey: "",
        // 污染物图标色值别名
        unitKey: "",
        // 污染物单位别名
        imgUrl: "",
        // 图标url
        style: {}
        // 仪表盘的样式设置
      })
    },
    // 表盘数据
    gaugeData: {
      type: Object,
      default: () => f(K)
    },
    // 等级颜色
    levelColors: {
      type: Object,
      default: () => f(C)
    }
  },
  setup(e) {
    const a = D(() => e.cStyle), u = D(() => n(f(K), e.gaugeData)), N = D(() => n(f(C), e.levelColors)), g = V(null);
    let o = null;
    const l = W("chart-dashboard-aqi"), c = M({
      // 左边污染物数据
      leftPollutantData: [],
      // 右边污染物数据
      rightPollutantData: []
    }), h = () => {
      if (e.data.length && e.seriesSetting) {
        const {
          valueKey: t,
          nameKey: d,
          colorKey: y,
          unitKey: v
        } = e.seriesSetting, m = e.data.map((i) => ({
          ...i,
          name: P(b, i[d]) || "-",
          value: i[t] || "-",
          color: i[y],
          unit: i[v] || ""
        }));
        c.leftPollutantData = m.slice(0, 3), c.rightPollutantData = m.slice(3, 6);
      } else
        c.leftPollutantData = [], c.rightPollutantData = [];
    }, O = () => {
      var t, d, y, v, m, i;
      if (g.value) {
        o || (o = H.init(g.value));
        const L = (t = e == null ? void 0 : e.options) == null ? void 0 : t.series;
        let S;
        L ? S = L.map((r) => n(
          {},
          r
        )) : e.data && e.seriesSetting && (S = [
          n(
            {
              type: "gauge",
              radius: "100%",
              startAngle: 235,
              endAngle: -125,
              axisLine: {
                show: !0,
                roundCap: !0,
                lineStyle: {
                  width: 5,
                  // 获取对应的等级色 如果传的等级颜色为空 则显示默认的值
                  color: [[1, ((y = N.value) == null ? void 0 : y[$((d = u.value) == null ? void 0 : d.aqiValue)]) || C[$((v = u.value) == null ? void 0 : v.aqiValue)]]]
                }
              },
              pointer: {
                show: !1
              },
              axisTick: {
                length: 3,
                distance: 5,
                lineStyle: {
                  width: 2,
                  color: "#FFF"
                }
              },
              splitLine: {
                show: !1
              },
              axisLabel: {
                show: !1
              },
              detail: {
                fontSize: 18,
                offsetCenter: [0, 0],
                valueAnimation: !0,
                formatter(r) {
                  var A;
                  return `{name|${((A = u.value) == null ? void 0 : A.aqiName) || "-"}}
{value|${r >= 0 ? r : "-"}}`;
                },
                // 富文本的样式调整
                rich: {
                  value: {
                    color: "#FFF",
                    fontSize: 32,
                    fontWeight: "bold",
                    fontFamily: "Kaiti"
                  },
                  name: {
                    fontSize: 14,
                    color: "#FFF"
                  }
                },
                color: "#FFF"
              },
              data: [
                {
                  value: (m = u.value) == null ? void 0 : m.aqiValue
                }
              ]
            },
            (i = e.seriesSetting) == null ? void 0 : i.style
          )
        ]);
        let w = n(
          {
            series: S
          },
          x
        );
        w = n(w, e.options), o.clear();
        try {
          o.setOption(w);
        } catch (r) {
          console.log(r);
        }
      }
    }, z = () => {
      let t = "";
      return a.value.wrapper && (t += s(a.value.wrapper, `.${l}`)), a.value.icon && (t += s(a.value.icon, `.${l} .item__icon`)), a.value.itemValue && (t += s(a.value.itemValue, `.${l} .item__mes_value`)), a.value.itemUnit && (t += s(a.value.itemUnit, `.${l} .item__mes_unit`)), a.value.itemName && (t += s(a.value.itemName, `.${l} .item__mes_name`)), a.value.primaryWrapper && (t += s(a.value.primaryWrapper, `.${l} .content-center .primary-pollutant`)), t;
    }, p = () => {
      const t = document.querySelector(`.${l} .style-container`);
      t && (t.innerHTML = `<style>${z()}</style>`);
    }, F = () => {
      q(() => {
        o.resize();
      });
    };
    return U(() => {
      p(), h(), O(), window.addEventListener("resize", F);
    }), _(() => e.cStyle, () => {
      p(), q(() => {
        o.resize();
      });
    }), _(
      () => [
        e.seriesSetting,
        e.options,
        e.gaugeData
      ],
      () => {
        h(), O();
      },
      {
        deep: !0
      }
    ), _(
      () => [
        e.data
      ],
      () => {
        h();
      },
      {
        deep: !0
      }
    ), j(() => {
      window.removeEventListener("resize", F);
    }), {
      ...I(c),
      getPollutionHtml: P,
      POLLUTION_HTML: b,
      curDescriptionData: u,
      chart: g,
      domOnlyClass: l
    };
  }
});
export {
  J as default
};
