// @ths/design/es/components/collapse/src/index.vue2.js
import { defineComponent as C, computed as y, reactive as h, onMounted as w, watch as p, ref as L, toRefs as _ } from "vue";
import { ComponentLoader as u } from "../../../utils/loader.js";
import { getDomRandomClass as $, setStyle as a, getCssPropertyString as S } from "../../../utils/index.js";
import B from "./header.vue.js";
import g from "./content.vue.js";
const N = C({
  name: "TCollapse",
  components: {
    CustomLHeader: B,
    CustomLContent: g
  },
  props: {
    // 手风琴模式
    accordion: {
      type: Boolean,
      default: !1
    },
    // 组件列表
    componentList: {
      type: Array,
      default: () => [
        {
          header: "CustomLHeader",
          content: "CustomLContent"
        }
      ]
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 单项容器
        item: {
          default: {}
        },
        // 头部容器
        header: {
          default: {}
        },
        // 内容容器
        content: {
          default: {}
        },
        // 操作按钮
        actionButton: {
          default: {},
          hover: {},
          active: {}
        }
      })
    }
  },
  setup(n, { emit: H }) {
    const t = y(() => n.cStyle), l = $("collapse"), s = h({
      // 已经展开的折叠面板索引
      activeNames: []
    }), d = () => {
      var o, c, r;
      let e = "";
      return t.value.wrapper && (e += a(t.value.wrapper, `.${l}`)), t.value.item && (e += a(t.value.item, `.${l} .el-collapse .el-collapse-item`)), t.value.header && (e += a(t.value.header, `.${l} .el-collapse .el-collapse-item>div[role=tab]`)), t.value.content && (e += a(t.value.content, `.${l} .el-collapse .el-collapse-item .el-collapse-item__wrap`)), t.value.actionButton && (e += a(t.value.actionButton, `.${l} .el-collapse .el-collapse-item .el-collapse-item__arrow.el-icon-arrow-right`)), (o = t.value.actionButton) != null && o.hover && (e += a(t.value.actionButton, `.${l} .el-collapse .el-collapse-item .el-collapse-item__arrow.el-icon-arrow-right:hover`)), (c = t.value.actionButton) != null && c.active && (e += `.${l} .el-collapse .el-collapse-item .el-collapse-item__arrow.el-icon-arrow-right.is-active {
          ${S((r = t.value.actionButton) == null ? void 0 : r.active)}
        }`), e;
    }, m = () => {
      const e = document.querySelector(`.${l}>.style-container`);
      e && (e.innerHTML = `<style>${d()}</style>`);
    }, f = () => {
      s.activeNames = n.componentList.map((e, o) => o);
    };
    w(() => {
      f(), m();
    }), p(() => n.cStyle, () => {
      m();
    }, { deep: !0 });
    const i = L(0);
    p(() => n.componentList, (e) => {
      i.value = 0, e && e.forEach((o) => {
        var c, r;
        o.header && u.loadComponent((c = o.header) == null ? void 0 : c.replace("Component", ""), () => {
          i.value += 1;
        }), o.content && u.loadComponent((r = o.content) == null ? void 0 : r.replace("Component", ""), () => {
          i.value += 1;
        });
      });
    }, { immediate: !0 });
    const v = () => n.componentList.reduce((e, o) => e + (o.header ? 1 : 0) + (o.content ? 1 : 0), 0);
    return {
      ..._(s),
      curStyle: t,
      domOnlyClass: l,
      numberOfLoadedComponents: i,
      getTotalComponentsToLoad: v
    };
  }
});
export {
  N as default
};
