// @ths/design/es/components/collapse/src/index.vue.js
import u from "./index.vue2.js";
import { resolveComponent as m, openBlock as t, createElementBlock as d, normalizeClass as f, createBlock as l, withCtx as s, Fragment as C, renderList as v, createElementVNode as a, withModifiers as k, resolveDynamicComponent as p, createCommentVNode as y } from "vue";
import "./index.vue3.js";
import L from "../../../_virtual/_plugin-vue_export-helper.js";
const V = { class: "t-collapse-content" };
function _(e, o, N, $, B, O) {
  const i = m("el-collapse-item"), c = m("el-collapse");
  return t(), d("div", {
    class: f(["t-collapse", e.domOnlyClass])
  }, [
    e.numberOfLoadedComponents && e.numberOfLoadedComponents === e.getTotalComponentsToLoad() ? (t(), l(c, {
      key: 0,
      modelValue: e.activeNames,
      "onUpdate:modelValue": o[1] || (o[1] = (n) => e.activeNames = n),
      accordion: e.accordion
    }, {
      default: s(() => [
        (t(!0), d(C, null, v(e.componentList, (n, r) => (t(), l(i, {
          key: r,
          name: r
        }, {
          title: s(() => [
            a("div", {
              class: "t-collapse-header",
              onClick: o[0] || (o[0] = k(() => {
              }, ["stop"]))
            }, [
              (t(), l(p(n.header)))
            ])
          ]),
          default: s(() => [
            a("div", V, [
              (t(), l(p(n.content)))
            ])
          ]),
          _: 2
        }, 1032, ["name"]))), 128))
      ]),
      _: 1
    }, 8, ["modelValue", "accordion"])) : y("", !0),
    o[2] || (o[2] = a("div", { class: "style-container" }, null, -1))
  ], 2);
}
const T = /* @__PURE__ */ L(u, [["render", _]]);
export {
  T as default
};
