// @ths/design/es/components/chart-wind-direction/src/index.vue2.js
import { defineComponent as te, ref as ae, computed as Q, reactive as ie, onMounted as ne, onUnmounted as le, watch as z, nextTick as V, toRefs as se } from "vue";
import * as re from "echarts";
import u from "lodash/merge";
import I from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as oe, DEFAULT_OPTIONS_AXIS as ce } from "../../../utils/constant.js";
import { getDomRandomClass as de, setStyle as L } from "../../../utils/index.js";
const ye = u(I(oe), I(ce)), h = u(
  I(ye),
  {
    grid: {
      top: "13%",
      bottom: "10%",
      left: "6%",
      right: "4%"
    },
    xAxis: {
      boundaryGap: !0
    },
    legend: {
      itemStyle: {
        borderWidth: 0
      }
    }
  }
), me = {
  north: 0,
  northeast: -45,
  east: -90,
  sortheast: -135,
  sorth: -180,
  sorthwest: -225,
  west: -270,
  northwest: -315
}, Oe = te({
  name: "TChartWindDirection",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 暂无数据容器
        emptyContainer: {
          default: {}
        },
        // 暂无数据图片
        emptyImg: {
          default: {}
        },
        // 暂无数据文字
        emptyText: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => h,
      required: !1
    },
    // 设置主轴
    mainAxis: {
      type: String,
      default: "x"
    },
    // 传入数组数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 折线主题配置
    line: {
      type: Object,
      default: () => ({})
    },
    // 是否显示背景
    showItemBg: {
      type: Boolean,
      default: !1
    },
    // 背景条样式
    backgroundStyle: {
      type: Object,
      default: () => ({})
    }
  },
  // 方法名定义
  emits: ["com-click", "legend-select"],
  setup(e, { emit: $ }) {
    const S = de("chart-wind-direction"), b = ae(null);
    let f = null, w = null;
    const m = Q(() => e.cStyle), i = ie({
      // 图表数据
      chartData: [],
      chartBgData: [],
      windObj: {},
      // 主题配置
      lineOptions: I(e.line)
    }), B = () => {
      var t, o, c, d, r, l;
      (o = (t = e.seriesSetting) == null ? void 0 : t.data) != null && o.length && e.data ? (i.chartData = e.seriesSetting.data.map((y) => {
        const { name: O, dataKey: g, windKey: k } = y;
        return {
          name: O || "--",
          data: e.data.map((s) => s[g]),
          windData: e.data.map((s) => s[k])
        };
      }), i.chartBgData = i.chartData) : (i.chartData = [], i.chartBgData = ((c = e.options) == null ? void 0 : c.series) || i.chartData), (r = Object.keys((d = e.seriesSetting) == null ? void 0 : d.windData)) != null && r.length && e.data && Object.keys((l = e.seriesSetting) == null ? void 0 : l.windData).forEach((y) => {
        var g;
        const O = (g = e.seriesSetting) == null ? void 0 : g.windData[y];
        i.windObj[O] = me[y];
      });
    }, Y = () => {
      let t = "";
      return m.value.wrapper && (t += L(m.value.wrapper, `.${S}`)), m.value.emptyContainer && (t += L(m.value.emptyContainer, `.${S} .empty`)), m.value.emptyImg && (t += L(m.value.emptyImg, `.${S} .empty .empty-img`)), m.value.emptyText && (t += L(m.value.emptyText, `.${S} .empty .empty-text`)), t;
    }, _ = () => {
      const t = document.querySelector(`.${S} .style-container`);
      t && (t.innerHTML = `<style>${Y()}</style>`);
    }, T = () => {
      var l, y, O, g, k;
      const { color: t, series: o } = e.options;
      let c = [], d = [];
      o ? c = o.map((s, a) => {
        var D, v, A, C, x, F, j;
        const n = u(
          {
            type: "line",
            itemStyle: {
              color: t && t[a] || h.color[a] || "#FFF"
            },
            lineStyle: {
              color: t && t[a] || h.color[a] || "#FFF"
            },
            emphasis: {
              focus: "series"
            },
            smooth: !0
          },
          i == null ? void 0 : i.lineOptions,
          (A = (v = (D = e.seriesSetting) == null ? void 0 : D.data) == null ? void 0 : v[a]) == null ? void 0 : A.style
        );
        return (F = (x = (C = e.seriesSetting) == null ? void 0 : C.data) == null ? void 0 : x[a]) != null && F.isOpenLineColor || "color" in (n == null ? void 0 : n.lineStyle) && ((j = n == null ? void 0 : n.lineStyle) == null || delete j.color), u({}, n, s);
      }) : e.data && ((y = (l = e.seriesSetting) == null ? void 0 : l.data) != null && y.length) && (c = (O = e.seriesSetting) == null ? void 0 : O.data.map((s, a) => {
        var D, v, A, C, x, F, j, R, q, M, N, P, U, H, K, W, G, X;
        const n = u(
          {
            name: (D = i.chartData[a]) == null ? void 0 : D.name,
            type: "line",
            label: {
              show: !1
            },
            itemStyle: {
              color: t && t[a] || h.color[a] || "#FFF"
            },
            lineStyle: {
              color: t && t[a] || h.color[a] || "#FFF"
            },
            emphasis: {
              focus: "series"
            },
            smooth: !0,
            data: e.data.length ? (v = i.chartData[a]) == null ? void 0 : v.data : []
          },
          i == null ? void 0 : i.lineOptions,
          (x = (C = (A = e.seriesSetting) == null ? void 0 : A.data) == null ? void 0 : C[a]) == null ? void 0 : x.style,
          {
            symbol: (M = (q = (R = (j = (F = e.seriesSetting) == null ? void 0 : F.data) == null ? void 0 : j[a]) == null ? void 0 : R.style) == null ? void 0 : q.symbol) != null && M.includes("http") ? `image://${(H = (U = (P = (N = e.seriesSetting) == null ? void 0 : N.data) == null ? void 0 : P[a]) == null ? void 0 : U.style) == null ? void 0 : H.symbol}` : "emptyCircle",
            symbolRotate: (ue, ee) => {
              var J;
              return e.data.length ? i.windObj[(J = i.chartData[a].windData) == null ? void 0 : J[ee.dataIndex]] : 0;
            }
          }
        );
        return (G = (W = (K = e.seriesSetting) == null ? void 0 : K.data) == null ? void 0 : W[a]) != null && G.isOpenLineColor || "color" in (n == null ? void 0 : n.lineStyle) && ((X = n == null ? void 0 : n.lineStyle) == null || delete X.color), u({}, n, s);
      }), (g = e.seriesSetting) != null && g.xAxisLabel ? d = (k = e.data) != null && k.length ? e.data.map((s) => s[e.seriesSetting.xAxisLabel]) : [] : d = e.data.map((s, a) => a));
      let r;
      e.mainAxis === "x" ? r = u(
        {
          series: c,
          xAxis: {
            data: d
          }
        },
        h
      ) : e.mainAxis === "y" && (r = u(
        {
          series: c,
          yAxis: {
            data: d
          }
        },
        h
      )), r = u(r, e.options), e.mainAxis === "y" && (r.xAxis.type = "value", r.yAxis.type = "category"), f.clear();
      try {
        f.setOption(r);
      } catch (s) {
        console.log(s);
      }
    }, Z = Q(() => {
      var t, o, c, d;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((l) => {
            var y;
            return (l == null ? void 0 : l.data) && ((y = l == null ? void 0 : l.data) == null ? void 0 : y.length) !== 0;
          });
        if (typeof ((o = e.options) == null ? void 0 : o.series) == "object")
          return ((d = (c = e.options.series) == null ? void 0 : c.data) == null ? void 0 : d.length) === 0;
      }
      return !!e.data.length;
    }), E = () => {
      V(() => {
        f.resize();
      });
    };
    function p() {
      b.value && (w = new ResizeObserver((t) => {
        t.forEach((o) => {
          o.target === b.value && E();
        });
      }), w.observe(b.value));
    }
    return ne(() => {
      _(), b.value && (f = re.init(b.value), B(), T(), f.on("click", (t) => {
        $("com-click", t);
      }), f.on("legendselectchanged", (t) => {
        $("legend-select", t);
      }), window.addEventListener("resize", E), p());
    }), le(() => {
      window.removeEventListener("resize", E), w && (w.disconnect(), w = null);
    }), z(
      [() => e.line, () => e.backgroundStyle, () => e.mainAxis],
      () => {
        i.lineOptions = I(e.line), T();
      },
      {
        deep: !0
      }
    ), z(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting,
        () => e.showItemBg
      ],
      () => {
        B(), T();
      },
      {
        deep: !0
      }
    ), z(
      () => e.cStyle,
      () => {
        _(), V(() => {
          f.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      curStyle: m,
      ...se(i),
      chartId: b,
      checkEmpty: Z,
      domOnlyClass: S
    };
  }
});
export {
  Oe as default
};
