// @ths/design/es/components/cascader/src/index.vue2.js
import { defineComponent as O, reactive as W, computed as b, onMounted as k, watch as s, toRefs as T } from "vue";
import B from "lodash/merge";
import f from "lodash/cloneDeep";
import { filterAttrs as D, getDomRandomClass as g, setStyle as d, getCssPropertyString as p } from "../../../utils/index.js";
const S = {
  label: "label",
  value: "value",
  disabled: "disabled"
}, q = O({
  name: "TCascader",
  props: {
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 高亮导航项
    activeValue: {
      type: [String, Number, Boolean, Object, Array],
      default: ""
    },
    // 是否禁用
    disabled: {
      type: Boolean,
      default: !1
    },
    // 空数据文本
    noDataText: {
      type: String,
      default: "无数据"
    },
    // 占位文本
    placeholder: {
      type: String,
      default: "请选择"
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => f(S),
      require: !1
    },
    // 分割符
    separator: {
      type: String,
      default: "/"
    },
    // 选择任意一项
    checkStrictly: {
      type: Boolean,
      default: !0
    },
    // 展开方式
    expandTrigger: {
      type: String,
      default: "click"
    },
    // 仅显示最后一级
    showAllLevels: {
      type: Boolean,
      default: !1
    },
    // 可清空
    clearable: {
      type: Boolean,
      default: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {},
          focus: {},
          hover: {}
        },
        arrow: {
          default: {}
        },
        clearIcon: {
          default: {},
          focus: {},
          hover: {}
        },
        placeholder: {
          default: {}
        },
        tag: {
          default: {}
        },
        tagClose: {
          default: {}
        },
        disabledWrapper: {
          default: {}
        },
        disabledArrow: {
          default: {}
        },
        disabledPlaceholder: {
          default: {}
        },
        dropdownWrapper: {
          default: {}
        },
        dropdownItem: {
          default: {},
          active: {},
          hover: {}
        },
        dropdownArrow: {
          default: {}
        },
        dropdownRadio: {
          default: {},
          active: {}
        },
        dropdownRadioCenter: {
          active: {}
        },
        dropdownMenu: {
          default: {}
        }
      })
    }
  },
  emits: ["change"],
  setup(r, { emit: A }) {
    const c = W({
      curOptions: r.data,
      curActiveValue: r.activeValue
    }), C = b(() => ({ expandTrigger: r.expandTrigger, checkStrictly: r.checkStrictly })), l = b(() => r.cStyle), a = g("cascader"), o = g("cascader-popper"), I = (e) => {
      let n = "", i = [], t = "";
      c.curActiveValue = e, n = e, i = [e], t = (e == null ? void 0 : e[(e == null ? void 0 : e.length) - 1]) || "", A("change", n, i, t);
    }, R = () => {
      let e = "";
      return l.value.wrapper && (e += d(l.value.wrapper, `.t-cascader.${a} .el-cascader .el-input .el-input__inner`)), l.value.arrow && (e += d(l.value.arrow, `.t-cascader.${a} .el-cascader .el-input .el-input__icon`)), l.value.dropdownWrapper && (e += d(l.value.dropdownWrapper, `.${a} .${o}.el-popper[role=tooltip]`), e += d(l.value.dropdownWrapper, `.${a} .${o}.el-popper[role=tooltip] .el-cascader-panel`), e += d(l.value.dropdownWrapper, `.${a} .${o}.el-popper[role=tooltip] .el-cascader-panel .el-cascader-menu__wrap.el-scrollbar__wrap`)), l.value.dropdownMenu && (e += d(l.value.dropdownMenu, `.${a} .${o}.el-popper[role=tooltip] .el-cascader-panel .el-cascader-menu`)), l.value.dropdownItem && (e += d(l.value.dropdownItem, `.${a} .${o}.el-popper[role=tooltip] .el-cascader-node`)), l.value.dropdownRadio && (e += d(l.value.dropdownRadio, `.${a} .${o}.el-popper[role=tooltip] .el-radio .el-radio__input .el-radio__inner`)), l.value.placeholder && (e += d(l.value.placeholder, `.${a} .el-input__inner::placeholder`)), l.value.disabledWrapper && (e += d(l.value.disabledWrapper, `.${a} .is-disabled .el-input__inner`)), l.value.disabledPlaceholder && (e += d(l.value.disabledPlaceholder, `.${a} .is-disabled .el-input__inner::placeholder`)), l.value.disabledArrow && (e += d(l.value.disabledArrow, `.${a} .el-input.is-disabled .el-input__icon`)), e;
    }, V = () => {
      var n, i, t, w, $, m, _, y, h;
      let e = "";
      return (n = l.value.wrapper) != null && n.focus && (e += `
          .${a} .el-input.is-focus .el-input__inner {
            ${p(l.value.wrapper.focus)}
          }
        `), (i = l.value.clearIcon) != null && i.focus && (e += `
          .${a} .el-input.is-focus .el-input__suffix .el-input_icon.el-icon-circle-close {
            ${p(l.value.clearIcon.focus)}
          }
        `), (t = l.value.dropdownArrow) != null && t.default && (e += `
          .${a} .${o}.el-popper[role=tooltip]>.el-popper__arrow::before {
            ${p((w = l.value.dropdownArrow) == null ? void 0 : w.default)}
          }
        `), ($ = l.value.dropdownRadio) != null && $.active && (e += `
          .${a} .${o}.el-popper[role=tooltip] .el-radio .el-radio__input.is-checked .el-radio__inner {
            ${p((m = l.value.dropdownRadio) == null ? void 0 : m.active)}
          }
        `), (_ = l.value.dropdownRadioCenter) != null && _.active && (e += `
          .${a} .${o}.el-popper[role=tooltip] .el-radio .el-radio__input.is-checked .el-radio__inner::after {
            ${p((y = l.value.dropdownRadioCenter) == null ? void 0 : y.active)}
          }
        `), (h = l.value.dropdownItem) != null && h.active && (e += `
          .${a} .${o}.el-popper[role=tooltip] .el-cascader-node.is-active {
            ${p(l.value.dropdownItem.active)}
          }
        `, e += `
          .${a} .${o}.el-popper[role=tooltip] .el-cascader-node.in-active-path {
            ${p(l.value.dropdownItem.active)}
          }
        `, e += `
          .${a} .${o}.el-popper[role=tooltip] .el-cascader-node:not(.is-disabled):focus {
            ${p(l.value.dropdownItem.active)}
          }
        `), e;
    }, v = () => {
      const e = document.querySelector(`.${a} .style-container`);
      e && (e.innerHTML = `<style>${R()}${V()}</style>`);
    }, u = (e, n) => {
      const i = B(S, n);
      Array.isArray(e) && e.forEach((t) => {
        t.disabled = t[i.disabled], t.label = t[i.label], t.value = t[i.value], t != null && t.children && t.children.length && u(t.children, n);
      });
    };
    return k(() => {
      v();
      const e = f(r.data);
      u(e, r.alias);
    }), s(() => [r.data, r.alias], () => {
      c.curOptions = f(r.data), u(c.curOptions, r.alias);
    }, {
      deep: !0,
      immediate: !0
    }), s(() => r.activeValue, (e) => {
      c.curActiveValue = e;
    }), s(() => r.cStyle, () => {
      v();
    }, { deep: !0 }), {
      ...T(c),
      curStyle: l,
      domOnlyClass: a,
      popperOnlyClass: o,
      handleSelectChange: I,
      filterAttrs: D,
      customProps: C
    };
  }
});
export {
  q as default
};
