// @ths/design/es/components/cascader/src/index.vue.js
import u from "./index.vue2.js";
import { resolveComponent as c, openBlock as f, createElementBlock as g, normalizeClass as y, normalizeStyle as h, createVNode as v, mergeProps as C, createElementVNode as b } from "vue";
import "./index.vue3.js";
import w from "../../../_virtual/_plugin-vue_export-helper.js";
function V(e, a, S, $, A, B) {
  var r, l, o, p, s, t, n, d;
  const i = c("el-cascader");
  return f(), g("div", {
    class: y(["t-cascader", e.domOnlyClass]),
    style: h({
      marginTop: (l = (r = e.curStyle.wrapper) == null ? void 0 : r.default) == null ? void 0 : l.marginTop,
      marginLeft: (p = (o = e.curStyle.wrapper) == null ? void 0 : o.default) == null ? void 0 : p.marginLeft,
      marginRight: (t = (s = e.curStyle.wrapper) == null ? void 0 : s.default) == null ? void 0 : t.marginRight,
      marginBottom: (d = (n = e.curStyle.wrapper) == null ? void 0 : n.default) == null ? void 0 : d.marginBottom
    })
  }, [
    v(i, C({
      modelValue: e.curActiveValue,
      "onUpdate:modelValue": a[0] || (a[0] = (m) => e.curActiveValue = m)
    }, e.filterAttrs(e.$attrs), {
      props: e.customProps,
      clearable: e.clearable,
      "popper-class": e.popperOnlyClass,
      "show-all-levels": !e.showAllLevels,
      disabled: e.disabled,
      "no-data-text": e.noDataText,
      placeholder: e.placeholder,
      separator: e.separator,
      options: e.curOptions,
      teleported: !1,
      "popper-append-to-body": !1,
      onChange: e.handleSelectChange
    }), null, 16, ["modelValue", "props", "clearable", "popper-class", "show-all-levels", "disabled", "no-data-text", "placeholder", "separator", "options", "onChange"]),
    a[1] || (a[1] = b("div", { class: "style-container" }, null, -1))
  ], 6);
}
const z = /* @__PURE__ */ w(u, [["render", V]]);
export {
  z as default
};
