// @ths/design/es/components/chart-dashboard-aqi-label/src/index.vue2.js
import { defineComponent as P, computed as y, ref as j, reactive as M, onMounted as N, onUnmounted as E, watch as p, nextTick as S, toRefs as H } from "vue";
import * as K from "echarts";
import s from "lodash/merge";
import n from "lodash/cloneDeep";
import { getDomRandomClass as R, getPollutionHtml as T, getAQILevel as r, setStyle as v } from "../../../utils/index.js";
import { DEFAULT_OPTIONS_COMMON as F, AQI_COLOR as g, AQI_TYPE as C, POLLUTION_HTML as U, POLLUTION_CHARTS as k } from "../../../utils/constant.js";
const A = s(
  n(F),
  {
    legend: {
      show: !1
    },
    tooltip: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    }
  }
), $ = {
  value: void 0,
  type: "",
  descTitle: "首要污染物",
  descValue: "PM10"
}, x = {
  value: "value",
  name: "name",
  color: "color"
}, z = "level7", B = P({
  name: "TChartDashboardAqiLabel",
  props: {
    // echarts原生配置
    options: {
      type: Object,
      default: () => A
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        },
        // 图表容器
        chartWrapper: {
          default: {}
        },
        // 下方内容容器
        contentWrapper: {
          default: {}
        },
        // 下方内容单个容器
        contentItem: {
          default: {}
        },
        // 值
        value: {
          default: {}
        },
        // 名称
        name: {
          default: {}
        }
      })
    },
    // 列表数据
    listData: {
      type: Array,
      default: () => []
    },
    // 表盘数据
    gaugeData: {
      type: Object,
      default: () => n($)
    },
    // 数据别名设置及样式设置
    seriesSetting: {
      type: Object,
      default: () => ({
        style: {},
        dataKey: n(x)
      })
    },
    // 等级颜色
    levelColors: {
      type: Object,
      default: () => n(g)
    },
    // 等级类型描述
    levelTypes: {
      type: Object,
      default: () => n(C)
    }
  },
  setup(t) {
    const l = y(() => t.cStyle), a = y(() => s(n($), t.gaugeData)), w = y(() => s(n(g), t.levelColors)), I = y(() => s(n(C), t.levelTypes)), i = R("chart-dashboard-aqi-label"), h = j(null);
    let u = null;
    const D = M({
      // 列表数据
      listData: []
    }), O = () => {
      var c;
      const e = s(n(x), (c = t.seriesSetting) == null ? void 0 : c.dataKey);
      if (t.listData && t.listData.length) {
        const { value: f, name: m, color: d } = e;
        D.listData = t.listData.map((o) => ({
          value: o[f] || "-",
          name: T(U, o[m]) || "-",
          color: o[d] || "#FFF"
        }));
      } else
        D.listData = [];
    }, b = () => {
      var e, c, f;
      if (h.value) {
        u || (u = K.init(h.value));
        let m;
        t.listData && t.gaugeData && (m = [
          s(
            {
              type: "gauge",
              startAngle: 180,
              endAngle: 0,
              min: 0,
              max: 500,
              center: ["50%", "94%"],
              radius: "160%",
              splitNumber: 12,
              itemStyle: {
                color: "#fff"
              },
              progress: {
                show: !0,
                roundCap: !0,
                width: 10,
                itemStyle: {
                  // 获取值颜色 如果传的等级颜色为空 则显示默认的值
                  color: ((e = w.value) == null ? void 0 : e[r(a.value.value)]) || g[r(a.value.value)]
                }
              },
              pointer: {
                show: !1
              },
              axisLine: {
                roundCap: !0,
                lineStyle: {
                  width: 10,
                  color: [[1, "rgba(0, 0, 0, .1)"]]
                }
              },
              axisTick: {
                splitNumber: 4,
                distance: 20,
                length: 8,
                lineStyle: {
                  width: 1,
                  color: "rgba(0, 0, 0, 0.2)"
                }
              },
              splitLine: {
                show: !1
              },
              axisLabel: {
                show: !1
              },
              title: {
                offsetCenter: [-5, -45],
                color: "#333",
                fontSize: 50,
                fontWeight: "bold",
                rich: {
                  a: {
                    fontWeight: "normal",
                    fontSize: 16,
                    color: "#5E4D4D",
                    padding: [20, 0, 10, 20]
                  }
                }
              },
              detail: {
                width: 50,
                height: 32,
                formatter() {
                  var o;
                  return ((o = I.value) == null ? void 0 : o[r(a.value.value)]) || C[r(a.value.value)];
                },
                color: "#fff",
                fontSize: 20,
                fontWeight: "normal",
                offsetCenter: [72, -95],
                // 背景色获取 如果传的等级颜色为空 则显示默认的值
                backgroundColor: ((c = w.value) == null ? void 0 : c[r(a.value.value)]) || g[r(a.value.value)],
                borderRadius: [10, 10, 10, 0],
                padding: [4, 12]
              },
              data: [
                {
                  value: r(a.value.value) !== z ? a.value.value : "-",
                  // 值、名称
                  // eslint-disable-next-line max-len
                  name: `${r(a.value.value) !== z ? a.value.value : "-"}
{a|${a.value.descTitle || "-"}:${T(k, a.value.descValue) || "-"}}`
                }
              ]
            },
            (f = t.seriesSetting) == null ? void 0 : f.style
          )
        ]);
        let d = s(
          {
            series: m
          },
          A
        );
        d = s(d, t.options), u.clear();
        try {
          u.setOption(d);
        } catch (o) {
          console.log(o);
        }
      }
    }, L = () => {
      S(() => {
        u.resize();
      });
    }, W = () => {
      let e = "";
      return l.value.chartWrapper && (e += v(l.value.chartWrapper, `.${i} .chart`)), l.value.contentWrapper && (e += v(l.value.contentWrapper, `.${i} .content`)), l.value.contentItem && (e += v(l.value.contentItem, `.${i} .content .content__item`)), l.value.value && (e += v(l.value.value, `.${i} .content .content__item__value`)), l.value.name && (e += v(l.value.name, `.${i} .content .content__item__name`)), e;
    }, _ = () => {
      const e = document.querySelector(`.${i} .style-container`);
      e && (e.innerHTML = `<style>${W()}</style>`);
    };
    return N(() => {
      O(), b(), _(), window.addEventListener("resize", L);
    }), E(() => {
      window.removeEventListener("resize", L);
    }), p(
      () => {
        var e;
        return [
          t.listData,
          (e = t.seriesSetting) == null ? void 0 : e.dataKey
        ];
      },
      () => {
        O();
      },
      {
        deep: !0
      }
    ), p(
      () => {
        var e;
        return [
          (e = t.seriesSetting) == null ? void 0 : e.style,
          t.gaugeData,
          t.options,
          t.levelColors
        ];
      },
      () => {
        b(), S(() => {
          u.resize();
        });
      },
      {
        deep: !0
      }
    ), p(
      () => [t.cStyle],
      () => {
        _(), S(() => {
          u.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      curStyle: l,
      ...H(D),
      chart: h,
      domOnlyClass: i
    };
  }
});
export {
  B as default
};
