// @ths/design/es/components/chart-dashboard/src/index.vue2.js
import { defineComponent as I, computed as w, ref as L, reactive as E, onMounted as $, onUnmounted as M, watch as h, nextTick as D } from "vue";
import * as A from "echarts";
import d from "lodash/merge";
import x from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as V } from "../../../utils/constant.js";
import { getDomRandomClass as _, setStyle as g } from "../../../utils/index.js";
const m = d(
  x(V),
  {
    animation: !0,
    tooltip: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    }
  }
), B = I({
  name: "TChartDashboard",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        }
      })
    },
    // 仪表盘数据
    data: {
      type: String,
      default: () => "0"
    },
    // 仪表盘文字数据
    name: {
      type: String,
      default: () => ""
    },
    // 仪表盘配置
    seriesStyle: {
      type: Object,
      default: () => {
      }
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => m,
      required: !1
    },
    // 仪表盘主题设置
    gauge: {
      type: Object,
      default: () => ({})
    },
    // 刻度值是否需要保留小数
    isFixed: {
      type: Boolean,
      default: !1
    },
    // 刻度值保留小数位数
    fixedNumber: {
      type: Number,
      default: 0
    }
  },
  // 事件定义
  emits: ["com-click"],
  setup(e, { emit: O }) {
    const u = _("chart-dashboard"), r = w(() => e.cStyle), T = w(() => e.seriesStyle), f = L(null);
    let i = null;
    const c = E({
      // 图表数据
      chartData: [],
      // 刻度值数据
      labelValueArray: []
    }), S = () => {
      if (e.data !== "") {
        const t = e.data, { name: a } = e;
        c.chartData = [
          {
            value: Number.isNaN(Number(t)) ? "0" : Number(t),
            name: a || "--"
          }
        ];
      }
    }, k = () => {
      let t = "";
      return r.value.wrapper && (t += g(r.value.wrapper, `.${u}`)), r.value.emptyContainer && (t += g(r.value.emptyContainer, `.${u} .empty`)), r.value.emptyImg && (t += g(r.value.emptyImg, `.${u} .empty .empty-img`)), r.value.emptyText && (t += g(r.value.emptyText, `.${u} .empty .empty-text`)), t;
    }, v = () => {
      const t = document.querySelector(`.${u} .style-container`);
      t && (t.innerHTML = `<style>${k()}</style>`);
    }, z = w(() => {
      var t, a, l, s;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((n) => {
            var o;
            return (n == null ? void 0 : n.data) && ((o = n == null ? void 0 : n.data) == null ? void 0 : o.length) !== 0;
          });
        if (typeof ((a = e.options) == null ? void 0 : a.series) == "object")
          return ((s = (l = e.options.series) == null ? void 0 : l.data) == null ? void 0 : s.length) === 0;
      }
      return !(typeof parseFloat(e.data) != "number" || Number.isNaN(parseFloat(e.data)));
    }), F = (t) => {
      c.labelValueArray.push(t);
      const a = c.labelValueArray.findIndex((s) => s === t);
      let l;
      return T.value.tickIntervalShow ? l = a % 2 === 0 : l = !0, l;
    }, C = (t) => ({
      type: "gauge",
      radius: "85%",
      startAngle: 225,
      endAngle: -45,
      splitNumber: 10,
      min: 0,
      max: 100,
      progress: {
        show: !1,
        roundCap: !0,
        width: 25,
        itemStyle: {
          color: {
            type: "linear",
            x: 0,
            y: 1,
            x2: 0,
            y2: 0
          }
        }
      },
      axisLine: {
        roundCap: !1,
        lineStyle: {
          width: 25,
          color: [
            [0.3, t && t[0] || m.color[0]],
            [0.7, t && t[1] || m.color[1]],
            [1, t && t[2] || m.color[2]]
          ]
        }
      },
      pointer: {
        show: !0,
        itemStyle: {
          color: "#005fff"
        },
        icon: "",
        offsetCenter: [0, 0],
        length: "80%",
        width: 8
      },
      axisTick: {
        show: !0,
        distance: -25,
        length: 8,
        lineStyle: {
          color: "#fff",
          width: 2
        }
      },
      splitLine: {
        show: !0,
        distance: -25,
        length: 25,
        lineStyle: {
          color: "#fff",
          width: 2
        }
      },
      axisLabel: {
        show: !0,
        color: "#000",
        distance: 40,
        fontSize: 20,
        fontWeight: "normal",
        formatter: (a) => F(a) ? e.isFixed ? a.toFixed(Number(e.fixedNumber) <= 0 ? 0 : Number(e.fixedNumber)) : a : ""
      },
      anchor: {
        show: !0,
        showAbove: !0,
        size: 6,
        icon: "circle",
        itemStyle: {
          color: "#fff"
        }
      },
      detail: {
        valueAnimation: !0,
        formatter: "{value}%",
        color: "inherit",
        fontSize: 30,
        fontWeight: "bold",
        offsetCenter: [0, "35%"]
      },
      title: {
        fontSize: 20,
        color: "#000",
        fontWeight: "normal",
        offsetCenter: [0, "58%"]
      }
    }), j = () => {
      var t;
      if (f.value) {
        i || (i = A.init(f.value)), (t = e.options) == null || t.color;
        const a = {
          data: c.chartData
        };
        i.setOption({
          series: a
        });
      }
    }, b = () => {
      var t, a;
      if (f.value) {
        i || (i = A.init(f.value));
        const l = (t = e.options) == null ? void 0 : t.series, s = (a = e.options) == null ? void 0 : a.color;
        c.labelValueArray = [];
        let y = {};
        l ? y = l.map((o) => d(
          x(C(s)),
          e == null ? void 0 : e.gauge,
          o
        )) : e.data !== "" && (y = d(
          x(C(s)),
          {
            data: c.chartData
          },
          e == null ? void 0 : e.gauge,
          e == null ? void 0 : e.seriesStyle
        ));
        let n = d(
          {
            series: y
          },
          m
        );
        n = d(n, e.options), i.clear();
        try {
          i.setOption(n);
        } catch (o) {
          console.log(o);
        }
        i.on("click", (o) => {
          O("com-click", o);
        });
      }
    }, N = () => {
      D(() => {
        i.resize();
      });
    };
    return $(() => {
      v(), S(), b(), window.addEventListener("resize", N);
    }), M(() => {
      window.removeEventListener("resize", N);
    }), h(
      () => [
        e.gauge,
        e.options,
        e.seriesStyle,
        e.fixedNumber,
        e.isFixed
      ],
      () => {
        b();
      },
      {
        deep: !0
      }
    ), h(
      () => [
        e.name
      ],
      () => {
        S(), b();
      },
      {
        immediate: !0
      }
    ), h(() => e.data, () => {
      S(), j();
    }), h(
      () => [e.cStyle],
      () => {
        v(), D(() => {
          i.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      chart: f,
      curStyle: r,
      checkEmpty: z,
      domOnlyClass: u
    };
  }
});
export {
  B as default
};
