// @ths/design/es/components/proportion-bar/src/index.vue2.js
import { defineComponent as m, computed as u, reactive as d, watch as c, onMounted as y, toRefs as s } from "vue";
import { getDomRandomClass as f, setStyle as g } from "../../../utils/index.js";
const p = m({
  name: "TProportionBar",
  props: {
    // 目标值/总数
    targetValue: {
      type: Number,
      default: 100
    },
    // 当前值
    currentValue: {
      type: Number,
      default: 0
    },
    // 百分数
    percentage: {
      type: String,
      default: "0%"
    },
    // 图标
    iconSrc: {
      type: String,
      default: ""
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    }
  },
  setup(t) {
    const a = u(() => t.cStyle), n = f("proportion-bar"), r = d({
      targetValue: 100,
      currentValue: 0,
      percentage: "0%"
    }), l = u(() => t.iconSrc), o = u(() => t.currentValue && t.targetValue ? `${Number(t.currentValue) / Number(t.targetValue) * 100}%` : t.percentage), i = () => {
      let e = "";
      return a.value.wrapper && (e += g(a.value.wrapper, `.${n}`)), e;
    };
    return c(() => t.targetValue, (e) => {
      r.targetValue = e;
    }), c(() => t.currentValue, (e) => {
      r.currentValue = e;
    }, {
      immediate: !0
    }), c(() => t.percentage, (e) => {
      r.percentage = e;
    }, {
      immediate: !0
    }), y(() => {
      const e = document.querySelector(`.${n} .style-container`);
      e && (e.innerHTML = `<style>${i()}</style>`);
    }), {
      innerWidth: o,
      curIconSrc: l,
      ...s(r),
      domOnlyClass: n,
      curStyle: a
    };
  }
});
export {
  p as default
};
