// @ths/design/es/components/timeline-base/src/back-end-svg.vue.js
import l from "./back-end-svg.vue2.js";
import { openBlock as e, createElementBlock as t, normalizeStyle as r, createElementVNode as n } from "vue";
import "./back-end-svg.vue3.js";
import s from "../../../_virtual/_plugin-vue_export-helper.js";
const a = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "11",
  height: "11",
  viewBox: "0 0 68 68"
}, i = ["fill"];
function c(o, d, m, p, f, _) {
  return e(), t("div", {
    class: "back-end-svg",
    style: r({ "--hover-fill-color": o.hoverFillColor })
  }, [
    (e(), t("svg", a, [
      n("path", {
        id: "矩形_1",
        fill: o.fillColor,
        "data-name": "矩形 1",
        class: "cls-1",
        d: `M4.5,0A4.5,4.5,0,0,1,9,4.5v62a4.5,4.5,0,0,1-9,
      0V4.5A4.5,4.5,0,0,1,4.5,0ZM28.668,
      35.483L56.71,7.5a4.4,4.4,0,0,0-6.231-6.217L19.321,32.374a4.389,
      4.389,0,0,0,0,6.217L50.479,69.678a4.4,4.4,0,0,0,6.231-6.217Zm0,0`
      }, null, 8, i)
    ]))
  ], 4);
}
const w = /* @__PURE__ */ s(l, [["render", c]]);
export {
  w as default
};
