// @ths/design/es/components/timeline-base/src/forward-end-svg.vue.js
import l from "./forward-end-svg.vue2.js";
import { openBlock as e, createElementBlock as r, normalizeStyle as t, createElementVNode as s } from "vue";
import "./forward-end-svg.vue3.js";
import a from "../../../_virtual/_plugin-vue_export-helper.js";
const i = {
  xmlns: "http://www.w3.org/2000/svg",
  width: "11",
  height: "11",
  viewBox: "0 0 68 68"
}, n = ["fill"];
function d(o, c, m, f, p, _) {
  return e(), r("div", {
    class: "forward-dnd-svg",
    style: t({ "--hover-fill-color": o.hoverFillColor })
  }, [
    (e(), r("svg", i, [
      s("path", {
        id: "矩形_1",
        fill: o.fillColor,
        "data-name": "矩形 1",
        class: "cls-1",
        d: `M53.5,0A4.5,4.5,0,0,1,58,4.5v62a4.5,4.5,0,0,1-9,0V4.5A4.5,4.5,0,0,1,53.5,
      0ZM29.347,35.483L1.321,7.5A4.4,4.4,0,0,1,7.549,1.287l31.14,31.087a4.391,4.391,0,0,1,0,6.217L7.549,69.678a4.4,4.4,0,0,1-6.228-6.217Zm0,0`
      }, null, 8, n)
    ]))
  ], 4);
}
const g = /* @__PURE__ */ a(l, [["render", d]]);
export {
  g as default
};
