// @ths/design/es/components/timeline-base/src/index.vue2.js
import { defineComponent as Oe, ref as _, reactive as Ae, computed as y, watch as S, nextTick as J, onMounted as oe, getCurrentInstance as De, defineExpose as Me, onBeforeUnmount as Fe, toRefs as He } from "vue";
import { getDomRandomClass as je, setStyle as P, getCssPropertyString as de } from "../../../utils/index.js";
import Ne from "./go-back-svg.vue.js";
import Re from "./go-forward-svg.vue.js";
import Xe from "./back-end-svg.vue.js";
import qe from "./forward-end-svg.vue.js";
var ue = /* @__PURE__ */ ((t) => (t.backToEnd = "backToEnd", t.goBack = "goBack", t.goForward = "goForward", t.forwardToEnd = "forwardToEnd", t))(ue || {});
const Qe = Oe({
  name: "TTimelineBase",
  components: {
    GoBackSvg: Ne,
    GoForwardSvg: Re,
    BackEndSvg: Xe,
    ForwardEndSvg: qe
  },
  props: {
    // 图标
    timeData: {
      type: Array,
      default: () => [],
      require: !0
    },
    // 日期区间颜色
    timeColors: {
      type: Array,
      default: () => []
    },
    // 播放按钮自定义图片信息
    controlButtonInfo: {
      type: Object,
      default: () => ({})
    },
    // 选中时间
    curTime: {
      type: String,
      default: "",
      require: !0
    },
    // 默认是否开始播放
    autoPlay: {
      type: Boolean,
      default: !1
    },
    // 是否循环播放
    loop: {
      type: Boolean,
      default: !1
    },
    // 播放速度 单位：ms
    playInterval: {
      type: Number,
      default: 3e3
    },
    // 刻度标签的显示间隔
    interval: {
      type: Number,
      default: 0
    },
    // 步进按钮是否显示
    stepButtonVisible: {
      type: Boolean,
      default: !1
    },
    // 是否反转
    inverse: {
      type: Boolean,
      default: !1
    },
    // 事件默认是否触发
    immediateTrigger: {
      type: Boolean,
      default: !1
    },
    // 前进后退按钮的颜色
    stepButtonColor: {
      type: Object,
      default: () => ({
        default: "#dee2e8",
        hover: "#02d6db",
        disable: "#3b5a84"
      })
    },
    // 缩放比例
    scalePercent: {
      type: Number,
      default: 1
    },
    // 缩略显示刻度文本的条件 就是时间轴数据大于30个之后才会开启缩略显示
    abbreviationsDisplayCondition: {
      type: Number,
      default: 30
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {},
        // 左侧模块样式
        wrapperLeftMoudle: {},
        // 右侧模块样式
        wrapperRightMoudle: {},
        // 轴线进度条容器样式
        axisLineProgressWrappper: {},
        // 轴线进度条当前进度样式
        axisLineProgressCurrent: {},
        // 轴线刻度文本样式
        axisLabel: {},
        // 轴线刻度二级文本样式
        axisLabelSecond: {},
        // 轴线一级刻度样式
        axisFirstLevelItem: {},
        // 轴线二级刻度样式
        axisSecondLevelItem: {},
        // 播放按钮样式
        controlButtonPlay: {},
        // 暂停按钮样式
        controlButtonPause: {},
        // 前进按钮样式
        goForwardButton: {},
        // 前进到底按钮样式
        forwardToEndButton: {},
        // 后退按钮样式
        goBackButton: {},
        // 后退到底按钮样式
        backToEndButton: {},
        // 指示当前时间的提示框样式
        toolTipCurrentTime: {},
        // 鼠标悬停在轴线上显示的提示框样式
        toolTipHover: {},
        // 轴线样式
        axisLine: {},
        // 刻度容器
        axisItemWrapper: {},
        // 操作按钮容器
        operationButtonWrapper: {},
        // 操作按钮
        operationButton: {}
      })
    },
    // 是否总是显示二级刻度
    secondTick: {
      type: Boolean,
      default: !1
    },
    // 数据自定义样式
    dataStyle: {
      type: Object,
      default: () => ({})
    },
    // 展示完整的时间
    showCompleteTime: {
      type: Boolean,
      default: !1
    },
    // 指定时间
    specifiedTime: {
      type: String,
      default: ""
    },
    // 操作按钮是否显示(预测、回溯)
    optionBtnVisible: {
      type: Boolean,
      default: !1
    },
    // 滑块颜色（当未配置 sliderIcon 时生效）
    sliderColor: {
      type: String,
      default: "#022E69"
    },
    // 滑块图标地址（可选），优先于 sliderColor
    sliderIcon: {
      type: String,
      default: ""
    },
    // 是否开启滑块及键盘滑动控制
    sliderEnabled: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["time-change", "gap-set", "component-loaded"],
  setup(t, { emit: E }) {
    var Z, p, ee, te;
    const f = je("timeline-base"), se = [], fe = _(null);
    function G(n) {
      return n.map((a) => {
        if (typeof a != "string")
          return a;
        const l = a.match(/\d+/g);
        return l ? l.length < 4 ? l.join("-") : l.length > 4 ? `${l.slice(0, 3).join("-")} ${l[l.length - 2]}:${l[l.length - 1]}` : `${l.slice(0, 3).join("-")} ${l[l.length - 1]}` : a;
      });
    }
    const e = Ae({
      // 播放按钮是否滑过
      playBtnHover: !1,
      // 暂停按钮是否滑过
      pauseBtnHover: !1,
      data: G(t.timeData),
      // 当前播放时间轴的状态 'play' | 'pause'
      playOrPause: "pause",
      // 时间轴宽度
      timelineWidth: 0,
      // 每个刻度间间隔的像素
      gapPixel: 0,
      // 时间轴刻度位置数组
      timelineItemLevelArr: se,
      // 当前选中的顶头的刻度下标
      currentSelectIndex: t.timeData.indexOf(t.curTime),
      // 鼠标悬停所在的刻度的下标
      hoverIndex: 0,
      // 鼠标悬停位置
      hoverPos: 0,
      // 提示框是否可见
      hoverTooltipVisible: !1,
      // 步进按钮是否高亮
      stepButtonActive: {
        backToEnd: !1,
        goBack: !1,
        goForward: !1,
        forwardToEnd: !1
      },
      // 一级刻度宽度
      firstLevelItemWidth: y(() => {
        var a, l, r;
        const n = (r = (l = (a = t.cStyle) == null ? void 0 : a.axisFirstLevelItem) == null ? void 0 : l.default) == null ? void 0 : r.width;
        return n ? Number(n.replace("px", "")) : 1;
      }),
      // 二级刻度宽度
      secondLevelItemWidth: y(() => {
        var a, l, r;
        const n = (r = (l = (a = t.cStyle) == null ? void 0 : a.axisSecondLevelItem) == null ? void 0 : l.default) == null ? void 0 : r.width;
        return n ? Number(n.replace("px", "")) : 1;
      }),
      // 要显示的刻度文本索引数组
      showScaleTextIndexList: [],
      predictBtnActive: !0,
      backBtnActive: !1
    }), me = _(
      y(() => {
        var a;
        if (C.value)
          return `${X.value}%`;
        if (e.currentSelectIndex === t.timeData.length - 1)
          return "100%";
        const n = ((a = e.timelineItemLevelArr[e.currentSelectIndex]) == null ? void 0 : a.level) === 1 ? e.firstLevelItemWidth : e.secondLevelItemWidth;
        return `calc(${100 / (e.data.length - 1) * e.currentSelectIndex}% + ${n}px)`;
      })
    ), ve = _(
      y(() => {
        var n, a;
        return {
          left: `${100 / (e.data.length - 1) * e.currentSelectIndex}%`,
          transition: "left 0.5s",
          ...(a = (n = t.cStyle) == null ? void 0 : n.toolTipCurrentTime) == null ? void 0 : a.default
        };
      })
    ), F = _(), x = _();
    let d;
    function m(n) {
      e.playOrPause = n;
      const a = e.data.length - 1;
      let l = !0;
      const r = () => {
        e.currentSelectIndex === a && !t.loop ? l ? e.currentSelectIndex = 0 : (clearInterval(d), d = null, e.playOrPause = "pause", l = !1) : e.currentSelectIndex !== a ? (e.currentSelectIndex += 1, l = !1) : e.currentSelectIndex === a && t.loop && (e.currentSelectIndex = 0, l = !1);
      };
      if (n === "play") {
        if (d)
          return;
        e.currentSelectIndex === a && t.loop && (clearInterval(d), setTimeout(r, 0)), d = setInterval(r, t.playInterval);
      } else
        clearInterval(d), d = null;
    }
    function j() {
      if (e.data.length) {
        e.gapPixel = F.value.offsetWidth / (e.data.length - 1), E("gap-set", e.gapPixel);
        const n = [];
        for (let l = 0; l < e.data.length; l += 1) {
          const r = (t.interval + 1) * l;
          r <= e.data.length && n.push(r);
        }
        let a = 0;
        e.timelineItemLevelArr = Array.from({ length: e.data.length }, (l, r) => {
          const i = { pos: a, level: n.includes(r) ? 1 : 2 };
          return a += e.gapPixel, i;
        });
      }
    }
    function ge(n) {
      switch (n) {
        case "goBack":
          e.currentSelectIndex > 0 ? e.currentSelectIndex -= 1 : e.currentSelectIndex === 0 && t.loop && (e.currentSelectIndex = e.data.length - 1);
          break;
        case "backToEnd":
          e.currentSelectIndex = 0;
          break;
        case "goForward":
          e.currentSelectIndex < e.data.length - 1 ? e.currentSelectIndex += 1 : e.currentSelectIndex === e.data.length - 1 && t.loop && (e.currentSelectIndex = 0);
          break;
        case "forwardToEnd":
          e.currentSelectIndex = e.data.length - 1;
          break;
      }
    }
    function ye(n) {
      e.playOrPause === "play" && m("pause"), e.currentSelectIndex = n;
    }
    function Se(n, a) {
      var u, c, b, L, B, T, $, W, O, A, D, s, M, g, v, w, k, H, ne, ae, le, ie, re, ce;
      const l = {}, r = {
        default: (c = (u = t.cStyle) == null ? void 0 : u.axisFirstLevelItem) == null ? void 0 : c.default,
        active: (L = (b = t.cStyle) == null ? void 0 : b.axisFirstLevelItem) == null ? void 0 : L.active
      }, i = {
        default: (T = (B = t.cStyle) == null ? void 0 : B.axisSecondLevelItem) == null ? void 0 : T.default,
        active: (W = ($ = t.cStyle) == null ? void 0 : $.axisSecondLevelItem) == null ? void 0 : W.active
      };
      (D = (A = (O = t.dataStyle) == null ? void 0 : O[a]) == null ? void 0 : A.axisLineStyle) != null && D.default && (n.level === 1 ? r.default = (s = t.dataStyle[a].axisLineStyle) == null ? void 0 : s.default : i.default = (M = t.dataStyle[a].axisLineStyle) == null ? void 0 : M.default), (w = (v = (g = t.dataStyle) == null ? void 0 : g[a]) == null ? void 0 : v.axisLineStyle) != null && w.active && (n.level === 1 ? r.active = { ...r.active, ...(k = t.dataStyle[a].axisLineStyle) == null ? void 0 : k.active } : i.active = { ...i.active, ...(H = t.dataStyle[a].axisLineStyle) == null ? void 0 : H.active });
      const o = y(() => n.level === 1 ? r : i);
      return a === e.data.length - 1 && ((ae = (ne = o.value) == null ? void 0 : ne.default) != null && ae.transform) && (l.right = `-${(ie = (le = o.value) == null ? void 0 : le.default) == null ? void 0 : ie.width}` || "0"), e.currentSelectIndex === a ? {
        ...l,
        ...(re = o.value) == null ? void 0 : re.default,
        ...(ce = o.value) == null ? void 0 : ce.active
      } : o.value ? { ...l, ...o.value.default } : l;
    }
    function xe(n) {
      e.stepButtonActive[n] = !e.stepButtonActive[n];
    }
    function U(n) {
      if (!t.sliderEnabled)
        return;
      const a = n.key || n.code;
      if (!(a !== "ArrowLeft" && a !== "ArrowRight" && a !== "Left" && a !== "Right") && e.data.length) {
        if (e.playOrPause === "play" && m("pause"), a === "ArrowLeft" || a === "Left") {
          if (e.currentSelectIndex <= 0)
            return;
          n.preventDefault(), e.currentSelectIndex -= 1;
          return;
        }
        if (a === "ArrowRight" || a === "Right") {
          if (e.currentSelectIndex >= e.data.length - 1)
            return;
          n.preventDefault(), e.currentSelectIndex += 1;
        }
      }
    }
    function he(n) {
      if (x.value) {
        const a = x.value.getBoundingClientRect(), l = n.clientX - a.left, r = a.width, i = Math.round(l / r * (e.data.length - 1));
        e.hoverPos = l, e.hoverIndex = Math.max(0, Math.min(i, e.data.length - 1)), e.hoverTooltipVisible = !0;
      }
    }
    function Ie(n) {
      if (x.value) {
        const a = x.value.getBoundingClientRect(), l = n.clientX - a.left, r = a.width, i = Math.round(l / r * (e.data.length - 1));
        e.hoverPos = l, e.hoverIndex = Math.max(0, Math.min(i, e.data.length - 1));
      }
    }
    function be(n, a) {
      e.hoverIndex = a, e.hoverPos = n.target.offsetLeft - n.target.offsetWidth / 2 + n.offsetX;
    }
    const Le = {
      backToEnd: (Z = t.cStyle) == null ? void 0 : Z.goForwardButton,
      goBack: (p = t.cStyle) == null ? void 0 : p.forwardToEndButton,
      forwardToEnd: (ee = t.cStyle) == null ? void 0 : ee.goBackButton,
      goForward: (te = t.cStyle) == null ? void 0 : te.backToEndButton
    };
    function Be(n) {
      const a = Le[n];
      if (!a)
        return;
      const l = y(() => e.stepButtonActive[n]), r = !e.currentSelectIndex, i = e.currentSelectIndex === e.data.length - 1;
      let o;
      a && (o = a.default);
      const c = {
        backToEnd: r,
        goBack: r && !t.loop,
        forwardToEnd: i,
        goForward: i && !t.loop
      }[n];
      return c ? o = { ...o, ...a == null ? void 0 : a.disable } : !c && l.value && (o = { ...o, ...a == null ? void 0 : a.hover }), o;
    }
    function N(n, a, l, r) {
      var b, L, B, T, $, W, O, A, D;
      const i = n == null ? void 0 : n.match(/\d+/g), o = a == null ? void 0 : a.match(/\d+/g);
      let u = "";
      if (!i)
        return u;
      const c = i.length;
      if (c > 0) {
        let s = "";
        const M = (B = (L = (b = t.dataStyle) == null ? void 0 : b[t.timeData.length - 1]) == null ? void 0 : L.labelStyle) == null ? void 0 : B.default;
        if (r && M ? (Object.entries(M).forEach(([g, v]) => {
          s += `${g}:${v};`;
        }), s = `style="${s}"`) : (W = ($ = (T = t.dataStyle) == null ? void 0 : T[l]) == null ? void 0 : $.labelStyle) != null && W.default && (Object.entries((O = t.dataStyle[l].labelStyle) == null ? void 0 : O.default).forEach(([g, v]) => {
          s += `${g}:${v};`;
        }), s = `style="${s}"`), t.showCompleteTime)
          return u = `<span class="timeline__label--fragment" ${s}>${n}</span>`, u;
        if (o) {
          let g = o.slice(0, c - 1).every((v, w) => v === i[w]);
          if (t.timeData.length > t.abbreviationsDisplayCondition && e.showScaleTextIndexList.includes(l)) {
            const v = e.showScaleTextIndexList.findIndex((k) => k === l), w = (A = e.data[e.showScaleTextIndexList[v - 1]]) == null ? void 0 : A.match(/\d+/g);
            g = ((D = w == null ? void 0 : w.slice(0, c - 1)) == null ? void 0 : D.every((k, H) => k === i[H])) || !1;
          }
          g && !t.secondTick ? c > 4 ? u = `<span class="timeline__label--fragment" ${s}>
              ${i[c - 2]}:${i[c - 1]}</span>` : u = `<span class="timeline__label--fragment" ${s}>${i[c - 1]}</span>` : c > 4 ? u = `
              <span class="timeline__label--fragment" ${s}>${i[c - 2]}:${i[c - 1]}</span>
              <span class="timeline__label--fragment second" ${s}>${i.slice(0, c - 2).join("-")}</span>
            ` : u = `
              <span class="timeline__label--fragment" ${s}>${i[c - 1]}</span>
              <span class="timeline__label--fragment second" ${s}>${i.slice(0, c - 1).join("-")}</span>
            `;
        } else
          c > 4 ? u = `
              <span class="timeline__label--fragment" ${s}>${i[c - 2]}:${i[c - 1]}</span>
              <span class="timeline__label--fragment second" ${s}>${i.slice(0, c - 2).join("-")}</span>
            ` : u = `
              <span class="timeline__label--fragment" ${s}>${i[c - 1]}</span>
              <span class="timeline__label--fragment second" ${s}>${i.slice(0, c - 1).join("-")}</span>
            `;
      }
      return u;
    }
    function Te(n) {
      n === "play" ? e.playBtnHover = !0 : n === "pause" && (e.pauseBtnHover = !0);
    }
    function $e(n) {
      n === "play" ? e.playBtnHover = !1 : n === "pause" && (e.pauseBtnHover = !1);
    }
    function V() {
      e.data.reverse();
      const n = t.timeData;
      e.currentSelectIndex = n.reverse().indexOf(t.curTime);
    }
    function z() {
      const n = document.querySelectorAll(`.${f} .timeline__label-wrapper-item`), a = document.querySelectorAll(`.${f} .timeline__label-wrapper`), l = n.length + a.length;
      n.forEach((r, i) => {
        const o = r.querySelector(".timeline__label--former"), u = N(e.data[i], i ? e.data[i - 1] : "", i, !1);
        o && (o.innerHTML = u);
      }), a.forEach((r, i) => {
        const o = r.querySelector(".timeline__label--last"), u = N(e.data[l - 1], i ? e.data[l - 2] : "", i, !0);
        o && (o.innerHTML = u);
      });
    }
    function R() {
      Array.from(x.value.children).forEach((n, a) => {
        const l = n;
        l.style.backgroundColor = t.timeColors[a];
      });
    }
    const C = _(!1), X = _(0);
    let h = null, I = null;
    function q(n) {
      if (!x.value || !e.data.length)
        return e.currentSelectIndex;
      const a = x.value.getBoundingClientRect();
      let l = n - a.left;
      l < 0 && (l = 0), l > a.width && (l = a.width);
      const r = a.width || 1, i = r ? l / r : 0;
      X.value = i * 100;
      const o = Math.round(i * (e.data.length - 1));
      return Math.max(0, Math.min(o, e.data.length - 1));
    }
    function we(n) {
      t.sliderEnabled && (n.preventDefault(), e.playOrPause === "play" && m("pause"), C.value = !0, q(n.clientX), h = (a) => {
        C.value && q(a.clientX);
      }, I = (a) => {
        const l = q(a.clientX);
        e.currentSelectIndex = l, C.value = !1, h && (document.removeEventListener("mousemove", h), h = null), I && (document.removeEventListener("mouseup", I), I = null);
      }, document.addEventListener("mousemove", h), document.addEventListener("mouseup", I));
    }
    const _e = y(() => {
      const n = e.data.length > 1 ? 100 / (e.data.length - 1) * e.currentSelectIndex : 0, l = {
        position: "absolute",
        top: "50%",
        left: `${C.value ? X.value : n}%`,
        width: "14px",
        height: "14px",
        transform: "translate(-50%, -50%)",
        borderRadius: "50%",
        border: "2px solid #ffffff",
        boxShadow: "0 0 6px rgba(255, 0, 0, 0.8)",
        cursor: "grabbing",
        zIndex: "5"
      };
      return t.sliderIcon ? l.backgroundColor = "transparent" : l.backgroundColor = t.sliderColor || "#ff0000", l;
    }), ke = y(() => t.sliderIcon), Pe = () => {
      var a, l, r, i, o, u, c, b, L, B, T, $;
      let n = "";
      return (a = t == null ? void 0 : t.cStyle) != null && a.axisLabel && (n += P((l = t == null ? void 0 : t.cStyle) == null ? void 0 : l.axisLabel, `.${f} .timeline__label--fragment`)), (r = t == null ? void 0 : t.cStyle) != null && r.axisLabelSecond && (n += P((i = t == null ? void 0 : t.cStyle) == null ? void 0 : i.axisLabelSecond, `.${f} .timeline__label--fragment.second`)), (o = t == null ? void 0 : t.cStyle) != null && o.axisLineProgressWrappper && (n += P((u = t == null ? void 0 : t.cStyle) == null ? void 0 : u.axisLineProgressWrappper, `.${f} .timeline__progress`)), (c = t == null ? void 0 : t.cStyle) != null && c.axisLine && (n += P((b = t == null ? void 0 : t.cStyle) == null ? void 0 : b.axisLine, `.${f} .timeline__axis`)), (L = t == null ? void 0 : t.cStyle) != null && L.axisItemWrapper && (n += P((B = t == null ? void 0 : t.cStyle) == null ? void 0 : B.axisItemWrapper, `.${f} .timeline__item-wrapper`)), (T = t == null ? void 0 : t.cStyle) != null && T.operationButtonWrapper && (n += P(($ = t == null ? void 0 : t.cStyle) == null ? void 0 : $.operationButtonWrapper, `.${f} .t-timeline-base__option`)), n;
    }, Ee = () => {
      var a, l, r, i;
      let n = "";
      return (l = (a = t.cStyle) == null ? void 0 : a.operationButton) != null && l.default && (n += `.${f} .t-timeline-base__option .operation-button {
          ${de(t.cStyle.operationButton.default)}
        }`), (i = (r = t.cStyle) == null ? void 0 : r.operationButton) != null && i.active && (n += `.${f} .t-timeline-base__option .operation-button.active{
          ${de(t.cStyle.operationButton.active)}
        }`), n;
    }, K = () => {
      const n = document.querySelector(`.${f} .style-container`);
      n && (n.innerHTML = `<style>${Pe()}${Ee()}</style>`);
    };
    S(
      () => t.cStyle,
      () => {
        K();
      },
      {
        immediate: !0
      }
    ), S(
      () => t.timeData,
      (n, a) => {
        var u;
        if (JSON.stringify(n) === JSON.stringify(a))
          return;
        e.data = G(n), t.inverse && V(), e.playOrPause === "play" && (t.loop ? (m("pause"), e.currentSelectIndex = 0, m("play")) : m("pause")), e.showScaleTextIndexList = [];
        const l = y(() => t.timeData.length), r = 12, i = Math.ceil(l.value / r);
        let o = 0;
        for (let c = 0; c < l.value; c++)
          c === o && (Math.abs(c - ((u = t.timeData) == null ? void 0 : u.length)) > i - 2 && e.showScaleTextIndexList.push(c), o += i);
      },
      {
        deep: !0,
        immediate: !0
      }
    ), S(
      () => t.curTime,
      (n) => {
        e.currentSelectIndex = e.data.indexOf(n), J(() => {
          e.currentSelectIndex = e.data.indexOf(n);
        });
      }
    ), S(
      () => e.currentSelectIndex,
      (n) => {
        e.currentSelectIndex > -1 && E("time-change", { time: e.data[n], index: e.currentSelectIndex });
      }
    ), S(
      () => [e.data, t.secondTick],
      (n, a) => {
        JSON.stringify(n[0]) !== JSON.stringify(a[0]) && (e.currentSelectIndex = e.data.indexOf(t.curTime)), j(), J(() => {
          z(), R();
        });
      },
      {
        deep: !0
      }
    ), S(
      () => t.timeColors,
      () => {
        R();
      }
    ), S(
      () => t.interval,
      () => {
        j();
      }
    ), S(
      () => e.gapPixel,
      (n) => {
        E("gap-set", n);
      }
    ), S(
      () => t.playInterval,
      () => {
        clearInterval(d), d = null, m("play");
      }
    ), oe(() => {
      K(), t.sliderEnabled && window.addEventListener("keydown", U), clearInterval(d), d = null, F.value && (e.timelineWidth = F.value.offsetWidth - 2 * e.firstLevelItemWidth, t.immediateTrigger && E("time-change", { time: e.data[e.currentSelectIndex], index: e.currentSelectIndex }), t.autoPlay && m("play"), t.inverse && V(), j()), J(() => {
        z(), t.timeColors.length && R();
      });
    }), oe(() => {
      const n = De();
      E("component-loaded", n == null ? void 0 : n.proxy);
    }), Me({
      changePlayStatus: m
    }), Fe(() => {
      d && (clearInterval(d), d = null), h && (document.removeEventListener("mousemove", h), h = null), I && (document.removeEventListener("mouseup", I), I = null), window.removeEventListener("keydown", U);
    });
    function Q(n) {
      const a = e.data.indexOf(n);
      if (a === -1)
        return;
      const l = a;
      d && (clearInterval(d), d = null), e.currentSelectIndex = 0, e.playOrPause = "play", d = setInterval(() => {
        e.currentSelectIndex < l ? e.currentSelectIndex += 1 : (clearInterval(d), d = null, e.playOrPause = "pause");
      }, t.playInterval);
    }
    function Y(n) {
      const a = e.data.indexOf(n);
      if (a === -1)
        return;
      e.currentSelectIndex = a, d && (clearInterval(d), d = null), e.playOrPause = "play", d = setInterval(() => {
        e.currentSelectIndex === e.data.length - 1 ? t.loop ? e.currentSelectIndex = 0 : (clearInterval(d), d = null, e.playOrPause = "pause") : e.currentSelectIndex += 1;
      }, t.playInterval);
    }
    const Ce = () => {
      e.predictBtnActive || (e.predictBtnActive = !0, e.backBtnActive = !1, Y(t.specifiedTime));
    }, We = () => {
      e.backBtnActive || (e.backBtnActive = !0, e.predictBtnActive = !1, Q(t.specifiedTime));
    };
    return {
      ...He(e),
      timelineDom: F,
      progressSelectorDom: x,
      TimelineStepButtonType: ue,
      thsTimelineBase: fe,
      currentProgressWidth: me,
      currentTimeTooltipStyle: ve,
      setLineItemStyle: Se,
      changePlayStatus: m,
      setProgress: ge,
      handleMouseMoveProgress: Ie,
      handleMouseEnterProgress: he,
      handleAxisItemMouseMove: be,
      handleSelectItem: ye,
      setStepButtonStyle: Be,
      setStepButtonStatus: xe,
      formatTime: N,
      domOnlyClass: f,
      defaultImg: $e,
      hoverImg: Te,
      playToTime: Q,
      playFromTime: Y,
      handlePredict: Ce,
      handleBack: We,
      sliderIconStyle: _e,
      sliderIconSrc: ke,
      handleSliderMouseDown: we
    };
  }
});
export {
  Qe as default
};
