// @ths/design/es/components/timeline-base/src/index.vue.js
import te from "./index.vue2.js";
import { resolveComponent as f, openBlock as n, createElementBlock as r, normalizeClass as u, normalizeStyle as i, createElementVNode as t, createCommentVNode as a, withDirectives as d, vShow as v, Fragment as p, createVNode as g, renderList as y, createTextVNode as c, toDisplayString as _ } from "vue";
import "./index.vue3.js";
import ne from "../../../_virtual/_plugin-vue_export-helper.js";
const re = {
  key: 0,
  class: "t-timeline-base__option"
}, ie = ["src"], se = ["src"], ue = {
  ref: "timelineDom",
  class: "timeline"
}, ae = ["select-index"], de = ["src"], ve = ["onClick"], pe = { class: "timeline__axis" }, me = { class: "timeline__item-wrapper" }, fe = ["onClick", "onMousemove"], ge = { class: "timeline__label-wrapper" }, ye = {
  key: 0,
  class: "timeline__label timeline__label--former"
}, Se = { class: "timeline__label timeline__label--last" };
function Be(e, l, Ie, be, Te, ke) {
  var S, B, I, b, T, k, M, C, A, w, $, L, P, h, E, V, D, F, U, H, N, O, z, W, R, j, q, G, J, K, Q, X, Y, Z;
  const x = f("back-end-svg"), ee = f("go-back-svg"), le = f("go-forward-svg"), oe = f("forward-end-svg");
  return n(), r("div", {
    ref: "thsTimelineBase",
    class: u(["t-timeline-base", e.domOnlyClass]),
    style: i({ ...(B = (S = e.cStyle) == null ? void 0 : S.wrapper) == null ? void 0 : B.default })
  }, [
    e.optionBtnVisible ? (n(), r("div", re, [
      t("button", {
        class: u(["operation-button predict", { active: e.predictBtnActive }]),
        onClick: l[0] || (l[0] = (...o) => e.handlePredict && e.handlePredict(...o))
      }, "预测", 2),
      t("button", {
        class: u(["operation-button back", { active: e.backBtnActive }]),
        onClick: l[1] || (l[1] = (...o) => e.handleBack && e.handleBack(...o))
      }, "回溯", 2)
    ])) : a("", !0),
    t("section", {
      class: "left-moddle",
      style: i({ ...(b = (I = e.cStyle) == null ? void 0 : I.wrapperLeftMoudle) == null ? void 0 : b.default })
    }, [
      d(t("div", {
        class: "control",
        style: i({ ...(k = (T = e.cStyle) == null ? void 0 : T.controlButtonPlay) == null ? void 0 : k.default }),
        onClick: l[4] || (l[4] = (o) => e.changePlayStatus("play"))
      }, [
        (M = e.controlButtonInfo) != null && M.playImgUrl ? (n(), r("img", {
          key: 0,
          class: "control--play",
          src: e.playBtnHover ? (A = e.controlButtonInfo) == null ? void 0 : A.hoverPlayUrl : (C = e.controlButtonInfo) == null ? void 0 : C.playImgUrl,
          alt: "play",
          draggable: "false",
          onMouseenter: l[2] || (l[2] = (o) => e.hoverImg("play")),
          onMouseleave: l[3] || (l[3] = (o) => e.defaultImg("play"))
        }, null, 40, ie)) : a("", !0)
      ], 4), [
        [v, e.playOrPause === "pause"]
      ]),
      d(t("div", {
        class: "control",
        style: i({ ...($ = (w = e.cStyle) == null ? void 0 : w.controlButtonPause) == null ? void 0 : $.default }),
        onClick: l[7] || (l[7] = (o) => e.changePlayStatus("pause"))
      }, [
        (L = e.controlButtonInfo) != null && L.playImgUrl ? (n(), r("img", {
          key: 0,
          class: "control--pause",
          src: e.pauseBtnHover ? (h = e.controlButtonInfo) == null ? void 0 : h.hoverPauseUrl : (P = e.controlButtonInfo) == null ? void 0 : P.pauseImgUrl,
          alt: "pause",
          draggable: "false",
          onMouseenter: l[5] || (l[5] = (o) => e.hoverImg("pause")),
          onMouseleave: l[6] || (l[6] = (o) => e.defaultImg("pause"))
        }, null, 40, se)) : a("", !0)
      ], 4), [
        [v, e.playOrPause === "play"]
      ])
    ], 4),
    t("section", {
      class: "right-moddle",
      style: i({ ...(V = (E = e.cStyle) == null ? void 0 : E.wrapperRightMoudle) == null ? void 0 : V.default })
    }, [
      e.stepButtonVisible ? (n(), r(p, { key: 0 }, [
        t("div", {
          class: u(["step-back-end", e.currentSelectIndex ? "step-back-end--default cursor-pointer" : "step-back-end--disable"]),
          style: i(e.setStepButtonStyle(e.TimelineStepButtonType.backToEnd)),
          onClick: l[8] || (l[8] = (o) => e.setProgress(e.TimelineStepButtonType.backToEnd)),
          onMouseenter: l[9] || (l[9] = (o) => e.stepButtonActive[e.TimelineStepButtonType.backToEnd] = !0),
          onMouseleave: l[10] || (l[10] = (o) => e.stepButtonActive[e.TimelineStepButtonType.backToEnd] = !1)
        }, [
          g(x, {
            "fill-color": e.currentSelectIndex ? e.stepButtonColor.default : e.stepButtonColor.disable,
            "hover-fill-color": e.stepButtonColor.hover
          }, null, 8, ["fill-color", "hover-fill-color"])
        ], 38),
        t("div", {
          class: u(["step-back", e.currentSelectIndex || e.loop ? "step-back--default cursor-pointer" : "step-back--disable"]),
          style: i(e.setStepButtonStyle(e.TimelineStepButtonType.goBack)),
          onClick: l[11] || (l[11] = (o) => e.setProgress(e.TimelineStepButtonType.goBack)),
          onMouseenter: l[12] || (l[12] = (o) => e.stepButtonActive[e.TimelineStepButtonType.goBack] = !0),
          onMouseleave: l[13] || (l[13] = (o) => e.stepButtonActive[e.TimelineStepButtonType.goBack] = !1)
        }, [
          g(ee, {
            "fill-color": e.currentSelectIndex || e.loop ? e.stepButtonColor.default : e.stepButtonColor.disable,
            "hover-fill-color": e.stepButtonColor.hover
          }, null, 8, ["fill-color", "hover-fill-color"])
        ], 38)
      ], 64)) : a("", !0),
      t("div", ue, [
        t("div", {
          class: "timeline__progress cursor-pointer",
          style: i({ ...(F = (D = e.cStyle) == null ? void 0 : D.axisLineProgressWrappper) == null ? void 0 : F.default })
        }, [
          t("div", {
            ref: "progressSelectorDom",
            class: "timeline__progress-selector",
            onMousemove: l[14] || (l[14] = (...o) => e.handleMouseMoveProgress && e.handleMouseMoveProgress(...o)),
            onMouseleave: l[15] || (l[15] = (o) => e.hoverTooltipVisible = !1),
            onMouseenter: l[16] || (l[16] = (...o) => e.handleMouseEnterProgress && e.handleMouseEnterProgress(...o))
          }, [
            (n(!0), r(p, null, y(e.data.slice(0, e.data.length - 1), (o, s) => (n(), r("section", {
              key: o,
              class: "timeline__progress-selector--option",
              "select-index": s + 1
            }, null, 8, ae))), 128))
          ], 544),
          t("div", {
            class: "timeline__progress--current",
            style: i({
              ...(H = (U = e.cStyle) == null ? void 0 : U.axisLineProgressCurrent) == null ? void 0 : H.default,
              width: e.currentProgressWidth,
              transition: "width 0.5s"
            })
          }, null, 4),
          e.sliderEnabled ? (n(), r("div", {
            key: 0,
            class: "timeline__slider-icon",
            style: i(e.sliderIconStyle),
            onMousedown: l[17] || (l[17] = (...o) => e.handleSliderMouseDown && e.handleSliderMouseDown(...o))
          }, [
            e.sliderIconSrc ? (n(), r("img", {
              key: 0,
              class: "timeline__slider-icon-img",
              src: e.sliderIconSrc,
              alt: "slider icon",
              draggable: "false"
            }, null, 8, de)) : a("", !0)
          ], 36)) : a("", !0),
          t("div", {
            class: "timeline__progress-selector",
            onMousemove: l[19] || (l[19] = (...o) => e.handleMouseMoveProgress && e.handleMouseMoveProgress(...o)),
            onMouseleave: l[20] || (l[20] = (o) => e.hoverTooltipVisible = !1),
            onMouseenter: l[21] || (l[21] = (...o) => e.handleMouseEnterProgress && e.handleMouseEnterProgress(...o))
          }, [
            t("section", {
              style: { width: "5px" },
              "select-index": 0,
              onClick: l[18] || (l[18] = (o) => e.handleSelectItem(0))
            }),
            (n(!0), r(p, null, y(e.data.slice(0, e.data.length - 1), (o, s) => (n(), r("section", {
              key: o,
              class: "timeline__progress-selector--option",
              onClick: (m) => e.handleSelectItem(s + 1)
            }, null, 8, ve))), 128))
          ], 32),
          d(t("div", {
            class: "timeline__tooltip",
            style: i(e.currentTimeTooltipStyle)
          }, [
            c(_(e.data[e.currentSelectIndex]) + " ", 1),
            t("div", {
              class: "timeline__tooltip--corner",
              style: i({
                "border-top-color": ((z = (O = (N = e.cStyle) == null ? void 0 : N.toolTipCurrentTime) == null ? void 0 : O.default) == null ? void 0 : z.backgroundColor) || ((j = (R = (W = e.cStyle) == null ? void 0 : W.toolTipCurrentTime) == null ? void 0 : R.default) == null ? void 0 : j.background)
              })
            }, null, 4)
          ], 4), [
            [v, e.currentSelectIndex >= 0]
          ]),
          d(t("div", {
            class: "timeline__tooltip timeline__tooltip-move",
            style: i({
              left: `${e.hoverPos}px`,
              transition: "left 0.1s linear",
              ...(G = (q = e.cStyle) == null ? void 0 : q.toolTipHover) == null ? void 0 : G.default
            })
          }, [
            c(_(e.data[e.hoverIndex]) + " ", 1),
            t("div", {
              class: "timeline__tooltip--corner",
              style: i({
                "border-top-color": ((Q = (K = (J = e.cStyle) == null ? void 0 : J.toolTipHover) == null ? void 0 : K.default) == null ? void 0 : Q.backgroundColor) || ((Z = (Y = (X = e.cStyle) == null ? void 0 : X.toolTipHover) == null ? void 0 : Y.default) == null ? void 0 : Z.background)
              })
            }, null, 4)
          ], 4), [
            [v, e.hoverTooltipVisible]
          ])
        ], 4),
        t("div", pe, [
          t("div", me, [
            (n(!0), r(p, null, y(e.timelineItemLevelArr.slice(0, e.timelineItemLevelArr.length - 1), (o, s) => (n(), r("div", {
              key: o.pos + s,
              class: "timeline__item-wrapper-item"
            }, [
              t("em", {
                ref_for: !0,
                ref: "axisItems",
                class: u(["timeline__item", {
                  "timeline__item--first-level": o.level === 1,
                  "timeline__item--second-level": o.level === 2,
                  "timeline__item--first-level-ative": e.currentSelectIndex === s && o.level === 1,
                  "timeline__item--second-level-ative": e.currentSelectIndex === s && o.level === 2
                }]),
                style: i({ ...e.setLineItemStyle(o, s) }),
                onClick: (m) => e.handleSelectItem(s),
                onMousemove: (m) => e.handleAxisItemMouseMove(m, s),
                onMouseleave: l[22] || (l[22] = (m) => e.hoverTooltipVisible = !1),
                onMouseenter: l[23] || (l[23] = (m) => e.hoverTooltipVisible = !0)
              }, null, 46, fe)
            ]))), 128)),
            e.timelineItemLevelArr.length ? (n(), r("em", {
              key: 0,
              ref: "axisItems",
              class: u(["timeline__item timeline__item--last", {
                "timeline__item--first-level": e.timelineItemLevelArr[e.timelineItemLevelArr.length - 1].level === 1,
                "timeline__item--second-level": e.timelineItemLevelArr[e.timelineItemLevelArr.length - 1].level === 2,
                "timeline__item--first-level-ative": e.currentSelectIndex === e.timelineItemLevelArr.length - 1 && e.timelineItemLevelArr[e.timelineItemLevelArr.length - 1].level === 1,
                "timeline__item--second-level-ative": e.currentSelectIndex === e.timelineItemLevelArr.length - 1 && e.timelineItemLevelArr[e.timelineItemLevelArr.length - 1].level === 2
              }]),
              style: i({
                ...e.setLineItemStyle(e.timelineItemLevelArr[e.timelineItemLevelArr.length - 1], e.timelineItemLevelArr.length - 1)
              }),
              onClick: l[24] || (l[24] = (o) => e.handleSelectItem(e.timelineItemLevelArr.length - 1)),
              onMousemove: l[25] || (l[25] = (o) => e.handleAxisItemMouseMove(o, e.timelineItemLevelArr.length - 1)),
              onMouseleave: l[26] || (l[26] = (o) => e.hoverTooltipVisible = !1),
              onMouseenter: l[27] || (l[27] = (o) => e.hoverTooltipVisible = !0)
            }, null, 38)) : a("", !0)
          ]),
          t("div", ge, [
            (n(!0), r(p, null, y(e.timelineItemLevelArr.slice(0, e.timelineItemLevelArr.length - 1), (o, s) => (n(), r("div", {
              key: o.pos + s,
              class: "timeline__label-wrapper-item"
            }, [
              e.timeData.length > e.abbreviationsDisplayCondition && e.showScaleTextIndexList.includes(s) || e.timeData.length <= e.abbreviationsDisplayCondition ? d((n(), r("div", ye, null, 512)), [
                [v, o.level === 1]
              ]) : a("", !0)
            ]))), 128)),
            d(t("div", Se, null, 512), [
              [v, e.timelineItemLevelArr.length && e.timelineItemLevelArr[e.timelineItemLevelArr.length - 1].level === 1]
            ])
          ])
        ])
      ], 512),
      e.stepButtonVisible ? (n(), r(p, { key: 1 }, [
        t("div", {
          class: u(["step-forward", e.currentSelectIndex === e.data.length - 1 && !e.loop ? "step-forward--disable" : "step-forward--default cursor-pointer"]),
          style: i(e.setStepButtonStyle(e.TimelineStepButtonType.goForward)),
          onClick: l[28] || (l[28] = (o) => e.setProgress(e.TimelineStepButtonType.goForward)),
          onMouseenter: l[29] || (l[29] = (o) => e.stepButtonActive[e.TimelineStepButtonType.goForward] = !0),
          onMouseleave: l[30] || (l[30] = (o) => e.stepButtonActive[e.TimelineStepButtonType.goForward] = !1)
        }, [
          g(le, {
            "fill-color": e.currentSelectIndex === e.data.length - 1 && !e.loop ? e.stepButtonColor.disable : e.stepButtonColor.default,
            "hover-fill-color": e.stepButtonColor.hover
          }, null, 8, ["fill-color", "hover-fill-color"])
        ], 38),
        t("div", {
          class: u(["step-forward-end", e.currentSelectIndex === e.data.length - 1 ? "step-forward-end--disable" : "step-forward-end--default cursor-pointer"]),
          style: i(e.setStepButtonStyle(e.TimelineStepButtonType.forwardToEnd)),
          onClick: l[31] || (l[31] = (o) => e.setProgress(e.TimelineStepButtonType.forwardToEnd)),
          onMouseenter: l[32] || (l[32] = (o) => e.stepButtonActive[e.TimelineStepButtonType.forwardToEnd] = !0),
          onMouseleave: l[33] || (l[33] = (o) => e.stepButtonActive[e.TimelineStepButtonType.forwardToEnd] = !1)
        }, [
          g(oe, {
            "fill-color": e.currentSelectIndex === e.data.length - 1 ? e.stepButtonColor.disable : e.stepButtonColor.default,
            "hover-fill-color": e.stepButtonColor.hover
          }, null, 8, ["fill-color", "hover-fill-color"])
        ], 38)
      ], 64)) : a("", !0)
    ], 4),
    l[34] || (l[34] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const $e = /* @__PURE__ */ ne(te, [["render", Be]]);
export {
  $e as default
};
