// @ths/design/es/components/user-multiple-dialog/src/index.vue.js
import k from "./index.vue2.js";
import { resolveComponent as n, openBlock as d, createBlock as f, withCtx as o, createVNode as a, createTextVNode as r, createElementVNode as u, createElementBlock as g, Fragment as b, renderList as c, toDisplayString as s } from "vue";
import "./index.vue3.js";
import "./index.vue4.js";
import E from "../../../_virtual/_plugin-vue_export-helper.js";
const B = { class: "content-container" }, I = { class: "content-container__lf" }, U = { class: "form-container" }, A = { class: "content-container__rt" }, R = { class: "selected-header" }, $ = { class: "selected-count" }, z = { class: "selected-person" };
function L(e, t, H, O, j, M) {
  const P = n("t-dialog-single-tree-input"), m = n("el-form-item"), p = n("el-col"), h = n("el-row"), D = n("el-option"), S = n("el-select"), C = n("el-input"), F = n("t-search-button"), w = n("el-form"), i = n("el-table-column"), v = n("el-table"), T = n("el-config-provider"), V = n("t-paging"), _ = n("el-tag"), N = n("el-button"), y = n("el-dialog");
  return d(), f(y, {
    modelValue: e.dialogVisible,
    "onUpdate:modelValue": t[9] || (t[9] = (l) => e.dialogVisible = l),
    width: "90%",
    title: "选择人员",
    "close-on-click-modal": !1,
    "close-on-press-escape": !1,
    "append-to-body": "",
    "custom-class": "user-multiple-dialog",
    onClose: e.handleClose
  }, {
    footer: o(() => [
      a(N, { onClick: e.handleClose }, {
        default: o(() => t[10] || (t[10] = [
          r(" 取消 ")
        ])),
        _: 1,
        __: [10]
      }, 8, ["onClick"]),
      a(N, {
        type: "primary",
        onClick: e.handleSure
      }, {
        default: o(() => t[11] || (t[11] = [
          r(" 确定 ")
        ])),
        _: 1,
        __: [11]
      }, 8, ["onClick"])
    ]),
    default: o(() => [
      u("div", B, [
        u("div", I, [
          u("div", U, [
            a(w, {
              ref: "searchFormRef",
              model: e.tableFormParams,
              "status-icon": "",
              "label-width": "80px",
              class: "change-form"
            }, {
              default: o(() => [
                a(h, null, {
                  default: o(() => [
                    a(p, { span: 12 }, {
                      default: o(() => [
                        a(m, { label: "单位" }, {
                          default: o(() => [
                            a(P, {
                              selectedId: e.tableFormParams.unitId,
                              "onUpdate:selectedId": t[0] || (t[0] = (l) => e.tableFormParams.unitId = l),
                              selectedName: e.tableFormParams.unitName,
                              "onUpdate:selectedName": t[1] || (t[1] = (l) => e.tableFormParams.unitName = l),
                              class: "dialog-input",
                              clearable: !1,
                              "dialog-title": e.unitSingleTreeParams.dialogTitle,
                              "default-expand-layer": e.unitSingleTreeParams.defaultExpandLayer,
                              url: e.unitSingleTreeParams.url,
                              "root-id": e.unitSingleTreeParams.rootId,
                              "simple-data": e.unitSingleTreeParams.simpleData,
                              "result-data-path": e.unitSingleTreeParams.resultDataPath,
                              param: e.unitSingleTreeParams.param,
                              placeholder: e.unitSingleTreeParams.placeholder
                            }, null, 8, ["selectedId", "selectedName", "dialog-title", "default-expand-layer", "url", "root-id", "simple-data", "result-data-path", "param", "placeholder"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    }),
                    a(p, { span: 12 }, {
                      default: o(() => [
                        a(m, { label: "部门" }, {
                          default: o(() => [
                            a(P, {
                              selectedId: e.tableFormParams.deptId,
                              "onUpdate:selectedId": t[2] || (t[2] = (l) => e.tableFormParams.deptId = l),
                              selectedName: e.tableFormParams.deptName,
                              "onUpdate:selectedName": t[3] || (t[3] = (l) => e.tableFormParams.deptName = l),
                              class: "dialog-input",
                              "dialog-title": e.deptSingleTreeParams.dialogTitle,
                              "default-expand-layer": e.deptSingleTreeParams.defaultExpandLayer,
                              url: e.deptSingleTreeParams.url,
                              "root-id": e.deptSingleTreeParams.rootId,
                              "simple-data": e.deptSingleTreeParams.simpleData,
                              "result-data-path": e.deptSingleTreeParams.resultDataPath,
                              param: e.deptSingleTreeParams.param,
                              placeholder: e.deptSingleTreeParams.placeholder
                            }, null, 8, ["selectedId", "selectedName", "dialog-title", "default-expand-layer", "url", "root-id", "simple-data", "result-data-path", "param", "placeholder"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    })
                  ]),
                  _: 1
                }),
                a(h, null, {
                  default: o(() => [
                    a(p, { span: 12 }, {
                      default: o(() => [
                        a(m, { label: "岗位" }, {
                          default: o(() => [
                            a(S, {
                              modelValue: e.tableFormParams.enforcePostCode,
                              "onUpdate:modelValue": t[4] || (t[4] = (l) => e.tableFormParams.enforcePostCode = l),
                              clearable: !0,
                              placeholder: "请选择岗位"
                            }, {
                              default: o(() => [
                                (d(!0), g(b, null, c(e.jobOptions, (l) => (d(), f(D, {
                                  key: l.enforcePostCode,
                                  label: l.enforcePostName,
                                  value: l.enforcePostCode
                                }, null, 8, ["label", "value"]))), 128))
                              ]),
                              _: 1
                            }, 8, ["modelValue"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    }),
                    a(p, { span: 12 }, {
                      default: o(() => [
                        a(m, { label: "人员标签" }, {
                          default: o(() => [
                            a(S, {
                              modelValue: e.tableFormParams.personTag,
                              "onUpdate:modelValue": t[5] || (t[5] = (l) => e.tableFormParams.personTag = l),
                              clearable: !0,
                              placeholder: "请选择人员标签"
                            }, {
                              default: o(() => [
                                (d(!0), g(b, null, c(e.personTagOptions, (l) => (d(), f(D, {
                                  key: l.value,
                                  label: l.label,
                                  value: l.value
                                }, null, 8, ["label", "value"]))), 128))
                              ]),
                              _: 1
                            }, 8, ["modelValue"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    }),
                    a(p, { span: 12 }, {
                      default: o(() => [
                        a(m, { label: "关键字" }, {
                          default: o(() => [
                            a(C, {
                              modelValue: e.tableFormParams.keyword,
                              "onUpdate:modelValue": t[6] || (t[6] = (l) => e.tableFormParams.keyword = l),
                              type: "text",
                              maxlength: "50",
                              placeholder: "请输入姓名、用户名、手机号等",
                              clearable: ""
                            }, null, 8, ["modelValue"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    }),
                    a(p, {
                      span: 12,
                      class: "flex-right"
                    }, {
                      default: o(() => [
                        a(F, {
                          "is-show-fold-btn": !1,
                          "search-show-all": !0,
                          onSearch: e.getTableData,
                          onReset: e.handleReset
                        }, null, 8, ["onSearch", "onReset"])
                      ]),
                      _: 1
                    })
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }, 8, ["model"])
          ]),
          a(T, { locale: e.locale }, {
            default: o(() => [
              a(v, {
                ref: "userTable",
                size: "mini",
                data: e.page.list,
                "tooltip-effect": "dark",
                border: "",
                stripe: "",
                "row-key": e.currentDefaultProps.userId,
                onSelectAll: e.handleSelectionAllChange,
                onSelect: e.handleSelectionChange,
                onRowClick: e.handleRowClick
              }, {
                default: o(() => [
                  a(i, {
                    align: "center",
                    type: "selection",
                    fixed: "",
                    "reserve-selection": !0
                  }),
                  a(i, {
                    align: "left",
                    width: "120",
                    label: "姓名",
                    prop: "userName",
                    "show-overflow-tooltip": ""
                  }, {
                    default: o(({ row: l }) => [
                      r(s(e.handleNoVal(l[e.currentDefaultProps.userName])), 1)
                    ]),
                    _: 1
                  }),
                  a(i, {
                    align: "left",
                    label: "单位",
                    prop: "orgName",
                    "show-overflow-tooltip": ""
                  }, {
                    default: o(({ row: l }) => [
                      r(s(e.handleNoVal(l[e.currentDefaultProps.orgName])), 1)
                    ]),
                    _: 1
                  }),
                  a(i, {
                    align: "left",
                    label: "部门",
                    prop: "deptName",
                    "show-overflow-tooltip": ""
                  }, {
                    default: o(({ row: l }) => [
                      r(s(e.handleNoVal(l[e.currentDefaultProps.deptName])), 1)
                    ]),
                    _: 1
                  }),
                  a(i, {
                    align: "left",
                    width: "140",
                    label: "岗位",
                    prop: "positionName",
                    "show-overflow-tooltip": ""
                  }, {
                    default: o(({ row: l }) => [
                      r(s(e.handleNoVal(l[e.currentDefaultProps.positionName])), 1)
                    ]),
                    _: 1
                  }),
                  a(i, {
                    align: "center",
                    width: "130",
                    label: "执法证号",
                    prop: "enforceCardCode",
                    "show-overflow-tooltip": ""
                  }, {
                    default: o(({ row: l }) => [
                      r(s(e.handleNoVal(l[e.currentDefaultProps.enforceCardCode])), 1)
                    ]),
                    _: 1
                  })
                ]),
                _: 1
              }, 8, ["data", "row-key", "onSelectAll", "onSelect", "onRowClick"])
            ]),
            _: 1
          }, 8, ["locale"]),
          a(V, {
            page: e.page,
            total: e.page.total,
            "pager-count": 5,
            class: "dialog-pagination",
            layout: "total, prev, pager, next",
            onHandleSearch: e.getTableData,
            "onUpdate:pageNum": t[7] || (t[7] = (l) => e.page.pageNum = l),
            "onUpdate:pageSize": t[8] || (t[8] = (l) => e.page.pageSize = l)
          }, null, 8, ["page", "total", "onHandleSearch"])
        ]),
        u("div", A, [
          u("div", R, [
            u("div", $, " 已选:" + s(e.tableSelectedData.length) + "个 ", 1)
          ]),
          u("div", z, [
            (d(!0), g(b, null, c(e.tableSelectedData, (l) => (d(), f(_, {
              key: l[e.currentDefaultProps.userId],
              closable: "",
              type: "primary",
              onClose: (q) => e.handleRemoveUser(l)
            }, {
              default: o(() => [
                r(s(e.handleNoVal(l[e.currentDefaultProps.userName])), 1)
              ]),
              _: 2
            }, 1032, ["onClose"]))), 128))
          ])
        ])
      ])
    ]),
    _: 1
  }, 8, ["modelValue", "onClose"]);
}
const X = /* @__PURE__ */ E(k, [["render", L], ["__scopeId", "data-v-5f10179a"]]);
export {
  X as default
};
