// @ths/design/es/components/user-multiple-dialog/src/index.vue2.js
import { defineComponent as k, ref as f, computed as L, reactive as A, onMounted as K, toRefs as j, nextTick as u } from "vue";
import z from "qs";
import J from "lodash/merge";
import q from "lodash/cloneDeep";
import { ElMessage as B } from "element-plus";
import G from "element-plus/lib/locale/lang/zh-cn";
import M from "@ths/c-common-base-paging";
import { TSearchButton as V } from "@ths/c-common-base-button";
import W from "../../dialog-single-tree-input/src/index.vue.js";
import { handleNoVal as $ } from "./utils/common.js";
import S from "./utils/http.js";
import { IC_SELECT_LAW_PERSON_INTERFACE as y, getDictList as h, getCurrentOrg as H } from "./services/user-multiple.js";
const Q = "http://192.168.24.189:9600/enforce-local-check/api/ouUserApi/listUser", I = {
  userId: "userId",
  userName: "userName",
  orgName: "orgName",
  deptName: "deptName",
  positionName: "positionName",
  enforceCardCode: "enforceCardCode"
}, de = k({
  name: "TUserMultipleDialog",
  components: {
    TPaging: M,
    TDialogSingleTreeInput: W,
    TSearchButton: V
  },
  props: {
    // 默认人员数据
    personData: {
      type: Array,
      default: () => []
    },
    // 人员列表数据-请求地址
    personDataUrl: {
      type: String,
      default: Q
    },
    // 人员列表数据-请求方式（GET、POST），目前只支持这两种方式
    requestType: {
      type: String,
      default: "GET"
    },
    // 人员列表数据-是否使用 requestBody 接收参数（post 请求时使用），如果传 false，则会使用 qs.stringify() 对参数进行序列化处理
    useRequestBody: {
      type: Boolean,
      default: !0
    },
    // 人员列表数据-返回数据在 data 中的位置，如：mapList
    resultDataPath: {
      type: String,
      default: "data"
    },
    // 人员列表数据-数据别名
    defaultProps: {
      type: Object,
      default: () => I
    }
  },
  emits: ["on-close", "on-sure"],
  setup(n, { emit: p }) {
    const m = f(), s = f(), D = {
      keyword: "",
      enforcePostCode: "",
      unitId: "",
      unitName: "",
      personTag: "",
      deptId: "",
      deptName: ""
    }, l = L(() => J(q(I), n.defaultProps)), a = A({
      // 弹窗显示
      dialogVisible: !0,
      // 分页数据和列表数据
      page: {
        pageNum: 1,
        pageSize: 7
      },
      tableFormParams: JSON.parse(JSON.stringify(D)),
      // 表格筛选条件-下拉数据
      jobOptions: [],
      // 人员标签
      personTagOptions: [],
      // 单位单选树
      unitSingleTreeParams: {
        dialogTitle: "选择单位",
        defaultExpandLayer: 1,
        url: y.getOrgList,
        rootId: "",
        simpleData: {
          idKey: "orgId",
          // 数据中id对应的key
          nameKey: "orgName",
          // 要展示的名称
          pIdKey: "orgParentId",
          // 数据中父节点标识key，懒加载时候使用
          spanKey: ""
          // 在节点后拼接内容 key
        },
        resultDataPath: "data",
        param: {},
        placeholder: "请选择单位"
      },
      // 部门单选树
      deptSingleTreeParams: {
        dialogTitle: "选择部门",
        defaultExpandLayer: 1,
        url: y.getDeptList,
        rootId: "",
        simpleData: {
          idKey: "deptId",
          // 数据中id对应的key
          nameKey: "deptName",
          // 要展示的名称
          pIdKey: "deptParentId",
          // 数据中父节点标识key，懒加载时候使用 PARENT
          spanKey: ""
          // 在节点后拼接内容 key
        },
        resultDataPath: "data",
        param: {},
        placeholder: "请选择部门"
      },
      // 表格选中人员数据
      tableSelectedData: JSON.parse(JSON.stringify(n.personData))
    });
    function b(e, t) {
      let r = e, o = t;
      o = o.replace(/\[(\w+)\]/g, ".$1"), o = o.replace(/^\./, "");
      const i = o.split(".");
      for (let c = 0; c < i.length && r; c++) {
        const g = i[c];
        if (g in r)
          r = r[g];
        else
          break;
      }
      return r;
    }
    const P = (e) => new Promise((t, r) => {
      try {
        t(S.get(n.personDataUrl, { params: e }));
      } catch (o) {
        r(o);
      }
    }), N = (e) => {
      let t = e;
      return n.useRequestBody || (t = z.stringify(e)), new Promise((r, o) => {
        try {
          r(S.post(n.personDataUrl, t));
        } catch (i) {
          o(i);
        }
      });
    }, O = async () => {
      a.jobOptions = [];
      try {
        const e = await h({
          parentId: "POSITION_DIC"
        });
        if (e != null && e.data && (e == null ? void 0 : e.status) === 200) {
          const { data: t } = e == null ? void 0 : e.data;
          a.jobOptions = t.map((r) => ({
            enforcePostCode: r.dictionarycode,
            enforcePostName: r.dictionaryname
          }));
        }
      } catch (e) {
        console.error(e);
      }
    }, T = async () => {
      a.personTagOptions = [];
      try {
        const e = await h({
          parentId: "PERSON_TAG"
        });
        if (e != null && e.data && (e == null ? void 0 : e.status) === 200) {
          const { data: t } = e == null ? void 0 : e.data;
          a.personTagOptions = t.map((r) => ({
            value: r.dictionarycode,
            label: r.dictionaryname
          }));
        }
      } catch (e) {
        console.error(e);
      }
    }, v = () => {
      s.value.clearSelection(), a.tableSelectedData.forEach((e, t) => {
        const r = a.page.list.find((o) => o[l.value.userId] === e[l.value.userId]);
        r && (a.tableSelectedData[t] = JSON.parse(JSON.stringify(r)), u(() => s.value.toggleRowSelection(r, !0)));
      });
    }, d = async () => {
      let e = {};
      n.requestType === "GET" ? e = await P({ ...a.tableFormParams, pageSize: a.page.pageSize, pageNum: a.page.pageNum }) : e = await N({ ...a.tableFormParams, pageSize: a.page.pageSize, pageNum: a.page.pageNum }), e != null && e.data && (e == null ? void 0 : e.status) === 200 ? a.page = b(e.data, n.resultDataPath) || {} : a.page = {}, v();
    }, R = (e) => {
      e.length ? e.forEach((t) => {
        a.tableSelectedData.findIndex((o) => o[l.value.userId] === t[l.value.userId]) === -1 && a.tableSelectedData.push(t);
      }) : a.page.list.forEach((t) => {
        const r = a.tableSelectedData.findIndex((o) => o[l.value.userId] === t[l.value.userId]);
        a.tableSelectedData.splice(r, 1);
      });
    }, C = (e, t) => {
      const r = a.tableSelectedData.findIndex((o) => o[l.value.userId] === t[l.value.userId]);
      r !== -1 ? a.tableSelectedData.splice(r, 1) : a.tableSelectedData.push(t);
    }, w = (e) => {
      a.tableSelectedData.splice(a.tableSelectedData.indexOf(e), 1);
      const t = a.page.list.find((r) => r[l.value.userId] === e[l.value.userId]);
      t && u(() => s.value.toggleRowSelection(t || e, !1));
    }, E = (e) => {
      u(() => s.value.toggleRowSelection(e, !0)), a.tableSelectedData.find((t) => t[l.value.userId] === e[l.value.userId]) || a.tableSelectedData.push(e);
    }, x = async () => {
      await m.value.resetFields(), d();
    }, F = () => {
      p("on-close");
    }, U = () => {
      if (!a.tableSelectedData.length) {
        B.error("请选择人员！");
        return;
      }
      p("on-sure", a.tableSelectedData);
    }, _ = async () => {
      try {
        const e = await H();
        if (e != null && e.data && (e == null ? void 0 : e.status) === 200) {
          const { data: t } = e == null ? void 0 : e.data;
          a.tableFormParams.unitId = t.orgId, a.tableFormParams.unitName = t.orgName;
        }
      } catch (e) {
        console.error(e);
      }
    };
    return K(async () => {
      a.page.pageNum = 1, a.page.pageSize = 7, await _(), await O(), await T(), d();
    }), {
      locale: {
        ...G
      },
      userTable: s,
      searchFormRef: m,
      ...j(a),
      handleClose: F,
      handleSelectionAllChange: R,
      handleSelectionChange: C,
      handleRowClick: E,
      getTableData: d,
      handleRemoveUser: w,
      handleSure: U,
      handleNoVal: $,
      handleReset: x,
      currentDefaultProps: l
    };
  }
});
export {
  de as default
};
