// @ths/design/es/components/two-dimension-text/src/index.vue2.js
import { defineComponent as m, computed as r, watch as f, onMounted as s } from "vue";
import { getDomRandomClass as p, setStyle as i } from "../../../utils/index.js";
import "../../two-dimension-container/index.js";
import y from "../../two-dimension-container/src/index.vue.js";
const S = m({
  name: "TTwoDimensionText",
  components: {
    TTwoDimensionContainer: y
  },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外容器
        wrapper: {
          default: {}
        },
        // 左容器
        sideLeftWrapper: {
          default: {}
        },
        // 右容器
        sideRightWrapper: {
          default: {}
        },
        // 中间容器
        centerWrapper: {
          default: {}
        },
        // 中心内容
        innerWrapper: {
          default: {}
        },
        // 两侧标题
        title: {
          default: {}
        },
        // 标识线
        line: {
          default: {}
        },
        // 内标识线
        lineInner: {
          default: {}
        },
        // 左值
        leftNum: {
          default: {}
        },
        // 右值
        rightNum: {
          default: {}
        },
        // 数值
        centerNum: {
          default: {}
        },
        // 单位
        unit: {
          default: {}
        },
        // 中间标题
        desc: {
          default: {}
        }
      })
    },
    // 左侧数据
    leftData: {
      type: Object,
      default: () => ({})
    },
    // 右侧数据
    rightData: {
      type: Object,
      default: () => ({})
    },
    // 中心数据
    centerData: {
      type: Object,
      default: () => ({})
    },
    // 中心背景
    bgUrl: {
      type: String,
      default: ""
    }
  },
  setup(n) {
    const e = r(() => n.cStyle), l = r(() => n.leftData), u = r(() => n.rightData), c = r(() => n.centerData), a = p("two-dimension-text"), d = () => {
      let t = "";
      return e.value.innerWrapper && (t += i(e.value.innerWrapper, `.${a} .two-dimension-container .center-content`)), e.value.unit && (t += i(e.value.unit, `.${a} .two-dimension-container .center-value .unit`)), e.value.centerNum && (t += i(e.value.centerNum, `.${a} .two-dimension-container .center-value .count`)), e.value.desc && (t += i(e.value.desc, `.${a} .two-dimension-container .center-content .desc`)), t;
    }, o = () => {
      const t = document.querySelector(`.${a} > .style-container`);
      t && (t.innerHTML = `<style>${d()}</style>`);
    };
    return f(() => [n.cStyle, n.centerData], () => {
      o();
    }, {
      deep: !0,
      immediate: !0
    }), s(() => {
      o();
    }), {
      curStyle: e,
      curLeftData: l,
      curRightData: u,
      curCenterData: c,
      domOnlyClass: a
    };
  }
});
export {
  S as default
};
