// @ths/design/es/components/list-card-rate/src/index.vue2.js
import { defineComponent as L, computed as i, reactive as N, onMounted as h, watch as C, toRefs as O } from "vue";
import P from "lodash/cloneDeep";
import R from "lodash/merge";
import { getDomRandomClass as U } from "../../../utils/index.js";
import k from "./item.vue.js";
const d = {
  name: "name",
  currentValue: "value",
  unit: "unit",
  rateName: "rateName",
  rateValue: "rate",
  rateUnit: "rateUnit",
  rateStatus: "status"
}, w = L({
  name: "TListCardRate",
  components: { TRateItem: k },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 单项容器
        itemWrapper: {
          default: {}
        },
        // 名称
        name: {
          default: {}
        },
        // 值
        value: {
          default: {}
        },
        // 比率名称
        rateName: {
          default: {}
        },
        // 图标
        rateIcon: {
          default: {}
        },
        // 比率值
        rateValue: {
          default: {}
        },
        // 单位
        unit: {
          default: {}
        },
        // 分割线
        line: {
          default: {}
        }
      })
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => d
    },
    // 状态定义
    statusList: {
      type: Array,
      default: () => []
    },
    // 是否交换值与文字位置
    changeValuePosition: {
      type: Boolean,
      default: () => !1
    }
  },
  // 事件定义
  emits: ["card-click"],
  setup(t, { emit: o }) {
    const f = i(() => t.cStyle), m = i(() => t.changeValuePosition), y = U("list-card-rate"), u = N({
      cardData: []
    }), g = (e) => {
      o("card-click", JSON.parse(JSON.stringify(e)));
    }, V = (e) => {
      var n;
      let r = "";
      return t.statusList && t.statusList.length && (r = (n = t.statusList.find((l) => l.name === e)) == null ? void 0 : n.url), r;
    }, s = () => {
      const e = R(P(d), t.dataAlias), {
        name: r,
        currentValue: n,
        unit: l,
        rateName: S,
        rateValue: c,
        rateStatus: A,
        rateUnit: D
      } = e;
      u.cardData = [], t.data && t.data.length && (u.cardData = t.data.map((a) => ({
        name: a[r] || "-",
        currentValue: a[n] !== 0 ? a[n] || "-" : 0,
        unit: a[l],
        rateName: a[S],
        rateUnit: a[D] || "%",
        rateValue: a[c] !== 0 ? a[c] || "-" : 0,
        url: V(a[A])
      })));
    };
    return h(() => {
      s();
    }), C(
      () => [t.data, t.dataAlias, t.statusList],
      () => {
        s();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      curStyle: f,
      curChangeValuePosition: m,
      domOnlyClass: y,
      clickListCard: g,
      ...O(u)
    };
  }
});
export {
  w as default
};
