// @ths/design/es/components/list-card-rate/src/item.vue2.js
import { defineComponent as f, computed as u, onMounted as d, watch as v } from "vue";
import { getDomRandomClass as y, setStyle as n } from "../../../utils/index.js";
const C = f({
  name: "TListCardRate",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 单项容器
        itemWrapper: {
          default: {}
        },
        // 名称
        name: {
          default: {}
        },
        // 值
        value: {
          default: {}
        },
        // 比率名称
        rateName: {
          default: {}
        },
        // 图标
        rateIcon: {
          default: {}
        },
        // 比率值
        rateValue: {
          default: {}
        },
        // 单位
        unit: {
          default: {}
        },
        // 分割线
        line: {
          default: {}
        }
      })
    },
    // 数据
    itemData: {
      type: Object,
      default: () => ({})
    },
    // 名称数值交换位置
    changeValuePosition: {
      type: Boolean,
      default: () => !1
    }
  },
  // 事件名定义
  emits: ["card-click"],
  setup(l, { emit: c }) {
    const e = u(() => l.cStyle), o = u(() => l.changeValuePosition), r = u(() => l.itemData), a = y("list-card-rate-item"), m = (t) => {
      c("card-click", t);
    }, s = () => {
      let t = "";
      return e.value.itemWrapper && (t += n(e.value.itemWrapper, `.${a}`)), e.value.name && (t += n(e.value.name, `.${a} .item-current-content .name`)), e.value.value && (t += n(e.value.value, `.${a} .item-current-content .value`)), e.value.rateName && (t += n(e.value.rateName, `.${a} .item-change-content .name`)), e.value.rateIcon && (t += n(e.value.rateIcon, `.${a} .item-change-content .change-icon`)), e.value.rateValue && (t += n(e.value.rateValue, `.${a} .item-change-content .value`)), e.value.unit && (t += n(e.value.unit, `.${a} .unit`)), e.value.line && (t += n(e.value.line, `.${a} .line`)), t;
    }, i = () => {
      const t = document.querySelector(`.${a} > .style-container`);
      t && (t.innerHTML = `<style>${s()}</style>`);
    };
    return d(() => {
      i();
    }), v(
      () => [l.cStyle],
      () => {
        i();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      curStyle: e,
      curChangeValuePosition: o,
      domOnlyClass: a,
      curItemData: r,
      clickListCard: m
    };
  }
});
export {
  C as default
};
