// @ths/design/es/components/chart-pie-border/src/index.vue2.js
import { defineComponent as he, computed as s, ref as A, reactive as ge, watch as _, onMounted as ve, onUnmounted as Se, nextTick as B, useCssVars as be } from "vue";
import * as we from "echarts";
import H from "lodash/merge";
import j from "lodash/cloneDeep";
import { splitArray as xe, getDomRandomClass as Ce, rgb2rgba as Ie, replacePollutionHtml as Te, setStyle as D } from "../../../utils/index.js";
import { DEFAULT_OPTIONS as K, POLLUTION_HTML as Ae } from "../../../utils/constant.js";
const M = { name: "name", value: "value" }, x = j(K.color), V = {
  root: {},
  wrapper: {
    default: {}
  },
  chart: {
    default: {}
  },
  legendRoot: {
    default: {}
  },
  legendRow: {
    default: {}
  },
  legendItem: {
    default: {}
  },
  legendIcon: {
    default: {}
  },
  legendName: {
    default: {}
  },
  legendValue: {
    default: {}
  },
  innerText: {
    default: {}
  }
}, Y = 0.5, $ = {
  borderWidth: 1,
  borderRadius: 1
}, De = {
  align: "top",
  color: "#ddd",
  fontSize: "14px",
  numberFontsize: "20px",
  numberFontWeight: "bold"
}, Q = he({
  name: "TChartPieBorder",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 饼图数据
    data: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    alias: {
      type: Object,
      default: () => M,
      required: !1
    },
    // 图形填充透明度
    opacity: {
      type: Number,
      default: Y
    },
    // 图形边框宽度
    borderStyle: {
      type: Object,
      default: $
    },
    // 整个图形宽度
    width: {
      type: String,
      default: "100%",
      required: !1
    },
    // 整个图形高度
    height: {
      type: String,
      default: "100%",
      required: !1
    },
    // 整个图形命名
    title: {
      type: String,
      default: "",
      required: !1
    },
    // 是否开启循环展示中间区域
    isShowCenter: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 中间区域循环时间
    showCenterTime: {
      type: Number,
      default: 1e3,
      required: !1
    },
    // 饼图半径
    radius: {
      type: Array,
      default: () => ["55%", "70%"],
      required: !1,
      validator: (e) => e.length === 2
    },
    // 饼图数据间分隔
    separation: {
      type: Object,
      default: () => ({ width: 2 }),
      required: !1,
      validator: (e) => {
        let f = !0;
        return typeof e.width == "string" && (f = /^\d+(\.\d+)?%$/.test(e.width)), f && e.color ? /#[0-9a-fA-F]{3,6}/.test(e.color) : f;
      }
    },
    // 数据单位
    unit: {
      type: String,
      default: "",
      required: !1
    },
    // 总数单位
    allCountUnit: {
      type: String,
      default: "",
      required: !1
    },
    // 内环
    showInner: {
      type: Boolean,
      default: !0,
      required: !1
    },
    innerStyle: {
      type: Object,
      default: () => De,
      required: !1
    },
    // 中心文字
    innerText: {
      type: String,
      default: "",
      required: !1
    },
    // 显示数据列表
    showList: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 数据列表排列方式
    listStyle: {
      type: Object,
      default: () => ({
        align: "right",
        value: !0,
        fontSize: 14,
        iconSize: 14,
        col: 2,
        cStyle: {}
      }),
      required: !1,
      validator: (e) => !!e
    },
    // 分隔线样式
    splitLineStyle: {
      type: Object,
      default: () => ({
        value: !1,
        cStyle: {}
      })
    },
    // 总数样式
    allCountStyle: {
      type: Object,
      default: () => ({
        value: !1,
        cStyle: {}
      })
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({ ...V }),
      required: !1,
      validator: (e) => !!e
    },
    // 列表容器自定义样式
    lStyle: {
      type: Object,
      default: () => ({}),
      required: !1,
      validator: (e) => !!e
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    }
  },
  emits: ["com-click", "legend-click"],
  setup(e, { emit: f }) {
    const u = s(() => e.cStyle), m = Ce("chart-pie-border"), L = j(K.color), h = A(0);
    let v = null;
    const C = A(null), g = A([]), X = s(() => e.innerText), c = ge({
      label: "",
      value: ""
    }), O = {
      dataColor: "#666666",
      fontColor: "#666666",
      separationColor: "#EEE",
      separationColorOpacity: 1,
      maskColor: "#FFF",
      maskColorOpacity: 0.8
    }, Z = s(() => ({
      ...V,
      ...e.cStyle
    })), R = s(() => (t) => {
      var r, a;
      return (r = t == null ? void 0 : t.default) != null && r.margin ? ((a = t == null ? void 0 : t.default) == null ? void 0 : a.margin).map((i) => `${i}px`).join(" ") : null;
    }), ee = s(() => (t) => t != null && t.margin ? (t == null ? void 0 : t.margin).map((l) => `${l}px`).join(" ") : null), te = () => {
      var a;
      return `${(((a = C.value) == null ? void 0 : a.clientWidth) || 0) * +`0.${parseInt(e.radius[0], 10)}` - 30}px`;
    }, E = (t) => {
      if (!/^#([0-9A-F]{3}|[0-9A-F]{6})$/i.test(t))
        return !1;
      let a = t.substr(1);
      if (a.length === 3) {
        let w = "";
        for (let y = 0; y < 3; y++)
          w += a[y] + a[y];
        a = w;
      }
      a = `0x${a}`;
      const l = a >> 16, o = a >> 8 & 255, i = a & 255, d = `rgb(${l}, ${o}, ${i})`, T = `rgba(${l}, ${o}, ${i}, ${e.opacity || Y})`;
      return {
        rgb: d,
        rgba: T
      };
    }, ae = {
      label: {
        show: !1
      },
      labelLine: {
        show: !1
      },
      color: "rgba(0, 0, 0, 0)",
      borderColor: "rgba(0, 0, 0, 0)",
      borderWidth: 0
    }, le = () => e.showList ? ["left", "right"].includes(e.listStyle.align) ? {
      show: "row",
      value: !!e.listStyle.value,
      style: {
        flexDirection: e.listStyle.align === "right" ? "row" : "row-reverse",
        alignItems: "center"
      },
      cStyle: e.listStyle.cStyle
    } : {
      show: "col",
      value: !!e.listStyle.value,
      style: {
        flexDirection: e.listStyle.align === "bottom" ? "column" : "column-reverse",
        alignItems: "center"
      },
      cStyle: e.listStyle.cStyle
    } : {
      show: "",
      value: !1,
      style: {},
      cStyle: {}
    }, I = (t = "") => {
      const r = t.toUpperCase().replace(/[-.+_]/g, ""), a = Ae.filter((l) => l.name === r);
      return a.length > 0 ? a[0].html : t;
    }, re = () => {
      let t = "";
      return u.value.wrapper && (t += D(u.value.wrapper, `.${m}`)), u.value.emptyContainer && (t += D(u.value.emptyContainer, `.${m} .empty`)), u.value.emptyImg && (t += D(u.value.emptyImg, `.${m} .empty .empty-img`)), u.value.emptyText && (t += D(u.value.emptyText, `.${m} .empty .empty-text`)), t;
    }, F = () => {
      const t = document.querySelector(`.${m} .style-container`);
      t && (t.innerHTML = `<style>${re()}</style>`);
    }, ne = s(() => {
      var t, r, a, l;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((i) => {
            var d;
            return (i == null ? void 0 : i.data) && ((d = i == null ? void 0 : i.data) == null ? void 0 : d.length) !== 0;
          });
        if (typeof ((r = e.options) == null ? void 0 : r.series) == "object")
          return ((l = (a = e.options.series) == null ? void 0 : a.data) == null ? void 0 : l.length) === 0;
      }
      return !!e.data.length;
    }), N = (t, r = x) => [...t].map((l, o) => ({
      ...l,
      index: o,
      name: I(l.name),
      tooltipName: l.name,
      style: {
        color: r[o % r.length],
        legend: {
          nameColor: O.fontColor,
          valueColor: O.dataColor,
          backgroundColor: r[o % r.length],
          borderColor: Ie(r[o % r.length], 0.5)
        }
      }
    })), ie = (t) => typeof t == "string" && t.includes("[") && t.includes("]") ? JSON.parse(t) : t, oe = (t) => String(t).includes("%") ? Number(String(t).split("%")[0]) : Number(t), W = () => {
      const t = [], r = g.value.reduce((a, l) => typeof l == "object" ? a + Number(l.value || 0) : a + Number(l || 0), 0);
      return g.value.forEach((a, l) => {
        E((e.options.color || x)[l]) && t.push(
          {
            value: typeof a == "object" ? a.value || 0 : a || 0,
            name: typeof a == "object" && a.name || "",
            itemStyle: {
              color: E((e.options.color || x)[l]).rgba,
              borderColor: E((e.options.color || x)[l]).rgb,
              borderWidth: e.borderStyle.borderWidth || $.borderWidth,
              borderRadius: ie(e.borderStyle.borderRadius || $.borderRadius)
            }
          },
          {
            value: e.separation.width === 0 || e.separation.width === "0%" ? 0 : r * (oe(e.separation.width) / 100),
            name: "interval",
            itemStyle: ae,
            tooltip: {
              show: !1
            }
          }
        );
      }), t;
    };
    let n = null;
    const se = s(() => xe(N(g.value, e.options.color || L), e.listStyle.col || 2)), ue = s(() => N(g.value, e.options.color || L)), S = () => {
      c.label = "", c.value = "";
    }, k = (t) => {
      h.value = 0;
      let r = 0, a = 0;
      const l = t.series[0].data.map((i, d) => d);
      n.dispatchAction({
        type: "highlight",
        seriesIndex: 0,
        dataIndex: 0
      });
      function o() {
        var i, d, T, w, y;
        c.label = ((i = t.series[0].data[r]) == null ? void 0 : i.name) !== "interval" ? I(t.series[0].data[r].name) : I(t.series[0].data[r - 1].name) ?? "", c.value = ((d = t.series[0].data[r]) == null ? void 0 : d.name) !== "interval" ? t.series[0].data[r].value : ((T = t.series[0].data[r - 1]) == null ? void 0 : T.value) ?? "", n.dispatchAction({
          type: "downplay",
          seriesIndex: 0,
          dataIndex: l
        }), a = ((w = t.series[0].data[r]) == null ? void 0 : w.name) !== "interval" ? r : r - 1, h.value = a / 2, n.dispatchAction({
          type: "highlight",
          seriesIndex: 0,
          dataIndex: l[a]
        }), r++, r > ((y = t.series[0]) == null ? void 0 : y.data.length) - 1 && (r = 0, a = 0);
      }
      v = setInterval(() => {
        o();
      }, e.showCenterTime / 2);
    }, p = (t) => {
      const r = t.series[0].data.map((a, l) => l);
      n.dispatchAction({
        type: "downplay",
        seriesIndex: 0,
        dataIndex: r
      });
    };
    let P;
    const b = () => {
      B(() => {
        n.resize();
      });
    };
    function q() {
      var t, r, a;
      if (C.value) {
        n || (n = we.init(C.value)), clearInterval(v);
        const l = H({
          color: e.options.color || x,
          title: {
            text: e.title,
            left: "center",
            padding: [0, 0, 40, 0]
          },
          legend: {
            show: !1
          },
          tooltip: {
            show: !0
          },
          xAxis: {
            show: !1
          },
          yAxis: {
            show: !1
          },
          series: []
        }, e.options), o = {
          name: e.title,
          type: "pie",
          radius: e.radius,
          avoidLabelOverlap: !0,
          clockwise: !0,
          label: {
            show: !1,
            position: "center"
          },
          ...e.options.dataSeries,
          data: W()
        };
        l.series = [
          o
          // 饼图数据series
        ], P = l, n.setOption(l), b(), e.isShowCenter ? (t = l == null ? void 0 : l.series[0]) != null && t.data && ((a = (r = l == null ? void 0 : l.series[0]) == null ? void 0 : r.data) != null && a.length) ? k(l) : (S(), p(l)) : (h.value = "", S(), p(l));
      }
    }
    const ce = (t) => {
      clearInterval(v), W().forEach((r, a) => {
        n.dispatchAction({
          type: "downplay",
          seriesIndex: 0,
          dataIndex: a
        });
      }), n.dispatchAction({
        type: "highlight",
        seriesIndex: 0,
        dataIndex: t.dataIndex
      }), c.label = t.name !== "interval" ? I(t.name) : "", c.value = t.name !== "interval" ? t.value : "", h.value = t.dataIndex / 2;
    }, de = (t) => {
      c.label = "", c.value = "", h.value = "", n.dispatchAction({
        type: "downplay",
        seriesIndex: 0,
        dataIndex: t.dataIndex
      }), clearInterval(v), e.isShowCenter && k(P);
    };
    function fe(t, r) {
      f("legend-click", {
        current: t,
        currentIndex: r,
        all: g.value
      });
    }
    function U() {
      const t = H(j(M), e.alias);
      g.value = e.data.map((r) => {
        const a = {};
        return Object.keys(t).forEach((l) => {
          l === "name" ? a[l] = Te(r[t[l]]) : a[l] = r[t[l]];
        }), { ...r, ...a };
      });
    }
    _(() => [
      e.data,
      e.alias
    ], () => {
      try {
        S(), U(), q(), b();
      } catch (t) {
        console.log(t);
      }
    }, { deep: !0 });
    const z = A(e.listStyle.col);
    _(() => [
      e.title,
      e.showInner,
      e.options,
      e.radius,
      e.borderStyle,
      e.innerStyle,
      e.separation,
      e.opacity,
      e.cStyle,
      e.listStyle,
      e.isShowCenter,
      e.showCenterTime,
      e.allCountStyle,
      e.splitLineStyle,
      e.allCountUnit
    ], () => {
      try {
        F(), z.value = e.listStyle.col, S(), q();
      } catch (t) {
        console.log(t);
      }
    }, { deep: !0 }), ve(() => {
      F(), S(), U(), q(), n.on("mouseover", ce), n.on("mouseout", de), b(), window.addEventListener("resize", b), n.on("click", (t) => {
        f("com-click", t);
      });
    }), Se(() => {
      window.removeEventListener("resize", b), clearInterval(v);
    });
    const ye = s(() => le()), me = s(() => {
      var t;
      return {
        width: e.showList ? e.height : e.width,
        height: e.height,
        background: (t = e.options) == null ? void 0 : t.backgroundColor
      };
    });
    return _(
      () => [e.width, e.height, e.cStyle],
      () => {
        B(() => {
          n.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      currentIndex: h,
      currentContent: c,
      chart: C,
      chartDataRow: se,
      chartData: ue,
      curInnerText: X,
      listDataStyle: ye,
      style: me,
      cStyles: Z,
      col: z,
      handleLegendItemClick: fe,
      marginAddPx: R,
      marginAddPxLine: ee,
      getTextWidth: te,
      defaultColors: L,
      checkEmpty: ne,
      curStyle: u,
      domOnlyClass: m
    };
  }
}), J = () => {
  be((e) => ({
    "17cd19a6": e.col
  }));
}, G = Q.setup;
Q.setup = G ? (e, f) => (J(), G(e, f)) : J;
export {
  Q as default
};
