// @ths/design/es/components/chart-transverse/src/index.vue.js
import a from "./index.vue2.js";
import { openBlock as n, createElementBlock as i, normalizeClass as m, normalizeStyle as r, withDirectives as o, createElementVNode as e, vShow as l } from "vue";
import "./index.vue3.js";
import c from "../../../_virtual/_plugin-vue_export-helper.js";
const p = {
  ref: "chartId",
  class: "chart"
}, d = { class: "empty" }, u = ["src"];
function y(t, s, f, h, v, E) {
  return n(), i("div", {
    class: m(["t-chart-transverse", t.domOnlyClass]),
    style: r({ ...t.cStyle.wrapper.default })
  }, [
    o(e("div", p, null, 512), [
      [l, t.checkEmpty]
    ]),
    o(e("div", d, [
      e("img", {
        class: "empty-img",
        style: r({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, u),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [l, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const w = /* @__PURE__ */ c(a, [["render", y]]);
export {
  w as default
};
