// @ths/design/es/components/chart-transverse/src/index.vue2.js
import { defineComponent as B, computed as _, ref as H, reactive as U, onMounted as W, onUnmounted as K, watch as F, nextTick as q, toRefs as X } from "vue";
import * as Y from "echarts";
import o from "lodash/merge";
import x from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as G, DEFAULT_OPTIONS_AXIS as J } from "../../../utils/constant.js";
import { getDomRandomClass as Q, setStyle as h } from "../../../utils/index.js";
const V = o(x(G), x(J.xAxis)), f = o(
  x(V),
  {
    legend: {
      show: !1
    },
    tooltip: {
      trigger: "item"
    },
    grid: {
      borderWidth: 0,
      top: "10%",
      left: "2%",
      right: "13%",
      bottom: "3%"
    },
    xAxis: {
      type: "value",
      axisTick: {
        show: !1
      },
      axisLine: {
        show: !1
      },
      splitLine: {
        show: !1
      },
      axisLabel: {
        show: !1
      }
    },
    yAxis: [
      o({}, {
        type: "category",
        inverse: !0,
        axisTick: {
          show: !1
        },
        axisLine: {
          show: !1
        },
        axisLabel: {
          show: !0,
          inside: !1,
          padding: [0, -60, 0, 0],
          fontSize: 16,
          color: "#666",
          verticalAlign: "bottom",
          lineHeight: 30
        }
      }),
      o({}, {
        type: "category",
        inverse: !0,
        axisLine: {
          show: !1
        },
        axisTick: {
          show: !1
        },
        axisLabel: {
          show: !0,
          inside: !1,
          formatter: "{value}",
          // 右侧Y轴文字显示
          color: "#538FFF",
          fontSize: "14",
          fontFamily: "PingFangSC-Regular"
        },
        splitArea: {
          show: !1
        },
        splitLine: {
          show: !1
        }
      })
    ]
  }
), Z = {
  name: "条",
  type: "bar",
  showBackground: !0,
  borderRadius: 30,
  yAxisIndex: 0,
  barWidth: 3,
  itemStyle: {
    color: {
      type: "linear",
      x: 0,
      y: 0,
      x2: 1,
      y2: 0,
      colorStops: [
        {
          offset: 0,
          color: "rgb(252, 237, 37,0)"
        },
        {
          offset: 0.3,
          color: "rgb(252, 237, 37,0.8)"
        },
        {
          offset: 1,
          color: "rgb(252, 237, 37,1)"
        }
      ],
      global: !1
      // 缺省为 false
    },
    borderRadius: 10
  },
  label: {
    show: !0,
    width: 7,
    height: 7,
    backgroundColor: "#538FFF",
    position: "right",
    distance: 0,
    fontSize: 16,
    borderRadius: 6,
    shadowColor: "#538FFF",
    shadowBlur: 8,
    formatter(t) {
      return "";
    }
  }
}, ne = B({
  name: "TChartTransverse",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => {
      },
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 左侧 Y 轴类目取resData数据中的别名设置
        yAxisLeftLabel: "",
        // 右侧 Y 轴类目取resData数据中的别名设置
        yAxisRightLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 单位
    unit: {
      type: String,
      default: "万人"
    },
    bar: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["legend-select"],
  setup(t, { emit: M }) {
    const d = Q("chart-transverse"), l = _(() => t.cStyle), b = H(null);
    let u = null;
    const m = U({
      // 图表数据
      chartData: []
    }), S = () => {
      var e, a;
      (a = (e = t.seriesSetting) == null ? void 0 : e.data) != null && a.length && t.data && (m.chartData = t.seriesSetting.data.map((i) => {
        const { name: r, dataKey: n } = i;
        return {
          name: r || "--",
          data: t.data.map((s) => s[n])
        };
      }));
    }, N = () => {
      let e = "";
      return l.value.wrapper && (e += h(l.value.wrapper, `.${d}`)), l.value.emptyContainer && (e += h(l.value.emptyContainer, `.${d} .empty`)), l.value.emptyImg && (e += h(l.value.emptyImg, `.${d} .empty .empty-img`)), l.value.emptyText && (e += h(l.value.emptyText, `.${d} .empty .empty-text`)), e;
    }, v = () => {
      const e = document.querySelector(`.${d} .style-container`);
      e && (e.innerHTML = `<style>${N()}</style>`);
    }, P = _(() => {
      var e, a, i, r;
      if (t.options) {
        if (Array.isArray((e = t.options) == null ? void 0 : e.series))
          return t.options.series.some((s) => {
            var g;
            return (s == null ? void 0 : s.data) && ((g = s == null ? void 0 : s.data) == null ? void 0 : g.length) !== 0;
          });
        if (typeof ((a = t.options) == null ? void 0 : a.series) == "object")
          return ((r = (i = t.options.series) == null ? void 0 : i.data) == null ? void 0 : r.length) === 0;
      }
      return !!t.data.length;
    }), y = (e, a) => {
      let i = e == null ? void 0 : e.toLowerCase();
      if (i && i.slice(0, 1) === "#") {
        if (i.length === 4) {
          let n = "#";
          for (let s = 1; s < 4; s += 1)
            n += i.slice(s, s + 1).concat(i.slice(s, s + 1));
          i = n;
        }
        const r = [];
        for (let n = 1; n < 7; n += 2)
          r.push(parseInt(`0x${i.slice(n, n + 2)}`, 16));
        return `rgba(${r.join(",")},${a})`;
      }
      return i;
    }, A = (e, a) => o(
      {
        ...Z,
        itemStyle: {
          color: {
            type: "linear",
            x: 0,
            y: 0,
            x2: 1,
            y2: 0,
            colorStops: [
              {
                offset: 0,
                color: y(e == null ? void 0 : e[a], 0) || y(f.color[a], 0) || "#FFF"
              },
              {
                offset: 0.5,
                color: y(e == null ? void 0 : e[a], 0.8) || y(f.color[a], 0.8) || "#FFF"
              },
              {
                offset: 1,
                color: y(e == null ? void 0 : e[a], 1) || y(f.color[a], 1) || "#FFF"
              }
            ],
            global: !1
            // 缺省为 false
          },
          borderRadius: 10
        },
        label: {
          show: !0,
          width: 7,
          height: 7,
          backgroundColor: (e == null ? void 0 : e[a]) || f.color[a] || "#FFF",
          position: "right",
          distance: 0,
          fontSize: 16,
          borderRadius: 6,
          shadowColor: (e == null ? void 0 : e[a]) || f.color[a] || "#FFF",
          shadowBlur: 8,
          formatter(i) {
            return "";
          }
        }
      },
      t == null ? void 0 : t.bar
    ), C = (e) => {
      var i;
      let a = [];
      return e ? a = (i = t.data) != null && i.length ? t.data.map((r) => r[e]) : [] : a = t.data.map((r, n) => n), a;
    }, w = () => {
      var g, p, D, O, T, R;
      const { color: e, series: a } = t.options;
      let i = [], r = [], n = [];
      a ? i = a.map(($, c) => o(A(e, c), $)) : (g = t.data) != null && g.length && ((D = (p = t.seriesSetting) == null ? void 0 : p.data) != null && D.length) && (i = (O = t.seriesSetting) == null ? void 0 : O.data.map(($, c) => {
        var z, I, k, j, E;
        return o({
          ...A(e, c),
          name: ((z = m.chartData[c]) == null ? void 0 : z.name) || "",
          data: ((I = m.chartData[c]) == null ? void 0 : I.data) || []
        }, (E = (j = (k = t.seriesSetting) == null ? void 0 : k.data) == null ? void 0 : j[c]) == null ? void 0 : E.style);
      }), r = C((T = t.seriesSetting) == null ? void 0 : T.yAxisLeftLabel), n = C((R = t.seriesSetting) == null ? void 0 : R.yAxisRightLabel)), f.yAxis[1].axisLabel.formatter = `{value} ${t.unit}`;
      let s = o(
        {
          series: i,
          yAxis: [{
            data: r
          }, {
            data: n
          }]
        },
        f
      );
      s = o(s, t.options), u.clear(), u.setOption(s);
    }, L = () => {
      q(() => {
        u.resize();
      });
    };
    return W(() => {
      v(), b.value && (u = Y.init(b.value), S(), w(), u.on("legendselectchanged", (e) => {
        M("legend-select", e);
      })), window.addEventListener("resize", L);
    }), K(() => {
      window.removeEventListener("resize", L);
    }), F(
      [() => t.options, () => t.data, () => t.seriesSetting],
      () => {
        S(), w();
      },
      {
        deep: !0
      }
    ), F(
      [() => t.unit, () => t.bar],
      () => {
        w();
      }
    ), F(
      () => t.cStyle,
      () => {
        v(), q(() => {
          u.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      ...X(m),
      chartId: b,
      dataProcessing: S,
      curStyle: l,
      checkEmpty: P,
      domOnlyClass: d
    };
  }
});
export {
  ne as default
};
