// @ths/design/es/components/regional-multiple-selector/src/index.vue2.js
import { defineComponent as Je, computed as we, reactive as Qe, onMounted as Ve, watch as I, toRefs as We } from "vue";
import Be from "lodash/merge";
import { getDomRandomClass as Xe, setStyle as Ze, getCssPropertyString as n } from "../../../utils/index.js";
const Fe = {
  nodePid: "nodePid",
  nodeId: "nodeId",
  pinyin: "pinyin",
  nodeName: "nodeName",
  pinyinAll: "pinyinAll"
}, tt = Je({
  name: "TRegionalMultipleSelector",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        /* 标签容器 */
        tagContainer: {
          default: {}
        },
        /* 标签 */
        tag: {
          default: {}
        },
        /* 标签-名称 */
        tagName: {
          default: {}
        },
        /* 标签-删除按钮 */
        tagDeleteIcon: {
          default: {},
          hover: {}
        },
        /* 拼音字母容器 */
        areaLitter: {
          default: {}
        },
        /* 拼音字母 */
        litter: {
          default: {},
          hover: {},
          active: {}
        },
        /* 区域选择容器 */
        regionContainer: {
          default: {}
        },
        /* 区域第一层容器 */
        firstLevel: {
          default: {}
        },
        /* 区域选择 - 省份首字母 */
        regionSelectFirstLitter: {
          default: {}
        },
        /* 区域选择 - 右侧模块 */
        regionSelectRightBlock: {
          default: {}
        },
        /* 区域选择 - 区域容器 */
        regionMain: {
          default: {}
        },
        /* 区域选择 - 省份 */
        province: {
          default: {},
          hover: {},
          active: {}
        },
        /* 区域选择 - 城市容器 */
        cityContainer: {
          default: {}
        },
        /* 区域选择 - 城市 */
        city: {
          default: {}
        },
        /* 区域选择 - 城市名称 */
        cityName: {
          default: {},
          hover: {},
          active: {}
        },
        /* 区域选择 - 区县容器 */
        districtContainer: {
          default: {}
        },
        /* 区域选择 - 区县 */
        district: {
          default: {},
          hover: {},
          active: {}
        }
      })
    },
    // 选中项
    selected: {
      type: Array,
      default: () => []
    },
    // 区域数据
    data: {
      type: Array,
      default: () => []
    },
    /* 数据别名 */
    alias: {
      type: Object,
      default: () => Fe
    }
  },
  emits: ["change"],
  setup(s, { emit: C }) {
    const t = we(() => s.cStyle), i = Xe("regional-multiple-selector"), l = we(() => Be({}, Fe, s.alias)), o = Qe({
      // 当前选中的首字母
      activeLetter: "",
      mapList: {},
      dataList: {},
      // 区域选择的显隐
      isShowBox: !1,
      // 选中区域
      radio1: "xz",
      // 区域列表dom
      list: null,
      // 默认选中类型
      selectedData: {}
    }), Ke = () => {
      let e = "";
      return t.value.wrapper && (e += Ze(t.value.wrapper, `.${i}`)), e;
    }, xe = (e) => {
      var c;
      o.activeLetter = e;
      const a = document.getElementById(e);
      a && ((c = o.list) == null || c.scroll(0, (a == null ? void 0 : a.offsetTop) - o.list.offsetTop));
    };
    function He(e) {
      const a = {};
      e.forEach((r) => {
        o.mapList[r == null ? void 0 : r[l.value.nodeId]] = { ...r, children: [] };
      });
      const c = e.filter((r) => r[l.value.nodePid] === "000000000000");
      return c.sort((r, d) => r[l.value.nodeId].localeCompare(d[l.value.nodeId])), c.forEach((r) => {
        var g;
        const d = (g = r[l.value.pinyin]) == null ? void 0 : g.charAt(0).toUpperCase(), u = {
          ...r,
          [l.value.pinyinAll]: r[l.value.nodeId],
          [l.value.pinyin]: d,
          children: []
        };
        e.filter((v) => v[l.value.nodePid] === r[l.value.nodeId]).forEach((v) => {
          var m;
          const p = {
            ...v,
            [l.value.pinyinAll]: v[l.value.nodeId],
            [l.value.pinyin]: (m = v[l.value.pinyin]) == null ? void 0 : m.charAt(0).toUpperCase(),
            children: []
          };
          u.children.push(p), e.filter((f) => f[l.value.nodePid] === v[l.value.nodeId]).forEach((f) => {
            var y;
            const h = {
              ...f,
              [l.value.pinyinAll]: f[l.value.nodeId],
              pinyin: (y = f[l.value.pinyin]) == null ? void 0 : y.charAt(0).toUpperCase()
            };
            p.children || (p.children = []), p.children.push(h);
          });
        }), a[d] = Be(a[d], {
          [r[l.value.nodeId]]: u
        });
      }), a;
    }
    const ke = [
      "A",
      "B",
      "C",
      "D",
      "E",
      "F",
      "G",
      "H",
      "I",
      "J",
      "K",
      "L",
      "M",
      "N",
      "O",
      "P",
      "Q",
      "R",
      "S",
      "T",
      "U",
      "V",
      "W",
      "X",
      "Y",
      "Z",
      "港澳"
    ], Ye = (e) => {
      var c, r;
      if (typeof e[l.value.nodeId] != "string")
        return;
      o.selectedData[e[l.value.nodeId]] = e;
      const a = l.value.nodePid;
      e[a] && ((c = o.mapList[e[a]]) != null && c[a] && delete o.selectedData[(r = o.mapList[e[a]]) == null ? void 0 : r[a]], delete o.selectedData[e[a]]), e != null && e.children && e.children.forEach((d) => {
        d != null && d.children && d.children.forEach((u) => {
          delete o.selectedData[u[l.value.nodeId]];
        }), delete o.selectedData[d[l.value.nodeId]];
      }), C("change", Object.values(o.selectedData));
    }, qe = (e) => {
      delete o.selectedData[e], C("change", Object.values(o.selectedData));
    }, ze = () => {
      var a, c, r, d, u, $, g, v, p, D, m, f, h, y, S, A, E, P, O, _, T, U, M, R, w, B, F, K, x, H, k, Y, q, z, G, J, Q, V, W, X, Z, j, b, ee, te, le, ae, ie, ne, re, oe, ce, de, ue, se, ve, fe, pe, ge, me, ye, he, $e, De, Ie, Ce, Le, Ne, Se, Ae, Ee, Pe, Oe, _e, Te, Ue, Me, Re;
      let e = "";
      return (c = (a = t.value) == null ? void 0 : a.tagContainer) != null && c.default && (e += `.${i}.t-regional-multiple-selector .selected-container {
          ${n((r = t.value) == null ? void 0 : r.tagContainer.default)}
        }`), (u = (d = t.value) == null ? void 0 : d.tag) != null && u.default && (e += `.${i}.t-regional-multiple-selector .selected-container .selected-item {
          ${n(($ = t.value) == null ? void 0 : $.tag.default)}
        }`), (v = (g = t.value) == null ? void 0 : g.tagName) != null && v.default && (e += `.${i}.t-regional-multiple-selector .selected-container .selected-item__name {
          ${n((p = t.value) == null ? void 0 : p.tagName.default)}
        }`), (m = (D = t.value) == null ? void 0 : D.tagDeleteIcon) != null && m.default && (e += `.${i}.t-regional-multiple-selector .selected-container .selected-item__delete {
          ${n((f = t.value) == null ? void 0 : f.tagDeleteIcon.default)}
        }`), (y = (h = t.value) == null ? void 0 : h.tagDeleteIcon) != null && y.hover && (e += `.${i}.t-regional-multiple-selector .selected-container .selected-item__delete:hover {
          ${n((S = t.value) == null ? void 0 : S.tagDeleteIcon.hover)}
        }`), (E = (A = t.value) == null ? void 0 : A.areaLitter) != null && E.default && (e += `.${i}.t-regional-multiple-selector .area-litter {
          ${n((P = t.value) == null ? void 0 : P.areaLitter.default)}
        }`), (_ = (O = t.value) == null ? void 0 : O.litter) != null && _.default && (e += `.${i}.t-regional-multiple-selector .area-litter .litter {
          ${n((T = t.value) == null ? void 0 : T.litter.default)}
        }`), (M = (U = t.value) == null ? void 0 : U.litter) != null && M.active && (e += `.${i}.t-regional-multiple-selector .area-litter .litter.active {
          ${n((R = t.value) == null ? void 0 : R.litter.active)}
        }`), (B = (w = t.value) == null ? void 0 : w.litter) != null && B.hover && (e += `.${i}.t-regional-multiple-selector .area-litter .litter:hover {
          ${n((F = t.value) == null ? void 0 : F.litter.hover)}
        }`), (x = (K = t.value) == null ? void 0 : K.regionContainer) != null && x.default && (e += `.${i}.t-regional-multiple-selector .region-container {
          ${n((H = t.value) == null ? void 0 : H.regionContainer.default)}
        }`), (Y = (k = t.value) == null ? void 0 : k.firstLevel) != null && Y.default && (e += `.${i}.t-regional-multiple-selector .first-level {
          ${n((q = t.value) == null ? void 0 : q.firstLevel.default)}
        }`), (G = (z = t.value) == null ? void 0 : z.regionSelectFirstLitter) != null && G.default && (e += `.${i}.t-regional-multiple-selector .initial {
          ${n((J = t.value) == null ? void 0 : J.regionSelectFirstLitter.default)}
        }`), (V = (Q = t.value) == null ? void 0 : Q.regionSelectRightBlock) != null && V.default && (e += `.${i}.t-regional-multiple-selector .full-width {
          ${n((W = t.value) == null ? void 0 : W.regionSelectRightBlock.default)}
        }`), (Z = (X = t.value) == null ? void 0 : X.regionMain) != null && Z.default && (e += `.${i}.t-regional-multiple-selector .region-main {
          ${n((j = t.value) == null ? void 0 : j.regionMain.default)}
        }`), (ee = (b = t.value) == null ? void 0 : b.province) != null && ee.default && (e += `.${i}.t-regional-multiple-selector .province {
          ${n((te = t.value) == null ? void 0 : te.province.default)}
        }`), (ae = (le = t.value) == null ? void 0 : le.province) != null && ae.active && (e += `.${i}.t-regional-multiple-selector .province.active {
          ${n((ie = t.value) == null ? void 0 : ie.province.active)}
        }`), (re = (ne = t.value) == null ? void 0 : ne.province) != null && re.hover && (e += `.${i}.t-regional-multiple-selector .province:hover {
          ${n((oe = t.value) == null ? void 0 : oe.province.hover)}
        }`), (de = (ce = t.value) == null ? void 0 : ce.cityContainer) != null && de.default && (e += `.${i}.t-regional-multiple-selector .city-container {
          ${n((ue = t.value) == null ? void 0 : ue.cityContainer.default)}
        }`), (ve = (se = t.value) == null ? void 0 : se.city) != null && ve.default && (e += `.${i}.t-regional-multiple-selector .city {
          ${n((fe = t.value) == null ? void 0 : fe.city.default)}
        }`), (ge = (pe = t.value) == null ? void 0 : pe.cityName) != null && ge.default && (e += `.${i}.t-regional-multiple-selector .city-name {
          ${n((me = t.value) == null ? void 0 : me.cityName.default)}
        }`), (he = (ye = t.value) == null ? void 0 : ye.cityName) != null && he.active && (e += `.${i}.t-regional-multiple-selector .city.active .city-name {
          ${n(($e = t.value) == null ? void 0 : $e.cityName.active)}
        }`), (Ie = (De = t.value) == null ? void 0 : De.cityName) != null && Ie.hover && (e += `.${i}.t-regional-multiple-selector .city-name:hover {
          ${n((Ce = t.value) == null ? void 0 : Ce.cityName.hover)}
        }`), (Ne = (Le = t.value) == null ? void 0 : Le.districtContainer) != null && Ne.default && (e += `.${i}.t-regional-multiple-selector .district-container {
          ${n((Se = t.value) == null ? void 0 : Se.districtContainer.default)}
        }`), (Ee = (Ae = t.value) == null ? void 0 : Ae.district) != null && Ee.default && (e += `.${i}.t-regional-multiple-selector .district {
          ${n((Pe = t.value) == null ? void 0 : Pe.district.default)}
        }`), (_e = (Oe = t.value) == null ? void 0 : Oe.district) != null && _e.active && (e += `.${i}.t-regional-multiple-selector .district.active {
          ${n((Te = t.value) == null ? void 0 : Te.district.active)}
        }`), (Me = (Ue = t.value) == null ? void 0 : Ue.district) != null && Me.hover && (e += `.${i}.t-regional-multiple-selector .district:hover {
          ${n((Re = t.value) == null ? void 0 : Re.district.hover)}
        }`), e;
    }, L = () => {
      const e = document.querySelector(`.${i} .style-container`);
      e && (e.innerHTML = `<style>${Ke()}${ze()}</style>`);
    }, N = () => {
      var e;
      o.selectedData = {}, (e = s.selected) != null && e.length && s.selected.forEach((a) => {
        const c = a == null ? void 0 : a[l.value.nodeId], r = a == null ? void 0 : a[l.value.nodeName];
        s.data.find((u) => (u == null ? void 0 : u[l.value.nodeId]) === c && (a == null ? void 0 : a[l.value.nodeName]) === r) && (o.selectedData[c] = {
          [l.value.nodeName]: r,
          [l.value.nodeId]: c
        });
      });
    }, Ge = () => {
      const e = {
        nodePid: !1,
        nodeId: !1,
        pinyin: !1,
        nodeName: !1
      };
      return Object.entries(l.value).forEach(([a, c]) => {
        Object.keys(e).includes(a) && (e[a] = s.data.some((r) => c in r));
      }), Object.values(e).every((a) => a);
    };
    return Ve(() => {
      L();
    }), I(() => s.cStyle, () => {
      L();
    }, { deep: !0 }), I(() => s.selected, () => {
      N();
    }, {
      immediate: !0,
      deep: !0
    }), I(() => [s.data, l.value], () => {
      Ge() ? (o.dataList = He(s.data), N()) : (o.dataList = {}, o.selectedData = {});
    }, {
      immediate: !0,
      deep: !0
    }), {
      filterList: ke,
      ...We(o),
      curStyle: t,
      domOnlyClass: i,
      aliasKey: l,
      selectLitter: xe,
      handleSelect: Ye,
      handleDelete: qe
    };
  }
});
export {
  tt as default
};
