// @ths/design/es/components/regional-multiple-selector/src/index.vue.js
import y from "./index.vue2.js";
import { openBlock as s, createElementBlock as t, normalizeClass as v, createElementVNode as a, Fragment as r, renderList as c, toDisplayString as u, withModifiers as m } from "vue";
import "./index.vue3.js";
import h from "../../../_virtual/_plugin-vue_export-helper.js";
const K = { class: "selected-container" }, f = {
  key: 0,
  class: "empty"
}, k = ["title"], C = ["onClick"], _ = { class: "area-litter" }, N = ["onClick"], $ = {
  ref: "list",
  class: "region-container"
}, D = ["id"], I = { class: "initial" }, g = { class: "full-width" }, L = ["id"], S = ["title", "onClick"], E = { class: "city-container" }, M = ["title", "onClick"], j = { class: "district-container" }, w = ["title", "onClick"];
function B(e, p, O, b, z, F) {
  return s(), t("div", {
    class: v(["t-regional-multiple-selector", e.domOnlyClass])
  }, [
    a("div", K, [
      Object.keys(e.selectedData).length ? (s(!0), t(r, { key: 1 }, c(e.selectedData, (o, l) => (s(), t("p", {
        key: l,
        class: "selected-item"
      }, [
        a("span", {
          class: "selected-item__name",
          title: o[e.aliasKey.nodeName]
        }, u(o[e.aliasKey.nodeName]), 9, k),
        a("i", {
          class: "el-icon-close selected-item__delete",
          onClick: (i) => e.handleDelete(o[e.aliasKey.nodeId])
        }, null, 8, C)
      ]))), 128)) : (s(), t("p", f, " 暂无数据 "))
    ]),
    a("div", _, [
      (s(!0), t(r, null, c(e.dataList, (o, l) => (s(), t("a", {
        key: l,
        href: "javascript:void(0)",
        class: v(["litter", e.activeLetter === l ? "active" : ""]),
        onClick: (i) => e.selectLitter(l)
      }, u(l), 11, N))), 128))
    ]),
    a("div", $, [
      (s(!0), t(r, null, c(e.dataList, (o, l) => (s(), t("div", {
        id: l,
        key: l,
        class: "first-level"
      }, [
        a("span", I, u(l), 1),
        a("div", g, [
          (s(!0), t(r, null, c(o, (i) => (s(), t("div", {
            id: i[e.aliasKey.nodeId],
            key: i[e.aliasKey.nodeId],
            class: "region-main"
          }, [
            a("span", {
              class: v(["province", e.selectedData[i[e.aliasKey.nodeId]] ? "active" : ""]),
              title: i[e.aliasKey.nodeName],
              onClick: (n) => e.handleSelect(i)
            }, u(i[e.aliasKey.nodeName]), 11, S),
            a("div", E, [
              (s(!0), t(r, null, c(i.children, (n) => (s(), t("div", {
                key: n[e.aliasKey.nodeId],
                class: v(["city", e.selectedData[n[e.aliasKey.nodeId]] ? "active" : ""]),
                onMouseover: p[0] || (p[0] = m(() => {
                }, ["stop"]))
              }, [
                a("span", {
                  class: "city-name",
                  title: n[e.aliasKey.nodeName],
                  onClick: (d) => e.handleSelect(n)
                }, u(n[e.aliasKey.nodeName]), 9, M),
                a("div", j, [
                  (s(!0), t(r, null, c(n.children, (d) => (s(), t("span", {
                    key: d[e.aliasKey.nodeId],
                    title: d[e.aliasKey.nodeName],
                    class: v(["district", e.selectedData[d[e.aliasKey.nodeId]] ? "active" : ""]),
                    onClick: (R) => e.handleSelect(d)
                  }, u(d[e.aliasKey.nodeName]), 11, w))), 128))
                ])
              ], 34))), 128))
            ])
          ], 8, L))), 128))
        ])
      ], 8, D))), 128))
    ], 512),
    p[1] || (p[1] = a("div", { class: "style-container" }, null, -1))
  ], 2);
}
const H = /* @__PURE__ */ h(y, [["render", B]]);
export {
  H as default
};
