// @ths/design/es/components/chart-custom-stacked/src/index.vue2.js
import { defineComponent as R, computed as B, ref as z, reactive as H, onMounted as U, nextTick as C, onUnmounted as K, watch as w, toRefs as X } from "vue";
import * as G from "echarts";
import p from "lodash/merge";
import x from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as J, DEFAULT_OPTIONS_AXIS as Q } from "../../../utils/constant.js";
import { getDomRandomClass as V, getCssPropertyString as i, setStyle as Y } from "../../../utils/index.js";
const Z = p(x(J), x(Q)), $ = p(
  x(Z),
  {
    grid: {
      top: 0,
      bottom: 0,
      left: 0,
      right: 0
    },
    xAxis: {
      type: "value",
      show: !1
    },
    yAxis: {
      type: "category",
      show: !1
    },
    legend: {
      show: !1
    },
    tooltip: {
      trigger: "axis",
      show: !0,
      axisPointer: {
        type: "shadow"
      }
    }
  }
), ne = R({
  name: "TChartCustomStacked",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 单项容器
        itemWrapper: {
          default: {},
          hover: {}
        },
        // 上部分容器
        topContainer: {
          default: {}
        },
        // 下部分容器
        bottomContainer: {
          default: {}
        },
        // 序号
        serial: {
          default: {}
        },
        // 名称
        seriesName: {
          default: {}
        },
        // 数量
        number: {
          default: {}
        },
        // 单位
        unit: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => $,
      required: !1
    },
    // 传入数组数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 是否显示背景
    showItemBg: {
      type: Boolean,
      default: !1
    },
    // 背景条样式
    backgroundStyle: {
      type: Object,
      default: () => ({})
    }
  },
  // 方法名定义
  emits: ["com-click", "legend-select"],
  setup(a, { emit: D }) {
    const e = B(() => a.cStyle), r = V("custom-stacked"), F = z(null), y = z([]), P = (t) => {
      t && y.value.push(t);
    }, f = [], o = H({
      // 图表数据
      chartData: []
    }), A = () => {
      var t, n;
      o.chartData = [], (n = (t = a.seriesSetting) == null ? void 0 : t.data) != null && n.length && a.data ? a.seriesSetting.data.forEach((m, b) => {
        const { name: S, dataKey: l } = m;
        a.data.forEach((s, u) => {
          typeof o.chartData[u] != "object" ? o.chartData[u] = {
            dataName: s[a.seriesSetting.yAxisLabel],
            data: [s[l]]
          } : o.chartData[u].data[o.chartData[u].data.length] = s[l];
        });
      }) : o.chartData = [];
    }, q = () => {
      let t = "";
      return e.value.wrapper && e.value.wrapper.hover && (t += `
            .${r}:hover {
              ${e.value.wrapper && e.value.wrapper.hover && i(e.value.wrapper.hover)}
            }
          `), e.value.itemWrapper && (e.value.itemWrapper.default && (t += `
            .${r} .chart-container {
              ${e.value.itemWrapper && e.value.itemWrapper.default && i(e.value.itemWrapper.default)}
            }
          `), e.value.itemWrapper.hover && (t += `
            .${r} .chart-container:hover {
              ${e.value.itemWrapper && e.value.itemWrapper.hover && i(e.value.itemWrapper.hover)}
            }
          `)), e.value.topContainer && e.value.topContainer.default && (t += `
            .${r} .chart-container .chart-container-top {
              ${e.value.topContainer && e.value.topContainer.default && i(e.value.topContainer.default)}
            }
          `), e.value.serial && e.value.serial.default && (t += `
            .${r} .chart-container .chart-container-top .top-left-serial {
              ${e.value.serial && e.value.serial.default && i(e.value.serial.default)}
            }
          `), e.value.seriesName && e.value.seriesName.default && (t += `
            .${r} .chart-container .chart-container-top .top-left-name {
              ${e.value.seriesName && e.value.seriesName.default && i(e.value.seriesName.default)}
            }
          `), e.value.number && e.value.number.default && (t += `
            .${r} .chart-container .chart-container-top .top-right-name {
              ${e.value.number && e.value.number.default && i(e.value.number.default)}
            }
          `), e.value.unit && e.value.unit.default && (t += `
            .${r} .chart-container .chart-container-top .top-right-unit {
              ${e.value.unit && e.value.unit.default && i(e.value.unit.default)}
            }
          `), e.value.bottomContainer && e.value.bottomContainer.default && (t += `
            .${r} .chart-container .chart-container-bottom {
              ${e.value.bottomContainer && e.value.bottomContainer.default && i(e.value.bottomContainer.default)}
            }
          `), t;
    }, M = () => {
      let t = "";
      return e.value.wrapper && (t += Y(e.value.wrapper, `.${r}`)), t;
    }, v = () => {
      const t = document.querySelector(`.${r} .style-container`);
      t && (t.innerHTML = `<style>${M()}${q()}</style>`);
    }, g = () => {
      var b, S;
      const { color: t, series: n } = a.options, m = [];
      n || a.data && ((S = (b = a.seriesSetting) == null ? void 0 : b.data) != null && S.length) && o.chartData.forEach((l, s) => {
        var h, E, W, N;
        const u = (h = a.seriesSetting) == null ? void 0 : h.data.map((k, c) => {
          var L, T, I, _, j;
          return p(
            {
              barWidth: 12,
              name: (T = (L = a.seriesSetting) == null ? void 0 : L.data[c]) == null ? void 0 : T.name,
              stack: "total",
              type: "bar",
              label: {
                show: !1
              },
              itemStyle: {
                color: t && t[c] || $.color[c] || "#FFF"
              },
              data: l.data.length ? [l.data[c]] : []
            },
            (j = (_ = (I = a.seriesSetting) == null ? void 0 : I.data) == null ? void 0 : _[c]) == null ? void 0 : j.style,
            k
          );
        });
        let d = [];
        (E = a.seriesSetting) != null && E.yAxisLabel ? d = (W = a.data) != null && W.length ? [a.data[s][(N = a.seriesSetting) == null ? void 0 : N.yAxisLabel]] : [] : d = a.data.map((k, c) => c), m.push({
          datas: u,
          yAxisData: d
        });
      }), f.forEach((l) => {
        l.clear();
      });
      try {
        f.forEach((l, s) => {
          var d, h;
          const u = p(
            {
              series: (d = m[s]) == null ? void 0 : d.datas,
              yAxis: {
                data: (h = m[s]) == null ? void 0 : h.yAxisData
              }
            },
            $,
            a.options,
            {
              legend: {
                show: !1
              }
            }
          );
          l.setOption(u);
        });
      } catch (l) {
        console.log(l);
      }
    }, O = () => {
      C(() => {
        f.forEach((t) => {
          t.resize();
        });
      });
    };
    return U(() => {
      A(), C(() => {
        v(), y.value.length && (y.value.forEach((t, n) => {
          f[n] = G.init(t);
        }), g(), f.forEach((t) => {
          t.on("click", (n) => {
            D("com-click", n);
          }), t.on("legendselectchanged", (n) => {
            D("legend-select", n);
          });
        }), window.addEventListener("resize", O));
      });
    }), K(() => {
      window.removeEventListener("resize", O);
    }), w(
      [() => a.backgroundStyle],
      () => {
        v(), g();
      },
      {
        deep: !0
      }
    ), w(
      [
        () => a.options,
        () => a.data,
        () => a.seriesSetting,
        () => a.showItemBg
      ],
      () => {
        A(), v(), g();
      },
      {
        deep: !0
      }
    ), w(
      () => a.cStyle,
      () => {
        v(), C(() => {
          f.forEach((t) => {
            t.resize();
          });
        });
      },
      {
        deep: !0
      }
    ), {
      curStyle: e,
      ...X(o),
      setRefs: P,
      domOnlyClass: r,
      chartId: F
    };
  }
});
export {
  ne as default
};
