// @ths/design/es/components/aqi/src/index.vue2.js
import { defineComponent as f, computed as p, reactive as y, watch as r, onMounted as _, toRefs as b } from "vue";
import q from "lodash/merge";
import c from "lodash/cloneDeep";
import { getDomRandomClass as v, setStyle as s } from "../../../utils/index.js";
const d = {
  value: "value",
  label: "label",
  title: "title",
  primaryPollutants: "primaryPollutants",
  desc: "desc",
  updateTime: "updateTime"
}, $ = f({
  name: "TAqi",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        value: {
          default: {}
        },
        // 优
        label: {
          default: {}
        },
        // 标题
        title: {
          default: {}
        },
        // 首要污染物
        primaryPollutants: {
          default: {}
        },
        // 描述
        desc: {
          default: {}
        },
        // 更新的时间
        updateTime: {
          default: {}
        }
      })
    },
    // 数据
    data: {
      type: Object,
      default: () => ({
        value: "",
        label: "",
        title: "",
        primaryPollutants: "",
        desc: "",
        updateTime: ""
      })
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => c(d),
      require: !1
    }
  },
  setup(a, { emit: S }) {
    const l = p(() => a.cStyle), i = v("aqi"), n = {
      wrapper: ".t-aqi",
      value: ".aqi-content__middle-value",
      label: ".aqi-content__middle-label",
      title: ".aqi-content__top-title",
      primaryPollutants: ".aqi-content__footer-primaryPollutants",
      desc: ".aqi-content__footer-desc",
      updateTime: ".aqi-content__top-updateTime"
    }, o = y({
      curData: a.data
    }), m = () => {
      let e = "";
      return Object.keys(n).forEach((t) => {
        l.value[t] && t === "wrapper" ? e += s(l.value[t], `.${i}${n[t]}`) : l.value[t] && (e += s(l.value[t], `.${i} ${n[t]}`));
      }), e;
    }, u = () => {
      const e = document.querySelector(`.${i} .style-container`);
      e && (e.innerHTML = `<style>${m()}</style>`);
    };
    return r(() => a.cStyle, () => {
      u();
    }), r(() => [a.data, a.alias], () => {
      const e = q(c(d), a.alias);
      Object.keys(e).forEach((t) => {
        o.curData[t] = a.data[e[t]];
      });
    }, {
      deep: !0,
      immediate: !0
    }), _(() => {
      u();
    }), {
      ...b(o),
      curStyle: l,
      domOnlyClass: i,
      curStyleMap: n
    };
  }
});
export {
  $ as default
};
