// @ths/design/es/components/video-monitor/src/index.vue.js
import M from "./index.vue2.js";
import { resolveComponent as d, openBlock as a, createElementBlock as r, normalizeClass as u, createElementVNode as n, normalizeStyle as i, createCommentVNode as s, withDirectives as C, vShow as S, createBlock as v, withCtx as p, Fragment as V, renderList as $, createVNode as t, toDisplayString as b, createTextVNode as P } from "vue";
import "./index.vue3.js";
import f from "../../../_virtual/_plugin-vue_export-helper.js";
const T = {
  key: 0,
  class: "loading__icon"
}, R = { class: "video-container__map" }, B = ["src"], F = {
  key: 1,
  class: "btn-turn video-container__steering-wheel"
}, I = {
  key: 2,
  class: "tip"
}, E = {
  key: 3,
  class: "tip"
}, N = { class: "toolbar" }, D = { class: "toolbar__left" }, U = ["src"], z = ["src"], G = ["src"], L = ["src"], O = ["src"], H = { class: "toolbar__right" }, J = {
  key: 0,
  class: "volume__panel"
}, W = ["src"], j = ["src"], q = ["src"], A = { class: "action-container" }, K = {
  key: 1,
  class: "progress-bar"
}, Q = ["src"], X = ["src"], Y = { class: "line" };
function Z(e, l, _, x, ee, le) {
  var g, k;
  const c = d("el-option"), h = d("el-select"), m = d("el-slider"), y = d("el-radio-button"), w = d("el-radio-group");
  return a(), r("div", {
    class: u(["t-video-monitor", e.domOnlyClass])
  }, [
    n("div", {
      class: "video-container",
      style: i((k = (g = e.cStyle) == null ? void 0 : g.video) == null ? void 0 : k.default)
    }, [
      n("div", {
        id: "player",
        class: u(["video-container__player", { loading: e.loading }])
      }, null, 2),
      e.loading ? (a(), r("div", T)) : s("", !0),
      C(n("div", R, [
        n("img", {
          class: "icon location-icon",
          src: e.iconPath.location,
          alt: "",
          width: 60,
          style: i({ transform: `rotate(${e.angle}deg)` })
        }, null, 12, B),
        l[20] || (l[20] = n("div", {
          id: "baiduMap",
          class: "map-container"
        }, null, -1))
      ], 512), [
        [S, e.playerMode === e.PlayerMode.RealTime && e.playerType === e.PlayerMode.RealTime]
      ]),
      e.playerMode === e.PlayerMode.RealTime && e.playerType === e.PlayerMode.RealTime ? (a(), r("div", F, [
        n("p", {
          class: "btn-turn__left direction-arrow",
          style: i({ backgroundImage: `url(${e.iconPath.arrow}` }),
          onClick: l[0] || (l[0] = (o) => e.handleSteer("LEFT"))
        }, [
          n("span", {
            class: "direction-arrow--hover",
            style: i({ backgroundImage: `url(${e.iconPath.circle}` })
          }, null, 4)
        ], 4),
        n("p", {
          class: "btn-turn__up direction-arrow",
          style: i({ backgroundImage: `url(${e.iconPath.arrow}` }),
          onClick: l[1] || (l[1] = (o) => e.handleSteer("UP"))
        }, [
          n("span", {
            class: "direction-arrow--hover",
            style: i({ backgroundImage: `url(${e.iconPath.circle}` })
          }, null, 4)
        ], 4),
        n("p", {
          class: "btn-turn__right direction-arrow",
          style: i({ backgroundImage: `url(${e.iconPath.arrow}` }),
          onClick: l[2] || (l[2] = (o) => e.handleSteer("RIGHT"))
        }, [
          n("span", {
            class: "direction-arrow--hover",
            style: i({ backgroundImage: `url(${e.iconPath.circle}` })
          }, null, 4)
        ], 4),
        n("p", {
          class: "btn-turn__down direction-arrow",
          style: i({ backgroundImage: `url(${e.iconPath.arrow}` }),
          onClick: l[3] || (l[3] = (o) => e.handleSteer("DOWN"))
        }, [
          n("span", {
            class: "direction-arrow--hover",
            style: i({ backgroundImage: `url(${e.iconPath.circle}` })
          }, null, 4)
        ], 4)
      ])) : s("", !0),
      e.recording ? (a(), r("p", I, " 正在录制视频 ")) : s("", !0),
      e.talking ? (a(), r("p", E, " 正在对讲 ")) : s("", !0)
    ], 4),
    n("div", N, [
      n("div", D, [
        n("span", null, [
          n("img", {
            width: "48",
            class: "btn-snapshot icon",
            src: e.iconPath.photo,
            onClick: l[4] || (l[4] = (o) => e.capture("JPEG"))
          }, null, 8, U)
        ]),
        e.recording ? s("", !0) : (a(), r("span", {
          key: 0,
          class: u({ "icon-disabled": e.recordingDisabled || e.talking })
        }, [
          n("img", {
            width: "48",
            class: "btn-record-video icon",
            src: e.iconPath.record,
            onClick: l[5] || (l[5] = (o) => e.recordStart("MP4"))
          }, null, 8, z)
        ], 2)),
        e.recording ? (a(), r("img", {
          key: 1,
          width: "48",
          class: "btn-record-video icon",
          src: e.iconPath.stopRecord,
          onClick: l[6] || (l[6] = (...o) => e.recordStop && e.recordStop(...o))
        }, null, 8, G)) : s("", !0),
        e.playerMode === e.PlayerMode.RealTime && !e.talking && e.playerType === e.PlayerMode.RealTime ? (a(), r("span", {
          key: 2,
          class: u({ "icon-disabled": e.intercomDisabled || e.recording })
        }, [
          n("img", {
            width: "48",
            class: "btn-speech icon",
            src: e.iconPath.talk,
            onClick: l[7] || (l[7] = (...o) => e.talkStart && e.talkStart(...o))
          }, null, 8, L)
        ], 2)) : s("", !0),
        e.playerMode === e.PlayerMode.RealTime && e.talking ? (a(), r("img", {
          key: 3,
          width: "48",
          class: "btn-speech icon",
          src: e.iconPath.stopTalk,
          onClick: l[8] || (l[8] = (...o) => e.talkStop && e.talkStop(...o))
        }, null, 8, O)) : s("", !0),
        e.playerMode === e.PlayerMode.Playback ? (a(), v(h, {
          key: 4,
          modelValue: e.playback.rate,
          "onUpdate:modelValue": l[9] || (l[9] = (o) => e.playback.rate = o),
          size: "small",
          class: "m-2",
          placeholder: "Select",
          placement: "top"
        }, {
          default: p(() => [
            (a(), r(V, null, $([
              { label: "正常", value: 0 },
              { label: "x1", value: 1 },
              { label: "x2", value: 2 }
            ], (o) => t(c, {
              key: o.value,
              label: o.label,
              value: o.value
            }, null, 8, ["label", "value"])), 64))
          ]),
          _: 1
        }, 8, ["modelValue"])) : s("", !0)
      ]),
      n("div", H, [
        n("div", {
          class: "volume-container",
          onMouseenter: l[13] || (l[13] = (o) => e.volumePanelVisible = !0),
          onMouseleave: l[14] || (l[14] = (o) => e.volumePanelVisible = !1)
        }, [
          e.volumePanelVisible ? (a(), r("div", J, [
            t(m, {
              modelValue: e.volume,
              "onUpdate:modelValue": l[10] || (l[10] = (o) => e.volume = o),
              max: 100,
              min: 0,
              vertical: "",
              height: "200px",
              onChange: e.setVolume
            }, null, 8, ["modelValue", "onChange"]),
            n("span", null, b(e.volume) + "%", 1)
          ])) : s("", !0),
          e.muted ? s("", !0) : (a(), r("img", {
            key: 1,
            width: "48",
            class: "volume__btn icon",
            src: e.iconPath.sound,
            onClick: l[11] || (l[11] = (...o) => e.closeSound && e.closeSound(...o))
          }, null, 8, W)),
          e.muted ? (a(), r("img", {
            key: 2,
            width: "48",
            class: "volume__btn icon",
            src: e.iconPath.soundNone,
            onClick: l[12] || (l[12] = (...o) => e.openSound && e.openSound(...o))
          }, null, 8, j)) : s("", !0)
        ], 32),
        n("img", {
          width: "48",
          class: "btn-full-screen icon",
          src: e.iconPath.fullScreen,
          onClick: l[15] || (l[15] = (...o) => e.wholeFullScreen && e.wholeFullScreen(...o))
        }, null, 8, q)
      ])
    ]),
    n("div", A, [
      e.playerType === e.PlayerMode.RealTime ? (a(), v(w, {
        key: 0,
        modelValue: e.playerMode,
        "onUpdate:modelValue": l[16] || (l[16] = (o) => e.playerMode = o),
        class: "btn-group",
        size: "small",
        onChange: e.switchPlayerMode
      }, {
        default: p(() => [
          t(y, {
            label: e.PlayerMode.RealTime
          }, {
            default: p(() => l[21] || (l[21] = [
              P(" 实时视频 ")
            ])),
            _: 1,
            __: [21]
          }, 8, ["label"]),
          t(y, {
            label: e.PlayerMode.Playback
          }, {
            default: p(() => l[22] || (l[22] = [
              P(" 历史录像 ")
            ])),
            _: 1,
            __: [22]
          }, 8, ["label"])
        ]),
        _: 1
      }, 8, ["modelValue", "onChange"])) : s("", !0),
      e.playerMode === e.PlayerMode.Playback || e.playerType === e.PlayerMode.Playback ? (a(), r("div", K, [
        e.playing ? s("", !0) : (a(), r("img", {
          key: 0,
          width: "30",
          class: "btn-play icon",
          src: e.iconPath.play,
          onClick: l[17] || (l[17] = (...o) => e.playbackResume && e.playbackResume(...o))
        }, null, 8, Q)),
        e.playing ? (a(), r("img", {
          key: 1,
          width: "30",
          class: "btn-play icon",
          src: e.iconPath.stopPlay,
          onClick: l[18] || (l[18] = (...o) => e.playbackPause && e.playbackPause(...o))
        }, null, 8, X)) : s("", !0),
        n("span", Y, [
          t(m, {
            modelValue: e.videoProgress,
            "onUpdate:modelValue": l[19] || (l[19] = (o) => e.videoProgress = o),
            max: 100,
            min: 0,
            height: "200px"
          }, null, 8, ["modelValue"]),
          n("span", null, b(e.videoProgress) + "%", 1)
        ])
      ])) : s("", !0)
    ]),
    l[23] || (l[23] = n("div", { class: "style-container" }, null, -1))
  ], 2);
}
const se = /* @__PURE__ */ f(M, [["render", Z]]);
export {
  se as default
};
