// @ths/design/es/components/video-monitor/src/index.vue2.js
import { defineComponent as ve, computed as f, reactive as me, ref as pe, onMounted as Se, watch as v, onBeforeUnmount as be, toRefs as ge } from "vue";
import I from "moment";
import { getDomRandomClass as ke, setStyle as c, getCssPropertyString as y } from "../../../utils/index.js";
import he from "../images/arrow.svg.js";
var z = /* @__PURE__ */ ((o) => (o.RealTime = "RealTime", o.Playback = "Playback", o))(z || {});
const Ue = ve({
  name: "TVideoMonitor",
  props: {
    // 监控点位ID
    cameraCode: {
      type: String,
      default: ""
    },
    // 视频插件存储的路径
    basePath: {
      type: String,
      default: "./"
    },
    // 播放器；类型
    playerType: {
      type: String,
      // 默认直播状态 可选仅回放状态
      default: "RealTime"
      /* RealTime */
    },
    // 坐标
    location: {
      type: Array,
      default: () => ["", ""]
    },
    // 经度
    longitude: {
      type: String,
      default: "116.404"
    },
    // 纬度
    latitude: {
      type: String,
      default: "39.915"
    },
    // 回放视频开始时间
    playbackStartTime: {
      type: String,
      default: ""
    },
    // 回放视频结束时间
    playbackEndTime: {
      type: String,
      default: ""
    },
    // 直播地址
    realPlayUrl: {
      type: [String, Array],
      default: ""
    },
    // 对讲地址
    talkUrl: {
      type: [String, Array],
      default: ""
    },
    // 回放地址
    playbackUrl: {
      type: [String, Array],
      default: ""
    },
    // 角度
    angle: {
      type: Number,
      default: 0
    },
    // 可录制功能
    recordingDisabled: {
      type: Boolean,
      default: !1
    },
    // 可对讲功能
    intercomDisabled: {
      type: Boolean,
      default: !1
    },
    iconPath: {
      type: Object,
      default: () => ({
        arrow: "",
        circle: "",
        location: "",
        photo: "",
        play: "",
        stopPlay: "",
        talk: "",
        stopTalk: "",
        record: "",
        stopRecord: "",
        sound: "",
        soundNone: "",
        fullScreen: ""
      })
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {
            width: "1130px"
          }
        },
        // 视频容器
        videoContainer: {
          default: {
            width: "1130px",
            height: "606px"
          }
        },
        // 加载中状态背景
        playerLoading: {
          default: {}
        },
        // 加载中状态的图标
        loadingIcon: {
          default: {}
        },
        // 视频播放器 video-container__player
        video: {
          default: {}
        },
        // 小地图 video-container__map
        miniMap: {
          default: {}
        },
        // 定位图标  location-icon
        locationIcon: {
          default: {}
        },
        // 方向盘  video-container__steering-wheel
        steeringWheel: {
          default: {}
        },
        // 方向箭头图标
        directionArrowIcon: {
          // direction-arrow
          default: {},
          // direction-arrow--hover
          hover: {}
        },
        // 录制、对话状态提示 tip
        statusTip: {
          default: {}
        },
        // 工具条容器 toolbar
        toolbarContainer: {
          default: {}
        },
        // 工具条容器左侧区域 toolbar__left
        toolbarLeft: {
          default: {}
        },
        // 工具条容器左侧区域 toolbar__right
        toolbarRight: {
          default: {}
        },
        // 工具条容器左侧区域 toolbar__right
        toolbarIcon: {
          // 默认状态 .toolbar img.icon
          default: {},
          // 滑过状态 .toolbar img.icon:hover
          hover: {},
          // 禁用状态 .toolbar  span.icon-disabled
          disabled: {}
        },
        // 音量调节容器 包含图标 .volume-container
        volumeContainer: {
          default: {}
        },
        // 音量调节面板 .volume__panel
        volumePanel: {
          default: {}
        },
        // 音量调节滑块 .el-slider__button
        volumeSliderButton: {
          default: {}
        },
        // 音量调节轨道 .el-slider__runway
        volumeSliderRunway: {
          default: {}
        },
        // 音量调节进度条 .el-slider__bar
        volumeSliderBar: {
          default: {}
        },
        // 音量调节显示的文字 .volume-panel span
        volumeLabel: {
          default: {}
        },
        // 操作容器 .action-container
        actionContainer: {
          default: {}
        },
        // 切换按钮
        switchButton: {
          // .el-radio-button__inner
          default: {},
          // .el-radio-button__original-radio:checked + .el-radio-button__inner
          active: {}
        }
      })
    }
  },
  emits: ["steer", "switchMode"],
  setup(o, { emit: R }) {
    const a = f(() => o.cStyle), n = ke("video-monitor"), Y = () => {
      var t, r, i, u, d, s, g, k, h, w, _, P, T, H, j, Z, G, N;
      let e = "";
      return (t = a.value) != null && t.wrapper && (e += c(a.value.wrapper, `.${n}`)), (r = a.value) != null && r.videoContainer && (e += c(a.value.videoContainer, `.${n} .video-container`)), (i = a.value) != null && i.video && (e += c(a.value.video, `.${n} .video-container__player`)), (u = a.value) != null && u.loadingIcon && (e += c(a.value.loadingIcon, `.${n} .loading__icon`)), (d = a.value) != null && d.miniMap && (e += c(a.value.miniMap, `.${n} .video-container__map`)), (s = a.value) != null && s.locationIcon && (e += c(a.value.locationIcon, `.${n} .location-icon`)), (g = a.value) != null && g.steeringWheel && (e += c(a.value.steeringWheel, `.${n} .video-container__steering-wheel`)), (k = a.value) != null && k.statusTip && (e += c(a.value.statusTip, `.${n} .tip`)), (h = a.value) != null && h.toolbarContainer && (e += c(a.value.toolbarContainer, `.${n} .toolbar`)), (w = a.value) != null && w.toolbarLeft && (e += c(a.value.toolbarLeft, `.${n} .toolbar__left`)), (_ = a.value) != null && _.toolbarRight && (e += c(a.value.toolbarRight, `.${n} .toolbar__right`)), (P = a.value) != null && P.volumeContainer && (e += c(a.value.volumeContainer, `.${n} .volume-container`)), (T = a.value) != null && T.volumePanel && (e += c(a.value.volumePanel, `.${n} .volume__panel`)), (H = a.value) != null && H.volumeSliderButton && (e += c(a.value.volumeSliderButton, `.${n} .el-slider__button`)), (j = a.value) != null && j.volumeSliderRunway && (e += c(a.value.volumeSliderRunway, `.${n} .el-slider__runway`)), (Z = a.value) != null && Z.volumeSliderBar && (e += c(a.value.volumeSliderBar, `.${n} .el-slider__bar`)), (G = a.value) != null && G.volumeLabel && (e += c(a.value.volumeLabel, `.${n} .volume__panel span`)), (N = a.value) != null && N.actionContainer && (e += c(a.value.actionContainer, `.${n} .action-container`)), e;
    }, q = () => {
      var t, r, i, u, d, s, g, k, h, w, _, P, T;
      let e = "";
      return (t = a.value.playerLoading) != null && t.default && (e += `
            .${n} .video-container__player.loading:after {
                content: '';
                width: 100%;
                height: 100%;
                position: absolute;
                top: 0;
                left: 0;
              ${y(a.value.playerLoading.default)}
            }
          `), (r = a.value) != null && r.directionArrowIcon && ((i = a.value.directionArrowIcon) != null && i.default && (e += `
            .${n} .direction-arrow {
              ${y(a.value.directionArrowIcon.default)}
            }
          `), (u = a.value.toolbarIcon) != null && u.hover && (e += `
            .${n} .direction-arrow--hover {
              ${y(a.value.directionArrowIcon.hover)}
            }
          `)), (d = a.value) != null && d.toolbarIcon && ((s = a.value.toolbarIcon) != null && s.default && (e += `
            .${n} .toolbar img.icon {
              ${y(a.value.toolbarIcon.default)}
            }
          `), (g = a.value.toolbarIcon) != null && g.hover && (e += `
            .${n} .toolbar img.icon:hover {
              ${y(a.value.toolbarIcon.hover)}
            }
          `), (k = a.value.toolbarIcon) != null && k.disabled && (e += `
            .${n} .toolbar span.icon-disabled {
              ${y(a.value.toolbarIcon.disabled)}
            }
          `)), (h = a.value) != null && h.switchButton && ((w = a.value.switchButton) != null && w.default && (e += `
            .${n} .el-radio-button__inner {
              ${y(a.value.switchButton.default)}
            }
          `), (_ = a.value.switchButton) != null && _.active && (e += `
            .${n} .el-radio-button__original-radio:checked + .el-radio-button__inner {
              ${y(a.value.switchButton.active)}
            }

            .${n} .el-radio-button:first-child .el-radio-button__inner {
              border-left: 1px solid  ${((P = a.value.switchButton.default) == null ? void 0 : P.borderLeftColor) || "#999"};
            }

            .${n} .el-radio-button:first-child .el-radio-button__original-radio:checked + .el-radio-button__inner {
              border-left: 1px solid  ${((T = a.value.switchButton.active) == null ? void 0 : T.borderLeftColor) || "rgba(255,255,255,0.50)"};
            }
          `)), e;
    }, A = () => {
      const e = document.querySelector(`.${n} .style-container`);
      e && (e.innerHTML = `<style>${Y()}${q()}</style>`);
    }, U = document.body.clientWidth < 992, L = !!window.MediaSource, $ = f(() => {
      var i, u, d;
      let e = "";
      typeof o.realPlayUrl == "string" && (e = o.realPlayUrl), typeof o.realPlayUrl == "object" && ((i = o.realPlayUrl) != null && i.length) && (e = o.realPlayUrl[0]);
      let t = "";
      typeof o.playbackUrl == "string" && (t = o.playbackUrl), typeof o.playbackUrl == "object" && ((u = o.playbackUrl) != null && u.length) && (t = o.playbackUrl[0]);
      let r = "";
      return typeof o.talkUrl == "string" && (r = o.talkUrl), typeof o.talkUrl == "object" && ((d = o.talkUrl) != null && d.length) && (r = o.talkUrl[0]), console.log({
        realPlay: e,
        talk: r,
        playback: t
      }), {
        realPlay: e,
        talk: r,
        playback: t
      };
    }), l = me({
      loading: !0,
      // 当前监控窗口索引  默认为0
      currentWindowIndex: 0,
      // 视频进度
      videoProgress: 0,
      // 播放状态
      playing: !1,
      playerMode: "RealTime",
      // 音量调节面板显示状态
      volumePanelVisible: !1,
      // 记录上一次操作的音量值
      oldVolume: 10,
      // 对讲状态
      talking: !1,
      // 录制状态
      recording: !1,
      isMoveDevice: U,
      player: null,
      splitNum: U ? 1 : 2,
      mseSupport: L,
      tabActive: L ? "mse" : "decoder",
      labelCol: { span: 5 },
      wrapperCol: { span: 18 },
      urls: $.value,
      playback: {
        startTime: "2023-10-10T00:00:00",
        endTime: "2023-10-10T23:59:59",
        valueFormat: I.HTML5_FMT.DATETIME_LOCAL_SECONDS,
        seekStart: "2023-10-10T00:00:00",
        // 播放速度
        rate: 1
      },
      // 静音
      muted: !0,
      // 音量大小
      volume: 0
    }), C = f(() => l.tabActive === "mse" ? 0 : 1);
    function W() {
      window.addEventListener("resize", () => {
        var e;
        (e = l.player) == null || e.JS_Resize();
      });
    }
    const K = f(() => Array.isArray(o.talkUrl)), B = f(() => Array.isArray(o.realPlayUrl)), Q = f(() => Array.isArray(o.playbackUrl)), X = f(() => {
      let e = 1;
      return B.value && (e = Math.ceil(Math.sqrt(o.realPlayUrl.length))), e;
    });
    function F() {
      l.player.JS_GetVolume(l.player.currentWindowIndex).then(
        (e) => {
          console.info("JS_GetVolume success", e), l.volume = e, M(e);
        },
        (e) => {
          console.info("JS_GetVolume failed");
        }
      );
    }
    async function E() {
      l.player.JS_OpenSound(l.player.currentWindowIndex).then(
        () => {
          console.log("openSound success"), F(), l.muted = !1;
        },
        (e) => {
          console.error(e);
        }
      );
    }
    function m() {
      l.player.JS_CloseSound(l.player.currentWindowIndex).then(
        () => {
          console.log("closeSound success"), l.muted = !0;
        },
        (e) => {
          console.error(e);
        }
      );
    }
    function M(e) {
      l.oldVolume = e, l.muted = !e, e ? l.player.JS_SetVolume(l.player.currentWindowIndex, e).then(
        () => {
          console.log("setVolume success", e);
        },
        (t) => {
          t === "0x12f900010" && l.player.JS_OpenSound(l.player.currentWindowIndex).then(
            () => {
              console.log("openSound success"), M(e);
            }
          ), console.error(t);
        }
      ) : (m(), l.muted = !0);
    }
    function D() {
      l.player = new window.JSPlugin({
        szId: "player",
        szBasePath: o.basePath,
        iMaxSplit: 9,
        iCurrentSplit: U ? 1 : X.value,
        openDebug: !0,
        oStyle: {
          borderSelect: U ? "#000" : "transparent"
        }
      }), l.player.JS_SetWindowControlCallback({
        windowEventSelect(e) {
          console.log("windowSelect callback: ", e), l.currentWindowIndex = e, F(), K.value && o.talkUrl.length && (l.urls.talk = o.talkUrl[e]), B.value && o.realPlayUrl.length && (l.urls.realPlay = o.realPlayUrl[e]), Q.value && o.playbackUrl.length && (l.urls.playback = o.playbackUrl[e]);
        },
        pluginErrorHandler(e, t, r) {
          console.log("pluginError callback: ", e, t, r);
        },
        windowEventOver(e) {
        },
        windowEventOut(e) {
        },
        windowEventUp(e) {
        },
        windowFullCcreenChange(e) {
          console.log("fullScreen callback: ", e);
        },
        firstFrameDisplay(e, t, r) {
          console.log("firstFrame loaded callback: ", e, t, r);
        },
        performanceLack() {
          console.log("performanceLack callback: ");
        }
      });
    }
    function ee() {
      var e;
      (e = l.player) == null || e.JS_FullScreenDisplay(!0).then(
        () => {
          console.log("wholeFullScreen success");
        },
        (t) => {
          console.error(t);
        }
      );
    }
    function le() {
      var e;
      (e = l.player) == null || e.JS_FullScreenSingle(l.player.currentWindowIndex).then(
        () => {
          console.info("JS_FullScreenSingle success");
        },
        (t) => {
          console.info("JS_FullScreenSingle failed");
        }
      );
    }
    function ae() {
      const e = l.urls.talk;
      l.player.JS_SetConnectTimeOut(0, 1e3), l.player.JS_StartTalk(e).then(
        () => {
          l.talking = !0, console.log("talkStart success");
        },
        (t) => {
          console.error(t);
        }
      );
    }
    function J() {
      l.talking = !1, l.player.JS_StopTalk().then(
        () => {
          console.log("talkStop success");
        },
        (e) => {
          console.error(e);
        }
      );
    }
    function oe() {
      l.player.JS_StoprealPlayAll().then(
        () => {
          l.playback.rate = 0, console.log("stopAllPlay success");
        },
        (e) => {
          console.error(e);
        }
      );
    }
    function te() {
      let { seekStart: e, endTime: t } = l.playback;
      e += "Z", t += "Z", l.player.JS_Seek(l.player.currentWindowIndex, e, t).then(
        () => {
          console.log("seekTo success");
        },
        (r) => {
          console.error(r);
        }
      );
    }
    function ne() {
      l.player.JS_Slow().then(
        (e) => {
          l.playback.rate = e;
        },
        (e) => {
          console.error(e);
        }
      );
    }
    function re() {
      l.player.JS_Fast().then(
        (e) => {
          l.playback.rate = e;
        },
        (e) => {
          console.error(e);
        }
      );
    }
    function ie() {
      l.player.JS_FrameForward(l.player.currentWindowIndex).then(
        () => {
          l.playback.rate = 1, console.log("frameForward success");
        },
        (e) => {
          console.error(e);
        }
      );
    }
    function p() {
      const { player: e, urls: t } = l, r = t.realPlay;
      if (typeof o.realPlayUrl == "object")
        for (let i = 0; i < o.realPlayUrl.length; i++)
          e.JS_Play(o.realPlayUrl[i], { playURL: o.realPlayUrl[0], mode: C.value }, i).then(
            async () => {
              console.log("realPlay success"), await m(), l.loading = !1;
            },
            (u) => {
              console.error(u);
            }
          );
      typeof o.realPlayUrl == "string" && e.JS_Play(r, { playURL: r, mode: C.value }, 0).then(
        async () => {
          console.log("realPlay success"), await m(), l.loading = !1;
        },
        (i) => {
          console.error(i);
        }
      );
    }
    function x() {
      l.player.JS_Stop().then(
        () => {
          l.playback.rate = 0, console.log("stop realPlay success");
        },
        (e) => {
          console.error(e);
        }
      );
    }
    function S() {
      l.playing = !0;
      const { player: e, urls: t, playback: r } = l, i = t.playback;
      let { startTime: u, endTime: d } = r;
      u += "Z", d += "Z", e.JS_Play(i, { playURL: i, mode: C.value }, l.player.currentWindowIndex, u, d).then(
        () => {
          console.log("playbackStart success"), l.playback.rate = 1, E(), l.loading = !1;
        },
        (s) => {
          l.loading = !1, console.error(s);
        }
      );
    }
    function ce() {
      l.playing = !1, l.player.JS_Pause().then(
        () => {
          console.log("playbackPause success");
        },
        (e) => {
          console.error(e);
        }
      );
    }
    function ue() {
      l.playing = !0, l.player.JS_Resume(l.player.currentWindowIndex).then(
        () => {
          console.log("playbackResume success");
        },
        (e) => {
          console.error(e);
        }
      );
    }
    function de(e) {
      const { player: t } = l, r = l.player.currentWindowIndex;
      t.JS_CapturePicture(r, "img", e).then(
        () => {
          console.log("capture success", e);
        },
        (i) => {
          console.error(i);
        }
      );
    }
    function se(e) {
      const t = { MP4: 5, PS: 2 }, { player: r } = l, i = l.player.currentWindowIndex, u = `${I().format("YYYYMMDDHHmm")}.mp4`, d = t[e];
      r.JS_StartSaveEx(i, u, d).then(
        () => {
          console.log("record start ..."), l.recording = !0;
        },
        (s) => {
          console.error(s);
        }
      );
    }
    function O() {
      const { player: e } = l, t = l.player.currentWindowIndex;
      e.JS_StopSave(t).then(
        () => {
          console.log("record stoped, saving ..."), l.recording = !1;
        },
        (r) => {
          console.error(r), l.recording = !1;
        }
      );
    }
    function ye(e) {
      R("switchMode", e), l.loading = !0, J(), O(), x(), e === "RealTime" ? p() : S();
    }
    function fe(e) {
      R("steer", {
        cameraCode: o.cameraCode,
        direction: e,
        // 当前选中监控索引
        currentIndex: l.player.currentWindowIndex
      });
    }
    const b = pe();
    function V() {
      b.value && (b.value.centerAndZoom(new window.BMapGL.Point(o.longitude, o.latitude), 12), b.value.disableDragging(), b.value.disableScrollWheelZoom());
    }
    return Se(() => {
      b.value = new window.BMapGL.Map("baiduMap"), V(), A(), W(), D(), o.playerType === "RealTime" ? p() : S();
    }), v(
      () => o.cStyle,
      () => {
        A();
      },
      { deep: !0, immediate: !0 }
    ), v(
      () => o.playerType,
      (e) => {
        l.playerMode = e, o.playerType === "RealTime" ? p() : S();
      }
    ), v(
      () => o.realPlayUrl,
      (e) => {
        console.log(e), l.urls = $.value, o.playerType === "RealTime" && l.playerMode === "RealTime" && p();
      }
    ), v(
      () => o.talkUrl,
      (e) => {
        console.log(e), l.urls = $.value;
      }
    ), v(
      () => o.playbackUrl,
      (e) => {
        l.urls = $.value, console.log(e), o.playerType === "Playback" && l.playerMode === "Playback" && S();
      }
    ), v([() => o.longitude, () => o.latitude], () => {
      V();
    }), v([() => o.playbackStartTime, () => o.playbackEndTime], () => {
      l.playback.startTime = I(o.playbackStartTime).format(l.playback.valueFormat), l.playback.endTime = I(o.playbackEndTime).format(l.playback.valueFormat);
    }, { immediate: !0 }), be(() => {
      l.loading = !1, J(), m(), x(), l.player = null;
    }), {
      ...ge(l),
      PlayerMode: z,
      curStyle: a,
      domOnlyClass: n,
      init: W,
      createPlayer: D,
      wholeFullScreen: ee,
      fullScreenSingle: le,
      realPlay: p,
      stopPlay: x,
      talkStart: ae,
      talkStop: J,
      stopAllPlay: oe,
      playbackStart: S,
      playbackPause: ce,
      playbackResume: ue,
      seekTo: te,
      playbackSlow: ne,
      playbackFast: re,
      frameForward: ie,
      openSound: E,
      closeSound: m,
      setVolume: M,
      capture: de,
      recordStart: se,
      recordStop: O,
      icon: he,
      handleSteer: fe,
      switchPlayerMode: ye
    };
  }
});
export {
  Ue as default
};
