// @ths/design/es/components/video-base/src/index.vue.js
import _ from "./index.vue2.js";
import { resolveComponent as g, openBlock as r, createElementBlock as s, normalizeClass as C, normalizeStyle as m, createElementVNode as o, createVNode as w, withCtx as f, toDisplayString as k, createCommentVNode as V } from "vue";
import "./index.vue3.js";
import B from "../../../_virtual/_plugin-vue_export-helper.js";
const S = { class: "video-play dark" }, h = {
  key: 1,
  class: "video autoplay-video"
}, E = ["src"], D = {
  key: 2,
  id: "t-video-viewer",
  class: "video-viewer"
}, L = { class: "video-viewer__name" }, $ = { class: "video-viewer__box" }, z = ["src"];
function F(e, i, N, b, j, M) {
  var l, a, d, n, u, v, p, c;
  const t = g("el-icon");
  return r(), s("div", {
    class: C(["t-video-base", e.domOnlyClass]),
    style: m({
      marginTop: (a = (l = e.curStyle.wrapper) == null ? void 0 : l.default) == null ? void 0 : a.marginTop,
      marginLeft: (n = (d = e.curStyle.wrapper) == null ? void 0 : d.default) == null ? void 0 : n.marginLeft,
      marginRight: (v = (u = e.curStyle.wrapper) == null ? void 0 : u.default) == null ? void 0 : v.marginRight,
      marginBottom: (c = (p = e.curStyle.wrapper) == null ? void 0 : p.default) == null ? void 0 : c.marginBottom
    })
  }, [
    e.autoplay ? (r(), s("div", h, [
      o("video", {
        ref: "autoplayVideo",
        class: "video-js video-viewer__video",
        src: e.videoSrc,
        controls: "controls",
        autoplay: ""
      }, " 浏览器不支持视频标签 ", 8, E)
    ])) : (r(), s("div", {
      key: 0,
      class: "video",
      style: m({ background: `url(${e.coverSrc}) 0% 0% / 100% 100% no-repeat` })
    }, [
      o("div", S, [
        w(t, { onClick: e.openViewer }, {
          default: f(() => i[1] || (i[1] = [
            o("svg", {
              xmlns: "http://www.w3.org/2000/svg",
              viewBox: "0 0 1024 1024"
            }, [
              o("path", {
                fill: "currentColor",
                d: "M384 192v640l384-320.064z"
              })
            ], -1)
          ])),
          _: 1,
          __: [1]
        }, 8, ["onClick"])
      ])
    ], 4)),
    e.showViewer ? (r(), s("div", D, [
      o("div", L, k(e.videoName), 1),
      o("div", {
        class: "video-viewer__close",
        onClick: i[0] || (i[0] = (...y) => e.closeViewer && e.closeViewer(...y))
      }, [
        w(t, null, {
          default: f(() => i[2] || (i[2] = [
            o("svg", {
              xmlns: "http://www.w3.org/2000/svg",
              viewBox: "0 0 1024 1024"
            }, [
              o("path", {
                fill: "currentColor",
                d: `M764.288 214.592 512 466.88 259.712 214.592a31.936 31.936 0 0 0-45.12 45.12L466.752 512 214.528 764.224a31.936 31.936
              0 1 0 45.12 45.184L512 557.184l252.288 252.288a31.936 31.936 0 0 0 45.12-45.12L557.12 512.064l252.288-252.352a31.936 31.936 0 1 0-45.12-45.184z`
              })
            ], -1)
          ])),
          _: 1,
          __: [2]
        })
      ]),
      o("div", $, [
        o("video", {
          ref: "realtimeVideo",
          class: "video-js video-viewer__video",
          src: e.videoSrc,
          controls: "controls",
          autoplay: ""
        }, " 浏览器不支持视频标签 ", 8, z)
      ])
    ])) : V("", !0),
    i[3] || (i[3] = o("div", { class: "style-container" }, null, -1))
  ], 6);
}
const A = /* @__PURE__ */ B(_, [["render", F]]);
export {
  A as default
};
