// @ths/design/es/components/video-base/src/index.vue2.js
import { defineComponent as S, reactive as w, ref as C, computed as V, onBeforeUnmount as B, watch as m, onMounted as b, toRefs as h, nextTick as M } from "vue";
import u from "video.js";
import R from "video.js/dist/lang/zh-CN.json";
import "video.js/dist/video-js.css";
import { getDomRandomClass as T, setStyle as J } from "../../../utils/index.js";
u.addLanguage("zh-CN", R);
const j = S({
  name: "TVideoBase",
  props: {
    // 自动播放
    autoplay: {
      type: Boolean,
      default: !1
    },
    // 封面路径
    coverSrc: {
      type: String,
      default: ""
    },
    // 视频路径
    videoSrc: {
      type: String,
      default: ""
    },
    // 视频名称描述
    videoName: {
      type: String,
      default: ""
    },
    // 播放器的动态大小
    aspectRatio: {
      type: String,
      default: "16:7"
    },
    // 是否拥有控制条
    controls: {
      type: Boolean,
      default: !0
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    }
  },
  setup(t) {
    const o = w({
      realtimeVideo: {},
      // 弹出层video元素
      autoplayVideo: {},
      // 自动播放video元素
      player: null,
      // video.js video对象
      videoJs: {}
      // video.js参数设置
    }), a = C(!1), s = window.videojs || u, l = V(() => t.cStyle), r = T("video-base"), p = (e) => {
      let n = "";
      const g = e.lastIndexOf("."), i = e.substr(g);
      return i === ".MP4" || i === ".mp4" ? n = "video/mp4" : i === ".m3u8" || i === ".M3U8" ? n = "application/x-mpegURL" : i === ".ogg" || i === ".OGG" ? n = "video/ogg" : (i === ".webm" || i === ".WEBM") && (n = "video/webm"), n;
    }, c = () => {
      o.videoJs = {
        aspectRatio: t.aspectRatio,
        // 宽高比
        controls: t.controls,
        // 显示控制条
        language: "zh-CN",
        // 设置语言
        sources: [
          {
            src: t.videoSrc,
            // 视频路径
            type: p(t.videoSrc)
            // 视频类型
          }
        ],
        controlBar: {
          remainingTimeDisplay: {
            displayNegative: !1
            // 剩余时间不显示负号
          }
        }
      };
    }, y = () => {
      let e = "";
      return l.value.wrapper && (e += J(l.value.wrapper, `.${r} .video`)), e;
    }, d = () => {
      const e = document.querySelector(`.${r} .style-container`);
      e && (e.innerHTML = `<style>${y()}</style>`);
    }, f = () => {
      a.value = !0, M(() => {
        const e = document.querySelector("body");
        e.append ? e.append(document.getElementById("t-video-viewer")) : e.appendChild(document.getElementById("t-video-viewer")), o.player = s(o.realtimeVideo, o.videoJs, () => {
        });
        const n = document.querySelectorAll(".vjs-remaining-time span");
        n && (n[1].innerHTML = n[1].innerHTML.replace("-", ""));
      });
    }, v = () => {
      a.value = !1;
    };
    return B(() => {
      o.player && o.player.dispose();
    }), m(() => t.cStyle, () => {
      d();
    }), m(() => [t.controls, t.aspectRatio, t.videoSrc], () => {
      c();
    }), b(() => {
      d(), c(), t.autoplay && (o.player = s(o.autoplayVideo, o.videoJs, () => {
      }));
    }), {
      domOnlyClass: r,
      ...h(o),
      curStyle: l,
      openViewer: f,
      closeViewer: v,
      showViewer: a
    };
  }
});
export {
  j as default
};
