// @ths/design/es/components/tabs-scroll/src/index.vue2.js
import { defineComponent as At, ref as m, reactive as Rt, watch as B, onMounted as xt, toRefs as Et } from "vue";
import { getDomRandomClass as Ft, setStyle as $t, getCssPropertyString as n } from "../../../utils/index.js";
const _t = At({
  name: "TTabsScroll",
  props: {
    // 数据
    data: {
      type: Object,
      default: () => ({})
    },
    // 是否显示图标
    showIcon: {
      type: Boolean,
      default: !0
    },
    // 是否是自定义图标
    customIcon: {
      type: Boolean,
      default: !1
    },
    // 当前选中项路径
    activePath: {
      type: String,
      default: ""
    },
    // 字段关键词
    keyField: {
      type: Object,
      default: () => ({
        label: "label",
        icon: "icon",
        path: "path"
      })
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层总容器
        wrapper: {
          default: {}
        },
        // 标签容器
        itemsWrapper: {
          default: {}
        },
        // 单项容器
        itemWrapper: {
          default: {},
          hover: {},
          active: {}
        },
        // 图标
        icon: {
          default: {}
        },
        // 两侧切换按钮
        tabsButton: {
          default: {},
          hover: {},
          active: {}
        },
        // 更多按钮
        moreButton: {
          default: {},
          hover: {},
          active: {}
        },
        // 下拉内容容器
        dropDownWrapper: {
          default: {}
        },
        // 下拉内容单项
        dropDownItem: {
          default: {},
          hover: {}
        }
      })
    }
  },
  emits: ["tab-click", "tab-close"],
  setup(l, h) {
    const o = Ft("tabs-scroll"), u = m(null), $ = m(!1), wt = m(!0), gt = m(!1), t = Rt({
      tabsData: [],
      // 组件数据
      hoverItemIndex: -1,
      // hover的索引
      customStyle: l.cStyle,
      // 组件样式
      hoverButtonIndex: -1
      // hover的索引
    }), p = m(!1), S = m(0);
    if (Array.isArray(l.data))
      t.tabsData = [...l.data];
    else {
      const e = { ...l.data };
      t.tabsData = Object.keys(e).map((a) => e[a]);
    }
    const s = m(0), D = (e) => {
      s.value = e && (t.tabsData.findIndex((a) => a[l.keyField.path] === e) !== -1 ? t.tabsData.findIndex((a) => a[l.keyField.path] === e) : 0) || 0;
    };
    D(l.activePath);
    const b = () => {
      u.value && ($.value = u.value.scrollWidth > u.value.clientWidth);
    }, It = () => {
      if (u.value) {
        const a = (u.value.scrollLeft || 0) - 200 || 0;
        u.value.scrollTo({
          left: a,
          behavior: "smooth"
        });
      }
    }, Tt = () => {
      if (u.value) {
        const a = (u.value.scrollLeft || 0) + 200 || 0;
        u.value.scrollTo({
          left: a,
          behavior: "smooth"
        });
      }
    }, Lt = (e, a) => {
      h.emit("tab-click", e), s.value = a;
    }, kt = (e, a) => {
      t.tabsData.length !== 1 && (t.tabsData.splice(a, 1), b(), s.value > a && (s.value -= 1), t.tabsData.length === a && (s.value = t.tabsData.length - 1), h.emit("tab-close", { selectedItem: t.tabsData[s.value], data: t.tabsData }));
    }, Wt = (e) => {
      let a = [];
      switch (e) {
        case "left":
          s.value !== 0 && (a = [], t.tabsData.forEach((c, i) => {
            i >= s.value && a.push(c);
          }), t.tabsData = [...a], s.value = 0);
          break;
        case "right":
          a = [], t.tabsData.forEach((c, i) => {
            i <= s.value && a.push(c);
          }), t.tabsData = [...a];
          break;
        case "other":
          a = [], a.push(t.tabsData[s.value]), t.tabsData = [...a], s.value = 0;
          break;
        case "all":
          a = [], a.push(t.tabsData[0]), t.tabsData = [...a], s.value = 0;
          break;
      }
      h.emit("tab-close", { selectedItem: t.tabsData[s.value], data: t.tabsData });
      const r = setTimeout(() => {
        b(), clearTimeout(r);
      }, 10);
    }, Ct = (e) => {
      e ? (S.value && clearTimeout(S.value), p.value = e) : S.value = window.setTimeout(() => {
        p.value = e, clearTimeout(S.value);
      }, 300);
    }, Ot = () => {
      var a, r, c, i, d, f, y, v;
      let e = "";
      return (a = t.customStyle) != null && a.itemWrapper && (e += $t(t.customStyle.itemWrapper, `.${o} .tabs-main .el-tag:not(.active)`), (c = (r = t.customStyle) == null ? void 0 : r.itemWrapper) != null && c.active && (e += `.${o} .tabs-main .el-tag.active {
            ${n((d = (i = t.customStyle) == null ? void 0 : i.itemWrapper) == null ? void 0 : d.active)}
          }`)), (f = t.customStyle) != null && f.icon && (e += $t(t.customStyle.icon, `.${o} .el-tag .tab-div .tab-div-icon`), e += `.${o} .tabs-main .el-tag:not(.active):hover .tab-div .tab-div-icon {
          ${n((v = (y = t.customStyle) == null ? void 0 : y.icon) == null ? void 0 : v.hover)}
        }`), e;
    }, jt = () => {
      var a, r, c, i, d, f, y, v, g, I, T, L, k, W, C, O, j, A, R, x, E, F, H, P, M, _, q, z, V, G, J, K, N, Q, U, X, Y, Z, tt, et, at, ot, lt, st, nt, ct, ut, rt, it, mt, bt, dt, ft, yt, vt, St, ht, pt, Dt, Bt;
      let e = "";
      return (a = t.customStyle) != null && a.tabsButton && ((c = (r = t.customStyle) == null ? void 0 : r.tabsButton) != null && c.default) && (e += `.${o}.t-tabs-scroll .left-btn {
          ${n((d = (i = t.customStyle) == null ? void 0 : i.tabsButton) == null ? void 0 : d.default)}
        }`), (f = t.customStyle) != null && f.tabsButton && ((v = (y = t.customStyle) == null ? void 0 : y.tabsButton) != null && v.hover) && (e += `
        .${o}.t-tabs-scroll .left-btn:not(.is-disabled):hover {
          ${n((I = (g = t.customStyle) == null ? void 0 : g.tabsButton) == null ? void 0 : I.hover)}
        }`), (T = t.customStyle) != null && T.tabsButton && ((k = (L = t.customStyle) == null ? void 0 : L.tabsButton) != null && k.active) && (e += `.${o}.t-tabs-scroll .left-btn:not(.is-disabled):focus {
          ${n((C = (W = t.customStyle) == null ? void 0 : W.tabsButton) == null ? void 0 : C.active)}
        }`), (O = t.customStyle) != null && O.tabsButton && ((A = (j = t.customStyle) == null ? void 0 : j.tabsButton) != null && A.default) && (e += `.${o}.t-tabs-scroll .right-btn {
          ${n((x = (R = t.customStyle) == null ? void 0 : R.tabsButton) == null ? void 0 : x.default)}
        }`), (E = t.customStyle) != null && E.tabsButton && ((H = (F = t.customStyle) == null ? void 0 : F.tabsButton) != null && H.hover) && (e += `
        .${o}.t-tabs-scroll .right-btn:not(.is-disabled):hover {
          ${n((M = (P = t.customStyle) == null ? void 0 : P.tabsButton) == null ? void 0 : M.hover)}
        }`), (_ = t.customStyle) != null && _.tabsButton && ((z = (q = t.customStyle) == null ? void 0 : q.tabsButton) != null && z.active) && (e += `.${o}.t-tabs-scroll .right-btn:not(.is-disabled):focus {
          ${n((G = (V = t.customStyle) == null ? void 0 : V.tabsButton) == null ? void 0 : G.active)}
        }`), (J = t.customStyle) != null && J.moreButton && ((N = (K = t.customStyle) == null ? void 0 : K.moreButton) != null && N.default) && (e += `.${o}.t-tabs-scroll .more-btn {
          ${n((U = (Q = t.customStyle) == null ? void 0 : Q.moreButton) == null ? void 0 : U.default)}
        }`), (X = t.customStyle) != null && X.moreButton && ((Z = (Y = t.customStyle) == null ? void 0 : Y.moreButton) != null && Z.hover) && (e += `
        .${o}.t-tabs-scroll .more-btn:not(.is-disabled):hover {
          ${n((et = (tt = t.customStyle) == null ? void 0 : tt.moreButton) == null ? void 0 : et.hover)}
        }`), (at = t.customStyle) != null && at.moreButton && ((lt = (ot = t.customStyle) == null ? void 0 : ot.moreButton) != null && lt.active) && (e += `.${o}.t-tabs-scroll .more-btn:not(.is-disabled):focus {
          ${n((nt = (st = t.customStyle) == null ? void 0 : st.moreButton) == null ? void 0 : nt.active)}
        }`), (ct = t.customStyle) != null && ct.dropDownWrapper && ((rt = (ut = t.customStyle) == null ? void 0 : ut.dropDownWrapper) != null && rt.default) && (e += `.${o} .tabs-dropdown-menu .dropdown-menu-content {
          ${n((mt = (it = t.customStyle) == null ? void 0 : it.dropDownWrapper) == null ? void 0 : mt.default)}
          }`), (bt = t.customStyle) != null && bt.dropDownItem && ((ft = (dt = t.customStyle) == null ? void 0 : dt.dropDownItem) != null && ft.default) && (e += `
        .${o} .tabs-dropdown-menu .dropdown-menu-content span {
          ${n((vt = (yt = t.customStyle) == null ? void 0 : yt.dropDownItem) == null ? void 0 : vt.default)}
          }`), (St = t.customStyle) != null && St.dropDownItem && ((pt = (ht = t.customStyle) == null ? void 0 : ht.dropDownItem) != null && pt.hover) && (e += `
        .${o} .tabs-dropdown-menu .dropdown-menu-content span:hover {
          ${n((Bt = (Dt = t.customStyle) == null ? void 0 : Dt.dropDownItem) == null ? void 0 : Bt.hover)}
          }`), e;
    }, w = () => {
      const e = document.querySelector(`.${o} > .style-container`);
      e && (e.innerHTML = `<style>${Ot()}${jt()}</style>`);
    };
    return B(
      () => l.activePath,
      (e) => {
        D(e);
      }
    ), B(
      () => l.data,
      (e) => {
        if (Array.isArray(l.data))
          t.tabsData = [...l.data];
        else {
          const r = { ...l.data };
          t.tabsData = Object.keys(r).map((c) => r[c]);
        }
        D(l.activePath);
        const a = setTimeout(() => {
          b(), clearTimeout(a);
        }, 10);
      },
      {
        deep: !0
      }
    ), B(() => l.cStyle, () => {
      t.customStyle = l.cStyle, w();
    }, {
      deep: !0
    }), xt(() => {
      b(), w(), window.onresize = () => {
        b();
      };
    }), {
      ...Et(t),
      domOnlyClass: o,
      tabClick: Lt,
      tabClose: kt,
      moveLeft: It,
      moveRight: Tt,
      showScroll: $,
      handleCloseTab: Wt,
      currentActiveIndex: s,
      tTab: u,
      disabledLeftButton: wt,
      disabledRightButton: gt,
      showMenu: p,
      showDropdownMenu: Ct
    };
  }
});
export {
  _t as default
};
