// @ths/design/es/components/tabs-scroll/src/index.vue.js
import T from "./index.vue2.js";
import { resolveComponent as w, openBlock as l, createElementBlock as f, normalizeClass as S, normalizeStyle as u, createElementVNode as t, createBlock as d, withCtx as s, createVNode as i, createCommentVNode as v, Fragment as h, renderList as W, resolveDynamicComponent as A, toDisplayString as N, withModifiers as V } from "vue";
import "./index.vue3.js";
import O from "../../../_virtual/_plugin-vue_export-helper.js";
const R = { class: "container" }, X = {
  ref: "tTab",
  class: "nav-content"
}, j = { class: "nav-content_scroll" }, q = ["src"], G = { class: "dropdown-menu-content" };
function H(e, o, J, K, P, Q) {
  var b, y, g, k, C, I, M, $, B, D, L, z, F;
  const a = w("el-icon"), p = w("el-button"), E = w("el-tag");
  return l(), f("div", {
    class: S(["t-tabs-scroll", e.domOnlyClass]),
    style: u({ ...(y = (b = e.customStyle) == null ? void 0 : b.wrapper) == null ? void 0 : y.default })
  }, [
    t("div", R, [
      t("div", {
        class: "tabs-main",
        style: u({ ...(k = (g = e.customStyle) == null ? void 0 : g.itemsWrapper) == null ? void 0 : k.default })
      }, [
        e.showScroll ? (l(), d(p, {
          key: 0,
          class: "tabs-btn left-btn",
          type: "text",
          size: "mini",
          onClick: e.moveLeft,
          onMouseenter: o[0] || (o[0] = (n) => e.hoverButtonIndex = 0),
          onMouseleave: o[1] || (o[1] = (n) => e.hoverButtonIndex = -1)
        }, {
          default: s(() => [
            i(a, null, {
              default: s(() => o[21] || (o[21] = [
                t("svg", {
                  xmlns: "http://www.w3.org/2000/svg",
                  viewBox: "0 0 1024 1024"
                }, [
                  t("path", {
                    fill: "currentColor",
                    d: `M609.408 149.376 277.76 489.6a32 32 0 0 0 0 44.672l331.648 340.352a29.12 29.12 0 0 0 41.728 0 30.592 30.592 0 0 0 0-42.752L339.264
              511.936l311.872-319.872a30.592 30.592 0 0 0 0-42.688 29.12 29.12 0 0 0-41.728 0z`
                  })
                ], -1)
              ])),
              _: 1,
              __: [21]
            })
          ]),
          _: 1
        }, 8, ["onClick"])) : v("", !0),
        t("div", X, [
          t("div", j, [
            (l(!0), f(h, null, W(e.tabsData, (n, r) => (l(), d(E, {
              key: r,
              class: S({ active: e.currentActiveIndex === r }),
              onClick: (m) => e.tabClick(n, r),
              onMouseenter: (m) => e.hoverItemIndex = r,
              onMouseleave: o[2] || (o[2] = (m) => e.hoverItemIndex = -1)
            }, {
              default: s(() => [
                t("div", {
                  class: "tab-div",
                  style: u({ marginLeft: e.hoverItemIndex === r ? "-6px" : "0px" })
                }, [
                  e.showIcon && n[e.keyField.icon] && e.customIcon ? (l(), f("img", {
                    key: 0,
                    class: "fa tab-div-icon",
                    src: n[e.keyField.icon],
                    draggable: "false"
                  }, null, 8, q)) : v("", !0),
                  e.showIcon && n[e.keyField.icon] && n[e.keyField.icon].indexOf("/") === -1 && !e.customIcon ? (l(), d(a, {
                    key: 1,
                    class: "tab-div-icon"
                  }, {
                    default: s(() => [
                      (l(), d(A(e.customIcon ? "" : n[e.keyField.icon])))
                    ]),
                    _: 2
                  }, 1024)) : v("", !0),
                  t("span", null, N(n[e.keyField.label]), 1),
                  e.tabsData.length !== 1 ? (l(), d(a, {
                    key: 2,
                    class: "el-icon-close",
                    style: u({
                      marginRight: e.hoverItemIndex === r ? "-6px" : "0px",
                      width: e.hoverItemIndex === r ? "12px" : "0px"
                    }),
                    onClick: V((m) => e.tabClose(n, r), ["stop"])
                  }, {
                    default: s(() => o[22] || (o[22] = [
                      t("svg", {
                        xmlns: "http://www.w3.org/2000/svg",
                        viewBox: "0 0 1024 1024"
                      }, [
                        t("path", {
                          fill: "currentColor",
                          d: `M195.2 195.2a64 64 0 0 1 90.496 0L512 421.504 738.304 195.2a64 64 0 0 1 90.496 90.496L602.496 512 828.8
                    738.304a64 64 0 0 1-90.496 90.496L512 602.496 285.696 828.8a64 64 0 0 1-90.496-90.496L421.504 512 195.2 285.696a64 64 0 0 1 0-90.496z`
                        })
                      ], -1)
                    ])),
                    _: 2,
                    __: [22]
                  }, 1032, ["style", "onClick"])) : v("", !0)
                ], 4)
              ]),
              _: 2
            }, 1032, ["class", "onClick", "onMouseenter"]))), 128))
          ])
        ], 512),
        e.showScroll ? (l(), d(p, {
          key: 1,
          class: "tabs-btn right-btn",
          type: "text",
          size: "mini",
          onClick: e.moveRight,
          onMouseenter: o[3] || (o[3] = (n) => e.hoverButtonIndex = 1),
          onMouseleave: o[4] || (o[4] = (n) => e.hoverButtonIndex = -1)
        }, {
          default: s(() => [
            i(a, null, {
              default: s(() => o[23] || (o[23] = [
                t("svg", {
                  xmlns: "http://www.w3.org/2000/svg",
                  viewBox: "0 0 1024 1024"
                }, [
                  t("path", {
                    fill: "currentColor",
                    d: `M340.864 149.312a30.592 30.592 0 0 0 0 42.752L652.736 512 340.864 831.872a30.592 30.592 0 0 0 0
              42.752 29.12 29.12 0 0 0 41.728 0L714.24 534.336a32 32 0 0 0 0-44.672L382.592 149.376a29.12 29.12 0 0 0-41.728 0z`
                  })
                ], -1)
              ])),
              _: 1,
              __: [23]
            })
          ]),
          _: 1
        }, 8, ["onClick"])) : v("", !0)
      ], 4),
      t("div", null, [
        i(p, {
          class: "tabs-btn more-btn",
          type: "text",
          size: "mini",
          onMouseenter: o[5] || (o[5] = (n) => {
            e.showDropdownMenu(!0), e.hoverButtonIndex = 2;
          }),
          onMouseleave: o[6] || (o[6] = (n) => {
            e.showDropdownMenu(!1), e.hoverButtonIndex = -1;
          })
        }, {
          default: s(() => [
            i(a, null, {
              default: s(() => o[24] || (o[24] = [
                t("svg", {
                  xmlns: "http://www.w3.org/2000/svg",
                  viewBox: "0 0 1024 1024"
                }, [
                  t("path", {
                    fill: "currentColor",
                    d: `M176 416a112 112 0 1 1 0 224 112 112 0 0 1 0-224zm336 0a112 112 0 1 1 0 224 112
              112 0 0 1 0-224zm336 0a112 112 0 1 1 0 224 112 112 0 0 1 0-224z`
                  })
                ], -1)
              ])),
              _: 1,
              __: [24]
            })
          ]),
          _: 1
        }),
        t("div", {
          style: u({
            opacity: e.showMenu ? 1 : 0,
            transform: e.showMenu ? "rotateX(0deg)" : "rotateX(90deg)"
          }),
          class: "tabs-dropdown-menu",
          onMouseenter: o[19] || (o[19] = (n) => e.showDropdownMenu(!0)),
          onMouseleave: o[20] || (o[20] = (n) => e.showDropdownMenu(!1))
        }, [
          t("div", {
            class: "arrow",
            style: u({
              backgroundColor: (M = (I = (C = e.customStyle) == null ? void 0 : C.dropDownWrapper) == null ? void 0 : I.default) == null ? void 0 : M.backgroundColor,
              borderTop: (D = (B = ($ = e.customStyle) == null ? void 0 : $.dropDownWrapper) == null ? void 0 : B.default) == null ? void 0 : D.borderTop,
              borderLeft: (F = (z = (L = e.customStyle) == null ? void 0 : L.dropDownWrapper) == null ? void 0 : z.default) == null ? void 0 : F.borderLeft
            })
          }, null, 4),
          t("div", G, [
            t("span", {
              onClick: o[7] || (o[7] = (n) => e.handleCloseTab("left")),
              onMouseenter: o[8] || (o[8] = (n) => e.hoverButtonIndex = 3),
              onMouseleave: o[9] || (o[9] = (n) => e.hoverButtonIndex = -1)
            }, "关闭左侧", 32),
            t("span", {
              onClick: o[10] || (o[10] = (n) => e.handleCloseTab("right")),
              onMouseenter: o[11] || (o[11] = (n) => e.hoverButtonIndex = 4),
              onMouseleave: o[12] || (o[12] = (n) => e.hoverButtonIndex = -1)
            }, "关闭右侧", 32),
            t("span", {
              onClick: o[13] || (o[13] = (n) => e.handleCloseTab("other")),
              onMouseenter: o[14] || (o[14] = (n) => e.hoverButtonIndex = 5),
              onMouseleave: o[15] || (o[15] = (n) => e.hoverButtonIndex = -1)
            }, "关闭其他", 32),
            t("span", {
              onClick: o[16] || (o[16] = (n) => e.handleCloseTab("all")),
              onMouseenter: o[17] || (o[17] = (n) => e.hoverButtonIndex = 6),
              onMouseleave: o[18] || (o[18] = (n) => e.hoverButtonIndex = -1)
            }, "关闭全部", 32)
          ])
        ], 36)
      ])
    ]),
    o[25] || (o[25] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const c = /* @__PURE__ */ O(T, [["render", H]]);
export {
  c as default
};
