// @ths/design/es/components/gradient-shadow-text/src/index.vue2.js
import { defineComponent as u, computed as d, onMounted as s, watch as c } from "vue";
import { filterAttrs as y, getDomRandomClass as x, setStyle as n } from "../../../utils/index.js";
const f = u({
  name: "TGradientShadowText",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 渐变文字样式
        textStyle: {
          default: {}
        },
        // 阴影文字样式
        textShadowStyle: {
          default: {}
        }
      })
    },
    // 文字
    label: {
      type: String,
      default: () => ""
    },
    // 文字阴影样式数据
    shadowStyle: {
      type: Object,
      default: () => ({})
    }
  },
  setup(o) {
    const l = d(() => o.cStyle), a = d(() => o.shadowStyle), e = x("gradient-shadow-text"), i = () => {
      let t = "";
      return l.value.wrapper && (t += n(l.value.wrapper, `.${e}`)), l.value.textStyle && (t += n(l.value.textStyle, `.${e} .text-container .text`)), l.value.textShadowStyle && (t += n(l.value.textShadowStyle, `.${e} .text-container .text-shadow`)), a.value.x && (t += `.${e}.t-gradient-shadow-text .text-container {
          --t-gradient-shadow-x: ${a.value.x || 0};
        }`), a.value.y && (t += `.${e}.t-gradient-shadow-text .text-container {
          --t-gradient-shadow-y: ${a.value.y || 0};
        }`), a.value.blur && (t += `.${e}.t-gradient-shadow-text .text-container {
          --t-gradient-shadow-blur: ${a.value.blur || 0};
        }`), a.value.color && (t += `.${e}.t-gradient-shadow-text .text-container {
          --t-gradient-shadow-color: ${a.value.color || "transparent"};
        }`), t;
    }, r = () => {
      const t = document.querySelector(`.${e} > .style-container`);
      t && (t.innerHTML = `<style>${i()}}</style>`);
    };
    return s(() => {
      r();
    }), c(() => [o.cStyle, o.shadowStyle], () => {
      r();
    }, {
      deep: !0,
      immediate: !0
    }), {
      curStyle: l,
      domOnlyClass: e,
      filterAttrs: y
    };
  }
});
export {
  f as default
};
