// @ths/design/es/components/six-dimension/src/index.vue2.js
import { defineComponent as v, computed as g, reactive as D, onMounted as S, watch as s, toRefs as W } from "vue";
import h from "lodash/merge";
import $ from "lodash/cloneDeep";
import A from "./item.vue.js";
import { getDomRandomClass as C, setStyle as r } from "../../../utils/index.js";
const f = {
  titleKey: "title",
  valueKey: "value",
  unitKey: "unit"
}, R = v({
  name: "TSixDimension",
  components: { TSixDimensionItem: A },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外容器
        wrapper: {
          default: {}
        },
        // 左侧总容器
        leftWrapper: {
          default: {}
        },
        // 右侧总容器
        rightWrapper: {
          default: {}
        },
        // 左侧单项容器
        leftItemWrapper: {
          default: {}
        },
        // 右侧单项容器
        rightItemWrapper: {
          default: {}
        },
        // 两侧值
        sideValue: {
          default: {}
        },
        // 两侧单位
        sideUnit: {
          default: {}
        },
        // 两侧标题
        sideTitle: {
          default: {}
        },
        // 中间容器
        centerWrapper: {
          default: {}
        },
        // 中间值
        centerValue: {
          default: {}
        },
        // 中间单位
        centerUnit: {
          default: {}
        }
      })
    },
    // 两侧数据
    sideData: {
      type: Array,
      default: () => []
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => f
    },
    // 中间数值
    centerValue: {
      type: String,
      default: () => ""
    },
    // 中间单位
    centerUnit: {
      type: String,
      default: () => "天"
    },
    // 中间背景
    centerBg: {
      type: String,
      default: () => ""
    }
  },
  setup(a) {
    const e = g(() => a.cStyle), n = C("six-dimension"), l = D({
      currentLeftData: [],
      // 左侧数据
      currentRightData: []
      // 右侧数据
    }), c = () => {
      const t = h($(f), a.dataAlias), { titleKey: p, valueKey: m, unitKey: y } = t;
      let u = [];
      a.sideData && a.sideData.length && (u = a.sideData.map((i) => ({
        ...i,
        title: i[p],
        unit: i[y],
        value: i[m]
      })), l.currentLeftData = u.slice(0, 3), l.currentRightData = u.slice(3, 6));
    }, d = () => {
      let t = "";
      return e.value.wrapper && (t += r(e.value.wrapper, `.${n}`)), e.value.leftWrapper && (t += r(e.value.leftWrapper, `.${n} .container .container-left`)), e.value.rightWrapper && (t += r(e.value.rightWrapper, `.${n} .container .container-right`)), e.value.centerWrapper && (t += r(e.value.centerWrapper, `.${n} .container .container-center`)), e.value.centerValue && (t += r(e.value.centerValue, `.${n} .container .container-center .value`)), e.value.centerUnit && (t += r(e.value.centerUnit, `.${n} .container .container-center .unit`)), t;
    }, o = () => {
      const t = document.querySelector(`.${n} > .style-container`);
      t && (t.innerHTML = `<style>${d()}</style>`);
    };
    return S(() => {
      o(), c();
    }), s(
      () => [a.cStyle],
      () => {
        o();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), s(
      () => [a.sideData, a.dataAlias],
      () => {
        c();
      },
      {
        immediate: !0
      }
    ), {
      curStyle: e,
      domOnlyClass: n,
      ...W(l)
    };
  }
});
export {
  R as default
};
