// @ths/design/es/components/six-dimension/src/index.vue.js
import c from "./index.vue2.js";
import { resolveComponent as m, openBlock as n, createElementBlock as s, normalizeClass as p, createElementVNode as t, Fragment as i, renderList as o, createBlock as d, normalizeStyle as y, toDisplayString as u } from "vue";
import "./index.vue3.js";
import f from "../../../_virtual/_plugin-vue_export-helper.js";
const S = { class: "container" }, v = { class: "container-left side-common" }, g = ["title"], h = { class: "unit" }, k = { class: "container-right side-common" };
function C(e, a, V, $, B, D) {
  const r = m("t-six-dimension-item");
  return n(), s("div", {
    class: p(["t-six-dimension", e.domOnlyClass])
  }, [
    t("div", S, [
      t("div", v, [
        (n(!0), s(i, null, o(e.currentLeftData, (l) => (n(), d(r, {
          key: l,
          "item-data": l,
          "parent-class": e.domOnlyClass,
          "c-style": {
            wrapper: e.curStyle.leftItemWrapper || { default: {} },
            value: e.curStyle.sideValue || { default: {} },
            title: e.curStyle.sideTitle || { default: {} },
            unit: e.curStyle.sideUnit || { default: {} }
          }
        }, null, 8, ["item-data", "parent-class", "c-style"]))), 128))
      ]),
      t("div", {
        class: "container-center g-flex-center",
        style: y({ backgroundImage: `url(${e.centerBg})` })
      }, [
        t("span", {
          class: "value",
          title: e.centerValue
        }, u(e.centerValue || "-"), 9, g),
        t("span", h, u(e.centerUnit || "-"), 1)
      ], 4),
      t("div", k, [
        (n(!0), s(i, null, o(e.currentRightData, (l) => (n(), d(r, {
          key: l,
          "item-data": l,
          "parent-class": e.domOnlyClass,
          "c-style": {
            wrapper: e.curStyle.rightItemWrapper || { default: {} },
            value: e.curStyle.sideValue || { default: {} },
            title: e.curStyle.sideTitle || { default: {} },
            unit: e.curStyle.sideUnit || { default: {} }
          }
        }, null, 8, ["item-data", "parent-class", "c-style"]))), 128))
      ])
    ]),
    a[0] || (a[0] = t("div", { class: "style-container" }, null, -1))
  ], 2);
}
const w = /* @__PURE__ */ f(c, [["render", C]]);
export {
  w as default
};
