// @ths/design/es/components/six-dimension/src/item.vue2.js
import { defineComponent as s, computed as n, onMounted as r, watch as c } from "vue";
import { getPollutionHtml as d, setStyle as a } from "../../../utils/index.js";
import { POLLUTION_HTML as f } from "../../../utils/constant.js";
const x = s({
  name: "TSixDimensionItem",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 值
        value: {
          default: {}
        },
        // 名称
        title: {
          default: {}
        },
        // 单位
        unit: {
          default: {}
        }
      })
    },
    // 单项数据
    itemData: {
      type: Object,
      default: () => ({
        bgUrl: "",
        // 整体背景url
        title: "",
        // 名称
        value: "",
        // 值
        unit: ""
        // 单位
      })
    },
    // 父级class
    parentClass: {
      type: String,
      default: () => ""
    }
  },
  emits: [""],
  setup(i, { emit: v }) {
    const e = n(() => i.cStyle), m = n(() => i.itemData), l = n(() => i.parentClass), o = () => {
      let t = "";
      return e.value.value && (t += a(e.value.value, `.${l.value} .t-six-dimension-item .item-value__box .value`)), e.value.unit && (t += a(e.value.unit, `.${l.value} .t-six-dimension-item .item-value__box .unit`)), e.value.title && (t += a(e.value.title, `.${l.value} .t-six-dimension-item .item-title`)), t;
    }, u = () => {
      const t = document.querySelector(`.${l.value} .t-six-dimension-item > .style-container`);
      t && (t.innerHTML = `<style>${o()}</style>`);
    };
    return r(() => {
      u();
    }), c(() => [i.cStyle], () => {
      u();
    }, {
      deep: !0,
      immediate: !0
    }), {
      curStyle: e,
      curItemData: m,
      getPollutionHtml: d,
      POLLUTION_HTML: f
    };
  }
});
export {
  x as default
};
