// @ths/design/es/components/select-law-person/src/index.vue.js
import _ from "./index.vue2.js";
import { resolveComponent as s, openBlock as r, createBlock as i, withCtx as t, createElementVNode as n, createElementBlock as h, createCommentVNode as m, createVNode as l, withDirectives as p, vShow as g, Fragment as v, renderList as F, createTextVNode as b, toDisplayString as d, normalizeClass as k, withModifiers as D } from "vue";
import "./index.vue3.js";
import $ from "../../../_virtual/_plugin-vue_export-helper.js";
const z = { class: "content-container" }, H = { class: "content-container__lf" }, L = { class: "form-container" }, M = { class: "content-container__rt" }, O = { class: "selected-header" }, j = { class: "selected-count" }, q = { class: "scroll-container" }, G = {
  key: 0,
  class: "form-container"
}, J = { class: "checkbox-container" }, K = ["title"], Q = ["title", "onClick", "onDblclick"], W = { class: "company-info" }, X = ["title"], Y = ["title"], Z = ["onClick"], x = { class: "footer-button" }, ee = { class: "footer-button__lf" }, ae = { class: "footer-button__rt" };
function le(e, o, oe, te, ne, re) {
  const w = s("t-dialog-single-tree-input"), u = s("el-form-item"), f = s("el-col"), N = s("el-option"), S = s("el-select"), y = s("el-input"), T = s("t-search-button"), E = s("el-row"), P = s("el-form"), c = s("el-table-column"), B = s("el-table"), I = s("el-config-provider"), U = s("t-paging"), A = s("el-checkbox"), V = s("t-button"), R = s("el-dialog");
  return r(), i(R, {
    modelValue: e.dialogVisible,
    "onUpdate:modelValue": o[12] || (o[12] = (a) => e.dialogVisible = a),
    width: "70%",
    title: "选择执法人员",
    "close-on-click-modal": !1,
    "close-on-press-escape": !1,
    "append-to-body": "",
    "custom-class": "user-multiple-dialog",
    onClose: e.handleClose
  }, {
    footer: t(() => [
      n("div", x, [
        n("div", ee, [
          e.hasRevertPrev ? (r(), h("span", {
            key: 0,
            class: "revert-button",
            onClick: o[11] || (o[11] = (...a) => e.handleRevert && e.handleRevert(...a))
          }, "恢复默认")) : m("", !0)
        ]),
        n("div", ae, [
          l(V, {
            text: "取消",
            onClick: e.handleClose
          }, null, 8, ["onClick"]),
          l(V, {
            text: "确定",
            type: "primary",
            onClick: e.handleSure
          }, null, 8, ["onClick"])
        ])
      ])
    ]),
    default: t(() => [
      n("div", z, [
        n("div", H, [
          n("div", L, [
            l(P, {
              ref: "tableFormRef",
              model: e.tableFormParams,
              "status-icon": "",
              "label-width": "80px",
              class: "change-form"
            }, {
              default: t(() => [
                l(E, null, {
                  default: t(() => [
                    p(l(f, { span: 12 }, {
                      default: t(() => [
                        l(u, {
                          label: "单位",
                          prop: "orgId"
                        }, {
                          default: t(() => [
                            l(w, {
                              selectedId: e.tableFormParams.orgId,
                              "onUpdate:selectedId": o[0] || (o[0] = (a) => e.tableFormParams.orgId = a),
                              selectedName: e.tableFormParams.orgName,
                              "onUpdate:selectedName": o[1] || (o[1] = (a) => e.tableFormParams.orgName = a),
                              class: "dialog-input",
                              "dialog-title": e.unitSingleTreeParams.dialogTitle,
                              "default-expand-layer": e.unitSingleTreeParams.defaultExpandLayer,
                              url: e.unitSingleTreeParams.url,
                              "root-id": e.unitSingleTreeParams.rootId,
                              "simple-data": e.unitSingleTreeParams.simpleData,
                              "result-data-path": e.unitSingleTreeParams.resultDataPath,
                              param: e.unitSingleTreeParams.param,
                              placeholder: e.unitSingleTreeParams.placeholder
                            }, null, 8, ["selectedId", "selectedName", "dialog-title", "default-expand-layer", "url", "root-id", "simple-data", "result-data-path", "param", "placeholder"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    }, 512), [
                      [g, e.searCriteriaConfig.unitSearch]
                    ]),
                    p(l(f, { span: 12 }, {
                      default: t(() => [
                        l(u, { label: "部门" }, {
                          default: t(() => [
                            l(w, {
                              selectedId: e.tableFormParams.deptId,
                              "onUpdate:selectedId": o[2] || (o[2] = (a) => e.tableFormParams.deptId = a),
                              selectedName: e.tableFormParams.deptName,
                              "onUpdate:selectedName": o[3] || (o[3] = (a) => e.tableFormParams.deptName = a),
                              class: "dialog-input",
                              "dialog-title": e.deptSingleTreeParams.dialogTitle,
                              "default-expand-layer": e.deptSingleTreeParams.defaultExpandLayer,
                              url: e.deptSingleTreeParams.url,
                              "root-id": e.deptSingleTreeParams.rootId,
                              "simple-data": e.deptSingleTreeParams.simpleData,
                              "result-data-path": e.deptSingleTreeParams.resultDataPath,
                              param: e.deptSingleTreeParams.param,
                              placeholder: e.deptSingleTreeParams.placeholder
                            }, null, 8, ["selectedId", "selectedName", "dialog-title", "default-expand-layer", "url", "root-id", "simple-data", "result-data-path", "param", "placeholder"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    }, 512), [
                      [g, e.searCriteriaConfig.deptSearch]
                    ]),
                    p(l(f, { span: 12 }, {
                      default: t(() => [
                        l(u, {
                          label: "岗位",
                          prop: "positionCode"
                        }, {
                          default: t(() => [
                            l(S, {
                              modelValue: e.tableFormParams.positionCode,
                              "onUpdate:modelValue": o[4] || (o[4] = (a) => e.tableFormParams.positionCode = a),
                              clearable: !0,
                              placeholder: "请选择岗位"
                            }, {
                              default: t(() => [
                                (r(!0), h(v, null, F(e.jobOptions, (a) => (r(), i(N, {
                                  key: a.enforcePostCode,
                                  label: a.enforcePostName,
                                  value: a.enforcePostCode
                                }, null, 8, ["label", "value"]))), 128))
                              ]),
                              _: 1
                            }, 8, ["modelValue"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    }, 512), [
                      [g, e.searCriteriaConfig.postSearch]
                    ]),
                    p(l(f, { span: 12 }, {
                      default: t(() => [
                        l(u, {
                          label: "人员标签",
                          prop: "personTag"
                        }, {
                          default: t(() => [
                            l(S, {
                              modelValue: e.tableFormParams.personTag,
                              "onUpdate:modelValue": o[5] || (o[5] = (a) => e.tableFormParams.personTag = a),
                              clearable: !0,
                              placeholder: "请选择人员标签"
                            }, {
                              default: t(() => [
                                (r(!0), h(v, null, F(e.personTagOptions, (a) => (r(), i(N, {
                                  key: a.value,
                                  label: a.label,
                                  value: a.value
                                }, null, 8, ["label", "value"]))), 128))
                              ]),
                              _: 1
                            }, 8, ["modelValue"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    }, 512), [
                      [g, e.searCriteriaConfig.userTagSearch]
                    ]),
                    p(l(f, { span: 12 }, {
                      default: t(() => [
                        l(u, {
                          label: "关键字",
                          prop: "keyword"
                        }, {
                          default: t(() => [
                            l(y, {
                              modelValue: e.tableFormParams.keyword,
                              "onUpdate:modelValue": o[6] || (o[6] = (a) => e.tableFormParams.keyword = a),
                              type: "text",
                              maxlength: "50",
                              placeholder: "请输入相关内容",
                              clearable: ""
                            }, null, 8, ["modelValue"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    }, 512), [
                      [g, e.searCriteriaConfig.keywordSearch]
                    ]),
                    l(f, {
                      span: e.searchBtnSpan,
                      class: "flex-right"
                    }, {
                      default: t(() => [
                        l(u, null, {
                          default: t(() => [
                            l(T, {
                              "is-show-fold-btn": !1,
                              "search-show-all": !0,
                              onSearch: e.handleSearch,
                              onReset: e.handleReset
                            }, null, 8, ["onSearch", "onReset"])
                          ]),
                          _: 1
                        })
                      ]),
                      _: 1
                    }, 8, ["span"])
                  ]),
                  _: 1
                })
              ]),
              _: 1
            }, 8, ["model"])
          ]),
          l(I, { locale: e.locale }, {
            default: t(() => [
              l(B, {
                ref: "userTable",
                size: "mini",
                data: e.page.list,
                "tooltip-effect": "dark",
                border: "",
                stripe: "",
                "row-key": "userId",
                onSelectAll: e.handleSelectionAllChange,
                onSelect: e.handleSelectionChange
              }, {
                default: t(() => [
                  e.tableColumnConfig.selectColumn ? (r(), i(c, {
                    key: 0,
                    align: "center",
                    fixed: "",
                    type: "selection",
                    "reserve-selection": !0
                  })) : m("", !0),
                  e.tableColumnConfig.name ? (r(), i(c, {
                    key: 1,
                    align: "left",
                    "min-width": "100",
                    label: "姓名",
                    prop: "userName",
                    "show-overflow-tooltip": ""
                  }, {
                    default: t(({ row: a }) => [
                      b(d(e.handleNoVal(a.userName)), 1)
                    ]),
                    _: 1
                  })) : m("", !0),
                  e.tableColumnConfig.unit ? (r(), i(c, {
                    key: 2,
                    align: "left",
                    "min-width": "180",
                    label: "单位",
                    prop: "orgName",
                    "show-overflow-tooltip": ""
                  }, {
                    default: t(({ row: a }) => [
                      b(d(e.handleNoVal(a.orgName)), 1)
                    ]),
                    _: 1
                  })) : m("", !0),
                  e.tableColumnConfig.dept ? (r(), i(c, {
                    key: 3,
                    align: "left",
                    "min-width": "140",
                    label: "部门",
                    prop: "deptName",
                    "show-overflow-tooltip": ""
                  }, {
                    default: t(({ row: a }) => [
                      b(d(e.handleNoVal(a.deptName)), 1)
                    ]),
                    _: 1
                  })) : m("", !0),
                  e.tableColumnConfig.post ? (r(), i(c, {
                    key: 4,
                    align: "left",
                    "min-width": "140",
                    label: "岗位",
                    prop: "positionName",
                    "show-overflow-tooltip": ""
                  }, {
                    default: t(({ row: a }) => [
                      b(d(e.handleNoVal(a.positionName)), 1)
                    ]),
                    _: 1
                  })) : m("", !0),
                  e.tableColumnConfig.lawEnforcement ? (r(), i(c, {
                    key: 5,
                    align: "center",
                    "min-width": "152",
                    label: "执法证号",
                    prop: "enforceCardCode",
                    "show-overflow-tooltip": ""
                  }, {
                    default: t(({ row: a }) => [
                      b(d(e.handleNoVal(a.enforceCardCode)), 1)
                    ]),
                    _: 1
                  })) : m("", !0)
                ]),
                _: 1
              }, 8, ["data", "onSelectAll", "onSelect"])
            ]),
            _: 1
          }, 8, ["locale"]),
          l(U, {
            page: e.page,
            total: e.page.total,
            "pager-count": 5,
            class: "dialog-pagination",
            layout: "total, prev, pager, next",
            onHandleSearch: e.getTableData,
            "onUpdate:pageNum": o[7] || (o[7] = (a) => e.page.pageNum = a),
            "onUpdate:pageSize": o[8] || (o[8] = (a) => e.page.pageSize = a)
          }, null, 8, ["page", "total", "onHandleSearch"])
        ]),
        o[14] || (o[14] = n("div", { class: "content-container__center" }, null, -1)),
        n("div", M, [
          n("div", O, [
            n("div", j, " 已选:" + d(e.selectedPersonList.length) + "个 ", 1)
          ]),
          n("div", q, [
            e.hasChangeForm ? (r(), h("div", G, [
              l(P, {
                ref: "changeFormRef",
                model: e.changeForm,
                "status-icon": "",
                "label-width": "80px",
                class: "change-form"
              }, {
                default: t(() => [
                  l(u, {
                    label: "变更理由",
                    prop: "changereason"
                  }, {
                    default: t(() => [
                      l(S, {
                        modelValue: e.changeForm.changereason,
                        "onUpdate:modelValue": o[9] || (o[9] = (a) => e.changeForm.changereason = a),
                        clearable: !0,
                        placeholder: "请选择变更理由",
                        disabled: e.changeInfoDisabled
                      }, {
                        default: t(() => [
                          (r(!0), h(v, null, F(e.changeReasonOptions, (a) => (r(), i(N, {
                            key: a.dictionaryCode,
                            label: a.dictionaryName,
                            value: a.dictionaryCode
                          }, null, 8, ["label", "value"]))), 128))
                        ]),
                        _: 1
                      }, 8, ["modelValue", "disabled"])
                    ]),
                    _: 1
                  }),
                  l(u, {
                    label: "变更说明",
                    prop: "changeexplain"
                  }, {
                    default: t(() => [
                      l(y, {
                        modelValue: e.changeForm.changeexplain,
                        "onUpdate:modelValue": o[10] || (o[10] = (a) => e.changeForm.changeexplain = a),
                        rows: 2,
                        autosize: {
                          minRows: 2
                        },
                        type: "textarea",
                        maxlength: "150",
                        "show-word-limit": "",
                        placeholder: "请输入变更说明",
                        disabled: e.changeInfoDisabled,
                        class: "textarea-modify"
                      }, null, 8, ["modelValue", "disabled"])
                    ]),
                    _: 1
                  })
                ]),
                _: 1
              }, 8, ["model"])
            ])) : m("", !0),
            n("div", {
              class: k(["selected-person", { "m-t-12": !e.hasChangeForm }])
            }, [
              (r(!0), h(v, null, F(e.selectedPersonList, (a) => (r(), h("div", {
                key: a.userId,
                class: k(["person-item", { "is-director": a.isleader === e.DirectorNameValue.yes.code }])
              }, [
                n("div", J, [
                  l(A, {
                    onChange: (C) => e.handleSelectedCheckbox(C, a)
                  }, {
                    default: t(() => [
                      n("span", {
                        class: "person-name",
                        title: e.handleTitleConvert(a.userName)
                      }, d(e.handleNoVal(a.userName)), 9, K),
                      n("span", null, d(a.isleader === e.DirectorNameValue.yes.code ? `（${e.dutyName}）` : ""), 1)
                    ]),
                    _: 2
                  }, 1032, ["onChange"]),
                  p(n("span", {
                    class: "text-button",
                    title: `设为${e.dutyName}`,
                    onClick: D((C) => e.handleSetDirector(a), ["stop"]),
                    onDblclick: D((C) => e.handleSetDirector(a), ["stop"])
                  }, "设为" + d(e.dutyName), 41, Q), [
                    [g, e.isShowHead && a.isleader !== e.DirectorNameValue.yes.code]
                  ])
                ]),
                n("div", W, [
                  p(n("span", {
                    class: "company-name name",
                    title: e.handleTitleConvert(a.orgName || a.deptName)
                  }, d(e.handleNoVal(a.orgName || a.deptName)), 9, X), [
                    [g, e.tableColumnConfig.dept]
                  ]),
                  p(n("span", {
                    class: "dept-name name",
                    title: e.handleTitleConvert(a.positionName)
                  }, d(e.handleNoVal(a.positionName)), 9, Y), [
                    [g, e.tableColumnConfig.post]
                  ]),
                  n("span", {
                    class: "delete-button",
                    onClick: (C) => e.handlePersonSingleRemove(a)
                  }, o[13] || (o[13] = [
                    n("svg", {
                      xmlns: "http://www.w3.org/2000/svg",
                      viewBox: "0 0 1024 1024"
                    }, [
                      n("path", {
                        fill: "currentColor",
                        d: `M160 256H96a32 32 0 0 1 0-64h256V95.936a32 32 0 0 1 32-32h256a32
                    32 0 0 1 32 32V192h256a32 32 0 1 1 0 64h-64v672a32 32 0 0 1-32 32H192a32
                    32 0 0 1-32-32zm448-64v-64H416v64zM224 896h576V256H224zm192-128a32 32 0 0
                    1-32-32V416a32 32 0 0 1 64 0v320a32 32 0 0 1-32 32m192 0a32 32 0 0 1-32-32V416a32 32 0 0 1 64 0v320a32 32 0 0 1-32 32`
                      })
                    ], -1)
                  ]), 8, Z)
                ])
              ], 2))), 128))
            ], 2)
          ])
        ])
      ])
    ]),
    _: 1
  }, 8, ["modelValue", "onClose"]);
}
const me = /* @__PURE__ */ $(_, [["render", le], ["__scopeId", "data-v-1eb0a732"]]);
export {
  me as default
};
