// @ths/design/es/components/select-law-person/src/index.vue2.js
import { defineComponent as Q, ref as C, reactive as X, watch as Y, onMounted as Z, toRefs as ee, nextTick as S } from "vue";
import { ElMessage as h } from "element-plus";
import ae from "element-plus/lib/locale/lang/zh-cn";
import te from "@ths/c-common-base-paging";
import { TButton as re, TSearchButton as se } from "@ths/c-common-base-button";
import ne from "../../dialog-single-tree-input/src/index.vue.js";
import { handleNoVal as L } from "./utils/common.js";
import { IC_SELECT_LAW_PERSON_INTERFACE as x, getDictList as P, getCurrentOrg as oe, getUserList as de } from "./services/law-person.js";
const he = Q({
  name: "TSelectLawPerson",
  components: {
    TPaging: te,
    TButton: re,
    TDialogSingleTreeInput: ne,
    TSearchButton: se
  },
  props: {
    // 请求根地址
    apiRootPath: {
      type: String,
      required: !0,
      default: ""
    },
    // 排除人员ID（逗号拼接的字符串），在请求单位、部门、人员列表接口中带入
    excludeUserIds: {
      type: String,
      default: () => ""
    },
    // appCode参数，在请求单位、部门、人员列表接口中带入
    appCode: {
      type: String,
      default: ""
    },
    // 请求单位、部门、人员列表接口额外需要的参数
    extraParams: {
      type: Object,
      default: () => {
      }
    },
    // 选中人员数据
    personData: {
      type: Array,
      default: () => []
    },
    // 人员初始默认值（用于恢复默认值时，由于目前需求是恢复原始值，所以单独新开了一个变量）
    initialPersonData: {
      type: Array,
      default: () => []
    },
    // 是否需要变更表单
    hasChangeForm: {
      type: Boolean,
      default: () => !1
    },
    // 是否需要恢复默认功能
    hasRevertPrev: {
      type: Boolean,
      default: () => !0
    },
    // 最少选择人数
    minSelectedPersonNum: {
      type: Number,
      default: 0
    },
    // 职责名称（右侧选中人员职责名称）
    dutyName: {
      type: String,
      default: "负责人"
    },
    // 返回数据在 data 中的位置，如：mapList
    resultDataPath: {
      type: String,
      default: "data"
    },
    // 单位信息
    org: {
      type: Object,
      default: () => {
      }
    },
    // 控制查询条件显隐
    searCriteriaConfig: {
      type: Object,
      default: () => ({
        unitSearch: !0,
        deptSearch: !0,
        postSearch: !0,
        userTagSearch: !0,
        keywordSearch: !0
      })
    },
    // 控制表格列显隐
    tableColumnConfig: {
      type: Object,
      default: () => ({
        selectColumn: !0,
        name: !0,
        unit: !0,
        dept: !0,
        post: !0,
        lawEnforcement: !0
      })
    },
    // 查询按钮所占的每行的份数(如果查询条件有奇数个建议12，偶数个建议24)
    searchBtnSpan: {
      type: Number,
      default: 12
    },
    // 是否显示负责人
    isShowHead: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["on-close", "on-sure"],
  setup(s, { emit: p }) {
    const i = C(), I = C(), o = {
      yes: {
        code: "1",
        // 是负责人
        name: "是"
      },
      no: {
        code: "0",
        // 不是负责人
        name: "否"
      }
    }, T = {
      isleader: "0",
      isleaderName: "否"
    }, y = {
      keyword: "",
      positionCode: "",
      // 所属岗位
      orgId: "",
      // 所属部门
      // orgName: '',
      personTag: "",
      deptId: ""
      // deptName: '',
    }, w = "POSITION_DIC", J = "PERSON_TAG", E = "UPDATE_REASON", a = X({
      // 弹窗显示
      dialogVisible: !0,
      // 分页数据和列表数据
      page: {
        pageNum: 1,
        pageSize: 7
      },
      // 是否全屏
      fullScreen: !1,
      // 表格 顶部左侧列 按钮的显隐控制
      tableLeftBtnSetting: {
        densityBtn: !1,
        // 表格间距设置（默认为 true）
        fullScreenBtn: !1
        // 表格全屏（默认为 true）
      },
      // 表格筛选条件
      tableFormParams: JSON.parse(JSON.stringify(y)),
      // 表格筛选条件-岗位下拉数据
      jobOptions: [],
      personTagOptions: [],
      // 单位单选树
      unitSingleTreeParams: {
        dialogTitle: "选择单位",
        defaultExpandLayer: 1,
        url: s.apiRootPath + x.getOrgList,
        rootId: "",
        simpleData: {
          idKey: "orgId",
          // 数据中id对应的key
          nameKey: "orgName",
          // 要展示的名称
          pIdKey: "orgParentId",
          // 数据中父节点标识key，懒加载时候使用
          spanKey: ""
          // 在节点后拼接内容 key
        },
        resultDataPath: "data",
        param: {
          appCode: s.appCode,
          excludeUserIds: s.excludeUserIds,
          ...s.extraParams
        },
        placeholder: "请选择单位"
      },
      // 部门单选树
      deptSingleTreeParams: {
        dialogTitle: "选择部门",
        defaultExpandLayer: 1,
        url: s.apiRootPath + x.getDeptList,
        rootId: "",
        simpleData: {
          idKey: "deptId",
          // 数据中id对应的key
          nameKey: "deptName",
          // 要展示的名称
          pIdKey: "deptParentId",
          // 数据中父节点标识key，懒加载时候使用 PARENT
          spanKey: ""
          // 在节点后拼接内容 key
        },
        resultDataPath: "data",
        param: {
          appCode: s.appCode,
          excludeUserIds: s.excludeUserIds,
          orgId: "",
          ...s.extraParams
        },
        placeholder: "请选择部门"
      },
      // 表格选中数据
      tableSelectedData: JSON.parse(JSON.stringify(s.personData)),
      // 默认选中人员数据
      defaultSelectedPersonData: JSON.parse(JSON.stringify(s.initialPersonData)).sort(
        (e, t) => Number(t.isleader) - Number(e.isleader)
      ),
      // 已选中人员
      selectedPersonList: JSON.parse(JSON.stringify(s.personData)),
      // 变更理由填写表单
      changeForm: {
        changeexplain: "",
        changereason: ""
      },
      // 变更理由
      changeReasonOptions: [],
      // 变更理由、变更说明是否可以编辑（选中的人员与初始数据不一致时，可以编辑，反之不可以）
      changeInfoDisabled: !0
    }), v = (e) => L(e), F = (e, t) => a.selectedPersonList.findIndex((n) => n.userId === e.userId) === -1, N = () => {
      var e;
      (e = i.value) == null || e.clearSelection(), a.tableSelectedData.forEach((t) => {
        const r = a.page.list.find((n) => n.userId === t.userId);
        r && S(() => i.value.toggleRowSelection(r, !0));
      });
    };
    function U(e, t) {
      let r = e, n = t;
      n = n.replace(/\[(\w+)\]/g, ".$1"), n = n.replace(/^\./, "");
      const d = n.split(".");
      for (let f = 0; f < d.length && r; f++) {
        const R = d[f];
        if (R in r)
          r = r[R];
        else
          break;
      }
      return r;
    }
    const B = async () => {
      a.changeReasonOptions = [];
      try {
        const e = await P(s.apiRootPath, {
          parentId: E
        });
        if (e != null && e.data && (e == null ? void 0 : e.status) === 200) {
          const { data: t } = e == null ? void 0 : e.data;
          a.changeReasonOptions = t.map((r) => ({
            dictionaryCode: r.dictionarycode,
            dictionaryName: r.dictionaryname
          }));
        }
      } catch (e) {
        console.error(e);
      }
    }, b = async () => {
      try {
        const e = await oe(s.apiRootPath, { appCode: s.appCode });
        if (e != null && e.data && (e == null ? void 0 : e.status) === 200) {
          const { data: t } = e == null ? void 0 : e.data;
          a.tableFormParams.orgId = t.orgId, a.tableFormParams.orgName = t.orgName;
        }
      } catch (e) {
        console.error(e);
      }
    }, j = async () => {
      a.jobOptions = [];
      try {
        const e = {
          parentId: w
        }, t = await P(s.apiRootPath, e);
        if (t != null && t.data && (t == null ? void 0 : t.status) === 200) {
          const { data: r } = t == null ? void 0 : t.data;
          a.jobOptions = r.map((n) => ({
            enforcePostCode: n.dictionarycode,
            enforcePostName: n.dictionaryname
          }));
        }
      } catch (e) {
        console.error(e);
      }
    }, k = async () => {
      a.personTagOptions = [];
      try {
        const e = {
          parentId: J
        }, t = await P(s.apiRootPath, e);
        if (t != null && t.data && (t == null ? void 0 : t.status) === 200) {
          const { data: r } = t == null ? void 0 : t.data;
          a.personTagOptions = r.map((n) => ({
            value: n.dictionarycode,
            label: n.dictionaryname
          }));
        }
      } catch (e) {
        console.error(e);
      }
    }, c = async () => {
      let e = {};
      e = await de(
        s.apiRootPath,
        {
          ...a.tableFormParams,
          pageSize: a.page.pageSize,
          pageNum: a.page.pageNum,
          appCode: s.appCode,
          excludeUserIds: s.excludeUserIds,
          ...s.extraParams
        }
      ), e != null && e.data && (e == null ? void 0 : e.status) === 200 ? a.page = U(e.data, s.resultDataPath) || {} : a.page = {}, N();
    }, O = (e) => {
      a.selectedPersonList = a.selectedPersonList.filter((r) => r.userId !== e.userId);
      const t = a.selectedPersonList.find((r) => r.userId === e.userId);
      t && t.isleader === o.yes.code && m();
    }, A = (e, t) => {
      const r = a.tableSelectedData.findIndex((n) => n.userId === t.userId);
      r !== -1 ? (a.tableSelectedData.splice(r, 1), O(t)) : (a.tableSelectedData.push(JSON.parse(JSON.stringify(t))), u());
    }, _ = (e) => {
      e.length ? e.forEach((t) => {
        a.tableSelectedData.findIndex((n) => n.userId === t.userId) === -1 && (a.tableSelectedData.push(JSON.parse(JSON.stringify(t))), u());
      }) : a.page.list.forEach((t) => {
        const r = a.tableSelectedData.findIndex((n) => n.userId === t.userId);
        a.tableSelectedData.splice(r, 1), O(t);
      });
    }, l = () => {
      a.changeInfoDisabled = JSON.stringify(a.selectedPersonList) === JSON.stringify(a.defaultSelectedPersonData);
    }, u = () => {
      if (!a.tableSelectedData.length) {
        h.error("请选择需要加入的人员");
        return;
      }
      a.tableSelectedData.length ? a.tableSelectedData.forEach((e) => {
        const t = a.defaultSelectedPersonData.findIndex((d) => d.userId === e.userId), r = a.defaultSelectedPersonData[t];
        if (a.selectedPersonList.findIndex((d) => d.userId === e.userId) === -1) {
          const d = t !== -1 ? {
            ...r,
            ...T
          } : e;
          a.selectedPersonList.push(d);
        }
      }) : a.page.list.forEach((e) => {
        const t = a.selectedPersonList.findIndex((r) => r.userId === e.userId);
        a.selectedPersonList.splice(t, 1);
      }), l();
    }, K = (e, t) => {
      t.isRemove = e;
    }, g = () => {
      a.selectedPersonList.sort((e, t) => Number(t.isleader) - Number(e.isleader));
    }, z = (e) => {
      a.selectedPersonList.forEach((t) => {
        t.isleader = o.no.code, t.isleaderName = o.no.name;
      }), e.isleader = o.yes.code, e.isleaderName = o.yes.name, g(), l();
    }, m = () => {
      a.selectedPersonList.length && (a.selectedPersonList[0].isleader = o.yes.code, a.selectedPersonList[0].isleaderName = o.yes.name), l();
    }, V = () => {
      const e = a.selectedPersonList.filter((t) => t.isRemove);
      if (!e.length) {
        h.error("请选择需要移除的人员");
        return;
      }
      e.forEach((t) => {
        const r = a.page.list.find((n) => n.userId === t.userId);
        r && S(() => i.value.toggleRowSelection(r, !1));
      }), a.selectedPersonList = a.selectedPersonList.filter((t) => !t.isRemove), a.tableSelectedData = JSON.parse(JSON.stringify(a.selectedPersonList)), m();
    }, M = (e) => {
      const t = a.page.list.find((r) => r.userId === e.userId);
      t && S(() => i.value.toggleRowSelection(t, !1)), a.selectedPersonList = a.selectedPersonList.filter((r) => r.userId !== e.userId), a.tableSelectedData = JSON.parse(JSON.stringify(a.selectedPersonList)), e.isleader === o.yes.code && m();
    }, $ = () => {
      a.tableSelectedData = JSON.parse(JSON.stringify(a.defaultSelectedPersonData)), a.selectedPersonList = JSON.parse(JSON.stringify(a.defaultSelectedPersonData)), g(), N(), l();
    }, q = () => {
      p("on-close");
    }, G = () => {
      var t;
      if (a.selectedPersonList.length < Number(s.minSelectedPersonNum)) {
        h.error(`请至少选择${s.minSelectedPersonNum}个执法人员`);
        return;
      }
      const e = {
        ...a.changeForm,
        changeReasonName: ((t = a.changeReasonOptions.filter((r) => r.dictionaryCode === a.changeForm.changereason)[0]) == null ? void 0 : t.dictionaryName) || "",
        userUpdate: a.changeInfoDisabled ? "0" : "1"
      };
      p("on-sure", a.selectedPersonList, e);
    }, H = () => {
      a.page.pageNum = 1, c();
    }, D = async () => {
      var e;
      (e = s.org) != null && e.orgId ? (a.tableFormParams.orgId = s.org.orgId, a.tableFormParams.orgName = s.org.orgName) : await b(), c();
    }, W = async () => {
      await I.value.resetFields(), a.tableFormParams = JSON.parse(JSON.stringify(y)), D();
    };
    return Y(() => a.tableFormParams.orgId, (e) => {
      a.deptSingleTreeParams.param.orgId = e;
    }), Z(async () => {
      a.page.pageNum = 1, a.page.pageSize = 7, s.searCriteriaConfig.postSearch && await j(), s.searCriteriaConfig.userTagSearch && await k(), D(), s.hasChangeForm && B(), g(), l();
    }), {
      locale: {
        ...ae
      },
      ...ee(a),
      handleClose: q,
      handleSelectionAllChange: _,
      handleSelectionChange: A,
      getTableData: c,
      getUnitOptions: b,
      userTable: i,
      tableFormRef: I,
      handlePersonJoin: u,
      handlePersonRemove: V,
      handleSelectedCheckbox: K,
      handlePersonSingleRemove: M,
      handleSetDirector: z,
      handleSure: G,
      handleRevert: $,
      handleRowSelectable: F,
      handleNoVal: L,
      DirectorNameValue: o,
      handleSearch: H,
      handleReset: W,
      handleTitleConvert: v
    };
  }
});
export {
  he as default
};
