// @ths/design/es/components/year-contrast/src/index.vue.js
import v from "./index.vue2.js";
import { openBlock as l, createElementBlock as o, normalizeClass as a, createElementVNode as e, toDisplayString as d, Fragment as c, renderList as i } from "vue";
import "./index.vue3.js";
import k from "../../../_virtual/_plugin-vue_export-helper.js";
const m = { class: "title" }, f = { class: "menu" }, h = ["onClick"], C = { class: "menu-item-text" }, _ = ["onClick"];
function w(t, r, g, y, $, D) {
  return l(), o("div", {
    class: a(["t-year-contrast", t.domOnlyClass])
  }, [
    e("div", m, d(t.title), 1),
    e("div", f, [
      (l(!0), o(c, null, i(t.curData, (n, u) => (l(), o("div", {
        key: u,
        class: "menu-collapse"
      }, [
        e("div", {
          class: "menu-item",
          onClick: (s) => t.showContent(n)
        }, [
          e("span", C, d(n.date), 1),
          e("div", {
            class: a(["menu-item-arrow", { rotate: n == null ? void 0 : n.clicked }])
          }, r[0] || (r[0] = [
            e("svg", {
              xmlns: "http://www.w3.org/2000/svg",
              viewBox: "0 0 1024 1024"
            }, [
              e("path", {
                fill: "currentColor",
                d: `M831.872 340.864 512 652.672 192.128 340.864a30.592 30.592 0 0 0-42.752 0 29.12 29.12 0 0 0 0 41.6L489.664 714.24a32 32 0 0 0
                44.672 0l340.288-331.712a29.12 29.12 0 0 0 0-41.728 30.592 30.592 0 0 0-42.752 0z`
              })
            ], -1)
          ]), 2)
        ], 8, h),
        e("div", {
          class: a(["menu-content", { hidden: !(n != null && n.clicked) }])
        }, [
          (l(!0), o(c, null, i(n.children, (s, p) => (l(), o("div", {
            key: p,
            class: a(["menu-children", { active: s == null ? void 0 : s.clicked, disabled: t.selectedDates.length === 2 && !(s != null && s.clicked) }]),
            onClick: (B) => t.addDate(s)
          }, [
            e("span", null, d(s.date), 1)
          ], 10, _))), 128))
        ], 2)
      ]))), 128))
    ]),
    r[1] || (r[1] = e("div", { class: "style-container" }, null, -1))
  ], 2);
}
const F = /* @__PURE__ */ k(v, [["render", w]]);
export {
  F as default
};
