// @ths/design/es/components/year-contrast/src/index.vue2.js
import { defineComponent as k, computed as D, ref as v, onMounted as M, watch as f } from "vue";
import O from "lodash/merge";
import y from "lodash/cloneDeep";
import { ElMessage as h, ElMessageBox as p } from "element-plus";
import { getDomRandomClass as x, getCssPropertyString as B, setStyle as r } from "../../../utils/index.js";
const T = {
  date: "date",
  children: "children"
}, L = k({
  name: "TYearContrast",
  props: {
    // 是否出提示框
    showTip: {
      type: Boolean,
      default: !1
    },
    // 标题
    title: {
      type: String,
      default: "年份预览"
    },
    // 选中的数据
    selectedData: {
      type: Array,
      default: () => []
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => y(T),
      require: !1
    },
    // 选中一个时的提示文案
    selectOneText: {
      type: String,
      default: "当前已选中一个时间，请再选中一个时间进行对比"
    },
    // 选中两个时的提示文案
    selectTwoText: {
      type: String,
      default: "是否进行比对"
    },
    // 选中两个后取消一个时的提示文案
    cancelOneText: {
      type: String,
      default: "是否取消选中时间"
    },
    // 确定框提示类型
    promptIconType: {
      type: String,
      default: "warning"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        title: {
          default: {}
        },
        menu: {
          default: {}
        },
        menuCollapse: {
          default: {}
        },
        menuItem: {
          default: {}
        },
        menuItemText: {
          default: {}
        },
        menuItemArrow: {
          default: {}
        },
        menuContent: {
          default: {}
        },
        subMenuItem: {
          default: {},
          active: {}
        }
      })
    }
  },
  /**
   * active-double-emit: 选择两个后抛出的事件
   */
  emits: ["active-double-emit", "cancel-one"],
  setup(u, { emit: s }) {
    const t = D(() => u.cStyle), c = x("year-contrast"), o = v([]), l = v([]), m = (e) => {
      try {
        const a = O(T, u.alias);
        if (!Array.isArray(e))
          return;
        e.forEach((n, A) => {
          typeof n != "object" ? e[A] = {
            date: n
          } : (n.date = n[a.date], n.children = n[a.children]), n != null && n.children && n.children.length && m(n.children);
        });
      } catch (a) {
        console.warn(a);
      }
    }, g = (e) => {
      e != null && e.clicked ? e.clicked = !1 : e.clicked = !0;
    }, i = (e) => {
      if (l.value.includes(e)) {
        const a = l.value.indexOf(e);
        l.value.splice(a, 1);
      }
    }, I = (e) => {
      l.value.length === 2 ? p.confirm(
        `${u.cancelOneText}`,
        "提示",
        {
          confirmButtonText: "确定",
          cancelButtonText: "取消",
          type: u.promptIconType
        }
      ).then(() => {
        e.clicked = !1, i(e.date), s("cancel-one", l.value);
      }).catch(() => !1) : (e.clicked = !1, i(e.date));
    }, w = (e) => {
      const a = l.value.length;
      a === 0 ? (e && (e.clicked = !0, l.value.push(e.date)), h(u.selectOneText)) : a >= 1 && p.confirm(
        `${u.selectTwoText}`,
        "提示",
        {
          confirmButtonText: "确定",
          cancelButtonText: "取消",
          type: u.promptIconType
        }
      ).then(() => {
        e && (e.clicked = !0, l.value.push(e.date)), s("active-double-emit", l.value);
      }).catch(() => !1);
    }, S = (e) => {
      l.value.length === 2 && !(e != null && e.clicked) || (e != null && e.clicked ? u.showTip ? I(e) : (e.clicked = !1, i(e.date)) : u.showTip ? w(e) : (e && (e.clicked = !0, l.value.push(e.date)), l.value.length === 2 && s("active-double-emit", l.value)));
    }, $ = (e) => {
      Array.isArray(e) && (o.value.forEach((a) => {
        a.children.forEach((n) => {
          (n.date === ((e == null ? void 0 : e[0]) ?? "") || n.date === ((e == null ? void 0 : e[1]) ?? "")) && (n.clicked = !0, l.value.push(n.date));
        });
      }), l.value.length === 1 && u.showTip && h(u.selectOneText), l.value.length === 2 && s("active-double-emit", l.value));
    }, b = () => {
      var a, n;
      let e = "";
      return (a = t.value.subMenuItem) != null && a.active && (e += `
          .${c} .menu .menu-collapse .menu-content .menu-children.active {
            ${B((n = t.value.subMenuItem) == null ? void 0 : n.active)}
          }
        `), e;
    }, C = () => {
      let e = "";
      return t.value.wrapper && (e += r(t.value.wrapper, `.${c}`)), t.value.title && (e += r(t.value.title, `.${c} .title`)), t.value.menu && (e += r(t.value.menu, `.${c} .menu`)), t.value.menuCollapse && (e += r(t.value.menuCollapse, `.${c} .menu .menu-collapse`)), t.value.menuItem && (e += r(t.value.menuItem, `.${c} .menu .menu-collapse .menu-item`)), t.value.menuItemText && (e += r(t.value.menuItemText, `.${c} .menu .menu-collapse .menu-item .menu-item-text`)), t.value.menuItemArrow && (e += r(t.value.menuItemArrow, `.${c} .menu .menu-collapse .menu-item .menu-item-arrow`)), t.value.menuContent && (e += r(t.value.menuContent, `.${c} .menu .menu-collapse .menu-content`)), t.value.subMenuItem && (e += r(t.value.subMenuItem, `.${c} .menu .menu-collapse .menu-content .menu-children`)), e;
    }, d = () => {
      const e = document.querySelector(`.${c} .style-container`);
      e && (e.innerHTML = `<style>${C()}${b()}</style>`);
    };
    return M(() => {
      d();
    }), f(() => u.cStyle, () => {
      d();
    }, { deep: !0 }), f([() => u.data, () => u.alias], () => {
      o.value = y(u.data), m(o.value);
    }, {
      deep: !0,
      immediate: !0
    }), f(() => u.selectedData, () => {
      l.value = [], $(u.selectedData);
    }, {
      immediate: !0
    }), {
      curStyle: t,
      domOnlyClass: c,
      curData: o,
      showContent: g,
      addDate: S,
      selectedDates: l
    };
  }
});
export {
  L as default
};
