// @ths/design/es/components/chart-line-single/src/index.vue2.js
import { defineComponent as te, ref as ae, computed as V, reactive as le, nextTick as E, onMounted as ne, onUnmounted as ie, watch as z, toRefs as re } from "vue";
import * as se from "echarts";
import d from "lodash/merge";
import $ from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as oe, DEFAULT_OPTIONS_AXIS as ce } from "../../../utils/constant.js";
import { getDomRandomClass as ye, setStyle as j } from "../../../utils/index.js";
const me = d($(oe), $(ce)), O = d(
  $(me),
  {
    grid: {
      top: "13%",
      bottom: "10%",
      left: "6%",
      right: "4%"
    },
    xAxis: {
      boundaryGap: !0
    },
    tooltip: {
      formatter: (e) => {
        let v = "";
        const { name: h } = e[0];
        return e.forEach((r) => {
          r.componentSubType === "line" && (v += `<div style="margin: 10px 0 0;line-height: 1">${r.marker}<span>${r.seriesName}</span>
          <span style="float: right;margin-left: 20px;">${r.value}</span><br/></div>`);
        }), `${h}<br/>${v}`;
      },
      textStyle: {
        fontWeight: "normal"
      }
    },
    legend: {
      itemStyle: {
        borderWidth: 0
      }
    }
  }
), Se = te({
  name: "TChartLineSingle",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 暂无数据容器
        emptyContainer: {
          default: {}
        },
        // 暂无数据图片
        emptyImg: {
          default: {}
        },
        // 暂无数据文字
        emptyText: {
          default: {}
        }
      })
    },
    // 设置主轴
    mainAxis: {
      type: String,
      default: "x"
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => O,
      required: !1
    },
    // 传入数组数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 标记线属性配置
    markLineList: {
      type: Array,
      default: () => []
    },
    // 标记线数据
    markLineObj: {
      type: Object,
      default: () => ({})
    },
    // 折线主题配置
    line: {
      type: Object,
      default: () => ({})
    },
    // 是否显示背景
    showItemBg: {
      type: Boolean,
      default: !1
    },
    // 背景条样式
    backgroundStyle: {
      type: Object,
      default: () => ({})
    }
  },
  // 方法名定义
  emits: ["com-click", "legend-select"],
  setup(e, { emit: v }) {
    const h = ye("chart-line-single"), r = ae(null);
    let x = null, S = null;
    const f = V(() => e.cStyle), a = le({
      // 图表数据
      chartData: [],
      chartBgData: [],
      // 主题配置
      lineOptions: $(e.line)
    }), B = () => {
      var t, c, s;
      (c = (t = e.seriesSetting) == null ? void 0 : t.data) != null && c.length && e.data ? (a.chartData = e.seriesSetting.data.map((g) => {
        const { name: y, dataKey: o } = g;
        return {
          name: y || "--",
          data: e.data.map((b) => b[o])
        };
      }), a.chartBgData = a.chartData) : (a.chartData = [], a.chartBgData = ((s = e.options) == null ? void 0 : s.series) || a.chartData);
    }, Y = () => {
      let t = "";
      return f.value.wrapper && (t += j(f.value.wrapper, `.${h}`)), f.value.emptyContainer && (t += j(f.value.emptyContainer, `.${h} .empty`)), f.value.emptyImg && (t += j(f.value.emptyImg, `.${h} .empty .empty-img`)), f.value.emptyText && (t += j(f.value.emptyText, `.${h} .empty .empty-text`)), t;
    }, _ = () => {
      const t = document.querySelector(`.${h} .style-container`);
      t && (t.innerHTML = `<style>${Y()}</style>`);
    }, I = () => {
      var o, b, q, M, N;
      const { color: t, series: c } = e.options;
      let s = [], g = [];
      c ? s = c.map((l, n) => {
        var u, k, L, D, C, w, F;
        const m = d(
          {
            type: "line",
            itemStyle: {
              color: t && t[n] || O.color[n] || "#FFF"
            },
            lineStyle: {
              color: t && t[n] || O.color[n] || "#FFF"
            },
            emphasis: {
              focus: "series"
            },
            smooth: !0
          },
          a == null ? void 0 : a.lineOptions,
          (L = (k = (u = e.seriesSetting) == null ? void 0 : u.data) == null ? void 0 : k[n]) == null ? void 0 : L.style
        );
        return (w = (C = (D = e.seriesSetting) == null ? void 0 : D.data) == null ? void 0 : C[n]) != null && w.isOpenLineColor || "color" in (m == null ? void 0 : m.lineStyle) && ((F = m == null ? void 0 : m.lineStyle) == null || delete F.color), d({}, m, l);
      }) : e.data && ((b = (o = e.seriesSetting) == null ? void 0 : o.data) != null && b.length) && (s = (q = e.seriesSetting) == null ? void 0 : q.data.map((l, n) => {
        var k, L, D, C, w, F, R, H, P, U, W, p, K;
        const m = [];
        e.markLineList.forEach((i) => {
          var G, X;
          ((G = i == null ? void 0 : i.mark) == null ? void 0 : G.series) === l.dataKey && m.push({
            yAxis: e.markLineObj[(X = i == null ? void 0 : i.mark) == null ? void 0 : X.markAlias],
            label: {
              formatter: (T) => {
                var J, Q;
                return `{text|${((J = i == null ? void 0 : i.label) == null ? void 0 : J.text) ?? ""}}{num|${(T == null ? void 0 : T.value) ?? ""}}{unit|${((Q = i == null ? void 0 : i.label) == null ? void 0 : Q.unit) ?? ""}}`;
              },
              ...i.label
            },
            lineStyle: {
              ...i.lineStyle
            }
          });
        });
        const u = d(
          {
            name: (k = a.chartData[n]) == null ? void 0 : k.name,
            type: "line",
            label: {
              show: !1
            },
            symbol: "emptyCircle",
            itemStyle: {
              color: t && t[n] || O.color[n] || "#FFF"
            },
            lineStyle: {
              color: t && t[n] || O.color[n] || "#FFF"
            },
            emphasis: {
              focus: "series"
            },
            smooth: !0,
            data: e.data.length ? (L = a.chartData[n]) == null ? void 0 : L.data : [],
            markLine: {
              symbol: [(D = l == null ? void 0 : l.style) == null ? void 0 : D.symbolStart, (C = l == null ? void 0 : l.style) == null ? void 0 : C.symbolEnd],
              symbolSize: [((w = l.style) == null ? void 0 : w.symbolWidth) ?? 5, ((F = l.style) == null ? void 0 : F.symbolHeight) ?? 5],
              data: m
            }
          },
          a == null ? void 0 : a.lineOptions,
          (P = (H = (R = e.seriesSetting) == null ? void 0 : R.data) == null ? void 0 : H[n]) == null ? void 0 : P.style
        );
        return (p = (W = (U = e.seriesSetting) == null ? void 0 : U.data) == null ? void 0 : W[n]) != null && p.isOpenLineColor || "color" in (u == null ? void 0 : u.lineStyle) && ((K = u == null ? void 0 : u.lineStyle) == null || delete K.color), d({}, u, l);
      }), (M = e.seriesSetting) != null && M.xAxisLabel ? g = (N = e.data) != null && N.length ? e.data.map((l) => l[e.seriesSetting.xAxisLabel]) : [] : g = e.data.map((l, n) => n)), a.chartBgData.length && e.showItemBg && s.push({
        name: "",
        type: "bar",
        showBackground: !0,
        backgroundStyle: d({
          color: {
            type: "linear",
            x: 0,
            y: 0,
            x2: 0,
            y2: 1,
            colorStops: [{
              offset: 0,
              color: "rgba(180, 180, 180, 0.2)"
              // 0% 处的颜色
            }, {
              offset: 1,
              color: "rgba(180, 180, 180, 0.2)"
              // 100% 处的颜色
            }]
          }
        }, e == null ? void 0 : e.backgroundStyle),
        data: e.data.length ? Array(e.data.length).fill(0) : []
      });
      let y;
      e.mainAxis === "x" ? y = d(
        {
          series: s,
          xAxis: {
            data: g
          }
        },
        O
      ) : e.mainAxis === "y" && (y = d(
        {
          series: s,
          yAxis: {
            data: g
          }
        },
        O
      )), y = d(y, e.options), e.mainAxis === "y" && (y.xAxis.type = "value", y.yAxis.type = "category"), S.clear();
      try {
        S.setOption(y);
      } catch (l) {
        console.log(l);
      }
    }, A = () => {
      E(() => {
        S.resize();
      });
    };
    function Z() {
      r.value && (x = new ResizeObserver((t) => {
        for (const c of t)
          c.target === r.value && A();
      }), x.observe(r.value));
    }
    const ee = V(() => {
      var t, c, s, g;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((o) => {
            var b;
            return (o == null ? void 0 : o.data) && ((b = o == null ? void 0 : o.data) == null ? void 0 : b.length) !== 0;
          });
        if (typeof ((c = e.options) == null ? void 0 : c.series) == "object")
          return ((g = (s = e.options.series) == null ? void 0 : s.data) == null ? void 0 : g.length) === 0;
      }
      return e.data.length ? (E(() => {
        A();
      }), !0) : !1;
    });
    return ne(() => {
      _(), r.value && (S = se.init(r.value), B(), I(), S.on("click", (t) => {
        v("com-click", t);
      }), S.on("legendselectchanged", (t) => {
        v("legend-select", t);
      }), window.addEventListener("resize", A), Z());
    }), ie(() => {
      window.removeEventListener("resize", A), x && (x.disconnect(), x = null);
    }), z(
      [() => e.line, () => e.backgroundStyle, () => e.mainAxis],
      () => {
        a.lineOptions = $(e.line), I();
      },
      {
        deep: !0
      }
    ), z(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting,
        () => e.showItemBg,
        () => e.markLineList,
        () => e.markLineObj
      ],
      () => {
        B(), I(), A();
      },
      {
        deep: !0
      }
    ), z(
      () => e.cStyle,
      () => {
        _(), E(() => {
          S.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      curStyle: f,
      ...re(a),
      chartId: r,
      checkEmpty: ee,
      domOnlyClass: h
    };
  }
});
export {
  Se as default
};
