// @ths/design/es/components/chart-line-single/src/index.vue.js
import a from "./index.vue2.js";
import { openBlock as m, createElementBlock as c, normalizeClass as p, normalizeStyle as o, withDirectives as i, createElementVNode as t, vShow as n } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  ref: "chartId",
  class: "chart"
}, y = { class: "empty" }, f = ["src"];
function h(e, s, v, E, g, k) {
  var l, r;
  return m(), c("div", {
    class: p(["t-chart-line-single", e.domOnlyClass]),
    style: o({ ...(r = (l = e.curStyle) == null ? void 0 : l.wrapper) == null ? void 0 : r.default })
  }, [
    i(t("div", u, null, 512), [
      [n, e.checkEmpty]
    ]),
    i(t("div", y, [
      t("img", {
        class: "empty-img",
        src: e.emptyImg,
        style: o({
          "object-fit": e.objectFit
        }),
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [n, !e.checkEmpty]
    ]),
    s[1] || (s[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const C = /* @__PURE__ */ d(a, [["render", h]]);
export {
  C as default
};
