// @ths/design/es/components/header-tabs/src/index.vue.js
import g from "./index.vue2.js";
import { openBlock as a, createElementBlock as r, normalizeClass as l, normalizeStyle as i, createElementVNode as s, Fragment as o, renderList as c, toDisplayString as d, createCommentVNode as h } from "vue";
import "./index.vue3.js";
import m from "../../../_virtual/_plugin-vue_export-helper.js";
const p = {
  ref: "headerWrapper",
  class: "header"
}, k = ["onClick"], f = ["onClick"];
function y(e, u, N, I, C, w) {
  return a(), r("div", {
    class: l(["t-header-tabs", e.domOnlyClass]),
    style: i({
      "background-image": e.curBgImg ? `url(${e.curBgImg})` : ""
    })
  }, [
    s("div", p, [
      e.curLeftData.length ? (a(), r("div", {
        key: 0,
        class: "left",
        style: i({
          width: e.sideWidth + "px"
        })
      }, [
        (a(!0), r(o, null, c(e.curLeftData, (t, n) => (a(), r("div", {
          key: n,
          class: l(["item left-item", e.curActiveNav === t.value ? "active" : ""]),
          style: i({
            height: e.curNavItem.height,
            width: e.curNavItem.width,
            "margin-left": e.curNavItem.space,
            "background-image": e.getItemBgImg(t)
          }),
          onClick: (v) => e.handleNavClick(t)
        }, [
          s("div", {
            class: l(["text", e.curActiveNav === t.value ? "active" : ""])
          }, d(t.label), 3)
        ], 14, k))), 128))
      ], 4)) : h("", !0),
      s("div", {
        class: "center",
        style: i({
          width: e.curTitle.width,
          height: e.curTitle.height
        })
      }, d(e.curTitle.text), 5),
      e.curRightData.length ? (a(), r("div", {
        key: 1,
        class: "right",
        style: i({
          width: e.sideWidth + "px"
        })
      }, [
        (a(!0), r(o, null, c(e.curRightData, (t, n) => (a(), r("div", {
          key: n,
          class: l(["item right-item", e.curActiveNav === t.value ? "active" : ""]),
          style: i({
            height: e.curNavItem.height,
            width: e.curNavItem.width,
            "margin-right": e.curNavItem.space,
            "background-image": e.getItemBgImg(t)
          }),
          onClick: (v) => e.handleNavClick(t)
        }, [
          s("div", {
            class: l(["text", e.curActiveNav === t.value ? "active" : ""])
          }, d(t.label), 3)
        ], 14, f))), 128))
      ], 4)) : h("", !0)
    ], 512),
    u[0] || (u[0] = s("div", { class: "style-container" }, null, -1))
  ], 6);
}
const A = /* @__PURE__ */ m(g, [["render", y]]);
export {
  A as default
};
