// @ths/design/es/components/header-tabs/src/index.vue2.js
import { defineComponent as T, ref as f, computed as u, watch as h, nextTick as D, onMounted as L } from "vue";
import { getDomRandomClass as R, setStyle as n, getCssPropertyString as d } from "../../../utils/index.js";
const H = T({
  name: "THeaderTabs",
  props: {
    /* 高亮导航项 */
    activeValue: {
      type: String,
      default: ""
    },
    /* 左侧菜单数据 */
    leftData: {
      type: Array,
      default: () => []
    },
    /* 右侧菜单数据 */
    rightData: {
      type: Array,
      default: () => []
    },
    /* 导航按钮属性设置 */
    navItem: {
      type: Object,
      default: () => ({
        width: 249,
        height: 80,
        space: -19,
        backgroundImage: "",
        activeBackgroundImage: ""
      })
    },
    /* 标题设置 */
    title: {
      type: Object,
      default: () => ({
        text: "",
        width: 914,
        height: 80
      })
    },
    /* 背景图片 */
    backgroundImage: {
      type: String,
      default: ""
    },
    /* 自定义组件样式 */
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 导航项容器
        navItemWrapper: {
          default: {},
          hover: {},
          active: {}
        },
        /* 导航项文字 */
        navItemText: {
          default: {},
          hover: {},
          active: {}
        },
        /* 标题 */
        title: {
          default: {}
        }
      })
    }
  },
  emits: ["nav-item-click"],
  setup(r, { emit: y }) {
    const a = R("header-tabs"), m = f(), t = u(() => r.cStyle), v = f(r.activeValue), p = u(() => {
      const { width: e, height: i, text: l } = r.title;
      return {
        width: (typeof e == "number" ? `${e}px` : e) || "0px",
        height: (typeof i == "number" ? `${i}px` : i) || "0px",
        text: l || "这里是标题"
      };
    }), g = u(() => r.leftData), o = u(() => r.rightData), c = u(() => {
      const {
        width: e,
        height: i,
        space: l,
        backgroundImage: S,
        activeBackgroundImage: w
      } = r.navItem;
      return {
        width: (typeof e == "number" ? `${e}px` : e) || "0px",
        height: (typeof i == "number" ? `${i}px` : i) || "0px",
        space: (typeof l == "number" ? `${l}px` : l) || "0px",
        backgroundImage: S,
        activeBackgroundImage: w
      };
    }), $ = u(() => r.backgroundImage), I = f(0);
    h(() => [r.leftData, r.rightData, m.value, r.navItem, r.title], () => {
      D(() => {
        var e, i;
        if ((e = m.value) != null && e.clientWidth) {
          const l = ((i = m.value) == null ? void 0 : i.clientWidth) - Number(typeof p.value.width == "number" ? p.value.width : p.value.width.replace("px", ""));
          I.value = g.value.length && o.value.length ? l / 2 : l;
        }
      });
    }, {
      immediate: !0,
      deep: !0
    }), h(() => r.activeValue, (e) => {
      e !== v.value && (v.value = e);
    });
    const W = (e) => v.value === e.value && c.value.activeBackgroundImage ? `url(${c.value.activeBackgroundImage})` : v.value !== e.value && c.value.backgroundImage ? `url(${c.value.backgroundImage})` : "", x = () => {
      let e = "";
      return t.value.wrapper && (e += n(t.value.wrapper, `.${a}`)), t.value.navItemWrapper && (e += n(t.value.navItemWrapper, `.${a} .header .item`)), t.value.navLeftWrapper && (e += n(t.value.navLeftWrapper, `.${a} .header .left`)), t.value.navRightWrapper && (e += n(t.value.navRightWrapper, `.${a} .header .right`)), t.value.navItemLeftWrapper && (e += n(t.value.navItemLeftWrapper, `.${a} .header .item.left-item`)), t.value.navItemRightWrapper && (e += n(t.value.navItemRightWrapper, `.${a} .header .item.right-item`)), t.value.navItemText && (e += n(t.value.navItemText, `.${a} .header .item .text`)), t.value.title && (e += n(t.value.title, `.${a} .header .center`)), e;
    }, b = () => {
      let e = "";
      return t.value.navItemWrapper && t.value.navItemWrapper.active && (e += `
            .${a}  .header .item.active {
              ${d(t.value.navItemWrapper.active)}
            }
          `), t.value.navItemLeftWrapper && t.value.navItemLeftWrapper.active && (e += `
            .${a}  .header .item.left-item.active {
              ${d(t.value.navItemLeftWrapper.active)}
            }
          `), t.value.navItemRightWrapper && t.value.navItemRightWrapper.active && (e += `
            .${a}  .header .item.right-item.active {
              ${d(t.value.navItemRightWrapper.active)}
            }
          `), t.value.navItemText && t.value.navItemText.active && (e += `
            .${a} .header .item .text.active {
              ${d(t.value.navItemText.active)}
            }
          `), e;
    }, s = () => {
      const e = document.querySelector(`.${a} .style-container`);
      e && (e.innerHTML = `<style>${x()}${b()}</style>`);
    };
    L(() => {
      s();
    });
    const k = (e) => {
      v.value = e.value, y("nav-item-click", e);
    };
    return h(() => r.cStyle, () => {
      s();
    }), {
      curStyle: t,
      headerWrapper: m,
      sideWidth: I,
      handleNavClick: k,
      domOnlyClass: a,
      curTitle: p,
      curActiveNav: v,
      curLeftData: g,
      curRightData: o,
      curNavItem: c,
      curBgImg: $,
      getItemBgImg: W
    };
  }
});
export {
  H as default
};
