// @ths/design/es/components/dialog-multiple-tree/src/index.vue.js
import m from "./index.vue2.js";
import { resolveComponent as o, openBlock as u, createBlock as _, withCtx as l, createElementVNode as i, createVNode as t, createTextVNode as s, toDisplayString as f, mergeProps as g } from "vue";
import "./index.vue3.js";
import h from "../../../_virtual/_plugin-vue_export-helper.js";
const C = { class: "dialog-body" }, b = { class: "dialog-main" }, k = { class: "dialog-button-box" };
function v(e, n, y, N, T, $) {
  const a = o("info-filled"), r = o("el-icon"), c = o("t-tree-multiple"), d = o("el-button"), p = o("el-dialog");
  return u(), _(p, {
    "model-value": e.dialogVisible,
    "close-on-press-escape": !1,
    "before-close": e.handleCancel,
    "destroy-on-close": "",
    "append-to-body": "",
    width: e.dialogWidth,
    "custom-class": "business-tree-dialog"
  }, {
    title: l(() => [
      i("span", null, [
        t(r, { size: 16 }, {
          default: l(() => [
            t(a)
          ]),
          _: 1
        }),
        s(" " + f(e.dialogTitle), 1)
      ])
    ]),
    default: l(() => [
      i("div", C, [
        i("div", b, [
          t(c, g({
            ref: "multipleTreeRef",
            "root-id": "root",
            "result-data-path": "mapList"
          }, e.$attrs, { onGetCheckedNodes: e.handleTreeNodeClick }), null, 16, ["onGetCheckedNodes"])
        ]),
        i("div", k, [
          t(d, {
            size: "small",
            onClick: e.handleCancel
          }, {
            default: l(() => n[0] || (n[0] = [
              s(" 取消 ")
            ])),
            _: 1,
            __: [0]
          }, 8, ["onClick"]),
          t(d, {
            size: "small",
            type: "primary",
            onClick: e.handleSure
          }, {
            default: l(() => n[1] || (n[1] = [
              s(" 确定 ")
            ])),
            _: 1,
            __: [1]
          }, 8, ["onClick"])
        ])
      ])
    ]),
    _: 1
  }, 8, ["model-value", "before-close", "width"]);
}
const B = /* @__PURE__ */ h(m, [["render", v]]);
export {
  B as default
};
