// @ths/design/es/components/dialog-multiple-tree/src/index.vue2.js
import { defineComponent as r, reactive as n, ref as p, watch as u, toRefs as c } from "vue";
import f from "./t-tree-multiple.vue.js";
const C = r({
  name: "TDialogMultipleTree",
  components: {
    TTreeMultiple: f
  },
  props: {
    // 弹窗是否显示
    dialogVisible: {
      type: Boolean,
      default: !1
    },
    // 弹窗标题
    dialogTitle: {
      type: String,
      default: "选择"
    },
    // 弹窗宽度
    dialogWidth: {
      type: String,
      default: "500px"
    }
  },
  emits: ["update:dialogVisible", "dialogClose", "dialogSure"],
  setup(g, { emit: l, attrs: m }) {
    const t = n({
      selectedData: []
    }), a = p();
    u(() => {
      var e, o;
      return (o = (e = a.value) == null ? void 0 : e.multipleTree) == null ? void 0 : o.getCheckedNodes();
    }, (e) => {
      t.selectedData = e;
    }, {
      immediate: !0,
      deep: !0
    });
    const i = () => {
      l("update:dialogVisible", !1), l("dialogClose");
    }, d = (e) => {
      t.selectedData = e;
    }, s = () => {
      l("dialogSure", t.selectedData);
    };
    return {
      multipleTreeRef: a,
      ...c(t),
      handleSure: s,
      handleTreeNodeClick: d,
      handleCancel: i
    };
  }
});
export {
  C as default
};
