// @ths/design/es/components/dialog-multiple-tree/src/t-tree-multiple.vue2.js
import { defineComponent as S, ref as w, reactive as z, computed as P, watch as o, toRefs as B } from "vue";
import j from "qs";
import k from "./utils/http.js";
const G = S({
  name: "TTreeMultiple",
  props: {
    // 请求 url 地址
    url: {
      type: String,
      default: "",
      required: !0
    },
    // 请求方式（GET、POST），目前只支持这两种方式
    type: {
      type: String,
      default: "GET"
    },
    // 请求参数
    param: {
      type: Object,
      default: () => ({})
    },
    // 是否使用 requestBody 接收参数（post 请求时使用），如果传 false，则会使用 qs.stringify() 对参数进行序列化处理
    useRequestBody: {
      type: Boolean,
      default: !0
    },
    // 返回数据在 data 中的位置，如：mapList
    resultDataPath: {
      type: String,
      default: "mapList"
    },
    // 主要业务字段
    simpleData: {
      type: Object,
      default: () => ({
        idKey: "treeId",
        // 数据中id对应的key
        nameKey: "treeName",
        // 要展示的名称
        pIdKey: "treePId",
        // 数据中父节点标识key
        iconKey: "icon",
        // 树节点图标key
        spanKey: "appendInfo",
        // 在节点后拼接内容key
        leafKey: "isLeaf",
        // 是否是父节点key
        disabled: "disabled"
      })
    },
    // 根节点id，默认 null
    rootId: {
      type: String,
      default: void 0
    },
    // 树节点前的图标
    defaultIcon: {
      type: Object,
      default: () => ({
        expandIcon: "folder-opened",
        // 展开图标
        collapseIcon: "folder"
        // 关闭时图标，叶子节点统一使用关闭时图标
      })
    },
    // 是否默认展开所有节点,默认 false
    defaultExpandAll: {
      type: Boolean,
      default: !1
    },
    // 是否展开选中节点
    openCheckedNode: {
      type: Boolean,
      default: !0
    },
    // 默认展开的层数，默认值 0
    defaultExpandLayer: {
      type: Number,
      default: 0
    },
    // 是否只允许选中叶子节点，默认 false
    onlyCheckLeaf: {
      type: Boolean,
      default: !1
    },
    // 是否是懒加载
    lazy: {
      type: Boolean,
      default: !1
    },
    // 逗号分隔的默认选中的数据id
    hiddenId: {
      type: String,
      default: ""
    },
    // 是否严格遵循父子不关联
    checkStrictly: {
      type: Boolean,
      default: !1
    },
    // 是否开启结构线
    openStructureLine: {
      type: Boolean,
      default: !0
    },
    // 根据那个字段搜索
    searchParams: {
      type: String,
      default: "TREE_NAME"
    },
    // 是否使用 elementplus 之外的自定义图标
    isCustomIcon: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["pre-load-tree", "get-checked-nodes"],
  setup(e, { emit: y, attrs: h }) {
    const u = w(), a = z({
      // 默认展开的节点 key，根据展开的层数计算得来
      defaultExpandKeys: [],
      // 点击节点的时候是否展开，在只允许勾选子节点时候展开
      expandOnClickNode: !1,
      // 树对应的 prop 属性，正常加载和懒加载不同，在初始化页面时初始化该值
      treeProps: {},
      // 平行结构数据
      listNodes: [],
      // 树型数据，根据 listNodes 转换来的数据
      treeNodes: [],
      // 是否展示加载动画
      loading: !1,
      // 异步树第一个节点内容
      lazyNodes: [],
      // 内容为空的时候展示的文本
      emptyText: "暂无数据",
      // 默认选中的节点
      defaultCheckedKeys: P(() => e.hiddenId.length > 0 ? e.hiddenId.split(",") : []),
      // 是否父子关联
      relationParent: !1,
      // 是否默认展开全部节点
      isDefaultExpandAll: P(() => e.hiddenId.length > 0 && e.lazy ? !1 : e.defaultExpandAll),
      filterText: ""
    }), E = () => {
      y("get-checked-nodes", u.value.getCheckedNodes());
    };
    function m(t, i) {
      let l = t, n = i;
      n = n.replace(/\[(\w+)\]/g, ".$1"), n = n.replace(/^\./, "");
      const d = n.split(".");
      for (let f = 0; f < d.length && l; f++) {
        const r = d[f];
        if (r in l)
          l = l[r];
        else
          break;
      }
      return l;
    }
    const g = (t) => new Promise((i, l) => {
      try {
        i(
          k.get(e.url, {
            params: t
          })
        );
      } catch (n) {
        l(n);
      }
    }), N = (t) => {
      let i = t;
      return e.useRequestBody || (i = j.stringify(t)), new Promise((l, n) => {
        try {
          l(k.post(e.url, i));
        } catch (d) {
          n(d);
        }
      });
    }, K = async (t) => {
      const i = { ...e.param };
      t.level === 0 ? i[e.simpleData.pIdKey] = e.rootId : Object.keys(e.param).forEach((n) => {
        i[n] = t.data[e.param[n]] ? t.data[e.param[n]] : i[n];
      });
      let l;
      if (e.type === "GET" && (l = await g(i)), e.type === "POST" && (l = await N(i)), l != null && l.data && (l == null ? void 0 : l.status) === 200) {
        const n = m(l.data, e.resultDataPath) || [];
        n.length > 0 ? h.onPreLoadTree ? y("pre-load-tree", n, (d) => {
          d.length > 0 && (d.forEach((f) => {
            f[e.simpleData.leafKey] === e.simpleData.leafValue ? f.isLeaf = !0 : f.isLeaf = !1;
          }), a.lazyNodes = d);
        }) : (n.forEach((d) => {
          d[e.simpleData.leafKey] === e.simpleData.leafValue ? d.isLeaf = !0 : d.isLeaf = !1;
        }), a.lazyNodes = n) : a.lazyNodes = [];
      } else
        a.lazyNodes = [];
    }, D = async (t, i) => {
      a.loading = !0, await K(t), i(a.lazyNodes), a.loading = !1;
    }, T = async () => {
      let t = {};
      e.type === "GET" ? t = await g(e.param) : e.type === "POST" && (t = await N(e.param)), t != null && t.data && (t == null ? void 0 : t.status) === 200 ? a.listNodes = m(t.data, e.resultDataPath) || [] : a.listNodes = [];
    }, b = (t) => {
      const i = [], { idKey: l, pIdKey: n } = e.simpleData;
      if (!Array.isArray(t))
        return i;
      const d = {};
      return t.forEach((f) => {
        f[l] && (d[f[l]] = f);
      }), t.forEach((f) => {
        const r = d[f[n]];
        r ? (r.children || (r.children = [])).push(f) : i.push(f);
      }), i;
    }, x = () => {
      let t = [];
      if (a.treeNodes.length > 0 && t.push(a.treeNodes[0]), t.length > 0)
        for (let i = 1; i <= e.defaultExpandLayer; i += 1) {
          const l = [];
          t.forEach((n) => {
            n.children && (e.openCheckedNode && a.defaultExpandKeys.push(n[e.simpleData.idKey]), n[e.simpleData.iconKey] || (n[e.simpleData.iconKey] = e.defaultIcon.expandIcon), n.children.forEach((d) => {
              l.push(d);
            }));
          }), t = l;
        }
      a.defaultCheckedKeys.forEach((i) => {
        e.openCheckedNode && a.defaultExpandKeys.push(i);
      });
    }, s = (t, i) => {
      t.length > 0 && t.forEach((l) => {
        l.children && (l.disabled = i, s(l.children, i));
      });
    }, L = async () => {
      a.treeNodes = await b(a.listNodes), x(), e.onlyCheckLeaf && (a.relationParent = !0, s(a.treeNodes, !0));
    }, I = async () => {
      a.loading = !0, a.treeProps = {
        label: e.simpleData.nameKey,
        children: "children",
        disabled: e.simpleData.disabled,
        isLeaf: e.simpleData.isLeaf
      }, a.relationParent = !e.checkStrictly, await T(), h.onPreLoadTree ? y("pre-load-tree", a.listNodes, (t) => {
        t.length > 0 && (a.listNodes = t), L();
      }) : L(), a.loading = !1;
    }, C = () => {
      u.value.filter(a.filterText);
    }, O = (t, i) => t ? i[e.simpleData.nameKey].indexOf(t) !== -1 : !0, c = () => {
      e.lazy ? (a.treeProps = {
        label: e.simpleData.nameKey,
        isLeaf: "isLeaf"
      }, a.emptyText = "", e.onlyCheckLeaf ? a.relationParent = !0 : a.relationParent = !e.checkStrictly) : I();
    };
    return c(), o(
      () => e.param,
      () => {
        e.lazy ? a.treeProps = {
          label: e.simpleData.nameKey,
          isLeaf: "isLeaf"
        } : c();
      }
    ), o(
      () => e.onlyCheckLeaf,
      (t) => {
        t ? s(a.treeNodes, !0) : s(a.treeNodes, !1);
      }
    ), o(
      () => e.checkStrictly,
      (t) => {
        e.onlyCheckLeaf ? a.relationParent = !0 : a.relationParent = !t;
      }
    ), o(() => e.rootId, (t) => {
      c();
    }), {
      ...B(a),
      getCheckedNodes: E,
      multipleTree: u,
      loadLazyTree: D,
      filterNode: O,
      initTree: c,
      filterNodeByText: C
    };
  }
});
export {
  G as default
};
