// @ths/design/es/components/dialog-multiple-tree/src/t-tree-multiple.vue.js
import T from "./t-tree-multiple.vue2.js";
import { resolveComponent as l, resolveDirective as K, openBlock as n, createElementBlock as u, normalizeClass as c, createVNode as r, withCtx as a, createTextVNode as m, withDirectives as N, createBlock as i, createElementVNode as d, createCommentVNode as f, resolveDynamicComponent as z, toDisplayString as y } from "vue";
import b from "../../../_virtual/_plugin-vue_export-helper.js";
const g = { class: "ths-tree__node" }, v = ["title"], B = { class: "ths-tree__node-append-text" };
function V(e, s, I, E, $, w) {
  const h = l("search"), p = l("el-icon"), k = l("el-button"), C = l("el-input"), D = l("el-tree"), _ = K("loading");
  return n(), u("div", {
    class: c({ "ths-structure-tree": e.openStructureLine })
  }, [
    r(C, {
      modelValue: e.filterText,
      "onUpdate:modelValue": s[0] || (s[0] = (t) => e.filterText = t),
      class: "m-t-sm m-b-xs",
      placeholder: "请输入关键字搜索",
      clearable: "",
      onClear: e.filterNodeByText
    }, {
      append: a(() => [
        r(k, {
          size: "small",
          onClick: e.filterNodeByText
        }, {
          default: a(() => [
            r(p, { size: 13 }, {
              default: a(() => [
                r(h)
              ]),
              _: 1
            }),
            s[1] || (s[1] = m(" 搜索 "))
          ]),
          _: 1,
          __: [1]
        }, 8, ["onClick"])
      ]),
      _: 1
    }, 8, ["modelValue", "onClear"]),
    N((n(), i(D, {
      ref: "multipleTree",
      "show-checkbox": "",
      "node-key": e.simpleData.idKey,
      data: e.treeNodes,
      props: e.treeProps,
      "default-expanded-keys": e.defaultExpandKeys,
      "default-expand-all": e.isDefaultExpandAll,
      "highlight-current": !0,
      "check-strictly": e.relationParent,
      "default-checked-keys": e.defaultCheckedKeys,
      lazy: e.lazy,
      indent: e.openStructureLine ? 0 : 20,
      load: e.loadLazyTree,
      "empty-text": e.emptyText,
      "filter-node-method": e.filterNode,
      onCheck: e.getCheckedNodes
    }, {
      default: a(({ node: t, data: o }) => [
        d("span", g, [
          d("span", {
            title: t.label
          }, [
            e.isCustomIcon ? (n(), u("i", {
              key: 0,
              class: c(`${o[e.simpleData.iconKey]} ths-tree__node-icon`)
            }, null, 2)) : f("", !0),
            e.isCustomIcon ? f("", !0) : (n(), i(p, {
              key: 1,
              size: 14,
              class: "ths-tree__node-icon"
            }, {
              default: a(() => [
                (n(), i(z(
                  o[e.simpleData.iconKey] ? o[e.simpleData.iconKey] : t.expanded ? e.defaultIcon.expandIcon : e.defaultIcon.collapseIcon
                )))
              ]),
              _: 2
            }, 1024)),
            m(" " + y(t.label), 1)
          ], 8, v),
          d("span", B, y(e.simpleData.spanKey && o[e.simpleData.spanKey] ? o[e.simpleData.spanKey] : ""), 1)
        ])
      ]),
      _: 1
    }, 8, ["node-key", "data", "props", "default-expanded-keys", "default-expand-all", "check-strictly", "default-checked-keys", "lazy", "indent", "load", "empty-text", "filter-node-method", "onCheck"])), [
      [_, e.loading]
    ])
  ], 2);
}
const P = /* @__PURE__ */ b(T, [["render", V]]);
export {
  P as default
};
