// @ths/design/es/components/tabs/src/index.vue2.js
import { defineComponent as L, ref as _, computed as v, watch as $, onMounted as W } from "vue";
import p from "lodash/cloneDeep";
import w from "lodash/merge";
import { filterAttrs as E, getDomRandomClass as P, getCssPropertyString as r, setStyle as c } from "../../../utils/index.js";
const h = {
  label: "label",
  name: "name",
  disabled: "disabled",
  iconSrc: "iconSrc",
  activeIconSrc: "activeIconSrc"
}, B = L({
  name: "TTabs",
  props: {
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 选中项
    activeTab: {
      type: String,
      default: ""
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => p(h),
      require: !1
    },
    // 选项卡所在位置(替换 element-plus 的 tab-position 属性)
    tabPositionExtend: {
      type: String,
      default: "top"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        prevIcon: {
          default: {},
          hover: {}
        },
        nextIcon: {
          default: {},
          hover: {}
        },
        itemWrapper: {
          default: {},
          hover: {},
          active: {},
          disabled: {}
        },
        itemActiveBorder: {
          default: {}
        },
        itemIcon: {
          default: {},
          hover: {},
          active: {}
        },
        itemLabel: {
          default: {},
          hover: {},
          active: {}
        }
      })
    }
  },
  emits: ["tab-click"],
  setup(i, { emit: y }) {
    const S = "50px", o = _(i.data), t = v(() => i.cStyle), a = P("tabs"), I = v(() => {
      var e, l;
      return (l = (e = i == null ? void 0 : i.tabPositionExtend) == null ? void 0 : e.toString()) == null ? void 0 : l.split("-")[0];
    }), T = v(() => {
      var e;
      return (e = i == null ? void 0 : i.tabPositionExtend) == null ? void 0 : e.toString();
    }), g = v(() => {
      var e;
      return i.activeTab || (((e = o == null ? void 0 : o.value[0]) == null ? void 0 : e.name) ?? "");
    }), m = _(!1), x = (e) => {
      const l = o.value.find((n) => e.props.name === n.name);
      y("tab-click", { ...l, item: l });
    }, A = () => {
      var l, n, s, b, d, f;
      let e = "";
      return (l = t.value.itemWrapper) != null && l.active && (e += `
          .${a} .el-tabs__item.is-active {
            ${r(t.value.itemWrapper.active)}
          }
        `), (n = t.value.itemWrapper) != null && n.disabled && (e += `
          .${a} .el-tabs__item.is-disabled {
            ${r(t.value.itemWrapper.disabled)}
          }
        `), (s = t.value.itemIcon) != null && s.active && (e += `
        .${a} .el-tabs__item.is-active .custom-tabs-label__icon {
            ${r(t.value.itemIcon.active)}
          }
        `), (b = t.value.itemLabel) != null && b.active && (e += `
          .${a} .el-tabs__item.is-active .custom-tabs-label__label {
            ${r(t.value.itemLabel.active)}
          }
        `), (d = t.value.itemIcon) != null && d.hover && (e += `
          .${a} .el-tabs__item:hover.not(.is-active) .custom-tabs-label__icon {
            ${r(t.value.itemIcon.hover)}
          }
        `), (f = t.value.itemLabel) != null && f.hover && (e += `
          .${a} .el-tabs__item:hover.not(.is-active) .custom-tabs-label__label {
            ${r(t.value.itemLabel.hover)}
          }
        `), e;
    }, C = () => {
      let e = "";
      return t.value.wrapper && (e += c(t.value.wrapper, `.${a}`), t.value.wrapper.default.height && (e += `
          .${a} .el-tabs__nav-prev,
          .${a} .el-tabs__nav-next{
            line-height: ${t.value.wrapper.default.height};
          }
        `)), t.value.itemWrapper && (e += c(t.value.itemWrapper, `.${a} .el-tabs__item`)), t.value.itemActiveBorder && (e += c(t.value.itemActiveBorder, `.${a}.t-tabs .el-tabs__item.is-active::after`)), t.value.prevIcon && (e += c(t.value.prevIcon, `.${a} .el-tabs__nav-prev`)), t.value.nextIcon && (e += c(t.value.nextIcon, `.${a} .el-tabs__nav-next`)), t.value.itemIcon && (e += c(t.value.itemIcon, `.${a} .el-tabs__item .custom-tabs-label__icon`)), t.value.itemLabel && (e += c(t.value.itemLabel, `.${a} .el-tabs__item .custom-tabs-label__label`)), e;
    }, u = () => {
      const e = document.querySelector(`.${a} .style-container`);
      e && (e.innerHTML = `<style>${C()}${A()}</style>`);
    };
    return $(() => i.data, () => {
      const e = w(p(h), i.alias);
      o.value = i.data.map((l) => {
        const n = l;
        return Object.keys(e).forEach((s) => {
          n[s] = l[e[s]];
        }), n;
      }), m.value = !m.value;
    }, {
      deep: !0,
      immediate: !0
    }), $(() => i.cStyle, () => {
      u();
    }), W(() => {
      u();
    }), {
      curStyle: t,
      curData: o,
      curActiveTab: g,
      elTabPosition: I,
      tTabPosition: T,
      domOnlyClass: a,
      refreshTab: m,
      filterAttrs: E,
      handleTabClick: x,
      ITEM_HEIGHT: S
    };
  }
});
export {
  B as default
};
