// @ths/design/es/components/tabs/src/index.vue.js
import C from "./index.vue2.js";
import { resolveComponent as f, openBlock as l, createElementBlock as o, normalizeClass as E, normalizeStyle as v, createBlock as T, mergeProps as g, withCtx as y, Fragment as _, renderList as $, createCommentVNode as A, createElementVNode as k, toDisplayString as I } from "vue";
import "./index.vue3.js";
import P from "../../../_virtual/_plugin-vue_export-helper.js";
const V = ["src"], H = ["title"];
function B(e, t, w, z, D, G) {
  var s, n, r, i, b, d, c, p, u, m;
  const S = f("el-tab-pane"), h = f("el-tabs");
  return l(), o("div", {
    class: E(["t-tabs", e.domOnlyClass]),
    style: v({
      "--font-line-height": (n = (s = e.tabPositionExtend) == null ? void 0 : s.toString()) != null && n.includes("top") || (i = (r = e.tabPositionExtend) == null ? void 0 : r.toString()) != null && i.includes("bottom") ? ((c = (d = (b = e.curStyle) == null ? void 0 : b.wrapper) == null ? void 0 : d.default) == null ? void 0 : c.height) ?? e.ITEM_HEIGHT : ((m = (u = (p = e.curStyle) == null ? void 0 : p.itemWrapper) == null ? void 0 : u.default) == null ? void 0 : m.height) ?? e.ITEM_HEIGHT
    })
  }, [
    (l(), T(h, g(e.filterAttrs(e.$attrs), {
      key: e.refreshTab,
      modelValue: e.curActiveTab,
      "onUpdate:modelValue": t[0] || (t[0] = (a) => e.curActiveTab = a),
      "tab-position": e.elTabPosition,
      class: `t-tabs__${e.tTabPosition}`,
      onTabClick: e.handleTabClick
    }), {
      default: y(() => [
        (l(!0), o(_, null, $(e.curData, (a) => (l(), T(S, g({ ref_for: !0 }, e.filterAttrs(e.$attrs), {
          key: a.name,
          label: a.label,
          name: a.name,
          disabled: a.disabled
        }), {
          label: y(() => [
            a.iconSrc ? (l(), o("img", {
              key: 0,
              class: "custom-tabs-label__icon",
              src: e.curActiveTab === a.name ? a.activeIconSrc : a.iconSrc,
              draggable: "false"
            }, null, 8, V)) : A("", !0),
            k("span", {
              title: a.label,
              class: "custom-tabs-label__label"
            }, I(a.label), 9, H)
          ]),
          _: 2
        }, 1040, ["label", "name", "disabled"]))), 128))
      ]),
      _: 1
    }, 16, ["modelValue", "tab-position", "class", "onTabClick"])),
    t[1] || (t[1] = k("div", { class: "style-container" }, null, -1))
  ], 6);
}
const O = /* @__PURE__ */ P(C, [["render", B]]);
export {
  O as default
};
