// @ths/design/es/components/chart-pie-doughnut-data/src/index.vue.js
import O from "./index.vue2.js";
import { openBlock as s, createElementBlock as i, normalizeStyle as l, normalizeClass as d, withDirectives as A, createElementVNode as t, toDisplayString as a, createCommentVNode as r, renderSlot as q, Fragment as C, renderList as k, vShow as B } from "vue";
import "./index.vue3.js";
import G from "../../../_virtual/_plugin-vue_export-helper.js";
const J = {
  ref: "chart",
  class: "chart"
}, K = { class: "slot" }, Q = ["innerHTML"], X = ["onClick"], Y = { class: "list-item_iconname" }, Z = ["title", "innerHTML"], x = { class: "list-item_valueunit" }, _ = ["onClick"], ee = ["title", "innerHTML"], le = { class: "empty" }, te = ["src"];
function ne(e, h, se, ie, oe, re) {
  var I, b, w, T, D, U, F, $, L, M, R, E;
  return s(), i("div", {
    style: l({
      ...(I = e.listDataStyle) == null ? void 0 : I.style,
      ...e.cStyle,
      ...(b = e.cStyles) == null ? void 0 : b.root,
      ...(T = (w = e.cStyles) == null ? void 0 : w.wrapper) == null ? void 0 : T.default
    }),
    class: d([e.domOnlyClass, "t-chart-pie-doughnut-data"])
  }, [
    A(t("main", {
      style: l({
        ...(D = e.listDataStyle) == null ? void 0 : D.style
      })
    }, [
      t("div", {
        style: l({ ...e.style, ...(U = e.cStyles) == null ? void 0 : U.chart, ...($ = (F = e.cStyles) == null ? void 0 : F.chart) == null ? void 0 : $.default }),
        class: "chart-box"
      }, [
        t("div", J, null, 512),
        t("div", K, [
          e.curInnerText ? (s(), i("div", {
            key: 0,
            class: "inner-text",
            style: l({ ...(M = (L = e.cStyles) == null ? void 0 : L.innerText) == null ? void 0 : M.default })
          }, a(e.curInnerText), 5)) : r("", !0),
          q(e.$slots, "default"),
          e.showInner ? (s(), i("div", {
            key: 1,
            class: "inner",
            style: l({ "flex-direction": e.innerStyle.align === "bottom" ? "column-reverse" : "column" })
          }, [
            t("div", {
              class: "inner-text text-custom",
              style: l({ "max-width": e.getTextWidth(), color: e.innerStyle.color })
            }, [
              t("span", {
                class: "inner-number",
                style: l({
                  "font-size": typeof e.innerStyle.numberFontsize == "number" ? `${e.innerStyle.numberFontsize}px` : e.innerStyle.numberFontsize,
                  "font-weight": e.innerStyle.numberFontWeight,
                  color: e.innerStyle.numberColor
                })
              }, a(e.currentContent.value), 5),
              e.currentContent.value ? (s(), i("span", {
                key: 0,
                class: "inner-unit",
                style: l({
                  "font-size": `${parseInt(e.innerStyle.unitFontSize)}px`,
                  "font-weight": e.innerStyle.unitFontWeight,
                  color: e.innerStyle.unitColor
                })
              }, a(e.unit), 5)) : r("", !0)
            ], 4),
            t("div", {
              class: "inner-text text-custom",
              style: l({ "max-width": e.getTextWidth(), color: e.innerStyle.textColor })
            }, [
              t("span", {
                class: "inner-legend",
                style: l({
                  "font-size": `${parseInt(e.innerStyle.textFontSize)}px`,
                  "font-weight": e.innerStyle.textFontWeight,
                  color: e.innerStyle.textColor
                }),
                innerHTML: e.currentContent.label
              }, null, 12, Q)
            ], 4)
          ], 4)) : r("", !0)
        ])
      ], 4),
      e.listDataStyle.show === "row" ? (s(), i("div", {
        key: 0,
        style: l({
          ...e.lStyle,
          ...e.cStyles.legendRoot,
          ...(R = e.cStyles.legendRoot) == null ? void 0 : R.default
        }),
        class: "row-list"
      }, [
        (s(!0), i(C, null, k(e.chartDataRow, (o, y) => {
          var u;
          return s(), i("div", {
            key: y,
            style: l({ ...e.cStyles.legendRow, ...(u = e.cStyles.legendRow) == null ? void 0 : u.default }),
            class: "list-wrap"
          }, [
            (s(!0), i(C, null, k(o, (n, v) => {
              var S, g, c, f, p, m;
              return s(), i("div", {
                key: n,
                style: l({
                  ...e.listStyle.cStyle,
                  ...e.cStyles.legendItem,
                  ...(S = e.cStyles.legendItem) == null ? void 0 : S.default,
                  "--highlight-color": ((g = e.options.color || e.defaultColors) == null ? void 0 : g[e.currentIndex]) ?? ""
                }),
                class: "list-item",
                onClick: (z) => e.handleLegendItemClick(
                  n,
                  y * Math.ceil(e.data.length / e.listStyle.col) + v
                )
              }, [
                t("div", Y, [
                  t("div", {
                    style: l({
                      backgroundColor: (c = n.style.legend) == null ? void 0 : c.borderColor,
                      width: e.listStyle.iconSize ? e.listStyle.iconSize + "px" : "14px",
                      height: e.listStyle.iconSize ? e.listStyle.iconSize + "px" : "14px",
                      ...(f = e.cStyles.legendIcon) == null ? void 0 : f.default
                    }),
                    class: "list-item__icon"
                  }, [
                    t("div", {
                      class: "icon-inner",
                      style: l({ backgroundColor: n.style.legend.backgroundColor })
                    }, null, 4)
                  ], 4),
                  t("div", {
                    style: l({
                      color: n.style.legend.nameColor,
                      fontSize: e.listStyle.fontSize ? e.listStyle.fontSize + "px" : "14px",
                      ...e.cStyles.legendName,
                      ...(p = e.cStyles.legendName) == null ? void 0 : p.default
                    }),
                    class: d(["list-item__name", { active: e.currentIndex === n.index }]),
                    title: n.name,
                    innerHTML: n.name
                  }, null, 14, Z)
                ]),
                t("div", x, [
                  e.listDataStyle.value ? (s(), i("div", {
                    key: 0,
                    style: l({
                      color: n.style.legend.valueColor,
                      fontSize: e.listStyle.fontSize ? e.listStyle.fontSize + "px" : "14px",
                      ...e.cStyles.legendValue,
                      ...(m = e.cStyles.legendValue) == null ? void 0 : m.default
                    }),
                    class: d(["list-item__value", { active: e.currentIndex === n.index }])
                  }, a(n.customText ? n.customText : n.value || 0), 7)) : r("", !0),
                  e.listDataStyle.value ? (s(), i("div", {
                    key: 1,
                    style: l({
                      color: e.cStyles.legendUnit.default.color,
                      ...e.cStyles.legendUnit,
                      ...e.cStyles.legendUnit.default,
                      fontSize: e.cStyles.legendUnit.default.fontSize ? e.cStyles.legendUnit.default.fontSize + "px" : "14px",
                      margin: e.marginAddPx(e.cStyles.legendUnit)
                    }),
                    class: d(["list-item__value", { active: e.currentIndex === n.index }])
                  }, a(n.unit || e.unit), 7)) : r("", !0)
                ])
              ], 12, X);
            }), 128))
          ], 4);
        }), 128))
      ], 4)) : r("", !0),
      e.listDataStyle.show === "col" ? (s(), i("div", {
        key: 1,
        style: l({
          ...e.lStyle,
          ...e.cStyles.legendRoot,
          ...(E = e.cStyles.legendRoot) == null ? void 0 : E.default
        }),
        class: "col-list"
      }, [
        (s(!0), i(C, null, k(e.chartData, (o, y) => {
          var u, n, v, S, g, c, f, p, m, z, H, N, V, W, P, j;
          return s(), i("div", {
            key: y,
            style: l({
              ...e.listStyle.cStyle,
              ...e.cStyles.legendItem,
              ...(u = e.cStyles.legendItem) == null ? void 0 : u.default,
              "--highlight-color": ((n = e.options.color || e.defaultColors) == null ? void 0 : n[e.currentIndex]) ?? ""
            }),
            class: "list-item",
            onClick: (ae) => e.handleLegendItemClick(o, y)
          }, [
            t("div", {
              style: l({
                backgroundColor: (S = (v = o.style) == null ? void 0 : v.legend) == null ? void 0 : S.borderColor,
                width: e.listStyle.iconSize ? ((g = e.listStyle) == null ? void 0 : g.iconSize) + "px" : "14px",
                height: e.listStyle.iconSize ? ((c = e.listStyle) == null ? void 0 : c.iconSize) + "px" : "14px",
                ...(p = (f = e.cStyles) == null ? void 0 : f.legendIcon) == null ? void 0 : p.default
              }),
              class: "list-item__icon"
            }, [
              t("div", {
                class: "icon-inner",
                style: l({ backgroundColor: (z = (m = o.style) == null ? void 0 : m.legend) == null ? void 0 : z.backgroundColor })
              }, null, 4)
            ], 4),
            t("div", {
              style: l({
                color: (N = (H = o.style) == null ? void 0 : H.legend) == null ? void 0 : N.nameColor,
                fontSize: (V = e.listStyle) != null && V.fontSize ? ((W = e.listStyle) == null ? void 0 : W.fontSize) + "px" : "14px",
                ...e.cStyles.legendName,
                ...(P = e.cStyles.legendName) == null ? void 0 : P.default
              }),
              class: "list-item__name",
              title: o.name,
              innerHTML: o.name
            }, null, 12, ee),
            e.listDataStyle.value ? (s(), i("div", {
              key: 0,
              style: l({
                color: o.style.legend.valueColor,
                fontSize: e.listStyle.fontSize ? e.listStyle.fontSize + "px" : "14px",
                ...e.cStyles.legendValue,
                ...(j = e.cStyles.legendValue) == null ? void 0 : j.default
              }),
              class: d(["list-item__value", { active: e.currentIndex === o.index }])
            }, a(o.customText ? o.customText : o.value || 0), 7)) : r("", !0),
            e.listDataStyle.value ? (s(), i("div", {
              key: 1,
              style: l({
                color: e.cStyles.legendUnit.default.color,
                fontSize: e.cStyles.legendUnit.default.fontSize ? e.cStyles.legendUnit.default.fontSize + "px" : "14px",
                ...e.cStyles.legendUnit,
                ...e.cStyles.legendUnit.default,
                margin: e.marginAddPx(e.cStyles.legendUnit)
              }),
              class: d(["list-item__value", { active: e.currentIndex === o.index }])
            }, a(o.unit || e.unit), 7)) : r("", !0)
          ], 12, _);
        }), 128))
      ], 4)) : r("", !0)
    ], 4), [
      [B, e.checkEmpty]
    ]),
    A(t("div", le, [
      t("img", {
        class: "empty-img",
        style: l({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, te),
      h[0] || (h[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [B, !e.checkEmpty]
    ]),
    h[1] || (h[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const ge = /* @__PURE__ */ G(O, [["render", ne]]);
export {
  ge as default
};
