// @ths/design/es/components/chart-pie-doughnut-data/src/index.vue2.js
import { defineComponent as oe, computed as f, ref as L, reactive as se, nextTick as U, watch as q, onMounted as ue, onUnmounted as de } from "vue";
import * as ce from "echarts";
import H from "lodash/merge";
import D from "lodash/cloneDeep";
import { splitArray as fe, getDomRandomClass as ye, rgb2rgba as he, replacePollutionHtml as me, setStyle as x } from "../../../utils/index.js";
import { DEFAULT_OPTIONS as B, POLLUTION_HTML as ve } from "../../../utils/constant.js";
const p = {
  name: "name",
  value: "value",
  customText: "customText",
  // 替代图例“数值➕单位”显示的文本内容
  unit: "unit"
  // 单位，优先级高于props.unit
}, C = D(B.color), ge = {
  align: "top",
  color: "#ddd",
  fontSize: "14px",
  numberFontsize: "20px",
  numberFontWeight: "bold"
}, M = {
  root: {},
  wrapper: {
    default: {}
  },
  chart: {
    default: {}
  },
  legendRoot: {
    default: {}
  },
  legendRow: {
    default: {}
  },
  legendItem: {
    default: {}
  },
  legendIcon: {
    default: {}
  },
  legendName: {
    default: {}
  },
  legendValue: {
    default: {}
  },
  legendUnit: {
    default: {}
  },
  innerText: {
    default: {}
  }
}, Ie = oe({
  name: "TChartPieDoughnutData",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 饼图数据
    data: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => p,
      required: !1
    },
    // 整个图形宽度
    width: {
      type: String,
      default: "100%",
      required: !1
    },
    // 整个图形高度
    height: {
      type: String,
      default: "100%",
      required: !1
    },
    // 整个图形命名
    title: {
      type: String,
      default: "",
      required: !1
    },
    // 图像颜色
    colors: {
      type: Array,
      default: () => C,
      required: !1
    },
    // 饼图半径
    radius: {
      type: Array,
      default: () => ["45%", "70%"],
      required: !1,
      validator: (e) => e.length === 2
    },
    // 饼图数据间分隔
    separation: {
      type: Object,
      default: () => ({ width: 2 }),
      required: !1,
      validator: (e) => {
        let s = !0;
        return typeof e.width == "string" && (s = /^\d+(\.\d+)?%$/.test(e.width)), s && e.color ? /#[0-9a-fA-F]{3,6}/.test(e.color) : s;
      }
    },
    // 数据单位
    unit: {
      type: String,
      default: "",
      required: !1
    },
    // 外环阴影宽度
    shadowRadius: {
      type: Array,
      default: () => ["43.5%", "85%"],
      required: !1,
      validator: (e) => e.length === 2
    },
    // 外环阴影颜色
    shadowStyle: {
      type: Object,
      default: () => ({}),
      required: !1,
      validator: (e) => {
        let s = !0;
        return e.opacity !== void 0 && (s = e.opacity >= 0 && e.opacity <= 1), s && e.color ? /#[0-9a-fA-F]{3,6}/.test(e.color) : s;
      }
    },
    // 内环
    showInner: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 中心文字
    innerText: {
      type: String,
      default: "",
      required: !1
    },
    // 显示数据列表
    showList: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // 数据列表排列方式
    listStyle: {
      type: Object,
      default: () => ({
        align: "right",
        value: !0,
        fontSize: 14,
        iconSize: 14,
        col: 2,
        cStyle: {}
      }),
      required: !1,
      validator: (e) => !!e
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({ ...M }),
      required: !1,
      validator: (e) => !!e
    },
    // 列表容器自定义样式
    lStyle: {
      type: Object,
      default: () => ({}),
      required: !1,
      validator: (e) => !!e
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 中间区域循环时间
    showCenterTime: {
      type: Number,
      default: 1e3,
      required: !1
    },
    // 是否开启循环展示中间区域
    isShowCenter: {
      type: Boolean,
      default: !0,
      required: !1
    },
    innerStyle: {
      type: Object,
      default: () => ge,
      required: !1
    }
  },
  emits: ["legend-click"],
  setup(e, { emit: s }) {
    const u = f(() => e.cStyle), h = ye("chart-pie-doughnut-data"), Y = D(B.color);
    let E;
    const m = L(0);
    let S = null;
    const d = se({
      label: "",
      value: ""
    }), v = L(null), g = L([]), G = f(() => e.innerText), w = {
      dataColor: "#666666",
      fontColor: "#666666",
      separationColor: "#EEE",
      separationColorOpacity: 1,
      maskColor: "#FFF",
      maskColorOpacity: 0.8
    }, J = f(() => ({
      ...M,
      ...e.cStyle
    })), K = {
      label: {
        show: !1
      },
      labelLine: {
        show: !1
      },
      color: "rgba(0, 0, 0, 0)",
      borderColor: "rgba(0, 0, 0, 0)",
      borderWidth: 0
    }, I = () => {
      d.label = "", d.value = "";
    }, T = (t = "") => {
      var l;
      const a = ((l = `${t}`.toUpperCase()) == null ? void 0 : l.replace(/[-.+_]/g, "")) || "", r = ve.filter((o) => o.name === a);
      return r.length > 0 ? r[0].html : t;
    }, Q = () => {
      let t = "";
      return u.value.wrapper && (t += x(u.value.wrapper, `.${h}`)), u.value.emptyContainer && (t += x(u.value.emptyContainer, `.${h} .empty`)), u.value.emptyImg && (t += x(u.value.emptyImg, `.${h} .empty .empty-img`)), u.value.emptyText && (t += x(u.value.emptyText, `.${h} .empty .empty-text`)), t;
    }, O = () => {
      const t = document.querySelector(`.${h} .style-container`);
      t && (t.innerHTML = `<style>${Q()}</style>`);
    };
    let i = null;
    const y = () => {
      v.value && (v.value.getBoundingClientRect().width ?? 0) && U(() => {
        i.resize();
      });
    }, R = f(() => {
      var t, a, r, l;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((n) => {
            var c;
            return (n == null ? void 0 : n.data) && ((c = n == null ? void 0 : n.data) == null ? void 0 : c.length) !== 0;
          });
        if (typeof ((a = e.options) == null ? void 0 : a.series) == "object")
          return ((l = (r = e.options.series) == null ? void 0 : r.data) == null ? void 0 : l.length) === 0;
      }
      return e.data.length ? (U(() => {
        y();
      }), !0) : !1;
    }), X = f(() => (t) => {
      var a, r;
      return (a = t == null ? void 0 : t.default) != null && a.margin ? ((r = t == null ? void 0 : t.default) == null ? void 0 : r.margin).map((n) => `${n}px`).join(" ") : null;
    }), Z = () => e.showList ? ["left", "right"].includes(e.listStyle.align) ? {
      show: "row",
      value: !!e.listStyle.value,
      style: {
        flexDirection: e.listStyle.align === "right" ? "row" : "row-reverse",
        alignItems: "center"
      },
      cStyle: e.listStyle.cStyle
    } : {
      show: "col",
      value: !!e.listStyle.value,
      style: {
        flexDirection: e.listStyle.align === "bottom" ? "column" : "column-reverse",
        alignItems: "center"
      },
      cStyle: e.listStyle.cStyle
    } : {
      show: "",
      value: !1,
      style: {},
      cStyle: {}
    }, j = (t, a = C) => [...t].map((l, o) => ({
      ...l,
      index: o,
      name: l.name,
      style: {
        color: a[o % a.length],
        legend: {
          nameColor: w.fontColor,
          valueColor: w.dataColor,
          backgroundColor: a[o % a.length],
          borderColor: he(a[o % a.length], 0.5)
        }
      }
    })), F = (t) => String(t).includes("%") ? Number(String(t).split("%")[0]) : Number(t), $ = (t, a) => {
      if (!t.includes("%"))
        return t;
      const r = F(t);
      if (r < a)
        throw new Error("减数太大");
      return `${r - a}%`;
    }, k = () => {
      const t = [], a = g.value.reduce((r, l) => typeof l == "object" ? r + Number(l.value || 0) : r + Number(l || 0), 0);
      return g.value.forEach((r, l) => {
        t.push(
          {
            ...r,
            value: typeof r == "object" ? r.value || 0 : r || 0,
            name: typeof r == "object" && r.name || "",
            itemStyle: {
              color: (e.colors || e.options.color || C)[l]
            }
          },
          {
            ...r,
            value: e.separation.width === 0 || e.separation.width === "0%" ? 0 : a * (F(e.separation.width) / 100),
            name: "interval",
            itemStyle: K
          }
        );
      }), t;
    }, z = (t) => {
      m.value = 0;
      let a = 0, r = 0;
      const l = t.series[2].data.map((n, c) => c);
      i.dispatchAction({
        type: "highlight",
        seriesIndex: 2,
        dataIndex: 0
      });
      function o() {
        var n, c, b, W, _;
        d.label = ((n = t.series[2].data[a]) == null ? void 0 : n.name) !== "interval" ? T(t.series[2].data[a].name) : T(t.series[2].data[a - 1].name) ?? "", d.value = ((c = t.series[2].data[a]) == null ? void 0 : c.name) !== "interval" ? t.series[2].data[a].value : ((b = t.series[2].data[a - 1]) == null ? void 0 : b.value) ?? "", i.dispatchAction({
          type: "downplay",
          seriesIndex: 2,
          dataIndex: l
        }), r = ((W = t.series[2].data[a]) == null ? void 0 : W.name) !== "interval" ? a : a - 1, m.value = r / 2, i.dispatchAction({
          type: "highlight",
          seriesIndex: 2,
          dataIndex: l[r]
        }), a++, a > ((_ = t.series[2]) == null ? void 0 : _.data.length) - 1 && (a = 0, r = 0);
      }
      S = setInterval(() => {
        o();
      }, e.showCenterTime / 2);
    }, V = () => {
      var r;
      return `${(((r = v.value) == null ? void 0 : r.clientWidth) || 0) * +`0.${parseInt(e.radius[0], 10)}` - 30}px`;
    }, N = (t) => {
      const a = t.series[2].data.map((r, l) => l);
      i.dispatchAction({
        type: "downplay",
        seriesIndex: 2,
        dataIndex: a
      });
    }, ee = f(() => fe(j(g.value, e.colors), e.listStyle.col || 2)), te = f(() => j(g.value, e.colors));
    function A() {
      var t, a, r;
      if (v.value) {
        i || (i = ce.init(v.value)), clearInterval(S);
        const l = H({
          color: e.colors || C,
          title: {
            text: e.title,
            left: "center",
            padding: [0, 0, 40, 0]
          },
          legend: {
            show: !1
          },
          tooltip: {
            show: !1
          },
          xAxis: {
            show: !1
          },
          yAxis: {
            show: !1
          },
          series: []
        }, e.options), o = {
          name: e.title,
          type: "pie",
          radius: e.radius,
          avoidLabelOverlap: !0,
          clockwise: !0,
          label: { show: !1 },
          ...e.options.dataSeries,
          data: k()
        }, n = {
          z: 1,
          name: "外环阴影",
          type: "pie",
          radius: e.shadowRadius,
          avoidLabelOverlap: !0,
          silent: !0,
          clockwise: !0,
          label: { show: !1 },
          itemStyle: {
            color: e.shadowStyle.color || w.separationColor,
            opacity: e.shadowStyle.opacity === void 0 ? w.separationColorOpacity : e.shadowStyle.opacity
          },
          data: [
            { value: 100, name: "外环阴影" }
          ],
          ...e.options.outerSeries
        }, c = {
          z: 4,
          name: "内环阴影",
          type: "pie",
          radius: ["0%", $(e.radius[1], 10)],
          avoidLabelOverlap: !0,
          silent: !0,
          clockwise: !0,
          label: { show: !1 },
          itemStyle: {
            color: w.maskColor,
            opacity: w.maskColorOpacity
          },
          data: [
            { value: 100, name: "内环遮罩" }
          ],
          ...e.options.innerSeries
        }, b = {
          z: 4,
          name: "默认中心",
          type: "pie",
          radius: ["0%", $(e.radius[0], 7)],
          avoidLabelOverlap: !0,
          silent: !0,
          clockwise: !0,
          label: { show: !1 },
          itemStyle: {
            color: "rgba(255,255,255,.1)"
          },
          data: [
            { value: 100, name: "内环遮罩" }
          ],
          ...e.options.centerSeries
        };
        l.series = [
          n,
          // 外部阴影
          c,
          // 内部遮罩
          o
          // 饼图数据series
        ], e.showInner || (l.series = [
          ...l.series,
          b
        ]), E = l, i.setOption(l), e.isShowCenter ? (t = l == null ? void 0 : l.series[2]) != null && t.data && ((r = (a = l == null ? void 0 : l.series[2]) == null ? void 0 : a.data) != null && r.length) ? z(l) : (I(), N(l)) : (m.value = "", I(), N(l));
      }
    }
    function ae(t, a) {
      s("legend-click", {
        current: t,
        currentIndex: a,
        all: g.value
      });
    }
    function P() {
      const t = H(D(p), e.alias);
      g.value = e.data.map((a) => {
        const r = {};
        return Object.keys(t).forEach((l) => {
          l === "name" ? a[t[l]] && (r[l] = me(a[t[l]])) : a[t[l]] && (r[l] = a[t[l]]);
        }), { ...a, ...r };
      });
    }
    const le = (t) => {
      clearInterval(S), k().forEach((a, r) => {
        i.dispatchAction({
          type: "downplay",
          seriesIndex: 2,
          dataIndex: r
        });
      }), i.dispatchAction({
        type: "highlight",
        seriesIndex: 2,
        dataIndex: t.dataIndex
      }), d.label = t.name !== "interval" ? T(t.name) : "", d.value = t.name !== "interval" ? t.value : "", m.value = t.dataIndex / 2;
    }, re = (t) => {
      d.label = "", d.value = "", m.value = "", i.dispatchAction({
        type: "downplay",
        seriesIndex: 2,
        dataIndex: t.dataIndex
      }), clearInterval(S), e.isShowCenter && z(E);
    };
    q(() => [
      e.data,
      e.alias
    ], () => {
      try {
        P(), A(), y();
      } catch (t) {
        console.log(t);
      }
    }, {
      deep: !0
    }), q(() => [
      e.title,
      e.shadowStyle,
      e.showInner,
      e.options,
      e.colors,
      e.radius,
      e.separation,
      e.shadowRadius,
      e.listStyle,
      e.innerStyle,
      e.isShowCenter,
      e.showCenterTime
    ], () => {
      try {
        A(), y();
      } catch (t) {
        console.log(t);
      }
    }, {
      deep: !0
    }), ue(() => {
      O(), I(), P(), A(), i.on("mouseover", le), i.on("mouseout", re), setTimeout(() => {
        y();
      }, 500), window.addEventListener("resize", y);
    }), de(() => {
      window.removeEventListener("resize", y), clearInterval(S);
    });
    const ne = f(() => Z()), ie = f(() => {
      var t;
      return {
        width: e.showList ? e.height : e.width,
        height: e.height,
        background: (t = e.options) == null ? void 0 : t.backgroundColor
      };
    });
    return q(
      () => [e.width, e.height, e.cStyle],
      () => {
        O(), y();
      },
      {
        deep: !0
      }
    ), {
      chart: v,
      chartDataRow: ee,
      chartData: te,
      curInnerText: G,
      listDataStyle: ne,
      style: ie,
      cStyles: J,
      handleLegendItemClick: ae,
      defaultColors: Y,
      currentIndex: m,
      currentContent: d,
      getTextWidth: V,
      marginAddPx: X,
      checkEmpty: R,
      curStyle: u,
      domOnlyClass: h
    };
  }
});
export {
  Ie as default
};
