// @ths/design/es/components/chart-pie-ring/src/index.vue2.js
import { defineComponent as T, computed as g, ref as E, reactive as K, nextTick as v, onMounted as k, onUnmounted as L, watch as C } from "vue";
import * as W from "echarts";
import c from "lodash/merge";
import y from "lodash/cloneDeep";
import { DEFAULT_OPTIONS as H } from "../../../utils/constant.js";
import { getDomRandomClass as q, setStyle as f } from "../../../utils/index.js";
const $ = {
  wrapper: {
    default: {}
  },
  chart: {
    default: {}
  }
}, D = c(
  y(H),
  {
    tooltip: {
      trigger: "item",
      formatter: (e) => {
        const { data: n, color: i } = e;
        return `<div style="display: flex;justify-content: center;align-items: center;">
        <span style="display: block;width: 8px;height: 8px;margin-right: 5px;border-radius: 50%;background: ${i};"></span>
          ${n.name} ${e.percent}%
      </div>`;
      }
    },
    legend: {
      padding: 0,
      orient: "vertical",
      left: "left",
      top: "top",
      icon: "circle"
    },
    grid: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    }
  }
), O = {
  type: "pie",
  radius: ["40%", "60%"],
  avoidLabelOverlap: !0,
  clockwise: !0,
  label: {
    show: !1,
    position: "center"
  },
  itemStyle: {
    borderColor: "#fff",
    borderWidth: 3
  },
  emphasis: {
    label: {
      show: !0,
      formatter: (e) => {
        var n;
        return `{value| ${e.percent}%}

{name| ${(n = e.data) == null ? void 0 : n.name}}`;
      },
      rich: {
        value: {
          color: "#000",
          fontSize: 12,
          fontWeight: "bold"
        },
        name: {
          color: "#000",
          fontSize: 12,
          fontWeight: "normal"
        }
      }
    }
  }
}, U = T({
  name: "TChartPieRing",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 系列样式设置
    seriesStyle: {
      type: Object,
      default: () => ({})
    },
    // 名称别名
    nameKey: {
      type: String,
      default: () => "name"
    },
    // 值别名
    dataKey: {
      type: String,
      default: () => "revenue"
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({ ...$ }),
      required: !1
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 饼图主题配置
    pie: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const n = q("chart-pie-ring"), i = g(() => e.cStyle), d = E(null), j = g(() => ({
      ...$,
      ...e.cStyle
    }));
    let r = null;
    const S = K({
      // 图表数据
      chartData: []
    }), p = () => {
      if (e.data && e.data.length) {
        const { nameKey: t, dataKey: a } = e, { color: l } = e.options;
        S.chartData = e.data.map((o, h) => ({
          name: o[t] || "-",
          value: o[a] || 0,
          itemStyle: {
            color: l && l[h] || D.color[h] || "#fff"
          }
        }));
      }
    }, z = () => {
      let t = "";
      return i.value.wrapper && (t += f(i.value.wrapper, `.${n}`)), i.value.emptyContainer && (t += f(i.value.emptyContainer, `.${n} .empty`)), i.value.emptyImg && (t += f(i.value.emptyImg, `.${n} .empty .empty-img`)), i.value.emptyText && (t += f(i.value.emptyText, `.${n} .empty .empty-text`)), t;
    }, w = () => {
      const t = document.querySelector(`.${n} .style-container`);
      t && (t.innerHTML = `<style>${z()}</style>`);
    }, u = () => {
      d.value && (d.value.getBoundingClientRect().width ?? 0) && v(() => {
        r.resize();
      });
    }, A = g(() => {
      var t, a, l, o;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((s) => {
            var x;
            return (s == null ? void 0 : s.data) && ((x = s == null ? void 0 : s.data) == null ? void 0 : x.length) !== 0;
          });
        if (typeof ((a = e.options) == null ? void 0 : a.series) == "object")
          return ((o = (l = e.options.series) == null ? void 0 : l.data) == null ? void 0 : o.length) === 0;
      }
      return e.data.length ? (v(() => {
        u();
      }), !0) : !1;
    }), m = (t, a) => {
      t.dispatchAction({
        type: a,
        seriesIndex: 0,
        dataIndex: 0
      });
    }, I = (t) => {
      m(t, "highlight"), t.on("mouseover", (a) => {
        a.dataIndex !== 0 && m(t, "downplay");
      }), t.on("mouseout", (a) => {
        m(t, "highlight");
      });
    }, b = () => {
      if (d.value) {
        r || (r = W.init(d.value));
        const { series: t } = e.options;
        let a = {};
        t ? a = t.map((o) => c(
          c(
            y(O),
            e == null ? void 0 : e.pie
          ),
          o
        )) : e.data && e.data.length && (a = c(
          y(O),
          {
            data: S.chartData
          },
          y(e.pie),
          e == null ? void 0 : e.seriesStyle
        ));
        let l = c(
          {
            series: a
          },
          D
        );
        l = c(l, e.options), r.clear();
        try {
          r.setOption(l);
        } catch (o) {
          console.log(o);
        }
        I(r);
      }
    };
    return k(() => {
      w(), p(), b(), window.addEventListener("resize", u);
    }), L(() => {
      window.removeEventListener("resize", u);
    }), C(
      () => [
        e.options,
        e.pie,
        e.data,
        e.nameKey,
        e.dataKey,
        e.seriesStyle
      ],
      () => {
        p(), b();
      },
      {
        deep: !0
      }
    ), C(
      () => [e.cStyle],
      () => {
        w(), v(() => {
          r.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      chart: d,
      cStyles: j,
      curStyle: i,
      checkEmpty: A,
      domOnlyClass: n
    };
  }
});
export {
  U as default
};
