// @ths/design/es/components/chart-pie-ring/src/index.vue.js
import a from "./index.vue2.js";
import { openBlock as m, createElementBlock as c, normalizeStyle as l, normalizeClass as p, withDirectives as i, createElementVNode as e, vShow as n } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const y = {
  ref: "chart",
  class: "chart"
}, u = { class: "empty" }, f = ["src"];
function h(t, s, v, E, g, k) {
  var r, o;
  return m(), c("div", {
    style: l({
      ...t.cStyle,
      ...(o = (r = t.cStyles) == null ? void 0 : r.wrapper) == null ? void 0 : o.default
    }),
    class: p([t.domOnlyClass, "t-chart-pie-ring"])
  }, [
    i(e("div", y, null, 512), [
      [n, t.checkEmpty]
    ]),
    i(e("div", u, [
      e("img", {
        class: "empty-img",
        style: l({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [n, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const _ = /* @__PURE__ */ d(a, [["render", h]]);
export {
  _ as default
};
