// @ths/design/es/components/chart-sankey/src/index.vue.js
import i from "./index.vue2.js";
import { openBlock as m, createElementBlock as c, normalizeClass as p, normalizeStyle as l, withDirectives as a, createElementVNode as e, vShow as n } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const y = {
  ref: "chartId",
  class: "chart"
}, u = { class: "empty" }, f = ["src"];
function h(t, s, k, v, E, S) {
  var r, o;
  return m(), c("div", {
    class: p(["t-chart-sankey", t.domOnlyClass]),
    style: l({ ...(o = (r = t.curStyle) == null ? void 0 : r.wrapper) == null ? void 0 : o.default })
  }, [
    a(e("div", y, null, 512), [
      [n, t.checkEmpty]
    ]),
    a(e("div", u, [
      e("img", {
        class: "empty-img",
        style: l({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [n, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const C = /* @__PURE__ */ d(i, [["render", h]]);
export {
  C as default
};
