// @ths/design/es/components/chart-sankey/src/index.vue2.js
import { defineComponent as p, computed as b, ref as $, reactive as q, onMounted as z, onUnmounted as I, watch as j, nextTick as w, toRefs as K } from "vue";
import * as E from "echarts";
import m from "lodash/merge";
import M from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as _ } from "../../../utils/constant.js";
import { getDomRandomClass as F, setStyle as k } from "../../../utils/index.js";
const x = m(
  M(_),
  {
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    },
    tooltip: {
      trigger: "item"
    }
  }
), G = p({
  name: "TChartSankey",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => x,
      required: !1
    },
    // 传入的节点数组数据
    nodesData: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 传入的连接数组数据
    linksData: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 取resData数据中的别名设置
        data: {
          type: Object,
          default: () => [],
          required: !1
        }
      })
    },
    // 标记线属性配置
    markLineList: {
      type: Array,
      default: () => []
    },
    // 标记线数据
    markLineObj: {
      type: Object,
      default: () => ({})
    },
    // 高亮数据
    highLight: {
      type: String,
      default: ""
    }
  },
  setup(e, { emit: H }) {
    const i = F("chart-sankey"), a = b(() => e.cStyle), S = $(null);
    let s = null;
    const n = q({
      // 图表数据
      chartData: [],
      // 节点数据
      nodesData: [],
      // 连接数据
      linksData: []
    }), O = () => {
      var t, o, l, r, c;
      if ((o = Object.keys((t = e.seriesSetting) == null ? void 0 : t.data)) != null && o.length && ((l = e.nodesData) != null && l.length) && ((r = e.linksData) != null && r.length)) {
        const { nodeKey: u, linkKey: { sourceKey: f, targetKey: g, valueKey: h } } = (c = e.seriesSetting) == null ? void 0 : c.data, D = e.nodesData.map((d) => ({
          name: d[u]
        })), y = e.linksData.map((d) => ({
          source: d[f],
          target: d[g],
          value: d[h]
        }));
        n.nodesData = D, n.linksData = y;
      } else
        n.nodesData = [], n.linksData = [];
    }, A = () => {
      let t = "";
      return a.value.wrapper && (t += k(a.value.wrapper, `.${i}`)), a.value.emptyContainer && (t += k(a.value.emptyContainer, `.${i} .empty`)), a.value.emptyImg && (t += k(a.value.emptyImg, `.${i} .empty .empty-img`)), a.value.emptyText && (t += k(a.value.emptyText, `.${i} .empty .empty-text`)), t;
    }, v = () => {
      const t = document.querySelector(`.${i} .style-container`);
      t && (t.innerHTML = `<style>${A()}</style>`);
    }, T = b(() => !(!e.linksData.length || !e.nodesData.length)), L = () => {
      var c, u, f, g, h, D;
      const { color: t, series: o } = e.options;
      let l = {};
      o ? l = m(
        o,
        (u = (c = e.seriesSetting) == null ? void 0 : c.data) == null ? void 0 : u.style
      ) : e.nodesData.length && e.linksData.length && ((g = Object.keys((f = e.seriesSetting) == null ? void 0 : f.data)) != null && g.length) && (l = m(
        {
          type: "sankey",
          draggable: !1,
          emphasis: {
            focus: "adjacency"
          },
          data: e.nodesData.length ? n.nodesData : [],
          links: e.linksData.length ? n.linksData : []
        },
        (D = (h = e.seriesSetting) == null ? void 0 : h.data) == null ? void 0 : D.style
      ));
      let r = m(
        {
          series: l
        },
        x
      );
      r = m(r, e.options), s.clear();
      try {
        s.setOption(r), e.highLight && s.dispatchAction({
          type: "highlight",
          name: e.highLight
        });
      } catch (y) {
        console.log(y);
      }
    }, C = () => {
      w(() => {
        s.resize();
      });
    };
    return z(() => {
      v(), S.value && (s = E.init(S.value), O(), L(), window.addEventListener("resize", C));
    }), I(() => {
      window.removeEventListener("resize", C);
    }), j(
      [
        () => e.options,
        () => e.nodesData,
        () => e.linksData,
        () => e.seriesSetting,
        () => e.markLineList,
        () => e.markLineObj,
        () => e.highLight
      ],
      () => {
        O(), L();
      },
      {
        deep: !0
      }
    ), j(
      () => e.cStyle,
      () => {
        v(), w(() => {
          s.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      curStyle: a,
      ...K(n),
      chartId: S,
      checkEmpty: T,
      domOnlyClass: i
    };
  }
});
export {
  G as default
};
