// @ths/design/es/components/secondary-menu/src/index.vue2.js
import { defineComponent as _, computed as j, ref as q, reactive as L, provide as k, onMounted as z, watch as I, toRefs as B } from "vue";
import F from "lodash/cloneDeep";
import G from "mitt";
import { getDomRandomClass as J, setStyle as l, getCssPropertyString as n } from "../../../utils/index.js";
import K from "./submenu.vue.js";
const Y = _({
  name: "TSecondaryMenu",
  components: {
    Submenu: K
  },
  props: {
    data: {
      type: Array,
      default: () => []
    },
    // 默认选中项
    activeMenu: {
      type: Array,
      default: () => []
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 一级菜单项
        menuItem: {
          default: {},
          hover: {},
          active: {}
        },
        // 一级菜单项的图标
        menuItemIcon: {
          default: {},
          hover: {},
          active: {}
        },
        // 一级菜单项的文本
        menuItemText: {
          default: {},
          hover: {},
          active: {}
        },
        // 二级菜单
        submenu: {
          default: {},
          hover: {},
          active: {}
        },
        // 二级菜单项
        submenuItem: {
          default: {},
          hover: {},
          active: {}
        },
        // 二级菜单项的图标
        submenuItemIcon: {
          default: {},
          hover: {},
          active: {}
        },
        // 二级菜单项的文本
        submenuItemText: {
          default: {},
          hover: {},
          active: {}
        },
        // 中间层级节点
        middleNode: {
          default: {},
          hover: {},
          active: {}
        },
        // 中间层级节点的图标
        middleNodeIcon: {
          default: {},
          hover: {},
          active: {}
        },
        // 中间层级节点的文本
        middleNodeText: {
          default: {},
          hover: {},
          active: {}
        }
      })
    }
  },
  emits: ["select"],
  setup(a, { emit: P }) {
    var $, x;
    const r = G(), e = j(() => a.cStyle), u = J("secondary-menu"), c = q(a.activeMenu || [(x = ($ = a == null ? void 0 : a.data[0]) == null ? void 0 : $.children[0]) == null ? void 0 : x.id]), f = L({
      // activeMenu: props.activeMenu || [props?.data[0]?.children[0]?.id],
      menuData: a.data
    }), H = () => {
      var m, v, d, o, i, s, T, N, S, M, p, C, g, w, D, A, E, O;
      let t = "";
      return e.value.wrapper && (t += l(e.value.wrapper, `div.t-secondary-menu.${u}`)), e.value.menuItem && (t += l(e.value.menuItem, `div.t-secondary-menu.${u} .menu-item`)), e.value.menuItem && ((m = e.value.menuItem) != null && m.hover) && (t += `div.t-secondary-menu.${u} .menu-item:hover {
           ${n(e.value.menuItem.hover)}
        }`), e.value.menuItem && ((v = e.value.menuItem) != null && v.active) && (t += `div.t-secondary-menu.${u} .menu-item.active {
           ${n(e.value.menuItem.active)}
        }`), e.value.menuItemIcon && (t += l(e.value.menuItemIcon, `div.t-secondary-menu.${u} .menu-item-icon`)), e.value.menuItemIcon && ((d = e.value.menuItemIcon) != null && d.hover) && (t += `div.t-secondary-menu.${u} .menu-item:hover .menu-item-icon {
           ${n(e.value.menuItemIcon.hover)}
        }`), e.value.menuItemIcon && ((o = e.value.menuItemIcon) != null && o.active) && (t += `div.t-secondary-menu.${u} .menu-item.active .menu-item-icon {
           ${n(e.value.menuItemIcon.active)}
        }`), e.value.menuItemText && (t += l(e.value.menuItemText, `div.t-secondary-menu.${u} .menu-item-text`)), e.value.menuItemText && ((i = e.value.menuItemText) != null && i.hover) && (t += `div.t-secondary-menu.${u} .menu-item:hover .menu-item-text {
           ${n(e.value.menuItemText.hover)}
        }`), e.value.menuItemText && ((s = e.value.menuItemText) != null && s.active) && (t += `div.t-secondary-menu.${u} .menu-item.active .menu-item-text {
           ${n(e.value.menuItemText.active)}
        }`), e.value.submenu && (t += l(e.value.submenu, `div.t-secondary-menu.${u} .submenu`)), e.value.submenuItem && (t += l(e.value.submenuItem, `div.t-secondary-menu.${u} .submenu-item`)), e.value.submenuItem && ((T = e.value.submenuItem) != null && T.hover) && (t += `div.t-secondary-menu.${u} .submenu-item.is-last-node:hover {
           ${n(e.value.submenuItem.hover)}
        }`), e.value.submenuItem && ((N = e.value.submenuItem) != null && N.active) && (t += `div.t-secondary-menu.${u} .submenu-item.is-last-node.active {
           ${n(e.value.submenuItem.active)}
        }`), e.value.middleNode && ((S = e.value.middleNode) != null && S.hover) && (t += `div.t-secondary-menu.${u} .submenu-item.middle-node:hover {
           ${n(e.value.middleNode.hover)}
        }`), e.value.middleNode && ((M = e.value.middleNode) != null && M.active) && (t += `div.t-secondary-menu.${u} .submenu-item.middle-node.active {
           ${n(e.value.middleNode.active)}
        }`), e.value.submenuItemIcon && (t += l(e.value.submenuItemIcon, `div.t-secondary-menu.${u} .submenu-item-icon`)), e.value.submenuItemIcon && ((p = e.value.submenuItemIcon) != null && p.hover) && (t += `div.t-secondary-menu.${u} .submenu-item.is-last-node:hover .submenu-item-icon {
           ${n(e.value.submenuItemIcon.hover)}
        }`), e.value.submenuItemIcon && ((C = e.value.submenuItemIcon) != null && C.active) && (t += `div.t-secondary-menu.${u} .submenu-item.is-last-node.active .submenu-item-icon {
           ${n(e.value.submenuItemIcon.active)}
        }`), e.value.middleNodeIcon && ((g = e.value.middleNodeIcon) != null && g.hover) && (t += `div.t-secondary-menu.${u} .submenu-item.middle-node:hover .submenu-item-icon {
           ${n(e.value.middleNodeIcon.hover)}
        }`), e.value.middleNodeIcon && ((w = e.value.middleNodeIcon) != null && w.active) && (t += `div.t-secondary-menu.${u} .submenu-item.middle-node.active .submenu-item-icon {
           ${n(e.value.middleNodeIcon.active)}
        }`), e.value.submenuItemText && (t += l(e.value.submenuItemText, `div.t-secondary-menu.${u} .submenu-item-text`)), e.value.submenuItemText && ((D = e.value.submenuItemText) != null && D.hover) && (t += `div.t-secondary-menu.${u} .submenu-item.is-last-node:hover .submenu-item-text {
           ${n(e.value.submenuItemText.hover)}
        }`), e.value.submenuItemText && ((A = e.value.submenuItemText) != null && A.active) && (t += `div.t-secondary-menu.${u} .submenu-item.is-last-node.active .submenu-item-text {
           ${n(e.value.submenuItemText.active)}
        }`), e.value.middleNodeText && ((E = e.value.middleNodeText) != null && E.hover) && (t += `div.t-secondary-menu.${u} .submenu-item.middle-node:hover .submenu-item-text {
           ${n(e.value.middleNodeText.hover)}
        }`), e.value.middleNodeText && ((O = e.value.middleNodeText) != null && O.active) && (t += `div.t-secondary-menu.${u} .submenu-item.middle-node.active .submenu-item-text {
           ${n(e.value.middleNodeText.active)}
        }`), t;
    }, b = () => {
      const t = document.querySelector(`.${u} .style-container`);
      t && (t.innerHTML = `<style>${H()}</style>`);
    };
    k("rootMenu", {
      props: a,
      selectedMenus: c,
      rootMenuEmit: r.emit,
      rootMenuOn: r.on
    }), z(() => {
      b();
    }), I(() => a.cStyle, () => {
      b();
    }, {
      deep: !0,
      immediate: !0
    }), I(() => a.activeMenu, (t) => {
      c.value = t;
    }), I(() => a.data, (t) => {
      f.menuData = t;
    }, {
      deep: !0,
      immediate: !0
    });
    function R(t) {
      const m = F(t), v = [], d = (o) => {
        o && o.length > 0 && o.forEach((i) => {
          i.children && (d(i.children), delete i.children), v.push(i);
        });
      };
      return d(m), v;
    }
    const h = (t, m, v) => {
      let d = [...v];
      const o = m.find((i) => i.id === t);
      if (o && o.parentId !== "root") {
        d.push(o.parentId);
        const i = m.find((s) => s.id === o.parentId);
        i && (d = [.../* @__PURE__ */ new Set([...d, ...h(i.id, m, d)])]);
      }
      return d;
    }, y = (t, m) => {
      var v;
      (v = t.children) != null && v.length || (c.value = h(m, R(a.data), [m]), P("select", {
        item: t,
        id: m
      }));
    };
    return r.on("submenuItem:select", (t) => y(t, t.id)), {
      ...B(f),
      selectedMenus: c,
      curStyle: e,
      domOnlyClass: u,
      handleSelect: y
    };
  }
});
export {
  Y as default
};
