// @ths/design/es/components/secondary-menu/src/index.vue.js
import h from "./index.vue2.js";
import { resolveComponent as v, openBlock as t, createElementBlock as s, normalizeClass as c, Fragment as r, renderList as u, createElementVNode as l, toDisplayString as p, normalizeStyle as _, createVNode as y } from "vue";
import "./index.vue3.js";
import f from "../../../_virtual/_plugin-vue_export-helper.js";
const M = ["src"], g = ["title"], k = { class: "second-menu" }, C = ["onClick"], $ = ["src"], S = ["title"];
function z(n, o, B, D, E, N) {
  const m = v("submenu");
  return t(), s("div", {
    class: c(["t-secondary-menu", n.domOnlyClass])
  }, [
    (t(!0), s(r, null, u(n.menuData, (i) => (t(), s("div", {
      key: i.id
    }, [
      l("h5", {
        class: c(["menu-item", { active: n.selectedMenus.includes(i.id) }])
      }, [
        l("img", {
          class: "menu-item-icon",
          src: n.selectedMenus.includes(i.id) ? i.activeIcon : i.icon,
          alt: ""
        }, null, 8, M),
        l("span", {
          class: "menu-item-text",
          title: i.text
        }, p(i.text), 9, g)
      ], 2),
      l("ul", k, [
        (t(!0), s(r, null, u(i.children, (e) => {
          var d, a;
          return t(), s("li", {
            key: e.id,
            onClick: (V) => n.handleSelect(e, e.id)
          }, [
            l("div", {
              class: c(["submenu-item", { "is-last-node": !((d = e.children) != null && d.length), "middle-node": (a = e.children) == null ? void 0 : a.length, active: n.selectedMenus.includes(e.id) }])
            }, [
              l("img", {
                class: "submenu-item-icon",
                src: n.selectedMenus.includes(e.id) ? e.activeIcon : e.icon,
                style: _({ opacity: !n.activeMenu.includes(e.id) && e.icon || n.activeMenu.includes(e.id) && e.activeIcon ? 1 : 0 }),
                alt: ""
              }, null, 12, $),
              l("span", {
                class: "submenu-item-text",
                title: e.text
              }, p(e.text), 9, S)
            ], 2),
            y(m, {
              data: e.children,
              index: 1
            }, null, 8, ["data"])
          ], 8, C);
        }), 128))
      ])
    ]))), 128)),
    o[0] || (o[0] = l("div", { class: "style-container" }, null, -1))
  ], 2);
}
const j = /* @__PURE__ */ f(h, [["render", z]]);
export {
  j as default
};
