// @ths/design/es/components/secondary-menu/src/submenu.vue.js
import r from "./submenu.vue2.js";
import { resolveComponent as u, openBlock as c, createElementBlock as l, Fragment as _, renderList as h, createElementVNode as o, normalizeClass as m, normalizeStyle as a, toDisplayString as x, createVNode as v } from "vue";
import f from "../../../_virtual/_plugin-vue_export-helper.js";
const b = { class: "submenu" }, g = ["onClick"], p = ["src"], y = ["title"];
function S(k, C, t, n, M, z) {
  const s = u("Submenu", !0);
  return c(), l("ul", b, [
    (c(!0), l(_, null, h(t.data, (e) => {
      var i, d;
      return c(), l("li", {
        key: e.id,
        onClick: (B) => n.handleSelect(e, e.id)
      }, [
        o("div", {
          class: m(["submenu-item", { "is-last-node": !((i = e.children) != null && i.length), "middle-node": (d = e.children) == null ? void 0 : d.length, active: n.activeMenu.includes(e.id) }])
        }, [
          o("img", {
            class: "submenu-item-icon",
            src: n.activeMenu.includes(e.id) ? e.activeIcon : e.icon,
            alt: "",
            style: a({ opacity: !n.activeMenu.includes(e.id) && e.icon || n.activeMenu.includes(e.id) && e.activeIcon ? 1 : 0 })
          }, null, 12, p),
          o("span", {
            class: "submenu-item-text",
            title: e.text,
            style: a({ "padding-left": `${t.index * 16}px` })
          }, x(e.text), 13, y)
        ], 2),
        v(s, {
          data: e.children,
          index: t.index + 1
        }, null, 8, ["data", "index"])
      ], 8, g);
    }), 128))
  ]);
}
const V = /* @__PURE__ */ f(r, [["render", S]]);
export {
  V as default
};
