// @ths/design/es/components/two-dimension-container/src/index.vue2.js
import { defineComponent as d, computed as l, onMounted as p, watch as s } from "vue";
import { getDomRandomClass as v, setStyle as a } from "../../../utils/index.js";
const $ = d({
  name: "TTwoDimensionContainer",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外容器
        wrapper: {
          default: {}
        },
        // 左容器
        sideLeftWrapper: {
          default: {}
        },
        // 右容器
        sideRightWrapper: {
          default: {}
        },
        // 中间容器
        centerWrapper: {
          default: {}
        },
        // 标题
        title: {
          default: {}
        },
        // 标识线
        line: {
          default: {}
        },
        // 内标识线
        lineInner: {
          default: {}
        },
        // 左值
        leftNum: {
          default: {}
        },
        // 右值
        rightNum: {
          default: {}
        },
        // 单位
        unit: {
          default: {}
        },
        // 百分数
        percent: {
          default: {}
        }
      })
    },
    // 左侧数据
    leftData: {
      type: Object,
      default: () => ({})
    },
    // 右侧数据
    rightData: {
      type: Object,
      default: () => ({})
    },
    // 中心图片背景
    bgUrl: {
      type: String,
      default: ""
    }
  },
  setup(i) {
    const e = l(() => i.cStyle), r = l(() => i.leftData), o = l(() => i.rightData), n = v("two-dimension-container"), f = (t) => {
      let c = t;
      return !t && t !== 0 && (c = "-"), c;
    }, m = () => {
      let t = "";
      return e.value.wrapper && (t += a(e.value.wrapper, `.${n}`)), e.value.sideLeftWrapper && (t += a(e.value.sideLeftWrapper, `.${n} .container-left`)), e.value.sideRightWrapper && (t += a(e.value.sideRightWrapper, `.${n} .container-right`)), e.value.centerWrapper && (t += a(e.value.centerWrapper, `.${n} .container-center`)), e.value.lineInner && (t += a(e.value.lineInner, `.${n} .common .line .line__inner`)), r.value.lineColor && (t += `.${n}.t-two-dimension-container .container-left .line .line__inner {
          background: ${r.value.lineColor}
        }`), o.value.lineColor && (t += `.${n}.t-two-dimension-container .container-right .line .line__inner {
          background: ${o.value.lineColor}
        }`), e.value.title && (t += a(e.value.title, `.${n} .common .title`)), e.value.line && (t += a(e.value.line, `.${n} .common .line`)), e.value.leftNum && (t += a(e.value.leftNum, `.${n}.t-two-dimension-container .container-left .value__box .num`)), e.value.rightNum && (t += a(e.value.rightNum, `.${n}.t-two-dimension-container .container-right .value__box .num`)), e.value.unit && (t += a(e.value.unit, `.${n} .common .value__box .unit`)), e.value.percent && (t += a(e.value.percent, `.${n} .common .value__box .percent`)), t;
    }, u = () => {
      const t = document.querySelector(`.${n} > .style-container`);
      t && (t.innerHTML = `<style>${m()}</style>`);
    };
    return p(() => {
      u();
    }), s(
      () => [i.leftData, i.rightData, i.cStyle],
      () => {
        u();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      curStyle: e,
      domOnlyClass: n,
      isNoVal: f
    };
  }
});
export {
  $ as default
};
