// @ths/design/es/components/chart-liquidfill/src/index.vue2.js
import { defineComponent as $, computed as p, ref as k, reactive as z, onMounted as A, onUnmounted as j, watch as v, nextTick as E, toRefs as I } from "vue";
import * as L from "echarts";
import d from "lodash/merge";
import g from "lodash/cloneDeep";
import "echarts-liquidfill";
import { DEFAULT_OPTIONS_COMMON as M } from "../../../utils/constant.js";
import { getDomRandomClass as U, setStyle as y } from "../../../utils/index.js";
const w = d(g(M), {
  legend: {
    show: !1
  },
  xAxis: {
    show: !1
  },
  yAxis: {
    show: !1
  }
}), R = $({
  name: "TChartLiquidfill",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => ({})
    },
    // 传入数组数据;如果options设置了 series 属性 则会忽略data的数据，用于水波的数量
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        borderStyle: {},
        // 外层border样式
        data: {
          // 水波球配置
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 文字值
    name: {
      type: String,
      default: ""
    },
    // 数字值
    value: {
      type: String,
      default: ""
    },
    // 主题样式字段
    liquidfill: {
      type: Object,
      default: () => ({})
    }
  },
  // 方法名
  emits: ["com-click", "legend-select"],
  setup(e, { emit: C }) {
    const c = U("chart-liquidfill"), l = p(() => e.cStyle), S = k(null);
    let s = null;
    const r = z({
      // 图表数据
      chartData: [],
      textValue: "",
      // 文字值
      liquidfillOptions: g(e.liquidfill)
      // 水波球配置
    }), q = () => {
      var t;
      if (e.seriesSetting && e.data.length) {
        const { color: o } = e.options;
        r.chartData = (t = e.seriesSetting) == null ? void 0 : t.data.map((n, a) => {
          var u, f;
          const { dataKey: m } = n, i = d({
            itemStyle: {
              color: o && o[a] || w.color[a] || "#fff"
            }
          }, (f = (u = e.seriesSetting) == null ? void 0 : u.data[a]) == null ? void 0 : f.style);
          return {
            value: e.data[a] && e.data[a][m] || 0,
            ...i
          };
        }), r.textValue = `${e.name ? e.name : ""}
${e.value !== null && e.value !== void 0 && e.value !== "" ? e.value : "-"}`;
      }
    }, x = () => {
      let t = "";
      return l.value.wrapper && (t += y(l.value.wrapper, `.${c}`)), l.value.emptyContainer && (t += y(l.value.emptyContainer, `.${c} .empty`)), l.value.emptyImg && (t += y(l.value.emptyImg, `.${c} .empty .empty-img`)), l.value.emptyText && (t += y(l.value.emptyText, `.${c} .empty .empty-text`)), t;
    }, D = () => {
      const t = document.querySelector(`.${c} .style-container`);
      t && (t.innerHTML = `<style>${x()}</style>`);
    }, F = p(() => {
      var t, o, n, a;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((i) => {
            var u;
            return (i == null ? void 0 : i.data) && ((u = i == null ? void 0 : i.data) == null ? void 0 : u.length) !== 0;
          });
        if (typeof ((o = e.options) == null ? void 0 : o.series) == "object")
          return ((a = (n = e.options.series) == null ? void 0 : n.data) == null ? void 0 : a.length) === 0;
      }
      return !!e.data.length;
    }), h = () => {
      var m;
      const { color: t, series: o } = e.options;
      let n = [];
      if (o)
        n = o.map((i, u) => {
          var O;
          const f = d(
            {
              type: "liquidFill",
              // 水位图
              radius: "90%",
              // 显示比例
              center: ["50%", "50%"],
              // 中心点
              amplitude: 8,
              // 水波振幅
              data: r.chartData,
              // data个数代表波浪数
              color: t || w.color,
              // 波浪颜色
              itemStyle: {
                shadowBlur: 0
                // 波浪上的阴影
              },
              direction: "right",
              period: (V, T) => T % 2 === 0 ? 1200 : 1500,
              // 周期：交错效果占时通过不同周期实现
              animationDuration: 2e3,
              // 向上填充的时间
              animationDurationUpdate: 2e3,
              backgroundStyle: {
                borderWidth: 1,
                // 外边框
                borderColor: "#00FF54",
                // 边框颜色
                color: "rgba(0,255,84,0.3)"
                // 边框内部填充部分颜色
              },
              label: {
                // 标签设置
                position: ["50%", "55%"],
                formatter: r.textValue,
                // 显示文本,
                fontSize: "26px",
                // 文本字号,
                color: "#fff"
              },
              outline: {
                show: !0,
                // 最外层边框显示控制
                borderDistance: 0,
                itemStyle: {
                  borderWidth: 20,
                  borderColor: "rgba(0,255,84,0.2)"
                }
              }
            },
            r.liquidfillOptions,
            (O = e.seriesSetting) == null ? void 0 : O.borderStyle
          );
          return d({}, f, i);
        });
      else if (e.data.length && e.seriesSetting) {
        const i = d(
          {
            type: "liquidFill",
            // 水位图
            radius: "90%",
            // 显示比例
            silent: !0,
            center: ["50%", "50%"],
            // 中心点
            amplitude: 8,
            // 水波振幅
            data: r.chartData,
            // data个数代表波浪数
            itemStyle: {
              shadowBlur: 0
              // 波浪上的阴影
            },
            direction: "right",
            period: (u, f) => f % 2 === 0 ? 1200 : 1500,
            // 周期：交错效果占时通过不同周期实现
            animationDuration: 2e3,
            // 向上填充的时间
            animationDurationUpdate: 2e3,
            backgroundStyle: {
              borderWidth: 1,
              // 外边框
              borderColor: "#00FF54",
              // 边框颜色
              color: "rgba(0,255,84,0.3)"
              // 边框内部填充部分颜色
            },
            label: {
              // 标签设置
              position: ["50%", "55%"],
              formatter: r.textValue,
              // 显示文本,
              fontSize: "26px",
              // 文本字号,
              color: "#fff"
            },
            outline: {
              show: !0,
              // 最外层边框显示控制
              borderDistance: 0,
              itemStyle: {
                borderWidth: 20,
                borderColor: "rgba(0,255,84,0.2)",
                shadowColor: "transparent"
              }
            }
          },
          g(e.liquidfill)
        );
        n = d({}, i, (m = e.seriesSetting) == null ? void 0 : m.borderStyle);
      }
      let a = d(
        {
          series: n
        },
        w
      );
      a = d(a, e.options), s.clear();
      try {
        s.setOption(a);
      } catch (i) {
        console.log(i);
      }
    }, b = () => {
      setTimeout(() => {
        s.resize();
      }, 500);
    };
    return A(() => {
      D(), S.value && (s = L.init(S.value), q(), h(), s.on("click", (t) => {
        C("com-click", t);
      }), s.on("legendselectchanged", (t) => {
        C("legend-select", t);
      })), window.addEventListener("resize", b);
    }), j(() => {
      window.removeEventListener("resize", b);
    }), v(
      () => e.liquidfill,
      () => {
        r.liquidfillOptions = g(e.liquidfill), h();
      },
      {
        deep: !0
      }
    ), v(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting,
        () => e.name,
        () => e.value
      ],
      () => {
        q(), h(), E(() => {
          s.resize();
        });
      },
      {
        deep: !0
      }
    ), v(
      () => e.cStyle,
      () => {
        D(), b();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      curStyle: l,
      ...I(r),
      chartId: S,
      checkEmpty: F,
      domOnlyClass: c
    };
  }
});
export {
  R as default
};
