// @ths/design/es/components/regulatory-target-selector/src/services/point-select-api.js
import g from "../utils/http.js";
async function d(s) {
  const r = (await g.get(s)).data, n = r.mdmEntitys.entityIds, c = r.mdmEntitys.entityNames, e = new Array(n.length);
  for (let a = 0; a < n.length; a++) {
    const t = {
      entityId: n[a],
      entityName: c[a],
      config: r.mdmSettings[n[a]],
      keyWord: "",
      regionCode: "",
      regionName: "",
      tagSearchs: [],
      tagJson: {},
      tableHeaderOptions: [],
      page: {
        pageNum: 1,
        pageSize: 10,
        list: [],
        total: 0
      }
    }, l = [];
    if (t.config.tagSearch && t.config.tagSearch.showCodes)
      for (let o = 0; o < t.config.tagSearch.showCodes.length; o++)
        l.push({
          prop: t.config.tagSearch.showCodes[o],
          label: t.config.tagSearch.showNames[o],
          align: t.config.tagSearch.showAligns[o] || "left",
          headerAlign: t.config.tagSearch.showAligns[o] || "left",
          width: t.config.tagSearch.showWidths[o] || void 0,
          minWidth: t.config.tagSearch.showWidths[o] || 150
        });
    t.tableHeaderOptions = l, e[a] = t;
  }
  return e;
}
async function f(s, i) {
  const n = (await g.get(s + i)).data.tags;
  function c(e) {
    if (e.children && e.children.length > 0)
      for (const a of e.children)
        c(a);
    else
      e.nodeType === "tag" && (e.tagValue = null, e.children = []);
  }
  for (let e = 0; e < n.length; e++)
    c(n[e]);
  return n;
}
async function p(s, i) {
  return (await g.post(s, {
    ENTITY_CODE: "",
    queryField: "",
    HasAllTag: !1,
    pageNum: 1,
    pageSize: 10,
    hasGather: !1,
    tagJson: {},
    keyword: "",
    REGION_CODE: "",
    orderBy: "",
    gisType: "",
    pointValue: "",
    distance: "",
    useGisShape: !1,
    locationField: "location",
    ...i
  })).data;
}
async function y(s) {
  return (await g.get(s)).data.data;
}
export {
  p as getDataListPage,
  f as getTagSearchs,
  y as getUserRegion,
  d as initPointSelectOptions
};
