// @ths/design/es/components/chart-line-area/src/index.vue2.js
import { defineComponent as N, computed as q, ref as U, reactive as H, watch as C, nextTick as D, onMounted as K, onUnmounted as R } from "vue";
import * as X from "echarts";
import i from "lodash/merge";
import F from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as B, DEFAULT_OPTIONS_AXIS as G } from "../../../utils/constant.js";
import { getDomRandomClass as J, setStyle as O } from "../../../utils/index.js";
const Q = i(F(B), F(G)), u = i(
  F(Q),
  {
    tooltip: {
      textStyle: {
        color: "#666"
      },
      axisPointer: {
        type: "line"
      }
    },
    grid: {
      top: "13%",
      left: "2%",
      right: "3%",
      bottom: "2%",
      containLabel: !0
    },
    yAxis: {
      name: "单位",
      splitLine: {
        show: !0
      }
    }
  }
), te = N({
  name: "TChartLineArea",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => u,
      required: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        }
      })
    },
    // 折线主题配置
    line: {
      type: Object,
      default: () => ({})
    }
  },
  // 方法名定义
  emits: ["com-click", "com-hover"],
  setup(e, { emit: L }) {
    const f = J("chart-line-area"), r = q(() => e.cStyle), v = U(null);
    let l = null;
    const n = H({
      // 图表数据
      chartData: [],
      lineOptions: F(e.line)
    }), w = () => {
      var t, s;
      (s = (t = e.seriesSetting) == null ? void 0 : t.data) != null && s.length && e.data && (n.chartData = e.seriesSetting.data.map((c) => {
        const { name: m, dataKey: d } = c;
        return {
          name: m || "--",
          data: e.data.map((o) => o[d])
        };
      }));
    }, M = () => {
      let t = "";
      return r.value.wrapper && (t += O(r.value.wrapper, `.${f}`)), r.value.emptyContainer && (t += O(r.value.emptyContainer, `.${f} .empty`)), r.value.emptyImg && (t += O(r.value.emptyImg, `.${f} .empty .empty-img`)), r.value.emptyText && (t += O(r.value.emptyText, `.${f} .empty .empty-text`)), t;
    }, T = () => {
      const t = document.querySelector(`.${f} .style-container`);
      t && (t.innerHTML = `<style>${M()}</style>`);
    }, P = q(() => {
      var t, s, c, m;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((o) => {
            var g;
            return (o == null ? void 0 : o.data) && ((g = o == null ? void 0 : o.data) == null ? void 0 : g.length) !== 0;
          });
        if (typeof ((s = e.options) == null ? void 0 : s.series) == "object")
          return ((m = (c = e.options.series) == null ? void 0 : c.data) == null ? void 0 : m.length) === 0;
      }
      return !!e.data.length;
    }), b = () => {
      var o, g, z, E, j, I, _;
      l.clear();
      const t = (o = e.options) == null ? void 0 : o.color, s = (g = e.options) == null ? void 0 : g.series;
      let c = [], m = [];
      s ? c = s.map((y, a) => {
        var S, h, p;
        const A = i(
          i(
            {
              type: "line",
              lineStyle: {
                color: t && t[a] || u.color[a] || "#FFF"
              },
              emphasis: {
                focus: "series"
              },
              areaStyle: {
                color: {
                  type: "linear",
                  x: 1,
                  y: 1.2,
                  r: 0,
                  colorStops: [
                    {
                      offset: 0,
                      color: "rgba(255,255,255,0)"
                      // 0% 处的颜色
                    },
                    {
                      offset: 1,
                      color: t && t[a] || u.color[a] || "rgba(255,255,255,0)"
                      // 100% 处的颜色
                    }
                  ]
                }
              }
            },
            n == null ? void 0 : n.lineOptions
          ),
          (p = (h = (S = e.seriesSetting) == null ? void 0 : S.data) == null ? void 0 : h[a]) == null ? void 0 : p.style
        );
        return i({}, A, y);
      }) : e.data && ((E = (z = e.seriesSetting) == null ? void 0 : z.data) != null && E.length) && (c = (j = e.seriesSetting) == null ? void 0 : j.data.map((y, a) => {
        var S, h, p, $, k;
        const A = i(
          {
            name: (S = n.chartData[a]) == null ? void 0 : S.name,
            type: "line",
            lineStyle: {
              color: t && t[a] || u.color[a] || "#FFF"
            },
            itemStyle: {
              color: t && t[a] || u.color[a] || "#FFF"
            },
            symbolSize: 4,
            emphasis: {
              focus: "series"
            },
            smooth: !0,
            areaStyle: {
              color: {
                type: "linear",
                x: 1,
                y: 1.2,
                r: 0,
                colorStops: [
                  {
                    offset: 0,
                    color: "rgba(255,255,255,0)"
                    // 0% 处的颜色
                  },
                  {
                    offset: 1,
                    color: t && t[a] || u.color[a] || "#FFFFFF00"
                    // 100% 处的颜色
                  }
                ]
              }
            },
            data: e.data.length ? (h = n.chartData[a]) == null ? void 0 : h.data : []
          },
          n == null ? void 0 : n.lineOptions,
          (k = ($ = (p = e.seriesSetting) == null ? void 0 : p.data) == null ? void 0 : $[a]) == null ? void 0 : k.style
        );
        return i({}, A, y);
      }), (I = e.seriesSetting) != null && I.xAxisLabel ? m = (_ = e.data) != null && _.length ? e.data.map((y) => y[e.seriesSetting.xAxisLabel]) : [] : m = e.data.map((y, a) => a));
      let d = i(
        {
          series: c,
          xAxis: {
            data: m
          }
        },
        u
      );
      d = i(d, e.options);
      try {
        D(() => {
          l.setOption(d), l.resize();
        });
      } catch (y) {
        console.log(y);
      }
    }, x = () => {
      D(() => {
        l.resize();
      });
    };
    return C(
      () => [
        e.line,
        e.options
      ],
      () => {
        n.lineOptions = F(e.line), b();
      },
      {
        deep: !0
      }
    ), C(
      () => [
        e.data,
        e.seriesSetting
      ],
      () => {
        w(), b();
      },
      {
        deep: !0
      }
    ), C(
      () => [e.cStyle],
      () => {
        T(), D(() => {
          l.resize();
        });
      },
      {
        deep: !0
      }
    ), K(() => {
      T(), v.value && (l = X.init(v.value), w(), b(), l.on("click", (t) => {
        L("com-click", t);
      }), l.on("mouseover", (t) => {
        L("com-hover", t);
      })), window.addEventListener("resize", x);
    }), R(() => {
      window.removeEventListener("resize", x);
    }), {
      chart: v,
      curStyle: r,
      checkEmpty: P,
      domOnlyClass: f
    };
  }
});
export {
  te as default
};
