// @ths/design/es/components/chart-line-area/src/index.vue.js
import n from "./index.vue2.js";
import { openBlock as m, createElementBlock as c, normalizeClass as p, normalizeStyle as l, withDirectives as a, createElementVNode as t, vShow as i } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  ref: "chart",
  class: "chart-box"
}, y = { class: "empty" }, f = ["src"];
function h(e, s, v, E, k, _) {
  var r, o;
  return m(), c("div", {
    class: p(["t-chart-line-area", e.domOnlyClass]),
    style: l({ ...(o = (r = e.curStyle) == null ? void 0 : r.wrapper) == null ? void 0 : o.default })
  }, [
    a(t("div", u, null, 512), [
      [i, e.checkEmpty]
    ]),
    a(t("div", y, [
      t("img", {
        class: "empty-img",
        style: l({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [i, !e.checkEmpty]
    ]),
    s[1] || (s[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const C = /* @__PURE__ */ d(n, [["render", h]]);
export {
  C as default
};
