// @ths/design/es/components/date-picker/src/index.vue.js
import d from "./index.vue2.js";
import { resolveComponent as o, openBlock as s, createElementBlock as p, normalizeClass as m, createVNode as l, withCtx as i, mergeProps as u, createElementVNode as f } from "vue";
import "./index.vue3.js";
import c from "../../../_virtual/_plugin-vue_export-helper.js";
function C(e, a, g, k, F, h) {
  const n = o("el-date-picker"), r = o("el-config-provider");
  return s(), p("div", {
    class: m(["t-date-picker", e.domOnlyClass])
  }, [
    l(r, { locale: e.locale }, {
      default: i(() => [
        l(n, u({
          modelValue: e.dateValue,
          "onUpdate:modelValue": a[0] || (a[0] = (t) => e.dateValue = t),
          type: e.dateType,
          format: e.dateFormat,
          "value-format": e.type !== "week" ? e.dateFormat : null,
          placeholder: "请选择日期",
          "popper-class": `${e.domOnlyClass} t-date-picker`,
          "disabled-date": e.disabledDateFunc
        }, e.filterAttrs(e.$attrs), {
          onCalendarChange: e.handleCalendarChange,
          onChange: e.changeTime,
          onFocus: e.handleFocus
        }), null, 16, ["modelValue", "type", "format", "value-format", "popper-class", "disabled-date", "onCalendarChange", "onChange", "onFocus"])
      ]),
      _: 1
    }, 8, ["locale"]),
    a[1] || (a[1] = f("div", { class: "style-container" }, null, -1))
  ], 2);
}
const E = /* @__PURE__ */ c(d, [["render", C]]);
export {
  E as default
};
