// @ths/design/es/components/date-picker/src/index.vue2.js
import { defineComponent as rr, computed as ze, reactive as ar, watch as ie, onMounted as pr, toRefs as tr } from "vue";
import { ElConfigProvider as lr } from "element-plus";
import ir from "element-plus/lib/locale/lang/zh-cn";
import { filterAttrs as nr, getDomRandomClass as or, setStyle as t, getCssPropertyString as l } from "../../../utils/index.js";
const kr = rr({
  name: "TDatePicker",
  components: {
    ElConfigProvider: lr
  },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({})
    },
    // 日期类型
    type: {
      type: String,
      default: "date"
    },
    // 日期框内默认选中的日期
    defaultDate: {
      type: String,
      default: ""
    },
    // 日期年月日间的分隔符号
    separator: {
      type: String,
      default: "-"
    },
    // 禁用时间
    disabledDate: {
      type: [String, Array],
      default: ""
    },
    // 是否禁用之前的时间
    isDisabledFront: {
      type: Boolean,
      default: !0
    }
  },
  emits: ["date-change", "date-focus", "calendar-change"],
  setup(p, { emit: ne }) {
    const r = ze(() => p.cStyle), e = or("date-picker"), Ee = [
      "year",
      "month",
      "date",
      "dates",
      "datetime",
      "datetimehour",
      "datetimeminute",
      "week",
      "datetimerange",
      "datetimerangehour",
      "datetimerangeminute",
      "daterange",
      "monthrange"
    ], _ = ar({
      // 日期类型
      dateType: "",
      // 日期框绑定的值
      dateValue: p.defaultDate,
      disabledDateFunc: null
    }), Ge = (a) => {
      if (!p.disabledDate)
        return !1;
      if (Array.isArray(p.disabledDate)) {
        const i = p.disabledDate.map((n) => {
          const [o, d, c] = n.split("-").map(Number);
          return new Date(o, d - 1, c, 0, 0, 0).getTime();
        });
        return a.setHours(0, 0, 0, 0), !i.includes(a.getTime());
      }
      if (typeof p.disabledDate == "string") {
        const i = new Date(p.disabledDate).getTime(), n = new Date(Number(new Date(p.disabledDate)) - 24 * 60 * 60 * 1e3).getTime();
        return p.isDisabledFront ? n > a.getTime() : i < a.getTime();
      }
      return !1;
    }, Je = (a) => {
      const i = new Date(a), o = (new Date(i.getFullYear(), i.getMonth(), 1).getDay() + 6) % 7, d = Math.ceil((i.getDate() + o) / 7);
      return `${i.getMonth() + 1}月-第${d}周`;
    }, Ke = ze(() => {
      const a = [
        "YYYY",
        `YYYY${p.separator}MM`,
        `YYYY${p.separator}MM${p.separator}DD`,
        `YYYY${p.separator}MM${p.separator}DD`,
        `YYYY${p.separator}MM${p.separator}DD HH:mm:ss`,
        `YYYY${p.separator}MM${p.separator}DD HH`,
        `YYYY${p.separator}MM${p.separator}DD HH:mm`,
        `YYYY${p.separator}${Je(_.dateValue)}`,
        `YYYY${p.separator}MM${p.separator}DD HH:mm:ss`,
        `YYYY${p.separator}MM${p.separator}DD HH`,
        `YYYY${p.separator}MM${p.separator}DD HH:mm`,
        `YYYY${p.separator}MM${p.separator}DD`,
        `YYYY${p.separator}MM`
      ], i = Ee.findIndex((n) => n === p.type) ?? 2;
      return a[i];
    }), Qe = (a) => {
      ne("date-change", a || "");
    }, oe = (a) => {
      const i = a.getFullYear(), n = String(a.getMonth() + 1).padStart(2, "0"), o = String(a.getDate()).padStart(2, "0");
      return `${i}-${n}-${o}`;
    }, Ue = (a) => {
      const i = [oe(new Date(a[0])), a[1] !== null ? oe(new Date(a[1])) : ""];
      ne("calendar-change", i);
    }, Xe = () => {
      ne("date-focus");
    }, Ze = () => {
      var i, n, o, d, c, u, k, $, b, s, v, h, f, m, g, y, W, D, Y, w, M, H, C, B, S, T, x, P, F, A, N, I, L, O, V, R, j, q, z, E, G, J, K, Q, U, X, Z, ee, re, ae, pe, te, le;
      let a = "";
      return (i = r.value) != null && i.wrapper && (a += t((n = r.value) == null ? void 0 : n.wrapper, `.${e}`)), (o = r.value) != null && o.dateWrapper && (a += t(
        (d = r.value) == null ? void 0 : d.dateWrapper,
        `.${e}.el-picker__popper.el-popper[role=tooltip],
        .${e}.el-picker__popper .el-month-table td.disabled .cell,
        .${e}.el-picker__popper .el-year-table td.disabled .cell,
        .${e}.el-picker__popper .el-picker-panel__footer,
        .${e}.el-picker__popper .el-button`
      ), (k = (u = (c = r.value) == null ? void 0 : c.dateWrapper) == null ? void 0 : u.default) != null && k.width && (a += `.${e}.el-picker__popper .el-picker-panel,
        .${e}.el-picker__popper .el-picker-panel .el-picker-panel__body-wrapper,
        .${e}.el-picker__popper .el-picker-panel .el-picker-panel__body,
        .${e}.el-picker__popper .el-picker-panel .el-picker-panel__content{
          ${l({ width: "100%" })}
        }`, a += `
        .${e}.el-picker__popper .el-picker-panel .el-picker-panel__content.el-date-range-picker__content{
          ${l({ width: "50%" })}
        }`, a += `
        .${e}.el-picker__popper .el-picker-panel .el-picker-panel__content{
          ${l({ margin: "0px", padding: "0px 15px" })}
        }`)), ($ = r.value) != null && $.weekWrapper && (a += t((b = r.value) == null ? void 0 : b.weekWrapper, `.t-date-picker.${e} .el-picker-panel__content tbody>tr th`)), (s = r.value) != null && s.splitLine && (a += t(
        (v = r.value) == null ? void 0 : v.splitLine,
        `.${e}.el-picker__popper .el-date-table tr:first-child th,
        .${e}.el-picker__popper .el-date-picker__header--bordered,
        .${e}.el-picker__popper .el-picker-panel__footer,
        .${e}.el-picker__popper .el-date-picker__time-header,
        .${e}.el-picker__popper .el-date-range-picker__time-header,
        .${e}.el-picker__popper .el-time-panel__content::after,
        .${e}.el-picker__popper .el-time-panel__content::before,
        .${e}.el-picker__popper .el-time-panel__footer`
      )), (h = r.value) != null && h.inputWrapper && (a += t(
        (f = r.value) == null ? void 0 : f.inputWrapper,
        `.${e} .el-input .el-input__inner,
          .${e} .el-range-editor.el-input__inner`
      ), a += t(
        { default: { width: "inherit", height: "auto" } },
        `.${e} .el-picker-panel__body-wrapper .el-picker-panel__body .el-input__inner`
      )), (m = r.value) != null && m.rangeInputWrapper && (a += t(
        (g = r.value) == null ? void 0 : g.rangeInputWrapper,
        `.${e} .el-date-editor input.el-range-input`
      )), (y = r.value) != null && y.operatorArrow && (a += t(
        (W = r.value) == null ? void 0 : W.operatorArrow,
        `.${e} .t-date-picker.el-picker__popper .el-date-picker__header-label, .t-date-picker.el-picker__popper .el-picker-panel__icon-btn`
      )), (D = r.value) != null && D.inputDateIcons && (a += t(
        (Y = r.value) == null ? void 0 : Y.inputDateIcons,
        `.${e} .el-input__prefix,
        .${e} .el-date-editor .el-range__icon`
      )), (w = r.value) != null && w.inputCloseIcons && (a += t(
        (M = r.value) == null ? void 0 : M.inputCloseIcons,
        `.${e} .el-input__suffix,
        .${e} .el-date-editor .el-range__close-icon`
      )), (H = r.value) != null && H.dateHeaderWrapper && (a += t(
        (C = r.value) == null ? void 0 : C.dateHeaderWrapper,
        `.${e}.el-picker__popper .el-date-picker__header-label,
        .${e}.el-picker__popper .el-date-range-picker__header`
      )), (B = r.value) != null && B.currentDateWrapper && (a += t(
        (S = r.value) == null ? void 0 : S.currentDateWrapper,
        `.${e}.el-picker__popper .el-date-table td.available,
        .${e}.el-picker__popper .el-year-table td:not(.disabled) .cell,
        .${e}.el-picker__popper .el-month-table td:not(.disabled) .cell,
        .${e}.el-picker__popper .el-date-table td.start-date span,
        .${e}.el-picker__popper .el-date-table td.end-date span`
      )), (T = r.value) != null && T.todayWrapper && (a += t(
        (x = r.value) == null ? void 0 : x.todayWrapper,
        `.${e}.el-picker__popper .el-picker-panel__content .available.today span,
        .${e}.el-picker__popper .el-month-table td.today .cell,
        .${e}.el-picker__popper .el-year-table td.today .cell`
      )), (P = r.value) != null && P.disableDateWrapper && (a += t(
        (F = r.value) == null ? void 0 : F.disableDateWrapper,
        `.${e}.el-picker__popper .el-date-table td.disabled div,
        .${e}.el-picker__popper .el-month-table td.disabled .cell,
        .${e}.el-picker__popper .el-year-table td.disabled .cell`
      )), (A = r.value) != null && A.okWrapper && (a += t((N = r.value) == null ? void 0 : N.okWrapper, `.${e}.el-picker__popper .el-button.el-button--default`)), (I = r.value) != null && I.timePanelWrapper && (a += t((L = r.value) == null ? void 0 : L.timePanelWrapper, `.${e}.el-picker__popper .el-picker-panel .el-time-panel`)), (O = r.value) != null && O.currentWeekWrapper && (a += t(
        (V = r.value) == null ? void 0 : V.currentWeekWrapper,
        `.${e}.el-picker__popper .el-date-table.is-week-mode .el-date-table__row.current div`
      )), (R = r.value) != null && R.rangeTimeWrapper && (a += t(
        (j = r.value) == null ? void 0 : j.rangeTimeWrapper,
        `.${e}.el-picker__popper .el-date-table td.in-range div,
          .${e}.el-picker__popper .el-month-table td.in-range div`
      )), (q = r.value) != null && q.textBtnWrapper && (a += t(
        (z = r.value) == null ? void 0 : z.textBtnWrapper,
        `.${e}.el-picker__popper .el-button--text,
          .${e}.el-picker__popper .el-time-panel__btn.confirm,
          .${e}.el-picker__popper .el-time-panel__btn`
      )), (E = r.value) != null && E.timePanelNumWrapper && (a += t((G = r.value) == null ? void 0 : G.timePanelNumWrapper, `.t-date-picker.${e} .el-picker-panel .el-time-spinner__item`)), (J = r.value) != null && J.rangeSplitLine && (a += t((K = r.value) == null ? void 0 : K.rangeSplitLine, `.${e}.el-picker__popper .el-date-range-picker__content.is-left`)), (Q = r.value) != null && Q.dateHeaderYearBtnWrapper && (a += t(
        (U = r.value) == null ? void 0 : U.dateHeaderYearBtnWrapper,
        `.${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-right,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-left,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-right,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-left
          `
      )), (X = r.value) != null && X.dateHeaderYearBtn && (a += t(
        (Z = r.value) == null ? void 0 : Z.dateHeaderYearBtn,
        `.${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-right::before,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-left::before,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-right::before,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-left::before
          `
      )), (ee = r.value) != null && ee.dateHeaderMonthBtnWrapper && (a += t(
        (re = r.value) == null ? void 0 : re.dateHeaderMonthBtnWrapper,
        `.${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-arrow-right,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-arrow-left,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-arrow-right,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-arrow-left
          `
      )), (ae = r.value) != null && ae.dateHeaderMonthBtn && (a += t(
        (pe = r.value) == null ? void 0 : pe.dateHeaderMonthBtn,
        `.${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-arrow-right::before,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-arrow-left::before,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-arrow-right::before,
           .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-arrow-left::before
          `
      )), (te = r.value) != null && te.prevAndNextMonthDate && (a += t(
        (le = r.value) == null ? void 0 : le.prevAndNextMonthDate,
        `
          .${e}.el-picker__popper .el-picker-panel .el-picker-panel__body-wrapper .el-picker-panel__body .el-picker-panel__content table.el-date-table td.prev-month,
          .${e}.el-picker__popper .el-picker-panel .el-picker-panel__body-wrapper .el-picker-panel__body .el-picker-panel__content table.el-date-table td.next-month
          `
      )), a;
    }, er = () => {
      var i, n, o, d, c, u, k, $, b, s, v, h, f, m, g, y, W, D, Y, w, M, H, C, B, S, T, x, P, F, A, N, I, L, O, V, R, j, q, z, E, G, J, K, Q, U, X, Z, ee, re, ae, pe, te, le, de, ce, ue, ke, $e, be, se, ve, he, fe, me, ge, ye, We, De, Ye, we, Me, He, Ce, Be, Se, Te, xe, Pe, Fe, Ae, Ne, Ie, Le, Oe, Ve, Re;
      let a = "";
      if ((n = (i = r.value) == null ? void 0 : i.wrapper) != null && n.default) {
        const { width: je, height: qe } = (d = (o = r.value) == null ? void 0 : o.wrapper) == null ? void 0 : d.default;
        a += `
          .${e}>.el-input .el-input__inner {
             ${l({ width: je, height: qe })}
           }
          .${e} .el-input .el-input__icon,
          .${e} .el-input {
            ${l({ "line-height": qe })}
          }
          .${e} .el-date-editor.el-input,
          .${e} .el-date-editor.el-input__inner {
            ${l({ width: je })}
          }
        `;
      }
      return (u = (c = r.value) == null ? void 0 : c.inputWrapper) != null && u.active && (a += `
          .${e} .el-input .el-input__inner:focus,
          .${e} .el-range-editor.el-input__inner.is-active{
          ${l(($ = (k = r.value) == null ? void 0 : k.inputWrapper) == null ? void 0 : $.active)}
         }`), (s = (b = r.value) == null ? void 0 : b.currentDateWrapper) != null && s.active && (a += `
          .${e}.el-picker__popper .el-year-table td.current:not(.disabled) .cell,
           .${e}.el-picker__popper .el-month-table td.current:not(.disabled) .cell {
             ${l((h = (v = r.value) == null ? void 0 : v.currentDateWrapper) == null ? void 0 : h.active)}
           }
        `), (m = (f = r.value) == null ? void 0 : f.timePanelNumWrapper) != null && m.hover && (a += `.t-date-picker.${e} .el-picker-panel .el-time-spinner__item:hover:not(.disabled):not(.active) {
          ${l((y = (g = r.value) == null ? void 0 : g.timePanelNumWrapper) == null ? void 0 : y.hover)}
        }`), (W = r.value) != null && W.selCurrentDateWrapper && ((Y = (D = r.value) == null ? void 0 : D.selCurrentDateWrapper) != null && Y.default) && (a += `
          .${e}.el-picker__popper .el-date-table td.current:not(.disabled) span,
          .${e}.el-picker__popper .el-date-table td.selected span,
          .${e}.el-picker__popper .el-date-table td.start-date span,
          .${e}.el-picker__popper .el-date-table td.end-date span,
          .${e}.el-picker__popper .el-month-table td.end-date .cell,
          .${e}.el-picker__popper .el-month-table td.start-date .cell {
            ${l((M = (w = r.value) == null ? void 0 : w.selCurrentDateWrapper) == null ? void 0 : M.default)}
          }
        `), (H = r.value) != null && H.selCurrentDateWrapper && ((B = (C = r.value) == null ? void 0 : C.selCurrentDateWrapper) != null && B.hover) && (a += `
          .${e}.el-picker__popper .el-date-table td.current:not(.disabled) span:hover,
          .${e}.el-picker__popper .el-date-table td.selected span:hover,
          .${e}.el-picker__popper .el-date-table td.start-date span:hover,
          .${e}.el-picker__popper .el-date-table td.end-date span:hover,
          .${e}.el-picker__popper .el-month-table td.end-date .cell:hover,
          .${e}.el-picker__popper .el-month-table td.start-date .cell:hover {
            ${l((T = (S = r.value) == null ? void 0 : S.selCurrentDateWrapper) == null ? void 0 : T.hover)}
          }
        `), (x = r.value) != null && x.timePanelWrapper && ((A = (F = (P = r.value) == null ? void 0 : P.timePanelWrapper) == null ? void 0 : F.default) != null && A.borderColor && (a += `
            .${e}.el-picker__popper .el-time-panel__footer {
              ${l({ borderColor: (L = (I = (N = r.value) == null ? void 0 : N.timePanelWrapper) == null ? void 0 : I.default) == null ? void 0 : L.borderColor })}
            }
          `), (R = (V = (O = r.value) == null ? void 0 : O.timePanelWrapper) == null ? void 0 : V.active) != null && R.color && (a += `
            .${e}.el-picker__popper .el-time-spinner__item.active {
              ${l({ color: (z = (q = (j = r.value) == null ? void 0 : j.timePanelWrapper) == null ? void 0 : q.active) == null ? void 0 : z.color })}
            }
          `), (G = (E = r.value) == null ? void 0 : E.currentWeekWrapper) != null && G.hover && (a += `
            .${e}.el-picker__popper .el-date-table.is-week-mode .el-date-table__row:hover div {
              ${l((K = (J = r.value) == null ? void 0 : J.currentWeekWrapper) == null ? void 0 : K.hover)}
            }
          `)), (U = (Q = r.value.inputWrapper) == null ? void 0 : Q.default) != null && U.backgroundColor && (a += `
          .${e} .el-range-editor .el-range-input {
            ${l({
        backgroundColor: (ee = (Z = (X = r.value) == null ? void 0 : X.inputWrapper) == null ? void 0 : Z.default) == null ? void 0 : ee.backgroundColor,
        color: (pe = (ae = (re = r.value) == null ? void 0 : re.inputWrapper) == null ? void 0 : ae.default) == null ? void 0 : pe.color
      })}
          }
        `), (de = (le = (te = r.value) == null ? void 0 : te.inputWrapper) == null ? void 0 : le.default) != null && de.color && (a += `
                .${e} .el-date-editor .el-range-separator {
                  ${l({
        color: (ke = (ue = (ce = r.value) == null ? void 0 : ce.inputWrapper) == null ? void 0 : ue.default) == null ? void 0 : ke.color
      })}
          }
        `), (se = (be = ($e = r.value) == null ? void 0 : $e.dateWrapper) == null ? void 0 : be.default) != null && se.background && (a += `
          .${e}.el-picker__popper.el-popper .el-popper__arrow::before {
                    ${l({
        borderLeftColor: (fe = (he = (ve = r.value) == null ? void 0 : ve.dateWrapper) == null ? void 0 : he.default) == null ? void 0 : fe.background,
        borderTopColor: (ye = (ge = (me = r.value) == null ? void 0 : me.dateWrapper) == null ? void 0 : ge.default) == null ? void 0 : ye.background
      })}
          }
        `), (De = (We = r.value) == null ? void 0 : We.dateHeaderYearBtnWrapper) != null && De.hover && (a += `
            .${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-right:hover {
              ${l((we = (Ye = r.value) == null ? void 0 : Ye.dateHeaderYearBtnWrapper) == null ? void 0 : we.hover)}
            }
          `, a += `
            .${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-left:hover {
              ${l((He = (Me = r.value) == null ? void 0 : Me.dateHeaderYearBtnWrapper) == null ? void 0 : He.hover)}
            }
          `, a += `
            .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-right:hover {
              ${l((Be = (Ce = r.value) == null ? void 0 : Ce.dateHeaderYearBtnWrapper) == null ? void 0 : Be.hover)}
            }
          `, a += `
            .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-d-arrow-left:hover {
              ${l((Te = (Se = r.value) == null ? void 0 : Se.dateHeaderYearBtnWrapper) == null ? void 0 : Te.hover)}
            }
          `), (Pe = (xe = r.value) == null ? void 0 : xe.dateHeaderMonthBtnWrapper) != null && Pe.hover && (a += `
            .${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-arrow-right:hover {
              ${l((Ae = (Fe = r.value) == null ? void 0 : Fe.dateHeaderMonthBtnWrapper) == null ? void 0 : Ae.hover)}
            }
          `, a += `
            .${e}.el-picker__popper .el-picker-panel__body .el-date-range-picker__header .el-picker-panel__icon-btn.el-icon-arrow-left:hover {
              ${l((Ie = (Ne = r.value) == null ? void 0 : Ne.dateHeaderMonthBtnWrapper) == null ? void 0 : Ie.hover)}
            }
          `, a += `
            .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-arrow-right:hover {
              ${l((Oe = (Le = r.value) == null ? void 0 : Le.dateHeaderMonthBtnWrapper) == null ? void 0 : Oe.hover)}
            }
          `, a += `
            .${e}.el-picker__popper .el-picker-panel__body .el-date-picker__header .el-picker-panel__icon-btn.el-icon-arrow-left:hover {
              ${l((Re = (Ve = r.value) == null ? void 0 : Ve.dateHeaderMonthBtnWrapper) == null ? void 0 : Re.hover)}
            }
          `), a;
    }, _e = () => {
      const a = document.querySelector(`.${e} .style-container`);
      a && (a.innerHTML = `<style>${Ze()}${er()}</style>`);
    };
    return ie(
      () => p.cStyle,
      () => {
        _e();
      },
      {
        immediate: !0
      }
    ), ie(
      () => p.type,
      (a) => {
        a === "datetimehour" || a === "datetimeminute" ? _.dateType = "datetime" : a === "datetimerangehour" || a === "datetimerangeminute" ? _.dateType = "datetimerange" : _.dateType = a, _.dateValue = "";
      },
      {
        immediate: !0
      }
    ), ie(
      () => p.defaultDate,
      (a) => {
        _.dateValue = a;
      },
      {
        immediate: !0
      }
    ), pr(() => {
      _e();
    }), ie(
      () => [p.disabledDate, p.isDisabledFront],
      () => {
        _.disabledDateFunc = Ge;
      },
      {
        immediate: !0
      }
    ), {
      ...tr(_),
      curStyle: r,
      domOnlyClass: e,
      locale: {
        ...ir
      },
      filterAttrs: nr,
      dateFormat: Ke,
      changeTime: Qe,
      handleFocus: Xe,
      handleCalendarChange: Ue,
      formatDate: oe
      // disabledDateFunc,
    };
  }
});
export {
  kr as default
};
