// @ths/design/es/components/dynamically-module-container/src/index.vue2.js
import { defineComponent as J, provide as K, ref as _, reactive as Q, computed as P, onMounted as R, nextTick as g, watch as S, onBeforeUnmount as X, toRefs as Y, render as Z } from "vue";
import v from "axios";
import x from "lodash/cloneDeep";
import { getDomRandomClass as ee, getCssPropertyString as T, setStyle as te } from "../../../utils/index.js";
import B from "./menu-tree.vue.js";
import "../../dialog/index.js";
import oe from "../../dialog/src/index.vue.js";
const ue = J({
  name: "TDynamicallyModuleContainer",
  components: {
    MenuTree: B,
    TDialog: oe
  },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        //  组件最外层
        wrapper: {
          default: {}
        },
        // 面板内容
        content: {
          default: {}
        },
        // 遮罩样式
        mask: {
          default: {}
        },
        // 单个模块容器
        moduleContainer: {
          default: {}
        },
        // 面板底部
        foot: {
          default: {}
        },
        // 弹窗最外层
        dialogWrapper: {
          default: {
            backgroundColor: "var(--t-dynamically-module-container-dialog-bg)",
            overFlow: "hidden",
            borderTopLeftRadius: "18px",
            borderTopRightRadius: "18px",
            borderBottomLeftRadius: "18px",
            borderBottomRightRadius: "18px",
            borderLeft: "1px solid  var(--t-dynamically-module-container-dialog-border-color)",
            borderTop: "1px solid var(--t-dynamically-module-container-dialog-border-color)",
            borderRight: "1px solid  var(--t-dynamically-module-container-dialog-border-color)",
            borderBottom: "1px solid  var(--t-dynamically-module-container-dialog-border-color)",
            zIndex: 9999
          }
        },
        // 弹窗头部样式
        dialogHeader: {
          default: {
            backgroundColor: "var(--t-dynamically-module-container-dialog-title-bg)",
            width: "100%",
            display: "flex",
            justifyContent: "space-between",
            paddingLeft: "24px",
            paddingTop: "12px",
            paddingRight: "24px",
            paddingBottom: "12px",
            fontSize: "18px",
            color: "var(--t-dynamically-module-container-dialog-title-text-color)",
            borderRadius: "16px 16px 0px 0px"
          }
        },
        // 弹窗标题样式
        dialogTitle: {
          default: {
            color: "var(--t-dynamically-module-container-dialog-title-text-color)"
          }
        },
        // 弹窗关闭按钮样式
        dialogCloseButton: {
          default: {
            border: "none",
            cursor: "pointer",
            width: "24px",
            height: "24px",
            color: "transparent",
            backgroundImage: "url(../.././packages/components/dynamically-module-container/src/assets/images/close.png)",
            backgroundRepeat: "no-repeat",
            backgroundPosition: "center",
            backgroundSize: "50% 50%",
            backgroundColor: "var( --t-dynamically-module-container-close-btn-close-bg)",
            borderTop: "1px solid  var(--t-dynamically-module-container-close-btn-border-color)",
            borderRight: "1px solid  var(--t-dynamically-module-container-close-btn-border-color)",
            borderBottom: "1px solid  var(--t-dynamically-module-container-close-btn-border-color)",
            borderLeft: "1px solid  var(--t-dynamically-module-container-close-btn-border-color)",
            borderRadius: "4px",
            transition: "all .2s"
          },
          hover: {
            color: "transparent",
            backgroundColor: "var(--t-dynamically-module-container-footer-btn-hover-bg)"
          }
        },
        //  弹窗最小化按钮样式
        dialogMinButton: {
          default: {
            display: "none",
            color: "transparent"
          }
        },
        // 弹窗内容
        dialogContent: {
          default: {
            height: "100%",
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            overflow: "hidden"
          }
        },
        // 弹窗底部
        dialogFooter: {
          default: {
            height: "52px",
            backgroundColor: "var(--t-dynamically-module-container-dialog-footer-bg)",
            borderRadius: "0px 0px 18px 18px",
            padding: "12px 20px",
            display: "flex",
            justifyContent: "flex-end",
            alignItems: "center"
          }
        },
        // 弹窗确认按钮
        confirmButton: {
          default: {
            width: "90px",
            height: "28px",
            borderRadius: "89px",
            border: "none",
            color: "var(--t-dynamically-module-container-footer-btn-text-color)",
            backgroundColor: "var(--t-dynamically-module-container-footer-btn-bg)",
            transition: "all .2s"
          },
          hover: {
            backgroundColor: "var(--t-dynamically-module-container-footer-btn-hover-bg)"
          }
        },
        // 弹窗取消按钮
        cancelButton: {
          default: {
            width: "90px",
            height: "28px",
            borderRadius: "89px",
            border: "none",
            color: "var(--t-dynamically-module-container-footer-btn-text-color)",
            backgroundColor: "var(--t-dynamically-module-container-footer-btn-bg)",
            transition: "all .2s"
          },
          hover: {
            backgroundColor: "var(--t-dynamically-module-container-footer-btn-hover-bg)"
          }
        },
        // 预览区域
        previewArea: {
          default: {}
        },
        // 编辑功能模块按钮
        editBtn: {
          default: {},
          hover: {}
        },
        // 菜单内容
        menuContent: {
          default: {}
        },
        // 左右菜单内容
        menuContentLeftAndRight: {
          default: {}
        },
        // 菜单头部
        menuHeader: {
          default: {}
        },
        // 右侧菜单头部
        menuHeaderRight: {
          default: {}
        },
        // 菜单搜索框
        menuSearch: {
          default: {},
          hover: {},
          focus: {}
        },
        // 菜单搜索框图标
        menuSearchIcon: {
          default: {},
          hover: {}
        },
        // 菜单项文字
        menuItemText: {
          default: {},
          hover: {}
        },
        // 右侧菜单项
        rightMenuItem: {
          default: {}
        },
        // 右侧菜单项前置图标
        rightMenuItemIcon: {
          default: {}
        }
      })
    },
    // 获取左侧菜单树url
    getMenuTreeUrl: {
      type: String,
      default: "/lego/api/category/page/authorization/category/list"
    },
    // 获取默认菜单树(右侧菜单树)url
    getDefaultMenuTreeUrl: {
      type: String,
      default: "/lego-generated-apps/lego_file/app/src"
    },
    // 获取用户自定义菜单树(右侧菜单树)url
    getUserMenuTreeUrl: {
      type: String,
      default: "/lego/api/component/list/custom"
    },
    // 用户token
    token: {
      type: String,
      default: ""
    },
    // 应用名称
    appName: {
      type: String,
      default: ""
    },
    // 应用appId
    appId: {
      type: String,
      default: ""
    },
    // 组件componentId
    componentId: {
      type: String,
      default: ""
    },
    // 保存选择页面url
    saveSelectedPageUrl: {
      type: String,
      default: "/lego/api/component/save"
    },
    // 右侧列表默认选中页面
    defaultSelectedPage: {
      type: Array,
      default: () => []
    },
    // 是否是编辑页面
    isEditPage: {
      type: Boolean,
      default: !1
    },
    // 弹窗大小配置
    dialogMaxPosition: {
      type: Object,
      default: () => ({
        width: "880px",
        height: "620px",
        minWidth: "880px",
        minHeight: "620px",
        top: "unset",
        left: "unset",
        overflow: "unset"
      })
    },
    // 页面地址
    pageUrl: {
      type: String,
      default: "/lego-generated-apps/lego_file/app/src/"
    },
    // 获取默认选中页面（编辑页面）
    getDefaultSelectedPageUrl: {
      type: String,
      default: "/lego/api/component/list/default"
    },
    // 编辑按钮文字
    btnText: {
      type: String,
      default: "编辑功能模块"
    }
  },
  emits: ["init", "confirm", "edit", "cancel"],
  setup(a, { emit: y }) {
    K("root", window.__lego_root);
    const E = _(), r = Q({
      dialogVisible: !1,
      dialogTitle: "编辑功能模块",
      rightMenuTreeData: [],
      previewVisible: !1,
      // 默认数据（用户默认选择数据）
      defaultData: [],
      // 返回数据(弹窗点击确定后返回的数据)
      selectedData: [],
      // 记录创建内容
      mountedComponents: []
    }), s = P(() => a.cStyle), c = ee("dynamically-module-container"), h = P(() => a.token || localStorage.getItem("token") || ""), b = _(), U = () => {
      let e = "";
      const o = `.${c}.t-dynamically-module-container`, t = {
        wrapper: `${o}`,
        content: `${o} .module-container__wrapper`,
        mask: `${o} .module-container__wrapper-mask`,
        foot: `${o} .module-container__foot`
      };
      return Object.entries(s.value).forEach(([n, l]) => {
        t[n] && l && (e += te(l, t[n]));
      }), e;
    }, L = () => {
      var o, t, n, l, i, d, u, p;
      let e = "";
      return (t = (o = s.value) == null ? void 0 : o.editBtn) != null && t.default && (e += `.${c}.t-dynamically-module-container .module-container__foot .el-button  {
          ${T((l = (n = s.value) == null ? void 0 : n.editBtn) == null ? void 0 : l.default)}
        }`), (d = (i = s.value) == null ? void 0 : i.editBtn) != null && d.hover && (e += `.${c}.t-dynamically-module-container .module-container__foot .el-button:hover {
          ${T((p = (u = s.value) == null ? void 0 : u.editBtn) == null ? void 0 : p.hover)}
        }`), e;
    }, I = () => {
      const e = document.querySelector(`.${c} .style-container`);
      e && (e.innerHTML = `<style>${U()}${L()}</style>`);
    }, k = () => {
      const e = document.querySelector(`.${c} .module-container__wrapper-mask`), o = document.querySelector(`.${c} .module-container__wrapper`);
      e && o && e.setAttribute("style", `width: ${o.clientWidth}px; height: ${o.clientHeight}px;`);
    }, H = () => {
      var e, o;
      b.value && (e = s.value.previewArea) != null && e.default && (b.value.style = T((o = s.value.previewArea) == null ? void 0 : o.default));
    };
    R(() => {
      I(), a.isEditPage && g(() => {
        k();
      });
    }), S(() => a.cStyle, () => {
      I(), k();
    }, { deep: !0 });
    const j = () => {
      r.dialogVisible = !0, y("edit");
    }, D = async () => {
      var o, t, n, l, i;
      r.rightMenuTreeData.forEach((d) => {
        d != null && d.createTime && (d == null || delete d.createTime);
      });
      const e = await v({
        url: a.saveSelectedPageUrl,
        method: "post",
        params: {
          token: h.value,
          appId: a.appId,
          componentId: a.componentId,
          isDefault: 2
        },
        data: r.rightMenuTreeData
      });
      return (n = (t = (o = e == null ? void 0 : e.data) == null ? void 0 : o.data) == null ? void 0 : t.pageList) != null && n.length ? (i = (l = e == null ? void 0 : e.data) == null ? void 0 : l.data) == null ? void 0 : i.pageList : [];
    }, $ = (e, o, t) => {
      var n;
      return (
        // 动态创建内容
        mountDynamicComponent({
          id: t,
          componentName: "TComponent",
          mountTo: e,
          style: (n = s.value.moduleContainer) == null ? void 0 : n.default,
          props: {
            name: `${o}Component`,
            cStyle: {
              wrapper: {
                default: {
                  width: "100%",
                  height: "auto"
                }
              }
            }
          }
        })
      );
    }, w = async (e) => {
      e == null || e.forEach((o) => {
        const t = document.getElementById(`${o == null ? void 0 : o.componentId}`);
        t && (Z(null, t), t.remove());
      });
    }, V = (e, o) => {
      const t = document.createElement("iframe");
      t.id = `${c}-${o.pageCode}`;
      const n = `${a.pageUrl}${a.appId}/index.html#/${o.pageCode}`;
      t.src = n, t.style.width = "100%", t.style.height = "0px", t.style.border = "none", e.appendChild(t);
      const l = document.getElementById(`${c}-${o.pageCode}`);
      l && l.addEventListener("load", () => {
        var d;
        const i = (d = l == null ? void 0 : l.contentWindow) == null ? void 0 : d.document;
        if (i) {
          const u = i.documentElement.scrollHeight;
          l.style.height = `${u}px`;
        }
      });
    }, f = async (e, o = !1) => {
      let t = document.querySelector(`.${c}.t-dynamically-module-container .module-container__wrapper`);
      o && (t = b.value), t && (t != null && t.hasChildNodes()) && (t.innerHTML = ""), !(!e || (e == null ? void 0 : e.length) === 0) && (e == null || e.forEach((n) => {
        if (a.isEditPage)
          V(t, n);
        else {
          const { id: l } = $(t, n == null ? void 0 : n.pageCode, n == null ? void 0 : n.id);
          r.mountedComponents.push({
            componentId: l
          });
        }
      }));
    }, C = async () => {
      var t, n;
      let e = [], o = [];
      try {
        const l = await v({
          url: `${a.getDefaultMenuTreeUrl}/${a.appId}/src/dynamicComponentConfig.json`,
          method: "get"
        });
        if (Object.keys(l.data).length) {
          const d = [];
          l.data[a == null ? void 0 : a.componentId] ? (Object.keys(l.data[a.componentId]).forEach((u) => {
            d.push(l.data[a.componentId][u]);
          }), e = d) : e = [];
        }
        const i = await v({
          url: a.getUserMenuTreeUrl,
          method: "get",
          params: {
            token: h.value,
            appId: a.appId,
            componentId: a.componentId,
            dataFlag: "3"
          }
        });
        if (((n = (t = i == null ? void 0 : i.data) == null ? void 0 : t.data) == null ? void 0 : n.success) === !0) {
          const { customConfigs: d } = i.data.data;
          if (d != null && d.length) {
            const u = d.filter((p) => (p == null ? void 0 : p.deleteFlagCode) === "2");
            o = x(u);
          } else
            o = x(e);
        } else
          o = x(e);
        return o.sort((d, u) => d.sort - u.sort);
      } catch (l) {
        return console.error("获取用户默认选择页面失败:", l), [];
      }
    }, O = async () => {
      const e = await C();
      r.defaultData = x(e), await w(r.mountedComponents), await g(), r.selectedData = await D(), await f(r.selectedData), await g(), y("confirm", r.defaultData, r.selectedData), r.dialogVisible = !1;
    }, A = () => {
      y("cancel");
    }, N = (e) => {
      r.rightMenuTreeData = e;
    }, M = async () => {
      r.rightMenuTreeData = await C(), g(() => {
        f(r.rightMenuTreeData);
      }), y("init", r.rightMenuTreeData);
    };
    R(() => {
      a.isEditPage || M();
    });
    let m = null;
    const q = (e) => {
      r.previewVisible = !0, m && (clearTimeout(m), m = null), g(() => {
        f([e], !0);
      });
    }, z = (e) => {
      m = setTimeout(() => {
        r.previewVisible = !1;
      }, 1e3);
    }, F = () => {
      m && (clearTimeout(m), m = null), r.previewVisible = !0;
    }, W = () => {
      setTimeout(() => {
        r.previewVisible = !1;
      }, 200);
    }, G = async () => {
      var e, o;
      try {
        const t = await v({
          url: a.getDefaultSelectedPageUrl,
          method: "get",
          params: {
            token: h.value,
            appId: a.appId,
            componentId: a.componentId
          }
        });
        if ((o = (e = t == null ? void 0 : t.data) == null ? void 0 : e.data) != null && o.length)
          return t.data.data;
      } catch (t) {
        console.error("获取默认选择页面失败:", t);
      }
      return [];
    };
    return S(() => a.defaultSelectedPage, async () => {
      if (a.isEditPage) {
        const e = await G();
        g(() => {
          f(e);
        });
      }
    }, { deep: !0, immediate: !0 }), S(() => r.previewVisible, (e) => {
      e && g(() => {
        H();
      });
    }, { immediate: !0 }), X(() => {
      w(r.mountedComponents);
    }), {
      ...Y(r),
      userToken: h,
      previewRef: b,
      menuTreeRef: E,
      curStyle: s,
      domOnlyClass: c,
      MenuTree: B,
      handleConfirm: O,
      handleClose: A,
      saveSelectedPage: D,
      handleUpdateSelectedPage: N,
      createContent: $,
      createPanelContent: f,
      initData: M,
      getSelectedPage: C,
      handleEditModule: j,
      handleMouseEnterTreeNode: q,
      handleMouseLeaveTreeNode: z,
      handlePreviewMouseOver: F,
      handlePreviewMouseLeave: W,
      destroyContent: w
    };
  }
});
export {
  ue as default
};
