// @ths/design/es/components/dynamically-module-container/src/index.vue.js
import g from "./index.vue2.js";
import { resolveComponent as t, openBlock as i, createElementBlock as r, normalizeClass as y, createElementVNode as l, createCommentVNode as u, createVNode as a, withCtx as d, createTextVNode as v, toDisplayString as S } from "vue";
import "./index.vue3.js";
import c from "../../../_virtual/_plugin-vue_export-helper.js";
const f = { class: "module-container" }, M = {
  key: 0,
  class: "module-container__wrapper-mask"
}, C = { class: "module-container__foot" };
function h(e, n, T, b, w, I) {
  const s = t("el-button"), m = t("menu-tree"), p = t("t-dialog");
  return i(), r("div", {
    class: y(["t-dynamically-module-container", e.domOnlyClass])
  }, [
    l("div", f, [
      n[3] || (n[3] = l("div", { class: "module-container__wrapper" }, null, -1)),
      e.isEditPage ? (i(), r("div", M)) : u("", !0),
      l("div", C, [
        a(s, { onClick: e.handleEditModule }, {
          default: d(() => [
            v(S(e.btnText || "编辑功能模块"), 1)
          ]),
          _: 1
        }, 8, ["onClick"])
      ])
    ]),
    a(p, {
      dialogVisible: e.dialogVisible,
      "onUpdate:dialogVisible": n[2] || (n[2] = (o) => e.dialogVisible = o),
      title: e.dialogTitle,
      "max-position": e.dialogMaxPosition,
      "c-style": {
        wrapper: e.cStyle.dialogWrapper,
        header: e.cStyle.dialogHeader,
        title: e.cStyle.dialogTitle,
        closeButton: e.cStyle.dialogCloseButton,
        minButton: e.cStyle.dialogMinButton,
        content: e.cStyle.dialogContent,
        footer: e.cStyle.dialogFooter,
        confirmButton: e.cStyle.confirmButton,
        cancelButton: e.cStyle.cancelButton
      },
      onConfirm: e.handleConfirm,
      onClose: e.handleClose
    }, {
      content: d(() => [
        a(m, {
          ref: "menuTreeRef",
          "c-style": {
            wrapper: e.cStyle.menuContent,
            menuContentLeftAndRight: e.cStyle.menuContentLeftAndRight,
            menuHeader: e.cStyle.menuHeader,
            menuHeaderRight: e.cStyle.menuHeaderRight,
            menuSearch: e.cStyle.menuSearch,
            menuSearchIcon: e.cStyle.menuSearchIcon,
            menuItemText: e.cStyle.menuItemText,
            rightMenuItem: e.cStyle.rightMenuItem,
            rightMenuItemIcon: e.cStyle.rightMenuItemIcon
          },
          "dialog-visible": e.dialogVisible,
          "get-menu-tree-url": e.getMenuTreeUrl,
          "get-user-menu-tree-url": e.getUserMenuTreeUrl,
          "get-default-menu-tree-url": e.getDefaultMenuTreeUrl,
          "app-name": e.appName,
          token: e.userToken,
          "app-id": e.appId,
          "component-id": e.componentId,
          "save-selected-page-url": e.saveSelectedPageUrl,
          onUpdateSelectedPage: e.handleUpdateSelectedPage,
          onMouseEnter: e.handleMouseEnterTreeNode,
          onMouseLeave: e.handleMouseLeaveTreeNode
        }, null, 8, ["c-style", "dialog-visible", "get-menu-tree-url", "get-user-menu-tree-url", "get-default-menu-tree-url", "app-name", "token", "app-id", "component-id", "save-selected-page-url", "onUpdateSelectedPage", "onMouseEnter", "onMouseLeave"]),
        e.previewVisible ? (i(), r("div", {
          key: 0,
          ref: "previewRef",
          class: "t-dynamically-module-container__preview",
          onMouseover: n[0] || (n[0] = (...o) => e.handlePreviewMouseOver && e.handlePreviewMouseOver(...o)),
          onMouseleave: n[1] || (n[1] = (...o) => e.handlePreviewMouseLeave && e.handlePreviewMouseLeave(...o))
        }, null, 544)) : u("", !0)
      ]),
      _: 1
    }, 8, ["dialogVisible", "title", "max-position", "c-style", "onConfirm", "onClose"]),
    n[4] || (n[4] = l("div", { class: "style-container" }, null, -1))
  ], 2);
}
const U = /* @__PURE__ */ c(g, [["render", h]]);
export {
  U as default
};
