// @ths/design/es/components/dynamically-module-container/src/menu-tree.vue2.js
import { defineComponent as Z, computed as ee, reactive as te, onMounted as ne, watch as X, toRefs as ae } from "vue";
import re from "vuedraggable";
import s from "lodash/cloneDeep";
import T from "axios";
import { getDomRandomClass as ue, setStyle as oe, getCssPropertyString as m } from "../../../utils/index.js";
import le from "./tree-dragable.vue.js";
const fe = Z({
  name: "MenuTree",
  components: {
    TreeDragable: le,
    Draggable: re
  },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 左右菜单内容
        menuContentLeftAndRight: {
          default: {}
        },
        // 菜单头部
        menuHeader: {
          default: {}
        },
        // 右侧菜单头部
        menuHeaderRight: {
          default: {}
        },
        // 菜单搜索框
        menuSearch: {
          default: {},
          hover: {},
          focus: {}
        },
        // 菜单搜索框图标
        menuSearchIcon: {
          default: {},
          hover: {}
        },
        // 菜单项文字
        menuItemText: {
          default: {},
          hover: {}
        },
        // 右侧菜单项
        rightMenuItem: {
          default: {}
        },
        // 右侧菜单项前置图标
        rightMenuItemIcon: {
          default: {}
        }
      })
    },
    // 弹窗是否显示
    dialogVisible: {
      type: Boolean,
      default: !1
    },
    // 获取菜单树(左侧)Url
    getMenuTreeUrl: {
      type: String,
      default: ""
    },
    // 获取默认菜单树(右侧菜单树)url
    getDefaultMenuTreeUrl: {
      type: String,
      default: ""
    },
    // 获取用户菜单树(右侧)Url
    getUserMenuTreeUrl: {
      type: String,
      default: ""
    },
    // 用户token
    token: {
      type: String,
      default: ""
    },
    // 应用appId
    appId: {
      type: String,
      default: ""
    },
    // 应用名称
    appName: {
      type: String,
      default: ""
    },
    // 组件componentId
    componentId: {
      type: String,
      default: ""
    },
    // 保存选择页面url
    saveSelectedPageUrl: {
      type: String,
      default: ""
    }
  },
  emits: ["title-click", "update-selected-page", "mouse-enter", "mouse-leave"],
  setup(o, { emit: f }) {
    const l = ee(() => o.cStyle), c = ue("menu-tree"), t = te({
      // 左侧菜单树搜索值
      leftSearchVal: "",
      // 右侧菜单树搜索值
      rightSearchVal: "",
      // 左侧菜单树数据
      leftMenuTreeData: [],
      // 右侧默认菜单树数据（编辑页面默认配置）
      rightDefaultMenuTreeData: [],
      // 右侧菜单树数据
      rightMenuTreeData: [],
      leftMenuTreeDataCopy: [],
      rightMenuTreeDataCopy: [],
      menuLoading: !1
    }), B = () => {
      let e = "";
      const n = `.${c}.menu-tree`, u = {
        wrapper: `${n} .menu-tree__container`,
        menuContentLeftAndRight: `${n} .menu-tree__container .left-menu-tree,
        ${n} .menu-tree__container .right-menu-tree`,
        menuHeader: `${n} .menu-tree__container .left-menu-tree__title,
        ${n} .menu-tree__container .right-menu-tree__title`,
        menuHeaderRight: `${n} .menu-tree__container .right-menu-tree__title`,
        rightMenuItem: `${n} .menu-tree__container .right-menu-tree .menu-list .menu-list-item`,
        rightMenuItemIcon: `${n} .menu-tree__container .right-menu-tree .menu-list .menu-list-item .pre-icon`
      };
      return Object.entries(l.value).forEach(([a, r]) => {
        u[a] && r && (e += oe(r, u[a]));
      }), e;
    }, F = () => {
      var n, u, a, r, _, h, i, d, p, I, y, $, v, b, C, x, E, U, L, N, A, k, H, P, j, w, O, V;
      let e = "";
      return (u = (n = l.value) == null ? void 0 : n.menuSearch) != null && u.default && (e += `.${c}.menu-tree .menu-tree__container .left-menu-tree__search-box .el-input__inner,
        .${c}.menu-tree .menu-tree__container .right-menu-tree__search-box .el-input__inner {
          ${m((r = (a = l.value) == null ? void 0 : a.menuSearch) == null ? void 0 : r.default)}
        }`), (h = (_ = l.value) == null ? void 0 : _.menuSearch) != null && h.hover && (e += `.${c}.menu-tree .menu-tree__container .left-menu-tree__search-box .el-input__inner:hover,
        .${c}.menu-tree .menu-tree__container .right-menu-tree__search-box .el-input__inner:hover {
          ${m((d = (i = l.value) == null ? void 0 : i.menuSearch) == null ? void 0 : d.hover)}
        }`), (I = (p = l.value) == null ? void 0 : p.menuSearch) != null && I.focus && (e += `.${c}.menu-tree .menu-tree__container .left-menu-tree__search-box .el-input__inner:focus,
        .${c}.menu-tree .menu-tree__container .right-menu-tree__search-box .el-input__inner:focus {
          ${m(($ = (y = l.value) == null ? void 0 : y.menuSearch) == null ? void 0 : $.focus)}
        }`), (b = (v = l.value) == null ? void 0 : v.menuSearchIcon) != null && b.default && (e += `.${c}.menu-tree .menu-tree__container .left-menu-tree__search-box .search-icon,
        .${c}.menu-tree .menu-tree__container .right-menu-tree__search-box .search-icon {
          ${m((x = (C = l.value) == null ? void 0 : C.menuSearchIcon) == null ? void 0 : x.default)}
        }`), (U = (E = l.value) == null ? void 0 : E.menuSearchIcon) != null && U.hover && (e += `.${c}.menu-tree .menu-tree__container .left-menu-tree__search-box .search-icon:hover,
        .${c}.menu-tree .menu-tree__container .right-menu-tree__search-box .search-icon:hover {
          ${m((N = (L = l.value) == null ? void 0 : L.menuSearchIcon) == null ? void 0 : N.hover)}
        }`), (k = (A = l.value) == null ? void 0 : A.menuItemText) != null && k.default && (e += `.${c}.menu-tree .menu-tree__container .left-menu-tree .dragArea p.label,
        .${c}.menu-tree .menu-tree__container .right-menu-tree .menu-list .menu-list-item .name{
          ${m((P = (H = l.value) == null ? void 0 : H.menuItemText) == null ? void 0 : P.default)}
        }`), (w = (j = l.value) == null ? void 0 : j.menuItemText) != null && w.hover && (e += `.${c}.menu-tree .menu-tree__container .left-menu-tree .dragArea p.label.item:hover{
          ${m((V = (O = l.value) == null ? void 0 : O.menuItemText) == null ? void 0 : V.hover)}
        }`), e;
    }, M = () => {
      const e = document.querySelector(`.${c} .style-container`);
      e && (e.innerHTML = `<style>${B()}${F()}</style>`);
    };
    ne(() => {
      M();
    }), X(() => o.cStyle, () => {
      M();
    }, { deep: !0 });
    const g = (e) => e.map((n, u) => ({
      ...n,
      sort: u + 1
    })), R = (e) => {
    }, q = () => {
      t.rightMenuTreeData = g(t.rightMenuTreeData), t.rightMenuTreeDataCopy = s(t.rightMenuTreeData), f("update-selected-page", t.rightMenuTreeData);
    }, z = (e) => {
      const n = t.rightMenuTreeData.findIndex((u) => u.pageId === e);
      n !== -1 && t.rightMenuTreeData.splice(n, 1), t.rightMenuTreeData = g(t.rightMenuTreeData), f("update-selected-page", t.rightMenuTreeData);
    }, G = (e) => {
      const n = /* @__PURE__ */ new Map(), u = [];
      return e == null || e.forEach((a) => {
        const r = {
          type: a == null ? void 0 : a.EXT2,
          pageId: a == null ? void 0 : a.RES_ID,
          pageName: a == null ? void 0 : a.RES_NAME,
          pageCode: (a == null ? void 0 : a.EXT2) === "page" ? a.EXT3 : "",
          parentId: a == null ? void 0 : a.PARENT,
          open: !1,
          // 默认设置为false,后面处理父节点时会修改
          children: []
        };
        n.set(a.RES_ID, r);
      }), n.forEach((a) => {
        const r = n.get(a.parentId);
        r ? (r.children.push(a), r.open = !0) : u.push(a);
      }), u;
    }, J = async () => {
      t.rightMenuTreeData.forEach((e) => {
        e != null && e.createTime && (e == null || delete e.createTime);
      }), await T({
        url: o.saveSelectedPageUrl,
        method: "post",
        params: {
          token: o.token,
          appId: o.appId,
          componentId: o.componentId,
          isDefault: 2
        },
        data: t.rightMenuTreeData
      });
    }, S = () => {
      t.menuLoading = !0, T({
        url: `${o.getDefaultMenuTreeUrl}/${o.appId}/src/dynamicComponentConfig.json`,
        method: "get"
      }).then((e) => {
        if (Object.keys(e.data).length) {
          const n = [];
          e.data[o.componentId] ? (Object.keys(e.data[o.componentId]).forEach((u) => {
            n.push(e.data[o.componentId][u]);
          }), t.rightDefaultMenuTreeData = n) : t.rightDefaultMenuTreeData = [];
        }
      }), Promise.all([
        T({
          url: o.getMenuTreeUrl,
          method: "get",
          params: {
            token: o.token,
            appId: o.appId,
            componentId: o.componentId
          }
        }),
        T({
          url: o.getUserMenuTreeUrl,
          method: "get",
          params: {
            token: o.token,
            appId: o.appId,
            componentId: o.componentId,
            dataFlag: "3"
          }
        })
      ]).then(async ([e, n]) => {
        var u, a, r, _;
        if (((a = (u = e == null ? void 0 : e.data) == null ? void 0 : u.data) == null ? void 0 : a.success) === !0) {
          const { selectedPage: h, tree: i } = e.data.data;
          i != null && i.length && (t.leftMenuTreeData = G(i)), t.leftMenuTreeDataCopy = s(t.leftMenuTreeData);
        }
        if (((_ = (r = n == null ? void 0 : n.data) == null ? void 0 : r.data) == null ? void 0 : _.success) === !0) {
          const { customConfigs: h } = n.data.data;
          if (h != null && h.length) {
            const i = h.filter((d) => (d == null ? void 0 : d.deleteFlagCode) === "2");
            t.rightMenuTreeData = g(i).sort((d, p) => d.sort - p.sort);
          } else
            t.rightMenuTreeData = g(t.rightDefaultMenuTreeData).sort((i, d) => i.sort - d.sort), await J();
        } else
          t.rightMenuTreeData = g(t.rightDefaultMenuTreeData).sort((h, i) => h.sort - i.sort);
        t.rightMenuTreeDataCopy = s(t.rightMenuTreeData), f("update-selected-page", t.rightMenuTreeData);
      }).catch((e) => {
        console.error("获取菜单树数据失败:", e);
      }).finally(() => {
        t.menuLoading = !1;
      });
    }, D = (e, n) => {
      const u = [];
      return Array.isArray(e) && e.forEach((a) => {
        const r = s(a);
        r.pageName && r.pageName.includes(n) ? u.push(r) : r.children && r.children.length > 0 && D(r.children, n).length && (r.children = D(r.children, n), u.push(r));
      }), u;
    }, K = (e) => {
      f("mouse-enter", e);
    }, Q = (e) => {
      f("mouse-leave", e);
    }, W = (e) => {
      e === "" ? t.leftMenuTreeData = s(t.leftMenuTreeDataCopy) : t.leftMenuTreeData = D(t.leftMenuTreeDataCopy, e);
    }, Y = (e) => {
      e === "" ? t.rightMenuTreeData = s(t.rightMenuTreeDataCopy) : t.rightMenuTreeData = t.rightMenuTreeData.filter((n) => n.pageName.includes(e));
    };
    return X(() => o.dialogVisible, (e) => {
      e === !0 && S();
    }, { immediate: !0 }), {
      ...ae(t),
      curStyle: l,
      domOnlyClass: c,
      handleDragStart: R,
      handleDragEnd: q,
      handleDeleteRightMenuItem: z,
      handleLeftSearchChange: W,
      handleRightSearchChange: Y,
      getMenuTreeList: S,
      handleMouseEnter: K,
      handleMouseLeave: Q
    };
  }
});
export {
  fe as default
};
