// @ths/design/es/components/dynamically-module-container/src/menu-tree.vue.js
import D from "./menu-tree.vue2.js";
import { resolveComponent as o, resolveDirective as F, openBlock as l, createElementBlock as s, normalizeClass as d, createElementVNode as t, createVNode as u, withCtx as i, withDirectives as c, createBlock as M, createCommentVNode as r, createTextVNode as m, toDisplayString as C } from "vue";
import "./menu-tree.vue3.js";
import b from "../../../_virtual/_plugin-vue_export-helper.js";
const V = { class: "menu-tree__container" }, k = { class: "left-menu-tree" }, _ = { class: "left-menu-tree__search-box" }, y = {
  key: 1,
  class: "no-data"
}, S = { class: "right-menu-tree" }, T = { class: "right-menu-tree__title" }, B = { class: "right-menu-tree__search-box" }, I = { class: "menu-list-item" }, L = { class: "left" }, A = { class: "name" }, N = ["onClick"], $ = {
  key: 0,
  class: "no-data"
};
function w(e, n, R, U, z, O) {
  const p = o("el-input"), h = o("tree-dragable"), v = o("el-tooltip"), E = o("el-icon"), f = o("draggable"), g = F("loading");
  return l(), s("div", {
    class: d(["menu-tree", e.domOnlyClass])
  }, [
    t("div", V, [
      t("div", k, [
        n[4] || (n[4] = t("div", { class: "left-menu-tree__title" }, " 全部专题 ", -1)),
        t("div", _, [
          u(p, {
            modelValue: e.leftSearchVal,
            "onUpdate:modelValue": n[0] || (n[0] = (a) => e.leftSearchVal = a),
            placeholder: "请输入关键词",
            clearable: "",
            onInput: e.handleLeftSearchChange
          }, {
            prefix: i(() => n[2] || (n[2] = [
              t("div", { class: "search-icon" }, null, -1)
            ])),
            _: 1
          }, 8, ["modelValue", "onInput"])
        ]),
        c((l(), s("div", {
          "element-loading-background": "rgba(0, 0, 0, 0)",
          "element-loading-custom-class": "menu-loading",
          class: d(["left-menu-tree-content", { empty: !e.leftMenuTreeData.length }])
        }, [
          e.leftMenuTreeData.length ? (l(), M(h, {
            key: 0,
            list: e.leftMenuTreeData,
            "app-id": e.appId,
            "app-name": e.appName,
            "component-id": e.componentId,
            onEnd: e.handleDragEnd,
            onMouseEnter: e.handleMouseEnter,
            onMouseLeave: e.handleMouseLeave
          }, null, 8, ["list", "app-id", "app-name", "component-id", "onEnd", "onMouseEnter", "onMouseLeave"])) : r("", !0),
          !e.leftMenuTreeData.length && !e.menuLoading ? (l(), s("div", y, n[3] || (n[3] = [
            t("div", { class: "no-data-icon" }, null, -1),
            t("div", { class: "no-data-text" }, " 暂无数据 ", -1)
          ]))) : r("", !0)
        ], 2)), [
          [g, e.menuLoading]
        ])
      ]),
      t("div", S, [
        t("div", T, [
          n[7] || (n[7] = m(" 右侧数据面板 ")),
          u(v, {
            class: "item",
            effect: "dark",
            placement: "top",
            "popper-class": "drag-menu-tip"
          }, {
            content: i(() => n[5] || (n[5] = [
              t("div", { class: "tip-content" }, [
                m(' 1. 拖拽 "左侧" 组件到 "右侧" 定制面板 '),
                t("br"),
                m(" 2. 拖拽组件可定制页面组件顺序 ")
              ], -1)
            ])),
            default: i(() => [
              n[6] || (n[6] = t("span", { class: "tip" }, null, -1))
            ]),
            _: 1,
            __: [6]
          })
        ]),
        t("div", B, [
          u(p, {
            modelValue: e.rightSearchVal,
            "onUpdate:modelValue": n[1] || (n[1] = (a) => e.rightSearchVal = a),
            placeholder: "请输入关键词",
            clearable: "",
            onInput: e.handleRightSearchChange
          }, {
            prefix: i(() => n[8] || (n[8] = [
              t("div", { class: "search-icon" }, null, -1)
            ])),
            _: 1
          }, 8, ["modelValue", "onInput"])
        ]),
        c((l(), s("div", {
          "element-loading-background": "rgba(0, 0, 0, 0)",
          "element-loading-custom-class": "menu-loading",
          class: d(["menu-list", { empty: !e.rightMenuTreeData.length }])
        }, [
          u(f, {
            class: d(["dragArea", { empty: !e.rightMenuTreeData.length }]),
            "item-key": "sort",
            list: e.rightMenuTreeData,
            group: { name: "drag-data" },
            "ghost-class": "ghost",
            "chosen-class": "chosenClass",
            "drag-class": "drag",
            animation: "200",
            "fallback-class": !0,
            onStart: e.handleDragStart,
            onEnd: e.handleDragEnd
          }, {
            item: i(({ element: a }) => [
              t("div", I, [
                t("div", L, [
                  n[9] || (n[9] = t("div", { class: "pre-icon" }, null, -1)),
                  t("div", A, C(a.pageName), 1)
                ]),
                t("div", {
                  class: "delete",
                  onClick: (j) => e.handleDeleteRightMenuItem(a.pageId)
                }, [
                  u(E, { class: "el-icon-delete" })
                ], 8, N)
              ])
            ]),
            _: 1
          }, 8, ["class", "list", "onStart", "onEnd"]),
          !e.rightMenuTreeData.length && !e.menuLoading ? (l(), s("div", $, n[10] || (n[10] = [
            t("div", { class: "no-data-icon" }, null, -1),
            t("div", { class: "no-data-text" }, " 暂无数据 ", -1)
          ]))) : r("", !0)
        ], 2)), [
          [g, e.menuLoading]
        ])
      ]),
      n[11] || (n[11] = t("div", { class: "style-container" }, null, -1))
    ])
  ], 2);
}
const K = /* @__PURE__ */ b(D, [["render", w]]);
export {
  K as default
};
