// @ths/design/es/components/dynamically-module-container/src/tree-dragable.vue2.js
import { reactive as g, toRefs as m } from "vue";
import u from "vuedraggable";
const N = {
  name: "TreeDraggable",
  components: { draggable: u },
  props: {
    list: {
      type: Array,
      default: () => []
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    // 应用名称
    appName: {
      type: String,
      default: ""
    },
    // 应用appId
    appId: {
      type: String,
      default: ""
    },
    // 组件componentId
    componentId: {
      type: String,
      default: ""
    }
  },
  emits: ["clone", "start", "end", "mouse-enter", "mouse-leave"],
  setup(n, { emit: a }) {
    const o = g({}), r = (e) => {
      const t = {
        appName: n.appName,
        appId: n.appId,
        componentId: n.componentId,
        pageId: e.pageId,
        pageName: e.pageName,
        pageCode: e.pageCode,
        sort: null
      };
      return a("clone", t), t;
    }, s = (e) => {
      var t;
      a("start", (t = e == null ? void 0 : e.item) == null ? void 0 : t.className);
    }, p = (e) => {
      a("start", e);
    }, l = () => {
      a("end", "");
    }, d = (e) => {
      a("mouse-enter", e);
    }, c = (e) => {
      a("mouse-leave", e);
    };
    return {
      ...m(o),
      handleDragStart: s,
      handleDragStartNested: p,
      handleDragEnd: l,
      cloneDrag: r,
      handleMouseEnter: d,
      handleMouseLeave: c
    };
  }
};
export {
  N as default
};
