// @ths/design/es/components/dynamically-module-container/src/tree-dragable.vue.js
import v from "./tree-dragable.vue2.js";
import { resolveComponent as p, openBlock as e, createElementBlock as r, createVNode as h, withCtx as f, createElementVNode as d, normalizeClass as _, toDisplayString as s, createCommentVNode as t } from "vue";
import "./tree-dragable.vue3.js";
import E from "../../../_virtual/_plugin-vue_export-helper.js";
const M = { class: "tree-dragable" }, N = ["onClick"], D = ["title"], k = ["title", "onMouseenter", "onMouseleave"], y = ["title"];
function C(S, x, n, o, L, T) {
  const b = p("TreeDraggable"), u = p("draggable");
  return e(), r("div", M, [
    h(u, {
      class: "dragArea",
      tag: "ul",
      list: n.list,
      "chosen-class": "chosenClass",
      "ghost-class": "ghost",
      "force-fallback": !1,
      animation: "300",
      group: {
        name: "drag-data",
        put: !1,
        // 允许拖入
        pull: "clone"
      },
      disabled: n.disabled,
      handle: ".move",
      "item-key": "name",
      clone: o.cloneDrag,
      sort: !1,
      onStart: o.handleDragStart,
      onEnd: o.handleDragEnd
    }, {
      item: f(({ element: a }) => {
        var i, c, l;
        return [
          d("li", {
            class: _(["item", { "drag-tree-node-open": a.open, "drag-tree-node-hide": !a.open }])
          }, [
            (i = a == null ? void 0 : a.children) != null && i.length ? (e(), r("div", {
              key: 0,
              class: "list-group",
              onClick: (g) => a.open = !a.open
            }, [
              d("p", {
                class: _(["icon", a.open ? "drag-tree-node-open" : "drag-tree-node-hide"])
              }, null, 2),
              d("p", {
                class: "label item",
                title: a.pageName
              }, s(a.pageName), 9, D)
            ], 8, N)) : t("", !0),
            !((c = a == null ? void 0 : a.children) != null && c.length) && (a == null ? void 0 : a.type) === "page" ? (e(), r("p", {
              key: 1,
              title: a.pageName,
              class: "label item move",
              onMouseenter: (g) => o.handleMouseEnter(a),
              onMouseleave: (g) => o.handleMouseLeave(a)
            }, s(a.pageName), 41, k)) : t("", !0),
            !((l = a == null ? void 0 : a.children) != null && l.length) && (a == null ? void 0 : a.type) === "category" ? (e(), r("p", {
              key: 2,
              title: a.pageName,
              class: "label item category"
            }, s(a.pageName), 9, y)) : t("", !0),
            h(b, {
              "app-name": n.appName,
              "app-id": n.appId,
              "component-id": n.componentId,
              list: a.children,
              disabled: n.disabled,
              onStart: o.handleDragStartNested,
              onEnd: o.handleDragEnd,
              onClone: o.cloneDrag,
              onMouseEnter: o.handleMouseEnter,
              onMouseLeave: o.handleMouseLeave
            }, null, 8, ["app-name", "app-id", "component-id", "list", "disabled", "onStart", "onEnd", "onClone", "onMouseEnter", "onMouseLeave"])
          ], 2)
        ];
      }),
      _: 1
    }, 8, ["list", "disabled", "clone", "onStart", "onEnd"])
  ]);
}
const w = /* @__PURE__ */ E(v, [["render", C]]);
export {
  w as default
};
