// @ths/design/es/components/calendar-pollution/src/index.vue2.js
import { defineComponent as E, computed as Y, ref as O, getCurrentInstance as R, reactive as k, onMounted as U, watch as p, toRefs as F } from "vue";
import N from "lodash/merge";
import S from "lodash/cloneDeep";
import { ElConfigProvider as b } from "element-plus";
import { getDomRandomClass as P, setStyle as l } from "../../../utils/index.js";
const T = {
  startValue: "startValue",
  endValue: "endValue",
  startColor: "startColor",
  endColor: "endColor",
  name: "name",
  date: "date"
}, g = "rgba(0, 140, 255, 0.3)", q = "rgba(0, 140, 255, 0.1)", $ = {
  // 外层容器
  wrapper: {
    default: {}
  },
  // 带日期、按钮的头部
  headerWrapper: {
    default: {}
  },
  // 头部文字
  headerDate: {
    default: {}
  },
  // 切换按钮
  button: {
    default: {},
    hover: {},
    focus: {}
  },
  // 表格
  tableWrapper: {
    default: {}
  },
  // 日历表头
  tableHeader: {
    default: {}
  },
  // 选中的日期单元格
  selectedDateWrapper: {
    default: {
      background: g
    }
  },
  // 选中的日期（文本）
  selectedDate: {
    default: {}
  },
  // 当月单元格
  currentDateWrapper: {
    default: {
      background: g
    }
  },
  // 其他单元格
  otherDateWrapper: {
    default: {
      background: q
    }
  },
  // 当月单元格日期数字
  currentDate: {
    default: {}
  },
  // 其他月单元格日期数字
  otherDate: {
    default: {}
  },
  // 污染物名称
  name: {
    default: {}
  },
  // 指引线
  line: {
    default: {}
  }
}, I = E({
  name: "TCalendarPollution",
  components: {
    [b.name]: b
  },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        ...$
      })
    },
    // 选中月份值
    dateValue: {
      type: String,
      default: () => ""
    },
    // 展示方向，vertical / horizontal
    direction: {
      type: String,
      default: () => "vertical "
    },
    // 数据
    dataList: {
      type: Array,
      default: () => []
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => T
    }
  },
  // 事件名
  emits: ["change-click"],
  setup(n, { emit: C }) {
    const t = Y(() => ({
      ...$,
      ...n.cStyle
    })), r = P("calendar-pollution"), m = O(), { proxy: i } = R(), H = ["日", "一", "二", "三", "四", "五", "六"], v = "#eee", h = (a) => {
      if (!a || a.split("-").length < 2)
        return "";
      const c = a.split("-")[0].slice(
        0,
        4
        /* yearLength */
      ), e = a.split("-")[1].slice(
        0,
        2
        /* dateLength */
      );
      return /^\d{4}$/.test(c) && /^\d{1,2}$/.test(e) && Number(e) >= 1 && Number(e) <= 12 ? `${c}-${e}` : "";
    }, D = (a, c) => {
      const e = {
        Year: 0,
        TYear: "0",
        Month: 0,
        TMonth: "0",
        Day: 0,
        TDay: "0",
        Hour: 0,
        THour: "0",
        hour: 0,
        Thour: "0",
        Minute: 0,
        TMinute: "0",
        Second: 0,
        TSecond: "0",
        Millisecond: 0
      };
      return e.Year = a.getFullYear(), e.TYear = String(e.Year).substr(2), e.Month = a.getMonth() + 1, e.TMonth = e.Month < 10 ? `0${e.Month}` : String(e.Month), e.Day = a.getDate(), e.TDay = e.Day < 10 ? `0${e.Day}` : String(e.Day), e.Hour = a.getHours(), e.THour = e.Hour < 10 ? `0${e.Hour}` : String(e.Hour), e.hour = e.Hour < 13 ? e.Hour : e.Hour - 12, e.Thour = e.hour < 10 ? `0${e.hour}` : String(e.hour), e.Minute = a.getMinutes(), e.TMinute = e.Minute < 10 ? `0${e.Minute}` : String(e.Minute), e.Second = a.getSeconds(), e.TSecond = e.Second < 10 ? `0${e.Second}` : String(e.Second), e.Millisecond = a.getMilliseconds(), c.replace(/yyyy/gi, String(e.Year)).replace(/yyy/gi, String(e.Year)).replace(/yy/gi, e.TYear).replace(/y/gi, e.TYear).replace(/MM/g, e.TMonth).replace(/M/g, String(e.Month)).replace(/dd/gi, e.TDay).replace(/d/gi, String(e.Day)).replace(/HH/g, e.THour).replace(/H/g, String(e.Hour)).replace(/hh/g, e.Thour).replace(/h/g, String(e.hour)).replace(/mm/g, e.TMinute).replace(/m/g, String(e.Minute)).replace(/ss/gi, e.TSecond).replace(/s/gi, String(e.Second)).replace(/fff/gi, String(e.Millisecond));
    }, d = k({
      curDateValue: new Date(h(n.dateValue) || /* @__PURE__ */ new Date()),
      // 当前月份值
      curData: [],
      // 当前月日历数据
      curDate: D(
        new Date(h(n.dateValue) || /* @__PURE__ */ new Date()),
        "yyyy年MM月"
        /* monthFormat */
      )
      // 头部日期显示
    }), V = (a) => {
      m.value.selectDate(a), d.curDate = D(
        new Date(d.curDateValue),
        "yyyy年MM月"
        /* monthFormat */
      );
      const c = D(
        new Date(d.curDateValue),
        "yyyy-MM"
        /* monthBreakFormat */
      );
      C("change-click", c);
    }, f = () => {
      const a = N(S(T), n.dataAlias), {
        startValue: c,
        endValue: e,
        startColor: o,
        endColor: s,
        name: A,
        date: L
      } = a;
      d.curData = [], n.dataList && n.dataList.length && (d.curData = n.dataList.map((y) => {
        const u = S(y);
        return !y[o] && !y[s] ? (u[o] = v, u[s] = v) : y[o] ? y[s] || (u[s] = u[o]) : u[o] = u[s], {
          startValue: u[c] || "",
          endValue: u[e] || "",
          startColor: u[o],
          endColor: u[s],
          name: u[A] || "-",
          date: D(
            new Date(u[L]),
            "yyyy-MM-dd"
            /* dayFormat */
          )
        };
      }));
    }, W = (a) => {
      const c = d.curData.length;
      let e = {
        startValue: "",
        endValue: "",
        startColor: "",
        endColor: "",
        name: " ",
        date: "-"
      };
      for (let o = 0; o < c; o++)
        if (d.curData[o].date === a)
          return e = d.curData[o], e;
      return e;
    }, w = () => {
      i == null || i.$el.querySelector("div.el-calendar__header").remove();
      const a = i == null ? void 0 : i.$el.querySelectorAll("table.el-calendar-table thead th");
      a.forEach((c, e) => {
        a[e].innerHTML = H[e];
      });
    }, _ = () => {
      let a = "";
      return t.value.wrapper && (a += l(t.value.wrapper, `.${r}`)), t.value.headerWrapper && (a += l(t.value.headerWrapper, `.${r} .calendar-header`)), t.value.headerDate && (a += l(t.value.headerDate, `.${r} .calendar-header .header-date`)), t.value.button && (a += l(t.value.button, `.${r} .calendar-header .el-button`)), t.value.tableWrapper && (a += l(t.value.tableWrapper, `.${r} .el-calendar .el-calendar-table`)), t.value.tableHeader && (a += l(t.value.tableHeader, `.${r} .el-calendar-table thead th`), a += l(t.value.tableHeader, `.${r} .el-calendar-table thead th:first-child`), a += l(t.value.tableHeader, `.${r} .el-calendar-table thead th:last-child`)), t.value.selectedDateWrapper && (a += l(t.value.selectedDateWrapper, `.${r} .el-calendar-table td.current.is-selected .el-calendar-day`)), t.value.selectedDate && (a += l(
        t.value.selectedDate,
        `.${r} .el-calendar-table td.current.is-selected .el-calendar-day .calendar-day .is-selected`
      )), t.value.currentDateWrapper && (a += l(t.value.currentDateWrapper, `.${r} .el-calendar-table td.current:not(.is-selected) .el-calendar-day`)), t.value.otherDateWrapper && (a += l(t.value.otherDateWrapper, `.${r} .el-calendar-table td.prev .el-calendar-day`), a += l(t.value.otherDateWrapper, `.${r} .el-calendar-table td.next .el-calendar-day`)), t.value.currentDate && (a += l(t.value.currentDate, `.${r} .el-calendar-table td.current .calendar-day p`)), t.value.otherDate && (a += l(t.value.otherDate, `.${r} .el-calendar-table td.prev .calendar-day p`), a += l(t.value.otherDate, `.${r} .el-calendar-table td.next .calendar-day p`)), t.value.name && (a += l(t.value.name, `.${r} .calendar-day .pollution-name`)), t.value.line && (a += l(t.value.line, `.${r} .calendar-day .line`)), a;
    }, M = () => {
      const a = document.querySelector(`.${r} > .style-container`);
      a && (a.innerHTML = `<style>${_()}</style>`);
    };
    return U(() => {
      w(), M(), f();
    }), p(
      () => [n.dataList, n.dataAlias],
      () => {
        f();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), p(
      () => [n.cStyle],
      () => {
        M();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), p(
      () => [n.dateValue],
      () => {
        d.curDate = D(
          new Date(h(n.dateValue) || /* @__PURE__ */ new Date()),
          "yyyy年MM月"
          /* monthFormat */
        ), d.curDateValue = new Date(h(n.dateValue) || /* @__PURE__ */ new Date()), f();
      },
      {
        immediate: !0
      }
    ), {
      curStyle: t,
      ...F(d),
      domOnlyClass: r,
      selectDate: V,
      calendar: m,
      getPollutionDate: W,
      DEFAULT_CURRENT_DATE_TD_BACKGROUND_COLOR: g
    };
  }
});
export {
  I as default
};
