// @ths/design/es/components/calendar-pollution/src/index.vue.js
import _ from "./index.vue2.js";
import { resolveComponent as a, openBlock as y, createElementBlock as g, normalizeClass as D, createElementVNode as t, toDisplayString as i, createVNode as n, withCtx as f, withDirectives as h, normalizeStyle as V, createCommentVNode as $, vShow as P } from "vue";
import "./index.vue3.js";
import k from "../../../_virtual/_plugin-vue_export-helper.js";
const w = { class: "calendar-header" }, z = { class: "header-date" }, b = { class: "calendar-day" }, S = { class: "pollution" }, N = ["title"];
function B(e, l, E, j, O, U) {
  const s = a("el-button"), v = a("el-button-group"), C = a("el-calendar");
  return y(), g("div", {
    class: D(["t-calendar-pollution", [e.domOnlyClass, { horizontal: e.direction === "horizontal" }]])
  }, [
    t("div", w, [
      t("span", z, i(e.curDate || "-"), 1),
      n(v, null, {
        default: f(() => [
          n(s, {
            class: "arrow-left",
            size: "mini",
            icon: "el-icon-arrow-left",
            onClick: l[0] || (l[0] = (o) => e.selectDate("prev-month"))
          }),
          n(s, {
            size: "mini",
            icon: "el-icon-arrow-right",
            onClick: l[1] || (l[1] = (o) => e.selectDate("next-month"))
          })
        ]),
        _: 1
      })
    ]),
    n(C, {
      ref: "calendar",
      modelValue: e.curDateValue,
      "onUpdate:modelValue": l[2] || (l[2] = (o) => e.curDateValue = o)
    }, {
      dateCell: f(({ data: o }) => {
        var r, d, u, c, m, p;
        return [
          t("div", b, [
            t("p", {
              class: D(o.isSelected ? "is-selected" : "")
            }, i(o.day.split("-").slice(2).join("").replace(/^0/, "")), 3),
            h(t("div", S, [
              t("span", {
                class: "pollution-name",
                title: (r = e.getPollutionDate(o.day)) == null ? void 0 : r.name
              }, i(((d = e.getPollutionDate(o.day)) == null ? void 0 : d.name) || "-"), 9, N),
              (c = (u = e.getPollutionDate(o.day)) == null ? void 0 : u.name) != null && c.trim() ? (y(), g("span", {
                key: 0,
                class: "line",
                style: V({ background: `linear-gradient(90deg, ${(m = e.getPollutionDate(o.day)) == null ? void 0 : m.startColor} 0%, ${(p = e.getPollutionDate(o.day)) == null ? void 0 : p.endColor} 100%` })
              }, null, 4)) : $("", !0)
            ], 512), [
              [P, e.getPollutionDate(o.day)]
            ])
          ])
        ];
      }),
      _: 1
    }, 8, ["modelValue"]),
    l[3] || (l[3] = t("div", { class: "style-container" }, null, -1))
  ], 2);
}
const H = /* @__PURE__ */ k(_, [["render", B]]);
export {
  H as default
};
