// @ths/design/es/components/user-multiple-dialog-input/src/index.vue.js
import m from "./index.vue2.js";
import { resolveComponent as o, openBlock as n, createElementBlock as c, createVNode as t, withCtx as a, createTextVNode as f, createBlock as _, mergeProps as b, createCommentVNode as C } from "vue";
import g from "../../../_virtual/_plugin-vue_export-helper.js";
const h = { class: "input-wrapper" };
function k(e, l, N, V, D, S) {
  const r = o("search"), s = o("el-icon"), p = o("el-button"), d = o("el-input"), u = o("t-user-multiple-dialog");
  return n(), c("div", h, [
    t(d, {
      modelValue: e.userNameState,
      "onUpdate:modelValue": l[0] || (l[0] = (i) => e.userNameState = i),
      title: e.userNameState,
      placeholder: e.placeholder,
      clearable: "",
      disabled: e.disabled,
      onFocus: e.openDialog,
      onClear: e.clearInput
    }, {
      append: a(() => [
        t(p, {
          size: "small",
          disabled: e.disabled,
          onClick: e.openDialog
        }, {
          default: a(() => [
            t(s, { size: 13 }, {
              default: a(() => [
                t(r)
              ]),
              _: 1
            }),
            l[1] || (l[1] = f(" 选择 "))
          ]),
          _: 1,
          __: [1]
        }, 8, ["disabled", "onClick"])
      ]),
      _: 1
    }, 8, ["modelValue", "title", "placeholder", "disabled", "onFocus", "onClear"]),
    e.visible ? (n(), _(u, b({ key: 0 }, e.$attrs, {
      "person-data": e.curUser,
      "result-data-path": e.resultDataPath,
      "default-props": e.defaultProps,
      onOnSure: e.selected,
      onOnClose: e.closeDialog
    }), null, 16, ["person-data", "result-data-path", "default-props", "onOnSure", "onOnClose"])) : C("", !0)
  ]);
}
const B = /* @__PURE__ */ g(m, [["render", k]]);
export {
  B as default
};
