// @ths/design/es/components/user-multiple-dialog-input/src/index.vue2.js
import { defineComponent as p, reactive as m, computed as u, ref as d, watchEffect as g, toRefs as D } from "vue";
import N from "../../user-multiple-dialog/src/index.vue.js";
const y = p({
  name: "TUserMultipleDialogInput",
  components: {
    TUserMultipleDialog: N
    // 多选用户
  },
  props: {
    // 人员数据
    user: {
      type: Array,
      default: () => []
    },
    // 是否不可编辑
    disabled: {
      type: Boolean,
      default: !1
    },
    // 输入框提示文字
    placeholder: {
      type: String,
      default: "请选择"
    },
    // 部门树请求地址
    deptDataUrl: {
      type: String,
      default: ""
    },
    // 人员列表数据-返回数据在 data 中的位置
    resultDataPath: {
      type: String,
      default: "data"
    },
    // 人员表格数据路径
    userResultDataPath: {
      type: String,
      default: "pageInfo"
    },
    // 人员列表数据-数据别名
    defaultProps: {
      type: Object,
      default: () => ({
        userId: "userId",
        userName: "userName",
        orgName: "orgName",
        deptName: "deptName",
        positionName: "positionName",
        enforceCardCode: "enforceCardCode"
      })
    }
  },
  emits: ["update:user", "selected", "clear"],
  setup(t, { emit: a }) {
    const r = m({
      // 多选用户弹框是否可见
      visible: !1,
      // userId
      user: u(() => t.user),
      deptDataUrl: t.deptDataUrl,
      // 树平行结构数据存储字段
      resultDataPath: u(() => t.resultDataPath),
      userResultDataPath: u(() => t.userResultDataPath)
    }), s = d(""), l = d([]);
    g(() => {
      s.value = t.user.map((e) => e[t.defaultProps.userName]).toString(), l.value = t.user.map((e) => ({
        userId: e[t.defaultProps.userId],
        userName: e[t.defaultProps.userName]
      }));
    });
    const n = (e) => {
      r.visible = !0, e.target.blur();
    }, c = () => {
      a("update:user", []), a("clear");
    }, o = () => {
      r.visible = !1;
    }, f = (e) => {
      s.value = e.map((i) => i.userName).toString(), a("update:user", e), a("selected", e), o();
    };
    return {
      curUser: l,
      userNameState: s,
      ...D(r),
      openDialog: n,
      clearInput: c,
      selected: f,
      closeDialog: o
    };
  }
});
export {
  y as default
};
