// @ths/design/es/components/chart-bar-capsule/src/index.vue2.js
import { defineComponent as F, computed as E, ref as H, reactive as U, watch as W, nextTick as M, onMounted as K, onUnmounted as V } from "vue";
import * as X from "echarts";
import m from "lodash/merge";
import w from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as J, DEFAULT_OPTIONS_AXIS as Q } from "../../../utils/constant.js";
import { getDomRandomClass as Y, setStyle as j } from "../../../utils/index.js";
const Z = m(w(J), w(Q)), D = m(
  w(Z),
  {
    // 气泡框
    tooltip: {
      axisPointer: {
        type: "line"
      },
      formatter: (e) => {
        var h, y, S, f;
        let g = "";
        for (let n = 0; n < e.length; n++)
          g += `${(h = e[n]) == null ? void 0 : h.marker} ${(y = e[n]) == null ? void 0 : y.seriesName}: ${(S = e[n]) == null ? void 0 : S.value} <br />`;
        return g = `${(f = e[0]) == null ? void 0 : f.name} <br /> ${g}`, g;
      }
    },
    // 网格配置
    grid: {
      top: "13%",
      left: "1%",
      right: "2%",
      bottom: "2%",
      containLabel: !0
    },
    // x轴
    xAxis: {
      boundaryGap: !0
    },
    // y轴
    yAxis: {
      name: "(单位)",
      splitNumber: 3,
      splitLine: {
        show: !0
      }
    }
  }
), ne = F({
  name: "TChartBarCapsule",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 传入数组数据 如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => []
    },
    // 设置主轴
    mainAxis: {
      type: String,
      default: "x"
    },
    // 包含 系列样式、别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // x 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => D,
      required: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 柱子样式
    barStyle: {
      type: Object,
      default: () => ({})
    },
    // 图表类型（用于接主题中的样式）
    bar: {
      type: Object,
      default: () => ({})
    }
  },
  // 方法名
  emits: ["com-click", "legend-select"],
  setup(e, { emit: g }) {
    const h = Y("chart-bar-capsule"), y = E(() => e.cStyle), S = H(null);
    let f = null;
    const n = U({
      // 图表数据
      chartData: [],
      // 处理后的数据
      handleAfterData: [],
      // 系列名列表
      nameData: []
    }), _ = () => {
      var r, o, c;
      let t = 0;
      const l = [], a = [];
      for (let s = 0; s < n.chartData.length; s++) {
        const d = n.handleAfterData[s] ? (r = n.handleAfterData[s]) == null ? void 0 : r.data : [], b = Math.max(...d);
        l.push((o = n.chartData[s]) == null ? void 0 : o.name), t = b > t ? b : t;
      }
      n.nameData = l;
      for (let s = 0; s < ((c = n.chartData[0]) == null ? void 0 : c.data).length; s++)
        a.push(t + 20);
      return a;
    }, G = () => {
      D.tooltip.formatter = (t) => {
        var c, s;
        const l = [], a = [], r = [];
        for (let d = 0, b = t.length; d < b; d++)
          ((c = t[d]) == null ? void 0 : c.seriesName) !== "gap" && (l.push(t[d].value), a.push(t[d].marker), r.push(t[d].seriesName));
        let o = "";
        for (let d = 0; d < l.length; d++) {
          const b = n.nameData.findIndex(($) => $ === r[d]);
          o += `${a[d]} ${r[d]} : ${((s = n.chartData[b]) == null ? void 0 : s.data)[t[d].dataIndex]}<br />`;
        }
        return o = `${t[0].name} <br /> ${o}`, o;
      };
    }, v = (t) => {
      n.handleAfterData = [];
      let l = [];
      l = t.map((a) => {
        const { data: r } = a;
        return [...r];
      });
      for (let a = 0; a < l.length; a++) {
        for (let r = 0; r < l[a].length; r++)
          l[a][r] = a === 0 ? Number(l[a][r]) : Number(l[a][r]) + Number(l[a - 1][r]);
        n.handleAfterData.push({
          name: t[a].name || "--",
          data: l[a]
        });
      }
    }, O = () => {
      var t, l, a, r, o;
      n.chartData = [], (l = (t = e.seriesSetting) == null ? void 0 : t.data) != null && l.length && ((a = e.data) != null && a.length) && !((r = e.options) != null && r.series) && (n.chartData = (o = e.seriesSetting) == null ? void 0 : o.data.map((c) => {
        const { name: s, dataKey: d } = c;
        return {
          name: s || "--",
          data: e.data.map((b) => b[d] || "")
        };
      }), G());
    }, P = () => {
      let t = "";
      return y.value.wrapper && (t += j(y.value.wrapper, `.${h}`)), y.value.emptyContainer && (t += j(y.value.emptyContainer, `.${h} .empty`)), y.value.emptyImg && (t += j(y.value.emptyImg, `.${h} .empty .empty-img`)), y.value.emptyText && (t += j(y.value.emptyText, `.${h} .empty .empty-text`)), t;
    }, L = () => {
      const t = document.querySelector(`.${h} .style-container`);
      t && (t.innerHTML = `<style>${P()}</style>`);
    }, q = E(() => {
      var t, l, a, r;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((c) => {
            var s;
            return (c == null ? void 0 : c.data) && ((s = c == null ? void 0 : c.data) == null ? void 0 : s.length) !== 0;
          });
        if (typeof ((l = e.options) == null ? void 0 : l.series) == "object")
          return ((r = (a = e.options.series) == null ? void 0 : a.data) == null ? void 0 : r.length) === 0;
      }
      return !!e.data.length;
    }), A = () => {
      var c, s, d, b, $, I, T, z;
      const t = (c = e.options) == null ? void 0 : c.color, l = (s = e.options) == null ? void 0 : s.series;
      let a = [], r = [];
      l ? a = l.map((i, u) => {
        var k, x, C;
        return m(
          {
            type: "bar",
            stack: "总量",
            z: 1e3 - u,
            barGap: "-100%",
            barWidth: 14,
            itemStyle: {
              color: ((k = i == null ? void 0 : i.itemStyle) == null ? void 0 : k.color) || t && t[u] || D.color[u] || "#fff",
              borderColor: ((x = i == null ? void 0 : i.itemStyle) == null ? void 0 : x.borderColor) || "#fff",
              // 边框颜色
              borderWidth: 2,
              borderRadius: ((C = i == null ? void 0 : i.itemStyle) == null ? void 0 : C.borderRadius) || 14
            },
            ...w(e == null ? void 0 : e.bar)
          },
          i
        );
      }) : (d = e.data) != null && d.length && (($ = (b = e.seriesSetting) == null ? void 0 : b.data) != null && $.length) && (a = (I = e.seriesSetting) == null ? void 0 : I.data.map((i, u) => {
        var x, C, R, B;
        const k = m({
          name: (x = n.chartData[u]) == null ? void 0 : x.name,
          type: "bar",
          stack: (C = n.chartData[u]) == null ? void 0 : C.name,
          z: 1e3 - u,
          barWidth: 14,
          barGap: "-100%",
          itemStyle: {
            color: ((R = i == null ? void 0 : i.itemStyle) == null ? void 0 : R.color) || t && t[u] || D.color[u] || "#fff",
            borderColor: "#fff",
            borderWidth: 2,
            borderRadius: 14,
            ...i == null ? void 0 : i.itemStyle
          },
          data: ((B = n.handleAfterData[u]) == null ? void 0 : B.data) || []
        }, e == null ? void 0 : e.bar);
        return m({}, k, {
          barWidth: e.barStyle.barWidth,
          itemStyle: {
            borderColor: e.barStyle.borderColor,
            borderWidth: e.barStyle.borderWidth,
            barBorderColor: e.barStyle.borderColor,
            barBorderWidth: e.barStyle.borderWidth,
            borderRadius: e.barStyle.borderRadius
          }
        });
      }), a.push({
        type: "bar",
        name: "",
        barGap: "-100%",
        stack: "",
        tooltip: {
          show: !1
        },
        data: _(),
        barWidth: a[0].barWidth,
        itemStyle: {
          color: "none",
          borderColor: e.barStyle.barBackBorderColor || "#b5bec3",
          borderWidth: e.barStyle.barBackBorderWidth || 1,
          borderRadius: e.barStyle.borderRadius || 14
        },
        z: 2e3,
        animation: !1
      }), (T = e.seriesSetting) != null && T.xAxisLabel ? r = (z = e.data) != null && z.length ? e.data.map((i) => {
        var u;
        return i[(u = e.seriesSetting) == null ? void 0 : u.xAxisLabel];
      }) : [] : r = e.data.map((i, u) => u));
      let o;
      e.mainAxis === "x" ? o = m(
        {
          series: a,
          xAxis: {
            data: r
          }
        },
        D
      ) : e.mainAxis === "y" && (o = m(
        {
          series: a,
          yAxis: {
            data: r
          }
        },
        D
      )), o = m(o, e.options), e.mainAxis === "y" && (o.xAxis.type = "value", o.yAxis.type = "category");
      try {
        f.setOption(o);
      } catch (i) {
        console.log(i);
      }
    }, N = () => {
      M(() => {
        f.resize();
      });
    };
    return W(
      () => [
        e.options,
        e.barStyle,
        e.bar,
        e.mainAxis
      ],
      () => {
        f.clear(), A();
      },
      {
        deep: !0
      }
    ), W(
      () => [
        e.seriesSetting.xAxisLabel
      ],
      () => {
        O(), v(n.chartData), A();
      },
      {
        deep: !0
      }
    ), W(
      () => [e.seriesSetting.data, e.data],
      () => {
        O(), v(n.chartData), f.clear(), A();
      },
      {
        deep: !0
      }
    ), W(
      () => [e.cStyle],
      () => {
        L(), M(() => {
          f.resize();
        });
      },
      {
        deep: !0
      }
    ), K(() => {
      L(), S.value && (f = X.init(S.value), O(), v(n.chartData), A(), f.on("click", (t) => {
        g("com-click", t);
      }), f.on("legendselectchanged", (t) => {
        var l;
        if (O(), !((l = e.options) != null && l.series)) {
          const { selected: a } = t, r = n.chartData;
          for (let o = 0; o < Object.keys(a).length; o++)
            r[o].data = a[Object.keys(a)[o]] ? n.chartData[o].data : r[o].data.fill(0);
          v(r);
        }
        A(), g("legend-select", t);
      })), window.addEventListener("resize", N);
    }), V(() => {
      window.removeEventListener("resize", N);
    }), {
      chartId: S,
      curStyle: y,
      checkEmpty: q,
      domOnlyClass: h
    };
  }
});
export {
  ne as default
};
