// @ths/design/es/components/chart-bar-capsule/src/index.vue.js
import n from "./index.vue2.js";
import { openBlock as c, createElementBlock as m, normalizeClass as p, normalizeStyle as o, withDirectives as a, createElementVNode as e, vShow as i } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  ref: "chartId",
  class: "chart"
}, y = { class: "empty" }, f = ["src"];
function h(t, s, v, E, k, C) {
  var r, l;
  return c(), m("div", {
    class: p(["t-chart-bar-capsule", t.domOnlyClass]),
    style: o({ ...(l = (r = t.curStyle) == null ? void 0 : r.wrapper) == null ? void 0 : l.default })
  }, [
    a(e("div", u, null, 512), [
      [i, t.checkEmpty]
    ]),
    a(e("div", y, [
      e("img", {
        class: "empty-img",
        style: o({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [i, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const w = /* @__PURE__ */ d(n, [["render", h]]);
export {
  w as default
};
