// @ths/design/es/components/t-area-selector-simple/src/index.vue2.js
import { defineComponent as $a, computed as da, reactive as ha, onMounted as ma, watch as c, toRefs as xa } from "vue";
import { getDomRandomClass as Sa, setStyle as Ia, getCssPropertyString as r } from "../../../utils/index.js";
const Ba = {
  REGION_CODE: "REGION_CODE",
  REGION_NAME: "REGION_NAME"
}, Ea = $a({
  name: "TTAreaSelectorSimple",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 选中项
    activeItem: {
      type: String,
      default: ""
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => Ba
    }
  },
  emits: ["search", "resetRegion"],
  setup(o, { emit: u }) {
    const e = da(() => o.cStyle), t = Sa("t-area-selector-simple"), l = ha({
      breadcrumbData: ["宝安区"],
      // 区域选择的显隐
      isShowBox: !1,
      // 当前选中项
      activeItem: {},
      // 数据
      curData: []
    }), ca = () => {
      l.isShowBox = !l.isShowBox;
    }, ua = () => {
      l.breadcrumbData = ["宝安区", l.activeItem.REGION_NAME], u("search", l.activeItem), l.isShowBox = !l.isShowBox;
    }, na = () => {
      l.activeItem = {}, l.breadcrumbData = ["宝安区"], u("resetRegion", l.activeItem), l.isShowBox = !l.isShowBox;
    }, va = () => {
      l.curData = o.data.map((a) => ({
        ...a,
        REGION_NAME: a[o.alias.REGION_NAME],
        REGION_CODE: a[o.alias.REGION_CODE]
      }));
    }, sa = (a) => {
      l.activeItem = a;
    }, n = () => {
      const a = l.curData.find((i) => i.REGION_CODE === o.activeItem);
      a ? (l.activeItem = a, l.breadcrumbData = ["宝安区", l.activeItem.REGION_NAME]) : (l.activeItem = {}, l.breadcrumbData = ["宝安区"]);
    }, fa = () => {
      let a = "";
      return e.value.wrapper && (a += Ia(e.value.wrapper, `.${t}`)), a;
    }, ba = () => {
      var i, s, f, b, $, d, h, m, x, S, I, B, _, C, E, R, y, O, w, N, g, D, p, G, A, T, M, H, j, k, q, L, P, z, F, J, K, Q, U, V, W, X, Y, Z, ee, ae, te, re, le, oe, ie, ce, ue, ne, ve, se, fe, be, $e, de, he, me, xe, Se, Ie, Be, _e, Ce, Ee, Re, ye, Oe, we, Ne, ge, De, pe, Ge, Ae, Te, Me, He, je, ke, qe, Le, Pe, ze, Fe, Je, Ke, Qe, Ue, Ve, We, Xe, Ye, Ze, ea, aa, ta, ra, la, oa, ia;
      let a = "";
      return (s = (i = e.value) == null ? void 0 : i.areaSelector) != null && s.default && (a += `.${t}.t-area-selector .area-selector {
          ${r((f = e.value) == null ? void 0 : f.areaSelector.default)}
        }`), ($ = (b = e.value) == null ? void 0 : b.areaSelector) != null && $.hover && (a += `.${t}.t-area-selector .area-selector:hover {
          ${r((d = e.value) == null ? void 0 : d.areaSelector.hover)}
        }`), (m = (h = e.value) == null ? void 0 : h.areaSelector) != null && m.active && (a += `.${t}.t-area-selector .area-selector.active {
          ${r((x = e.value) == null ? void 0 : x.areaSelector.active)}
        }`), (I = (S = e.value) == null ? void 0 : S.selector) != null && I.default && (a += `.${t}.t-area-selector .area-selector .el-button{
          ${r((B = e.value) == null ? void 0 : B.selector.default)}
        }`), (C = (_ = e.value) == null ? void 0 : _.selector) != null && C.hover && (a += `.${t}.t-area-selector .area-selector .el-button:hover {
          ${r((E = e.value) == null ? void 0 : E.selector.hover)}
        }`), (y = (R = e.value) == null ? void 0 : R.selector) != null && y.focus && (a += `.${t}.t-area-selector .area-selector .el-button:focus {
          ${r((O = e.value) == null ? void 0 : O.selector.focus)}
        }`), (N = (w = e.value) == null ? void 0 : w.selector) != null && N.active && (a += `.${t}.t-area-selector .area-selector .el-button:active {
          ${r((g = e.value) == null ? void 0 : g.selector.active)}
        }`), (p = (D = e.value) == null ? void 0 : D.lastRegion) != null && p.default && (a += `.${t}.t-area-selector .area-selector .area-breadcrumb .last-region .el-breadcrumb__inner{
          ${r((G = e.value) == null ? void 0 : G.lastRegion.default)}
        }`), (T = (A = e.value) == null ? void 0 : A.lastRegion) != null && T.hover && (a += `.${t}.t-area-selector .area-selector .area-breadcrumb .last-region .el-breadcrumb__inner:hover {
          ${r((M = e.value) == null ? void 0 : M.lastRegion.hover)}
        }`), (j = (H = e.value) == null ? void 0 : H.lastRegion) != null && j.active && (a += `.${t}.t-area-selector .area-selector .area-breadcrumb .last-region .el-breadcrumb__inner:active {
          ${r((k = e.value) == null ? void 0 : k.lastRegion.active)}
        }`), (L = (q = e.value) == null ? void 0 : q.areaBox) != null && L.default && (a += `.${t}.t-area-selector .area-box {
          ${r((P = e.value) == null ? void 0 : P.areaBox.default)}
        }`), (F = (z = e.value) == null ? void 0 : z.areaBox) != null && F.hover && (a += `.${t}.t-area-selector .area-box:hover {
          ${r((J = e.value) == null ? void 0 : J.areaBox.hover)}
        }`), (Q = (K = e.value) == null ? void 0 : K.areaBox) != null && Q.active && (a += `.${t}.t-area-selector .area-box.active {
          ${r((U = e.value) == null ? void 0 : U.areaBox.active)}
        }`), (W = (V = e.value) == null ? void 0 : V.areaTitle) != null && W.default && (a += `.${t}.t-area-selector .area-box .area-title{
          ${r((X = e.value) == null ? void 0 : X.areaTitle.default)}
        }`), (Z = (Y = e.value) == null ? void 0 : Y.areaTitle) != null && Z.hover && (a += `.${t}.t-area-selector .area-box .area-title:hover {
          ${r((ee = e.value) == null ? void 0 : ee.areaTitle.hover)}
        }`), (te = (ae = e.value) == null ? void 0 : ae.areaTitle) != null && te.active && (a += `.${t}.t-area-selector .area-box .area-title.active {
          ${r((re = e.value) == null ? void 0 : re.areaTitle.active)}
        }`), (oe = (le = e.value) == null ? void 0 : le.titleIcon) != null && oe.default && (a += `.${t}.t-area-selector .area-box .area-title .title-icon{
          ${r((ie = e.value) == null ? void 0 : ie.titleIcon.default)}
        }`), (ue = (ce = e.value) == null ? void 0 : ce.titleIcon) != null && ue.hover && (a += `.${t}.t-area-selector .area-box .area-title .title-icon:hover {
          ${r((ne = e.value) == null ? void 0 : ne.titleIcon.hover)}
        }`), (se = (ve = e.value) == null ? void 0 : ve.titleIcon) != null && se.active && (a += `.${t}.t-area-selector .area-box .area-title .title-icon.active {
          ${r((fe = e.value) == null ? void 0 : fe.titleIcon.active)}
        }`), ($e = (be = e.value) == null ? void 0 : be.titleClose) != null && $e.default && (a += `.${t}.t-area-selector .area-box .area-title .title-close{
          ${r((de = e.value) == null ? void 0 : de.titleClose.default)}
        }`), (me = (he = e.value) == null ? void 0 : he.titleClose) != null && me.hover && (a += `.${t}.t-area-selector .area-box .area-title .title-close:hover {
          ${r((xe = e.value) == null ? void 0 : xe.titleClose.hover)}
        }`), (Ie = (Se = e.value) == null ? void 0 : Se.titleClose) != null && Ie.active && (a += `.${t}.t-area-selector .area-box .area-title .title-close.active {
          ${r((Be = e.value) == null ? void 0 : Be.titleClose.active)}
        }`), (Ce = (_e = e.value) == null ? void 0 : _e.areaContainer) != null && Ce.default && (a += `.${t}.t-area-selector .area-box .area-container{
          ${r((Ee = e.value) == null ? void 0 : Ee.areaContainer.default)}
        }`), (ye = (Re = e.value) == null ? void 0 : Re.areaContainer) != null && ye.hover && (a += `.${t}.t-area-selector .area-box .area-container:hover {
          ${r((Oe = e.value) == null ? void 0 : Oe.areaContainer.hover)}
        }`), (Ne = (we = e.value) == null ? void 0 : we.areaContainer) != null && Ne.active && (a += `.${t}.t-area-selector .area-box .area-container.active {
          ${r((ge = e.value) == null ? void 0 : ge.areaContainer.active)}
        }`), (pe = (De = e.value) == null ? void 0 : De.areaButton) != null && pe.default && (a += `.${t}.t-area-selector .area-box .area-container .area-button{
          ${r((Ge = e.value) == null ? void 0 : Ge.areaButton.default)}
        }`), (Te = (Ae = e.value) == null ? void 0 : Ae.areaButton) != null && Te.hover && (a += `.${t}.t-area-selector .area-box .area-container .area-button:hover {
          ${r((Me = e.value) == null ? void 0 : Me.areaButton.hover)}
        }`), (je = (He = e.value) == null ? void 0 : He.areaButton) != null && je.active && (a += `.${t}.t-area-selector .area-box .area-container .area-button.active {
          ${r((ke = e.value) == null ? void 0 : ke.areaButton.active)}
        }`), (Le = (qe = e.value) == null ? void 0 : qe.areaSwitch) != null && Le.default && (a += `.${t}.t-area-selector .area-box .area-container .area-button .el-radio-button__inner{
          ${r((Pe = e.value) == null ? void 0 : Pe.areaSwitch.default)}
        }`), (Fe = (ze = e.value) == null ? void 0 : ze.areaSwitch) != null && Fe.hover && (a += `.${t}.t-area-selector .area-box .area-container .area-button .el-radio-button__inner:hover {
          ${r((Je = e.value) == null ? void 0 : Je.areaSwitch.hover)}
        }`), (Qe = (Ke = e.value) == null ? void 0 : Ke.areaSwitch) != null && Qe.active && (a += `.${t}  .area-box .area-container .area-button .el-radio-button__original-radio:checked + .el-radio-button__inner {
          ${r((Ue = e.value) == null ? void 0 : Ue.areaSwitch.active)}
        }`), (We = (Ve = e.value) == null ? void 0 : Ve.handleButton) != null && We.default && (a += `.${t}.t-area-selector .area-box .area-container .area-button .handle-button .el-button{
          ${r((Xe = e.value) == null ? void 0 : Xe.handleButton.default)}
        }`), (Ze = (Ye = e.value) == null ? void 0 : Ye.handleButton) != null && Ze.hover && (a += `.${t}.t-area-selector .area-box .area-container .area-button .handle-button .el-button:hover {
          ${r((ea = e.value) == null ? void 0 : ea.handleButton.hover)}
        }`), (ta = (aa = e.value) == null ? void 0 : aa.handleButton) != null && ta.focus && (a += `.${t}.t-area-selector .area-box .area-container .area-button .handle-button .el-button:focus {
          ${r((ra = e.value) == null ? void 0 : ra.handleButton.focus)}
        }`), (oa = (la = e.value) == null ? void 0 : la.handleButton) != null && oa.active && (a += `.${t}.t-area-selector .area-box .area-container .area-button .handle-button .el-button:active{
          ${r((ia = e.value) == null ? void 0 : ia.handleButton.active)}
        }`), a;
    }, v = () => {
      const a = document.querySelector(`.${t} .style-container`);
      a && (a.innerHTML = `<style>${fa()}${ba()}</style>`);
    };
    return ma(() => {
      v();
    }), c(() => o.cStyle, () => {
      v();
    }, { deep: !0 }), c(() => o.data, () => {
      va(), n();
    }, { deep: !0, immediate: !0 }), c(() => o.activeItem, () => {
      n();
    }, { deep: !0, immediate: !0 }), {
      ...xa(l),
      curStyle: e,
      domOnlyClass: t,
      switchSelector: ca,
      resetRegion: na,
      search: ua,
      selectItem: sa
    };
  }
});
export {
  Ea as default
};
