// @ths/design/es/components/timeline/src/index.vue.js
import P from "./index.vue2.js";
import { resolveComponent as u, openBlock as t, createElementBlock as d, normalizeClass as v, normalizeStyle as i, createElementVNode as o, createBlock as h, withCtx as C, createVNode as k, createCommentVNode as S, withDirectives as p, vModelText as T, Fragment as V, renderList as H, toDisplayString as R, vShow as B } from "vue";
import "./index.vue3.js";
import N from "../../../_virtual/_plugin-vue_export-helper.js";
const z = { class: "timeline" }, D = ["max"];
function E(e, r, F, L, M, O) {
  var y, b, g, $;
  const K = u("video-play"), m = u("el-icon"), w = u("video-pause");
  return t(), d("div", {
    class: v(["t-timeline", e.domOnlyClass]),
    style: i({ ...(b = (y = e.curStyle) == null ? void 0 : y.wrapper) == null ? void 0 : b.default })
  }, [
    o("div", {
      class: "play-btn",
      style: i({ ...($ = (g = e.curStyle) == null ? void 0 : g.playButton) == null ? void 0 : $.default }),
      onClick: r[0] || (r[0] = (...l) => e.changePlayStatus && e.changePlayStatus(...l))
    }, [
      !e.autoPlay && e.play ? (t(), h(m, { key: 0 }, {
        default: C(() => [
          k(K)
        ]),
        _: 1
      })) : S("", !0),
      e.autoPlay && e.play ? (t(), h(m, { key: 1 }, {
        default: C(() => [
          k(w)
        ]),
        _: 1
      })) : S("", !0)
    ], 4),
    o("div", z, [
      p(o("input", {
        "onUpdate:modelValue": r[1] || (r[1] = (l) => e.location = l),
        min: 0,
        max: e.data.length - 1,
        type: "range",
        class: "timeline__range",
        onChange: r[2] || (r[2] = (...l) => e.getVal && e.getVal(...l))
      }, null, 40, D), [
        [
          T,
          e.location,
          void 0,
          { number: !0 }
        ]
      ]),
      o("ul", {
        style: i({ height: `${e.lineHeight}px` }),
        class: "timeline__blocks"
      }, [
        (t(!0), d(V, null, H(e.data, (l, a) => {
          var n, s;
          return t(), d("li", {
            key: l[e.lineValueK],
            style: i({
              backgroundColor: l[e.lineColorK],
              borderRadius: a === 0 ? `${e.borderRadus}px 0 0 ${e.borderRadus}px` : a === e.data.length - 1 ? `0 ${e.borderRadus}px ${e.borderRadus}px 0` : "",
              borderTop: `${e.borderHeight}px solid ${e.borderColor}`,
              borderBottom: `${e.borderHeight}px solid ${e.borderColor}`,
              borderLeft: a === 0 ? `${e.borderHeight}px solid ${e.borderColor}` : "",
              borderRight: a === e.data.length - 1 ? `${e.borderHeight}px solid ${e.borderColor}` : ""
            }),
            class: "timeline__blocks__item"
          }, [
            p(o("div", {
              class: v(["time", { "time-hour": !1 }])
            }, [
              o("p", {
                style: i({ ...(s = (n = e.curStyle) == null ? void 0 : n.curdateWrapper) == null ? void 0 : s.default }),
                class: "time__block"
              }, R(e.getCurrentTime(e.dateFormat, l[e.lineValueK])), 5)
            ], 512), [
              [B, a === e.location]
            ])
          ], 4);
        }), 128))
      ], 4),
      o("ul", {
        style: i({ padding: `0 ${e.borderHeight}px` }),
        class: "timeline__date"
      }, [
        (t(!0), d(V, null, H(e.data, (l, a) => {
          var n, s, f;
          return t(), d("li", {
            key: l[e.lineValueK],
            class: "timeline__date__text"
          }, [
            p(o("span", {
              class: "timeline__datetext",
              style: i({ ...(s = (n = e.curStyle) == null ? void 0 : n.fontStyle) == null ? void 0 : s.default })
            }, R((f = l[e.lineValueK]) == null ? void 0 : f.slice(0, 7)), 5), [
              [B, a % (e.dataInterval + 1) === 0]
            ])
          ]);
        }), 128))
      ], 4)
    ]),
    r[3] || (r[3] = o("div", { class: "style-container" }, null, -1))
  ], 6);
}
const A = /* @__PURE__ */ N(P, [["render", E]]);
export {
  A as default
};
