// @ths/design/es/components/timeline/src/index.vue2.js
import { defineComponent as R, computed as v, reactive as E, watch as b, onMounted as M, toRefs as I } from "vue";
import { VideoPlay as T, VideoPause as V } from "@element-plus/icons";
import { getDomRandomClass as w, setStyle as D } from "../../../utils/index.js";
const L = R({
  name: "TTimeline",
  components: {
    VideoPlay: T,
    VideoPause: V
  },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 当前日期容器
        curdateWrapper: {
          default: {
            width: "92px"
          }
        },
        // 字体样式
        fontStyle: {
          default: {}
        },
        // 播放按钮样式
        playButton: {
          default: {
            fontSize: "40px"
          }
        }
      })
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 当前日期
    currentDate: {
      type: String,
      default: "2018-08-01 13:34:23"
    },
    // 数据间隔
    dataInterval: {
      type: Number,
      default: 1
    },
    // 显示播放
    play: {
      type: Boolean,
      default: !0
    },
    // 播放速度
    playSpeed: {
      type: Number,
      default: 1e3
    },
    // 自动播放
    autoplay: {
      type: Boolean,
      default: !1
    },
    // 循环播放
    loop: {
      type: Boolean,
      default: !1
    },
    // 日期格式
    dateFormat: {
      type: String,
      default: "yyyy-MM-dd HH:mm:ss"
    },
    // 时间轴数值Key
    lineValueK: {
      type: String,
      default: "linevalue"
    },
    // 时间轴颜色Key
    lineColorK: {
      type: String,
      default: "linecolor"
    },
    // 时间轴高度
    lineHeight: {
      type: Number,
      default: 20
    },
    // 分割线颜色
    splitlineColor: {
      type: String,
      default: ""
    },
    // 日期文字分割线颜色
    textSplitlineColor: {
      type: String,
      default: ""
    },
    // 边框高度
    borderHeight: {
      type: Number,
      default: 3
    },
    // 边框颜色
    borderColor: {
      type: String,
      default: "#4b83d8"
    },
    // 边框圆角
    borderRadus: {
      type: Number,
      default: 20
    }
  },
  emits: ["date-change"],
  setup(e, { emit: x }) {
    const _ = v(() => e.cStyle), i = w("timeline");
    let S = null;
    const a = E({
      location: 0,
      // 滑块当前位置
      autoPlay: e.autoplay
      // 自动播放
    }), $ = () => {
      x("date-change", a.location);
    }, g = (l, t) => {
      S = setInterval(() => {
        a.location++, a.location %= e.data.length, a.location === 0 && !t && (clearInterval(S), a.autoPlay = !1);
      }, l);
    }, h = () => {
      clearInterval(S);
    }, N = () => {
      a.autoPlay = !a.autoPlay, a.autoPlay ? g(e.playSpeed, e.loop) : h();
    }, P = (l, t) => {
      if (!t)
        return "";
      const [o, n] = t == null ? void 0 : t.split(" "), [c, d, y] = o == null ? void 0 : o.split("-"), [f, m, s] = n == null ? void 0 : n.split(":");
      let r = l;
      const u = {
        "[Mm]+": Number(d),
        // 月份
        "[Dd]+": Number(y),
        // 日
        "[Hh]+": Number(f),
        // 小时
        "[m]+": Number(m),
        // 分
        "[Ss]+": Number(s),
        // 秒
        "[Qq]+": Math.floor((Number(d) + 3) / 3)
        // 季度
      };
      return /([Yy]+)/.test(r) && (r = r.replace(RegExp.$1, `${c}`.substr(4 - RegExp.$1.length))), Object.keys(u).forEach((p) => {
        new RegExp(`(${p})`).test(r) && (r = r.replace(RegExp.$1, RegExp.$1.length === 1 ? u[p] : `00${u[p]}`.substr(`${u[p]}`.length)));
      }), r;
    }, H = (l, t) => {
      var n;
      let o = "";
      return (n = _.value) != null && n.wrapper && (o += D(l, t)), o;
    }, C = () => {
      var c, d, y, f, m, s;
      let l = "";
      const t = document.querySelector(`.${i} .style-container`), o = [
        `.${i} .timeline__blocks__item:not(:first-child)::before`,
        `.${i} .timeline__date__text:not(:first-child)::before`,
        `.${i} .timeline__blocks__item .time__block::before`,
        `.${i} .timeline__blocks__item .time::before`
      ], n = [
        {
          default: { borderColor: e.splitlineColor, backgroundColor: e.splitlineColor }
        },
        {
          default: { borderColor: e.textSplitlineColor }
        },
        {
          default: { backgroundColor: (y = (d = (c = e.cStyle) == null ? void 0 : c.curdateWrapper) == null ? void 0 : d.default) == null ? void 0 : y.color }
        },
        {
          default: { borderColor: (s = (m = (f = e.cStyle) == null ? void 0 : f.curdateWrapper) == null ? void 0 : m.default) == null ? void 0 : s.color }
        }
      ];
      o.forEach((r, u) => {
        t && (l += H(n[u], r));
      }), t && (t.innerHTML = `<style>${l}</style>`);
    };
    return b(() => [e.splitlineColor, e.textSplitlineColor], () => {
      C();
    }), b(() => [e.playSpeed, e.autoplay, e.loop], (l) => {
      const [t, o, n] = l;
      a.autoPlay = o, h(), o && g(t, n);
    }), b(() => e.currentDate, (l) => {
      a.location = e.data.findIndex((t) => t[e.lineValueK] === l);
    }), b(() => e.cStyle, () => {
      C();
    }, { deep: !0 }), M(() => {
      C(), e.autoplay && g(e.playSpeed, e.loop), a.location = e.data.findIndex((l) => l[e.lineValueK] === e.currentDate);
    }), {
      ...I(a),
      curStyle: _,
      getVal: $,
      domOnlyClass: i,
      changePlayStatus: N,
      getCurrentTime: P
    };
  }
});
export {
  L as default
};
