// @ths/design/es/components/line/src/index.vue2.js
import { defineComponent as i, computed as l, watch as p, onMounted as u } from "vue";
import { getDomRandomClass as m, getCssPropertyString as y, setStyle as v } from "../../../utils/index.js";
const d = i({
  name: "TLine",
  props: {
    // 选中状态
    active: {
      type: Boolean,
      default: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {},
          active: {},
          hover: {}
        }
      })
    }
  },
  setup(r) {
    const o = l(() => r.active), t = l(() => r.cStyle), n = m("line"), c = () => {
      let e = "";
      return t.value.wrapper && t.value.wrapper.active && (e += `
          .${n}.active {
            ${y(t.value.wrapper.active)}
          }
        `), e;
    }, s = () => {
      let e = "";
      return t.value.wrapper && (e += v(t.value.wrapper, `.${n}`)), e;
    }, a = () => {
      const e = document.querySelector(`.${n} .style-container`);
      e && (e.innerHTML = `<style>${s()}${c()}</style>`);
    };
    return p(() => r.cStyle, () => {
      a();
    }), u(() => {
      a();
    }), {
      isActive: o,
      domOnlyClass: n
    };
  }
});
export {
  d as default
};
