// @ths/design/es/components/chart-word-cloud/src/index.vue.js
import a from "./index.vue2.js";
import { openBlock as c, createElementBlock as m, normalizeClass as d, normalizeStyle as l, withDirectives as i, createElementVNode as e, vShow as n } from "vue";
import "./index.vue3.js";
import p from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  ref: "chart",
  class: "word-cloud"
}, y = { class: "empty" }, f = ["src"];
function h(t, o, v, E, w, k) {
  var s, r;
  return c(), m("div", {
    class: d(["t-chart-word-cloud", t.domOnlyClass]),
    style: l({ ...(r = (s = t.curStyle) == null ? void 0 : s.wrapper) == null ? void 0 : r.default })
  }, [
    i(e("div", u, null, 512), [
      [n, t.checkEmpty]
    ]),
    i(e("div", y, [
      e("img", {
        class: "empty-img",
        style: l({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, f),
      o[0] || (o[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [n, !t.checkEmpty]
    ]),
    o[1] || (o[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const S = /* @__PURE__ */ p(a, [["render", h]]);
export {
  S as default
};
