// @ths/design/es/components/chart-word-cloud/src/index.vue2.js
import { defineComponent as z, computed as M, ref as I, reactive as $, nextTick as g, onMounted as T, onUnmounted as k, watch as w } from "vue";
import * as A from "echarts";
import "echarts-wordcloud";
import m from "lodash/merge";
import E from "lodash/cloneDeep";
import { DEFAULT_OPTIONS as R } from "../../../utils/constant.js";
import { getDomRandomClass as W, setStyle as y } from "../../../utils/index.js";
const S = m(
  E(R),
  {
    tooltip: {
      backgroundColor: "rgba(33, 41, 71, .8)",
      borderColor: "rgba(33, 41, 71, .8)",
      textStyle: {
        color: "#fff",
        fontWeight: "normal"
      },
      trigger: "item"
    },
    legend: {
      show: !1
    },
    grid: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    }
  }
), H = z({
  name: "TChartWordCloud",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => []
    },
    // 别名 及 样式设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 词云文本样式
        wordStyle: {
          type: Object,
          default: () => ({})
        },
        // 取resData数据中的别名设置
        data: {
          nameKey: "",
          // 名称别名
          dataKey: ""
          // 数据别名
        }
      })
    },
    wordCloud: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["com-click"],
  setup(t, { emit: O }) {
    const c = W("chart-word-cloud"), a = M(() => t.cStyle), s = I(null);
    let u = null;
    const C = $({
      // 图表数据
      chartData: []
    }), v = () => {
      var e;
      if ((e = t.seriesSetting) != null && e.data && t.data) {
        const { nameKey: d, dataKey: i } = t.seriesSetting.data;
        C.chartData = t.data.map((n) => ({
          name: n[d] || "--",
          value: n[i] || 0
        }));
      }
    }, j = () => {
      let e = "";
      return a.value.wrapper && (e += y(a.value.wrapper, `.${c}`)), a.value.emptyContainer && (e += y(a.value.emptyContainer, `.${c} .empty`)), a.value.emptyImg && (e += y(a.value.emptyImg, `.${c} .empty .empty-img`)), a.value.emptyText && (e += y(a.value.emptyText, `.${c} .empty .empty-text`)), e;
    }, l = () => {
      g(() => {
        s.value && (s.value.getBoundingClientRect().width ?? 0) && g(() => {
          u.resize();
        });
      });
    }, b = () => {
      const e = document.querySelector(`.${c} .style-container`);
      e && (e.innerHTML = `<style>${j()}</style>`);
    }, D = M(() => {
      var e, d, i, n;
      if (t.options) {
        if (Array.isArray((e = t.options) == null ? void 0 : e.series))
          return t.options.series.some((o) => {
            var r;
            return (o == null ? void 0 : o.data) && ((r = o == null ? void 0 : o.data) == null ? void 0 : r.length) !== 0;
          });
        if (typeof ((d = t.options) == null ? void 0 : d.series) == "object")
          return ((n = (i = t.options.series) == null ? void 0 : i.data) == null ? void 0 : n.length) === 0;
      }
      return t.data.length ? (g(() => {
        l();
      }), !0) : !1;
    }), f = () => {
      var h, o;
      const { color: e, series: d } = t.options;
      let i;
      d ? i = d.map((r) => m(
        {
          type: "wordCloud",
          gridSize: 12,
          // 单词之间的间隔大小
          // 最小字体和最大字体 必须写一个范围不能直接写每个字的大小
          sizeRange: [12, 30],
          rotationRange: [-45, 0, 45, 90],
          // 字体旋转角度的范围，水平
          shape: "circle",
          // 形状
          width: "80%",
          height: "100%",
          left: "center",
          top: "center",
          right: null,
          bottom: null,
          drawOutOfBound: !1,
          // 词云显示完整，超出画布的也显示
          textStyle: {
            fontWeight: "500",
            // 字体随机颜色
            color: (x) => e && e[x.dataIndex] || S.color[x.dataIndex] || `rgb(${[Math.round(Math.random() * 255), Math.round(Math.random() * 255), Math.round(Math.random() * 255)].join(",")})`
          },
          ...t == null ? void 0 : t.wordCloud
        },
        r
      )) : t.data && ((h = t.seriesSetting) != null && h.data) && (i = m(
        {
          type: "wordCloud",
          gridSize: 12,
          // 单词之间的间隔大小
          // 最小字体和最大字体 必须写一个范围不能直接写每个字的大小
          sizeRange: [12, 30],
          rotationRange: [-45, 0, 45, 90],
          // 字体旋转角度的范围，水平
          shape: "circle",
          // 形状
          width: "80%",
          height: "100%",
          left: "center",
          top: "center",
          right: null,
          bottom: null,
          drawOutOfBound: !1,
          // 字是否叠加
          textStyle: {
            fontWeight: "normal",
            // 字体随即色
            color: (r) => e && e[r.dataIndex] || S.color[r.dataIndex] || `rgb(${[Math.round(Math.random() * 255), Math.round(Math.random() * 255), Math.round(Math.random() * 255)].join(",")})`
          },
          data: C.chartData
        },
        t == null ? void 0 : t.wordCloud,
        (o = t == null ? void 0 : t.seriesSetting) == null ? void 0 : o.wordStyle
      ));
      let n = m(
        {
          series: i
        },
        S
      );
      n = m(n, t.options), u.clear();
      try {
        u.setOption(n);
      } catch (r) {
        console.log(r);
      }
    };
    return T(() => {
      b(), s.value && (u = A.init(s.value), v(), f(), u.on("click", (e) => {
        O("com-click", e);
      })), l(), window.addEventListener("resize", l);
    }), k(() => {
      window.removeEventListener("resize", l);
    }), w(
      () => [t.options, t.seriesSetting.wordStyle, t.wordCloud],
      () => {
        f(), l();
      },
      {
        deep: !0
      }
    ), w(
      () => [t.data, t.seriesSetting.data],
      () => {
        v(), f(), l();
      },
      {
        deep: !0
      }
    ), w(
      () => [t.cStyle],
      () => {
        b(), l();
      },
      {
        deep: !0
      }
    ), {
      curStyle: a,
      chart: s,
      checkEmpty: D,
      domOnlyClass: c
    };
  }
});
export {
  H as default
};
