// @ths/design/es/components/chart-curve-card/src/index.vue2.js
import { defineComponent as Q, computed as R, ref as Y, reactive as ee, onMounted as te, onUnmounted as ae, watch as D, nextTick as K, toRefs as le } from "vue";
import * as A from "echarts";
import u from "lodash/merge";
import L from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as oe, DEFAULT_OPTIONS_AXIS as re } from "../../../utils/constant.js";
import { getDomRandomClass as ne, setStyle as S } from "../../../utils/index.js";
const se = u(L(oe), L(re)), s = u(
  L(se),
  {
    // 气泡框
    tooltip: {
      trigger: "axis",
      padding: [10, 10],
      backgroundColor: "rgba(0, 0, 0, .5)",
      borderColor: "transparent",
      textStyle: {
        color: "#ffffff"
      },
      axisPointer: {
        type: "line",
        lineStyle: {
          color: {
            type: "linear",
            x: 0,
            y: 0,
            x2: 0,
            y2: 2,
            colorStops: [{
              offset: 0,
              color: "rgba(126,199,255,0)"
              // 0% 处的颜色
            }, {
              offset: 0.5,
              color: "rgba(126,199,255,1)"
              // 100% 处的颜色
            }, {
              offset: 1,
              color: "rgba(126,199,255,0)"
              // 100% 处的颜色
            }],
            global: !1
            // 缺省为 false
          }
        }
      }
    },
    // 网格
    grid: {
      top: "25%",
      left: "8%",
      bottom: "15%",
      right: "10%"
    },
    // 图例
    legend: {
      show: !1
    },
    // x轴
    xAxis: {
      axisLabel: {
        // 坐标轴刻度标签的相关设置
        padding: [0, 0],
        fontSize: 14
      },
      splitLine: {
        show: !1
      }
    },
    // y轴
    yAxis: {
      name: "(个)",
      nameTextStyle: {
        fontSize: 14,
        padding: [0, 50, 10, 0]
      },
      min: 0,
      splitNumber: 3,
      splitLine: {
        show: !0
      },
      axisLabel: {
        show: !0,
        fontSize: 14,
        padding: [0, 8, 0, 0]
      },
      axisTick: {
        show: !1
      }
    },
    dataZoom: [
      {
        // 禁用缩放
        disabled: !1,
        type: "inside"
      },
      {
        // 隐藏滑块
        show: !1,
        type: "slider"
      }
    ]
  }
), ie = {
  listText: {},
  // 文字项
  listValue: {},
  // 值项
  listShowHeight: {
    showNumber: 4,
    // 展示的列数
    height: 45
    // 每列的高度
  }
}, he = Q({
  name: "TChartCurveCard",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        },
        // 图表容器
        chartContainer: {
          default: {}
        },
        // 表容器
        tableContainer: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => []
    },
    // 包含 别名设置、系列样式
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 表格名称及数据别名设置
    tableData: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 卡片展示的表头
    listName: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 列表样式
    listStyle: {
      type: Object,
      default: () => ie
    },
    // 折线主题设置
    line: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["legend-select"],
  setup(e, { emit: V }) {
    const h = ne("chart-curve-card"), f = R(() => e.cStyle), b = Y(null);
    let d = null, w = null;
    const g = ee({
      // 图表数据
      chartData: [],
      // 列表表头数据
      listName: [],
      // 消耗量列表数据
      consumptionData: []
    }), C = (o = 0) => {
      var i, r;
      let a = {};
      (i = e.tableData) != null && i.length && ((r = e.data) != null && r.length) && (a = e.data[o], g.consumptionData = e.tableData.map((n) => ({
        name: n.label || "-",
        value: a[n.valueAlias]
      })));
    }, O = () => {
      var o, a, i, r;
      (a = (o = e.seriesSetting) == null ? void 0 : o.data) != null && a.length && ((i = e.data) != null && i.length) && (g.chartData = e.seriesSetting.data.map((n) => {
        const { name: c, dataKey: y } = n;
        return {
          name: c || "--",
          data: e.data.map((m) => m[y])
        };
      }), C()), (r = e.listName) != null && r.length && (g.listName = e.listName);
    }, X = () => {
      let o = "";
      return f.value.wrapper && (o += S(f.value.wrapper, `.${h}`)), f.value.emptyContainer && (o += S(f.value.emptyContainer, `.${h} .empty`)), f.value.emptyImg && (o += S(f.value.emptyImg, `.${h} .empty .empty-img`)), f.value.emptyText && (o += S(f.value.emptyText, `.${h} .empty .empty-text`)), o;
    }, T = () => {
      const o = document.querySelector(`.${h} .style-container`);
      o && (o.innerHTML = `<style>${X()}</style>`);
    }, Z = R(() => {
      var o, a, i, r;
      if (e.options) {
        if (Array.isArray((o = e.options) == null ? void 0 : o.series))
          return e.options.series.some((c) => {
            var y;
            return (c == null ? void 0 : c.data) && ((y = c == null ? void 0 : c.data) == null ? void 0 : y.length) !== 0;
          });
        if (typeof ((a = e.options) == null ? void 0 : a.series) == "object")
          return ((r = (i = e.options.series) == null ? void 0 : i.data) == null ? void 0 : r.length) === 0;
      }
      return !!e.data.length;
    }), v = () => {
      var c, y, m, p, N, z, x, j;
      d.clear();
      let o = {};
      const { color: a, series: i } = e.options;
      let r = [];
      i ? r = i.map((t, l) => u(
        {
          name: "平均能量消费量",
          type: "line",
          symbol: "circle",
          // 默认是空心圆（中间是白色的），改成实心圆
          showAllSymbol: !0,
          symbolSize: 6,
          smooth: !0,
          lineStyle: {
            width: 3,
            color: a && a[l] || s.color[l] || "#fff"
            // 线条颜色
          },
          itemStyle: {
            color: a && a[l] || s.color[l] || "#fff",
            borderWidth: 3,
            borderColor: a && a[l] || s.color[l] || "#fff",
            shadowColor: a && a[l] || s.color[l] || "#fff",
            shadowBlur: 10
          },
          tooltip: {
            show: !0
          },
          areaStyle: {
            // 区域填充样式
            // 线性渐变，前4个参数分别是x0,y0,x2,y2(范围0~1);相当于图形包围盒中的百分比。如果最后一个参数是‘true’，则该四个值是绝对像素位置。
            color: new A.graphic.LinearGradient(0, 0, 0, 1, [
              {
                offset: 0,
                color: a && a[l] || s.color[l] || "#fff"
              },
              {
                offset: 1,
                color: "transparent"
              }
            ], !1),
            opacity: 0.4,
            shadowColor: a && a[l] || s.color[l] || "#fff",
            // 阴影颜色
            shadowBlur: 20
            // shadowBlur设图形阴影的模糊大小。配合shadowColor,shadowOffsetX/Y, 设置图形的阴影效果。
          },
          ...e == null ? void 0 : e.line
        },
        t
      )) : (y = (c = e.seriesSetting) == null ? void 0 : c.data) != null && y.length && ((m = e.data) != null && m.length) && (r = (p = e.seriesSetting) == null ? void 0 : p.data.map((t, l) => {
        var E, _, $, k, q, M, P, B, F, H, U, W, G;
        const J = u(
          {
            name: (E = g.chartData[l]) == null ? void 0 : E.name,
            type: "line",
            symbol: "circle",
            // 默认是空心圆（中间是白色的），改成实心圆
            showAllSymbol: !0,
            symbolSize: 6,
            smooth: !0,
            lineStyle: {
              width: 3,
              color: a && a[l] || s.color[l] || "#fff"
              // 线条颜色
            },
            itemStyle: {
              color: a && a[l] || s.color[l] || "#fff",
              borderWidth: 3,
              borderColor: a && a[l] || s.color[l] || "#fff",
              shadowColor: a && a[l] || s.color[l] || "#fff",
              shadowBlur: 10
            },
            tooltip: {
              show: !0
            },
            areaStyle: {
              // 区域填充样式
              // 线性渐变，前4个参数分别是x0,y0,x2,y2(范围0~1);相当于图形包围盒中的百分比。如果最后一个参数是‘true’，则该四个值是绝对像素位置。
              color: new A.graphic.LinearGradient(0, 0, 0, 1, [
                {
                  offset: 0,
                  color: ((_ = t == null ? void 0 : t.itemStyle) == null ? void 0 : _.color) || a && a[l] || s.color[l] || "#fff"
                },
                {
                  offset: 1,
                  color: (($ = t == null ? void 0 : t.itemStyle) == null ? void 0 : $.endLinearColor) || "transparent"
                }
              ], !1),
              opacity: 0.4
            },
            data: ((k = g.chartData[l]) == null ? void 0 : k.data) || []
          },
          e == null ? void 0 : e.line,
          (P = (M = (q = e.seriesSetting) == null ? void 0 : q.data) == null ? void 0 : M[l]) == null ? void 0 : P.style
        );
        return u({}, J, {
          lineStyle: {
            width: (B = t == null ? void 0 : t.itemStyle) == null ? void 0 : B.width,
            color: (F = t == null ? void 0 : t.itemStyle) == null ? void 0 : F.color
            // 线条颜色
          },
          itemStyle: {
            color: (H = t == null ? void 0 : t.itemStyle) == null ? void 0 : H.color,
            borderWidth: (U = t == null ? void 0 : t.itemStyle) == null ? void 0 : U.width,
            borderColor: (W = t == null ? void 0 : t.itemStyle) == null ? void 0 : W.color,
            shadowColor: (G = t == null ? void 0 : t.itemStyle) == null ? void 0 : G.color
          }
        });
      }), (N = e.seriesSetting) != null && N.xAxisLabel ? o = (z = e.data) != null && z.length ? e.data.map((t) => t[e.seriesSetting.xAxisLabel]) : [] : o = e.data.map((t, l) => l));
      let n = u(
        {
          series: r,
          xAxis: {
            data: o
          }
        },
        s
      );
      n = u(n, e.options);
      try {
        d.setOption(n);
      } catch (t) {
        console.log(t);
      }
      if (clearInterval(w), (x = n.series[0]) != null && x.data && ((j = n.series[0]) != null && j.data.length)) {
        let t = 0;
        w = setInterval(() => {
          var l;
          d.dispatchAction({
            type: "showTip",
            seriesIndex: 0,
            dataIndex: t
          }), t++, t > ((l = n.series[0]) == null ? void 0 : l.data.length) && (t = 0), C(t - 1 < 0 ? 0 : t - 1);
        }, 1e3);
      }
    }, I = () => {
      K(() => {
        d.resize();
      });
    };
    return te(() => {
      T(), b.value && (d = A.init(b.value), O(), v(), d.on("legendselectchanged", (o) => {
        V("legend-select", o);
      }), d.on("click", (o) => {
        if (o.componentType === "series") {
          const { dataIndex: a } = o;
          clearInterval(w), C(a);
        }
      })), window.addEventListener("resize", I);
    }), ae(() => {
      window.removeEventListener("resize", I);
    }), D(
      [
        () => e.listStyle,
        () => e.line
      ],
      (o) => {
        v();
      },
      {
        deep: !0
      }
    ), D(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting,
        () => e.listName,
        () => e.tableData
      ],
      () => {
        O(), v();
      },
      {
        deep: !0
      }
    ), D(
      [
        () => e.cStyle,
        () => e.tableData
      ],
      () => {
        T(), K(() => {
          d.resize();
        });
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      ...le(g),
      chartId: b,
      curStyle: f,
      checkEmpty: Z,
      domOnlyClass: h
    };
  }
});
export {
  he as default
};
