// @ths/design/es/components/card-pollution-base/src/index.vue2.js
import { defineComponent as m, computed as s, reactive as y, onMounted as p, watch as K, toRefs as b } from "vue";
import g from "lodash/merge";
import v from "lodash/cloneDeep";
import { getPollutionHtml as A } from "../../../utils/index.js";
import { POLLUTION_HTML as C } from "../../../utils/constant.js";
const n = {
  nameKey: "name",
  valueKey: "value",
  colorKey: "borderColor",
  unitKey: "unit"
}, w = m({
  name: "TCardPollutionBase",
  props: {
    // 标题和值排列方向 x 在一行排列 y 在不同行排列
    direction: {
      type: String,
      default: "x"
    },
    // 横向间隔
    spacingX: {
      type: Number,
      default: 17
    },
    // 纵向间隔
    spacingY: {
      type: Number,
      default: 12
    },
    // 列数
    col: {
      type: Number,
      default: 3
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 自定义属性
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        },
        // 单项容器
        wrapperContent: {
          default: {}
        },
        // 污染物名称
        name: {
          default: {}
        },
        // 污染物含量容器
        levelsContent: {
          default: {}
        },
        // 单位
        unit: {
          default: {}
        },
        // 值
        value: {
          default: {}
        }
      })
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => n
    }
  },
  setup(e) {
    const u = s(() => e.cStyle), a = y({
      listData: []
      // 当前数据
    }), l = () => {
      var o;
      if (a.listData = [], (o = e.data) != null && o.length && e.dataAlias) {
        const r = g(v(n), e.dataAlias), {
          nameKey: d,
          valueKey: i,
          colorKey: c,
          unitKey: f
        } = r;
        a.listData = e.data.map((t) => ({
          name: A(C, t[d]),
          value: t[i],
          unit: t[f],
          borderColor: t[c] || "#FFF"
        }));
      }
    };
    return p(() => {
      l();
    }), K(() => [e.data, e.dataAlias], () => {
      l();
    }, {
      deep: !0,
      immediate: !0
    }), {
      ...b(a),
      curStyle: u
    };
  }
});
export {
  w as default
};
