// @ths/design/es/components/chart-line-simple/src/index.vue2.js
import { defineComponent as q, computed as _, ref as M, reactive as K, nextTick as C, onMounted as N, onUnmounted as P, watch as D, toRefs as U } from "vue";
import * as H from "echarts";
import s from "lodash/merge";
import h from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as R, DEFAULT_OPTIONS_AXIS as X } from "../../../utils/constant.js";
import { getDomRandomClass as B, setStyle as p } from "../../../utils/index.js";
const G = s(h(R), h(X)), r = s(
  h(G),
  {
    grid: {
      top: "13%",
      bottom: "10%",
      left: "6%",
      right: "4%"
    }
  }
), Z = q({
  name: "TChartLineSimple",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 暂无数据容器
        emptyContainer: {
          default: {}
        },
        // 暂无数据图片
        emptyImg: {
          default: {}
        },
        // 暂无数据文字
        emptyText: {
          default: {}
        }
      })
    },
    // 设置主轴
    mainAxis: {
      type: String,
      default: "x"
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => r,
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    line: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: L }) {
    const f = _(() => e.cStyle), d = B("chart-line-simple"), O = M(null);
    let m = null;
    const o = K({
      // 图表数据
      chartData: [],
      lineOptions: h(e.line)
    });
    function w() {
      var t, u;
      (u = (t = e.seriesSetting) == null ? void 0 : t.data) != null && u.length && e.data ? o.chartData = e.seriesSetting.data.map((c) => {
        const { name: y, dataKey: n } = c;
        return {
          name: y || "--",
          data: e.data.map((i) => i[n])
        };
      }) : o.chartData = [];
    }
    function b() {
      var i, g, E, I, j;
      const { color: t, series: u } = e.options;
      let c = [], y = {};
      u ? c = u.map((l, a) => {
        var A, S, F;
        const v = s(
          {
            type: "line",
            itemStyle: {
              color: t && t[a] || r.color[a] || "#FFF"
            },
            lineStyle: {
              color: t && t[a] || r.color[a] || "#FFF"
            },
            emphasis: {
              focus: "series"
            },
            smooth: !0,
            areaStyle: {
              color: {
                type: "linear",
                x: 1,
                y: 1.2,
                r: 0,
                colorStops: [
                  {
                    offset: 0,
                    color: "#FFFFFF00"
                    // 0% 处的颜色
                  },
                  {
                    offset: 1,
                    color: t && t[a] || r.color[a] || "#FFF"
                    // 100% 处的颜色
                  }
                ]
              }
            },
            ...o == null ? void 0 : o.lineOptions
          },
          (F = (S = (A = e.seriesSetting) == null ? void 0 : A.data) == null ? void 0 : S[a]) == null ? void 0 : F.style
        );
        return s({}, v, l);
      }) : e.data && ((g = (i = e.seriesSetting) == null ? void 0 : i.data) != null && g.length) && ((E = e.seriesSetting) == null || E.data.forEach((l, a) => {
        var S, F;
        const v = s(
          {
            name: (S = o.chartData[a]) == null ? void 0 : S.name,
            type: "line",
            itemStyle: {
              color: t && t[a] || r.color[a] || "#FFF"
            },
            lineStyle: {
              color: t && t[a] || r.color[a] || "#FFF"
            },
            emphasis: {
              focus: "series"
            },
            smooth: !1,
            areaStyle: {
              color: {
                type: "linear",
                x: 1,
                y: 1.2,
                r: 0,
                colorStops: [
                  {
                    offset: 0,
                    color: "rgba(255,255,255,0)"
                    // 0% 处的颜色
                  },
                  {
                    offset: 1,
                    color: t && t[a] || r.color[a] || "#FFFFFF00"
                    // 100% 处的颜色
                  }
                ]
              }
            },
            data: e.data.length ? (F = o.chartData[a]) == null ? void 0 : F.data : [],
            ...o == null ? void 0 : o.lineOptions
          },
          l == null ? void 0 : l.style
        ), A = s({}, v, l);
        c.push(A), e.data.length > 1 && s({
          type: "lines",
          coordinateSystem: "cartesian2d",
          zlevel: 1,
          polyline: !0,
          smooth: !0,
          symbol: "circle",
          data: [
            {
              coords: e.data.map((x) => e.mainAxis === "x" ? [x[e.seriesSetting.xAxisLabel], x[l.dataKey]] : [x[l.dataKey], x[e.seriesSetting.xAxisLabel]])
            }
          ],
          effect: {
            show: !1,
            // 拖尾长度0-1
            trailLength: 0.4,
            symbol: "circle",
            // 配置特效图形的移动动画是否是固定速度，
            // 单位像素/秒，设置为大于 0 的值后会忽略 period 配置项。
            // constantSpeed: 280,
            period: 8,
            symbolSize: 4
          },
          lineStyle: {
            // color: (color && color[index]) || defaultOptions.color[index] || '#FFF',
            color: {
              type: "linear",
              x: 1,
              y: 1.2,
              r: 0,
              colorStops: [
                {
                  offset: 0,
                  color: "rgba(255,255,255,0)"
                  // 0% 处的颜色
                },
                {
                  offset: 1,
                  color: t && t[a] || r.color[a] || "#FFFFFF00"
                  // 100% 处的颜色
                }
              ]
            },
            width: 0,
            cap: "round",
            opacity: 0,
            curveness: 0
          }
        }, l == null ? void 0 : l.linesStyle);
      }), (I = e.seriesSetting) != null && I.xAxisLabel ? y = (j = e.data) != null && j.length ? e.data.map((l) => l[e.seriesSetting.xAxisLabel]) : [] : y = e.data.map((l, a) => a));
      let n;
      e.mainAxis === "x" ? n = s(
        {
          series: c,
          xAxis: {
            data: y
          }
        },
        r
      ) : e.mainAxis === "y" && (n = s(
        {
          series: c,
          yAxis: {
            data: y
          }
        },
        r
      )), n = s(n, e.options), e.mainAxis === "y" && (n.xAxis.type = "value", n.yAxis.type = "category"), m.clear();
      try {
        console.log(n, "简单折线图"), m.setOption(n);
      } catch (l) {
        console.log(l);
      }
    }
    function T() {
      C(() => {
        m.resize();
      });
    }
    const $ = () => {
      let t = "";
      return f.value.wrapper && (t += p(f.value.wrapper, `.${d}`)), f.value.emptyContainer && (t += p(f.value.emptyContainer, `.${d} .empty`)), f.value.emptyImg && (t += p(f.value.emptyImg, `.${d} .empty .empty-img`)), f.value.emptyText && (t += p(f.value.emptyText, `.${d} .empty .empty-text`)), t;
    }, z = () => {
      const t = document.querySelector(`.${d} .style-container`);
      t && (t.innerHTML = `<style>${$()}</style>`);
    }, k = _(() => {
      var t, u, c, y;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((i) => {
            var g;
            return (i == null ? void 0 : i.data) && ((g = i == null ? void 0 : i.data) == null ? void 0 : g.length) !== 0;
          });
        if (typeof ((u = e.options) == null ? void 0 : u.series) == "object")
          return ((y = (c = e.options.series) == null ? void 0 : c.data) == null ? void 0 : y.length) === 0;
      }
      return e.data.length ? (C(() => {
        m.resize();
      }), !0) : !1;
    });
    return N(() => {
      z(), O.value && (m = H.init(O.value), w(), b(), m.on("click", (t) => {
        L("com-click", t);
      }), m.on("legendselectchanged", (t) => {
        L("legend-select", t);
      })), window.addEventListener("resize", T);
    }), P(() => {
      window.removeEventListener("resize", T);
    }), D(
      () => e.line,
      () => {
        o.lineOptions = h(e.line), b();
      },
      {
        deep: !0
      }
    ), D(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting
      ],
      () => {
        w(), b();
      },
      {
        deep: !0
      }
    ), D(
      () => e.cStyle,
      () => {
        z(), C(() => {
          m.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      ...U(o),
      chartId: O,
      checkEmpty: k,
      domOnlyClass: d
    };
  }
});
export {
  Z as default
};
