// @ths/design/es/components/chart-line-simple/src/index.vue.js
import m from "./index.vue2.js";
import { openBlock as a, createElementBlock as p, normalizeClass as c, normalizeStyle as r, withDirectives as i, createElementVNode as t, vShow as n } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  ref: "chartId",
  class: "chart"
}, y = { class: "empty" }, f = ["src"];
function h(e, s, v, E, k, S) {
  var l, o;
  return a(), p("div", {
    class: c(["t-chart-line-simple", e.domOnlyClass]),
    style: r({ ...(o = (l = e.cStyle) == null ? void 0 : l.wrapper) == null ? void 0 : o.default })
  }, [
    i(t("div", u, null, 512), [
      [n, e.checkEmpty]
    ]),
    i(t("div", y, [
      t("img", {
        class: "empty-img",
        style: r({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [n, !e.checkEmpty]
    ]),
    s[1] || (s[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const C = /* @__PURE__ */ d(m, [["render", h]]);
export {
  C as default
};
