// @ths/design/es/components/background-box/src/index.vue2.js
import { defineComponent as c, computed as l, onMounted as s, watch as u } from "vue";
import { getDomRandomClass as y, setStyle as m } from "../../../utils/index.js";
const S = c({
  name: "TBackgroundBox",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器默认
        wrapper: {
          default: {}
        }
      })
    },
    // 背景图片路径数据
    data: {
      type: String,
      default: () => ""
    },
    // 边框、背景样式数据
    borderStyle: {
      type: Object,
      default: () => ({
        borderWidth: {},
        // 边框宽
        borderImageWidth: {},
        // 边框图像宽
        borderFill: ""
        // 填充方式
      })
    }
  },
  setup(t) {
    const r = l(() => t.cStyle), a = l(() => t.borderStyle), o = y("background-box"), d = () => {
      let e = "";
      return r.value.wrapper && (e += m(r.value.wrapper, `.${o}`)), e;
    }, n = () => {
      const e = document.querySelector(`.${o} .style-container`);
      e && (e.innerHTML = `<style>${d()}</style>`);
    };
    return s(() => {
      n();
    }), u(
      () => [t.cStyle],
      () => {
        n();
      },
      {
        deep: !0
      }
    ), {
      curStyle: r,
      curBorderStyle: a,
      domOnlyClass: o
    };
  }
});
export {
  S as default
};
