// @ths/design/es/components/component/src/index.vue.js
import m from "./index.vue2.js";
import { openBlock as r, createElementBlock as s, normalizeClass as t, normalizeStyle as d, createElementVNode as l, createCommentVNode as a, createBlock as p, resolveDynamicComponent as u, mergeProps as f, Fragment as g, renderList as y, withModifiers as v } from "vue";
import "./index.vue3.js";
import k from "../../../_virtual/_plugin-vue_export-helper.js";
const w = {
  key: 0,
  class: "el-loading-spinner"
}, $ = ["onMousedown"];
function b(e, o, c, M, z, B) {
  return r(), s("div", {
    ref: "containerRef",
    class: t(["t-component", { empty: !e.name, draggable: e.draggable }]),
    style: d({ ...e.curStyle.wrapper.default }),
    onMouseenter: o[0] || (o[0] = (n) => e.showHandles = e.draggable),
    onMouseleave: o[1] || (o[1] = (n) => e.showHandles = !1)
  }, [
    e.loading && !e.name ? (r(), s("div", w, o[2] || (o[2] = [
      l("svg", {
        class: "circular",
        viewBox: "25 25 50 50"
      }, [
        l("circle", {
          class: "path",
          cx: "50",
          cy: "50",
          r: "20",
          fill: "none"
        })
      ], -1),
      l("p", { class: "el-loading-text" }, " 加载中... ", -1)
    ]))) : a("", !0),
    e.comName && e.componentLoaded ? (r(), p(u(e.comName), f({ key: e.internalKey }, e.filterAttrs(e.$attrs)), null, 16)) : a("", !0),
    e.draggable ? (r(!0), s(g, { key: 2 }, y(e.visibleHandles, (n) => (r(), s("div", {
      key: n,
      class: t(["resize-handle", n]),
      onMousedown: v((i) => e.startResize(n, i), ["stop"])
    }, null, 42, $))), 128)) : a("", !0)
  ], 38);
}
const H = /* @__PURE__ */ k(m, [["render", b]]);
export {
  H as default
};
