// @ths/design/es/components/component/src/index.vue2.js
import { defineComponent as U, ref as t, computed as _, reactive as I, onUnmounted as K, watch as N } from "vue";
import { ComponentLoader as q } from "../../../utils/loader.js";
import { filterAttrs as G } from "../../../utils/index.js";
const ee = U({
  name: "TComponent",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 动态组件名称
    name: {
      type: String,
      default: ""
    },
    // 加载中的状态
    loading: {
      type: Boolean,
      default: !1
    },
    // 是否开启拖拽拉伸功能
    draggable: {
      type: Boolean,
      default: !1
    },
    // 拖拽拉伸的边界限制
    dragLimit: {
      type: Object,
      default: () => ({})
    },
    // 挂载节点（用于计算缩放比例）
    mountTo: {
      type: String,
      default: "body"
    },
    // 可拉伸方向配置
    // 'all' - 全部方向, 'none' - 禁止拉伸, string[] - 指定方向
    resizeDirections: {
      type: [String, Array],
      default: () => ["top", "bottom", "left", "right", "top-left", "top-right", "bottom-left", "bottom-right"]
    }
  },
  emits: ["resize"],
  setup(C, { emit: T }) {
    const o = t(null), D = t(!1), j = _(() => {
      const e = C.resizeDirections;
      return e === "none" ? [] : e === "all" ? ["top", "bottom", "left", "right", "top-left", "top-right", "bottom-left", "bottom-right"] : Array.isArray(e) ? e : [];
    }), c = I(JSON.parse(JSON.stringify(C.cStyle))), R = t(0), k = t(!1), E = t(!1), X = t(""), z = t(!1), h = t(""), Y = t(0), O = t(0), s = t(0), i = t(0), g = t(0), y = t(0);
    let W = 1, $ = 1, x = 0, M = 0;
    const d = t(null), H = () => {
      d.value && o.value && (o.value.contains(d.value) && o.value.removeChild(d.value), d.value = null);
    }, B = () => {
      var u;
      let e = 1, m = 1, r = (u = o.value) == null ? void 0 : u.parentElement;
      for (; r && r !== document.body; ) {
        const b = window.getComputedStyle(r).transform;
        if (b && b !== "none") {
          const S = b.match(/matrix(3d)?\(([-\d\.eE, ]+)\)/);
          if (S) {
            const a = S[2].split(",").map((n) => parseFloat(n.trim()));
            e *= a[0], m *= a[3];
          }
        }
        r = r.parentElement;
      }
      return { scaleX: e, scaleY: m };
    }, L = () => {
      z.value && (z.value = !1, document.removeEventListener("mousemove", A), document.removeEventListener("mouseup", L), H(), document.body.style.removeProperty("user-select"), x = 0, M = 0);
    }, J = () => {
      if (!o.value)
        return;
      H(), d.value = document.createElement("div");
      const e = d.value;
      switch (e.style.position = "fixed", e.style.top = "0", e.style.left = "0", e.style.width = "100%", e.style.height = "100%", e.style.zIndex = "10000", e.style.pointerEvents = "auto", e.style.background = "transparent", h.value) {
        case "top":
        case "bottom":
          e.style.cursor = "ns-resize";
          break;
        case "left":
        case "right":
          e.style.cursor = "ew-resize";
          break;
        case "top-left":
        case "bottom-right":
          e.style.cursor = "nwse-resize";
          break;
        case "top-right":
        case "bottom-left":
          e.style.cursor = "nesw-resize";
          break;
      }
      o.value.appendChild(e);
    }, A = (e) => {
      if (!z.value || !o.value)
        return;
      const m = W || 1, r = $ || 1, u = (e.clientX - Y.value) / m, v = (e.clientY - O.value) / r, b = x || document.documentElement.clientWidth, S = M || document.documentElement.clientHeight;
      let a = s.value, n = i.value, f = g.value, p = y.value;
      switch (h.value) {
        case "right":
          a = s.value + u;
          break;
        case "bottom":
          n = i.value + v;
          break;
        case "left":
          {
            const l = Math.min(u, s.value - 1);
            a = s.value - l, f = g.value + l;
          }
          break;
        case "top":
          {
            const l = Math.min(v, i.value - 1);
            n = i.value - l, p = y.value + l;
          }
          break;
        case "top-left":
          {
            const l = Math.min(u, s.value - 1), F = Math.min(v, i.value - 1);
            a = s.value - l, n = i.value - F, f = g.value + l, p = y.value + F;
          }
          break;
        case "top-right":
          a = s.value + u;
          {
            const l = Math.min(v, i.value - 1);
            n = i.value - l, p = y.value + l;
          }
          break;
        case "bottom-left":
          {
            const l = Math.min(u, s.value - 1);
            a = s.value - l, f = g.value + l;
          }
          n = i.value + v;
          break;
        case "bottom-right":
          a = s.value + u, n = i.value + v;
          break;
      }
      a < 1 && (a = 1), n < 1 && (n = 1), f = Math.max(0, Math.min(f, b - a)), p = Math.max(0, Math.min(p, S - n)), c.wrapper.default.width = `${a}px`, c.wrapper.default.height = `${n}px`;
      const w = {};
      h.value.includes("left") && (w.left = `${f}px`, c.wrapper.default.left = w.left), h.value.includes("top") && (w.top = `${p}px`, c.wrapper.default.top = w.top), T("resize", {
        width: `${a}px`,
        height: `${n}px`,
        ...w
      });
    }, P = (e, m) => {
      if (!o.value)
        return;
      const r = B();
      W = r.scaleX, $ = r.scaleY, x = document.documentElement.clientWidth, M = document.documentElement.clientHeight, document.body.style.userSelect = "none", z.value = !0, h.value = e, J(), Y.value = m.clientX, O.value = m.clientY, g.value = parseFloat(c.wrapper.default.left) || 0, y.value = parseFloat(c.wrapper.default.top) || 0, s.value = o.value.offsetWidth, i.value = o.value.offsetHeight, document.addEventListener("mousemove", A), document.addEventListener("mouseup", L);
    };
    return K(() => {
      L(), H();
    }), N(() => C.name, (e) => {
      R.value += 1, k.value = !1, E.value = !0, e ? q.loadComponent(e.replace("Component", ""), () => {
        X.value = e, k.value = !0, E.value = !1;
      }) : E.value = !1;
    }, { immediate: !0 }), {
      containerRef: o,
      showHandles: D,
      visibleHandles: j,
      comName: X,
      curStyle: c,
      filterAttrs: G,
      componentLoaded: k,
      internalKey: R,
      startResize: P
    };
  }
});
export {
  ee as default
};
