// @ths/c-common-base-table/lib/index.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('https://unpkg.com/vue@next')) :
  typeof define === 'function' && define.amd ? define(['exports', 'https://unpkg.com/vue@next'], factory) :
  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global["C-common-base-table"] = {}, global.Vue));
})(this, (function (exports, vue) { 'use strict';

  var script$3 = vue.defineComponent({
      name: 'TDensityButton',
      /*
       * props: {
       * },
       */
      emits: ['table-size-change'],
      setup(props, { emit }) {
          // 响应式数据
          const state = vue.reactive({
              // 表格尺寸
              tableSize: 'mini',
          });
          /**
           * 设置表格尺寸（密度）
           * @param command 表格尺寸（密度）
           */
          const setTableDensity = (command) => {
              state.tableSize = command;
              // 发送事件，告诉父组件表单尺寸改变了
              emit('table-size-change', state.tableSize);
          };
          return {
              ...vue.toRefs(state),
              setTableDensity,
          };
      },
  });

  function render$3(_ctx, _cache, $props, $setup, $data, $options) {
    const _component_operation = vue.resolveComponent("operation");
    const _component_el_icon = vue.resolveComponent("el-icon");
    const _component_el_tooltip = vue.resolveComponent("el-tooltip");
    const _component_el_dropdown_item = vue.resolveComponent("el-dropdown-item");
    const _component_el_dropdown_menu = vue.resolveComponent("el-dropdown-menu");
    const _component_el_dropdown = vue.resolveComponent("el-dropdown");

    return (vue.openBlock(), vue.createBlock(_component_el_dropdown, {
      class: "table-handle-dropdown",
      trigger: "click",
      onCommand: _ctx.setTableDensity
    }, {
      dropdown: vue.withCtx(() => [
        vue.createVNode(_component_el_dropdown_menu, null, {
          default: vue.withCtx(() => [
            (vue.openBlock(), vue.createElementBlock(vue.Fragment, null, vue.renderList([
              { label: '正常', value: 'small' },
              { label: '紧凑', value: 'mini' },
              { label: '宽松', value: 'medium' },
            ], (item) => {
              return vue.createVNode(_component_el_dropdown_item, {
                key: item.label,
                class: vue.normalizeClass({ active: _ctx.tableSize === item.value }),
                command: item.value
              }, {
                default: vue.withCtx(() => [
                  vue.createTextVNode(vue.toDisplayString(item.label), 1 /* TEXT */)
                ]),
                _: 2 /* DYNAMIC */
              }, 1032 /* PROPS, DYNAMIC_SLOTS */, ["class", "command"])
            }), 64 /* STABLE_FRAGMENT */))
          ]),
          _: 1 /* STABLE */
        })
      ]),
      default: vue.withCtx(() => [
        vue.createVNode(_component_el_tooltip, {
          content: "密度",
          effect: "dark",
          placement: "top"
        }, {
          default: vue.withCtx(() => [
            vue.createVNode(_component_el_icon, {
              class: "pointer m-r-xs tooltip-icon",
              size: 18
            }, {
              default: vue.withCtx(() => [
                vue.createVNode(_component_operation)
              ]),
              _: 1 /* STABLE */
            })
          ]),
          _: 1 /* STABLE */
        })
      ]),
      _: 1 /* STABLE */
    }, 8 /* PROPS */, ["onCommand"]))
  }

  script$3.render = render$3;
  script$3.__file = "package/ths/components/c-common-base-table/src/view/t-density-button.vue";

  var script$2 = vue.defineComponent({
      name: 'TFullScreenButton',
      /*
       * props: {
       * },
       */
      emits: ['full-screen-change'],
      setup(props, { emit }) {
          // 响应式数据
          const state = vue.reactive({
              // 是否全屏
              fullScreen: false,
          });
          /**
           * esc 按键事件
           * @param event
           */
          const keydownEvent = (event) => {
              if (document.fullscreenElement) {
                  document.exitFullscreen();
              }
              if (event.keyCode === 27) {
                  state.fullScreen = false;
              }
              // 发送事件，切换全屏状态
              emit('full-screen-change', state.fullScreen);
          };
          /**
           * esc 窗口尺寸变化事件
           */
          const windowResizeEvent = () => {
              state.fullScreen = document.fullscreen;
              // 发送事件，切换全屏状态
              emit('full-screen-change', state.fullScreen);
          };
          /**
           * 设置控制全屏显示状态值
           */
          const handleSetFullScreen = () => {
              // 设置状态值
              state.fullScreen = !state.fullScreen;
              // 浏览器全屏模式
              if (state.fullScreen && document.body.requestFullscreen) {
                  document.body.requestFullscreen();
              }
              else if (document.fullscreenElement) {
                  // 浏览器退出全屏模式
                  document.exitFullscreen();
              }
              // 发送事件，切换全屏状态
              emit('full-screen-change', state.fullScreen);
          };
          vue.onMounted(async () => {
              // 监听 esc 按键事件  退出全屏
              window.addEventListener('keyup', keydownEvent, false);
              // 监听窗口尺寸改变事件  退出全屏
              window.addEventListener('resize', windowResizeEvent, false);
          });
          vue.onUnmounted(() => {
              // 移除监听事件
              window.removeEventListener('keyup', keydownEvent);
              window.removeEventListener('resize', windowResizeEvent);
          });
          return {
              ...vue.toRefs(state),
              handleSetFullScreen,
          };
      },
  });

  const _hoisted_1$2 = {
    key: 1,
    class: "fa fa-compress pointer m-r-xs tooltip-icon"
  };

  function render$2(_ctx, _cache, $props, $setup, $data, $options) {
    const _component_full_screen = vue.resolveComponent("full-screen");
    const _component_el_icon = vue.resolveComponent("el-icon");
    const _component_el_tooltip = vue.resolveComponent("el-tooltip");

    return (vue.openBlock(), vue.createBlock(_component_el_tooltip, {
      effect: "dark",
      content: _ctx.fullScreen ? '取消全屏' : '全屏',
      placement: "top",
      "auto-close": 1000
    }, {
      default: vue.withCtx(() => [
        vue.createElementVNode("span", {
          class: "tooltip-outline-none",
          onClick: _cache[0] || (_cache[0] = $event => (_ctx.handleSetFullScreen()))
        }, [
          (!_ctx.fullScreen)
            ? (vue.openBlock(), vue.createBlock(_component_el_icon, {
                key: 0,
                class: "pointer m-r-xs tooltip-icon",
                size: 18
              }, {
                default: vue.withCtx(() => [
                  vue.createVNode(_component_full_screen)
                ]),
                _: 1 /* STABLE */
              }))
            : vue.createCommentVNode("v-if", true),
          (_ctx.fullScreen)
            ? (vue.openBlock(), vue.createElementBlock("i", _hoisted_1$2))
            : vue.createCommentVNode("v-if", true)
        ])
      ]),
      _: 1 /* STABLE */
    }, 8 /* PROPS */, ["content"]))
  }

  script$2.render = render$2;
  script$2.__file = "package/ths/components/c-common-base-table/src/view/t-full-screen-button.vue";

  var script$1 = vue.defineComponent({
      name: 'TTable',
      components: {
          TDensityButton: script$3,
          TFullScreenButton: script$2, // 表格全屏设置按钮
      },
      inheritAttrs: true,
      props: {
          // 表格数据
          tableData: {
              type: Array,
              required: true,
              default: () => [],
          },
          // 表头配置项
          tableHeaderOpitons: {
              type: Array,
              required: true,
              default: () => [],
          },
          // 判断是否禁用多选方法（返回true不禁用）
          selectable: {
              type: Function,
              default: (row, index) => true,
          },
          // 页码
          pageNum: {
              type: Number,
              default: 1,
          },
          // 每页条数
          pageSize: {
              type: Number,
              default: 10,
          },
          // 是否开启序号列(开启的话,必须传入每页条数/页码)
          openIndexColumn: {
              type: Boolean,
              default: false,
          },
          // 是否开启选择列
          openSelectColumn: {
              type: Boolean,
              default: false,
          },
          // 要修改背景的列 index 数组
          changeBgColumnIndex: {
              type: Array,
              default: () => [],
          },
          // 要修改的背景色
          changeBgColumnColor: {
              type: String,
              default: '#F5F7FA',
          },
          // 操作列是否显示
          showOperationColumn: {
              type: Boolean,
              default: true,
          },
          // 表格顶部左侧列显隐控制
          showTableLeftBtn: {
              type: Object,
              default: () => ({
                  densityBtn: true,
                  fullScreenBtn: true,
              }),
          },
          // 是否开启斑马纹
          openStripe: {
              type: Boolean,
              default: true,
          },
          // 关闭表头背景色是否根据主题色变化
          closeThemeColorHeader: {
              type: Boolean,
              default: false,
          },
          // 序号列宽度
          orderNoWidth: {
              type: Number,
              default: 55,
          },
          // 是否禁用当前页面的多选框
          isDisableSelectionAll: {
              type: Boolean,
              default: false,
          },
      },
      emits: ['table-sort-change', 'table-choose-change', 'update:fullScreen'],
      setup(props, { emit }) {
          // 响应式数据
          const state = vue.reactive({
              // 用户选择的列表项数组
              multipleSelection: [],
              // 表格尺寸
              tableSize: 'mini',
              // 是否全屏
              fullScreen: false,
              // 列表类名
              tableClassName: '',
          });
          const tableRef = vue.ref();
          // 过滤 操作项 之后的列配置
          const tableNormalOptions = vue.computed(() => props.tableHeaderOpitons.filter((item) => !item.action));
          // 操作项列配置
          const tableActionOptions = vue.computed(() => props.tableHeaderOpitons.find((item) => item.action));
          /*
           * console.log('过滤 操作项 之后的配置', tableNormalOptions.value);
           * console.log('操作项', tableActionOptions.value);
           */
          /**
           * 改变表格列背景色
           * @param column 表格单元格的 style 的回调方法接受的数据
           */
          const changeCellStyle = (column) => {
              if (props.changeBgColumnIndex && props.changeBgColumnIndex.includes(column.columnIndex)) {
                  // console.log(column);
                  return `background-color: ${props.changeBgColumnColor}`;
              }
              return null;
          };
          /**
           * 表格排序
           * @param sortColumn
           */
          const onSortChange = (sortColumn) => {
              let orderBy = '';
              if (sortColumn && sortColumn.column && sortColumn.column.sortable) {
                  orderBy = `${sortColumn.column.sortable} ${sortColumn.order.replace('ending', '')}`;
              }
              else {
                  orderBy = '';
              }
              emit('table-sort-change', orderBy);
          };
          /**
           * 表格多选改变
           * @description 当选择多条表格数据时，被选中的数据会被存储到 state 响应数据中去
           * @param val 当前选中的全部表格数据（以数组的形式进行存储）
           */
          const onSelectChange = (val) => {
              state.multipleSelection = val;
              emit('table-choose-change', state.multipleSelection);
          };
          /**
           * element计算的高度，没有带上小数，导致边框线被压着，这里重新设置一下
           */
          const setTableBodyTop = () => {
              setTimeout(() => {
                  const tableDom = document.getElementsByClassName('el-table');
                  if (tableDom && tableDom.length > 0) {
                      for (let i = 0; i < tableDom.length; i++) {
                          // 获取高度
                          const tableHeaderDom = tableDom[i].getElementsByClassName('el-table__header-wrapper')[0];
                          const tableHeaderHeight = tableHeaderDom.getBoundingClientRect().height + 'px';
                          // 设置top值
                          const tableBody = tableDom[i].getElementsByClassName('el-table__fixed-body-wrapper');
                          if (tableBody && tableBody[0]) {
                              tableBody[0].style.top = tableHeaderHeight;
                          }
                          if (tableBody && tableBody[1]) {
                              tableBody[1].style.top = tableHeaderHeight;
                          }
                          const tablePatch = tableDom[i].getElementsByClassName('el-table__fixed-right-patch');
                          if (tablePatch && tablePatch[0]) {
                              tablePatch[0].style.height = tableHeaderHeight;
                          }
                      }
                  }
              }, 200);
          };
          /**
           * 设置表格尺寸（密度）
           * @param command 表格尺寸（密度）
           */
          const setTableDensity = (command) => {
              state.tableSize = command;
              setTableBodyTop();
          };
          /**
           * 控制全屏状态
           */
          const handleSetFullScreen = (val) => {
              state.fullScreen = val;
              emit('update:fullScreen', val);
          };
          /**
           * 是否有值
           */
          const isNoVal = (val) => {
              let newVal = val;
              if (!val && val !== 0) {
                  newVal = '-';
              }
              return newVal;
          };
          vue.onMounted(() => {
              // element计算的高度，没有带上小数，导致边框线被压着，这里重新设置一下
              setTableBodyTop();
          });
          // 设置列表全选是否禁用
          vue.watch(() => props.isDisableSelectionAll, () => {
              if (props.isDisableSelectionAll) {
                  state.tableClassName = 'disable-selection';
              }
              else {
                  state.tableClassName = '';
              }
          }, {
              immediate: true,
          });
          /*
           * 表头背景是否根据主题色变化
           * watch(
           *   () => props.closeThemeColorHeader,
           *   (val) => {
           *     const themeName = JSON.parse(localStorage.getItem('userStyle') || '{}').theme ?? '';
           *     console.log('在表格中，获取主题色 ===', themeName);
           *     if (val && themeName) {
           *       document.body.style.setProperty(
           *         '--table-header-background-color',
           *         `var(--thead-${themeName}-background)`,
           *       );
           *     } else {
           *       document.body.style.setProperty('--table-header-background-color', 'var(--thead-background)');
           *     }
           *   },
           *   {
           *     immediate: true,
           *   },
           * );
           */
          return {
              ...vue.toRefs(state),
              tableNormalOptions,
              tableActionOptions,
              tableRef,
              isNoVal,
              changeCellStyle,
              onSortChange,
              onSelectChange,
              setTableDensity,
              handleSetFullScreen,
              // cellClass,
          };
      },
  });

  const _hoisted_1$1 = { class: "flex flex-end" };

  function render$1(_ctx, _cache, $props, $setup, $data, $options) {
    const _component_t_density_button = vue.resolveComponent("t-density-button");
    const _component_t_full_screen_button = vue.resolveComponent("t-full-screen-button");
    const _component_el_col = vue.resolveComponent("el-col");
    const _component_el_row = vue.resolveComponent("el-row");
    const _component_el_table_column = vue.resolveComponent("el-table-column");
    const _component_el_table = vue.resolveComponent("el-table");

    return (vue.openBlock(), vue.createElementBlock(vue.Fragment, null, [
      vue.createElementVNode("div", {
        class: vue.normalizeClass(["form-list-container", { 'close-theme-color-header': _ctx.closeThemeColorHeader }])
      }, [
        vue.createCommentVNode(" 表格操作列 "),
        (_ctx.showOperationColumn)
          ? (vue.openBlock(), vue.createBlock(_component_el_row, {
              key: 0,
              align: "middle",
              justify: "space-between",
              type: "flex",
              class: "m-b-sm"
            }, {
              default: vue.withCtx(() => [
                vue.createCommentVNode(" 左侧按钮组（密度、全屏） "),
                vue.createVNode(_component_el_col, {
                  span: 8,
                  class: "flex"
                }, {
                  default: vue.withCtx(() => [
                    vue.createCommentVNode(" 表格密度设置按钮 "),
                    (_ctx.showTableLeftBtn.densityBtn)
                      ? (vue.openBlock(), vue.createBlock(_component_t_density_button, {
                          key: 0,
                          onTableSizeChange: _ctx.setTableDensity
                        }, null, 8 /* PROPS */, ["onTableSizeChange"]))
                      : vue.createCommentVNode("v-if", true),
                    vue.createCommentVNode(" 表格全屏设置按钮 "),
                    (_ctx.showTableLeftBtn.fullScreenBtn)
                      ? (vue.openBlock(), vue.createBlock(_component_t_full_screen_button, {
                          key: 1,
                          onFullScreenChange: _ctx.handleSetFullScreen
                        }, null, 8 /* PROPS */, ["onFullScreenChange"]))
                      : vue.createCommentVNode("v-if", true),
                    vue.createCommentVNode(" 表格标题 "),
                    vue.renderSlot(_ctx.$slots, "tableTitleSlot")
                  ]),
                  _: 3 /* FORWARDED */
                }),
                vue.createCommentVNode(" 右侧按钮组 "),
                vue.createVNode(_component_el_col, { span: 16 }, {
                  default: vue.withCtx(() => [
                    vue.createElementVNode("div", _hoisted_1$1, [
                      vue.createCommentVNode(" 表格按钮组 - 插槽 "),
                      vue.renderSlot(_ctx.$slots, "tableBtnSlot")
                    ])
                  ]),
                  _: 3 /* FORWARDED */
                })
              ]),
              _: 3 /* FORWARDED */
            }))
          : vue.createCommentVNode("v-if", true),
        vue.createCommentVNode(" 表格 "),
        vue.createVNode(_component_el_table, vue.mergeProps({
          ref: "tableRef",
          data: _ctx.tableData,
          "cell-style": _ctx.changeCellStyle,
          size: _ctx.tableSize,
          "tooltip-effect": "dark",
          sortable: "",
          border: "",
          stripe: _ctx.openStripe,
          class: "table-component"
        }, _ctx.$attrs, {
          onSelectionChange: _ctx.onSelectChange,
          onSortChange: _ctx.onSortChange,
          "header-cell-class-name": _ctx.tableClassName
        }), {
          default: vue.withCtx(() => [
            vue.createCommentVNode(" 选择列（多选按钮） "),
            (_ctx.openSelectColumn)
              ? (vue.openBlock(), vue.createBlock(_component_el_table_column, {
                  key: 0,
                  type: 'selection',
                  "header-align": 'center',
                  align: 'center',
                  fixed: 'left',
                  selectable: _ctx.selectable,
                  resizable: false
                }, null, 8 /* PROPS */, ["selectable"]))
              : vue.createCommentVNode("v-if", true),
            vue.createCommentVNode(" 序号列 "),
            (_ctx.openIndexColumn && _ctx.pageSize && _ctx.pageNum)
              ? (vue.openBlock(), vue.createBlock(_component_el_table_column, {
                  key: 1,
                  type: 'index',
                  align: 'center',
                  fixed: 'left',
                  label: '序号',
                  width: _ctx.orderNoWidth,
                  "min-width": _ctx.orderNoWidth,
                  resizable: false
                }, {
                  default: vue.withCtx((scope) => [
                    vue.createElementVNode("span", null, vue.toDisplayString((_ctx.pageNum - 1) * _ctx.pageSize + scope.$index + 1), 1 /* TEXT */)
                  ]),
                  _: 1 /* STABLE */
                }, 8 /* PROPS */, ["width", "min-width"]))
              : vue.createCommentVNode("v-if", true),
            vue.createCommentVNode(" 普通列 "),
            (vue.openBlock(true), vue.createElementBlock(vue.Fragment, null, vue.renderList(_ctx.tableNormalOptions, (tOptions, tIndex) => {
              return (vue.openBlock(), vue.createElementBlock(vue.Fragment, { key: tIndex }, [
                vue.createCommentVNode(" 单表头 "),
                (!tOptions.child)
                  ? (vue.openBlock(), vue.createBlock(_component_el_table_column, {
                      key: tIndex,
                      prop: tOptions.prop,
                      label: tOptions.label,
                      width: tOptions.width ? tOptions.width : '',
                      "min-width": tOptions.minWidth ? tOptions.minWidth : '',
                      align: tOptions.align ? tOptions.align : 'center',
                      "header-align": tOptions.headerAlign ? tOptions.headerAlign : 'center',
                      fixed: tOptions.fixed ? tOptions.fixed : false,
                      sortable: tOptions.sortable ? tOptions.sortable : false,
                      "show-overflow-tooltip": tOptions.slot ? false : true,
                      resizable: tOptions.resizable !== null ? tOptions.resizable : true
                    }, vue.createSlots({ _: 2 /* DYNAMIC */ }, [
                      (tOptions.slot)
                        ? {
                            name: "default",
                            fn: vue.withCtx((scope) => [
                              vue.renderSlot(_ctx.$slots, tOptions.slot, { scope: scope })
                            ])
                          }
                        : {
                            name: "default",
                            fn: vue.withCtx((scope) => [
                              vue.createTextVNode(vue.toDisplayString(`${_ctx.isNoVal(scope.row[tOptions.prop])}`), 1 /* TEXT */)
                            ])
                          }
                    ]), 1032 /* PROPS, DYNAMIC_SLOTS */, ["prop", "label", "width", "min-width", "align", "header-align", "fixed", "sortable", "show-overflow-tooltip", "resizable"]))
                  : (vue.openBlock(), vue.createElementBlock(vue.Fragment, { key: 1 }, [
                      vue.createCommentVNode(" 多表头 "),
                      vue.createVNode(_component_el_table_column, {
                        label: tOptions.label,
                        "header-align": 'center'
                      }, {
                        default: vue.withCtx(() => [
                          (vue.openBlock(true), vue.createElementBlock(vue.Fragment, null, vue.renderList(tOptions.child, (childOptions, cIndex) => {
                            return (vue.openBlock(), vue.createBlock(_component_el_table_column, {
                              key: cIndex,
                              prop: childOptions.prop,
                              label: childOptions.label,
                              width: childOptions.width ? childOptions.width : '',
                              "min-width": childOptions.minWidth ? childOptions.minWidth : '',
                              align: childOptions.align ? childOptions.align : 'center',
                              "header-align": childOptions.headerAlign ? childOptions.headerAlign : 'center',
                              sortable: childOptions.sortable ? childOptions.sortable : false,
                              "show-overflow-tooltip": childOptions.slot ? false : true,
                              resizable: childOptions.resizable !== null ? childOptions.resizable : true
                            }, vue.createSlots({ _: 2 /* DYNAMIC */ }, [
                              (childOptions.slot)
                                ? {
                                    name: "default",
                                    fn: vue.withCtx((scope) => [
                                      vue.renderSlot(_ctx.$slots, childOptions.slot, { scope: scope })
                                    ])
                                  }
                                : {
                                    name: "default",
                                    fn: vue.withCtx((scope) => [
                                      vue.createTextVNode(vue.toDisplayString(`${_ctx.isNoVal(scope.row[childOptions.prop])}`), 1 /* TEXT */)
                                    ])
                                  }
                            ]), 1032 /* PROPS, DYNAMIC_SLOTS */, ["prop", "label", "width", "min-width", "align", "header-align", "sortable", "show-overflow-tooltip", "resizable"]))
                          }), 128 /* KEYED_FRAGMENT */))
                        ]),
                        _: 2 /* DYNAMIC */
                      }, 1032 /* PROPS, DYNAMIC_SLOTS */, ["label"])
                    ], 2112 /* STABLE_FRAGMENT, DEV_ROOT_FRAGMENT */))
              ], 64 /* STABLE_FRAGMENT */))
            }), 128 /* KEYED_FRAGMENT */)),
            vue.createCommentVNode(" 操作列 "),
            (_ctx.tableActionOptions)
              ? (vue.openBlock(), vue.createBlock(_component_el_table_column, {
                  key: 2,
                  width: _ctx.tableActionOptions.width ? _ctx.tableActionOptions.width : '',
                  "min-width": _ctx.tableActionOptions.minWidth ? _ctx.tableActionOptions.minWidth : '',
                  "header-align": 'center',
                  align: 'center',
                  fixed: _ctx.tableActionOptions.fixed ? _ctx.tableActionOptions.fixed : 'right',
                  label: "操作",
                  resizable: false
                }, {
                  default: vue.withCtx((scope) => [
                    vue.renderSlot(_ctx.$slots, 'actionSolt', { scope: scope })
                  ]),
                  _: 3 /* FORWARDED */
                }, 8 /* PROPS */, ["width", "min-width", "fixed"]))
              : vue.createCommentVNode("v-if", true)
          ]),
          _: 3 /* FORWARDED */
        }, 16 /* FULL_PROPS */, ["data", "cell-style", "size", "stripe", "onSelectionChange", "onSortChange", "header-cell-class-name"])
      ], 2 /* CLASS */),
      vue.createCommentVNode(" <span>表格组件接收到的 $attrs: {{ $attrs }}</span> ")
    ], 2112 /* STABLE_FRAGMENT, DEV_ROOT_FRAGMENT */))
  }

  script$1.render = render$1;
  script$1.__scopeId = "data-v-275cda5d";
  script$1.__file = "package/ths/components/c-common-base-table/src/view/t-table.vue";

  var script = vue.defineComponent({
      name: 'TTableMega',
      components: {
          TDensityButton: script$3,
          TFullScreenButton: script$2, // 表格全屏设置按钮
      },
      inheritAttrs: true,
      props: {
          // 表格数据
          tableData: {
              type: Array,
              required: true,
              default: () => [],
          },
          // 表头配置项
          tableHeaderOpitons: {
              type: Array,
              required: true,
              default: () => [],
          },
          // 页码
          pageNum: {
              type: Number,
              default: 1,
          },
          // 每页条数
          pageSize: {
              type: Number,
              default: 10,
          },
          // 是否开启序号列(开启的话,必须传入每页条数/页码)
          openIndexColumn: {
              type: Boolean,
              default: false,
          },
          // 是否开启选择列
          openSelectColumn: {
              type: Boolean,
              default: false,
          },
          // 操作列是否显示
          showOperationColumn: {
              type: Boolean,
              default: true,
          },
          // 表格顶部左侧列显隐控制
          showTableLeftBtn: {
              type: Object,
              default: () => ({
                  densityBtn: true,
                  fullScreenBtn: true,
              }),
          },
          // 是否开启斑马纹
          openStripe: {
              type: Boolean,
              default: true,
          },
          // 关闭表头背景色是否根据主题色变化
          closeThemeColorHeader: {
              type: Boolean,
              default: false,
          },
          // 序号列宽度
          orderNoWidth: {
              type: String,
              default: '52',
          },
          // 是否禁用当前页面的多选框
          isDisableSelectionAll: {
              type: Boolean,
              default: false,
          },
      },
      emits: ['table-sort-change', 'table-choose-change', 'update:fullScreen'],
      setup(props, { emit }) {
          // 响应式数据
          const state = vue.reactive({
              // 用户选择的列表项数组
              multipleSelection: [],
              // 表格尺寸
              tableSize: 'mini',
              // 是否全屏
              fullScreen: false,
              // 列表类名
              tableClassName: '',
          });
          // 过滤 操作项 之后的列配置
          const tableNormalOptions = vue.computed(() => props.tableHeaderOpitons.filter((item) => !item.action));
          // 操作项列配置
          const tableActionOptions = vue.computed(() => props.tableHeaderOpitons.find((item) => item.action));
          /*
           * console.log('过滤 操作项 之后的配置', tableNormalOptions.value);
           * console.log('操作项', tableActionOptions.value);
           */
          /**
           * 表格排序
           * @param sortColumn
           */
          const onSortChange = (sortColumn) => {
              let orderBy = '';
              if (sortColumn && sortColumn.column && sortColumn.column.sortBy && sortColumn.order) {
                  orderBy = `${sortColumn.column.sortBy} ${sortColumn.order.replace('ending', '')}`;
              }
              else {
                  orderBy = '';
              }
              emit('table-sort-change', orderBy);
          };
          /**
           * 表格多选改变
           * @description 当选择多条表格数据时，被选中的数据会被存储到 state 响应数据中去
           * @param val 当前选中的全部表格数据（以数组的形式进行存储）
           */
          const onSelectChange = (val) => {
              if (Array.isArray(val)) {
                  state.multipleSelection = val;
              }
              else {
                  state.multipleSelection = val.records;
              }
              emit('table-choose-change', state.multipleSelection);
          };
          /**
           * 设置表格尺寸（密度）
           * @param command 表格尺寸（密度）
           */
          const setTableDensity = (command) => {
              state.tableSize = command;
          };
          /**
           * 控制全屏状态
           */
          const handleSetFullScreen = (val) => {
              state.fullScreen = val;
              emit('update:fullScreen', val);
          };
          /**
           * 是否有值
           */
          const isNoVal = (val) => {
              let newVal = val;
              if (!val && val !== 0) {
                  newVal = '-';
              }
              return newVal;
          };
          const xTable = vue.ref({});
          // 监听数据的变化，重新加载数据
          vue.watch(() => props.tableData, () => {
              const tableRef = xTable.value;
              if (tableRef) {
                  tableRef.loadData(props.tableData);
              }
          }, {
              deep: true,
          });
          /**
           * @description: 设置序列号的值
           * @param {*} data 单条数据
           * @return {*}返回处理后的值
           */
          const setSeqStartIndex = (data) => {
              const { rowIndex } = data;
              return (props.pageNum - 1) * props.pageSize + rowIndex + 1;
          };
          // 设置列表全选是否禁用
          vue.watch(() => props.isDisableSelectionAll, () => {
              if (props.isDisableSelectionAll) {
                  state.tableClassName = 'disable-selection';
              }
              else {
                  state.tableClassName = '';
              }
          }, {
              immediate: true,
          });
          return {
              ...vue.toRefs(state),
              tableNormalOptions,
              tableActionOptions,
              isNoVal,
              onSortChange,
              onSelectChange,
              setTableDensity,
              handleSetFullScreen,
              xTable,
              setSeqStartIndex,
          };
      },
  });

  const _hoisted_1 = { class: "flex flex-end" };

  function render(_ctx, _cache, $props, $setup, $data, $options) {
    const _component_t_density_button = vue.resolveComponent("t-density-button");
    const _component_t_full_screen_button = vue.resolveComponent("t-full-screen-button");
    const _component_el_col = vue.resolveComponent("el-col");
    const _component_el_row = vue.resolveComponent("el-row");
    const _component_vxe_column = vue.resolveComponent("vxe-column");
    const _component_vxe_colgroup = vue.resolveComponent("vxe-colgroup");
    const _component_vxe_table = vue.resolveComponent("vxe-table");

    return (vue.openBlock(), vue.createElementBlock(vue.Fragment, null, [
      vue.createElementVNode("div", {
        class: vue.normalizeClass(["form-list-container", { 'close-theme-color-header': _ctx.closeThemeColorHeader }])
      }, [
        vue.createCommentVNode(" 表格操作列 "),
        (_ctx.showOperationColumn)
          ? (vue.openBlock(), vue.createBlock(_component_el_row, {
              key: 0,
              align: "middle",
              justify: "space-between",
              type: "flex",
              class: "m-b-sm"
            }, {
              default: vue.withCtx(() => [
                vue.createCommentVNode(" 左侧按钮组（密度、全屏） "),
                vue.createVNode(_component_el_col, {
                  span: 8,
                  class: "flex"
                }, {
                  default: vue.withCtx(() => [
                    vue.createCommentVNode(" 表格密度设置按钮 "),
                    (_ctx.showTableLeftBtn.densityBtn)
                      ? (vue.openBlock(), vue.createBlock(_component_t_density_button, {
                          key: 0,
                          onTableSizeChange: _ctx.setTableDensity
                        }, null, 8 /* PROPS */, ["onTableSizeChange"]))
                      : vue.createCommentVNode("v-if", true),
                    vue.createCommentVNode(" 表格全屏设置按钮 "),
                    (_ctx.showTableLeftBtn.fullScreenBtn)
                      ? (vue.openBlock(), vue.createBlock(_component_t_full_screen_button, {
                          key: 1,
                          onFullScreenChange: _ctx.handleSetFullScreen
                        }, null, 8 /* PROPS */, ["onFullScreenChange"]))
                      : vue.createCommentVNode("v-if", true),
                    vue.createCommentVNode(" 表格标题 "),
                    vue.renderSlot(_ctx.$slots, "tableTitleSlot")
                  ]),
                  _: 3 /* FORWARDED */
                }),
                vue.createCommentVNode(" 右侧按钮组 "),
                vue.createVNode(_component_el_col, { span: 16 }, {
                  default: vue.withCtx(() => [
                    vue.createElementVNode("div", _hoisted_1, [
                      vue.createCommentVNode(" 表格按钮组 - 插槽 "),
                      vue.renderSlot(_ctx.$slots, "tableBtnSlot")
                    ])
                  ]),
                  _: 3 /* FORWARDED */
                })
              ]),
              _: 3 /* FORWARDED */
            }))
          : vue.createCommentVNode("v-if", true),
        vue.createCommentVNode(" 表格 "),
        vue.createVNode(_component_vxe_table, vue.mergeProps({
          ref: "xTable",
          size: _ctx.tableSize,
          border: "",
          stripe: _ctx.openStripe,
          "show-overflow": "",
          "row-config": {isHover: true},
          "seq-config": {seqMethod:_ctx.setSeqStartIndex},
          "column-config": {resizable: true},
          "scroll-x": {oSize: 60}
        }, _ctx.$attrs, {
          onSortChange: _ctx.onSortChange,
          onCheckboxChange: _ctx.onSelectChange,
          onCheckboxAll: _ctx.onSelectChange,
          "header-cell-class-name": _ctx.tableClassName
        }), {
          default: vue.withCtx(() => [
            vue.createCommentVNode(" 选择列（多选按钮） "),
            (_ctx.openSelectColumn)
              ? (vue.openBlock(), vue.createBlock(_component_vxe_column, {
                  key: 0,
                  type: "checkbox",
                  width: "48",
                  align: "center",
                  fixed: "left",
                  resizable: false
                }))
              : vue.createCommentVNode("v-if", true),
            vue.createCommentVNode(" 序号列 "),
            (_ctx.openIndexColumn && _ctx.pageSize && _ctx.pageNum)
              ? (vue.openBlock(), vue.createBlock(_component_vxe_column, {
                  key: 1,
                  type: "seq",
                  title: "序号",
                  width: _ctx.orderNoWidth,
                  fixed: "left",
                  align: "center",
                  resizable: false
                }, null, 8 /* PROPS */, ["width"]))
              : vue.createCommentVNode("v-if", true),
            vue.createCommentVNode(" 普通列 "),
            (vue.openBlock(true), vue.createElementBlock(vue.Fragment, null, vue.renderList(_ctx.tableNormalOptions, (tOptions, tIndex) => {
              return (vue.openBlock(), vue.createElementBlock(vue.Fragment, { key: tIndex }, [
                vue.createCommentVNode(" 单表头 "),
                (!tOptions.child)
                  ? (vue.openBlock(), vue.createBlock(_component_vxe_column, {
                      key: tIndex,
                      title: tOptions.label,
                      field: tOptions.prop,
                      width: tOptions.width ? tOptions.width : '',
                      "min-width": tOptions.minWidth ? tOptions.minWidth : '',
                      "header-align": tOptions.headerAlign ? tOptions.headerAlign : 'center',
                      align: tOptions.align ? tOptions.align : 'center',
                      fixed: tOptions.fixed ? tOptions.fixed : '',
                      sortable: tOptions.sortable ? true : false,
                      "sort-by": tOptions.sortable ? tOptions.sortable : '',
                      resizable: tOptions.resizable !== null ? tOptions.resizable : true
                    }, vue.createSlots({ _: 2 /* DYNAMIC */ }, [
                      (tOptions.slot)
                        ? {
                            name: "default",
                            fn: vue.withCtx((scope) => [
                              vue.renderSlot(_ctx.$slots, tOptions.slot, { scope: scope })
                            ])
                          }
                        : {
                            name: "default",
                            fn: vue.withCtx((scope) => [
                              vue.createTextVNode(vue.toDisplayString(`${_ctx.isNoVal(scope.row[tOptions.prop])}`), 1 /* TEXT */)
                            ])
                          }
                    ]), 1032 /* PROPS, DYNAMIC_SLOTS */, ["title", "field", "width", "min-width", "header-align", "align", "fixed", "sortable", "sort-by", "resizable"]))
                  : (vue.openBlock(), vue.createElementBlock(vue.Fragment, { key: 1 }, [
                      vue.createCommentVNode(" 多表头 "),
                      vue.createVNode(_component_vxe_colgroup, {
                        title: tOptions.label,
                        "header-align": tOptions.headerAlign ? tOptions.headerAlign : 'center'
                      }, {
                        default: vue.withCtx(() => [
                          (vue.openBlock(true), vue.createElementBlock(vue.Fragment, null, vue.renderList(tOptions.child, (childOptions, cIndex) => {
                            return (vue.openBlock(), vue.createBlock(_component_vxe_column, {
                              key: cIndex,
                              title: childOptions.label,
                              field: childOptions.prop,
                              width: childOptions.width ? childOptions.width : '',
                              "min-width": childOptions.minWidth ? childOptions.minWidth : '',
                              "header-align": childOptions.headerAlign ? childOptions.headerAlign : 'center',
                              align: childOptions.align ? childOptions.align : 'center',
                              fixed: childOptions.fixed ? childOptions.fixed : '',
                              sortable: childOptions.sortable ? true : false,
                              "sort-by": childOptions.sortable ? childOptions.sortable : '',
                              resizable: childOptions.resizable !== null ? childOptions.resizable : true
                            }, vue.createSlots({ _: 2 /* DYNAMIC */ }, [
                              (childOptions.slot)
                                ? {
                                    name: "default",
                                    fn: vue.withCtx((scope) => [
                                      vue.renderSlot(_ctx.$slots, childOptions.slot, { scope: scope })
                                    ])
                                  }
                                : {
                                    name: "default",
                                    fn: vue.withCtx((scope) => [
                                      vue.createTextVNode(vue.toDisplayString(`${_ctx.isNoVal(scope.row[childOptions.prop])}`), 1 /* TEXT */)
                                    ])
                                  }
                            ]), 1032 /* PROPS, DYNAMIC_SLOTS */, ["title", "field", "width", "min-width", "header-align", "align", "fixed", "sortable", "sort-by", "resizable"]))
                          }), 128 /* KEYED_FRAGMENT */))
                        ]),
                        _: 2 /* DYNAMIC */
                      }, 1032 /* PROPS, DYNAMIC_SLOTS */, ["title", "header-align"])
                    ], 2112 /* STABLE_FRAGMENT, DEV_ROOT_FRAGMENT */))
              ], 64 /* STABLE_FRAGMENT */))
            }), 128 /* KEYED_FRAGMENT */)),
            vue.createCommentVNode(" 操作列 "),
            (_ctx.tableActionOptions)
              ? (vue.openBlock(), vue.createBlock(_component_vxe_column, {
                  key: 2,
                  width: _ctx.tableActionOptions.width ? _ctx.tableActionOptions.width : '',
                  "min-width": _ctx.tableActionOptions.minWidth ? _ctx.tableActionOptions.minWidth : '',
                  title: "操作",
                  fixed: "right",
                  align: "center",
                  resizable: false
                }, {
                  default: vue.withCtx((scope) => [
                    vue.renderSlot(_ctx.$slots, 'actionSolt', { scope: scope })
                  ]),
                  _: 3 /* FORWARDED */
                }, 8 /* PROPS */, ["width", "min-width"]))
              : vue.createCommentVNode("v-if", true)
          ]),
          _: 3 /* FORWARDED */
        }, 16 /* FULL_PROPS */, ["size", "stripe", "seq-config", "onSortChange", "onCheckboxChange", "onCheckboxAll", "header-cell-class-name"])
      ], 2 /* CLASS */),
      vue.createCommentVNode(" <span>表格组件接收到的 $attrs: {{ $attrs }}</span> ")
    ], 2112 /* STABLE_FRAGMENT, DEV_ROOT_FRAGMENT */))
  }

  script.render = render;
  script.__scopeId = "data-v-3092ac92";
  script.__file = "package/ths/components/c-common-base-table/src/view/t-table-mega.vue";

  // 存储组件列表
  const components = [
      script$1,
      script,
      script$3,
      script$2,
  ];
  const install = (app) => {
      components.forEach((component) => {
          app.component(component.name, component);
      });
  };

  exports.TDensityButton = script$3;
  exports.TFullScreenButton = script$2;
  exports.TTable = script$1;
  exports.TTableMega = script;
  exports["default"] = script$1;
  exports.install = install;

  Object.defineProperty(exports, '__esModule', { value: true });

}));
