// @ths/design/es/components/chart-stacked-curve/src/index.vue2.js
import { defineComponent as q, computed as j, ref as M, reactive as H, onMounted as P, onUnmounted as U, watch as E, nextTick as v } from "vue";
import * as G from "echarts";
import n from "lodash/merge";
import D from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as W, DEFAULT_OPTIONS_AXIS as K } from "../../../utils/constant.js";
import { getDomRandomClass as R, setStyle as L } from "../../../utils/index.js";
const X = n(D(W), D(K)), o = n(
  D(X),
  {
    tooltip: {
      show: !0,
      trigger: "axis",
      formatter(e) {
        const s = new Array(e.length).fill(""), i = new Array(e.length).fill(0);
        let d = 0;
        for (let t = 0; t < e.length; t++) {
          const r = e[t];
          r.marker = `<span style="display:inline-block;margin-right:4px;border-radius:
            10px;width:10px;height:10px;background-color:${r.color};"></span>`, String(r.data) === "1" ? s[t] = "Ⅰ类" : String(r.data) === "2" ? s[t] = "Ⅱ类" : String(r.data) === "3" ? s[t] = "Ⅲ类" : String(r.data) === "4" ? s[t] = "Ⅳ类" : String(r.data) === "5" ? s[t] = "Ⅴ类" : String(r.data) === "6" && (s[t] = "劣Ⅴ类"), i[t] = r.seriesName.length + s[t].length, d < i[t] && (d = i[t]);
        }
        let u = `${e[0].name}<br>`;
        for (let t = 0; t < s.length; t++)
          u += `${e[t].marker}${e[t].seriesName}`, u += new Array(d - i[t] + 2).fill("&emsp;").join(""), u += `${s[t]}<br>`;
        return u;
      }
    },
    legend: {
      show: !0,
      icon: "rect",
      itemWidth: 8,
      itemHeight: 8,
      itemGap: 20,
      color: "#fff",
      textStyle: {
        color: "#fff",
        fontSize: 20
      },
      itemStyle: {
        borderWidth: 4
      },
      data: [
        {
          itemStyle: {
            borderColor: "rgba(0, 255, 255, 0.4)"
          }
        },
        {
          itemStyle: {
            borderColor: "rgba(255, 130, 80, 0.4)"
          }
        },
        {
          itemStyle: {
            borderColor: "rgba(35, 129, 248, 0.4)"
          }
        },
        {
          itemStyle: {
            borderColor: "rgba(255, 221, 0, 0.4)"
          }
        }
      ]
    },
    calculable: !0,
    grid: {
      left: "70px",
      right: "20px",
      bottom: "30px"
    },
    xAxis: {
      type: "category",
      boundaryGap: !1,
      axisLine: {
        show: !0,
        lineStyle: {
          color: "rgba(173, 197, 255, 0.3)"
        }
      },
      axisLabel: {
        color: "rgba(255, 255, 255, 0.8)",
        fontSize: 20,
        interval: 0
      },
      splitLine: {
        show: !0,
        lineStyle: {
          type: "dashed",
          color: "rgba(173, 197, 255, 0)"
        }
      }
    },
    yAxis: {
      type: "value",
      splitLine: {
        show: !0,
        lineStyle: {
          type: "dashed",
          color: "rgba(173, 197, 255, 0.3)"
        }
      },
      axisLabel: {
        color: "white",
        fontSize: 20,
        formatter(e, s) {
          let i = "";
          switch (e) {
            case 1:
              i = "Ⅰ类";
              break;
            case 2:
              i = "Ⅱ类";
              break;
            case 3:
              i = "Ⅲ类";
              break;
            case 4:
              i = "Ⅳ类";
              break;
            case 5:
              i = "Ⅴ类";
              break;
            case 6:
              i = "劣Ⅴ类";
              break;
          }
          return i;
        }
      }
    },
    series: [
      {}
    ]
  }
), Z = q({
  name: "TChartStackedCurve",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 设置主轴
    mainAxis: {
      type: String,
      default: "x"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 计数文本
    countLabel: {
      type: Object,
      default: () => ({})
    },
    bar: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["title-click"],
  setup(e, { emit: s }) {
    const i = j(() => e.cStyle), d = R("chart-stacked-curve"), u = M(null);
    let t = null;
    const r = H({
      option: {},
      // 图表数据
      chartData: [],
      selectedLegendData: []
    });
    function C() {
      var a, f;
      (f = (a = e.seriesSetting) == null ? void 0 : a.data) != null && f.length && e.data && (r.chartData = e.seriesSetting.data.map((S) => {
        const { name: m, dataKey: g } = S;
        return {
          name: m || "--",
          data: e.data.map((c) => c[g])
        };
      }), r.selectedLegendData = r.chartData), r.selectedLegendData = e.options.series || r.chartData;
    }
    const _ = () => {
      let a = "";
      return i.value.wrapper && (a += L(i.value.wrapper, `.${d}`)), i.value.emptyContainer && (a += L(i.value.emptyContainer, `.${d} .empty`)), i.value.emptyImg && (a += L(i.value.emptyImg, `.${d} .empty .empty-img`)), i.value.emptyText && (a += L(i.value.emptyText, `.${d} .empty .empty-text`)), a;
    }, k = () => {
      const a = document.querySelector(`.${d} .style-container`);
      a && (a.innerHTML = `<style>${_()}</style>`);
    }, N = j(() => {
      var a, f, S, m;
      if (e.options) {
        if (Array.isArray((a = e.options) == null ? void 0 : a.series))
          return e.options.series.some((c) => {
            var b;
            return (c == null ? void 0 : c.data) && ((b = c == null ? void 0 : c.data) == null ? void 0 : b.length) !== 0;
          });
        if (typeof ((f = e.options) == null ? void 0 : f.series) == "object")
          return ((m = (S = e.options.series) == null ? void 0 : S.data) == null ? void 0 : m.length) === 0;
      }
      return !!e.data.length;
    });
    function O() {
      var c, b, $, I, T;
      t.clear();
      const { color: a, series: f, xAxis: S } = e.options;
      let m = [], g = {};
      f ? m = f.map((y, l) => {
        var F, x, w;
        o.legend = n(o.legend, e.options.legend), o.legend.data.push({
          name: y.name,
          itemStyle: {
            color: a && a[l] || o.color[l] || "#FFFFFF00"
          }
        }), o.legend.data[l].name = y.name;
        const h = {
          type: "line",
          symbolSize: 0,
          showSymbol: !1,
          smooth: !0,
          lineStyle: {
            color: a && a[l] || o.color[l] || "#FFFFFF00",
            // 100% 处的颜色
            width: 1.5
          }
        };
        l === f.length - 1 && (h.showSymbol = !0, h.lineStyle.type = "dashed", h.label = {
          color: a && a[l] || o.color[l] || "#FFFFFF00",
          show: !0,
          fontSize: "20px",
          position: [5, 3],
          formatter: (A) => A.dataIndex === S.data.length - 1 ? A.data : ""
        }), y.noAreaStyle || (h.areaStyle = {
          color: {
            type: "linear",
            x: 0,
            y: 0,
            x2: 0,
            y2: 1,
            colorStops: [
              {
                offset: 0,
                color: a && a[l] || o.color[l] || "#FFFFFF00"
                // 100% 处的颜色
              },
              {
                offset: 0.7,
                color: "transparent"
                // 100% 处的颜色
              }
            ]
          }
        });
        const p = n(
          n(
            h,
            e == null ? void 0 : e.bar
          ),
          (w = (x = (F = e.seriesSetting) == null ? void 0 : F.data) == null ? void 0 : x[l]) == null ? void 0 : w.style
        );
        return n({}, p, y);
      }) : e.data && ((b = (c = e.seriesSetting) == null ? void 0 : c.data) != null && b.length) && (m = ($ = e.seriesSetting) == null ? void 0 : $.data.map((y, l) => {
        var p, F, x, w, A;
        const h = n(
          n(
            {
              name: (p = r.chartData[l]) == null ? void 0 : p.name,
              type: "line",
              symbolSize: 0,
              showSymbol: !1,
              smooth: !0,
              lineStyle: {
                color: a && a[l] || o.color[l] || "#FFFFFF00",
                // 100% 处的颜色
                width: 1.5
              },
              areaStyle: {
                color: {
                  type: "linear",
                  x: 0,
                  y: 0,
                  x2: 0,
                  y2: 1,
                  colorStops: [
                    {
                      offset: 0,
                      color: a && a[l] || o.color[l] || "#FFFFFF00"
                      // 100% 处的颜色
                    },
                    {
                      offset: 0.7,
                      color: "transparent"
                      // 100% 处的颜色
                    }
                  ]
                }
              },
              data: e.data.length ? (F = r.chartData[l]) == null ? void 0 : F.data : []
            },
            e == null ? void 0 : e.bar
          ),
          (A = (w = (x = e.seriesSetting) == null ? void 0 : x.data) == null ? void 0 : w[l]) == null ? void 0 : A.style
        );
        return n({}, h, y);
      }), (I = e.seriesSetting) != null && I.xAxisLabel ? g = (T = e.data) != null && T.length ? e.data.map((y) => y[e.seriesSetting.xAxisLabel]) : [] : g = e.data.map((y, l) => l)), e.mainAxis === "x" ? r.option = n(
        {
          series: m,
          xAxis: {
            data: g
          }
        },
        o
      ) : e.mainAxis === "y" && (r.option = n(
        {
          series: m,
          yAxis: {
            data: g
          }
          // xAxis: props.options.yAxis,
        },
        o
      )), r.option = n(r.option, e.options), e.mainAxis === "y" && (r.option.xAxis.type = "value", r.option.yAxis.type = "category");
      try {
        v(() => {
          t.setOption(r.option), t.resize();
        });
      } catch (y) {
        console.log(y);
      }
    }
    const z = () => {
      v(() => {
        t.resize();
      });
    };
    return P(() => {
      k(), u.value && (t = G.init(u.value), C(), O()), window.addEventListener("resize", z);
    }), U(() => {
      window.removeEventListener("resize", z);
    }), E(() => e.cStyle, () => {
      k(), v(() => {
        t.resize();
      });
    }, { deep: !0 }), E(
      [() => e.data, () => e.seriesSetting, () => e.options],
      () => {
        C(), O();
      },
      { deep: !0 }
    ), {
      curStyle: i,
      domOnlyClass: d,
      chartId: u,
      checkEmpty: N
    };
  }
});
export {
  Z as default
};
