// @ths/design/es/components/chart-stacked-curve/src/index.vue.js
import l from "./index.vue2.js";
import { openBlock as c, createElementBlock as i, normalizeClass as a, withDirectives as o, createElementVNode as e, vShow as r, normalizeStyle as n } from "vue";
import "./index.vue3.js";
import m from "../../../_virtual/_plugin-vue_export-helper.js";
const p = {
  ref: "chartId",
  class: "chart"
}, d = { class: "empty" }, u = ["src"];
function f(t, s, y, h, v, k) {
  return c(), i("div", {
    class: a(["t-chart-stacked-curve", t.domOnlyClass])
  }, [
    o(e("div", p, null, 512), [
      [r, t.checkEmpty]
    ]),
    o(e("div", d, [
      e("img", {
        class: "empty-img",
        style: n({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, u),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [r, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 2);
}
const g = /* @__PURE__ */ m(l, [["render", f]]);
export {
  g as default
};
