// @ths/design/es/components/dialog/src/index.vue.js
import F from "./index.vue2.js";
import { openBlock as r, createBlock as u, Teleport as O, createElementBlock as i, createElementVNode as n, normalizeStyle as s, Fragment as j, withModifiers as l, createCommentVNode as d, toDisplayString as a, resolveDynamicComponent as A, renderSlot as q } from "vue";
import "./index.vue3.js";
import G from "../../../_virtual/_plugin-vue_export-helper.js";
const I = ["data-id"], J = { class: "header-container" };
function K(e, o, P, Q, U, W) {
  var m, v, y, f, p, g, M, z, $, S, k, b, B, C, w, H, R, D, N, T, E, V, L;
  return r(), u(O, {
    to: e.mountTo || "body"
  }, [
    e.dialogVisible ? (r(), i("div", {
      key: 0,
      ref: "dialogContainerRef",
      class: "t-dialog overlay",
      "data-id": e.$attrs["data-id"]
    }, [
      n("div", {
        class: "mask",
        style: s({ ...(v = (m = e.currentStyle) == null ? void 0 : m.mask) == null ? void 0 : v.default, pointerEvents: e.isDragging ? "all" : ((p = (f = (y = e.currentStyle) == null ? void 0 : y.mask) == null ? void 0 : f.default) == null ? void 0 : p.pointerEvents) || "none" })
      }, null, 4),
      n("div", {
        ref: "dialogRef",
        class: "dialog draggable",
        style: s({ ...(M = (g = e.currentStyle) == null ? void 0 : g.wrapper) == null ? void 0 : M.default, ...e.dialogStyle })
      }, [
        e.resizable ? (r(), i(j, { key: 0 }, [
          n("div", {
            class: "resize-handle top",
            onMousedown: o[0] || (o[0] = l((t) => e.startResize("top"), ["stop"]))
          }, null, 32),
          n("div", {
            class: "resize-handle right",
            onMousedown: o[1] || (o[1] = l((t) => e.startResize("right"), ["stop"]))
          }, null, 32),
          n("div", {
            class: "resize-handle bottom",
            onMousedown: o[2] || (o[2] = l((t) => e.startResize("bottom"), ["stop"]))
          }, null, 32),
          n("div", {
            class: "resize-handle left",
            onMousedown: o[3] || (o[3] = l((t) => e.startResize("left"), ["stop"]))
          }, null, 32),
          n("div", {
            class: "resize-handle top-left",
            onMousedown: o[4] || (o[4] = l((t) => e.startResize("top-left"), ["stop"]))
          }, null, 32),
          n("div", {
            class: "resize-handle top-right",
            onMousedown: o[5] || (o[5] = l((t) => e.startResize("top-right"), ["stop"]))
          }, null, 32),
          n("div", {
            class: "resize-handle bottom-left",
            onMousedown: o[6] || (o[6] = l((t) => e.startResize("bottom-left"), ["stop"]))
          }, null, 32),
          n("div", {
            class: "resize-handle bottom-right",
            onMousedown: o[7] || (o[7] = l((t) => e.startResize("bottom-right"), ["stop"]))
          }, null, 32)
        ], 64)) : d("", !0),
        n("div", {
          class: "title-bar",
          style: s(e.dialogMinimized ? { ...($ = (z = e.currentStyle) == null ? void 0 : z.header) == null ? void 0 : $.default, ...e.minimizedHeaderStyle } : (k = (S = e.currentStyle) == null ? void 0 : S.header) == null ? void 0 : k.default),
          onMousedown: o[21] || (o[21] = l((...t) => e.startDrag && e.startDrag(...t), ["stop"]))
        }, [
          n("div", J, [
            n("div", {
              class: "title",
              style: s(e.dialogMinimized ? (B = (b = e.currentStyle) == null ? void 0 : b.minimizedTitle) == null ? void 0 : B.default : (w = (C = e.currentStyle) == null ? void 0 : C.title) == null ? void 0 : w.default)
            }, a(e.dialogMinimized ? e.minimizedTitle : e.title), 5),
            n("div", {
              style: s((R = (H = e.currentStyle) == null ? void 0 : H.headerCustomContainer) == null ? void 0 : R.default),
              onMousedown: o[8] || (o[8] = l(() => {
              }, ["stop"]))
            }, [
              e.headerLoaded && e.headerComName ? (r(), u(A(e.headerComName), { key: 0 })) : d("", !0)
            ], 36)
          ]),
          n("div", {
            class: "controls",
            style: s([{ display: "flex" }, (N = (D = e.currentStyle) == null ? void 0 : D.buttonControls) == null ? void 0 : N.default])
          }, [
            e.dialogMinimized ? (r(), i("button", {
              key: 0,
              style: s(e.maxButtonStyle),
              onClick: o[9] || (o[9] = l((...t) => e.restore && e.restore(...t), ["stop"])),
              onMousedown: o[10] || (o[10] = l(() => {
              }, ["stop"])),
              onMouseenter: o[11] || (o[11] = (t) => e.maxButtonHover = !0),
              onMouseleave: o[12] || (o[12] = (t) => e.maxButtonHover = !1)
            }, " □ ", 36)) : (r(), i("button", {
              key: 1,
              style: s(e.minButtonStyle),
              onClick: o[13] || (o[13] = l((...t) => e.minimize && e.minimize(...t), ["stop"])),
              onMousedown: o[14] || (o[14] = l(() => {
              }, ["stop"])),
              onMouseenter: o[15] || (o[15] = (t) => e.minButtonHover = !0),
              onMouseleave: o[16] || (o[16] = (t) => e.minButtonHover = !1)
            }, " - ", 36)),
            n("button", {
              style: s(e.closeButtonStyle),
              onClick: o[17] || (o[17] = l((...t) => e.close && e.close(...t), ["stop"])),
              onMousedown: o[18] || (o[18] = l(() => {
              }, ["stop"])),
              onMouseenter: o[19] || (o[19] = (t) => e.closeButtonHover = !0),
              onMouseleave: o[20] || (o[20] = (t) => e.closeButtonHover = !1)
            }, " × ", 36)
          ], 4)
        ], 36),
        n("div", {
          class: "content",
          style: s((E = (T = e.currentStyle) == null ? void 0 : T.content) == null ? void 0 : E.default)
        }, [
          n("div", {
            class: "scale-container",
            style: s(e.isContentScale ? {} : { width: "100%", height: "100%" })
          }, [
            Object.keys(e.contentComName).length > 0 ? (r(), i(j, { key: 0 }, [
              e.contentLoaded && e.contentComName ? (r(), u(A(e.contentComName), { key: 0 })) : d("", !0)
            ], 64)) : q(e.$slots, "content", { key: 1 })
          ], 4)
        ], 4),
        n("div", {
          class: "footer",
          style: s((L = (V = e.currentStyle) == null ? void 0 : V.footer) == null ? void 0 : L.default)
        }, [
          n("button", {
            style: s(e.cancelButtonStyle),
            onClick: o[22] || (o[22] = (...t) => e.close && e.close(...t)),
            onMouseenter: o[23] || (o[23] = (t) => e.cancelButtonHover = !0),
            onMouseleave: o[24] || (o[24] = (t) => e.cancelButtonHover = !1)
          }, a(e.cancelText), 37),
          n("button", {
            style: s(e.confirmButtonStyle),
            onClick: o[25] || (o[25] = (...t) => e.confirm && e.confirm(...t)),
            onMouseenter: o[26] || (o[26] = (t) => e.confirmButtonHover = !0),
            onMouseleave: o[27] || (o[27] = (t) => e.confirmButtonHover = !1)
          }, a(e.confirmText), 37)
        ], 4)
      ], 4)
    ], 8, I)) : d("", !0)
  ], 8, ["to"]);
}
const x = /* @__PURE__ */ G(F, [["render", K]]);
export {
  x as default
};
