// @ths/design/es/components/dialog/src/index.vue2.js
import { defineComponent as Xe, ref as s, watch as O, provide as Ye, reactive as $e, computed as y, nextTick as P, onMounted as je, onUpdated as Ve, onBeforeUnmount as _e, toRefs as Ne } from "vue";
import { cloneDeep as G } from "lodash";
import Ue from "lodash/merge";
import { ComponentLoader as xe } from "../../../utils/loader.js";
const J = {
  // 对话框样式
  wrapper: {
    default: {
      width: "500px",
      height: "auto",
      border: "1px solid #ccc",
      backgroundColor: "white",
      pointerEvents: "all",
      zIndex: 9999
    }
  },
  // 蒙版样式
  mask: {
    default: {
      position: "absolute",
      width: "100%",
      height: "100%",
      top: "0",
      left: "0",
      backgroundColor: "rgba(0,0,0,.1)",
      pointerEvents: "none",
      zIndex: 9999
    }
  },
  // 对话框头部、顶栏样式
  header: {
    default: {
      display: "flex",
      justifyContent: "space-between",
      alignItems: "center",
      padding: "5px 10px",
      backgroundColor: "#f0f0f0",
      cursor: "move"
    }
  },
  // 对话框顶栏中的标题样式
  title: {
    default: {}
  },
  // 对话框顶栏中的最小化的标题样式
  minimizedTitle: {
    default: {}
  },
  // 标题栏自定义容器
  headerCustomContainer: {
    default: {}
  },
  // 对话框顶栏中的右侧按钮容器样式
  buttonControls: {
    default: {},
    hover: { color: "red" }
  },
  // 对话框顶栏中的关闭按钮样式
  closeButton: {
    default: {
      paddingLeft: "10px"
    },
    hover: { color: "var(--t-brand8)" }
  },
  // 对话框顶栏中的最小化按钮样式
  minButton: {
    default: {
      marginLeft: "5px",
      cursor: "pointer"
    },
    hover: { color: "var(--t-brand8)" }
  },
  // 对话框顶栏中的最大化按钮样式
  maxButton: {
    default: {
      marginLeft: "5px",
      cursor: "pointer"
    },
    hover: { color: "var(--t-brand8)" }
  },
  // 对话框内容区域样式
  content: {
    default: {
      padding: "10px",
      width: "100%",
      height: "200px"
    }
  },
  // 对话框底栏样式
  footer: {
    default: {
      display: "flex",
      justifyContent: "center",
      alignItems: "center",
      padding: "5px",
      backgroundColor: "#f0f0f0",
      fontSize: "14px",
      cursor: "pointer"
    }
  },
  // 对话框底栏中的确认按钮样式
  confirmButton: {
    default: {
      marginLeft: "15px",
      border: "1px solid #ccc",
      padding: "0 6px",
      borderRadius: "4px",
      backgroundColor: "var(--t-brand3)"
    },
    hover: {
      color: "var(--t-white)"
    }
  },
  // 对话框底栏中的取消按钮样式
  cancelButton: {
    default: {
      marginLeft: "5px",
      border: "1px solid #ccc",
      padding: "0 6px",
      borderRadius: "4px"
    },
    hover: { color: "var(--t-brand8)" }
  }
}, be = {
  width: "400px",
  height: "auto",
  left: "calc(50% - 200px)",
  top: "calc(50% - 150px)"
}, Se = {
  width: "200px",
  height: "30px",
  right: "100px",
  bottom: "100px",
  overflow: "hidden"
}, Ze = {
  width: "100%",
  height: "auto",
  right: "100px",
  bottom: "100px"
}, et = Xe({
  name: "TDialog",
  props: {
    // 挂载的节点 | 默认为 body
    mountTo: {
      type: String,
      default: "body"
    },
    // 显示状态
    dialogVisible: {
      type: Boolean,
      default: !1
    },
    // 最小化状态
    dialogMinimized: {
      type: Boolean,
      default: !1
    },
    // 最大化状态
    dialogMaximized: {
      type: Boolean,
      default: !1
    },
    // 标题
    title: {
      type: String,
      default: "标题"
    },
    // 最小化的标题
    minimizedTitle: {
      type: String,
      default: "最小化标题"
    },
    // 取消按钮的文本
    cancelText: {
      type: String,
      default: "取消"
    },
    // 确认按钮的文本
    confirmText: {
      type: String,
      default: "确认"
    },
    // 标题栏区域的组件
    headerComName: {
      type: String,
      default: ""
    },
    // 内容区域的组件
    contentComName: {
      type: String,
      default: ""
    },
    // 最大化时的宽高、定位
    maxPosition: {
      type: Object,
      default: () => be
    },
    // 最小化时的宽高、定位
    minPosition: {
      type: Object,
      default: () => Se
    },
    // 最小化时头部的样式
    minimizedHeaderStyle: {
      type: Object,
      default: () => Ze
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => J
    },
    // 内容是否缩放
    isContentScale: {
      type: Boolean,
      default: !1
    },
    // 开启后允许所有边/角拖拽
    resizable: {
      type: Boolean,
      default: !0
    }
  },
  // 更新dialogVisible变量 | 关闭窗口 | 确认 | 最小化 | 还原
  emits: ["update:dialogVisible", "close", "confirm", "update:dialogMinimized", "update:dialogMaximized", "minimize", "restore"],
  setup(a, { emit: f }) {
    const K = s(!1), Q = s(!1);
    let p = 1;
    O(() => a.headerComName, (e) => {
      e && xe.loadComponent(e.replace("Component", ""), () => {
        K.value = !0;
      });
    }, { immediate: !0 }), O(() => a.contentComName, (e) => {
      e && xe.loadComponent(e.replace("Component", ""), () => {
        Q.value = !0;
      });
    }, { immediate: !0 }), Ye("root", window.__lego_root);
    const d = $e({
      closeButtonHover: !1,
      minButtonHover: !1,
      maxButtonHover: !1,
      confirmButtonHover: !1,
      cancelButtonHover: !1,
      isDragging: !1,
      contentWidth: 0,
      isContentScale: y(() => a.isContentScale)
      // 内容是否缩放
    }), ze = y(() => Ue({}, J, a.cStyle)), k = (e) => {
      var u, m;
      let n = J[e];
      (u = a.cStyle) != null && u[e] && (n = { ...n, ...(m = a.cStyle) == null ? void 0 : m[e] });
      const o = n.hover || {}, i = { ...n.default }, r = { ...i, ...o };
      let l = i;
      return d[`${e}Hover`] && o && Object.keys(o).length && (l = r), l;
    }, Ce = y(() => k("closeButton")), we = y(() => k("minButton")), Me = y(() => k("maxButton")), Ee = y(() => k("confirmButton")), Be = y(() => k("cancelButton")), ee = y(() => a.maxPosition), He = y(() => a.minPosition), x = s({
      position: "absolute",
      ...ee.value,
      transform: "none"
      // 确保不使用 transform
    }), te = y(() => ({
      position: "absolute",
      ...be,
      ...ee.value
    })), We = y(() => ({
      position: "fixed",
      ...Se,
      ...He.value
    })), t = s(null), c = s(null), w = s(!1), F = s(!1), $ = s(""), I = s(0), D = s(0), b = s(0), S = s(0), v = s(0), h = s(0), M = 200, E = 100, ne = () => {
      var e, n, o, i;
      if (d.contentWidth = ((o = (n = (e = t.value) == null ? void 0 : e.querySelector(".content")) == null ? void 0 : n.getBoundingClientRect()) == null ? void 0 : o.width) ?? 0, d.contentWidth > 0 && ((i = t.value) != null && i.querySelector(".content"))) {
        const r = t.value.querySelector(".content"), l = window.getComputedStyle(r), u = parseInt(l.paddingLeft, 10) || 0, m = parseInt(l.paddingRight, 10) || 0;
        d.contentWidth -= u + m;
      }
    };
    async function X() {
      d.isContentScale && (ne(), P(() => {
        var C, T, R, A;
        const e = (C = t.value) == null ? void 0 : C.querySelector(".scale-container");
        if (!e)
          return;
        const n = e == null ? void 0 : e.offsetWidth, o = d.contentWidth, i = e == null ? void 0 : e.offsetHeight, r = ((A = (R = (T = t.value) == null ? void 0 : T.querySelector(".content")) == null ? void 0 : R.getBoundingClientRect()) == null ? void 0 : A.height) ?? 0, l = o / n, u = r / i, m = Math.min(l, u);
        let g = "";
        const z = "50% 50%";
        g = `scale(${m})`, e.style.transform = g, e.style.transformOrigin = z;
      }));
    }
    const j = s([]), V = () => {
      t.value && (j.value = Array.from(t.value.querySelectorAll("iframe")), j.value.forEach((e) => {
        e.dataset.originalSrc = e.src;
      }));
    }, _ = () => {
      j.value = [];
    }, B = () => {
      c != null && c.value && (V(), requestAnimationFrame(() => {
        const e = document.querySelector((a == null ? void 0 : a.mountTo) || "body"), n = e ? e.querySelectorAll(".t-dialog") : [];
        let o = 1001;
        if (n.forEach((i) => {
          const r = window.getComputedStyle(i), l = parseInt(r.zIndex, 10);
          !isNaN(l) && l > o && (o = l);
        }), c.value) {
          c.value.style.zIndex = `${o + 1}`;
          const i = c.value.getAttribute("data-id") || "";
          document.querySelectorAll(".t-dialog").forEach((l) => {
            l.classList.remove("t-dialog-active");
          }), c.value.classList.add("t-dialog-active");
          const r = new CustomEvent("dialog-activated", {
            detail: { id: i, zIndex: o + 1 }
          });
          document.dispatchEvent(r);
        }
        _();
      }));
    }, Le = () => {
      c != null && c.value && (V(), requestAnimationFrame(() => {
        const e = document.createDocumentFragment();
        c.value && e.appendChild(c.value);
        const n = document.querySelector((a == null ? void 0 : a.mountTo) || "body");
        n && n.appendChild(e), _();
      }));
    }, qe = () => {
      if (!(c != null && c.value))
        return;
      const e = document.querySelector((a == null ? void 0 : a.mountTo) || "body");
      !e || !e.contains(c.value) || (V(), requestAnimationFrame(() => {
        c.value && e.removeChild(c.value), _();
      }));
    }, ke = () => {
      const e = document.querySelector((a == null ? void 0 : a.mountTo) || "body");
      if (!e)
        return 1;
      const { transform: n } = window.getComputedStyle(e);
      if (n && n !== "none") {
        const o = n.match(/matrix(3d)?\(([-\d\.eE, ]+)\)/);
        if (o) {
          const i = o[2].split(",");
          return parseFloat(i[0]);
        }
      }
      return 1;
    }, Y = (e) => {
      p = ke(), d.isDragging = e;
    }, H = s(null);
    let oe = 0, ae = 0, W = null;
    const N = () => {
      H.value && t.value && (t.value.contains(H.value) && t.value.removeChild(H.value), H.value = null);
    }, Ie = () => {
      if (!t.value)
        return;
      N(), H.value = document.createElement("div");
      const e = H.value;
      switch (e.style.position = "absolute", e.style.top = "0", e.style.left = "0", e.style.width = "100%", e.style.height = "100%", e.style.zIndex = "10000", e.style.background = "transparent", $.value) {
        case "top":
        case "bottom":
          e.style.cursor = "ns-resize";
          break;
        case "left":
        case "right":
          e.style.cursor = "ew-resize";
          break;
        case "top-left":
        case "bottom-right":
          e.style.cursor = "nwse-resize";
          break;
        case "top-right":
        case "bottom-left":
          e.style.cursor = "nesw-resize";
          break;
      }
      t.value.appendChild(e);
    }, De = (e) => {
      if (!F.value || !t.value)
        return;
      const n = e.clientX - I.value, o = e.clientY - D.value, i = oe || document.documentElement.clientWidth, r = ae || document.documentElement.clientHeight;
      let l = v.value, u = h.value, m = b.value, g = S.value;
      switch ($.value) {
        case "right":
          l = Math.max(v.value + n, M);
          break;
        case "bottom":
          u = Math.max(h.value + o, E);
          break;
        case "left":
          const T = Math.min(n, v.value - M);
          l = v.value - T, m = b.value + T;
          break;
        case "top":
          const R = Math.min(o, h.value - E);
          u = h.value - R, g = S.value + R;
          break;
        case "top-left":
          const A = Math.min(n, v.value - M), ge = Math.min(o, h.value - E);
          l = v.value - A, u = h.value - ge, m = b.value + A, g = S.value + ge;
          break;
        case "top-right":
          l = Math.max(v.value + n, M);
          const ye = Math.min(o, h.value - E);
          u = h.value - ye, g = S.value + ye;
          break;
        case "bottom-left":
          const pe = Math.min(n, v.value - M);
          l = v.value - pe, u = Math.max(h.value + o, E), m = b.value + pe;
          break;
        case "bottom-right":
          l = Math.max(v.value + n, M), u = Math.max(h.value + o, E);
          break;
      }
      m = Math.max(0, Math.min(m, i - l)), g = Math.max(0, Math.min(g, r - u));
      const { style: z } = t.value;
      z.width = `${l}px`, z.height = `${u}px`, z.left = `${m}px`, z.top = `${g}px`;
      const C = t.value.querySelector(".content");
      C && (C.style.overflow = "hidden"), d.isContentScale && X();
    }, le = (e) => {
      W && cancelAnimationFrame(W), W = requestAnimationFrame(() => De(e));
    }, ie = () => {
      if (Y(!1), !!F.value) {
        if (F.value = !1, document.removeEventListener("mousemove", le), document.removeEventListener("mouseup", ie), W && (cancelAnimationFrame(W), W = null), N(), t.value) {
          const e = t.value.querySelector(".content");
          e && (e.style.overflow = ""), x.value = {
            ...x.value,
            width: t.value.style.width,
            height: t.value.style.height,
            left: t.value.style.left,
            top: t.value.style.top
          };
        }
        d.isContentScale && X(), document.body.style.removeProperty("user-select");
      }
    }, Te = (e) => {
      Y(!0), t.value && (document.body.style.userSelect = "none", F.value = !0, $.value = e, B(), Ie(), I.value = (event == null ? void 0 : event.clientX) || 0, D.value = (event == null ? void 0 : event.clientY) || 0, b.value = t.value.offsetLeft, S.value = t.value.offsetTop, v.value = t.value.offsetWidth, h.value = t.value.offsetHeight, oe = document.documentElement.clientWidth, ae = document.documentElement.clientHeight, document.addEventListener("mousemove", le, { passive: !0 }), document.addEventListener("mouseup", ie));
    }, L = s(null), U = () => {
      L.value && t.value && (t.value.contains(L.value) && t.value.removeChild(L.value), L.value = null);
    }, Re = () => {
      if (!t.value)
        return;
      U(), L.value = document.createElement("div");
      const e = L.value;
      e.style.position = "absolute", e.style.top = "0", e.style.left = "0", e.style.width = "100%", e.style.height = "100%", e.style.zIndex = "10000", e.style.background = "transparent", e.style.cursor = "move", t.value.appendChild(e);
    };
    let ce = 0, ue = 0, se = 0, re = 0;
    const Ae = (e) => {
      if (!w.value || !t.value)
        return;
      const n = e.clientX / p - I.value, o = e.clientY / p - D.value, i = b.value + n, r = S.value + o, l = (ce || document.documentElement.clientWidth) / p, u = (ue || document.documentElement.clientHeight) / p, m = (se || t.value.offsetWidth) / 1, g = (re || t.value.offsetHeight) / 1, z = Math.max(0, Math.min(i, l - m)), C = Math.max(0, Math.min(r, u - g));
      t.value.style.left = `${z}px`, t.value.style.top = `${C}px`;
    };
    let q = null;
    const de = (e) => {
      q && cancelAnimationFrame(q), q = requestAnimationFrame(() => Ae(e));
    }, me = () => {
      if (Y(!1), !w.value)
        return;
      w.value = !1, document.removeEventListener("mousemove", de), document.removeEventListener("mouseup", me), U(), q && (cancelAnimationFrame(q), q = null), t.value && (x.value = {
        ...x.value,
        left: t.value.style.left,
        top: t.value.style.top
      });
      const e = document.querySelector((a == null ? void 0 : a.mountTo) || "body");
      e ? (e.style.cursor = "default", e.style.removeProperty("user-select")) : (document.body.style.cursor = "default", document.body.style.removeProperty("user-select"));
    }, Oe = (e) => {
      Y(!0);
      const n = document.querySelector((a == null ? void 0 : a.mountTo) || "body");
      if (n ? (n.style.userSelect = "none", n.style.cursor = "move") : (document.body.style.userSelect = "none", document.body.style.cursor = "move"), Re(), B(), w.value = !0, I.value = e.clientX / p, D.value = e.clientY / p, t.value) {
        const o = t.value.getBoundingClientRect();
        b.value = o.left / p, S.value = o.top / p, ce = document.documentElement.clientWidth, ue = document.documentElement.clientHeight, se = t.value.offsetWidth, re = t.value.offsetHeight;
      }
      document.addEventListener("mousemove", de, { passive: !0 }), document.addEventListener("mouseup", me);
    }, Z = () => {
      d.closeButtonHover = !1, d.minButtonHover = !1, d.maxButtonHover = !1, d.confirmButtonHover = !1, d.cancelButtonHover = !1;
    }, fe = () => {
      x.value = G(We.value), Z(), f("minimize"), f("update:dialogMinimized", !0), f("update:dialogMaximized", !1);
    }, ve = () => {
      x.value = G(te.value), Z(), f("restore"), f("update:dialogMinimized", !1), f("update:dialogMaximized", !0);
    }, he = () => {
      x.value = G(te.value), Z(), f("update:dialogMinimized", !1), f("update:dialogMaximized", !1), w.value = !1;
    }, Pe = () => {
      f("close"), f("update:dialogVisible", !1), f("update:dialogMinimized", !1), he();
    }, Fe = () => {
      f("confirm"), f("update:dialogVisible", !1), f("update:dialogMinimized", !1), he();
    };
    return O(
      () => a.dialogVisible,
      (e) => {
        e && P(() => {
          var n, o, i;
          if (Le(), B(), ne(), a.isContentScale) {
            const r = (o = (n = t.value) == null ? void 0 : n.querySelector(".scale-container")) == null ? void 0 : o.getBoundingClientRect().width, l = (i = t.value) == null ? void 0 : i.querySelector(".scale-container");
            l && r && (l.style.width = `${r}px`, X());
          }
        });
      },
      { immediate: !0 }
    ), O(
      () => a.dialogMinimized,
      (e) => {
        e && fe();
      }
    ), O(
      () => a.dialogMaximized,
      (e) => {
        e && ve();
      }
    ), je(() => {
      P(() => {
        B();
      });
    }), Ve(() => {
      P(() => {
        B();
      });
    }), _e(() => {
      P(() => {
        qe(), U(), N();
      });
    }), {
      headerLoaded: K,
      contentLoaded: Q,
      currentStyle: ze,
      ...Ne(d),
      closeButtonStyle: Ce,
      minButtonStyle: we,
      maxButtonStyle: Me,
      confirmButtonStyle: Ee,
      cancelButtonStyle: Be,
      close: Pe,
      confirm: Fe,
      minimize: fe,
      restore: ve,
      startDrag: Oe,
      dialogRef: t,
      dragging: w,
      initialMouseX: I,
      initialMouseY: D,
      initialDialogX: b,
      initialDialogY: S,
      dialogContainerRef: c,
      dialogStyle: x,
      startResize: Te,
      moveToTop: B,
      setPageScale: X
    };
  }
});
export {
  et as default
};
