// @ths/design/es/components/quality-monitor/src/index.vue2.js
import { defineComponent as s, computed as a, onMounted as u, watch as d } from "vue";
import { getDomRandomClass as y, setStyle as m } from "../../../utils/index.js";
const S = s({
  name: "TQualityMonitor",
  props: {
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        listBody: {
          default: {}
        },
        card: {
          default: {}
        },
        notLastCard: {
          default: {}
        }
      })
    }
  },
  emits: ["title-click"],
  setup(e, { emit: r }) {
    const l = a(() => e.cStyle), o = y("quality-monitor"), i = a(() => e.data), c = () => {
      let t = "";
      return l.value.wrapper && (t += m(l.value.wrapper, `.${o} .t-quality-monitor`)), t;
    }, n = () => {
      const t = document.querySelector(`.${o} .style-container`);
      t && (t.innerHTML = `<style>${c()}</style>`);
    };
    return u(() => {
      n();
    }), d(() => e.cStyle, () => {
      n();
    }, { deep: !0 }), {
      curStyle: l,
      domOnlyClass: o,
      titleClick: () => {
        r("title-click");
      },
      dataList: i
    };
  }
});
export {
  S as default
};
