// @ths/design/es/components/chart-map-base/src/index.vue.js
import y from "./index.vue2.js";
import { openBlock as a, createElementBlock as l, normalizeClass as f, normalizeStyle as n, withDirectives as c, createElementVNode as t, Fragment as g, renderList as h, toDisplayString as r, createCommentVNode as v, vShow as d } from "vue";
import "./index.vue3.js";
import k from "../../../_virtual/_plugin-vue_export-helper.js";
const E = {
  ref: "mapBase",
  class: "map-box"
}, _ = { class: "item-name" }, S = { class: "item-value" }, b = { class: "item-value" }, C = { class: "empty" }, B = ["src"];
function D(e, s, L, $, j, w) {
  var i, m;
  return a(), l("div", {
    class: f(["t-chart-map-base", e.domOnlyClass]),
    style: n({ ...(m = (i = e.curStyle) == null ? void 0 : i.wrapper) == null ? void 0 : m.default })
  }, [
    c(t("main", null, [
      t("div", E, null, 512),
      Object.keys(e.mapData).length ? (a(), l("div", {
        key: 0,
        class: "legend-list",
        style: n({ ...e.curLegendContainer })
      }, [
        (a(!0), l(g, null, h(e.chartData, (o, u) => {
          var p;
          return a(), l("div", {
            key: u,
            class: "legend-list__item",
            style: n({ ...e.curLegendStyle })
          }, [
            t("span", _, r(o.name), 1),
            t("span", S, r(o.percent) + "%", 1),
            t("span", b, r(o.value) + r(((p = e.curLegendStyle) == null ? void 0 : p.unit) || "个"), 1)
          ], 4);
        }), 128))
      ], 4)) : v("", !0)
    ], 512), [
      [d, e.checkEmpty]
    ]),
    c(t("div", C, [
      t("img", {
        class: "empty-img",
        style: n({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, B),
      s[0] || (s[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [d, !e.checkEmpty]
    ]),
    s[1] || (s[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const V = /* @__PURE__ */ k(y, [["render", D]]);
export {
  V as default
};
