// @ths/design/es/components/chart-map-base/src/index.vue2.js
import { defineComponent as q, computed as w, ref as H, reactive as J, onMounted as B, onUnmounted as R, watch as M, nextTick as F, toRefs as U } from "vue";
import * as P from "echarts";
import s from "lodash/merge";
import G from "lodash/cloneDeep";
import { DEFAULT_OPTIONS as Q } from "../../../utils/constant.js";
import { getDomRandomClass as V, setStyle as v } from "../../../utils/index.js";
const z = s(
  G(Q),
  {
    // color: ['#BCEFBB', '#A3E1A9', '#5BB774', '#018432', '#199949'], 设计图颜色
    tooltip: {
      trigger: "item",
      axisPointer: {
        type: "none"
      },
      formatter: "{b}: {c}"
    },
    legend: {
      show: !1
    },
    grid: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    },
    visualMap: {
      type: "piecewise",
      show: !1,
      realtime: !1,
      calculable: !1
    },
    geo: {
      map: "chartMap",
      // 地图类型
      zoom: 1.1,
      roam: !1,
      animation: !1,
      silent: !0,
      itemStyle: {
        // 区域样式
        areaColor: "#fff",
        borderColor: "#BEC8C0",
        borderWidth: 4
      }
    }
  }
), ae = q({
  name: "TChartMapBase",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件容器的样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => z,
      required: !1
    },
    // 传入数组数据； 如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据 - 地图
    mapData: {
      type: Object,
      default: () => ({})
    },
    // 别名 及 样式设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 地图图形的配置
        mapStyle: {
          type: Object,
          default: () => ({})
        },
        // 图例列表的配置
        legendStyle: {
          type: Object,
          default: () => ({})
        },
        // 取resData数据中的别名设置
        data: {
          nameKey: "",
          // 名称别名
          dataKey: "",
          // 列表数据别名
          mapDataKey: ""
          // 地图数据别名
        }
      })
    },
    // 地图主题设置
    map: {
      type: Object,
      default: () => ({})
    }
  },
  setup(e) {
    const n = w(() => e.cStyle), W = w(() => {
      var t, a;
      return (a = (t = e.seriesSetting) == null ? void 0 : t.legendStyle) == null ? void 0 : a.legendContainer;
    }), D = w(() => {
      var t, a;
      return (a = (t = e.seriesSetting) == null ? void 0 : t.legendStyle) == null ? void 0 : a.legendItem;
    }), p = H(null);
    let r = null;
    const O = J({
      // 图表数据
      chartData: []
    }), m = V("chart-map-base"), x = () => {
      var t, a;
      if (e.mapData && ((t = e.seriesSetting) != null && t.data) && (e != null && e.data)) {
        const { nameKey: c, dataKey: l } = e.seriesSetting.data, y = (a = e.options) == null ? void 0 : a.color;
        let o = 0;
        e.data.forEach((i) => {
          o += i[l] || 0;
        }), O.chartData = e.data.map((i, f) => {
          var d, u, S, g, h, b;
          return {
            name: i[c] || "--",
            value: i[l] || 0,
            // 计算百分比值
            percent: ((i[l] || 0) / o * 100).toFixed(
              Number(((d = D.value) == null ? void 0 : d.fixedNumber) <= 0 ? 0 : (u = D.value) == null ? void 0 : u.fixedNumber) || 0
            ),
            // 每块样式设置
            itemStyle: {
              color: y && y[f] || z.color[f],
              borderColor: ((g = (S = e.seriesSetting) == null ? void 0 : S.mapStyle) == null ? void 0 : g.borderColor) || "#fff",
              borderWidth: Number((b = (h = e.seriesSetting) == null ? void 0 : h.mapStyle) == null ? void 0 : b.borderWidth)
            }
          };
        });
      }
    }, _ = () => {
      let t = "";
      return n.value.wrapper && (t += v(n.value.wrapper, `.${m}`)), n.value.emptyContainer && (t += v(n.value.emptyContainer, `.${m} .empty`)), n.value.emptyImg && (t += v(n.value.emptyImg, `.${m} .empty .empty-img`)), n.value.emptyText && (t += v(n.value.emptyText, `.${m} .empty .empty-text`)), t;
    }, A = () => {
      const t = document.querySelector(`.${m} .style-container`);
      t && (t.innerHTML = `<style>${_()}</style>`);
    }, k = w(() => {
      var t, a, c, l;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((o) => {
            var i;
            return (o == null ? void 0 : o.data) && ((i = o == null ? void 0 : o.data) == null ? void 0 : i.length) !== 0;
          });
        if (typeof ((a = e.options) == null ? void 0 : a.series) == "object")
          return ((l = (c = e.options.series) == null ? void 0 : c.data) == null ? void 0 : l.length) === 0;
      }
      return !!e.data.length;
    }), j = () => {
      var y, o, i, f, d, u, S, g, h, b, L, N, T, I, $, K;
      const t = (y = e.options) == null ? void 0 : y.series;
      let a;
      const c = e.mapData;
      if (P.registerMap("chartMap", JSON.stringify(c)), t)
        a = t.map((C) => s(
          {
            type: "map",
            map: "chartMap",
            zoom: 1.1,
            label: {
              show: !1
            },
            showLegendSymbol: !1,
            // 系列标识的小圆点
            top: "10%",
            left: "10%",
            emphasis: {
              label: {
                show: !1
              },
              itemStyle: {
                areaColor: "#36A259"
              }
            },
            select: {
              label: {
                show: !1
              },
              itemStyle: {
                areaColor: "#36A259"
              }
            },
            ...e == null ? void 0 : e.map
          },
          C
        ));
      else if (e.mapData && e.data && ((o = e.seriesSetting) != null && o.data)) {
        const C = s(
          {
            type: "map",
            map: "chartMap",
            zoom: 1.1,
            label: {
              show: !1
            },
            showLegendSymbol: !1,
            // 系列标识的小圆点
            top: "10%",
            left: "10%",
            emphasis: {
              label: {
                show: !1
              },
              itemStyle: {
                areaColor: void 0
              }
            },
            select: {
              label: {
                show: !1
              },
              itemStyle: {
                areaColor: void 0,
                color: void 0
              }
            },
            data: O.chartData
          },
          e == null ? void 0 : e.map
        );
        a = s(
          C,
          {
            label: {
              show: !1
            },
            top: (f = (i = e.seriesSetting) == null ? void 0 : i.mapStyle) == null ? void 0 : f.top,
            left: (u = (d = e.seriesSetting) == null ? void 0 : d.mapStyle) == null ? void 0 : u.left,
            emphasis: {
              label: {
                show: !1
              },
              itemStyle: {
                areaColor: (g = (S = e.seriesSetting) == null ? void 0 : S.mapStyle) == null ? void 0 : g.hoverColor
              }
            },
            select: {
              label: {
                show: !1
              },
              itemStyle: {
                areaColor: (b = (h = e.seriesSetting) == null ? void 0 : h.mapStyle) == null ? void 0 : b.selectColor,
                color: (N = (L = e.seriesSetting) == null ? void 0 : L.mapStyle) == null ? void 0 : N.selectColor
              }
            }
          }
        );
      }
      let l = s(
        {
          series: a
        },
        z
      );
      l = s(l, {
        geo: {
          top: ((I = (T = e.seriesSetting) == null ? void 0 : T.mapStyle) == null ? void 0 : I.top) || "10%",
          left: ((K = ($ = e.seriesSetting) == null ? void 0 : $.mapStyle) == null ? void 0 : K.left) || "10%"
        }
      }), l = s(l, e.options), r.clear();
      try {
        r.setOption(l);
      } catch (C) {
        console.log(C);
      }
    }, E = () => {
      F(() => {
        r.resize();
      });
    };
    return B(() => {
      A(), p.value && e.mapData && (r = P.init(p.value), x(), j()), window.addEventListener("resize", E);
    }), R(() => {
      window.removeEventListener("resize", E);
    }), M(
      () => [e.options, e.map],
      () => {
        x(), j();
      },
      {
        deep: !0
      }
    ), M(
      () => {
        var t, a;
        return [
          e.data,
          e.seriesSetting.mapStyle,
          (a = (t = e.seriesSetting.legendStyle) == null ? void 0 : t.legendItem) == null ? void 0 : a.fixedNumber
        ];
      },
      () => {
        x(), r.clear(), e.mapData && j();
      },
      {
        deep: !0
      }
    ), M(
      () => [e.cStyle],
      () => {
        A(), F(() => {
          r.resize();
        });
      },
      {
        deep: !0
      }
    ), B(() => {
    }), {
      mapBase: p,
      ...U(O),
      curStyle: n,
      curLegendStyle: D,
      curLegendContainer: W,
      domOnlyClass: m,
      checkEmpty: k
    };
  }
});
export {
  ae as default
};
