// @ths/design/es/components/module-tabs/src/index.vue2.js
import { defineComponent as P, computed as N, reactive as S, onMounted as I, watch as p, toRefs as b } from "vue";
import { getDomRandomClass as g, getCssPropertyString as u, setStyle as n } from "../../../utils/index.js";
const D = {
  title: "title",
  num: "num",
  numUnit: "numUnit",
  percent: "percent",
  percentUnit: "percentUnit",
  numTooltip: "numTooltip",
  percentTooltip: "percentTooltip"
}, A = P({
  name: "TModuleTabs",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => D
    },
    // 选中项
    activeItem: {
      type: String,
      default: ""
    }
  },
  emits: ["tab-click"],
  setup(m, { emit: _ }) {
    const t = N(() => m.cStyle), i = g("module-tabs"), s = S({
      curData: [],
      activeIndex: null
    }), C = () => {
      s.curData = m.data.map((e) => {
        var l, a, o, c, d, r, v;
        return {
          ...e,
          title: e[((l = m.alias) == null ? void 0 : l.title) || "title"] || "",
          num: e[((a = m.alias) == null ? void 0 : a.num) || "num"] || "",
          numUnit: e[((o = m.alias) == null ? void 0 : o.numUnit) || "numUnit"] || "",
          percent: e[((c = m.alias) == null ? void 0 : c.percent) || "percent"] || "",
          percentUnit: e[((d = m.alias) == null ? void 0 : d.percentUnit) || "percentUnit"] || "",
          numTooltip: e[((r = m.alias) == null ? void 0 : r.numTooltip) || "numTooltip"] || "",
          percentTooltip: e[((v = m.alias) == null ? void 0 : v.percentTooltip) || "percentTooltip"] || ""
        };
      });
    }, f = (e) => {
      const l = s.curData.findIndex((a) => a.title === e);
      l > -1 && (s.activeIndex = l);
    }, y = (e) => {
      f(e.title), _("tab-click", e);
    }, U = () => {
      var l, a, o, c, d, r, v, T;
      let e = "";
      return (l = t.value.itemContainer) != null && l.active && (e += `
          .${i} .module-item.active {
            ${u(t.value.itemContainer.active)}
          }
        `), (a = t.value.itemTitle) != null && a.active && (e += `
          .${i} .module-item.active .module-item__title {
            ${u(t.value.itemTitle.active)}
          }
        `), (o = t.value.itemNumContainer) != null && o.active && (e += `
          .${i} .module-item.active .module-item-content .module-item-num {
            ${u(t.value.itemNumContainer.active)}
          }
        `), (c = t.value.itemNumText) != null && c.active && (e += `
          .${i} .module-item.active .module-item-content .module-item-num .module-item-num__text {
            ${u(t.value.itemNumText.active)}
          }
        `), (d = t.value.itemNumUnit) != null && d.active && (e += `
          .${i} .module-item.active .module-item-content .module-item-num .module-item-num__unit {
            ${u(t.value.itemNumUnit.active)}
          }
        `), (r = t.value.itemPercentContainer) != null && r.active && (e += `
          .${i} .module-item.active .module-item-content .module-item-percent {
            ${u(t.value.itemPercentContainer.active)}
          }
        `), (v = t.value.itemPercentText) != null && v.active && (e += `
          .${i} .module-item.active .module-item-content .module-item-percent .module-item-percent__text {
            ${u(t.value.itemPercentText.active)}
          }
        `), (T = t.value.itemPercentUnit) != null && T.active && (e += `
          .${i} .module-item.active .module-item-content .module-item-percent .module-item-percent__unit {
            ${u(t.value.itemPercentUnit.active)}
          }
        `), e;
    }, x = () => {
      let e = "";
      return t.value.wrapper && (e += n(t.value.wrapper, `.${i}`)), t.value.itemContainer && (e += n(t.value.itemContainer, `.${i} .module-item`)), t.value.itemTitle && (e += n(t.value.itemTitle, `.${i} .module-item .module-item__title`)), t.value.itemContent && (e += n(t.value.itemContent, `.${i} .module-item .module-item-content`)), t.value.itemNumContainer && (e += n(t.value.itemNumContainer, `.${i} .module-item .module-item-content .module-item-num`)), t.value.itemNumText && (e += n(t.value.itemNumText, `.${i} .module-item .module-item-content .module-item-num
        .module-item-num__text`)), t.value.itemNumUnit && (e += n(t.value.itemNumUnit, `.${i} .module-item .module-item-content .module-item-num
        .module-item-num__unit`)), t.value.itemPercentContainer && (e += n(t.value.itemPercentContainer, `.${i} .module-item .module-item-content .module-item-percent`)), t.value.itemPercentText && (e += n(t.value.itemPercentText, `.${i} .module-item .module-item-content .module-item-percent
        .module-item-percent__text`)), t.value.itemPercentUnit && (e += n(t.value.itemPercentUnit, `.${i} .module-item .module-item-content .module-item-percent
        .module-item-percent__unit`)), e;
    }, $ = () => {
      const e = document.querySelector(`.${i} .style-container`);
      e && (e.innerHTML = `<style>${x()}${U()}</style>`);
    };
    return I(() => {
      $();
    }), p([() => m.data, () => m.alias], () => {
      C(), f(m.activeItem);
    }, { deep: !0, immediate: !0 }), p(() => m.activeItem, () => {
      f(m.activeItem);
    }), p(() => m.cStyle, () => {
      $();
    }, { deep: !0 }), {
      ...b(s),
      curStyle: t,
      domOnlyClass: i,
      changeActiveItem: f,
      clickItem: y
    };
  }
});
export {
  A as default
};
