// @ths/design/es/components/breadcrumb-base/src/index.vue2.js
import { defineComponent as G, computed as M, reactive as J, watch as m, onMounted as N, toRefs as Q } from "vue";
import U from "lodash/merge";
import V from "lodash/cloneDeep";
import { getDomRandomClass as X, getCssPropertyString as r } from "../../../utils/index.js";
const T = {
  label: "label",
  // 名称
  icon: "icon",
  // 图标
  activeIcon: "activeIcon",
  // 激活图标
  path: "path",
  // 路径
  active: "label"
  // 激活判断标识
}, ce = G({
  name: "TBreadcrumbBase",
  props: {
    // 间隔符
    separator: {
      type: String,
      default: ">"
    },
    // 激活项
    activeItem: {
      type: String,
      default: ""
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 是否显示图标
    showIcon: {
      type: Boolean,
      default: !0
    },
    // 是否是自定义图标
    customIcon: {
      type: Boolean,
      default: !1
    },
    // 字段关键词
    keyField: {
      type: Object,
      default: () => T
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器默认
        wrapper: {
          default: {}
        },
        // 单项容器 默认/hover/选中
        itemWrapper: {
          default: {},
          hover: {},
          active: {}
        },
        // 图标
        icon: { default: {} }
      })
    }
  },
  // 事件名
  emits: ["breadcrumb-click"],
  setup(c, { emit: q }) {
    const a = X("breadcrumb-base"), L = c.separator === ">", P = M(() => c.separator), i = M(() => U(V(T), c.keyField)), e = J({
      curData: [],
      customStyle: c.cStyle,
      // 组件样式
      hoverItemIndex: -1,
      // hover的索引
      curActive: c.activeItem
      // 当前选中项
    }), z = () => {
      var o, n, u, d, s, y, b, S, f, p, v, h, _, $, I, W, C, g, D, k, w, A, B, F, O, R, j, x, H, K;
      let t = "";
      return (o = e.customStyle) != null && o.itemWrapper && ((u = (n = e.customStyle) == null ? void 0 : n.itemWrapper) != null && u.default) && (t += `.${a} .breadcrumb-main .el-breadcrumb__item {
          ${r((s = (d = e.customStyle) == null ? void 0 : d.itemWrapper) == null ? void 0 : s.default)}
        }`), (y = e.customStyle) != null && y.itemWrapper && ((S = (b = e.customStyle) == null ? void 0 : b.itemWrapper) != null && S.hover) && (t += `.${a} .breadcrumb-main .el-breadcrumb__item.hover:not(.active) {
          ${r((p = (f = e.customStyle) == null ? void 0 : f.itemWrapper) == null ? void 0 : p.hover)}
        }`), (v = e.customStyle) != null && v.itemWrapper && ((_ = (h = e.customStyle) == null ? void 0 : h.itemWrapper) != null && _.active) && (t += `.${a} .breadcrumb-main .el-breadcrumb__item.active {
          ${r((I = ($ = e.customStyle) == null ? void 0 : $.itemWrapper) == null ? void 0 : I.active)}
        }`), (W = e.customStyle) != null && W.icon && ((g = (C = e.customStyle) == null ? void 0 : C.icon) != null && g.default) && (t += `.${a} .breadcrumb-main .el-breadcrumb__inner .icon-common {
          ${r((k = (D = e.customStyle) == null ? void 0 : D.icon) == null ? void 0 : k.default)}
        }`), (w = e.customStyle) != null && w.icon && ((B = (A = e.customStyle) == null ? void 0 : A.icon) != null && B.hover) && (t += `.${a} .breadcrumb-main .el-breadcrumb__item.hover:not(.active) .icon-common {
          ${r((O = (F = e.customStyle) == null ? void 0 : F.icon) == null ? void 0 : O.hover)}
        }`), (R = e.customStyle) != null && R.icon && ((x = (j = e.customStyle) == null ? void 0 : j.icon) != null && x.active) && (t += `.${a} .breadcrumb-main .el-breadcrumb__item.active .icon-common {
          ${r((K = (H = e.customStyle) == null ? void 0 : H.icon) == null ? void 0 : K.active)}
        }`), t;
    }, l = () => {
      const t = document.querySelector(`.${a} > .style-container`);
      t && (t.innerHTML = `<style>${z()}</style>`);
    }, E = (t, o) => {
      e.curData = e.curData.slice(0, o + 1), e.curActive = t[i.value.active], q("breadcrumb-click", t);
    };
    return m(() => c.cStyle, () => {
      e.customStyle = c.cStyle, l();
    }, {
      deep: !0
    }), m(() => c.activeItem, () => {
      e.curActive = c.activeItem;
    }), m(() => c.data, () => {
      e.curData = c.data;
    }, { immediate: !0, deep: !0 }), N(() => {
      l();
    }), {
      ...Q(e),
      curSeparator: P,
      currentKeyField: i,
      domOnlyClass: a,
      isShowRightIcon: L,
      breadcrumbClick: E
    };
  }
});
export {
  ce as default
};
