// @ths/design/es/components/breadcrumb-base/src/index.vue.js
import I from "./index.vue2.js";
import { resolveComponent as d, openBlock as n, createElementBlock as c, normalizeClass as t, normalizeStyle as h, createBlock as o, withCtx as a, Fragment as K, renderList as g, createCommentVNode as i, resolveDynamicComponent as m, createTextVNode as b, toDisplayString as f, createElementVNode as C } from "vue";
import "./index.vue3.js";
import w from "../../../_virtual/_plugin-vue_export-helper.js";
const $ = ["src"], B = ["src"];
function O(e, s, S, A, D, M) {
  var p, F;
  const y = d("el-icon"), v = d("el-breadcrumb-item"), k = d("el-breadcrumb");
  return n(), c("div", {
    class: t(["t-breadcrumb-base", e.domOnlyClass]),
    style: h({ ...(F = (p = e.customStyle) == null ? void 0 : p.wrapper) == null ? void 0 : F.default })
  }, [
    (n(), o(k, {
      key: e.separator,
      class: "breadcrumb-main",
      "separator-icon": e.isShowRightIcon ? "arrow-right" : "",
      separator: e.curSeparator
    }, {
      default: a(() => [
        (n(!0), c(K, null, g(e.curData, (r, l) => (n(), c(K, { key: l }, [
          r[e.currentKeyField.path] ? (n(), o(v, {
            key: 0,
            class: t({ active: r[e.currentKeyField.active] && r[e.currentKeyField.active] === e.curActive, hover: e.hoverItemIndex === l }),
            to: { path: r[e.currentKeyField.path] },
            onClick: (u) => e.breadcrumbClick(r, l),
            onMouseenter: (u) => e.hoverItemIndex = l,
            onMouseleave: s[0] || (s[0] = (u) => e.hoverItemIndex = -1)
          }, {
            default: a(() => [
              e.showIcon && r[e.currentKeyField.icon] && r[e.currentKeyField.icon].indexOf("/") !== -1 && e.customIcon ? (n(), c("img", {
                key: 0,
                class: "fa icon-common",
                src: r[e.currentKeyField.active] && r[e.currentKeyField.active] === e.curActive ? r[e.currentKeyField.activeIcon] : r[e.currentKeyField.icon],
                draggable: "false"
              }, null, 8, $)) : i("", !0),
              e.showIcon && r[e.currentKeyField.icon] && r[e.currentKeyField.icon].indexOf("/") === -1 && !e.customIcon ? (n(), o(y, {
                key: 1,
                class: "tag-div-icon icon-common"
              }, {
                default: a(() => [
                  (n(), o(m(r[e.currentKeyField.icon])))
                ]),
                _: 2
              }, 1024)) : i("", !0),
              b(" " + f(r[e.currentKeyField.label]), 1)
            ]),
            _: 2
          }, 1032, ["class", "to", "onClick", "onMouseenter"])) : (n(), o(v, {
            key: 1,
            class: t({ active: r[e.currentKeyField.active] && r[e.currentKeyField.active] === e.curActive }),
            onClick: (u) => e.breadcrumbClick(r, l)
          }, {
            default: a(() => [
              e.showIcon && r[e.currentKeyField.icon] && r[e.currentKeyField.icon].indexOf("/") !== -1 && e.customIcon ? (n(), c("img", {
                key: 0,
                class: "fa icon-common",
                src: r[e.currentKeyField.active] && r[e.currentKeyField.active] === e.curActive ? r[e.currentKeyField.activeIcon] : r[e.currentKeyField.icon],
                draggable: "false"
              }, null, 8, B)) : i("", !0),
              e.showIcon && r[e.currentKeyField.icon] && r[e.currentKeyField.icon].indexOf("/") === -1 && !e.customIcon ? (n(), o(y, {
                key: 1,
                class: "tag-div-icon icon-common"
              }, {
                default: a(() => [
                  (n(), o(m(r[e.currentKeyField.icon])))
                ]),
                _: 2
              }, 1024)) : i("", !0),
              b(" " + f(r[e.currentKeyField.label]), 1)
            ]),
            _: 2
          }, 1032, ["class", "onClick"]))
        ], 64))), 128))
      ]),
      _: 1
    }, 8, ["separator-icon", "separator"])),
    s[1] || (s[1] = C("div", { class: "style-container" }, null, -1))
  ], 6);
}
const L = /* @__PURE__ */ w(I, [["render", O]]);
export {
  L as default
};
