// @ths/design/es/components/chart-pie-doughnut/src/index.vue2.js
import { defineComponent as D, ref as k, reactive as A, onMounted as E, onUnmounted as T, watch as S, nextTick as z, toRefs as W } from "vue";
import * as _ from "echarts";
import m from "lodash/merge";
import f from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as j, DEFAULT_OPTIONS_AXIS as I } from "../../../utils/constant.js";
const K = m(f(j), f(I)), c = m(
  f(K),
  {
    legend: {
      show: !0,
      icon: "circle",
      align: "left",
      left: "right",
      top: "middle"
    },
    tooltip: {
      trigger: "item",
      formatter: "{b} : {c} ({d}%)"
    },
    grid: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    },
    graphic: {
      elements: [
        {
          type: "image",
          z: 3,
          style: {
            image: "",
            width: 178,
            height: 178
          },
          left: "center",
          top: "center"
        }
      ]
    }
  }
), $ = D({
  name: "TChartPieDoughnut",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => c,
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    unit: {
      type: String,
      default: ""
    },
    separation: {
      type: Number,
      default: 2
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 取resData数据中的别名设置
        data: {
          nameKey: "",
          // 名称别名
          dataKey: ""
          // 数据别名
        }
      })
    },
    pie: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: v }) {
    const b = k(null);
    let a = null;
    const o = A({
      // 图表数据
      chartData: [],
      pieOptions: f(e.pie)
    });
    function O() {
      var t, d;
      if ((t = e.seriesSetting) != null && t.data && e.data) {
        const { nameKey: n, dataKey: w } = e.seriesSetting.data, l = (d = e.options) == null ? void 0 : d.color;
        o.chartData = [], e.data.forEach((u, i) => {
          o.chartData.push(
            {
              name: u[n] || "--",
              value: u[w] || 0,
              itemStyle: {
                normal: {
                  borderWidth: 8,
                  shadowBlur: 8,
                  borderColor: l && l[i] || c.color[i] || "#fff",
                  shadowColor: l && l[i] || c.color[i] || "#fff"
                }
              }
            },
            {
              value: e.separation,
              name: "",
              itemStyle: {
                normal: {
                  label: {
                    show: !1
                  },
                  labelLine: {
                    show: !1
                  },
                  color: "rgba(0, 0, 0, 0)",
                  borderColor: "rgba(0, 0, 0, 0)",
                  borderWidth: 0
                }
              }
            }
          );
        });
      }
    }
    function y() {
      var l, u, i;
      if (!a)
        return;
      const { color: t, series: d } = e.options;
      let n = [];
      d ? n = d.map((r) => {
        var h, C;
        const s = [];
        return r.data.forEach((L, g) => {
          s.push(
            {
              name: L.name || "--",
              value: L.value || 0,
              itemStyle: {
                normal: {
                  borderWidth: 8,
                  shadowBlur: 8,
                  borderColor: t && t[g] || c.color[g] || "#fff",
                  shadowColor: t && t[g] || c.color[g] || "#fff"
                }
              }
            },
            {
              value: e.separation,
              name: "",
              itemStyle: {
                normal: {
                  label: {
                    show: !1
                  },
                  labelLine: {
                    show: !1
                  },
                  color: "rgba(0, 0, 0, 0)",
                  borderColor: "rgba(0, 0, 0, 0)",
                  borderWidth: 0
                }
              }
            }
          );
        }), m(
          {
            type: "pie",
            center: ["25%", "50%"],
            avoidLabelOverlap: !1,
            clockwise: !1,
            radius: [105, 109],
            itemStyle: {
              normal: {
                label: {
                  show: !1
                },
                labelLine: {
                  show: !1
                }
              }
            },
            ...r,
            data: s,
            ...o == null ? void 0 : o.pieOptions
          },
          (C = (h = e.seriesSetting) == null ? void 0 : h.data) == null ? void 0 : C.style
        );
      }) : e.data && ((l = e.seriesSetting) != null && l.data) && (n = [m(
        {
          type: "pie",
          center: ["25%", "50%"],
          avoidLabelOverlap: !1,
          clockwise: !1,
          radius: [105, 109],
          itemStyle: {
            normal: {
              label: {
                show: !1
              },
              labelLine: {
                show: !1
              }
            }
          },
          data: o.chartData
        },
        f(e.pie),
        (i = (u = e.seriesSetting) == null ? void 0 : u.data) == null ? void 0 : i.style
      )]);
      const w = m(
        c,
        {
          legend: {
            show: !0,
            textStyle: {
              rich: {
                name: {
                  fontSize: 14,
                  color: "rgba(255, 255, 255, 0.7)",
                  width: 115
                },
                value: {
                  fontSize: 14,
                  fontWeight: "bold",
                  padding: [0, 5, 0, 15],
                  color: "#fff"
                },
                unit: {
                  fontWeight: "bold",
                  fontSize: 14,
                  color: "#fff"
                }
              }
            },
            formatter(r) {
              let s = n[0].data.filter((h) => h.name === r);
              return s = s[0] || {}, `{name|${r}}{value|${s.value || ""}}{unit|${e.unit || ""}}`;
            }
          }
        },
        e.options,
        {
          series: n
        }
      );
      a.clear();
      try {
        a.setOption(w);
      } catch (r) {
        console.log(r);
      }
    }
    function p() {
      z(() => {
        a.resize();
      });
    }
    return E(() => {
      b.value && (a = _.init(b.value), O(), y(), a.on("click", (t) => {
        v("com-click", t);
      }), a.on("legendselectchanged", (t) => {
        v("legend-select", t);
      })), window.addEventListener("resize", p);
    }), T(() => {
      window.removeEventListener("resize", p);
    }), S(
      () => e.pie,
      () => {
        o.pieOptions = f(e.pie), y();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), S(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting,
        () => e.unit,
        () => e.separation
      ],
      () => {
        O(), y();
      },
      {
        deep: !0
      }
    ), S(
      () => e.cStyle,
      () => {
        z(() => {
          a.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      ...W(o),
      chartId: b
    };
  }
});
export {
  $ as default
};
