// @ths/design/es/components/chart-bar-compare/src/index.vue2.js
import { defineComponent as M, computed as W, ref as G, reactive as N, nextTick as q, onMounted as P, onUnmounted as R, watch as C, toRefs as U } from "vue";
import * as H from "echarts";
import l from "lodash/merge";
import S from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as K, DEFAULT_OPTIONS_AXIS as X } from "../../../utils/constant.js";
import { getDomRandomClass as J, setStyle as v } from "../../../utils/index.js";
const Q = l(S(K), S(X)), A = l(
  S(Q),
  {
    // x轴
    xAxis: {
      boundaryGap: !0
    },
    // 去掉轴文字、数值的负号，暂时隐藏
    // tooltip: {
    //   formatter: (value) => {
    //     let str = '';
    //     const { seriesName } = value[0];
    //     value.forEach((item: any) => {
    //       str += `${item.marker}${item.name}  ${item.value.toString().replace('-', '')}<br/>`;
    //     });
    //     return `${seriesName}<br/>${str}`;
    //   },
    // },
    // yAxis: {
    //   axisLabel: {
    //     formatter: (value) => value.toString().replace('-', ''),
    //   },
    // },
    // 图例
    legend: {
      show: !1
    },
    // 网格线
    grid: {
      top: "12%",
      bottom: "2%",
      left: "1%",
      right: "2%",
      containLabel: !0
    }
  }
), ae = M({
  name: "TChartBarCompare",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        }
      })
    },
    // 设置主轴
    mainAxis: {
      type: String,
      default: "x"
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => A,
      required: !1
    },
    // 传入数组数据 如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 柱状图主题样式，用于平台里设置
    bar: {
      type: Object,
      default: () => ({})
    }
  },
  // 事件名定义
  emits: ["com-click", "legend-select"],
  setup(e, { emit: w }) {
    const u = J("chart-bar-compare"), d = W(() => e.cStyle), h = G(null);
    let g = null;
    const n = N({
      // 图表数据
      chartData: [],
      // 柱状图配置
      barOptions: S(e.bar),
      // 图例数据
      legendData: []
    }), T = () => {
      var t, y;
      (y = (t = e.seriesSetting) == null ? void 0 : t.data) != null && y.length && e.data && (n.chartData = e.seriesSetting.data.map((a) => {
        const { name: s, dataKey: r } = a;
        let i = [];
        return i = e.data.map((o) => ({
          value: o[r],
          itemStyle: {
            color: {
              type: "linear",
              x: 0,
              y: 1,
              x2: 0,
              y2: 0,
              colorStops: [
                {
                  offset: 1,
                  color: o[r] > 0 ? a.positiveEnd || "#eee" : a.negativeEnd || "#eee"
                  // 100% 处的颜色
                },
                {
                  offset: 0,
                  color: o[r] > 0 ? a.positiveStart || "#eee" : a.negativeStart || "#eee"
                  // 0% 处的颜色
                }
              ]
            },
            borderColor: o[r] > 0 ? a.positiveBorder || "transparent" : a.negativeBorder || "transparent"
          }
        })), {
          name: s || "--",
          data: i
        };
      }), n.legendData = e.seriesSetting.data.map((a) => ({
        name: a == null ? void 0 : a.name,
        itemStyle: l(S({
          // 可以设置渐变色
          color: {
            type: "linear",
            x: 0,
            y: 1,
            x2: 0,
            y2: 0,
            colorStops: [
              {
                offset: 1,
                color: "transparent"
                // 100% 处的颜色
              },
              {
                offset: 0,
                color: "transparent"
                // 0% 处的颜色
              }
            ]
          }
        }), a.legendStyle)
      })));
    }, z = () => {
      let t = "";
      return d.value.wrapper && (t += v(d.value.wrapper, `.${u}`)), d.value.emptyContainer && (t += v(d.value.emptyContainer, `.${u} .empty`)), d.value.emptyImg && (t += v(d.value.emptyImg, `.${u} .empty .empty-img`)), d.value.emptyText && (t += v(d.value.emptyText, `.${u} .empty .empty-text`)), t;
    }, E = () => {
      const t = document.querySelector(`.${u} .style-container`);
      t && (t.innerHTML = `<style>${z()}</style>`);
    }, m = () => {
      h.value && (h.value.getBoundingClientRect().width ?? 0) && q(() => {
        g.resize();
      });
    }, B = W(() => {
      var t, y, a, s;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((i) => {
            var o;
            return (i == null ? void 0 : i.data) && ((o = i == null ? void 0 : i.data) == null ? void 0 : o.length) !== 0;
          });
        if (typeof ((y = e.options) == null ? void 0 : y.series) == "object")
          return ((s = (a = e.options.series) == null ? void 0 : a.data) == null ? void 0 : s.length) === 0;
      }
      return e.data.length ? (q(() => {
        m();
      }), !0) : !1;
    }), O = () => {
      var i, o, F, L, I, j, _;
      const t = (i = e.options) == null ? void 0 : i.color, y = (o = e.options) == null ? void 0 : o.series;
      let a = [], s = [];
      y ? a = y.map((f, c) => {
        var b, x, p;
        const D = l(
          {
            type: "bar",
            barWidth: 14,
            barGap: "80%",
            itemStyle: {
              color: t && t[c] || A.color[c] || "#FFFFFF00"
            }
          },
          n == null ? void 0 : n.barOptions,
          (p = (x = (b = e.seriesSetting) == null ? void 0 : b.data) == null ? void 0 : x[c]) == null ? void 0 : p.style
        );
        return l({}, D, f);
      }) : e.data && ((L = (F = e.seriesSetting) == null ? void 0 : F.data) != null && L.length) && (a = (I = e.seriesSetting) == null ? void 0 : I.data.map((f, c) => {
        var b, x, p, $, k;
        const D = l(
          {
            name: (b = n.chartData[c]) == null ? void 0 : b.name,
            type: "bar",
            barWidth: 14,
            barGap: "80%",
            label: {
              show: !1,
              position: "top"
            },
            itemStyle: {
              borderWidth: 0
            },
            data: e.data.length ? (x = n.chartData[c]) == null ? void 0 : x.data : []
          },
          n == null ? void 0 : n.barOptions,
          (k = ($ = (p = e.seriesSetting) == null ? void 0 : p.data) == null ? void 0 : $[c]) == null ? void 0 : k.style
        );
        return l({}, D, f);
      }), (j = e.seriesSetting) != null && j.xAxisLabel ? s = (_ = e.data) != null && _.length ? e.data.map((f) => f[e.seriesSetting.xAxisLabel]) : [] : s = e.data.map((f, c) => c));
      let r;
      e.mainAxis === "x" ? r = l(
        {
          series: a,
          xAxis: {
            data: s
          },
          legend: {
            data: n.legendData
          }
        },
        A
      ) : e.mainAxis === "y" && (r = l(
        {
          series: a,
          yAxis: {
            data: s
          },
          legend: {
            data: n.legendData
          }
        },
        A
      )), r = l(r, e.options), e.mainAxis === "y" && (r.xAxis.type = "value", r.yAxis.type = "category"), g.clear();
      try {
        g.setOption(r);
      } catch (f) {
        console.log(f);
      }
    };
    return P(() => {
      E(), h.value && (g = H.init(h.value), T(), O(), g.on("click", (t) => {
        w("com-click", t);
      }), setTimeout(() => {
        m();
      }, 500), g.on("legendselectchanged", (t) => {
        w("legend-select", t);
      })), window.addEventListener("resize", m);
    }), R(() => {
      window.removeEventListener("resize", m);
    }), C(
      [
        () => e.bar,
        () => e.mainAxis
      ],
      () => {
        n.barOptions = S(e.bar), O(), m();
      },
      {
        deep: !0
      }
    ), C(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting
      ],
      () => {
        T(), O(), m();
      },
      {
        deep: !0
      }
    ), C(
      () => e.cStyle,
      () => {
        E(), m();
      },
      {
        deep: !0
      }
    ), {
      ...U(n),
      chartId: h,
      checkEmpty: B,
      domOnlyClass: u
    };
  }
});
export {
  ae as default
};
