// @ths/design/es/components/chart-area-3d/src/index.vue.js
import y from "./index.vue2.js";
import { openBlock as s, createElementBlock as r, normalizeClass as k, normalizeStyle as d, Fragment as i, renderList as g, createElementVNode as e, toDisplayString as c, createCommentVNode as f, withDirectives as h, vShow as p } from "vue";
import "./index.vue3.js";
import C from "../../../_virtual/_plugin-vue_export-helper.js";
const L = {
  key: 0,
  class: "t-chart-area-3d-legend"
}, b = ["onClick"], w = { class: "t-chart-area-3d-legend-item_text" }, E = { class: "t-chart-area-3d-container" }, $ = {
  id: "t-chart-area-3d-canvas",
  class: "t-chart-area-3d-canvas"
}, S = { class: "t-chart-area-3d-no-data" }, V = { class: "t-chart-area-3d-tooltip" }, z = { class: "t-chart-area-3d-tooltip-left" }, A = { class: "t-chart-area-3d-tooltip-left__text" }, B = { class: "t-chart-area-3d-tooltip__right" };
function D(t, o, N, F, I, O) {
  var u, v, _, m;
  return s(), r("div", {
    ref: "chartArea3dRef",
    class: k(["t-chart-area-3d", t.domOnlyClass]),
    style: d({ ...(v = (u = t.curStyle) == null ? void 0 : u.wrapper) == null ? void 0 : v.default })
  }, [
    t.showLegend ? (s(), r("div", L, [
      (s(!0), r(i, null, g(t.legendList, (a, l) => (s(), r("div", {
        key: a.name,
        class: "t-chart-area-3d-legend-item",
        onClick: (n) => t.legendItemClick(l)
      }, [
        e("span", {
          class: "t-chart-area-3d-legend-item_icon",
          style: d({ backgroundColor: a.visible ? a.color : "#aeafb2" })
        }, null, 4),
        e("span", w, c(a.name), 1)
      ], 8, b))), 128))
    ])) : f("", !0),
    e("div", E, [
      h(e("canvas", $, null, 512), [
        [p, ((_ = t.data) == null ? void 0 : _.length) > 0]
      ]),
      h(e("div", S, " 暂无数据 ", 512), [
        [p, ((m = t.data) == null ? void 0 : m.length) === 0]
      ]),
      h(e("div", V, [
        o[0] || (o[0] = e("div", { class: "t-chart-area-3d-tooltip-title" }, " 2023年 ", -1)),
        (s(!0), r(i, null, g(t.tooltipList, (a, l) => {
          var n;
          return s(), r("div", {
            key: a.name,
            class: "t-chart-area-3d-tooltip-series"
          }, [
            t.showLegend && ((n = t.legendList[l]) != null && n.visible) || !t.showLegend ? (s(), r(i, { key: 0 }, [
              e("div", z, [
                e("span", {
                  class: "t-chart-area-3d-tooltip-left__icon",
                  style: d({ background: a.color })
                }, null, 4),
                e("span", A, c(a.name), 1)
              ]),
              e("div", B, c(a.value), 1)
            ], 64)) : f("", !0)
          ]);
        }), 128))
      ], 512), [
        [p, t.tooltipVisible]
      ])
    ]),
    o[1] || (o[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const H = /* @__PURE__ */ C(y, [["render", D]]);
export {
  H as default
};
