// @ths/design/es/components/chart-area-3d/src/index.vue2.js
import { defineComponent as be, computed as we, ref as Le, reactive as De, onMounted as Ce, nextTick as se, onUnmounted as Se, watch as ae, toRefs as Ae } from "vue";
import I from "lodash/merge";
import W from "lodash/cloneDeep";
import "lodash/sortBy";
import Pe from "lodash/debounce";
import * as i from "three";
import { OrbitControls as Ee } from "three/addons/controls/OrbitControls";
import { BoxLineGeometry as He } from "three/addons/geometries/BoxLineGeometry.js";
import { FontLoader as ke } from "three/addons/loaders/FontLoader";
import { TextGeometry as ze } from "three/addons/geometries/TextGeometry";
import "three/addons/postprocessing/EffectComposer.js";
import "three/addons/postprocessing/RenderPass.js";
import "three/addons/postprocessing/UnrealBloomPass.js";
import "three/addons/postprocessing/OutputPass.js";
import "three/addons/postprocessing/ShaderPass.js";
import "three/addons/objects/ReflectorForSSRPass.js";
import "three/addons/postprocessing/SSRPass.js";
import { getDomRandomClass as Te, setStyle as Oe } from "../../../utils/index.js";
const Re = {
  key: "YEARNUMBER",
  fontSize: 24,
  color: "white",
  fontWeight: "normal",
  yAxisDistance: 0.3,
  zAxisDistance: 0,
  interval: "auto",
  showNum: 6
}, Ve = {
  fontSize: 24,
  color: "white",
  fontWeight: "normal",
  yAxisDistance: 0.5,
  zAxisDistance: 0,
  segment: 5,
  min: 0,
  decimal: 0
}, Be = {
  sceneBgColor: "#041c3e",
  yEdgeDistance: 1,
  boxLineColor: "#314765",
  outLineWidth: 1,
  OLHighlightWidth: 1,
  zoom: 1,
  areaOpacity: 0.3
}, nt = be({
  name: "TChartArea3d",
  props: {
    // 系列配置
    series: {
      type: Array,
      default: () => []
    },
    // x轴配置
    xAxis: {
      type: Object,
      default: () => ({})
    },
    // y轴配置
    yAxis: {
      type: Object,
      default: () => ({})
    },
    // 是否显示tooltip
    showTooltip: {
      type: Boolean,
      default: !0
    },
    // 是否显示图例
    showLegend: {
      type: Boolean,
      default: !0
    },
    // 图例的位置
    legendPosition: {
      type: String,
      default: "top"
      /* top */
    },
    // 三维场景参数
    threeParam: {
      type: Object,
      default: () => ({})
    },
    // 传入数据
    data: {
      type: Array,
      default: () => []
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    }
  },
  emits: ["title-click"],
  setup(s, { emit: We }) {
    const Y = we(() => s.cStyle), _ = Te("chart-area-3d"), z = I(Re, s.xAxis), M = I(Ve, s.yAxis), S = I(Be, s.threeParam), T = Le();
    let l, b, g, H, O;
    const re = new ke(), $ = {};
    let R, q, G, j, P = 0, K = null, V = null, d = null;
    const e = De({
      // 图例数据
      legendList: [],
      // y轴坐标比例 3
      yAxisRate: 1,
      // x 轴标签集合
      xAxisData: [],
      // y 轴坐标集合
      // Y轴最大坐标
      yMaxData: 0,
      // 曲线坐标集合
      curvePointsMap: {},
      // 曲线数值集合
      curveValuesMap: {},
      // series 系列数据
      seriesData: [],
      // 接口返回的数据
      dataList: [],
      // 最小面key值
      minKey: "",
      // 控制tooltip 显示隐藏
      tooltipVisible: !1,
      tooltipList: [],
      // 保存系列key值与高亮边网格集合映射关系
      highlightMap: {},
      mouseEnter: !1,
      boxLong: 10,
      boxDept: 2,
      boxHeight: 6,
      resizeObserve: null
    }), X = 1;
    new i.Layers().set(X), new i.MeshBasicMaterial({ color: "black" });
    const le = () => {
      let o = "";
      return Y.value.wrapper && (o += Oe(Y.value.wrapper, `.${_}`)), o;
    }, Z = () => {
      const o = document.querySelector(
        `.${_} .style-container`
      );
      o && (o.innerHTML = `<style>${le()}</style>`);
    }, ce = () => {
      var o;
      if (M.max && M.max > 0)
        e.yMaxData = M.max;
      else {
        let t = 0;
        (o = s.data) == null || o.forEach((r) => {
          var a;
          (a = s.series) == null || a.forEach((h) => {
            Number(r[h.key]) > t && (t = Number(r[h.key]));
          });
        });
        let n = "";
        String(t).includes(".") ? n = String(t).split(".")[0] : n = String(t);
        const c = Math.pow(10, n.length - 1);
        e.yMaxData = Math.ceil(t / c) * c, e.yAxisRate = Math.round(e.yMaxData / e.boxHeight * 100) / 100;
      }
    }, he = () => {
      var t, n;
      if (((t = s.series) == null ? void 0 : t.length) < 2 || !((n = s.data) != null && n.length)) {
        e.seriesData = W(s.series);
        return;
      }
      e.seriesData = [];
      const o = s.data[s.data.length - 1];
      Number(o[s.series[0].key]) > Number(o[s.series[1].key]) ? e.seriesData = [W(s.series[1]), W(s.series[0])] : e.seriesData = W(s.series);
    }, ue = (o, t) => {
      $[o] ? t($[o]) : re.load(o, (n) => {
        $[o] = n, t(n);
      });
    }, U = (o, t) => {
      ue(
        "./FZZongYi-M05S_Regular.json",
        (n) => {
          const c = new ze(o, {
            font: n,
            size: 0.3,
            // 字体大小
            depth: 0.1
            // 挤出文本的厚度
          });
          c.center();
          const r = new i.MeshBasicMaterial({
            color: 4294967295,
            transparent: !0,
            opacity: 0.7
          }), a = new i.Mesh(c, r);
          a.position.set(t[0], t[1], t[2]), g.add(a), a.lookAt(new i.Vector3(t[0], b.position.y, b.position.z));
        }
      );
    }, de = () => {
      const o = e.boxHeight / M.segment;
      if (M.segment)
        for (let t = 0; t <= M.segment; t++) {
          const n = t * o, c = String(e.yMaxData / M.segment * t), r = -M.yAxisDistance, a = e.boxDept / 2 - M.zAxisDistance;
          U(c, [r, n, a]);
        }
    }, ge = () => {
      var o, t, n;
      if (e.xAxisData = [], z.interval === "auto") {
        const c = (e.boxLong - S.yEdgeDistance * 2) / (((o = s.data) == null ? void 0 : o.length) - 1);
        (t = s.data) == null || t.forEach((r, a) => {
          const h = -z.yAxisDistance, u = e.boxDept / 2 - z.zAxisDistance, w = a * c + S.yEdgeDistance, y = {
            value: r[z.key],
            point: [w, h, u]
          };
          U(y.value, y.point), e.xAxisData.push(y);
        });
      } else
        (n = s.data) == null || n.forEach((c, r) => {
          r + 1, z.interval;
        });
    }, me = () => {
      var o;
      (o = e.seriesData) == null || o.forEach((t, n) => {
        var c;
        e.curveValuesMap[t.key] = [], e.curvePointsMap[t.key] = [], (c = s.data) == null || c.forEach((r, a) => {
          const h = {
            name: t.name,
            // value: item[sItem.key],
            value: 0
          };
          h.value = r[t.key], e.curveValuesMap[t.key].push(W(h));
          const u = {
            x: 0,
            y: 0
          };
          u.x = e.xAxisData[a].point[0], u.y = Number(r[t.key]) / e.yAxisRate, e.curvePointsMap[t.key].push(new i.Vector2(u.x, u.y));
        });
      });
    }, J = (o, t, n, c) => {
      l = document.createElement("canvas"), l.width = n, l.height = c;
      const r = l.getContext("2d"), a = r == null ? void 0 : r.createLinearGradient(0, 0, n, 0);
      return a == null || a.addColorStop(0, o), a == null || a.addColorStop(1, t), r.fillStyle = a, r == null || r.fillRect(0, 0, n, c), new i.Texture(l);
    }, fe = () => {
      var o;
      (o = e.seriesData) == null || o.forEach((t) => {
        var f;
        const n = new i.CurvePath(), c = new i.SplineCurve(
          e.curvePointsMap[t.key]
        );
        n.curves.push(c);
        const r = ((f = e.curvePointsMap[t.key]) == null ? void 0 : f.length) || 0, a = new i.LineCurve(
          e.curvePointsMap[t.key][r - 1],
          new i.Vector2(e.curvePointsMap[t.key][r - 1].x, 0)
        );
        n.curves.push(a);
        const h = new i.LineCurve(
          new i.Vector2(e.curvePointsMap[t.key][r - 1].x, 0),
          new i.Vector2(e.curvePointsMap[t.key][0].x, 0)
        );
        n.curves.push(h);
        let u = null;
        u = new i.LineCurve(
          new i.Vector2(e.curvePointsMap[t.key][0].x, 0),
          e.curvePointsMap[t.key][0]
        ), n.curves.push(u);
        const w = new i.Shape(n.getPoints(100)), y = {
          depth: e.boxDept,
          steps: 1,
          bevelEnabled: !1,
          curveSegments: 100,
          bevelThickness: 0,
          bevelSegments: 0
        }, x = new i.ExtrudeGeometry(w, y);
        x.raycast = !0;
        const m = [], p = J(
          "rgba(0,0,0,0.4)",
          t.color,
          e.boxLong,
          e.boxHeight
        );
        p.needsUpdate = !0, p.offset.set(-1.8, 4), p.center.set(5, 3.5), p.rotation = Math.PI / 2;
        const L = new i.MeshPhongMaterial({
          map: p,
          transparent: !0,
          depthWrite: !1,
          opacity: 0.3
        });
        m.push(L);
        const E = J(
          t.color,
          t.color,
          e.boxLong,
          e.boxDept
        );
        E.needsUpdate = !0;
        const D = new i.MeshPhongMaterial({
          map: E,
          transparent: !0,
          depthWrite: !1,
          opacity: 0.3
        });
        m.push(D);
        const v = new i.Mesh(x, m);
        e.highlightMap[t.key].push(v), v.position.set(0, 0, -1), g.add(v);
      });
    }, F = (o, t, n) => ({
      blending: i.AdditiveBlending,
      transparent: !0,
      side: i.DoubleSide,
      depthWrite: !1,
      // color: 0xcdcccc,
      uniforms: {
        time: { value: 0 },
        // 运动时间
        len: { value: o },
        // 运动点距离范围
        size: { value: 0.01 },
        // 管道增加宽度
        color1: { value: new i.Color(t) },
        color2: { value: new i.Color(n) }
      },
      // 顶点着色器
      vertexShader: `uniform float time;
          uniform float size;
          uniform float len;
          uniform vec3 color1;
          uniform vec3 color2;
          varying vec3 vColor;
          void main() {
            vColor = color1;
            vec3 newPosition = position;
            float d = uv.x - time;

            if(abs(d) < len) {
              newPosition = newPosition + normal * size;
              vColor = color2;
            }
            gl_Position = projectionMatrix * modelViewMatrix * vec4(newPosition, 1.0);
          }`,
      // 片元着色器
      fragmentShader: `
          varying vec3 vColor;
          void main() {
          gl_FragColor =vec4(vColor, 1.0);
          }`
    }), A = (o, t) => {
      const n = new i.CatmullRomCurve3(t), c = new i.TubeGeometry(
        n,
        100,
        0.01,
        100,
        !1
      ), r = new i.ShaderMaterial(o), a = new i.Mesh(
        c,
        r
      );
      return a.layers.toggle(X), g.add(a), a;
    }, ye = () => {
      var k, N, B;
      R = F(
        0.5,
        e.seriesData[1].outLineColor,
        e.seriesData[1].OLHighlightColor
      ), G = F(
        0.5,
        e.seriesData[1].outLineColor,
        e.seriesData[1].OLHighlightColor
      ), q = F(
        0.5,
        e.seriesData[0].outLineColor,
        e.seriesData[0].OLHighlightColor
      ), j = F(
        0.5,
        e.seriesData[0].outLineColor,
        e.seriesData[0].OLHighlightColor
      );
      const o = e.seriesData[((k = e.seriesData) == null ? void 0 : k.length) - 1].key, t = [], n = [];
      (N = e.curvePointsMap[o]) == null || N.forEach((C) => {
        t.push(new i.Vector3(C.x, C.y, e.boxDept / 2)), n.push(new i.Vector3(C.x, C.y, -e.boxDept / 2));
      }), e.highlightMap[o] = [];
      const c = A(R, t);
      e.highlightMap[o].push(c);
      const r = A(R, n);
      e.highlightMap[o].push(r);
      const a = e.seriesData[0].key, h = [], u = [];
      (B = e.curvePointsMap[a]) == null || B.forEach((C) => {
        h.push(new i.Vector3(C.x, C.y, e.boxDept / 2)), u.push(new i.Vector3(C.x, C.y, -e.boxDept / 2));
      }), e.highlightMap[a] = [];
      const w = A(q, h);
      e.highlightMap[a].push(w);
      const y = A(q, u);
      e.highlightMap[a].push(y);
      const x = [n[0], t[0]], m = A(G, x);
      e.highlightMap[o].push(m);
      const p = [u[0], h[0]], L = A(j, p);
      e.highlightMap[a].push(L);
      const E = [
        n[(n == null ? void 0 : n.length) - 1],
        t[(t == null ? void 0 : t.length) - 1]
      ], D = A(G, E);
      e.highlightMap[o].push(D);
      const v = [
        u[(u == null ? void 0 : u.length) - 1],
        h[(h == null ? void 0 : h.length) - 1]
      ], f = A(j, v);
      e.highlightMap[a].push(f);
    }, ve = () => {
      const o = new i.PlaneGeometry(e.boxDept, e.boxHeight);
      o.rotateY(Math.PI / 2);
      const t = new i.MeshBasicMaterial({
        color: "#d1cdcd",
        // color: '#747576',
        // color: 'yellow',
        transparent: !0,
        opacity: 0.3,
        // 透明度
        side: i.DoubleSide,
        // 两面都可见
        envMap: void 0,
        depthWrite: !1
      });
      d = new i.Mesh(o, t), d.position.set(S.yEdgeDistance, e.boxHeight / 2, 0), g.add(d);
    }, xe = () => {
      R && (P >= 1.5 && (P = 0), P += 5e-3, R.uniforms.time.value = P, q.uniforms.time.value = P, G.uniforms.time.value = P, j.uniforms.time.value = P);
    }, Q = () => {
      O && O.update(), xe(), H.render(g, b), K = requestAnimationFrame(Q);
    }, ee = () => {
      l = document.querySelector(".t-chart-area-3d-canvas"), b.aspect = (l == null ? void 0 : l.offsetWidth) / (l == null ? void 0 : l.offsetHeight), b.updateProjectionMatrix(), H.setSize(l == null ? void 0 : l.offsetWidth, l == null ? void 0 : l.offsetHeight);
    }, te = (o) => {
      e.tooltipVisible = !1, e.mouseEnter = !1, d && (d.visible = !1);
    }, oe = (o) => {
      var t;
      (s.showLegend && ((t = e.legendList) != null && t.some((n) => n.visible)) || !s.showLegend) && (e.mouseEnter = !0, d ? d.visible || (d.visible = !0) : ve());
    };
    function pe(o) {
      const t = l.getBoundingClientRect();
      return {
        x: (o.clientX - t.left) * l.width / t.width,
        y: (o.clientY - t.top) * l.height / t.height
      };
    }
    const ne = Pe((o) => {
      var t, n, c, r, a, h, u, w;
      if (o.preventDefault(), s.showTooltip) {
        V || (V = document.querySelector(".t-chart-area-3d-tooltip")), l = document.querySelector("#t-chart-area-3d-canvas"), l.getBoundingClientRect();
        const y = pe(o), x = new i.Vector2();
        x.x = y.x / l.offsetWidth * 2 - 1, x.y = -(y.y / l.offsetHeight) * 2 + 1;
        const m = new i.Raycaster();
        m.linePrecision = 200, m == null || m.setFromCamera(x, b);
        const p = m == null ? void 0 : m.intersectObjects(g.children);
        let L = 0;
        if (p.length > 0) {
          p[0].object;
          const { point: E } = p[0];
          let D;
          if (E.x > S.yEdgeDistance) {
            const { yEdgeDistance: f } = S, { x: k } = E, N = (k - f) / ((e.boxLong - 2 * f) / (2 * (((t = s.data) == null ? void 0 : t.length) - 1)));
            L = Math.ceil((Math.ceil(N) + 1) / 2), L > ((n = s.data) == null ? void 0 : n.length) && (L = (c = s.data) == null ? void 0 : c.length);
            const B = (r = e.xAxisData[L - 1]) == null ? void 0 : r.point[0];
            d == null || d.position.set(B, e.boxHeight / 2, 0), D = new i.Vector3(B, e.boxHeight / 2, 0);
          } else
            L = 1, d == null || d.position.set(S.yEdgeDistance, e.boxHeight / 2, 0), D = new i.Vector3(S.yEdgeDistance, e.boxHeight / 2, 0);
          const v = (a = D == null ? void 0 : D.clone()) == null ? void 0 : a.project(b);
          v.x = (v.x + 1) * l.offsetWidth / 2, v.y = -(v.y - 1) * l.offsetHeight / 2, V.style.left = `${v.x + 20}px`, V.style.top = `${v.y - V.offsetHeight / 2}px`, e.tooltipList = [];
          for (let f = 0; f < ((h = s.series) == null ? void 0 : h.length); f++) {
            const k = {
              name: s.series[f].name,
              color: (u = s.series[f]) == null ? void 0 : u.color,
              value: s.data[L - 1][(w = s.series[f]) == null ? void 0 : w.key]
            };
            e.tooltipList.push(k);
          }
          e.tooltipVisible = s.showTooltip && e.mouseEnter;
        }
      }
    }, 100), Me = (o) => {
      var t, n;
      e.legendList[o].visible = !((t = e.legendList[o]) != null && t.visible), (n = e.legendList) == null || n.forEach((c) => {
        var r;
        (r = e.highlightMap[c.key]) == null || r.forEach((a) => {
          a.visible = c.visible;
        });
      });
    }, ie = () => {
      if (l = document.querySelector("#t-chart-area-3d-canvas"), l) {
        H = new i.WebGLRenderer({
          antialias: !0,
          alpha: !0,
          canvas: l
        }), H.setPixelRatio(window.devicePixelRatio), H.setSize(l.offsetWidth, l.offsetHeight), H.setClearColor(0, 0);
        const o = 20, t = l.offsetWidth / l.offsetHeight, n = 0.1, c = 200;
        b = new i.PerspectiveCamera(o, t, n, c), b.position.set(5, 10, 20), O = new Ee(b, l), O.target.set(e.boxLong / 2, e.boxHeight / 2, 0), O.update(), g = new i.Scene(), d && g.add(d), g.add(new i.HemisphereLight(12303291, 8947848, 3));
        const r = new i.DirectionalLight(16777215, 3);
        r.position.set(0, 0, 1).normalize(), g.add(r);
        const a = new i.DirectionalLight(16777215, 3);
        a.position.set(0, -1, 0).normalize(), g.add(a);
        const h = new i.DirectionalLight(16777215, 3);
        h.position.set(0, 0, -1).normalize(), g.add(h);
        const u = new He(
          e.boxLong,
          e.boxHeight,
          e.boxDept,
          1,
          M.segment,
          1
        ), w = new i.Color(S.boxLineColor), y = new i.LineBasicMaterial({
          color: w,
          linewidth: 0.01,
          side: i.DoubleSide
        }), x = new i.LineSegments(
          u,
          y
        );
        x.position.set(e.boxLong / 2, e.boxHeight / 2, 0), g.add(x), Q();
      }
    };
    return Ce(() => {
      var o, t, n;
      se(() => {
        e.resizeObserve || (e.resizeObserve = new ResizeObserver((c) => {
          for (const r of c)
            r.target === T.value && (l = document.querySelector(".t-chart-area-3d-canvas"), l.style.width = "100%", l.style.height = "100%");
        })), e.resizeObserve.observe(T.value);
      }), Z(), ie(), window.addEventListener("resize", ee), (o = document.querySelector(".t-chart-area-3d-canvas")) == null || o.addEventListener("mousemove", ne), (t = document.querySelector(".t-chart-area-3d-canvas")) == null || t.addEventListener("mouseleave", te), (n = document.querySelector(".t-chart-area-3d-canvas")) == null || n.addEventListener("mouseenter", oe);
    }), Se(() => {
      var o, t, n, c;
      window.removeEventListener("resize", ee), (o = document.querySelector(".t-chart-area-3d-canvas")) == null || o.removeEventListener("mousemove", ne), (t = document.querySelector(".t-chart-area-3d-canvas")) == null || t.removeEventListener("mouseleave", te), (n = document.querySelector(".t-chart-area-3d-canvas")) == null || n.removeEventListener("mouseenter", oe), cancelAnimationFrame(K), T.value && ((c = e.resizeObserve) == null || c.unobserve(T.value));
    }), ae(
      () => s.cStyle,
      () => {
        Z();
      },
      { deep: !0 }
    ), ae(
      () => [
        s.series,
        s.data,
        s.showLegend,
        s.showTooltip,
        s.legendPosition
      ],
      () => {
        var o;
        (o = s.data) != null && o.length && se(() => {
          var t;
          cancelAnimationFrame(K), e.legendList = [], s.showLegend ? (t = s.series) == null || t.forEach((n) => {
            const c = {
              name: n.name,
              color: n.color,
              key: n.key,
              visible: n.visible
              // 默认显示
            };
            e.legendList.push(c);
          }) : e.legendList = [], ie(), he(), ce(), ge(), de(), me(), ye(), fe();
        });
      },
      { immediate: !0, deep: !0 }
    ), {
      ...Ae(e),
      curStyle: Y,
      domOnlyClass: _,
      chartArea3dRef: T,
      legendItemClick: Me
    };
  }
});
export {
  nt as default
};
