// @ths/design/es/components/rank-contrast/src/index.vue2.js
import { defineComponent as s, computed as u, watch as c, onMounted as m } from "vue";
import { getDomRandomClass as p, setStyle as e } from "../../../utils/index.js";
import "../../proportion-bar/index.js";
import f from "../../proportion-bar/src/index.vue.js";
const B = s({
  name: "TRankContrast",
  components: { TProportionBar: f },
  props: {
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        name: {
          default: {}
        },
        arrow: {
          default: {}
        },
        proportionBarBg: {
          default: {}
        },
        currentProportionBar: {
          default: {}
        },
        targetProportionBar: {
          default: {}
        },
        currentValue: {
          default: {}
        },
        targetValue: {
          default: {}
        }
      })
    }
  },
  setup(n) {
    const r = u(() => n.cStyle), a = p("rank-contrast"), l = u(() => n.data), i = () => {
      let t = "";
      return r.value.wrapper && (t += e(r.value.wrapper, `.${a}`)), r.value.name && (t += e(r.value.name, `.${a} .rank-contrast-item .name`)), r.value.arrow && (t += e(r.value.arrow, `.${a} .rank-contrast-item .arrow`)), r.value.proportionBarBg && (t += e(r.value.proportionBarBg, `.${a} .rank-contrast-item .ranks .ranks-item .proportion-bar__outer`)), r.value.currentProportionBar && (t += e(r.value.currentProportionBar, `.${a} .rank-contrast-item .ranks .ranks-item--current .proportion-bar__inner`)), r.value.targetProportionBar && (t += e(r.value.targetProportionBar, `.${a} .rank-contrast-item .ranks .ranks-item--target .proportion-bar__inner`)), r.value.currentValue && (t += e(r.value.currentValue, `.${a} .rank-contrast-item .ranks .ranks-item--current .value`)), r.value.targetValue && (t += e(r.value.targetValue, `.${a} .rank-contrast-item .ranks .ranks-item--target .value`)), t;
    }, o = () => {
      const t = document.querySelector(`.${a} .style-container`);
      t && (t.innerHTML = `<style>${i()}}</style>`);
    };
    return c(() => n.cStyle, () => {
      o();
    }), m(() => {
      o();
    }), {
      curData: l,
      curStyle: r,
      domOnlyClass: a
    };
  }
});
export {
  B as default
};
