// @ths/design/es/components/button-full-screen/src/index.vue2.js
import { defineComponent as f, computed as s, reactive as p, onMounted as y, watch as S, toRefs as v } from "vue";
import { getDomRandomClass as F, setStyle as u } from "../../../utils/index.js";
const C = f({
  name: "TButtonFullScreen",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {},
          hover: {}
        },
        // 展开按钮容器
        fullWrapper: {
          default: {},
          hover: {}
        },
        // 收缩按钮容器
        shrinkWrapper: {
          default: {},
          hover: {}
        }
      })
    },
    // 需要单独全屏的容器id
    containerId: {
      type: String,
      default: ""
    },
    // 是否自定义按钮图片
    designButton: {
      type: Boolean,
      default: !1
    }
  },
  setup(l) {
    const n = s(() => l.cStyle), o = s(() => l.containerId), e = p({
      isFullScreen: !0,
      // 是否显示展开按钮，默认true显示
      docElement: {}
      // html元素
    }), r = F("button-full-screen"), a = () => {
      let t = "";
      return n.value.wrapper && (t += u(n.value.wrapper, `.${r}`)), n.value.fullWrapper && l.designButton && (t += u(n.value.fullWrapper, `.${r} .full-button`)), n.value.shrinkWrapper && l.designButton && (t += u(n.value.shrinkWrapper, `.${r} .shrink-button`)), t;
    }, c = () => {
      const t = document.querySelector(`.${r} .style-container`);
      t && (t.innerHTML = `<style>${a()}}</style>`);
    }, d = () => {
      e.docElement = {}, o.value ? e.docElement = document.getElementById(o.value) : e.docElement = document.documentElement;
    }, i = () => {
      e.docElement && e.docElement.requestFullscreen && !document.fullscreen && (e.docElement.requestFullscreen(), e.isFullScreen = !1);
    }, m = () => {
      document.fullscreen && document.exitFullscreen(), e.isFullScreen = !0;
    };
    return window.onresize = () => {
      e.isFullScreen = !document.fullscreen;
    }, y(() => {
      e.isFullScreen = !0, d(), c();
    }), S(
      () => l.cStyle,
      () => {
        c();
      },
      {
        deep: !0
      }
    ), {
      ...v(e),
      domOnlyClass: r,
      handleFullScreenClick: i,
      handleCloseFullClick: m
    };
  }
});
export {
  C as default
};
