// @ths/design/es/components/button-full-screen/src/index.vue.js
import d from "./index.vue2.js";
import { resolveComponent as C, openBlock as o, createElementBlock as i, normalizeClass as m, createBlock as a, withCtx as u, createElementVNode as n, createCommentVNode as t } from "vue";
import "./index.vue3.js";
import k from "../../../_virtual/_plugin-vue_export-helper.js";
function v(l, e, p, c, f, V) {
  const s = C("el-icon");
  return o(), i("div", {
    class: m(["t-button-full-screen", l.domOnlyClass])
  }, [
    l.isFullScreen && !l.designButton ? (o(), a(s, {
      key: 0,
      onClick: l.handleFullScreenClick
    }, {
      default: u(() => e[2] || (e[2] = [
        n("svg", {
          xmlns: "http://www.w3.org/2000/svg",
          viewBox: "0 0 1024 1024"
        }, [
          n("path", {
            fill: "currentColor",
            d: `m160 96.064 192 .192a32 32 0 0 1 0 64l-192-.192V352a32 32 0 0 1-64 0V96h64v.064zm0 831.872V928H96V672a32 32 0 1 1 64 0v191.936l192-.192a32 32
          0 1 1 0 64l-192 .192zM864 96.064V96h64v256a32 32 0 1 1-64 0V160.064l-192 .192a32 32 0 1 1 0-64l192-.192zm0 831.872-192-.192a32 32 0 0 1 0-64l192
          .192V672a32 32 0 1 1 64 0v256h-64v-.064z`
          })
        ], -1)
      ])),
      _: 1,
      __: [2]
    }, 8, ["onClick"])) : t("", !0),
    l.isFullScreen && l.designButton ? (o(), i("div", {
      key: 1,
      class: "full-button button-container",
      onClick: e[0] || (e[0] = (...r) => l.handleFullScreenClick && l.handleFullScreenClick(...r))
    })) : t("", !0),
    !l.isFullScreen && !l.designButton ? (o(), a(s, {
      key: 2,
      onClick: l.handleCloseFullClick
    }, {
      default: u(() => e[3] || (e[3] = [
        n("svg", {
          xmlns: "http://www.w3.org/2000/svg",
          viewBox: "0 0 1024 1024"
        }, [
          n("path", {
            fill: "currentColor",
            d: `M768 832a128 128 0 0 1-128 128H192A128 128 0 0 1 64 832V384a128 128 0 0 1
          128-128v64a64 64 0 0 0-64 64v448a64 64 0 0 0 64 64h448a64 64 0 0 0 64-64h64z`
          }),
          n("path", {
            fill: "currentColor",
            d: `M384 128a64 64 0 0 0-64 64v448a64 64 0 0 0 64 64h448a64 64 0 0 0 64-64V192a64
          64 0 0 0-64-64H384zm0-64h448a128 128 0 0 1 128 128v448a128 128 0 0 1-128 128H384a128 128 0 0 1-128-128V192A128 128 0 0 1 384 64z`
          })
        ], -1)
      ])),
      _: 1,
      __: [3]
    }, 8, ["onClick"])) : t("", !0),
    !l.isFullScreen && l.designButton ? (o(), i("div", {
      key: 3,
      class: "shrink-button button-container",
      onClick: e[1] || (e[1] = (...r) => l.handleCloseFullClick && l.handleCloseFullClick(...r))
    })) : t("", !0),
    e[4] || (e[4] = n("div", { class: "style-container" }, null, -1))
  ], 2);
}
const B = /* @__PURE__ */ k(d, [["render", v]]);
export {
  B as default
};
