// @ths/design/es/components/chart-curve/src/index.vue2.js
import { defineComponent as P, computed as k, ref as U, reactive as B, nextTick as x, onMounted as H, onUnmounted as J, watch as z, toRefs as W } from "vue";
import * as w from "echarts";
import r from "lodash/merge";
import s from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as K, DEFAULT_OPTIONS_AXIS as X } from "../../../utils/constant.js";
import { getDomRandomClass as Z, setStyle as A } from "../../../utils/index.js";
const Q = r(s(K), s(X)), N = {
  show: !0,
  name: "(人/平方公里)",
  nameTextStyle: {
    color: "#666",
    align: "right",
    fontSize: 12,
    padding: [0, -32, 5, 0]
  },
  type: "value",
  splitNumber: 4,
  splitLine: {
    show: !0,
    lineStyle: {
      color: "rgba(202,202,202,0.3)"
    }
  },
  axisLine: {
    show: !1,
    lineStyle: {
      color: "#cacaca",
      width: 1,
      type: "solid"
    }
  },
  axisLabel: {
    show: !0,
    margin: 20,
    color: "#666",
    fontSize: 12
  },
  axisTick: {
    show: !1
  }
}, _ = {
  type: "value",
  name: "(万人)",
  nameTextStyle: {
    color: "#666",
    align: "left",
    fontSize: 12,
    padding: [0, 0, 5, 15]
  },
  position: "right",
  splitLine: {
    show: !1
  },
  axisLabel: {
    show: !0,
    margin: 20,
    fontSize: 12,
    color: "#666"
  },
  axisLine: {
    show: !1,
    lineStyle: {
      color: "#cacaca",
      width: 1,
      type: "solid"
    }
  }
}, i = r(
  s(Q),
  {
    // 气泡框
    tooltip: {
      padding: [10, 10],
      axisPointer: {
        type: "line"
      }
    },
    // 网格线
    grid: {
      containLabel: !0,
      top: "13%",
      bottom: "3%",
      left: "2%",
      right: "2%"
    },
    // x轴
    xAxis: {
      axisLine: {
        show: !0
      },
      splitArea: {
        // 坐标轴在 grid 区域中的分隔区域,默认不显示。
        color: "#f00",
        lineStyle: {
          color: "#f00"
        }
      },
      axisLabel: {
        interval: "auto"
      },
      splitLine: {
        show: !1
      }
    },
    dataZoom: [
      {
        // 禁用缩放
        disabled: !1,
        type: "inside"
      },
      {
        // 隐藏滑块
        show: !1,
        type: "slider"
      }
    ]
  }
), oe = P({
  name: "TChartCurve",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据 如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [
        {
          key1: 2,
          key2: 2,
          key3: "label"
        }
      ],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 左y轴配置
    yAxisLeftOptions: {
      type: Object,
      default: () => ({})
    },
    // 右y轴配置
    yAxisRightOptions: {
      type: Object,
      default: () => ({})
    },
    // 折线主题配置项
    line: {
      type: Object,
      default: () => ({})
    }
  },
  // 方法名
  emits: ["legend-select"],
  setup(t, { emit: $ }) {
    const g = Z("chart-curve"), f = k(() => t.cStyle), h = U(null);
    let S = null, d = null;
    const n = B({
      // 图表数据
      chartData: [],
      // 主题配置项数据
      lineOptions: {}
    });
    function D() {
      var e, c;
      (c = (e = t.seriesSetting) == null ? void 0 : e.data) != null && c.length && t.data && (n.chartData = t.seriesSetting.data.map((y) => {
        const { name: u, dataKey: m } = y;
        return {
          name: u || "--",
          data: t.data.map((l) => l[m])
        };
      }));
    }
    const q = () => {
      let e = "";
      return f.value.wrapper && (e += A(f.value.wrapper, `.${g}`)), f.value.emptyContainer && (e += A(f.value.emptyContainer, `.${g} .empty`)), f.value.emptyImg && (e += A(f.value.emptyImg, `.${g} .empty .empty-img`)), f.value.emptyText && (e += A(f.value.emptyText, `.${g} .empty .empty-text`)), e;
    }, T = () => {
      const e = document.querySelector(`.${g} .style-container`);
      e && (e.innerHTML = `<style>${q()}</style>`);
    }, G = k(() => {
      var e, c, y, u;
      if (t.options) {
        if (Array.isArray((e = t.options) == null ? void 0 : e.series))
          return t.options.series.some((l) => {
            var p;
            return (l == null ? void 0 : l.data) && ((p = l == null ? void 0 : l.data) == null ? void 0 : p.length) !== 0;
          });
        if (typeof ((c = t.options) == null ? void 0 : c.series) == "object")
          return ((u = (y = t.options.series) == null ? void 0 : y.data) == null ? void 0 : u.length) === 0;
      }
      return t.data.length ? (x(() => {
        d.resize();
      }), !0) : !1;
    });
    function L() {
      var p, I, R, j;
      const { color: e, series: c } = t.options;
      let y = [], u = {}, m = [];
      if (c ? y = c.map((o, a) => {
        var O, F, b;
        const C = r(
          r(
            {
              name: "",
              // 使用的 y 轴的 index，在单个图表实例中存在多个 y轴的时候有用
              yAxisIndex: a,
              type: "line",
              smooth: !0,
              symbol: "circle",
              symbolSize: 8,
              areaStyle: {
                color: new w.graphic.LinearGradient(0, 0, 0, 1, [
                  {
                    offset: 0,
                    color: e && e[a] || i.color[a]
                    // 100% 处的颜色
                  },
                  {
                    offset: 1,
                    color: "transparent"
                  }
                ])
              },
              itemStyle: {
                color: new w.graphic.LinearGradient(1, 1, 0, 1, [
                  {
                    offset: 0,
                    color: e && e[a] || i.color[a]
                    // 100% 处的颜色
                  },
                  {
                    offset: 0.5,
                    color: "#fff"
                  },
                  {
                    offset: 1,
                    color: e && e[a] || i.color[a] || "#FFF"
                  }
                ]),
                borderWidth: 2,
                borderColor: e && e[a] || i.color[a] || "#FFF",
                shadowColor: e && e[a] || i.color[a] || "#FFF",
                shadowBlur: 10
              },
              lineStyle: {
                color: e && e[a] || i.color[a] || "#FFF"
              }
            },
            n == null ? void 0 : n.lineOptions
          ),
          (b = (F = (O = t.seriesSetting) == null ? void 0 : O.data) == null ? void 0 : F[a]) == null ? void 0 : b.style
        );
        return r(s(C), s(o));
      }) : (I = (p = t.seriesSetting) == null ? void 0 : p.data) != null && I.length && t.data && (y = t.seriesSetting.data.map((o, a) => {
        var O, F, b, E;
        const C = r(
          {
            name: o == null ? void 0 : o.name,
            // 使用的 y 轴的 index，在单个图表实例中存在多个 y轴的时候有用
            yAxisIndex: a,
            type: "line",
            smooth: !0,
            symbol: "circle",
            symbolSize: 8,
            areaStyle: {
              color: new w.graphic.LinearGradient(0, 0, 0, 1, [
                {
                  offset: 0,
                  color: e && e[a] || i.color[a] || "#FFF"
                },
                {
                  offset: 1,
                  color: "transparent"
                }
              ])
            },
            itemStyle: {
              color: new w.graphic.LinearGradient(1, 1, 0, 1, [
                {
                  offset: 0,
                  color: e && e[a] || i.color[a] || "#FFF"
                },
                {
                  offset: 0.5,
                  color: "#fff"
                },
                {
                  offset: 1,
                  color: e && e[a] || i.color[a] || "#FFF"
                }
              ]),
              borderWidth: 2,
              borderColor: e && e[a] || i.color[a] || "#FFF",
              shadowColor: e && e[a] || i.color[a] || "#FFF",
              shadowBlur: 10
            },
            lineStyle: {
              color: e && e[a] || i.color[a] || "#FFF"
            },
            data: t.data.length ? (O = n.chartData[a]) == null ? void 0 : O.data : []
          },
          n == null ? void 0 : n.lineOptions,
          (E = (b = (F = t.seriesSetting) == null ? void 0 : F.data) == null ? void 0 : b[a]) == null ? void 0 : E.style
        );
        return r(s(C), s(o));
      }), (R = t.seriesSetting) != null && R.xAxisLabel ? u = (j = t.data) != null && j.length ? t.data.map((o) => o[t.seriesSetting.xAxisLabel]) : [] : u = t.data.map((o, a) => a)), m = r([], [N, _]), JSON.stringify(t.yAxisLeftOptions) !== "{}") {
        let o = s(N);
        o = r(o, t.yAxisLeftOptions), m[0] = o;
      }
      if (JSON.stringify(t.yAxisRightOptions) !== "{}") {
        let o = s(_);
        o = r(o, t.yAxisRightOptions), m[1] = o;
      }
      let l = r(
        {
          series: y,
          xAxis: {
            data: u
          },
          yAxis: m
        },
        i
      );
      l = r(l, t.options), d.clear();
      try {
        d.setOption(l);
      } catch (o) {
        console.log(o);
      }
    }
    function v() {
      x(() => {
        d.resize();
      });
    }
    function M() {
      h.value && (S = new ResizeObserver((e) => {
        for (const c of e)
          c.target === h.value && v();
      }), S.observe(h.value));
    }
    return H(() => {
      T(), h.value && (d = w.init(h.value), n.lineOptions = s(t.line), D(), L(), d.on("legendselectchanged", (e) => {
        $("legend-select", e);
      }), M()), window.addEventListener("resize", v);
    }), J(() => {
      window.removeEventListener("resize", v), S && (S.disconnect(), S = null);
    }), z(
      () => [t.yAxisLeftOptions, t.yAxisRightOptions, t.line],
      (e) => {
        n.lineOptions = s(t.line), L(), v();
      },
      {
        deep: !0
      }
    ), z(
      [
        () => t.options,
        () => t.data,
        () => t.seriesSetting
      ],
      () => {
        D(), L();
      },
      {
        deep: !0
      }
    ), z(
      () => t.cStyle,
      () => {
        T(), x(() => {
          d.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      ...W(n),
      chartId: h,
      curStyle: f,
      checkEmpty: G,
      domOnlyClass: g
    };
  }
});
export {
  oe as default
};
