// @ths/design/es/components/drawer/src/index.vue.js
import m from "./index.vue2.js";
import { resolveComponent as n, openBlock as t, createElementBlock as u, createVNode as f, withCtx as d, createElementVNode as l, normalizeStyle as s, renderSlot as r, createBlock as y, createCommentVNode as w } from "vue";
import "./index.vue3.js";
import v from "../../../_virtual/_plugin-vue_export-helper.js";
const C = { class: "header" };
function E(e, o, c, S, V, $) {
  const a = n("el-icon"), i = n("el-drawer");
  return t(), u("div", null, [
    f(i, {
      modelValue: e.dialog,
      "onUpdate:modelValue": o[1] || (o[1] = (p) => e.dialog = p),
      size: e.width,
      "with-header": !1,
      "before-close": () => e.handleEmit("close"),
      modal: e.modal,
      "append-to-body": e.appendToBody,
      "close-on-press-escape": e.closeOnPressEscape,
      "destroy-on-close": e.destroyOnClose,
      direction: e.openDirection,
      onOpen: o[2] || (o[2] = () => e.handleEmit("open")),
      onOpened: o[3] || (o[3] = () => e.handleEmit("opened")),
      onClosed: o[4] || (o[4] = () => e.handleEmit("closed"))
    }, {
      default: d(() => [
        l("div", {
          class: "t-drawer",
          style: s(e.customStyle.wrapper.default)
        }, [
          l("div", C, [
            r(e.$slots, "title"),
            o[6] || (o[6] = l("span", null, null, -1)),
            e.isShowClose ? (t(), y(a, {
              key: 0,
              class: "el-icon-close",
              style: s(e.customStyle.icon.default),
              onClick: o[0] || (o[0] = () => e.handleEmit("close"))
            }, {
              default: d(() => o[5] || (o[5] = [
                l("svg", {
                  xmlns: "http://www.w3.org/2000/svg",
                  viewBox: "0 0 1024 1024"
                }, [
                  l("path", {
                    fill: "currentColor",
                    d: `M764.288 214.592 512 466.88 259.712 214.592a31.936 31.936 0 0 0-45.12 45.12L466.752 512 214.528 764.224a31.936 31.936 0 1 0 45.12
                45.184L512 557.184l252.288 252.288a31.936 31.936 0 0 0 45.12-45.12L557.12 512.064l252.288-252.352a31.936 31.936 0 1 0-45.12-45.184z`
                  })
                ], -1)
              ])),
              _: 1,
              __: [5]
            }, 8, ["style"])) : w("", !0)
          ]),
          l("div", {
            class: "content",
            style: s(e.customStyle.content.default)
          }, [
            r(e.$slots, "default")
          ], 4)
        ], 4)
      ]),
      _: 3
    }, 8, ["modelValue", "size", "before-close", "modal", "append-to-body", "close-on-press-escape", "destroy-on-close", "direction"])
  ]);
}
const z = /* @__PURE__ */ v(m, [["render", E]]);
export {
  z as default
};
