// @ths/design/es/components/drawer/src/index.vue2.js
import { defineComponent as d, reactive as i, computed as r, watch as s, toRefs as c } from "vue";
const f = d({
  name: "TDrawer",
  props: {
    // 弹窗是否可见
    visible: {
      type: Boolean,
      default: !1
    },
    // 宽度
    width: {
      type: String,
      default: "30%"
    },
    // 是否需要遮罩层
    modal: {
      type: Boolean,
      default: !0
    },
    // 是否可以通过按下 ESC 关闭 Drawer
    closeOnPressEscape: {
      type: Boolean,
      default: !0
    },
    // 控制是否在关闭 Drawer 之后将子元素全部销毁 ?
    destroyOnClose: {
      type: Boolean,
      default: !0
    },
    // Drawer 自身是否插入至 body 元素上。嵌套的 Drawer 必须指定该属性并赋值为 true
    appendToBody: {
      type: Boolean,
      default: !1
    },
    // Drawer 打开的方向 可选值：rtl（从右往左开） / ltr（从左往右开） / ttb（从上往下开） / btt（从下往上开）
    direction: {
      type: String,
      default: "rtl"
    },
    // 是否显示关闭按钮
    isShowClose: {
      type: Boolean,
      default: !0
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        content: {
          default: {}
        },
        icon: {
          default: {}
        }
      })
    }
  },
  emits: ["close", "open", "opened", "closed"],
  setup(e, l) {
    const a = ["rtl", "ltr", "ttb", "btt"], o = i({
      customStyle: e.cStyle,
      // 组件样式
      // 弹窗是否可见
      dialog: !1,
      // Drawer 打开的方向
      openDirection: r(() => a.includes(e.direction) ? e.direction : "rtl")
    });
    function n(t) {
      l.emit(t);
    }
    return s(
      () => e.visible,
      (t) => {
        o.dialog = t;
      }
    ), {
      ...c(o),
      handleEmit: n
    };
  }
});
export {
  f as default
};
