// @ths/design/es/components/chart-dashboard-aqi-segment/src/default-options.js
import { merge as o, cloneDeep as e } from "lodash";
import { DEFAULT_OPTIONS_COMMON as t, AQI_COLOR as i, AQI_LEVEL_NAME as s } from "../../../utils/constant.js";
const r = o(
  e(t),
  {
    tooltip: {
      show: !1
    },
    xAxis: {
      show: !1
    },
    yAxis: {
      show: !1
    }
  }
), n = {
  axisTickWidth: 2,
  // 刻度宽度
  axisTickHeight: 5,
  // 刻度高度
  axisTickColor: "rgba(76,179,228,0.15)",
  // 刻度背景色
  axisTickDistance: 10,
  // 刻度和轴线之间的距离
  splitNumber: 1,
  // 刻度的分割段数
  showAxisTick: !0
}, a = {
  left: "59%",
  top: "36%",
  fontSize: "18px",
  lineHeight: "36px",
  minWidth: "56px",
  height: "36px",
  padding: "0 10px",
  borderWidth: "1px",
  borderStyle: "solid",
  borderRadius: "20px 20px 20px 0px",
  color: "#fff",
  backgroundImage: "",
  backgroundPosition: "center",
  backgroundSize: "cover",
  backgroundRepeat: "no-repeat"
}, p = {
  color: "#fff",
  // 字体颜色
  fontSize: "48px",
  // 字体大小
  fontWeight: "normal",
  // 字体大小
  offsetCenter: [0, -60]
  // 距离中心点的偏移量
}, l = {
  fontSize: 16,
  color: "#B7E3FF",
  fontWeight: "normal",
  offsetCenter: [0, -20]
}, x = {
  data: "",
  // 仪表盘数据
  name: "",
  // 仪表盘文字数据
  options: r,
  // echarts原生配置
  levelColors: e(i),
  // 等级颜色
  levelDesc: e(s),
  // 等级文字描述
  isFixed: !0,
  // 刻度值是否需要保留小数
  fixedNumber: 1,
  // 刻度值保留小数位数
  bgColor: "rgb(5,13,21)",
  // 背景色
  splitNumber: 40,
  // 刻度个数
  showSegments: !0,
  // 是否展示分隔线
  splitLineWidth: 2,
  // 分隔线宽度
  progressWidth: 15,
  // 进度条总体宽度
  axisTickOptions: { ...n },
  // 刻度的配置
  aqiValueOptions: { ...p },
  // AQI 数值部分的配置
  labelOptions: { ...l },
  // 文本的配置
  statusStyle: { ...a },
  // 状态的样式
  statusBgColorOpacity: {
    start: 0,
    end: 0.53
  }
}, f = {
  wrapper: {
    default: {
      width: "400px",
      height: "228px"
    }
  }
};
export {
  p as DEFAULT_AQI_VALUE_OPTIONS,
  n as DEFAULT_AXISTICK_OPTIONS,
  r as DEFAULT_CHART_OPTIONS,
  f as DEFAULT_CUSTOM_STYLE,
  l as DEFAULT_LABEL_OPTIONS,
  x as DEFAULT_PROPS_VALUE,
  a as DEFAULT_STATUS_STYLE
};
