// @ths/design/es/components/chart-dashboard-aqi-segment/src/index.vue2.js
import { defineComponent as Q, computed as d, ref as R, onMounted as Y, onUnmounted as K, watch as N, nextTick as D, useCssVars as X } from "vue";
import * as G from "echarts";
import { merge as s, cloneDeep as k } from "lodash";
import { getAQILevel as x, colorRgb as E, getDomRandomClass as J, getValueOrDefaultValue as Z, setStyle as W } from "../../../utils/index.js";
import { AQI_COLOR as tt, AQI_LEVEL_NAME as et } from "../../../utils/constant.js";
import { DEFAULT_CUSTOM_STYLE as V, DEFAULT_PROPS_VALUE as i, DEFAULT_CHART_OPTIONS as it, DEFAULT_AXISTICK_OPTIONS as m, DEFAULT_AQI_VALUE_OPTIONS as C, DEFAULT_LABEL_OPTIONS as T, DEFAULT_STATUS_STYLE as at } from "./default-options.js";
const ot = "#6E7683", lt = "level7", L = J("chart-dashboard-aqi-segment"), q = Q({
  name: "TChartDashboardAqiSegment",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: V.wrapper
      })
    },
    // 仪表盘数据
    data: {
      type: String,
      default: () => i.data
    },
    // 仪表盘文字数据
    name: {
      type: String,
      default: () => i.name
    },
    // echarts原生配置（优先级比seriesStyle和gauge高）
    options: {
      type: Object,
      default: () => i.options,
      required: !1
    },
    // 等级颜色
    levelColors: {
      type: Object,
      default: () => i.levelColors
    },
    // 等级文字描述
    levelDesc: {
      type: Object,
      default: () => i.levelDesc
    },
    // 背景色
    bgColor: {
      type: String,
      default: i.bgColor
    },
    // 分隔线个数
    splitNumber: {
      type: Number,
      default: i.splitNumber
    },
    // 是否展示分隔线
    showSegments: {
      type: Boolean,
      default: i.showSegments
    },
    // 分隔线宽度
    splitLineWidth: {
      type: Number,
      default: i.splitLineWidth
    },
    // 进度条宽度
    progressWidth: {
      type: Number,
      default: i.progressWidth
    },
    // 刻度相关配置
    axisTickOptions: {
      type: Object,
      default: i.axisTickOptions
    },
    // AQI数值部分的配置
    aqiValueOptions: {
      type: Object,
      default: i.aqiValueOptions
    },
    // 文本选项
    labelOptions: {
      type: Object,
      default: i.labelOptions
    },
    // 状态的样式
    statusStyle: {
      type: Object,
      default: i.statusStyle
    },
    // 状态的背景色的颜色透明度，从中间开始渐变。仅支持传入两个参数的情况
    statusBgColorOpacity: {
      type: Object,
      default: i.statusBgColorOpacity
    }
  },
  setup(t) {
    const b = d(() => s(
      t.cStyle,
      V
    )), U = d(() => s(k(tt), t.levelColors)), z = d(() => s(k(et), t.levelDesc)), v = () => {
      var o, r, c, u, l, a, g, O, y;
      let e;
      if ((o = t.options) != null && o.series) {
        const n = Object.prototype.toString.call((r = t.options) == null ? void 0 : r.series);
        let S;
        if (n === "[object Array]" && ((u = (c = t.options) == null ? void 0 : c.series) != null && u.length) && (e = (a = (l = t.options) == null ? void 0 : l.series[0]) == null ? void 0 : a.data), n === "[object Object]" && ((g = t.options.series) != null && g.data) && (e = (O = t.options.series) == null ? void 0 : O.data), !e)
          return "";
        if (Object.prototype.toString.call(e) === "[object Array]" && (e != null && e.length)) {
          if (S = Object.prototype.toString.call(e[0]), ["[object String]", "[object Number]"].includes(S))
            return e[0].toString();
          if (S === "[object Object]")
            return (y = e[0]) == null ? void 0 : y.value.toString();
        }
      }
      return t.data ? t.data.toString() : "";
    }, h = d(() => U.value[x(v())]), $ = d(
      () => {
        var r, c, u, l;
        const e = t.statusBgColorOpacity.start || i.statusBgColorOpacity.start, o = t.statusBgColorOpacity.end || i.statusBgColorOpacity.end;
        return (r = t.statusStyle) != null && r.background || (c = t.statusStyle) != null && c.backgroundColor ? ((u = t.statusStyle) == null ? void 0 : u.background) || ((l = t.statusStyle) == null ? void 0 : l.backgroundColor) : `radial-gradient(50% 50% at 50% 50%, rgba(${E(h.value)},${e}) 0%, rgba(${E(h.value)},${o}) 100%)`;
      }
    ), F = d(() => z.value[x(v())]), p = d(() => t.bgColor || i.bgColor), A = R(null);
    let f = null;
    const M = () => ({
      type: "gauge",
      radius: "160%",
      startAngle: 180,
      center: ["50%", "90%"],
      endAngle: 0,
      // 刻度的个数
      splitNumber: t.splitNumber,
      min: 0,
      max: 300,
      axisLine: {
        lineStyle: {
          width: t.progressWidth,
          color: [
            // 判断填充色使用等级颜色或自定义颜色 如果传的等级颜色为空 则显示默认的值
            [0, h.value],
            [1, "rgb(16,38,52)"]
          ]
        }
      },
      axisTick: {
        show: Z(t.axisTickOptions.showAxisTick, m.showAxisTick),
        splitNumber: t.axisTickOptions.splitNumber || m.splitNumber,
        length: t.axisTickOptions.axisTickHeight || m.axisTickHeight,
        distance: t.axisTickOptions.axisTickDistance || m.axisTickDistance,
        lineStyle: {
          color: t.axisTickOptions.axisTickColor || m.axisTickColor,
          width: t.axisTickOptions.axisTickWidth || m.axisTickWidth
        }
      },
      // 分隔线样式
      splitLine: {
        show: t.showSegments,
        distance: -t.progressWidth,
        length: t.progressWidth,
        lineStyle: {
          color: t.bgColor || i.bgColor,
          width: t.splitLineWidth
        }
      },
      // AQI数值部分
      detail: {
        valueAnimation: !0,
        formatter: "{value}",
        // 数值颜色为等级色 如果传的等级颜色为空 则显示默认的值
        color: t.aqiValueOptions.color || C.color,
        // color: curLevelColorMap.value?.[getAQILevel(props.data)] || AQI_COLOR[getAQILevel(props.data)],
        fontSize: t.aqiValueOptions.fontSize || C.fontSize,
        offsetCenter: t.aqiValueOptions.offsetCenter || C.offsetCenter,
        fontWeight: t.aqiValueOptions.fontWeight || C.fontWeight
      },
      // AQI文本部分
      title: {
        fontSize: t.labelOptions.fontSize || T.fontSize,
        color: t.labelOptions.color || T.color,
        fontWeight: t.labelOptions.fontWeight || T.fontWeight,
        offsetCenter: t.labelOptions.offsetCenter || T.offsetCenter
      },
      pointer: {
        show: !1
      },
      axisLabel: {
        show: !1
      },
      anchor: {
        show: !1
      }
    }), _ = () => {
      var e, o, r, c, u;
      if (A.value) {
        f || (f = G.init(A.value));
        const l = (e = t.options) == null ? void 0 : e.series;
        let a = {};
        const g = k(M());
        l ? Array.isArray(l) ? a = l.map((n) => s(
          g,
          n
        )) : a = s(
          g,
          l
        ) : t.data !== "" && (a = s(
          g,
          {
            data: [
              {
                name: t.name || i.name,
                value: Number.isNaN(Number(t.data)) ? "0" : Number(t.data)
              }
            ]
          }
        ));
        const O = v();
        if (Object.prototype.toString.call((r = (o = a == null ? void 0 : a.axisLine) == null ? void 0 : o.lineStyle) == null ? void 0 : r.color) === "[object Array]" && ((c = a.axisLine.lineStyle.color) != null && c.length) && ((u = a.axisLine.lineStyle.color[0]) != null && u.length) && typeof O == "string") {
          const n = Number(O), S = n > 300 ? 1 : n / 300;
          a.axisLine.lineStyle.color[0][0] = S;
        }
        let y = s(
          {
            series: a
          },
          it
        );
        y = s(y, t.options), f.clear();
        try {
          (x(t.data) !== lt || l) && f.setOption(y);
        } catch (n) {
          console.error(n);
        }
      }
    }, P = (e) => {
      const o = e["background-image"] || e.backgroundImage;
      return o ? {
        ...e,
        "background-image": o.indexOf("url") === -1 ? `url(${o})` : o
      } : e;
    }, H = () => {
      let e = "";
      return b.value.wrapper && (e += W(b.value.wrapper, `.${L}`)), t.statusStyle && (e += W({ default: s({ ...at }, P(t.statusStyle)) }, `.${L} .status`)), e;
    }, w = () => {
      const e = document.querySelector(`.${L} .style-container`);
      e && (e.innerHTML = `<style>${H()}</style>`);
    }, j = () => {
      D(() => {
        f.resize();
      });
    };
    return Y(() => {
      w(), _(), window.addEventListener("resize", j);
    }), K(() => {
      window.removeEventListener("resize", j);
    }), N(
      () => [
        t.options,
        t.data,
        t.name,
        t.levelColors,
        t.levelDesc,
        t.bgColor,
        t.splitNumber,
        t.showSegments,
        t.splitLineWidth,
        t.progressWidth,
        t.axisTickOptions,
        t.aqiValueOptions,
        t.labelOptions,
        t.statusBgColorOpacity
      ],
      () => {
        _();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), N(
      () => [t.cStyle, t.statusStyle],
      () => {
        w(), D(() => {
          f.resize();
        });
      },
      {
        deep: !0
      }
    ), {
      chart: A,
      curStyle: b,
      curLevelDescription: F,
      curLevelColor: h,
      statusBgColor: $,
      defaultLineColor: ot,
      wrapperBgColor: p,
      domOnlyClass: L
    };
  }
}), I = () => {
  X((t) => ({
    "238a3520": t.wrapperBgColor,
    "1bf046ea": t.curLevelColor,
    "0d59ec17": t.statusBgColor
  }));
}, B = q.setup;
q.setup = B ? (t, b) => (I(), B(t, b)) : I;
export {
  q as default
};
