// @ths/design/es/components/rolling-number/src/index.vue2.js
import { defineComponent as y, computed as m, reactive as S, watch as c, onMounted as p, toRefs as N } from "vue";
const w = y({
  name: "TRollingNumber",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器样式
        wrapper: {
          default: {}
        },
        // 前缀样式
        prefix: {
          default: {}
        },
        // 后缀样式
        suffix: {
          default: {}
        },
        // 分隔符样式
        separator: {
          default: {}
        },
        // 数字样式
        number: {
          default: {
            width: "40px",
            height: "64px"
          }
        }
      })
    },
    // 最小长度
    minLength: {
      type: Number,
      default: 1
    },
    // 数字
    number: {
      type: Number,
      default: 641.17
    },
    // 前缀
    prefix: {
      type: String,
      default: ""
    },
    // 后缀
    suffix: {
      type: String,
      default: ""
    },
    // 动画
    animationDuration: {
      type: Number,
      default: 2
    }
  },
  setup(i, { emit: x }) {
    const d = m(() => i.cStyle), s = S({
      list: [],
      // 当前数字分解成的数组
      numList: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, "."],
      animationDuration: m(() => i.animationDuration)
      // 动画时间
    }), h = (e) => e === "." ? "." : ",", g = (e) => {
      if (!e)
        return "";
      let n = "", t = "";
      const r = e.toString(), a = r.indexOf("."), l = [];
      let f = 0;
      if (a > -1 ? (n = r.substring(a), t = r.substring(0, a)) : t = r, t = t.split(""), i.minLength > 1) {
        let u = 0;
        a > -1 ? u = i.minLength - (t.length + n.length - 1) : u = i.minLength - t.length, u > 0 && (t = new Array(u).fill(0).concat(t));
      }
      for (let u = t.length - 1; u >= 0; u--)
        f++, l.unshift(t[u]), f % 3 === 0 && u !== 0 && l.unshift(",");
      return l.join("") + n;
    }, b = (e) => {
      const n = [];
      return String(g(e)).split("").forEach((r) => {
        n.push({ num: r, translateY: 0 });
      }), n;
    }, o = (e) => {
      var t, r, a;
      const n = parseInt((a = (r = (t = i.cStyle) == null ? void 0 : t.number) == null ? void 0 : r.default) == null ? void 0 : a.height, 10);
      s.list = b(e), s.list.forEach((l) => {
        setTimeout(() => {
          l.translateY = parseFloat(String(Number(l.num) * n + n * 10));
        });
      });
    };
    return c(() => i.number, (e) => {
      o(e);
    }), c(() => {
      var e, n, t;
      return (t = (n = (e = i.cStyle) == null ? void 0 : e.number) == null ? void 0 : n.default) == null ? void 0 : t.height;
    }, () => {
      o(i.number);
    }), p(() => {
      o(i.number);
    }), {
      ...N(s),
      curStyle: d,
      getNumberSeparator: h
    };
  }
});
export {
  w as default
};
