// @ths/design/es/components/tabs-base/src/index.vue2.js
import V from "lodash/cloneDeep";
import M from "lodash/merge";
import { defineComponent as R, computed as B, reactive as E, watch as h, onMounted as N, toRefs as P } from "vue";
import { getDomRandomClass as z, setStyle as f, getCssPropertyString as H } from "../../../utils/index.js";
const O = {
  label: "label",
  value: "value",
  disabled: "disabled"
}, U = R({
  name: "TTabsBase",
  components: {},
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // tabs 容器 (ul.tabs)
        tabsContainer: {
          default: {},
          hover: {}
        },
        // tab 选项卡
        itemWrapper: {
          default: {},
          hover: {},
          active: {},
          disabled: {}
        },
        label: {
          default: {},
          hover: {},
          active: {},
          disabled: {}
        }
      })
    },
    // tabs 标签页数据
    data: {
      type: Array,
      default: () => []
    },
    // tab 之间的间距
    space: {
      type: Number,
      default: 5
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => {
        V(O);
      },
      required: !1
    },
    // 选中项
    selectedTabVal: {
      type: String,
      default: "SO2"
    },
    // 内阴影颜色
    innerShadowColor: {
      type: String,
      default: "transparent"
    },
    // active 状态内阴影颜色
    activeInnerShadowColor: {
      type: String,
      default: "transparent"
    }
  },
  emits: ["tab-click"],
  setup(i, { emit: j }) {
    const t = B(() => i.cStyle), l = z("tabs-base"), v = E({
      data: i.data,
      // 选项卡数据
      selectedVal: i.selectedTabVal,
      // tab 选中的索引
      hoveredTab: null
      // 当前hover的tab
    }), D = (e, a) => {
      j("tab-click", e, a), v.selectedVal = e.value;
    }, k = (e, a) => {
      e.disabled || (v.hoveredTab = a ? e.value : null);
    }, A = (e) => {
      var d, n, u, c, o, b, m, p;
      const a = e.value === v.selectedVal, r = v.hoveredTab === e.value;
      return e.disabled && ((n = (d = t.value) == null ? void 0 : d.label) != null && n.disabled) ? t.value.label.disabled : r && ((c = (u = t.value) == null ? void 0 : u.label) != null && c.hover) ? t.value.label.hover : a && ((b = (o = t.value) == null ? void 0 : o.label) != null && b.active) ? t.value.label.active : ((p = (m = t.value) == null ? void 0 : m.label) == null ? void 0 : p.default) || {};
    }, q = () => {
      var a, r, s, d, n, u, c, o, b, m, p, S, _, $, g, C, W, T, w;
      let e = "";
      if ((a = t.value) != null && a.wrapper && (e += f((r = t.value) == null ? void 0 : r.wrapper, `.${l}`), (n = (d = (s = t.value) == null ? void 0 : s.wrapper) == null ? void 0 : d.default) != null && n.height && (e += `
          .${l} .tabs__item {
            line-height: ${(o = (c = (u = t.value) == null ? void 0 : u.wrapper) == null ? void 0 : c.default) == null ? void 0 : o.height};
          }`)), (b = t.value) != null && b.tabsContainer && (e += f((m = t.value) == null ? void 0 : m.tabsContainer, `.${l} .tabs`)), (p = t.value) != null && p.itemWrapper) {
        const { hover: F, ...L } = (S = t.value) == null ? void 0 : S.itemWrapper;
        e += f((_ = t.value) == null ? void 0 : _.itemWrapper, `.${l} .tabs__item:not(.tabs__item--selected):not(.tabs__item--disabled)`), e += f(L, `.${l} .tabs__item--selected`), (C = (g = ($ = t.value) == null ? void 0 : $.itemWrapper) == null ? void 0 : g.default) != null && C.height && (e += `
          .${l} .tabs__item {
            line-height: ${(w = (T = (W = t.value) == null ? void 0 : W.itemWrapper) == null ? void 0 : T.default) == null ? void 0 : w.height};
          }`);
      }
      return e;
    }, x = () => {
      var a, r, s, d, n, u, c, o, b;
      let e = "";
      return (a = t.value) != null && a.itemWrapper && ((s = (r = t.value) == null ? void 0 : r.itemWrapper) != null && s.active) && (e += `
          .${l} .tabs__item--selected {
            ${H((n = (d = t.value) == null ? void 0 : d.itemWrapper) == null ? void 0 : n.active)}
          }
        `), (c = (u = t.value) == null ? void 0 : u.itemWrapper) != null && c.disabled && (e += `
          .${l} .tabs__item--disabled {
            ${H((b = (o = t.value) == null ? void 0 : o.itemWrapper) == null ? void 0 : b.disabled)}
          }
        `), e;
    }, y = () => {
      const e = document.querySelector(`.${l} .style-container`);
      e && (e.innerHTML = `<style>${q()}${x()}</style>`);
    };
    return h(() => i.cStyle, () => {
      y();
    }), h(() => i.selectedTabVal, (e) => {
      v.selectedVal = e;
    }), h(() => [i.data, i.alias], () => {
      const e = M(V(O), i.alias);
      v.data = i.data.map((a) => {
        const r = a;
        return Object.keys(e).forEach((s) => {
          r[s] = a[e[s]];
        }), r;
      });
    }, {
      deep: !0,
      immediate: !0
    }), N(() => {
      y();
    }), {
      ...P(v),
      curStyle: t,
      domOnlyClass: l,
      tabClick: D,
      setHoverState: k,
      getLabelStyle: A
    };
  }
});
export {
  U as default
};
