// @ths/design/es/components/tabs-base/src/index.vue.js
import m from "./index.vue2.js";
import { openBlock as o, createElementBlock as l, normalizeClass as i, normalizeStyle as r, createElementVNode as n, Fragment as c, renderList as p, toDisplayString as u } from "vue";
import "./index.vue3.js";
import v from "../../../_virtual/_plugin-vue_export-helper.js";
const f = ["title", "onClick", "onMouseenter", "onMouseleave"];
function y(s, t, C, S, h, $) {
  return o(), l("div", {
    class: i(["t-tabs-base", s.domOnlyClass]),
    style: r({
      "--t-tabs-base-box-shadow-color": s.innerShadowColor,
      "--t-tabs-base-hover-box-shadow-color": s.activeInnerShadowColor,
      "--t-tabs-base-active-box-shadow-color": s.activeInnerShadowColor
    })
  }, [
    n("ul", {
      class: "tabs",
      style: r({ "--tab-space": s.space + "px" })
    }, [
      (o(!0), l(c, null, p(s.data, (e, d) => (o(), l("li", {
        key: e.value,
        class: i(["tabs__item", { "tabs__item--selected": !(e != null && e.disabled) && (e == null ? void 0 : e.value) === s.selectedVal, "tabs__item--disabled": e == null ? void 0 : e.disabled }]),
        title: e.label,
        onClick: (a) => s.tabClick(e, d),
        onMouseenter: (a) => s.setHoverState(e, !0),
        onMouseleave: (a) => s.setHoverState(e, !1)
      }, [
        n("span", {
          style: r(s.getLabelStyle(e))
        }, u(e.label), 5)
      ], 42, f))), 128))
    ], 4),
    t[0] || (t[0] = n("div", { class: "style-container" }, null, -1))
  ], 6);
}
const g = /* @__PURE__ */ v(m, [["render", y]]);
export {
  g as default
};
