// @ths/design/es/components/three-circle-chart/src/index.vue2.js
import { defineComponent as B, computed as m, ref as F, onMounted as x, watch as v } from "vue";
import o from "lodash/cloneDeep";
import C from "lodash/merge";
import { getDomRandomClass as A, setStyle as i } from "../../../utils/index.js";
import D from "./item.vue.js";
var p = /* @__PURE__ */ ((t) => (t.Top = "top", t.Left = "left", t.Right = "right", t))(p || {});
const h = {
  small: "155px",
  // 小圆
  medium: "180px",
  // 中圆
  large: "194px"
  // 大圆
}, y = {
  topNameColor: "#00BFFF",
  // 上圆名称颜色
  topUnitColor: "rgba(138, 210, 255, 0.8)",
  // 上圆单位颜色
  leftNameColor: "#F2994A",
  // 左圆名称颜色
  leftUnitColor: "#FFD0A7",
  // 左圆单位颜色
  rightNameColor: "#47BC78",
  // 右圆名称颜色
  rightUnitColor: "#47BC78"
  // 右圆单位颜色
}, O = ["large", "medium", "small"], M = B({
  name: "TThreeCircleChart",
  components: { TCircleItem: D },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外容器
        wrapper: {
          default: {}
        },
        // 小圆名称、值
        smallName: {
          default: {}
        },
        smallValue: {
          default: {}
        },
        // 中圆名称、值
        mediumName: {
          default: {}
        },
        mediumValue: {
          default: {}
        },
        // 大圆名称、值
        largeName: {
          default: {}
        },
        largeValue: {
          default: {}
        },
        // 单位
        unit: {
          default: {}
        },
        // 图片
        image: {
          default: {}
        }
      })
    },
    // 大中小尺寸数据
    sizeList: {
      type: Object,
      default: () => h
    },
    // 上圆 名称、值、背景图
    topCircleName: {
      type: String,
      default: ""
    },
    topCircleValue: {
      type: Number,
      default: 0
    },
    topCircleBg: {
      type: String,
      default: ""
    },
    // 左圆 名称、值、背景图
    leftCircleName: {
      type: String,
      default: ""
    },
    leftCircleValue: {
      type: Number,
      default: 0
    },
    leftCircleBg: {
      type: String,
      default: ""
    },
    // 右圆 名称、值、背景图
    rightCircleName: {
      type: String,
      default: ""
    },
    rightCircleValue: {
      type: Number,
      default: 0
    },
    rightCircleBg: {
      type: String,
      default: ""
    },
    // 单位
    unit: {
      type: String,
      default: "个"
    },
    // 确定图片如何适应容器框
    fit: {
      type: String,
      default: "cover"
    },
    // 不同圆中的文字、单位颜色
    colors: {
      type: Object,
      default: () => y
    }
  },
  emits: ["click"],
  setup(t, { emit: N }) {
    const l = m(() => t.cStyle), a = A("three-circle-chart"), f = F([]), r = m(() => C(o(h), t.sizeList)), V = m(() => C(o(y), t.colors)), $ = () => {
      let e = "";
      return l.value.wrapper && (e += i(l.value.wrapper, `.${a}`)), l.value.smallName && (e += i(l.value.smallName, `.${a} .circle-container .content.small .center-container .name`)), l.value.smallValue && (e += i(l.value.smallValue, `.${a} .circle-container .content.small .center-container .num`)), l.value.mediumName && (e += i(l.value.mediumName, `.${a} .circle-container .content.medium .center-container .name`)), l.value.mediumValue && (e += i(l.value.mediumValue, `.${a} .circle-container .content.medium .center-container .num`)), l.value.largeName && (e += i(l.value.largeName, `.${a} .circle-container .content.large .center-container .name`)), l.value.largeValue && (e += i(l.value.largeValue, `.${a} .circle-container .content.large .center-container .num`)), l.value.unit && (e += i(l.value.unit, `.${a} .circle-container .content .center-container .unit`)), l.value.image && (e += i(l.value.image, `.${a} .circle-container img`)), e;
    }, S = () => {
      let e = "";
      return r.value.small && (e += `.${a} .circle-container .content.small {
          width: ${r.value.small};
          height: ${r.value.small};
        }`), r.value.medium && (e += `.${a} .circle-container .content.medium {
          width: ${r.value.medium};
          height: ${r.value.medium};
        }`), r.value.large && (e += `.${a} .circle-container .content.large {
          width: ${r.value.large};
          height: ${r.value.large};
        }`), e;
    }, s = () => {
      const e = [
        { name: "top", value: Number(t.topCircleValue) || 0 },
        { name: "left", value: Number(t.leftCircleValue) || 0 },
        { name: "right", value: Number(t.rightCircleValue) || 0 }
      ], n = [Number(t.topCircleValue) || 0, Number(t.leftCircleValue) || 0, Number(t.rightCircleValue) || 0], c = [...new Set(o(n).sort((u, z) => z - u))], g = [];
      e.forEach((u) => {
        g.push({
          ...u,
          level: O[c.indexOf(u.value)]
          // 获取level的class名
        });
      }), f.value = g;
    }, b = (e) => e == null || e === "" ? "-" : Number(e), w = (e) => {
      var n;
      return (n = f.value.find((c) => c.name === e)) == null ? void 0 : n.level;
    }, d = () => {
      const e = document.querySelector(`.${a} > .style-container`);
      e && (e.innerHTML = `<style>${$()}${S()}</style>`);
    }, L = (e) => {
      N("click", e);
    };
    return x(() => {
      d(), s();
    }), v(
      () => [t.topCircleValue, t.rightCircleValue, t.leftCircleValue],
      () => {
        s();
      },
      {
        immediate: !0
      }
    ), v(
      () => [t.sizeList, t.cStyle],
      () => {
        d();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      curStyle: l,
      curColors: V,
      domOnlyClass: a,
      getLevelClass: w,
      handleNumberValue: b,
      LocationName: p,
      handleClick: L
    };
  }
});
export {
  M as default
};
