// @ths/design/es/components/three-circle-chart/src/item.vue.js
import r from "./item.vue2.js";
import { openBlock as a, createElementBlock as i, normalizeClass as m, createElementVNode as l, normalizeStyle as o, toDisplayString as t } from "vue";
import "./item.vue3.js";
import c from "../../../_virtual/_plugin-vue_export-helper.js";
const p = ["src"], d = { class: "center-container" }, u = ["title"], f = { class: "num" };
function C(e, s, g, h, k, y) {
  return a(), i("div", {
    class: m(["content", e.className]),
    onClick: s[0] || (s[0] = (...n) => e.handleClick && e.handleClick(...n))
  }, [
    l("img", {
      src: e.bgSrc,
      style: o({ "object-fit": e.imgFit }),
      alt: "",
      draggable: "false"
    }, null, 12, p),
    l("div", d, [
      l("div", null, [
        l("span", {
          class: "name",
          title: e.textName,
          style: o({ color: e.nameColor })
        }, t(e.textName || "-"), 13, u),
        l("p", null, [
          l("span", f, t(e.numberValue), 1),
          l("span", {
            class: "unit",
            style: o({ color: e.unitColor })
          }, t(e.unit || "个"), 5)
        ])
      ])
    ])
  ], 2);
}
const E = /* @__PURE__ */ c(r, [["render", C]]);
export {
  E as default
};
