// @ths/design/es/components/three-circle-chart/src/item.vue2.js
import { defineComponent as n } from "vue";
const u = n({
  name: "TCircleItem",
  props: {
    // class类名 大、中、小class名
    className: {
      type: String,
      default: ""
    },
    // 图片适应方式
    imgFit: {
      type: String,
      default: "cover"
    },
    // 图片路径地址
    bgSrc: {
      type: String,
      default: ""
    },
    // 数值
    numberValue: {
      type: Number,
      default: 0
    },
    // 文本名称
    textName: {
      type: String,
      default: ""
    },
    // 单位
    unit: {
      type: String,
      default: "个"
    },
    // 文本名称颜色
    nameColor: {
      type: String,
      default: ""
    },
    // 单位颜色
    unitColor: {
      type: String,
      default: ""
    }
  },
  emits: ["click"],
  setup(e, { emit: t }) {
    return {
      handleClick: () => {
        const a = {
          type: e.className,
          value: e.numberValue,
          textName: e.textName,
          unit: e.unit
        };
        t("click", a);
      }
    };
  }
});
export {
  u as default
};
