// @ths/design/es/components/chart-line-discolor/src/index.vue.js
import a from "./index.vue2.js";
import { openBlock as c, createElementBlock as m, normalizeClass as p, normalizeStyle as l, withDirectives as i, createElementVNode as e, vShow as n } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  ref: "chartId",
  class: "chart"
}, y = { class: "empty" }, f = ["src"];
function h(t, s, v, E, k, _) {
  var o, r;
  return c(), m("div", {
    class: p(["t-chart-line-discolor", t.domOnlyClass]),
    style: l({ ...(r = (o = t.curStyle) == null ? void 0 : o.wrapper) == null ? void 0 : r.default })
  }, [
    i(e("div", u, null, 512), [
      [n, t.checkEmpty]
    ]),
    i(e("div", y, [
      e("img", {
        class: "empty-img",
        style: l({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [n, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const S = /* @__PURE__ */ d(a, [["render", h]]);
export {
  S as default
};
