// @ths/design/es/components/chart-line-discolor/src/index.vue2.js
import { defineComponent as N, ref as Y, computed as B, reactive as H, onMounted as P, onUnmounted as U, watch as x, nextTick as F, toRefs as W } from "vue";
import * as G from "echarts";
import f from "lodash/merge";
import h from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as K, DEFAULT_OPTIONS_AXIS as R } from "../../../utils/constant.js";
import { getDomRandomClass as X, setStyle as p } from "../../../utils/index.js";
const J = f(h(K), h(R)), v = f(
  h(J),
  {
    grid: {
      top: "13%",
      bottom: "10%",
      left: "6%",
      right: "4%"
    },
    xAxis: {
      boundaryGap: !0
    },
    tooltip: {
      formatter: (e) => {
        let g = "";
        const { name: l } = e[0];
        return e.forEach((o) => {
          o.componentSubType === "line" && (g += `<div style="margin: 10px 0 0;line-height: 1">${o.marker}<span>${o.seriesName}</span>
          <span style="float: right;margin-left: 20px;">${o.value}</span><br/></div>`);
        }), `${l}<br/>${g}`;
      },
      textStyle: {
        fontWeight: "normal"
      }
    },
    legend: {
      show: !1,
      itemStyle: {
        borderWidth: 0
      }
    }
  }
), ae = N({
  name: "TChartLineDiscolor",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 暂无数据容器
        emptyContainer: {
          default: {}
        },
        // 暂无数据图片
        emptyImg: {
          default: {}
        },
        // 暂无数据文字
        emptyText: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => v,
      required: !1
    },
    // 传入数组数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    gridcolorStops: {
      type: Array,
      default: () => [],
      required: !1
    },
    dataxAlias: {
      type: String,
      default: "dataX"
    },
    datayAlias: {
      type: String,
      default: "dataY"
    },
    name: {
      type: String,
      default: "水质等级"
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // X 轴类目取resData数据中的别名设置
        xAxisLabel: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 折线主题配置
    line: {
      type: Object,
      default: () => ({})
    },
    // 是否显示背景
    showItemBg: {
      type: Boolean,
      default: !1
    },
    // 背景条样式
    backgroundStyle: {
      type: Object,
      default: () => ({})
    }
  },
  // 方法名定义
  emits: ["com-click", "legend-select"],
  setup(e, { emit: g }) {
    const l = X("chart-line-discolor"), o = Y(null);
    let d = null;
    const r = B(() => e.cStyle), a = H({
      labelY: "",
      // 图表数据
      chartData: [],
      chartBgData: [],
      // 主题配置
      lineOptions: h(e.line)
    }), D = () => {
      var t, u, s;
      (u = (t = e.seriesSetting) == null ? void 0 : t.data) != null && u.length && e.data ? (a.labelY = e.seriesSetting.data.map((i) => i.dataKey), a.chartData = e.seriesSetting.data.map((i) => {
        const { name: S, dataKey: n } = i;
        return {
          name: S || "--",
          data: e.data.map((m) => m[n])
        };
      }), a.chartBgData = a.chartData) : (a.chartData = [], a.chartBgData = ((s = e.options) == null ? void 0 : s.series) || a.chartData);
    }, M = () => {
      let t = "";
      return r.value.wrapper && (t += p(r.value.wrapper, `.${l}`)), r.value.emptyContainer && (t += p(r.value.emptyContainer, `.${l} .empty`)), r.value.emptyImg && (t += p(r.value.emptyImg, `.${l} .empty .empty-img`)), r.value.emptyText && (t += p(r.value.emptyText, `.${l} .empty .empty-text`)), t;
    }, O = () => {
      const t = document.querySelector(`.${l} .style-container`);
      t && (t.innerHTML = `<style>${M()}</style>`);
    }, b = () => {
      var m, C, w, L, T, $, k;
      const { color: t, series: u } = e.options;
      let s = [], i = [];
      const S = e.data.map((c) => c[a.labelY]);
      e.data && ((C = (m = e.seriesSetting) == null ? void 0 : m.data) != null && C.length) && (s = (w = e.seriesSetting) == null ? void 0 : w.data.map((c, y) => {
        var I, E, j, z;
        const q = f(
          {
            name: (I = a.chartData[y]) == null ? void 0 : I.name,
            type: "line",
            itemStyle: {
              color: t && t[y] || v.color[y] || "#FFF"
            },
            // lineStyle: {
            //   color: (color && color[index]) || defaultOptions.color[index] || '#FFF',
            // },
            showSymbol: !1,
            emphasis: {
              focus: "series"
            },
            smooth: !0,
            data: S,
            ...a == null ? void 0 : a.lineOptions
          },
          (z = (j = (E = e.seriesSetting) == null ? void 0 : E.data) == null ? void 0 : j[y]) == null ? void 0 : z.style
        );
        return f({}, q, c);
      }), (L = e.seriesSetting) != null && L.xAxisLabel ? i = (T = e.data) != null && T.length ? e.data.map((c) => c[e.seriesSetting.xAxisLabel]) : [] : i = e.data.map((c, y) => y));
      let n = f(
        v,
        {
          series: s,
          xAxis: {
            data: i,
            axisLabel: {
              color: "rgba(164, 232, 232, 1)"
            }
          },
          yAxis: {
            axisLabel: {
              color: "rgba(164, 232, 232, 1)"
            }
          },
          backgroundColor: "rgba(0,0,0,1)",
          grid: {
            left: "10",
            right: "10",
            bottom: "6%",
            top: "15%",
            containLabel: !0,
            show: !0,
            borderColor: "transparent",
            backgroundColor: {
              type: "linear",
              x: 0,
              y: 0,
              x2: 0,
              y2: 1,
              colorStops: e.gridcolorStops,
              globalCoord: !1
              // 缺省为 false
            }
          },
          visualMap: {
            show: !0,
            top: 0,
            right: "10%",
            orient: "horizontal",
            inverse: !1,
            itemGap: 20,
            itemHeight: 3,
            itemWidth: 15,
            textStyle: {
              color: "#fff",
              fontSize: 14
            },
            pieces: (k = ($ = e.options) == null ? void 0 : $.visualMap) == null ? void 0 : k.pieces
          }
        }
      );
      n = f(n, e.options), d.clear();
      try {
        d.setOption(n);
      } catch (c) {
        console.log(c);
      }
    }, _ = B(() => {
      var t, u, s, i;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((n) => {
            var m;
            return (n == null ? void 0 : n.data) && ((m = n == null ? void 0 : n.data) == null ? void 0 : m.length) !== 0;
          });
        if (typeof ((u = e.options) == null ? void 0 : u.series) == "object")
          return ((i = (s = e.options.series) == null ? void 0 : s.data) == null ? void 0 : i.length) === 0;
      }
      return !!e.data.length;
    }), A = () => {
      F(() => {
        d.resize();
      });
    };
    return P(() => {
      O(), o.value && (d = G.init(o.value), D(), b(), d.on("click", (t) => {
        g("com-click", t);
      }), d.on("legendselectchanged", (t) => {
        g("legend-select", t);
      }), window.addEventListener("resize", A));
    }), U(() => {
      window.removeEventListener("resize", A);
    }), x(
      [() => e.line, () => e.backgroundStyle],
      () => {
        a.lineOptions = h(e.line), b();
      },
      {
        deep: !0
      }
    ), x(
      [
        () => e.options,
        () => e.data,
        () => e.seriesSetting,
        () => e.showItemBg
      ],
      () => {
        D(), b();
      },
      {
        deep: !0
      }
    ), x(
      () => e.cStyle,
      () => {
        O(), F(() => {
          d.resize();
        });
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      curStyle: r,
      ...W(a),
      chartId: o,
      checkEmpty: _,
      domOnlyClass: l
    };
  }
});
export {
  ae as default
};
