// @ths/design/es/components/self-adaption-card/src/index.vue.js
import b from "./index.vue2.js";
import { openBlock as s, createElementBlock as a, normalizeClass as $, Fragment as C, renderList as F, normalizeStyle as n, createElementVNode as t, toDisplayString as L, withModifiers as w } from "vue";
import "./index.vue3.js";
import B from "../../../_virtual/_plugin-vue_export-helper.js";
const M = ["onClick"], z = ["src"], D = ["src"], E = ["src"], N = ["title"], O = ["onClick"];
function A(e, r, T, V, W, h) {
  return s(), a("div", {
    class: $(["t-self-adaption-card", e.domOnlyClass])
  }, [
    (s(!0), a(C, null, F(e.curData, (o, l) => {
      var d, u, m, g, c, f, I, v, p, y, k, S;
      return s(), a("div", {
        key: l,
        ref_for: !0,
        ref: (i) => e.refNodes[l] = i,
        class: $(["trigger-title menu-title", { "title-active": e.currentIndex === l, vertical: e.triggerTextWriteMode === "vertical" }]),
        style: n({
          opacity: e.imgOnloaded ? 1 : 0
        }),
        onClick: (i) => e.clickItem(o, l)
      }, [
        t("img", {
          class: "title-front",
          src: e.currentIndex === l ? (d = e.titleFrontImg) == null ? void 0 : d.active : (u = e.titleFrontImg) == null ? void 0 : u.default,
          alt: "",
          onLoad: r[0] || (r[0] = (i) => e.imgLoaded())
        }, null, 40, z),
        t("div", {
          class: "title-fill",
          style: n({
            backgroundImage: e.currentIndex === l ? (m = e.curStyle.titleFillImg) == null ? void 0 : m.active : (g = e.curStyle.titleFillImg) == null ? void 0 : g.default
          })
        }, null, 4),
        t("img", {
          class: "title-mid",
          src: e.currentIndex === l ? (c = e.titleMidImg) == null ? void 0 : c.active : (f = e.titleMidImg) == null ? void 0 : f.default,
          alt: "",
          onLoad: r[1] || (r[1] = (i) => e.imgLoaded())
        }, null, 40, D),
        t("div", {
          class: "title-fill",
          style: n({
            backgroundImage: e.currentIndex === l ? (I = e.curStyle.titleFillImg) == null ? void 0 : I.active : (v = e.curStyle.titleFillImg) == null ? void 0 : v.default
          })
        }, null, 4),
        t("img", {
          class: "title-below",
          src: e.currentIndex === l ? (p = e.titleBelowImg) == null ? void 0 : p.active : (y = e.titleBelowImg) == null ? void 0 : y.default,
          alt: "",
          onLoad: r[2] || (r[2] = (i) => e.imgLoaded())
        }, null, 40, E),
        t("span", {
          class: "label",
          title: o.label
        }, [
          t("span", null, L(o.label), 1),
          t("div", {
            class: "title-delete",
            style: n({
              backgroundImage: e.currentIndex === l ? (k = e.curStyle.deleteImg) == null ? void 0 : k.active : (S = e.curStyle.deleteImg) == null ? void 0 : S.default
            }),
            onClick: w((i) => e.deleteItem(o, l), ["stop"])
          }, null, 12, O)
        ], 8, N)
      ], 14, M);
    }), 128)),
    r[3] || (r[3] = t("div", { class: "style-container" }, null, -1))
  ], 2);
}
const J = /* @__PURE__ */ B(b, [["render", A]]);
export {
  J as default
};
