// @ths/design/es/components/self-adaption-card/src/index.vue2.js
import { defineComponent as G, computed as q, ref as P, reactive as R, onMounted as U, watch as s, toRefs as W, nextTick as N } from "vue";
import z from "lodash/merge";
import C from "lodash/cloneDeep";
import { getDomRandomClass as J, getCssPropertyString as m, setStyle as g } from "../../../utils/index.js";
const H = {
  label: "label",
  value: "value",
  children: "children"
}, B = 0, K = 1, Z = G({
  name: "TSelfAdaptionCard",
  props: {
    // 触发按钮文字排位方式
    triggerTextWriteMode: {
      type: String,
      default: "vertical"
    },
    // 标题按钮前部分图片
    titleFrontImg: {
      type: Object,
      default: () => ({})
    },
    // 标题按钮后部分图片
    titleBelowImg: {
      type: Object,
      default: () => ({})
    },
    // 标题按钮中部分图片
    titleMidImg: {
      type: Object,
      default: () => ({})
    },
    // 删除按钮图片
    deleteImg: {
      type: Object,
      default: () => ({})
    },
    // 已选中的菜单项
    activeItem: {
      type: String,
      default: ""
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => C(H),
      require: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 标题
        title: {
          default: {},
          hover: {},
          active: {}
        },
        // 标题前半部分图片
        titleFrontImg: {
          default: {},
          active: {}
        },
        // 标题后半部分图片
        titleBelowImg: {
          default: {},
          active: {}
        },
        // 标题中间部分图片
        titleMidImg: {
          default: {},
          active: {}
        },
        // 标题填充图片
        titleFillImg: {
          default: {},
          active: {}
        },
        // 标题关闭按钮
        deleteImg: {
          default: {},
          active: {}
        }
      })
    }
  },
  /**
   * item-click 点击单项卡片事件
   * delete-item 点击单项删除按钮事件
   */
  emits: ["item-click", "delete-item"],
  setup(d, { emit: I }) {
    const t = q(() => d.cStyle), i = J("self-adaption-card"), y = P([]), l = R({
      showMenuAnimation: !1,
      // 控制菜单的展开收起动画
      showMenu: !1,
      // 控制菜单的展开收起
      curData: [],
      // 菜单列表
      currentItem: "",
      // 选中的菜单项
      menuWidth: "",
      // 菜单项动态宽度
      menuHeight: "",
      // 菜单项动态高度
      imgOnloaded: !1,
      // 菜单按钮图片加载完毕
      currentIndex: null
      // 当前选中的图片的索引
    }), L = (e, r) => {
      l.currentItem = e.value, l.currentIndex = r, I("item-click", e);
    }, j = (e) => {
      try {
        const r = z(H, d.alias);
        if (!Array.isArray(e))
          return;
        e.forEach((a, c) => {
          a.label = a[r.label] ?? "", a.value = a[r.value] ?? "";
        });
      } catch (r) {
        console.warn(r);
      }
    }, v = () => {
      N(() => {
        y.value.filter((a) => !!a).forEach((a) => {
          let c = 0, u = 0, o;
          Array.from(a.children).forEach((n) => {
            n.className === "title-front" && n.clientHeight && (c = n.clientHeight), n.className === "title-below" && n.clientHeight && (u = n.clientHeight), n.nodeName === "SPAN" && (o = n, o.style.paddingTop = `${c}px`, o.style.paddingBottom = `${u}px`, a.style.height = `${o.clientHeight}px`);
          });
        });
      });
    }, $ = (e) => {
      l.currentItem = e;
      const r = l.curData.findIndex((a) => a.value === l.currentItem);
      r >= 0 ? l.currentIndex = r : l.currentIndex = -1;
    }, _ = (e, r) => {
      var c;
      let a;
      if (l.curData.splice(r, 1), e.value !== l.currentItem) {
        const u = l.curData.findIndex((o) => o.value === l.currentItem);
        u >= 0 && (l.currentIndex = u, a = l.curData[u]);
      } else
        l.currentIndex = l.curData.length - 1, l.currentItem = ((c = l.curData[l.curData.length - 1]) == null ? void 0 : c.value) ?? "", a = l.curData[l.curData.length - 1] ?? "";
      N(() => {
        v();
      }), I("delete-item", a, l.curData);
    };
    let f = B;
    const T = () => {
      l.imgOnloaded = !1, f++, f === K && (f = B, l.imgOnloaded = !0, v());
    }, k = () => {
      var r, a, c, u, o, n, p, D, b, w, x, S, M, O, A, F;
      let e = "";
      return (r = t.value.title) != null && r.hover && (e += `
          .${i} .trigger-title:hover {
            ${m((a = t.value.title) == null ? void 0 : a.hover)}
          }
        `), (c = t.value.title) != null && c.active && (e += `
          .${i} .trigger-title.title-active {
            ${m((u = t.value.title) == null ? void 0 : u.active)}
          }
        `), (o = t.value.titleFrontImg) != null && o.active && (e += `
          .${i} .trigger-title.title-active .title-front {
            ${m((n = t.value.titleFrontImg) == null ? void 0 : n.active)}
          }
        `), (p = t.value.titleBelowImg) != null && p.active && (e += `
          .${i} .trigger-title.title-active .title-below {
            ${m((D = t.value.titleBelowImg) == null ? void 0 : D.active)}
          }
        `), (b = t.value.titleMidImg) != null && b.active && (e += `
          .${i} .trigger-title.title-active .title-mid {
            ${m((w = t.value.titleMidImg) == null ? void 0 : w.active)}
          }
        `), (x = t.value.titleFillImg) != null && x.active && (e += `
          .${i} .trigger-title.title-active .title-fill {
            ${m((S = t.value.titleFillImg) == null ? void 0 : S.active)}
          }
        `), (M = t.value.deleteImg) != null && M.hover && (e += `
          .${i} .trigger-title .title-delete:hover {
            ${m((O = t.value.deleteImg) == null ? void 0 : O.hover)}
          }
        `), (A = t.value.deleteImg) != null && A.active && (e += `
          .${i} .trigger-title.title-active .title-delete {
            ${m((F = t.value.deleteImg) == null ? void 0 : F.active)}
          }
        `), e;
    }, E = () => {
      let e = "";
      return t.value.wrapper && (e += g(t.value.wrapper, `.${i}`)), t.value.title && (e += g(t.value.title, `.${i} .trigger-title`)), t.value.titleFrontImg && (e += g(t.value.titleFrontImg, `.${i} .trigger-title .title-front`)), t.value.titleBelowImg && (e += g(t.value.titleBelowImg, `.${i} .trigger-title .title-below`)), t.value.titleMidImg && (e += g(t.value.titleMidImg, `.${i} .trigger-title .title-mid`)), t.value.titleFillImg && (e += g(t.value.titleFillImg, `.${i} .trigger-title .title-fill`)), t.value.textContainer && (e += g(t.value.textContainer, `.${i} .trigger-title .label`)), t.value.deleteImg && (e += g(t.value.deleteImg, `.${i} .trigger-title .title-delete`)), e;
    }, h = () => {
      const e = document.querySelector(`.${i} .style-container`);
      e && (e.innerHTML = `<style>${E()}${k()}</style>`);
    };
    return U(() => {
      h();
    }), s([() => d.data, () => d.alias], () => {
      l.curData = C(d.data), j(l.curData), $(d.activeItem), v();
    }, {
      deep: !0,
      immediate: !0
    }), s(() => d.activeItem, (e) => {
      $(e), v();
    }, {
      immediate: !0
    }), s(() => d.cStyle, () => {
      h(), v();
    }, { deep: !0 }), {
      ...W(l),
      curStyle: t,
      domOnlyClass: i,
      clickItem: L,
      imgLoaded: T,
      refNodes: y,
      deleteItem: _
    };
  }
});
export {
  Z as default
};
