// @ths/design/es/components/button/src/index.vue2.js
import { defineComponent as f, computed as y, watch as b, onMounted as $ } from "vue";
import { filterAttrs as w, getDomRandomClass as S, getCssPropertyString as a, setStyle as c } from "../../../utils/index.js";
import "../../text/index.js";
import x from "../../text/src/index.vue.js";
const H = f({
  name: "TButton",
  components: {
    TText: x
  },
  props: {
    // 文字
    label: {
      type: String,
      default: ""
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {},
          hover: {},
          active: {},
          disabled: {}
        },
        text: {
          default: {}
        }
      })
    }
  },
  setup(r) {
    const e = y(() => r.cStyle), l = S("button"), m = () => {
      var o, u, i, p, s, d;
      let t = "";
      return e.value.wrapper && e.value.wrapper.active && (t += `
          .${l} .el-button:active {
            ${a(e.value.wrapper.active)}
          }
        `), e.value.wrapper && e.value.wrapper.disabled && (t += `
          .${l} .el-button.is-disabled{
            ${a(e.value.wrapper.disabled)}
          }
        `, t += `
          .${l} .el-button.is-disabled:hover{
            ${a(e.value.wrapper.disabled)}
          }
        `), e.value.text && (t += `
          .${l} .el-button span{
            display: ${(u = (o = e.value.text) == null ? void 0 : o.default) == null ? void 0 : u.display};
            width: ${(p = (i = e.value.text) == null ? void 0 : i.default) == null ? void 0 : p.width};
            height: ${(d = (s = e.value.text) == null ? void 0 : s.default) == null ? void 0 : d.height};
          }
        `), t;
    }, v = () => {
      let t = "";
      return e.value.wrapper && (t += c(e.value.wrapper, `.${l} .el-button`)), e.value.text && (t += c(e.value.text, `.${l} .el-button .t-text`)), t;
    }, n = () => {
      const t = document.querySelector(`.${l} .style-container`);
      t && (t.innerHTML = `<style>${v()}${m()}</style>`);
    };
    return b(() => r.cStyle, () => {
      n();
    }), $(() => {
      n();
    }), {
      curStyle: e,
      domOnlyClass: l,
      filterAttrs: w
    };
  }
});
export {
  H as default
};
