// @ths/design/es/components/button/src/index.vue.js
import S from "./index.vue2.js";
import { resolveComponent as f, openBlock as b, createElementBlock as w, normalizeClass as $, normalizeStyle as B, createVNode as d, mergeProps as y, withCtx as C, createElementVNode as v } from "vue";
import "./index.vue3.js";
import z from "../../../_virtual/_plugin-vue_export-helper.js";
function h(t, e, k, A, E, L) {
  var r, l, o, a, n, p, s, m;
  const c = f("t-text"), g = f("el-button");
  return b(), w("div", {
    class: $(["t-button", t.domOnlyClass]),
    style: B({
      marginTop: (l = (r = t.curStyle.wrapper) == null ? void 0 : r.default) == null ? void 0 : l.marginTop,
      marginLeft: (a = (o = t.curStyle.wrapper) == null ? void 0 : o.default) == null ? void 0 : a.marginLeft,
      marginRight: (p = (n = t.curStyle.wrapper) == null ? void 0 : n.default) == null ? void 0 : p.marginRight,
      marginBottom: (m = (s = t.curStyle.wrapper) == null ? void 0 : s.default) == null ? void 0 : m.marginBottom
    })
  }, [
    d(g, y(t.filterAttrs(t.$attrs, ["size", "type"]), { type: "primary" }), {
      default: C(() => {
        var i, u;
        return [
          d(c, y(t.filterAttrs(t.$attrs), {
            label: t.label,
            "c-style": {
              wrapper: (i = t.curStyle) != null && i.text ? { ...(u = t.curStyle) == null ? void 0 : u.text } : { default: {} }
            }
          }), null, 16, ["label", "c-style"])
        ];
      }),
      _: 1
    }, 16),
    e[0] || (e[0] = v("div", { class: "style-container" }, null, -1))
  ], 6);
}
const O = /* @__PURE__ */ z(S, [["render", h]]);
export {
  O as default
};
