// @ths/design/es/components/list/src/index.vue2.js
import { defineComponent as h, computed as y, reactive as S, watch as r, onMounted as C, toRefs as W } from "vue";
import A from "lodash/merge";
import m from "lodash/cloneDeep";
import { getDomRandomClass as g, setStyle as v, getCssPropertyString as a } from "../../../utils/index.js";
const f = {
  active: "active",
  icon: "icon",
  iconActive: "iconActive",
  state: "state",
  stateActive: "stateActive",
  info: "info",
  desc: "desc"
}, j = h({
  name: "TList",
  props: {
    // 列表数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => m(f),
      require: !1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        itemWrapper: {
          default: {},
          active: {},
          hover: {}
        },
        info: {
          default: {},
          active: {},
          hover: {}
        },
        unit: {
          default: {},
          active: {},
          hover: {}
        },
        icon: {
          default: {},
          active: {}
        },
        state: {
          default: {},
          active: {},
          hover: {}
        },
        desc: {
          default: {},
          hover: {}
        }
      })
    }
  },
  emits: ["item-click"],
  setup(l, { emit: d }) {
    const e = y(() => l.cStyle), c = S({
      curData: l.data
    }), i = g("list");
    r(() => [l.data, l.alias], () => {
      const t = A(m(f), l.alias);
      c.curData = l.data.map((s) => {
        const n = s;
        return Object.keys(t).forEach((u) => {
          n[u] = s[t[u]];
        }), n;
      });
    }, {
      deep: !0,
      immediate: !0
    });
    const $ = () => {
      let t = "";
      return e.value.wrapper && (t += v(e.value.wrapper, `.${i}`)), e.value.itemWrapper && (t += v(e.value.itemWrapper, `.${i} .list-item`)), e.value.info && (t += v(e.value.info, `.${i} .list-item__info .info`)), e.value.unit && (t += v(e.value.unit, `.${i} .list-item__info .unit`)), e.value.icon && (t += v(e.value.icon, `.${i} .list-item__icon`)), e.value.state && (t += v(e.value.state, `.${i} .list-item__state`)), e.value.desc && (t += v(e.value.desc, `.${i} .list-item__desc`)), t;
    }, _ = () => {
      let t = "";
      return e.value.itemWrapper && (e.value.itemWrapper.active && (t += `
            .${i} .list-item.active {
              ${a(e.value.itemWrapper.active)}
            }
            .${i} .list-item.active .list-item__info .info {
              ${e.value.info && e.value.info.active && a(e.value.info.active)}
            }
            .${i} .list-item.active .list-item__info .unit {
              ${e.value.unit && e.value.unit.active && a(e.value.unit.active)}
            }
            .${i} .list-item.active .list-item__desc {
              ${e.value.desc && e.value.desc.active && a(e.value.desc.active)}
            }
          `), e.value.itemWrapper.hover && (t += `
            .${i} .list-item:hover {
              ${a(e.value.itemWrapper.hover)}
            }
            .${i} .list-item:hover .list-item__info .info {
              ${e.value.info && e.value.info.hover && a(e.value.info.hover)}
            }
            .${i} .list-item:hover .list-item__state {
              ${e.value.state && e.value.state.hover && a(e.value.state.hover)}
            }

            .${i} .list-item:hover .list-item__info .unit {
              ${e.value.unit && e.value.unit.hover && a(e.value.unit.hover)}
            }
            .${i} .list-item:hover .list-item__desc {
              ${e.value.desc && e.value.desc.hover && a(e.value.desc.hover)}
            }
          `)), e.value.info && e.value.info.active && (t += `
          .${i} .list-item__info.active {
            ${a(e.value.info.active)}
          }
        `), e.value.icon && e.value.icon.active && (t += `
          .${i} .list-item__icon.active {
            ${a(e.value.icon.active)}
          }
        `), e.value.state && e.value.state.active && (t += `
          .${i} .list-item__state.active {
            ${a(e.value.state.active)}
          }
        `), t;
    }, o = () => {
      const t = document.querySelector(`.${i} .style-container`);
      t && (t.innerHTML = `<style>${$()}${_()}</style>`);
    };
    r(() => l.cStyle, () => {
      o();
    }), C(() => {
      o();
    });
    const p = (t) => {
      d("item-click", t);
    };
    return {
      ...W(c),
      domOnlyClass: i,
      curStyle: e,
      handleListItemClick: p
    };
  }
});
export {
  j as default
};
