// @ths/design/es/components/list/src/index.vue.js
import r from "./index.vue2.js";
import { openBlock as t, createElementBlock as e, normalizeClass as o, Fragment as _, renderList as d, createElementVNode as i, createCommentVNode as l, toDisplayString as n } from "vue";
import "./index.vue3.js";
import u from "../../../_virtual/_plugin-vue_export-helper.js";
const p = ["onClick"], f = { class: "list-item__left" }, v = ["src"], k = ["title"], m = { class: "info" }, h = { class: "unit" }, y = {
  key: 0,
  class: "list-item__desc"
};
function C(c, a, g, $, A, L) {
  return t(), e("ul", {
    class: o(["t-list", c.domOnlyClass])
  }, [
    (t(!0), e(_, null, d(c.curData, (s) => (t(), e("li", {
      key: s,
      class: o(["list-item", s.active ? "active" : ""]),
      onClick: (B) => c.handleListItemClick(s)
    }, [
      i("div", f, [
        s.icon ? (t(), e("img", {
          key: 0,
          src: s.iconActive || s.icon,
          class: o(["list-item__icon", s.iconActive ? "active" : ""]),
          draggable: "false"
        }, null, 10, v)) : l("", !0),
        s.state ? (t(), e("span", {
          key: 1,
          class: o(["list-item__state", s.stateActive ? "active" : ""])
        }, n(s.state), 3)) : l("", !0),
        i("div", {
          class: "list-item__info",
          title: s.info
        }, [
          i("span", m, n(s.info), 1),
          i("span", h, n(s.unit), 1)
        ], 8, k)
      ]),
      s.desc ? (t(), e("div", y, n(s.desc), 1)) : l("", !0)
    ], 10, p))), 128)),
    a[0] || (a[0] = i("div", { class: "style-container" }, null, -1))
  ], 2);
}
const b = /* @__PURE__ */ u(r, [["render", C]]);
export {
  b as default
};
