// @ths/design/es/components/chart-type-treemap/src/index.vue2.js
import { defineComponent as O, computed as C, ref as W, reactive as $, watch as f, nextTick as T, onMounted as x, onUnmounted as E, toRefs as j } from "vue";
import * as z from "echarts";
import g from "lodash/merge";
import b from "lodash/cloneDeep";
import { DEFAULT_OPTIONS as k } from "../../../utils/constant.js";
import { getDomRandomClass as N, setStyle as y } from "../../../utils/index.js";
const q = {
  id: "id",
  name: "name",
  parentId: "parentId",
  value: "value"
}, F = O({
  name: "TChartTypeTreemap",
  props: {
    // 动画更新时长
    animationUpdateTime: {
      type: Number,
      default: 800
    },
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    props: {
      type: Object,
      default: () => q,
      required: !1
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => {
      },
      required: !1
    },
    // 是否展示文字内容数值
    showNum: {
      type: Boolean,
      default: !1
    },
    // 是否展示文字内容单位
    showUnit: {
      type: Boolean,
      default: !1
    },
    // 文字单位
    unit: {
      type: String,
      default: ""
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 矩形树图主题配置
    treemap: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["com-click"],
  setup(e, { emit: D }) {
    const I = g(b(k), {
      animation: !1,
      xAxis: {
        show: !1
      },
      yAxis: {
        show: !1
      },
      legend: {
        show: !1
      },
      series: {
        name: "全部",
        type: "treemap",
        roam: !1,
        width: "100%",
        height: "100%",
        leafDepth: 1,
        // 呈现层级，若为1加载时仅展开一层，接下来的每一层通过单击进入，如上面的效果图所示，
        // 每一层级呈现的样式
        label: {
          align: "center",
          verticalAlign: "top",
          padding: [-5, 0, 0, 0],
          ellipsis: !0,
          fontSize: 16,
          formatter: (t) => [
            `{label|${t.data.name}}`,
            `{num|${e.showNum ? t.data.value : ""}}{unit|${e.showUnit ? e.unit : ""}}`
          ].join(`

`),
          rich: {}
        },
        levels: [
          {
            itemStyle: {
              normal: {
                borderColor: "transparent",
                borderWidth: 4,
                gapWidth: 4
              }
            }
          },
          {
            colorSaturation: [0.3, 0.6],
            itemStyle: {
              normal: {
                borderColorSaturation: 0.7,
                gapWidth: 2,
                borderWidth: 2
              }
            }
          },
          {
            colorSaturation: [0.3, 0.5],
            itemStyle: {
              normal: {
                borderColorSaturation: 0.6,
                gapWidth: 1
              }
            }
          },
          {
            colorSaturation: [0.3, 0.5]
          }
        ]
      }
    }), s = N("chart-type-treemap"), o = C(() => e.cStyle), l = W(null);
    let d = null;
    const h = $({
      // 图表数据
      chartData: []
    }), U = (t) => {
      const n = [], { id: r, parentId: u } = e.props;
      if (!Array.isArray(t))
        return n;
      const m = {};
      return t.forEach((a) => {
        a[r] && (m[a[r]] = a);
      }), t.forEach((a) => {
        const i = m[a[u]];
        i ? (i.children || (i.children = [])).push(a) : n.push(a);
      }), n;
    }, v = () => {
      var t;
      if (e.data && ((t = e.data) != null && t.length)) {
        const {
          id: n,
          name: r,
          value: u,
          parentId: m
        } = e.props, a = e.data.map((i) => ({
          id: i[n],
          name: i[r],
          value: i[u],
          parentId: i[m]
        }));
        h.chartData = U(a);
      }
    }, A = () => {
      let t = "";
      return o.value.wrapper && (t += y(o.value.wrapper, `.${s}`)), o.value.emptyContainer && (t += y(o.value.emptyContainer, `.${s} .empty`)), o.value.emptyImg && (t += y(o.value.emptyImg, `.${s} .empty .empty-img`)), o.value.emptyText && (t += y(o.value.emptyText, `.${s} .empty .empty-text`)), t;
    }, w = () => {
      const t = document.querySelector(`.${s} .style-container`);
      t && (t.innerHTML = `<style>${A()}</style>`);
    }, S = C(() => {
      var t, n, r, u;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((a) => {
            var i;
            return (a == null ? void 0 : a.data) && ((i = a == null ? void 0 : a.data) == null ? void 0 : i.length) !== 0;
          });
        if (typeof ((n = e.options) == null ? void 0 : n.series) == "object" && "data" in e.options.series)
          return ((u = (r = e.options.series) == null ? void 0 : r.data) == null ? void 0 : u.length) === 0;
      }
      return !!e.data.length;
    }), p = () => {
      var r;
      if (!d)
        return;
      let t = g(e == null ? void 0 : e.treemap, (r = e.options) == null ? void 0 : r.series);
      t = {
        ...b(t),
        data: h.chartData || [],
        animationDurationUpdate: e.animationUpdateTime
      };
      const n = g(
        {
          series: t
        },
        I,
        e.options,
        {
          animation: !1
        }
      );
      d.setOption(n);
    }, c = () => {
      l.value && (l.value.getBoundingClientRect().width ?? 0) && T(() => {
        d.resize();
      });
    };
    return f(S, (t) => {
      t && setTimeout(() => {
        l.value && (l.value.getBoundingClientRect().width ?? 0) && T(() => {
          d.resize();
        });
      }, e.animationUpdateTime + 100);
    }), x(() => {
      w(), l.value && (d = z.init(l.value), v(), p(), d.on("click", (t) => {
        D("com-click", t);
      })), window.addEventListener("resize", c);
    }), E(() => {
      window.removeEventListener("resize", c);
    }), f(
      [
        () => e.options,
        () => e.treemap,
        () => e.unit,
        () => e.showUnit,
        () => e.showNum,
        () => e.animationUpdateTime
      ],
      (t) => {
        p(), c();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), f(
      () => [e.data, e.props],
      () => {
        v(), p(), c();
      },
      {
        deep: !0
      }
    ), f(
      () => e.cStyle,
      () => {
        w(), c();
      },
      {
        deep: !0
      }
    ), {
      curStyle: o,
      ...j(h),
      chartId: l,
      checkEmpty: S,
      domOnlyClass: s
    };
  }
});
export {
  F as default
};
