// @ths/design/es/components/chart-type-treemap/src/index.vue.js
import m from "./index.vue2.js";
import { openBlock as p, createElementBlock as n, normalizeClass as c, normalizeStyle as l, withDirectives as a, createElementVNode as t, vShow as i } from "vue";
import "./index.vue3.js";
import d from "../../../_virtual/_plugin-vue_export-helper.js";
const y = {
  ref: "chartId",
  class: "chart"
}, u = { class: "empty" }, f = ["src"];
function h(e, s, v, E, k, _) {
  var r, o;
  return p(), n("div", {
    class: c(["t-chart-type-treemap", e.domOnlyClass]),
    style: l({ ...(o = (r = e.curStyle) == null ? void 0 : r.wrapper) == null ? void 0 : o.default })
  }, [
    a(t("div", y, null, 512), [
      [i, e.checkEmpty]
    ]),
    a(t("div", u, [
      t("img", {
        class: "empty-img",
        style: l({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, f),
      s[0] || (s[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [i, !e.checkEmpty]
    ]),
    s[1] || (s[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const S = /* @__PURE__ */ d(m, [["render", h]]);
export {
  S as default
};
