// @ths/design/es/components/tree/src/index.vue.js
import _ from "./index.vue2.js";
import { resolveComponent as n, openBlock as a, createElementBlock as r, normalizeClass as k, createElementVNode as o, toDisplayString as p, normalizeStyle as C, createVNode as s, withModifiers as b, withCtx as i, withDirectives as w, mergeProps as c, vShow as g } from "vue";
import "./index.vue3.js";
import v from "../../../_virtual/_plugin-vue_export-helper.js";
const S = { class: "tree__content__wrap" }, y = ["title"], V = { class: "tree__content" }, N = { class: "label" }, E = { class: "tree__bottom" };
function A(e, t, D, $, B, L) {
  var d;
  const f = n("el-icon"), h = n("el-input"), u = n("el-tooltip"), m = n("el-tree");
  return a(), r("div", {
    class: k(["t-tree", e.domOnlyClass])
  }, [
    o("div", S, [
      o("div", {
        class: "tree__top",
        onClick: t[0] || (t[0] = (...l) => e.changeVisibleStatus && e.changeVisibleStatus(...l))
      }, [
        e.finSelectedNodeLabel.length ? (a(), r("span", {
          key: 0,
          class: "tree__top__text",
          title: e.preEchoVal
        }, p(e.preEchoVal), 9, y)) : (a(), r("span", {
          key: 1,
          class: "tree__top__text placeholder__top",
          style: C({
            ...(d = e.cStyle.placeholder) == null ? void 0 : d.default
          })
        }, " 请选择 ", 4)),
        s(f, {
          class: "tree__top__btn",
          onClick: b(e.initSelectData, ["stop"])
        }, {
          default: i(() => t[4] || (t[4] = [
            o("svg", {
              xmlns: "http://www.w3.org/2000/svg",
              viewBox: "0 0 1024 1024"
            }, [
              o("path", {
                fill: "currentColor",
                d: `M764.288 214.592 512 466.88 259.712 214.592a31.936 31.936 0 0 0-45.12 45.12L466.752 512 214.528 764.224a31.936 31.936 0 1 0 45.12
            45.184L512 557.184l252.288 252.288a31.936 31.936 0 0 0 45.12-45.12L557.12 512.064l252.288-252.352a31.936 31.936 0 1 0-45.12-45.184z`
              })
            ], -1)
          ])),
          _: 1,
          __: [4]
        }, 8, ["onClick"])
      ]),
      w(o("div", V, [
        s(h, {
          modelValue: e.filterText,
          "onUpdate:modelValue": t[1] || (t[1] = (l) => e.filterText = l),
          placeholder: "搜索"
        }, null, 8, ["modelValue"]),
        s(m, c({
          ref: "treeRef",
          data: e.data,
          props: e.dataAlias,
          "show-checkbox": e.isCheckbox,
          "filter-node-method": e.filterNode,
          "node-key": "id",
          "default-expand-all": e.showAllNodes,
          "highlight-current": "",
          class: "filter-tree"
        }, e.filterAttrs(e.$attrs), {
          onNodeClick: e.selectNode,
          onCheckChange: e.selectCheckbox,
          onCheck: e.handleCheck
        }), {
          default: i(({ node: l }) => [
            s(u, c({
              effect: "dark",
              content: l.label,
              placement: "right",
              "show-after": 800
            }, e.filterAttrs(e.$attrs)), {
              default: i(() => [
                o("span", N, p(l.label), 1)
              ]),
              _: 2
            }, 1040, ["content"])
          ]),
          _: 1
        }, 16, ["data", "props", "show-checkbox", "filter-node-method", "default-expand-all", "onNodeClick", "onCheckChange", "onCheck"]),
        o("div", E, [
          o("span", {
            class: "tree__bottom__btn",
            onClick: t[2] || (t[2] = (...l) => e.defineSelected && e.defineSelected(...l))
          }, "确定"),
          o("span", {
            class: "tree__bottom__btn",
            onClick: t[3] || (t[3] = (...l) => e.cancelSelected && e.cancelSelected(...l))
          }, "取消")
        ])
      ], 512), [
        [g, e.isExpandAll]
      ])
    ]),
    t[5] || (t[5] = o("div", { class: "style-container" }, null, -1))
  ], 2);
}
const O = /* @__PURE__ */ v(_, [["render", A]]);
export {
  O as default
};
