// @ths/design/es/components/tree/src/index.vue2.js
import { defineComponent as be, computed as O, ref as ae, reactive as Ne, watch as _, onMounted as oe, getCurrentInstance as ve, defineExpose as Se, toRefs as xe } from "vue";
import { filterAttrs as $e, getDomRandomClass as ke, setStyle as s, getCssPropertyString as h } from "../../../utils/index.js";
const ge = be({
  name: "TTree",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 顶部数据回显容器
        dataEchoWrapper: {
          default: {}
        },
        // 内容容器
        contentWrapper: {
          default: {}
        },
        // 搜索框容器
        searchWrapper: {
          default: {},
          focus: {}
        },
        // 树结构内容容器
        treeConWrapper: {
          default: {}
        },
        // 复选框容器
        checkBoxWrapper: {
          default: {},
          hover: {},
          active: {}
        },
        // 每行的树节点容器
        treeNodeWrapper: {
          default: {},
          hover: {},
          active: {}
        },
        // 图标容器
        nodeIconWrapper: {
          default: {}
        },
        // 树节点文本
        nodeText: {
          default: {},
          hover: {},
          active: {}
        },
        // placeholder
        placeholder: {
          default: {}
        },
        // 确定取消按钮
        controlButton: {
          default: {}
        }
      })
    },
    // 默认是否展开
    showAllNodes: {
      type: Boolean,
      default: !1
    },
    // 树结构数据
    data: {
      type: Array,
      default: () => []
    },
    // 是否复选
    isCheckbox: {
      type: Boolean,
      default: !0
    },
    // 边框颜色
    borderColor: {
      type: String,
      default: ""
    },
    // 数据别名
    defaultProps: {
      type: Object,
      default: () => ({
        children: "children",
        label: "label",
        disabled: "disabled"
      })
    },
    // 默认选中节点
    checkedNodes: {
      type: Array,
      default: () => []
    },
    // 数据别名
    props: {
      type: Object,
      default: () => ({
        label: "label",
        children: "children",
        disabled: "disabled"
      })
    }
  },
  emits: ["data-change", "node-click", "check", "cancel", "component-loaded", "focus", "clear"],
  setup(o, { emit: u }) {
    const a = O(() => o.cStyle), l = ke("tree"), A = ae(""), d = ae(), t = Ne({
      data: O(() => o.data),
      // 树数据
      isCheckbox: O(() => o.isCheckbox),
      // 是否多选
      selectedNodeLabel: JSON.parse(JSON.stringify(o.checkedNodes)),
      // 当前被选中的节点内容
      finSelectedNodeLabel: JSON.parse(JSON.stringify(o.checkedNodes)),
      // 最终要回显的数据
      preEchoVal: O(() => {
        var n, r;
        const e = [];
        return (n = t.finSelectedNodeLabel) == null || n.filter((c) => c.label), (r = t.finSelectedNodeLabel) == null || r.forEach((c) => {
          e.push(c.label);
        }), e.join(",") ? e.join(",") : "请选择";
      }),
      // 页面回显值
      isExpandAll: !1,
      // 是否展开所有内容
      dataAlias: o.props
    }), ne = (e, n) => e ? n.label.indexOf(e) !== -1 : !0, re = () => {
      t.isExpandAll = !t.isExpandAll, t.isExpandAll && t.isCheckbox && d.value && t.finSelectedNodeLabel.length > 0 && d.value.setCheckedNodes(
        t.finSelectedNodeLabel,
        !1
      ), u("focus", t.isExpandAll);
    }, ce = (e) => {
      if (!e.length) {
        t.isCheckbox ? u("data-change", []) : u("data-change", { id: -1, [o.defaultProps.label]: "" });
        return;
      }
      t.isCheckbox ? u("data-change", e) : u("data-change", e[0]);
    }, de = () => {
      t.isExpandAll = !1, t.finSelectedNodeLabel = JSON.parse(JSON.stringify(t.selectedNodeLabel)), ce(t.finSelectedNodeLabel);
    }, ie = () => {
      t.isExpandAll = !1, t.selectedNodeLabel = JSON.parse(JSON.stringify(t.finSelectedNodeLabel)), u("cancel");
    }, se = (e) => {
      const { id: n, label: r } = e;
      t.isCheckbox || (t.selectedNodeLabel.length = 0, t.selectedNodeLabel.push({ id: n, label: r })), u("node-click", e);
    }, ue = () => {
      A.value = "", t.selectedNodeLabel = JSON.parse(JSON.stringify([])), t.finSelectedNodeLabel = JSON.parse(JSON.stringify([])), d.value.setCheckedKeys([], !1), u("clear");
    }, pe = (e, n) => {
      const { id: r, label: c } = e, i = t.selectedNodeLabel.findIndex((p) => r === p.id);
      if (n) {
        if (i > -1)
          return;
        t.selectedNodeLabel.push({ id: r, label: c });
      } else {
        if (i < 0)
          return;
        t.selectedNodeLabel.splice(i, 1);
      }
    }, fe = (e, n) => {
      const c = n.checkedNodes.findIndex((p) => p.id === e.id) > -1, i = n.checkedNodes;
      u("check", e, c, i);
    }, he = () => {
      var n, r, c, i, p, b, N, v, S, x, $, k, y, C, f, L, W, g, T, w, B, I, j, P, H, D, R, q, M, V, z, K, F, G, Q, U, X, Y, Z, ee, te;
      let e = "";
      if ((n = a.value) != null && n.wrapper && (e += s(
        (r = a.value) == null ? void 0 : r.wrapper,
        `.${l} .tree__content__wrap`
      )), (c = a.value) != null && c.contentWrapper && (e += s(
        (i = a.value) == null ? void 0 : i.contentWrapper,
        `.${l} .tree__content__wrap .tree__content`
      )), o.borderColor) {
        const { borderColor: m } = o;
        e += s(
          { default: { borderColor: m } },
          `.${l} .tree__top,
          .${l} .tree__content,
          .${l} .el-input__inner,
          .${l} .tree__bottom,
          .${l} .tree__bottom__btn:nth-child(1)`
        );
      }
      if ((p = a.value) != null && p.dataEchoWrapper && (e += s(
        (b = a.value) == null ? void 0 : b.dataEchoWrapper,
        `.${l}.t-tree .tree__top`
      )), (N = a.value) != null && N.searchWrapper && (e += s(
        (v = a.value) == null ? void 0 : v.searchWrapper,
        `.${l} .el-input__inner`
      ), ($ = (x = (S = a.value) == null ? void 0 : S.searchWrapper) == null ? void 0 : x.default) != null && $.color && (e += `
            .${l} .el-input__inner::-webkit-input-placeholder,
            .${l} .el-input__inner::-moz-placeholder,
            .${l} .el-input__inner::-ms-input-placeholder,
            .${l} .el-input__inner::placeholder {
              color: ${(C = (y = (k = a.value) == null ? void 0 : k.searchWrapper) == null ? void 0 : y.default) == null ? void 0 : C.color}
            }
          `)), (f = a.value) != null && f.treeConWrapper && (e += s((L = a.value) == null ? void 0 : L.treeConWrapper, `.${l}.t-tree .el-tree`)), (W = a.value) != null && W.checkBoxWrapper && (e += s(
        (g = a.value) == null ? void 0 : g.checkBoxWrapper,
        `.${l} .el-checkbox__inner`
      ), (w = (T = a.value) == null ? void 0 : T.checkBoxWrapper) != null && w.active && (e += `
            .${l} .el-checkbox__input.is-checked .el-checkbox__inner,
            .${l} .el-checkbox__input.is-indeterminate .el-checkbox__inner {
              ${h((I = (B = a.value) == null ? void 0 : B.checkBoxWrapper) == null ? void 0 : I.active)}
            }
          `)), (j = a.value) != null && j.treeNodeWrapper && (e += s(
        (P = a.value) == null ? void 0 : P.treeNodeWrapper,
        `.${l} .el-tree-node__content`
      ), (D = (H = a.value) == null ? void 0 : H.treeNodeWrapper) != null && D.active && (e += `
            .${l} .el-tree--highlight-current .el-tree-node.is-current>.el-tree-node__content {
              background-color: ${(M = (q = (R = a.value) == null ? void 0 : R.treeNodeWrapper) == null ? void 0 : q.active) == null ? void 0 : M.backgroundColor};
            }
          `)), (V = a.value) != null && V.nodeIconWrapper && (e += s(
        (z = a.value) == null ? void 0 : z.nodeIconWrapper,
        `.${l} .el-tree-node__content :not(.is-leaf)`
      )), (K = a.value) != null && K.nodeText && (e += s(
        (F = a.value) == null ? void 0 : F.nodeText,
        `.${l} .el-tree-node__content .el-tree-node__label,
          .${l} .el-tree-node__content .label,
          .${l} .el-tree__empty-text`
      )), (U = (Q = (G = a.value) == null ? void 0 : G.placeholder) == null ? void 0 : Q.default) != null && U.color) {
        const m = (Z = (Y = (X = a.value) == null ? void 0 : X.placeholder) == null ? void 0 : Y.default) == null ? void 0 : Z.color, le = document.querySelector(`.${l}`);
        m && le && le.style.setProperty("--t-tree-placeholder-text-color", m);
      }
      return (ee = a.value) != null && ee.controlButton && (e += s(
        (te = a.value) == null ? void 0 : te.controlButton,
        `.${l} .tree__bottom .tree__bottom__btn`
      )), e;
    }, _e = () => {
      var n, r, c, i, p, b, N, v, S, x, $, k, y, C;
      let e = "";
      if ((r = (n = a.value) == null ? void 0 : n.wrapper) != null && r.default) {
        const {
          backgroundColor: f,
          color: L,
          width: W,
          height: g
        } = (i = (c = a.value) == null ? void 0 : c.wrapper) == null ? void 0 : i.default;
        e += `
          .${l} .el-input__inner,
          .${l} .el-tree,
          .${l} .tree__content,
          .${l} .tree__top {
            ${h({ backgroundColor: f, color: L })}
          }
        `, e += `
        .${l} .tree__content {
          ${h({ width: W })}
        }
        `, e += `
        .${l} .tree__top {
          ${h({ height: g })}
        }
        `, e += `
        .${l} .tree__top__text {
          ${h({ lineHeight: g })}
        }
        `;
      }
      if ((N = (b = (p = a.value) == null ? void 0 : p.nodeText) == null ? void 0 : b.hover) != null && N.color) {
        const { color: f } = (S = (v = a.value) == null ? void 0 : v.nodeText) == null ? void 0 : S.hover;
        e += `
          .${l} .el-tree-node__content:hover .el-tree-node__label {
            ${h({ color: f })}
          }
        `;
      }
      if ((k = ($ = (x = a.value) == null ? void 0 : x.nodeText) == null ? void 0 : $.active) != null && k.color) {
        const { color: f } = (C = (y = a.value) == null ? void 0 : y.nodeText) == null ? void 0 : C.active;
        e += `
          .${l} .is-current>.el-tree-node__content .el-tree-node__label {
            ${h({ color: f })}
          }
        `;
      }
      return e;
    }, E = () => {
      const e = document.querySelector(`.${l} .style-container`);
      e && (e.innerHTML = `<style>${he()}${_e()}</style>`);
    };
    _(A, (e) => {
      d.value.filter(e);
    }), _(() => [o.cStyle, o.borderColor], () => {
      E();
    }), _(() => o.props.label, (e) => {
      t.dataAlias.label = e;
    }), _(() => o.checkedNodes, () => {
      t.selectedNodeLabel = JSON.parse(JSON.stringify(o.checkedNodes)), t.finSelectedNodeLabel = JSON.parse(JSON.stringify(o.checkedNodes)), t.isCheckbox && d.value && t.finSelectedNodeLabel.length > 0 && d.value.setCheckedNodes(
        t.finSelectedNodeLabel,
        !1
      );
    }), _(() => o.props.children, (e) => {
      t.dataAlias.children = e;
    }), _(() => o.props.disabled, (e) => {
      t.dataAlias.disabled = e;
    }), oe(() => {
      E(), t.isCheckbox && t.finSelectedNodeLabel.length > 0 && d.value && d.value.setCheckedNodes(
        t.finSelectedNodeLabel,
        !1
      );
    });
    const J = (e) => {
      t.isExpandAll = e, t.isExpandAll && t.isCheckbox && d.value && t.finSelectedNodeLabel.length > 0 && d.value.setCheckedNodes(
        t.finSelectedNodeLabel,
        !1
      );
    };
    return oe(() => {
      const e = ve();
      u("component-loaded", e == null ? void 0 : e.proxy);
    }), Se({
      changePanelStatus: J
    }), {
      ...xe(t),
      curStyle: a,
      filterText: A,
      filterNode: ne,
      treeRef: d,
      changeVisibleStatus: re,
      selectNode: se,
      defineSelected: de,
      cancelSelected: ie,
      initSelectData: ue,
      filterAttrs: $e,
      domOnlyClass: l,
      selectCheckbox: pe,
      handleCheck: fe,
      changePanelStatus: J
    };
  }
});
export {
  ge as default
};
