// @ths/design/es/components/button-back/src/index.vue2.js
import { defineComponent as d, computed as n } from "vue";
import { getDomRandomClass as f } from "../../../utils/index.js";
import "../../button/index.js";
import i from "../../button/src/index.vue.js";
const S = d({
  name: "TButtonBack",
  components: {
    TButton: i
  },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        },
        // 按钮
        button: {
          default: {},
          focus: {},
          hover: {}
        },
        // 文字
        text: {
          default: {},
          focus: {},
          hover: {}
        }
      })
    },
    // 按钮文字
    label: {
      type: String,
      default: "返回"
    }
  },
  setup(e) {
    const l = n(() => e.cStyle), s = n(() => e.label);
    return {
      domOnlyClass: f("button-back"),
      curStyle: l,
      curLabel: s,
      handleBackClick: () => {
        window.history.back();
      },
      filterAttrs: (o, a = []) => {
        const c = {};
        return Object.keys(o).forEach((t) => {
          ["data-id", "cStyle", "class", "style", "onClick"].concat(a).reduce((r, u) => r || t === u, !1) || (c[t] = o[t]);
        }), c;
      }
    };
  }
});
export {
  S as default
};
