// @ths/design/es/components/dialog-multiple-tree-input/src/index.vue2.js
import { defineComponent as m, reactive as f, watch as c, toRefs as g } from "vue";
import I from "../../dialog-multiple-tree/src/index.vue.js";
const b = m({
  name: "TDialogMultipleTreeInput",
  components: {
    TDialogMultipleTree: I
  },
  props: {
    // 选中的节点id
    selectedId: {
      type: String,
      default: ""
    },
    // 选中的节点名称
    selectedName: {
      type: String,
      default: ""
    },
    // 是否不可编辑
    disabled: {
      type: Boolean,
      default: !1
    },
    // 输入框提示文字
    placeholder: {
      type: String,
      default: "请选择"
    }
  },
  emits: ["update:selectedId", "update:selectedName", "selected", "clear"],
  setup(d, { emit: t, attrs: s }) {
    const l = f({
      // 单选角色弹框是否可见
      visible: !1,
      // 选中的节点名称
      curSelectedName: "",
      // 选中的节点id
      curSelectedId: ""
    });
    c(() => d.selectedName, (e) => {
      l.curSelectedName = e;
    }, {
      deep: !0,
      immediate: !0
    }), c(() => d.selectedId, (e) => {
      l.curSelectedId = e;
    }, {
      deep: !0,
      immediate: !0
    });
    const o = (e) => {
      l.visible = !0, e.target.blur();
    }, r = () => {
      l.visible = !1;
    }, i = () => {
      t("update:selectedId", ""), t("update:selectedName", ""), t("clear");
    }, p = (e) => {
      const u = e.map((a) => a[s["simple-data"].idKey]).toString(), n = e.map((a) => a[s["simple-data"].nameKey]).toString();
      t("update:selectedId", u), t("update:selectedName", n), t("selected", e), l.visible = !1;
    };
    return {
      ...g(l),
      clearInput: i,
      handleDialogSure: p,
      openDialog: o,
      closeDialog: r
    };
  }
});
export {
  b as default
};
