// @ths/design/es/components/dialog-multiple-tree-input/src/index.vue.js
import u from "./index.vue2.js";
import { resolveComponent as l, openBlock as m, createElementBlock as c, createVNode as n, withCtx as t, createTextVNode as _, mergeProps as g } from "vue";
import b from "../../../_virtual/_plugin-vue_export-helper.js";
const f = { class: "input-wrapper" };
function V(e, o, C, D, S, h) {
  const a = l("search"), r = l("el-icon"), d = l("el-button"), p = l("el-input"), s = l("t-dialog-Multiple-tree");
  return m(), c("div", f, [
    n(p, {
      modelValue: e.curSelectedName,
      "onUpdate:modelValue": o[0] || (o[0] = (i) => e.curSelectedName = i),
      placeholder: e.placeholder,
      clearable: "",
      disabled: e.disabled,
      onFocus: e.openDialog,
      onClear: e.clearInput
    }, {
      append: t(() => [
        n(d, {
          size: "small",
          onClick: e.openDialog
        }, {
          default: t(() => [
            n(r, { size: 13 }, {
              default: t(() => [
                n(a)
              ]),
              _: 1
            }),
            o[2] || (o[2] = _(" 选择 "))
          ]),
          _: 1,
          __: [2]
        }, 8, ["onClick"])
      ]),
      _: 1
    }, 8, ["modelValue", "placeholder", "disabled", "onFocus", "onClear"]),
    n(s, g({
      dialogVisible: e.visible,
      "onUpdate:dialogVisible": o[1] || (o[1] = (i) => e.visible = i),
      "hidden-id": e.curSelectedId
    }, e.$attrs, { onDialogSure: e.handleDialogSure }), null, 16, ["dialogVisible", "hidden-id", "onDialogSure"])
  ]);
}
const N = /* @__PURE__ */ b(u, [["render", V]]);
export {
  N as default
};
