// @ths/design/es/components/chart-funnel/src/index.vue.js
import r from "./index.vue2.js";
import { openBlock as i, createElementBlock as m, normalizeClass as a, withDirectives as o, createElementVNode as e, normalizeStyle as l, vShow as n, toDisplayString as p } from "vue";
import "./index.vue3.js";
import c from "../../../_virtual/_plugin-vue_export-helper.js";
const y = { class: "empty" }, d = ["src"], u = { class: "empty-text" };
function f(t, s, h, v, E, g) {
  return i(), m("div", {
    class: a(["t-chart-funnel", t.domOnlyClass])
  }, [
    o(e("div", {
      ref: "chart",
      style: l({ ...t.style })
    }, null, 4), [
      [n, t.checkEmpty]
    ]),
    o(e("div", y, [
      e("img", {
        class: "empty-img",
        style: l({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, d),
      e("span", u, p(t.emptyText), 1),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [n, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 2);
}
const _ = /* @__PURE__ */ c(r, [["render", f]]);
export {
  _ as default
};
