// @ths/design/es/components/chart-funnel/src/index.vue2.js
import { defineComponent as j, computed as u, ref as z, reactive as q, onMounted as E, watch as $, nextTick as B } from "vue";
import * as H from "echarts";
import I from "lodash/merge";
import { getDomRandomClass as K, setStyle as m } from "../../../utils/index.js";
var b = /* @__PURE__ */ ((t) => (t.LIGHT = "light", t.DARK = "dark", t))(b || {});
const R = j({
  name: "TChartFunnel",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 暂无数据图片
        emptyImg: {
          default: {}
        },
        // 暂无数据文字
        emptyText: {
          default: {}
        }
      })
    },
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据文字
    emptyText: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 图表数据
    data: {
      type: Array,
      default: () => []
    },
    // 名称别名
    nameKey: {
      type: String,
      default: () => "name"
    },
    // 值别名
    dataKey: {
      type: String,
      default: () => "value"
    },
    // 整个图表宽度
    width: {
      type: String,
      default: "100%",
      required: !1
    },
    // 整个图表高度
    height: {
      type: String,
      default: "300px",
      required: !1
    },
    // 整个图表标题
    title: {
      type: String,
      default: "",
      required: !1
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 默认图例排列方式
    legendStyle: {
      type: Object,
      default: () => ({
        position: "top",
        align: "center",
        fontSize: 12,
        iconSize: 12,
        col: 1
      }),
      required: !1,
      validator: (t) => !!t
    },
    // 主题风格
    themeStyle: {
      type: String,
      default: "light",
      validator: (t) => t.toUpperCase() in b
    }
  },
  emits: ["com-click", "legend-select"],
  setup(t, { emit: h }) {
    const a = u(() => t.cStyle), c = K("chart-funnel"), s = z(null), f = u(() => {
      var e, o, i, d;
      if (t.options) {
        if (Array.isArray((e = t.options) == null ? void 0 : e.series))
          return t.options.series.some((l) => {
            var w;
            return (l == null ? void 0 : l.data) && ((w = l == null ? void 0 : l.data) == null ? void 0 : w.length) !== 0;
          });
        if (typeof ((o = t.options) == null ? void 0 : o.series) == "object")
          return ((d = (i = t.options.series) == null ? void 0 : i.data) == null ? void 0 : d.length) === 0;
      }
      return !!t.data.length;
    }), k = z({
      width: f.value ? t.width : "0",
      height: f.value ? t.height : "0"
    }), A = {
      light: {
        fontColor: "#000",
        tooltipBackgroundColor: "rgba(255, 255, 255, 1)",
        tooltipFontColor: "#000"
      },
      dark: {
        fontColor: "#fff",
        tooltipBackgroundColor: "rgba(34, 68, 99, 1)",
        tooltipFontColor: "#fff"
      }
    }, r = () => A[t.themeStyle], y = (e) => e ? String(e).split(" ").map((i) => `${i}px`).join(" ") : "", T = u(() => ({
      position: t.legendStyle.position || "top",
      align: t.legendStyle.align || "center",
      margin: y(t.legendStyle.margin),
      fontSize: t.legendStyle.fontSize ? y(t.legendStyle.fontSize) : "14px",
      iconSize: t.legendStyle.iconSize ? y(t.legendStyle.iconSize) : "14px",
      cStyle: t.legendStyle.cStyle
    })), D = () => {
      let e = "";
      return a.value.wrapper && (e += m(a.value.wrapper, `.${c}`)), a.value.emptyImg && (e += m(a.value.emptyImg, `.${c} .empty .empty-img`)), a.value.emptyText && (e += m(a.value.emptyText, `.${c} .empty .empty-text`)), e;
    }, S = () => {
      const e = document.querySelector(`.${c} .style-container`);
      e && (e.innerHTML = `<style>${D()}</style>`);
    };
    let p = [];
    const O = () => {
      const e = [];
      t.data.forEach((o, i) => {
        o.name && e.push(o.name);
      }), p = e;
    }, C = q({
      // 图表数据
      chartData: []
    }), x = () => {
      var e;
      if (t.data && t.data.length) {
        const { nameKey: o, dataKey: i } = t, d = (e = t.options) == null ? void 0 : e.color;
        C.chartData = t.data.map((g, l) => ({
          name: g[o] || "-",
          value: g[i] || 0,
          itemStyle: {
            color: d && d[l] || "#fcc"
          }
        }));
      }
    };
    let n = null;
    const v = () => {
      if (O(), s.value) {
        n || (n = H.init(s.value, t.themeStyle));
        const e = I({
          title: {
            show: !1,
            text: t.title,
            left: "50%",
            padding: [0, 5, 0, 5],
            textAlign: "center",
            textStyle: {
              color: r().fontColor,
              fontSize: 16,
              fontWeight: "bold"
            }
          },
          legend: {
            itemWidth: 15,
            itemHeight: 15,
            itemGap: 20,
            top: "top",
            left: "center",
            orient: "horizontal",
            textStyle: {
              color: r().fontColor,
              fontSize: 16,
              height: 14,
              verticalAlign: "top",
              rich: {
                a: {
                  verticalAlign: "top"
                }
              }
            },
            data: p
          },
          tooltip: {
            trigger: "item",
            formatter: "{b} : {c}%",
            backgroundColor: r().tooltipBackgroundColor,
            borderColor: r().tooltipBackgroundColor,
            textStyle: {
              color: r().fontColor
            }
          },
          series: [
            {
              type: "funnel",
              data: C.chartData,
              label: {
                show: !0,
                position: "inside",
                formatter: "{c}%",
                color: r().fontColor
              }
            }
          ],
          xAxis: {
            show: !1
          },
          yAxis: {
            show: !1
          }
        }, t.options);
        n.clear(), n.setOption(e), n.on("click", (o) => {
          h("com-click", o);
        }), n.on("legendselectchanged", (o) => {
          h("legend-select", o);
        }), window.onresize = () => {
          n.resize();
        };
      }
    };
    return E(() => {
      S(), x(), v();
    }), $(() => t, () => {
      S(), x(), v(), B(() => {
        n.resize(), window.onresize = () => {
          n.resize();
        };
      });
    }, { deep: !0, immediate: !0 }), {
      chart: s,
      style: k,
      curStyle: a,
      domOnlyClass: c,
      checkEmpty: f,
      legendStyles: T
    };
  }
});
export {
  R as default
};
