// @ths/design/es/components/background-animation/src/index.vue2.js
import { defineComponent as y, computed as _, watch as W, onMounted as v } from "vue";
import { getDomRandomClass as S, setStyle as n } from "../../../utils/index.js";
const g = y({
  name: "TBackgroundAnimation",
  props: {
    // 图片数据
    images: {
      type: Object,
      default: () => ({}),
      require: !0
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 污染源图片容器
        polluteWrapper: {
          default: {}
        },
        // 底图图片容器
        baseWrapper: {
          default: {}
        },
        // 动态线图片容器
        lineWrapper: {
          default: {}
        }
      })
    }
  },
  setup(r, { emit: b }) {
    const e = _(() => r.cStyle), a = S("background-animation"), d = () => {
      var o, p, u, i, c, s, m, f;
      let t = "";
      return (o = e.value) != null && o.wrapper && (t += n((p = e.value) == null ? void 0 : p.wrapper, `.${a} .container`)), (u = e.value) != null && u.polluteWrapper && (t += n((i = e.value) == null ? void 0 : i.polluteWrapper, `.${a} .container__pollute`)), (c = e.value) != null && c.baseWrapper && (t += n((s = e.value) == null ? void 0 : s.baseWrapper, `.${a} .container__base`)), (m = e.value) != null && m.lineWrapper && (t += n((f = e.value) == null ? void 0 : f.lineWrapper, `.${a} .container__line1, .${a} .container__line2`)), t;
    }, l = () => {
      const t = document.querySelector(`.${a} .style-container`);
      t && (t.innerHTML = `<style>${d()}</style>`);
    };
    return W(() => r.cStyle, () => {
      l();
    }), v(() => {
      l();
    }), {
      curStyle: e,
      domOnlyClass: a
    };
  }
});
export {
  g as default
};
