// @ths/design/es/components/scrollable-cards/src/index.vue.js
import V from "./index.vue2.js";
import { openBlock as r, createElementBlock as o, normalizeClass as v, createElementVNode as s, Fragment as g, renderList as C, normalizeStyle as t, createCommentVNode as d, createTextVNode as p, toDisplayString as n } from "vue";
import "./index.vue3.js";
import m from "../../../_virtual/_plugin-vue_export-helper.js";
const k = {
  ref: "cardContainerRef",
  class: "scrollable-cards-container"
}, A = {
  ref: "cardBodyRef",
  class: "scrollable-cards-body"
}, S = ["onClick"], D = ["title"], O = { class: "scrollable-cards-item-value_unit" }, $ = ["title"];
function w(l, a, h, R, B, H) {
  return r(), o("div", {
    ref: "scrollCardsRef",
    class: v(["t-scrollable-cards", l.domOnlyClass])
  }, [
    s("div", {
      class: "scrollable-cards-left",
      onClick: a[0] || (a[0] = (e) => l.slidCardHandler("left"))
    }, a[2] || (a[2] = [
      s("i", { class: "el-icon-arrow-left scrollable-cards-arrow-left" }, null, -1)
    ])),
    s("div", k, [
      s("div", A, [
        (r(!0), o(g, null, C(l.dataList, (e, i) => {
          var c, u, f, y, b;
          return r(), o("div", {
            key: i,
            class: v(["scrollable-cards-item", { "scrollable-cards-item-checked": l.stateCollection[i] }]),
            style: t({ marginRight: l.cardSpace + "px", ...(c = e[l.curAlias.style]) == null ? void 0 : c.cardStyle }),
            onClick: (I) => l.clickCardHandler(i)
          }, [
            l.showIcon ? (r(), o("span", {
              key: 0,
              class: "scrollable-cards-item-icon",
              style: t({ ...(u = e[l.curAlias.style]) == null ? void 0 : u.cardIconStyle })
            }, null, 4)) : d("", !0),
            l.getValueOrDefaultValue(e[l.curAlias.value], !1) ? (r(), o("div", {
              key: 1,
              class: "scrollable-cards-item-value",
              title: l.getValueOrDefaultValue(e[l.curAlias.value], "-"),
              style: t((f = e[l.curAlias.style]) == null ? void 0 : f.valueStyle)
            }, [
              p(n(l.getValueOrDefaultValue(e[l.curAlias.value], "-")) + " ", 1),
              s("div", O, n(e[l.curAlias.unit]), 1)
            ], 12, D)) : d("", !0),
            l.getValueOrDefaultValue(e[l.curAlias.label], !1) ? (r(), o("div", {
              key: 2,
              title: l.getValueOrDefaultValue(e[l.curAlias.label], "-"),
              class: "scrollable-cards-item-label",
              style: t((y = e[l.curAlias.style]) == null ? void 0 : y.labelStyle)
            }, [
              p(n(l.getValueOrDefaultValue(e[l.curAlias.label], "-")) + " ", 1),
              s("div", {
                class: "scrollable-cards-item-label_icon",
                style: t((b = e[l.curAlias.style]) == null ? void 0 : b.labelIconStyle)
              }, null, 4)
            ], 12, $)) : d("", !0)
          ], 14, S);
        }), 128))
      ], 512)
    ], 512),
    s("div", {
      class: "scrollable-cards-right",
      onClick: a[1] || (a[1] = (e) => l.slidCardHandler("right"))
    }, a[3] || (a[3] = [
      s("i", { class: "el-icon-arrow-right scrollable-cards-arrow-right" }, null, -1)
    ])),
    a[4] || (a[4] = s("div", { class: "style-container" }, null, -1))
  ], 2);
}
const F = /* @__PURE__ */ m(V, [["render", w]]);
export {
  F as default
};
