// @ths/design/es/components/scrollable-cards/src/index.vue2.js
import { defineComponent as X, computed as W, ref as i, onMounted as Y, watch as C, nextTick as S, onUnmounted as Z } from "vue";
import { getValueOrDefaultValue as ee, getDomRandomClass as le, setStyle as c, getCssPropertyString as o } from "../../../utils/index.js";
const te = {
  value: "value",
  // 卡片数值
  label: "label",
  // 卡片标签
  unit: "unit",
  // 单位
  style: "style"
  // 卡片样式
}, ue = X({
  name: "TScrollableCards",
  // 滚动卡片列表组件
  props: {
    // 关键字，用于设置默认选中
    keyWord: {
      type: String,
      default: ""
    },
    /**
     * 设置默认选中字段，当通过关键字，以及defaultChecked设置默认选中
     */
    defaultChecked: {
      type: String,
      default: ""
    },
    // 是否开启默认选中
    tab: {
      type: Boolean,
      default: !1
    },
    // 是否显示角标
    showIcon: {
      type: Boolean,
      default: !1
    },
    // 列表数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => {
      }
    },
    // 组件是否自动滚动
    autoScroll: {
      type: Boolean,
      default: !1
    },
    // 自动滚动速度
    speed: {
      type: Number,
      default: () => 3
    },
    // 时间间隔
    interval: {
      type: Number,
      default: () => 0.5
    },
    // 卡片间隔
    cardSpace: {
      type: Number,
      default: () => 0
    },
    // 滚动方向
    direction: {
      type: String,
      default: () => "left"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层父元素
        wrapper: {
          default: {}
        },
        // 卡片组视口容器
        container: {
          default: {}
        },
        // 单个卡片外层元素
        itemWrapper: {
          default: {},
          hover: {},
          active: {}
        },
        // 单个卡片图标样式
        cardItemIcon: {
          default: {}
        },
        // 左侧箭头按钮
        leftArrowButton: {
          default: {},
          hover: {}
        },
        // 左侧箭头图标
        leftArrow: {
          default: {},
          hover: {}
        },
        // 右侧箭头按钮
        rightArrowButton: {
          default: {},
          hover: {}
        },
        // 右侧箭头图标
        rightArrow: {
          default: {},
          hover: {}
        },
        // 卡片label
        cardLabel: {
          default: {}
        },
        // 卡片labelIcon
        cardLabelIcon: {
          default: {}
        },
        // 卡片内容
        cardValue: {
          default: {}
        },
        // 卡片数据 单位
        cardValueUnit: {
          default: {}
        }
      })
    }
  },
  emits: ["card-click"],
  setup(a, { emit: j }) {
    const t = W(() => a.cStyle), r = le("scrollable-cards"), A = i(null), e = i(null), w = i(null), s = i([]);
    let b = null;
    const v = i(), d = i(!1), h = i(0), B = W(() => Math.round(a.interval * 60)), g = i(!0);
    let u = 0;
    const y = i([]), f = i(-1), P = W(() => ({ ...te, ...a.dataAlias })), z = () => {
      var n, m, H, O, V, R, M, _, q, D, J, U;
      let l = "";
      return t.value.wrapper && (l += c(t.value.wrapper, `.${r}`)), t.value.container && (l += c(t.value.container, `.${r} .scrollable-cards-container`)), t.value.itemWrapper && (l += c(t.value.itemWrapper, `.${r} .scrollable-cards-item`)), (n = t.value.itemWrapper) != null && n.hover && (l += `.${r} .scrollable-cards-item:hover {
          ${o((m = t.value.itemWrapper) == null ? void 0 : m.hover)}
        }`), (H = t.value.itemWrapper) != null && H.active && (l += `.${r} .scrollable-cards-item-checked {
          ${o((O = t.value.itemWrapper) == null ? void 0 : O.active)}
        }`), t.value.cardItemIcon && (l += c(t.value.cardItemIcon, `.${r} .scrollable-cards-item-icon`)), (V = t.value.leftArrowButton) != null && V.hover && (l += `.${r} .scrollable-cards-left:hover {
          ${o((R = t.value.leftArrowButton) == null ? void 0 : R.hover)}
        }`), t.value.leftArrowButton && (l += `.${r} .scrollable-cards-left {
          ${o(t.value.leftArrowButton.default)}
        }`), t.value.rightArrowButton && (l += `.${r} .scrollable-cards-right {
          ${o(t.value.rightArrowButton.default)}
        }`), (M = t.value.rightArrowButton) != null && M.hover && (l += `.${r} .scrollable-cards-right:hover {
          ${o((_ = t.value.rightArrowButton) == null ? void 0 : _.hover)}
        }`), (q = t.value.leftArrow) != null && q.default && (l += `.${r} .scrollable-cards-arrow-left {
          ${o(t.value.leftArrow.default)}
        }`), (D = t.value.leftArrow) != null && D.hover && (l += `.${r} .scrollable-cards-arrow-left:hover {
          ${o(t.value.leftArrow.hover)}
        }`), (J = t.value.rightArrow) != null && J.default && (l += `.${r} .scrollable-cards-arrow-right {
          ${o(t.value.rightArrow.default)}
        }`), (U = t.value.rightArrow) != null && U.hover && (l += `.${r} .scrollable-cards-arrow-right:hover {
          ${o(t.value.rightArrow.hover)}
        }`), t.value.cardLabel && (l += c(t.value.cardLabel, `.${r} .scrollable-cards-item-label`)), t.value.cardLabelIcon && (l += c(t.value.cardLabelIcon, `.${r} .scrollable-cards-item-label_icon`)), t.value.cardValue && (l += c(t.value.cardValue, `.${r} .scrollable-cards-item-value`)), t.value.cardValueUnit && (l += c(t.value.cardValueUnit, `.${r} .scrollable-cards-item-value_unit`)), l;
    }, I = () => {
      const l = document.querySelector(`.${r} .style-container`);
      l && (l.innerHTML = `<style>${z()}</style>`);
    }, G = (l) => {
      f.value !== -1 && l !== f.value && (y.value[f.value] = !1), l !== f.value && (y.value[l] = !0, f.value = l), j("card-click", JSON.parse(JSON.stringify(s.value[l])));
    }, L = () => {
      e.value.style.transition = "", e.value.appendChild(e.value.children[0]), e.value.style.left = "0px", g.value = !0;
    }, $ = () => {
      g.value = !0;
    }, K = (l) => {
      if (!g.value)
        return;
      g.value = !1, e.value.removeEventListener("transitionend", L), e.value.removeEventListener("transitionend", $), u = Number(e.value.style.left.slice(0, -2));
      const n = e.value.children[0].offsetWidth;
      l === "right" ? (-u === n && (e.value.style.transition = "", e.value.appendChild(e.value.children[0]), e.value.style.left = "0px"), clearTimeout(b), b = setTimeout(() => {
        e.value.style.transition = "left 0.5s ease-in-out";
        const m = n + u + a.cardSpace;
        e.value.style.left = `${u - m}px`, e.value.addEventListener("transitionend", L);
      }, 0)) : u === 0 ? (e.value.style.transition = "", e.value.style.left = `${-n - a.cardSpace}px`, e.value.insertBefore(e.value.children[s.value.length - 1], e.value.children[0]), clearTimeout(b), b = setTimeout(() => {
        e.value.style.transition = "left 0.5s ease-in-out", e.value.style.left = "0px", e.value.addEventListener("transitionend", $);
      }, 1)) : (e.value.style.transition = "left 0.5s ease-in-out", e.value.style.left = "0px", e.value.addEventListener("transitionend", $));
    }, k = () => {
      u = Number(e.value.style.left.slice(0, -2));
      const l = e.value.children[0].offsetWidth;
      e.value.style.transition = "left 0s ease-in-out", d.value ? e.value.style.left = `${u - a.speed}px` : h.value < B.value ? h.value++ : (h.value = 0, d.value = !0), u = Number(e.value.style.left.slice(0, -2)), -u > l + a.cardSpace && (d.value = !1, e.value.style.transition = "", e.value.style.left = "0px", e.value.appendChild(e.value.children[0])), u = Number(e.value.style.left.slice(0, -2)), v.value = requestAnimationFrame(k);
    }, N = () => {
      w.value.offsetWidth, e.value.offsetWidth, e.value.children[0].offsetWidth, v.value = requestAnimationFrame(k);
    }, p = () => {
      const l = e.value.children[0].offsetWidth;
      u = Number(e.value.style.left.slice(0, -2)), e.value.style.transition = "left 0s ease-in-out", d.value ? e.value.style.left = `${u + a.speed}px` : h.value <= B.value ? h.value++ : (h.value = 0, d.value = !0, e.value.style.transition = "", e.value.style.left = `${-l}px`, e.value.insertBefore(e.value.children[s.value.length - 1], e.value.children[0])), u = Number(e.value.style.left.slice(0, -2)), u >= 0 && (d.value = !1), u = Number(e.value.style.left.slice(0, -2)), v.value = requestAnimationFrame(p);
    }, E = () => {
      const l = w.value.offsetWidth, n = e.value.offsetWidth, m = e.value.children[0].offsetWidth;
      u = Number(e.value.style.left.slice(0, -2)), n - l >= m && (d.value && (e.value.style.transition = "", u === 0 && (e.value.style.left = `${-m}px`, e.value.insertBefore(e.value.children[s.value.length - 1], e.value.children[0]))), v.value = requestAnimationFrame(p));
    }, x = () => {
      cancelAnimationFrame(v.value), g.value = !0;
    }, T = () => {
      cancelAnimationFrame(v.value), a.direction === "left" ? N() : E();
    }, Q = () => {
      S(() => {
        const l = A.value;
        a.direction === "left" ? N() : E(), l.addEventListener("mouseenter", x), l.addEventListener("mouseleave", T);
      });
    }, F = () => {
      clearTimeout(b), cancelAnimationFrame(v.value), S(() => {
        A.value.removeEventListener("mouseenter", x), A.value.removeEventListener("mouseleave", T), e.value.removeEventListener("transitionend", L), e.value.removeEventListener("transitionend", $);
      });
    };
    return Y(() => {
      I();
    }), C(() => a.cStyle, () => {
      I();
    }, { deep: !0 }), C(() => [a.data, a.keyWord, a.defaultChecked, a.tab], () => {
      if (a.data.length ? (s.value = JSON.parse(JSON.stringify(a.data)), y.value = new Array(s.value.length).fill(!1)) : s.value = [], a.data.length > 0 && a.keyWord !== "" && a.defaultChecked !== "") {
        const l = s.value.findIndex((n) => String(n[a.keyWord]) === String(a.defaultChecked));
        l >= 0 && S(() => {
          y.value[l] = !0, f.value = l;
        });
      } else
        a.tab && S(() => {
          y.value[0] = !0, f.value = 0;
        });
    }, { deep: !0, immediate: !0 }), C(() => [a.data, a.autoScroll, a.direction, a.interval, a.cardSpace], () => {
      F(), a.autoScroll && a.data.length && Q();
    }, { deep: !0, immediate: !0 }), Z(() => {
      F();
    }), {
      curStyle: t,
      domOnlyClass: r,
      scrollCardsRef: A,
      cardBodyRef: e,
      cardContainerRef: w,
      slidCardHandler: K,
      dataList: s,
      clickCardHandler: G,
      getValueOrDefaultValue: ee,
      stateCollection: y,
      curAlias: P
    };
  }
});
export {
  ue as default
};
