// @ths/design/es/components/four-dimension/src/index.vue2.js
import { defineComponent as v, computed as D, reactive as g, onMounted as W, watch as o, toRefs as R } from "vue";
import h from "lodash/merge";
import S from "lodash/cloneDeep";
import T from "./item.vue.js";
import { getDomRandomClass as $, setStyle as l } from "../../../utils/index.js";
const d = {
  titleKey: "title",
  valueKey: "value",
  unitKey: "unit"
}, b = v({
  name: "TFourDimension",
  components: { TFourDimensionItem: T },
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 左侧容器
        sideLeftWrapper: {
          default: {}
        },
        // 右侧容器
        sideRightWrapper: {
          default: {}
        },
        // 两侧单个容器
        sideItemWrapper: {
          default: {}
        },
        // 中间容器
        centerWrapper: {
          default: {}
        },
        // 中间值
        centerValue: {
          default: {}
        },
        // 中间标题
        centerTitle: {
          default: {}
        },
        // 两侧值
        sideValue: {
          default: {}
        },
        // 两侧单位
        sideUnit: {
          default: {}
        },
        // 两侧标题容器
        sideTitleBox: {
          default: {}
        },
        // 两侧标题
        sideTitle: {
          default: {}
        }
      })
    },
    // 两侧数据
    sideData: {
      type: Object,
      default: () => ({})
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => d
    },
    // 中心数据
    centerData: {
      type: Object,
      default: () => ({})
    },
    // 中心数据是否颠倒展示
    centerReverse: {
      type: Boolean,
      default: !1
    },
    // 两侧数据是否颠倒展示
    sideReverse: {
      type: Boolean,
      default: !1
    }
  },
  setup(a) {
    const e = D(() => a.cStyle), r = $("four-dimension"), i = g({
      currentLeftData: [],
      // 左侧数据
      currentRightData: []
      // 右侧数据
    }), c = () => {
      const t = h(S(d), a.dataAlias), { titleKey: p, valueKey: m, unitKey: y } = t;
      let u = [];
      a.sideData && a.sideData.length && (u = a.sideData.map((n) => ({
        ...n,
        title: n[p],
        unit: n[y],
        value: n[m]
      })), i.currentLeftData = u.slice(0, 2), i.currentRightData = u.slice(2, 4));
    }, f = () => {
      let t = "";
      return e.value.wrapper && (t += l(e.value.wrapper, `.${r}`)), e.value.sideLeftWrapper && (t += l(e.value.sideLeftWrapper, `.${r} .container-left`)), e.value.sideRightWrapper && (t += l(e.value.sideRightWrapper, `.${r} .container-right`)), e.value.centerWrapper && (t += l(e.value.centerWrapper, `.${r} .container-center`)), e.value.centerValue && (t += l(e.value.centerValue, `.${r} .container-center .value`)), e.value.centerTitle && (t += l(e.value.centerTitle, `.${r} .container-center .title`)), t;
    }, s = () => {
      const t = document.querySelector(`.${r} > .style-container`);
      t && (t.innerHTML = `<style>${f()}</style>`);
    };
    return W(() => {
      s(), c();
    }), o(
      () => [a.cStyle],
      () => {
        s();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), o(
      () => [a.sideData, a.dataAlias],
      () => {
        c();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      domOnlyClass: r,
      curStyle: e,
      ...R(i)
    };
  }
});
export {
  b as default
};
