// @ths/design/es/components/four-dimension/src/index.vue.js
import c from "./index.vue2.js";
import { resolveComponent as m, openBlock as r, createElementBlock as a, normalizeClass as f, createElementVNode as t, Fragment as o, renderList as u, createBlock as d, normalizeStyle as p, toDisplayString as i } from "vue";
import "./index.vue3.js";
import y from "../../../_virtual/_plugin-vue_export-helper.js";
const v = { class: "container-left side-common" }, S = ["title"], D = ["title"], g = { class: "container-right side-common" };
function B(e, n, k, h, $, T) {
  const s = m("t-four-dimension-item");
  return r(), a("div", {
    class: f(["t-four-dimension", e.domOnlyClass])
  }, [
    t("div", v, [
      (r(!0), a(o, null, u(e.currentLeftData, (l) => (r(), d(s, {
        key: l,
        "item-data": l,
        "c-style": {
          wrapper: e.curStyle.sideItemWrapper || { default: {} },
          value: e.curStyle.sideValue || { default: {} },
          titleBox: e.curStyle.sideTitleBox || { default: {} },
          title: e.curStyle.sideTitle || { default: {} },
          unit: e.curStyle.sideUnit || { default: {} }
        }
      }, null, 8, ["item-data", "c-style"]))), 128))
    ]),
    t("div", {
      class: "container-center g-flex-y-center",
      style: p({ backgroundImage: `url(${e.centerData.bgUrl})`, flexDirection: e.centerReverse ? "column-reverse" : "column" })
    }, [
      t("span", {
        class: "value",
        title: e.centerData.value
      }, i(e.centerData.value || "-"), 9, S),
      t("span", {
        class: "title",
        title: e.centerData.title
      }, i(e.centerData.title || "-") + i(e.centerData.unit || "-"), 9, D)
    ], 4),
    t("div", g, [
      (r(!0), a(o, null, u(e.currentRightData, (l) => (r(), d(s, {
        key: l,
        "item-data": l,
        "c-style": {
          wrapper: e.curStyle.sideItemWrapper || { default: {} },
          value: e.curStyle.sideValue || { default: {} },
          titleBox: e.curStyle.sideTitleBox || { default: {} },
          title: e.curStyle.sideTitle || { default: {} },
          unit: e.curStyle.sideUnit || { default: {} }
        }
      }, null, 8, ["item-data", "c-style"]))), 128))
    ]),
    n[0] || (n[0] = t("div", { class: "style-container" }, null, -1))
  ], 2);
}
const b = /* @__PURE__ */ y(c, [["render", B]]);
export {
  b as default
};
