// @ths/design/es/components/four-dimension/src/item.vue2.js
import { defineComponent as r, computed as n, onMounted as s, watch as c } from "vue";
import { getDomRandomClass as f, setStyle as i } from "../../../utils/index.js";
const y = r({
  name: "TFourDimensionItem",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 值
        value: {
          default: {}
        },
        // 标题容器
        titleBox: {
          default: {}
        },
        // 标题
        title: {
          default: {}
        },
        // 单位
        unit: {
          default: {}
        }
      })
    },
    // 单项数据
    itemData: {
      type: Object,
      default: () => ({
        bgUrl: "",
        // 整体背景url
        titleBgUrl: "",
        // 标题背景url
        title: "",
        // 名称
        value: "",
        // 值
        unit: ""
        // 单位
      })
    }
  },
  setup(u) {
    const t = n(() => u.cStyle), l = f("four-dimension-item"), o = n(() => u.itemData), m = () => {
      let e = "";
      return t.value.value && (e += i(t.value.value, `.${l} .item-value__box .value`)), t.value.unit && (e += i(t.value.unit, `.${l} .item-value__box .unit`)), t.value.titleBox && (e += i(t.value.titleBox, `.${l} .item-title__box`)), t.value.title && (e += i(t.value.title, `.${l} .item-title__box .item-title`)), e;
    }, a = () => {
      const e = document.querySelector(`.${l} > .style-container`);
      e && (e.innerHTML = `<style>${m()}</style>`);
    };
    return s(() => {
      a();
    }), c(() => [u.cStyle], () => {
      a();
    }, {
      deep: !0,
      immediate: !0
    }), {
      domOnlyClass: l,
      curStyle: t,
      curItemData: o
    };
  }
});
export {
  y as default
};
