// @ths/design/es/components/chart-liquidfill-rect/src/index.vue.js
import f from "./index.vue2.js";
import { openBlock as y, createElementBlock as h, normalizeClass as b, normalizeStyle as n, withDirectives as m, createElementVNode as t, vShow as u } from "vue";
import "./index.vue3.js";
import v from "../../../_virtual/_plugin-vue_export-helper.js";
const E = {
  ref: "chartId",
  class: "chart"
}, $ = { class: "empty" }, w = ["src"];
function S(e, r, k, g, C, R) {
  var s, l, o, i, a, d, p, c;
  return y(), h("div", {
    class: b(["t-chart-liquidfill-rect", e.domOnlyClass]),
    style: n({
      ...(l = (s = e.curStyle) == null ? void 0 : s.wrapper) == null ? void 0 : l.default,
      borderRadius: `${(a = (i = (o = e.curStyle) == null ? void 0 : o.wrapper) == null ? void 0 : i.default) == null ? void 0 : a.borderRadius}px`,
      borderWidth: `${(c = (p = (d = e.curStyle) == null ? void 0 : d.wrapper) == null ? void 0 : p.default) == null ? void 0 : c.borderWidth}px`
    })
  }, [
    m(t("div", E, null, 512), [
      [u, e.checkEmpty]
    ]),
    m(t("div", $, [
      t("img", {
        class: "empty-img",
        style: n({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, w),
      r[0] || (r[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [u, !e.checkEmpty]
    ]),
    r[1] || (r[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const I = /* @__PURE__ */ v(f, [["render", S]]);
export {
  I as default
};
