// @ths/design/es/components/chart-liquidfill-rect/src/index.vue2.js
import { defineComponent as L, computed as j, ref as k, reactive as I, nextTick as q, onMounted as M, onUnmounted as R, watch as C, toRefs as U } from "vue";
import * as p from "echarts";
import d from "lodash/merge";
import v from "lodash/cloneDeep";
import "echarts-liquidfill";
import { DEFAULT_OPTIONS_COMMON as V } from "../../../utils/constant.js";
import { getDomRandomClass as W, setStyle as b } from "../../../utils/index.js";
const J = L({
  name: "TChartLiquidfillRect",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 统计图的options配置项
    options: {
      type: Object,
      default: () => ({})
    },
    // 传入数组数据;如果options设置了 series 属性 则会忽略data的数据，用于水波的数量
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        borderStyle: {},
        // 外层border样式
        data: {
          // 水波球配置
          type: Array,
          default: () => [],
          required: !1
        }
      })
    },
    // 文字值
    name: {
      type: String,
      default: ""
    },
    // 数字值
    value: {
      type: String,
      default: ""
    },
    // 单位
    unit: {
      type: String,
      default: ""
    },
    // 主题样式字段
    liquidfill: {
      type: Object,
      default: () => ({})
    }
  },
  emits: ["title-click"],
  setup(t, { emit: _ }) {
    const n = j(() => t.cStyle), c = W("chart-liquidfill-rect"), O = d(v(V), {
      legend: {
        show: !1
      },
      xAxis: {
        show: !1
      },
      yAxis: {
        show: !1
      },
      series: {}
    }), f = k(null);
    let S = null;
    const o = I({
      // 图表数据
      chartData: [],
      liquidfillOptions: v(t.liquidfill),
      // 水波球配置
      textValue: ""
      // 文字值
    }), D = () => {
      var e;
      if (t.seriesSetting && t.data.length) {
        const { color: u } = t.options;
        o.chartData = (e = t.seriesSetting) == null ? void 0 : e.data.map((i, l) => {
          var r, s, m, g, $, F, A, T;
          const { dataKey: h } = i, a = d({
            itemStyle: {
              color: ((m = (s = (r = i == null ? void 0 : i.style) == null ? void 0 : r.itemStyle) == null ? void 0 : s.color) == null ? void 0 : m.colorStops) && new p.graphic.LinearGradient(
                0,
                0,
                0,
                1,
                (F = ($ = (g = i == null ? void 0 : i.style) == null ? void 0 : g.itemStyle) == null ? void 0 : $.color) == null ? void 0 : F.colorStops
              ) || u && u[l] || O.color[l] || "#fff"
            }
          }, (T = (A = t.seriesSetting) == null ? void 0 : A.data[l]) == null ? void 0 : T.style);
          return {
            value: t.data[l] && t.data[l][h] || 0,
            ...a
          };
        }), o.textValue = `${t.name ? t.name : ""}
${t.value !== null && t.value !== void 0 && t.value !== "" ? t.value : ""}${t.unit !== null && t.unit !== void 0 && t.unit !== "" ? t.unit : ""}`;
      }
    }, z = () => {
      let e = "";
      return n.value.wrapper && (e += b(n.value.wrapper, `.${c}`)), n.value.emptyContainer && (e += b(n.value.emptyContainer, `.${c} .empty`)), n.value.emptyImg && (e += b(n.value.emptyImg, `.${c} .empty .empty-img`)), n.value.emptyText && (e += b(n.value.emptyText, `.${c} .empty .empty-text`)), e;
    }, x = () => {
      const e = document.querySelector(`.${c} .style-container`);
      e && (e.innerHTML = `<style>${z()}</style>`);
    }, y = () => {
      f.value && (f.value.getBoundingClientRect().width ?? 0) && q(() => {
        S.resize();
      });
    }, E = j(() => {
      var e, u, i, l;
      if (t.options) {
        if (Array.isArray((e = t.options) == null ? void 0 : e.series))
          return t.options.series.some((a) => {
            var r;
            return (a == null ? void 0 : a.data) && ((r = a == null ? void 0 : a.data) == null ? void 0 : r.length) !== 0;
          });
        if (typeof ((u = t.options) == null ? void 0 : u.series) == "object")
          return ((l = (i = t.options.series) == null ? void 0 : i.data) == null ? void 0 : l.length) === 0;
      }
      return t.data.length ? (q(() => {
        y();
      }), !0) : !1;
    }), w = () => {
      var a, r;
      const { color: e, series: u } = t.options;
      let i = {};
      if (u)
        i = d(
          {
            type: "liquidFill",
            // 水位图
            silent: !0,
            // center: ['50%', '50%'], // 中心点
            shape: "container",
            amplitude: 4,
            // 水波振幅
            data: o.chartData,
            // data个数代表波浪数
            itemStyle: {
              shadowBlur: 0
              // 波浪上的阴影
            },
            direction: "left",
            period: (s, m) => m % 2 === 0 ? 1200 : 1500,
            // 周期：交错效果占时通过不同周期实现
            animationDuration: 2e3,
            // 向上填充的时间
            animationDurationUpdate: 2e3,
            backgroundStyle: {
              borderWidth: 0,
              // 外边框
              borderColor: "#00FF54",
              // 边框颜色
              color: "rgba(0,255,84,0.3)"
              // 边框内部填充部分颜色
            },
            label: {
              // 标签设置
              position: ["50%", "55%"],
              formatter: o.textValue,
              // 显示文本,
              fontSize: "26px",
              // 文本字号,
              color: "#fff"
            },
            outline: {
              show: !1
            }
          },
          o.liquidfillOptions,
          (a = t.seriesSetting) == null ? void 0 : a.borderStyle
        );
      else if (t.data.length && t.seriesSetting) {
        const s = d(
          {
            type: "liquidFill",
            // 水位图
            silent: !0,
            // center: ['50%', '50%'], // 中心点
            shape: "container",
            amplitude: 4,
            // 水波振幅
            data: o.chartData,
            // data个数代表波浪数
            itemStyle: {
              shadowBlur: 0
              // 波浪上的阴影
            },
            direction: "left",
            period: (m, g) => g % 2 === 0 ? 1200 : 1500,
            // 周期：交错效果占时通过不同周期实现
            animationDuration: 2e3,
            // 向上填充的时间
            animationDurationUpdate: 2e3,
            backgroundStyle: {
              borderWidth: 0,
              // 外边框
              borderColor: "#00FF54",
              // 边框颜色
              color: "rgba(0,255,84,0.3)"
              // 边框内部填充部分颜色
            },
            label: {
              // 标签设置
              position: ["50%", "55%"],
              formatter: o.textValue,
              // 显示文本,
              fontSize: "26px",
              // 文本字号,
              color: "#fff"
            },
            outline: {
              show: !1
            }
          },
          v(t.liquidfill)
        );
        i = d({}, s, (r = t.seriesSetting) == null ? void 0 : r.borderStyle);
      }
      const l = d(
        {
          series: i
        },
        O
      ), h = d(l, t.options);
      S.clear();
      try {
        S.setOption(h);
      } catch (s) {
        console.log(s);
      }
    };
    return M(() => {
      x(), f.value && (S = p.init(f.value), D(), w()), window.addEventListener("resize", y);
    }), R(() => {
      window.removeEventListener("resize", y);
    }), C(
      () => t.liquidfill,
      () => {
        o.liquidfillOptions = v(t.liquidfill), w();
      },
      {
        deep: !0
      }
    ), C(
      [
        () => t.options,
        () => t.data,
        () => t.seriesSetting,
        () => t.name,
        () => t.value,
        () => t.unit
      ],
      () => {
        D(), q(() => {
          w(), y();
        });
      },
      {
        deep: !0
      }
    ), C(
      () => t.cStyle,
      () => {
        x(), y();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), {
      curStyle: n,
      ...U(o),
      chartId: f,
      checkEmpty: E,
      domOnlyClass: c
    };
  }
});
export {
  J as default
};
