// @ths/design/es/components/chart-tag-cloud-3d/src/index.vue2.js
import { defineComponent as V, ref as g, computed as _, onMounted as U, watch as H, onUnmounted as W } from "vue";
import { getDomRandomClass as G, setStyle as K } from "../../../utils/index.js";
const tt = V({
  name: "TChartTagCloud3d",
  props: {
    colors: {
      type: Array,
      default: () => [
        "#F7DF1E",
        "#E34F26",
        "#1572B6",
        "#61DAFB",
        "#4FC08D",
        "#DD0031",
        "#339933",
        "#007ACC"
      ]
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // 标签数据
    tags: {
      type: Array,
      default: () => [
        {
          text: "JavaScript",
          value: 10,
          weight: 10,
          color: "#F7DF1E",
          tooltip: "编程语言"
        },
        {
          text: "HTML5",
          value: 10,
          weight: 8,
          color: "#E34F26",
          tooltip: "超文本标记语言"
        },
        {
          text: "CSS3",
          value: 10,
          weight: 7,
          color: "#1572B6",
          tooltip: "层叠样式表"
        },
        {
          text: "React",
          value: 10,
          weight: 9,
          color: "#61DAFB",
          tooltip: "前端框架"
        },
        {
          text: "Vue",
          value: 10,
          weight: 7,
          color: "#4FC08D",
          tooltip: "前端框架"
        },
        {
          text: "Angular",
          value: 10,
          weight: 6,
          color: "#DD0031",
          tooltip: "前端框架"
        },
        {
          text: "Node.js",
          value: 10,
          weight: 8,
          color: "#339933",
          tooltip: "后端运行环境"
        },
        {
          text: "TypeScript",
          value: 10,
          weight: 7,
          color: "#007ACC",
          tooltip: "JavaScript的超集"
        }
      ]
    },
    // 配置参数
    config: {
      type: Object,
      default: () => ({
        radius: 150,
        speedX: 2e-3,
        speedY: 3e-3,
        reverse: !1,
        autoRotate: !0,
        hoverScale: 1.5
      })
    }
  },
  emits: ["tag-click"],
  setup(o, { emit: k }) {
    const i = g(null), v = /* @__PURE__ */ new Map();
    let d = 0, f = 0, p = null, w = !1, M = 0, C = 0;
    const S = _(() => o.cStyle), T = G("chart-tag-cloud-3d"), q = g(null), h = g(!1), A = g(""), z = g({ x: 0, y: 0 }), j = () => {
      let t = "";
      return S.value.wrapper && (t += K(S.value.wrapper, `.${T}`)), t;
    }, D = () => {
      const t = document.querySelector(`.${T} .style-container`);
      t && (t.innerHTML = `<style>${j()}</style>`);
    }, E = () => {
      i.value && (i.value.innerHTML = "", v.clear());
    }, I = (t) => {
      k("tag-click", t);
    };
    function F(t, a, r, l, n) {
      const u = Math.cos(l), s = Math.sin(l), c = a * u - r * s, e = a * s + r * u, m = Math.cos(n), y = Math.sin(n), R = t * m - e * y, X = t * y + e * m;
      return { x: R, y: c, z: X };
    }
    function $(t, a, r) {
      const l = v.get(t);
      if (!l || !i.value)
        return;
      const n = i.value.getBoundingClientRect(), u = n.width / 2, s = n.height / 2, c = F(l.x, l.y, l.z, a, r), e = (c.z + o.config.radius) / (2 * o.config.radius);
      t.style.transform = `translate3d(${c.x + u}px, ${c.y + s}px, ${c.z}px) scale(${0.5 + e * 0.8})`, t.style.opacity = (0.2 + e * 0.8).toString(), t.style.zIndex = Math.floor(c.z + o.config.radius).toString();
    }
    function O(t, a) {
      A.value = a, z.value = {
        x: t.clientX + 10,
        y: t.clientY + 10
      }, h.value = !0;
    }
    function J() {
      h.value = !1;
    }
    function b(t) {
      h.value && (z.value = {
        x: t.clientX + 10,
        y: t.clientY + 10
      });
    }
    function L() {
      if (!i.value)
        return;
      const t = o.tags.length, a = i.value.getBoundingClientRect(), r = a.width / 2, l = a.height / 2;
      o.tags.forEach((n, u) => {
        const s = Math.acos(-1 + 2 * u / t), c = Math.sqrt(t * Math.PI) * s, e = document.createElement("div");
        e.className = "tag", e.innerHTML = n.text, e.style.color = o.colors[u], e.style.fontSize = `${n.weight * 2 + 8}px`, e.style.fontWeight = "bold", e.style.textShadow = "0 0 8px rgba(0,0,0,0.6)";
        const m = o.config.radius * Math.cos(c) * Math.sin(s), y = o.config.radius * Math.sin(c) * Math.sin(s), R = o.config.radius * Math.cos(s);
        v.set(e, {
          x: m,
          y,
          z: R,
          baseSize: n.weight * 2 + 8
        }), $(e, d, f), e.addEventListener("mouseenter", (X) => {
          const x = v.get(e);
          if (!x)
            return;
          O(X, `${n.text}: ${n.value}` || `查看 ${n.text} 相关内容`);
          const Y = F(x.x, x.y, x.z, d, f);
          e.style.transform = `translate3d(${Y.x + r}px,
           ${Y.y + l}px,
            ${Y.z + 100}px) scale(${o.config.hoverScale})`, e.style.opacity = "1", o.config.autoRotate = !1;
        }), e.addEventListener("mouseleave", () => {
          $(e, d, f), o.config.autoRotate = !0, J();
        }), e.addEventListener("click", () => {
          I(n);
        }), i.value.appendChild(e);
      });
    }
    function B() {
      if (o.config.autoRotate) {
        const t = o.config.reverse ? -1 : 1;
        d += o.config.speedX * t, f += o.config.speedY * t;
      }
      document.querySelectorAll(".tag").forEach((t) => {
        $(t, d, f);
      }), p = requestAnimationFrame(B);
    }
    const P = () => {
      E(), L();
    };
    function N() {
      i.value && (E(), i.value.addEventListener("mousedown", (t) => {
        w = !0, M = t.clientX, C = t.clientY, o.config.autoRotate = !1;
      }), document.addEventListener("mousemove", b), document.addEventListener("mouseup", () => {
        w = !1;
      }), document.addEventListener("mousemove", (t) => {
        if (w) {
          const a = t.clientX - M, r = t.clientY - C;
          f -= a * 5e-3, d -= r * 5e-3, M = t.clientX, C = t.clientY;
        }
      }), L(), B());
    }
    return U(() => {
      D(), window.addEventListener("resize", P), N();
    }), H(
      () => o.cStyle,
      () => {
        D();
      },
      { deep: !0 }
    ), H(
      [
        () => o.tags,
        () => o.colors
      ],
      () => {
        E(), L();
      },
      { deep: !0 }
    ), W(() => {
      p && cancelAnimationFrame(p), window.removeEventListener("resize", P), document.removeEventListener("mousemove", b);
    }), {
      tagCloud: i,
      curStyle: S,
      domOnlyClass: T,
      tooltip: q,
      tooltipVisible: h,
      tooltipText: A,
      tooltipPos: z
    };
  }
});
export {
  tt as default
};
