// @ths/design/es/utils/constant.js
const e = "t-hairline", a = `${e}--top`, t = `${e}--left`, o = [
  {
    name: "O3",
    html: "O<small><sub>3</sub></small>"
  },
  {
    name: "PM10",
    html: "PM<small><sub>10</sub></small>"
  },
  {
    name: "PM25",
    html: "PM<small><sub>2.5</sub></small>"
  },
  {
    name: "SO2",
    html: "SO<small><sub>2</sub></small>"
  },
  {
    name: "SO3",
    html: "SO<small><sub>3</sub></small>"
  },
  {
    name: "CO",
    html: "CO"
  },
  {
    name: "CODMN",
    html: "COD<small><sub>mn</sub></small>"
  },
  {
    name: "VOCS",
    html: "VOC<small><sub>s</sub></small>"
  },
  {
    name: "NO",
    html: "NO"
  },
  {
    name: "NO2",
    html: "NO<small><sub>2</sub></small>"
  },
  {
    name: "NH3",
    html: "NH<small><sub>3</sub></small>"
  },
  {
    name: "NOX",
    html: "NO<small><sub>x</sub></small>"
  },
  {
    name: "NH4",
    html: "NH<small><sub>4</sub><sup>+</sup></small>"
  },
  {
    name: "MNO4",
    html: "MnO<small><sub>4</sub><sup>-</sup></small>"
  },
  {
    name: "PH",
    html: "pH"
  }
], m = [
  {
    name: "O3",
    html: "O‚āÉ"
  },
  {
    name: "PM10",
    html: "PM‚āĀ‚āÄ"
  },
  {
    name: "PM25",
    html: "PM‚āā.‚āÖ"
  },
  {
    name: "SO2",
    html: "SO‚āā"
  },
  {
    name: "SO3",
    html: "SO‚āÉ"
  },
  {
    name: "CO",
    html: "CO"
  },
  {
    name: "CODMN",
    html: "CODmn"
  },
  {
    name: "VOCS",
    html: "VOCs"
  },
  {
    name: "NO",
    html: "NO"
  },
  {
    name: "NO2",
    html: "NO‚āā"
  },
  {
    name: "NH3",
    html: "NH‚āÉ"
  },
  {
    name: "NOX",
    html: "NOx"
  },
  {
    name: "NH4",
    html: "NH‚āĄ‚Āļ"
  },
  {
    name: "MNO4",
    html: "MnO‚āĄ¬Į"
  },
  {
    name: "PH",
    html: "pH"
  }
], s = [
  {
    name: ["O3", "o3"],
    html: "O‚āÉ"
  },
  {
    name: ["PM10", "pm10"],
    html: "PM‚āĀ‚āÄ"
  },
  {
    name: ["PM25", "PM2_5", "PM2.5", "pm25", "pm2_5", "pm25"],
    html: "PM‚āā.‚āÖ"
  },
  {
    name: ["SO2", "so2"],
    html: "SO‚āā"
  },
  {
    name: ["SO3", "so3"],
    html: "SO‚āÉ"
  },
  {
    name: ["CO2", "co2"],
    html: "CO‚āā"
  },
  {
    name: ["CODMN", "codmn"],
    html: "CODmn"
  },
  {
    name: ["VOCS", "vocs"],
    html: "VOCs"
  },
  {
    name: ["no"],
    html: "NO"
  },
  {
    name: ["NO2", "no2"],
    html: "NO‚āā"
  },
  {
    name: ["NH3", "nh3"],
    html: "NH‚āÉ"
  },
  {
    name: ["NOX", "nox"],
    html: "NOx"
  },
  {
    name: ["NH4", "nh4"],
    html: "NH‚āĄ‚Āļ"
  },
  {
    name: ["MNO4", "mno4"],
    html: "MnO‚āĄ¬Į"
  },
  {
    name: ["PH", "ph"],
    html: "pH"
  }
], n = {
  level1: "šľė",
  level2: "ŤČĮ",
  level3: "ŤĹĽŚļ¶śĪ°śüď",
  level4: "šł≠Śļ¶śĪ°śüď",
  level5: "ťáćŚļ¶śĪ°śüď",
  level6: "šł•ťáćśĪ°śüď",
  level7: "-"
}, h = {
  level1: "#05CE04",
  level2: "#FCFE03",
  level3: "#FF6600",
  level4: "#FE0000",
  level5: "#9820F4",
  level6: "#800000",
  level7: "#777"
}, l = Object.freeze({
  color: [
    "#2ec7c9",
    "#b6a2de",
    "#5ab1ef",
    "#ffb980",
    "#d87a80",
    "#8d98b3",
    "#e5cf0d",
    "#97b552",
    "#95706d",
    "#dc69aa",
    "#07a2a4",
    "#9a7fd1",
    "#588dd5",
    "#f5994e",
    "#c05050",
    "#59678c",
    "#c9ab00",
    "#7eb00a",
    "#6f5553",
    "#c14089"
  ],
  backgroundColor: "rgba(0,0,0,0)",
  textStyle: {},
  title: {
    show: !1,
    text: "ś†áťĘė",
    padding: [5, 5, 5, 5],
    left: "left",
    top: "top",
    textStyle: {
      color: "#008acd"
    },
    subtextStyle: {
      color: "#aaaaaa"
    }
  },
  toolbox: {
    iconStyle: {
      borderColor: "#2ec7c9"
    },
    emphasis: {
      iconStyle: {
        borderColor: "#18a4a6"
      }
    }
  },
  legend: {
    show: !0,
    padding: [10, 5, 5, 5],
    textStyle: {
      color: "#333333"
    }
  },
  tooltip: {
    trigger: "axis",
    axisPointer: {
      lineStyle: {
        color: "#008acd",
        width: "1"
      },
      crossStyle: {
        color: "#008acd",
        width: "1"
      }
    }
  },
  markPoint: {
    label: {
      color: "#eeeeee"
    },
    emphasis: {
      label: {
        color: "#eeeeee"
      }
    }
  },
  valueAxis: {
    axisLine: {
      show: !1
    },
    axisTick: {
      show: !1
    },
    axisLabel: {
      show: !0
    },
    splitLine: {
      show: !1
    },
    splitArea: {
      show: !1
    }
  },
  /* ************************** śöāśó∂šŅĚÁēô šłćŤ¶ĀŚą†ťô§ start ************************ */
  dataZoom: [
    {
      // Á¶ĀÁĒ®Áľ©śĒĺ
      disabled: !0,
      type: "inside",
      start: 0,
      end: 100,
      preventDefaultMouseMove: !0
    },
    {
      // ťöźŤóŹśĽĎŚĚó
      show: !1,
      type: "slider",
      start: 0,
      end: 100,
      height: 12,
      borderColor: "transparent",
      backgroundColor: "#f8f8f8",
      dataBackground: {
        lineStyle: {
          width: 0
        },
        areaStyle: {
          color: "transparent"
        }
      },
      fillerColor: "#e1e1e1",
      handleIcon: "path://path://M100, 100m -75, 0a75,75 0 1,0 150,0a75,75 0 1,0 -150,0",
      handleStyle: {
        color: "#fff",
        shadowColor: "rgba(0, 0, 0, 0.2)",
        shadowBlur: 4
      },
      handleSize: "100%",
      textStyle: {
        color: "#888"
      },
      bottom: "10px"
    }
  ]
  /* ************************** śöāśó∂šŅĚÁēô šłćŤ¶ĀŚą†ťô§ end ************************ */
}), r = l, i = Object.freeze({
  // XŤĹīťÖćÁĹģ
  xAxis: {
    show: !0,
    name: "",
    type: "category",
    nameTextStyle: {
      align: "left",
      color: "#666",
      fontSize: "12",
      fontFamily: "PF"
    },
    nameGap: 5,
    boundaryGap: !1,
    axisLine: {
      show: !0,
      lineStyle: {
        color: "#cacaca",
        width: 1,
        type: "solid"
      }
    },
    axisTick: {
      show: !1,
      lineStyle: {
        color: "#333"
      }
    },
    splitLine: {
      show: !1,
      lineStyle: {
        color: "rgba(202,202,202,0.3)",
        width: 1,
        type: "solid"
      }
    },
    axisLabel: {
      show: !0,
      rotate: 0,
      color: "#666",
      fontSize: "12",
      fontFamily: "PF",
      margin: 10
    }
  },
  // yŤĹīťÖćÁĹģ
  yAxis: {
    show: !0,
    name: "",
    nameLocation: "end",
    type: "value",
    nameTextStyle: {
      color: "#666",
      fontSize: "12",
      fontFamily: "PF",
      align: "left",
      padding: [0, 0, 7, -24]
    },
    nameGap: 10,
    inverse: !1,
    boundaryGap: !1,
    splitNumber: 5,
    axisLine: {
      show: !0,
      lineStyle: {
        color: "#cacaca",
        width: 1,
        type: "solid"
      }
    },
    axisTick: {
      show: !1
    },
    splitLine: {
      show: !0,
      lineStyle: {
        color: "rgba(202,202,202,0.3)",
        width: 1,
        type: "solid"
      }
    },
    axisLabel: {
      show: !0,
      interval: 0,
      rotate: 0,
      color: "#666",
      fontSize: "12",
      fontFamily: "PF",
      formatter: "{value}",
      align: "right"
    }
  }
}), O = {
  level1: "#2CE493",
  // šľė
  // level1: '#00e400', // šľė
  level2: "#FFD878",
  // ŤČĮ
  // level2: '#edc24a', // ŤČĮ
  level3: "#FFA076",
  // ŤĹĽŚļ¶
  // level3: '#ff7e00', // ŤĹĽŚļ¶
  level4: "#FF5A5A",
  // šł≠Śļ¶
  // level4: '#ff0000', // šł≠Śļ¶
  level5: "#DB2C83",
  // ťáćŚļ¶
  // level5: '#990000', // ťáćŚļ¶
  level6: "#96002A",
  // šł•ťáć
  // level6: '#7e0000', // šł•ťáć
  level7: "#aaa"
  // śó†śēįśćģ
}, c = {
  level1: "šľė",
  level2: "ŤČĮ",
  level3: "ŤĹĽŚļ¶",
  level4: "šł≠Śļ¶",
  level5: "ťáćŚļ¶",
  level6: "šł•ťáć",
  level7: "-"
  // śó†śēįśćģ
};
export {
  h as AQI_COLOR,
  O as AQI_LEVEL_COLOR,
  c as AQI_LEVEL_NAME,
  n as AQI_TYPE,
  e as BORDER,
  t as BORDER_LEFT,
  a as BORDER_TOP,
  l as DEFAULT_OPTIONS,
  i as DEFAULT_OPTIONS_AXIS,
  r as DEFAULT_OPTIONS_COMMON,
  m as POLLUTION_CHARTS,
  s as POLLUTION_CHARTS_ALL,
  o as POLLUTION_HTML
};
