// @ths/design/es/utils/index.js
import { isDef as u, isNumeric as d } from "./validate.js";
import { POLLUTION_CHARTS_ALL as c } from "./constant.js";
const g = (e) => {
  const r = [
    "a",
    "b",
    "c",
    "d",
    "e",
    "f",
    "g",
    "h",
    "i",
    "j",
    "k",
    "l",
    "m",
    "n",
    "o",
    "p",
    "q",
    "r",
    "s",
    "t",
    "u",
    "v",
    "w",
    "x",
    "y",
    "z",
    "A",
    "B",
    "C",
    "D",
    "E",
    "F",
    "G",
    "H",
    "I",
    "J",
    "K",
    "L",
    "M",
    "N",
    "O",
    "P",
    "Q",
    "R",
    "S",
    "T",
    "U",
    "V",
    "W",
    "X",
    "Y",
    "Z",
    "0",
    "1",
    "2",
    "3",
    "4",
    "5",
    "6",
    "7",
    "8",
    "9"
  ];
  let t = "";
  const o = r.length;
  for (let n = 0; n < e; n++) {
    const i = Math.floor(Math.random() * o);
    t += r[i];
  }
  return t;
}, F = (e = {}) => {
  let r = "";
  return Object.keys(e).forEach((t) => {
    if (t && e) {
      const o = t.replace(/([A-Z])/g, (n, i) => `-${i.toLowerCase()}`);
      r += `${o}: ${e[t]};`;
    }
  }), r;
}, b = (e) => `${e}-${g(16)}`, A = (e, r) => {
  let t = "";
  return Object.keys(e).forEach((o) => {
    const n = o;
    if (n === "active")
      return;
    const i = r.split(",").map((l) => `${l}:${n}`).join(",");
    t += n === "default" ? `
            ${r} {
              ${F(e[n])}
            }
          ` : `
            ${i} {
              ${F(e[n])}
            }
          `;
  }), t;
};
function E(e) {
  return u(e) ? d(e) ? `${e}px` : String(e) : "";
}
function D(e, r = []) {
  const t = {};
  return Object.keys(e).forEach((o) => {
    [
      "data-id",
      "id",
      "cStyle",
      "class",
      "style",
      "onClick",
      "onDblclick",
      "onMousedown",
      "onMouseup",
      "onMousemove",
      "onMouseover",
      "onMouseout",
      "onMouseenter",
      "onMouseleave",
      "onAnimationstart",
      "onAnimationend",
      "onAnimationiteration",
      "onTransitionend"
    ].concat(r).reduce((l, s) => l || o === s, !1) || (t[o] = e[o]);
  }), t;
}
const C = (e) => {
  const r = [
    "width",
    "height",
    "top",
    "left",
    "rotate",
    "fontSize",
    "marginTop",
    "marginLeft",
    "marginBottom",
    "marginRight",
    "paddingTop",
    "paddingLeft",
    "paddingBottom",
    "paddingRight",
    "lineHeight",
    "textIndent",
    "borderWidth",
    "borderRadius",
    "borderTopWidth",
    "borderBottomWidth",
    "borderLeftWidth",
    "borderRightWidth",
    "borderTopLeftRadius",
    "borderTopRightRadius",
    "borderBottomLeftRadius",
    "borderBottomRightRadius"
  ], t = {};
  return e && Object.keys(e).forEach((o) => {
    if (r.includes(o) && typeof e[o] != "string") {
      let n = 0;
      o === "fontSize" && (n = 12), o === "width" || o === "height" ? t[o] = e[o] ? `${e[o]}px` : "" : t[o] = `${e[o] ? e[o] : n}px`;
    } else
      t[o] = e[o];
  }), t;
};
function a(e) {
  let r = e.toLowerCase();
  if (r && /^#([0-9a-fA-f]{3}|[0-9a-fA-f]{6})$/.test(r)) {
    if (r.length === 4) {
      let n = "#";
      for (let i = 1; i < 4; i += 1)
        n += r.slice(i, i + 1).concat(r.slice(i, i + 1));
      r = n;
    }
    const o = [];
    for (let n = 1; n < 7; n += 2)
      o.push(parseInt(`0x${r.slice(n, n + 2)}`));
    return o.join(",");
  }
  if (r.substring(0, 4) === "rgba") {
    const n = r.replace("rgba(", "").replace(")", "").split(",");
    return n.pop(), n.join(",");
  }
  return r.substring(0, 3) === "rgb" ? r.replace("rgb(", "").replace(")", "") : r;
}
const B = (e, r = 1) => {
  if (e.indexOf("rgb") !== -1)
    return e;
  let t = "255,255,255";
  const o = {
    aliceblue: "#F0F8FF",
    antiquewhite: "#FAEBD7",
    aqua: "#00FFFF",
    aquamarine: "#7FFFD4",
    azure: "#F0FFFF",
    beige: "#F5F5DC",
    bisque: "#FFE4C4",
    black: "#000000",
    blanchedalmond: "#FFEBCD",
    blue: "#0000FF",
    blueviolet: "#8A2BE2",
    brown: "#A52A2A",
    burlywood: "#DEB887",
    cadetblue: "#5F9EA0",
    chartreuse: "#7FFF00",
    chocolate: "#D2691E",
    coral: "#FF7F50",
    cornflowerblue: "#6495ED",
    cornsilk: "#FFF8DC",
    crimson: "#DC143C",
    cyan: "#00FFFF",
    darkblue: "#00008B",
    darkcyan: "#008B8B",
    darkgoldenrod: "#B8860B",
    darkgray: "#A9A9A9",
    darkgreen: "#006400",
    darkkhaki: "#BDB76B",
    darkmagenta: "#8B008B",
    darkolivegreen: "#556B2F",
    darkorange: "#FF8C00",
    darkorchid: "#9932CC",
    darkred: "#8B0000",
    darksalmon: "#E9967A",
    darkseagreen: "#8FBC8F",
    darkslateblue: "#483D8B",
    darkslategray: "#2F4F4F",
    darkturquoise: "#00CED1",
    darkviolet: "#9400D3",
    deeppink: "#FF1493",
    deepskyblue: "#00BFFF",
    dimgray: "#696969",
    dodgerblue: "#1E90FF",
    firebrick: "#B22222",
    floralwhite: "#FFFAF0",
    forestgreen: "#228B22",
    fuchsia: "#FF00FF",
    gainsboro: "#DCDCDC",
    ghostwhite: "#F8F8FF",
    gold: "#FFD700",
    goldenrod: "#DAA520",
    gray: "#808080",
    green: "#008000",
    greenyellow: "#ADFF2F",
    honeydew: "#F0FFF0",
    hotpink: "#FF69B4",
    indianred: "#CD5C5C",
    indigo: "#4B0082",
    ivory: "#FFFFF0",
    khaki: "#F0E68C",
    lavender: "#E6E6FA",
    lavenderblush: "#FFF0F5",
    lawngreen: "#7CFC00",
    lemonchiffon: "#FFFACD",
    lightblue: "#ADD8E6",
    lightcoral: "#F08080",
    lightcyan: "#E0FFFF",
    lightgoldenrodyellow: "#FAFAD2",
    lightgreen: "#90EE90",
    lightgray: "#D3D3D3",
    lightpink: "#FFB6C1",
    lightsalmon: "#FFA07A",
    lightseagreen: "#20B2AA",
    lightskyblue: "#87CEFA",
    lightslategray: "#778899",
    lightsteelblue: "#B0C4DE",
    lightyellow: "#FFFFE0",
    lime: "#00FF00",
    limegreen: "#32CD32",
    linen: "#FAF0E6",
    magenta: "#FF00FF",
    maroon: "#800000",
    mediumaquamarine: "#66CDAA",
    mediumblue: "#0000CD",
    mediumorchid: "#BA55D3",
    mediumpurple: "#9370DB",
    mediumseagreen: "#3CB371",
    mediumslateblue: "#7B68EE",
    mediumspringgreen: "#00FA9A",
    mediumturquoise: "#48D1CC",
    mediumvioletred: "#C71585",
    midnightblue: "#191970",
    mintcream: "#F5FFFA",
    mistyrose: "#FFE4E1",
    moccasin: "#FFE4B5",
    navajowhite: "#FFDEAD",
    navy: "#000080",
    oldlace: "#FDF5E6",
    olive: "#808000",
    olivedrab: "#6B8E23",
    orange: "#FFA500",
    orangered: "#FF4500",
    orchid: "#DA70D6",
    palegoldenrod: "#EEE8AA",
    palegreen: "#98FB98",
    paleturquoise: "#AFEEEE",
    palevioletred: "#DB7093",
    papayawhip: "#FFEFD5",
    peachpuff: "#FFDAB9",
    peru: "#CD853F",
    pink: "#FFC0CB",
    plum: "#DDA0DD",
    powderblue: "#B0E0E6",
    purple: "#800080",
    red: "#FF0000",
    rosybrown: "#BC8F8F",
    royalblue: "#4169E1",
    saddlebrown: "#8B4513",
    salmon: "#FA8072",
    sandybrown: "#FAA460",
    seagreen: "#2E8B57",
    seashell: "#FFF5EE",
    sienna: "#A0522D",
    silver: "#C0C0C0",
    skyblue: "#87CEEB",
    slateblue: "#6A5ACD",
    slategray: "#708090",
    snow: "#FFFAFA",
    springgreen: "#00FF7F",
    steelblue: "#4682B4",
    tan: "#D2B48C",
    teal: "#008080",
    thistle: "#D8BFD8",
    tomato: "#FF6347",
    turquoise: "#40E0D0",
    violet: "#EE82EE",
    wheat: "#F5DEB3",
    white: "#FFFFFF",
    whitesmoke: "#F5F5F5",
    yellow: "#FFFF00",
    yellowgreen: "#9ACD32"
  }, n = Object.keys(o).find((i) => i === e.toLowerCase());
  return t = a(n ? o[n] : e), `rgba(${t},${r})`;
}, w = (e, r = 1) => {
  if (!r)
    throw new Error(`error in params n: ${r}`);
  const t = [...e], o = Math.ceil(t.length / r) || 1, n = [];
  for (; ; ) {
    const i = t.splice(0, o);
    if (!i || i.length === 0)
      break;
    n.push(i);
  }
  return n;
}, k = (e) => {
  if (e == null || Number.isNaN(Number(e)) || Number(e) < 0 || e === "")
    return "level7";
  const r = Number(e);
  return r <= 50 && r >= 0 ? "level1" : r > 50 && r <= 100 ? "level2" : r <= 150 && r > 100 ? "level3" : r <= 200 && r > 150 ? "level4" : r <= 300 && r > 200 ? "level5" : "level6";
}, v = (e, r = "") => {
  const t = r.toUpperCase().replace(/[-.+_]/g, ""), o = e.filter((n) => n.name === t);
  return o.length > 0 ? o[0].html : r;
}, $ = (e) => {
  let r = e;
  return c.forEach((t) => {
    t.name.forEach((o) => {
      r = r.replace(o, t.html);
    });
  }), r;
}, y = (e, r = "-") => e == null || Number.isNaN(e) || e === "" ? r : e;
function m(e) {
  if (typeof e == "object" && e !== null) {
    const r = Array.isArray(e) ? [] : {};
    return Object.keys(e).forEach((t) => {
      r[t] = m(e[t]);
    }), r;
  }
  return e;
}
function h(e, r) {
  return Object.keys(r).forEach((t) => {
    e[t] = e[t] && e[t].toString() === "[object Object]" ? h(e[t], r[t]) : e[t] = r[t];
  }), e;
}
function L(e, r = 1e3) {
  let t;
  return (...n) => {
    t && clearTimeout(t), t = setTimeout(() => {
      e(...n), t = void 0;
    }, r);
  };
}
export {
  E as addUnit,
  a as colorRgb,
  m as deepCloneFun,
  h as deepMergeFun,
  D as filterAttrs,
  k as getAQILevel,
  F as getCssPropertyString,
  b as getDomRandomClass,
  v as getPollutionHtml,
  g as getRandomStr,
  C as getStyle,
  y as getValueOrDefaultValue,
  $ as replacePollutionHtml,
  B as rgb2rgba,
  A as setStyle,
  w as splitArray,
  L as useDebounce
};
