// @ths/design/es/components/calendar-base/src/index.vue2.js
import { defineComponent as W, computed as L, ref as V, getCurrentInstance as Y, reactive as C, onMounted as A, watch as D, toRefs as k } from "vue";
import I from "lodash/merge";
import p from "lodash/cloneDeep";
import { getDomRandomClass as O, setStyle as u } from "../../../utils/index.js";
const S = {
  id: "id",
  date: "date",
  name: "name",
  value: "value"
}, F = W({
  name: "TCalendarBase",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        },
        // 带日期、按钮的头部
        headerWrapper: {
          default: {}
        },
        // 头部文字
        headerDate: {
          default: {}
        },
        // 切换按钮
        button: {
          default: {},
          hover: {},
          focus: {}
        },
        // 表格
        tableWrapper: {
          default: {}
        },
        // 日历表头
        tableHeader: {
          default: {}
        },
        // 当月单元格
        currentDateWrapper: {
          default: {}
        },
        // 其他单元格
        otherDateWrapper: {
          default: {}
        },
        // 当月单元格日期数字
        currentDate: {
          default: {}
        },
        // 其他月单元格日期数字
        otherDate: {
          default: {}
        }
      })
    },
    // 选中月份值
    dateValue: {
      type: String,
      default: () => ""
    },
    // 数据list
    dataList: {
      type: Array,
      default: () => []
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => p(S)
    },
    // 不同状态下的样式列表
    stateList: {
      type: Array,
      default: () => []
    }
  },
  // 事件名
  emits: ["change-click"],
  setup(r, { emit: M }) {
    const t = L(() => r.cStyle), l = O("calendar-base"), f = V(), { proxy: i } = Y(), v = ["日", "一", "二", "三", "四", "五", "六"], s = (a) => {
      if (!a || a.split("-").length < 2)
        return "";
      const n = a.split("-")[0].slice(
        0,
        4
        /* yearLength */
      ), e = a.split("-")[1].slice(
        0,
        2
        /* dateLength */
      );
      return /^\d{4}$/.test(n) && /^\d{1,2}$/.test(e) && Number(e) >= 1 && Number(e) <= 12 ? `${n}-${e}` : "";
    }, o = (a, n) => {
      const e = {
        Year: 0,
        TYear: "0",
        Month: 0,
        TMonth: "0",
        Day: 0,
        TDay: "0",
        Hour: 0,
        THour: "0",
        hour: 0,
        Thour: "0",
        Minute: 0,
        TMinute: "0",
        Second: 0,
        TSecond: "0",
        Millisecond: 0
      };
      return e.Year = a.getFullYear(), e.TYear = String(e.Year).substr(2), e.Month = a.getMonth() + 1, e.TMonth = e.Month < 10 ? `0${e.Month}` : String(e.Month), e.Day = a.getDate(), e.TDay = e.Day < 10 ? `0${e.Day}` : String(e.Day), e.Hour = a.getHours(), e.THour = e.Hour < 10 ? `0${e.Hour}` : String(e.Hour), e.hour = e.Hour < 13 ? e.Hour : e.Hour - 12, e.Thour = e.hour < 10 ? `0${e.hour}` : String(e.hour), e.Minute = a.getMinutes(), e.TMinute = e.Minute < 10 ? `0${e.Minute}` : String(e.Minute), e.Second = a.getSeconds(), e.TSecond = e.Second < 10 ? `0${e.Second}` : String(e.Second), e.Millisecond = a.getMilliseconds(), n.replace(/yyyy/gi, String(e.Year)).replace(/yyy/gi, String(e.Year)).replace(/yy/gi, e.TYear).replace(/y/gi, e.TYear).replace(/MM/g, e.TMonth).replace(/M/g, String(e.Month)).replace(/dd/gi, e.TDay).replace(/d/gi, String(e.Day)).replace(/HH/g, e.THour).replace(/H/g, String(e.Hour)).replace(/hh/g, e.Thour).replace(/h/g, String(e.hour)).replace(/mm/g, e.TMinute).replace(/m/g, String(e.Minute)).replace(/ss/gi, e.TSecond).replace(/s/gi, String(e.Second)).replace(/fff/gi, String(e.Millisecond));
    }, c = C({
      curDateValue: new Date(s(r.dateValue) || /* @__PURE__ */ new Date()),
      // 当前月份值
      curDate: o(
        new Date(s(r.dateValue) || /* @__PURE__ */ new Date()),
        "yyyy年M月"
        /* monthFormat */
      ),
      // 头部日期显示
      curData: []
      // 当月数据
    }), b = () => {
      i == null || i.$el.querySelector("div.el-calendar__header").remove();
      const a = i == null ? void 0 : i.$el.querySelectorAll("table.el-calendar-table thead th");
      a.forEach((n, e) => {
        a[e].innerHTML = v[e];
      });
    }, $ = (a) => {
      f.value.selectDate(a), c.curDate = o(
        new Date(c.curDateValue),
        "yyyy年M月"
        /* monthFormat */
      );
      const n = o(
        new Date(c.curDateValue),
        "yyyy-MM"
        /* monthBreakFormat */
      );
      M("change-click", n);
    }, g = () => {
      const a = I(p(S), r.dataAlias), {
        id: n,
        date: e,
        name: h,
        value: d
      } = a;
      c.curData = [], r.dataList && r.dataList.length && (c.curData = r.dataList.map((y) => ({
        id: y[n],
        name: y[h] || "-",
        date: o(
          new Date(y[e]),
          "yyyy-MM-dd"
          /* dayFormat */
        ),
        // 格式化日期，格式为 yyyy-MM-dd
        value: y[d] || ""
      })));
    }, T = (a) => {
      const n = a["background-image"] || a.backgroundImage;
      return n ? {
        ...a,
        "background-image": n.indexOf("url") === -1 ? `url(${n})` : n
      } : a;
    }, H = (a) => {
      var h;
      const n = r.stateList.length, e = c.curData.find((d) => d.date === a.day);
      for (let d = 0; d < n; d++)
        if ((e == null ? void 0 : e.id) === ((h = r.stateList[d]) == null ? void 0 : h.id))
          return T(r.stateList[d].style);
      return {};
    }, w = () => {
      let a = "";
      return t.value.wrapper && (a += u(t.value.wrapper, `.${l}`)), t.value.headerWrapper && (a += u(t.value.headerWrapper, `.${l} .calendar-header`)), t.value.headerDate && (a += u(t.value.headerDate, `.${l} .calendar-header .header-date`)), t.value.button && (a += u(t.value.button, `.${l} .calendar-header .el-button`)), t.value.tableWrapper && (a += u(t.value.tableWrapper, `.${l} .el-calendar .el-calendar-table`)), t.value.tableHeader && (a += u(t.value.tableHeader, `.${l} .el-calendar .el-calendar-table thead th`), a += u(t.value.tableHeader, `.${l} .el-calendar .el-calendar-table thead th:first-child`), a += u(t.value.tableHeader, `.${l} .el-calendar .el-calendar-table thead th:last-child`)), t.value.currentDateWrapper && (a += u(t.value.currentDateWrapper, `.${l} .el-calendar .el-calendar-table td.current`)), t.value.otherDateWrapper && (a += u(t.value.otherDateWrapper, `.${l} .el-calendar .el-calendar-table td.prev`), a += u(t.value.otherDateWrapper, `.${l} .el-calendar .el-calendar-table td.next`)), t.value.currentDate && (a += u(t.value.currentDate, `.${l} .el-calendar .el-calendar-table td.current .calendar-day`)), t.value.otherDate && (a += u(t.value.otherDate, `.${l} .el-calendar .el-calendar-table td.prev .calendar-day`), a += u(t.value.otherDate, `.${l} .el-calendar .el-calendar-table td.next .calendar-day`)), a;
    }, m = () => {
      const a = document.querySelector(`.${l} > .style-container`);
      a && (a.innerHTML = `<style>${w()}</style>`);
    };
    return A(() => {
      b(), g(), m();
    }), D(
      () => [r.dataList, r.dataAlias],
      () => {
        g();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), D(
      () => [r.cStyle],
      () => {
        m();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), D(
      () => [r.dateValue],
      () => {
        c.curDate = o(
          new Date(s(r.dateValue) || /* @__PURE__ */ new Date()),
          "yyyy年M月"
          /* monthFormat */
        ), c.curDateValue = new Date(s(r.dateValue) || /* @__PURE__ */ new Date()), g();
      },
      {
        immediate: !0
      }
    ), {
      curStyle: t,
      domOnlyClass: l,
      ...k(c),
      calendar: f,
      selectDate: $,
      getCellStyle: H
    };
  }
});
export {
  F as default
};
