// @ths/design/es/components/calendar-base/src/index.vue.js
import m from "./index.vue2.js";
import { resolveComponent as a, openBlock as p, createElementBlock as u, normalizeClass as f, createElementVNode as n, createVNode as t, withCtx as r, toDisplayString as d, normalizeStyle as _ } from "vue";
import "./index.vue3.js";
import C from "../../../_virtual/_plugin-vue_export-helper.js";
const v = { class: "calendar-header" }, y = { class: "header-date" };
function V(l, e, g, D, $, b) {
  const s = a("el-button"), i = a("el-button-group"), c = a("el-calendar");
  return p(), u("div", {
    class: f(["t-calendar-base", l.domOnlyClass])
  }, [
    n("div", v, [
      t(i, null, {
        default: r(() => [
          t(s, {
            class: "arrow-left",
            size: "mini",
            icon: "el-icon-arrow-left",
            onClick: e[0] || (e[0] = (o) => l.selectDate("prev-month"))
          })
        ]),
        _: 1
      }),
      n("span", y, d(l.curDate || "-"), 1),
      t(i, null, {
        default: r(() => [
          t(s, {
            class: "arrow-right",
            size: "mini",
            icon: "el-icon-arrow-right",
            onClick: e[1] || (e[1] = (o) => l.selectDate("next-month"))
          })
        ]),
        _: 1
      })
    ]),
    t(c, {
      ref: "calendar",
      modelValue: l.curDateValue,
      "onUpdate:modelValue": e[2] || (e[2] = (o) => l.curDateValue = o)
    }, {
      dateCell: r(({ data: o }) => [
        n("div", {
          class: "calendar-day",
          style: _(l.getCellStyle(o))
        }, d(o.day.split("-").slice(2).join("").replace(/^0/, "")), 5)
      ]),
      _: 1
    }, 8, ["modelValue"]),
    e[3] || (e[3] = n("div", { class: "style-container" }, null, -1))
  ], 2);
}
const S = /* @__PURE__ */ C(m, [["render", V]]);
export {
  S as default
};
