// @ths/design/es/components/gradient-text/src/index.vue2.js
import { defineComponent as p, computed as m, onMounted as S, watch as C } from "vue";
import { filterAttrs as x, getDomRandomClass as g, setStyle as $ } from "../../../utils/index.js";
import "../../text/index.js";
import v from "../../text/src/index.vue.js";
const O = p({
  name: "TGradientText",
  components: {
    TText: v
  },
  props: {
    // 文字
    label: {
      type: String,
      default: ""
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        }
      })
    },
    // 渐变颜色
    colorStyle: {
      type: Object,
      default: () => ({})
    }
  },
  setup(l) {
    const r = m(() => l.cStyle), e = m(() => l.colorStyle), o = g("gradient-text"), y = () => {
      let t = "";
      return r.value.wrapper && (t += $(r.value.wrapper, `.${o}`)), t;
    }, f = () => {
      var a, c, i, s, u, d;
      let t = "";
      return (a = e.value) != null && a.startColor && (t += `.${o} .t-text {
          --t-gradient-text-start-color: ${(c = e.value) == null ? void 0 : c.startColor};
        }`), (i = e.value) != null && i.endColor && (t += `.${o} .t-text {
          --t-gradient-text-end-color: ${(s = e.value) == null ? void 0 : s.endColor};
        }`), (u = e.value) != null && u.angle && (t += `.${o} .t-text {
          --t-gradient-text-angle: ${(d = e.value) == null ? void 0 : d.angle}deg;
        }`), t;
    }, n = () => {
      const t = document.querySelector(`.${o} .style-container`);
      t && (t.innerHTML = `<style>${y()} ${f()}</style>`);
    };
    return S(() => {
      n();
    }), C(() => [l.cStyle, l.colorStyle], () => {
      n();
    }, {
      deep: !0,
      immediate: !0
    }), {
      curStyle: r,
      filterAttrs: x,
      domOnlyClass: o
    };
  }
});
export {
  O as default
};
