// @ths/design/es/components/search/src/index.vue.js
import I from "./index.vue2.js";
import { resolveComponent as w, openBlock as y, createElementBlock as $, normalizeClass as k, normalizeStyle as z, createBlock as F, mergeProps as L, withModifiers as C, withCtx as B, createVNode as N, createElementVNode as t, createCommentVNode as b } from "vue";
import "./index.vue3.js";
import E from "../../../_virtual/_plugin-vue_export-helper.js";
function M(e, r, R, T, A, O) {
  var n, l, o, a, i, p, u, s, m, d, f, g, h, c;
  const S = w("el-icon"), V = w("el-input");
  return y(), $("div", {
    class: k(["t-search", e.domOnlyClass]),
    style: z({
      width: (o = (l = (n = e.curStyle) == null ? void 0 : n.wrapper) == null ? void 0 : l.default) == null ? void 0 : o.width,
      height: (p = (i = (a = e.curStyle) == null ? void 0 : a.wrapper) == null ? void 0 : i.default) == null ? void 0 : p.height,
      marginTop: (s = (u = e.curStyle.wrapper) == null ? void 0 : u.default) == null ? void 0 : s.marginTop,
      marginLeft: (d = (m = e.curStyle.wrapper) == null ? void 0 : m.default) == null ? void 0 : d.marginLeft,
      marginRight: (g = (f = e.curStyle.wrapper) == null ? void 0 : f.default) == null ? void 0 : g.marginRight,
      marginBottom: (c = (h = e.curStyle.wrapper) == null ? void 0 : h.default) == null ? void 0 : c.marginBottom
    })
  }, [
    e.batchSearch ? b("", !0) : (y(), F(V, L({
      key: 0,
      modelValue: e.curInputVal,
      "onUpdate:modelValue": r[0] || (r[0] = (v) => e.curInputVal = v)
    }, e.filterAttrs(e.$attrs), {
      clearable: "",
      title: e.curInputVal,
      onInput: C(e.handleInputChange, ["self"]),
      onFocus: e.handleFocus,
      onBlur: e.handleBlur,
      onChange: C(e.handleChange, ["self"])
    }), {
      prefix: B(() => [
        N(S, { class: "el-input__icon" }, {
          default: B(() => r[1] || (r[1] = [
            t("svg", {
              xmlns: "http://www.w3.org/2000/svg",
              viewBox: "0 0 1024 1024"
            }, [
              t("path", {
                fill: "currentColor",
                d: `m795.904 750.72 124.992 124.928a32 32 0 0 1-45.248
          45.248L750.656 795.904a416 416 0 1 1 45.248-45.248zM480 832a352 352 0 1 0 0-704 352 352 0 0 0 0 704z`
              })
            ], -1)
          ])),
          _: 1,
          __: [1]
        })
      ]),
      _: 1
    }, 16, ["modelValue", "title", "onInput", "onFocus", "onBlur", "onChange"])),
    r[2] || (r[2] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const D = /* @__PURE__ */ E(I, [["render", M]]);
export {
  D as default
};
