// @ths/design/es/components/search/src/index.vue2.js
import { defineComponent as x, computed as V, ref as i, reactive as g, onMounted as C, watch as p, toRefs as b } from "vue";
import { filterAttrs as w, getDomRandomClass as $, setStyle as u } from "../../../utils/index.js";
const T = x({
  name: "TSearch",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {},
          focus: {},
          hover: {}
        },
        prefix: {
          default: {},
          hover: {}
        },
        suffix: {
          default: {},
          hover: {}
        },
        placeholder: {
          default: {}
        }
      })
    },
    // 输入值
    inputValue: {
      type: String,
      default: ""
    },
    batchSearch: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["input", "change"],
  setup(a, { emit: s }) {
    const t = V(() => a.cStyle), l = $("search"), c = i(a.inputValue), n = g({
      searchKey: null,
      isFocus: !1,
      keysList: [],
      batchSearchVal: "",
      showBatchSearchVal: ""
    }), o = i(null), f = () => {
      let e = "";
      return t.value.wrapper && (e += u(t.value.wrapper, `.${l} .el-input .el-input__inner`)), t.value.prefix && (e += u(t.value.prefix, `.${l} .el-input .el-input__prefix .el-input__icon`)), t.value.suffix && (e += u(t.value.suffix, `.${l} .el-input .el-input__suffix .el-input__icon`)), t.value.placeholder && (e += u(t.value.placeholder, `.${l} .el-input .el-input__inner::placeholder`)), e;
    }, r = () => {
      const e = document.querySelector(`.${l} .style-container`);
      e && (e.innerHTML = `<style>${f()}</style>`);
    }, h = (e) => {
      c.value = e, s("input", e);
    }, d = (e) => {
      s("change", e);
    }, m = () => {
      n.isFocus = !0;
    }, y = () => {
      n.isFocus = !1;
    }, S = () => {
      o.value && o.value.focus();
    }, v = () => {
      n.batchSearchVal = "";
    }, _ = () => {
      const e = n.batchSearchVal.split(/[\n]/g);
      n.showBatchSearchVal = e.toString();
    };
    return C(() => {
      r();
    }), p(() => a.cStyle, () => {
      r();
    }, { deep: !0 }), p(() => a.inputValue, (e) => {
      c.value = e;
    }), {
      ...b(n),
      curStyle: t,
      domOnlyClass: l,
      curInputVal: c,
      inputTextarea: o,
      filterAttrs: w,
      handleInputChange: h,
      handleChange: d,
      handleFocus: m,
      handleBlur: y,
      batchSearchFocus: S,
      clear: v,
      determine: _
    };
  }
});
export {
  T as default
};
