// @ths/design/es/components/steps/src/index.vue2.js
import { defineComponent as F, computed as r, watch as L, onMounted as T } from "vue";
import { filterAttrs as $, getDomRandomClass as D, getCssPropertyString as V } from "../../../utils/index.js";
const O = F({
  name: "TSteps",
  props: {
    data: {
      type: Array,
      default: () => []
    },
    // 当前激活步骤
    active: {
      type: Number,
      default: 1
    },
    // 当前进行中是否出错
    errorState: {
      type: Boolean,
      default: !1
    },
    // 设置是否有序号
    serial: {
      type: Boolean,
      default: !1
    },
    // 设置步骤条方向
    direcition: {
      type: String,
      default: "horizontal"
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        // 步骤所属框样式
        wrapper: {
          default: {}
        },
        // 步骤信息
        itemWrapper: {
          default: {}
        },
        // 标题样式
        title: {
          default: {},
          active: {},
          finish: {}
        },
        // 描述样式
        desc: {
          default: {},
          active: {},
          finish: {}
        },
        // 线的样式
        line: {
          default: {},
          active: {},
          finish: {}
        },
        // 图标的样式
        icon: {
          default: {},
          active: {},
          finish: {}
        }
      })
    }
  },
  setup(t) {
    const c = r(() => t.cStyle), s = D("steps"), u = () => {
      let e = "";
      return c.value.wrapper && (e += `
          .${s} {
            ${V(c.value.wrapper.default)}
          }
        `), e;
    }, o = () => {
      const e = document.querySelector(
        `.${s} .style-container`
      );
      e && (e.innerHTML = `<style>${u()}</style>`);
    };
    L(
      () => t.cStyle,
      () => {
        o();
      }
    ), T(() => {
      o();
    });
    const f = r(() => t.data), i = r(() => t.active - 1), S = (e) => i.value > e, d = (e) => i.value > e && t.serial === !1, v = (e) => i.value < e, y = (e) => i.value === e && t.errorState === !1, m = (e) => i.value === e && t.errorState && !t.serial, h = (e) => i.value > e && t.serial, N = (e) => i.value < e && t.serial, C = (e) => i.value === e && t.serial, P = (e) => i.value > e && t.serial && t.direcition === "vertical", g = (e) => i.value < e && t.serial && t.direcition === "vertical", w = (e) => i.value === e && t.serial && t.direcition === "vertical", l = (e, n) => {
      const a = t.cStyle[e];
      return a != null && a.default && i.value < n ? a.default : a != null && a.active && i.value === n ? a.active : a != null && a.finish && i.value > n ? a.finish : null;
    };
    return {
      curStyle: c,
      domOnlyClass: s,
      filterAttrs: $,
      stepsList: f,
      activeKey: i,
      isFinish: S,
      isSerialFinish: d,
      isNoStart: v,
      isProcess: y,
      isProcessError: m,
      isFinishNoSerial: h,
      isProcessNoSerial: C,
      isNoStartNoSerial: N,
      isVerticalFinishNoSerial: P,
      isVerticalNoStartNoSerial: g,
      isVerticalProcessNoSerial: w,
      customDescStyle: (e) => l("desc", e),
      customTitlecStyle: (e) => l("title", e),
      customLinecStyle: (e) => l("line", e),
      customIconcStyle: (e) => l("icon", e)
    };
  }
});
export {
  O as default
};
