// @ths/design/es/components/target-assessment/src/index.vue.js
import T from "./index.vue2.js";
import { openBlock as n, createElementBlock as o, normalizeStyle as l, createElementVNode as r, Fragment as j, renderList as q, toDisplayString as K } from "vue";
import "./index.vue3.js";
import G from "../../../_virtual/_plugin-vue_export-helper.js";
function H(e, s, J, M, P, Q) {
  var d, g, c, u, i, p;
  return n(), o("div", {
    class: "t-target-assessment",
    style: l({ ...(g = (d = e.cStyle) == null ? void 0 : d.wrapper) == null ? void 0 : g.default })
  }, [
    r("div", {
      ref: "chartId",
      class: "chart",
      style: l({ ...(u = (c = e.cStyle) == null ? void 0 : c.chartWrapper) == null ? void 0 : u.default })
    }, null, 4),
    r("div", {
      class: "legend",
      style: l({ ...(p = (i = e.cStyle) == null ? void 0 : i.legendWrapper) == null ? void 0 : p.default })
    }, [
      (n(!0), o(j, null, q(e.chartSetting.dataSetting, (t, a) => {
        var y, f, S, m, h, v, k, I, $, B, C, N, O, V, b, E, L, W, w, z, A, D, F;
        return n(), o("div", {
          key: a,
          class: "legend-item",
          style: l({ ...(f = (y = e.cStyle) == null ? void 0 : y.legendItem) == null ? void 0 : f.default })
        }, [
          r("div", {
            class: "icon",
            style: l({
              backgroundColor: e.getLegendBgColor(((S = t.style) == null ? void 0 : S.color) || ((v = (h = (m = e.chartSetting) == null ? void 0 : m.defaultSetting) == null ? void 0 : h.color) == null ? void 0 : v[a]) || e.defaultOptions.color[a]),
              ...((k = t.style) == null ? void 0 : k.legendIcon) || (($ = (I = e.chartSetting) == null ? void 0 : I.defaultSetting) == null ? void 0 : $.legendIcon)
            })
          }, [
            r("div", {
              class: "icon-inner",
              style: l({
                backgroundColor: ((B = t.style) == null ? void 0 : B.color) || ((O = (N = (C = e.chartSetting) == null ? void 0 : C.defaultSetting) == null ? void 0 : N.color) == null ? void 0 : O[a]) || e.defaultOptions.color[a]
              })
            }, null, 4)
          ], 4),
          r("span", {
            class: "name",
            style: l({
              ...((V = t.style) == null ? void 0 : V.legendName) || ((E = (b = e.chartSetting) == null ? void 0 : b.defaultSetting) == null ? void 0 : E.legendName)
            })
          }, K(t.name), 5),
          r("span", {
            class: "value",
            style: l({
              color: ((L = t.style) == null ? void 0 : L.color) || ((z = (w = (W = e.chartSetting) == null ? void 0 : W.defaultSetting) == null ? void 0 : w.color) == null ? void 0 : z[a]) || e.defaultOptions.color[a],
              ...((A = t.style) == null ? void 0 : A.legendValue) || ((F = (D = e.chartSetting) == null ? void 0 : D.defaultSetting) == null ? void 0 : F.legendValue)
            })
          }, K(e.data[t.dataKey]), 5)
        ], 4);
      }), 128))
    ], 4),
    s[0] || (s[0] = r("div", { class: "style-container" }, null, -1))
  ], 4);
}
const Z = /* @__PURE__ */ G(T, [["render", H]]);
export {
  Z as default
};
