// @ths/design/es/components/target-assessment/src/index.vue2.js
import { defineComponent as D, ref as E, reactive as I, onMounted as M, onUnmounted as $, watch as k, toRefs as K, nextTick as U } from "vue";
import * as F from "echarts";
import G from "lodash/merge";
import P from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as R } from "../../../utils/constant.js";
import { rgb2rgba as N } from "../../../utils/index.js";
const n = {
  color: R.color,
  tooltip: {
    show: !1
  },
  title: {
    show: !1
  },
  legend: {
    show: !1
  },
  xAxis: {
    show: !1
  },
  yAxis: {
    show: !1
  },
  polar: {
    center: ["50%", "50%"]
  },
  angleAxis: {
    max: 100,
    show: !1
  },
  grid: {
    top: 0,
    bottom: 0,
    left: 0,
    right: 0
  },
  radiusAxis: {
    type: "category",
    show: !0,
    axisLabel: {
      show: !1
    },
    axisLine: {
      show: !1
    },
    axisTick: {
      show: !1
    }
  },
  graphic: {
    elements: [
      {
        type: "image",
        z: 3,
        style: {
          image: "",
          width: 178,
          height: 178
        },
        left: "center",
        top: "center"
      }
    ]
  }
}, i = {
  space: "40%",
  // 圆环间隙
  itemWidth: 10,
  // 圆环宽度
  radius: "156%",
  // 圆环半径
  itemBgColor: "rgba(66, 66, 66, .3)"
  // 圆环背景色
}, X = D({
  name: "TTargetAssessment",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        chartWrapper: {
          default: {}
        },
        legendWrapper: {
          default: {}
        },
        legendItem: {
          default: {}
        }
      })
    },
    // 数据
    data: {
      type: Object,
      default: () => {
      }
    },
    // 图表设置，包含系列样式、别名设置
    chartSetting: {
      type: Object,
      default: () => ({
        defaultSetting: P(i),
        dataSetting: []
      })
    }
  },
  setup(t) {
    const o = E(null);
    let l = null;
    const d = I({
      cStyle: {}
    }), _ = (a) => {
      if (a.indexOf("rgba") !== -1) {
        const e = a.replace(/rgba?\(/, "").replace(/\)/, "").replace(/[\s+]/g, "").split(",");
        return Number(e[3]) === 1 ? `rgba(${e[0]}, ${e[1]}, ${e[2]}, 0.2)` : a;
      }
      return N(a, 0.2);
    }, g = () => {
      var e, c, u, f, m, h;
      const a = G(
        {
          polar: {
            radius: ((c = (e = t.chartSetting) == null ? void 0 : e.defaultSetting) == null ? void 0 : c.radius) || i.radius
          },
          series: ((f = (u = t.chartSetting) == null ? void 0 : u.dataSetting) == null ? void 0 : f.map((r, p) => {
            var S, y, w, b, x, C, O, v, W, A, B, L, T, z;
            const j = (S = t.data) != null && S[r.dataKey] ? (y = t.data) == null ? void 0 : y[r.dataKey].split("%")[0] : 0;
            return {
              name: r.name || "--",
              type: "bar",
              roundCap: !0,
              barWidth: ((w = r.style) == null ? void 0 : w.itemWidth) || ((x = (b = t.chartSetting) == null ? void 0 : b.defaultSetting) == null ? void 0 : x.itemWidth) || i.itemWidth,
              showBackground: !0,
              data: [j],
              coordinateSystem: "polar",
              barGap: ((O = (C = t.chartSetting) == null ? void 0 : C.defaultSetting) == null ? void 0 : O.space) || i.space,
              backgroundStyle: {
                color: ((v = r.style) == null ? void 0 : v.itemBgColor) || ((A = (W = t.chartSetting) == null ? void 0 : W.defaultSetting) == null ? void 0 : A.itemBgColor) || i.itemBgColor
              },
              itemStyle: {
                normal: {
                  color: ((B = r.style) == null ? void 0 : B.color) || ((z = (T = (L = t.chartSetting) == null ? void 0 : L.defaultSetting) == null ? void 0 : T.color) == null ? void 0 : z[p]) || n.color[p]
                }
              }
            };
          })) || [],
          graphic: {
            elements: [
              {
                style: ((h = (m = t.chartSetting) == null ? void 0 : m.defaultSetting) == null ? void 0 : h.graphic) || n.graphic.elements[0].style
              }
            ]
          }
        },
        n
      );
      l.clear(), l.setOption(a);
    }, s = () => {
      U(() => {
        l.resize();
      });
    };
    return M(() => {
      o.value && (l = F.init(o.value), g()), window.addEventListener("resize", s);
    }), $(() => {
      window.removeEventListener("resize", s);
    }), k(
      [
        () => t.chartSetting,
        () => t.data
      ],
      () => {
        g();
      },
      {
        deep: !0
      }
    ), k(
      () => t.cStyle,
      () => {
        s(), d.cStyle = t.cStyle;
      },
      {
        immediate: !0,
        deep: !0
      }
    ), {
      defaultOptions: n,
      ...K(d),
      chartId: o,
      rgb2rgba: N,
      getLegendBgColor: _
    };
  }
});
export {
  X as default
};
