// @ths/design/es/components/text-unit/src/index.vue2.js
import { defineComponent as i, computed as r, onMounted as s, watch as p } from "vue";
import { getDomRandomClass as f, setStyle as l } from "../../../utils/index.js";
const m = i({
  name: "TTextUnit",
  props: {
    // 水平排列方式
    justify: {
      type: String,
      default: ""
    },
    // 垂直排列方式
    align: {
      type: String,
      default: ""
    },
    // 文本内容
    content: {
      type: String,
      default: ""
    },
    // 文本单位
    unit: {
      type: String,
      default: ""
    },
    // 超长...
    showOverflowTooltip: {
      type: Boolean,
      default: !0
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        text: {
          default: {}
        },
        unit: {
          default: {}
        }
      })
    }
  },
  setup(o, { emit: y }) {
    const t = r(() => o.cStyle), n = f("text-unit"), a = () => {
      let e = "";
      return t.value.wrapper && (e += l(t.value.wrapper, `.${n}`)), t.value.text && (e += l(t.value.text, `.${n} .t-text-unit__content`)), t.value.unit && (e += l(t.value.unit, `.${n} .t-text-unit__unit`)), e;
    }, u = () => {
      const e = document.querySelector(`.${n} .style-container`);
      e && (e.innerHTML = `<style>${a()}</style>`);
    };
    return s(() => {
      u();
    }), p(() => o.cStyle, () => {
      u();
    }, { deep: !0 }), {
      curStyle: t,
      domOnlyClass: n
    };
  }
});
export {
  m as default
};
