// @ths/design/es/components/chart-line-base/src/index.vue2.js
import { defineComponent as I, ref as B, computed as h, watch as m, nextTick as S, onMounted as O } from "vue";
import * as A from "echarts";
import { rgb2rgba as x } from "../../../utils/index.js";
var w = /* @__PURE__ */ ((e) => (e.LIGHT = "light", e.DARK = "dark", e))(w || {});
const T = [
  "#43d401",
  "#ffea00",
  "#00f6ff",
  "#be4cff",
  "#0167ff",
  "#fd7800",
  "#5c25d2",
  "#ffb540"
], V = {
  light: {
    axisLine: "rgba(202, 202, 202, .8)",
    lineColor: "rgba(202, 202, 202, .3)",
    fontColor: "#666",
    tooltipBackgroundColor: "rgba(255, 255, 255, 1)",
    tooltipFontColor: "#666"
  },
  dark: {
    axisLine: "rgba(202, 202, 202, .3)",
    lineColor: "rgba(202, 202, 202, .1)",
    fontColor: "#fff",
    tooltipBackgroundColor: "rgba(34, 68, 99, 1)",
    tooltipFontColor: "#fff"
  }
}, C = {
  wrapper: {
    default: {}
  }
}, j = I({
  name: "TChartLineBase",
  props: {
    // ж•°жҚ®
    data: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    // жЁӘеқҗж Үж•°жҚ®
    dataX: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    // жҜҸжқЎжҠҳзәҝе‘ҪеҗҚ
    names: {
      type: Array,
      default: () => [],
      required: !0,
      validator: (e) => !!e
    },
    // ж•ҙдёӘеӣҫеҪўе®ҪеәҰ
    width: {
      type: String,
      default: "100%",
      required: !1
    },
    // ж•ҙдёӘеӣҫеҪўй«ҳеәҰ
    height: {
      type: String,
      default: "300px",
      required: !1
    },
    // ж•ҙдёӘеӣҫеҪўе‘ҪеҗҚ
    title: {
      type: String,
      default: "",
      required: !1
    },
    // жҳҜеҗҰе№іж»‘жӣІзәҝжҳҫзӨә
    smooth: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // еӣҫеғҸйўңиүІ
    colors: {
      type: Array,
      default: () => T,
      required: !1,
      validator: (e) => !e.some((s) => !/#[0-9a-fA-F]{6}/.test(s.toString()))
    },
    // жҜҸжқЎжҠҳзәҝзҡ„еҚ•дҪҚ
    units: {
      type: Array,
      default: () => [],
      required: !1
    },
    // ж Үзәҝ
    markLine: {
      type: Array,
      default: () => [],
      required: !1
    },
    // жҠҳзәҝдёҠжҳҜеҗҰжҳҫзӨәж•°жҚ®
    showValue: {
      type: Array,
      default: () => [],
      required: !1
    },
    // еҢәеҹҹй”Ғе®ҡ
    zoomLock: {
      type: Boolean,
      default: !0,
      required: !1
    },
    // жЁӘеҗ‘жҳҫзӨәеҢәеҹҹ
    scope: {
      type: Array,
      default: () => ["0%", "100%"],
      required: !1,
      validator: (e) => e.length === 2 ? /%/.test(e[0].toString()) ? /%/.test(e[1].toString()) : !/%/.test(e[1].toString()) : !1
    },
    // ж•°жҚ®жһҒеҖјеҢәй—ҙ
    range: {
      type: Array,
      default: () => [0, void 0],
      required: !1,
      validator: (e) => !(e.length > 2)
    },
    // жҳҜеҗҰжҳҫзӨәжҠҳзәҝдёӢж–№иғҢжҷҜ
    showAreas: {
      type: Array,
      default: () => [],
      required: !1
    },
    // yиҪҙеҗҜз”Ё
    showYaxis: {
      type: Array,
      default: () => [!0, !1],
      required: !1
    },
    // з»ҷзі»еҲ—жҢҮе®ҡyиҪҙ
    yIndexs: {
      type: Array,
      default: () => [],
      required: !1
    },
    // echartsеҺҹз”ҹй…ҚзҪ®
    options: {
      type: Object,
      default: () => ({
        title: {},
        grid: {},
        xAxis: {},
        yAxis: {},
        tooltip: {},
        color: null,
        series: {}
      }),
      required: !1
    },
    // дё»йўҳйЈҺж јж·ұжө…
    themeStyle: {
      type: String,
      default: "light",
      validator: (e) => e.toUpperCase() in w
    },
    // жҳҜеҗҰжҳҫзӨәеӣҫдҫӢ
    showLegend: {
      type: Boolean,
      default: !0
    },
    legendIcon: {
      type: String,
      default: ""
    },
    // й»ҳи®ӨеӣҫдҫӢжҺ’еҲ—ж–№ејҸ
    legendStyle: {
      type: Object,
      default: () => ({
        position: "top",
        align: "center",
        fontSize: 12,
        iconSize: 12
      }),
      required: !1,
      validator: (e) => !!e
    },
    // иҮӘе®ҡд№үж ·ејҸ
    cStyle: {
      type: Object,
      default: () => C
    }
  },
  emits: ["col-click"],
  setup(e, s) {
    const c = B(null), L = h(() => ({
      width: e.width,
      height: e.height
    })), b = h(() => ({
      ...C,
      ...e.cStyle
    })), k = () => /%/.test(e.scope[0].toString()) ? {
      start: Number(e.scope[0].toString().replace(/%/, "")),
      end: Number(e.scope[1].toString().replace(/%/, ""))
    } : {
      startValue: e.scope[0],
      endValue: e.scope[1]
    }, t = () => V[e.themeStyle], q = (i) => {
      const r = {
        dataX: i.name,
        dataIndex: i.dataIndex,
        seriesIndex: i.seriesIndex,
        data: e.data.map((l, a) => ({
          name: e.names[a],
          value: l[i.dataIndex]
        }))
      };
      s.emit("col-click", r);
    };
    let n = null;
    function v() {
      var r, l, a, o, d, f, g;
      let { yAxis: i } = e.options;
      if (Array.isArray(e.options.yAxis))
        i = e.options.yAxis;
      else {
        let { data: u } = e;
        Array.isArray((r = e.options) == null ? void 0 : r.series) && ((a = (l = e.options) == null ? void 0 : l.series) != null && a.length) && Array.isArray((d = (o = e.options) == null ? void 0 : o.series[0]) == null ? void 0 : d.data) && (u = (g = (f = e.options) == null ? void 0 : f.series[0]) == null ? void 0 : g.data), i = Array.from({ length: Array.isArray(u) ? u.length : 0 }, (F, z) => ({
          id: z,
          type: "value",
          min: e.range[0],
          max: e.range[1],
          name: e.units[0] || "",
          nameTextStyle: {
            align: "center",
            fontSize: 12,
            color: t().fontColor,
            fontWeight: "normal",
            padding: [0, 0, 0, 10]
          },
          splitLine: {
            show: !1,
            lineStyle: {
              color: [t().lineColor]
            }
          },
          axisLabel: {
            show: !0,
            fontSize: 12,
            color: t().fontColor
          },
          axisLine: {
            show: !0,
            lineStyle: {
              color: t().axisLine
            }
          },
          ...e.options.yAxis
        }));
      }
      return i;
    }
    function y() {
      var i;
      if (c.value) {
        n || (n = A.init(c.value));
        const r = {
          type: "inside",
          zoomLock: e.zoomLock,
          // й”Ғе®ҡеҢәеҹҹпјҡеҸӘиғҪе№із§»пјҢдёҚиғҪж”ҫеӨ§зј©е°Ҹ
          ...k()
        }, l = {
          ...e.options,
          dataZoom: [r],
          grid: {
            top: "16%",
            left: 10,
            right: 10,
            bottom: "15%",
            containLabel: !0,
            ...e.options.grid
          },
          tooltip: {
            trigger: "axis",
            confine: !0,
            backgroundColor: t().tooltipBackgroundColor,
            borderColor: t().tooltipBackgroundColor,
            textStyle: {
              color: t().tooltipFontColor
            },
            axisPointer: {
              type: "line",
              lineStyle: {
                color: t().lineColor
              }
            },
            ...e.options.tooltip
          },
          legend: {
            show: e.showLegend,
            icon: e.legendIcon,
            textStyle: {
              color: t().fontColor,
              fontSize: e.legendStyle.fontSize
            },
            itemWidth: e.legendStyle.iconSize,
            itemHeight: e.legendStyle.iconSize,
            [e.legendStyle.position]: "6%",
            left: e.legendStyle.align
          },
          xAxis: {
            type: "category",
            boundaryGap: !1,
            axisLabel: {
              fontSize: 12,
              color: t().fontColor,
              align: "center"
            },
            axisLine: {
              show: !0,
              lineStyle: {
                color: t().axisLine
              }
            },
            axisTick: { show: !1 },
            data: e.dataX,
            ...(i = e.options) == null ? void 0 : i.xAxis
          },
          yAxis: v(),
          color: e.options.color || e.colors,
          series: e.data.map((a, o) => {
            var d, f;
            return {
              name: e.names[o],
              type: "line",
              yAxisIndex: e.yIndexs.length > 0 ? e.yIndexs[o] : 0,
              symbol: e.showValue[o] ? "emptyCircle" : "none",
              // йңҖиҰҒжҳҫзӨәиҠӮзӮ№з¬ҰеҸ·жүҚиғҪжҳҫзӨәvalue
              data: a,
              smooth: e.smooth,
              label: {
                show: !!e.showValue[o],
                color: t().fontColor
              },
              areaStyle: e.showAreas[o] ? {
                color: new A.graphic.LinearGradient(0, 0, 0, 1, [{
                  offset: 0,
                  color: x(e.colors[o], 0.4)
                }, {
                  offset: 1,
                  color: x(e.colors[o], 0)
                }])
              } : void 0,
              markLine: e.markLine[o] ? {
                silent: !0,
                symbol: ["none"],
                label: { show: !1 },
                lineStyle: e.markLine[o].lineStyle,
                data: [{
                  name: "",
                  // TODO: ж ҮжіЁзәҝж Үйўҳ
                  yAxis: e.markLine[o].data
                }]
              } : void 0,
              ...(d = e.options) == null ? void 0 : d.series,
              ...(f = e.options) == null ? void 0 : f.lineSeries
            };
          })
        };
        e.title && (l.title = {
          text: e.title,
          left: "50%",
          textAlign: "center",
          textStyle: {
            color: t().fontColor,
            fontSize: 16,
            fontWeight: "bold"
          },
          padding: [0, 5, 0, 5],
          ...e.options.title
        }), n.setOption(l), n.on("click", (a) => {
          q(a);
        }), window.onresize = () => {
          S(() => {
            n.resize();
          });
        };
      }
    }
    return m(() => [
      e.data,
      e.dataX,
      e.names,
      e.markLine,
      e.showValue,
      e.zoomLock,
      e.scope,
      e.range,
      e.showYaxis,
      e.showAreas,
      e.yIndexs,
      e.options,
      e.units,
      e.colors,
      e.smooth,
      e.title,
      e.showLegend
    ], () => {
      y();
    }), m(() => [e.width, e.height, e.cStyle], () => {
      S(() => {
        n.resize();
      });
    }), O(() => {
      y();
    }), {
      chart: c,
      style: L,
      cStyles: b
    };
  }
});
export {
  j as default
};
