// @ths/design/es/components/magnifier/src/index.vue2.js
import { defineComponent as p, computed as g, reactive as y, onMounted as h, watch as s, ref as B, toRefs as S } from "vue";
import { ComponentLoader as $ } from "../../../utils/loader.js";
import { getStyle as _, getDomRandomClass as C, setStyle as n, getCssPropertyString as d } from "../../../utils/index.js";
const k = p({
  name: "TMagnifier",
  props: {
    title: {
      type: String,
      default: "标题"
    },
    componentName: {
      type: String,
      default: ""
    },
    scale: {
      type: Number,
      default: 1.5
    },
    autoScroll: {
      type: Boolean,
      default: !0
    },
    width: {
      type: Number,
      default: 800
    },
    height: {
      type: Number,
      default: 500
    },
    zoomInButtonIcon: {
      type: String,
      default: "components/images/fullsreen.svg"
    },
    closeButtonIcon: {
      type: String,
      default: "components/images/close.svg"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        closeButton: {
          default: {},
          hover: {}
        },
        zoomInButton: {
          default: {},
          hover: {}
        },
        dialog: {
          default: {}
        },
        header: {
          default: {}
        },
        title: {
          default: {}
        },
        content: {
          default: {}
        },
        mask: {
          default: {}
        }
      })
    }
  },
  emits: ["close"],
  setup(l, { emit: f }) {
    const e = g(() => l.cStyle), o = C("magnifier"), i = y({
      contentVisible: !1
    }), c = () => {
      var r, m;
      let t = "";
      return e.value.wrapper && (t += n(e.value.wrapper, `.${o}`)), e.value.mask && (t += n(e.value.mask, `div.t-magnifier.${o} .mask`)), e.value.dialog && (t += n(e.value.dialog, `div.t-magnifier.${o} .dialog`)), e.value.closeButton && (t += n(e.value.closeButton, `div.t-magnifier.${o} .dialog__close-button`)), e.value.closeButton && ((r = e.value.closeButton) != null && r.hover) && (t += `div.t-magnifier.${o} .dialog__close-button:hover {
           ${d(e.value.closeButton.hover)}
        }`), e.value.zoomInButton && (t += n(e.value.zoomInButton, `.${o} .zoom-in-button`)), e.value.zoomInButton && ((m = e.value.zoomInButton) != null && m.hover) && (t += `.${o} .zoom-in-button:hover {
           ${d(e.value.zoomInButton.hover)}
        }`), e.value.header && (t += n(e.value.header, `div.t-magnifier.${o} .dialog__header`)), e.value.title && (t += n(e.value.title, `div.t-magnifier.${o} .dialog__title`)), e.value.content && (t += n(e.value.content, `div.t-magnifier.${o} .dialog__content`)), t;
    }, u = () => {
      const t = document.querySelector(`.${o} .style-container`);
      t && (t.innerHTML = `<style>${c()}</style>`);
    }, v = () => {
      i.contentVisible = !1, f("close");
    };
    h(() => {
      u();
    }), s(() => l.cStyle, () => {
      u();
    }, {
      deep: !0,
      immediate: !0
    });
    const a = B(!1);
    return s(() => l.componentName, (t) => {
      a.value = !1, t && $.loadComponent(t.replace("Component", ""), () => {
        a.value = !0;
      });
    }, { immediate: !0 }), {
      componentLoaded: a,
      ...S(i),
      getStyle: _,
      curStyle: e,
      domOnlyClass: o,
      handleClose: v
    };
  }
});
export {
  k as default
};
