// @ths/design/es/components/magnifier/src/index.vue.js
import r from "./index.vue2.js";
import { openBlock as t, createElementBlock as i, normalizeClass as d, createElementVNode as o, normalizeStyle as n, toDisplayString as m, createBlock as p, resolveDynamicComponent as g, createCommentVNode as a } from "vue";
import "./index.vue3.js";
import c from "../../../_virtual/_plugin-vue_export-helper.js";
const u = {
  key: 0,
  class: "t-magnifier__container"
}, h = { class: "dialog__header" }, f = { class: "dialog__title" }, y = { class: "dialog__content" }, $ = { class: "dialog__content--scroll" };
function k(e, l, v, C, w, b) {
  return t(), i("div", {
    class: d(["t-magnifier", e.domOnlyClass])
  }, [
    o("span", {
      class: "zoom-in-button",
      style: n({
        backgroundImage: `url(${e.zoomInButtonIcon})`,
        ...e.getStyle({ width: e.cStyle.wrapper.default.width, height: e.cStyle.wrapper.default.height })
      }),
      onClick: l[0] || (l[0] = (s) => e.contentVisible = !0)
    }, null, 4),
    e.contentVisible ? (t(), i("div", u, [
      l[2] || (l[2] = o("p", { class: "mask" }, null, -1)),
      o("div", {
        class: "dialog",
        style: n({
          width: `${e.width + 20}px`,
          height: `${e.height}px`,
          marginTop: `${-e.height / 2}px`,
          marginLeft: `${20 - e.width / 2}px`
        })
      }, [
        o("p", h, [
          o("span", f, m(e.title), 1),
          o("span", {
            class: "dialog__close-button",
            style: n({ backgroundImage: `url(${e.closeButtonIcon})` }),
            onClick: l[1] || (l[1] = (...s) => e.handleClose && e.handleClose(...s))
          }, null, 4)
        ]),
        o("div", y, [
          o("div", $, [
            e.componentName && e.componentLoaded ? (t(), p(g(e.componentName), {
              key: 0,
              class: "dialog__content--scale",
              style: n({ transform: `scale(${e.scale})` })
            }, null, 8, ["style"])) : a("", !0)
          ])
        ])
      ], 4)
    ])) : a("", !0),
    l[3] || (l[3] = o("div", { class: "style-container" }, null, -1))
  ], 2);
}
const z = /* @__PURE__ */ c(r, [["render", k]]);
export {
  z as default
};
