// @ths/design/es/components/chart-design/src/index.vue.js
import o from "./index.vue2.js";
import { openBlock as n, createElementBlock as a, normalizeClass as m, normalizeStyle as i, createElementVNode as t } from "vue";
import "./index.vue3.js";
import c from "../../../_virtual/_plugin-vue_export-helper.js";
const p = {
  ref: "chart",
  class: "chart"
}, d = ["src"];
function y(e, s, u, f, v, h) {
  var l, r;
  return n(), a("div", {
    class: m(["t-chart-design", e.domOnlyClass]),
    style: i({ ...(r = (l = e.curStyle) == null ? void 0 : l.wrapper) == null ? void 0 : r.default })
  }, [
    t("div", {
      class: "chart-box",
      style: i({
        visibility: e.checkEmpty ? "visible" : "hidden"
      })
    }, [
      t("div", p, null, 512)
    ], 4),
    t("div", {
      class: "empty",
      style: i({
        visibility: e.checkEmpty ? "hidden" : "visible"
      })
    }, [
      t("img", {
        class: "empty-img",
        style: i({
          "object-fit": e.objectFit
        }),
        src: e.emptyImg,
        alt: ""
      }, null, 12, d),
      s[0] || (s[0] = t("span", { class: "empty-text" }, "暂无数据", -1))
    ], 4),
    s[1] || (s[1] = t("div", { class: "style-container" }, null, -1))
  ], 6);
}
const $ = /* @__PURE__ */ c(o, [["render", y]]);
export {
  $ as default
};
