// @ths/design/es/components/chart-design/src/index.vue2.js
import { defineComponent as L, computed as C, ref as M, nextTick as o, watch as S, onMounted as T, onUnmounted as $ } from "vue";
import * as g from "echarts";
import j from "lodash/cloneDeep";
import { getDomRandomClass as k, setStyle as d } from "../../../utils/index.js";
const R = L({
  name: "TChartDesign",
  props: {
    // 地图json数据
    geoJson: {
      type: Object,
      default: () => ({})
    },
    // 注册地图名称
    registerMapName: {
      type: String,
      default: ""
    },
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 外层容器
        wrapper: {
          default: {}
        }
      })
    },
    // echart自定义的配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 高亮数据
    highLight: {
      type: String,
      default: ""
    }
  },
  emits: ["com-click", "legend-select", "get-chart-instance"],
  setup(t, { emit: y }) {
    const l = k("chart-design"), a = C(() => t.cStyle), r = M(null);
    let u = null;
    const w = () => {
      let e = "";
      return a.value.wrapper && (e += d(a.value.wrapper, `.${l}`)), a.value.emptyContainer && (e += d(a.value.emptyContainer, `.${l} .empty`)), a.value.emptyImg && (e += d(a.value.emptyImg, `.${l} .empty .empty-img`)), a.value.emptyText && (e += d(a.value.emptyText, `.${l} .empty .empty-text`)), e;
    }, m = () => {
      const e = document.querySelector(`.${l} .style-container`);
      e && (e.innerHTML = `<style>${w()}</style>`);
    };
    let n = null;
    const i = () => {
      r.value && (r.value.getBoundingClientRect().width ?? 0) && o(() => {
        n.resize();
      });
    }, b = C(() => {
      var e, s, f, p;
      if (t.options) {
        if (Array.isArray((e = t.options) == null ? void 0 : e.series)) {
          const z = t.options.series.some((c) => {
            var v;
            return (c == null ? void 0 : c.data) && ((v = c == null ? void 0 : c.data) == null ? void 0 : v.length) !== 0;
          });
          return o(() => {
            i();
          }), z;
        }
        if (typeof ((s = t.options) == null ? void 0 : s.series) == "object")
          return o(() => {
            i();
          }), ((p = (f = t.options.series) == null ? void 0 : f.data) == null ? void 0 : p.length) === 0;
      }
      return o(() => {
        i();
      }), !0;
    }), h = () => {
      if (r.value) {
        n || (n = g.init(r.value), y("get-chart-instance", n)), n.clear(), g == null || g.registerMap(t.registerMapName, t.geoJson);
        const e = j(t.options);
        try {
          n.setOption(e), t.highLight && n.dispatchAction({
            type: "highlight",
            name: t.highLight
          });
        } catch (s) {
          console.log(s);
        }
        o(() => {
          i();
        });
      }
    };
    S(() => [
      t.options,
      t.highLight,
      t.registerMapName,
      t.geoJson
    ], () => {
      h(), o(() => {
        i();
      });
    }, { deep: !0 });
    function O() {
      r.value && (u = new ResizeObserver((e) => {
        e.forEach((s) => {
          s.target === r.value && i();
        });
      }), u.observe(r.value));
    }
    return S(() => [t.cStyle], () => {
      m(), o(() => {
        i();
      });
    }, { deep: !0 }), T(() => {
      m(), n = g.init(r.value), y("get-chart-instance", n), h(), n.on("click", (e) => {
        y("com-click", e);
      }), n.on("legendselectchanged", (e) => {
        y("legend-select", e);
      }), O();
    }), setTimeout(() => {
      i();
    }, 500), window.addEventListener("resize", i), $(() => {
      window.removeEventListener("resize", i), u && (u.disconnect(), u = null);
    }), {
      chart: r,
      curStyle: a,
      checkEmpty: b,
      domOnlyClass: l
    };
  }
});
export {
  R as default
};
