// @ths/design/es/components/list-card-multiple-choice/src/index.vue2.js
import { defineComponent as T, computed as N, ref as o, onMounted as j, watch as w } from "vue";
import { getValueOrDefaultValue as P, getDomRandomClass as q, setStyle as u, getCssPropertyString as i } from "../../../utils/index.js";
const E = {
  value: "value",
  // 卡片数值
  label: "label",
  // 卡片标签
  icon: "icon",
  // 是否额外需要icon
  unit: "unit"
  // value 单位
}, z = T({
  name: "TListCardMultipleChoice",
  props: {
    // 关键字，用于设置默认选中
    keyWord: {
      type: String,
      default: ""
    },
    // 默认选中项
    defaultChecked: {
      type: Array,
      default: () => []
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 单位
    unit: {
      type: String,
      default: ""
    },
    // 数据别名
    dataAlias: {
      type: Object,
      default: () => {
      }
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 卡片中的每一项
        cardItem: {
          default: {},
          hover: {},
          active: {}
        },
        // 未被选中的卡片容器
        cardUnchecked: {
          default: {},
          hover: {}
        },
        // 卡片上方的value元素
        cardItemValue: {
          default: {}
        },
        // 卡片上方的value内容元素
        cardItemValueContent: {
          default: {}
        },
        // 卡片上方的value 单位元素
        cardItemValueUnit: {
          default: {}
        },
        // 卡片下方的label元素
        cardItemLabel: {
          default: {}
        },
        // 卡片下方的label内容元素
        cardItemLabelContent: {
          // 标签内容默认样式
          default: {},
          hover: {},
          active: {}
        },
        // 卡片下方的label图标元素
        cardItemLabelIcon: {
          default: {}
        },
        // 卡片中的icon
        cardItemIcon: {
          default: {},
          hover: {},
          active: {}
        }
      })
    }
  },
  emits: ["card-click"],
  setup(c, { emit: D }) {
    const e = N(() => c.cStyle), a = q("t-list-card-multiple-choice"), r = o([]), n = o([]), v = o(), M = N(() => ({ ...E, ...c.dataAlias })), d = o([]), W = () => {
      var l, h, f, I, s, C, $, y, k, p, b, S, g, L, V, O, U, _, A, J;
      let t = "";
      return e.value.wrapper && (t += u(e.value.wrapper, `.${a}`)), e.value.cardItem && (t += u(e.value.cardItem, `.${a} .multiple-choice-card-item`)), e.value.cardUnchecked && (t += u(e.value.cardUnchecked, `.${a} .item-unchecked`)), (l = e.value.cardUnchecked) != null && l.hover && (t += `.${a} .item-unchecked:hover {
          ${i((h = e.value.cardUnchecked) == null ? void 0 : h.hover)}
        }`), (f = e.value.cardItem) != null && f.active && (t += `.${a} .item-checked {
          ${i((I = e.value.cardItem) == null ? void 0 : I.active)}
        }`), e.value.cardItemValue && (t += u(e.value.cardItemValue, `.${a} .multiple-choice-card-item-value`)), e.value.cardItemValueContent && (t += u(e.value.cardItemValueContent, `.${a} .multiple-choice-card-item-value_content`)), (s = e.value.cardItemValueContent) != null && s.hover && (t += `.${a} .multiple-choice-card-item:hover .multiple-choice-card-item-value_content {
          ${i((C = e.value.cardItemValueContent) == null ? void 0 : C.hover)}
        }`), ($ = e.value.cardItemValueContent) != null && $.active && (t += `.${a} .item-value-content-checked {
          ${i((y = e.value.cardItemValueContent) == null ? void 0 : y.active)}
        }`), e.value.cardItemValueUnit && (t += u(e.value.cardItemValueUnit, `.${a} .multiple-choice-card-item-value_unit`)), (k = e.value.cardItemValueUnit) != null && k.hover && (t += `.${a} .multiple-choice-card-item:hover .multiple-choice-card-item-value_unit {
          ${i((p = e.value.cardItemValueUnit) == null ? void 0 : p.hover)}
        }`), (b = e.value.cardItemValueUnit) != null && b.active && (t += `.${a} .item-value-unit-checked {
          ${i((S = e.value.cardItemValueUnit) == null ? void 0 : S.active)}
        }`), e.value.cardItemLabel && (t += u(e.value.cardItemLabel, `.${a} .multiple-choice-card-item-label`)), e.value.cardItemLabelContent && (t += u(e.value.cardItemLabelContent, `.${a} .multiple-choice-card-item-label_content`)), (g = e.value.cardItemLabelContent) != null && g.hover && (t += `.${a} .multiple-choice-card-item:hover .multiple-choice-card-item-label_content {
          ${i((L = e.value.cardItemLabelContent) == null ? void 0 : L.hover)}
        }`), (V = e.value.cardItemLabelContent) != null && V.active && (t += `.${a} .item-label-content-checked {
          ${i((O = e.value.cardItemLabelContent) == null ? void 0 : O.active)}
        }`), e.value.cardItemLabelIcon && (t += u(e.value.cardItemLabelIcon, `.${a} .multiple-choice-card-item-label_icon`)), e.value.cardItemIcon && (t += u(e.value.cardItemIcon, `.${a} .multiple-choice-card-item-icon`)), (U = e.value.cardItemIcon) != null && U.hover && (t += `.${a} .multiple-choice-card-item:hover .multiple-choice-card-item-icon {
          ${i((_ = e.value.cardItemIcon) == null ? void 0 : _.hover)}
        }`), (A = e.value.cardItemIcon) != null && A.active && (t += `.${a} .item-icon-checked {
          ${i((J = e.value.cardItemIcon) == null ? void 0 : J.active)}
        }`), t;
    }, m = () => {
      const t = document.querySelector(`.${a} .style-container`);
      t && (t.innerHTML = `<style>${W()}</style>`);
    }, H = (t) => {
      d.value[t] = !d.value[t], v.value = JSON.parse(JSON.stringify(r.value[t])), n.value = [];
      for (let l = 0; l < d.value.length; l++)
        d.value[l] && n.value.push(JSON.parse(JSON.stringify(r.value[l])));
      D("card-click", v.value, n.value);
    };
    return j(() => {
      m();
    }), w(() => c.cStyle, () => {
      m();
    }, { deep: !0 }), w(() => [c.data, c.keyWord, c.defaultChecked], () => {
      c.data.length ? (r.value = JSON.parse(JSON.stringify(c.data)), n.value = [], d.value = new Array(r.value.length).fill(!1)) : (r.value = [], n.value = []);
      let t = [];
      if (c.defaultChecked.length > 0 && (t = c.defaultChecked.map((l) => String(l))), c.data.length > 0 && c.keyWord !== "" && c.defaultChecked.length > 0)
        for (let l = 0; l < r.value.length; l++)
          t.includes(String(r.value[l][c.keyWord])) ? (d.value[l] = !0, n.value.push(JSON.parse(JSON.stringify(r.value[l])))) : d.value[l] = !1;
    }, { deep: !0, immediate: !0 }), {
      curStyle: e,
      domOnlyClass: a,
      dataList: r,
      clickCardHandler: H,
      getValueOrDefaultValue: P,
      curAlias: M,
      stateCollection: d
    };
  }
});
export {
  z as default
};
