// @ths/design/es/components/composite-timeline/src/index.vue2.js
import { defineComponent as ie, computed as re, reactive as ue, ref as b, watch as T, onMounted as ce, onBeforeUnmount as se, toRefs as de, nextTick as y } from "vue";
import g from "lodash/cloneDeep";
import { getDomRandomClass as me, getCssPropertyString as f, setStyle as c } from "../../../utils/index.js";
const G = "transform 0.38s", ve = "width 0.38s", ye = {
  label: "label",
  value: "value",
  children: "children"
}, fe = {
  interval: 3,
  autoplay: !0
}, he = {
  interval: 3,
  autoplay: !0
}, xe = ie({
  name: "TCompositeTimeline",
  props: {
    // 主要时间和次要时间的拼接符
    jointSymbol: {
      type: String,
      default: "-"
    },
    // 选中项
    activeItem: {
      type: Object,
      default: () => ({})
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 数据别名
    alias: {
      type: Object,
      default: () => ({})
    },
    // 主要时间轴播放属性
    mainAutoPlay: {
      type: Object,
      default: () => fe
    },
    // 次要时间轴播放属性
    secondaryAutoPlay: {
      type: Object,
      default: () => he
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    }
  },
  /**
   * 主要时间变化
   * 次要时间变化
   */
  emits: ["main-click", "secondary-click"],
  setup(r, { emit: x }) {
    const a = re(() => r.cStyle), o = me("composite-timeline"), e = ue({
      curData: [],
      dateIndex: "",
      // 高亮日期的索引
      secondaryIndex: "",
      // 高亮次要时间轴的索引
      currentItem: "",
      toggleHourTimeline: !1,
      // 是否展示小时时间轴
      hoursList: [],
      mainAutoPlay: {},
      // 主体时间轴自动播放属性
      secondaryAutoPlay: {}
      // 次要时间轴自动播放属性
    }), h = {
      x: 0
    }, m = {
      x: 0
    }, B = b(), I = b(), s = b([]), q = (t) => {
      t && s.value.push(t);
    }, $ = b(), p = (t) => {
      y(() => {
        I.value.style.transform = `translateX(calc(${t}px - 50%))`, B.value.style.transform = `translateX(calc(${t}px - 50%))`, $.value.style.width = `${t}px`;
      });
    }, D = () => {
      I.value.style.transition = G, B.value.style.transition = G, $.value.style.transition = ve;
    }, C = () => {
      h.x = 0, m.x = 0;
    }, E = (t) => {
      y(() => {
        var l, n, d, u, A, S;
        const i = ((n = (l = I.value.style) == null ? void 0 : l.transform) == null ? void 0 : n.match(/(-?)\d+/g)[0]) ?? null;
        parseFloat(i) < ((u = (d = s.value) == null ? void 0 : d[0]) == null ? void 0 : u.offsetLeft) || parseFloat(i) > ((S = (A = s.value) == null ? void 0 : A[s.value.length - 1]) == null ? void 0 : S.offsetLeft) || p(t.clientX - h.x + m.x);
      });
    }, U = (t) => {
      var u, A;
      const i = ((A = (u = I.value.style) == null ? void 0 : u.transform) == null ? void 0 : A.match(/(-?)\d+/g)[0]) ?? null;
      let l = 0, n = 0, d = null;
      if (i !== null) {
        t.forEach((R, oe) => {
          const O = R.offsetLeft, k = Math.abs(O - parseFloat(i));
          (oe === 0 || k < l) && (l = k, n = O, d = R);
        }), m.x = n, D(), p(n);
        const S = d.dataset.index ?? null, F = e.hoursList[S].value, le = `${e.currentItem}${r.jointSymbol}${F}`, ne = {
          main: e.currentItem,
          secondary: F
        };
        x("secondary-click", le, ne);
      }
    };
    let P, v;
    const N = (t) => {
      e.secondaryAutoPlay.autoplay = !1, clearInterval(v), m.x = t.clientX - h.x + m.x, y(() => {
        U(s.value), document.removeEventListener("mousemove", E), document.removeEventListener("mouseup", N);
      });
    }, z = (t) => {
      h.x = t.clientX, y(() => {
        I.value.style.transition = "", $.value.style.transition = "";
      }), document.addEventListener("mousemove", E), document.addEventListener("mouseup", N);
    }, w = (t, i) => {
      const l = s.value[t];
      if (D(), l) {
        const n = l.offsetLeft;
        h.x = n, m.x = n;
        const d = e.hoursList[t].value, u = `${e.currentItem}${r.jointSymbol}${d}`;
        p(n), i(u);
      }
    }, J = (t, i) => {
      e.secondaryAutoPlay.autoplay = !1, clearInterval(v), e.secondaryIndex = i, y(() => {
        w(i, (l) => {
          const n = {
            main: e.currentItem,
            secondary: l
          };
          x("secondary-click", l, n);
        });
      });
    }, _ = () => {
      P = setInterval(() => {
        e.dateIndex === "" && (e.dateIndex = 0), e.currentItem = e.curData[e.dateIndex].value, x("main-click", e.currentItem), e.dateIndex = e.dateIndex === e.curData.length - 1 ? 0 : e.dateIndex += 1;
      }, e.mainAutoPlay.interval * 1e3);
    }, M = () => {
      v = setInterval(() => {
        y(() => {
          var i;
          e.secondaryIndex === "" && (e.secondaryIndex = 0);
          const t = s.value[e.secondaryIndex];
          if (D(), t) {
            const l = t.offsetLeft;
            h.x = l, m.x = l;
            const n = ((i = e.hoursList[e.secondaryIndex]) == null ? void 0 : i.value) ?? "";
            if (n) {
              const d = `${e.currentItem}${r.jointSymbol}${n}`, u = {
                main: e.currentItem,
                secondary: n
              };
              x("secondary-click", d, u), p(l), e.secondaryIndex = e.secondaryIndex === s.value.length - 1 ? 0 : e.secondaryIndex += 1;
            }
          }
        });
      }, e.secondaryAutoPlay.interval * 1e3);
    }, L = () => {
      clearInterval(P), e.mainAutoPlay.autoplay = g(r.mainAutoPlay.autoplay), !e.toggleHourTimeline && e.mainAutoPlay.autoplay && _();
    }, j = () => {
      clearInterval(v), e.secondaryIndex = "", e.toggleHourTimeline && (e.secondaryAutoPlay.autoplay = g(r.secondaryAutoPlay.autoplay), e.secondaryAutoPlay.autoplay && M());
    }, K = () => {
      clearInterval(P), e.mainAutoPlay.autoplay = !e.mainAutoPlay.autoplay, e.mainAutoPlay.autoplay && _();
    }, Q = () => {
      s.value = [], clearInterval(v), e.secondaryAutoPlay.autoplay = !e.secondaryAutoPlay.autoplay, e.secondaryAutoPlay.autoplay && M();
    }, V = (t) => {
      x("main-click", t);
    }, W = (t, i) => {
      s.value = [], e.hoursList = t.children, e.dateIndex === i && e.toggleHourTimeline ? (e.dateIndex = "", e.currentItem = "", e.toggleHourTimeline = !1) : (e.dateIndex = i, e.currentItem = t.value, e.toggleHourTimeline = e.dateIndex >= 0), V(e.currentItem), L(), p(0), j(), C();
    }, Y = () => {
      const t = {
        ...ye,
        ...r.alias
      }, i = g(r.data);
      i.forEach((l) => {
        l.label = l[t.label] ?? "", l.value = l[t.value] ?? "", l.children = l[t.children] ?? [], l.children.forEach((n) => {
          n.label = n[t.label], n.value = n[t.value];
        });
      }), e.curData = i;
    }, H = () => {
      clearInterval(P), clearInterval(v);
    }, Z = () => {
      e.toggleHourTimeline = !1, e.hoursList = [];
    };
    T([() => r.alias, () => r.data], () => {
      H(), Z(), L(), Y();
    }, { immediate: !0, deep: !0 });
    const ee = (t) => {
      const i = e.curData.findIndex((u) => u.value === t.main) ?? "", l = e.curData[i] ?? "";
      s.value = [], l && (e.mainAutoPlay.autoplay = !1, clearInterval(P), e.dateIndex = i, e.currentItem = l.value, e.toggleHourTimeline = e.dateIndex >= 0, e.hoursList = l.children ?? [], p(0), j(), C());
      const n = e.hoursList.findIndex((u) => u.value === t.secondary) ?? "";
      (e.hoursList[n] ?? "") && (e.secondaryAutoPlay.autoplay = !1, clearInterval(v), y(() => {
        w(n, (u) => {
        });
      }));
    }, te = () => {
      let t = "";
      return a.value.date && a.value.date.active && (t += `
          .${o} .relative .date-timeline .date-timeline-item.active .date-timeline-item__date {
            ${f(a.value.date.active)}
          }
        `), a.value.dateText && a.value.dateText.active && (t += `
          .${o} .relative .date-timeline .date-timeline-item.active .date-timeline-item__date span {
            ${f(a.value.dateText.active)}
          }
        `), a.value.dateIcon && a.value.dateIcon.active && (t += `
          .${o} .relative .date-timeline .date-timeline-item.active .date-timeline-item__icon {
            ${f(a.value.dateIcon.active)}
          }
        `), a.value.dateAutoPlayBtn && a.value.dateAutoPlayBtn.hover && (t += `
          .${o} .relative .date-autoplay:hover {
            ${f(a.value.dateAutoPlayBtn.hover)}
          }
        `), a.value.dateAutoPlayBtn && a.value.dateAutoPlayBtn.active && (t += `
          .${o} .relative .date-autoplay.date-active {
            ${f(a.value.dateAutoPlayBtn.active)}
          }
        `), a.value.hourAutoPlayBtn && a.value.hourAutoPlayBtn.hover && (t += `
          .${o} .relative .hour-timeline .hour-timeline-play:hover {
            ${f(a.value.hourAutoPlayBtn.hover)}
          }
        `), a.value.hourAutoPlayBtn && a.value.hourAutoPlayBtn.active && (t += `
          .${o} .relative .hour-timeline .hour-timeline-play.hour-active {
            ${f(a.value.hourAutoPlayBtn.active)}
          }
        `), t;
    }, ae = () => {
      let t = "";
      return a.value.wrapper && (t += c(a.value.wrapper, `.${o}`)), a.value.dateTimeline && (t += c(a.value.dateTimeline, `.${o} .relative .date-timeline`)), a.value.date && (t += c(a.value.date, `.${o} .relative .date-timeline .date-timeline-item .date-timeline-item__date`)), a.value.dateText && (t += c(a.value.dateText, `.${o} .relative .date-timeline .date-timeline-item .date-timeline-item__date span`)), a.value.dateIcon && (t += c(a.value.dateIcon, `.${o} .relative .date-timeline .date-timeline-item .date-timeline-item__icon`)), a.value.dateAutoPlayBtn && (t += c(a.value.dateAutoPlayBtn, `.${o} .relative .date-autoplay`)), a.value.hourTimeLine && (t += c(a.value.hourTimeLine, `.${o} .relative .hour-timeline`)), a.value.hourAutoPlayBtn && (t += c(a.value.hourAutoPlayBtn, `.${o} .relative .hour-timeline .hour-timeline-play`)), a.value.hourTimeLineScale && (t += c(a.value.hourTimeLineScale, `.${o} .relative .hour-timeline .hour-timeline-item`)), a.value.hourTimeLineScaleText && (t += c(a.value.hourTimeLineScaleText, `.${o} .relative .hour-timeline .hour-timeline-item span`)), a.value.trigger && (t += c(a.value.trigger, `.${o} .relative .hour-timeline .hour-timeline-trigger`)), a.value.progress && (t += c(a.value.progress, `.${o} .relative .hour-timeline .hour-timeline-progress`)), a.value.tooltip && (t += c(a.value.progress, `.${o} .relative .hour-timeline .hour-timeline-tooltip`)), t;
    }, X = () => {
      const t = document.querySelector(`.${o} .style-container`);
      t && (t.innerHTML = `<style>${ae()}${te()}</style>`);
    };
    return ce(() => {
      X();
    }), se(() => {
      H();
    }), T(() => r.mainAutoPlay, (t) => {
      e.mainAutoPlay = g(t), clearInterval(P), e.mainAutoPlay.autoplay && _();
    }, { immediate: !0, deep: !0 }), T(() => r.secondaryAutoPlay, (t) => {
      e.secondaryAutoPlay = g(t), L(), j();
    }, { immediate: !0, deep: !0 }), T(() => r.cStyle, () => {
      X();
    }, { deep: !0 }), T(() => r.activeItem, (t) => {
      H(), L(), ee(t);
    }, { immediate: !0, deep: !0 }), {
      ...de(e),
      curStyle: a,
      domOnlyClass: o,
      showHourTimeline: W,
      translateTrigger: z,
      triggerRef: I,
      scaleRef: q,
      progressRef: $,
      getTime: J,
      playDateLoop: K,
      playHourLoop: Q,
      tooltipRef: B
    };
  }
});
export {
  xe as default
};
