// @ths/design/es/components/composite-timeline/src/index.vue.js
import p from "./index.vue2.js";
import { openBlock as s, createElementBlock as r, normalizeClass as n, createElementVNode as i, Fragment as a, renderList as d, toDisplayString as u, withDirectives as v, withModifiers as c, vShow as f } from "vue";
import "./index.vue3.js";
import y from "../../../_virtual/_plugin-vue_export-helper.js";
const g = { class: "relative" }, h = { class: "date-timeline" }, k = { class: "date-timeline-item__date" }, C = ["onClick"], L = { class: "hour-timeline" }, T = ["data-index", "onClick"], $ = {
  ref: "tooltipRef",
  class: "hour-timeline-tooltip"
}, w = {
  ref: "progressRef",
  class: "hour-timeline-progress"
};
function D(e, l, _, H, R, b) {
  return s(), r("div", {
    class: n(["t-composite-timeline", e.domOnlyClass])
  }, [
    i("div", g, [
      i("div", {
        class: n(["date-autoplay", e.mainAutoPlay.autoplay ? "date-active" : ""]),
        onClick: l[0] || (l[0] = (...o) => e.playDateLoop && e.playDateLoop(...o))
      }, null, 2),
      i("div", h, [
        (s(!0), r(a, null, d(e.curData, (o, t) => (s(), r("div", {
          key: t,
          class: n(["date-timeline-item", e.currentItem === o.value ? "active" : ""])
        }, [
          i("div", k, [
            i("span", null, u(o.label), 1)
          ]),
          i("div", {
            class: "date-timeline-item__icon",
            onClick: (m) => e.showHourTimeline(o, t)
          }, null, 8, C)
        ], 2))), 128))
      ]),
      l[3] || (l[3] = i("div", { class: "style-container" }, null, -1)),
      v(i("div", L, [
        (s(!0), r(a, null, d(e.hoursList, (o, t) => (s(), r("div", {
          ref_for: !0,
          ref: e.scaleRef,
          key: t,
          "data-index": t,
          class: "hour-timeline-item",
          onClick: c((m) => e.getTime(o, t), ["stop"])
        }, [
          i("span", null, u(o.label), 1)
        ], 8, T))), 128)),
        i("div", {
          ref: "triggerRef",
          class: "hour-timeline-trigger",
          onMousedown: l[1] || (l[1] = (...o) => e.translateTrigger && e.translateTrigger(...o))
        }, null, 544),
        i("div", $, " 12423526 ", 512),
        i("div", w, null, 512),
        i("div", {
          class: n(["hour-timeline-play", e.secondaryAutoPlay.autoplay ? "hour-active" : ""]),
          onClick: l[2] || (l[2] = (...o) => e.playHourLoop && e.playHourLoop(...o))
        }, null, 2)
      ], 512), [
        [f, e.toggleHourTimeline]
      ])
    ])
  ], 2);
}
const P = /* @__PURE__ */ y(p, [["render", D]]);
export {
  P as default
};
