// @ths/design/es/components/swiper-base/src/index.vue.js
import g from "./index.vue2.js";
import { resolveComponent as p, openBlock as t, createElementBlock as r, normalizeClass as v, normalizeStyle as k, createElementVNode as l, createBlock as u, mergeProps as w, withCtx as f, Fragment as b, renderList as C, toDisplayString as S, createCommentVNode as o } from "vue";
import "./index.vue3.js";
import $ from "../../../_virtual/_plugin-vue_export-helper.js";
const B = { class: "swiper-container" }, z = {
  key: 0,
  class: "text-container"
}, D = {
  key: 1,
  class: "img-container"
}, E = ["src"], I = {
  key: 2,
  class: "iframe"
}, N = ["src"];
function V(e, i, A, F, L, O) {
  var a, n, c, d, m;
  const h = p("el-carousel-item"), y = p("el-carousel");
  return t(), r("div", {
    class: v(["t-swiper-base", e.domOnlyClass]),
    style: k({ ...(n = (a = e.curStyle) == null ? void 0 : a.wrapper) == null ? void 0 : n.default })
  }, [
    l("div", B, [
      (t(), u(y, w(e.filterAttrs(e.$attrs), {
        ref: "carousel",
        key: e.count,
        height: (m = (d = (c = e.cStyle) == null ? void 0 : c.wrapper) == null ? void 0 : d.default) == null ? void 0 : m.height,
        interval: e.curInterval,
        "initial-index": e.initialIndex
      }), {
        default: f(() => [
          (t(!0), r(b, null, C(e.imageData, (s, _) => (t(), u(h, {
            key: _,
            label: s.desc
          }, {
            default: f(() => [
              e.type === "text" ? (t(), r("div", z, S(s), 1)) : o("", !0),
              e.type === "image" ? (t(), r("div", D, [
                l("img", {
                  draggable: "false",
                  src: s.src
                }, null, 8, E)
              ])) : o("", !0),
              e.type === "iframe" ? (t(), r("div", I, [
                l("iframe", {
                  src: s.src,
                  frameborder: "0",
                  scrolling: "no",
                  width: "100%",
                  height: "100%"
                }, null, 8, N)
              ])) : o("", !0)
            ]),
            _: 2
          }, 1032, ["label"]))), 128))
        ]),
        _: 1
      }, 16, ["height", "interval", "initial-index"]))
    ]),
    i[0] || (i[0] = l("div", { class: "style-container" }, null, -1))
  ], 6);
}
const H = /* @__PURE__ */ $(g, [["render", V]]);
export {
  H as default
};
