// @ths/design/es/components/swiper-base/src/index.vue2.js
import { defineComponent as $, ref as _, computed as C, reactive as I, onMounted as T, watch as l, toRefs as W } from "vue";
import h from "lodash/cloneDeep";
import { filterAttrs as b, getDomRandomClass as w, setStyle as i } from "../../../utils/index.js";
const H = $({
  name: "TSwiperBase",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 组件容器
        wrapper: {
          default: {}
        },
        // 每一项的容器
        itemWrapper: {
          default: {}
        },
        // 指示器的容器
        indicatorWrapper: {
          default: {},
          hover: {}
        },
        // 指示器
        indicator: {
          default: {},
          hover: {}
        },
        // 切换按钮
        toggleButton: {
          default: {},
          hover: {}
        }
      })
    },
    // 数据
    data: {
      type: Array,
      default: () => []
    },
    // 别名及样式设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 轮播速度
        interval: {
          type: Number,
          default: 3e3
        },
        // 取resData数据中的别名设置
        data: {
          nameKey: "",
          // 名称别名
          dataKey: ""
          // 数据别名
        }
      })
    },
    // 是否轮播文字
    showText: {
      type: Boolean,
      default: !1
    },
    // 显示类型
    type: {
      type: String,
      default: "image"
    },
    // 当前显示的索引
    initialIndex: {
      type: Number,
      default: 0
    }
  },
  setup(t) {
    const o = _(null), a = C(() => t.cStyle), n = w("swiper-base"), r = I({
      imageData: [],
      // 数据
      curInterval: t.seriesSetting.interval,
      count: 1
      // 计数器-确保DOM重新渲染
    }), u = (e) => {
      var s;
      if (r.imageData = [], (s = t.seriesSetting) != null && s.data && e) {
        const { nameKey: p, dataKey: S } = t.seriesSetting.data;
        r.imageData = e.map((d) => ({
          desc: d[p] || "",
          src: d[S] || ""
        }));
      }
    }, m = () => {
      let e = "";
      return a.value.indicatorWrapper && (e += i(a.value.indicatorWrapper, `.${n} .el-carousel__indicator`)), e;
    }, f = () => {
      let e = "";
      return a.value.indicator && (e += i(a.value.indicator, `.${n} .el-carousel__indicator .el-carousel__button`)), e;
    }, y = () => {
      let e = "";
      return a.value.toggleButton && (e += i(a.value.toggleButton, `.${n} .el-carousel__arrow`)), e;
    }, g = () => {
      let e = "";
      return a.value.itemWrapper && (t.type === "text" ? e += i(a.value.itemWrapper, `.${n} .text-container`) : t.type === "image" ? e += i(a.value.itemWrapper, `.${n} .img-container img`) : t.type === "iframe" && (e += i(a.value.itemWrapper, `.${n} .iframe`))), e;
    }, c = () => {
      const e = document.querySelector(`.${n} .style-container`);
      e && (e.innerHTML = `<style>${g()}${y()}
          ${m()}${f()}</style>`);
    }, v = () => {
      r.count += 1;
      const e = h(t.data);
      u(e), setTimeout(() => {
        o.value.setActiveItem(t.initialIndex);
      }, 100);
    };
    return T(() => {
      c(), u(t.data);
    }), l(() => t.cStyle, () => {
      c();
    }), l(() => [t.data, t.seriesSetting.data], () => {
      v();
    }, {
      deep: !0
    }), l(() => t.seriesSetting.interval, () => {
      r.count += 1, r.curInterval = t.seriesSetting.interval;
    }, {
      deep: !0
    }), {
      ...W(r),
      carousel: o,
      curStyle: a,
      domOnlyClass: n,
      filterAttrs: b
    };
  }
});
export {
  H as default
};
