// @ths/design/es/components/area-selector-custom-data/src/custom-list.vue.js
import m from "./custom-list.vue2.js";
import { resolveComponent as f, openBlock as o, createElementBlock as s, createElementVNode as i, createVNode as p, withCtx as _, createTextVNode as l, createCommentVNode as n, normalizeClass as h, Fragment as v, renderList as k, toDisplayString as r, withModifiers as C } from "vue";
import "./custom-list.vue3.js";
import w from "../../../_virtual/_plugin-vue_export-helper.js";
const y = { class: "list-wrapper" }, b = { class: "top" }, E = {
  key: 0,
  class: "add-button"
}, $ = {
  key: 1,
  class: "vertical-line"
}, N = { class: "list" }, g = ["onClick"], F = {
  key: 0,
  class: "line"
}, A = ["onClick"], L = ["onClick"], V = { class: "fold" }, B = {
  key: 0,
  class: "arrow arrow-up"
}, M = {
  key: 1,
  class: "arrow arrow-down"
}, z = { class: "bottom" }, S = { class: "children-list" };
function D(e, d, I, T, j, q) {
  const a = f("el-button");
  return o(), s("div", y, [
    i("div", b, [
      e.isEdit ? (o(), s("div", E, [
        p(a, {
          round: "",
          size: "small",
          onClick: e.handleAdd
        }, {
          default: _(() => d[1] || (d[1] = [
            l(" 新建 "),
            i("div", { class: "add-icon" }, null, -1)
          ])),
          _: 1,
          __: [1]
        }, 8, ["onClick"])
      ])) : n("", !0),
      e.isEdit ? (o(), s("div", $)) : n("", !0),
      i("div", N, [
        i("div", {
          class: h(["button-list", { expanded: e.showFold.showMore }])
        }, [
          (o(!0), s(v, null, k(e.customAreaList, (t) => {
            var c;
            return o(), s("div", {
              key: t.areaCode,
              class: h(["button", { active: t.areaCode === ((c = e.currentArea) == null ? void 0 : c.areaCode) }]),
              onClick: (u) => e.handleClick(t)
            }, [
              l(r(t.areaName) + " ", 1),
              e.isEdit ? (o(), s("div", F)) : n("", !0),
              e.isEdit ? (o(), s("div", {
                key: 1,
                class: "icon-delete",
                onClick: C((u) => e.close(t), ["stop"])
              }, null, 8, A)) : n("", !0),
              e.isEdit ? (o(), s("div", {
                key: 2,
                class: "icon-edit",
                onClick: C((u) => e.edit(t), ["stop"])
              }, null, 8, L)) : n("", !0)
            ], 10, g);
          }), 128))
        ], 2)
      ]),
      i("div", V, [
        p(a, {
          class: "fold-btn",
          type: "text",
          onClick: d[0] || (d[0] = (t) => e.toggleFoldState())
        }, {
          default: _(() => [
            e.showFold.showMore ? (o(), s("span", B)) : (o(), s("span", M)),
            l(" " + r(e.showFold.buttonName), 1)
          ]),
          _: 1
        })
      ])
    ]),
    i("div", z, [
      i("div", S, [
        (o(!0), s(v, null, k(e.childrenList, (t) => (o(), s("div", {
          key: t.regionCode,
          class: "children-list-item"
        }, r(t.regionName), 1))), 128))
      ])
    ])
  ]);
}
const O = /* @__PURE__ */ w(m, [["render", D], ["__scopeId", "data-v-2f27051f"]]);
export {
  O as default
};
