// @ths/design/es/components/area-selector-custom-data/src/custom-add.vue2.js
import { defineComponent as p, reactive as O, onMounted as S, toRefs as M } from "vue";
import d from "qs";
import s from "axios";
const A = p({
  name: "CustomAdd",
  props: {
    currentCategory: {
      type: Object,
      default: () => {
      }
    },
    // 获取地区数据列表url
    getRegionListUrl: {
      type: String,
      default: ""
    },
    // 添加单个区域url
    addSingleRegionUrl: {
      type: String,
      default: ""
    },
    // 删除单个区域url
    deleteSingleRegionUrl: {
      type: String,
      default: ""
    },
    // 修改区域url
    updateRegionUrl: {
      type: String,
      default: ""
    },
    // 添加区域url
    addRegionUrl: {
      type: String,
      default: ""
    },
    // 获取子级区域url
    getChildElementsUrl: {
      type: String,
      default: ""
    },
    // 行政区域默认类型
    defaultRegionLevel: {
      type: String,
      default: "province"
    }
  },
  emits: ["back", "selected"],
  setup(r, { emit: g }) {
    const e = O({
      areaLevel: "province",
      selectedTags: [],
      thsDefaultRegionCode: "000000000000",
      buttons: [
        { label: "省份", value: "province" },
        { label: "城市", value: "city" },
        { label: "区县", value: "county" }
      ],
      selectedButton: 0,
      // 当前数据
      regionTool: {
        thsDefaultLevel: "",
        // 默认首字母编码
        currentProvinceName: "",
        // 当前渲染省名称
        currentProvinceCode: "",
        // 当前渲染省编码
        currentCityName: "",
        // 当前渲染市名称
        currentCityCode: "",
        // 当前渲染市编码
        currentCountyName: "",
        // 当前渲染县名称
        currentCountyCode: "",
        // 当前渲染县编码
        currentTownCode: "",
        // 当前乡镇编码
        currentTownName: "",
        // 当前乡镇名称
        currentRegionCode: "",
        // 当前渲染行政区编码
        currentRegionName: "",
        // 当前渲染行政区名称
        currentRegionType: "",
        // 当前渲染行政区类型
        tempRegionCode: "",
        // 临时行政区编码，选中未执行查询
        allRegionMap: [],
        // 所有行政区map
        regionLevelShowList: [],
        // 行政区的树结构
        regionType: ""
      },
      // 区域列表dom
      list: null,
      areaName: "",
      // 区域名称
      areaCode: "",
      // 区域编码
      currentOperation: "添加",
      // 当前操作
      regionLevels: "1",
      currentRegionLevel: "1",
      loading: !1,
      addButtonsStatus: !0,
      isAdd: !1,
      currentRegionType: "",
      templateSelectedTags: [],
      deleteList: [],
      addList: [],
      disabled: !0
    }), v = (t) => {
      var o;
      e.regionTool.regionLevelShowList.forEach((a) => {
        a.isActive = !1;
      }), e.regionTool.regionLevelShowList.find((a) => a.REGION_CODE === t).isActive = !0;
      const n = document.getElementById(`${t}`);
      n && ((o = e.list) == null || o.scroll(0, (n == null ? void 0 : n.offsetTop) - e.list.offsetTop));
    }, N = (t) => {
      e.areaName = t, e.areaName === "" ? e.addButtonsStatus = !0 : e.selectedTags.length > 0 && (e.addButtonsStatus = !1);
    }, E = async (t) => {
      e.addList.push({ regionName: t.REGION_NAME, regionCode: t.REGION_CODE, areaCode: t.AREA_CODE }), !e.selectedTags.find((n) => n.regionCode === t.REGION_CODE) && (e.selectedTags.push({ regionName: t.REGION_NAME, regionCode: t.REGION_CODE, areaCode: t.AREA_CODE }), e.areaName !== "" && (e.addButtonsStatus = !1));
    }, C = () => {
      const t = Math.floor(Math.random() * 9) + 1, n = Math.floor(Math.random() * 1e4);
      return t * 1e4 + n;
    }, L = () => new Promise((t, n) => {
      s({
        url: r.getRegionListUrl,
        method: "post",
        data: d.stringify({ regionLevels: e.regionLevels })
      }).then((o) => {
        t(o);
      }).catch((o) => {
        n(o);
      });
    }), c = async () => {
      e.loading = !0;
      const t = await L();
      if (t.data.head.statusCode === "200" && t.data.data.length > 0) {
        e.loading = !1, e.regionTool.currentRegionCode = t.data.data[0].REGION_CODE, e.regionTool.currentRegionName = t.data.data[0].REGION_NAME, e.regionTool.regionLevelShowList = [], e.regionTool.allRegionMap = [];
        const n = [
          "A",
          "B",
          "C",
          "D",
          "E",
          "F",
          "G",
          "H",
          "I",
          "J",
          "K",
          "L",
          "M",
          "N",
          "O",
          "P",
          "Q",
          "R",
          "S",
          "T",
          "U",
          "V",
          "W",
          "X",
          "Y",
          "Z",
          "港澳"
        ];
        let o = [];
        o = t.data.data;
        for (let a = 0; a < o.length; a++)
          o[a] && o[a].REGION_PY && (o[a].CHILD = []), e.regionTool.allRegionMap[o[a].REGION_CODE] = o[a];
        for (let a = 0; a < o.length; a++) {
          const i = o[a].PARENT_REGION_CODE, l = o[a].REGION_PY.substring(0, 1);
          o[a] && (o[a].REGION_LEVEL === e.currentRegionLevel ? (e.regionTool.allRegionMap[l] || (e.regionTool.allRegionMap[l] = {}, e.regionTool.allRegionMap[l].CHILD = [], e.regionTool.allRegionMap[l].REGION_CODE = l, e.regionTool.allRegionMap[l].REGION_NAME = l), e.regionTool.allRegionMap[l].CHILD.push(o[a])) : e.regionTool.allRegionMap[i] && e.regionTool.allRegionMap[i].CHILD.push(o[a]));
        }
        for (let a = 0; a < n.length; a++) {
          const i = n[a];
          if (e.regionTool.allRegionMap[i] && e.regionTool.allRegionMap[i].CHILD != null && e.regionTool.allRegionMap[i].CHILD.length > 0) {
            const l = e.regionTool.allRegionMap[i];
            e.regionTool.regionLevelShowList.push(l);
          }
        }
      } else
        e.loading = !1;
    }, u = async (t, n) => {
      if (e.selectedTags = [], e.currentRegionType === t) {
        const { areaName: o, areaCode: a, areaLevel: i } = r.currentCategory;
        s({
          url: r.getChildElementsUrl,
          method: "post",
          data: d.stringify({
            areaCode: a
          })
        }).then((l) => {
          const { data: T } = l.data;
          e.selectedTags = T.map((m) => ({
            regionName: m.REGION_NAME,
            regionCode: m.REGION_CODE
          }));
        });
      }
      e.areaLevel = t, e.selectedButton = n, t === "province" ? (e.regionLevels = "1", e.currentRegionLevel = "1", e.disabled = !1) : t === "city" ? (e.regionLevels = "1,2", e.currentRegionLevel = "1", e.disabled = !0) : t === "county" && (e.regionLevels = "2,3", e.currentRegionLevel = "2", e.disabled = !0), await c();
    }, h = (t) => {
      e.deleteList.push(t), e.selectedTags = e.selectedTags.filter((n) => n.regionCode !== t.regionCode), e.selectedTags.length === 0 && (e.addButtonsStatus = !0);
    }, f = () => {
      e.currentOperation === "添加" && !e.isAdd && (e.selectedTags.forEach((t) => {
        s({
          url: r.deleteSingleRegionUrl,
          method: "post",
          data: d.stringify({ areaCode: e.areaCode, regionCode: t.regionCode })
        });
      }), e.selectedTags = []), g("back");
    }, y = async () => {
      e.isAdd = !0, e.areaName !== "" && (e.currentOperation === "修改" ? (e.currentRegionType !== e.areaLevel && e.templateSelectedTags.forEach(async (t) => {
        await s({
          url: r.deleteSingleRegionUrl,
          method: "post",
          data: d.stringify({ areaCode: e.areaCode, regionCode: t.regionCode })
        });
      }), e.deleteList.length && e.deleteList.forEach(async (t) => {
        await s({
          url: r.deleteSingleRegionUrl,
          method: "post",
          data: d.stringify({ areaCode: e.areaCode, regionCode: t.regionCode })
        });
      }), e.addList.length && e.addList.forEach(async (t) => {
        await s({
          url: r.addSingleRegionUrl,
          method: "post",
          data: d.stringify({ regionCode: t.regionCode, regionName: t.regionName, areaCode: e.areaCode })
        });
      }), await s({
        url: r.updateRegionUrl,
        method: "post",
        data: d.stringify({ areaCode: e.areaCode, areaName: e.areaName, areaLevel: e.areaLevel })
      }).then((t) => {
        t.data.data.result === 1 && (g("selected", { areaCode: e.areaCode, areaName: e.areaName, areaLevel: e.areaLevel }), f());
      })) : (await e.selectedTags.forEach((t) => {
        s({
          url: r.addSingleRegionUrl,
          method: "post",
          data: d.stringify({ regionCode: t.regionCode, regionName: t.regionName, areaCode: e.areaCode })
        }).then((n) => {
        });
      }), await s({
        url: r.addRegionUrl,
        method: "post",
        data: d.stringify({ areaCode: e.areaCode, areaName: e.areaName, areaLevel: e.areaLevel })
      }).then((t) => {
        t.data.data.result === 1 && (g("selected", { areaCode: e.areaCode, areaName: e.areaName, areaLevel: e.areaLevel }), f());
      })));
    }, R = () => {
      if (JSON.stringify(r.currentCategory) !== "{}") {
        const { areaName: t, areaCode: n, areaLevel: o } = r.currentCategory;
        e.areaCode = n, e.areaName = t, e.areaLevel = o, e.currentOperation = "修改", e.addButtonsStatus = !1, e.currentRegionType = o, e.areaLevel === "province" ? (e.selectedButton = 0, e.regionLevels = "1") : e.areaLevel === "city" ? (r.defaultRegionLevel === "province" ? (e.buttons = e.buttons.slice(1), e.selectedButton = 0) : e.selectedButton = 1, e.regionLevels = "1,2") : (r.defaultRegionLevel === "city" ? (e.buttons = e.buttons.slice(2), e.selectedButton = 0) : e.selectedButton = 2, e.regionLevels = "2,3"), s({
          url: r.getChildElementsUrl,
          method: "post",
          data: d.stringify({
            areaCode: n
          })
        }).then((a) => {
          const { data: i } = a.data;
          e.selectedTags = i.map((l) => ({
            regionName: l.REGION_NAME,
            regionCode: l.REGION_CODE
          })), e.templateSelectedTags = [...e.selectedTags];
        }), u(e.areaLevel, e.selectedButton);
      } else
        e.areaName = "", e.areaCode = String(C()), r.defaultRegionLevel === "province" ? (e.buttons = e.buttons.slice(1), e.selectedButton = 0, e.areaLevel = "city") : r.defaultRegionLevel === "city" ? (e.buttons = e.buttons.slice(2), e.selectedButton = 0, e.areaLevel = "county") : (e.areaLevel = "province", e.disabled = !1), u(e.areaLevel, e.selectedButton);
    };
    return S(() => {
      R(), c();
    }), {
      ...M(e),
      getData: L,
      handleBack: f,
      selectLitter: v,
      checkRegion: E,
      handleCloseTag: h,
      updateArea: y,
      getRandomAreaCode: C,
      initCurrentArea: R,
      getRegionData: c,
      handleAdd: N,
      selectButton: u
    };
  }
});
export {
  A as default
};
