// @ths/design/es/components/area-selector-custom-data/src/custom-list.vue2.js
import { defineComponent as g, reactive as C, computed as y, watch as c, onMounted as w, toRefs as L } from "vue";
import i from "axios";
import u from "qs";
const b = g({
  name: "List",
  props: {
    defaultArea: {
      type: Object,
      default: () => {
      }
    },
    // 当前选中区域
    selectedArea: {
      type: Object,
      default: () => {
      }
    },
    selectedCode: {
      type: String,
      default: ""
    },
    isEdit: {
      type: Boolean,
      default: !0
    },
    // 获取自定义区域列表
    getListUrl: {
      type: String,
      default: ""
    },
    // 删除自定义区域列表
    deleteListUrl: {
      type: String,
      default: ""
    },
    // 获取子级区域
    getChildElementsUrl: {
      type: String,
      default: ""
    },
    // 是否打开弹窗
    visible: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["update", "select"],
  setup(e, { emit: r }) {
    const a = C({
      customAreaList: [],
      childrenList: [],
      showFold: {
        showMore: !1,
        buttonName: "展开",
        buttonIcon: "arrow-down"
      },
      currentArea: y(() => JSON.stringify(e.selectedArea) !== "{}" ? e.selectedArea : e.defaultArea)
    }), f = () => {
      a.showFold.showMore = !a.showFold.showMore, a.showFold.buttonName = a.showFold.showMore ? "收起" : "展开";
    }, s = (t) => {
      i({
        url: e.getChildElementsUrl,
        method: "post",
        data: u.stringify({
          areaCode: t.areaCode
        })
      }).then((d) => {
        const { data: l } = d.data;
        a.childrenList = l.map((n) => ({
          regionName: n.REGION_NAME,
          regionCode: n.REGION_CODE
        }));
      }), r("select", t);
    }, m = () => {
      r("update");
    }, o = () => {
      i({
        url: e.getListUrl,
        method: "get"
      }).then((t) => {
        const { data: d } = t.data;
        a.customAreaList = d.map((l) => ({
          areaName: l.AREA_NAME,
          areaCode: l.AREA_CODE,
          areaLevel: l.AREA_LEVEL
        }));
      });
    }, h = async (t) => {
      a.customAreaList = a.customAreaList.filter((d) => d.areaCode !== t.areaCode), a.childrenList = [], await i({
        url: e.deleteListUrl,
        method: "post",
        data: u.stringify({
          areaCode: t.areaCode
        })
      }), await o();
    }, A = (t) => {
      r("update", t);
    };
    return c(() => e.getListUrl, () => {
      o();
    }), c(() => e.visible, (t) => {
      t && (o(), Object.prototype.hasOwnProperty.call(e.selectedArea, "areaName") && e.selectedArea.areaCode !== "" && s(e.selectedArea));
    }, { immediate: !0 }), w(() => {
      o(), Object.prototype.hasOwnProperty.call(e.selectedArea, "areaName") && e.selectedArea.areaCode !== "" && s(e.selectedArea);
    }), {
      ...L(a),
      toggleFoldState: f,
      handleAdd: m,
      handleClick: s,
      close: h,
      edit: A,
      getCategoryDate: o
    };
  }
});
export {
  b as default
};
