// @ths/design/es/components/area-selector-custom-data/src/custom-add.vue.js
import b from "./custom-add.vue2.js";
import { resolveComponent as _, resolveDirective as O, openBlock as s, createElementBlock as n, createElementVNode as o, createVNode as u, withCtx as c, createTextVNode as k, toDisplayString as a, Fragment as d, renderList as r, normalizeClass as v, createBlock as I, withDirectives as D } from "vue";
import "./custom-add.vue3.js";
import R from "../../../_virtual/_plugin-vue_export-helper.js";
const y = { class: "add-wrapper" }, G = { class: "header" }, A = { class: "header-content" }, B = { class: "search" }, $ = { class: "add-btn" }, L = { class: "tab" }, T = { class: "button-group" }, w = ["onClick"], M = { class: "selected-list" }, S = { class: "data-list" }, V = { class: "area-litter" }, z = ["onClick"], F = {
  ref: "list",
  class: "area-city-list",
  "element-loading-background": "rgba(255, 255, 255, .0)"
}, H = ["id"], j = { class: "citys" }, q = ["onClick"], J = { class: "city_text" }, K = { class: "countys" }, P = ["onClick"];
function Q(e, p, U, W, X, Y) {
  const C = _("el-input"), E = _("el-button"), m = _("el-page-header"), f = _("el-tag"), h = _("el-tooltip"), N = O("loading");
  return s(), n("div", y, [
    o("div", G, [
      u(m, { onBack: e.handleBack }, {
        title: c(() => p[0] || (p[0] = [
          o("span", { class: "back" }, "返回", -1)
        ])),
        content: c(() => [
          o("div", A, [
            o("div", B, [
              u(C, {
                type: "text",
                "model-value": e.areaName,
                clearable: "",
                onInput: e.handleAdd
              }, null, 8, ["model-value", "onInput"])
            ]),
            o("div", $, [
              u(E, {
                disabled: e.addButtonsStatus,
                onClick: e.updateArea
              }, {
                default: c(() => [
                  k(a(e.currentOperation), 1)
                ]),
                _: 1
              }, 8, ["disabled", "onClick"])
            ])
          ])
        ]),
        _: 1
      }, 8, ["onBack"])
    ]),
    p[1] || (p[1] = o("div", { class: "line" }, null, -1)),
    o("div", L, [
      o("div", T, [
        (s(!0), n(d, null, r(e.buttons, (t, l) => (s(), n("button", {
          key: l,
          class: v(["button", { selected: e.selectedButton === l }]),
          onClick: (i) => e.selectButton(t.value, l)
        }, a(t.label), 11, w))), 128))
      ]),
      o("div", M, [
        (s(!0), n(d, null, r(e.selectedTags, (t) => (s(), I(f, {
          key: t.regionCode,
          class: "tag",
          size: "large",
          closable: "",
          round: "",
          onClose: (l) => e.handleCloseTag(t)
        }, {
          default: c(() => [
            k(a(t.regionName), 1)
          ]),
          _: 2
        }, 1032, ["onClose"]))), 128))
      ])
    ]),
    o("div", S, [
      o("div", V, [
        (s(!0), n(d, null, r(e.regionTool.regionLevelShowList, (t) => (s(), n("a", {
          key: t,
          href: "javascript:void(0)",
          class: v(["litter", { "litter-active": t.isActive }]),
          onClick: (l) => e.selectLitter(t.REGION_CODE)
        }, a(t.REGION_CODE), 11, z))), 128))
      ]),
      D((s(), n("div", F, [
        (s(!0), n(d, null, r(e.regionTool.regionLevelShowList, (t) => (s(), n("div", {
          id: t.REGION_CODE,
          key: t,
          class: "row"
        }, [
          o("div", {
            class: v(["tag", { "tag-active": t.isActive }])
          }, a(t.REGION_CODE), 3),
          o("div", j, [
            (s(!0), n(d, null, r(t.CHILD, (l) => (s(), n("div", {
              key: l,
              class: "city-container"
            }, [
              o("div", {
                class: v(["city", {
                  "city-active": !!e.selectedTags.find((i) => i.regionCode === l.REGION_CODE),
                  "city-province": !e.disabled
                }]),
                onClick: (i) => e.checkRegion(l)
              }, [
                u(h, {
                  class: "box-item",
                  effect: "dark",
                  content: l.REGION_NAME,
                  placement: "top"
                }, {
                  default: c(() => [
                    o("div", J, a(l.REGION_NAME), 1)
                  ]),
                  _: 2
                }, 1032, ["content"])
              ], 10, q),
              o("div", K, [
                (s(!0), n(d, null, r(l.CHILD, (i) => (s(), n("div", {
                  key: i,
                  class: "county",
                  onClick: (g) => e.checkRegion(i)
                }, [
                  u(h, {
                    class: "box-item",
                    effect: "dark",
                    content: i.REGION_NAME,
                    placement: "top"
                  }, {
                    default: c(() => [
                      o("div", {
                        class: v(["county_text", { "county-active": !!e.selectedTags.find((g) => g.regionCode === i.REGION_CODE) }])
                      }, a(i.REGION_NAME), 3)
                    ]),
                    _: 2
                  }, 1032, ["content"])
                ], 8, P))), 128))
              ])
            ]))), 128))
          ])
        ], 8, H))), 128))
      ])), [
        [N, e.loading]
      ])
    ])
  ]);
}
const oe = /* @__PURE__ */ R(b, [["render", Q]]);
export {
  oe as default
};
