// @ths/design/es/components/area-selector-custom-data/src/index.vue.js
import N from "./index.vue2.js";
import { resolveComponent as m, openBlock as t, createElementBlock as s, normalizeClass as y, createElementVNode as r, toDisplayString as u, createVNode as c, withCtx as i, createTextVNode as g, Fragment as k, renderList as p, createBlock as C, createCommentVNode as n } from "vue";
import "./index.vue3.js";
import w from "../../../_virtual/_plugin-vue_export-helper.js";
const G = {
  key: 0,
  class: "area-selector"
}, L = { class: "area-breadcrumb" }, I = { class: "last-region" }, T = {
  key: 1,
  class: "area-selector"
}, U = { class: "area-breadcrumb" }, $ = {
  key: 2,
  class: "area-box"
}, B = {
  key: 0,
  class: "area-title"
}, _ = { class: "area-breadcrumb custom-layout" }, j = {
  key: 1,
  class: "area-title"
}, W = {
  key: 2,
  class: "add-list"
}, K = {
  key: 3,
  class: "area-container"
}, P = {
  key: 0,
  class: "area-button"
}, M = { class: "button-group" }, F = ["onClick"], V = { class: "handle-button" }, q = {
  key: 1,
  class: "custom-area-list"
}, z = {
  key: 2,
  class: "custom-area-list"
}, H = {
  key: 3,
  class: "custom-area-list"
}, J = {
  key: 4,
  class: "area-litter"
}, Q = ["onClick"], X = {
  key: 5,
  ref: "list",
  class: "area-city-list"
}, Y = ["id"], Z = { class: "citys" }, x = ["onClick"], ee = {
  key: 0,
  class: "city_text"
}, te = { class: "city_text" }, oe = { class: "countys" }, se = ["onClick"], le = {
  key: 0,
  class: "county_text"
}, re = { class: "county_text" }, ae = {
  key: 6,
  class: "footer custom-layout"
}, ne = { class: "handle-button" };
function ie(e, l, de, ue, ce, me) {
  var f;
  const h = m("el-button"), v = m("el-breadcrumb-item"), b = m("el-breadcrumb"), O = m("custom-add"), A = m("custom-list"), S = m("key-area"), R = m("key-watershed-group"), D = m("key-watershed"), E = m("el-tooltip");
  return t(), s("div", {
    class: y(["t-area-selector-custom-data", e.domOnlyClass])
  }, [
    e.isCustomLayout ? (t(), s("div", G, [
      r("div", {
        class: "area-icon",
        onClick: l[0] || (l[0] = (...o) => e.switchSelector && e.switchSelector(...o))
      }),
      r("div", L, [
        r("span", I, u((f = e.breadcrumbObj.path[e.breadcrumbObj.path.length - 1]) == null ? void 0 : f.name), 1)
      ])
    ])) : (t(), s("div", T, [
      r("div", {
        class: "area-icon",
        onClick: l[1] || (l[1] = (...o) => e.switchSelector && e.switchSelector(...o))
      }),
      c(h, {
        class: "selector",
        onClick: e.switchSelector
      }, {
        default: i(() => l[5] || (l[5] = [
          g(" 区域选择 ")
        ])),
        _: 1,
        __: [5]
      }, 8, ["onClick"]),
      r("div", U, [
        c(b, { separator: e.separator }, {
          default: i(() => [
            (t(!0), s(k, null, p(e.breadcrumbObj.path, (o, a) => (t(), C(v, {
              key: o,
              class: y(a + 1 === e.breadcrumbObj.path.length ? "last-region" : ""),
              onClick: (d) => e.select(o, a)
            }, {
              default: i(() => [
                g(u(o.name), 1)
              ]),
              _: 2
            }, 1032, ["class", "onClick"]))), 128))
          ]),
          _: 1
        }, 8, ["separator"])
      ])
    ])),
    e.isShowBox ? (t(), s("div", $, [
      e.isCustomLayout ? (t(), s("div", B, [
        r("div", _, [
          c(b, { separator: e.separator }, {
            default: i(() => [
              (t(!0), s(k, null, p(e.breadcrumbObj.path, (o, a) => (t(), C(v, {
                key: o,
                class: y(a + 1 === e.breadcrumbObj.path.length ? "last-region" : ""),
                onClick: (d) => e.select(o, a)
              }, {
                default: i(() => [
                  g(u(o.name), 1)
                ]),
                _: 2
              }, 1032, ["class", "onClick"]))), 128))
            ]),
            _: 1
          }, 8, ["separator"])
        ]),
        r("div", {
          class: "title-close",
          onClick: l[2] || (l[2] = (...o) => e.close && e.close(...o))
        }, l[6] || (l[6] = [
          r("span", { class: "close-icon" }, null, -1)
        ]))
      ])) : (t(), s("div", j, [
        l[8] || (l[8] = r("div", { class: "title-icon" }, null, -1)),
        l[9] || (l[9] = r("div", { class: "title-text" }, " 区域查询 ", -1)),
        r("div", {
          class: "title-close",
          onClick: l[3] || (l[3] = (...o) => e.close && e.close(...o))
        }, l[7] || (l[7] = [
          r("span", { class: "close-icon" }, null, -1)
        ]))
      ])),
      e.isShowEditPage ? (t(), s("div", W, [
        c(O, {
          "current-category": e.currentCategory,
          "get-region-list-url": e.customGetDataListUrl,
          "add-single-region-url": e.customAddSingleUrl,
          "delete-single-region-url": e.customDeleteSingleUrl,
          "update-region-url": e.customUpdateUrl,
          "add-region-url": e.customAddUrl,
          "get-child-elements-url": e.customGetChildElementsUrl,
          "default-region-level": e.defaultRegionLevel,
          onBack: l[4] || (l[4] = (o) => e.isShowEditPage = !1),
          onSelected: e.customSelectedArea
        }, null, 8, ["current-category", "get-region-list-url", "add-single-region-url", "delete-single-region-url", "update-region-url", "add-region-url", "get-child-elements-url", "default-region-level", "onSelected"])
      ])) : n("", !0),
      e.isShowEditPage ? n("", !0) : (t(), s("div", K, [
        e.isShowEditPage ? n("", !0) : (t(), s("div", P, [
          r("div", M, [
            (t(!0), s(k, null, p(e.buttons, (o, a) => (t(), s(k, { key: a }, [
              o.isShow ? (t(), s("button", {
                key: 0,
                class: y(["button", { selected: e.selectedButton === a }]),
                onClick: (d) => e.selectButton(o.value, a)
              }, u(o.label), 11, F)) : n("", !0)
            ], 64))), 128))
          ]),
          r("div", V, [
            c(h, {
              label: "cx",
              class: "search",
              onClick: e.search
            }, {
              default: i(() => l[10] || (l[10] = [
                g(" 查询 ")
              ])),
              _: 1,
              __: [10]
            }, 8, ["onClick"]),
            c(h, {
              class: "reset",
              label: "cz",
              onClick: e.resetRegion
            }, {
              default: i(() => l[11] || (l[11] = [
                g(" 重置 ")
              ])),
              _: 1,
              __: [11]
            }, 8, ["onClick"])
          ])
        ])),
        e.currentAreaType === "custom" && !e.isShowEditPage ? (t(), s("div", q, [
          c(A, {
            "is-edit": e.isEdit,
            "default-area": e.defaultArea,
            "selected-area": e.currentCustomArea,
            "selected-code": e.regionTool.currentRegionCode,
            "get-list-url": e.customGetListUrl,
            "delete-list-url": e.customDeleteListUrl,
            "get-child-elements-url": e.customGetChildElementsUrl,
            visible: e.isShowCustomArea,
            onUpdate: e.updateCategoryArea,
            onSelect: e.customSelected
          }, null, 8, ["is-edit", "default-area", "selected-area", "selected-code", "get-list-url", "delete-list-url", "get-child-elements-url", "visible", "onUpdate", "onSelect"])
        ])) : n("", !0),
        e.currentAreaType === "qy" ? (t(), s("div", z, [
          c(S, {
            "is-key-area": e.isKeyAreaList,
            "default-area": e.defaultArea,
            "selected-area": e.currentKeyArea,
            url: e.keyAreaUrl,
            onSelected: e.selectedKeyArea
          }, null, 8, ["is-key-area", "default-area", "selected-area", "url", "onSelected"])
        ])) : n("", !0),
        e.currentAreaType === "ly" ? (t(), s("div", H, [
          e.isKeyWatershedGroup ? (t(), C(R, {
            key: 0,
            "default-area": e.defaultArea,
            "root-code": e.keyWatershedRootCode,
            "selected-area": e.currentKeyWatershed,
            url: e.keyWatershedUrl,
            onSelected: e.selectedKeyWatershed
          }, null, 8, ["default-area", "root-code", "selected-area", "url", "onSelected"])) : (t(), C(D, {
            key: 1,
            "default-area": e.defaultArea,
            "root-code": e.keyWatershedRootCode,
            "selected-area": e.currentKeyWatershed,
            url: e.keyWatershedUrl,
            onSelected: e.selectedKeyWatershed,
            onOnDataLoaded: e.handleDataLoaded
          }, null, 8, ["default-area", "root-code", "selected-area", "url", "onSelected", "onOnDataLoaded"]))
        ])) : n("", !0),
        e.showInitialSort && e.currentAreaType !== "custom" && e.currentAreaType !== "qy" && e.currentAreaType !== "ly" || e.currentAreaType === "custom" && e.isShowEditPage ? (t(), s("div", J, [
          (t(!0), s(k, null, p(e.regionTool.regionLevelShowList, (o) => (t(), s("a", {
            key: o,
            href: "javascript:void(0)",
            class: y(["litter", e.breadcrumbObj.code === o.REGION_CODE && e.regionTool.tempRegionCode != e.thsDefaultRegionCode ? "litter-active" : ""]),
            onClick: (a) => e.selectLitter(o.REGION_CODE)
          }, u(o.REGION_CODE), 11, Q))), 128))
        ])) : n("", !0),
        e.currentAreaType !== "custom" && e.currentAreaType !== "qy" && e.currentAreaType !== "ly" || e.currentAreaType === "custom" && e.isShowEditPage ? (t(), s("div", X, [
          (t(!0), s(k, null, p(e.regionTool.regionLevelShowList, (o) => (t(), s("div", {
            id: o.REGION_CODE,
            key: o,
            class: "row"
          }, [
            e.showInitialSort ? (t(), s("div", {
              key: 0,
              class: y(["tag", e.breadcrumbObj.code === o.REGION_CODE && e.regionTool.tempRegionCode != e.thsDefaultRegionCode ? "tag-active" : ""])
            }, u(o.REGION_CODE), 3)) : n("", !0),
            r("div", Z, [
              (t(!0), s(k, null, p(o.CHILD, (a) => (t(), s("div", {
                key: a,
                class: "city-container"
              }, [
                r("div", {
                  class: y(["city", e.regionTool.tempRegionCode === a.REGION_CODE ? "city-active" : ""]),
                  onClick: (d) => e.checkRegion(a.REGION_CODE)
                }, [
                  e.isCustomLayout ? (t(), s("div", ee, u(a.REGION_NAME), 1)) : (t(), C(E, {
                    key: 1,
                    class: "box-item",
                    content: a.REGION_NAME,
                    placement: "top-start"
                  }, {
                    default: i(() => [
                      r("div", te, u(a.REGION_NAME), 1)
                    ]),
                    _: 2
                  }, 1032, ["content"]))
                ], 10, x),
                r("div", oe, [
                  (t(!0), s(k, null, p(a.CHILD, (d) => (t(), s("div", {
                    key: d,
                    class: y(["county", e.regionTool.tempRegionCode === d.REGION_CODE ? "county-active" : ""]),
                    onClick: (ye) => e.checkRegion(d.REGION_CODE)
                  }, [
                    e.isCustomLayout ? (t(), s("div", le, u(d.REGION_NAME), 1)) : (t(), C(E, {
                      key: 1,
                      class: "box-item",
                      content: d.REGION_NAME,
                      placement: "top"
                    }, {
                      default: i(() => [
                        r("div", re, u(d.REGION_NAME), 1)
                      ]),
                      _: 2
                    }, 1032, ["content"]))
                  ], 10, se))), 128))
                ])
              ]))), 128))
            ])
          ], 8, Y))), 128))
        ], 512)) : n("", !0),
        e.isCustomLayout ? (t(), s("div", ae, [
          r("div", ne, [
            c(h, {
              class: "reset",
              label: "cz",
              onClick: e.resetRegion
            }, {
              default: i(() => l[12] || (l[12] = [
                g(" 重置 ")
              ])),
              _: 1,
              __: [12]
            }, 8, ["onClick"]),
            c(h, {
              label: "cx",
              class: "search",
              onClick: e.search
            }, {
              default: i(() => l[13] || (l[13] = [
                g(" 确定 ")
              ])),
              _: 1,
              __: [13]
            }, 8, ["onClick"])
          ])
        ])) : n("", !0)
      ]))
    ])) : n("", !0),
    l[14] || (l[14] = r("div", { class: "style-container" }, null, -1))
  ], 2);
}
const he = /* @__PURE__ */ w(N, [["render", ie]]);
export {
  he as default
};
