// @ths/design/es/components/area-selector-custom-data/src/index.vue2.js
import { defineComponent as $r, computed as b, reactive as pr, watch as E, onMounted as rr, nextTick as lr, toRefs as Rr } from "vue";
import Cr from "axios";
import Ar from "qs";
import Er from "./custom-list.vue.js";
import Nr from "./custom-add.vue.js";
import Or from "./key-area.vue.js";
import Ir from "./key-watershed.vue.js";
import wr from "./key-watershed-group.vue.js";
import { getDomRandomClass as xr, setStyle as Sr, getCssPropertyString as r } from "../../../utils/index.js";
const _r = {
  REGION_NAME: "REGION_NAME",
  REGION_CODE: "REGION_CODE",
  REGION_PY: "REGION_PY",
  PARENT_REGION_CODE: "PARENT_REGION_CODE"
}, Yr = $r({
  name: "TAreaSelectorCustomData",
  components: {
    CustomList: Er,
    CustomAdd: Nr,
    KeyArea: Or,
    KeyWatershed: Ir,
    keyWatershedGroup: wr
  },
  props: {
    // 面包屑箭头样子
    separatorType: {
      type: String,
      default: ""
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 标题
        areaSelector: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域选择按钮
        selector: {
          default: {},
          hover: {},
          focus: {},
          active: {}
        },
        // 最后区域高亮
        lastRegion: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询容器
        areaBox: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询标题
        areaTitle: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询标题图标
        titleIcon: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域查询标题关闭按钮
        titleClose: {
          default: {},
          hover: {},
          active: {}
        },
        // 区域容器
        areaContainer: {
          default: {},
          hover: {},
          active: {}
        },
        // 按钮容器
        areaButton: {
          default: {},
          hover: {},
          active: {}
        },
        // radio按钮
        areaSwitch: {
          default: {},
          hover: {},
          active: {}
        },
        // 操作按钮
        handleButton: {
          default: {},
          hover: {},
          focus: {},
          active: {}
        },
        // 选中高亮
        Active: {
          default: {},
          hover: {}
        }
      })
    },
    selected: {
      type: String,
      default: ""
    },
    // 行政区划顶级编码
    thsDefaultRegionCode: {
      type: String,
      default: "000000000000"
    },
    // 行政区划默认区域编码
    regionCode: {
      type: String,
      default: ""
    },
    // 行政区域默认类型
    defaultRegionLevel: {
      type: String,
      default: "city"
    },
    // 行政区划数据别名
    alias: {
      type: Object,
      default: () => _r
    },
    // 行政区划数据
    divisionData: {
      type: Array,
      default: () => []
    },
    // 重点区域数据
    keyAreaData: {
      type: Array,
      default: () => []
    },
    // 重点流域数据
    keyWatershedData: {
      type: Array,
      default: () => []
    },
    // 自定义区域数据
    customAreaData: {
      type: Array,
      default: () => []
    },
    // 是否展示首字母排序
    showInitialSort: {
      type: Boolean,
      default: !0
    },
    // 是否显示行政区划
    isShowDivision: {
      type: Boolean,
      default: !0
    },
    // 是否显示重点区域
    isShowKeyArea: {
      type: Boolean,
      default: !0
    },
    // 是否显示重点流域
    isShowKeyWatershed: {
      type: Boolean,
      default: !0
    },
    // 是否显示自定义区域
    isShowCustomArea: {
      type: Boolean,
      default: !0
    },
    // 是否具有编辑、删除权限
    isEdit: {
      type: Boolean,
      default: !0
    },
    // 重点流域顶级编码
    keyWatershedRootCode: {
      type: String,
      default: "000000000000"
    },
    // 是否重点区域列表
    isKeyAreaList: {
      type: String,
      default: "1"
    },
    // 当前选中类型
    selectedType: {
      type: String,
      default: "region"
    },
    // 默认选中区域
    defaultArea: {
      type: Object,
      default: () => {
      }
    },
    // 行政区划请求url
    divisionUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=91e1f92fa791802de1ea4d5245915021&token=dcp"
    },
    // 重点流域请求url
    keyWatershedUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=8bdb487456849c97a8e74152ca1a8439&token=dcp"
    },
    // 重点区域请求url
    keyAreaUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=29bbddede70f5293c79109d8edc2dcac&token=dcp"
    },
    // 获取自定义区域列表请求url
    customGetListUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=8e24536413f30eebe085bf105574b5f1&token=dcp"
    },
    // 删除自定义区域列表请求url
    customDeleteListUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=e8ea54caf8d5a7bb517e6cad0913c33d&token=dcp"
    },
    // 获取自定义区域子集区域url
    customGetChildElementsUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=167a6c03c8b6bd3ed71187d47ec5f7b3&token=dcp"
    },
    // 获取自定义区域数据列表url
    customGetDataListUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=e480d2c59e99935a33bed17d0aa56b4b&token=dcp"
    },
    // 添加单个区域url
    customAddSingleUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=92431ad0be517aa026de9153d27e5bfc&token=dcp"
    },
    // 删除单个区域
    customDeleteSingleUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=de27b19c3207ad79b003a5cb7dc98ce1&token=dcp"
    },
    // 修改区域url
    customUpdateUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=5b36e65d1fda6ade7de55ca795566ae4&token=dcp"
    },
    // 添加区域
    customAddUrl: {
      type: String,
      default: "http://192.168.0.202:7777/service/serviceinterface/search/run.action?interfaceId=6497a42ee6aef7b523f445b72b95ef52&token=dcp"
    },
    // 重点流域字母分组
    isKeyWatershedGroup: {
      type: Boolean,
      default: !1
    },
    // 行政区划默认选中区域
    defaultAreaCode: {
      type: String,
      default: "130100000000"
    },
    // 是否特殊布局
    isCustomLayout: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["search", "resetRegion", "select"],
  setup(c, { emit: N }) {
    const t = b(() => c.cStyle), o = xr("area-selector"), e = pr({
      separator: "",
      // 面包屑箭头
      // 区域选择的显隐
      isShowBox: !1,
      // 区域列表dom
      list: null,
      // 默认选中类型
      selected: c.selected,
      // 默认渲染顶级区域编码
      thsDefaultRegionCode: b(() => c.thsDefaultRegionCode),
      // 当前选中区域编码
      REGION_CODE: "",
      // 当前数据
      regionTool: {
        thsDefaultLevel: "",
        // 默认首字母编码
        currentProvinceName: "",
        // 当前渲染省名称
        currentProvinceCode: "",
        // 当前渲染省编码
        currentCityName: "",
        // 当前渲染市名称
        currentCityCode: "",
        // 当前渲染市编码
        currentCountyName: "",
        // 当前渲染县名称
        currentCountyCode: "",
        // 当前渲染县编码
        currentTownCode: "",
        // 当前乡镇编码
        currentTownName: "",
        // 当前乡镇名称
        currentRegionCode: "",
        // 当前渲染行政区编码
        currentRegionName: "",
        // 当前渲染行政区名称
        currentRegionType: "",
        // 当前渲染行政区类型
        tempRegionCode: "",
        // 临时行政区编码，选中未执行查询
        allRegionMap: [],
        // 所有行政区map
        regionLevelShowList: [],
        // 行政区的树结构
        regionType: "",
        defaultRegionCode: "",
        defaultRegionName: "",
        shortName: ""
      },
      // 面包屑数据
      breadcrumbObj: {
        path: [{ name: "", code: c.regionCode === "" ? c.thsDefaultRegionCode : c.regionCode }],
        code: ""
      },
      currentAreaType: "xz",
      // 新增、编辑页面
      isShowEditPage: !1,
      // 当前自定义区域
      currentCategory: {},
      // 当前选中自定义区域
      divisionData: [],
      currentCustomArea: {},
      selectCustomAreaCode: "",
      currentKeyArea: {},
      selectKeyAreaCode: "",
      currentKeyWatershed: {},
      selectKeyWatershedCode: "",
      buttons: [
        { label: "行政区划", value: "xz", isShow: b(() => c.isShowDivision) },
        { label: "重点区域", value: "qy", isShow: b(() => c.isShowKeyArea) },
        { label: "重点流域", value: "ly", isShow: b(() => c.isShowKeyWatershed) },
        { label: "自定义区域", value: "custom", isShow: b(() => c.isShowCustomArea) }
      ],
      selectedButton: 0,
      currentSelectedType: "region",
      // 已选择类型
      searchAreaType: "region",
      // 已查询区域类型
      isShowCustomArea: !1,
      watershedList: []
    }), cr = () => {
      let a = "";
      return t.value.wrapper && (a += Sr(t.value.wrapper, `.${o}`)), a;
    }, nr = () => {
      var l, n, d, i, u, s, m, f, g, h, T, y, $, p, R, C, A, S, B, M, L, j, P, K, H, Y, k, U, W, z, q, F, J, Z, Q, V, X, ee, ae, te, oe, re, le, ce, ne, ie, ue, se, de, me, fe, be, ve, ge, he, Te, ye, $e, pe, Re, Ce, Ae, Ee, Ne, Oe, Ie, we, xe, Se, _e, De, Ge, Be, Me, Le, je, Pe, Ke, He, Ye, ke, Ue, We, ze, qe, Fe, Je, Ze, Qe, Ve, Xe, ea, aa, ta, oa, ra, la, ca, na, ia, ua, sa, da, ma, fa, ba, va, ga, ha, Ta, ya, $a, pa, Ra, Ca, Aa, Ea, Na, Oa, Ia, wa, xa, Sa, _a, Da, Ga, Ba, Ma, La, ja, Pa, Ka, Ha, Ya, ka, Ua, Wa, za, qa, Fa, Ja, Za, Qa, Va, Xa, et, at, tt, ot, rt, lt, ct, nt, it, ut, st, dt, mt, ft, bt, vt, gt, ht, Tt, yt, $t, pt, Rt, Ct, At, Et, Nt, Ot, It, wt, xt, St, _t, Dt, Gt, Bt, Mt, Lt, jt, Pt, Kt, Ht, Yt, kt, Ut, Wt, zt, qt, Ft, Jt, Zt, Qt, Vt, Xt, eo, ao, to, oo, ro, lo, co, no, io, uo, so, mo, fo, bo, vo, go, ho, To, yo, $o, po, Ro, Co, Ao, Eo, No, Oo, Io, wo, xo, So, _o, Do, Go, Bo, Mo, Lo, jo, Po, Ko, Ho, Yo, ko, Uo, Wo, zo, qo, Fo, Jo, Zo, Qo, Vo, Xo, er, ar, tr, or;
      let a = "";
      return (n = (l = t.value) == null ? void 0 : l.areaIcon) != null && n.default && (a += `.${o}.t-area-selector-custom-data .area-selector .area-icon {
          ${r((i = (d = t.value) == null ? void 0 : d.areaIcon) == null ? void 0 : i.default)}
        }`), (s = (u = t.value) == null ? void 0 : u.areaIcon) != null && s.hover && (a += `.${o}.t-area-selector-custom-data .area-selector .area-icon:hover {
          ${r((f = (m = t.value) == null ? void 0 : m.areaIcon) == null ? void 0 : f.hover)}
        }`), (h = (g = t.value) == null ? void 0 : g.areaSelector) != null && h.default && (a += `.${o}.t-area-selector-custom-data .area-selector {
          ${r((T = t.value) == null ? void 0 : T.areaSelector.default)}
        }`), ($ = (y = t.value) == null ? void 0 : y.areaSelector) != null && $.hover && (a += `.${o}.t-area-selector-custom-data .area-selector:hover {
          ${r((p = t.value) == null ? void 0 : p.areaSelector.hover)}
        }`), (C = (R = t.value) == null ? void 0 : R.areaSelector) != null && C.active && (a += `.${o}.t-area-selector-custom-data .area-selector.active {
          ${r((A = t.value) == null ? void 0 : A.areaSelector.active)}
        }`), (B = (S = t.value) == null ? void 0 : S.selector) != null && B.default && (a += `.${o}.t-area-selector-custom-data .area-selector .el-button{
          ${r((M = t.value) == null ? void 0 : M.selector.default)}
        }`), (j = (L = t.value) == null ? void 0 : L.selector) != null && j.hover && (a += `.${o}.t-area-selector-custom-data .area-selector .el-button:hover {
          ${r((P = t.value) == null ? void 0 : P.selector.hover)}
        }`), (H = (K = t.value) == null ? void 0 : K.selector) != null && H.focus && (a += `.${o}.t-area-selector-custom-data .area-selector .el-button:focus {
          ${r((Y = t.value) == null ? void 0 : Y.selector.focus)}
        }`), (U = (k = t.value) == null ? void 0 : k.selector) != null && U.active && (a += `.${o}.t-area-selector-custom-data .area-selector .el-button:active {
          ${r((W = t.value) == null ? void 0 : W.selector.active)}
        }`), (q = (z = t.value) == null ? void 0 : z.breadCrumbArrow) != null && q.default && (a += `.${o}.t-area-selector-custom-data .area-selector .area-breadcrumb .el-breadcrumb__item .el-breadcrumb__separator {
          ${r((F = t.value) == null ? void 0 : F.breadCrumbArrow.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-breadcrumb .el-breadcrumb__item .el-breadcrumb__separator {
          ${r((J = t.value) == null ? void 0 : J.breadCrumbArrow.default)}
        }`), (Q = (Z = t.value) == null ? void 0 : Z.lastRegion) != null && Q.default && (a += `.${o}.t-area-selector-custom-data .area-selector .area-breadcrumb .el-breadcrumb__item.last-region .el-breadcrumb__inner{
          ${r((V = t.value) == null ? void 0 : V.lastRegion.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-selector .area-breadcrumb .last-region{
          ${r((X = t.value) == null ? void 0 : X.lastRegion.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-title .area-breadcrumb .el-breadcrumb__item.last-region .el-breadcrumb__inner{
          ${r((ee = t.value) == null ? void 0 : ee.lastRegion.default)}
        }`), (te = (ae = t.value) == null ? void 0 : ae.lastRegion) != null && te.hover && (a += `.${o}.t-area-selector-custom-data .area-selector .area-breadcrumb .el-breadcrumb__item.last-region
         .el-breadcrumb__inner:hover {
          ${r((oe = t.value) == null ? void 0 : oe.lastRegion.hover)}
        }`), (le = (re = t.value) == null ? void 0 : re.lastRegion) != null && le.active && (a += `.${o}.t-area-selector-custom-data .area-selector .area-breadcrumb
        .el-breadcrumb__item.last-region .el-breadcrumb__inner:active {
          ${r((ce = t.value) == null ? void 0 : ce.lastRegion.active)}
        }`), (ie = (ne = t.value) == null ? void 0 : ne.areaBox) != null && ie.default && (a += `.${o}.t-area-selector-custom-data .area-box {
          ${r((ue = t.value) == null ? void 0 : ue.areaBox.default)}
        }`), (de = (se = t.value) == null ? void 0 : se.areaBox) != null && de.hover && (a += `.${o}.t-area-selector-custom-data .area-box:hover {
          ${r((me = t.value) == null ? void 0 : me.areaBox.hover)}
        }`), (be = (fe = t.value) == null ? void 0 : fe.areaBox) != null && be.active && (a += `.${o}.t-area-selector-custom-data .area-box.active {
          ${r((ve = t.value) == null ? void 0 : ve.areaBox.active)}
        }`), (he = (ge = t.value) == null ? void 0 : ge.areaTitle) != null && he.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-title{
          ${r((Te = t.value) == null ? void 0 : Te.areaTitle.default)}
        }`), ($e = (ye = t.value) == null ? void 0 : ye.areaTitle) != null && $e.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-title:hover {
          ${r((pe = t.value) == null ? void 0 : pe.areaTitle.hover)}
        }`), (Ce = (Re = t.value) == null ? void 0 : Re.areaTitle) != null && Ce.active && (a += `.${o}.t-area-selector-custom-data .area-box .area-title.active {
          ${r((Ae = t.value) == null ? void 0 : Ae.areaTitle.active)}
        }`), (Ne = (Ee = t.value) == null ? void 0 : Ee.titleIcon) != null && Ne.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-title .title-icon{
          ${r((Oe = t.value) == null ? void 0 : Oe.titleIcon.default)}
        }`), (we = (Ie = t.value) == null ? void 0 : Ie.titleIcon) != null && we.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-title .title-icon:hover {
          ${r((xe = t.value) == null ? void 0 : xe.titleIcon.hover)}
        }`), (_e = (Se = t.value) == null ? void 0 : Se.titleIcon) != null && _e.active && (a += `.${o}.t-area-selector-custom-data .area-box .area-title .title-icon.active {
          ${r((De = t.value) == null ? void 0 : De.titleIcon.active)}
        }`), (Be = (Ge = t.value) == null ? void 0 : Ge.titleClose) != null && Be.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-title .title-close{
          ${r((Me = t.value) == null ? void 0 : Me.titleClose.default)}
        }`), (je = (Le = t.value) == null ? void 0 : Le.titleClose) != null && je.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-title .title-close:hover {
          ${r((Pe = t.value) == null ? void 0 : Pe.titleClose.hover)}
        }`), (He = (Ke = t.value) == null ? void 0 : Ke.titleClose) != null && He.active && (a += `.${o}.t-area-selector-custom-data .area-box .area-title .title-close.active {
          ${r((Ye = t.value) == null ? void 0 : Ye.titleClose.active)}
        }`), (Ue = (ke = t.value) == null ? void 0 : ke.closeIcon) != null && Ue.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-title .title-close .close-icon{
          ${r((We = t.value) == null ? void 0 : We.closeIcon.default)}
        }`), (qe = (ze = t.value) == null ? void 0 : ze.closeIcon) != null && qe.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-title .title-close .close-icon:hover {
          ${r((Fe = t.value) == null ? void 0 : Fe.closeIcon.hover)}
        }`), (Ze = (Je = t.value) == null ? void 0 : Je.addIcon) != null && Ze.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .add-button .add-icon{
          ${r((Qe = t.value) == null ? void 0 : Qe.addIcon.default)}
        }`), (Xe = (Ve = t.value) == null ? void 0 : Ve.addIcon) != null && Xe.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .add-button .add-icon:hover {
          ${r((ea = t.value) == null ? void 0 : ea.addIcon.hover)}
        }`), (ta = (aa = t.value) == null ? void 0 : aa.areaContainer) != null && ta.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container{
          ${r((oa = t.value) == null ? void 0 : oa.areaContainer.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list{
          ${r((ra = t.value) == null ? void 0 : ra.areaContainer.default)}
        }`), (ca = (la = t.value) == null ? void 0 : la.areaContainer) != null && ca.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container:hover {
          ${r((na = t.value) == null ? void 0 : na.areaContainer.hover)}
        }`), (ua = (ia = t.value) == null ? void 0 : ia.areaContainer) != null && ua.active && (a += `.${o}.t-area-selector-custom-data .area-box .area-container.active {
          ${r((sa = t.value) == null ? void 0 : sa.areaContainer.active)}
        }`), (ma = (da = t.value) == null ? void 0 : da.areaButton) != null && ma.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-button{
          ${r((fa = t.value) == null ? void 0 : fa.areaButton.default)}
        }`), (va = (ba = t.value) == null ? void 0 : ba.areaButton) != null && va.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-button:hover {
          ${r((ga = t.value) == null ? void 0 : ga.areaButton.hover)}
        }`), (Ta = (ha = t.value) == null ? void 0 : ha.areaButton) != null && Ta.active && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-button.active {
          ${r((ya = t.value) == null ? void 0 : ya.areaButton.active)}
        }`), (pa = ($a = t.value) == null ? void 0 : $a.areaSwitch) != null && pa.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-button .button-group .button{
          ${r((Ra = t.value) == null ? void 0 : Ra.areaSwitch.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list .button-group .button{
          ${r((Ca = t.value) == null ? void 0 : Ca.areaSwitch.default)}
        }`), (Ea = (Aa = t.value) == null ? void 0 : Aa.areaSwitch) != null && Ea.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-button .button-group .button:hover {
          ${r((Na = t.value) == null ? void 0 : Na.areaSwitch.hover)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list .area-button .button-group .button:hover{
          ${r((Oa = t.value) == null ? void 0 : Oa.areaSwitch.hover)}
        }`), (wa = (Ia = t.value) == null ? void 0 : Ia.areaSwitch) != null && wa.active && (a += `.${o}  .area-box .area-container .area-button .button-group .button.selected {
          ${r((xa = t.value) == null ? void 0 : xa.areaSwitch.active)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list .area-button .button-group .button.selected{
          ${r((Sa = t.value) == null ? void 0 : Sa.areaSwitch.active)}
        }`), (Da = (_a = t.value) == null ? void 0 : _a.handleButton) != null && Da.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-button .handle-button .el-button{
          ${r((Ga = t.value) == null ? void 0 : Ga.handleButton.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .add-button .el-button{
          ${r((Ba = t.value) == null ? void 0 : Ba.handleButton.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .header .add-btn .el-button{
          ${r((Ma = t.value) == null ? void 0 : Ma.handleButton.default)}
        }`), (ja = (La = t.value) == null ? void 0 : La.handleButton) != null && ja.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-button .handle-button .el-button:hover {
          ${r((Pa = t.value) == null ? void 0 : Pa.handleButton.hover)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .add-button .el-button:hover{
          ${r((Ka = t.value) == null ? void 0 : Ka.handleButton.hover)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .header .add-btn .el-button:hover{
          ${r((Ha = t.value) == null ? void 0 : Ha.handleButton.hover)}
        }`), (ka = (Ya = t.value) == null ? void 0 : Ya.handleButton) != null && ka.focus && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-button .handle-button .el-button:focus {
          ${r((Ua = t.value) == null ? void 0 : Ua.handleButton.focus)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .add-button .el-button:focus{
          ${r((Wa = t.value) == null ? void 0 : Wa.handleButton.focus)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .header .add-btn .el-button:focus{
          ${r((za = t.value) == null ? void 0 : za.handleButton.focus)}
        }`), (Fa = (qa = t.value) == null ? void 0 : qa.handleButton) != null && Fa.active && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-button .handle-button .el-button:active{
          ${r((Ja = t.value) == null ? void 0 : Ja.handleButton.active)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .add-button .el-button:active{
          ${r((Za = t.value) == null ? void 0 : Za.handleButton.active)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .header .add-btn .el-button:active{
          ${r((Qa = t.value) == null ? void 0 : Qa.handleButton.active)}
        }`), (Xa = (Va = t.value) == null ? void 0 : Va.cityContainer) != null && Xa.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list{
          ${r((et = t.value) == null ? void 0 : et.cityContainer.default)}
        }`), (tt = (at = t.value) == null ? void 0 : at.breadcrumbItem) != null && tt.default && (a += `.${o}.t-area-selector-custom-data .area-selector .area-breadcrumb .el-breadcrumb__item .el-breadcrumb__inner{
          ${r((ot = t.value) == null ? void 0 : ot.breadcrumbItem.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-title .area-breadcrumb .el-breadcrumb__item .el-breadcrumb__inner{
          ${r((rt = t.value) == null ? void 0 : rt.breadcrumbItem.default)}
        }`), (ct = (lt = t.value) == null ? void 0 : lt.litters) != null && ct.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-litter{
          ${r((nt = t.value) == null ? void 0 : nt.litters.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .area-litter{
          ${r((it = t.value) == null ? void 0 : it.litters.default)}
        }`), (st = (ut = t.value) == null ? void 0 : ut.litterItem) != null && st.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-litter .litter{
          ${r((dt = t.value) == null ? void 0 : dt.litterItem.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .area-litter .litter{
          ${r((mt = t.value) == null ? void 0 : mt.litterItem.default)}
        }`), (bt = (ft = t.value) == null ? void 0 : ft.litterItemTag) != null && bt.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .tag{
          ${r((vt = t.value) == null ? void 0 : vt.litterItemTag.default)}
        }`), (ht = (gt = t.value) == null ? void 0 : gt.countyTop) != null && ht.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .citys .city-container .city {
          ${r((Tt = t.value) == null ? void 0 : Tt.countyTop.default)}
        }`), ($t = (yt = t.value) == null ? void 0 : yt.countyItem) != null && $t.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .citys .city-container .countys .county {
          ${r((pt = t.value) == null ? void 0 : pt.countyItem.default)}
        }`), (Ct = (Rt = t.value) == null ? void 0 : Rt.dataList) != null && Ct.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list{
          ${r((At = t.value) == null ? void 0 : At.dataList.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list{
          ${r((Et = t.value) == null ? void 0 : Et.dataList.default)}
        }`), (Ot = (Nt = t.value) == null ? void 0 : Nt.customDataList) != null && Ot.default && (a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .data-list .area-city-list{
          ${r((It = t.value) == null ? void 0 : It.customDataList.default)}
        }`), (xt = (wt = t.value) == null ? void 0 : wt.customAreaList) != null && xt.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .bottom{
          ${r((St = t.value) == null ? void 0 : St.customAreaList.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .tab .selected-list{
          ${r((_t = t.value) == null ? void 0 : _t.customAreaList.default)}
        }`), (Gt = (Dt = t.value) == null ? void 0 : Dt.customAreaTag) != null && Gt.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .list .button{
          ${r((Bt = t.value) == null ? void 0 : Bt.customAreaTag.default)}
        }`), (Lt = (Mt = t.value) == null ? void 0 : Mt.customAreaTag) != null && Lt.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .list .button:hover{
          ${r((jt = t.value) == null ? void 0 : jt.customAreaTag.hover)}
        }`), (Kt = (Pt = t.value) == null ? void 0 : Pt.customAreaTag) != null && Kt.active && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .list .button.active{
          ${r((Ht = t.value) == null ? void 0 : Ht.customAreaTag.active)}
        }`), (kt = (Yt = t.value) == null ? void 0 : Yt.customAreaTagDeleteIcon) != null && kt.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .list .button .icon-delete{
          ${r((Ut = t.value) == null ? void 0 : Ut.customAreaTagDeleteIcon.default)}
        }`), (zt = (Wt = t.value) == null ? void 0 : Wt.customAreaTagDeleteIcon) != null && zt.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .list .button .icon-delete:hover{
          ${r((qt = t.value) == null ? void 0 : qt.customAreaTagDeleteIcon.hover)}
        }`), (Jt = (Ft = t.value) == null ? void 0 : Ft.customAreaTagEditIcon) != null && Jt.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .list .button .icon-edit{
          ${r((Zt = t.value) == null ? void 0 : Zt.customAreaTagEditIcon.default)}
        }`), (Vt = (Qt = t.value) == null ? void 0 : Qt.customAreaTagEditIcon) != null && Vt.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .list .button .icon-edit:hover{
          ${r((Xt = t.value) == null ? void 0 : Xt.customAreaTagEditIcon.hover)}
        }`), (ao = (eo = t.value) == null ? void 0 : eo.customAreaArrowDown) != null && ao.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .top .fold .arrow-down{
          ${r((to = t.value) == null ? void 0 : to.customAreaArrowDown.default)}
        }`), (ro = (oo = t.value) == null ? void 0 : oo.customAreaArrowUp) != null && ro.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .top .fold .arrow-up{
          ${r((lo = t.value) == null ? void 0 : lo.customAreaArrowUp.default)}
        }`), (no = (co = t.value) == null ? void 0 : co.customAreaAddTag) != null && no.default && (a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .tab .selected-list .el-tag{
          ${r((io = t.value) == null ? void 0 : io.customAreaAddTag.default)}
        }`), (so = (uo = t.value) == null ? void 0 : uo.customAreaAddTagIcon) != null && so.default && (a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .tab .selected-list .el-tag .el-icon-close{
          ${r((mo = t.value) == null ? void 0 : mo.customAreaAddTagIcon.default)}
        }`), (bo = (fo = t.value) == null ? void 0 : fo.customAreaAddTagIcon) != null && bo.hover && (a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .tab .selected-list .el-tag .el-icon-close:hover{
          ${r((vo = t.value) == null ? void 0 : vo.customAreaAddTagIcon.hover)}
        }`), (ho = (go = t.value) == null ? void 0 : go.customAreaAddTagIcon) != null && ho.active && (a += `.${o}.t-area-selector-custom-data .area-box .add-list .add-wrapper .tab .selected-list .el-tag .el-icon-close.active{
          ${r((To = t.value) == null ? void 0 : To.customAreaAddTagIcon.active)}
        }`), ($o = (yo = t.value) == null ? void 0 : yo.customAreaExpand) != null && $o.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .list-wrapper .top .fold .el-button{
          ${r((po = t.value) == null ? void 0 : po.customAreaExpand.default)}
        }`), (Co = (Ro = t.value) == null ? void 0 : Ro.Active) != null && Co.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-litter .litter-active{
          ${r((Ao = t.value) == null ? void 0 : Ao.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .tag-active{
          ${r((Eo = t.value) == null ? void 0 : Eo.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .citys .city-container .city-active {
          ${r((No = t.value) == null ? void 0 : No.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .watershed .city-container .city-active {
          ${r((Oo = t.value) == null ? void 0 : Oo.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .citys .city-container .countys
         .county-active {
          ${r((Io = t.value) == null ? void 0 : Io.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .key-area .area-item-active {
          ${r((wo = t.value) == null ? void 0 : wo.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .key-watershed .area-item-active {
          ${r((xo = t.value) == null ? void 0 : xo.Active.default)}
        }`), (_o = (So = t.value) == null ? void 0 : So.Active) != null && _o.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-litter .litter:hover{
          ${r((Do = t.value) == null ? void 0 : Do.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .tag:hover{
          ${r((Go = t.value) == null ? void 0 : Go.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .citys .city-container .city:hover {
          ${r((Bo = t.value) == null ? void 0 : Bo.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .watershed .city-container .city:hover {
          ${r((Mo = t.value) == null ? void 0 : Mo.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .area-city-list .row .citys .city-container .countys .county:hover {
          ${r((Lo = t.value) == null ? void 0 : Lo.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .key-area .area-item:hover {
          ${r((jo = t.value) == null ? void 0 : jo.Active.default)}
        }`, a += `.${o}.t-area-selector-custom-data .area-box .area-container .custom-area-list .key-watershed .area-item:hover {
          ${r((Po = t.value) == null ? void 0 : Po.Active.default)}
        }`), (Ho = (Ko = t.value) == null ? void 0 : Ko.bottomContainer) != null && Ho.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .footer{
          ${r((Yo = t.value) == null ? void 0 : Yo.bottomContainer.default)}
        }`), (Uo = (ko = t.value) == null ? void 0 : ko.bottomButton) != null && Uo.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .footer .el-button{
          ${r((Wo = t.value) == null ? void 0 : Wo.bottomButton.default)}
        }`), (qo = (zo = t.value) == null ? void 0 : zo.bottomResetButton) != null && qo.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .footer .el-button.reset{
          ${r((Fo = t.value) == null ? void 0 : Fo.bottomResetButton.default)}
        }`), (Zo = (Jo = t.value) == null ? void 0 : Jo.bottomResetButton) != null && Zo.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .footer .el-button.reset:hover{
          ${r((Qo = t.value) == null ? void 0 : Qo.bottomResetButton.hover)}
        }`), (Xo = (Vo = t.value) == null ? void 0 : Vo.bottomConfirmButton) != null && Xo.default && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .footer .el-button.search{
          ${r((er = t.value) == null ? void 0 : er.bottomConfirmButton.default)}
        }`), (tr = (ar = t.value) == null ? void 0 : ar.bottomConfirmButton) != null && tr.hover && (a += `.${o}.t-area-selector-custom-data .area-box .area-container .footer .el-button.search:hover{
          ${r((or = t.value) == null ? void 0 : or.bottomConfirmButton.hover)}
        }`), a;
    }, _ = () => {
      const a = document.querySelector(`.${o} .style-container`);
      a && (a.innerHTML = `<style>${cr()}${nr()}</style>`);
    }, ir = () => {
      e.isShowBox = !e.isShowBox, lr(() => {
        var l;
        const a = document.getElementById(`${e.breadcrumbObj.code}`);
        a && ((l = e.list) == null || l.scroll(0, (a == null ? void 0 : a.offsetTop) - e.list.offsetTop));
      });
    }, ur = (a) => {
      var n;
      e.breadcrumbObj.code = a;
      const l = document.getElementById(`${a}`);
      l && ((n = e.list) == null || n.scroll(0, (l == null ? void 0 : l.offsetTop) - e.list.offsetTop));
    }, O = (a) => {
      let l;
      return a.substr(4, 2) !== "00" ? l = "country" : a.substr(2, 2) !== "00" ? l = "city" : l = "province", l;
    }, v = (a) => {
      var n, d, i, u;
      if (e.breadcrumbObj.path = [], e.regionTool.currentRegionCode = a, e.regionTool.currentRegionName = ((d = (n = e.regionTool) == null ? void 0 : n.allRegionMap[a]) == null ? void 0 : d.REGION_NAME) || "", e.regionTool.shortName = ((u = (i = e.regionTool) == null ? void 0 : i.allRegionMap[a]) == null ? void 0 : u.SHORT_NAME) || "", e.regionTool.currentRegionType = O(a), !e.regionTool.currentRegionName)
        return;
      e.regionTool.currentProvinceCode = `${e.regionTool.currentRegionCode.substr(0, 2)}0000000000`, a.substr(2, 2) !== "00" && e.regionTool.allRegionMap[e.regionTool.currentProvinceCode] && (e.regionTool.currentProvinceName = e.regionTool.allRegionMap[e.regionTool.currentProvinceCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentProvinceCode, name: e.regionTool.currentProvinceName })), e.regionTool.currentCityCode = `${e.regionTool.currentRegionCode.substr(0, 4)}00000000`, a.substr(4, 2) !== "00" && e.regionTool.allRegionMap[e.regionTool.currentCityCode] && (e.regionTool.currentCityName = e.regionTool.allRegionMap[e.regionTool.currentCityCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentCityCode, name: e.regionTool.currentCityName })), e.regionTool.currentCountyCode = `${e.regionTool.currentRegionCode.substr(0, 6)}000000`, a.substr(6, 3) !== "000" && e.regionTool.allRegionMap[e.regionTool.currentCountyCode] && (e.regionTool.currentCountyName = e.regionTool.allRegionMap[e.regionTool.currentCountyCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentCountyCode, name: e.regionTool.currentCountyName })), e.regionTool.currentTownCode = `${e.regionTool.currentRegionCode.substr(0, 9)}000`, a.substr(9, 3) !== "000" && e.regionTool.allRegionMap[e.regionTool.currentTownCode] && (e.regionTool.currentTownName = e.regionTool.allRegionMap[e.regionTool.currentTownCode].REGION_NAME, e.breadcrumbObj.path.push({ code: e.regionTool.currentTownCode, name: e.regionTool.currentTownName }));
      const l = e.regionTool.allRegionMap[e.regionTool.currentRegionCode].REGION_NAME;
      l.length > 8 && `${l.substr(0, 8)}`, e.breadcrumbObj.path.push({ code: e.regionTool.currentRegionCode, name: e.regionTool.currentRegionName });
    }, sr = () => {
      var l, n;
      let a = {};
      if (e.currentAreaType === "custom" && e.currentSelectedType === "custom")
        e.searchAreaType = "custom", e.breadcrumbObj.path = [{ code: e.regionTool.currentRegionCode, name: e.regionTool.currentRegionName }], a = {
          selected: "custom",
          REGION_CODE: e.regionTool.currentRegionCode,
          REGION_NAME: e.regionTool.currentRegionName,
          REGION_TYPE: e.regionTool.currentRegionType,
          SHORT_NAME: e.regionTool.shortName
        };
      else if (e.currentAreaType === "xz" && e.currentSelectedType === "region")
        e.searchAreaType = "region", v(e.regionTool.tempRegionCode), a = {
          selected: "region",
          REGION_CODE: e.regionTool.currentRegionCode,
          REGION_NAME: e.regionTool.currentRegionName,
          REGION_TYPE: O(e.regionTool.currentRegionCode),
          SHORT_NAME: e.regionTool.shortName
        };
      else if (e.currentAreaType === "qy")
        e.searchAreaType = "area", e.breadcrumbObj.path = [{ code: e.regionTool.currentRegionCode, name: e.regionTool.currentRegionName }], a = {
          selected: "area",
          REGION_CODE: e.regionTool.currentRegionCode,
          REGION_NAME: e.regionTool.currentRegionName,
          REGION_TYPE: "",
          SHORT_NAME: e.regionTool.shortName
        };
      else if (e.currentAreaType === "ly" && e.currentSelectedType === "" && !e.watershedList.includes((n = (l = e.breadcrumbObj) == null ? void 0 : l.path[0]) == null ? void 0 : n.code))
        e.searchAreaType = "wsystem", e.breadcrumbObj.path = [{ code: "", name: "重点流域" }], a = {
          selected: "wsystem",
          REGION_CODE: "",
          REGION_NAME: "",
          REGION_TYPE: "",
          SHORT_NAME: e.regionTool.shortName
        };
      else if (e.currentAreaType === "ly")
        e.searchAreaType = "wsystem", e.breadcrumbObj.path = [{ code: e.regionTool.currentRegionCode, name: e.regionTool.currentRegionName }], a = {
          selected: "wsystem",
          REGION_CODE: e.regionTool.currentRegionCode,
          REGION_NAME: e.regionTool.currentRegionName,
          REGION_TYPE: "",
          SHORT_NAME: e.regionTool.shortName
        };
      else {
        e.isShowBox = !e.isShowBox;
        return;
      }
      N("search", a), e.isShowBox = !e.isShowBox;
    }, I = (a) => {
      if (a !== "" && (e.currentSelectedType = "region"), e.regionTool.tempRegionCode = a, !e.regionTool.allRegionMap[a] || a === e.thsDefaultRegionCode)
        return;
      const l = e.regionTool.allRegionMap[a].PARENT_REGION_CODE;
      e.breadcrumbObj.code = l === e.thsDefaultRegionCode ? e.regionTool.allRegionMap[a].REGION_PY.substring(0, 1) : e.regionTool.allRegionMap[l].REGION_PY.substring(0, 1);
    }, w = () => {
      Cr({
        url: c.divisionUrl,
        method: "post",
        data: Ar.stringify({ REGION_PARENT_CODE: c.thsDefaultRegionCode })
      }).then((a) => {
        var d;
        e.regionTool.regionLevelShowList = [], e.regionTool.allRegionMap = [];
        const l = a.data.data;
        for (let i = 0; i < l.length; i++)
          l[i] && l[i].REGION_PY && (l[i].CHILD = []), e.regionTool.allRegionMap[l[i].REGION_CODE] = l[i];
        for (let i = 0; i < l.length; i++) {
          const u = l[i].PARENT_REGION_CODE, s = l[i].REGION_PY.substring(0, 1);
          l[i] && u && (u === e.thsDefaultRegionCode ? (e.regionTool.allRegionMap[s] || (e.regionTool.allRegionMap[s] = {}, e.regionTool.allRegionMap[s].CHILD = [], e.regionTool.allRegionMap[s].REGION_CODE = s, e.regionTool.allRegionMap[s].REGION_NAME = s), e.regionTool.allRegionMap[s].CHILD.push(l[i])) : e.regionTool.allRegionMap[u] && e.regionTool.allRegionMap[u].CHILD.push(l[i]));
        }
        if (c.showInitialSort) {
          const i = [
            "A",
            "B",
            "C",
            "D",
            "E",
            "F",
            "G",
            "H",
            "I",
            "J",
            "K",
            "L",
            "M",
            "N",
            "O",
            "P",
            "Q",
            "R",
            "S",
            "T",
            "U",
            "V",
            "W",
            "X",
            "Y",
            "Z",
            "港澳"
          ];
          for (let u = 0; u < i.length; u++) {
            const s = i[u];
            if (e.regionTool.allRegionMap[s] && e.regionTool.allRegionMap[s].CHILD != null && e.regionTool.allRegionMap[s].CHILD.length > 0) {
              const m = e.regionTool.allRegionMap[s];
              e.regionTool.regionLevelShowList.push(m);
            }
          }
        } else {
          const i = /* @__PURE__ */ new Set();
          for (let u = 0; u < l.length; u++) {
            const s = l[u];
            if (s.PARENT_REGION_CODE === e.thsDefaultRegionCode && s.REGION_PY) {
              const m = s.REGION_PY.substring(0, 1);
              if (m && !i.has(m)) {
                i.add(m);
                const f = e.regionTool.allRegionMap[m];
                f && f.CHILD && f.CHILD.length > 0 && e.regionTool.regionLevelShowList.push(f);
              }
            }
          }
        }
        e.searchAreaType !== "custom" && e.searchAreaType !== "area" && e.searchAreaType !== "wsystem" ? (e.REGION_CODE && I(e.REGION_CODE), v(e.REGION_CODE ? e.REGION_CODE : e.thsDefaultRegionCode)) : (e.breadcrumbObj.code = "", I(""));
        const n = document.getElementById(`${e.breadcrumbObj.code}`);
        n && ((d = e.list) == null || d.scroll(0, (n == null ? void 0 : n.offsetTop) - e.list.offsetTop));
      });
    }, D = (a, l) => {
      var n, d, i;
      if (e.currentSelectedType = "", e.currentAreaType = a, e.selectedButton = l, a === "xz") {
        const u = (i = (d = e.breadcrumbObj) == null ? void 0 : d.path[((n = e.breadcrumbObj) == null ? void 0 : n.path.length) - 1]) == null ? void 0 : i.code;
        e.REGION_CODE = u || c.regionCode, w();
      } else
        a === "qy" ? (e.currentKeyArea = {
          areaCode: e.breadcrumbObj.path[0].code
        }, e.regionTool.regionLevelShowList = []) : a === "ly" ? (e.currentKeyWatershed = {
          areaCode: e.breadcrumbObj.path[0].code
        }, e.regionTool.regionLevelShowList = []) : a === "custom" && (e.isShowCustomArea = !0, e.currentCustomArea = {
          areaCode: e.breadcrumbObj.path[0].code,
          areaName: e.breadcrumbObj.path[0].name
        }, e.regionTool.regionLevelShowList = []);
    }, dr = (a, l) => {
      var i;
      if (a.code === "")
        return;
      e.regionTool.tempRegionCode = a.code;
      let n = {};
      e.currentAreaType === "custom" ? (e.breadcrumbObj.path = [{ code: e.regionTool.currentRegionCode, name: e.regionTool.currentRegionName }], n = {
        selected: "custom",
        REGION_CODE: e.regionTool.currentRegionCode,
        REGION_NAME: e.regionTool.currentRegionName,
        REGION_TYPE: e.regionTool.currentRegionType,
        SHORT_NAME: e.regionTool.shortName
      }) : e.currentAreaType === "xz" ? (v(e.regionTool.tempRegionCode), n = {
        selected: "region",
        REGION_CODE: e.regionTool.currentRegionCode,
        REGION_NAME: e.regionTool.currentRegionName,
        REGION_TYPE: O(e.regionTool.currentRegionCode),
        SHORT_NAME: e.regionTool.shortName
      }) : e.currentAreaType === "qy" ? (e.breadcrumbObj.path = [{ code: e.regionTool.currentRegionCode, name: e.regionTool.currentRegionName }], n = {
        selected: "area",
        REGION_CODE: e.regionTool.currentRegionCode,
        REGION_NAME: e.regionTool.currentRegionName,
        REGION_TYPE: "",
        SHORT_NAME: e.regionTool.shortName
      }) : e.currentAreaType === "ly" && (e.breadcrumbObj.path = [{ code: e.regionTool.currentRegionCode, name: e.regionTool.currentRegionName }], n = {
        selected: "wsystem",
        REGION_CODE: e.regionTool.currentRegionCode,
        REGION_NAME: e.regionTool.currentRegionName,
        REGION_TYPE: "",
        SHORT_NAME: e.regionTool.shortName
      }), N("select", n);
      const d = document.getElementById(`${e.breadcrumbObj.code}`);
      d && ((i = e.list) == null || i.scroll(0, (d == null ? void 0 : d.offsetTop) - e.list.offsetTop));
    }, G = () => {
      e.currentKeyWatershed = {}, e.breadcrumbObj.path = [{ code: "", name: "重点流域" }];
      const a = {
        selected: "wsystem",
        REGION_CODE: "",
        REGION_NAME: "",
        REGION_TYPE: ""
      };
      e.isShowBox = !e.isShowBox, e.currentSelectedType = "", N("resetRegion", a);
    }, mr = () => {
      var n;
      if (e.currentAreaType === "ly") {
        G();
        return;
      }
      I(e.regionTool.defaultRegionCode), v(e.regionTool.defaultRegionCode);
      const a = {
        selected: "region",
        REGION_CODE: e.regionTool.currentRegionCode,
        REGION_NAME: e.regionTool.currentRegionName,
        REGION_TYPE: O(e.regionTool.currentRegionCode)
      };
      N("resetRegion", a), e.currentKeyArea = {}, e.currentKeyWatershed = {}, e.currentCustomArea = {}, e.selectedButton = 0, e.currentAreaType = "xz", e.searchAreaType = "region";
      const l = document.getElementById(`${e.breadcrumbObj.code}`);
      l && ((n = e.list) == null || n.scroll(0, (l == null ? void 0 : l.offsetTop) - e.list.offsetTop)), e.isShowBox = !e.isShowBox, e.REGION_CODE = e.regionTool.defaultRegionCode, w();
    }, fr = (a) => {
      e.isShowEditPage = !0, a && JSON.stringify(a) !== "{}" ? e.currentCategory = a : e.currentCategory = {};
    }, br = (a) => {
      e.currentSelectedType = "custom", e.currentCustomArea = a;
      const { areaName: l, areaCode: n, areaLevel: d } = a;
      e.regionTool.currentRegionCode = n, e.regionTool.currentRegionName = l, e.regionTool.currentRegionType = d;
    }, vr = (a) => {
      e.currentSelectedType = "area", e.currentKeyArea = a;
      const { areaName: l, areaCode: n } = a;
      e.regionTool.currentRegionCode = n, e.regionTool.currentRegionName = l, e.selectKeyAreaCode !== "" && (e.breadcrumbObj.path = [{ code: e.regionTool.currentRegionCode, name: e.regionTool.currentRegionName }]);
    }, gr = (a) => {
      e.currentSelectedType = "wsystem", e.currentKeyWatershed = a;
      const { areaName: l, areaCode: n } = a;
      e.regionTool.currentRegionCode = n, e.regionTool.currentRegionName = l;
    }, x = (a) => {
      var d, i, u, s, m, f, g, h, T, y, $, p, R, C, A;
      let l = "", n = 0;
      if (a === "region") {
        const S = (u = (i = e.breadcrumbObj) == null ? void 0 : i.path[((d = e.breadcrumbObj) == null ? void 0 : d.path.length) - 1]) == null ? void 0 : u.code;
        e.REGION_CODE = S || c.regionCode, l = "xz", n = 0;
      } else
        a === "area" ? (((s = c.defaultArea) == null ? void 0 : s.areaCode) !== "" && ((m = c.defaultArea) == null ? void 0 : m.areaCode) !== void 0 && (e.breadcrumbObj.path = [{ code: (f = c.defaultArea) == null ? void 0 : f.areaCode, name: (g = c.defaultArea) == null ? void 0 : g.areaName }]), l = "qy", n = 1) : a === "wsystem" ? (((h = c.defaultArea) == null ? void 0 : h.areaCode) !== "" && ((T = c.defaultArea) == null ? void 0 : T.areaCode) !== void 0 && (e.breadcrumbObj.path = [{ code: (y = c.defaultArea) == null ? void 0 : y.areaCode, name: ($ = c.defaultArea) == null ? void 0 : $.areaName }]), l = "ly", n = 2) : a === "custom" && (((p = c.defaultArea) == null ? void 0 : p.areaCode) !== "" && ((R = c.defaultArea) == null ? void 0 : R.areaCode) !== void 0 && (e.breadcrumbObj.path = [{ code: (C = c.defaultArea) == null ? void 0 : C.areaCode, name: (A = c.defaultArea) == null ? void 0 : A.areaName }]), l = "custom", n = 3);
      D(l, n);
    }, hr = () => {
      e.currentKeyArea = {}, e.currentKeyWatershed = {}, e.currentCustomArea = {}, x(e.searchAreaType), e.isShowBox = !e.isShowBox, e.isShowEditPage = !1;
    }, Tr = (a) => {
      e.currentCustomArea = a;
    }, yr = (a) => {
      e.watershedList = a;
    };
    return E(
      () => c.separatorType,
      () => {
        e.separator = c.separatorType, v(c.regionCode);
      },
      { immediate: !0 }
    ), E(
      () => c.divisionUrl,
      () => {
        w();
      }
    ), rr(async () => {
      _();
    }), E(
      () => c.cStyle,
      () => {
        _();
      },
      { deep: !0, immediate: !0 }
    ), E(
      () => [c.thsDefaultRegionCode, c.selected],
      () => {
        e.thsDefaultRegionCode = c.thsDefaultRegionCode, e.selected = c.selected, w();
      },
      { deep: !0 }
    ), E(
      [() => c.selectedType, () => c.defaultArea, () => c.regionCode],
      () => {
        lr(() => {
          c.selectedType === "region" && (e.breadcrumbObj.path[e.breadcrumbObj.path.length - 1].code = c.regionCode, e.searchAreaType = c.selectedType), x(c.selectedType);
        });
      },
      { deep: !0 }
    ), rr(() => {
      x(c.selectedType), e.regionTool.defaultRegionCode = c.regionCode;
    }), {
      curStyle: t,
      domOnlyClass: o,
      checkRegion: I,
      select: dr,
      search: sr,
      resetRegion: mr,
      switchSelector: ir,
      close: hr,
      selectLitter: ur,
      checkedAreaType: x,
      updateCategoryArea: fr,
      customSelected: br,
      selectedKeyArea: vr,
      selectedKeyWatershed: gr,
      selectButton: D,
      customSelectedArea: Tr,
      resetWatershed: G,
      handleDataLoaded: yr,
      ...Rr(e),
      showInitialSort: b(() => c.showInitialSort)
    };
  }
});
export {
  Yr as default
};
