// @ths/design/es/components/area-selector-custom-data/src/key-watershed-group.vue2.js
import { defineComponent as L, ref as h, reactive as R, computed as T, watch as p, onMounted as M, toRefs as C } from "vue";
import E from "qs";
import O from "axios";
const I = L({
  name: "KeyWatershedGroup",
  props: {
    defaultArea: {
      type: Object,
      default: () => {
      }
    },
    // 当前选中区域
    selectedArea: {
      type: Object,
      default: () => {
      }
    },
    // 重点流域顶级编码
    rootCode: {
      type: String,
      default: ""
    },
    url: {
      type: String,
      default: ""
    }
  },
  emits: ["selected"],
  setup(i, { emit: s }) {
    const r = h(null), e = R({
      list: [],
      // 区域列表dom
      currentArea: T(() => JSON.stringify(i.selectedArea) !== "{}" ? i.selectedArea : i.defaultArea),
      loading: !1,
      // 当前数据
      regionTool: {
        regionLevelShowList: [],
        // 行政区的树结构
        allRegionMap: []
        // 行政区的map结构
      }
    }), d = (a) => {
      s("selected", a);
    }, g = async () => {
      e.loading = !0, O({
        url: i.url,
        method: "post",
        data: E.stringify({ REGION_CODE: i.rootCode })
      }).then((a) => {
        if (a.data.head.statusCode === "200" && a.data.data.length > 0) {
          e.loading = !1, e.regionTool.regionLevelShowList = [], e.regionTool.allRegionMap = [];
          const n = [
            "A",
            "B",
            "C",
            "D",
            "E",
            "F",
            "G",
            "H",
            "I",
            "J",
            "K",
            "L",
            "M",
            "N",
            "O",
            "P",
            "Q",
            "R",
            "S",
            "T",
            "U",
            "V",
            "W",
            "X",
            "Y",
            "Z"
          ];
          let l = [];
          l = a.data.data;
          for (let o = 0; o < l.length; o++)
            l[o] && l[o].REGION_PY && (l[o].CHILD = []);
          for (let o = 0; o < l.length; o++) {
            const t = l[o].REGION_PY.substring(0, 1);
            l[o] && (e.regionTool.allRegionMap[t] || (e.regionTool.allRegionMap[t] = {}, e.regionTool.allRegionMap[t].CHILD = [], e.regionTool.allRegionMap[t].REGION_CODE = t, e.regionTool.allRegionMap[t].REGION_NAME = t), e.regionTool.allRegionMap[t].CHILD.push(l[o]));
          }
          for (let o = 0; o < n.length; o++) {
            const t = n[o];
            if (e.regionTool.allRegionMap[t] && e.regionTool.allRegionMap[t].CHILD != null && e.regionTool.allRegionMap[t].CHILD.length > 0) {
              const u = e.regionTool.allRegionMap[t];
              e.regionTool.regionLevelShowList.push(u);
            }
          }
        } else
          e.loading = !1;
      });
    }, c = (a) => {
      const n = {
        areaCode: a.WSYSTEM_CODE,
        areaName: a.WSYSTEM_NAME
      };
      s("selected", n);
    }, f = (a) => {
      var l;
      e.regionTool.regionLevelShowList.forEach((o) => {
        o.isActive = !1;
      }), e.regionTool.regionLevelShowList.find((o) => o.REGION_CODE === a).isActive = !0;
      const n = document.getElementById(`${a}`);
      n && ((l = r.value) == null || l.scroll(0, (n == null ? void 0 : n.offsetTop) - r.value.offsetTop));
    };
    return p([() => i.rootCode, i.url], () => {
      g();
    }), M(() => {
      g();
    }), {
      ...C(e),
      checkRegion: c,
      getKeyWatershed: g,
      selectLitter: f,
      handleClick: d,
      listRef: r
    };
  }
});
export {
  I as default
};
