// @ths/design/es/components/area-selector-custom-data/src/key-area.vue2.js
import { defineComponent as i, reactive as n, computed as A, watch as c, onMounted as o, toRefs as u } from "vue";
import f from "qs";
import m from "axios";
const K = i({
  name: "KeyArea",
  props: {
    defaultArea: {
      type: Object,
      default: () => {
      }
    },
    // 当前选中区域
    selectedArea: {
      type: Object,
      default: () => {
      }
    },
    isKeyArea: {
      type: String,
      default: ""
    },
    url: {
      type: String,
      default: ""
    }
  },
  emits: ["selected"],
  setup(e, { emit: l }) {
    const r = n({
      list: [],
      currentArea: A(() => JSON.stringify(e.selectedArea) !== "{}" ? e.selectedArea : e.defaultArea)
    }), s = (a) => {
      l("selected", a);
    }, t = () => {
      m({
        url: e.url,
        method: "post",
        data: f.stringify({ IS_AREA: e.isKeyArea })
      }).then((a) => {
        r.list = a.data.data.map((d) => ({
          areaCode: d.AREA_CODE,
          areaName: d.AREA_NAME
        }));
      });
    };
    return c([() => e.isKeyArea, () => e.url], () => {
      t();
    }), o(() => {
      t();
    }), {
      ...u(r),
      getKeyArea: t,
      handleClick: s
    };
  }
});
export {
  K as default
};
