// @ths/design/es/components/area-selector-custom-data/src/key-watershed-group.vue.js
import p from "./key-watershed-group.vue2.js";
import { resolveComponent as h, resolveDirective as u, openBlock as t, createElementBlock as o, createElementVNode as r, Fragment as n, renderList as c, normalizeClass as a, toDisplayString as l, withDirectives as g, createVNode as E, withCtx as f } from "vue";
import "./key-watershed-group.vue3.js";
import k from "../../../_virtual/_plugin-vue_export-helper.js";
const C = { class: "key-watershed-group" }, m = { class: "area-litter" }, D = ["onClick"], O = {
  ref: "listRef",
  class: "area-city-list",
  "element-loading-background": "rgba(255, 255, 255, .0)"
}, S = ["id"], N = { class: "watershed" }, w = ["onClick"], y = { class: "city_text" };
function L(i, R, $, A, G, I) {
  const _ = h("el-tooltip"), v = u("loading");
  return t(), o("div", C, [
    r("div", m, [
      (t(!0), o(n, null, c(i.regionTool.regionLevelShowList, (e) => (t(), o("a", {
        key: e,
        href: "javascript:void(0)",
        class: a(["litter", { "litter-active": e.isActive }]),
        onClick: (s) => i.selectLitter(e.REGION_CODE)
      }, l(e.REGION_CODE), 11, D))), 128))
    ]),
    g((t(), o("div", O, [
      (t(!0), o(n, null, c(i.regionTool.regionLevelShowList, (e) => (t(), o("div", {
        id: e.REGION_CODE,
        key: e,
        class: "row"
      }, [
        r("div", {
          class: a(["tag", { "tag-active": e.isActive }])
        }, l(e.REGION_CODE), 3),
        r("div", N, [
          (t(!0), o(n, null, c(e.CHILD, (s) => {
            var d;
            return t(), o("div", {
              key: s,
              class: "city-container"
            }, [
              r("div", {
                class: a(["city", {
                  "city-active": s.WSYSTEM_CODE === ((d = i.currentArea) == null ? void 0 : d.areaCode)
                }]),
                onClick: (M) => i.checkRegion(s)
              }, [
                E(_, {
                  class: "box-item",
                  effect: "dark",
                  content: s.WSYSTEM_NAME,
                  placement: "top"
                }, {
                  default: f(() => [
                    r("div", y, l(s.WSYSTEM_NAME), 1)
                  ]),
                  _: 2
                }, 1032, ["content"])
              ], 10, w)
            ]);
          }), 128))
        ])
      ], 8, S))), 128))
    ])), [
      [v, i.loading]
    ])
  ]);
}
const B = /* @__PURE__ */ k(p, [["render", L]]);
export {
  B as default
};
