// @ths/design/es/components/area-selector-custom-data/src/key-watershed.vue2.js
import { defineComponent as s, reactive as i, computed as f, watch as c, onMounted as u, toRefs as m } from "vue";
import C from "qs";
import g from "axios";
const S = s({
  name: "KeyWatershed",
  props: {
    defaultArea: {
      type: Object,
      default: () => {
      }
    },
    // 当前选中区域
    selectedArea: {
      type: Object,
      default: () => {
      }
    },
    // 重点流域顶级编码
    rootCode: {
      type: String,
      default: ""
    },
    url: {
      type: String,
      default: ""
    }
  },
  emits: ["selected", "onDataLoaded"],
  setup(e, { emit: l }) {
    const t = i({
      list: [],
      currentArea: f(() => JSON.stringify(e.selectedArea) !== "{}" ? e.selectedArea : e.defaultArea),
      loading: !1
    }), r = (a) => {
      l("selected", a);
    }, o = () => {
      t.loading = !0, g({
        url: e.url,
        method: "post",
        data: C.stringify({ REGION_CODE: e.rootCode })
      }).then((a) => {
        if (a.data.head.statusCode === "200" && a.data.data.length > 0) {
          t.loading = !1, t.list = a.data.data.map((d) => ({
            areaCode: d.WSYSTEM_CODE,
            areaName: d.WSYSTEM_NAME
          }));
          const n = t.list.map((d) => d.areaCode);
          l("onDataLoaded", n);
        } else
          t.loading = !1;
      });
    };
    return c([() => e.rootCode, e.url], () => {
      o();
    }), u(() => {
      o();
    }), {
      ...m(t),
      getKeyWatershed: o,
      handleClick: r
    };
  }
});
export {
  S as default
};
