// @ths/design/es/components/rank-single/src/index.vue.js
import p from "./index.vue2.js";
import { resolveComponent as _, openBlock as n, createElementBlock as r, normalizeClass as t, createElementVNode as s, Fragment as u, renderList as g, withModifiers as i, toDisplayString as o, createCommentVNode as c, createVNode as V } from "vue";
import "./index.vue3.js";
import v from "../../../_virtual/_plugin-vue_export-helper.js";
const h = {
  ref: "rankSingleRef",
  class: "rank-single"
}, y = {
  ref: "rankBodyRef",
  class: "rank-single-body"
}, C = ["onClick"], b = { class: "rank-info" }, D = { class: "rank-info__lf" }, N = ["onClick"], $ = ["title"], S = { class: "rank-info__rt" }, B = {
  key: 0,
  class: "rank-value"
}, R = {
  key: 1,
  class: "rank-unit"
}, E = {
  key: 1,
  class: "t-rank-single-rank-line"
}, I = ["onClick"], P = {
  key: 1,
  class: "rank-unit"
}, w = { class: "t-rank-single-difference-text" };
function z(a, k, F, H, L, M) {
  const f = _("t-proportion-bar");
  return n(), r("div", {
    class: t(["t-rank-single", a.domOnlyClass])
  }, [
    s("div", h, [
      s("div", y, [
        (n(!0), r(u, null, g(a.curData, (e, l) => (n(), r("div", {
          key: l,
          class: t(["rank-single-item", [`rank-single-item-${l}`, { "rank-single-item-checked": a.stateCollection[l] }]]),
          onClick: (d) => a.cardClickHandler(l)
        }, [
          s("div", b, [
            s("div", D, [
              s("div", {
                class: "rank-index",
                onClick: i((d) => a.handleClickIndex(l), ["stop"])
              }, o(e != null && e.rank ? e.rank : l + 1), 9, N),
              s("span", {
                class: "rank-name",
                title: e.name
              }, o(e.name || "--"), 9, $)
            ]),
            s("div", S, [
              e.currentValue || Number(e.currentValue) === 0 ? (n(), r(u, { key: 0 }, [
                e.currentValue || Number(e.currentValue) === 0 ? (n(), r("span", B, o(e.currentValue), 1)) : c("", !0),
                e.unit ? (n(), r("span", R, o(e.unit), 1)) : c("", !0)
              ], 64)) : c("", !0),
              e.currentValue && e.percentage || Number(e.currentValue) === 0 || e.currentValue && e.differenceValue || Number(e.currentValue) === 0 ? (n(), r("div", E)) : c("", !0),
              e.percentage ? (n(), r(u, { key: 2 }, [
                e.currentValue || Number(e.currentValue) === 0 ? (n(), r("span", {
                  key: 0,
                  class: "rank-percentage",
                  onClick: i((d) => a.handleClickPercentage(l), ["stop"])
                }, o(a.getPercentValue(e.percentage)), 9, I)) : c("", !0),
                e.unit ? (n(), r("span", P, "%")) : c("", !0)
              ], 64)) : c("", !0),
              e.differenceValue ? (n(), r(u, { key: 3 }, [
                s("span", w, o(a.differenceText || "-"), 1),
                e.isDifference !== 0 ? (n(), r("span", {
                  key: 0,
                  class: t({
                    "t-rank-single-increase-icon": (e == null ? void 0 : e.isDifference) === 1,
                    "t-rank-single-decrease-icon": (e == null ? void 0 : e.isDifference) === -1
                  })
                }, null, 2)) : c("", !0),
                s("span", {
                  class: t({
                    "t-rank-single-increase-text": (e == null ? void 0 : e.isDifference) === 1,
                    "t-rank-single-decrease-text": (e == null ? void 0 : e.isDifference) === -1,
                    "t-rank-single-no-difference": (e == null ? void 0 : e.isDifference) === 0
                  })
                }, o(e.differenceValue || "-"), 3)
              ], 64)) : c("", !0)
            ])
          ]),
          V(f, {
            "target-value": e.targetValue,
            "current-value": e.currentValue,
            percentage: e.percentage,
            "icon-src": e.iconSrc || a.curIconSrc,
            class: "t-rank-single-proportion-bar"
          }, null, 8, ["target-value", "current-value", "percentage", "icon-src"])
        ], 10, C))), 128))
      ], 512),
      k[0] || (k[0] = s("div", { class: "style-container" }, null, -1))
    ], 512)
  ], 2);
}
const A = /* @__PURE__ */ v(p, [["render", z]]);
export {
  A as default
};
