// @ths/design/es/components/rank-single/src/index.vue2.js
import { defineComponent as z, computed as D, ref as g, watch as S, nextTick as h, onMounted as G, onUnmounted as Q } from "vue";
import J from "lodash/cloneDeep";
import X from "lodash/merge";
import "../../proportion-bar/index.js";
import { getDomRandomClass as Z, setStyle as i, getCssPropertyString as s } from "../../../utils/index.js";
import ee from "../../proportion-bar/src/index.vue.js";
const A = {
  name: "name",
  targetValue: "targetValue",
  currentValue: "currentValue",
  percentage: "percentage",
  unit: "unit",
  rank: "rank",
  isDifference: "isDifference",
  differenceValue: "differenceValue"
}, oe = z({
  name: "TRankSingle",
  components: { TProportionBar: ee },
  props: {
    // 设置关键字，用于设置默认单选
    keyWord: {
      type: String,
      default: ""
    },
    // 设置默认选中
    defaultChecked: {
      type: String,
      default: ""
    },
    // 差异文本
    differenceText: {
      type: String,
      default: ""
    },
    // 如果是选项卡状态，又当没有设置默认选中项时，会自动初始默认选中第一项
    tab: {
      type: Boolean,
      default: !1
    },
    data: {
      type: Array,
      default: () => []
    },
    // 别名
    alias: {
      type: Object,
      default: () => J(A)
    },
    // 每行数据样式
    dataStyle: {
      type: Array,
      default: () => []
    },
    // 进度条图标路径
    iconSrc: {
      type: String,
      default: ""
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        },
        // 内层容器
        innerContainer: {
          default: {}
        },
        // 列表集合容器
        listContainer: {
          default: {}
        },
        itemWrapper: {
          default: {},
          hover: {},
          active: {}
        },
        index: {
          default: {}
        },
        name: {
          default: {}
        },
        proportionBarBg: {
          default: {}
        },
        proportionBarInner: {
          default: {}
        },
        // 进度条外层父元素容器
        proportionBar: {
          default: {}
        },
        value: {
          default: {}
        },
        unit: {
          default: {}
        },
        percentage: {
          default: {}
        },
        icon: {
          default: {}
        },
        // 分割符
        separator: {
          default: {}
        },
        // 差异文本
        differenceText: {
          default: {}
        },
        // 增长图标
        increaseIcon: {
          default: {}
        },
        // 减少图标
        decreaseIcon: {
          default: {}
        },
        // 增加状态数值
        increaseValue: {
          default: {}
        },
        // 降低数值
        decreaseValue: {
          default: {}
        },
        // 没有变化，数据持平
        noDifference: {
          default: {}
        }
      })
    },
    // 开启自动滚动
    autoScroll: {
      type: Boolean,
      default: !1
    },
    // 滚动速度 数字越大越快
    speed: {
      type: Number,
      default: () => 10
    }
  },
  emits: [
    "card-click",
    // 单项点击
    "click-index",
    // 点击索引
    "click-percentage"
    // 点击百分比
  ],
  setup(t, { emit: I }) {
    const a = D(() => t.cStyle), r = Z("rank-single"), d = g(t.data), L = D(() => t.iconSrc), R = (e) => {
      var o;
      return ((o = e.toString()) == null ? void 0 : o.indexOf("%")) !== -1 ? e == null ? void 0 : e.split("%")[0] : e;
    };
    let f = null;
    const l = g(null), p = g(null), y = g(0), v = g([]), k = g(-1), U = () => {
      var o, n, u, c;
      let e = "";
      return a.value.wrapper && (e += i(a.value.wrapper, `.${r}`)), a.value.innerContainer && (e += i(a.value.innerContainer, `.${r} .rank-single`)), a.value.listContainer && (e += i(a.value.listContainer, `.${r} .rank-single-body`)), a.value.itemWrapper && (e += i(a.value.itemWrapper, `.${r} .rank-single-item`)), (o = a.value.itemWrapper) != null && o.hover && (e += `.${r} .rank-single-item:hover {
          ${s((n = a.value.itemWrapper) == null ? void 0 : n.hover)}
        }`), (u = a.value.itemWrapper) != null && u.active && (e += `.${r} .rank-single-item-checked {
          ${s((c = a.value.itemWrapper) == null ? void 0 : c.active)}
        }`), a.value.index && (e += i(a.value.index, `.${r} .rank-single-item .rank-info .rank-info__lf .rank-index`)), a.value.name && (e += i(a.value.name, `.${r} .rank-single-item .rank-info .rank-info__lf .rank-name`)), a.value.proportionBar && (e += i(a.value.proportionBar, `.${r} .t-rank-single-proportion-bar`)), a.value.proportionBarBg && (e += i(a.value.proportionBarBg, `.${r} .rank-single-item .proportion-bar__outer`)), a.value.proportionBarInner && (e += i(a.value.proportionBarInner, `.${r} .rank-single-item .proportion-bar__inner`)), a.value.value && (e += i(a.value.value, `.${r} .rank-single-item .rank-info .rank-info__rt .rank-value`)), a.value.percentage && (e += i(a.value.percentage, `.${r} .rank-single-item .rank-info .rank-info__rt .rank-percentage`)), a.value.unit && (e += i(a.value.unit, `.${r} .rank-single-item .rank-info .rank-info__rt .rank-unit`)), a.value.icon && (e += i(a.value.icon, `.${r} .t-proportion-bar .proportion-bar__icon`)), a.value.separator && (e += i(a.value.separator, `.${r} .rank-single-item .t-rank-single-rank-line`)), a.value.differenceText && (e += i(a.value.differenceText, `.${r} .t-rank-single-difference-text`)), a.value.increaseIcon && (e += i(a.value.increaseIcon, `.${r} .t-rank-single-increase-icon`)), a.value.decreaseIcon && (e += i(a.value.decreaseIcon, `.${r} .t-rank-single-decrease-icon`)), a.value.increaseValue && (e += i(a.value.increaseValue, `.${r} .t-rank-single-increase-text`)), a.value.decreaseValue && (e += i(a.value.decreaseValue, `.${r} .t-rank-single-decrease-text`)), a.value.noDifference && (e += i(a.value.noDifference, `.${r} .t-rank-single-no-difference`)), e;
    }, j = () => {
      let e = "";
      const o = `
        -webkit-background-clip: text;
        -webkit-text-fill-color: transparent;
      `;
      return d.value.forEach((n, u) => {
        var c, m, C, V, O, N, W, H, w, E, P, T;
        if ((c = n.style) != null && c.rankIndex) {
          const K = `url(${n.style.rankIndex.backgroundImage || ((m = n.style.rankIndex) == null ? void 0 : m.background)})`;
          e += `
          .${r} .rank-single-item.rank-single-item-${u} .rank-info__lf .rank-index {
            ${s(n.style.rankIndex)}
            background-image:
            ${(((C = n.style.rankIndex) == null ? void 0 : C.backgroundImage) || ((V = n.style.rankIndex) == null ? void 0 : V.background) || "").indexOf("url") === -1 ? K : n.style.rankIndex.backgroundImage || ((O = n.style.rankIndex) == null ? void 0 : O.background) || ""};
          }
        `;
        }
        (N = n.style) != null && N.rankName && (e += `
          .${r} .rank-single-item.rank-single-item-${u} .rank-info__lf .rank-name {
            ${s(n.style.rankName)}
          }
        `), (W = n.style) != null && W.rankValue && (e += `
            .${r} .rank-single-item.rank-single-item-${u} .rank-info__rt .rank-value {
              ${s(n.style.rankValue)}
              ${(n.style.rankValue.background || n.style.rankValue.backgroundImage || "").indexOf("linear-gradient") !== -1 ? o : ""}
            }
          `), (H = n.style) != null && H.rankUnit && (e += `
            .${r} .rank-single-item.rank-single-item-${u} .rank-info__rt .rank-unit {
              ${s(n.style.rankUnit)}
              ${(n.style.rankUnit.background || n.style.rankUnit.backgroundImage || "").indexOf("linear-gradient") !== -1 ? o : ""}
            }
          `), (w = n.style) != null && w.rankPercentage && (e += `
            .${r} .rank-single-item.rank-single-item-${u} .rank-info__rt .rank-percentage {
              ${s(n.style.rankPercentage)}
              ${(n.style.rankPercentage.background || n.style.rankPercentage.backgroundImage || "").indexOf("linear-gradient") !== -1 ? o : ""}
            }
          `), (E = n.style) != null && E.proportionBarInner && (e += `
          .${r} .rank-single-item.rank-single-item-${u} .proportion-bar__inner {
            ${s(n.style.proportionBarInner)}
          }
        `), (P = n.style) != null && P.proportionBarBg && (e += `
          .${r} .rank-single-item.rank-single-item-${u} .proportion-bar__outer {
            ${s(n.style.proportionBarBg)}
          }
        `), (T = n.style) != null && T.proportionBarIcon && (e += `
          .${r} .rank-single-item.rank-single-item-${u} .proportion-bar__icon {
            ${s(n.style.proportionBarIcon)}
          }
        `);
      }), e;
    }, b = () => {
      const e = document.querySelector(`.${r} .style-container`);
      e && (e.innerHTML = `<style>${U()}${j()}}</style>`);
    }, $ = (e) => {
      k.value !== -1 && e !== k.value && (v.value[k.value] = !1), e !== k.value && (v.value[e] = !0, k.value = e);
    }, M = (e) => {
      $(e), I("card-click", JSON.parse(JSON.stringify(d.value[e])));
    }, _ = (e) => {
      const o = p.value.offsetHeight, n = l.value.offsetHeight, u = l.value.children[0].offsetHeight;
      let c = Number(l.value.style.top.slice(0, -2));
      n - o >= u && (clearInterval(f), f = setInterval(() => {
        c = Number(l.value.style.top.slice(0, -2)), l.value.style.transition = `top ${1 - t.speed / 10 + 0.2}s ease-in-out`, y.value === 0 && (l.value.style.top = `${c - u}px`), c = Number(l.value.style.top.slice(0, -2)), y.value === 1 && (l.value.style.transition = "", l.value.style.top = "0px", l.value.appendChild(l.value.children[0]), y.value = -1), y.value++;
      }, 100 * (10 - t.speed + 2)));
    }, B = () => {
      clearInterval(f), l.value.style.top = "0px", y.value = 0, l.value.style.transition = `top ${1 - t.speed / 10 + 0.2}s ease-in-out`, p.value.style.overflowY = "auto";
    }, x = () => {
      clearInterval(f), p.value.style.overflowY = "hidden", _(l.value);
    }, Y = () => {
      h(() => {
        l.value, _(), l.value.addEventListener("mouseenter", B), l.value.addEventListener("mouseleave", x);
      });
    }, q = (e) => {
      $(e), I("click-index", JSON.parse(JSON.stringify(d.value[e])));
    }, F = (e) => {
      $(e), I("click-percentage", JSON.parse(JSON.stringify(d.value[e])));
    };
    return S(() => [t.cStyle], () => {
      b();
    }), S(() => [t.autoScroll, t.data, t.speed], () => {
      var e;
      clearImmediate(f), t.autoScroll && ((e = t.data) != null && e.length) ? Y() : clearInterval(f);
    }, { immediate: !0 }), S(() => [t.data, t.alias, t.dataStyle], () => {
      const e = X(J(A), t.alias);
      d.value = t.data.map((o, n) => {
        var c;
        const u = {};
        return Object.keys(e).forEach((m) => {
          u[m] = o[e[m]];
        }), {
          ...o,
          ...u,
          // 单条进度条图标
          iconSrc: t.dataStyle.length && ((c = t.dataStyle[n]) == null ? void 0 : c.iconSrc) || "",
          // 单条进度条样式
          style: t.dataStyle.length ? t.dataStyle[n] || {} : {}
        };
      }), b();
    }, {
      deep: !0,
      immediate: !0
    }), S(() => [t.data, t.keyWord, t.defaultChecked, t.tab], () => {
      if (t.data.length > 0 && (v.value = new Array(t.data.length).fill(!1)), t.data.length > 0 && t.keyWord !== "" && t.defaultChecked !== "") {
        const e = d.value.findIndex((o) => String(o[t.keyWord]) === String(t.defaultChecked));
        e >= 0 && h(() => {
          v.value[e] = !0, k.value = e;
        });
      } else
        t.tab && h(() => {
          v.value[0] = !0, k.value = 0;
        });
    }, {
      deep: !0,
      immediate: !0
    }), G(() => {
      b();
    }), Q(() => {
      var e, o;
      f && clearInterval(f), (e = l.value) == null || e.removeEventListener("mouseenter", B), (o = l.value) == null || o.removeEventListener("mouseleave", x);
    }), {
      curData: d,
      curStyle: a,
      domOnlyClass: r,
      curIconSrc: L,
      getPercentValue: R,
      rankBodyRef: l,
      rankSingleRef: p,
      cardClickHandler: M,
      stateCollection: v,
      handleClickIndex: q,
      handleClickPercentage: F,
      setCheckedItem: $
    };
  }
});
export {
  oe as default
};
