// @ths/design/es/components/chart-angle-axis/src/index.vue2.js
import { defineComponent as j, computed as T, ref as k, reactive as K, onMounted as q, onUnmounted as M, watch as C, toRefs as _, nextTick as N } from "vue";
import * as F from "echarts";
import A from "lodash/merge";
import z from "lodash/cloneDeep";
import { DEFAULT_OPTIONS_COMMON as H } from "../../../utils/constant.js";
import { getDomRandomClass as P, setStyle as S } from "../../../utils/index.js";
const R = A(z(H), {
  tooltip: {
    trigger: "item",
    formatter: "{a}<br />{c0}"
  },
  legend: {
    left: "right"
  },
  xAxis: {
    show: !1
  },
  yAxis: {
    show: !1
  },
  polar: {},
  angleAxis: {
    type: "category",
    boundaryGap: !1,
    splitLine: {
      show: !0,
      lineStyle: {
        color: "#999",
        type: "dashed"
      }
    },
    axisLine: {
      show: !1
    }
  },
  radiusAxis: {
    type: "category",
    axisLine: {
      show: !1
    },
    axisLabel: {
      rotate: 45
    }
  }
}), V = j({
  name: "TChartAngleAxis",
  props: {
    // 暂无数据图片路径
    emptyImg: {
      type: String,
      default: ""
    },
    // 暂无数据图片适应指定容器的高度与宽度
    objectFit: {
      type: String,
      default: "contain"
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    },
    // echarts原生配置
    options: {
      type: Object,
      default: () => ({}),
      required: !1
    },
    // 传入数组数据 如果不传入数据 会使用默认数据  如果options设置了 series 属性 则会忽略data的数据
    data: {
      type: Array,
      default: () => [],
      required: !1
    },
    // 包含 系列样式 别名设置
    seriesSetting: {
      type: Object,
      default: () => ({
        // 角度轴类目取resData数据中的别名设置
        angleAxisKey: "",
        // 经向轴类目取resData数据中的别名设置
        radiusAxisKey: "",
        // 取resData数据中的别名设置
        data: {
          type: Array,
          default: () => [],
          required: !1
        }
      })
    }
  },
  emits: ["com-click", "legend-select"],
  setup(e, { emit: O }) {
    const f = P("chart-angle-axis"), r = T(() => e.cStyle), v = k(null);
    let u = null;
    const a = K({
      // 图表数据
      chartData: [],
      // 径向轴数据
      radiusAxisData: [],
      // 角度轴数据
      angleAxisData: [],
      // 平均值
      avgData: 0
    }), I = () => {
      let t = "";
      return r.value.wrapper && (t += S(r.value.wrapper, `.${f}`)), r.value.emptyContainer && (t += S(r.value.emptyContainer, `.${f} .empty`)), r.value.emptyImg && (t += S(r.value.emptyImg, `.${f} .empty .empty-img`)), r.value.emptyText && (t += S(r.value.emptyText, `.${f} .empty .empty-text`)), t;
    }, E = () => {
      const t = document.querySelector(`.${f} .style-container`);
      t && (t.innerHTML = `<style>${I()}</style>`);
    }, $ = T(() => {
      var t, l, h, y;
      if (e.options) {
        if (Array.isArray((t = e.options) == null ? void 0 : t.series))
          return e.options.series.some((i) => {
            var c;
            return (i == null ? void 0 : i.data) && ((c = i == null ? void 0 : i.data) == null ? void 0 : c.length) !== 0;
          });
        if (typeof ((l = e.options) == null ? void 0 : l.series) == "object")
          return ((y = (h = e.options.series) == null ? void 0 : h.data) == null ? void 0 : y.length) === 0;
      }
      return !!e.data.length;
    }), L = () => {
      var t, l;
      if ((l = (t = e.seriesSetting) == null ? void 0 : t.data) != null && l.length && e.data) {
        a.radiusAxisData = [], a.angleAxisData = [];
        const { radiusAxisKey: h, angleAxisKey: y } = e.seriesSetting;
        e.seriesSetting.data.forEach((p, i) => {
          const { dataKey: c, name: m } = p;
          a.chartData.push({ name: m || "序列名", data: [] });
          let x = 0, s = z(e.data);
          e.data.forEach((n, o) => {
            const d = n[h] || "径向轴名", D = n[y] || "角度轴名", g = Number(n[c] || 0);
            x += g, a.radiusAxisData.includes(d) || a.radiusAxisData.push(d), a.angleAxisData.includes(D) || a.angleAxisData.push(D), a.chartData[i].data[o] = [d, D, g];
          }), s.length > 7 && (s.sort((n, o) => n - o), s = s.slice(1).slice(0, s.length - 2), a.avgData = x / s.length);
        });
      }
    }, w = () => {
      var y, p, i;
      if (!u)
        return;
      const { series: t } = e.options;
      let l = [];
      t ? l = t.map((c, m) => {
        var s, n, o;
        const x = A(
          {
            type: "scatter",
            // 散点气泡图
            coordinateSystem: "polar",
            // 使用极坐标系
            symbol: "circle",
            symbolSize(d) {
              return d[2] * 2;
            }
          },
          (o = (n = (s = e.seriesSetting) == null ? void 0 : s.data) == null ? void 0 : n[m]) == null ? void 0 : o.style
        );
        return A({}, x, c);
      }) : e.data && ((p = (y = e.seriesSetting) == null ? void 0 : y.data) != null && p.length) && (l = (i = e.seriesSetting) == null ? void 0 : i.data.map((c, m) => {
        var s, n, o, d, D;
        const x = A(
          {
            name: (s = a.chartData[m]) == null ? void 0 : s.name,
            type: "scatter",
            // 散点气泡图
            coordinateSystem: "polar",
            // 使用极坐标系
            symbol: "circle",
            symbolSize(g) {
              return g[2] > a.avgData ? 20 : g[2] <= a.avgData ? g[2] / a.avgData * 20 : g[2];
            },
            data: e.data.length ? (n = a.chartData[m]) == null ? void 0 : n.data : []
          },
          (D = (d = (o = e.seriesSetting) == null ? void 0 : o.data) == null ? void 0 : d[m]) == null ? void 0 : D.style
        );
        return A({}, x, c);
      }));
      const h = A(
        {
          series: l,
          angleAxis: {
            data: a.angleAxisData
          },
          radiusAxis: {
            data: a.radiusAxisData
          }
        },
        R,
        e.options
      );
      u.clear(), u.setOption(h);
    }, b = () => {
      N(() => {
        u.resize();
      });
    };
    return q(() => {
      E(), v.value && (u = F.init(v.value), L(), w(), u.on("click", (t) => {
        O("com-click", t);
      }), u.on("legendselectchanged", (t) => {
        O("legend-select", t);
      })), window.addEventListener("resize", b);
    }), M(() => {
      window.removeEventListener("resize", b);
    }), C(
      [
        () => e.options
      ],
      () => {
        w();
      },
      {
        deep: !0,
        immediate: !0
      }
    ), C(
      [
        () => e.data,
        () => e.seriesSetting
      ],
      () => {
        L(), w();
      },
      {
        deep: !0
      }
    ), C(() => e.cStyle, () => {
      E(), b();
    }, { deep: !0 }), {
      ..._(a),
      chartId: v,
      curStyle: r,
      checkEmpty: $,
      domOnlyClass: f
    };
  }
});
export {
  V as default
};
