// @ths/design/es/components/chart-angle-axis/src/index.vue.js
import a from "./index.vue2.js";
import { openBlock as n, createElementBlock as m, normalizeClass as c, normalizeStyle as o, withDirectives as r, createElementVNode as e, vShow as i } from "vue";
import "./index.vue3.js";
import p from "../../../_virtual/_plugin-vue_export-helper.js";
const d = {
  ref: "chartId",
  class: "chart"
}, u = { class: "empty" }, y = ["src"];
function f(t, s, h, v, E, g) {
  var l;
  return n(), m("div", {
    class: c(["t-chart-angle-axis", t.domOnlyClass]),
    style: o({ ...(l = t.cStyle) == null ? void 0 : l.wrapper.default })
  }, [
    r(e("div", d, null, 512), [
      [i, t.checkEmpty]
    ]),
    r(e("div", u, [
      e("img", {
        class: "empty-img",
        style: o({
          "object-fit": t.objectFit
        }),
        src: t.emptyImg,
        alt: ""
      }, null, 12, y),
      s[0] || (s[0] = e("span", { class: "empty-text" }, "暂无数据", -1))
    ], 512), [
      [i, !t.checkEmpty]
    ]),
    s[1] || (s[1] = e("div", { class: "style-container" }, null, -1))
  ], 6);
}
const C = /* @__PURE__ */ p(a, [["render", f]]);
export {
  C as default
};
