// @ths/design/es/components/pagination/src/index.vue.js
import E from "./index.vue2.js";
import { resolveComponent as i, openBlock as n, createElementBlock as t, normalizeClass as p, normalizeStyle as m, createBlock as o, createCommentVNode as s, createElementVNode as r, createVNode as u, withCtx as g, Fragment as P, renderList as J, toDisplayString as N } from "vue";
import "./index.vue3.js";
import h from "../../../_virtual/_plugin-vue_export-helper.js";
const b = { class: "pagination" };
function c(e, a, w, O, V, j) {
  const d = i("total"), y = i("page-size"), v = i("Pre"), C = i("el-option"), k = i("el-select"), S = i("Pager"), f = i("Next"), z = i("Jump");
  return n(), t("div", {
    ref: "pagination",
    class: p(["pagination-block", [e.paginationBlock, e.disable ? e.disableStyle : ""]]),
    style: m([e.isShowAllNum ? "justify-content: space-between;" : "", e.mini ? e.miniStyle : ""])
  }, [
    e.isShowAllNum ? (n(), o(d, {
      key: 0,
      mini: e.mini,
      "tag-class": e.tagClass,
      total: e.total,
      "show-all-data-example": e.showAllDataExample
    }, null, 8, ["mini", "tag-class", "total", "show-all-data-example"])) : s("", !0),
    r("div", b, [
      u(y, {
        "page-size": e.pageSizes,
        zx: e.sizeStyle,
        onPagesizeChange: e.handlePagesizeChange
      }, null, 8, ["page-size", "zx", "onPagesizeChange"]),
      e.isOver || e.isOverLong || e.isExtremeSimple ? (n(), o(v, {
        key: 0,
        mini: e.mini,
        "current-page": e.currentPage,
        "prev-text": e.prevText,
        "direction-style": e.directionClass,
        onClick: a[0] || (a[0] = (l) => e.isExtremeSimple ? e.toPre("simple") : e.toPre("ordinary"))
      }, null, 8, ["mini", "current-page", "prev-text", "direction-style"])) : s("", !0),
      e.isExtremeSimple ? (n(), t("div", {
        key: 1,
        class: p(["extreme-model", e.extremeClass])
      }, [
        u(k, {
          modelValue: e.selectedExtreme,
          "onUpdate:modelValue": a[1] || (a[1] = (l) => e.selectedExtreme = l),
          onChange: e.handleExtremevalChange
        }, {
          default: g(() => [
            (n(!0), t(P, null, J(e.extremeData, (l) => (n(), o(C, {
              key: l.val,
              label: l.text,
              value: l.val
            }, {
              default: g(() => [
                r("span", {
                  style: m(e.innerContentStyle)
                }, N(l.text), 5)
              ]),
              _: 2
            }, 1032, ["label", "value"]))), 128))
          ]),
          _: 1
        }, 8, ["modelValue", "onChange"])
      ], 2)) : s("", !0),
      e.isExtremeSimple ? s("", !0) : (n(), o(S, {
        key: 2,
        mini: e.mini,
        "page-list-value": e.pageListValue,
        "page-class": e.pageClass,
        "direction-style": e.directionClass,
        onJumpQuick: e.quickJump,
        onJumpOrinary: e.jumpTo
      }, null, 8, ["mini", "page-list-value", "page-class", "direction-style", "onJumpQuick", "onJumpOrinary"])),
      a[3] || (a[3] = r("div", { class: "style-container" }, null, -1)),
      e.isOver || e.isOverLong || e.isExtremeSimple ? (n(), o(f, {
        key: 3,
        mini: e.mini,
        page: e.page,
        "current-page": e.currentPage,
        "next-text": e.nextText,
        "page-sum": e.page,
        "direction-style": e.directionClass,
        onClick: a[2] || (a[2] = (l) => e.isExtremeSimple ? e.toNext("simple") : e.toNext("ordinary"))
      }, null, 8, ["mini", "page", "current-page", "next-text", "page-sum", "direction-style"])) : s("", !0),
      e.isOpenQuickJump ? (n(), o(z, {
        key: 4,
        "jump-area-class": e.jumpAreaStyle,
        onJumpAcross: e.jumpTo
      }, null, 8, ["jump-area-class", "onJumpAcross"])) : s("", !0)
    ])
  ], 6);
}
const T = /* @__PURE__ */ h(E, [["render", c], ["__scopeId", "data-v-3b64df99"]]);
export {
  T as default
};
