// @ths/design/es/components/pagination/src/index.vue2.js
import { defineComponent as G, computed as L, reactive as J, ref as n, watch as K, onMounted as U } from "vue";
import { getDomRandomClass as d, setStyle as O, getCssPropertyString as i } from "../../../utils/index.js";
import W from "./components/total.vue.js";
import Y from "./components/size.vue.js";
import Z from "./components/prev.vue.js";
import ee from "./components/next.vue.js";
import le from "./components/pager.vue.js";
import te from "./components/jump.vue.js";
const ge = G({
  name: "TPagination",
  components: {
    Total: W,
    PageSize: Y,
    Pre: Z,
    Next: ee,
    Pager: le,
    Jump: te
  },
  props: {
    // 总数
    total: {
      type: Number,
      default: 0
    },
    // 每页显示条数
    pageSizes: {
      type: Array,
      default: () => []
    },
    // 是否展示数据总条数
    isShowAllNum: {
      type: Boolean,
      default: !1
    },
    // 是否开启快速跳转
    isOpenQuickJump: {
      type: Boolean,
      default: !1
    },
    // 是否开启极简模式
    isExtremeSimple: {
      type: Boolean,
      default: !1
    },
    // 是否开启迷你模式
    mini: {
      type: Boolean,
      default: !1
    },
    // 是否开启禁用模式
    disable: {
      type: Boolean,
      default: !1
    },
    // 跳转个数
    skipNum: {
      type: Number,
      default: 5
    },
    // 默认跳转页
    defaultPage: {
      type: Number,
      default: 1
    },
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {
            "background-color": "#f3f3f3;"
          }
        },
        // 页码部分
        page: {
          default: {
            border: "1px solid #dcdcdc;"
          },
          hover: {
            border: "1px solid var(--theme-color)",
            color: "var(--theme-color)"
          },
          focus: {
            backgroundColor: "red",
            color: "yellow"
          }
        },
        // 下拉框部分
        selectDropArea: {
          default: {},
          hover: {},
          active: {}
        },
        // 左右标签样式
        directionClass: {
          default: {}
        },
        // 数据总量展示标签
        countTag: {
          default: {
            fontSize: "14px",
            color: "#000"
          }
        },
        // 页面数量选择器自定义样式
        selectNum: {
          default: {},
          hover: {},
          focus: {}
        },
        // 跳转框样式
        jumpAreaStyle: {
          default: {}
        },
        // 跳转框下的输入框样式
        inputStyle: {
          default: {},
          hover: {},
          focus: {}
        },
        // 极简模式
        extreme: {
          default: {},
          hover: {},
          focus: {}
        }
      })
    },
    // 数据总量展示模板
    showAllDataExample: {
      type: String,
      default: "总共XXX条数据"
    },
    // 自定义上一页内容
    prevText: {
      type: String,
      default: "-"
    },
    // 自定义下一页内容
    nextText: {
      type: String,
      default: "-"
    }
  },
  emits: ["page-change", "size-change"],
  setup(r, b) {
    const u = L(() => r.cStyle), _ = d("pagination"), w = d("page-text"), F = d("tag-class"), h = d("extremeClass"), $ = d("selectNum"), g = d("jumpArea"), j = d("directionClass"), q = () => {
      let l = "";
      return u.value.selectNum && u.value.selectNum.focus && u.value.selectNum.hover && (l += `
          .${$}  .el-select .el-input.is-focus .el-input__inner {
            ${i(u.value.selectNum.focus)}
          }
          .${$}  .el-input--small .el-input__inner {
            ${i(u.value.selectNum.default)}
          }
          .${$}  .el-input--small .el-input__inner:hover {
            ${i(u.value.selectNum.hover)}
          }
          .el-select-dropdown__list {
            ${i(u.value.selectDropArea.default)}
          }
          .el-select-dropdown__item{
            ${i(u.value.selectDropArea.default)}
          }
          .el-select-dropdown__item:hover{
            ${i(u.value.selectDropArea.hover)}
          }
          .el-select-dropdown__item:active{
            ${i(u.value.selectDropArea.active)}
          }
        `), u.value.jumpAreaStyle && u.value.inputStyle && (l += `
          .${g} {
            ${i(u.value.jumpAreaStyle.default)}
          }
          .${g} .el-input--small .el-input__inner {
            ${i(u.value.inputStyle.default)}
          }
          .${g} .el-input--small .el-input__inner:hover {
            ${i(u.value.inputStyle.hover)}
          }
          .${g} .el-input--small .el-input__inner:focus {
            ${i(u.value.inputStyle.focus)}
          }
        `), u.value.extreme && (l += `
          .${h} .el-input--small .el-input__inner {
            ${i(u.value.extreme.default)}
          }
          .${h} .el-input--small .el-input__inner:hover {
            ${i(u.value.extreme.hover)}
          }
          .${h} .el-input--small .el-input__inner:focus {
            ${i(u.value.extreme.focus)}
          }
        `), u.value.directionClass && (l += `
          .${j} {
            ${i(u.value.directionClass.default)}
          }
        `), l;
    }, k = () => {
      let l = "";
      return u.value.wrapper && (l += O(u.value.wrapper, `.${_}`)), u.value.page && (l += O(u.value.page, `.${w}`)), l;
    }, D = () => {
      const l = document.querySelector(
        `.${_} .style-container`
      );
      l && (l.innerHTML = `<style>${k()}${q()}</style>`);
    }, H = J({
      height: "56px"
    }), M = J({
      "justify-content": "space-around;"
    }), X = n("disableClass"), V = n("innerContentStyle"), I = n(!1), C = n(!1), s = n(1), S = n(1), P = n(10), a = L(() => Math.ceil(r.total / P.value)), Q = n(!1), R = n(null), N = n(`1/${a.value}`), o = n([]), t = n([]), z = n([]), x = () => {
      const l = (e) => e.val === o.value[s.value - 1].val;
      S.value = t.value.findIndex(l);
    }, y = (l) => {
      const e = (f) => f.val === l, c = t.value.findIndex(e);
      t.value[c].selected = !0;
    }, v = (l, e) => {
      const c = [];
      for (let f = l; f <= e; f++)
        c.push(o.value[f - 1]);
      return c;
    }, m = (l) => {
      let e = l;
      (e > a.value || e < 1) && (e < 1 ? e = 1 : e = a.value), x(), t.value[S.value].selected = !1, typeof e == "string" && (e = parseInt(e, 10)), a.value <= 5 ? y(e) : a.value > 5 && a.value <= 10 ? s.value >= 1 && s.value <= 5 && e >= 1 && e <= 5 || s.value > 5 && s.value <= 10 && e > 5 ? y(e) : s.value >= 1 && s.value <= 5 && e > 5 ? (t.value.splice(1, 6), t.value.splice(1, 0, {
        text: "...",
        val: "...",
        selected: !1
      }), t.value.splice(
        2,
        0,
        ...v(a.value - 4, a.value)
      )) : s.value > 5 && s.value <= 10 && e <= 5 && (t.value.splice(1, 5, {
        text: "...",
        val: "...",
        selected: !1
      }), t.value.splice(1, 0, ...v(2, 5))) : a.value > 10 && (t.value.length === 7 ? t.value[1].val === "..." ? e >= a.value - 4 && e <= a.value ? y(e) : e >= 1 && e <= 5 ? (t.value.splice(1, 5), t.value.splice(1, 0, {
        text: "...",
        val: "...",
        selected: !1
      }), t.value.splice(1, 0, ...v(2, 5))) : (t.value.splice(2, 5), t.value[1].text = "...", t.value.splice(1, 0, {
        text: "...",
        val: "...",
        selected: !1
      }), t.value.splice(
        2,
        0,
        ...v(e - 2, e + 2)
      ), t.value.push(o.value[a.value - 1])) : e >= 1 && e <= 5 ? y(e) : e > 5 && (e <= a.value && e >= a.value - 4 ? (t.value.splice(1, 4), t.value.splice(
        2,
        0,
        ...v(a.value - 4, a.value - 1)
      )) : (t.value.splice(1, 4), t.value.splice(1, 0, {
        text: "...",
        val: "...",
        selected: !1
      }), t.value.splice(
        2,
        0,
        ...v(e - 2, e + 2)
      ))) : e >= 1 && e <= 5 ? (t.value.splice(1, 6), t.value.splice(1, 0, ...v(2, 5))) : e > 5 && e <= a.value - 5 ? t.value.splice(
        2,
        5,
        ...v(e - 2, e + 2)
      ) : e >= a.value - 4 && e <= a.value && (t.value.splice(1, 6), t.value[1].text = "...", t.value.splice(
        2,
        0,
        ...v(a.value - 4, a.value - 1)
      )));
      const c = (p) => p.val === e, f = t.value.findIndex(c);
      t.value[f].selected = !0, s.value = e, b.emit("page-change", e);
    }, T = () => {
      o.value.length && (o.value = []);
      const l = a.value;
      let e = "", c = 0;
      const f = !1;
      for (let p = 0; p < l; p++)
        e = `${p + 1}`, c = p + 1, p === 0 ? o.value.push({ text: e, val: c, selected: !0 }) : o.value.push({ text: e, val: c, selected: f });
      l > 10 ? C.value = !0 : l > 5 && (I.value = !0);
    }, E = () => {
      t.value = o.value, C.value ? (t.value = t.value.slice(0, 5), t.value.push({ text: "...", val: "...", selected: !1 }), t.value.push(o.value[a.value - 1])) : a.value > 5 && (t.value = t.value.slice(0, 5), t.value[5] = { text: "...", val: "...", selected: !1 }, t.value.push(o.value[a.value - 1]));
    }, B = () => {
      let l;
      for (let e = 1; e <= a.value; e++)
        l = {
          text: "",
          val: ""
        }, l.text = `${e}/${a.value}`, l.val = e, z.value.push(l);
    }, A = () => {
      T(), E(), r.isExtremeSimple && B();
    };
    return K(
      () => [r.cStyle],
      () => {
        D();
      },
      {
        deep: !0
      }
    ), U(() => {
      if (D(), A(), r.defaultPage !== 1) {
        let l = r.defaultPage;
        r.defaultPage > a.value ? l = a.value : r.defaultPage < 1 && (l = 1), m(l);
      }
    }), {
      /* 基本变量 */
      page: a,
      pageList: o,
      pageListValue: t,
      currentPage: s,
      isOver: I,
      currentIndex: S,
      isOverLong: C,
      selectedPageNum: P,
      extremeData: z,
      selectedExtreme: N,
      pagination: R,
      paginationBlock: _,
      pageClass: w,
      isFocus: Q,
      tagClass: F,
      extremeClass: h,
      sizeStyle: $,
      jumpAreaStyle: g,
      directionClass: j,
      /* 样式 */
      isShowAllNumStyle: M,
      miniStyle: H,
      disableStyle: X,
      innerContentStyle: V,
      /* 方法 */
      toPre: (l) => {
        x(), s.value !== 1 && (l === "ordinary" ? m(s.value - 1) : l === "simple" && (N.value = `${s.value - 1}/${a.value}`, s.value -= 1));
      },
      jumpTo: m,
      toNext: (l) => {
        x(), s.value !== a.value && (l === "ordinary" ? m(s.value + 1) : l === "simple" && (N.value = `${s.value + 1}/${a.value}`, s.value += 1));
      },
      getPageIndex: x,
      handlePagesizeChange: (l) => {
        P.value = l, A(), b.emit("size-change", l);
      },
      handleExtremevalChange: (l) => {
        s.value = l;
      },
      constructTempPageArr: v,
      directJump: y,
      initPageList: T,
      initPageListValue: E,
      initAll: A,
      quickJump: (l) => {
        const e = r.skipNum;
        x(), t.value[S.value].selected = !1, l.text === "《" ? m(s.value - e) : m(s.value + e);
      },
      initExtremeData: B,
      setCommonStyle: k
    };
  }
});
export {
  ge as default
};
