// @ths/design/es/components/time-axis-complex/src/index.vue.js
import n from "./index.vue2.js";
import { openBlock as l, createElementBlock as o, normalizeClass as c, createElementVNode as s, Fragment as d, renderList as m, toDisplayString as a } from "vue";
import "./index.vue3.js";
import p from "../../../_virtual/_plugin-vue_export-helper.js";
const g = { class: "t-time-axis-complex-box" }, _ = ["src"], u = { class: "t-time-axis-complex-box_center" }, f = { class: "center-yellow" }, h = ["onClick"], k = { class: "center-date" }, w = { class: "center-date_start" }, v = { class: "center-date_end" }, C = { class: "center-arrow" }, y = { class: "center-arrow_top" }, D = ["src"], b = ["src"];
function S(e, t, $, A, L, B) {
  return l(), o("div", {
    class: c(["t-time-axis-complex", e.domOnlyClass])
  }, [
    s("div", g, [
      s("img", {
        class: "image-left",
        src: e.imageLeft,
        alt: "左",
        onClick: t[0] || (t[0] = (...i) => e.clickLeftSwitch && e.clickLeftSwitch(...i)),
        draggable: "false"
      }, null, 8, _),
      s("div", u, [
        s("div", f, [
          (l(!0), o(d, null, m(e.data, (i, r) => (l(), o("span", {
            key: r,
            class: "center-yellow_span",
            onClick: (E) => e.clickArrow(i)
          }, null, 8, h))), 128))
        ]),
        s("div", k, [
          s("span", w, a(e.startDate.slice(0, 4)), 1),
          s("span", v, a(e.endDate.slice(0, 4)), 1)
        ]),
        s("div", C, [
          s("div", y, a(e.arrowTopData), 1),
          s("div", null, [
            s("img", {
              class: "image-arrow",
              src: e.imageArrow,
              alt: "箭头",
              draggable: "false"
            }, null, 8, D)
          ])
        ])
      ]),
      s("img", {
        class: "image-right",
        src: e.imageRight,
        alt: "右",
        draggable: "false",
        onClick: t[1] || (t[1] = (...i) => e.clickRightSwitch && e.clickRightSwitch(...i))
      }, null, 8, b)
    ]),
    t[2] || (t[2] = s("div", { class: "style-container" }, null, -1))
  ], 2);
}
const N = /* @__PURE__ */ p(n, [["render", S]]);
export {
  N as default
};
