// @ths/design/es/components/time-axis-complex/src/index.vue2.js
import { defineComponent as E, computed as _, reactive as I, onMounted as W, watch as S, toRefs as C, nextTick as k } from "vue";
import { getDomRandomClass as A, setStyle as n } from "../../../utils/index.js";
const P = E({
  name: "TTimeAxisComplex",
  props: {
    // 自定义组件样式
    cStyle: {
      type: Object,
      default: () => ({
        // 容器
        wrapper: {
          default: {}
        },
        // 中间区域样式
        centerBox: {
          default: {}
        },
        // 有数据区域样式
        centerBoxDate: {
          default: {}
        },
        // 开始时间和结束时间样式
        startAndEndTime: {
          default: {}
        },
        // 箭头上方区域样式
        arrowTop: {
          default: {}
        },
        // 开始时间样式
        startTime: {
          default: {}
        },
        // 结束时间样式
        endTime: {
          default: {}
        },
        // 箭头图片样式
        arrowImg: {
          default: {}
        },
        // 左侧切换图片样式
        leftSwitchImg: {
          default: {}
        },
        // 右侧切换图片样式
        rightSwitchImg: {
          default: {}
        }
      })
    },
    // 图片数据-左侧切换
    imageLeft: {
      type: String,
      default: ""
    },
    // 图片数据-右侧切换
    imageRight: {
      type: String,
      default: ""
    },
    // 图片数据-点击箭头
    imageArrow: {
      type: String,
      default: ""
    },
    // 时间轴数据
    data: {
      type: Array,
      default: () => []
    },
    // 开始时间
    startDate: {
      type: String,
      default: ""
    },
    // 结束时间
    endDate: {
      type: String,
      default: ""
    },
    // 箭头默认点击的时间
    arrowDate: {
      type: String,
      default: ""
    }
  },
  // 事件名
  emits: ["time-change"],
  setup(t, { emit: b }) {
    const a = _(() => t.cStyle), o = A("time-axis-complex"), r = I({
      // 起始时间和结束时间相差的天数
      startToEndDays: 0,
      // 箭头上方显示的数据
      arrowTopData: "",
      // 箭头所定位到的数据
      arrowNowDate: "",
      // 中间区域的宽度
      centerWidth: 1112,
      // 有数据的单个区块宽度
      singleBlockWidth: 4,
      // 箭头图片的宽度
      arrowImgWidth: 20,
      // 箭头上方元素的宽度
      arrowTopWidth: 166
    }), v = () => {
      let e = "";
      return a.value.wrapper && (e += n(a.value.wrapper, `.${o}`)), a.value.centerBox && (e += n(a.value.centerBox, `.${o} .t-time-axis-complex-box .t-time-axis-complex-box_center`)), a.value.centerBoxDate && (e += n(a.value.centerBoxDate, `.${o} .t-time-axis-complex-box .t-time-axis-complex-box_center .center-yellow_span`)), a.value.startAndEndTime && (e += n(a.value.startAndEndTime, `.${o} .t-time-axis-complex-box .t-time-axis-complex-box_center .center-date`)), a.value.startTime && (e += n(a.value.startTime, `.${o} .t-time-axis-complex-box .t-time-axis-complex-box_center .center-date .center-date_start`)), a.value.endTime && (e += n(a.value.endTime, `.${o} .t-time-axis-complex-box .t-time-axis-complex-box_center .center-date .center-date_end`)), a.value.arrowTop && (e += n(a.value.arrowTop, `.${o} .t-time-axis-complex-box .t-time-axis-complex-box_center .center-arrow_top`)), a.value.arrowImg && (e += n(a.value.arrowImg, `.${o} .t-time-axis-complex-box .t-time-axis-complex-box_center .center-arrow .image-arrow`)), a.value.leftSwitchImg && (e += n(a.value.leftSwitchImg, `.${o} .t-time-axis-complex-box .image-left`)), a.value.rightSwitchImg && (e += n(a.value.rightSwitchImg, `.${o} .t-time-axis-complex-box .image-right`)), e;
    }, T = () => {
      const e = document.querySelector(`.${o} .style-container`);
      e && (e.innerHTML = `<style>${v()}</style>`);
    }, B = (e, l, c, i, d) => {
      const m = new Date(e), s = new Date(l) - m;
      return Math.floor(s / (24 * 3600 * 1e3)) / c * i - d / 2;
    }, x = (e) => {
      const l = document.getElementsByClassName(`t-time-axis-complex ${o}`)[0], c = l.getElementsByClassName("center-arrow")[0], i = l.getElementsByClassName("center-arrow_top")[0], d = B(t.startDate, e, r.startToEndDays, r.centerWidth, r.arrowImgWidth);
      c.style.left = `${d}px`;
      const m = r.arrowTopWidth / 2;
      i.style.left = `-${m}px`, r.arrowTopData = e.replace(/-/g, "/"), r.arrowNowDate = e, b("time-change", e);
    }, p = () => {
      const e = new Date(t.startDate), c = new Date(t.endDate) - e, i = Math.floor(c / (24 * 3600 * 1e3));
      r.startToEndDays = i;
      const m = document.getElementsByClassName(`t-time-axis-complex ${o}`)[0].getElementsByClassName("center-yellow_span");
      t.data.forEach((w, s) => {
        k(() => {
          const f = new Date(t.startDate), y = new Date(w) - f, g = Math.floor(y / (24 * 3600 * 1e3)) / r.startToEndDays * r.centerWidth - r.singleBlockWidth / 2;
          try {
            m[s].style.left = `${g}px`;
          } catch (h) {
            console.error(h);
          }
        });
      }), x(t.arrowDate);
    }, $ = () => {
      t.data.forEach((e, l) => {
        r.arrowNowDate === e && l - 1 >= 0 && (r.arrowNowDate = t.data[l - 1]);
      }), x(r.arrowNowDate);
    }, N = () => {
      let e = r.arrowNowDate;
      t.data.forEach((l, c) => {
        r.arrowNowDate === l && c + 1 < t.data.length && (e = t.data[c + 1]);
      }), x(e);
    };
    return W(() => {
      var e, l, c, i, d, m, w, s, f, u, y, D, g, h;
      (l = (e = t.cStyle.centerBox) == null ? void 0 : e.default) != null && l.width && (r.centerWidth = Number(String((c = t.cStyle.centerBox.default) == null ? void 0 : c.width).replace("px", ""))), (d = (i = t.cStyle.centerBoxDate) == null ? void 0 : i.default) != null && d.width && (r.singleBlockWidth = Number(String((m = t.cStyle.centerBoxDate.default) == null ? void 0 : m.width).replace("px", ""))), (s = (w = t.cStyle.arrowImg) == null ? void 0 : w.default) != null && s.width && (r.arrowImgWidth = Number(String((u = (f = t.cStyle.arrowImg) == null ? void 0 : f.default) == null ? void 0 : u.width).replace("px", ""))), (D = (y = t.cStyle.arrowTop) == null ? void 0 : y.default) != null && D.width && (r.arrowTopWidth = Number(String((h = (g = t.cStyle.arrowTop) == null ? void 0 : g.default) == null ? void 0 : h.width).replace("px", ""))), p(), T();
    }), S(() => t.cStyle, () => {
      T();
    }, { deep: !0 }), S([() => t.data, () => t.startDate, () => t.endDate, () => t.cStyle], () => {
      p();
    }, { deep: !0 }), S(() => t.arrowDate, () => {
      x(t.arrowDate);
    }, { deep: !0 }), {
      ...C(r),
      curStyle: a,
      domOnlyClass: o,
      clickArrow: x,
      clickLeftSwitch: $,
      clickRightSwitch: N
    };
  }
});
export {
  P as default
};
