// @ths/design/es/components/menu/src/menu-item.vue.js
import Me from "./menu-item.vue2.js";
import { resolveComponent as ve, openBlock as n, createElementBlock as l, normalizeClass as r, normalizeStyle as u, createElementVNode as fe, createBlock as pe, withCtx as ye, renderSlot as ce, createCommentVNode as m } from "vue";
import "./menu-item.vue3.js";
import Ie from "../../../_virtual/_plugin-vue_export-helper.js";
const ge = ["title"];
function he(e, t, Le, $e, be, ke) {
  var i, a, s, d, f, y, c, S, M, v, p, I, g, h, L, $, b, k, C, z, B, H, W, w, R, T, E, N, V, j, q, A, D, F, G, J, K, O, P, Q, U, X, Y, Z, _, x, ee, te, oe, ne, re, le, ue, me, ie, ae, se, de;
  const Se = ve("el-tooltip");
  return n(), l("div", {
    class: r(["t-menu-item", [
      e.currentMenuStyle,
      `t-menu-item__mode-${e.currentMode}`,
      e.currentBorderLine && `t-menu-item__mode-${e.currentMode}${e.fold ? "" : "--line"}`,
      `${e.currentMenuStyle}-${e.currentThemeStyle}`,
      e.active && `${e.currentMenuStyle}-${e.currentThemeStyle}--active`,
      e.active && `t-menu-item__mode-${e.currentMode}--active`
    ]]),
    style: u((i = e.customStyle) != null && i.menuItem && ((a = e.customStyle) != null && a.menuItem.active) && (e.menuItemHover || e.active) ? {
      // marginLeft: currentMode === 'vertical' && fold ? '15px' : 0,
      ...(s = e.customStyle) == null ? void 0 : s.menuItem.active,
      "--t-border-bottom-color": (y = (f = (d = e.customStyle) == null ? void 0 : d.borderBottomLine) == null ? void 0 : f.default) == null ? void 0 : y.backgroundColor,
      "--split-line-color": (M = (S = (c = e.customStyle) == null ? void 0 : c.splitLine) == null ? void 0 : S.default) == null ? void 0 : M.backgroundColor
    } : (v = e.customStyle) != null && v.menuItem && ((p = e.customStyle) != null && p.menuItem.default) ? {
      // marginLeft: currentMode === 'vertical' && fold ? '15px' : 0,
      ...(I = e.customStyle) == null ? void 0 : I.menuItem.default,
      "--t-border-bottom-color": (L = (h = (g = e.customStyle) == null ? void 0 : g.borderBottomLine) == null ? void 0 : h.default) == null ? void 0 : L.backgroundColor,
      "--split-line-color": (k = (b = ($ = e.customStyle) == null ? void 0 : $.splitLine) == null ? void 0 : b.default) == null ? void 0 : k.backgroundColor
    } : {
      // marginLeft: currentMode === 'vertical' && fold ? '15px' : 0,
      "--t-border-bottom-color": (B = (z = (C = e.customStyle) == null ? void 0 : C.borderBottomLine) == null ? void 0 : z.default) == null ? void 0 : B.backgroundColor,
      "--split-line-color": (w = (W = (H = e.customStyle) == null ? void 0 : H.splitLine) == null ? void 0 : W.default) == null ? void 0 : w.backgroundColor
    }),
    onClick: t[0] || (t[0] = (...o) => e.handleClickMenuItem && e.handleClickMenuItem(...o)),
    onMouseenter: t[1] || (t[1] = (...o) => e.handleMouseenterMenuItem && e.handleMouseenterMenuItem(...o)),
    onMouseleave: t[2] || (t[2] = (...o) => e.handleMouseleaveMenuItem && e.handleMouseleaveMenuItem(...o))
  }, [
    fe("div", {
      class: r([`mode-${e.currentMode}`, `mode-${e.currentMode}--${e.currentMenuLayout}`])
    }, [
      e.currentMode === "vertical" && e.fold ? (n(), pe(Se, {
        key: 0,
        effect: e.currentThemeStyle,
        placement: "right",
        offset: 25
      }, {
        content: ye(() => [
          ce(e.$slots, "default")
        ]),
        default: ye(() => [
          e.currentShowIcon ? (n(), l("i", {
            key: 0,
            class: r(["fa", e.currentIcon])
          }, null, 2)) : m("", !0)
        ]),
        _: 3
      }, 8, ["effect"])) : m("", !0),
      !e.fold && e.currentShowIcon ? (n(), l("i", {
        key: 1,
        class: r(["fa", e.currentIcon]),
        style: u((R = e.customStyle) != null && R.icon && ((T = e.customStyle) != null && T.icon.active) && (e.menuItemHover || e.active) ? {
          marginBottom: e.currentMode === "horizontal" && e.currentMenuLayout === "vertical" ? "5px" : 0,
          marginRight: e.currentMenuLayout === "horizontal" || e.currentMode === "vertical" ? "5px" : 0,
          ...(E = e.customStyle) == null ? void 0 : E.icon.active
        } : (N = e.customStyle) != null && N.icon && ((V = e.customStyle) != null && V.icon.default) ? {
          marginBottom: e.currentMode === "horizontal" && e.currentMenuLayout === "vertical" ? "5px" : 0,
          marginRight: e.currentMenuLayout === "horizontal" || e.currentMode === "vertical" ? "5px" : 0,
          ...(j = e.customStyle) == null ? void 0 : j.icon.default
        } : {
          marginBottom: e.currentMode === "horizontal" && e.currentMenuLayout === "vertical" ? "5px" : 0,
          marginRight: e.currentMenuLayout === "horizontal" || e.currentMode === "vertical" ? "5px" : 0
        })
      }, null, 6)) : m("", !0),
      fe("span", {
        class: "label",
        title: e.slots,
        style: u((q = e.customStyle) != null && q.menuItem && ((A = e.customStyle) != null && A.menuItem.active) && (e.menuItemHover || e.active) ? {
          marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
          color: (G = (F = (D = e.customStyle) == null ? void 0 : D.menuItem) == null ? void 0 : F.active) == null ? void 0 : G.color,
          fontWeight: (O = (K = (J = e.customStyle) == null ? void 0 : J.menuItem) == null ? void 0 : K.active) == null ? void 0 : O.fontWeight,
          fontSize: (U = (Q = (P = e.customStyle) == null ? void 0 : P.menuItem) == null ? void 0 : Q.active) == null ? void 0 : U.fontSize,
          lineHeight: (Z = (Y = (X = e.customStyle) == null ? void 0 : X.menuItem) == null ? void 0 : Y.active) == null ? void 0 : Z.lineHeight
        } : (_ = e.customStyle) != null && _.menuItem && ((x = e.customStyle) != null && x.menuItem.default) ? {
          marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
          color: (oe = (te = (ee = e.customStyle) == null ? void 0 : ee.menuItem) == null ? void 0 : te.default) == null ? void 0 : oe.color,
          fontWeight: (le = (re = (ne = e.customStyle) == null ? void 0 : ne.menuItem) == null ? void 0 : re.default) == null ? void 0 : le.fontWeight,
          fontSize: (ie = (me = (ue = e.customStyle) == null ? void 0 : ue.menuItem) == null ? void 0 : me.default) == null ? void 0 : ie.fontSize,
          lineHeight: (de = (se = (ae = e.customStyle) == null ? void 0 : ae.menuItem) == null ? void 0 : se.default) == null ? void 0 : de.lineHeight
        } : {
          marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0
        })
      }, [
        ce(e.$slots, "default")
      ], 12, ge)
    ], 2)
  ], 38);
}
const We = /* @__PURE__ */ Ie(Me, [["render", he]]);
export {
  We as default
};
