// @ths/design/es/components/menu/src/menu.vue.js
import k from "./menu.vue2.js";
import { resolveComponent as z, openBlock as S, createElementBlock as $, normalizeClass as l, normalizeStyle as C, renderSlot as B, createElementVNode as r, createVNode as M, withCtx as H, createCommentVNode as N } from "vue";
import "./menu.vue3.js";
import E from "../../../_virtual/_plugin-vue_export-helper.js";
function F(e, o, I, O, R, T) {
  var t, n, d, a, i, s, m, p, u, f, b, y, h, c, w, v, L;
  const g = z("el-icon");
  return S(), $("section", {
    ref: "menuRef",
    class: l(["t-menu", [`t-menu__${e.mode}`, `t-menu__${e.mode}--${e.themeStyle}`]]),
    style: C({
      ...(n = (t = e.cStyle) == null ? void 0 : t.wrapper) == null ? void 0 : n.default,
      ...(a = (d = e.cStyle) == null ? void 0 : d.menu) == null ? void 0 : a.default,
      width: e.mode === "vertical" ? e.fold ? "45px" : `${parseInt((m = (s = (i = e.cStyle) == null ? void 0 : i.wrapper) == null ? void 0 : s.default) == null ? void 0 : m.width) || parseInt(e.width)}px` : "",
      "--vertical-bottom-line-color": (f = (u = (p = e.customStyle) == null ? void 0 : p.borderBottomLine) == null ? void 0 : u.default) == null ? void 0 : f.backgroundColor,
      "--split-line-color": (h = (y = (b = e.customStyle) == null ? void 0 : b.splitLine) == null ? void 0 : y.default) == null ? void 0 : h.backgroundColor
    })
  }, [
    B(e.$slots, "default"),
    r("span", {
      class: l(["bar", [`bar--${e.mode}`]]),
      style: C(
        e.mode === "vertical" ? {
          top: e.borderLineTop + "px",
          height: e.borderLineHeight + "px",
          opacity: e.borderLineOpacity,
          ...(w = (c = e.customStyle) == null ? void 0 : c.borderBottomLine) == null ? void 0 : w.default
        } : {
          left: e.borderLineLeft + "px",
          width: e.borderLineWidth + "px",
          opacity: e.borderLineOpacity,
          ...(L = (v = e.customStyle) == null ? void 0 : v.borderBottomLine) == null ? void 0 : L.default
        }
      )
    }, null, 6),
    e.mode === "vertical" && e.collapsible ? (S(), $("span", {
      key: 0,
      class: "folder-btn",
      onClick: o[0] || (o[0] = (...V) => e.handleFoldMenu && e.handleFoldMenu(...V))
    }, [
      M(g, {
        class: l({ unfold: e.fold })
      }, {
        default: H(() => o[1] || (o[1] = [
          r("svg", {
            xmlns: "http://www.w3.org/2000/svg",
            viewBox: "0 0 1024 1024",
            class: ""
          }, [
            r("path", {
              fill: "currentColor",
              d: "M896 192H128v128h768V192zm0 256H384v128h512V448zm0 256H128v128h768V704zM320 384 128 512l192 128V384z"
            })
          ], -1)
        ])),
        _: 1,
        __: [1]
      }, 8, ["class"])
    ])) : N("", !0)
  ], 6);
}
const D = /* @__PURE__ */ E(k, [["render", F]]);
export {
  D as default
};
