// @ths/design/es/components/menu/src/menu.vue2.js
import { defineComponent as R, ref as o, computed as f, reactive as _, watch as S, provide as j, onMounted as F, onUnmounted as U, toRefs as q } from "vue";
import D from "mitt";
const K = R({
  name: "TMenu",
  props: {
    // 主题风格
    themeStyle: {
      type: String,
      default: "light"
    },
    // 自定义主题色
    themeColor: {
      type: String,
      default: "light"
    },
    // 模式 horizontal-水平 vertical-垂直
    mode: {
      type: String,
      default: "horizontal"
    },
    // 是否显示图标
    showIcon: {
      type: Boolean,
      default: !0
    },
    // 图标文字排列方向 horizontal-水平 vertical-垂直
    menuLayout: {
      type: String,
      default: "vertical"
    },
    // 风格样式  下划线 line | 背景色 surface
    menuStyle: {
      type: String,
      default: "line"
    },
    // 当前选中项
    defaultActive: {
      type: String,
      default: ""
    },
    // 选中项数组
    activeMenus: {
      type: Array,
      default: () => [""]
    },
    // 背景色
    backgroundColor: { type: String, default: "" },
    // 文本颜色
    textColor: { type: String, default: "" },
    // 当前文本颜色
    activeTextColor: { type: String, default: "" },
    // 纵向菜单 菜单宽度
    width: {
      type: Number,
      default: 200
    },
    // 纵向菜单 是否显示下划线
    borderLine: {
      type: Boolean,
      default: !0
    },
    // 纵向菜单 是否可以折叠
    collapsible: {
      type: Boolean,
      default: !0
    },
    // 纵向菜单 折叠状态
    folded: {
      type: Boolean,
      default: !1
    },
    // 图标名
    icon: {
      type: String,
      default: ""
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        // 菜单
        wrapper: {
          default: {}
        },
        menu: {
          default: {}
        },
        // 菜单项
        menuItem: {
          default: {},
          active: {}
        },
        icon: {
          default: {},
          active: {}
        },
        // 子菜单
        submenu: {
          default: {},
          active: {}
        },
        // 子菜单项
        submenuItem: {
          default: {},
          active: {}
        },
        // 下划线
        borderBottomLine: {
          default: {},
          active: {}
        },
        // 更多图标
        arrowIcon: {
          default: {},
          active: {}
        },
        // 子菜单更多图标
        subArrowIcon: {
          default: {},
          active: {}
        },
        // 分割线
        splitLine: {
          default: {},
          active: {}
        }
      })
    }
  },
  // 选择菜单项事件，折叠菜单事件-只支持纵向菜单
  emits: ["select", "fold"],
  setup(u, { emit: h }) {
    var x, W;
    const i = o(), p = o(0), g = o(0), b = o(0), L = o(0), m = o(0), d = o(!1), v = o(""), y = o(0), a = D(), l = o([]), r = o([]), M = f(() => u.cStyle), B = _({
      // 下划线自定义样式
      borderBottomLineStyle: {
        ...(W = (x = M.value) == null ? void 0 : x.borderBottomLine) == null ? void 0 : W.default,
        top: u.mode === "vertical" ? f(() => g.value) : `${f(() => p.value)}px`,
        height: u.mode === "vertical" ? f(() => L.value) : `${f(() => b.value)}px`,
        opacity: f(() => m.value)
      }
    }), I = () => {
      var n;
      const e = (n = i.value) == null ? void 0 : n.clientWidth;
      y.value = e || 0;
    }, s = () => {
      var e;
      (e = i.value) == null || e.addEventListener("transitionend", I);
    };
    function C() {
      var e, n, c;
      d.value = !d.value, d.value ? ((e = i.value) == null || e.removeEventListener("resize", s), (n = i.value) == null || n.removeEventListener("transitionend", I)) : (y.value = 0, s(), (c = i.value) == null || c.addEventListener("resize", s)), h("fold", d.value);
    }
    function E(e) {
      v.value = e.index, l.value = [], r.value = [e.index], h("select", e);
    }
    function O(e) {
      u.menuStyle === "line" && (u.mode === "horizontal" ? (p.value = e.target.offsetLeft, b.value = e.target.offsetWidth, m.value = 1) : (g.value = e.target.offsetTop, L.value = e.target.offsetHeight, m.value = 1));
    }
    function A() {
      b.value = 0, m.value = 0;
    }
    function k(e) {
      l.value = [.../* @__PURE__ */ new Set([e])];
    }
    function w(e) {
      r.value = [.../* @__PURE__ */ new Set([...l.value, e])];
    }
    function z(e) {
      l.value = l.value.filter((n) => n !== e && n);
    }
    function T(e) {
      l.value = [.../* @__PURE__ */ new Set([...l.value, ...e])];
    }
    function H(e) {
      v.value = e.index, w(e.index), h("select", e);
    }
    function N(e) {
      l.value = e, r.value = e;
    }
    return S(
      () => u.folded,
      (e) => {
        d.value = e;
      }
    ), S(
      () => u.activeMenus,
      (e) => {
        l.value = e, r.value = e;
      }
    ), S(
      () => u.defaultActive,
      (e) => {
        v.value = e;
      }
    ), j("rootMenu", {
      props: u,
      fold: d,
      activeIndex: v,
      openedSubmenus: l,
      activeSubmenus: r,
      menuWidth: y,
      rootMenuEmit: a.emit,
      rootMenuOn: a.on
    }), F(() => {
      var n, c;
      v.value = u.defaultActive, d.value = u.folded, r.value = u.activeMenus, a.on("menuItem:item-click", (t) => E(t)), a.on("menuItem:item-mouseenter", (t) => O(t)), a.on("menuItem:item-mouseleave", () => A()), a.on("submenuItem:item-click", (t) => H(t)), a.on("submenu:init", (t) => k(t)), a.on("submenu:add", (t) => T(t)), a.on("submenu:update", (t) => w(t)), a.on("submenu:remove", (t) => z(t)), a.on("submenu:set-value", (t) => N(t));
      const e = (n = i.value) == null ? void 0 : n.clientWidth;
      y.value = e || 0, (c = i.value) == null || c.addEventListener("resize", s);
    }), S(() => {
      var e, n, c;
      return (c = (n = (e = u.cStyle) == null ? void 0 : e.wrapper) == null ? void 0 : n.default) == null ? void 0 : c.width;
    }, () => {
      s();
    }), U(() => {
      a.all.clear();
    }), {
      ...q(B),
      fold: d,
      borderLineLeft: p,
      borderLineWidth: b,
      borderLineTop: g,
      borderLineHeight: L,
      borderLineOpacity: m,
      handleFoldMenu: C,
      customStyle: M,
      openedSubmenus: l,
      activeSubmenus: r,
      activeIndex: v,
      menuRef: i
    };
  }
});
export {
  K as default
};
