// @ths/design/es/components/menu/src/menu-item.vue2.js
import { defineComponent as g, inject as L, ref as a, computed as t, getCurrentInstance as w, watch as C } from "vue";
const j = g({
  name: "TMenuItem",
  props: {
    icon: {
      type: String,
      default: ""
    },
    index: {
      type: String || Number,
      default: ""
    },
    path: {
      type: String,
      default: ""
    },
    route: {
      type: Object,
      default: () => {
        "index";
      }
    }
  },
  setup(n) {
    const e = L("rootMenu"), c = a(!1), l = a(n.index === (e == null ? void 0 : e.activeIndex.value) || (e == null ? void 0 : e.activeSubmenus.value.includes(n.index))), m = t(() => e == null ? void 0 : e.fold.value), u = t(() => e == null ? void 0 : e.props.cStyle), d = t(() => e == null ? void 0 : e.props.menuStyle), f = t(() => e == null ? void 0 : e.props.showIcon), v = t(() => e == null ? void 0 : e.props.themeStyle), p = t(() => e == null ? void 0 : e.props.menuLayout), I = t(() => e == null ? void 0 : e.props.borderLine), h = t(() => e == null ? void 0 : e.props.mode), r = t(() => n.icon);
    function S() {
      e == null || e.rootMenuEmit("menuItem:item-click", {
        index: n.index,
        path: n.path,
        route: n.route
      });
    }
    function y(s) {
      e == null || e.rootMenuEmit("menuItem:item-mouseenter", s), c.value = !0;
    }
    const { proxy: i } = w(), x = a(i == null ? void 0 : i.$slots.default()[0].children);
    function b() {
      e == null || e.rootMenuEmit("menuItem:item-mouseleave"), c.value = !1;
    }
    return C(
      () => e == null ? void 0 : e.activeIndex.value,
      (s) => {
        l.value = n.index === s || (e == null ? void 0 : e.activeSubmenus.value.includes(n.index));
      }
    ), {
      ...e == null ? void 0 : e.props,
      fold: m,
      active: l,
      menuItemHover: c,
      customStyle: u,
      currentMenuStyle: d,
      currentShowIcon: f,
      currentThemeStyle: v,
      currentMenuLayout: p,
      currentMode: h,
      currentBorderLine: I,
      currentIcon: r,
      handleClickMenuItem: S,
      handleMouseenterMenuItem: y,
      handleMouseleaveMenuItem: b,
      slots: x
    };
  }
});
export {
  j as default
};
