// @ths/design/es/components/menu/src/submenu-item.vue2.js
import { defineComponent as y, inject as o, computed as t, getCurrentInstance as g, ref as l, watch as W } from "vue";
const $ = y({
  name: "TSubmenuItem",
  props: {
    icon: {
      type: String,
      default: ""
    },
    index: {
      type: String || Number,
      default: ""
    },
    path: {
      type: String,
      default: ""
    },
    route: {
      type: Object,
      default: () => ({})
    },
    vMenuWidth: {
      type: Number,
      default: 0
    }
  },
  setup(n) {
    const e = o("rootMenu"), u = t(() => n.index === (e == null ? void 0 : e.activeIndex.value)), a = t(() => e == null ? void 0 : e.fold.value), m = t(() => e == null ? void 0 : e.props.showIcon), s = t(() => e == null ? void 0 : e.props.themeStyle), f = t(() => n.icon), p = t(() => e == null ? void 0 : e.props.borderLine), h = t(() => e == null ? void 0 : e.props.cStyle), v = t(() => e == null ? void 0 : e.props.mode), { proxy: i } = g(), r = l(i == null ? void 0 : i.$slots.default()[0].children);
    function S() {
      e == null || e.rootMenuEmit("submenuItem:item-click", {
        index: n.index,
        path: n.path,
        route: n.route
      }), e == null || e.rootMenuEmit("submenu:update", {
        index: n.index,
        path: n.path,
        route: n.route
      });
    }
    function b(c) {
      e != null && e.fold.value;
    }
    function x() {
      e != null && e.fold.value;
    }
    const d = l("0px"), I = t(() => a.value ? "200px" : d.value);
    return W(() => {
      var c;
      return (c = e == null ? void 0 : e.menuWidth) == null ? void 0 : c.value;
    }, (c, w) => {
      d.value = `${c}px`;
    }, { immediate: !0 }), {
      rootMenu: e,
      ...e == null ? void 0 : e.props,
      fold: a,
      active: u,
      currentShowIcon: m,
      currentThemeStyle: s,
      currentIcon: f,
      currentBorderLine: p,
      customStyle: h,
      currentMode: v,
      handleClickSubmenuItem: S,
      handleMouseenterSubmenuItem: b,
      handleMouseleaveSubmenuItem: x,
      slots: r,
      vMenuWidths: d,
      computMenuWidth: I
    };
  }
});
export {
  $ as default
};
