// @ths/design/es/components/menu/src/submenu.vue2.js
import { defineComponent as F, inject as H, ref as v, computed as a, getCurrentInstance as _, watch as n, onUnmounted as $ } from "vue";
const A = F({
  name: "TSubmenu",
  props: {
    icon: {
      type: String,
      default: ""
    },
    index: {
      type: [String, Number],
      default: ""
    },
    showArrow: {
      type: Boolean,
      default: !0
    }
  },
  emits: ["mouseenter", "mouseleave", "click"],
  setup(l) {
    var b;
    const e = H("rootMenu"), i = v(!1), m = a(() => e == null ? void 0 : e.fold.value), f = v(!1), x = v(!0), t = v((e == null ? void 0 : e.activeSubmenus.value.includes(e == null ? void 0 : e.activeIndex.value)) && (e == null ? void 0 : e.activeSubmenus.value.includes(l.index))), S = v(null), h = v(0), E = a(() => e == null ? void 0 : e.props.cStyle), y = a(() => e == null ? void 0 : e.props.menuStyle), I = a(() => e == null ? void 0 : e.props.showIcon), p = a(() => e == null ? void 0 : e.props.themeStyle), L = a(() => e == null ? void 0 : e.props.menuLayout), w = a(() => e == null ? void 0 : e.props.borderLine), T = a(() => e == null ? void 0 : e.props.mode), g = a(() => l.icon), c = _(), z = v((b = c.proxy) == null ? void 0 : b.$slots.title()[0].children), s = a(() => {
      let d = !0, u = c == null ? void 0 : c.parent;
      for (; u && u.type.name !== "TMenu"; )
        if (["TSubmenu"].includes(u.type.name)) {
          d = !1;
          break;
        } else
          u = u.parent;
      return d;
    });
    function C() {
      i.value = !i.value, (e == null ? void 0 : e.props.mode) === "vertical" && !(e != null && e.fold.value) && (!i.value && !t.value && (e == null || e.rootMenuEmit("submenu:remove", l.index)), i.value && s.value && (e == null || e.rootMenuEmit("submenu:init", l.index)), i.value && !s.value && (e == null || e.rootMenuEmit("submenu:add", [l.index]))), ((e == null ? void 0 : e.props.mode) === "horizontal" || e != null && e.fold.value) && (i.value ? s.value ? e == null || e.rootMenuEmit("submenu:init", l.index) : e == null || e.rootMenuEmit("submenu:add", [l.index]) : e == null || e.rootMenuEmit("submenu:remove", l.index));
    }
    function k(d) {
      ((e == null ? void 0 : e.props.mode) === "horizontal" || e != null && e.fold.value) && (i.value = !0, i.value ? s.value ? e == null || e.rootMenuEmit("submenu:init", l.index) : e == null || e.rootMenuEmit("submenu:add", [l.index]) : e == null || e.rootMenuEmit("submenu:remove", l.index)), s.value && (f.value = !0);
    }
    function B() {
      ((e == null ? void 0 : e.props.mode) === "horizontal" || e != null && e.fold.value) && (i.value = !1, i.value ? s.value ? e == null || e.rootMenuEmit("submenu:init", l.index) : e == null || e.rootMenuEmit("submenu:add", [l.index]) : e == null || e.rootMenuEmit("submenu:remove", l.index)), s.value && (f.value = !1);
    }
    return n(
      () => e == null ? void 0 : e.fold.value,
      () => {
        i.value = !1, i.value ? s.value ? e == null || e.rootMenuEmit("submenu:init", l.index) : e == null || e.rootMenuEmit("submenu:add", [l.index]) : e == null || e.rootMenuEmit("submenu:remove", l.index);
      }
    ), n(
      () => e == null ? void 0 : e.openedSubmenus.value,
      () => {
        (e == null ? void 0 : e.props.mode) === "vertical" && !m.value && (e != null && e.openedSubmenus.value.includes(l.index) || (i.value = !1));
      }
    ), i.value = (e == null ? void 0 : e.props.mode) === "vertical" && !m.value && (e == null ? void 0 : e.openedSubmenus.value.includes(l.index)), n(
      () => e == null ? void 0 : e.activeIndex.value,
      () => {
        t.value = (e == null ? void 0 : e.activeSubmenus.value.includes(e == null ? void 0 : e.activeIndex.value)) && (e == null ? void 0 : e.activeSubmenus.value.includes(l.index)), (e == null ? void 0 : e.props.mode) === "vertical" && !m.value && (i.value = (e == null ? void 0 : e.activeSubmenus.value.includes(e == null ? void 0 : e.activeIndex.value)) && (e == null ? void 0 : e.activeSubmenus.value.includes(l.index)));
      }
    ), n(
      () => e == null ? void 0 : e.props.mode,
      (d) => {
        d === "horizontal" || d === "vertical" && m.value ? i.value = !1 : i.value = (e == null ? void 0 : e.activeSubmenus.value.includes(e == null ? void 0 : e.activeIndex.value)) && (e == null ? void 0 : e.activeSubmenus.value.includes(l.index));
      }
    ), $(() => {
      i.value = !1;
    }), {
      rootMenu: e,
      ...e == null ? void 0 : e.props,
      active: t,
      menuItemHover: f,
      opened: i,
      selfOpened: x,
      fold: m,
      isFirstLevel: s,
      customStyle: E,
      currentMenuStyle: y,
      currentShowIcon: I,
      currentThemeStyle: p,
      currentMenuLayout: L,
      currentBorderLine: w,
      currentMode: T,
      currentIcon: g,
      handleClickSubmenu: C,
      handleMouseenterSubmenu: k,
      handleMouseleaveSubmenu: B,
      submenuItemsContainerHeight: h,
      submenuList: S,
      slotsTitle: z
    };
  }
});
export {
  A as default
};
