// @ths/design/es/components/menu/src/submenu.vue.js
import Eo from "./submenu.vue2.js";
import { resolveComponent as Ro, openBlock as n, createElementBlock as i, normalizeClass as r, normalizeStyle as l, createBlock as To, withCtx as s, createElementVNode as t, createCommentVNode as m, renderSlot as a, createVNode as Uo } from "vue";
import "./submenu.vue3.js";
import jo from "../../../_virtual/_plugin-vue_export-helper.js";
const qo = ["title"], Do = { class: "mode-vertical__header-left" }, Go = ["title"];
function Jo(e, o, Ko, Oo, Po, Qo) {
  var v, y, S, f, b, c, I, g, h, p, L, M, w, z, H, F, B, $, k, W, C, A, R, T, N, V, E, U, j, q, D, G, J, K, O, P, Q, X, Y, Z, x, _, ee, oe, te, ue, le, re, ne, me, ie, se, ae, de, ve, ye, Se, fe, be;
  const d = Ro("el-icon"), No = Ro("el-popover");
  return n(), i("div", {
    class: r(["t-submenu", [e.isFirstLevel && e.currentBorderLine && !e.fold && e.currentMode === "vertical" && "t-submenu--line"]]),
    style: l({ "--split-line-color": (S = (y = (v = e.customStyle) == null ? void 0 : v.splitLine) == null ? void 0 : y.default) == null ? void 0 : S.backgroundColor }),
    onMouseleave: o[4] || (o[4] = (...u) => e.handleMouseleaveSubmenu && e.handleMouseleaveSubmenu(...u))
  }, [
    e.currentMode === "horizontal" || e.fold ? (n(), To(No, {
      key: 0,
      visible: e.opened,
      "onUpdate:visible": o[1] || (o[1] = (u) => e.opened = u),
      class: "mode-horizontal",
      "popper-class": "t-submenu",
      placement: e.isFirstLevel ? e.fold ? "right-start" : "bottom-start" : "right-start",
      width: 200,
      "show-arrow": !1,
      "manual-mode": !0,
      effect: "dark",
      pure: !0,
      offset: 0,
      "append-to-body": !1
    }, {
      reference: s(() => {
        var u, ce, Ie, ge, he, pe, Le, Me, we, ze, He, Fe, Be, $e, ke, We, Ce, Ae, Re, Te, Ne, Ve, Ee, Ue, je, qe, De, Ge, Je, Ke, Oe, Pe, Qe, Xe, Ye, Ze, xe, _e, eo, oo, to, uo, lo, ro, no, mo, io, so, ao, vo, yo, So, fo, bo, co, Io, go, ho, po, Lo, Mo, wo, zo, Ho, Fo, Bo, $o, ko, Wo, Co, Ao;
        return [
          t("div", {
            class: r(["t-menu-item", [
              e.currentMenuStyle,
              `t-menu-item__mode-${e.currentMode}`,
              e.isFirstLevel && e.currentBorderLine && `t-menu-item__mode-${e.currentMode}${e.fold ? "" : "--line"}`,
              `${e.currentMenuStyle}-${e.currentThemeStyle}`,
              (e.opened || e.active) && e.isFirstLevel && `${e.currentMenuStyle}-${e.currentThemeStyle}--active`,
              (e.opened || e.active) && "active"
            ]]),
            style: l(
              e.isFirstLevel && e.customStyle.menuItem && e.customStyle.menuItem.active && (e.menuItemHover || e.active) && {
                ...e.customStyle.menuItem.active,
                "--t-border-bottom-color": (Ie = (ce = (u = e.customStyle) == null ? void 0 : u.borderBottomLine) == null ? void 0 : ce.default) == null ? void 0 : Ie.backgroundColor
              } || e.isFirstLevel && e.customStyle.menuItem && e.customStyle.menuItem.default && {
                ...e.customStyle.menuItem.default,
                "--t-border-bottom-color": (pe = (he = (ge = e.customStyle) == null ? void 0 : ge.borderBottomLine) == null ? void 0 : he.default) == null ? void 0 : pe.backgroundColor
              } || !e.isFirstLevel && e.customStyle.submenu && e.customStyle.submenu.active && (e.menuItemHover || e.active) && {
                ...e.customStyle.submenu.active,
                "--t-border-bottom-color": (we = (Me = (Le = e.customStyle) == null ? void 0 : Le.borderBottomLine) == null ? void 0 : Me.default) == null ? void 0 : we.backgroundColor
              } || !e.isFirstLevel && e.customStyle.submenu && e.customStyle.submenu.default && {
                ...e.customStyle.submenu.default,
                "--t-border-bottom-color": (Fe = (He = (ze = e.customStyle) == null ? void 0 : ze.borderBottomLine) == null ? void 0 : He.default) == null ? void 0 : Fe.backgroundColor
              } || {
                "--t-border-bottom-color": (ke = ($e = (Be = e.customStyle) == null ? void 0 : Be.borderBottomLine) == null ? void 0 : $e.default) == null ? void 0 : ke.backgroundColor
              }
            ),
            onMouseenter: o[0] || (o[0] = (...Vo) => e.handleMouseenterSubmenu && e.handleMouseenterSubmenu(...Vo))
          }, [
            t("div", {
              class: r([`mode-${e.currentMode}`, e.isFirstLevel && `mode-${e.currentMode}--${e.currentMenuLayout}`])
            }, [
              e.currentShowIcon ? (n(), i("i", {
                key: 0,
                class: r(["fa", e.currentIcon]),
                style: l(
                  e.customStyle.icon && e.customStyle.icon.active && (e.menuItemHover || e.active) && {
                    marginBottom: e.currentMode === "horizontal" && e.currentMenuLayout === "vertical" ? "5px" : 0,
                    marginRight: e.currentMenuLayout === "horizontal" || e.currentMode === "vertical" ? "5px" : 0,
                    ...e.customStyle.icon.active
                  } || e.customStyle.icon && e.customStyle.icon.default && {
                    marginBottom: e.currentMode === "horizontal" && e.currentMenuLayout === "vertical" ? "5px" : 0,
                    marginRight: e.currentMenuLayout === "horizontal" || e.currentMode === "vertical" ? "5px" : 0,
                    ...e.customStyle.icon.default
                  } || {
                    marginBottom: e.currentMode === "horizontal" && e.currentMenuLayout === "vertical" ? "5px" : 0,
                    marginRight: e.currentMenuLayout === "horizontal" || e.currentMode === "vertical" ? "5px" : 0
                  }
                )
              }, null, 6)) : m("", !0),
              t("span", {
                class: "label",
                title: e.slotsTitle,
                style: l(
                  (e.menuItemHover || e.active) && e.isFirstLevel && e.customStyle.menuItem && e.customStyle.menuItem.active && {
                    marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
                    color: (Ae = (Ce = (We = e.customStyle) == null ? void 0 : We.menuItem) == null ? void 0 : Ce.active) == null ? void 0 : Ae.color,
                    fontWeight: (Ne = (Te = (Re = e.customStyle) == null ? void 0 : Re.menuItem) == null ? void 0 : Te.active) == null ? void 0 : Ne.fontWeight,
                    fontSize: (Ue = (Ee = (Ve = e.customStyle) == null ? void 0 : Ve.menuItem) == null ? void 0 : Ee.active) == null ? void 0 : Ue.fontSize,
                    lineHeight: (De = (qe = (je = e.customStyle) == null ? void 0 : je.menuItem) == null ? void 0 : qe.active) == null ? void 0 : De.lineHeight
                  } || e.isFirstLevel && e.customStyle.menuItem && e.customStyle.menuItem.default && {
                    marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
                    color: (Ke = (Je = (Ge = e.customStyle) == null ? void 0 : Ge.menuItem) == null ? void 0 : Je.default) == null ? void 0 : Ke.color,
                    fontWeight: (Qe = (Pe = (Oe = e.customStyle) == null ? void 0 : Oe.menuItem) == null ? void 0 : Pe.default) == null ? void 0 : Qe.fontWeight,
                    fontSize: (Ze = (Ye = (Xe = e.customStyle) == null ? void 0 : Xe.menuItem) == null ? void 0 : Ye.default) == null ? void 0 : Ze.fontSize,
                    lineHeight: (eo = (_e = (xe = e.customStyle) == null ? void 0 : xe.menuItem) == null ? void 0 : _e.default) == null ? void 0 : eo.lineHeight
                  } || (e.menuItemHover || e.active) && !e.isFirstLevel && e.customStyle.submenu && e.customStyle.submenu.active && {
                    marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
                    color: (uo = (to = (oo = e.customStyle) == null ? void 0 : oo.submenu) == null ? void 0 : to.active) == null ? void 0 : uo.color,
                    fontWeight: (no = (ro = (lo = e.customStyle) == null ? void 0 : lo.submenu) == null ? void 0 : ro.active) == null ? void 0 : no.fontWeight,
                    fontSize: (so = (io = (mo = e.customStyle) == null ? void 0 : mo.submenu) == null ? void 0 : io.active) == null ? void 0 : so.fontSize,
                    lineHeight: (yo = (vo = (ao = e.customStyle) == null ? void 0 : ao.submenu) == null ? void 0 : vo.active) == null ? void 0 : yo.lineHeight
                  } || !e.isFirstLevel && e.customStyle.submenu && e.customStyle.submenu.default && {
                    marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
                    color: (bo = (fo = (So = e.customStyle) == null ? void 0 : So.submenu) == null ? void 0 : fo.default) == null ? void 0 : bo.color,
                    fontWeight: (go = (Io = (co = e.customStyle) == null ? void 0 : co.submenu) == null ? void 0 : Io.default) == null ? void 0 : go.fontWeight,
                    fontSize: (Lo = (po = (ho = e.customStyle) == null ? void 0 : ho.submenu) == null ? void 0 : po.default) == null ? void 0 : Lo.fontSize,
                    lineHeight: (zo = (wo = (Mo = e.customStyle) == null ? void 0 : Mo.submenu) == null ? void 0 : wo.default) == null ? void 0 : zo.lineHeight
                  } || {
                    marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0
                  }
                )
              }, [
                a(e.$slots, "title")
              ], 12, qo)
            ], 2),
            e.currentMode === "vertical" && !e.isFirstLevel || e.currentMode !== "vertical" && e.showArrow ? (n(), To(d, {
              key: 0,
              class: r(["arrow", e.opened && "rotate"]),
              style: l(
                (e.menuItemHover || e.active) && e.isFirstLevel && e.customStyle.arrowIcon && e.customStyle.arrowIcon.active && { ...(Fo = (Ho = e.customStyle) == null ? void 0 : Ho.arrowIcon) == null ? void 0 : Fo.active } || e.isFirstLevel && e.customStyle.arrowIcon && e.customStyle.arrowIcon.default && { ...($o = (Bo = e.customStyle) == null ? void 0 : Bo.arrowIcon) == null ? void 0 : $o.default } || (e.menuItemHover || e.active) && !e.isFirstLevel && e.customStyle.subArrowIcon && e.customStyle.subArrowIcon.active && { ...(Wo = (ko = e.customStyle) == null ? void 0 : ko.subArrowIcon) == null ? void 0 : Wo.active } || !e.isFirstLevel && e.customStyle.subArrowIcon && e.customStyle.subArrowIcon.default && { ...(Ao = (Co = e.customStyle) == null ? void 0 : Co.subArrowIcon) == null ? void 0 : Ao.default }
              )
            }, {
              default: s(() => o[5] || (o[5] = [
                t("svg", {
                  xmlns: "http://www.w3.org/2000/svg",
                  viewBox: "0 0 1024 1024"
                }, [
                  t("path", {
                    fill: "currentColor",
                    d: `M831.872 340.864 512 652.672 192.128 340.864a30.592 30.592 0 0 0-42.752 0 29.12 29.12 0 0 0 0 41.6L489.664 714.24a32 32 0 0 0
                44.672 0l340.288-331.712a29.12 29.12 0 0 0 0-41.728 30.592 30.592 0 0 0-42.752 0z`
                  })
                ], -1)
              ])),
              _: 1,
              __: [5]
            }, 8, ["class", "style"])) : m("", !0),
            o[6] || (o[6] = t("i", null, null, -1))
          ], 38)
        ];
      }),
      default: s(() => [
        a(e.$slots, "default")
      ]),
      _: 3
    }, 8, ["visible", "placement"])) : m("", !0),
    e.currentMode === "vertical" && !e.fold ? (n(), i("div", {
      key: 1,
      class: "mode-vertical",
      style: l(
        e.isFirstLevel && e.customStyle.menuItem && e.customStyle.menuItem.active && (e.menuItemHover || e.active) && {
          // marginLeft: currentMode === 'vertical' && fold ? '15px' : 0,
          ...e.customStyle.menuItem.active
        } || e.isFirstLevel && e.customStyle.menuItem && e.customStyle.menuItem.default && {
          // marginLeft: currentMode === 'vertical' && fold ? '15px' : 0,
          ...e.customStyle.menuItem.default
        } || !e.isFirstLevel && e.customStyle.submenu && e.customStyle.submenu.active && (e.menuItemHover || e.active) && {
          // marginLeft: currentMode === 'vertical' && fold ? '15px' : 0,
          ...e.customStyle.submenu.active
        } || !e.isFirstLevel && e.customStyle.submenu && e.customStyle.submenu.default && {
          // marginLeft: currentMode === 'vertical' && fold ? '15px' : 0,
          ...e.customStyle.submenu.default
        } || {
          // marginLeft: currentMode === 'vertical' && fold ? '15px' : 0,
        }
      )
    }, [
      t("div", {
        class: r(["mode-vertical__header", [
          (e.opened || e.active) && "mode-vertical__header--active",
          e.currentBorderLine && "mode-vertical__header--line"
        ]]),
        onClick: o[2] || (o[2] = (...u) => e.handleClickSubmenu && e.handleClickSubmenu(...u)),
        onMouseenter: o[3] || (o[3] = (...u) => e.handleMouseenterSubmenu && e.handleMouseenterSubmenu(...u))
      }, [
        t("div", Do, [
          e.currentShowIcon ? (n(), i("i", {
            key: 0,
            class: r(["fa", e.currentIcon]),
            style: l(
              e.customStyle.icon && e.customStyle.icon.active && (e.menuItemHover || e.active) && {
                marginBottom: e.currentMode === "horizontal" && e.currentMenuLayout === "vertical" ? "5px" : 0,
                marginRight: e.currentMenuLayout === "horizontal" || e.currentMode === "vertical" ? "5px" : 0,
                ...e.customStyle.icon.active
              } || e.customStyle.icon && e.customStyle.icon.default && {
                marginBottom: e.currentMode === "horizontal" && e.currentMenuLayout === "vertical" ? "5px" : 0,
                marginRight: e.currentMenuLayout === "horizontal" || e.currentMode === "vertical" ? "5px" : 0,
                ...e.customStyle.icon.default
              } || {
                marginBottom: e.currentMode === "horizontal" && e.currentMenuLayout === "vertical" ? "5px" : 0,
                marginRight: e.currentMenuLayout === "horizontal" || e.currentMode === "vertical" ? "5px" : 0
              }
            )
          }, null, 6)) : m("", !0),
          t("span", {
            class: "label",
            title: e.slotsTitle,
            style: l(
              (e.menuItemHover || e.active) && e.isFirstLevel && e.customStyle.menuItem && e.customStyle.menuItem.active && {
                marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
                color: (c = (b = (f = e.customStyle) == null ? void 0 : f.menuItem) == null ? void 0 : b.active) == null ? void 0 : c.color,
                fontWeight: (h = (g = (I = e.customStyle) == null ? void 0 : I.menuItem) == null ? void 0 : g.active) == null ? void 0 : h.fontWeight,
                fontSize: (M = (L = (p = e.customStyle) == null ? void 0 : p.menuItem) == null ? void 0 : L.active) == null ? void 0 : M.fontSize,
                lineHeight: (H = (z = (w = e.customStyle) == null ? void 0 : w.menuItem) == null ? void 0 : z.active) == null ? void 0 : H.lineHeight
              } || e.isFirstLevel && e.customStyle.menuItem && e.customStyle.menuItem.default && {
                marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
                color: ($ = (B = (F = e.customStyle) == null ? void 0 : F.menuItem) == null ? void 0 : B.default) == null ? void 0 : $.color,
                fontWeight: (C = (W = (k = e.customStyle) == null ? void 0 : k.menuItem) == null ? void 0 : W.default) == null ? void 0 : C.fontWeight,
                fontSize: (T = (R = (A = e.customStyle) == null ? void 0 : A.menuItem) == null ? void 0 : R.default) == null ? void 0 : T.fontSize,
                lineHeight: (E = (V = (N = e.customStyle) == null ? void 0 : N.menuItem) == null ? void 0 : V.default) == null ? void 0 : E.lineHeight
              } || (e.menuItemHover || e.active) && !e.isFirstLevel && e.customStyle.submenu && e.customStyle.submenu.active && {
                marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
                color: (q = (j = (U = e.customStyle) == null ? void 0 : U.submenu) == null ? void 0 : j.active) == null ? void 0 : q.color,
                fontWeight: (J = (G = (D = e.customStyle) == null ? void 0 : D.submenu) == null ? void 0 : G.active) == null ? void 0 : J.fontWeight,
                fontSize: (P = (O = (K = e.customStyle) == null ? void 0 : K.submenu) == null ? void 0 : O.active) == null ? void 0 : P.fontSize,
                lineHeight: (Y = (X = (Q = e.customStyle) == null ? void 0 : Q.submenu) == null ? void 0 : X.active) == null ? void 0 : Y.lineHeight
              } || !e.isFirstLevel && e.customStyle.submenu && e.customStyle.submenu.default && {
                marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
                color: (_ = (x = (Z = e.customStyle) == null ? void 0 : Z.submenu) == null ? void 0 : x.default) == null ? void 0 : _.color,
                fontWeight: (te = (oe = (ee = e.customStyle) == null ? void 0 : ee.submenu) == null ? void 0 : oe.default) == null ? void 0 : te.fontWeight,
                fontSize: (re = (le = (ue = e.customStyle) == null ? void 0 : ue.submenu) == null ? void 0 : le.default) == null ? void 0 : re.fontSize,
                lineHeight: (ie = (me = (ne = e.customStyle) == null ? void 0 : ne.submenu) == null ? void 0 : me.default) == null ? void 0 : ie.lineHeight
              } || {
                marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0
              }
            )
          }, [
            a(e.$slots, "title")
          ], 12, Go)
        ]),
        Uo(d, {
          class: r(["arrow", e.opened && "rotate"]),
          style: l(
            (e.menuItemHover || e.active) && e.isFirstLevel && e.customStyle.arrowIcon && e.customStyle.arrowIcon.active && { ...(ae = (se = e.customStyle) == null ? void 0 : se.arrowIcon) == null ? void 0 : ae.active } || e.isFirstLevel && e.customStyle.arrowIcon && e.customStyle.arrowIcon.default && { ...(ve = (de = e.customStyle) == null ? void 0 : de.arrowIcon) == null ? void 0 : ve.default } || (e.menuItemHover || e.active) && !e.isFirstLevel && e.customStyle.subArrowIcon && e.customStyle.subArrowIcon.active && { ...(Se = (ye = e.customStyle) == null ? void 0 : ye.subArrowIcon) == null ? void 0 : Se.active } || !e.isFirstLevel && e.customStyle.subArrowIcon && e.customStyle.subArrowIcon.default && { ...(be = (fe = e.customStyle) == null ? void 0 : fe.subArrowIcon) == null ? void 0 : be.default }
          )
        }, {
          default: s(() => o[7] || (o[7] = [
            t("svg", {
              xmlns: "http://www.w3.org/2000/svg",
              viewBox: "0 0 1024 1024"
            }, [
              t("path", {
                fill: "currentColor",
                d: `M340.864 149.312a30.592 30.592 0 0 0 0 42.752L652.736 512 340.864 831.872a30.592 30.592 0 0 0 0
              42.752 29.12 29.12 0 0 0 41.728 0L714.24 534.336a32 32 0 0 0 0-44.672L382.592 149.376a29.12 29.12 0 0 0-41.728 0z`
              })
            ], -1)
          ])),
          _: 1,
          __: [7]
        }, 8, ["class", "style"])
      ], 34),
      t("div", {
        ref: "submenuList",
        class: r(["mode-vertical__content", `mode-vertical__content--${e.currentThemeStyle}`]),
        style: l({ minHeight: e.opened ? "40px" : 0, opacity: e.opened ? 1 : 0, height: e.opened ? "auto" : "0" })
      }, [
        a(e.$slots, "default", { parentIndex: e.index })
      ], 6)
    ], 4)) : m("", !0)
  ], 38);
}
const _o = /* @__PURE__ */ jo(Eo, [["render", Jo]]);
export {
  _o as default
};
