// @ths/design/es/components/menu/src/submenu-item.vue.js
import E from "./submenu-item.vue2.js";
import { openBlock as H, createElementBlock as B, normalizeStyle as o, createElementVNode as L, normalizeClass as w, createCommentVNode as N, renderSlot as T } from "vue";
import "./submenu-item.vue3.js";
import V from "../../../_virtual/_plugin-vue_export-helper.js";
const j = ["title"];
function q(e, t, A, D, F, G) {
  var m, n, l, s, i, r, a, d, S, f, y, b, I, v, c, h, p, M, $, g, k, z, C, W;
  return H(), B("div", {
    class: "t-submenu-item",
    style: o(
      e.active && e.customStyle.submenuItem && e.customStyle.submenuItem.active && { ...e.customStyle.submenuItem.active } || e.customStyle.submenuItem && e.customStyle.submenuItem.default && { ...e.customStyle.submenuItem.default }
    ),
    onClick: t[0] || (t[0] = (...u) => e.handleClickSubmenuItem && e.handleClickSubmenuItem(...u)),
    onMouseenter: t[1] || (t[1] = (...u) => e.handleMouseenterSubmenuItem && e.handleMouseenterSubmenuItem(...u)),
    onMouseleave: t[2] || (t[2] = (...u) => e.handleMouseleaveSubmenuItem && e.handleMouseleaveSubmenuItem(...u))
  }, [
    L("div", {
      style: o({ "--width": e.computMenuWidth }),
      class: w([
        `mode-${e.currentMode}`,
        !e.fold && "unfold",
        e.active && `mode-${e.currentMode}--active`,
        `mode-${e.currentMode}__${e.currentThemeStyle}`,
        e.currentBorderLine && `mode-${e.currentMode}${e.fold ? "" : "--line"}`,
        e.active && `mode-${e.currentMode}__${e.currentThemeStyle}--active`
      ])
    }, [
      e.currentShowIcon ? (H(), B("i", {
        key: 0,
        class: w(["fa", e.currentIcon]),
        style: o(
          e.active && e.customStyle.icon && e.customStyle.icon.active && { ...e.customStyle.icon.active } || e.customStyle.icon && e.customStyle.icon.default && { ...e.customStyle.icon.default }
        )
      }, null, 6)) : N("", !0),
      L("span", {
        class: "label",
        title: e.slots,
        style: o(
          e.active && e.customStyle.submenuItem && e.customStyle.submenuItem.active && {
            marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
            color: (l = (n = (m = e.customStyle) == null ? void 0 : m.submenuItem) == null ? void 0 : n.active) == null ? void 0 : l.color,
            fontWeight: (r = (i = (s = e.customStyle) == null ? void 0 : s.submenuItem) == null ? void 0 : i.active) == null ? void 0 : r.fontWeight,
            fontSize: (S = (d = (a = e.customStyle) == null ? void 0 : a.submenuItem) == null ? void 0 : d.active) == null ? void 0 : S.fontSize,
            lineHeight: (b = (y = (f = e.customStyle) == null ? void 0 : f.submenuItem) == null ? void 0 : y.active) == null ? void 0 : b.lineHeight
          } || e.customStyle.submenuItem && e.customStyle.submenuItem.default && {
            marginLeft: e.currentMode === "vertical" && e.fold ? "15px" : 0,
            color: (c = (v = (I = e.customStyle) == null ? void 0 : I.submenuItem) == null ? void 0 : v.default) == null ? void 0 : c.color,
            fontWeight: (M = (p = (h = e.customStyle) == null ? void 0 : h.submenuItem) == null ? void 0 : p.default) == null ? void 0 : M.fontWeight,
            fontSize: (k = (g = ($ = e.customStyle) == null ? void 0 : $.submenuItem) == null ? void 0 : g.default) == null ? void 0 : k.fontSize,
            lineHeight: (W = (C = (z = e.customStyle) == null ? void 0 : z.submenuItem) == null ? void 0 : C.default) == null ? void 0 : W.lineHeight
          }
        )
      }, [
        T(e.$slots, "default")
      ], 12, j)
    ], 6)
  ], 36);
}
const Q = /* @__PURE__ */ V(E, [["render", q]]);
export {
  Q as default
};
