// @ths/design/es/components/high-light-text/src/index.vue2.js
import { defineComponent as h, ref as o, watch as p, onMounted as x, computed as m } from "vue";
const w = h({
  name: "THighLightText",
  props: {
    // 是否显示标题
    showTitle: {
      type: Boolean,
      default: !1
    },
    // 分段文本列表
    fragmentList: {
      type: Array,
      default: () => []
    },
    // 分段样式配置
    styles: {
      type: Array,
      default: () => []
    },
    // 自定义样式
    cStyle: {
      type: Object,
      default: () => ({
        wrapper: {
          default: {}
        }
      })
    }
  },
  // 修改文本事件
  emits: ["modify-text", "blur-text"],
  setup(s, { emit: S }) {
    const c = o(null), y = o([]), f = (l) => l.map((t) => (Object.keys(t.style).forEach((e) => {
      const n = e.replace(/[A-Z]/g, (a) => `-${a.toLowerCase()}`);
      t.style[n] = t.style[e], e.match(/[A-Z]/g) && delete t.style[e];
    }), {
      name: t.name,
      style: t.style
    })), u = (l, r) => {
      if (!(l != null && l.length)) {
        y.value = [];
        return;
      }
      const t = f(r), e = l.map((n) => {
        const a = t.find((i) => i.name === n.styleName);
        return a ? {
          text: n.text,
          style: a.style
        } : {
          text: n.text,
          style: {}
        };
      });
      e.length && (y.value = e);
    };
    p(() => s, () => {
      u(s.fragmentList, s.styles);
    }, { deep: !0 }), x(() => {
      u(s.fragmentList, s.styles);
    });
    const d = m(() => {
      var t, e;
      return { ...(e = (t = s.cStyle) == null ? void 0 : t.wrapper) == null ? void 0 : e.default };
    });
    return {
      textRef: c,
      textList: y,
      customStyle: d
    };
  }
});
export {
  w as default
};
